{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# take mature cells from Calderon umap and subset, rename cells and find nearest neighbors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“package ‘S4Vectors’ was built under R version 4.1.3”\n"
     ]
    }
   ],
   "source": [
    "suppressPackageStartupMessages({\n",
    "    library(monocle3)\n",
    "    library(tidyverse)\n",
    "    library(ggplot2)\n",
    "    library(reshape2)\n",
    "    library(viridis)\n",
    "})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "R version 4.1.2 (2021-11-01)\n",
       "Platform: x86_64-apple-darwin17.0 (64-bit)\n",
       "Running under: macOS Big Sur 10.16\n",
       "\n",
       "Matrix products: default\n",
       "BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib\n",
       "LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib\n",
       "\n",
       "locale:\n",
       "[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8\n",
       "\n",
       "attached base packages:\n",
       "[1] stats4    stats     graphics  grDevices utils     datasets  methods  \n",
       "[8] base     \n",
       "\n",
       "other attached packages:\n",
       " [1] viridis_0.6.3               viridisLite_0.4.2          \n",
       " [3] reshape2_1.4.4              forcats_0.5.2              \n",
       " [5] stringr_1.5.0               dplyr_1.1.2                \n",
       " [7] purrr_1.0.1                 readr_2.1.3                \n",
       " [9] tidyr_1.3.0                 tibble_3.2.1               \n",
       "[11] ggplot2_3.4.2               tidyverse_1.3.2            \n",
       "[13] monocle3_1.3.1              SingleCellExperiment_1.16.0\n",
       "[15] SummarizedExperiment_1.24.0 GenomicRanges_1.46.1       \n",
       "[17] GenomeInfoDb_1.30.1         IRanges_2.28.0             \n",
       "[19] S4Vectors_0.32.4            MatrixGenerics_1.6.0       \n",
       "[21] matrixStats_1.0.0           Biobase_2.54.0             \n",
       "[23] BiocGenerics_0.40.0        \n",
       "\n",
       "loaded via a namespace (and not attached):\n",
       " [1] nlme_3.1-153           bitops_1.0-7           fs_1.6.2              \n",
       " [4] lubridate_1.9.0        httr_1.4.6             repr_1.1.4            \n",
       " [7] tools_4.1.2            backports_1.4.1        utf8_1.2.3            \n",
       "[10] R6_2.5.1               DBI_1.1.3              colorspace_2.1-0      \n",
       "[13] withr_2.5.0            gridExtra_2.3          tidyselect_1.2.0      \n",
       "[16] compiler_4.1.2         rvest_1.0.3            cli_3.6.1             \n",
       "[19] xml2_1.3.4             DelayedArray_0.20.0    scales_1.2.1          \n",
       "[22] pbdZMQ_0.3-8           digest_0.6.31          minqa_1.2.5           \n",
       "[25] XVector_0.34.0         base64enc_0.1-3        pkgconfig_2.0.3       \n",
       "[28] htmltools_0.5.5        parallelly_1.33.0      lme4_1.1-31           \n",
       "[31] dbplyr_2.2.1           fastmap_1.1.1          readxl_1.4.1          \n",
       "[34] rlang_1.1.1            generics_0.1.3         jsonlite_1.8.5        \n",
       "[37] googlesheets4_1.0.1    RCurl_1.98-1.12        magrittr_2.0.3        \n",
       "[40] GenomeInfoDbData_1.2.7 Matrix_1.5-3           Rcpp_1.0.10           \n",
       "[43] IRkernel_1.3.1         munsell_0.5.0          fansi_1.0.4           \n",
       "[46] lifecycle_1.0.3        terra_1.6-47           stringi_1.7.12        \n",
       "[49] MASS_7.3-54            zlibbioc_1.40.0        plyr_1.8.8            \n",
       "[52] grid_4.1.2             parallel_4.1.2         listenv_0.9.0         \n",
       "[55] crayon_1.5.2           lattice_0.20-45        IRdisplay_1.1         \n",
       "[58] haven_2.5.1            splines_4.1.2          hms_1.1.2             \n",
       "[61] pillar_1.9.0           igraph_1.3.5           uuid_1.1-0            \n",
       "[64] boot_1.3-28            codetools_0.2-18       reprex_2.0.2          \n",
       "[67] glue_1.6.2             evaluate_0.21          modelr_0.1.10         \n",
       "[70] vctrs_0.6.3            nloptr_2.0.3           tzdb_0.3.0            \n",
       "[73] cellranger_1.1.0       gtable_0.3.3           future_1.30.0         \n",
       "[76] assertthat_0.2.1       broom_1.0.2            googledrive_2.0.0     \n",
       "[79] gargle_1.2.1           timechange_0.1.1       globals_0.16.2        \n",
       "[82] ellipsis_0.3.2        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sessionInfo()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: ‘philentropy’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:SummarizedExperiment’:\n",
      "\n",
      "    distance\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    distance\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    distance\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(philentropy)\n",
    "library(cowplot)\n",
    "library(ggrepel)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "load(\"data/utility.functions.RData\")\n",
    "load(\"data/moreUtilities.RData\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "load(\"data/utility.functions.3.RData\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# these are the cells that were subsetted\n",
    "# probably should have added cluster 19, 20 the cells at lower left.  Could redo this if it looks like it is working"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## commands from old notebook"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(data_df_100, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(data_df_100, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(data_df_100 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = data_df_100$cluster %in% c(1, 16, 18,22, 23, 5, 8, 21, 11, 15, 9, 12)), size =  0.02, stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(data_df_100, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(data_df_100, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(data_df_100 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = data_df_100$inferred_time_window), size =  0.02, stroke = 0.005) +\n",
    "# scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "#      labs(color = cell_type) +\n",
    "guides(color = guide_legend(override.aes = list(size = 4))) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(data_df_100, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(data_df_100, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(data_df_100 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = data_df_100$inferred_time_window == \"0-2\"), size =  0.1, stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 955,
   "metadata": {},
   "outputs": [],
   "source": [
    "cds_sub = cds_pcg[, data_df$cluster %in% c(1, 16, 18,22, 23, 5, 8, 21, 11, 15, 9, 12) ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 956,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>13938</li><li>238445</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 13938\n",
       "\\item 238445\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 13938\n",
       "2. 238445\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1]  13938 238445"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(cds_sub)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 957,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>238445</li><li>32</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 238445\n",
       "\\item 32\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 238445\n",
       "2. 32\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 238445     32"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "data_df  %>% filter(cluster %in% c(1, 16, 18,22, 23, 5, 8, 21, 11, 15, 9, 12))  %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub$inferred_time_window == \"12-14\"), size =  0.1, stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### remove steps of making different PCAs since went forward with PCA 300"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# corrected plot expr for more than one gene\n",
    "plot.expr_clus.br2 = function (cds, gene_short_name, coexpr = F, ncol = NULL, x = 1, \n",
    "    y = 2, size_pos = NULL, stroke = NULL, non.expr.alpha = 1/3, \n",
    "    color.pal = c(\"grey85\", \"red\", \"orange\", \"yellow\"), sample.n = F, \n",
    "    label_size = NULL, limit = NULL, size_neg = NULL) \n",
    "{\n",
    "    if (sample.n != F) {\n",
    "        set.seed(42)\n",
    "        cds = cds[, sample(1:ncol(cds), sample.n)]\n",
    "    }\n",
    "    S_matrix <- reducedDims(cds)[[\"UMAP\"]]\n",
    "    data_df <- data.frame(S_matrix)\n",
    "    colnames(data_df) <- c(\"umap_1\", \"umap_2\")\n",
    "    data_df$cluster = as.vector(clusters(cds))\n",
    "    data_df$sample_name <- row.names(data_df)\n",
    "    data_df$Size_Factor <- colData(cds)$Size_Factor\n",
    "    data_df$lineage <- colData(cds)$lineage\n",
    "    data_df$species <- colData(cds)$species\n",
    "    if (length(nchar(gene_short_name)) > 1) {\n",
    "        label_df <- data_df %>% group_by(cluster) %>% summarize(umap_1 = median(umap_1), \n",
    "            umap_2 = median(umap_2))\n",
    "        a <- as.data.frame(as.matrix(t(counts(cds)[gene_short_name, \n",
    "            ])))\n",
    "        a <- a/data_df$Size_Factor\n",
    "        a$min <- apply(a, 1, min)\n",
    "        data_df$min <- a$min\n",
    "        if (is.null(size_pos)) {\n",
    "            size_pos = 0.1\n",
    "        }\n",
    "        if (is.null(stroke)) {\n",
    "            stroke = 0.4\n",
    "        }\n",
    "        if (is.null(label_size)) {\n",
    "            label_size = 4\n",
    "        }\n",
    "        if (is.null(limit)) {\n",
    "            limit = 1\n",
    "        }\n",
    "        if (is.null(size_neg)) {\n",
    "            size_neg = 0.1\n",
    "        }\n",
    "        plot = ggplot(data_df, alpha = min > 0) + geom_point(aes(x = umap_1, \n",
    "            y = umap_2, color = log2(min + 1)), size = ifelse(data_df$min > \n",
    "            limit, size_pos, size_neg), stroke = stroke) + scale_color_gradientn(colors = color.pal) + \n",
    "            scale_alpha_manual(values = c(non.expr.alpha, 1)) + \n",
    "            guides(alpha = F, color = guide_colorbar(title = \"Minimum\\nnormalized\\nlog2 expression\\nfor specified genes\")) + \n",
    "            geom_text(data = label_df, aes(label = cluster, x = umap_1 + \n",
    "                0.25, y = umap_2 + 0.25), size = label_size) + \n",
    "            monocle3:::monocle_theme_opts()\n",
    "        return(plot)\n",
    "    }\n",
    "    if (length(nchar(gene_short_name)) == 1) {\n",
    "        label_df <- data_df %>% group_by(cluster) %>% summarize(umap_1 = median(umap_1), \n",
    "            umap_2 = median(umap_2))\n",
    "        data_df$tmp.expr = counts(cds)[gene_short_name, ]/data_df$Size_Factor\n",
    "        if (is.null(size_pos)) {\n",
    "            size_pos = 0\n",
    "        }\n",
    "        if (is.null(stroke)) {\n",
    "            stroke = 0.4\n",
    "        }\n",
    "        if (is.null(label_size)) {\n",
    "            label_size = 4\n",
    "        }\n",
    "        if (is.null(limit)) {\n",
    "            limit = 1\n",
    "        }\n",
    "        if (is.null(size_neg)) {\n",
    "            size_neg = 0.1\n",
    "        }\n",
    "        plot = ggplot(data_df) + geom_point(aes(x = umap_1, y = umap_2, \n",
    "            color = log2(tmp.expr/Size_Factor + 1), alpha = tmp.expr > \n",
    "                0), size = ifelse(data_df$tmp.exp > limit, size_pos, \n",
    "            size_neg), stroke = stroke) + scale_color_gradientn(colors = color.pal) + \n",
    "            scale_alpha_manual(values = c(non.expr.alpha, 1)) + \n",
    "            guides(alpha = F, color = guide_colorbar(title = paste(gene_short_name, \n",
    "                \"\\nnormalized\\nlog2 expression\", sep = \"\"))) + \n",
    "            geom_text(data = label_df, aes(label = cluster, x = umap_1 + \n",
    "                0.25, y = umap_2 + 0.25), size = label_size) + \n",
    "            monocle3:::monocle_theme_opts()\n",
    "        return(plot)\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Try 300 PCAs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1155,
   "metadata": {},
   "outputs": [],
   "source": [
    "cds_sub <- preprocess_cds(cds_sub, num_dim = 300)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1156,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No preprocess_method specified, using preprocess_method = 'PCA'\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cds_sub = reduce_dimension(cds_sub)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot_cells(cds_sub, cell_size = 0.04, color_cells_by = \"annot\"): could not find function \"plot_cells\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot_cells(cds_sub, cell_size = 0.04, color_cells_by = \"annot\"): could not find function \"plot_cells\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot_cells(cds_sub, cell_size = 0.04,  color_cells_by = \"annot\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1158,
   "metadata": {},
   "outputs": [],
   "source": [
    "cds_sub = cluster_cells(cds_sub, resolution=4e-4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot_cells(cds_sub, cell_size = 0.04): could not find function \"plot_cells\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot_cells(cds_sub, cell_size = 0.04): could not find function \"plot_cells\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot_cells(cds_sub, cell_size = 0.04)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "cds_sub = readRDS( \"data/cds_fly_sub_PCA300_older_cells.rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sub300 = make_df(cds_sub)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "options(repr.plot.width=14, repr.plot.height=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot_cells(cds_sub, cell_size = 0.04): could not find function \"plot_cells\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot_cells(cds_sub, cell_size = 0.04): could not find function \"plot_cells\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot_cells(cds_sub, cell_size = 0.04)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Pick up here to redo"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use PCA to find singleton cells and remove"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "cells <- colData(cds_sub)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "cellTypes <- cells$annot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "cellIDs <- cells$cell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "pca <- cds_sub@int_colData$reducedDims$PCA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(RANN)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "nnMatches <- function(ids,labels,df,k=10){\n",
    "   k1 <- k + 1  # need one more since always finds itself\n",
    "   rann <- RANN::nn2(df,query=df,k1)\n",
    "   df <- as.data.frame(rann$nn.idx)\n",
    "   df <- mutate(df,ID=ids,.before=1)  # add the ids to the df\n",
    "   df <- mutate(df,Label=labels[V1],.before=2) %>% select(-V1)  # put the labels into the df\n",
    "   df <- pivot_longer(df,cols=3:k1,names_to = \"NN\")\n",
    "   df <- mutate(df,nnLabel=labels[value])  # label the nearest neighbors\n",
    "   df <- select(df,-NN,-value)\n",
    "   res <- group_by(df,ID) %>% summarise(nk=sum (Label==nnLabel)  )  # count the matches in the nearest neighors\n",
    "   return(res)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Do individual steps of Lou's function above"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "k = 20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "k1 <- k + 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pca"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "rann <- RANN::nn2(df,query=df,k1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "saveRDS(rann, \"data/fly_rann_PCA300.rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "ids = cellIDs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "df <- as.data.frame(rann$nn.idx)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 21</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V4</th><th scope=col>V5</th><th scope=col>V6</th><th scope=col>V7</th><th scope=col>V8</th><th scope=col>V9</th><th scope=col>V10</th><th scope=col>⋯</th><th scope=col>V12</th><th scope=col>V13</th><th scope=col>V14</th><th scope=col>V15</th><th scope=col>V16</th><th scope=col>V17</th><th scope=col>V18</th><th scope=col>V19</th><th scope=col>V20</th><th scope=col>V21</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>  1084</td><td>117474</td><td> 33297</td><td>  2934</td><td> 2355</td><td>   854</td><td>  2437</td><td>  4221</td><td> 63518</td><td>⋯</td><td>116766</td><td> 48132</td><td> 20116</td><td>115992</td><td> 34544</td><td> 31818</td><td> 50700</td><td>143608</td><td>   571</td><td>  3626</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>2</td><td> 51177</td><td> 19874</td><td> 69268</td><td> 44378</td><td>67519</td><td>212700</td><td>189174</td><td>131569</td><td> 24279</td><td>⋯</td><td> 20273</td><td>130972</td><td>  9391</td><td>  3473</td><td>109102</td><td>132457</td><td>130068</td><td> 46597</td><td> 53857</td><td>183430</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>3</td><td>137166</td><td> 35175</td><td>  4381</td><td>123690</td><td>99621</td><td> 35423</td><td>123344</td><td>108376</td><td>126736</td><td>⋯</td><td>123710</td><td> 47008</td><td>134782</td><td>136306</td><td>137709</td><td>123045</td><td> 33837</td><td> 32261</td><td>136304</td><td>135305</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>4</td><td> 68440</td><td> 71253</td><td> 67238</td><td> 73868</td><td>67236</td><td> 71880</td><td> 64019</td><td> 64022</td><td> 66768</td><td>⋯</td><td> 64612</td><td> 63786</td><td> 70436</td><td> 65552</td><td>  2853</td><td> 65551</td><td> 67168</td><td> 39930</td><td> 28538</td><td> 74114</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>5</td><td>137728</td><td> 19780</td><td>120926</td><td> 91533</td><td> 5730</td><td>144152</td><td>143823</td><td> 35446</td><td> 54161</td><td>⋯</td><td> 51952</td><td>  4980</td><td>134914</td><td> 19964</td><td>132236</td><td> 52185</td><td> 34199</td><td>214925</td><td>117952</td><td> 68538</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>6</td><td> 36849</td><td>  1387</td><td>  2385</td><td> 63108</td><td>36476</td><td>  2887</td><td> 67810</td><td>  3884</td><td> 35089</td><td>⋯</td><td>  5792</td><td> 75285</td><td>  2730</td><td> 19767</td><td> 34789</td><td> 35981</td><td>  1882</td><td> 64621</td><td> 49619</td><td>   781</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 21\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & V1 & V2 & V3 & V4 & V5 & V6 & V7 & V8 & V9 & V10 & ⋯ & V12 & V13 & V14 & V15 & V16 & V17 & V18 & V19 & V20 & V21\\\\\n",
       "  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 1 &   1084 & 117474 &  33297 &   2934 &  2355 &    854 &   2437 &   4221 &  63518 & ⋯ & 116766 &  48132 &  20116 & 115992 &  34544 &  31818 &  50700 & 143608 &    571 &   3626\\\\\n",
       "\t2 & 2 &  51177 &  19874 &  69268 &  44378 & 67519 & 212700 & 189174 & 131569 &  24279 & ⋯ &  20273 & 130972 &   9391 &   3473 & 109102 & 132457 & 130068 &  46597 &  53857 & 183430\\\\\n",
       "\t3 & 3 & 137166 &  35175 &   4381 & 123690 & 99621 &  35423 & 123344 & 108376 & 126736 & ⋯ & 123710 &  47008 & 134782 & 136306 & 137709 & 123045 &  33837 &  32261 & 136304 & 135305\\\\\n",
       "\t4 & 4 &  68440 &  71253 &  67238 &  73868 & 67236 &  71880 &  64019 &  64022 &  66768 & ⋯ &  64612 &  63786 &  70436 &  65552 &   2853 &  65551 &  67168 &  39930 &  28538 &  74114\\\\\n",
       "\t5 & 5 & 137728 &  19780 & 120926 &  91533 &  5730 & 144152 & 143823 &  35446 &  54161 & ⋯ &  51952 &   4980 & 134914 &  19964 & 132236 &  52185 &  34199 & 214925 & 117952 &  68538\\\\\n",
       "\t6 & 6 &  36849 &   1387 &   2385 &  63108 & 36476 &   2887 &  67810 &   3884 &  35089 & ⋯ &   5792 &  75285 &   2730 &  19767 &  34789 &  35981 &   1882 &  64621 &  49619 &    781\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 21\n",
       "\n",
       "| <!--/--> | V1 &lt;int&gt; | V2 &lt;int&gt; | V3 &lt;int&gt; | V4 &lt;int&gt; | V5 &lt;int&gt; | V6 &lt;int&gt; | V7 &lt;int&gt; | V8 &lt;int&gt; | V9 &lt;int&gt; | V10 &lt;int&gt; | ⋯ ⋯ | V12 &lt;int&gt; | V13 &lt;int&gt; | V14 &lt;int&gt; | V15 &lt;int&gt; | V16 &lt;int&gt; | V17 &lt;int&gt; | V18 &lt;int&gt; | V19 &lt;int&gt; | V20 &lt;int&gt; | V21 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 |   1084 | 117474 |  33297 |   2934 |  2355 |    854 |   2437 |   4221 |  63518 | ⋯ | 116766 |  48132 |  20116 | 115992 |  34544 |  31818 |  50700 | 143608 |    571 |   3626 |\n",
       "| 2 | 2 |  51177 |  19874 |  69268 |  44378 | 67519 | 212700 | 189174 | 131569 |  24279 | ⋯ |  20273 | 130972 |   9391 |   3473 | 109102 | 132457 | 130068 |  46597 |  53857 | 183430 |\n",
       "| 3 | 3 | 137166 |  35175 |   4381 | 123690 | 99621 |  35423 | 123344 | 108376 | 126736 | ⋯ | 123710 |  47008 | 134782 | 136306 | 137709 | 123045 |  33837 |  32261 | 136304 | 135305 |\n",
       "| 4 | 4 |  68440 |  71253 |  67238 |  73868 | 67236 |  71880 |  64019 |  64022 |  66768 | ⋯ |  64612 |  63786 |  70436 |  65552 |   2853 |  65551 |  67168 |  39930 |  28538 |  74114 |\n",
       "| 5 | 5 | 137728 |  19780 | 120926 |  91533 |  5730 | 144152 | 143823 |  35446 |  54161 | ⋯ |  51952 |   4980 | 134914 |  19964 | 132236 |  52185 |  34199 | 214925 | 117952 |  68538 |\n",
       "| 6 | 6 |  36849 |   1387 |   2385 |  63108 | 36476 |   2887 |  67810 |   3884 |  35089 | ⋯ |   5792 |  75285 |   2730 |  19767 |  34789 |  35981 |   1882 |  64621 |  49619 |    781 |\n",
       "\n"
      ],
      "text/plain": [
       "  V1 V2     V3     V4     V5     V6    V7     V8     V9     V10    ⋯ V12   \n",
       "1 1    1084 117474  33297   2934  2355    854   2437   4221  63518 ⋯ 116766\n",
       "2 2   51177  19874  69268  44378 67519 212700 189174 131569  24279 ⋯  20273\n",
       "3 3  137166  35175   4381 123690 99621  35423 123344 108376 126736 ⋯ 123710\n",
       "4 4   68440  71253  67238  73868 67236  71880  64019  64022  66768 ⋯  64612\n",
       "5 5  137728  19780 120926  91533  5730 144152 143823  35446  54161 ⋯  51952\n",
       "6 6   36849   1387   2385  63108 36476   2887  67810   3884  35089 ⋯   5792\n",
       "  V13    V14    V15    V16    V17    V18    V19    V20    V21   \n",
       "1  48132  20116 115992  34544  31818  50700 143608    571   3626\n",
       "2 130972   9391   3473 109102 132457 130068  46597  53857 183430\n",
       "3  47008 134782 136306 137709 123045  33837  32261 136304 135305\n",
       "4  63786  70436  65552   2853  65551  67168  39930  28538  74114\n",
       "5   4980 134914  19964 132236  52185  34199 214925 117952  68538\n",
       "6  75285   2730  19767  34789  35981   1882  64621  49619    781"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "df <- mutate(df,ID=ids,.before=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 22</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ID</th><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V4</th><th scope=col>V5</th><th scope=col>V6</th><th scope=col>V7</th><th scope=col>V8</th><th scope=col>V9</th><th scope=col>⋯</th><th scope=col>V12</th><th scope=col>V13</th><th scope=col>V14</th><th scope=col>V15</th><th scope=col>V16</th><th scope=col>V17</th><th scope=col>V18</th><th scope=col>V19</th><th scope=col>V20</th><th scope=col>V21</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>1</td><td>  1084</td><td>117474</td><td> 33297</td><td>  2934</td><td> 2355</td><td>   854</td><td>  2437</td><td>  4221</td><td>⋯</td><td>116766</td><td> 48132</td><td> 20116</td><td>115992</td><td> 34544</td><td> 31818</td><td> 50700</td><td>143608</td><td>   571</td><td>  3626</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG</td><td>2</td><td> 51177</td><td> 19874</td><td> 69268</td><td> 44378</td><td>67519</td><td>212700</td><td>189174</td><td>131569</td><td>⋯</td><td> 20273</td><td>130972</td><td>  9391</td><td>  3473</td><td>109102</td><td>132457</td><td>130068</td><td> 46597</td><td> 53857</td><td>183430</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA</td><td>3</td><td>137166</td><td> 35175</td><td>  4381</td><td>123690</td><td>99621</td><td> 35423</td><td>123344</td><td>108376</td><td>⋯</td><td>123710</td><td> 47008</td><td>134782</td><td>136306</td><td>137709</td><td>123045</td><td> 33837</td><td> 32261</td><td>136304</td><td>135305</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG</td><td>4</td><td> 68440</td><td> 71253</td><td> 67238</td><td> 73868</td><td>67236</td><td> 71880</td><td> 64019</td><td> 64022</td><td>⋯</td><td> 64612</td><td> 63786</td><td> 70436</td><td> 65552</td><td>  2853</td><td> 65551</td><td> 67168</td><td> 39930</td><td> 28538</td><td> 74114</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA</td><td>5</td><td>137728</td><td> 19780</td><td>120926</td><td> 91533</td><td> 5730</td><td>144152</td><td>143823</td><td> 35446</td><td>⋯</td><td> 51952</td><td>  4980</td><td>134914</td><td> 19964</td><td>132236</td><td> 52185</td><td> 34199</td><td>214925</td><td>117952</td><td> 68538</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA</td><td>6</td><td> 36849</td><td>  1387</td><td>  2385</td><td> 63108</td><td>36476</td><td>  2887</td><td> 67810</td><td>  3884</td><td>⋯</td><td>  5792</td><td> 75285</td><td>  2730</td><td> 19767</td><td> 34789</td><td> 35981</td><td>  1882</td><td> 64621</td><td> 49619</td><td>   781</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 22\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & ID & V1 & V2 & V3 & V4 & V5 & V6 & V7 & V8 & V9 & ⋯ & V12 & V13 & V14 & V15 & V16 & V17 & V18 & V19 & V20 & V21\\\\\n",
       "  & <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & 1 &   1084 & 117474 &  33297 &   2934 &  2355 &    854 &   2437 &   4221 & ⋯ & 116766 &  48132 &  20116 & 115992 &  34544 &  31818 &  50700 & 143608 &    571 &   3626\\\\\n",
       "\t2 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG & 2 &  51177 &  19874 &  69268 &  44378 & 67519 & 212700 & 189174 & 131569 & ⋯ &  20273 & 130972 &   9391 &   3473 & 109102 & 132457 & 130068 &  46597 &  53857 & 183430\\\\\n",
       "\t3 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA & 3 & 137166 &  35175 &   4381 & 123690 & 99621 &  35423 & 123344 & 108376 & ⋯ & 123710 &  47008 & 134782 & 136306 & 137709 & 123045 &  33837 &  32261 & 136304 & 135305\\\\\n",
       "\t4 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG & 4 &  68440 &  71253 &  67238 &  73868 & 67236 &  71880 &  64019 &  64022 & ⋯ &  64612 &  63786 &  70436 &  65552 &   2853 &  65551 &  67168 &  39930 &  28538 &  74114\\\\\n",
       "\t5 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA & 5 & 137728 &  19780 & 120926 &  91533 &  5730 & 144152 & 143823 &  35446 & ⋯ &  51952 &   4980 & 134914 &  19964 & 132236 &  52185 &  34199 & 214925 & 117952 &  68538\\\\\n",
       "\t6 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA & 6 &  36849 &   1387 &   2385 &  63108 & 36476 &   2887 &  67810 &   3884 & ⋯ &   5792 &  75285 &   2730 &  19767 &  34789 &  35981 &   1882 &  64621 &  49619 &    781\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 22\n",
       "\n",
       "| <!--/--> | ID &lt;chr&gt; | V1 &lt;int&gt; | V2 &lt;int&gt; | V3 &lt;int&gt; | V4 &lt;int&gt; | V5 &lt;int&gt; | V6 &lt;int&gt; | V7 &lt;int&gt; | V8 &lt;int&gt; | V9 &lt;int&gt; | ⋯ ⋯ | V12 &lt;int&gt; | V13 &lt;int&gt; | V14 &lt;int&gt; | V15 &lt;int&gt; | V16 &lt;int&gt; | V17 &lt;int&gt; | V18 &lt;int&gt; | V19 &lt;int&gt; | V20 &lt;int&gt; | V21 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | 1 |   1084 | 117474 |  33297 |   2934 |  2355 |    854 |   2437 |   4221 | ⋯ | 116766 |  48132 |  20116 | 115992 |  34544 |  31818 |  50700 | 143608 |    571 |   3626 |\n",
       "| 2 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG | 2 |  51177 |  19874 |  69268 |  44378 | 67519 | 212700 | 189174 | 131569 | ⋯ |  20273 | 130972 |   9391 |   3473 | 109102 | 132457 | 130068 |  46597 |  53857 | 183430 |\n",
       "| 3 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA | 3 | 137166 |  35175 |   4381 | 123690 | 99621 |  35423 | 123344 | 108376 | ⋯ | 123710 |  47008 | 134782 | 136306 | 137709 | 123045 |  33837 |  32261 | 136304 | 135305 |\n",
       "| 4 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG | 4 |  68440 |  71253 |  67238 |  73868 | 67236 |  71880 |  64019 |  64022 | ⋯ |  64612 |  63786 |  70436 |  65552 |   2853 |  65551 |  67168 |  39930 |  28538 |  74114 |\n",
       "| 5 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA | 5 | 137728 |  19780 | 120926 |  91533 |  5730 | 144152 | 143823 |  35446 | ⋯ |  51952 |   4980 | 134914 |  19964 | 132236 |  52185 |  34199 | 214925 | 117952 |  68538 |\n",
       "| 6 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA | 6 |  36849 |   1387 |   2385 |  63108 | 36476 |   2887 |  67810 |   3884 | ⋯ |   5792 |  75285 |   2730 |  19767 |  34789 |  35981 |   1882 |  64621 |  49619 |    781 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID                                                     V1 V2     V3    \n",
       "1 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 1    1084 117474\n",
       "2 exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 2   51177  19874\n",
       "3 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 3  137166  35175\n",
       "4 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 4   68440  71253\n",
       "5 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 5  137728  19780\n",
       "6 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 6   36849   1387\n",
       "  V4     V5     V6    V7     V8     V9     ⋯ V12    V13    V14    V15    V16   \n",
       "1  33297   2934  2355    854   2437   4221 ⋯ 116766  48132  20116 115992  34544\n",
       "2  69268  44378 67519 212700 189174 131569 ⋯  20273 130972   9391   3473 109102\n",
       "3   4381 123690 99621  35423 123344 108376 ⋯ 123710  47008 134782 136306 137709\n",
       "4  67238  73868 67236  71880  64019  64022 ⋯  64612  63786  70436  65552   2853\n",
       "5 120926  91533  5730 144152 143823  35446 ⋯  51952   4980 134914  19964 132236\n",
       "6   2385  63108 36476   2887  67810   3884 ⋯   5792  75285   2730  19767  34789\n",
       "  V17    V18    V19    V20    V21   \n",
       "1  31818  50700 143608    571   3626\n",
       "2 132457 130068  46597  53857 183430\n",
       "3 123045  33837  32261 136304 135305\n",
       "4  65551  67168  39930  28538  74114\n",
       "5  52185  34199 214925 117952  68538\n",
       "6  35981   1882  64621  49619    781"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "labels = cells$annot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "df <- mutate(df,Label=labels[V1],.before=2) %>% select(-V1) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 22</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ID</th><th scope=col>Label</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V4</th><th scope=col>V5</th><th scope=col>V6</th><th scope=col>V7</th><th scope=col>V8</th><th scope=col>V9</th><th scope=col>⋯</th><th scope=col>V12</th><th scope=col>V13</th><th scope=col>V14</th><th scope=col>V15</th><th scope=col>V16</th><th scope=col>V17</th><th scope=col>V18</th><th scope=col>V19</th><th scope=col>V20</th><th scope=col>V21</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>  1084</td><td>117474</td><td> 33297</td><td>  2934</td><td> 2355</td><td>   854</td><td>  2437</td><td>  4221</td><td>⋯</td><td>116766</td><td> 48132</td><td> 20116</td><td>115992</td><td> 34544</td><td> 31818</td><td> 50700</td><td>143608</td><td>   571</td><td>  3626</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG</td><td>visceral muscle primordium       </td><td> 51177</td><td> 19874</td><td> 69268</td><td> 44378</td><td>67519</td><td>212700</td><td>189174</td><td>131569</td><td>⋯</td><td> 20273</td><td>130972</td><td>  9391</td><td>  3473</td><td>109102</td><td>132457</td><td>130068</td><td> 46597</td><td> 53857</td><td>183430</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA</td><td>brain                            </td><td>137166</td><td> 35175</td><td>  4381</td><td>123690</td><td>99621</td><td> 35423</td><td>123344</td><td>108376</td><td>⋯</td><td>123710</td><td> 47008</td><td>134782</td><td>136306</td><td>137709</td><td>123045</td><td> 33837</td><td> 32261</td><td>136304</td><td>135305</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG</td><td>epidermis                        </td><td> 68440</td><td> 71253</td><td> 67238</td><td> 73868</td><td>67236</td><td> 71880</td><td> 64019</td><td> 64022</td><td>⋯</td><td> 64612</td><td> 63786</td><td> 70436</td><td> 65552</td><td>  2853</td><td> 65551</td><td> 67168</td><td> 39930</td><td> 28538</td><td> 74114</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA</td><td>sensory nervous system           </td><td>137728</td><td> 19780</td><td>120926</td><td> 91533</td><td> 5730</td><td>144152</td><td>143823</td><td> 35446</td><td>⋯</td><td> 51952</td><td>  4980</td><td>134914</td><td> 19964</td><td>132236</td><td> 52185</td><td> 34199</td><td>214925</td><td>117952</td><td> 68538</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA</td><td>sensory nervous system           </td><td> 36849</td><td>  1387</td><td>  2385</td><td> 63108</td><td>36476</td><td>  2887</td><td> 67810</td><td>  3884</td><td>⋯</td><td>  5792</td><td> 75285</td><td>  2730</td><td> 19767</td><td> 34789</td><td> 35981</td><td>  1882</td><td> 64621</td><td> 49619</td><td>   781</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 22\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & ID & Label & V2 & V3 & V4 & V5 & V6 & V7 & V8 & V9 & ⋯ & V12 & V13 & V14 & V15 & V16 & V17 & V18 & V19 & V20 & V21\\\\\n",
       "  & <chr> & <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium &   1084 & 117474 &  33297 &   2934 &  2355 &    854 &   2437 &   4221 & ⋯ & 116766 &  48132 &  20116 & 115992 &  34544 &  31818 &  50700 & 143608 &    571 &   3626\\\\\n",
       "\t2 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG & visceral muscle primordium        &  51177 &  19874 &  69268 &  44378 & 67519 & 212700 & 189174 & 131569 & ⋯ &  20273 & 130972 &   9391 &   3473 & 109102 & 132457 & 130068 &  46597 &  53857 & 183430\\\\\n",
       "\t3 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA & brain                             & 137166 &  35175 &   4381 & 123690 & 99621 &  35423 & 123344 & 108376 & ⋯ & 123710 &  47008 & 134782 & 136306 & 137709 & 123045 &  33837 &  32261 & 136304 & 135305\\\\\n",
       "\t4 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG & epidermis                         &  68440 &  71253 &  67238 &  73868 & 67236 &  71880 &  64019 &  64022 & ⋯ &  64612 &  63786 &  70436 &  65552 &   2853 &  65551 &  67168 &  39930 &  28538 &  74114\\\\\n",
       "\t5 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA & sensory nervous system            & 137728 &  19780 & 120926 &  91533 &  5730 & 144152 & 143823 &  35446 & ⋯ &  51952 &   4980 & 134914 &  19964 & 132236 &  52185 &  34199 & 214925 & 117952 &  68538\\\\\n",
       "\t6 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA & sensory nervous system            &  36849 &   1387 &   2385 &  63108 & 36476 &   2887 &  67810 &   3884 & ⋯ &   5792 &  75285 &   2730 &  19767 &  34789 &  35981 &   1882 &  64621 &  49619 &    781\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 22\n",
       "\n",
       "| <!--/--> | ID &lt;chr&gt; | Label &lt;chr&gt; | V2 &lt;int&gt; | V3 &lt;int&gt; | V4 &lt;int&gt; | V5 &lt;int&gt; | V6 &lt;int&gt; | V7 &lt;int&gt; | V8 &lt;int&gt; | V9 &lt;int&gt; | ⋯ ⋯ | V12 &lt;int&gt; | V13 &lt;int&gt; | V14 &lt;int&gt; | V15 &lt;int&gt; | V16 &lt;int&gt; | V17 &lt;int&gt; | V18 &lt;int&gt; | V19 &lt;int&gt; | V20 &lt;int&gt; | V21 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium |   1084 | 117474 |  33297 |   2934 |  2355 |    854 |   2437 |   4221 | ⋯ | 116766 |  48132 |  20116 | 115992 |  34544 |  31818 |  50700 | 143608 |    571 |   3626 |\n",
       "| 2 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG | visceral muscle primordium        |  51177 |  19874 |  69268 |  44378 | 67519 | 212700 | 189174 | 131569 | ⋯ |  20273 | 130972 |   9391 |   3473 | 109102 | 132457 | 130068 |  46597 |  53857 | 183430 |\n",
       "| 3 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA | brain                             | 137166 |  35175 |   4381 | 123690 | 99621 |  35423 | 123344 | 108376 | ⋯ | 123710 |  47008 | 134782 | 136306 | 137709 | 123045 |  33837 |  32261 | 136304 | 135305 |\n",
       "| 4 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG | epidermis                         |  68440 |  71253 |  67238 |  73868 | 67236 |  71880 |  64019 |  64022 | ⋯ |  64612 |  63786 |  70436 |  65552 |   2853 |  65551 |  67168 |  39930 |  28538 |  74114 |\n",
       "| 5 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA | sensory nervous system            | 137728 |  19780 | 120926 |  91533 |  5730 | 144152 | 143823 |  35446 | ⋯ |  51952 |   4980 | 134914 |  19964 | 132236 |  52185 |  34199 | 214925 | 117952 |  68538 |\n",
       "| 6 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA | sensory nervous system            |  36849 |   1387 |   2385 |  63108 | 36476 |   2887 |  67810 |   3884 | ⋯ |   5792 |  75285 |   2730 |  19767 |  34789 |  35981 |   1882 |  64621 |  49619 |    781 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID                                                    \n",
       "1 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "2 exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG\n",
       "3 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA\n",
       "4 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG\n",
       "5 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA\n",
       "6 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA\n",
       "  Label                             V2     V3     V4     V5     V6    V7    \n",
       "1 sensory nervous system primordium   1084 117474  33297   2934  2355    854\n",
       "2 visceral muscle primordium         51177  19874  69268  44378 67519 212700\n",
       "3 brain                             137166  35175   4381 123690 99621  35423\n",
       "4 epidermis                          68440  71253  67238  73868 67236  71880\n",
       "5 sensory nervous system            137728  19780 120926  91533  5730 144152\n",
       "6 sensory nervous system             36849   1387   2385  63108 36476   2887\n",
       "  V8     V9     ⋯ V12    V13    V14    V15    V16    V17    V18    V19   \n",
       "1   2437   4221 ⋯ 116766  48132  20116 115992  34544  31818  50700 143608\n",
       "2 189174 131569 ⋯  20273 130972   9391   3473 109102 132457 130068  46597\n",
       "3 123344 108376 ⋯ 123710  47008 134782 136306 137709 123045  33837  32261\n",
       "4  64019  64022 ⋯  64612  63786  70436  65552   2853  65551  67168  39930\n",
       "5 143823  35446 ⋯  51952   4980 134914  19964 132236  52185  34199 214925\n",
       "6  67810   3884 ⋯   5792  75285   2730  19767  34789  35981   1882  64621\n",
       "  V20    V21   \n",
       "1    571   3626\n",
       "2  53857 183430\n",
       "3 136304 135305\n",
       "4  28538  74114\n",
       "5 117952  68538\n",
       "6  49619    781"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“\u001b[1m\u001b[22mUsing an external vector in selections was deprecated in tidyselect 1.1.0.\n",
      "\u001b[36mℹ\u001b[39m Please use `all_of()` or `any_of()` instead.\n",
      "  # Was:\n",
      "  data %>% select(k1)\n",
      "\n",
      "  # Now:\n",
      "  data %>% select(all_of(k1))\n",
      "\n",
      "See <https://tidyselect.r-lib.org/reference/faq-external-vector.html>.”\n"
     ]
    }
   ],
   "source": [
    "df <- pivot_longer(df,cols=3:k1,names_to = \"NN\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "df <- mutate(df,nnLabel=labels[value])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>Label</th><th scope=col>V21</th><th scope=col>NN</th><th scope=col>value</th><th scope=col>nnLabel</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>3626</td><td>V2</td><td>  1084</td><td>plasmatocytes anlage             </td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>3626</td><td>V3</td><td>117474</td><td>plasmatocytes anlage             </td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>3626</td><td>V4</td><td> 33297</td><td>sensory nervous system primordium</td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>3626</td><td>V5</td><td>  2934</td><td>plasmatocytes anlage             </td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>3626</td><td>V6</td><td>  2355</td><td>plasmatocytes anlage             </td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>3626</td><td>V7</td><td>   854</td><td>plasmatocytes anlage             </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 6\n",
       "\\begin{tabular}{llllll}\n",
       " ID & Label & V21 & NN & value & nnLabel\\\\\n",
       " <chr> & <chr> & <int> & <chr> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & 3626 & V2 &   1084 & plasmatocytes anlage             \\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & 3626 & V3 & 117474 & plasmatocytes anlage             \\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & 3626 & V4 &  33297 & sensory nervous system primordium\\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & 3626 & V5 &   2934 & plasmatocytes anlage             \\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & 3626 & V6 &   2355 & plasmatocytes anlage             \\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & 3626 & V7 &    854 & plasmatocytes anlage             \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 6\n",
       "\n",
       "| ID &lt;chr&gt; | Label &lt;chr&gt; | V21 &lt;int&gt; | NN &lt;chr&gt; | value &lt;int&gt; | nnLabel &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | 3626 | V2 |   1084 | plasmatocytes anlage              |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | 3626 | V3 | 117474 | plasmatocytes anlage              |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | 3626 | V4 |  33297 | sensory nervous system primordium |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | 3626 | V5 |   2934 | plasmatocytes anlage              |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | 3626 | V6 |   2355 | plasmatocytes anlage              |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | 3626 | V7 |    854 | plasmatocytes anlage              |\n",
       "\n"
      ],
      "text/plain": [
       "  ID                                                    \n",
       "1 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "2 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "3 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "4 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "5 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "6 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "  Label                             V21  NN value \n",
       "1 sensory nervous system primordium 3626 V2   1084\n",
       "2 sensory nervous system primordium 3626 V3 117474\n",
       "3 sensory nervous system primordium 3626 V4  33297\n",
       "4 sensory nervous system primordium 3626 V5   2934\n",
       "5 sensory nervous system primordium 3626 V6   2355\n",
       "6 sensory nervous system primordium 3626 V7    854\n",
       "  nnLabel                          \n",
       "1 plasmatocytes anlage             \n",
       "2 plasmatocytes anlage             \n",
       "3 sensory nervous system primordium\n",
       "4 plasmatocytes anlage             \n",
       "5 plasmatocytes anlage             \n",
       "6 plasmatocytes anlage             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df, 6)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "df <- select(df,-NN,-value)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>Label</th><th scope=col>V21</th><th scope=col>nnLabel</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>3626</td><td>plasmatocytes anlage             </td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>3626</td><td>plasmatocytes anlage             </td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>3626</td><td>sensory nervous system primordium</td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>3626</td><td>plasmatocytes anlage             </td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>3626</td><td>plasmatocytes anlage             </td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>3626</td><td>plasmatocytes anlage             </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 4\n",
       "\\begin{tabular}{llll}\n",
       " ID & Label & V21 & nnLabel\\\\\n",
       " <chr> & <chr> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & 3626 & plasmatocytes anlage             \\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & 3626 & plasmatocytes anlage             \\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & 3626 & sensory nervous system primordium\\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & 3626 & plasmatocytes anlage             \\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & 3626 & plasmatocytes anlage             \\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & 3626 & plasmatocytes anlage             \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 4\n",
       "\n",
       "| ID &lt;chr&gt; | Label &lt;chr&gt; | V21 &lt;int&gt; | nnLabel &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | 3626 | plasmatocytes anlage              |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | 3626 | plasmatocytes anlage              |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | 3626 | sensory nervous system primordium |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | 3626 | plasmatocytes anlage              |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | 3626 | plasmatocytes anlage              |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | 3626 | plasmatocytes anlage              |\n",
       "\n"
      ],
      "text/plain": [
       "  ID                                                    \n",
       "1 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "2 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "3 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "4 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "5 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "6 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "  Label                             V21  nnLabel                          \n",
       "1 sensory nervous system primordium 3626 plasmatocytes anlage             \n",
       "2 sensory nervous system primordium 3626 plasmatocytes anlage             \n",
       "3 sensory nervous system primordium 3626 sensory nervous system primordium\n",
       "4 sensory nervous system primordium 3626 plasmatocytes anlage             \n",
       "5 sensory nervous system primordium 3626 plasmatocytes anlage             \n",
       "6 sensory nervous system primordium 3626 plasmatocytes anlage             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>4530455</li><li>4</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 4530455\n",
       "\\item 4\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 4530455\n",
       "2. 4\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 4530455       4"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 50 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Label</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Malpighian tubule                   </td><td>  6405</td></tr>\n",
       "\t<tr><td>amnioserosa                         </td><td>117350</td></tr>\n",
       "\t<tr><td>anlage in statu nascendi            </td><td>    35</td></tr>\n",
       "\t<tr><td>brain                               </td><td>229655</td></tr>\n",
       "\t<tr><td>brain primordium                    </td><td> 52907</td></tr>\n",
       "\t<tr><td>cellular blastoderm / pole cell     </td><td>     7</td></tr>\n",
       "\t<tr><td>crystal cell                        </td><td>   160</td></tr>\n",
       "\t<tr><td>ectoderm anlage in statu nascendi   </td><td>    76</td></tr>\n",
       "\t<tr><td>ectoderm primordium                 </td><td>  9695</td></tr>\n",
       "\t<tr><td>endoderm primordium                 </td><td>  1384</td></tr>\n",
       "\t<tr><td>epidermis                           </td><td>316327</td></tr>\n",
       "\t<tr><td>epidermis primordium                </td><td> 27379</td></tr>\n",
       "\t<tr><td>fat body                            </td><td> 66813</td></tr>\n",
       "\t<tr><td>foregut / hindgut / anal pad        </td><td> 32872</td></tr>\n",
       "\t<tr><td>foregut / hindgut primordium        </td><td>  6393</td></tr>\n",
       "\t<tr><td>germ cell                           </td><td>  9648</td></tr>\n",
       "\t<tr><td>glia                                </td><td>  5065</td></tr>\n",
       "\t<tr><td>hindgut / anal pad                  </td><td> 21056</td></tr>\n",
       "\t<tr><td>hindgut primordium                  </td><td>  7860</td></tr>\n",
       "\t<tr><td>maternal                            </td><td>    42</td></tr>\n",
       "\t<tr><td>mesoderm primordium                 </td><td>  1246</td></tr>\n",
       "\t<tr><td>midgut                              </td><td>102045</td></tr>\n",
       "\t<tr><td>midgut primordium                   </td><td>  8842</td></tr>\n",
       "\t<tr><td>midline primordium                  </td><td>  7726</td></tr>\n",
       "\t<tr><td>muscle                              </td><td> 33260</td></tr>\n",
       "\t<tr><td>peripheral nervous system           </td><td> 22567</td></tr>\n",
       "\t<tr><td>peripheral nervous system primordium</td><td>  1916</td></tr>\n",
       "\t<tr><td>plasmatocytes                       </td><td> 64848</td></tr>\n",
       "\t<tr><td>plasmatocytes anlage                </td><td> 11913</td></tr>\n",
       "\t<tr><td>procrystal cell                     </td><td> 12483</td></tr>\n",
       "\t<tr><td>proventriculus                      </td><td> 16907</td></tr>\n",
       "\t<tr><td>salivary gland                      </td><td> 47966</td></tr>\n",
       "\t<tr><td>salivary gland body primordium      </td><td>  8120</td></tr>\n",
       "\t<tr><td>salivary gland primordium           </td><td>    36</td></tr>\n",
       "\t<tr><td>sensory nervous system              </td><td> 25159</td></tr>\n",
       "\t<tr><td>sensory nervous system primordium   </td><td>  5060</td></tr>\n",
       "\t<tr><td>somatic muscle                      </td><td>167973</td></tr>\n",
       "\t<tr><td>somatic muscle primordium           </td><td>  6485</td></tr>\n",
       "\t<tr><td>spiracle (tracheal system)          </td><td>  2074</td></tr>\n",
       "\t<tr><td>spiracles (tracheal system)         </td><td>  5503</td></tr>\n",
       "\t<tr><td>tracheal primordium                 </td><td> 11655</td></tr>\n",
       "\t<tr><td>tracheal system                     </td><td> 82411</td></tr>\n",
       "\t<tr><td>trunk mesoderm primordium           </td><td>  2205</td></tr>\n",
       "\t<tr><td>ubiquitous                          </td><td>260372</td></tr>\n",
       "\t<tr><td>unknown                             </td><td>  2910</td></tr>\n",
       "\t<tr><td>ventral nerve cord                  </td><td> 69616</td></tr>\n",
       "\t<tr><td>ventral nerve cord primordium       </td><td>  1780</td></tr>\n",
       "\t<tr><td>visceral muscle                     </td><td>  2797</td></tr>\n",
       "\t<tr><td>visceral muscle primordium          </td><td> 14705</td></tr>\n",
       "\t<tr><td>yolk nuclei                         </td><td>   847</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 50 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Label & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Malpighian tubule                    &   6405\\\\\n",
       "\t amnioserosa                          & 117350\\\\\n",
       "\t anlage in statu nascendi             &     35\\\\\n",
       "\t brain                                & 229655\\\\\n",
       "\t brain primordium                     &  52907\\\\\n",
       "\t cellular blastoderm / pole cell      &      7\\\\\n",
       "\t crystal cell                         &    160\\\\\n",
       "\t ectoderm anlage in statu nascendi    &     76\\\\\n",
       "\t ectoderm primordium                  &   9695\\\\\n",
       "\t endoderm primordium                  &   1384\\\\\n",
       "\t epidermis                            & 316327\\\\\n",
       "\t epidermis primordium                 &  27379\\\\\n",
       "\t fat body                             &  66813\\\\\n",
       "\t foregut / hindgut / anal pad         &  32872\\\\\n",
       "\t foregut / hindgut primordium         &   6393\\\\\n",
       "\t germ cell                            &   9648\\\\\n",
       "\t glia                                 &   5065\\\\\n",
       "\t hindgut / anal pad                   &  21056\\\\\n",
       "\t hindgut primordium                   &   7860\\\\\n",
       "\t maternal                             &     42\\\\\n",
       "\t mesoderm primordium                  &   1246\\\\\n",
       "\t midgut                               & 102045\\\\\n",
       "\t midgut primordium                    &   8842\\\\\n",
       "\t midline primordium                   &   7726\\\\\n",
       "\t muscle                               &  33260\\\\\n",
       "\t peripheral nervous system            &  22567\\\\\n",
       "\t peripheral nervous system primordium &   1916\\\\\n",
       "\t plasmatocytes                        &  64848\\\\\n",
       "\t plasmatocytes anlage                 &  11913\\\\\n",
       "\t procrystal cell                      &  12483\\\\\n",
       "\t proventriculus                       &  16907\\\\\n",
       "\t salivary gland                       &  47966\\\\\n",
       "\t salivary gland body primordium       &   8120\\\\\n",
       "\t salivary gland primordium            &     36\\\\\n",
       "\t sensory nervous system               &  25159\\\\\n",
       "\t sensory nervous system primordium    &   5060\\\\\n",
       "\t somatic muscle                       & 167973\\\\\n",
       "\t somatic muscle primordium            &   6485\\\\\n",
       "\t spiracle (tracheal system)           &   2074\\\\\n",
       "\t spiracles (tracheal system)          &   5503\\\\\n",
       "\t tracheal primordium                  &  11655\\\\\n",
       "\t tracheal system                      &  82411\\\\\n",
       "\t trunk mesoderm primordium            &   2205\\\\\n",
       "\t ubiquitous                           & 260372\\\\\n",
       "\t unknown                              &   2910\\\\\n",
       "\t ventral nerve cord                   &  69616\\\\\n",
       "\t ventral nerve cord primordium        &   1780\\\\\n",
       "\t visceral muscle                      &   2797\\\\\n",
       "\t visceral muscle primordium           &  14705\\\\\n",
       "\t yolk nuclei                          &    847\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 50 × 2\n",
       "\n",
       "| Label &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Malpighian tubule                    |   6405 |\n",
       "| amnioserosa                          | 117350 |\n",
       "| anlage in statu nascendi             |     35 |\n",
       "| brain                                | 229655 |\n",
       "| brain primordium                     |  52907 |\n",
       "| cellular blastoderm / pole cell      |      7 |\n",
       "| crystal cell                         |    160 |\n",
       "| ectoderm anlage in statu nascendi    |     76 |\n",
       "| ectoderm primordium                  |   9695 |\n",
       "| endoderm primordium                  |   1384 |\n",
       "| epidermis                            | 316327 |\n",
       "| epidermis primordium                 |  27379 |\n",
       "| fat body                             |  66813 |\n",
       "| foregut / hindgut / anal pad         |  32872 |\n",
       "| foregut / hindgut primordium         |   6393 |\n",
       "| germ cell                            |   9648 |\n",
       "| glia                                 |   5065 |\n",
       "| hindgut / anal pad                   |  21056 |\n",
       "| hindgut primordium                   |   7860 |\n",
       "| maternal                             |     42 |\n",
       "| mesoderm primordium                  |   1246 |\n",
       "| midgut                               | 102045 |\n",
       "| midgut primordium                    |   8842 |\n",
       "| midline primordium                   |   7726 |\n",
       "| muscle                               |  33260 |\n",
       "| peripheral nervous system            |  22567 |\n",
       "| peripheral nervous system primordium |   1916 |\n",
       "| plasmatocytes                        |  64848 |\n",
       "| plasmatocytes anlage                 |  11913 |\n",
       "| procrystal cell                      |  12483 |\n",
       "| proventriculus                       |  16907 |\n",
       "| salivary gland                       |  47966 |\n",
       "| salivary gland body primordium       |   8120 |\n",
       "| salivary gland primordium            |     36 |\n",
       "| sensory nervous system               |  25159 |\n",
       "| sensory nervous system primordium    |   5060 |\n",
       "| somatic muscle                       | 167973 |\n",
       "| somatic muscle primordium            |   6485 |\n",
       "| spiracle (tracheal system)           |   2074 |\n",
       "| spiracles (tracheal system)          |   5503 |\n",
       "| tracheal primordium                  |  11655 |\n",
       "| tracheal system                      |  82411 |\n",
       "| trunk mesoderm primordium            |   2205 |\n",
       "| ubiquitous                           | 260372 |\n",
       "| unknown                              |   2910 |\n",
       "| ventral nerve cord                   |  69616 |\n",
       "| ventral nerve cord primordium        |   1780 |\n",
       "| visceral muscle                      |   2797 |\n",
       "| visceral muscle primordium           |  14705 |\n",
       "| yolk nuclei                          |    847 |\n",
       "\n"
      ],
      "text/plain": [
       "   Label                                n     \n",
       "1  Malpighian tubule                      6405\n",
       "2  amnioserosa                          117350\n",
       "3  anlage in statu nascendi                 35\n",
       "4  brain                                229655\n",
       "5  brain primordium                      52907\n",
       "6  cellular blastoderm / pole cell           7\n",
       "7  crystal cell                            160\n",
       "8  ectoderm anlage in statu nascendi        76\n",
       "9  ectoderm primordium                    9695\n",
       "10 endoderm primordium                    1384\n",
       "11 epidermis                            316327\n",
       "12 epidermis primordium                  27379\n",
       "13 fat body                              66813\n",
       "14 foregut / hindgut / anal pad          32872\n",
       "15 foregut / hindgut primordium           6393\n",
       "16 germ cell                              9648\n",
       "17 glia                                   5065\n",
       "18 hindgut / anal pad                    21056\n",
       "19 hindgut primordium                     7860\n",
       "20 maternal                                 42\n",
       "21 mesoderm primordium                    1246\n",
       "22 midgut                               102045\n",
       "23 midgut primordium                      8842\n",
       "24 midline primordium                     7726\n",
       "25 muscle                                33260\n",
       "26 peripheral nervous system             22567\n",
       "27 peripheral nervous system primordium   1916\n",
       "28 plasmatocytes                         64848\n",
       "29 plasmatocytes anlage                  11913\n",
       "30 procrystal cell                       12483\n",
       "31 proventriculus                        16907\n",
       "32 salivary gland                        47966\n",
       "33 salivary gland body primordium         8120\n",
       "34 salivary gland primordium                36\n",
       "35 sensory nervous system                25159\n",
       "36 sensory nervous system primordium      5060\n",
       "37 somatic muscle                       167973\n",
       "38 somatic muscle primordium              6485\n",
       "39 spiracle (tracheal system)             2074\n",
       "40 spiracles (tracheal system)            5503\n",
       "41 tracheal primordium                   11655\n",
       "42 tracheal system                       82411\n",
       "43 trunk mesoderm primordium              2205\n",
       "44 ubiquitous                           260372\n",
       "45 unknown                                2910\n",
       "46 ventral nerve cord                    69616\n",
       "47 ventral nerve cord primordium          1780\n",
       "48 visceral muscle                        2797\n",
       "49 visceral muscle primordium            14705\n",
       "50 yolk nuclei                             847"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df  %>% filter(Label==nnLabel)  %>% count(Label)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "res <- group_by(df,ID) %>% summarise(nk=sum (Label==nnLabel)  )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>nk</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAACTCCAATCGTCCGTCCT</td><td> 6</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AACTAGTTGTAACGGACCAA</td><td> 5</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATAACTAGGCGC </td><td> 5</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATCCTAGCGCCT </td><td>15</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG </td><td> 1</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td> 0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " ID & nk\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAACTCCAATCGTCCGTCCT &  6\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AACTAGTTGTAACGGACCAA &  5\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGACGATAACTAGGCGC  &  5\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGACGATCCTAGCGCCT  & 15\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG  &  1\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA &  0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 2\n",
       "\n",
       "| ID &lt;chr&gt; | nk &lt;int&gt; |\n",
       "|---|---|\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAACTCCAATCGTCCGTCCT |  6 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AACTAGTTGTAACGGACCAA |  5 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATAACTAGGCGC  |  5 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATCCTAGCGCCT  | 15 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG  |  1 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA |  0 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID                                                       nk\n",
       "1 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAACTCCAATCGTCCGTCCT  6\n",
       "2 exp10_idx1AATACGATAA_idx2AACTACCGAC.AACTAGTTGTAACGGACCAA  5\n",
       "3 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATAACTAGGCGC   5\n",
       "4 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATCCTAGCGCCT  15\n",
       "5 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG   1\n",
       "6 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA  0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(res)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1214,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>238445</li><li>2</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 238445\n",
       "\\item 2\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 238445\n",
       "2. 2\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 238445      2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(res)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 20 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>nk</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 0</td><td>26947</td></tr>\n",
       "\t<tr><td> 1</td><td>20080</td></tr>\n",
       "\t<tr><td> 2</td><td>16807</td></tr>\n",
       "\t<tr><td> 3</td><td>14894</td></tr>\n",
       "\t<tr><td> 4</td><td>13366</td></tr>\n",
       "\t<tr><td> 5</td><td>11919</td></tr>\n",
       "\t<tr><td> 6</td><td>11043</td></tr>\n",
       "\t<tr><td> 7</td><td>10165</td></tr>\n",
       "\t<tr><td> 8</td><td> 9928</td></tr>\n",
       "\t<tr><td> 9</td><td> 9572</td></tr>\n",
       "\t<tr><td>10</td><td> 9351</td></tr>\n",
       "\t<tr><td>11</td><td> 8829</td></tr>\n",
       "\t<tr><td>12</td><td> 8498</td></tr>\n",
       "\t<tr><td>13</td><td> 8114</td></tr>\n",
       "\t<tr><td>14</td><td> 7685</td></tr>\n",
       "\t<tr><td>15</td><td> 7889</td></tr>\n",
       "\t<tr><td>16</td><td> 7796</td></tr>\n",
       "\t<tr><td>17</td><td> 8233</td></tr>\n",
       "\t<tr><td>18</td><td> 9824</td></tr>\n",
       "\t<tr><td>19</td><td>17505</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 20 × 2\n",
       "\\begin{tabular}{ll}\n",
       " nk & n\\\\\n",
       " <int> & <int>\\\\\n",
       "\\hline\n",
       "\t  0 & 26947\\\\\n",
       "\t  1 & 20080\\\\\n",
       "\t  2 & 16807\\\\\n",
       "\t  3 & 14894\\\\\n",
       "\t  4 & 13366\\\\\n",
       "\t  5 & 11919\\\\\n",
       "\t  6 & 11043\\\\\n",
       "\t  7 & 10165\\\\\n",
       "\t  8 &  9928\\\\\n",
       "\t  9 &  9572\\\\\n",
       "\t 10 &  9351\\\\\n",
       "\t 11 &  8829\\\\\n",
       "\t 12 &  8498\\\\\n",
       "\t 13 &  8114\\\\\n",
       "\t 14 &  7685\\\\\n",
       "\t 15 &  7889\\\\\n",
       "\t 16 &  7796\\\\\n",
       "\t 17 &  8233\\\\\n",
       "\t 18 &  9824\\\\\n",
       "\t 19 & 17505\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 20 × 2\n",
       "\n",
       "| nk &lt;int&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "|  0 | 26947 |\n",
       "|  1 | 20080 |\n",
       "|  2 | 16807 |\n",
       "|  3 | 14894 |\n",
       "|  4 | 13366 |\n",
       "|  5 | 11919 |\n",
       "|  6 | 11043 |\n",
       "|  7 | 10165 |\n",
       "|  8 |  9928 |\n",
       "|  9 |  9572 |\n",
       "| 10 |  9351 |\n",
       "| 11 |  8829 |\n",
       "| 12 |  8498 |\n",
       "| 13 |  8114 |\n",
       "| 14 |  7685 |\n",
       "| 15 |  7889 |\n",
       "| 16 |  7796 |\n",
       "| 17 |  8233 |\n",
       "| 18 |  9824 |\n",
       "| 19 | 17505 |\n",
       "\n"
      ],
      "text/plain": [
       "   nk n    \n",
       "1   0 26947\n",
       "2   1 20080\n",
       "3   2 16807\n",
       "4   3 14894\n",
       "5   4 13366\n",
       "6   5 11919\n",
       "7   6 11043\n",
       "8   7 10165\n",
       "9   8  9928\n",
       "10  9  9572\n",
       "11 10  9351\n",
       "12 11  8829\n",
       "13 12  8498\n",
       "14 13  8114\n",
       "15 14  7685\n",
       "16 15  7889\n",
       "17 16  7796\n",
       "18 17  8233\n",
       "19 18  9824\n",
       "20 19 17505"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "res  %>% count(nk)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1221,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>nk</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG </td><td>1</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.CAAGCAACTGCTCTCGCCT </td><td>1</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.CAGCAAGTTGCCGCAACGA </td><td>1</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.GACCAGGATCCTAGCGCCT </td><td>1</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.TAATACCAGTATCCATGACT</td><td>1</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.TAGCCAATTTCATCTCTGCA</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " ID & nk\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG  & 1\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.CAAGCAACTGCTCTCGCCT  & 1\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.CAGCAAGTTGCCGCAACGA  & 1\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.GACCAGGATCCTAGCGCCT  & 1\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.TAATACCAGTATCCATGACT & 1\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.TAGCCAATTTCATCTCTGCA & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 2\n",
       "\n",
       "| ID &lt;chr&gt; | nk &lt;int&gt; |\n",
       "|---|---|\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG  | 1 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.CAAGCAACTGCTCTCGCCT  | 1 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.CAGCAAGTTGCCGCAACGA  | 1 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.GACCAGGATCCTAGCGCCT  | 1 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.TAATACCAGTATCCATGACT | 1 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.TAGCCAATTTCATCTCTGCA | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID                                                       nk\n",
       "1 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG  1 \n",
       "2 exp10_idx1AATACGATAA_idx2AACTACCGAC.CAAGCAACTGCTCTCGCCT  1 \n",
       "3 exp10_idx1AATACGATAA_idx2AACTACCGAC.CAGCAAGTTGCCGCAACGA  1 \n",
       "4 exp10_idx1AATACGATAA_idx2AACTACCGAC.GACCAGGATCCTAGCGCCT  1 \n",
       "5 exp10_idx1AATACGATAA_idx2AACTACCGAC.TAATACCAGTATCCATGACT 1 \n",
       "6 exp10_idx1AATACGATAA_idx2AACTACCGAC.TAGCCAATTTCATCTCTGCA 1 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "res %>% filter(nk == 1)  %>% head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1220,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 19 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>Label</th><th scope=col>V21</th><th scope=col>nnLabel</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>210714</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>210714</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>210714</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>210714</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>210714</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>210714</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>210714</td><td>midgut</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>210714</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>210714</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>210714</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>210714</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>210714</td><td>midgut</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>210714</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>210714</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>210714</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>210714</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>210714</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>210714</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>210714</td><td>midgut</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 19 × 4\n",
       "\\begin{tabular}{llll}\n",
       " ID & Label & V21 & nnLabel\\\\\n",
       " <chr> & <chr> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & 210714 & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & 210714 & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & 210714 & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & 210714 & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & 210714 & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & 210714 & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & 210714 & midgut\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & 210714 & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & 210714 & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & 210714 & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & 210714 & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & 210714 & midgut\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & 210714 & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & 210714 & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & 210714 & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & 210714 & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & 210714 & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & 210714 & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & 210714 & midgut\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 19 × 4\n",
       "\n",
       "| ID &lt;chr&gt; | Label &lt;chr&gt; | V21 &lt;int&gt; | nnLabel &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | 210714 | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | 210714 | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | 210714 | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | 210714 | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | 210714 | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | 210714 | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | 210714 | midgut |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | 210714 | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | 210714 | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | 210714 | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | 210714 | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | 210714 | midgut |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | 210714 | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | 210714 | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | 210714 | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | 210714 | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | 210714 | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | 210714 | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | 210714 | midgut |\n",
       "\n"
      ],
      "text/plain": [
       "   ID                                                       Label  V21   \n",
       "1  exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle 210714\n",
       "2  exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle 210714\n",
       "3  exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle 210714\n",
       "4  exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle 210714\n",
       "5  exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle 210714\n",
       "6  exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle 210714\n",
       "7  exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle 210714\n",
       "8  exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle 210714\n",
       "9  exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle 210714\n",
       "10 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle 210714\n",
       "11 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle 210714\n",
       "12 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle 210714\n",
       "13 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle 210714\n",
       "14 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle 210714\n",
       "15 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle 210714\n",
       "16 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle 210714\n",
       "17 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle 210714\n",
       "18 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle 210714\n",
       "19 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle 210714\n",
       "   nnLabel\n",
       "1  brain  \n",
       "2  brain  \n",
       "3  brain  \n",
       "4  brain  \n",
       "5  brain  \n",
       "6  brain  \n",
       "7  midgut \n",
       "8  brain  \n",
       "9  brain  \n",
       "10 brain  \n",
       "11 brain  \n",
       "12 midgut \n",
       "13 brain  \n",
       "14 brain  \n",
       "15 brain  \n",
       "16 brain  \n",
       "17 brain  \n",
       "18 brain  \n",
       "19 midgut "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df  %>% filter(ID == \"exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'ID'. You can override using the `.groups` argument.\n"
     ]
    }
   ],
   "source": [
    "res_label <- group_by(df,ID, Label) %>% summarise(nk=sum (Label==nnLabel)  )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>Label</th><th scope=col>nk</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAACTCCAATCGTCCGTCCT</td><td>ubiquitous</td><td> 6</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AACTAGTTGTAACGGACCAA</td><td>midgut    </td><td> 5</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATAACTAGGCGC </td><td>muscle    </td><td> 5</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATCCTAGCGCCT </td><td>brain     </td><td>15</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG </td><td>epidermis </td><td> 1</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle    </td><td> 0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 6 × 3\n",
       "\\begin{tabular}{lll}\n",
       " ID & Label & nk\\\\\n",
       " <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAACTCCAATCGTCCGTCCT & ubiquitous &  6\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AACTAGTTGTAACGGACCAA & midgut     &  5\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGACGATAACTAGGCGC  & muscle     &  5\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGACGATCCTAGCGCCT  & brain      & 15\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG  & epidermis  &  1\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle     &  0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 6 × 3\n",
       "\n",
       "| ID &lt;chr&gt; | Label &lt;chr&gt; | nk &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAACTCCAATCGTCCGTCCT | ubiquitous |  6 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AACTAGTTGTAACGGACCAA | midgut     |  5 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATAACTAGGCGC  | muscle     |  5 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATCCTAGCGCCT  | brain      | 15 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG  | epidermis  |  1 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle     |  0 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID                                                       Label      nk\n",
       "1 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAACTCCAATCGTCCGTCCT ubiquitous  6\n",
       "2 exp10_idx1AATACGATAA_idx2AACTACCGAC.AACTAGTTGTAACGGACCAA midgut      5\n",
       "3 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATAACTAGGCGC  muscle      5\n",
       "4 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATCCTAGCGCCT  brain      15\n",
       "5 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG  epidermis   1\n",
       "6 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle      0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(res_label)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 51 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Label</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Malpighian tubule                   </td><td> 252</td></tr>\n",
       "\t<tr><td>amnioserosa                         </td><td> 487</td></tr>\n",
       "\t<tr><td>anlage in statu nascendi            </td><td>  21</td></tr>\n",
       "\t<tr><td>brain                               </td><td> 935</td></tr>\n",
       "\t<tr><td>brain primordium                    </td><td> 904</td></tr>\n",
       "\t<tr><td>cellular blastoderm / pole cell     </td><td>  12</td></tr>\n",
       "\t<tr><td>crystal cell                        </td><td> 110</td></tr>\n",
       "\t<tr><td>ectoderm anlage                     </td><td>   5</td></tr>\n",
       "\t<tr><td>ectoderm anlage in statu nascendi   </td><td>  18</td></tr>\n",
       "\t<tr><td>ectoderm primordium                 </td><td> 149</td></tr>\n",
       "\t<tr><td>endoderm primordium                 </td><td> 103</td></tr>\n",
       "\t<tr><td>epidermis                           </td><td>2534</td></tr>\n",
       "\t<tr><td>epidermis primordium                </td><td> 745</td></tr>\n",
       "\t<tr><td>fat body                            </td><td> 480</td></tr>\n",
       "\t<tr><td>foregut / hindgut / anal pad        </td><td>1803</td></tr>\n",
       "\t<tr><td>foregut / hindgut primordium        </td><td> 225</td></tr>\n",
       "\t<tr><td>germ cell                           </td><td> 284</td></tr>\n",
       "\t<tr><td>glia                                </td><td> 236</td></tr>\n",
       "\t<tr><td>hindgut / anal pad                  </td><td> 380</td></tr>\n",
       "\t<tr><td>hindgut primordium                  </td><td> 395</td></tr>\n",
       "\t<tr><td>maternal                            </td><td>   5</td></tr>\n",
       "\t<tr><td>mesoderm primordium                 </td><td> 100</td></tr>\n",
       "\t<tr><td>midgut                              </td><td>1227</td></tr>\n",
       "\t<tr><td>midgut primordium                   </td><td> 419</td></tr>\n",
       "\t<tr><td>midline primordium                  </td><td> 231</td></tr>\n",
       "\t<tr><td>muscle                              </td><td>1568</td></tr>\n",
       "\t<tr><td>peripheral nervous system           </td><td> 403</td></tr>\n",
       "\t<tr><td>peripheral nervous system primordium</td><td> 142</td></tr>\n",
       "\t<tr><td>plasmatocytes                       </td><td> 693</td></tr>\n",
       "\t<tr><td>plasmatocytes anlage                </td><td> 249</td></tr>\n",
       "\t<tr><td>procrystal cell                     </td><td> 246</td></tr>\n",
       "\t<tr><td>proventriculus                      </td><td> 395</td></tr>\n",
       "\t<tr><td>salivary gland                      </td><td> 247</td></tr>\n",
       "\t<tr><td>salivary gland body primordium      </td><td> 140</td></tr>\n",
       "\t<tr><td>salivary gland primordium           </td><td>  93</td></tr>\n",
       "\t<tr><td>sensory nervous system              </td><td>1091</td></tr>\n",
       "\t<tr><td>sensory nervous system primordium   </td><td> 386</td></tr>\n",
       "\t<tr><td>somatic muscle                      </td><td>1017</td></tr>\n",
       "\t<tr><td>somatic muscle primordium           </td><td> 389</td></tr>\n",
       "\t<tr><td>spiracle (tracheal system)          </td><td> 345</td></tr>\n",
       "\t<tr><td>spiracles (tracheal system)         </td><td> 135</td></tr>\n",
       "\t<tr><td>tracheal primordium                 </td><td> 538</td></tr>\n",
       "\t<tr><td>tracheal system                     </td><td>1149</td></tr>\n",
       "\t<tr><td>trunk mesoderm primordium           </td><td> 310</td></tr>\n",
       "\t<tr><td>ubiquitous                          </td><td>2755</td></tr>\n",
       "\t<tr><td>unknown                             </td><td> 565</td></tr>\n",
       "\t<tr><td>ventral nerve cord                  </td><td> 480</td></tr>\n",
       "\t<tr><td>ventral nerve cord primordium       </td><td> 145</td></tr>\n",
       "\t<tr><td>visceral muscle                     </td><td> 284</td></tr>\n",
       "\t<tr><td>visceral muscle primordium          </td><td> 656</td></tr>\n",
       "\t<tr><td>yolk nuclei                         </td><td> 466</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 51 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Label & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Malpighian tubule                    &  252\\\\\n",
       "\t amnioserosa                          &  487\\\\\n",
       "\t anlage in statu nascendi             &   21\\\\\n",
       "\t brain                                &  935\\\\\n",
       "\t brain primordium                     &  904\\\\\n",
       "\t cellular blastoderm / pole cell      &   12\\\\\n",
       "\t crystal cell                         &  110\\\\\n",
       "\t ectoderm anlage                      &    5\\\\\n",
       "\t ectoderm anlage in statu nascendi    &   18\\\\\n",
       "\t ectoderm primordium                  &  149\\\\\n",
       "\t endoderm primordium                  &  103\\\\\n",
       "\t epidermis                            & 2534\\\\\n",
       "\t epidermis primordium                 &  745\\\\\n",
       "\t fat body                             &  480\\\\\n",
       "\t foregut / hindgut / anal pad         & 1803\\\\\n",
       "\t foregut / hindgut primordium         &  225\\\\\n",
       "\t germ cell                            &  284\\\\\n",
       "\t glia                                 &  236\\\\\n",
       "\t hindgut / anal pad                   &  380\\\\\n",
       "\t hindgut primordium                   &  395\\\\\n",
       "\t maternal                             &    5\\\\\n",
       "\t mesoderm primordium                  &  100\\\\\n",
       "\t midgut                               & 1227\\\\\n",
       "\t midgut primordium                    &  419\\\\\n",
       "\t midline primordium                   &  231\\\\\n",
       "\t muscle                               & 1568\\\\\n",
       "\t peripheral nervous system            &  403\\\\\n",
       "\t peripheral nervous system primordium &  142\\\\\n",
       "\t plasmatocytes                        &  693\\\\\n",
       "\t plasmatocytes anlage                 &  249\\\\\n",
       "\t procrystal cell                      &  246\\\\\n",
       "\t proventriculus                       &  395\\\\\n",
       "\t salivary gland                       &  247\\\\\n",
       "\t salivary gland body primordium       &  140\\\\\n",
       "\t salivary gland primordium            &   93\\\\\n",
       "\t sensory nervous system               & 1091\\\\\n",
       "\t sensory nervous system primordium    &  386\\\\\n",
       "\t somatic muscle                       & 1017\\\\\n",
       "\t somatic muscle primordium            &  389\\\\\n",
       "\t spiracle (tracheal system)           &  345\\\\\n",
       "\t spiracles (tracheal system)          &  135\\\\\n",
       "\t tracheal primordium                  &  538\\\\\n",
       "\t tracheal system                      & 1149\\\\\n",
       "\t trunk mesoderm primordium            &  310\\\\\n",
       "\t ubiquitous                           & 2755\\\\\n",
       "\t unknown                              &  565\\\\\n",
       "\t ventral nerve cord                   &  480\\\\\n",
       "\t ventral nerve cord primordium        &  145\\\\\n",
       "\t visceral muscle                      &  284\\\\\n",
       "\t visceral muscle primordium           &  656\\\\\n",
       "\t yolk nuclei                          &  466\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 51 × 2\n",
       "\n",
       "| Label &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Malpighian tubule                    |  252 |\n",
       "| amnioserosa                          |  487 |\n",
       "| anlage in statu nascendi             |   21 |\n",
       "| brain                                |  935 |\n",
       "| brain primordium                     |  904 |\n",
       "| cellular blastoderm / pole cell      |   12 |\n",
       "| crystal cell                         |  110 |\n",
       "| ectoderm anlage                      |    5 |\n",
       "| ectoderm anlage in statu nascendi    |   18 |\n",
       "| ectoderm primordium                  |  149 |\n",
       "| endoderm primordium                  |  103 |\n",
       "| epidermis                            | 2534 |\n",
       "| epidermis primordium                 |  745 |\n",
       "| fat body                             |  480 |\n",
       "| foregut / hindgut / anal pad         | 1803 |\n",
       "| foregut / hindgut primordium         |  225 |\n",
       "| germ cell                            |  284 |\n",
       "| glia                                 |  236 |\n",
       "| hindgut / anal pad                   |  380 |\n",
       "| hindgut primordium                   |  395 |\n",
       "| maternal                             |    5 |\n",
       "| mesoderm primordium                  |  100 |\n",
       "| midgut                               | 1227 |\n",
       "| midgut primordium                    |  419 |\n",
       "| midline primordium                   |  231 |\n",
       "| muscle                               | 1568 |\n",
       "| peripheral nervous system            |  403 |\n",
       "| peripheral nervous system primordium |  142 |\n",
       "| plasmatocytes                        |  693 |\n",
       "| plasmatocytes anlage                 |  249 |\n",
       "| procrystal cell                      |  246 |\n",
       "| proventriculus                       |  395 |\n",
       "| salivary gland                       |  247 |\n",
       "| salivary gland body primordium       |  140 |\n",
       "| salivary gland primordium            |   93 |\n",
       "| sensory nervous system               | 1091 |\n",
       "| sensory nervous system primordium    |  386 |\n",
       "| somatic muscle                       | 1017 |\n",
       "| somatic muscle primordium            |  389 |\n",
       "| spiracle (tracheal system)           |  345 |\n",
       "| spiracles (tracheal system)          |  135 |\n",
       "| tracheal primordium                  |  538 |\n",
       "| tracheal system                      | 1149 |\n",
       "| trunk mesoderm primordium            |  310 |\n",
       "| ubiquitous                           | 2755 |\n",
       "| unknown                              |  565 |\n",
       "| ventral nerve cord                   |  480 |\n",
       "| ventral nerve cord primordium        |  145 |\n",
       "| visceral muscle                      |  284 |\n",
       "| visceral muscle primordium           |  656 |\n",
       "| yolk nuclei                          |  466 |\n",
       "\n"
      ],
      "text/plain": [
       "   Label                                n   \n",
       "1  Malpighian tubule                     252\n",
       "2  amnioserosa                           487\n",
       "3  anlage in statu nascendi               21\n",
       "4  brain                                 935\n",
       "5  brain primordium                      904\n",
       "6  cellular blastoderm / pole cell        12\n",
       "7  crystal cell                          110\n",
       "8  ectoderm anlage                         5\n",
       "9  ectoderm anlage in statu nascendi      18\n",
       "10 ectoderm primordium                   149\n",
       "11 endoderm primordium                   103\n",
       "12 epidermis                            2534\n",
       "13 epidermis primordium                  745\n",
       "14 fat body                              480\n",
       "15 foregut / hindgut / anal pad         1803\n",
       "16 foregut / hindgut primordium          225\n",
       "17 germ cell                             284\n",
       "18 glia                                  236\n",
       "19 hindgut / anal pad                    380\n",
       "20 hindgut primordium                    395\n",
       "21 maternal                                5\n",
       "22 mesoderm primordium                   100\n",
       "23 midgut                               1227\n",
       "24 midgut primordium                     419\n",
       "25 midline primordium                    231\n",
       "26 muscle                               1568\n",
       "27 peripheral nervous system             403\n",
       "28 peripheral nervous system primordium  142\n",
       "29 plasmatocytes                         693\n",
       "30 plasmatocytes anlage                  249\n",
       "31 procrystal cell                       246\n",
       "32 proventriculus                        395\n",
       "33 salivary gland                        247\n",
       "34 salivary gland body primordium        140\n",
       "35 salivary gland primordium              93\n",
       "36 sensory nervous system               1091\n",
       "37 sensory nervous system primordium     386\n",
       "38 somatic muscle                       1017\n",
       "39 somatic muscle primordium             389\n",
       "40 spiracle (tracheal system)            345\n",
       "41 spiracles (tracheal system)           135\n",
       "42 tracheal primordium                   538\n",
       "43 tracheal system                      1149\n",
       "44 trunk mesoderm primordium             310\n",
       "45 ubiquitous                           2755\n",
       "46 unknown                               565\n",
       "47 ventral nerve cord                    480\n",
       "48 ventral nerve cord primordium         145\n",
       "49 visceral muscle                       284\n",
       "50 visceral muscle primordium            656\n",
       "51 yolk nuclei                           466"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "res_label  %>% filter(nk == 0)  %>% ungroup  %>% count(Label)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 51 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>annot</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Malpighian tubule                   </td><td> 1090</td></tr>\n",
       "\t<tr><td>amnioserosa                         </td><td> 7610</td></tr>\n",
       "\t<tr><td>anlage in statu nascendi            </td><td>   40</td></tr>\n",
       "\t<tr><td>brain                               </td><td>26617</td></tr>\n",
       "\t<tr><td>brain primordium                    </td><td> 9816</td></tr>\n",
       "\t<tr><td>cellular blastoderm / pole cell     </td><td>   16</td></tr>\n",
       "\t<tr><td>crystal cell                        </td><td>  177</td></tr>\n",
       "\t<tr><td>ectoderm anlage                     </td><td>    5</td></tr>\n",
       "\t<tr><td>ectoderm anlage in statu nascendi   </td><td>   51</td></tr>\n",
       "\t<tr><td>ectoderm primordium                 </td><td> 1601</td></tr>\n",
       "\t<tr><td>endoderm primordium                 </td><td>  400</td></tr>\n",
       "\t<tr><td>epidermis                           </td><td>36935</td></tr>\n",
       "\t<tr><td>epidermis primordium                </td><td> 6976</td></tr>\n",
       "\t<tr><td>fat body                            </td><td> 5550</td></tr>\n",
       "\t<tr><td>foregut / hindgut / anal pad        </td><td> 7035</td></tr>\n",
       "\t<tr><td>foregut / hindgut primordium        </td><td> 1218</td></tr>\n",
       "\t<tr><td>germ cell                           </td><td> 1309</td></tr>\n",
       "\t<tr><td>glia                                </td><td>  899</td></tr>\n",
       "\t<tr><td>hindgut / anal pad                  </td><td> 2662</td></tr>\n",
       "\t<tr><td>hindgut primordium                  </td><td> 1916</td></tr>\n",
       "\t<tr><td>maternal                            </td><td>   17</td></tr>\n",
       "\t<tr><td>mesoderm primordium                 </td><td>  384</td></tr>\n",
       "\t<tr><td>midgut                              </td><td>11649</td></tr>\n",
       "\t<tr><td>midgut primordium                   </td><td> 1848</td></tr>\n",
       "\t<tr><td>midline primordium                  </td><td>  982</td></tr>\n",
       "\t<tr><td>muscle                              </td><td> 6184</td></tr>\n",
       "\t<tr><td>peripheral nervous system           </td><td> 3591</td></tr>\n",
       "\t<tr><td>peripheral nervous system primordium</td><td>  559</td></tr>\n",
       "\t<tr><td>plasmatocytes                       </td><td> 5575</td></tr>\n",
       "\t<tr><td>plasmatocytes anlage                </td><td> 1658</td></tr>\n",
       "\t<tr><td>procrystal cell                     </td><td> 1556</td></tr>\n",
       "\t<tr><td>proventriculus                      </td><td> 2480</td></tr>\n",
       "\t<tr><td>salivary gland                      </td><td> 3487</td></tr>\n",
       "\t<tr><td>salivary gland body primordium      </td><td> 1042</td></tr>\n",
       "\t<tr><td>salivary gland primordium           </td><td>  115</td></tr>\n",
       "\t<tr><td>sensory nervous system              </td><td> 4816</td></tr>\n",
       "\t<tr><td>sensory nervous system primordium   </td><td> 1627</td></tr>\n",
       "\t<tr><td>somatic muscle                      </td><td>14715</td></tr>\n",
       "\t<tr><td>somatic muscle primordium           </td><td> 1674</td></tr>\n",
       "\t<tr><td>spiracle (tracheal system)          </td><td>  986</td></tr>\n",
       "\t<tr><td>spiracles (tracheal system)         </td><td>  751</td></tr>\n",
       "\t<tr><td>tracheal primordium                 </td><td> 2526</td></tr>\n",
       "\t<tr><td>tracheal system                     </td><td> 8052</td></tr>\n",
       "\t<tr><td>trunk mesoderm primordium           </td><td>  880</td></tr>\n",
       "\t<tr><td>ubiquitous                          </td><td>34589</td></tr>\n",
       "\t<tr><td>unknown                             </td><td> 1444</td></tr>\n",
       "\t<tr><td>ventral nerve cord                  </td><td> 7542</td></tr>\n",
       "\t<tr><td>ventral nerve cord primordium       </td><td>  671</td></tr>\n",
       "\t<tr><td>visceral muscle                     </td><td>  799</td></tr>\n",
       "\t<tr><td>visceral muscle primordium          </td><td> 3546</td></tr>\n",
       "\t<tr><td>yolk nuclei                         </td><td>  777</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 51 × 2\n",
       "\\begin{tabular}{ll}\n",
       " annot & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Malpighian tubule                    &  1090\\\\\n",
       "\t amnioserosa                          &  7610\\\\\n",
       "\t anlage in statu nascendi             &    40\\\\\n",
       "\t brain                                & 26617\\\\\n",
       "\t brain primordium                     &  9816\\\\\n",
       "\t cellular blastoderm / pole cell      &    16\\\\\n",
       "\t crystal cell                         &   177\\\\\n",
       "\t ectoderm anlage                      &     5\\\\\n",
       "\t ectoderm anlage in statu nascendi    &    51\\\\\n",
       "\t ectoderm primordium                  &  1601\\\\\n",
       "\t endoderm primordium                  &   400\\\\\n",
       "\t epidermis                            & 36935\\\\\n",
       "\t epidermis primordium                 &  6976\\\\\n",
       "\t fat body                             &  5550\\\\\n",
       "\t foregut / hindgut / anal pad         &  7035\\\\\n",
       "\t foregut / hindgut primordium         &  1218\\\\\n",
       "\t germ cell                            &  1309\\\\\n",
       "\t glia                                 &   899\\\\\n",
       "\t hindgut / anal pad                   &  2662\\\\\n",
       "\t hindgut primordium                   &  1916\\\\\n",
       "\t maternal                             &    17\\\\\n",
       "\t mesoderm primordium                  &   384\\\\\n",
       "\t midgut                               & 11649\\\\\n",
       "\t midgut primordium                    &  1848\\\\\n",
       "\t midline primordium                   &   982\\\\\n",
       "\t muscle                               &  6184\\\\\n",
       "\t peripheral nervous system            &  3591\\\\\n",
       "\t peripheral nervous system primordium &   559\\\\\n",
       "\t plasmatocytes                        &  5575\\\\\n",
       "\t plasmatocytes anlage                 &  1658\\\\\n",
       "\t procrystal cell                      &  1556\\\\\n",
       "\t proventriculus                       &  2480\\\\\n",
       "\t salivary gland                       &  3487\\\\\n",
       "\t salivary gland body primordium       &  1042\\\\\n",
       "\t salivary gland primordium            &   115\\\\\n",
       "\t sensory nervous system               &  4816\\\\\n",
       "\t sensory nervous system primordium    &  1627\\\\\n",
       "\t somatic muscle                       & 14715\\\\\n",
       "\t somatic muscle primordium            &  1674\\\\\n",
       "\t spiracle (tracheal system)           &   986\\\\\n",
       "\t spiracles (tracheal system)          &   751\\\\\n",
       "\t tracheal primordium                  &  2526\\\\\n",
       "\t tracheal system                      &  8052\\\\\n",
       "\t trunk mesoderm primordium            &   880\\\\\n",
       "\t ubiquitous                           & 34589\\\\\n",
       "\t unknown                              &  1444\\\\\n",
       "\t ventral nerve cord                   &  7542\\\\\n",
       "\t ventral nerve cord primordium        &   671\\\\\n",
       "\t visceral muscle                      &   799\\\\\n",
       "\t visceral muscle primordium           &  3546\\\\\n",
       "\t yolk nuclei                          &   777\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 51 × 2\n",
       "\n",
       "| annot &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Malpighian tubule                    |  1090 |\n",
       "| amnioserosa                          |  7610 |\n",
       "| anlage in statu nascendi             |    40 |\n",
       "| brain                                | 26617 |\n",
       "| brain primordium                     |  9816 |\n",
       "| cellular blastoderm / pole cell      |    16 |\n",
       "| crystal cell                         |   177 |\n",
       "| ectoderm anlage                      |     5 |\n",
       "| ectoderm anlage in statu nascendi    |    51 |\n",
       "| ectoderm primordium                  |  1601 |\n",
       "| endoderm primordium                  |   400 |\n",
       "| epidermis                            | 36935 |\n",
       "| epidermis primordium                 |  6976 |\n",
       "| fat body                             |  5550 |\n",
       "| foregut / hindgut / anal pad         |  7035 |\n",
       "| foregut / hindgut primordium         |  1218 |\n",
       "| germ cell                            |  1309 |\n",
       "| glia                                 |   899 |\n",
       "| hindgut / anal pad                   |  2662 |\n",
       "| hindgut primordium                   |  1916 |\n",
       "| maternal                             |    17 |\n",
       "| mesoderm primordium                  |   384 |\n",
       "| midgut                               | 11649 |\n",
       "| midgut primordium                    |  1848 |\n",
       "| midline primordium                   |   982 |\n",
       "| muscle                               |  6184 |\n",
       "| peripheral nervous system            |  3591 |\n",
       "| peripheral nervous system primordium |   559 |\n",
       "| plasmatocytes                        |  5575 |\n",
       "| plasmatocytes anlage                 |  1658 |\n",
       "| procrystal cell                      |  1556 |\n",
       "| proventriculus                       |  2480 |\n",
       "| salivary gland                       |  3487 |\n",
       "| salivary gland body primordium       |  1042 |\n",
       "| salivary gland primordium            |   115 |\n",
       "| sensory nervous system               |  4816 |\n",
       "| sensory nervous system primordium    |  1627 |\n",
       "| somatic muscle                       | 14715 |\n",
       "| somatic muscle primordium            |  1674 |\n",
       "| spiracle (tracheal system)           |   986 |\n",
       "| spiracles (tracheal system)          |   751 |\n",
       "| tracheal primordium                  |  2526 |\n",
       "| tracheal system                      |  8052 |\n",
       "| trunk mesoderm primordium            |   880 |\n",
       "| ubiquitous                           | 34589 |\n",
       "| unknown                              |  1444 |\n",
       "| ventral nerve cord                   |  7542 |\n",
       "| ventral nerve cord primordium        |   671 |\n",
       "| visceral muscle                      |   799 |\n",
       "| visceral muscle primordium           |  3546 |\n",
       "| yolk nuclei                          |   777 |\n",
       "\n"
      ],
      "text/plain": [
       "   annot                                n    \n",
       "1  Malpighian tubule                     1090\n",
       "2  amnioserosa                           7610\n",
       "3  anlage in statu nascendi                40\n",
       "4  brain                                26617\n",
       "5  brain primordium                      9816\n",
       "6  cellular blastoderm / pole cell         16\n",
       "7  crystal cell                           177\n",
       "8  ectoderm anlage                          5\n",
       "9  ectoderm anlage in statu nascendi       51\n",
       "10 ectoderm primordium                   1601\n",
       "11 endoderm primordium                    400\n",
       "12 epidermis                            36935\n",
       "13 epidermis primordium                  6976\n",
       "14 fat body                              5550\n",
       "15 foregut / hindgut / anal pad          7035\n",
       "16 foregut / hindgut primordium          1218\n",
       "17 germ cell                             1309\n",
       "18 glia                                   899\n",
       "19 hindgut / anal pad                    2662\n",
       "20 hindgut primordium                    1916\n",
       "21 maternal                                17\n",
       "22 mesoderm primordium                    384\n",
       "23 midgut                               11649\n",
       "24 midgut primordium                     1848\n",
       "25 midline primordium                     982\n",
       "26 muscle                                6184\n",
       "27 peripheral nervous system             3591\n",
       "28 peripheral nervous system primordium   559\n",
       "29 plasmatocytes                         5575\n",
       "30 plasmatocytes anlage                  1658\n",
       "31 procrystal cell                       1556\n",
       "32 proventriculus                        2480\n",
       "33 salivary gland                        3487\n",
       "34 salivary gland body primordium        1042\n",
       "35 salivary gland primordium              115\n",
       "36 sensory nervous system                4816\n",
       "37 sensory nervous system primordium     1627\n",
       "38 somatic muscle                       14715\n",
       "39 somatic muscle primordium             1674\n",
       "40 spiracle (tracheal system)             986\n",
       "41 spiracles (tracheal system)            751\n",
       "42 tracheal primordium                   2526\n",
       "43 tracheal system                       8052\n",
       "44 trunk mesoderm primordium              880\n",
       "45 ubiquitous                           34589\n",
       "46 unknown                               1444\n",
       "47 ventral nerve cord                    7542\n",
       "48 ventral nerve cord primordium          671\n",
       "49 visceral muscle                        799\n",
       "50 visceral muscle primordium            3546\n",
       "51 yolk nuclei                            777"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub300  %>% count(annot)  %>% arrange(annot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1359,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 50 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Label</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Malpighian tubule                   </td><td>  90</td></tr>\n",
       "\t<tr><td>amnioserosa                         </td><td> 279</td></tr>\n",
       "\t<tr><td>anlage in statu nascendi            </td><td>  11</td></tr>\n",
       "\t<tr><td>brain                               </td><td>1363</td></tr>\n",
       "\t<tr><td>brain primordium                    </td><td>1127</td></tr>\n",
       "\t<tr><td>cellular blastoderm / pole cell     </td><td>   2</td></tr>\n",
       "\t<tr><td>crystal cell                        </td><td>  23</td></tr>\n",
       "\t<tr><td>ectoderm anlage in statu nascendi   </td><td>  14</td></tr>\n",
       "\t<tr><td>ectoderm primordium                 </td><td> 164</td></tr>\n",
       "\t<tr><td>endoderm primordium                 </td><td>  70</td></tr>\n",
       "\t<tr><td>epidermis                           </td><td>2618</td></tr>\n",
       "\t<tr><td>epidermis primordium                </td><td> 862</td></tr>\n",
       "\t<tr><td>fat body                            </td><td> 288</td></tr>\n",
       "\t<tr><td>foregut / hindgut / anal pad        </td><td>1296</td></tr>\n",
       "\t<tr><td>foregut / hindgut primordium        </td><td> 141</td></tr>\n",
       "\t<tr><td>germ cell                           </td><td> 127</td></tr>\n",
       "\t<tr><td>glia                                </td><td>  75</td></tr>\n",
       "\t<tr><td>hindgut / anal pad                  </td><td> 210</td></tr>\n",
       "\t<tr><td>hindgut primordium                  </td><td> 321</td></tr>\n",
       "\t<tr><td>maternal                            </td><td>   3</td></tr>\n",
       "\t<tr><td>mesoderm primordium                 </td><td>  75</td></tr>\n",
       "\t<tr><td>midgut                              </td><td> 892</td></tr>\n",
       "\t<tr><td>midgut primordium                   </td><td> 286</td></tr>\n",
       "\t<tr><td>midline primordium                  </td><td>  68</td></tr>\n",
       "\t<tr><td>muscle                              </td><td> 960</td></tr>\n",
       "\t<tr><td>peripheral nervous system           </td><td> 389</td></tr>\n",
       "\t<tr><td>peripheral nervous system primordium</td><td>  84</td></tr>\n",
       "\t<tr><td>plasmatocytes                       </td><td> 369</td></tr>\n",
       "\t<tr><td>plasmatocytes anlage                </td><td> 158</td></tr>\n",
       "\t<tr><td>procrystal cell                     </td><td> 143</td></tr>\n",
       "\t<tr><td>proventriculus                      </td><td> 211</td></tr>\n",
       "\t<tr><td>salivary gland                      </td><td> 114</td></tr>\n",
       "\t<tr><td>salivary gland body primordium      </td><td>  69</td></tr>\n",
       "\t<tr><td>salivary gland primordium           </td><td>  12</td></tr>\n",
       "\t<tr><td>sensory nervous system              </td><td> 643</td></tr>\n",
       "\t<tr><td>sensory nervous system primordium   </td><td> 272</td></tr>\n",
       "\t<tr><td>somatic muscle                      </td><td> 754</td></tr>\n",
       "\t<tr><td>somatic muscle primordium           </td><td> 290</td></tr>\n",
       "\t<tr><td>spiracle (tracheal system)          </td><td> 202</td></tr>\n",
       "\t<tr><td>spiracles (tracheal system)         </td><td>  36</td></tr>\n",
       "\t<tr><td>tracheal primordium                 </td><td> 359</td></tr>\n",
       "\t<tr><td>tracheal system                     </td><td> 627</td></tr>\n",
       "\t<tr><td>trunk mesoderm primordium           </td><td> 139</td></tr>\n",
       "\t<tr><td>ubiquitous                          </td><td>2188</td></tr>\n",
       "\t<tr><td>unknown                             </td><td> 322</td></tr>\n",
       "\t<tr><td>ventral nerve cord                  </td><td> 366</td></tr>\n",
       "\t<tr><td>ventral nerve cord primordium       </td><td> 135</td></tr>\n",
       "\t<tr><td>visceral muscle                     </td><td> 104</td></tr>\n",
       "\t<tr><td>visceral muscle primordium          </td><td> 589</td></tr>\n",
       "\t<tr><td>yolk nuclei                         </td><td> 140</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 50 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Label & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Malpighian tubule                    &   90\\\\\n",
       "\t amnioserosa                          &  279\\\\\n",
       "\t anlage in statu nascendi             &   11\\\\\n",
       "\t brain                                & 1363\\\\\n",
       "\t brain primordium                     & 1127\\\\\n",
       "\t cellular blastoderm / pole cell      &    2\\\\\n",
       "\t crystal cell                         &   23\\\\\n",
       "\t ectoderm anlage in statu nascendi    &   14\\\\\n",
       "\t ectoderm primordium                  &  164\\\\\n",
       "\t endoderm primordium                  &   70\\\\\n",
       "\t epidermis                            & 2618\\\\\n",
       "\t epidermis primordium                 &  862\\\\\n",
       "\t fat body                             &  288\\\\\n",
       "\t foregut / hindgut / anal pad         & 1296\\\\\n",
       "\t foregut / hindgut primordium         &  141\\\\\n",
       "\t germ cell                            &  127\\\\\n",
       "\t glia                                 &   75\\\\\n",
       "\t hindgut / anal pad                   &  210\\\\\n",
       "\t hindgut primordium                   &  321\\\\\n",
       "\t maternal                             &    3\\\\\n",
       "\t mesoderm primordium                  &   75\\\\\n",
       "\t midgut                               &  892\\\\\n",
       "\t midgut primordium                    &  286\\\\\n",
       "\t midline primordium                   &   68\\\\\n",
       "\t muscle                               &  960\\\\\n",
       "\t peripheral nervous system            &  389\\\\\n",
       "\t peripheral nervous system primordium &   84\\\\\n",
       "\t plasmatocytes                        &  369\\\\\n",
       "\t plasmatocytes anlage                 &  158\\\\\n",
       "\t procrystal cell                      &  143\\\\\n",
       "\t proventriculus                       &  211\\\\\n",
       "\t salivary gland                       &  114\\\\\n",
       "\t salivary gland body primordium       &   69\\\\\n",
       "\t salivary gland primordium            &   12\\\\\n",
       "\t sensory nervous system               &  643\\\\\n",
       "\t sensory nervous system primordium    &  272\\\\\n",
       "\t somatic muscle                       &  754\\\\\n",
       "\t somatic muscle primordium            &  290\\\\\n",
       "\t spiracle (tracheal system)           &  202\\\\\n",
       "\t spiracles (tracheal system)          &   36\\\\\n",
       "\t tracheal primordium                  &  359\\\\\n",
       "\t tracheal system                      &  627\\\\\n",
       "\t trunk mesoderm primordium            &  139\\\\\n",
       "\t ubiquitous                           & 2188\\\\\n",
       "\t unknown                              &  322\\\\\n",
       "\t ventral nerve cord                   &  366\\\\\n",
       "\t ventral nerve cord primordium        &  135\\\\\n",
       "\t visceral muscle                      &  104\\\\\n",
       "\t visceral muscle primordium           &  589\\\\\n",
       "\t yolk nuclei                          &  140\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 50 × 2\n",
       "\n",
       "| Label &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Malpighian tubule                    |   90 |\n",
       "| amnioserosa                          |  279 |\n",
       "| anlage in statu nascendi             |   11 |\n",
       "| brain                                | 1363 |\n",
       "| brain primordium                     | 1127 |\n",
       "| cellular blastoderm / pole cell      |    2 |\n",
       "| crystal cell                         |   23 |\n",
       "| ectoderm anlage in statu nascendi    |   14 |\n",
       "| ectoderm primordium                  |  164 |\n",
       "| endoderm primordium                  |   70 |\n",
       "| epidermis                            | 2618 |\n",
       "| epidermis primordium                 |  862 |\n",
       "| fat body                             |  288 |\n",
       "| foregut / hindgut / anal pad         | 1296 |\n",
       "| foregut / hindgut primordium         |  141 |\n",
       "| germ cell                            |  127 |\n",
       "| glia                                 |   75 |\n",
       "| hindgut / anal pad                   |  210 |\n",
       "| hindgut primordium                   |  321 |\n",
       "| maternal                             |    3 |\n",
       "| mesoderm primordium                  |   75 |\n",
       "| midgut                               |  892 |\n",
       "| midgut primordium                    |  286 |\n",
       "| midline primordium                   |   68 |\n",
       "| muscle                               |  960 |\n",
       "| peripheral nervous system            |  389 |\n",
       "| peripheral nervous system primordium |   84 |\n",
       "| plasmatocytes                        |  369 |\n",
       "| plasmatocytes anlage                 |  158 |\n",
       "| procrystal cell                      |  143 |\n",
       "| proventriculus                       |  211 |\n",
       "| salivary gland                       |  114 |\n",
       "| salivary gland body primordium       |   69 |\n",
       "| salivary gland primordium            |   12 |\n",
       "| sensory nervous system               |  643 |\n",
       "| sensory nervous system primordium    |  272 |\n",
       "| somatic muscle                       |  754 |\n",
       "| somatic muscle primordium            |  290 |\n",
       "| spiracle (tracheal system)           |  202 |\n",
       "| spiracles (tracheal system)          |   36 |\n",
       "| tracheal primordium                  |  359 |\n",
       "| tracheal system                      |  627 |\n",
       "| trunk mesoderm primordium            |  139 |\n",
       "| ubiquitous                           | 2188 |\n",
       "| unknown                              |  322 |\n",
       "| ventral nerve cord                   |  366 |\n",
       "| ventral nerve cord primordium        |  135 |\n",
       "| visceral muscle                      |  104 |\n",
       "| visceral muscle primordium           |  589 |\n",
       "| yolk nuclei                          |  140 |\n",
       "\n"
      ],
      "text/plain": [
       "   Label                                n   \n",
       "1  Malpighian tubule                      90\n",
       "2  amnioserosa                           279\n",
       "3  anlage in statu nascendi               11\n",
       "4  brain                                1363\n",
       "5  brain primordium                     1127\n",
       "6  cellular blastoderm / pole cell         2\n",
       "7  crystal cell                           23\n",
       "8  ectoderm anlage in statu nascendi      14\n",
       "9  ectoderm primordium                   164\n",
       "10 endoderm primordium                    70\n",
       "11 epidermis                            2618\n",
       "12 epidermis primordium                  862\n",
       "13 fat body                              288\n",
       "14 foregut / hindgut / anal pad         1296\n",
       "15 foregut / hindgut primordium          141\n",
       "16 germ cell                             127\n",
       "17 glia                                   75\n",
       "18 hindgut / anal pad                    210\n",
       "19 hindgut primordium                    321\n",
       "20 maternal                                3\n",
       "21 mesoderm primordium                    75\n",
       "22 midgut                                892\n",
       "23 midgut primordium                     286\n",
       "24 midline primordium                     68\n",
       "25 muscle                                960\n",
       "26 peripheral nervous system             389\n",
       "27 peripheral nervous system primordium   84\n",
       "28 plasmatocytes                         369\n",
       "29 plasmatocytes anlage                  158\n",
       "30 procrystal cell                       143\n",
       "31 proventriculus                        211\n",
       "32 salivary gland                        114\n",
       "33 salivary gland body primordium         69\n",
       "34 salivary gland primordium              12\n",
       "35 sensory nervous system                643\n",
       "36 sensory nervous system primordium     272\n",
       "37 somatic muscle                        754\n",
       "38 somatic muscle primordium             290\n",
       "39 spiracle (tracheal system)            202\n",
       "40 spiracles (tracheal system)            36\n",
       "41 tracheal primordium                   359\n",
       "42 tracheal system                       627\n",
       "43 trunk mesoderm primordium             139\n",
       "44 ubiquitous                           2188\n",
       "45 unknown                               322\n",
       "46 ventral nerve cord                    366\n",
       "47 ventral nerve cord primordium         135\n",
       "48 visceral muscle                       104\n",
       "49 visceral muscle primordium            589\n",
       "50 yolk nuclei                           140"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "res_label  %>% filter(nk == 1)  %>% ungroup  %>% count(Label)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "singletons = (res %>% filter(nk == 0) )$ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "doubletons = (res %>% filter(nk == 1) )$ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# create the dataframe from the cds from 300 PCAs\n",
    "df_sub300 = make_df(cds_sub)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "cell_type = \"glia\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 459,
   "metadata": {},
   "outputs": [],
   "source": [
    "options(repr.plot.width=14, repr.plot.height=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "# cell.type = \"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% singletons), size = ifelse(df_sub300$cell %in% singletons, 0.2, 0.05), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"singletons\") + \n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "#cell.type = \"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% singletons & df_sub300$annot == \"Malpighian tubule\"), \n",
    "           size = ifelse( df_sub300$cell %in% singletons & df_sub300$annot == \"Malpighian tubule\", 2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Malpighian tubule singletons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "#cell.type = \"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% singletons & df_sub300$annot == \"Malpighian tubule\"), \n",
    "           size = ifelse( df_sub300$cell %in% singletons & df_sub300$annot == \"Malpighian tubule\", 2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Malpighian tubule singletons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "#cell.type = \"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = !(df_sub300$cell %in% singletons) & df_sub300$annot == \"Malpighian tubule\"), \n",
    "           size = ifelse( !(df_sub300$cell %in% singletons) & df_sub300$annot == \"Malpighian tubule\", 2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Malpighian tubule not singletons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "#cell.type = \"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% doubletons & df_sub300$annot == \"Malpighian tubule\"), \n",
    "           size = ifelse( df_sub300$cell %in% doubletons & df_sub300$annot == \"Malpighian tubule\", 2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Malpighian tubule doubletons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "#cell.type = \"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = !(df_sub300$cell %in% doubletons) & df_sub300$annot == \"Malpighian tubule\" ), \n",
    "           size = ifelse( !(df_sub300$cell %in% doubletons) & df_sub300$annot == \"Malpighian tubule\", 2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Malpighian tubule not doubletons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "#cell.type = \"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% doubletons), size = ifelse(df_sub300$cell %in% doubletons, 0.1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"doubletons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "nineteentons = (res %>% filter(nk == 19) )$ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "#cell.type = \"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% nineteentons), size = ifelse(df_sub300$cell %in% nineteentons, 0.1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"nineteentons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Subset on singletons and also on doubletons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'cds_sub'</li><li>'cell_type'</li><li>'cell.type'</li><li>'cellIDs'</li><li>'cells'</li><li>'cellTypes'</li><li>'df'</li><li>'df_sub300'</li><li>'doubletons'</li><li>'expresses.gene'</li><li>'get.gene.id'</li><li>'get.gene.id2'</li><li>'get.norm.expr.matrix'</li><li>'get.norm.expr.matrix.3'</li><li>'ids'</li><li>'k'</li><li>'k1'</li><li>'labels'</li><li>'make_df'</li><li>'nineteentons'</li><li>'nnMatches'</li><li>'pca'</li><li>'peedeef'</li><li>'plot.expr'</li><li>'plot.expr_clus'</li><li>'plot.expr_clus.br'</li><li>'plot.expr_clus.br2'</li><li>'plot.expr3'</li><li>'rann'</li><li>'res'</li><li>'res_label'</li><li>'singletons'</li><li>'two.set.differential.gene.test'</li><li>'two.set.differential.gene.test.3'</li><li>'two.set.differential.gene.test.3a'</li><li>'two.set.differential.gene.test.3b'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'cds\\_sub'\n",
       "\\item 'cell\\_type'\n",
       "\\item 'cell.type'\n",
       "\\item 'cellIDs'\n",
       "\\item 'cells'\n",
       "\\item 'cellTypes'\n",
       "\\item 'df'\n",
       "\\item 'df\\_sub300'\n",
       "\\item 'doubletons'\n",
       "\\item 'expresses.gene'\n",
       "\\item 'get.gene.id'\n",
       "\\item 'get.gene.id2'\n",
       "\\item 'get.norm.expr.matrix'\n",
       "\\item 'get.norm.expr.matrix.3'\n",
       "\\item 'ids'\n",
       "\\item 'k'\n",
       "\\item 'k1'\n",
       "\\item 'labels'\n",
       "\\item 'make\\_df'\n",
       "\\item 'nineteentons'\n",
       "\\item 'nnMatches'\n",
       "\\item 'pca'\n",
       "\\item 'peedeef'\n",
       "\\item 'plot.expr'\n",
       "\\item 'plot.expr\\_clus'\n",
       "\\item 'plot.expr\\_clus.br'\n",
       "\\item 'plot.expr\\_clus.br2'\n",
       "\\item 'plot.expr3'\n",
       "\\item 'rann'\n",
       "\\item 'res'\n",
       "\\item 'res\\_label'\n",
       "\\item 'singletons'\n",
       "\\item 'two.set.differential.gene.test'\n",
       "\\item 'two.set.differential.gene.test.3'\n",
       "\\item 'two.set.differential.gene.test.3a'\n",
       "\\item 'two.set.differential.gene.test.3b'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'cds_sub'\n",
       "2. 'cell_type'\n",
       "3. 'cell.type'\n",
       "4. 'cellIDs'\n",
       "5. 'cells'\n",
       "6. 'cellTypes'\n",
       "7. 'df'\n",
       "8. 'df_sub300'\n",
       "9. 'doubletons'\n",
       "10. 'expresses.gene'\n",
       "11. 'get.gene.id'\n",
       "12. 'get.gene.id2'\n",
       "13. 'get.norm.expr.matrix'\n",
       "14. 'get.norm.expr.matrix.3'\n",
       "15. 'ids'\n",
       "16. 'k'\n",
       "17. 'k1'\n",
       "18. 'labels'\n",
       "19. 'make_df'\n",
       "20. 'nineteentons'\n",
       "21. 'nnMatches'\n",
       "22. 'pca'\n",
       "23. 'peedeef'\n",
       "24. 'plot.expr'\n",
       "25. 'plot.expr_clus'\n",
       "26. 'plot.expr_clus.br'\n",
       "27. 'plot.expr_clus.br2'\n",
       "28. 'plot.expr3'\n",
       "29. 'rann'\n",
       "30. 'res'\n",
       "31. 'res_label'\n",
       "32. 'singletons'\n",
       "33. 'two.set.differential.gene.test'\n",
       "34. 'two.set.differential.gene.test.3'\n",
       "35. 'two.set.differential.gene.test.3a'\n",
       "36. 'two.set.differential.gene.test.3b'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"cds_sub\"                           \"cell_type\"                        \n",
       " [3] \"cell.type\"                         \"cellIDs\"                          \n",
       " [5] \"cells\"                             \"cellTypes\"                        \n",
       " [7] \"df\"                                \"df_sub300\"                        \n",
       " [9] \"doubletons\"                        \"expresses.gene\"                   \n",
       "[11] \"get.gene.id\"                       \"get.gene.id2\"                     \n",
       "[13] \"get.norm.expr.matrix\"              \"get.norm.expr.matrix.3\"           \n",
       "[15] \"ids\"                               \"k\"                                \n",
       "[17] \"k1\"                                \"labels\"                           \n",
       "[19] \"make_df\"                           \"nineteentons\"                     \n",
       "[21] \"nnMatches\"                         \"pca\"                              \n",
       "[23] \"peedeef\"                           \"plot.expr\"                        \n",
       "[25] \"plot.expr_clus\"                    \"plot.expr_clus.br\"                \n",
       "[27] \"plot.expr_clus.br2\"                \"plot.expr3\"                       \n",
       "[29] \"rann\"                              \"res\"                              \n",
       "[31] \"res_label\"                         \"singletons\"                       \n",
       "[33] \"two.set.differential.gene.test\"    \"two.set.differential.gene.test.3\" \n",
       "[35] \"two.set.differential.gene.test.3a\" \"two.set.differential.gene.test.3b\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ls()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "cds_sub_no_sing = cds_sub[, !(df_sub300$cell %in% singletons)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>211498</li><li>32</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 211498\n",
       "\\item 32\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 211498\n",
       "2. 32\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 211498     32"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub300  %>% filter(!(cell %in% singletons))  %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>13938</li><li>211498</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 13938\n",
       "\\item 211498\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 13938\n",
       "2. 211498\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1]  13938 211498"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(cds_sub_no_sing)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sub_no_singletons = make_df(cds_sub_no_sing)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "## read in the previously calculated marker files\n",
    "load(\"data/markers for fly datasets.RData\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'cds_sub'</li><li>'cds_sub_no_sing'</li><li>'cell_type'</li><li>'cell.type'</li><li>'cell.type.2.markers'</li><li>'cell.type.markers.sub300_no_singletons.cell.type'</li><li>'cell.type.markers.sub300.cell.type'</li><li>'cellIDs'</li><li>'cells'</li><li>'cellTypes'</li><li>'cluster.markers'</li><li>'cluster.markers.sub300'</li><li>'cluster.markers.sub300_no_doubletons'</li><li>'cluster.markers.sub300_no_doubletons_all'</li><li>'cluster.markers.sub300_no_singletons'</li><li>'df'</li><li>'df_sub_no_singletons'</li><li>'df_sub300'</li><li>'doubletons'</li><li>'expresses.gene'</li><li>'get.gene.id'</li><li>'get.gene.id2'</li><li>'get.norm.expr.matrix'</li><li>'get.norm.expr.matrix.3'</li><li>'ids'</li><li>'k'</li><li>'k1'</li><li>'labels'</li><li>'make_df'</li><li>'nineteentons'</li><li>'nnMatches'</li><li>'pca'</li><li>'peedeef'</li><li>'plot.expr'</li><li>'plot.expr_clus'</li><li>'plot.expr_clus.br'</li><li>'plot.expr_clus.br2'</li><li>'plot.expr3'</li><li>'rann'</li><li>'res'</li><li>'res_label'</li><li>'singletons'</li><li>'two.set.differential.gene.test'</li><li>'two.set.differential.gene.test.3'</li><li>'two.set.differential.gene.test.3a'</li><li>'two.set.differential.gene.test.3b'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'cds\\_sub'\n",
       "\\item 'cds\\_sub\\_no\\_sing'\n",
       "\\item 'cell\\_type'\n",
       "\\item 'cell.type'\n",
       "\\item 'cell.type.2.markers'\n",
       "\\item 'cell.type.markers.sub300\\_no\\_singletons.cell.type'\n",
       "\\item 'cell.type.markers.sub300.cell.type'\n",
       "\\item 'cellIDs'\n",
       "\\item 'cells'\n",
       "\\item 'cellTypes'\n",
       "\\item 'cluster.markers'\n",
       "\\item 'cluster.markers.sub300'\n",
       "\\item 'cluster.markers.sub300\\_no\\_doubletons'\n",
       "\\item 'cluster.markers.sub300\\_no\\_doubletons\\_all'\n",
       "\\item 'cluster.markers.sub300\\_no\\_singletons'\n",
       "\\item 'df'\n",
       "\\item 'df\\_sub\\_no\\_singletons'\n",
       "\\item 'df\\_sub300'\n",
       "\\item 'doubletons'\n",
       "\\item 'expresses.gene'\n",
       "\\item 'get.gene.id'\n",
       "\\item 'get.gene.id2'\n",
       "\\item 'get.norm.expr.matrix'\n",
       "\\item 'get.norm.expr.matrix.3'\n",
       "\\item 'ids'\n",
       "\\item 'k'\n",
       "\\item 'k1'\n",
       "\\item 'labels'\n",
       "\\item 'make\\_df'\n",
       "\\item 'nineteentons'\n",
       "\\item 'nnMatches'\n",
       "\\item 'pca'\n",
       "\\item 'peedeef'\n",
       "\\item 'plot.expr'\n",
       "\\item 'plot.expr\\_clus'\n",
       "\\item 'plot.expr\\_clus.br'\n",
       "\\item 'plot.expr\\_clus.br2'\n",
       "\\item 'plot.expr3'\n",
       "\\item 'rann'\n",
       "\\item 'res'\n",
       "\\item 'res\\_label'\n",
       "\\item 'singletons'\n",
       "\\item 'two.set.differential.gene.test'\n",
       "\\item 'two.set.differential.gene.test.3'\n",
       "\\item 'two.set.differential.gene.test.3a'\n",
       "\\item 'two.set.differential.gene.test.3b'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'cds_sub'\n",
       "2. 'cds_sub_no_sing'\n",
       "3. 'cell_type'\n",
       "4. 'cell.type'\n",
       "5. 'cell.type.2.markers'\n",
       "6. 'cell.type.markers.sub300_no_singletons.cell.type'\n",
       "7. 'cell.type.markers.sub300.cell.type'\n",
       "8. 'cellIDs'\n",
       "9. 'cells'\n",
       "10. 'cellTypes'\n",
       "11. 'cluster.markers'\n",
       "12. 'cluster.markers.sub300'\n",
       "13. 'cluster.markers.sub300_no_doubletons'\n",
       "14. 'cluster.markers.sub300_no_doubletons_all'\n",
       "15. 'cluster.markers.sub300_no_singletons'\n",
       "16. 'df'\n",
       "17. 'df_sub_no_singletons'\n",
       "18. 'df_sub300'\n",
       "19. 'doubletons'\n",
       "20. 'expresses.gene'\n",
       "21. 'get.gene.id'\n",
       "22. 'get.gene.id2'\n",
       "23. 'get.norm.expr.matrix'\n",
       "24. 'get.norm.expr.matrix.3'\n",
       "25. 'ids'\n",
       "26. 'k'\n",
       "27. 'k1'\n",
       "28. 'labels'\n",
       "29. 'make_df'\n",
       "30. 'nineteentons'\n",
       "31. 'nnMatches'\n",
       "32. 'pca'\n",
       "33. 'peedeef'\n",
       "34. 'plot.expr'\n",
       "35. 'plot.expr_clus'\n",
       "36. 'plot.expr_clus.br'\n",
       "37. 'plot.expr_clus.br2'\n",
       "38. 'plot.expr3'\n",
       "39. 'rann'\n",
       "40. 'res'\n",
       "41. 'res_label'\n",
       "42. 'singletons'\n",
       "43. 'two.set.differential.gene.test'\n",
       "44. 'two.set.differential.gene.test.3'\n",
       "45. 'two.set.differential.gene.test.3a'\n",
       "46. 'two.set.differential.gene.test.3b'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"cds_sub\"                                         \n",
       " [2] \"cds_sub_no_sing\"                                 \n",
       " [3] \"cell_type\"                                       \n",
       " [4] \"cell.type\"                                       \n",
       " [5] \"cell.type.2.markers\"                             \n",
       " [6] \"cell.type.markers.sub300_no_singletons.cell.type\"\n",
       " [7] \"cell.type.markers.sub300.cell.type\"              \n",
       " [8] \"cellIDs\"                                         \n",
       " [9] \"cells\"                                           \n",
       "[10] \"cellTypes\"                                       \n",
       "[11] \"cluster.markers\"                                 \n",
       "[12] \"cluster.markers.sub300\"                          \n",
       "[13] \"cluster.markers.sub300_no_doubletons\"            \n",
       "[14] \"cluster.markers.sub300_no_doubletons_all\"        \n",
       "[15] \"cluster.markers.sub300_no_singletons\"            \n",
       "[16] \"df\"                                              \n",
       "[17] \"df_sub_no_singletons\"                            \n",
       "[18] \"df_sub300\"                                       \n",
       "[19] \"doubletons\"                                      \n",
       "[20] \"expresses.gene\"                                  \n",
       "[21] \"get.gene.id\"                                     \n",
       "[22] \"get.gene.id2\"                                    \n",
       "[23] \"get.norm.expr.matrix\"                            \n",
       "[24] \"get.norm.expr.matrix.3\"                          \n",
       "[25] \"ids\"                                             \n",
       "[26] \"k\"                                               \n",
       "[27] \"k1\"                                              \n",
       "[28] \"labels\"                                          \n",
       "[29] \"make_df\"                                         \n",
       "[30] \"nineteentons\"                                    \n",
       "[31] \"nnMatches\"                                       \n",
       "[32] \"pca\"                                             \n",
       "[33] \"peedeef\"                                         \n",
       "[34] \"plot.expr\"                                       \n",
       "[35] \"plot.expr_clus\"                                  \n",
       "[36] \"plot.expr_clus.br\"                               \n",
       "[37] \"plot.expr_clus.br2\"                              \n",
       "[38] \"plot.expr3\"                                      \n",
       "[39] \"rann\"                                            \n",
       "[40] \"res\"                                             \n",
       "[41] \"res_label\"                                       \n",
       "[42] \"singletons\"                                      \n",
       "[43] \"two.set.differential.gene.test\"                  \n",
       "[44] \"two.set.differential.gene.test.3\"                \n",
       "[45] \"two.set.differential.gene.test.3a\"               \n",
       "[46] \"two.set.differential.gene.test.3b\"               "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ls()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### skip steps to find marker genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1255,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finding markers for cluster 1\n",
      "\n",
      "# of cells sampled in set 1: 2548\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 2\n",
      "\n",
      "# of cells sampled in set 1: 2423\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 3\n",
      "\n",
      "# of cells sampled in set 1: 2409\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 4\n",
      "\n",
      "# of cells sampled in set 1: 2403\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 5\n",
      "\n",
      "# of cells sampled in set 1: 2376\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 6\n",
      "\n",
      "# of cells sampled in set 1: 2357\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 7\n",
      "\n",
      "# of cells sampled in set 1: 2346\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 8\n",
      "\n",
      "# of cells sampled in set 1: 2344\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 9\n",
      "\n",
      "# of cells sampled in set 1: 2336\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 10\n",
      "\n",
      "# of cells sampled in set 1: 2316\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 11\n",
      "\n",
      "# of cells sampled in set 1: 2236\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 12\n",
      "\n",
      "# of cells sampled in set 1: 2212\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 13\n",
      "\n",
      "# of cells sampled in set 1: 2156\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 14\n",
      "\n",
      "# of cells sampled in set 1: 2146\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 15\n",
      "\n",
      "# of cells sampled in set 1: 2141\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 16\n",
      "\n",
      "# of cells sampled in set 1: 2140\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 17\n",
      "\n",
      "# of cells sampled in set 1: 2128\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 18\n",
      "\n",
      "# of cells sampled in set 1: 2123\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 19\n",
      "\n",
      "# of cells sampled in set 1: 2073\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 20\n",
      "\n",
      "# of cells sampled in set 1: 2047\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 21\n",
      "\n",
      "# of cells sampled in set 1: 2041\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 22\n",
      "\n",
      "# of cells sampled in set 1: 2004\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 23\n",
      "\n",
      "# of cells sampled in set 1: 2001\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 24\n",
      "\n",
      "# of cells sampled in set 1: 2000\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 25\n",
      "\n",
      "# of cells sampled in set 1: 1977\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 26\n",
      "\n",
      "# of cells sampled in set 1: 1960\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 27\n",
      "\n",
      "# of cells sampled in set 1: 1949\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 28\n",
      "\n",
      "# of cells sampled in set 1: 1882\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 29\n",
      "\n",
      "# of cells sampled in set 1: 1878\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 30\n",
      "\n",
      "# of cells sampled in set 1: 1864\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 31\n",
      "\n",
      "# of cells sampled in set 1: 1857\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 32\n",
      "\n",
      "# of cells sampled in set 1: 1854\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 33\n",
      "\n",
      "# of cells sampled in set 1: 1838\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 34\n",
      "\n",
      "# of cells sampled in set 1: 1812\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 35\n",
      "\n",
      "# of cells sampled in set 1: 1801\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 36\n",
      "\n",
      "# of cells sampled in set 1: 1789\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 37\n",
      "\n",
      "# of cells sampled in set 1: 1784\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 38\n",
      "\n",
      "# of cells sampled in set 1: 1768\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 39\n",
      "\n",
      "# of cells sampled in set 1: 1767\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 40\n",
      "\n",
      "# of cells sampled in set 1: 1766\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 41\n",
      "\n",
      "# of cells sampled in set 1: 1736\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 42\n",
      "\n",
      "# of cells sampled in set 1: 1722\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 43\n",
      "\n",
      "# of cells sampled in set 1: 1721\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 44\n",
      "\n",
      "# of cells sampled in set 1: 1700\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 45\n",
      "\n",
      "# of cells sampled in set 1: 1695\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 46\n",
      "\n",
      "# of cells sampled in set 1: 1693\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 47\n",
      "\n",
      "# of cells sampled in set 1: 1686\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 48\n",
      "\n",
      "# of cells sampled in set 1: 1676\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 49\n",
      "\n",
      "# of cells sampled in set 1: 1674\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 50\n",
      "\n",
      "# of cells sampled in set 1: 1669\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 51\n",
      "\n",
      "# of cells sampled in set 1: 1663\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 52\n",
      "\n",
      "# of cells sampled in set 1: 1657\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 53\n",
      "\n",
      "# of cells sampled in set 1: 1655\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 54\n",
      "\n",
      "# of cells sampled in set 1: 1655\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 55\n",
      "\n",
      "# of cells sampled in set 1: 1645\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 56\n",
      "\n",
      "# of cells sampled in set 1: 1637\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 57\n",
      "\n",
      "# of cells sampled in set 1: 1624\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 58\n",
      "\n",
      "# of cells sampled in set 1: 1621\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 59\n",
      "\n",
      "# of cells sampled in set 1: 1614\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 60\n",
      "\n",
      "# of cells sampled in set 1: 1613\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 61\n",
      "\n",
      "# of cells sampled in set 1: 1606\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 62\n",
      "\n",
      "# of cells sampled in set 1: 1593\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 63\n",
      "\n",
      "# of cells sampled in set 1: 1570\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 64\n",
      "\n",
      "# of cells sampled in set 1: 1565\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 65\n",
      "\n",
      "# of cells sampled in set 1: 1565\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 66\n",
      "\n",
      "# of cells sampled in set 1: 1564\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 67\n",
      "\n",
      "# of cells sampled in set 1: 1559\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 68\n",
      "\n",
      "# of cells sampled in set 1: 1557\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 69\n",
      "\n",
      "# of cells sampled in set 1: 1557\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 70\n",
      "\n",
      "# of cells sampled in set 1: 1555\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 71\n",
      "\n",
      "# of cells sampled in set 1: 1553\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 72\n",
      "\n",
      "# of cells sampled in set 1: 1544\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 73\n",
      "\n",
      "# of cells sampled in set 1: 1528\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 74\n",
      "\n",
      "# of cells sampled in set 1: 1517\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 75\n",
      "\n",
      "# of cells sampled in set 1: 1495\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 76\n",
      "\n",
      "# of cells sampled in set 1: 1494\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 77\n",
      "\n",
      "# of cells sampled in set 1: 1482\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 78\n",
      "\n",
      "# of cells sampled in set 1: 1480\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 79\n",
      "\n",
      "# of cells sampled in set 1: 1474\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 80\n",
      "\n",
      "# of cells sampled in set 1: 1467\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 81\n",
      "\n",
      "# of cells sampled in set 1: 1455\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 82\n",
      "\n",
      "# of cells sampled in set 1: 1451\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 83\n",
      "\n",
      "# of cells sampled in set 1: 1445\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 84\n",
      "\n",
      "# of cells sampled in set 1: 1416\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 85\n",
      "\n",
      "# of cells sampled in set 1: 1409\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 86\n",
      "\n",
      "# of cells sampled in set 1: 1406\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 87\n",
      "\n",
      "# of cells sampled in set 1: 1403\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 88\n",
      "\n",
      "# of cells sampled in set 1: 1402\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 89\n",
      "\n",
      "# of cells sampled in set 1: 1378\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 90\n",
      "\n",
      "# of cells sampled in set 1: 1377\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 91\n",
      "\n",
      "# of cells sampled in set 1: 1374\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 92\n",
      "\n",
      "# of cells sampled in set 1: 1369\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 93\n",
      "\n",
      "# of cells sampled in set 1: 1357\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 94\n",
      "\n",
      "# of cells sampled in set 1: 1355\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 95\n",
      "\n",
      "# of cells sampled in set 1: 1352\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 96\n",
      "\n",
      "# of cells sampled in set 1: 1351\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 97\n",
      "\n",
      "# of cells sampled in set 1: 1347\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 98\n",
      "\n",
      "# of cells sampled in set 1: 1339\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 99\n",
      "\n",
      "# of cells sampled in set 1: 1330\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 100\n",
      "\n",
      "# of cells sampled in set 1: 1316\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 101\n",
      "\n",
      "# of cells sampled in set 1: 1299\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 102\n",
      "\n",
      "# of cells sampled in set 1: 1280\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 103\n",
      "\n",
      "# of cells sampled in set 1: 1265\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 104\n",
      "\n",
      "# of cells sampled in set 1: 1260\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 105\n",
      "\n",
      "# of cells sampled in set 1: 1244\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 106\n",
      "\n",
      "# of cells sampled in set 1: 1244\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 107\n",
      "\n",
      "# of cells sampled in set 1: 1241\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 108\n",
      "\n",
      "# of cells sampled in set 1: 1225\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 109\n",
      "\n",
      "# of cells sampled in set 1: 1221\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 110\n",
      "\n",
      "# of cells sampled in set 1: 1209\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 111\n",
      "\n",
      "# of cells sampled in set 1: 1199\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 112\n",
      "\n",
      "# of cells sampled in set 1: 1196\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 113\n",
      "\n",
      "# of cells sampled in set 1: 1189\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 114\n",
      "\n",
      "# of cells sampled in set 1: 1186\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 115\n",
      "\n",
      "# of cells sampled in set 1: 1184\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 116\n",
      "\n",
      "# of cells sampled in set 1: 1168\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 117\n",
      "\n",
      "# of cells sampled in set 1: 1167\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 118\n",
      "\n",
      "# of cells sampled in set 1: 1164\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 119\n",
      "\n",
      "# of cells sampled in set 1: 1160\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 120\n",
      "\n",
      "# of cells sampled in set 1: 1148\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 121\n",
      "\n",
      "# of cells sampled in set 1: 1146\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 122\n",
      "\n",
      "# of cells sampled in set 1: 1140\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 123\n",
      "\n",
      "# of cells sampled in set 1: 1125\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 124\n",
      "\n",
      "# of cells sampled in set 1: 1101\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 125\n",
      "\n",
      "# of cells sampled in set 1: 1101\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 126\n",
      "\n",
      "# of cells sampled in set 1: 1081\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 127\n",
      "\n",
      "# of cells sampled in set 1: 1063\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 128\n",
      "\n",
      "# of cells sampled in set 1: 1062\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 129\n",
      "\n",
      "# of cells sampled in set 1: 1056\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 130\n",
      "\n",
      "# of cells sampled in set 1: 1052\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 131\n",
      "\n",
      "# of cells sampled in set 1: 1043\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 132\n",
      "\n",
      "# of cells sampled in set 1: 1028\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 133\n",
      "\n",
      "# of cells sampled in set 1: 1003\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 134\n",
      "\n",
      "# of cells sampled in set 1: 1000\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 135\n",
      "\n",
      "# of cells sampled in set 1: 998\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 136\n",
      "\n",
      "# of cells sampled in set 1: 987\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 137\n",
      "\n",
      "# of cells sampled in set 1: 981\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 138\n",
      "\n",
      "# of cells sampled in set 1: 979\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 139\n",
      "\n",
      "# of cells sampled in set 1: 970\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 140\n",
      "\n",
      "# of cells sampled in set 1: 962\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 141\n",
      "\n",
      "# of cells sampled in set 1: 956\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 142\n",
      "\n",
      "# of cells sampled in set 1: 931\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 143\n",
      "\n",
      "# of cells sampled in set 1: 898\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 144\n",
      "\n",
      "# of cells sampled in set 1: 890\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 145\n",
      "\n",
      "# of cells sampled in set 1: 888\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 146\n",
      "\n",
      "# of cells sampled in set 1: 887\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 147\n",
      "\n",
      "# of cells sampled in set 1: 850\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 148\n",
      "\n",
      "# of cells sampled in set 1: 748\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 149\n",
      "\n",
      "# of cells sampled in set 1: 717\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 150\n",
      "\n",
      "# of cells sampled in set 1: 697\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 151\n",
      "\n",
      "# of cells sampled in set 1: 673\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 152\n",
      "\n",
      "# of cells sampled in set 1: 651\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 153\n",
      "\n",
      "# of cells sampled in set 1: 645\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 154\n",
      "\n",
      "# of cells sampled in set 1: 640\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 155\n",
      "\n",
      "# of cells sampled in set 1: 636\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 156\n",
      "\n",
      "# of cells sampled in set 1: 617\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 157\n",
      "\n",
      "# of cells sampled in set 1: 610\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 158\n",
      "\n",
      "# of cells sampled in set 1: 607\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 159\n",
      "\n",
      "# of cells sampled in set 1: 581\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finding markers for cluster 160\n",
      "\n",
      "# of cells sampled in set 1: 528\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 161\n",
      "\n",
      "# of cells sampled in set 1: 526\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 162\n",
      "\n",
      "# of cells sampled in set 1: 485\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 163\n",
      "\n",
      "# of cells sampled in set 1: 466\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 164\n",
      "\n",
      "# of cells sampled in set 1: 453\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 165\n",
      "\n",
      "# of cells sampled in set 1: 432\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 166\n",
      "\n",
      "# of cells sampled in set 1: 371\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 167\n",
      "\n",
      "# of cells sampled in set 1: 370\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 168\n",
      "\n",
      "# of cells sampled in set 1: 225\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 169\n",
      "\n",
      "# of cells sampled in set 1: 184\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 170\n",
      "\n",
      "# of cells sampled in set 1: 149\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 171\n",
      "\n",
      "# of cells sampled in set 1: 147\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 172\n",
      "\n",
      "# of cells sampled in set 1: 121\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 173\n",
      "\n",
      "# of cells sampled in set 1: 115\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 174\n",
      "\n",
      "# of cells sampled in set 1: 59\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    }
   ],
   "source": [
    "DEG.results.sub300 = lapply(sort((unique(df_sub300$cluster))),\n",
    "                     function(this.cluster) {\n",
    "    message(\"Finding markers for cluster \", this.cluster)\n",
    "    cbind(\n",
    "        two.set.differential.gene.test.3a(\n",
    "            cds_sub,\n",
    "            df_sub300$cluster == this.cluster,\n",
    "            df_sub300$cluster != this.cluster,\n",
    "            sample.n.max = 50000),\n",
    "        data.frame(cluster = this.cluster))\n",
    "                         })"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1256,
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster.markers.sub300 = do.call(rbind, DEG.results.sub300) %>% filter(higher.expr == \"Set 1\") %>%\n",
    "    select(\n",
    "        cluster = cluster, gene, \n",
    "        cluster.n.umi = set.1.n.umi,\n",
    "        other.n.umi = set.2.n.umi,\n",
    "        cluster.tpm = set.1.tpm,\n",
    "        other.tpm = set.2.tpm,\n",
    "        log2.ratio, heuristic.score) %>%\n",
    "    arrange(-heuristic.score)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1281,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cluster</th><th scope=col>gene</th><th scope=col>cluster.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cluster.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG43896158</th><td>159</td><td>CG43896</td><td>6181</td><td>103</td><td>169782.689</td><td> 3.9803178</td><td>15.057092</td><td>261.5920</td></tr>\n",
       "\t<tr><th scope=row>Muc26B156</th><td>157</td><td>Muc26B </td><td>3922</td><td>129</td><td> 78413.767</td><td> 6.2073293</td><td>13.409355</td><td>218.0205</td></tr>\n",
       "\t<tr><th scope=row>CG7715158</th><td>159</td><td>CG7715 </td><td>5655</td><td>607</td><td>150265.856</td><td>22.3719218</td><td>12.650453</td><td>217.5520</td></tr>\n",
       "\t<tr><th scope=row>Muc96D156</th><td>157</td><td>Muc96D </td><td>2574</td><td> 92</td><td> 49862.918</td><td> 4.3355259</td><td>13.190049</td><td>205.8401</td></tr>\n",
       "\t<tr><th scope=row>CG7017156</th><td>157</td><td>CG7017 </td><td>3325</td><td>328</td><td> 77239.149</td><td>13.8080423</td><td>12.348736</td><td>200.5072</td></tr>\n",
       "\t<tr><th scope=row>CG17826156</th><td>157</td><td>CG17826</td><td>1070</td><td> 30</td><td> 21551.482</td><td> 1.5907303</td><td>13.022141</td><td>187.4611</td></tr>\n",
       "\t<tr><th scope=row>Cda9158</th><td>159</td><td>Cda9   </td><td> 569</td><td> 11</td><td> 15473.694</td><td> 0.4088698</td><td>13.422992</td><td>186.8161</td></tr>\n",
       "\t<tr><th scope=row>obst-F158</th><td>159</td><td>obst-F </td><td> 466</td><td>  6</td><td> 12801.082</td><td> 0.2458294</td><td>13.326872</td><td>181.8330</td></tr>\n",
       "\t<tr><th scope=row>CG13545157</th><td>158</td><td>CG13545</td><td> 900</td><td> 65</td><td> 12518.634</td><td> 2.5552825</td><td>11.781825</td><td>160.3731</td></tr>\n",
       "\t<tr><th scope=row>CG11370153</th><td>154</td><td>CG11370</td><td>4908</td><td>237</td><td> 17775.077</td><td> 5.9962850</td><td>11.310979</td><td>159.6844</td></tr>\n",
       "\t<tr><th scope=row>CG14424147</th><td>148</td><td>CG14424</td><td>3853</td><td> 92</td><td>  7742.030</td><td> 2.8889280</td><td>10.959124</td><td>141.5774</td></tr>\n",
       "\t<tr><th scope=row>CG11672156</th><td>157</td><td>CG11672</td><td> 185</td><td> 24</td><td>  5400.371</td><td> 1.0200242</td><td>11.384470</td><td>141.1573</td></tr>\n",
       "\t<tr><th scope=row>CG14421147</th><td>148</td><td>CG14421</td><td>3950</td><td>102</td><td>  7777.688</td><td> 3.0694983</td><td>10.900275</td><td>140.8894</td></tr>\n",
       "\t<tr><th scope=row>Muc11A148</th><td>149</td><td>Muc11A </td><td>4377</td><td> 28</td><td>  5517.634</td><td> 1.2092005</td><td>11.286310</td><td>140.2899</td></tr>\n",
       "\t<tr><th scope=row>CG7409155</th><td>156</td><td>CG7409 </td><td> 592</td><td> 20</td><td>  4170.495</td><td> 0.5616132</td><td>11.382966</td><td>136.8955</td></tr>\n",
       "\t<tr><th scope=row>CG7203157</th><td>158</td><td>CG7203 </td><td> 236</td><td> 48</td><td>  4674.008</td><td> 1.7812605</td><td>10.714706</td><td>130.6203</td></tr>\n",
       "\t<tr><th scope=row>Alp8148</th><td>149</td><td>Alp8   </td><td>4112</td><td> 34</td><td>  3884.164</td><td> 1.0801927</td><td>10.866671</td><td>129.5716</td></tr>\n",
       "\t<tr><th scope=row>CG13044157</th><td>158</td><td>CG13044</td><td> 477</td><td>340</td><td>  9651.846</td><td>11.3795358</td><td> 9.606704</td><td>127.1614</td></tr>\n",
       "\t<tr><th scope=row>CG32793147</th><td>148</td><td>CG32793</td><td>2075</td><td> 67</td><td>  4550.359</td><td> 2.3260443</td><td>10.417957</td><td>126.5999</td></tr>\n",
       "\t<tr><th scope=row>CG34282156</th><td>157</td><td>CG34282</td><td> 164</td><td> 21</td><td>  3215.156</td><td> 0.7353207</td><td>10.855471</td><td>126.4784</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cluster & gene & cluster.n.umi & other.n.umi & cluster.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <fct> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG43896158 & 159 & CG43896 & 6181 & 103 & 169782.689 &  3.9803178 & 15.057092 & 261.5920\\\\\n",
       "\tMuc26B156 & 157 & Muc26B  & 3922 & 129 &  78413.767 &  6.2073293 & 13.409355 & 218.0205\\\\\n",
       "\tCG7715158 & 159 & CG7715  & 5655 & 607 & 150265.856 & 22.3719218 & 12.650453 & 217.5520\\\\\n",
       "\tMuc96D156 & 157 & Muc96D  & 2574 &  92 &  49862.918 &  4.3355259 & 13.190049 & 205.8401\\\\\n",
       "\tCG7017156 & 157 & CG7017  & 3325 & 328 &  77239.149 & 13.8080423 & 12.348736 & 200.5072\\\\\n",
       "\tCG17826156 & 157 & CG17826 & 1070 &  30 &  21551.482 &  1.5907303 & 13.022141 & 187.4611\\\\\n",
       "\tCda9158 & 159 & Cda9    &  569 &  11 &  15473.694 &  0.4088698 & 13.422992 & 186.8161\\\\\n",
       "\tobst-F158 & 159 & obst-F  &  466 &   6 &  12801.082 &  0.2458294 & 13.326872 & 181.8330\\\\\n",
       "\tCG13545157 & 158 & CG13545 &  900 &  65 &  12518.634 &  2.5552825 & 11.781825 & 160.3731\\\\\n",
       "\tCG11370153 & 154 & CG11370 & 4908 & 237 &  17775.077 &  5.9962850 & 11.310979 & 159.6844\\\\\n",
       "\tCG14424147 & 148 & CG14424 & 3853 &  92 &   7742.030 &  2.8889280 & 10.959124 & 141.5774\\\\\n",
       "\tCG11672156 & 157 & CG11672 &  185 &  24 &   5400.371 &  1.0200242 & 11.384470 & 141.1573\\\\\n",
       "\tCG14421147 & 148 & CG14421 & 3950 & 102 &   7777.688 &  3.0694983 & 10.900275 & 140.8894\\\\\n",
       "\tMuc11A148 & 149 & Muc11A  & 4377 &  28 &   5517.634 &  1.2092005 & 11.286310 & 140.2899\\\\\n",
       "\tCG7409155 & 156 & CG7409  &  592 &  20 &   4170.495 &  0.5616132 & 11.382966 & 136.8955\\\\\n",
       "\tCG7203157 & 158 & CG7203  &  236 &  48 &   4674.008 &  1.7812605 & 10.714706 & 130.6203\\\\\n",
       "\tAlp8148 & 149 & Alp8    & 4112 &  34 &   3884.164 &  1.0801927 & 10.866671 & 129.5716\\\\\n",
       "\tCG13044157 & 158 & CG13044 &  477 & 340 &   9651.846 & 11.3795358 &  9.606704 & 127.1614\\\\\n",
       "\tCG32793147 & 148 & CG32793 & 2075 &  67 &   4550.359 &  2.3260443 & 10.417957 & 126.5999\\\\\n",
       "\tCG34282156 & 157 & CG34282 &  164 &  21 &   3215.156 &  0.7353207 & 10.855471 & 126.4784\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cluster &lt;fct&gt; | gene &lt;chr&gt; | cluster.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cluster.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG43896158 | 159 | CG43896 | 6181 | 103 | 169782.689 |  3.9803178 | 15.057092 | 261.5920 |\n",
       "| Muc26B156 | 157 | Muc26B  | 3922 | 129 |  78413.767 |  6.2073293 | 13.409355 | 218.0205 |\n",
       "| CG7715158 | 159 | CG7715  | 5655 | 607 | 150265.856 | 22.3719218 | 12.650453 | 217.5520 |\n",
       "| Muc96D156 | 157 | Muc96D  | 2574 |  92 |  49862.918 |  4.3355259 | 13.190049 | 205.8401 |\n",
       "| CG7017156 | 157 | CG7017  | 3325 | 328 |  77239.149 | 13.8080423 | 12.348736 | 200.5072 |\n",
       "| CG17826156 | 157 | CG17826 | 1070 |  30 |  21551.482 |  1.5907303 | 13.022141 | 187.4611 |\n",
       "| Cda9158 | 159 | Cda9    |  569 |  11 |  15473.694 |  0.4088698 | 13.422992 | 186.8161 |\n",
       "| obst-F158 | 159 | obst-F  |  466 |   6 |  12801.082 |  0.2458294 | 13.326872 | 181.8330 |\n",
       "| CG13545157 | 158 | CG13545 |  900 |  65 |  12518.634 |  2.5552825 | 11.781825 | 160.3731 |\n",
       "| CG11370153 | 154 | CG11370 | 4908 | 237 |  17775.077 |  5.9962850 | 11.310979 | 159.6844 |\n",
       "| CG14424147 | 148 | CG14424 | 3853 |  92 |   7742.030 |  2.8889280 | 10.959124 | 141.5774 |\n",
       "| CG11672156 | 157 | CG11672 |  185 |  24 |   5400.371 |  1.0200242 | 11.384470 | 141.1573 |\n",
       "| CG14421147 | 148 | CG14421 | 3950 | 102 |   7777.688 |  3.0694983 | 10.900275 | 140.8894 |\n",
       "| Muc11A148 | 149 | Muc11A  | 4377 |  28 |   5517.634 |  1.2092005 | 11.286310 | 140.2899 |\n",
       "| CG7409155 | 156 | CG7409  |  592 |  20 |   4170.495 |  0.5616132 | 11.382966 | 136.8955 |\n",
       "| CG7203157 | 158 | CG7203  |  236 |  48 |   4674.008 |  1.7812605 | 10.714706 | 130.6203 |\n",
       "| Alp8148 | 149 | Alp8    | 4112 |  34 |   3884.164 |  1.0801927 | 10.866671 | 129.5716 |\n",
       "| CG13044157 | 158 | CG13044 |  477 | 340 |   9651.846 | 11.3795358 |  9.606704 | 127.1614 |\n",
       "| CG32793147 | 148 | CG32793 | 2075 |  67 |   4550.359 |  2.3260443 | 10.417957 | 126.5999 |\n",
       "| CG34282156 | 157 | CG34282 |  164 |  21 |   3215.156 |  0.7353207 | 10.855471 | 126.4784 |\n",
       "\n"
      ],
      "text/plain": [
       "           cluster gene    cluster.n.umi other.n.umi cluster.tpm other.tpm \n",
       "CG43896158 159     CG43896 6181          103         169782.689   3.9803178\n",
       "Muc26B156  157     Muc26B  3922          129          78413.767   6.2073293\n",
       "CG7715158  159     CG7715  5655          607         150265.856  22.3719218\n",
       "Muc96D156  157     Muc96D  2574           92          49862.918   4.3355259\n",
       "CG7017156  157     CG7017  3325          328          77239.149  13.8080423\n",
       "CG17826156 157     CG17826 1070           30          21551.482   1.5907303\n",
       "Cda9158    159     Cda9     569           11          15473.694   0.4088698\n",
       "obst-F158  159     obst-F   466            6          12801.082   0.2458294\n",
       "CG13545157 158     CG13545  900           65          12518.634   2.5552825\n",
       "CG11370153 154     CG11370 4908          237          17775.077   5.9962850\n",
       "CG14424147 148     CG14424 3853           92           7742.030   2.8889280\n",
       "CG11672156 157     CG11672  185           24           5400.371   1.0200242\n",
       "CG14421147 148     CG14421 3950          102           7777.688   3.0694983\n",
       "Muc11A148  149     Muc11A  4377           28           5517.634   1.2092005\n",
       "CG7409155  156     CG7409   592           20           4170.495   0.5616132\n",
       "CG7203157  158     CG7203   236           48           4674.008   1.7812605\n",
       "Alp8148    149     Alp8    4112           34           3884.164   1.0801927\n",
       "CG13044157 158     CG13044  477          340           9651.846  11.3795358\n",
       "CG32793147 148     CG32793 2075           67           4550.359   2.3260443\n",
       "CG34282156 157     CG34282  164           21           3215.156   0.7353207\n",
       "           log2.ratio heuristic.score\n",
       "CG43896158 15.057092  261.5920       \n",
       "Muc26B156  13.409355  218.0205       \n",
       "CG7715158  12.650453  217.5520       \n",
       "Muc96D156  13.190049  205.8401       \n",
       "CG7017156  12.348736  200.5072       \n",
       "CG17826156 13.022141  187.4611       \n",
       "Cda9158    13.422992  186.8161       \n",
       "obst-F158  13.326872  181.8330       \n",
       "CG13545157 11.781825  160.3731       \n",
       "CG11370153 11.310979  159.6844       \n",
       "CG14424147 10.959124  141.5774       \n",
       "CG11672156 11.384470  141.1573       \n",
       "CG14421147 10.900275  140.8894       \n",
       "Muc11A148  11.286310  140.2899       \n",
       "CG7409155  11.382966  136.8955       \n",
       "CG7203157  10.714706  130.6203       \n",
       "Alp8148    10.866671  129.5716       \n",
       "CG13044157  9.606704  127.1614       \n",
       "CG32793147 10.417957  126.5999       \n",
       "CG34282156 10.855471  126.4784       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cluster</th><th scope=col>gene</th><th scope=col>cluster.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cluster.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Muc26B159</th><td>160</td><td>Muc26B </td><td>3993</td><td>111</td><td>10368.322</td><td> 4.1576480</td><td>10.973182</td><td>146.3826</td></tr>\n",
       "\t<tr><th scope=row>Muc11A162</th><td>163</td><td>Muc11A </td><td>4284</td><td> 73</td><td> 6913.090</td><td> 2.4916789</td><td>10.951194</td><td>139.6860</td></tr>\n",
       "\t<tr><th scope=row>Alp8162</th><td>163</td><td>Alp8   </td><td>4089</td><td> 31</td><td> 4923.237</td><td> 0.9517154</td><td>11.300649</td><td>138.6102</td></tr>\n",
       "\t<tr><th scope=row>CG7409171</th><td>172</td><td>CG7409 </td><td> 600</td><td> 24</td><td> 4121.228</td><td> 0.4739611</td><td>11.449160</td><td>137.4954</td></tr>\n",
       "\t<tr><th scope=row>CG11370167</th><td>168</td><td>CG11370</td><td>3708</td><td>446</td><td>15737.252</td><td>17.5042494</td><td> 9.732111</td><td>135.6850</td></tr>\n",
       "\t<tr><th scope=row>CG7017159</th><td>160</td><td>CG7017 </td><td>3759</td><td>265</td><td>11685.814</td><td>10.6958361</td><td> 9.964547</td><td>134.6469</td></tr>\n",
       "\t<tr><th scope=row>ppk503</th><td>170</td><td>ppk    </td><td> 456</td><td> 59</td><td> 5091.065</td><td> 1.7152860</td><td>10.872648</td><td>133.8862</td></tr>\n",
       "\t<tr><th scope=row>CG11350140</th><td>141</td><td>CG11350</td><td>7260</td><td>239</td><td> 9507.727</td><td> 8.1099484</td><td>10.027442</td><td>132.5130</td></tr>\n",
       "\t<tr><th scope=row>Mur18B162</th><td>163</td><td>Mur18B </td><td>2971</td><td> 77</td><td> 5330.350</td><td> 2.6636099</td><td>10.506749</td><td>130.0765</td></tr>\n",
       "\t<tr><th scope=row>CG14421150</th><td>151</td><td>CG14421</td><td>3722</td><td>236</td><td> 7381.196</td><td> 5.7984872</td><td>10.084425</td><td>129.5832</td></tr>\n",
       "\t<tr><th scope=row>CG14292162</th><td>163</td><td>CG14292</td><td>2862</td><td> 67</td><td> 4914.489</td><td> 2.6403965</td><td>10.398730</td><td>127.5209</td></tr>\n",
       "\t<tr><th scope=row>CG14424150</th><td>151</td><td>CG14424</td><td>3521</td><td>187</td><td> 7099.158</td><td> 6.1741798</td><td> 9.950618</td><td>127.3046</td></tr>\n",
       "\t<tr><th scope=row>Muc96D159</th><td>160</td><td>Muc96D </td><td>2590</td><td>118</td><td> 6427.809</td><td> 5.1071969</td><td>10.039601</td><td>127.0043</td></tr>\n",
       "\t<tr><th scope=row>CG13133171</th><td>172</td><td>CG13133</td><td> 405</td><td> 14</td><td> 2679.090</td><td> 0.1829864</td><td>11.145094</td><td>126.9211</td></tr>\n",
       "\t<tr><th scope=row>CG14566167</th><td>168</td><td>CG14566</td><td> 518</td><td> 60</td><td> 3985.539</td><td> 1.6730948</td><td>10.542048</td><td>126.0926</td></tr>\n",
       "\t<tr><th scope=row>CG43149171</th><td>172</td><td>CG43149</td><td> 527</td><td> 40</td><td> 3407.622</td><td> 1.1778353</td><td>10.611655</td><td>124.5275</td></tr>\n",
       "\t<tr><th scope=row>Nplp4167</th><td>168</td><td>Nplp4  </td><td> 585</td><td> 98</td><td> 3518.479</td><td> 1.4273695</td><td>10.501343</td><td>123.7179</td></tr>\n",
       "\t<tr><th scope=row>CG43666150</th><td>151</td><td>CG43666</td><td>1759</td><td> 58</td><td> 3440.841</td><td> 1.4061733</td><td>10.481805</td><td>123.1504</td></tr>\n",
       "\t<tr><th scope=row>ppk26169</th><td>170</td><td>ppk26  </td><td> 303</td><td> 40</td><td> 3407.935</td><td> 1.3971566</td><td>10.473358</td><td>122.9060</td></tr>\n",
       "\t<tr><th scope=row>CG3546150</th><td>151</td><td>CG3546 </td><td>1809</td><td> 70</td><td> 3732.656</td><td> 2.3612670</td><td>10.116982</td><td>120.0519</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cluster & gene & cluster.n.umi & other.n.umi & cluster.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <fct> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tMuc26B159 & 160 & Muc26B  & 3993 & 111 & 10368.322 &  4.1576480 & 10.973182 & 146.3826\\\\\n",
       "\tMuc11A162 & 163 & Muc11A  & 4284 &  73 &  6913.090 &  2.4916789 & 10.951194 & 139.6860\\\\\n",
       "\tAlp8162 & 163 & Alp8    & 4089 &  31 &  4923.237 &  0.9517154 & 11.300649 & 138.6102\\\\\n",
       "\tCG7409171 & 172 & CG7409  &  600 &  24 &  4121.228 &  0.4739611 & 11.449160 & 137.4954\\\\\n",
       "\tCG11370167 & 168 & CG11370 & 3708 & 446 & 15737.252 & 17.5042494 &  9.732111 & 135.6850\\\\\n",
       "\tCG7017159 & 160 & CG7017  & 3759 & 265 & 11685.814 & 10.6958361 &  9.964547 & 134.6469\\\\\n",
       "\tppk503 & 170 & ppk     &  456 &  59 &  5091.065 &  1.7152860 & 10.872648 & 133.8862\\\\\n",
       "\tCG11350140 & 141 & CG11350 & 7260 & 239 &  9507.727 &  8.1099484 & 10.027442 & 132.5130\\\\\n",
       "\tMur18B162 & 163 & Mur18B  & 2971 &  77 &  5330.350 &  2.6636099 & 10.506749 & 130.0765\\\\\n",
       "\tCG14421150 & 151 & CG14421 & 3722 & 236 &  7381.196 &  5.7984872 & 10.084425 & 129.5832\\\\\n",
       "\tCG14292162 & 163 & CG14292 & 2862 &  67 &  4914.489 &  2.6403965 & 10.398730 & 127.5209\\\\\n",
       "\tCG14424150 & 151 & CG14424 & 3521 & 187 &  7099.158 &  6.1741798 &  9.950618 & 127.3046\\\\\n",
       "\tMuc96D159 & 160 & Muc96D  & 2590 & 118 &  6427.809 &  5.1071969 & 10.039601 & 127.0043\\\\\n",
       "\tCG13133171 & 172 & CG13133 &  405 &  14 &  2679.090 &  0.1829864 & 11.145094 & 126.9211\\\\\n",
       "\tCG14566167 & 168 & CG14566 &  518 &  60 &  3985.539 &  1.6730948 & 10.542048 & 126.0926\\\\\n",
       "\tCG43149171 & 172 & CG43149 &  527 &  40 &  3407.622 &  1.1778353 & 10.611655 & 124.5275\\\\\n",
       "\tNplp4167 & 168 & Nplp4   &  585 &  98 &  3518.479 &  1.4273695 & 10.501343 & 123.7179\\\\\n",
       "\tCG43666150 & 151 & CG43666 & 1759 &  58 &  3440.841 &  1.4061733 & 10.481805 & 123.1504\\\\\n",
       "\tppk26169 & 170 & ppk26   &  303 &  40 &  3407.935 &  1.3971566 & 10.473358 & 122.9060\\\\\n",
       "\tCG3546150 & 151 & CG3546  & 1809 &  70 &  3732.656 &  2.3612670 & 10.116982 & 120.0519\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cluster &lt;fct&gt; | gene &lt;chr&gt; | cluster.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cluster.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Muc26B159 | 160 | Muc26B  | 3993 | 111 | 10368.322 |  4.1576480 | 10.973182 | 146.3826 |\n",
       "| Muc11A162 | 163 | Muc11A  | 4284 |  73 |  6913.090 |  2.4916789 | 10.951194 | 139.6860 |\n",
       "| Alp8162 | 163 | Alp8    | 4089 |  31 |  4923.237 |  0.9517154 | 11.300649 | 138.6102 |\n",
       "| CG7409171 | 172 | CG7409  |  600 |  24 |  4121.228 |  0.4739611 | 11.449160 | 137.4954 |\n",
       "| CG11370167 | 168 | CG11370 | 3708 | 446 | 15737.252 | 17.5042494 |  9.732111 | 135.6850 |\n",
       "| CG7017159 | 160 | CG7017  | 3759 | 265 | 11685.814 | 10.6958361 |  9.964547 | 134.6469 |\n",
       "| ppk503 | 170 | ppk     |  456 |  59 |  5091.065 |  1.7152860 | 10.872648 | 133.8862 |\n",
       "| CG11350140 | 141 | CG11350 | 7260 | 239 |  9507.727 |  8.1099484 | 10.027442 | 132.5130 |\n",
       "| Mur18B162 | 163 | Mur18B  | 2971 |  77 |  5330.350 |  2.6636099 | 10.506749 | 130.0765 |\n",
       "| CG14421150 | 151 | CG14421 | 3722 | 236 |  7381.196 |  5.7984872 | 10.084425 | 129.5832 |\n",
       "| CG14292162 | 163 | CG14292 | 2862 |  67 |  4914.489 |  2.6403965 | 10.398730 | 127.5209 |\n",
       "| CG14424150 | 151 | CG14424 | 3521 | 187 |  7099.158 |  6.1741798 |  9.950618 | 127.3046 |\n",
       "| Muc96D159 | 160 | Muc96D  | 2590 | 118 |  6427.809 |  5.1071969 | 10.039601 | 127.0043 |\n",
       "| CG13133171 | 172 | CG13133 |  405 |  14 |  2679.090 |  0.1829864 | 11.145094 | 126.9211 |\n",
       "| CG14566167 | 168 | CG14566 |  518 |  60 |  3985.539 |  1.6730948 | 10.542048 | 126.0926 |\n",
       "| CG43149171 | 172 | CG43149 |  527 |  40 |  3407.622 |  1.1778353 | 10.611655 | 124.5275 |\n",
       "| Nplp4167 | 168 | Nplp4   |  585 |  98 |  3518.479 |  1.4273695 | 10.501343 | 123.7179 |\n",
       "| CG43666150 | 151 | CG43666 | 1759 |  58 |  3440.841 |  1.4061733 | 10.481805 | 123.1504 |\n",
       "| ppk26169 | 170 | ppk26   |  303 |  40 |  3407.935 |  1.3971566 | 10.473358 | 122.9060 |\n",
       "| CG3546150 | 151 | CG3546  | 1809 |  70 |  3732.656 |  2.3612670 | 10.116982 | 120.0519 |\n",
       "\n"
      ],
      "text/plain": [
       "           cluster gene    cluster.n.umi other.n.umi cluster.tpm other.tpm \n",
       "Muc26B159  160     Muc26B  3993          111         10368.322    4.1576480\n",
       "Muc11A162  163     Muc11A  4284           73          6913.090    2.4916789\n",
       "Alp8162    163     Alp8    4089           31          4923.237    0.9517154\n",
       "CG7409171  172     CG7409   600           24          4121.228    0.4739611\n",
       "CG11370167 168     CG11370 3708          446         15737.252   17.5042494\n",
       "CG7017159  160     CG7017  3759          265         11685.814   10.6958361\n",
       "ppk503     170     ppk      456           59          5091.065    1.7152860\n",
       "CG11350140 141     CG11350 7260          239          9507.727    8.1099484\n",
       "Mur18B162  163     Mur18B  2971           77          5330.350    2.6636099\n",
       "CG14421150 151     CG14421 3722          236          7381.196    5.7984872\n",
       "CG14292162 163     CG14292 2862           67          4914.489    2.6403965\n",
       "CG14424150 151     CG14424 3521          187          7099.158    6.1741798\n",
       "Muc96D159  160     Muc96D  2590          118          6427.809    5.1071969\n",
       "CG13133171 172     CG13133  405           14          2679.090    0.1829864\n",
       "CG14566167 168     CG14566  518           60          3985.539    1.6730948\n",
       "CG43149171 172     CG43149  527           40          3407.622    1.1778353\n",
       "Nplp4167   168     Nplp4    585           98          3518.479    1.4273695\n",
       "CG43666150 151     CG43666 1759           58          3440.841    1.4061733\n",
       "ppk26169   170     ppk26    303           40          3407.935    1.3971566\n",
       "CG3546150  151     CG3546  1809           70          3732.656    2.3612670\n",
       "           log2.ratio heuristic.score\n",
       "Muc26B159  10.973182  146.3826       \n",
       "Muc11A162  10.951194  139.6860       \n",
       "Alp8162    11.300649  138.6102       \n",
       "CG7409171  11.449160  137.4954       \n",
       "CG11370167  9.732111  135.6850       \n",
       "CG7017159   9.964547  134.6469       \n",
       "ppk503     10.872648  133.8862       \n",
       "CG11350140 10.027442  132.5130       \n",
       "Mur18B162  10.506749  130.0765       \n",
       "CG14421150 10.084425  129.5832       \n",
       "CG14292162 10.398730  127.5209       \n",
       "CG14424150  9.950618  127.3046       \n",
       "Muc96D159  10.039601  127.0043       \n",
       "CG13133171 11.145094  126.9211       \n",
       "CG14566167 10.542048  126.0926       \n",
       "CG43149171 10.611655  124.5275       \n",
       "Nplp4167   10.501343  123.7179       \n",
       "CG43666150 10.481805  123.1504       \n",
       "ppk26169   10.473358  122.9060       \n",
       "CG3546150  10.116982  120.0519       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(cluster.markers, 20)\n",
    "head(cluster.markers.sub300, 20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1283,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 12 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cluster</th><th scope=col>gene</th><th scope=col>cluster.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cluster.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG43896158</th><td>159</td><td>CG43896</td><td>6181</td><td> 103</td><td>169782.68940</td><td> 3.980318</td><td>15.0570921</td><td>261.591955</td></tr>\n",
       "\t<tr><th scope=row>Muc26B156</th><td>157</td><td>Muc26B </td><td>3922</td><td> 129</td><td> 78413.76680</td><td> 6.207329</td><td>13.4093546</td><td>218.020520</td></tr>\n",
       "\t<tr><th scope=row>CG7715158</th><td>159</td><td>CG7715 </td><td>5655</td><td> 607</td><td>150265.85566</td><td>22.371922</td><td>12.6504532</td><td>217.551961</td></tr>\n",
       "\t<tr><th scope=row>Muc11A148</th><td>149</td><td>Muc11A </td><td>4377</td><td>  28</td><td>  5517.63414</td><td> 1.209200</td><td>11.2863097</td><td>140.289908</td></tr>\n",
       "\t<tr><th scope=row>Alp8148</th><td>149</td><td>Alp8   </td><td>4112</td><td>  34</td><td>  3884.16447</td><td> 1.080193</td><td>10.8666714</td><td>129.571581</td></tr>\n",
       "\t<tr><th scope=row>CG7715156</th><td>157</td><td>CG7715 </td><td> 945</td><td>2442</td><td> 20187.57353</td><td>48.503146</td><td> 8.6717316</td><td>124.016613</td></tr>\n",
       "\t<tr><th scope=row>CG771551</th><td>52 </td><td>CG7715 </td><td> 452</td><td>3769</td><td>   715.97945</td><td>70.468913</td><td> 3.3245304</td><td> 31.535791</td></tr>\n",
       "\t<tr><th scope=row>Muc26B45</th><td>46 </td><td>Muc26B </td><td>  41</td><td> 427</td><td>    77.99196</td><td>13.425134</td><td> 2.4347407</td><td> 15.347714</td></tr>\n",
       "\t<tr><th scope=row>CG7715132</th><td>133</td><td>CG7715 </td><td>  74</td><td>3239</td><td>   148.38053</td><td>57.158770</td><td> 1.3512331</td><td>  9.759752</td></tr>\n",
       "\t<tr><th scope=row>CG43896156</th><td>157</td><td>CG43896</td><td>   2</td><td>1841</td><td>    88.33128</td><td>30.385515</td><td> 1.4928256</td><td>  9.675142</td></tr>\n",
       "\t<tr><th scope=row>Muc26B51</th><td>52 </td><td>Muc26B </td><td>  42</td><td>1176</td><td>    72.63858</td><td>32.968398</td><td> 1.0965429</td><td>  6.801187</td></tr>\n",
       "\t<tr><th scope=row>Muc26B132</th><td>133</td><td>Muc26B </td><td>  28</td><td>1780</td><td>    59.04057</td><td>32.207752</td><td> 0.8301867</td><td>  4.904631</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 12 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cluster & gene & cluster.n.umi & other.n.umi & cluster.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <fct> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG43896158 & 159 & CG43896 & 6181 &  103 & 169782.68940 &  3.980318 & 15.0570921 & 261.591955\\\\\n",
       "\tMuc26B156 & 157 & Muc26B  & 3922 &  129 &  78413.76680 &  6.207329 & 13.4093546 & 218.020520\\\\\n",
       "\tCG7715158 & 159 & CG7715  & 5655 &  607 & 150265.85566 & 22.371922 & 12.6504532 & 217.551961\\\\\n",
       "\tMuc11A148 & 149 & Muc11A  & 4377 &   28 &   5517.63414 &  1.209200 & 11.2863097 & 140.289908\\\\\n",
       "\tAlp8148 & 149 & Alp8    & 4112 &   34 &   3884.16447 &  1.080193 & 10.8666714 & 129.571581\\\\\n",
       "\tCG7715156 & 157 & CG7715  &  945 & 2442 &  20187.57353 & 48.503146 &  8.6717316 & 124.016613\\\\\n",
       "\tCG771551 & 52  & CG7715  &  452 & 3769 &    715.97945 & 70.468913 &  3.3245304 &  31.535791\\\\\n",
       "\tMuc26B45 & 46  & Muc26B  &   41 &  427 &     77.99196 & 13.425134 &  2.4347407 &  15.347714\\\\\n",
       "\tCG7715132 & 133 & CG7715  &   74 & 3239 &    148.38053 & 57.158770 &  1.3512331 &   9.759752\\\\\n",
       "\tCG43896156 & 157 & CG43896 &    2 & 1841 &     88.33128 & 30.385515 &  1.4928256 &   9.675142\\\\\n",
       "\tMuc26B51 & 52  & Muc26B  &   42 & 1176 &     72.63858 & 32.968398 &  1.0965429 &   6.801187\\\\\n",
       "\tMuc26B132 & 133 & Muc26B  &   28 & 1780 &     59.04057 & 32.207752 &  0.8301867 &   4.904631\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 12 × 8\n",
       "\n",
       "| <!--/--> | cluster &lt;fct&gt; | gene &lt;chr&gt; | cluster.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cluster.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG43896158 | 159 | CG43896 | 6181 |  103 | 169782.68940 |  3.980318 | 15.0570921 | 261.591955 |\n",
       "| Muc26B156 | 157 | Muc26B  | 3922 |  129 |  78413.76680 |  6.207329 | 13.4093546 | 218.020520 |\n",
       "| CG7715158 | 159 | CG7715  | 5655 |  607 | 150265.85566 | 22.371922 | 12.6504532 | 217.551961 |\n",
       "| Muc11A148 | 149 | Muc11A  | 4377 |   28 |   5517.63414 |  1.209200 | 11.2863097 | 140.289908 |\n",
       "| Alp8148 | 149 | Alp8    | 4112 |   34 |   3884.16447 |  1.080193 | 10.8666714 | 129.571581 |\n",
       "| CG7715156 | 157 | CG7715  |  945 | 2442 |  20187.57353 | 48.503146 |  8.6717316 | 124.016613 |\n",
       "| CG771551 | 52  | CG7715  |  452 | 3769 |    715.97945 | 70.468913 |  3.3245304 |  31.535791 |\n",
       "| Muc26B45 | 46  | Muc26B  |   41 |  427 |     77.99196 | 13.425134 |  2.4347407 |  15.347714 |\n",
       "| CG7715132 | 133 | CG7715  |   74 | 3239 |    148.38053 | 57.158770 |  1.3512331 |   9.759752 |\n",
       "| CG43896156 | 157 | CG43896 |    2 | 1841 |     88.33128 | 30.385515 |  1.4928256 |   9.675142 |\n",
       "| Muc26B51 | 52  | Muc26B  |   42 | 1176 |     72.63858 | 32.968398 |  1.0965429 |   6.801187 |\n",
       "| Muc26B132 | 133 | Muc26B  |   28 | 1780 |     59.04057 | 32.207752 |  0.8301867 |   4.904631 |\n",
       "\n"
      ],
      "text/plain": [
       "           cluster gene    cluster.n.umi other.n.umi cluster.tpm  other.tpm\n",
       "CG43896158 159     CG43896 6181           103        169782.68940  3.980318\n",
       "Muc26B156  157     Muc26B  3922           129         78413.76680  6.207329\n",
       "CG7715158  159     CG7715  5655           607        150265.85566 22.371922\n",
       "Muc11A148  149     Muc11A  4377            28          5517.63414  1.209200\n",
       "Alp8148    149     Alp8    4112            34          3884.16447  1.080193\n",
       "CG7715156  157     CG7715   945          2442         20187.57353 48.503146\n",
       "CG771551   52      CG7715   452          3769           715.97945 70.468913\n",
       "Muc26B45   46      Muc26B    41           427            77.99196 13.425134\n",
       "CG7715132  133     CG7715    74          3239           148.38053 57.158770\n",
       "CG43896156 157     CG43896    2          1841            88.33128 30.385515\n",
       "Muc26B51   52      Muc26B    42          1176            72.63858 32.968398\n",
       "Muc26B132  133     Muc26B    28          1780            59.04057 32.207752\n",
       "           log2.ratio heuristic.score\n",
       "CG43896158 15.0570921 261.591955     \n",
       "Muc26B156  13.4093546 218.020520     \n",
       "CG7715158  12.6504532 217.551961     \n",
       "Muc11A148  11.2863097 140.289908     \n",
       "Alp8148    10.8666714 129.571581     \n",
       "CG7715156   8.6717316 124.016613     \n",
       "CG771551    3.3245304  31.535791     \n",
       "Muc26B45    2.4347407  15.347714     \n",
       "CG7715132   1.3512331   9.759752     \n",
       "CG43896156  1.4928256   9.675142     \n",
       "Muc26B51    1.0965429   6.801187     \n",
       "Muc26B132   0.8301867   4.904631     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 12 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cluster</th><th scope=col>gene</th><th scope=col>cluster.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cluster.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Muc26B159</th><td>160</td><td>Muc26B </td><td>3993</td><td> 111</td><td>10368.32230</td><td> 4.1576480</td><td>10.9731815</td><td>146.382615</td></tr>\n",
       "\t<tr><th scope=row>Muc11A162</th><td>163</td><td>Muc11A </td><td>4284</td><td>  73</td><td> 6913.08960</td><td> 2.4916789</td><td>10.9511940</td><td>139.686023</td></tr>\n",
       "\t<tr><th scope=row>Alp8162</th><td>163</td><td>Alp8   </td><td>4089</td><td>  31</td><td> 4923.23702</td><td> 0.9517154</td><td>11.3006488</td><td>138.610192</td></tr>\n",
       "\t<tr><th scope=row>CG438961</th><td>2  </td><td>CG43896</td><td>6210</td><td>  83</td><td> 2734.47541</td><td> 2.9518611</td><td> 9.4345161</td><td>107.719304</td></tr>\n",
       "\t<tr><th scope=row>CG77151</th><td>2  </td><td>CG7715 </td><td>6105</td><td> 486</td><td> 2912.44871</td><td>16.6408119</td><td> 7.3671719</td><td> 84.785187</td></tr>\n",
       "\t<tr><th scope=row>CG7715159</th><td>160</td><td>CG7715 </td><td>1011</td><td>2690</td><td> 2863.91837</td><td>63.7625614</td><td> 5.4666865</td><td> 62.780949</td></tr>\n",
       "\t<tr><th scope=row>CG7715170</th><td>171</td><td>CG7715 </td><td>  12</td><td>2034</td><td>  310.73237</td><td>53.9292147</td><td> 2.5000270</td><td> 20.710634</td></tr>\n",
       "\t<tr><th scope=row>CG43896170</th><td>171</td><td>CG43896</td><td>   4</td><td>1348</td><td>  108.84354</td><td>34.2476549</td><td> 1.6266566</td><td> 11.027603</td></tr>\n",
       "\t<tr><th scope=row>CG7715168</th><td>169</td><td>CG7715 </td><td>  13</td><td>1553</td><td>  122.02448</td><td>48.4356305</td><td> 1.3035475</td><td>  9.050271</td></tr>\n",
       "\t<tr><th scope=row>CG771531</th><td>32 </td><td>CG7715 </td><td>  72</td><td>1460</td><td>  115.05603</td><td>45.5284719</td><td> 1.3061508</td><td>  8.958468</td></tr>\n",
       "\t<tr><th scope=row>Muc26B1</th><td>2  </td><td>Muc26B </td><td>  43</td><td> 839</td><td>   53.10378</td><td>22.0104848</td><td> 1.2065231</td><td>  6.946747</td></tr>\n",
       "\t<tr><th scope=row>CG7715134</th><td>135</td><td>CG7715 </td><td>  32</td><td>1803</td><td>   90.12055</td><td>50.1681292</td><td> 0.8166106</td><td>  5.315894</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 12 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cluster & gene & cluster.n.umi & other.n.umi & cluster.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <fct> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tMuc26B159 & 160 & Muc26B  & 3993 &  111 & 10368.32230 &  4.1576480 & 10.9731815 & 146.382615\\\\\n",
       "\tMuc11A162 & 163 & Muc11A  & 4284 &   73 &  6913.08960 &  2.4916789 & 10.9511940 & 139.686023\\\\\n",
       "\tAlp8162 & 163 & Alp8    & 4089 &   31 &  4923.23702 &  0.9517154 & 11.3006488 & 138.610192\\\\\n",
       "\tCG438961 & 2   & CG43896 & 6210 &   83 &  2734.47541 &  2.9518611 &  9.4345161 & 107.719304\\\\\n",
       "\tCG77151 & 2   & CG7715  & 6105 &  486 &  2912.44871 & 16.6408119 &  7.3671719 &  84.785187\\\\\n",
       "\tCG7715159 & 160 & CG7715  & 1011 & 2690 &  2863.91837 & 63.7625614 &  5.4666865 &  62.780949\\\\\n",
       "\tCG7715170 & 171 & CG7715  &   12 & 2034 &   310.73237 & 53.9292147 &  2.5000270 &  20.710634\\\\\n",
       "\tCG43896170 & 171 & CG43896 &    4 & 1348 &   108.84354 & 34.2476549 &  1.6266566 &  11.027603\\\\\n",
       "\tCG7715168 & 169 & CG7715  &   13 & 1553 &   122.02448 & 48.4356305 &  1.3035475 &   9.050271\\\\\n",
       "\tCG771531 & 32  & CG7715  &   72 & 1460 &   115.05603 & 45.5284719 &  1.3061508 &   8.958468\\\\\n",
       "\tMuc26B1 & 2   & Muc26B  &   43 &  839 &    53.10378 & 22.0104848 &  1.2065231 &   6.946747\\\\\n",
       "\tCG7715134 & 135 & CG7715  &   32 & 1803 &    90.12055 & 50.1681292 &  0.8166106 &   5.315894\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 12 × 8\n",
       "\n",
       "| <!--/--> | cluster &lt;fct&gt; | gene &lt;chr&gt; | cluster.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cluster.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Muc26B159 | 160 | Muc26B  | 3993 |  111 | 10368.32230 |  4.1576480 | 10.9731815 | 146.382615 |\n",
       "| Muc11A162 | 163 | Muc11A  | 4284 |   73 |  6913.08960 |  2.4916789 | 10.9511940 | 139.686023 |\n",
       "| Alp8162 | 163 | Alp8    | 4089 |   31 |  4923.23702 |  0.9517154 | 11.3006488 | 138.610192 |\n",
       "| CG438961 | 2   | CG43896 | 6210 |   83 |  2734.47541 |  2.9518611 |  9.4345161 | 107.719304 |\n",
       "| CG77151 | 2   | CG7715  | 6105 |  486 |  2912.44871 | 16.6408119 |  7.3671719 |  84.785187 |\n",
       "| CG7715159 | 160 | CG7715  | 1011 | 2690 |  2863.91837 | 63.7625614 |  5.4666865 |  62.780949 |\n",
       "| CG7715170 | 171 | CG7715  |   12 | 2034 |   310.73237 | 53.9292147 |  2.5000270 |  20.710634 |\n",
       "| CG43896170 | 171 | CG43896 |    4 | 1348 |   108.84354 | 34.2476549 |  1.6266566 |  11.027603 |\n",
       "| CG7715168 | 169 | CG7715  |   13 | 1553 |   122.02448 | 48.4356305 |  1.3035475 |   9.050271 |\n",
       "| CG771531 | 32  | CG7715  |   72 | 1460 |   115.05603 | 45.5284719 |  1.3061508 |   8.958468 |\n",
       "| Muc26B1 | 2   | Muc26B  |   43 |  839 |    53.10378 | 22.0104848 |  1.2065231 |   6.946747 |\n",
       "| CG7715134 | 135 | CG7715  |   32 | 1803 |    90.12055 | 50.1681292 |  0.8166106 |   5.315894 |\n",
       "\n"
      ],
      "text/plain": [
       "           cluster gene    cluster.n.umi other.n.umi cluster.tpm other.tpm \n",
       "Muc26B159  160     Muc26B  3993           111        10368.32230  4.1576480\n",
       "Muc11A162  163     Muc11A  4284            73         6913.08960  2.4916789\n",
       "Alp8162    163     Alp8    4089            31         4923.23702  0.9517154\n",
       "CG438961   2       CG43896 6210            83         2734.47541  2.9518611\n",
       "CG77151    2       CG7715  6105           486         2912.44871 16.6408119\n",
       "CG7715159  160     CG7715  1011          2690         2863.91837 63.7625614\n",
       "CG7715170  171     CG7715    12          2034          310.73237 53.9292147\n",
       "CG43896170 171     CG43896    4          1348          108.84354 34.2476549\n",
       "CG7715168  169     CG7715    13          1553          122.02448 48.4356305\n",
       "CG771531   32      CG7715    72          1460          115.05603 45.5284719\n",
       "Muc26B1    2       Muc26B    43           839           53.10378 22.0104848\n",
       "CG7715134  135     CG7715    32          1803           90.12055 50.1681292\n",
       "           log2.ratio heuristic.score\n",
       "Muc26B159  10.9731815 146.382615     \n",
       "Muc11A162  10.9511940 139.686023     \n",
       "Alp8162    11.3006488 138.610192     \n",
       "CG438961    9.4345161 107.719304     \n",
       "CG77151     7.3671719  84.785187     \n",
       "CG7715159   5.4666865  62.780949     \n",
       "CG7715170   2.5000270  20.710634     \n",
       "CG43896170  1.6266566  11.027603     \n",
       "CG7715168   1.3035475   9.050271     \n",
       "CG771531    1.3061508   8.958468     \n",
       "Muc26B1     1.2065231   6.946747     \n",
       "CG7715134   0.8166106   5.315894     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cluster.markers  %>% filter(gene %in% (c(\"Muc26B\", \"Muc11A\", \"Alp8\", \"CG43896\", \"CG7715\" )))  %>%  head(12)\n",
    "cluster.markers.sub300  %>% filter(gene %in% (c(\"Muc26B\", \"Muc11A\", \"Alp8\", \"CG43896\", \"CG7715\" )))  %>%  head(12)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1246,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finding markers for cluster 1\n",
      "\n",
      "# of cells sampled in set 1: 2463\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 2\n",
      "\n",
      "# of cells sampled in set 1: 2227\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 3\n",
      "\n",
      "# of cells sampled in set 1: 1396\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 4\n",
      "\n",
      "# of cells sampled in set 1: 2093\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 5\n",
      "\n",
      "# of cells sampled in set 1: 2174\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 6\n",
      "\n",
      "# of cells sampled in set 1: 2123\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 7\n",
      "\n",
      "# of cells sampled in set 1: 2023\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 8\n",
      "\n",
      "# of cells sampled in set 1: 2101\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 9\n",
      "\n",
      "# of cells sampled in set 1: 2221\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 10\n",
      "\n",
      "# of cells sampled in set 1: 2100\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 11\n",
      "\n",
      "# of cells sampled in set 1: 1633\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 12\n",
      "\n",
      "# of cells sampled in set 1: 1977\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 13\n",
      "\n",
      "# of cells sampled in set 1: 2016\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 14\n",
      "\n",
      "# of cells sampled in set 1: 1992\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 15\n",
      "\n",
      "# of cells sampled in set 1: 1937\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 16\n",
      "\n",
      "# of cells sampled in set 1: 1833\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 17\n",
      "\n",
      "# of cells sampled in set 1: 1890\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 18\n",
      "\n",
      "# of cells sampled in set 1: 1958\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 19\n",
      "\n",
      "# of cells sampled in set 1: 1624\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 20\n",
      "\n",
      "# of cells sampled in set 1: 1954\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 21\n",
      "\n",
      "# of cells sampled in set 1: 1808\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 22\n",
      "\n",
      "# of cells sampled in set 1: 1586\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 23\n",
      "\n",
      "# of cells sampled in set 1: 1759\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 24\n",
      "\n",
      "# of cells sampled in set 1: 1700\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 25\n",
      "\n",
      "# of cells sampled in set 1: 1790\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 26\n",
      "\n",
      "# of cells sampled in set 1: 1683\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 27\n",
      "\n",
      "# of cells sampled in set 1: 1463\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 28\n",
      "\n",
      "# of cells sampled in set 1: 1778\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 29\n",
      "\n",
      "# of cells sampled in set 1: 1670\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 30\n",
      "\n",
      "# of cells sampled in set 1: 1614\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 31\n",
      "\n",
      "# of cells sampled in set 1: 1741\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 32\n",
      "\n",
      "# of cells sampled in set 1: 1695\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 33\n",
      "\n",
      "# of cells sampled in set 1: 1523\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 34\n",
      "\n",
      "# of cells sampled in set 1: 1763\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 35\n",
      "\n",
      "# of cells sampled in set 1: 1516\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 36\n",
      "\n",
      "# of cells sampled in set 1: 1610\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 37\n",
      "\n",
      "# of cells sampled in set 1: 1514\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 38\n",
      "\n",
      "# of cells sampled in set 1: 1572\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 39\n",
      "\n",
      "# of cells sampled in set 1: 1698\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 40\n",
      "\n",
      "# of cells sampled in set 1: 1642\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 41\n",
      "\n",
      "# of cells sampled in set 1: 1588\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 42\n",
      "\n",
      "# of cells sampled in set 1: 1470\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 43\n",
      "\n",
      "# of cells sampled in set 1: 1597\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 44\n",
      "\n",
      "# of cells sampled in set 1: 1556\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 45\n",
      "\n",
      "# of cells sampled in set 1: 1636\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 46\n",
      "\n",
      "# of cells sampled in set 1: 1585\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 47\n",
      "\n",
      "# of cells sampled in set 1: 1452\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 48\n",
      "\n",
      "# of cells sampled in set 1: 1414\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 49\n",
      "\n",
      "# of cells sampled in set 1: 1514\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 50\n",
      "\n",
      "# of cells sampled in set 1: 1384\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 51\n",
      "\n",
      "# of cells sampled in set 1: 1635\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 52\n",
      "\n",
      "# of cells sampled in set 1: 1470\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 53\n",
      "\n",
      "# of cells sampled in set 1: 1416\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 54\n",
      "\n",
      "# of cells sampled in set 1: 1481\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 55\n",
      "\n",
      "# of cells sampled in set 1: 1456\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 56\n",
      "\n",
      "# of cells sampled in set 1: 1584\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 57\n",
      "\n",
      "# of cells sampled in set 1: 1445\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 58\n",
      "\n",
      "# of cells sampled in set 1: 1586\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 59\n",
      "\n",
      "# of cells sampled in set 1: 1250\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 60\n",
      "\n",
      "# of cells sampled in set 1: 1532\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 61\n",
      "\n",
      "# of cells sampled in set 1: 1209\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 62\n",
      "\n",
      "# of cells sampled in set 1: 1236\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 63\n",
      "\n",
      "# of cells sampled in set 1: 1402\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 64\n",
      "\n",
      "# of cells sampled in set 1: 1446\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 65\n",
      "\n",
      "# of cells sampled in set 1: 1470\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 66\n",
      "\n",
      "# of cells sampled in set 1: 1417\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 67\n",
      "\n",
      "# of cells sampled in set 1: 1427\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 68\n",
      "\n",
      "# of cells sampled in set 1: 1386\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 69\n",
      "\n",
      "# of cells sampled in set 1: 1305\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 70\n",
      "\n",
      "# of cells sampled in set 1: 1420\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 71\n",
      "\n",
      "# of cells sampled in set 1: 1252\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 72\n",
      "\n",
      "# of cells sampled in set 1: 1339\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 73\n",
      "\n",
      "# of cells sampled in set 1: 1442\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 74\n",
      "\n",
      "# of cells sampled in set 1: 1211\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 75\n",
      "\n",
      "# of cells sampled in set 1: 1288\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 76\n",
      "\n",
      "# of cells sampled in set 1: 891\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 77\n",
      "\n",
      "# of cells sampled in set 1: 1333\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 78\n",
      "\n",
      "# of cells sampled in set 1: 1200\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 79\n",
      "\n",
      "# of cells sampled in set 1: 1242\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 80\n",
      "\n",
      "# of cells sampled in set 1: 1346\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 81\n",
      "\n",
      "# of cells sampled in set 1: 1354\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 82\n",
      "\n",
      "# of cells sampled in set 1: 1315\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 83\n",
      "\n",
      "# of cells sampled in set 1: 1224\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 84\n",
      "\n",
      "# of cells sampled in set 1: 1246\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 85\n",
      "\n",
      "# of cells sampled in set 1: 1266\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 86\n",
      "\n",
      "# of cells sampled in set 1: 1272\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 87\n",
      "\n",
      "# of cells sampled in set 1: 1162\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 88\n",
      "\n",
      "# of cells sampled in set 1: 1237\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 89\n",
      "\n",
      "# of cells sampled in set 1: 1222\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 90\n",
      "\n",
      "# of cells sampled in set 1: 1152\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 91\n",
      "\n",
      "# of cells sampled in set 1: 1221\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 92\n",
      "\n",
      "# of cells sampled in set 1: 1244\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 93\n",
      "\n",
      "# of cells sampled in set 1: 1147\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 94\n",
      "\n",
      "# of cells sampled in set 1: 1238\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 95\n",
      "\n",
      "# of cells sampled in set 1: 1327\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 96\n",
      "\n",
      "# of cells sampled in set 1: 1257\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 97\n",
      "\n",
      "# of cells sampled in set 1: 1227\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 98\n",
      "\n",
      "# of cells sampled in set 1: 1321\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 99\n",
      "\n",
      "# of cells sampled in set 1: 1227\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 100\n",
      "\n",
      "# of cells sampled in set 1: 1025\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 101\n",
      "\n",
      "# of cells sampled in set 1: 1041\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 102\n",
      "\n",
      "# of cells sampled in set 1: 1022\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 103\n",
      "\n",
      "# of cells sampled in set 1: 1124\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 104\n",
      "\n",
      "# of cells sampled in set 1: 1205\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 105\n",
      "\n",
      "# of cells sampled in set 1: 1056\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 106\n",
      "\n",
      "# of cells sampled in set 1: 1127\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 107\n",
      "\n",
      "# of cells sampled in set 1: 1064\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 108\n",
      "\n",
      "# of cells sampled in set 1: 1128\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 109\n",
      "\n",
      "# of cells sampled in set 1: 1130\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 110\n",
      "\n",
      "# of cells sampled in set 1: 1120\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 111\n",
      "\n",
      "# of cells sampled in set 1: 1140\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 112\n",
      "\n",
      "# of cells sampled in set 1: 983\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 113\n",
      "\n",
      "# of cells sampled in set 1: 1123\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 114\n",
      "\n",
      "# of cells sampled in set 1: 1062\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 115\n",
      "\n",
      "# of cells sampled in set 1: 1147\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 116\n",
      "\n",
      "# of cells sampled in set 1: 981\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 117\n",
      "\n",
      "# of cells sampled in set 1: 1029\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 118\n",
      "\n",
      "# of cells sampled in set 1: 1054\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 119\n",
      "\n",
      "# of cells sampled in set 1: 940\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 120\n",
      "\n",
      "# of cells sampled in set 1: 1054\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 121\n",
      "\n",
      "# of cells sampled in set 1: 1053\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 122\n",
      "\n",
      "# of cells sampled in set 1: 1113\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 123\n",
      "\n",
      "# of cells sampled in set 1: 1066\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 124\n",
      "\n",
      "# of cells sampled in set 1: 1038\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 125\n",
      "\n",
      "# of cells sampled in set 1: 1013\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 126\n",
      "\n",
      "# of cells sampled in set 1: 1035\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 127\n",
      "\n",
      "# of cells sampled in set 1: 929\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 128\n",
      "\n",
      "# of cells sampled in set 1: 970\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 129\n",
      "\n",
      "# of cells sampled in set 1: 1009\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 130\n",
      "\n",
      "# of cells sampled in set 1: 912\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 131\n",
      "\n",
      "# of cells sampled in set 1: 945\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 132\n",
      "\n",
      "# of cells sampled in set 1: 917\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 133\n",
      "\n",
      "# of cells sampled in set 1: 878\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 134\n",
      "\n",
      "# of cells sampled in set 1: 965\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 135\n",
      "\n",
      "# of cells sampled in set 1: 899\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 136\n",
      "\n",
      "# of cells sampled in set 1: 818\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 137\n",
      "\n",
      "# of cells sampled in set 1: 944\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 138\n",
      "\n",
      "# of cells sampled in set 1: 898\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 139\n",
      "\n",
      "# of cells sampled in set 1: 941\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 140\n",
      "\n",
      "# of cells sampled in set 1: 895\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 141\n",
      "\n",
      "# of cells sampled in set 1: 898\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 142\n",
      "\n",
      "# of cells sampled in set 1: 893\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 143\n",
      "\n",
      "# of cells sampled in set 1: 774\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 144\n",
      "\n",
      "# of cells sampled in set 1: 768\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 145\n",
      "\n",
      "# of cells sampled in set 1: 833\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 146\n",
      "\n",
      "# of cells sampled in set 1: 750\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 147\n",
      "\n",
      "# of cells sampled in set 1: 756\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 148\n",
      "\n",
      "# of cells sampled in set 1: 700\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 149\n",
      "\n",
      "# of cells sampled in set 1: 662\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 150\n",
      "\n",
      "# of cells sampled in set 1: 621\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 151\n",
      "\n",
      "# of cells sampled in set 1: 622\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 152\n",
      "\n",
      "# of cells sampled in set 1: 590\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 153\n",
      "\n",
      "# of cells sampled in set 1: 541\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 154\n",
      "\n",
      "# of cells sampled in set 1: 604\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 155\n",
      "\n",
      "# of cells sampled in set 1: 527\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 156\n",
      "\n",
      "# of cells sampled in set 1: 567\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 157\n",
      "\n",
      "# of cells sampled in set 1: 548\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 158\n",
      "\n",
      "# of cells sampled in set 1: 555\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 159\n",
      "\n",
      "# of cells sampled in set 1: 538\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finding markers for cluster 160\n",
      "\n",
      "# of cells sampled in set 1: 438\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 161\n",
      "\n",
      "# of cells sampled in set 1: 488\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 162\n",
      "\n",
      "# of cells sampled in set 1: 410\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 163\n",
      "\n",
      "# of cells sampled in set 1: 435\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 164\n",
      "\n",
      "# of cells sampled in set 1: 376\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 165\n",
      "\n",
      "# of cells sampled in set 1: 387\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 166\n",
      "\n",
      "# of cells sampled in set 1: 328\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 167\n",
      "\n",
      "# of cells sampled in set 1: 355\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 168\n",
      "\n",
      "# of cells sampled in set 1: 198\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 169\n",
      "\n",
      "# of cells sampled in set 1: 177\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 170\n",
      "\n",
      "# of cells sampled in set 1: 136\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 171\n",
      "\n",
      "# of cells sampled in set 1: 134\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 172\n",
      "\n",
      "# of cells sampled in set 1: 107\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 173\n",
      "\n",
      "# of cells sampled in set 1: 103\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 174\n",
      "\n",
      "# of cells sampled in set 1: 54\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    }
   ],
   "source": [
    "DEG.results.sub300_no_singletons = lapply(sort((unique(df_sub_no_singletons$cluster))),\n",
    "                     function(this.cluster) {\n",
    "    message(\"Finding markers for cluster \", this.cluster)\n",
    "    cbind(\n",
    "        two.set.differential.gene.test.3a(\n",
    "            cds_sub_no_sing,\n",
    "            df_sub_no_singletons$cluster == this.cluster,\n",
    "            df_sub_no_singletons$cluster != this.cluster,\n",
    "            sample.n.max = 50000),\n",
    "        data.frame(cluster = this.cluster))\n",
    "                         })"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1249,
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster.markers.sub300_no_singletons = do.call(rbind, DEG.results.sub300_no_singletons) %>% filter(higher.expr == \"Set 1\") %>%\n",
    "    select(\n",
    "        cluster = cluster, gene, \n",
    "        cluster.n.umi = set.1.n.umi,\n",
    "        other.n.umi = set.2.n.umi,\n",
    "        cluster.tpm = set.1.tpm,\n",
    "        other.tpm = set.2.tpm,\n",
    "        log2.ratio, heuristic.score) %>%\n",
    "    arrange(-heuristic.score)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1250,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cluster</th><th scope=col>gene</th><th scope=col>cluster.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cluster.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Muc26B159</th><td>160</td><td>Muc26B </td><td>3048</td><td> 85</td><td>10265.684</td><td> 2.9018029</td><td>11.361401</td><td>151.3984</td></tr>\n",
       "\t<tr><th scope=row>Muc11A162</th><td>163</td><td>Muc11A </td><td>4202</td><td> 57</td><td> 7240.091</td><td> 1.2972972</td><td>11.621855</td><td>149.0153</td></tr>\n",
       "\t<tr><th scope=row>CG7017159</th><td>160</td><td>CG7017 </td><td>3024</td><td>200</td><td>12257.985</td><td> 7.1512453</td><td>10.554414</td><td>143.3453</td></tr>\n",
       "\t<tr><th scope=row>CG7409171</th><td>172</td><td>CG7409 </td><td> 526</td><td> 14</td><td> 4108.171</td><td> 0.2547309</td><td>11.676902</td><td>140.1769</td></tr>\n",
       "\t<tr><th scope=row>CG11370167</th><td>168</td><td>CG11370</td><td>3506</td><td>434</td><td>15867.110</td><td>17.1741523</td><td> 9.769936</td><td>136.3281</td></tr>\n",
       "\t<tr><th scope=row>Mur18B162</th><td>163</td><td>Mur18B </td><td>2928</td><td> 61</td><td> 5529.166</td><td> 1.8673438</td><td>10.913131</td><td>135.6841</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cluster & gene & cluster.n.umi & other.n.umi & cluster.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <fct> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tMuc26B159 & 160 & Muc26B  & 3048 &  85 & 10265.684 &  2.9018029 & 11.361401 & 151.3984\\\\\n",
       "\tMuc11A162 & 163 & Muc11A  & 4202 &  57 &  7240.091 &  1.2972972 & 11.621855 & 149.0153\\\\\n",
       "\tCG7017159 & 160 & CG7017  & 3024 & 200 & 12257.985 &  7.1512453 & 10.554414 & 143.3453\\\\\n",
       "\tCG7409171 & 172 & CG7409  &  526 &  14 &  4108.171 &  0.2547309 & 11.676902 & 140.1769\\\\\n",
       "\tCG11370167 & 168 & CG11370 & 3506 & 434 & 15867.110 & 17.1741523 &  9.769936 & 136.3281\\\\\n",
       "\tMur18B162 & 163 & Mur18B  & 2928 &  61 &  5529.166 &  1.8673438 & 10.913131 & 135.6841\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 8\n",
       "\n",
       "| <!--/--> | cluster &lt;fct&gt; | gene &lt;chr&gt; | cluster.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cluster.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Muc26B159 | 160 | Muc26B  | 3048 |  85 | 10265.684 |  2.9018029 | 11.361401 | 151.3984 |\n",
       "| Muc11A162 | 163 | Muc11A  | 4202 |  57 |  7240.091 |  1.2972972 | 11.621855 | 149.0153 |\n",
       "| CG7017159 | 160 | CG7017  | 3024 | 200 | 12257.985 |  7.1512453 | 10.554414 | 143.3453 |\n",
       "| CG7409171 | 172 | CG7409  |  526 |  14 |  4108.171 |  0.2547309 | 11.676902 | 140.1769 |\n",
       "| CG11370167 | 168 | CG11370 | 3506 | 434 | 15867.110 | 17.1741523 |  9.769936 | 136.3281 |\n",
       "| Mur18B162 | 163 | Mur18B  | 2928 |  61 |  5529.166 |  1.8673438 | 10.913131 | 135.6841 |\n",
       "\n"
      ],
      "text/plain": [
       "           cluster gene    cluster.n.umi other.n.umi cluster.tpm other.tpm \n",
       "Muc26B159  160     Muc26B  3048           85         10265.684    2.9018029\n",
       "Muc11A162  163     Muc11A  4202           57          7240.091    1.2972972\n",
       "CG7017159  160     CG7017  3024          200         12257.985    7.1512453\n",
       "CG7409171  172     CG7409   526           14          4108.171    0.2547309\n",
       "CG11370167 168     CG11370 3506          434         15867.110   17.1741523\n",
       "Mur18B162  163     Mur18B  2928           61          5529.166    1.8673438\n",
       "           log2.ratio heuristic.score\n",
       "Muc26B159  11.361401  151.3984       \n",
       "Muc11A162  11.621855  149.0153       \n",
       "CG7017159  10.554414  143.3453       \n",
       "CG7409171  11.676902  140.1769       \n",
       "CG11370167  9.769936  136.3281       \n",
       "Mur18B162  10.913131  135.6841       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cluster.markers.sub300_no_singletons  %>%  head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1284,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 12 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cluster</th><th scope=col>gene</th><th scope=col>cluster.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cluster.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Muc26B159</th><td>160</td><td>Muc26B </td><td>3993</td><td> 111</td><td>10368.32230</td><td> 4.1576480</td><td>10.9731815</td><td>146.382615</td></tr>\n",
       "\t<tr><th scope=row>Muc11A162</th><td>163</td><td>Muc11A </td><td>4284</td><td>  73</td><td> 6913.08960</td><td> 2.4916789</td><td>10.9511940</td><td>139.686023</td></tr>\n",
       "\t<tr><th scope=row>Alp8162</th><td>163</td><td>Alp8   </td><td>4089</td><td>  31</td><td> 4923.23702</td><td> 0.9517154</td><td>11.3006488</td><td>138.610192</td></tr>\n",
       "\t<tr><th scope=row>CG438961</th><td>2  </td><td>CG43896</td><td>6210</td><td>  83</td><td> 2734.47541</td><td> 2.9518611</td><td> 9.4345161</td><td>107.719304</td></tr>\n",
       "\t<tr><th scope=row>CG77151</th><td>2  </td><td>CG7715 </td><td>6105</td><td> 486</td><td> 2912.44871</td><td>16.6408119</td><td> 7.3671719</td><td> 84.785187</td></tr>\n",
       "\t<tr><th scope=row>CG7715159</th><td>160</td><td>CG7715 </td><td>1011</td><td>2690</td><td> 2863.91837</td><td>63.7625614</td><td> 5.4666865</td><td> 62.780949</td></tr>\n",
       "\t<tr><th scope=row>CG7715170</th><td>171</td><td>CG7715 </td><td>  12</td><td>2034</td><td>  310.73237</td><td>53.9292147</td><td> 2.5000270</td><td> 20.710634</td></tr>\n",
       "\t<tr><th scope=row>CG43896170</th><td>171</td><td>CG43896</td><td>   4</td><td>1348</td><td>  108.84354</td><td>34.2476549</td><td> 1.6266566</td><td> 11.027603</td></tr>\n",
       "\t<tr><th scope=row>CG7715168</th><td>169</td><td>CG7715 </td><td>  13</td><td>1553</td><td>  122.02448</td><td>48.4356305</td><td> 1.3035475</td><td>  9.050271</td></tr>\n",
       "\t<tr><th scope=row>CG771531</th><td>32 </td><td>CG7715 </td><td>  72</td><td>1460</td><td>  115.05603</td><td>45.5284719</td><td> 1.3061508</td><td>  8.958468</td></tr>\n",
       "\t<tr><th scope=row>Muc26B1</th><td>2  </td><td>Muc26B </td><td>  43</td><td> 839</td><td>   53.10378</td><td>22.0104848</td><td> 1.2065231</td><td>  6.946747</td></tr>\n",
       "\t<tr><th scope=row>CG7715134</th><td>135</td><td>CG7715 </td><td>  32</td><td>1803</td><td>   90.12055</td><td>50.1681292</td><td> 0.8166106</td><td>  5.315894</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 12 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cluster & gene & cluster.n.umi & other.n.umi & cluster.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <fct> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tMuc26B159 & 160 & Muc26B  & 3993 &  111 & 10368.32230 &  4.1576480 & 10.9731815 & 146.382615\\\\\n",
       "\tMuc11A162 & 163 & Muc11A  & 4284 &   73 &  6913.08960 &  2.4916789 & 10.9511940 & 139.686023\\\\\n",
       "\tAlp8162 & 163 & Alp8    & 4089 &   31 &  4923.23702 &  0.9517154 & 11.3006488 & 138.610192\\\\\n",
       "\tCG438961 & 2   & CG43896 & 6210 &   83 &  2734.47541 &  2.9518611 &  9.4345161 & 107.719304\\\\\n",
       "\tCG77151 & 2   & CG7715  & 6105 &  486 &  2912.44871 & 16.6408119 &  7.3671719 &  84.785187\\\\\n",
       "\tCG7715159 & 160 & CG7715  & 1011 & 2690 &  2863.91837 & 63.7625614 &  5.4666865 &  62.780949\\\\\n",
       "\tCG7715170 & 171 & CG7715  &   12 & 2034 &   310.73237 & 53.9292147 &  2.5000270 &  20.710634\\\\\n",
       "\tCG43896170 & 171 & CG43896 &    4 & 1348 &   108.84354 & 34.2476549 &  1.6266566 &  11.027603\\\\\n",
       "\tCG7715168 & 169 & CG7715  &   13 & 1553 &   122.02448 & 48.4356305 &  1.3035475 &   9.050271\\\\\n",
       "\tCG771531 & 32  & CG7715  &   72 & 1460 &   115.05603 & 45.5284719 &  1.3061508 &   8.958468\\\\\n",
       "\tMuc26B1 & 2   & Muc26B  &   43 &  839 &    53.10378 & 22.0104848 &  1.2065231 &   6.946747\\\\\n",
       "\tCG7715134 & 135 & CG7715  &   32 & 1803 &    90.12055 & 50.1681292 &  0.8166106 &   5.315894\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 12 × 8\n",
       "\n",
       "| <!--/--> | cluster &lt;fct&gt; | gene &lt;chr&gt; | cluster.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cluster.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Muc26B159 | 160 | Muc26B  | 3993 |  111 | 10368.32230 |  4.1576480 | 10.9731815 | 146.382615 |\n",
       "| Muc11A162 | 163 | Muc11A  | 4284 |   73 |  6913.08960 |  2.4916789 | 10.9511940 | 139.686023 |\n",
       "| Alp8162 | 163 | Alp8    | 4089 |   31 |  4923.23702 |  0.9517154 | 11.3006488 | 138.610192 |\n",
       "| CG438961 | 2   | CG43896 | 6210 |   83 |  2734.47541 |  2.9518611 |  9.4345161 | 107.719304 |\n",
       "| CG77151 | 2   | CG7715  | 6105 |  486 |  2912.44871 | 16.6408119 |  7.3671719 |  84.785187 |\n",
       "| CG7715159 | 160 | CG7715  | 1011 | 2690 |  2863.91837 | 63.7625614 |  5.4666865 |  62.780949 |\n",
       "| CG7715170 | 171 | CG7715  |   12 | 2034 |   310.73237 | 53.9292147 |  2.5000270 |  20.710634 |\n",
       "| CG43896170 | 171 | CG43896 |    4 | 1348 |   108.84354 | 34.2476549 |  1.6266566 |  11.027603 |\n",
       "| CG7715168 | 169 | CG7715  |   13 | 1553 |   122.02448 | 48.4356305 |  1.3035475 |   9.050271 |\n",
       "| CG771531 | 32  | CG7715  |   72 | 1460 |   115.05603 | 45.5284719 |  1.3061508 |   8.958468 |\n",
       "| Muc26B1 | 2   | Muc26B  |   43 |  839 |    53.10378 | 22.0104848 |  1.2065231 |   6.946747 |\n",
       "| CG7715134 | 135 | CG7715  |   32 | 1803 |    90.12055 | 50.1681292 |  0.8166106 |   5.315894 |\n",
       "\n"
      ],
      "text/plain": [
       "           cluster gene    cluster.n.umi other.n.umi cluster.tpm other.tpm \n",
       "Muc26B159  160     Muc26B  3993           111        10368.32230  4.1576480\n",
       "Muc11A162  163     Muc11A  4284            73         6913.08960  2.4916789\n",
       "Alp8162    163     Alp8    4089            31         4923.23702  0.9517154\n",
       "CG438961   2       CG43896 6210            83         2734.47541  2.9518611\n",
       "CG77151    2       CG7715  6105           486         2912.44871 16.6408119\n",
       "CG7715159  160     CG7715  1011          2690         2863.91837 63.7625614\n",
       "CG7715170  171     CG7715    12          2034          310.73237 53.9292147\n",
       "CG43896170 171     CG43896    4          1348          108.84354 34.2476549\n",
       "CG7715168  169     CG7715    13          1553          122.02448 48.4356305\n",
       "CG771531   32      CG7715    72          1460          115.05603 45.5284719\n",
       "Muc26B1    2       Muc26B    43           839           53.10378 22.0104848\n",
       "CG7715134  135     CG7715    32          1803           90.12055 50.1681292\n",
       "           log2.ratio heuristic.score\n",
       "Muc26B159  10.9731815 146.382615     \n",
       "Muc11A162  10.9511940 139.686023     \n",
       "Alp8162    11.3006488 138.610192     \n",
       "CG438961    9.4345161 107.719304     \n",
       "CG77151     7.3671719  84.785187     \n",
       "CG7715159   5.4666865  62.780949     \n",
       "CG7715170   2.5000270  20.710634     \n",
       "CG43896170  1.6266566  11.027603     \n",
       "CG7715168   1.3035475   9.050271     \n",
       "CG771531    1.3061508   8.958468     \n",
       "Muc26B1     1.2065231   6.946747     \n",
       "CG7715134   0.8166106   5.315894     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 12 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cluster</th><th scope=col>gene</th><th scope=col>cluster.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cluster.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Muc26B159</th><td>160</td><td>Muc26B </td><td>3048</td><td>  85</td><td>10265.68401</td><td> 2.901803</td><td>11.361401</td><td>151.398428</td></tr>\n",
       "\t<tr><th scope=row>Muc11A162</th><td>163</td><td>Muc11A </td><td>4202</td><td>  57</td><td> 7240.09127</td><td> 1.297297</td><td>11.621855</td><td>149.015321</td></tr>\n",
       "\t<tr><th scope=row>Alp8162</th><td>163</td><td>Alp8   </td><td>3913</td><td>  44</td><td> 5133.35092</td><td> 1.623859</td><td>10.933995</td><td>134.772054</td></tr>\n",
       "\t<tr><th scope=row>CG438961</th><td>2  </td><td>CG43896</td><td>6037</td><td> 112</td><td> 2810.31908</td><td> 4.218980</td><td> 9.072750</td><td>103.946788</td></tr>\n",
       "\t<tr><th scope=row>CG77151</th><td>2  </td><td>CG7715 </td><td>5938</td><td> 708</td><td> 2979.30561</td><td>23.065451</td><td> 6.951869</td><td> 80.233219</td></tr>\n",
       "\t<tr><th scope=row>CG7715159</th><td>160</td><td>CG7715 </td><td> 779</td><td>2504</td><td> 2983.00308</td><td>71.100741</td><td> 5.370608</td><td> 61.993102</td></tr>\n",
       "\t<tr><th scope=row>CG7715168</th><td>169</td><td>CG7715 </td><td>  13</td><td>1216</td><td>  126.85031</td><td>41.764080</td><td> 1.568656</td><td> 10.977941</td></tr>\n",
       "\t<tr><th scope=row>CG43896168</th><td>169</td><td>CG43896</td><td>   7</td><td> 646</td><td>   68.30401</td><td>23.090034</td><td> 1.503534</td><td>  9.193909</td></tr>\n",
       "\t<tr><th scope=row>CG771531</th><td>32 </td><td>CG7715 </td><td>  66</td><td>1667</td><td>  118.95351</td><td>48.529265</td><td> 1.264045</td><td>  8.729912</td></tr>\n",
       "\t<tr><th scope=row>CG7715134</th><td>135</td><td>CG7715 </td><td>  30</td><td>2122</td><td>   94.14070</td><td>38.625683</td><td> 1.248383</td><td>  8.204360</td></tr>\n",
       "\t<tr><th scope=row>Alp8163</th><td>164</td><td>Alp8   </td><td>  12</td><td> 609</td><td>   25.18934</td><td> 7.084000</td><td> 1.639672</td><td>  7.724344</td></tr>\n",
       "\t<tr><th scope=row>CG7715165</th><td>166</td><td>CG7715 </td><td>  11</td><td>1351</td><td>   91.58318</td><td>42.170141</td><td> 1.085049</td><td>  7.088275</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 12 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cluster & gene & cluster.n.umi & other.n.umi & cluster.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <fct> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tMuc26B159 & 160 & Muc26B  & 3048 &   85 & 10265.68401 &  2.901803 & 11.361401 & 151.398428\\\\\n",
       "\tMuc11A162 & 163 & Muc11A  & 4202 &   57 &  7240.09127 &  1.297297 & 11.621855 & 149.015321\\\\\n",
       "\tAlp8162 & 163 & Alp8    & 3913 &   44 &  5133.35092 &  1.623859 & 10.933995 & 134.772054\\\\\n",
       "\tCG438961 & 2   & CG43896 & 6037 &  112 &  2810.31908 &  4.218980 &  9.072750 & 103.946788\\\\\n",
       "\tCG77151 & 2   & CG7715  & 5938 &  708 &  2979.30561 & 23.065451 &  6.951869 &  80.233219\\\\\n",
       "\tCG7715159 & 160 & CG7715  &  779 & 2504 &  2983.00308 & 71.100741 &  5.370608 &  61.993102\\\\\n",
       "\tCG7715168 & 169 & CG7715  &   13 & 1216 &   126.85031 & 41.764080 &  1.568656 &  10.977941\\\\\n",
       "\tCG43896168 & 169 & CG43896 &    7 &  646 &    68.30401 & 23.090034 &  1.503534 &   9.193909\\\\\n",
       "\tCG771531 & 32  & CG7715  &   66 & 1667 &   118.95351 & 48.529265 &  1.264045 &   8.729912\\\\\n",
       "\tCG7715134 & 135 & CG7715  &   30 & 2122 &    94.14070 & 38.625683 &  1.248383 &   8.204360\\\\\n",
       "\tAlp8163 & 164 & Alp8    &   12 &  609 &    25.18934 &  7.084000 &  1.639672 &   7.724344\\\\\n",
       "\tCG7715165 & 166 & CG7715  &   11 & 1351 &    91.58318 & 42.170141 &  1.085049 &   7.088275\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 12 × 8\n",
       "\n",
       "| <!--/--> | cluster &lt;fct&gt; | gene &lt;chr&gt; | cluster.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cluster.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Muc26B159 | 160 | Muc26B  | 3048 |   85 | 10265.68401 |  2.901803 | 11.361401 | 151.398428 |\n",
       "| Muc11A162 | 163 | Muc11A  | 4202 |   57 |  7240.09127 |  1.297297 | 11.621855 | 149.015321 |\n",
       "| Alp8162 | 163 | Alp8    | 3913 |   44 |  5133.35092 |  1.623859 | 10.933995 | 134.772054 |\n",
       "| CG438961 | 2   | CG43896 | 6037 |  112 |  2810.31908 |  4.218980 |  9.072750 | 103.946788 |\n",
       "| CG77151 | 2   | CG7715  | 5938 |  708 |  2979.30561 | 23.065451 |  6.951869 |  80.233219 |\n",
       "| CG7715159 | 160 | CG7715  |  779 | 2504 |  2983.00308 | 71.100741 |  5.370608 |  61.993102 |\n",
       "| CG7715168 | 169 | CG7715  |   13 | 1216 |   126.85031 | 41.764080 |  1.568656 |  10.977941 |\n",
       "| CG43896168 | 169 | CG43896 |    7 |  646 |    68.30401 | 23.090034 |  1.503534 |   9.193909 |\n",
       "| CG771531 | 32  | CG7715  |   66 | 1667 |   118.95351 | 48.529265 |  1.264045 |   8.729912 |\n",
       "| CG7715134 | 135 | CG7715  |   30 | 2122 |    94.14070 | 38.625683 |  1.248383 |   8.204360 |\n",
       "| Alp8163 | 164 | Alp8    |   12 |  609 |    25.18934 |  7.084000 |  1.639672 |   7.724344 |\n",
       "| CG7715165 | 166 | CG7715  |   11 | 1351 |    91.58318 | 42.170141 |  1.085049 |   7.088275 |\n",
       "\n"
      ],
      "text/plain": [
       "           cluster gene    cluster.n.umi other.n.umi cluster.tpm other.tpm\n",
       "Muc26B159  160     Muc26B  3048            85        10265.68401  2.901803\n",
       "Muc11A162  163     Muc11A  4202            57         7240.09127  1.297297\n",
       "Alp8162    163     Alp8    3913            44         5133.35092  1.623859\n",
       "CG438961   2       CG43896 6037           112         2810.31908  4.218980\n",
       "CG77151    2       CG7715  5938           708         2979.30561 23.065451\n",
       "CG7715159  160     CG7715   779          2504         2983.00308 71.100741\n",
       "CG7715168  169     CG7715    13          1216          126.85031 41.764080\n",
       "CG43896168 169     CG43896    7           646           68.30401 23.090034\n",
       "CG771531   32      CG7715    66          1667          118.95351 48.529265\n",
       "CG7715134  135     CG7715    30          2122           94.14070 38.625683\n",
       "Alp8163    164     Alp8      12           609           25.18934  7.084000\n",
       "CG7715165  166     CG7715    11          1351           91.58318 42.170141\n",
       "           log2.ratio heuristic.score\n",
       "Muc26B159  11.361401  151.398428     \n",
       "Muc11A162  11.621855  149.015321     \n",
       "Alp8162    10.933995  134.772054     \n",
       "CG438961    9.072750  103.946788     \n",
       "CG77151     6.951869   80.233219     \n",
       "CG7715159   5.370608   61.993102     \n",
       "CG7715168   1.568656   10.977941     \n",
       "CG43896168  1.503534    9.193909     \n",
       "CG771531    1.264045    8.729912     \n",
       "CG7715134   1.248383    8.204360     \n",
       "Alp8163     1.639672    7.724344     \n",
       "CG7715165   1.085049    7.088275     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cluster.markers.sub300  %>% filter(gene %in% (c(\"Muc26B\", \"Muc11A\", \"Alp8\", \"CG43896\", \"CG7715\" )))  %>%  head(12)\n",
    "cluster.markers.sub300_no_singletons  %>% filter(gene %in% (c(\"Muc26B\", \"Muc11A\", \"Alp8\", \"CG43896\", \"CG7715\" )))  %>%  head(12)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### compare prior cell type annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1261,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finding markers for cluster amnioserosa\n",
      "\n",
      "# of cells sampled in set 1: 7610\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster anlage in statu nascendi\n",
      "\n",
      "# of cells sampled in set 1: 40\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster brain\n",
      "\n",
      "# of cells sampled in set 1: 26617\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster brain primordium\n",
      "\n",
      "# of cells sampled in set 1: 9816\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster cellular blastoderm / pole cell\n",
      "\n",
      "# of cells sampled in set 1: 16\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster crystal cell\n",
      "\n",
      "# of cells sampled in set 1: 177\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster ectoderm anlage\n",
      "\n",
      "# of cells sampled in set 1: 5\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster ectoderm anlage in statu nascendi\n",
      "\n",
      "# of cells sampled in set 1: 51\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster ectoderm primordium\n",
      "\n",
      "# of cells sampled in set 1: 1601\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster endoderm primordium\n",
      "\n",
      "# of cells sampled in set 1: 400\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster epidermis\n",
      "\n",
      "# of cells sampled in set 1: 36935\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster epidermis primordium\n",
      "\n",
      "# of cells sampled in set 1: 6976\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster fat body\n",
      "\n",
      "# of cells sampled in set 1: 5550\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster foregut / hindgut / anal pad\n",
      "\n",
      "# of cells sampled in set 1: 7035\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster foregut / hindgut primordium\n",
      "\n",
      "# of cells sampled in set 1: 1218\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster germ cell\n",
      "\n",
      "# of cells sampled in set 1: 1309\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster glia\n",
      "\n",
      "# of cells sampled in set 1: 899\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster hindgut / anal pad\n",
      "\n",
      "# of cells sampled in set 1: 2662\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster hindgut primordium\n",
      "\n",
      "# of cells sampled in set 1: 1916\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster Malpighian tubule\n",
      "\n",
      "# of cells sampled in set 1: 1090\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster maternal\n",
      "\n",
      "# of cells sampled in set 1: 17\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster mesoderm primordium\n",
      "\n",
      "# of cells sampled in set 1: 384\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster midgut\n",
      "\n",
      "# of cells sampled in set 1: 11649\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster midgut primordium\n",
      "\n",
      "# of cells sampled in set 1: 1848\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster midline primordium\n",
      "\n",
      "# of cells sampled in set 1: 982\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster muscle\n",
      "\n",
      "# of cells sampled in set 1: 6184\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster peripheral nervous system\n",
      "\n",
      "# of cells sampled in set 1: 3591\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster peripheral nervous system primordium\n",
      "\n",
      "# of cells sampled in set 1: 559\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster plasmatocytes\n",
      "\n",
      "# of cells sampled in set 1: 5575\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster plasmatocytes anlage\n",
      "\n",
      "# of cells sampled in set 1: 1658\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster procrystal cell\n",
      "\n",
      "# of cells sampled in set 1: 1556\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster proventriculus\n",
      "\n",
      "# of cells sampled in set 1: 2480\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster salivary gland\n",
      "\n",
      "# of cells sampled in set 1: 3487\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster salivary gland body primordium\n",
      "\n",
      "# of cells sampled in set 1: 1042\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster salivary gland primordium\n",
      "\n",
      "# of cells sampled in set 1: 115\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster sensory nervous system\n",
      "\n",
      "# of cells sampled in set 1: 4816\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster sensory nervous system primordium\n",
      "\n",
      "# of cells sampled in set 1: 1627\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster somatic muscle\n",
      "\n",
      "# of cells sampled in set 1: 14715\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster somatic muscle primordium\n",
      "\n",
      "# of cells sampled in set 1: 1674\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster spiracle (tracheal system)\n",
      "\n",
      "# of cells sampled in set 1: 986\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster spiracles (tracheal system)\n",
      "\n",
      "# of cells sampled in set 1: 751\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster tracheal primordium\n",
      "\n",
      "# of cells sampled in set 1: 2526\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster tracheal system\n",
      "\n",
      "# of cells sampled in set 1: 8052\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster trunk mesoderm primordium\n",
      "\n",
      "# of cells sampled in set 1: 880\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster ubiquitous\n",
      "\n",
      "# of cells sampled in set 1: 34589\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster unknown\n",
      "\n",
      "# of cells sampled in set 1: 1444\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster ventral nerve cord\n",
      "\n",
      "# of cells sampled in set 1: 7542\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster ventral nerve cord primordium\n",
      "\n",
      "# of cells sampled in set 1: 671\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster visceral muscle\n",
      "\n",
      "# of cells sampled in set 1: 799\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster visceral muscle primordium\n",
      "\n",
      "# of cells sampled in set 1: 3546\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster yolk nuclei\n",
      "\n",
      "# of cells sampled in set 1: 777\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    }
   ],
   "source": [
    "DEG.results.sub300.cell.type = lapply(sort((unique(df_sub300$annot))),\n",
    "                     function(this.cell.type) {\n",
    "    message(\"Finding markers for cluster \", this.cell.type)\n",
    "    cbind(\n",
    "        two.set.differential.gene.test.3a(\n",
    "            cds_sub,\n",
    "            df_sub300$annot == this.cell.type,\n",
    "            df_sub300$annot != this.cell.type,\n",
    "            sample.n.max = 50000),\n",
    "        data.frame(cell.type = this.cell.type))\n",
    "                         })"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1263,
   "metadata": {},
   "outputs": [],
   "source": [
    "cell.type.markers.sub300.cell.type = do.call(rbind, DEG.results.sub300.cell.type) %>% filter(higher.expr == \"Set 1\") %>%\n",
    "    select(\n",
    "        cell.type = cell.type, gene, \n",
    "        cell.type.n.umi = set.1.n.umi,\n",
    "        other.n.umi = set.2.n.umi,\n",
    "        cell.type.tpm = set.1.tpm,\n",
    "        other.tpm = set.2.tpm,\n",
    "        log2.ratio, heuristic.score) %>%\n",
    "    arrange(-heuristic.score)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1270,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type</th><th scope=col>gene</th><th scope=col>cell.type.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>pip32</th><td>salivary gland                </td><td>pip    </td><td>38398</td><td> 9911</td><td>29916.415</td><td>412.274111</td><td>6.177695</td><td>91.85427</td></tr>\n",
       "\t<tr><th scope=row>Ppn28</th><td>plasmatocytes                 </td><td>Ppn    </td><td>73383</td><td>14116</td><td>35715.779</td><td>536.739395</td><td>6.053511</td><td>91.55520</td></tr>\n",
       "\t<tr><th scope=row>CG43332</th><td>amnioserosa                   </td><td>CG43332</td><td>23490</td><td> 1608</td><td> 7785.421</td><td> 56.955284</td><td>7.069691</td><td>91.38809</td></tr>\n",
       "\t<tr><th scope=row>CG12011</th><td>amnioserosa                   </td><td>CG12011</td><td>82838</td><td>15148</td><td>29343.847</td><td>535.995432</td><td>5.772004</td><td>85.66128</td></tr>\n",
       "\t<tr><th scope=row>nur32</th><td>salivary gland                </td><td>nur    </td><td>26904</td><td>10233</td><td>22058.236</td><td>378.677940</td><td>5.860397</td><td>84.56023</td></tr>\n",
       "\t<tr><th scope=row>Muc30E</th><td>amnioserosa                   </td><td>Muc30E </td><td>15253</td><td> 1442</td><td> 4852.606</td><td> 45.768575</td><td>6.697076</td><td>82.00464</td></tr>\n",
       "\t<tr><th scope=row>CG1475632</th><td>salivary gland                </td><td>CG14756</td><td>15387</td><td> 5802</td><td>12488.709</td><td>206.762957</td><td>5.909542</td><td>80.41972</td></tr>\n",
       "\t<tr><th scope=row>CG1429219</th><td>Malpighian tubule             </td><td>CG14292</td><td> 1706</td><td>  461</td><td> 1335.417</td><td>  5.328381</td><td>7.721238</td><td>80.17853</td></tr>\n",
       "\t<tr><th scope=row>CG1135041</th><td>tracheal primordium           </td><td>CG11350</td><td> 4932</td><td>  540</td><td> 2300.934</td><td> 17.603434</td><td>6.950507</td><td>77.62765</td></tr>\n",
       "\t<tr><th scope=row>Mur18B19</th><td>Malpighian tubule             </td><td>Mur18B </td><td> 1689</td><td>  536</td><td> 1372.886</td><td>  7.201720</td><td>7.387070</td><td>77.00316</td></tr>\n",
       "\t<tr><th scope=row>pip33</th><td>salivary gland body primordium</td><td>pip    </td><td>12416</td><td>15041</td><td>27014.334</td><td>719.297781</td><td>5.228988</td><td>76.97850</td></tr>\n",
       "\t<tr><th scope=row>Pxn28</th><td>plasmatocytes                 </td><td>Pxn    </td><td>12668</td><td> 2411</td><td> 6173.778</td><td> 93.268241</td><td>6.033238</td><td>75.97157</td></tr>\n",
       "\t<tr><th scope=row>wrapper16</th><td>glia                          </td><td>wrapper</td><td> 1155</td><td> 1040</td><td> 3520.441</td><td> 42.361026</td><td>6.343213</td><td>74.73542</td></tr>\n",
       "\t<tr><th scope=row>Pde1c5</th><td>crystal cell                  </td><td>Pde1c  </td><td>  908</td><td> 5838</td><td>12521.122</td><td>278.241204</td><td>5.486708</td><td>74.68612</td></tr>\n",
       "\t<tr><th scope=row>Col4a128</th><td>plasmatocytes                 </td><td>Col4a1 </td><td>35872</td><td>11438</td><td>17492.963</td><td>449.982027</td><td>5.277561</td><td>74.38495</td></tr>\n",
       "\t<tr><th scope=row>Cad96Cb40</th><td>spiracles (tracheal system)   </td><td>Cad96Cb</td><td>  794</td><td>  719</td><td> 2900.128</td><td> 32.672669</td><td>6.428394</td><td>73.94195</td></tr>\n",
       "\t<tr><th scope=row>CG18088</th><td>amnioserosa                   </td><td>CG18088</td><td> 7126</td><td>  704</td><td> 2375.209</td><td> 25.823683</td><td>6.468403</td><td>72.53956</td></tr>\n",
       "\t<tr><th scope=row>ZnT35C19</th><td>Malpighian tubule             </td><td>ZnT35C </td><td>  987</td><td>  675</td><td> 2453.358</td><td> 28.777066</td><td>6.364413</td><td>71.67048</td></tr>\n",
       "\t<tr><th scope=row>CG4389631</th><td>proventriculus                </td><td>CG43896</td><td> 3511</td><td>  300</td><td> 1464.652</td><td> 12.160526</td><td>6.798197</td><td>71.49886</td></tr>\n",
       "\t<tr><th scope=row>CG1315932</th><td>salivary gland                </td><td>CG13159</td><td> 6768</td><td> 2707</td><td> 5500.193</td><td>106.760465</td><td>5.673582</td><td>70.49726</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type & gene & cell.type.n.umi & other.n.umi & cell.type.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tpip32 & salivary gland                 & pip     & 38398 &  9911 & 29916.415 & 412.274111 & 6.177695 & 91.85427\\\\\n",
       "\tPpn28 & plasmatocytes                  & Ppn     & 73383 & 14116 & 35715.779 & 536.739395 & 6.053511 & 91.55520\\\\\n",
       "\tCG43332 & amnioserosa                    & CG43332 & 23490 &  1608 &  7785.421 &  56.955284 & 7.069691 & 91.38809\\\\\n",
       "\tCG12011 & amnioserosa                    & CG12011 & 82838 & 15148 & 29343.847 & 535.995432 & 5.772004 & 85.66128\\\\\n",
       "\tnur32 & salivary gland                 & nur     & 26904 & 10233 & 22058.236 & 378.677940 & 5.860397 & 84.56023\\\\\n",
       "\tMuc30E & amnioserosa                    & Muc30E  & 15253 &  1442 &  4852.606 &  45.768575 & 6.697076 & 82.00464\\\\\n",
       "\tCG1475632 & salivary gland                 & CG14756 & 15387 &  5802 & 12488.709 & 206.762957 & 5.909542 & 80.41972\\\\\n",
       "\tCG1429219 & Malpighian tubule              & CG14292 &  1706 &   461 &  1335.417 &   5.328381 & 7.721238 & 80.17853\\\\\n",
       "\tCG1135041 & tracheal primordium            & CG11350 &  4932 &   540 &  2300.934 &  17.603434 & 6.950507 & 77.62765\\\\\n",
       "\tMur18B19 & Malpighian tubule              & Mur18B  &  1689 &   536 &  1372.886 &   7.201720 & 7.387070 & 77.00316\\\\\n",
       "\tpip33 & salivary gland body primordium & pip     & 12416 & 15041 & 27014.334 & 719.297781 & 5.228988 & 76.97850\\\\\n",
       "\tPxn28 & plasmatocytes                  & Pxn     & 12668 &  2411 &  6173.778 &  93.268241 & 6.033238 & 75.97157\\\\\n",
       "\twrapper16 & glia                           & wrapper &  1155 &  1040 &  3520.441 &  42.361026 & 6.343213 & 74.73542\\\\\n",
       "\tPde1c5 & crystal cell                   & Pde1c   &   908 &  5838 & 12521.122 & 278.241204 & 5.486708 & 74.68612\\\\\n",
       "\tCol4a128 & plasmatocytes                  & Col4a1  & 35872 & 11438 & 17492.963 & 449.982027 & 5.277561 & 74.38495\\\\\n",
       "\tCad96Cb40 & spiracles (tracheal system)    & Cad96Cb &   794 &   719 &  2900.128 &  32.672669 & 6.428394 & 73.94195\\\\\n",
       "\tCG18088 & amnioserosa                    & CG18088 &  7126 &   704 &  2375.209 &  25.823683 & 6.468403 & 72.53956\\\\\n",
       "\tZnT35C19 & Malpighian tubule              & ZnT35C  &   987 &   675 &  2453.358 &  28.777066 & 6.364413 & 71.67048\\\\\n",
       "\tCG4389631 & proventriculus                 & CG43896 &  3511 &   300 &  1464.652 &  12.160526 & 6.798197 & 71.49886\\\\\n",
       "\tCG1315932 & salivary gland                 & CG13159 &  6768 &  2707 &  5500.193 & 106.760465 & 5.673582 & 70.49726\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type &lt;chr&gt; | gene &lt;chr&gt; | cell.type.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| pip32 | salivary gland                 | pip     | 38398 |  9911 | 29916.415 | 412.274111 | 6.177695 | 91.85427 |\n",
       "| Ppn28 | plasmatocytes                  | Ppn     | 73383 | 14116 | 35715.779 | 536.739395 | 6.053511 | 91.55520 |\n",
       "| CG43332 | amnioserosa                    | CG43332 | 23490 |  1608 |  7785.421 |  56.955284 | 7.069691 | 91.38809 |\n",
       "| CG12011 | amnioserosa                    | CG12011 | 82838 | 15148 | 29343.847 | 535.995432 | 5.772004 | 85.66128 |\n",
       "| nur32 | salivary gland                 | nur     | 26904 | 10233 | 22058.236 | 378.677940 | 5.860397 | 84.56023 |\n",
       "| Muc30E | amnioserosa                    | Muc30E  | 15253 |  1442 |  4852.606 |  45.768575 | 6.697076 | 82.00464 |\n",
       "| CG1475632 | salivary gland                 | CG14756 | 15387 |  5802 | 12488.709 | 206.762957 | 5.909542 | 80.41972 |\n",
       "| CG1429219 | Malpighian tubule              | CG14292 |  1706 |   461 |  1335.417 |   5.328381 | 7.721238 | 80.17853 |\n",
       "| CG1135041 | tracheal primordium            | CG11350 |  4932 |   540 |  2300.934 |  17.603434 | 6.950507 | 77.62765 |\n",
       "| Mur18B19 | Malpighian tubule              | Mur18B  |  1689 |   536 |  1372.886 |   7.201720 | 7.387070 | 77.00316 |\n",
       "| pip33 | salivary gland body primordium | pip     | 12416 | 15041 | 27014.334 | 719.297781 | 5.228988 | 76.97850 |\n",
       "| Pxn28 | plasmatocytes                  | Pxn     | 12668 |  2411 |  6173.778 |  93.268241 | 6.033238 | 75.97157 |\n",
       "| wrapper16 | glia                           | wrapper |  1155 |  1040 |  3520.441 |  42.361026 | 6.343213 | 74.73542 |\n",
       "| Pde1c5 | crystal cell                   | Pde1c   |   908 |  5838 | 12521.122 | 278.241204 | 5.486708 | 74.68612 |\n",
       "| Col4a128 | plasmatocytes                  | Col4a1  | 35872 | 11438 | 17492.963 | 449.982027 | 5.277561 | 74.38495 |\n",
       "| Cad96Cb40 | spiracles (tracheal system)    | Cad96Cb |   794 |   719 |  2900.128 |  32.672669 | 6.428394 | 73.94195 |\n",
       "| CG18088 | amnioserosa                    | CG18088 |  7126 |   704 |  2375.209 |  25.823683 | 6.468403 | 72.53956 |\n",
       "| ZnT35C19 | Malpighian tubule              | ZnT35C  |   987 |   675 |  2453.358 |  28.777066 | 6.364413 | 71.67048 |\n",
       "| CG4389631 | proventriculus                 | CG43896 |  3511 |   300 |  1464.652 |  12.160526 | 6.798197 | 71.49886 |\n",
       "| CG1315932 | salivary gland                 | CG13159 |  6768 |  2707 |  5500.193 | 106.760465 | 5.673582 | 70.49726 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type                      gene    cell.type.n.umi other.n.umi\n",
       "pip32     salivary gland                 pip     38398            9911      \n",
       "Ppn28     plasmatocytes                  Ppn     73383           14116      \n",
       "CG43332   amnioserosa                    CG43332 23490            1608      \n",
       "CG12011   amnioserosa                    CG12011 82838           15148      \n",
       "nur32     salivary gland                 nur     26904           10233      \n",
       "Muc30E    amnioserosa                    Muc30E  15253            1442      \n",
       "CG1475632 salivary gland                 CG14756 15387            5802      \n",
       "CG1429219 Malpighian tubule              CG14292  1706             461      \n",
       "CG1135041 tracheal primordium            CG11350  4932             540      \n",
       "Mur18B19  Malpighian tubule              Mur18B   1689             536      \n",
       "pip33     salivary gland body primordium pip     12416           15041      \n",
       "Pxn28     plasmatocytes                  Pxn     12668            2411      \n",
       "wrapper16 glia                           wrapper  1155            1040      \n",
       "Pde1c5    crystal cell                   Pde1c     908            5838      \n",
       "Col4a128  plasmatocytes                  Col4a1  35872           11438      \n",
       "Cad96Cb40 spiracles (tracheal system)    Cad96Cb   794             719      \n",
       "CG18088   amnioserosa                    CG18088  7126             704      \n",
       "ZnT35C19  Malpighian tubule              ZnT35C    987             675      \n",
       "CG4389631 proventriculus                 CG43896  3511             300      \n",
       "CG1315932 salivary gland                 CG13159  6768            2707      \n",
       "          cell.type.tpm other.tpm  log2.ratio heuristic.score\n",
       "pip32     29916.415     412.274111 6.177695   91.85427       \n",
       "Ppn28     35715.779     536.739395 6.053511   91.55520       \n",
       "CG43332    7785.421      56.955284 7.069691   91.38809       \n",
       "CG12011   29343.847     535.995432 5.772004   85.66128       \n",
       "nur32     22058.236     378.677940 5.860397   84.56023       \n",
       "Muc30E     4852.606      45.768575 6.697076   82.00464       \n",
       "CG1475632 12488.709     206.762957 5.909542   80.41972       \n",
       "CG1429219  1335.417       5.328381 7.721238   80.17853       \n",
       "CG1135041  2300.934      17.603434 6.950507   77.62765       \n",
       "Mur18B19   1372.886       7.201720 7.387070   77.00316       \n",
       "pip33     27014.334     719.297781 5.228988   76.97850       \n",
       "Pxn28      6173.778      93.268241 6.033238   75.97157       \n",
       "wrapper16  3520.441      42.361026 6.343213   74.73542       \n",
       "Pde1c5    12521.122     278.241204 5.486708   74.68612       \n",
       "Col4a128  17492.963     449.982027 5.277561   74.38495       \n",
       "Cad96Cb40  2900.128      32.672669 6.428394   73.94195       \n",
       "CG18088    2375.209      25.823683 6.468403   72.53956       \n",
       "ZnT35C19   2453.358      28.777066 6.364413   71.67048       \n",
       "CG4389631  1464.652      12.160526 6.798197   71.49886       \n",
       "CG1315932  5500.193     106.760465 5.673582   70.49726       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.markers.sub300.cell.type  %>% filter(cell.type.n.umi > 100)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1265,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finding markers for cluster amnioserosa\n",
      "\n",
      "# of cells sampled in set 1: 7123\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster anlage in statu nascendi\n",
      "\n",
      "# of cells sampled in set 1: 19\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster brain\n",
      "\n",
      "# of cells sampled in set 1: 25682\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster brain primordium\n",
      "\n",
      "# of cells sampled in set 1: 8912\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster cellular blastoderm / pole cell\n",
      "\n",
      "# of cells sampled in set 1: 4\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster crystal cell\n",
      "\n",
      "# of cells sampled in set 1: 67\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster ectoderm anlage in statu nascendi\n",
      "\n",
      "# of cells sampled in set 1: 33\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster ectoderm primordium\n",
      "\n",
      "# of cells sampled in set 1: 1452\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster endoderm primordium\n",
      "\n",
      "# of cells sampled in set 1: 297\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster epidermis\n",
      "\n",
      "# of cells sampled in set 1: 34401\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster epidermis primordium\n",
      "\n",
      "# of cells sampled in set 1: 6231\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster fat body\n",
      "\n",
      "# of cells sampled in set 1: 5070\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster foregut / hindgut / anal pad\n",
      "\n",
      "# of cells sampled in set 1: 5232\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster foregut / hindgut primordium\n",
      "\n",
      "# of cells sampled in set 1: 993\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster germ cell\n",
      "\n",
      "# of cells sampled in set 1: 1025\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster glia\n",
      "\n",
      "# of cells sampled in set 1: 663\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster hindgut / anal pad\n",
      "\n",
      "# of cells sampled in set 1: 2282\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster hindgut primordium\n",
      "\n",
      "# of cells sampled in set 1: 1521\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster Malpighian tubule\n",
      "\n",
      "# of cells sampled in set 1: 838\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster maternal\n",
      "\n",
      "# of cells sampled in set 1: 12\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster mesoderm primordium\n",
      "\n",
      "# of cells sampled in set 1: 284\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster midgut\n",
      "\n",
      "# of cells sampled in set 1: 10422\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster midgut primordium\n",
      "\n",
      "# of cells sampled in set 1: 1429\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster midline primordium\n",
      "\n",
      "# of cells sampled in set 1: 751\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster muscle\n",
      "\n",
      "# of cells sampled in set 1: 4616\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster peripheral nervous system\n",
      "\n",
      "# of cells sampled in set 1: 3188\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster peripheral nervous system primordium\n",
      "\n",
      "# of cells sampled in set 1: 417\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster plasmatocytes\n",
      "\n",
      "# of cells sampled in set 1: 4882\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster plasmatocytes anlage\n",
      "\n",
      "# of cells sampled in set 1: 1409\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster procrystal cell\n",
      "\n",
      "# of cells sampled in set 1: 1310\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster proventriculus\n",
      "\n",
      "# of cells sampled in set 1: 2085\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster salivary gland\n",
      "\n",
      "# of cells sampled in set 1: 3240\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster salivary gland body primordium\n",
      "\n",
      "# of cells sampled in set 1: 902\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster salivary gland primordium\n",
      "\n",
      "# of cells sampled in set 1: 22\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster sensory nervous system\n",
      "\n",
      "# of cells sampled in set 1: 3725\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster sensory nervous system primordium\n",
      "\n",
      "# of cells sampled in set 1: 1241\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster somatic muscle\n",
      "\n",
      "# of cells sampled in set 1: 13698\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster somatic muscle primordium\n",
      "\n",
      "# of cells sampled in set 1: 1285\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster spiracle (tracheal system)\n",
      "\n",
      "# of cells sampled in set 1: 641\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster spiracles (tracheal system)\n",
      "\n",
      "# of cells sampled in set 1: 616\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster tracheal primordium\n",
      "\n",
      "# of cells sampled in set 1: 1988\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster tracheal system\n",
      "\n",
      "# of cells sampled in set 1: 6903\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster trunk mesoderm primordium\n",
      "\n",
      "# of cells sampled in set 1: 570\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster ubiquitous\n",
      "\n",
      "# of cells sampled in set 1: 31834\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster unknown\n",
      "\n",
      "# of cells sampled in set 1: 879\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster ventral nerve cord\n",
      "\n",
      "# of cells sampled in set 1: 7062\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster ventral nerve cord primordium\n",
      "\n",
      "# of cells sampled in set 1: 526\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster visceral muscle\n",
      "\n",
      "# of cells sampled in set 1: 515\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster visceral muscle primordium\n",
      "\n",
      "# of cells sampled in set 1: 2890\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster yolk nuclei\n",
      "\n",
      "# of cells sampled in set 1: 311\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    }
   ],
   "source": [
    "DEG.results.sub300_no_singletons.cell.type = lapply(sort((unique(df_sub_no_singletons$annot))),\n",
    "                     function(this.cell.type) {\n",
    "    message(\"Finding markers for cluster \", this.cell.type)\n",
    "    cbind(\n",
    "        two.set.differential.gene.test.3a(\n",
    "            cds_sub_no_sing,\n",
    "            df_sub_no_singletons$annot == this.cell.type,\n",
    "            df_sub_no_singletons$annot != this.cell.type,\n",
    "            sample.n.max = 50000),\n",
    "        data.frame(cell.type = this.cell.type))\n",
    "                         })"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1266,
   "metadata": {},
   "outputs": [],
   "source": [
    "cell.type.markers.sub300_no_singletons.cell.type = do.call(rbind, DEG.results.sub300_no_singletons.cell.type) %>% filter(higher.expr == \"Set 1\") %>%\n",
    "    select(\n",
    "        cell.type = cell.type, gene, \n",
    "        cell.type.n.umi = set.1.n.umi,\n",
    "        other.n.umi = set.2.n.umi,\n",
    "        cell.type.tpm = set.1.tpm,\n",
    "        other.tpm = set.2.tpm,\n",
    "        log2.ratio, heuristic.score) %>%\n",
    "    arrange(-heuristic.score)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1271,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type</th><th scope=col>gene</th><th scope=col>cell.type.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Ppn27</th><td>plasmatocytes                 </td><td>Ppn    </td><td>71774</td><td>14120</td><td>39945.582</td><td>536.442522</td><td>6.215782</td><td>95.01310</td></tr>\n",
       "\t<tr><th scope=row>pip31</th><td>salivary gland                </td><td>pip    </td><td>37863</td><td> 9930</td><td>31750.702</td><td>397.472602</td><td>6.316164</td><td>94.45537</td></tr>\n",
       "\t<tr><th scope=row>Pde1c5</th><td>crystal cell                  </td><td>Pde1c  </td><td>  582</td><td> 5906</td><td>23364.987</td><td>270.162694</td><td>6.429046</td><td>93.29910</td></tr>\n",
       "\t<tr><th scope=row>CG43332</th><td>amnioserosa                   </td><td>CG43332</td><td>23047</td><td> 1551</td><td> 8171.342</td><td> 57.964400</td><td>7.114585</td><td>92.46495</td></tr>\n",
       "\t<tr><th scope=row>Mur18B18</th><td>Malpighian tubule             </td><td>Mur18B </td><td> 1677</td><td>  313</td><td> 1762.054</td><td>  5.027292</td><td>8.191532</td><td>88.33634</td></tr>\n",
       "\t<tr><th scope=row>CG1429218</th><td>Malpighian tubule             </td><td>CG14292</td><td> 1694</td><td>  274</td><td> 1716.660</td><td>  5.258797</td><td>8.099503</td><td>87.03912</td></tr>\n",
       "\t<tr><th scope=row>CG12011</th><td>amnioserosa                   </td><td>CG12011</td><td>81662</td><td>15596</td><td>30913.826</td><td>541.220982</td><td>5.833227</td><td>87.00849</td></tr>\n",
       "\t<tr><th scope=row>nur31</th><td>salivary gland                </td><td>nur    </td><td>26383</td><td>10064</td><td>23290.655</td><td>368.527442</td><td>5.977926</td><td>86.72491</td></tr>\n",
       "\t<tr><th scope=row>CG1135040</th><td>tracheal primordium           </td><td>CG11350</td><td> 4883</td><td>  828</td><td> 2882.860</td><td> 16.759489</td><td>7.342767</td><td>84.39619</td></tr>\n",
       "\t<tr><th scope=row>Muc11A18</th><td>Malpighian tubule             </td><td>Muc11A </td><td> 1400</td><td>  191</td><td> 1533.557</td><td>  5.277513</td><td>7.932473</td><td>83.95417</td></tr>\n",
       "\t<tr><th scope=row>Muc30E</th><td>amnioserosa                   </td><td>Muc30E </td><td>15015</td><td> 1404</td><td> 5098.146</td><td> 45.186317</td><td>6.786363</td><td>83.58112</td></tr>\n",
       "\t<tr><th scope=row>Muc96D44</th><td>unknown                       </td><td>Muc96D </td><td> 1243</td><td>  304</td><td> 2142.152</td><td> 11.673103</td><td>7.401147</td><td>81.89754</td></tr>\n",
       "\t<tr><th scope=row>CG1475631</th><td>salivary gland                </td><td>CG14756</td><td>15108</td><td> 5856</td><td>13190.646</td><td>208.816219</td><td>5.974245</td><td>81.77151</td></tr>\n",
       "\t<tr><th scope=row>Muc26B44</th><td>unknown                       </td><td>Muc26B </td><td> 1574</td><td>  543</td><td> 2593.095</td><td> 17.781491</td><td>7.109220</td><td>80.62577</td></tr>\n",
       "\t<tr><th scope=row>wrapper15</th><td>glia                          </td><td>wrapper</td><td> 1023</td><td>  995</td><td> 4151.796</td><td> 39.427475</td><td>6.682256</td><td>80.31982</td></tr>\n",
       "\t<tr><th scope=row>Pxn27</th><td>plasmatocytes                 </td><td>Pxn    </td><td>12200</td><td> 2357</td><td> 6795.435</td><td> 90.346121</td><td>6.217079</td><td>79.14691</td></tr>\n",
       "\t<tr><th scope=row>pip32</th><td>salivary gland body primordium</td><td>pip    </td><td>12041</td><td>15871</td><td>30212.212</td><td>760.092411</td><td>5.310916</td><td>79.04179</td></tr>\n",
       "\t<tr><th scope=row>PPO25</th><td>crystal cell                  </td><td>PPO2   </td><td>  158</td><td> 2591</td><td> 6696.101</td><td> 89.850590</td><td>6.203682</td><td>78.84458</td></tr>\n",
       "\t<tr><th scope=row>Col4a127</th><td>plasmatocytes                 </td><td>Col4a1 </td><td>34974</td><td>11468</td><td>19506.194</td><td>449.208424</td><td>5.437195</td><td>77.48938</td></tr>\n",
       "\t<tr><th scope=row>ZnT35C18</th><td>Malpighian tubule             </td><td>ZnT35C </td><td>  856</td><td>  634</td><td> 2725.608</td><td> 26.213955</td><td>6.646087</td><td>75.85107</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type & gene & cell.type.n.umi & other.n.umi & cell.type.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tPpn27 & plasmatocytes                  & Ppn     & 71774 & 14120 & 39945.582 & 536.442522 & 6.215782 & 95.01310\\\\\n",
       "\tpip31 & salivary gland                 & pip     & 37863 &  9930 & 31750.702 & 397.472602 & 6.316164 & 94.45537\\\\\n",
       "\tPde1c5 & crystal cell                   & Pde1c   &   582 &  5906 & 23364.987 & 270.162694 & 6.429046 & 93.29910\\\\\n",
       "\tCG43332 & amnioserosa                    & CG43332 & 23047 &  1551 &  8171.342 &  57.964400 & 7.114585 & 92.46495\\\\\n",
       "\tMur18B18 & Malpighian tubule              & Mur18B  &  1677 &   313 &  1762.054 &   5.027292 & 8.191532 & 88.33634\\\\\n",
       "\tCG1429218 & Malpighian tubule              & CG14292 &  1694 &   274 &  1716.660 &   5.258797 & 8.099503 & 87.03912\\\\\n",
       "\tCG12011 & amnioserosa                    & CG12011 & 81662 & 15596 & 30913.826 & 541.220982 & 5.833227 & 87.00849\\\\\n",
       "\tnur31 & salivary gland                 & nur     & 26383 & 10064 & 23290.655 & 368.527442 & 5.977926 & 86.72491\\\\\n",
       "\tCG1135040 & tracheal primordium            & CG11350 &  4883 &   828 &  2882.860 &  16.759489 & 7.342767 & 84.39619\\\\\n",
       "\tMuc11A18 & Malpighian tubule              & Muc11A  &  1400 &   191 &  1533.557 &   5.277513 & 7.932473 & 83.95417\\\\\n",
       "\tMuc30E & amnioserosa                    & Muc30E  & 15015 &  1404 &  5098.146 &  45.186317 & 6.786363 & 83.58112\\\\\n",
       "\tMuc96D44 & unknown                        & Muc96D  &  1243 &   304 &  2142.152 &  11.673103 & 7.401147 & 81.89754\\\\\n",
       "\tCG1475631 & salivary gland                 & CG14756 & 15108 &  5856 & 13190.646 & 208.816219 & 5.974245 & 81.77151\\\\\n",
       "\tMuc26B44 & unknown                        & Muc26B  &  1574 &   543 &  2593.095 &  17.781491 & 7.109220 & 80.62577\\\\\n",
       "\twrapper15 & glia                           & wrapper &  1023 &   995 &  4151.796 &  39.427475 & 6.682256 & 80.31982\\\\\n",
       "\tPxn27 & plasmatocytes                  & Pxn     & 12200 &  2357 &  6795.435 &  90.346121 & 6.217079 & 79.14691\\\\\n",
       "\tpip32 & salivary gland body primordium & pip     & 12041 & 15871 & 30212.212 & 760.092411 & 5.310916 & 79.04179\\\\\n",
       "\tPPO25 & crystal cell                   & PPO2    &   158 &  2591 &  6696.101 &  89.850590 & 6.203682 & 78.84458\\\\\n",
       "\tCol4a127 & plasmatocytes                  & Col4a1  & 34974 & 11468 & 19506.194 & 449.208424 & 5.437195 & 77.48938\\\\\n",
       "\tZnT35C18 & Malpighian tubule              & ZnT35C  &   856 &   634 &  2725.608 &  26.213955 & 6.646087 & 75.85107\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type &lt;chr&gt; | gene &lt;chr&gt; | cell.type.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Ppn27 | plasmatocytes                  | Ppn     | 71774 | 14120 | 39945.582 | 536.442522 | 6.215782 | 95.01310 |\n",
       "| pip31 | salivary gland                 | pip     | 37863 |  9930 | 31750.702 | 397.472602 | 6.316164 | 94.45537 |\n",
       "| Pde1c5 | crystal cell                   | Pde1c   |   582 |  5906 | 23364.987 | 270.162694 | 6.429046 | 93.29910 |\n",
       "| CG43332 | amnioserosa                    | CG43332 | 23047 |  1551 |  8171.342 |  57.964400 | 7.114585 | 92.46495 |\n",
       "| Mur18B18 | Malpighian tubule              | Mur18B  |  1677 |   313 |  1762.054 |   5.027292 | 8.191532 | 88.33634 |\n",
       "| CG1429218 | Malpighian tubule              | CG14292 |  1694 |   274 |  1716.660 |   5.258797 | 8.099503 | 87.03912 |\n",
       "| CG12011 | amnioserosa                    | CG12011 | 81662 | 15596 | 30913.826 | 541.220982 | 5.833227 | 87.00849 |\n",
       "| nur31 | salivary gland                 | nur     | 26383 | 10064 | 23290.655 | 368.527442 | 5.977926 | 86.72491 |\n",
       "| CG1135040 | tracheal primordium            | CG11350 |  4883 |   828 |  2882.860 |  16.759489 | 7.342767 | 84.39619 |\n",
       "| Muc11A18 | Malpighian tubule              | Muc11A  |  1400 |   191 |  1533.557 |   5.277513 | 7.932473 | 83.95417 |\n",
       "| Muc30E | amnioserosa                    | Muc30E  | 15015 |  1404 |  5098.146 |  45.186317 | 6.786363 | 83.58112 |\n",
       "| Muc96D44 | unknown                        | Muc96D  |  1243 |   304 |  2142.152 |  11.673103 | 7.401147 | 81.89754 |\n",
       "| CG1475631 | salivary gland                 | CG14756 | 15108 |  5856 | 13190.646 | 208.816219 | 5.974245 | 81.77151 |\n",
       "| Muc26B44 | unknown                        | Muc26B  |  1574 |   543 |  2593.095 |  17.781491 | 7.109220 | 80.62577 |\n",
       "| wrapper15 | glia                           | wrapper |  1023 |   995 |  4151.796 |  39.427475 | 6.682256 | 80.31982 |\n",
       "| Pxn27 | plasmatocytes                  | Pxn     | 12200 |  2357 |  6795.435 |  90.346121 | 6.217079 | 79.14691 |\n",
       "| pip32 | salivary gland body primordium | pip     | 12041 | 15871 | 30212.212 | 760.092411 | 5.310916 | 79.04179 |\n",
       "| PPO25 | crystal cell                   | PPO2    |   158 |  2591 |  6696.101 |  89.850590 | 6.203682 | 78.84458 |\n",
       "| Col4a127 | plasmatocytes                  | Col4a1  | 34974 | 11468 | 19506.194 | 449.208424 | 5.437195 | 77.48938 |\n",
       "| ZnT35C18 | Malpighian tubule              | ZnT35C  |   856 |   634 |  2725.608 |  26.213955 | 6.646087 | 75.85107 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type                      gene    cell.type.n.umi other.n.umi\n",
       "Ppn27     plasmatocytes                  Ppn     71774           14120      \n",
       "pip31     salivary gland                 pip     37863            9930      \n",
       "Pde1c5    crystal cell                   Pde1c     582            5906      \n",
       "CG43332   amnioserosa                    CG43332 23047            1551      \n",
       "Mur18B18  Malpighian tubule              Mur18B   1677             313      \n",
       "CG1429218 Malpighian tubule              CG14292  1694             274      \n",
       "CG12011   amnioserosa                    CG12011 81662           15596      \n",
       "nur31     salivary gland                 nur     26383           10064      \n",
       "CG1135040 tracheal primordium            CG11350  4883             828      \n",
       "Muc11A18  Malpighian tubule              Muc11A   1400             191      \n",
       "Muc30E    amnioserosa                    Muc30E  15015            1404      \n",
       "Muc96D44  unknown                        Muc96D   1243             304      \n",
       "CG1475631 salivary gland                 CG14756 15108            5856      \n",
       "Muc26B44  unknown                        Muc26B   1574             543      \n",
       "wrapper15 glia                           wrapper  1023             995      \n",
       "Pxn27     plasmatocytes                  Pxn     12200            2357      \n",
       "pip32     salivary gland body primordium pip     12041           15871      \n",
       "PPO25     crystal cell                   PPO2      158            2591      \n",
       "Col4a127  plasmatocytes                  Col4a1  34974           11468      \n",
       "ZnT35C18  Malpighian tubule              ZnT35C    856             634      \n",
       "          cell.type.tpm other.tpm  log2.ratio heuristic.score\n",
       "Ppn27     39945.582     536.442522 6.215782   95.01310       \n",
       "pip31     31750.702     397.472602 6.316164   94.45537       \n",
       "Pde1c5    23364.987     270.162694 6.429046   93.29910       \n",
       "CG43332    8171.342      57.964400 7.114585   92.46495       \n",
       "Mur18B18   1762.054       5.027292 8.191532   88.33634       \n",
       "CG1429218  1716.660       5.258797 8.099503   87.03912       \n",
       "CG12011   30913.826     541.220982 5.833227   87.00849       \n",
       "nur31     23290.655     368.527442 5.977926   86.72491       \n",
       "CG1135040  2882.860      16.759489 7.342767   84.39619       \n",
       "Muc11A18   1533.557       5.277513 7.932473   83.95417       \n",
       "Muc30E     5098.146      45.186317 6.786363   83.58112       \n",
       "Muc96D44   2142.152      11.673103 7.401147   81.89754       \n",
       "CG1475631 13190.646     208.816219 5.974245   81.77151       \n",
       "Muc26B44   2593.095      17.781491 7.109220   80.62577       \n",
       "wrapper15  4151.796      39.427475 6.682256   80.31982       \n",
       "Pxn27      6795.435      90.346121 6.217079   79.14691       \n",
       "pip32     30212.212     760.092411 5.310916   79.04179       \n",
       "PPO25      6696.101      89.850590 6.203682   78.84458       \n",
       "Col4a127  19506.194     449.208424 5.437195   77.48938       \n",
       "ZnT35C18   2725.608      26.213955 6.646087   75.85107       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.markers.sub300_no_singletons.cell.type %>% filter(cell.type.n.umi > 100)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type</th><th scope=col>gene</th><th scope=col>cell.type.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Ppn28</th><td>plasmatocytes</td><td>Ppn        </td><td>73383</td><td>14116</td><td>35715.7788</td><td>536.739395</td><td>6.053511</td><td>91.55520</td></tr>\n",
       "\t<tr><th scope=row>Pxn28</th><td>plasmatocytes</td><td>Pxn        </td><td>12668</td><td> 2411</td><td> 6173.7776</td><td> 93.268241</td><td>6.033238</td><td>75.97157</td></tr>\n",
       "\t<tr><th scope=row>Col4a128</th><td>plasmatocytes</td><td>Col4a1     </td><td>35872</td><td>11438</td><td>17492.9629</td><td>449.982027</td><td>5.277561</td><td>74.38495</td></tr>\n",
       "\t<tr><th scope=row>vkg28</th><td>plasmatocytes</td><td>vkg        </td><td>22952</td><td> 7448</td><td>11136.7554</td><td>296.224930</td><td>5.227630</td><td>70.27592</td></tr>\n",
       "\t<tr><th scope=row>Ance-528</th><td>plasmatocytes</td><td>Ance-5     </td><td> 1061</td><td>  196</td><td>  521.5562</td><td>  7.292113</td><td>5.974939</td><td>53.95037</td></tr>\n",
       "\t<tr><th scope=row>NtR28</th><td>plasmatocytes</td><td>NtR        </td><td> 1803</td><td>  486</td><td>  900.4277</td><td> 19.621909</td><td>5.448361</td><td>53.48148</td></tr>\n",
       "\t<tr><th scope=row>Plod28</th><td>plasmatocytes</td><td>Plod       </td><td>15887</td><td>11368</td><td> 7821.5455</td><td>486.544675</td><td>4.003847</td><td>51.78345</td></tr>\n",
       "\t<tr><th scope=row>CG3105128</th><td>plasmatocytes</td><td>CG31051    </td><td> 1142</td><td>  279</td><td>  567.3650</td><td> 10.717179</td><td>5.597580</td><td>51.22163</td></tr>\n",
       "\t<tr><th scope=row>CG109229</th><td>plasmatocytes</td><td>CG1092     </td><td> 1278</td><td>  335</td><td>  637.4403</td><td> 13.417458</td><td>5.466402</td><td>50.93816</td></tr>\n",
       "\t<tr><th scope=row>SPARC28</th><td>plasmatocytes</td><td>SPARC      </td><td> 9754</td><td> 7184</td><td> 4783.0811</td><td>291.396665</td><td>4.031941</td><td>49.28656</td></tr>\n",
       "\t<tr><th scope=row>Karl28</th><td>plasmatocytes</td><td>Karl       </td><td>  838</td><td>  189</td><td>  411.2801</td><td>  7.183737</td><td>5.651218</td><td>49.09485</td></tr>\n",
       "\t<tr><th scope=row>fat-spondin28</th><td>plasmatocytes</td><td>fat-spondin</td><td> 4061</td><td> 2268</td><td> 1997.4921</td><td> 99.991078</td><td>4.305890</td><td>47.21278</td></tr>\n",
       "\t<tr><th scope=row>NimC428</th><td>plasmatocytes</td><td>NimC4      </td><td> 4697</td><td> 3273</td><td> 2384.4128</td><td>129.740666</td><td>4.188854</td><td>46.99904</td></tr>\n",
       "\t<tr><th scope=row>Tig28</th><td>plasmatocytes</td><td>Tig        </td><td>10842</td><td> 9085</td><td> 5301.0267</td><td>383.327720</td><td>3.785863</td><td>46.83994</td></tr>\n",
       "\t<tr><th scope=row>firl28</th><td>plasmatocytes</td><td>firl       </td><td> 1390</td><td>  585</td><td>  699.1032</td><td> 23.190692</td><td>4.852981</td><td>45.86758</td></tr>\n",
       "\t<tr><th scope=row>hoe128</th><td>plasmatocytes</td><td>hoe1       </td><td> 1716</td><td>  779</td><td>  814.4804</td><td> 29.782489</td><td>4.725698</td><td>45.70462</td></tr>\n",
       "\t<tr><th scope=row>CG631028</th><td>plasmatocytes</td><td>CG6310     </td><td> 1622</td><td>  669</td><td>  800.1739</td><td> 29.532193</td><td>4.711910</td><td>45.45096</td></tr>\n",
       "\t<tr><th scope=row>nAChRbeta328</th><td>plasmatocytes</td><td>nAChRbeta3 </td><td>  850</td><td>  344</td><td>  431.7903</td><td> 13.217218</td><td>4.924620</td><td>43.12748</td></tr>\n",
       "\t<tr><th scope=row>CG4379928</th><td>plasmatocytes</td><td>CG43799    </td><td>  380</td><td>   76</td><td>  191.8828</td><td>  3.153807</td><td>5.529647</td><td>41.97876</td></tr>\n",
       "\t<tr><th scope=row>Nepl1928</th><td>plasmatocytes</td><td>Nepl19     </td><td> 1214</td><td>  627</td><td>  585.3566</td><td> 23.787657</td><td>4.561622</td><td>41.94701</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type & gene & cell.type.n.umi & other.n.umi & cell.type.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tPpn28 & plasmatocytes & Ppn         & 73383 & 14116 & 35715.7788 & 536.739395 & 6.053511 & 91.55520\\\\\n",
       "\tPxn28 & plasmatocytes & Pxn         & 12668 &  2411 &  6173.7776 &  93.268241 & 6.033238 & 75.97157\\\\\n",
       "\tCol4a128 & plasmatocytes & Col4a1      & 35872 & 11438 & 17492.9629 & 449.982027 & 5.277561 & 74.38495\\\\\n",
       "\tvkg28 & plasmatocytes & vkg         & 22952 &  7448 & 11136.7554 & 296.224930 & 5.227630 & 70.27592\\\\\n",
       "\tAnce-528 & plasmatocytes & Ance-5      &  1061 &   196 &   521.5562 &   7.292113 & 5.974939 & 53.95037\\\\\n",
       "\tNtR28 & plasmatocytes & NtR         &  1803 &   486 &   900.4277 &  19.621909 & 5.448361 & 53.48148\\\\\n",
       "\tPlod28 & plasmatocytes & Plod        & 15887 & 11368 &  7821.5455 & 486.544675 & 4.003847 & 51.78345\\\\\n",
       "\tCG3105128 & plasmatocytes & CG31051     &  1142 &   279 &   567.3650 &  10.717179 & 5.597580 & 51.22163\\\\\n",
       "\tCG109229 & plasmatocytes & CG1092      &  1278 &   335 &   637.4403 &  13.417458 & 5.466402 & 50.93816\\\\\n",
       "\tSPARC28 & plasmatocytes & SPARC       &  9754 &  7184 &  4783.0811 & 291.396665 & 4.031941 & 49.28656\\\\\n",
       "\tKarl28 & plasmatocytes & Karl        &   838 &   189 &   411.2801 &   7.183737 & 5.651218 & 49.09485\\\\\n",
       "\tfat-spondin28 & plasmatocytes & fat-spondin &  4061 &  2268 &  1997.4921 &  99.991078 & 4.305890 & 47.21278\\\\\n",
       "\tNimC428 & plasmatocytes & NimC4       &  4697 &  3273 &  2384.4128 & 129.740666 & 4.188854 & 46.99904\\\\\n",
       "\tTig28 & plasmatocytes & Tig         & 10842 &  9085 &  5301.0267 & 383.327720 & 3.785863 & 46.83994\\\\\n",
       "\tfirl28 & plasmatocytes & firl        &  1390 &   585 &   699.1032 &  23.190692 & 4.852981 & 45.86758\\\\\n",
       "\thoe128 & plasmatocytes & hoe1        &  1716 &   779 &   814.4804 &  29.782489 & 4.725698 & 45.70462\\\\\n",
       "\tCG631028 & plasmatocytes & CG6310      &  1622 &   669 &   800.1739 &  29.532193 & 4.711910 & 45.45096\\\\\n",
       "\tnAChRbeta328 & plasmatocytes & nAChRbeta3  &   850 &   344 &   431.7903 &  13.217218 & 4.924620 & 43.12748\\\\\n",
       "\tCG4379928 & plasmatocytes & CG43799     &   380 &    76 &   191.8828 &   3.153807 & 5.529647 & 41.97876\\\\\n",
       "\tNepl1928 & plasmatocytes & Nepl19      &  1214 &   627 &   585.3566 &  23.787657 & 4.561622 & 41.94701\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type &lt;chr&gt; | gene &lt;chr&gt; | cell.type.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Ppn28 | plasmatocytes | Ppn         | 73383 | 14116 | 35715.7788 | 536.739395 | 6.053511 | 91.55520 |\n",
       "| Pxn28 | plasmatocytes | Pxn         | 12668 |  2411 |  6173.7776 |  93.268241 | 6.033238 | 75.97157 |\n",
       "| Col4a128 | plasmatocytes | Col4a1      | 35872 | 11438 | 17492.9629 | 449.982027 | 5.277561 | 74.38495 |\n",
       "| vkg28 | plasmatocytes | vkg         | 22952 |  7448 | 11136.7554 | 296.224930 | 5.227630 | 70.27592 |\n",
       "| Ance-528 | plasmatocytes | Ance-5      |  1061 |   196 |   521.5562 |   7.292113 | 5.974939 | 53.95037 |\n",
       "| NtR28 | plasmatocytes | NtR         |  1803 |   486 |   900.4277 |  19.621909 | 5.448361 | 53.48148 |\n",
       "| Plod28 | plasmatocytes | Plod        | 15887 | 11368 |  7821.5455 | 486.544675 | 4.003847 | 51.78345 |\n",
       "| CG3105128 | plasmatocytes | CG31051     |  1142 |   279 |   567.3650 |  10.717179 | 5.597580 | 51.22163 |\n",
       "| CG109229 | plasmatocytes | CG1092      |  1278 |   335 |   637.4403 |  13.417458 | 5.466402 | 50.93816 |\n",
       "| SPARC28 | plasmatocytes | SPARC       |  9754 |  7184 |  4783.0811 | 291.396665 | 4.031941 | 49.28656 |\n",
       "| Karl28 | plasmatocytes | Karl        |   838 |   189 |   411.2801 |   7.183737 | 5.651218 | 49.09485 |\n",
       "| fat-spondin28 | plasmatocytes | fat-spondin |  4061 |  2268 |  1997.4921 |  99.991078 | 4.305890 | 47.21278 |\n",
       "| NimC428 | plasmatocytes | NimC4       |  4697 |  3273 |  2384.4128 | 129.740666 | 4.188854 | 46.99904 |\n",
       "| Tig28 | plasmatocytes | Tig         | 10842 |  9085 |  5301.0267 | 383.327720 | 3.785863 | 46.83994 |\n",
       "| firl28 | plasmatocytes | firl        |  1390 |   585 |   699.1032 |  23.190692 | 4.852981 | 45.86758 |\n",
       "| hoe128 | plasmatocytes | hoe1        |  1716 |   779 |   814.4804 |  29.782489 | 4.725698 | 45.70462 |\n",
       "| CG631028 | plasmatocytes | CG6310      |  1622 |   669 |   800.1739 |  29.532193 | 4.711910 | 45.45096 |\n",
       "| nAChRbeta328 | plasmatocytes | nAChRbeta3  |   850 |   344 |   431.7903 |  13.217218 | 4.924620 | 43.12748 |\n",
       "| CG4379928 | plasmatocytes | CG43799     |   380 |    76 |   191.8828 |   3.153807 | 5.529647 | 41.97876 |\n",
       "| Nepl1928 | plasmatocytes | Nepl19      |  1214 |   627 |   585.3566 |  23.787657 | 4.561622 | 41.94701 |\n",
       "\n"
      ],
      "text/plain": [
       "              cell.type     gene        cell.type.n.umi other.n.umi\n",
       "Ppn28         plasmatocytes Ppn         73383           14116      \n",
       "Pxn28         plasmatocytes Pxn         12668            2411      \n",
       "Col4a128      plasmatocytes Col4a1      35872           11438      \n",
       "vkg28         plasmatocytes vkg         22952            7448      \n",
       "Ance-528      plasmatocytes Ance-5       1061             196      \n",
       "NtR28         plasmatocytes NtR          1803             486      \n",
       "Plod28        plasmatocytes Plod        15887           11368      \n",
       "CG3105128     plasmatocytes CG31051      1142             279      \n",
       "CG109229      plasmatocytes CG1092       1278             335      \n",
       "SPARC28       plasmatocytes SPARC        9754            7184      \n",
       "Karl28        plasmatocytes Karl          838             189      \n",
       "fat-spondin28 plasmatocytes fat-spondin  4061            2268      \n",
       "NimC428       plasmatocytes NimC4        4697            3273      \n",
       "Tig28         plasmatocytes Tig         10842            9085      \n",
       "firl28        plasmatocytes firl         1390             585      \n",
       "hoe128        plasmatocytes hoe1         1716             779      \n",
       "CG631028      plasmatocytes CG6310       1622             669      \n",
       "nAChRbeta328  plasmatocytes nAChRbeta3    850             344      \n",
       "CG4379928     plasmatocytes CG43799       380              76      \n",
       "Nepl1928      plasmatocytes Nepl19       1214             627      \n",
       "              cell.type.tpm other.tpm  log2.ratio heuristic.score\n",
       "Ppn28         35715.7788    536.739395 6.053511   91.55520       \n",
       "Pxn28          6173.7776     93.268241 6.033238   75.97157       \n",
       "Col4a128      17492.9629    449.982027 5.277561   74.38495       \n",
       "vkg28         11136.7554    296.224930 5.227630   70.27592       \n",
       "Ance-528        521.5562      7.292113 5.974939   53.95037       \n",
       "NtR28           900.4277     19.621909 5.448361   53.48148       \n",
       "Plod28         7821.5455    486.544675 4.003847   51.78345       \n",
       "CG3105128       567.3650     10.717179 5.597580   51.22163       \n",
       "CG109229        637.4403     13.417458 5.466402   50.93816       \n",
       "SPARC28        4783.0811    291.396665 4.031941   49.28656       \n",
       "Karl28          411.2801      7.183737 5.651218   49.09485       \n",
       "fat-spondin28  1997.4921     99.991078 4.305890   47.21278       \n",
       "NimC428        2384.4128    129.740666 4.188854   46.99904       \n",
       "Tig28          5301.0267    383.327720 3.785863   46.83994       \n",
       "firl28          699.1032     23.190692 4.852981   45.86758       \n",
       "hoe128          814.4804     29.782489 4.725698   45.70462       \n",
       "CG631028        800.1739     29.532193 4.711910   45.45096       \n",
       "nAChRbeta328    431.7903     13.217218 4.924620   43.12748       \n",
       "CG4379928       191.8828      3.153807 5.529647   41.97876       \n",
       "Nepl1928        585.3566     23.787657 4.561622   41.94701       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.markers.sub300.cell.type %>% filter(cell.type.n.umi > 100 & \n",
    "                                                            cell.type == \"plasmatocytes\")  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1272,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type</th><th scope=col>gene</th><th scope=col>cell.type.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Ppn27</th><td>plasmatocytes</td><td>Ppn        </td><td>71774</td><td>14120</td><td>39945.5823</td><td> 536.442522</td><td>6.215782</td><td>95.01310</td></tr>\n",
       "\t<tr><th scope=row>Pxn27</th><td>plasmatocytes</td><td>Pxn        </td><td>12200</td><td> 2357</td><td> 6795.4350</td><td>  90.346121</td><td>6.217079</td><td>79.14691</td></tr>\n",
       "\t<tr><th scope=row>Col4a127</th><td>plasmatocytes</td><td>Col4a1     </td><td>34974</td><td>11468</td><td>19506.1940</td><td> 449.208424</td><td>5.437195</td><td>77.48938</td></tr>\n",
       "\t<tr><th scope=row>vkg27</th><td>plasmatocytes</td><td>vkg        </td><td>22254</td><td> 7348</td><td>12342.0464</td><td> 294.444644</td><td>5.384550</td><td>73.18363</td></tr>\n",
       "\t<tr><th scope=row>NtR27</th><td>plasmatocytes</td><td>NtR        </td><td> 1781</td><td>  488</td><td> 1016.9215</td><td>  18.334900</td><td>5.716857</td><td>57.11947</td></tr>\n",
       "\t<tr><th scope=row>Ance-527</th><td>plasmatocytes</td><td>Ance-5     </td><td> 1000</td><td>  232</td><td>  563.7018</td><td>   7.820584</td><td>5.997914</td><td>54.82900</td></tr>\n",
       "\t<tr><th scope=row>Plod27</th><td>plasmatocytes</td><td>Plod       </td><td>15175</td><td>11106</td><td> 8535.6323</td><td> 472.477755</td><td>4.172129</td><td>54.48572</td></tr>\n",
       "\t<tr><th scope=row>CG3105127</th><td>plasmatocytes</td><td>CG31051    </td><td> 1108</td><td>  268</td><td>  630.5397</td><td>  10.444006</td><td>5.783923</td><td>53.80627</td></tr>\n",
       "\t<tr><th scope=row>CG109228</th><td>plasmatocytes</td><td>CG1092     </td><td> 1260</td><td>  360</td><td>  719.0644</td><td>  14.628470</td><td>5.523872</td><td>52.43250</td></tr>\n",
       "\t<tr><th scope=row>Karl27</th><td>plasmatocytes</td><td>Karl       </td><td>  827</td><td>  192</td><td>  463.3877</td><td>   7.248140</td><td>5.812007</td><td>51.48965</td></tr>\n",
       "\t<tr><th scope=row>SPARC27</th><td>plasmatocytes</td><td>SPARC      </td><td> 9321</td><td> 7052</td><td> 5219.4311</td><td> 289.787244</td><td>4.165857</td><td>51.44814</td></tr>\n",
       "\t<tr><th scope=row>Tig27</th><td>plasmatocytes</td><td>Tig        </td><td>10503</td><td> 8881</td><td> 5871.8546</td><td> 372.153969</td><td>3.975973</td><td>49.77857</td></tr>\n",
       "\t<tr><th scope=row>fat-spondin27</th><td>plasmatocytes</td><td>fat-spondin</td><td> 3969</td><td> 2271</td><td> 2232.3905</td><td>  99.971451</td><td>4.466570</td><td>49.69068</td></tr>\n",
       "\t<tr><th scope=row>NimC427</th><td>plasmatocytes</td><td>NimC4      </td><td> 4559</td><td> 3200</td><td> 2644.8287</td><td> 129.002663</td><td>4.346561</td><td>49.41824</td></tr>\n",
       "\t<tr><th scope=row>CG631027</th><td>plasmatocytes</td><td>CG6310     </td><td> 1607</td><td>  649</td><td>  905.5086</td><td>  28.732764</td><td>4.928603</td><td>48.41947</td></tr>\n",
       "\t<tr><th scope=row>hoe127</th><td>plasmatocytes</td><td>hoe1       </td><td> 1636</td><td>  734</td><td>  888.7940</td><td>  27.978898</td><td>4.938774</td><td>48.38679</td></tr>\n",
       "\t<tr><th scope=row>firl27</th><td>plasmatocytes</td><td>firl       </td><td> 1326</td><td>  590</td><td>  766.3311</td><td>  23.721817</td><td>4.954111</td><td>47.47874</td></tr>\n",
       "\t<tr><th scope=row>nAChRbeta327</th><td>plasmatocytes</td><td>nAChRbeta3 </td><td>  839</td><td>  311</td><td>  487.9391</td><td>  12.209054</td><td>5.207102</td><td>46.51770</td></tr>\n",
       "\t<tr><th scope=row>Nepl1927</th><td>plasmatocytes</td><td>Nepl19     </td><td> 1180</td><td>  556</td><td>  648.0506</td><td>  22.540331</td><td>4.782900</td><td>44.68275</td></tr>\n",
       "\t<tr><th scope=row>LanA27</th><td>plasmatocytes</td><td>LanA       </td><td>23411</td><td>34640</td><td>13013.0184</td><td>1359.323334</td><td>3.257934</td><td>44.52872</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type & gene & cell.type.n.umi & other.n.umi & cell.type.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tPpn27 & plasmatocytes & Ppn         & 71774 & 14120 & 39945.5823 &  536.442522 & 6.215782 & 95.01310\\\\\n",
       "\tPxn27 & plasmatocytes & Pxn         & 12200 &  2357 &  6795.4350 &   90.346121 & 6.217079 & 79.14691\\\\\n",
       "\tCol4a127 & plasmatocytes & Col4a1      & 34974 & 11468 & 19506.1940 &  449.208424 & 5.437195 & 77.48938\\\\\n",
       "\tvkg27 & plasmatocytes & vkg         & 22254 &  7348 & 12342.0464 &  294.444644 & 5.384550 & 73.18363\\\\\n",
       "\tNtR27 & plasmatocytes & NtR         &  1781 &   488 &  1016.9215 &   18.334900 & 5.716857 & 57.11947\\\\\n",
       "\tAnce-527 & plasmatocytes & Ance-5      &  1000 &   232 &   563.7018 &    7.820584 & 5.997914 & 54.82900\\\\\n",
       "\tPlod27 & plasmatocytes & Plod        & 15175 & 11106 &  8535.6323 &  472.477755 & 4.172129 & 54.48572\\\\\n",
       "\tCG3105127 & plasmatocytes & CG31051     &  1108 &   268 &   630.5397 &   10.444006 & 5.783923 & 53.80627\\\\\n",
       "\tCG109228 & plasmatocytes & CG1092      &  1260 &   360 &   719.0644 &   14.628470 & 5.523872 & 52.43250\\\\\n",
       "\tKarl27 & plasmatocytes & Karl        &   827 &   192 &   463.3877 &    7.248140 & 5.812007 & 51.48965\\\\\n",
       "\tSPARC27 & plasmatocytes & SPARC       &  9321 &  7052 &  5219.4311 &  289.787244 & 4.165857 & 51.44814\\\\\n",
       "\tTig27 & plasmatocytes & Tig         & 10503 &  8881 &  5871.8546 &  372.153969 & 3.975973 & 49.77857\\\\\n",
       "\tfat-spondin27 & plasmatocytes & fat-spondin &  3969 &  2271 &  2232.3905 &   99.971451 & 4.466570 & 49.69068\\\\\n",
       "\tNimC427 & plasmatocytes & NimC4       &  4559 &  3200 &  2644.8287 &  129.002663 & 4.346561 & 49.41824\\\\\n",
       "\tCG631027 & plasmatocytes & CG6310      &  1607 &   649 &   905.5086 &   28.732764 & 4.928603 & 48.41947\\\\\n",
       "\thoe127 & plasmatocytes & hoe1        &  1636 &   734 &   888.7940 &   27.978898 & 4.938774 & 48.38679\\\\\n",
       "\tfirl27 & plasmatocytes & firl        &  1326 &   590 &   766.3311 &   23.721817 & 4.954111 & 47.47874\\\\\n",
       "\tnAChRbeta327 & plasmatocytes & nAChRbeta3  &   839 &   311 &   487.9391 &   12.209054 & 5.207102 & 46.51770\\\\\n",
       "\tNepl1927 & plasmatocytes & Nepl19      &  1180 &   556 &   648.0506 &   22.540331 & 4.782900 & 44.68275\\\\\n",
       "\tLanA27 & plasmatocytes & LanA        & 23411 & 34640 & 13013.0184 & 1359.323334 & 3.257934 & 44.52872\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type &lt;chr&gt; | gene &lt;chr&gt; | cell.type.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Ppn27 | plasmatocytes | Ppn         | 71774 | 14120 | 39945.5823 |  536.442522 | 6.215782 | 95.01310 |\n",
       "| Pxn27 | plasmatocytes | Pxn         | 12200 |  2357 |  6795.4350 |   90.346121 | 6.217079 | 79.14691 |\n",
       "| Col4a127 | plasmatocytes | Col4a1      | 34974 | 11468 | 19506.1940 |  449.208424 | 5.437195 | 77.48938 |\n",
       "| vkg27 | plasmatocytes | vkg         | 22254 |  7348 | 12342.0464 |  294.444644 | 5.384550 | 73.18363 |\n",
       "| NtR27 | plasmatocytes | NtR         |  1781 |   488 |  1016.9215 |   18.334900 | 5.716857 | 57.11947 |\n",
       "| Ance-527 | plasmatocytes | Ance-5      |  1000 |   232 |   563.7018 |    7.820584 | 5.997914 | 54.82900 |\n",
       "| Plod27 | plasmatocytes | Plod        | 15175 | 11106 |  8535.6323 |  472.477755 | 4.172129 | 54.48572 |\n",
       "| CG3105127 | plasmatocytes | CG31051     |  1108 |   268 |   630.5397 |   10.444006 | 5.783923 | 53.80627 |\n",
       "| CG109228 | plasmatocytes | CG1092      |  1260 |   360 |   719.0644 |   14.628470 | 5.523872 | 52.43250 |\n",
       "| Karl27 | plasmatocytes | Karl        |   827 |   192 |   463.3877 |    7.248140 | 5.812007 | 51.48965 |\n",
       "| SPARC27 | plasmatocytes | SPARC       |  9321 |  7052 |  5219.4311 |  289.787244 | 4.165857 | 51.44814 |\n",
       "| Tig27 | plasmatocytes | Tig         | 10503 |  8881 |  5871.8546 |  372.153969 | 3.975973 | 49.77857 |\n",
       "| fat-spondin27 | plasmatocytes | fat-spondin |  3969 |  2271 |  2232.3905 |   99.971451 | 4.466570 | 49.69068 |\n",
       "| NimC427 | plasmatocytes | NimC4       |  4559 |  3200 |  2644.8287 |  129.002663 | 4.346561 | 49.41824 |\n",
       "| CG631027 | plasmatocytes | CG6310      |  1607 |   649 |   905.5086 |   28.732764 | 4.928603 | 48.41947 |\n",
       "| hoe127 | plasmatocytes | hoe1        |  1636 |   734 |   888.7940 |   27.978898 | 4.938774 | 48.38679 |\n",
       "| firl27 | plasmatocytes | firl        |  1326 |   590 |   766.3311 |   23.721817 | 4.954111 | 47.47874 |\n",
       "| nAChRbeta327 | plasmatocytes | nAChRbeta3  |   839 |   311 |   487.9391 |   12.209054 | 5.207102 | 46.51770 |\n",
       "| Nepl1927 | plasmatocytes | Nepl19      |  1180 |   556 |   648.0506 |   22.540331 | 4.782900 | 44.68275 |\n",
       "| LanA27 | plasmatocytes | LanA        | 23411 | 34640 | 13013.0184 | 1359.323334 | 3.257934 | 44.52872 |\n",
       "\n"
      ],
      "text/plain": [
       "              cell.type     gene        cell.type.n.umi other.n.umi\n",
       "Ppn27         plasmatocytes Ppn         71774           14120      \n",
       "Pxn27         plasmatocytes Pxn         12200            2357      \n",
       "Col4a127      plasmatocytes Col4a1      34974           11468      \n",
       "vkg27         plasmatocytes vkg         22254            7348      \n",
       "NtR27         plasmatocytes NtR          1781             488      \n",
       "Ance-527      plasmatocytes Ance-5       1000             232      \n",
       "Plod27        plasmatocytes Plod        15175           11106      \n",
       "CG3105127     plasmatocytes CG31051      1108             268      \n",
       "CG109228      plasmatocytes CG1092       1260             360      \n",
       "Karl27        plasmatocytes Karl          827             192      \n",
       "SPARC27       plasmatocytes SPARC        9321            7052      \n",
       "Tig27         plasmatocytes Tig         10503            8881      \n",
       "fat-spondin27 plasmatocytes fat-spondin  3969            2271      \n",
       "NimC427       plasmatocytes NimC4        4559            3200      \n",
       "CG631027      plasmatocytes CG6310       1607             649      \n",
       "hoe127        plasmatocytes hoe1         1636             734      \n",
       "firl27        plasmatocytes firl         1326             590      \n",
       "nAChRbeta327  plasmatocytes nAChRbeta3    839             311      \n",
       "Nepl1927      plasmatocytes Nepl19       1180             556      \n",
       "LanA27        plasmatocytes LanA        23411           34640      \n",
       "              cell.type.tpm other.tpm   log2.ratio heuristic.score\n",
       "Ppn27         39945.5823     536.442522 6.215782   95.01310       \n",
       "Pxn27          6795.4350      90.346121 6.217079   79.14691       \n",
       "Col4a127      19506.1940     449.208424 5.437195   77.48938       \n",
       "vkg27         12342.0464     294.444644 5.384550   73.18363       \n",
       "NtR27          1016.9215      18.334900 5.716857   57.11947       \n",
       "Ance-527        563.7018       7.820584 5.997914   54.82900       \n",
       "Plod27         8535.6323     472.477755 4.172129   54.48572       \n",
       "CG3105127       630.5397      10.444006 5.783923   53.80627       \n",
       "CG109228        719.0644      14.628470 5.523872   52.43250       \n",
       "Karl27          463.3877       7.248140 5.812007   51.48965       \n",
       "SPARC27        5219.4311     289.787244 4.165857   51.44814       \n",
       "Tig27          5871.8546     372.153969 3.975973   49.77857       \n",
       "fat-spondin27  2232.3905      99.971451 4.466570   49.69068       \n",
       "NimC427        2644.8287     129.002663 4.346561   49.41824       \n",
       "CG631027        905.5086      28.732764 4.928603   48.41947       \n",
       "hoe127          888.7940      27.978898 4.938774   48.38679       \n",
       "firl27          766.3311      23.721817 4.954111   47.47874       \n",
       "nAChRbeta327    487.9391      12.209054 5.207102   46.51770       \n",
       "Nepl1927        648.0506      22.540331 4.782900   44.68275       \n",
       "LanA27        13013.0184    1359.323334 3.257934   44.52872       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.markers.sub300_no_singletons.cell.type %>% filter(cell.type.n.umi > 100 & \n",
    "                                                            cell.type == \"plasmatocytes\")  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1279,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cluster</th><th scope=col>annot</th><th scope=col>n</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>103</td><td>Malpighian tubule</td><td>456</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>163</td><td>Malpighian tubule</td><td>200</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>144</td><td>Malpighian tubule</td><td> 20</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>55 </td><td>Malpighian tubule</td><td> 12</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>8  </td><td>Malpighian tubule</td><td> 10</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>17 </td><td>Malpighian tubule</td><td> 10</td></tr>\n",
       "\t<tr><th scope=row>7</th><td>143</td><td>Malpighian tubule</td><td>  9</td></tr>\n",
       "\t<tr><th scope=row>8</th><td>54 </td><td>Malpighian tubule</td><td>  8</td></tr>\n",
       "\t<tr><th scope=row>9</th><td>106</td><td>Malpighian tubule</td><td>  8</td></tr>\n",
       "\t<tr><th scope=row>10</th><td>12 </td><td>Malpighian tubule</td><td>  7</td></tr>\n",
       "\t<tr><th scope=row>11</th><td>49 </td><td>Malpighian tubule</td><td>  7</td></tr>\n",
       "\t<tr><th scope=row>12</th><td>25 </td><td>Malpighian tubule</td><td>  6</td></tr>\n",
       "\t<tr><th scope=row>13</th><td>155</td><td>Malpighian tubule</td><td>  6</td></tr>\n",
       "\t<tr><th scope=row>14</th><td>31 </td><td>Malpighian tubule</td><td>  5</td></tr>\n",
       "\t<tr><th scope=row>15</th><td>80 </td><td>Malpighian tubule</td><td>  5</td></tr>\n",
       "\t<tr><th scope=row>16</th><td>91 </td><td>Malpighian tubule</td><td>  5</td></tr>\n",
       "\t<tr><th scope=row>17</th><td>84 </td><td>Malpighian tubule</td><td>  3</td></tr>\n",
       "\t<tr><th scope=row>18</th><td>85 </td><td>Malpighian tubule</td><td>  3</td></tr>\n",
       "\t<tr><th scope=row>19</th><td>109</td><td>Malpighian tubule</td><td>  3</td></tr>\n",
       "\t<tr><th scope=row>20</th><td>142</td><td>Malpighian tubule</td><td>  3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & cluster & annot & n\\\\\n",
       "  & <fct> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 103 & Malpighian tubule & 456\\\\\n",
       "\t2 & 163 & Malpighian tubule & 200\\\\\n",
       "\t3 & 144 & Malpighian tubule &  20\\\\\n",
       "\t4 & 55  & Malpighian tubule &  12\\\\\n",
       "\t5 & 8   & Malpighian tubule &  10\\\\\n",
       "\t6 & 17  & Malpighian tubule &  10\\\\\n",
       "\t7 & 143 & Malpighian tubule &   9\\\\\n",
       "\t8 & 54  & Malpighian tubule &   8\\\\\n",
       "\t9 & 106 & Malpighian tubule &   8\\\\\n",
       "\t10 & 12  & Malpighian tubule &   7\\\\\n",
       "\t11 & 49  & Malpighian tubule &   7\\\\\n",
       "\t12 & 25  & Malpighian tubule &   6\\\\\n",
       "\t13 & 155 & Malpighian tubule &   6\\\\\n",
       "\t14 & 31  & Malpighian tubule &   5\\\\\n",
       "\t15 & 80  & Malpighian tubule &   5\\\\\n",
       "\t16 & 91  & Malpighian tubule &   5\\\\\n",
       "\t17 & 84  & Malpighian tubule &   3\\\\\n",
       "\t18 & 85  & Malpighian tubule &   3\\\\\n",
       "\t19 & 109 & Malpighian tubule &   3\\\\\n",
       "\t20 & 142 & Malpighian tubule &   3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 3\n",
       "\n",
       "| <!--/--> | cluster &lt;fct&gt; | annot &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | 103 | Malpighian tubule | 456 |\n",
       "| 2 | 163 | Malpighian tubule | 200 |\n",
       "| 3 | 144 | Malpighian tubule |  20 |\n",
       "| 4 | 55  | Malpighian tubule |  12 |\n",
       "| 5 | 8   | Malpighian tubule |  10 |\n",
       "| 6 | 17  | Malpighian tubule |  10 |\n",
       "| 7 | 143 | Malpighian tubule |   9 |\n",
       "| 8 | 54  | Malpighian tubule |   8 |\n",
       "| 9 | 106 | Malpighian tubule |   8 |\n",
       "| 10 | 12  | Malpighian tubule |   7 |\n",
       "| 11 | 49  | Malpighian tubule |   7 |\n",
       "| 12 | 25  | Malpighian tubule |   6 |\n",
       "| 13 | 155 | Malpighian tubule |   6 |\n",
       "| 14 | 31  | Malpighian tubule |   5 |\n",
       "| 15 | 80  | Malpighian tubule |   5 |\n",
       "| 16 | 91  | Malpighian tubule |   5 |\n",
       "| 17 | 84  | Malpighian tubule |   3 |\n",
       "| 18 | 85  | Malpighian tubule |   3 |\n",
       "| 19 | 109 | Malpighian tubule |   3 |\n",
       "| 20 | 142 | Malpighian tubule |   3 |\n",
       "\n"
      ],
      "text/plain": [
       "   cluster annot             n  \n",
       "1  103     Malpighian tubule 456\n",
       "2  163     Malpighian tubule 200\n",
       "3  144     Malpighian tubule  20\n",
       "4  55      Malpighian tubule  12\n",
       "5  8       Malpighian tubule  10\n",
       "6  17      Malpighian tubule  10\n",
       "7  143     Malpighian tubule   9\n",
       "8  54      Malpighian tubule   8\n",
       "9  106     Malpighian tubule   8\n",
       "10 12      Malpighian tubule   7\n",
       "11 49      Malpighian tubule   7\n",
       "12 25      Malpighian tubule   6\n",
       "13 155     Malpighian tubule   6\n",
       "14 31      Malpighian tubule   5\n",
       "15 80      Malpighian tubule   5\n",
       "16 91      Malpighian tubule   5\n",
       "17 84      Malpighian tubule   3\n",
       "18 85      Malpighian tubule   3\n",
       "19 109     Malpighian tubule   3\n",
       "20 142     Malpighian tubule   3"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub_no_singletons  %>% count(cluster, annot)  %>% \n",
    "filter(annot == \"Malpighian tubule\")  %>% arrange(-n)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1280,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cluster</th><th scope=col>annot</th><th scope=col>n</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>103</td><td>Malpighian tubule</td><td>469</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>163</td><td>Malpighian tubule</td><td>206</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>144</td><td>Malpighian tubule</td><td> 23</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>12 </td><td>Malpighian tubule</td><td> 20</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>55 </td><td>Malpighian tubule</td><td> 20</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>49 </td><td>Malpighian tubule</td><td> 17</td></tr>\n",
       "\t<tr><th scope=row>7</th><td>5  </td><td>Malpighian tubule</td><td> 16</td></tr>\n",
       "\t<tr><th scope=row>8</th><td>17 </td><td>Malpighian tubule</td><td> 16</td></tr>\n",
       "\t<tr><th scope=row>9</th><td>8  </td><td>Malpighian tubule</td><td> 15</td></tr>\n",
       "\t<tr><th scope=row>10</th><td>143</td><td>Malpighian tubule</td><td> 15</td></tr>\n",
       "\t<tr><th scope=row>11</th><td>54 </td><td>Malpighian tubule</td><td> 13</td></tr>\n",
       "\t<tr><th scope=row>12</th><td>15 </td><td>Malpighian tubule</td><td> 10</td></tr>\n",
       "\t<tr><th scope=row>13</th><td>80 </td><td>Malpighian tubule</td><td> 10</td></tr>\n",
       "\t<tr><th scope=row>14</th><td>106</td><td>Malpighian tubule</td><td>  9</td></tr>\n",
       "\t<tr><th scope=row>15</th><td>25 </td><td>Malpighian tubule</td><td>  8</td></tr>\n",
       "\t<tr><th scope=row>16</th><td>31 </td><td>Malpighian tubule</td><td>  8</td></tr>\n",
       "\t<tr><th scope=row>17</th><td>21 </td><td>Malpighian tubule</td><td>  7</td></tr>\n",
       "\t<tr><th scope=row>18</th><td>85 </td><td>Malpighian tubule</td><td>  7</td></tr>\n",
       "\t<tr><th scope=row>19</th><td>91 </td><td>Malpighian tubule</td><td>  7</td></tr>\n",
       "\t<tr><th scope=row>20</th><td>150</td><td>Malpighian tubule</td><td>  7</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & cluster & annot & n\\\\\n",
       "  & <fct> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 103 & Malpighian tubule & 469\\\\\n",
       "\t2 & 163 & Malpighian tubule & 206\\\\\n",
       "\t3 & 144 & Malpighian tubule &  23\\\\\n",
       "\t4 & 12  & Malpighian tubule &  20\\\\\n",
       "\t5 & 55  & Malpighian tubule &  20\\\\\n",
       "\t6 & 49  & Malpighian tubule &  17\\\\\n",
       "\t7 & 5   & Malpighian tubule &  16\\\\\n",
       "\t8 & 17  & Malpighian tubule &  16\\\\\n",
       "\t9 & 8   & Malpighian tubule &  15\\\\\n",
       "\t10 & 143 & Malpighian tubule &  15\\\\\n",
       "\t11 & 54  & Malpighian tubule &  13\\\\\n",
       "\t12 & 15  & Malpighian tubule &  10\\\\\n",
       "\t13 & 80  & Malpighian tubule &  10\\\\\n",
       "\t14 & 106 & Malpighian tubule &   9\\\\\n",
       "\t15 & 25  & Malpighian tubule &   8\\\\\n",
       "\t16 & 31  & Malpighian tubule &   8\\\\\n",
       "\t17 & 21  & Malpighian tubule &   7\\\\\n",
       "\t18 & 85  & Malpighian tubule &   7\\\\\n",
       "\t19 & 91  & Malpighian tubule &   7\\\\\n",
       "\t20 & 150 & Malpighian tubule &   7\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 3\n",
       "\n",
       "| <!--/--> | cluster &lt;fct&gt; | annot &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | 103 | Malpighian tubule | 469 |\n",
       "| 2 | 163 | Malpighian tubule | 206 |\n",
       "| 3 | 144 | Malpighian tubule |  23 |\n",
       "| 4 | 12  | Malpighian tubule |  20 |\n",
       "| 5 | 55  | Malpighian tubule |  20 |\n",
       "| 6 | 49  | Malpighian tubule |  17 |\n",
       "| 7 | 5   | Malpighian tubule |  16 |\n",
       "| 8 | 17  | Malpighian tubule |  16 |\n",
       "| 9 | 8   | Malpighian tubule |  15 |\n",
       "| 10 | 143 | Malpighian tubule |  15 |\n",
       "| 11 | 54  | Malpighian tubule |  13 |\n",
       "| 12 | 15  | Malpighian tubule |  10 |\n",
       "| 13 | 80  | Malpighian tubule |  10 |\n",
       "| 14 | 106 | Malpighian tubule |   9 |\n",
       "| 15 | 25  | Malpighian tubule |   8 |\n",
       "| 16 | 31  | Malpighian tubule |   8 |\n",
       "| 17 | 21  | Malpighian tubule |   7 |\n",
       "| 18 | 85  | Malpighian tubule |   7 |\n",
       "| 19 | 91  | Malpighian tubule |   7 |\n",
       "| 20 | 150 | Malpighian tubule |   7 |\n",
       "\n"
      ],
      "text/plain": [
       "   cluster annot             n  \n",
       "1  103     Malpighian tubule 469\n",
       "2  163     Malpighian tubule 206\n",
       "3  144     Malpighian tubule  23\n",
       "4  12      Malpighian tubule  20\n",
       "5  55      Malpighian tubule  20\n",
       "6  49      Malpighian tubule  17\n",
       "7  5       Malpighian tubule  16\n",
       "8  17      Malpighian tubule  16\n",
       "9  8       Malpighian tubule  15\n",
       "10 143     Malpighian tubule  15\n",
       "11 54      Malpighian tubule  13\n",
       "12 15      Malpighian tubule  10\n",
       "13 80      Malpighian tubule  10\n",
       "14 106     Malpighian tubule   9\n",
       "15 25      Malpighian tubule   8\n",
       "16 31      Malpighian tubule   8\n",
       "17 21      Malpighian tubule   7\n",
       "18 85      Malpighian tubule   7\n",
       "19 91      Malpighian tubule   7\n",
       "20 150     Malpighian tubule   7"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub300  %>% count(cluster, annot)  %>% \n",
    "filter(annot == \"Malpighian tubule\")  %>% arrange(-n)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "#  looking at specific cluster\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = !(df_sub300$cell %in% doubletons) & df_sub300$annot == \"Malpighian tubule\"  & df_sub300$cluster == 163), \n",
    "           size = ifelse( !(df_sub300$cell %in% doubletons) & df_sub300$annot == \"Malpighian tubule\", 2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Malpighian tubule not doubletons cluster 163\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "#cell.type = \"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = !(df_sub300$cell %in% doubletons) & df_sub300$annot == \"Malpighian tubule\" ), \n",
    "           size = ifelse( !(df_sub300$cell %in% doubletons) & df_sub300$annot == \"Malpighian tubule\", 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Malpighian tubule not doubletons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'visceral muscle'"
      ],
      "text/latex": [
       "'visceral muscle'"
      ],
      "text/markdown": [
       "'visceral muscle'"
      ],
      "text/plain": [
       "[1] \"visceral muscle\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "\"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = !(df_sub300$cell %in% singletons) & df_sub300$annot == \"Malpighian tubule\" ), \n",
    "           size = ifelse( !(df_sub300$cell %in% singletons) & df_sub300$annot == \"Malpighian tubule\", 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Malpighian tubule not singletons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot_cells(cds_sub_no_sing, genes = \"Muc11A\"): could not find function \"plot_cells\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot_cells(cds_sub_no_sing, genes = \"Muc11A\"): could not find function \"plot_cells\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot_cells(cds_sub_no_sing, genes = \"Muc11A\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub, c(\"Muc11A\"), size_pos = 3, size_neg = 0.06, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub, c(\"Muc11A\"), size_pos = 3, size_neg = 0.06, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub, c(\"Muc11A\"), size_pos = 3, size_neg = 0.06, stroke = 0.02, limit = 0.01)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_sing, c(\"Muc11A\"), size_pos = 3, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_sing, c(\"Muc11A\"), size_pos = 3, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_sing, c(\"Muc11A\"), size_pos = 3, size_neg = 0.06, stroke = 0.02, limit = 0.01)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Remove both singletons and doubletons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "cds_sub_no_doubletons = cds_sub_no_sing[, !(df_sub_no_singletons$cell %in% doubletons)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Muc11A\"), size_pos = 3, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Muc11A\"), size_pos = 3, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Muc11A\"), size_pos = 3, size_neg = 0.2, stroke = 0.02, limit = 0.01)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = \n",
    "               df_sub300$annot == \"Malpighian tubule\" ), \n",
    "           size = ifelse( \n",
    "                         df_sub300$annot == \"Malpighian tubule\", 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### try muscle as another example"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "#cell.type = \"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = !(df_sub300$cell %in% doubletons) & df_sub300$annot == \"muscle\"  & df_sub300$cluster == 163), \n",
    "           size = ifelse( !(df_sub300$cell %in% doubletons) & df_sub300$annot == \"muscle\", 2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "#cell.type = \"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = !(df_sub300$cell %in% doubletons) & df_sub300$annot == \"muscle\" ), \n",
    "           size = ifelse( !(df_sub300$cell %in% doubletons) & df_sub300$annot == \"muscle\", 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'visceral muscle'"
      ],
      "text/latex": [
       "'visceral muscle'"
      ],
      "text/markdown": [
       "'visceral muscle'"
      ],
      "text/plain": [
       "[1] \"visceral muscle\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "\"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = !(df_sub300$cell %in% singletons) & df_sub300$annot == \"muscle\" ), \n",
    "           size = ifelse( !(df_sub300$cell %in% singletons) & df_sub300$annot == \"muscle\", 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = !(df_sub300$cell %in% singletons) & !(df_sub300$cell %in% doubletons) &\n",
    "               df_sub300$annot == \"muscle\" ), \n",
    "           size = ifelse( !(df_sub300$cell %in% singletons) & !(df_sub300$cell %in% doubletons) &\n",
    "                         df_sub300$annot == \"muscle\", 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sub_no_doubletons = make_df(cds_sub_no_doubletons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1341,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finding markers for cluster 1\n",
      "\n",
      "# of cells sampled in set 1: 2401\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 2\n",
      "\n",
      "# of cells sampled in set 1: 2045\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 3\n",
      "\n",
      "# of cells sampled in set 1: 1067\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 4\n",
      "\n",
      "# of cells sampled in set 1: 1806\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 5\n",
      "\n",
      "# of cells sampled in set 1: 2079\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 6\n",
      "\n",
      "# of cells sampled in set 1: 1871\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 7\n",
      "\n",
      "# of cells sampled in set 1: 1770\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 8\n",
      "\n",
      "# of cells sampled in set 1: 1895\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 9\n",
      "\n",
      "# of cells sampled in set 1: 2148\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 10\n",
      "\n",
      "# of cells sampled in set 1: 1960\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 11\n",
      "\n",
      "# of cells sampled in set 1: 1297\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 12\n",
      "\n",
      "# of cells sampled in set 1: 1828\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 13\n",
      "\n",
      "# of cells sampled in set 1: 1872\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 14\n",
      "\n",
      "# of cells sampled in set 1: 1851\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 15\n",
      "\n",
      "# of cells sampled in set 1: 1832\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 16\n",
      "\n",
      "# of cells sampled in set 1: 1593\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 17\n",
      "\n",
      "# of cells sampled in set 1: 1742\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 18\n",
      "\n",
      "# of cells sampled in set 1: 1864\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 19\n",
      "\n",
      "# of cells sampled in set 1: 1413\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 20\n",
      "\n",
      "# of cells sampled in set 1: 1895\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 21\n",
      "\n",
      "# of cells sampled in set 1: 1601\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 22\n",
      "\n",
      "# of cells sampled in set 1: 1301\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 23\n",
      "\n",
      "# of cells sampled in set 1: 1631\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 24\n",
      "\n",
      "# of cells sampled in set 1: 1436\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 25\n",
      "\n",
      "# of cells sampled in set 1: 1660\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 26\n",
      "\n",
      "# of cells sampled in set 1: 1496\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 27\n",
      "\n",
      "# of cells sampled in set 1: 1162\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 28\n",
      "\n",
      "# of cells sampled in set 1: 1714\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 29\n",
      "\n",
      "# of cells sampled in set 1: 1460\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 30\n",
      "\n",
      "# of cells sampled in set 1: 1419\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 31\n",
      "\n",
      "# of cells sampled in set 1: 1637\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 32\n",
      "\n",
      "# of cells sampled in set 1: 1555\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 33\n",
      "\n",
      "# of cells sampled in set 1: 1266\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 34\n",
      "\n",
      "# of cells sampled in set 1: 1716\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 35\n",
      "\n",
      "# of cells sampled in set 1: 1321\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 36\n",
      "\n",
      "# of cells sampled in set 1: 1476\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 37\n",
      "\n",
      "# of cells sampled in set 1: 1401\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 38\n",
      "\n",
      "# of cells sampled in set 1: 1405\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 39\n",
      "\n",
      "# of cells sampled in set 1: 1616\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 40\n",
      "\n",
      "# of cells sampled in set 1: 1502\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 41\n",
      "\n",
      "# of cells sampled in set 1: 1493\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 42\n",
      "\n",
      "# of cells sampled in set 1: 1220\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 43\n",
      "\n",
      "# of cells sampled in set 1: 1436\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 44\n",
      "\n",
      "# of cells sampled in set 1: 1465\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 45\n",
      "\n",
      "# of cells sampled in set 1: 1580\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 46\n",
      "\n",
      "# of cells sampled in set 1: 1520\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 47\n",
      "\n",
      "# of cells sampled in set 1: 1276\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 48\n",
      "\n",
      "# of cells sampled in set 1: 1148\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 49\n",
      "\n",
      "# of cells sampled in set 1: 1405\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 50\n",
      "\n",
      "# of cells sampled in set 1: 1323\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 51\n",
      "\n",
      "# of cells sampled in set 1: 1611\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 52\n",
      "\n",
      "# of cells sampled in set 1: 1396\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 53\n",
      "\n",
      "# of cells sampled in set 1: 1142\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 54\n",
      "\n",
      "# of cells sampled in set 1: 1349\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 55\n",
      "\n",
      "# of cells sampled in set 1: 1303\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 56\n",
      "\n",
      "# of cells sampled in set 1: 1538\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 57\n",
      "\n",
      "# of cells sampled in set 1: 1303\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 58\n",
      "\n",
      "# of cells sampled in set 1: 1558\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 59\n",
      "\n",
      "# of cells sampled in set 1: 952\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 60\n",
      "\n",
      "# of cells sampled in set 1: 1438\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 61\n",
      "\n",
      "# of cells sampled in set 1: 917\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 62\n",
      "\n",
      "# of cells sampled in set 1: 1177\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 63\n",
      "\n",
      "# of cells sampled in set 1: 1309\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 64\n",
      "\n",
      "# of cells sampled in set 1: 1321\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 65\n",
      "\n",
      "# of cells sampled in set 1: 1327\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 66\n",
      "\n",
      "# of cells sampled in set 1: 1325\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 67\n",
      "\n",
      "# of cells sampled in set 1: 1308\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 68\n",
      "\n",
      "# of cells sampled in set 1: 1219\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 69\n",
      "\n",
      "# of cells sampled in set 1: 1099\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 70\n",
      "\n",
      "# of cells sampled in set 1: 1274\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 71\n",
      "\n",
      "# of cells sampled in set 1: 1180\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 72\n",
      "\n",
      "# of cells sampled in set 1: 1178\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 73\n",
      "\n",
      "# of cells sampled in set 1: 1397\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 74\n",
      "\n",
      "# of cells sampled in set 1: 936\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 75\n",
      "\n",
      "# of cells sampled in set 1: 1136\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 76\n",
      "\n",
      "# of cells sampled in set 1: 669\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 77\n",
      "\n",
      "# of cells sampled in set 1: 1190\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 78\n",
      "\n",
      "# of cells sampled in set 1: 942\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 79\n",
      "\n",
      "# of cells sampled in set 1: 1047\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 80\n",
      "\n",
      "# of cells sampled in set 1: 1247\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finding markers for cluster 81\n",
      "\n",
      "# of cells sampled in set 1: 1277\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 82\n",
      "\n",
      "# of cells sampled in set 1: 1180\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 83\n",
      "\n",
      "# of cells sampled in set 1: 987\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 84\n",
      "\n",
      "# of cells sampled in set 1: 1137\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 85\n",
      "\n",
      "# of cells sampled in set 1: 1188\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 86\n",
      "\n",
      "# of cells sampled in set 1: 1131\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 87\n",
      "\n",
      "# of cells sampled in set 1: 1017\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 88\n",
      "\n",
      "# of cells sampled in set 1: 1100\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 89\n",
      "\n",
      "# of cells sampled in set 1: 1102\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 90\n",
      "\n",
      "# of cells sampled in set 1: 952\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 91\n",
      "\n",
      "# of cells sampled in set 1: 1142\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 92\n",
      "\n",
      "# of cells sampled in set 1: 1134\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 93\n",
      "\n",
      "# of cells sampled in set 1: 992\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 94\n",
      "\n",
      "# of cells sampled in set 1: 1103\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 95\n",
      "\n",
      "# of cells sampled in set 1: 1317\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 96\n",
      "\n",
      "# of cells sampled in set 1: 1181\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 97\n",
      "\n",
      "# of cells sampled in set 1: 1158\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 98\n",
      "\n",
      "# of cells sampled in set 1: 1294\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 99\n",
      "\n",
      "# of cells sampled in set 1: 1177\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 100\n",
      "\n",
      "# of cells sampled in set 1: 742\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 101\n",
      "\n",
      "# of cells sampled in set 1: 857\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 102\n",
      "\n",
      "# of cells sampled in set 1: 876\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 103\n",
      "\n",
      "# of cells sampled in set 1: 1013\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 104\n",
      "\n",
      "# of cells sampled in set 1: 1144\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 105\n",
      "\n",
      "# of cells sampled in set 1: 913\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 106\n",
      "\n",
      "# of cells sampled in set 1: 1040\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 107\n",
      "\n",
      "# of cells sampled in set 1: 940\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 108\n",
      "\n",
      "# of cells sampled in set 1: 1070\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 109\n",
      "\n",
      "# of cells sampled in set 1: 1048\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 110\n",
      "\n",
      "# of cells sampled in set 1: 1049\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 111\n",
      "\n",
      "# of cells sampled in set 1: 1098\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 112\n",
      "\n",
      "# of cells sampled in set 1: 806\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 113\n",
      "\n",
      "# of cells sampled in set 1: 1063\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 114\n",
      "\n",
      "# of cells sampled in set 1: 931\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 115\n",
      "\n",
      "# of cells sampled in set 1: 1111\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 116\n",
      "\n",
      "# of cells sampled in set 1: 819\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 117\n",
      "\n",
      "# of cells sampled in set 1: 911\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 118\n",
      "\n",
      "# of cells sampled in set 1: 958\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 119\n",
      "\n",
      "# of cells sampled in set 1: 770\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 120\n",
      "\n",
      "# of cells sampled in set 1: 986\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 121\n",
      "\n",
      "# of cells sampled in set 1: 979\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 122\n",
      "\n",
      "# of cells sampled in set 1: 1094\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 123\n",
      "\n",
      "# of cells sampled in set 1: 1028\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 124\n",
      "\n",
      "# of cells sampled in set 1: 979\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 125\n",
      "\n",
      "# of cells sampled in set 1: 967\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 126\n",
      "\n",
      "# of cells sampled in set 1: 977\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 127\n",
      "\n",
      "# of cells sampled in set 1: 797\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 128\n",
      "\n",
      "# of cells sampled in set 1: 910\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 129\n",
      "\n",
      "# of cells sampled in set 1: 964\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 130\n",
      "\n",
      "# of cells sampled in set 1: 814\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 131\n",
      "\n",
      "# of cells sampled in set 1: 831\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 132\n",
      "\n",
      "# of cells sampled in set 1: 853\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 133\n",
      "\n",
      "# of cells sampled in set 1: 755\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 134\n",
      "\n",
      "# of cells sampled in set 1: 921\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 135\n",
      "\n",
      "# of cells sampled in set 1: 787\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 136\n",
      "\n",
      "# of cells sampled in set 1: 647\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 137\n",
      "\n",
      "# of cells sampled in set 1: 898\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 138\n",
      "\n",
      "# of cells sampled in set 1: 826\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 139\n",
      "\n",
      "# of cells sampled in set 1: 925\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 140\n",
      "\n",
      "# of cells sampled in set 1: 845\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 141\n",
      "\n",
      "# of cells sampled in set 1: 842\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 142\n",
      "\n",
      "# of cells sampled in set 1: 846\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 143\n",
      "\n",
      "# of cells sampled in set 1: 681\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 144\n",
      "\n",
      "# of cells sampled in set 1: 695\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 145\n",
      "\n",
      "# of cells sampled in set 1: 798\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 146\n",
      "\n",
      "# of cells sampled in set 1: 619\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 147\n",
      "\n",
      "# of cells sampled in set 1: 687\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 148\n",
      "\n",
      "# of cells sampled in set 1: 652\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 149\n",
      "\n",
      "# of cells sampled in set 1: 622\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 150\n",
      "\n",
      "# of cells sampled in set 1: 564\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 151\n",
      "\n",
      "# of cells sampled in set 1: 570\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 152\n",
      "\n",
      "# of cells sampled in set 1: 535\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 153\n",
      "\n",
      "# of cells sampled in set 1: 446\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 154\n",
      "\n",
      "# of cells sampled in set 1: 573\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 155\n",
      "\n",
      "# of cells sampled in set 1: 441\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 156\n",
      "\n",
      "# of cells sampled in set 1: 519\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 157\n",
      "\n",
      "# of cells sampled in set 1: 490\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 158\n",
      "\n",
      "# of cells sampled in set 1: 511\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 159\n",
      "\n",
      "# of cells sampled in set 1: 514\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 160\n",
      "\n",
      "# of cells sampled in set 1: 374\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 161\n",
      "\n",
      "# of cells sampled in set 1: 457\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 162\n",
      "\n",
      "# of cells sampled in set 1: 361\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 163\n",
      "\n",
      "# of cells sampled in set 1: 382\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 164\n",
      "\n",
      "# of cells sampled in set 1: 304\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 165\n",
      "\n",
      "# of cells sampled in set 1: 356\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 166\n",
      "\n",
      "# of cells sampled in set 1: 285\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 167\n",
      "\n",
      "# of cells sampled in set 1: 318\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 168\n",
      "\n",
      "# of cells sampled in set 1: 170\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 169\n",
      "\n",
      "# of cells sampled in set 1: 171\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 170\n",
      "\n",
      "# of cells sampled in set 1: 117\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 171\n",
      "\n",
      "# of cells sampled in set 1: 119\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 172\n",
      "\n",
      "# of cells sampled in set 1: 89\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 173\n",
      "\n",
      "# of cells sampled in set 1: 91\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cluster 174\n",
      "\n",
      "# of cells sampled in set 1: 50\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    }
   ],
   "source": [
    "DEG.results.sub300_no_doubletons = lapply(sort((unique(df_sub_no_doubletons$cluster))),\n",
    "                     function(this.cluster) {\n",
    "    message(\"Finding markers for cluster \", this.cluster)\n",
    "    cbind(\n",
    "        two.set.differential.gene.test.3a(\n",
    "            cds_sub_no_doubletons,\n",
    "            df_sub_no_doubletons$cluster == this.cluster,\n",
    "            df_sub_no_doubletons$cluster != this.cluster,\n",
    "            sample.n.max = 50000),\n",
    "        data.frame(cluster = this.cluster))\n",
    "                         })"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1342,
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster.markers.sub300_no_doubletons = do.call(rbind, DEG.results.sub300_no_doubletons) %>% filter(higher.expr == \"Set 1\") %>%\n",
    "    select(\n",
    "        cluster = cluster, gene, \n",
    "        cluster.n.umi = set.1.n.umi,\n",
    "        other.n.umi = set.2.n.umi,\n",
    "        cluster.tpm = set.1.tpm,\n",
    "        other.tpm = set.2.tpm,\n",
    "        log2.ratio, heuristic.score) %>%\n",
    "    arrange(-heuristic.score)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1343,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cluster</th><th scope=col>gene</th><th scope=col>cluster.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cluster.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Muc26B159</th><td>160</td><td>Muc26B </td><td>2947</td><td> 71</td><td>11399.564</td><td> 2.5494796</td><td>11.64908</td><td>156.9926</td></tr>\n",
       "\t<tr><th scope=row>Muc11A162</th><td>163</td><td>Muc11A </td><td>3780</td><td> 45</td><td> 7151.767</td><td> 0.9844656</td><td>11.81533</td><td>151.2869</td></tr>\n",
       "\t<tr><th scope=row>CG7409171</th><td>172</td><td>CG7409 </td><td> 453</td><td> 20</td><td> 4089.077</td><td> 0.2765578</td><td>11.64530</td><td>139.7193</td></tr>\n",
       "\t<tr><th scope=row>CG14421150</th><td>151</td><td>CG14421</td><td>3521</td><td>192</td><td> 8224.650</td><td> 4.0385755</td><td>10.67272</td><td>138.8085</td></tr>\n",
       "\t<tr><th scope=row>Alp8162</th><td>163</td><td>Alp8   </td><td>3709</td><td> 27</td><td> 5261.148</td><td> 1.1925089</td><td>11.22858</td><td>138.8014</td></tr>\n",
       "\t<tr><th scope=row>CG7017159</th><td>160</td><td>CG7017 </td><td>2812</td><td>241</td><td>12742.012</td><td>10.3560468</td><td>10.13192</td><td>138.1732</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cluster & gene & cluster.n.umi & other.n.umi & cluster.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <fct> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tMuc26B159 & 160 & Muc26B  & 2947 &  71 & 11399.564 &  2.5494796 & 11.64908 & 156.9926\\\\\n",
       "\tMuc11A162 & 163 & Muc11A  & 3780 &  45 &  7151.767 &  0.9844656 & 11.81533 & 151.2869\\\\\n",
       "\tCG7409171 & 172 & CG7409  &  453 &  20 &  4089.077 &  0.2765578 & 11.64530 & 139.7193\\\\\n",
       "\tCG14421150 & 151 & CG14421 & 3521 & 192 &  8224.650 &  4.0385755 & 10.67272 & 138.8085\\\\\n",
       "\tAlp8162 & 163 & Alp8    & 3709 &  27 &  5261.148 &  1.1925089 & 11.22858 & 138.8014\\\\\n",
       "\tCG7017159 & 160 & CG7017  & 2812 & 241 & 12742.012 & 10.3560468 & 10.13192 & 138.1732\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 8\n",
       "\n",
       "| <!--/--> | cluster &lt;fct&gt; | gene &lt;chr&gt; | cluster.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cluster.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Muc26B159 | 160 | Muc26B  | 2947 |  71 | 11399.564 |  2.5494796 | 11.64908 | 156.9926 |\n",
       "| Muc11A162 | 163 | Muc11A  | 3780 |  45 |  7151.767 |  0.9844656 | 11.81533 | 151.2869 |\n",
       "| CG7409171 | 172 | CG7409  |  453 |  20 |  4089.077 |  0.2765578 | 11.64530 | 139.7193 |\n",
       "| CG14421150 | 151 | CG14421 | 3521 | 192 |  8224.650 |  4.0385755 | 10.67272 | 138.8085 |\n",
       "| Alp8162 | 163 | Alp8    | 3709 |  27 |  5261.148 |  1.1925089 | 11.22858 | 138.8014 |\n",
       "| CG7017159 | 160 | CG7017  | 2812 | 241 | 12742.012 | 10.3560468 | 10.13192 | 138.1732 |\n",
       "\n"
      ],
      "text/plain": [
       "           cluster gene    cluster.n.umi other.n.umi cluster.tpm other.tpm \n",
       "Muc26B159  160     Muc26B  2947           71         11399.564    2.5494796\n",
       "Muc11A162  163     Muc11A  3780           45          7151.767    0.9844656\n",
       "CG7409171  172     CG7409   453           20          4089.077    0.2765578\n",
       "CG14421150 151     CG14421 3521          192          8224.650    4.0385755\n",
       "Alp8162    163     Alp8    3709           27          5261.148    1.1925089\n",
       "CG7017159  160     CG7017  2812          241         12742.012   10.3560468\n",
       "           log2.ratio heuristic.score\n",
       "Muc26B159  11.64908   156.9926       \n",
       "Muc11A162  11.81533   151.2869       \n",
       "CG7409171  11.64530   139.7193       \n",
       "CG14421150 10.67272   138.8085       \n",
       "Alp8162    11.22858   138.8014       \n",
       "CG7017159  10.13192   138.1732       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cluster.markers.sub300_no_doubletons  %>%  head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>191418</li><li>32</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 191418\n",
       "\\item 32\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 191418\n",
       "2. 32\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 191418     32"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(df_sub_no_doubletons)  # using pca 300 cds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>13938</li><li>191418</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 13938\n",
       "\\item 191418\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 13938\n",
       "2. 191418\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1]  13938 191418"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(cds_sub_no_doubletons)  # "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1346,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>13938</li><li>211498</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 13938\n",
       "\\item 211498\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 13938\n",
       "2. 211498\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1]  13938 211498"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(cds_sub_no_sing)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "20080"
      ],
      "text/latex": [
       "20080"
      ],
      "text/markdown": [
       "20080"
      ],
      "text/plain": [
       "[1] 20080"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "26947"
      ],
      "text/latex": [
       "26947"
      ],
      "text/markdown": [
       "26947"
      ],
      "text/plain": [
       "[1] 26947"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(doubletons)\n",
    "length(singletons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1348,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>nk</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAACTCCAATCGTCCGTCCT</td><td> 6</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AACTAGTTGTAACGGACCAA</td><td> 5</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATAACTAGGCGC </td><td> 5</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATCCTAGCGCCT </td><td>15</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG </td><td> 1</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td> 0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " ID & nk\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAACTCCAATCGTCCGTCCT &  6\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AACTAGTTGTAACGGACCAA &  5\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGACGATAACTAGGCGC  &  5\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGACGATCCTAGCGCCT  & 15\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG  &  1\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA &  0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 2\n",
       "\n",
       "| ID &lt;chr&gt; | nk &lt;int&gt; |\n",
       "|---|---|\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAACTCCAATCGTCCGTCCT |  6 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AACTAGTTGTAACGGACCAA |  5 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATAACTAGGCGC  |  5 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATCCTAGCGCCT  | 15 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG  |  1 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA |  0 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID                                                       nk\n",
       "1 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAACTCCAATCGTCCGTCCT  6\n",
       "2 exp10_idx1AATACGATAA_idx2AACTACCGAC.AACTAGTTGTAACGGACCAA  5\n",
       "3 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATAACTAGGCGC   5\n",
       "4 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATCCTAGCGCCT  15\n",
       "5 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG   1\n",
       "6 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA  0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(res)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1349,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>Label</th><th scope=col>V21</th><th scope=col>nnLabel</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>3626</td><td>plasmatocytes anlage             </td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>3626</td><td>plasmatocytes anlage             </td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>3626</td><td>sensory nervous system primordium</td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>3626</td><td>plasmatocytes anlage             </td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>3626</td><td>plasmatocytes anlage             </td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>3626</td><td>plasmatocytes anlage             </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 4\n",
       "\\begin{tabular}{llll}\n",
       " ID & Label & V21 & nnLabel\\\\\n",
       " <chr> & <chr> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & 3626 & plasmatocytes anlage             \\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & 3626 & plasmatocytes anlage             \\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & 3626 & sensory nervous system primordium\\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & 3626 & plasmatocytes anlage             \\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & 3626 & plasmatocytes anlage             \\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & 3626 & plasmatocytes anlage             \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 4\n",
       "\n",
       "| ID &lt;chr&gt; | Label &lt;chr&gt; | V21 &lt;int&gt; | nnLabel &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | 3626 | plasmatocytes anlage              |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | 3626 | plasmatocytes anlage              |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | 3626 | sensory nervous system primordium |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | 3626 | plasmatocytes anlage              |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | 3626 | plasmatocytes anlage              |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | 3626 | plasmatocytes anlage              |\n",
       "\n"
      ],
      "text/plain": [
       "  ID                                                    \n",
       "1 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "2 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "3 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "4 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "5 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "6 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "  Label                             V21  nnLabel                          \n",
       "1 sensory nervous system primordium 3626 plasmatocytes anlage             \n",
       "2 sensory nervous system primordium 3626 plasmatocytes anlage             \n",
       "3 sensory nervous system primordium 3626 sensory nervous system primordium\n",
       "4 sensory nervous system primordium 3626 plasmatocytes anlage             \n",
       "5 sensory nervous system primordium 3626 plasmatocytes anlage             \n",
       "6 sensory nervous system primordium 3626 plasmatocytes anlage             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1369,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type</th><th scope=col>gene</th><th scope=col>cell.type.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>tey25</th><td>muscle</td><td>tey     </td><td> 1964</td><td> 1638</td><td> 762.34753</td><td>  70.041469</td><td>3.423716</td><td>32.78617</td></tr>\n",
       "\t<tr><th scope=row>Mp2025</th><td>muscle</td><td>Mp20    </td><td> 4137</td><td> 1896</td><td> 504.73719</td><td>  54.475322</td><td>3.185614</td><td>28.61397</td></tr>\n",
       "\t<tr><th scope=row>CG4507825</th><td>muscle</td><td>CG45078 </td><td> 1824</td><td>  714</td><td> 244.08650</td><td>  19.842287</td><td>3.549807</td><td>28.17534</td></tr>\n",
       "\t<tr><th scope=row>Mf25</th><td>muscle</td><td>Mf      </td><td> 7493</td><td> 5314</td><td> 919.35534</td><td> 178.201118</td><td>2.359043</td><td>23.22725</td></tr>\n",
       "\t<tr><th scope=row>beat-IIb25</th><td>muscle</td><td>beat-IIb</td><td> 1474</td><td> 2182</td><td> 579.46721</td><td> 100.363721</td><td>2.515186</td><td>23.09210</td></tr>\n",
       "\t<tr><th scope=row>Gapdh125</th><td>muscle</td><td>Gapdh1  </td><td> 4423</td><td> 3080</td><td> 514.39818</td><td>  92.681725</td><td>2.457046</td><td>22.13686</td></tr>\n",
       "\t<tr><th scope=row>Zasp6625</th><td>muscle</td><td>Zasp66  </td><td> 1662</td><td>  907</td><td> 222.79963</td><td>  30.772827</td><td>2.809881</td><td>21.93411</td></tr>\n",
       "\t<tr><th scope=row>beat-IIa25</th><td>muscle</td><td>beat-IIa</td><td>  904</td><td> 1603</td><td> 371.61340</td><td>  69.495814</td><td>2.398193</td><td>20.48425</td></tr>\n",
       "\t<tr><th scope=row>Act57B25</th><td>muscle</td><td>Act57B  </td><td>32620</td><td>47559</td><td>4906.25710</td><td>1577.983536</td><td>1.635627</td><td>20.05393</td></tr>\n",
       "\t<tr><th scope=row>Mpcp125</th><td>muscle</td><td>Mpcp1   </td><td>  534</td><td>  172</td><td>  61.92248</td><td>   5.369622</td><td>3.281183</td><td>19.60672</td></tr>\n",
       "\t<tr><th scope=row>Mlc225</th><td>muscle</td><td>Mlc2    </td><td>20354</td><td>26770</td><td>2933.22729</td><td> 913.416045</td><td>1.681566</td><td>19.36957</td></tr>\n",
       "\t<tr><th scope=row>CG929725</th><td>muscle</td><td>CG9297  </td><td> 1030</td><td>  601</td><td> 139.10186</td><td>  21.152874</td><td>2.650576</td><td>18.89949</td></tr>\n",
       "\t<tr><th scope=row>CG502325</th><td>muscle</td><td>CG5023  </td><td>  513</td><td>  198</td><td>  62.03037</td><td>   5.973473</td><td>3.153025</td><td>18.84871</td></tr>\n",
       "\t<tr><th scope=row>Mhc25</th><td>muscle</td><td>Mhc     </td><td>29613</td><td>43781</td><td>4800.48805</td><td>1700.914991</td><td>1.496022</td><td>18.29525</td></tr>\n",
       "\t<tr><th scope=row>Eno25</th><td>muscle</td><td>Eno     </td><td> 6862</td><td> 7834</td><td> 944.82065</td><td> 268.935125</td><td>1.807428</td><td>17.86719</td></tr>\n",
       "\t<tr><th scope=row>Ald125</th><td>muscle</td><td>Ald1    </td><td>18114</td><td>26937</td><td>2659.63378</td><td> 916.315767</td><td>1.535737</td><td>17.47293</td></tr>\n",
       "\t<tr><th scope=row>Argk25</th><td>muscle</td><td>Argk    </td><td>10739</td><td>14581</td><td>1575.09047</td><td> 503.601718</td><td>1.642218</td><td>17.44386</td></tr>\n",
       "\t<tr><th scope=row>Mlc125</th><td>muscle</td><td>Mlc1    </td><td> 6296</td><td> 6744</td><td> 826.82123</td><td> 243.514585</td><td>1.757655</td><td>17.03726</td></tr>\n",
       "\t<tr><th scope=row>Adh25</th><td>muscle</td><td>Adh     </td><td> 3422</td><td> 3048</td><td> 390.13937</td><td>  99.003300</td><td>1.963942</td><td>16.91256</td></tr>\n",
       "\t<tr><th scope=row>CG1137025</th><td>muscle</td><td>CG11370 </td><td>  920</td><td> 1027</td><td> 152.73782</td><td>  29.646137</td><td>2.317280</td><td>16.83348</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type & gene & cell.type.n.umi & other.n.umi & cell.type.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\ttey25 & muscle & tey      &  1964 &  1638 &  762.34753 &   70.041469 & 3.423716 & 32.78617\\\\\n",
       "\tMp2025 & muscle & Mp20     &  4137 &  1896 &  504.73719 &   54.475322 & 3.185614 & 28.61397\\\\\n",
       "\tCG4507825 & muscle & CG45078  &  1824 &   714 &  244.08650 &   19.842287 & 3.549807 & 28.17534\\\\\n",
       "\tMf25 & muscle & Mf       &  7493 &  5314 &  919.35534 &  178.201118 & 2.359043 & 23.22725\\\\\n",
       "\tbeat-IIb25 & muscle & beat-IIb &  1474 &  2182 &  579.46721 &  100.363721 & 2.515186 & 23.09210\\\\\n",
       "\tGapdh125 & muscle & Gapdh1   &  4423 &  3080 &  514.39818 &   92.681725 & 2.457046 & 22.13686\\\\\n",
       "\tZasp6625 & muscle & Zasp66   &  1662 &   907 &  222.79963 &   30.772827 & 2.809881 & 21.93411\\\\\n",
       "\tbeat-IIa25 & muscle & beat-IIa &   904 &  1603 &  371.61340 &   69.495814 & 2.398193 & 20.48425\\\\\n",
       "\tAct57B25 & muscle & Act57B   & 32620 & 47559 & 4906.25710 & 1577.983536 & 1.635627 & 20.05393\\\\\n",
       "\tMpcp125 & muscle & Mpcp1    &   534 &   172 &   61.92248 &    5.369622 & 3.281183 & 19.60672\\\\\n",
       "\tMlc225 & muscle & Mlc2     & 20354 & 26770 & 2933.22729 &  913.416045 & 1.681566 & 19.36957\\\\\n",
       "\tCG929725 & muscle & CG9297   &  1030 &   601 &  139.10186 &   21.152874 & 2.650576 & 18.89949\\\\\n",
       "\tCG502325 & muscle & CG5023   &   513 &   198 &   62.03037 &    5.973473 & 3.153025 & 18.84871\\\\\n",
       "\tMhc25 & muscle & Mhc      & 29613 & 43781 & 4800.48805 & 1700.914991 & 1.496022 & 18.29525\\\\\n",
       "\tEno25 & muscle & Eno      &  6862 &  7834 &  944.82065 &  268.935125 & 1.807428 & 17.86719\\\\\n",
       "\tAld125 & muscle & Ald1     & 18114 & 26937 & 2659.63378 &  916.315767 & 1.535737 & 17.47293\\\\\n",
       "\tArgk25 & muscle & Argk     & 10739 & 14581 & 1575.09047 &  503.601718 & 1.642218 & 17.44386\\\\\n",
       "\tMlc125 & muscle & Mlc1     &  6296 &  6744 &  826.82123 &  243.514585 & 1.757655 & 17.03726\\\\\n",
       "\tAdh25 & muscle & Adh      &  3422 &  3048 &  390.13937 &   99.003300 & 1.963942 & 16.91256\\\\\n",
       "\tCG1137025 & muscle & CG11370  &   920 &  1027 &  152.73782 &   29.646137 & 2.317280 & 16.83348\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type &lt;chr&gt; | gene &lt;chr&gt; | cell.type.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| tey25 | muscle | tey      |  1964 |  1638 |  762.34753 |   70.041469 | 3.423716 | 32.78617 |\n",
       "| Mp2025 | muscle | Mp20     |  4137 |  1896 |  504.73719 |   54.475322 | 3.185614 | 28.61397 |\n",
       "| CG4507825 | muscle | CG45078  |  1824 |   714 |  244.08650 |   19.842287 | 3.549807 | 28.17534 |\n",
       "| Mf25 | muscle | Mf       |  7493 |  5314 |  919.35534 |  178.201118 | 2.359043 | 23.22725 |\n",
       "| beat-IIb25 | muscle | beat-IIb |  1474 |  2182 |  579.46721 |  100.363721 | 2.515186 | 23.09210 |\n",
       "| Gapdh125 | muscle | Gapdh1   |  4423 |  3080 |  514.39818 |   92.681725 | 2.457046 | 22.13686 |\n",
       "| Zasp6625 | muscle | Zasp66   |  1662 |   907 |  222.79963 |   30.772827 | 2.809881 | 21.93411 |\n",
       "| beat-IIa25 | muscle | beat-IIa |   904 |  1603 |  371.61340 |   69.495814 | 2.398193 | 20.48425 |\n",
       "| Act57B25 | muscle | Act57B   | 32620 | 47559 | 4906.25710 | 1577.983536 | 1.635627 | 20.05393 |\n",
       "| Mpcp125 | muscle | Mpcp1    |   534 |   172 |   61.92248 |    5.369622 | 3.281183 | 19.60672 |\n",
       "| Mlc225 | muscle | Mlc2     | 20354 | 26770 | 2933.22729 |  913.416045 | 1.681566 | 19.36957 |\n",
       "| CG929725 | muscle | CG9297   |  1030 |   601 |  139.10186 |   21.152874 | 2.650576 | 18.89949 |\n",
       "| CG502325 | muscle | CG5023   |   513 |   198 |   62.03037 |    5.973473 | 3.153025 | 18.84871 |\n",
       "| Mhc25 | muscle | Mhc      | 29613 | 43781 | 4800.48805 | 1700.914991 | 1.496022 | 18.29525 |\n",
       "| Eno25 | muscle | Eno      |  6862 |  7834 |  944.82065 |  268.935125 | 1.807428 | 17.86719 |\n",
       "| Ald125 | muscle | Ald1     | 18114 | 26937 | 2659.63378 |  916.315767 | 1.535737 | 17.47293 |\n",
       "| Argk25 | muscle | Argk     | 10739 | 14581 | 1575.09047 |  503.601718 | 1.642218 | 17.44386 |\n",
       "| Mlc125 | muscle | Mlc1     |  6296 |  6744 |  826.82123 |  243.514585 | 1.757655 | 17.03726 |\n",
       "| Adh25 | muscle | Adh      |  3422 |  3048 |  390.13937 |   99.003300 | 1.963942 | 16.91256 |\n",
       "| CG1137025 | muscle | CG11370  |   920 |  1027 |  152.73782 |   29.646137 | 2.317280 | 16.83348 |\n",
       "\n"
      ],
      "text/plain": [
       "           cell.type gene     cell.type.n.umi other.n.umi cell.type.tpm\n",
       "tey25      muscle    tey       1964            1638        762.34753   \n",
       "Mp2025     muscle    Mp20      4137            1896        504.73719   \n",
       "CG4507825  muscle    CG45078   1824             714        244.08650   \n",
       "Mf25       muscle    Mf        7493            5314        919.35534   \n",
       "beat-IIb25 muscle    beat-IIb  1474            2182        579.46721   \n",
       "Gapdh125   muscle    Gapdh1    4423            3080        514.39818   \n",
       "Zasp6625   muscle    Zasp66    1662             907        222.79963   \n",
       "beat-IIa25 muscle    beat-IIa   904            1603        371.61340   \n",
       "Act57B25   muscle    Act57B   32620           47559       4906.25710   \n",
       "Mpcp125    muscle    Mpcp1      534             172         61.92248   \n",
       "Mlc225     muscle    Mlc2     20354           26770       2933.22729   \n",
       "CG929725   muscle    CG9297    1030             601        139.10186   \n",
       "CG502325   muscle    CG5023     513             198         62.03037   \n",
       "Mhc25      muscle    Mhc      29613           43781       4800.48805   \n",
       "Eno25      muscle    Eno       6862            7834        944.82065   \n",
       "Ald125     muscle    Ald1     18114           26937       2659.63378   \n",
       "Argk25     muscle    Argk     10739           14581       1575.09047   \n",
       "Mlc125     muscle    Mlc1      6296            6744        826.82123   \n",
       "Adh25      muscle    Adh       3422            3048        390.13937   \n",
       "CG1137025  muscle    CG11370    920            1027        152.73782   \n",
       "           other.tpm   log2.ratio heuristic.score\n",
       "tey25        70.041469 3.423716   32.78617       \n",
       "Mp2025       54.475322 3.185614   28.61397       \n",
       "CG4507825    19.842287 3.549807   28.17534       \n",
       "Mf25        178.201118 2.359043   23.22725       \n",
       "beat-IIb25  100.363721 2.515186   23.09210       \n",
       "Gapdh125     92.681725 2.457046   22.13686       \n",
       "Zasp6625     30.772827 2.809881   21.93411       \n",
       "beat-IIa25   69.495814 2.398193   20.48425       \n",
       "Act57B25   1577.983536 1.635627   20.05393       \n",
       "Mpcp125       5.369622 3.281183   19.60672       \n",
       "Mlc225      913.416045 1.681566   19.36957       \n",
       "CG929725     21.152874 2.650576   18.89949       \n",
       "CG502325      5.973473 3.153025   18.84871       \n",
       "Mhc25      1700.914991 1.496022   18.29525       \n",
       "Eno25       268.935125 1.807428   17.86719       \n",
       "Ald125      916.315767 1.535737   17.47293       \n",
       "Argk25      503.601718 1.642218   17.44386       \n",
       "Mlc125      243.514585 1.757655   17.03726       \n",
       "Adh25        99.003300 1.963942   16.91256       \n",
       "CG1137025    29.646137 2.317280   16.83348       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.markers.sub300.cell.type  %>% filter(cell.type == \"muscle\")  %>%  head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1370,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type</th><th scope=col>gene</th><th scope=col>cell.type.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>tey24</th><td>muscle</td><td>tey     </td><td> 1597</td><td> 1643</td><td> 845.04318</td><td>  69.559791</td><td>3.582107</td><td>34.83451</td></tr>\n",
       "\t<tr><th scope=row>Mp2024</th><td>muscle</td><td>Mp20    </td><td> 4063</td><td> 1966</td><td> 649.10988</td><td>  56.567685</td><td>3.495132</td><td>32.66040</td></tr>\n",
       "\t<tr><th scope=row>CG4507824</th><td>muscle</td><td>CG45078 </td><td> 1788</td><td>  868</td><td> 312.90491</td><td>  24.917110</td><td>3.593748</td><td>29.80720</td></tr>\n",
       "\t<tr><th scope=row>Mf24</th><td>muscle</td><td>Mf      </td><td> 7225</td><td> 5557</td><td>1143.41562</td><td> 177.853242</td><td>2.676502</td><td>27.19432</td></tr>\n",
       "\t<tr><th scope=row>Zasp6624</th><td>muscle</td><td>Zasp66  </td><td> 1619</td><td>  952</td><td> 279.69292</td><td>  28.290679</td><td>3.255330</td><td>26.47511</td></tr>\n",
       "\t<tr><th scope=row>Gapdh124</th><td>muscle</td><td>Gapdh1  </td><td> 4270</td><td> 3104</td><td> 636.48976</td><td>  90.331424</td><td>2.800954</td><td>26.09441</td></tr>\n",
       "\t<tr><th scope=row>Mpcp124</th><td>muscle</td><td>Mpcp1   </td><td>  525</td><td>  186</td><td>  79.67202</td><td>   5.247212</td><td>3.672789</td><td>23.26343</td></tr>\n",
       "\t<tr><th scope=row>beat-IIb24</th><td>muscle</td><td>beat-IIb</td><td> 1105</td><td> 2228</td><td> 587.96653</td><td> 101.074582</td><td>2.526110</td><td>23.24537</td></tr>\n",
       "\t<tr><th scope=row>Act57B24</th><td>muscle</td><td>Act57B  </td><td>29143</td><td>48429</td><td>5529.94972</td><td>1551.159977</td><td>1.832989</td><td>22.79012</td></tr>\n",
       "\t<tr><th scope=row>CG502324</th><td>muscle</td><td>CG5023  </td><td>  506</td><td>  233</td><td>  80.14003</td><td>   5.693488</td><td>3.581693</td><td>22.71632</td></tr>\n",
       "\t<tr><th scope=row>Mlc224</th><td>muscle</td><td>Mlc2    </td><td>18352</td><td>27398</td><td>3332.35920</td><td> 897.138059</td><td>1.891535</td><td>22.13618</td></tr>\n",
       "\t<tr><th scope=row>CG1137024</th><td>muscle</td><td>CG11370 </td><td>  894</td><td> 1032</td><td> 194.71480</td><td>  26.744400</td><td>2.811094</td><td>21.39976</td></tr>\n",
       "\t<tr><th scope=row>CG929724</th><td>muscle</td><td>CG9297  </td><td> 1000</td><td>  712</td><td> 174.36611</td><td>  23.538953</td><td>2.828974</td><td>21.08781</td></tr>\n",
       "\t<tr><th scope=row>Unc-8924</th><td>muscle</td><td>Unc-89  </td><td>  965</td><td>  739</td><td> 173.03389</td><td>  24.545117</td><td>2.759935</td><td>20.54282</td></tr>\n",
       "\t<tr><th scope=row>Adh24</th><td>muscle</td><td>Adh     </td><td> 3328</td><td> 3050</td><td> 490.35281</td><td> 100.088096</td><td>2.278207</td><td>20.36857</td></tr>\n",
       "\t<tr><th scope=row>Eno24</th><td>muscle</td><td>Eno     </td><td> 6334</td><td> 8165</td><td>1101.18450</td><td> 271.471284</td><td>2.014880</td><td>20.36268</td></tr>\n",
       "\t<tr><th scope=row>Argk24</th><td>muscle</td><td>Argk    </td><td> 9835</td><td>14877</td><td>1825.55634</td><td> 499.200202</td><td>1.867759</td><td>20.23700</td></tr>\n",
       "\t<tr><th scope=row>Ald124</th><td>muscle</td><td>Ald1    </td><td>16321</td><td>27513</td><td>3026.70938</td><td> 907.825437</td><td>1.735675</td><td>20.07137</td></tr>\n",
       "\t<tr><th scope=row>beat-IIa24</th><td>muscle</td><td>beat-IIa</td><td>  672</td><td> 1609</td><td> 368.71607</td><td>  71.408369</td><td>2.348282</td><td>20.03149</td></tr>\n",
       "\t<tr><th scope=row>Mlc124</th><td>muscle</td><td>Mlc1    </td><td> 5898</td><td> 6988</td><td> 991.17052</td><td> 245.206380</td><td>2.009265</td><td>20.00112</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type & gene & cell.type.n.umi & other.n.umi & cell.type.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\ttey24 & muscle & tey      &  1597 &  1643 &  845.04318 &   69.559791 & 3.582107 & 34.83451\\\\\n",
       "\tMp2024 & muscle & Mp20     &  4063 &  1966 &  649.10988 &   56.567685 & 3.495132 & 32.66040\\\\\n",
       "\tCG4507824 & muscle & CG45078  &  1788 &   868 &  312.90491 &   24.917110 & 3.593748 & 29.80720\\\\\n",
       "\tMf24 & muscle & Mf       &  7225 &  5557 & 1143.41562 &  177.853242 & 2.676502 & 27.19432\\\\\n",
       "\tZasp6624 & muscle & Zasp66   &  1619 &   952 &  279.69292 &   28.290679 & 3.255330 & 26.47511\\\\\n",
       "\tGapdh124 & muscle & Gapdh1   &  4270 &  3104 &  636.48976 &   90.331424 & 2.800954 & 26.09441\\\\\n",
       "\tMpcp124 & muscle & Mpcp1    &   525 &   186 &   79.67202 &    5.247212 & 3.672789 & 23.26343\\\\\n",
       "\tbeat-IIb24 & muscle & beat-IIb &  1105 &  2228 &  587.96653 &  101.074582 & 2.526110 & 23.24537\\\\\n",
       "\tAct57B24 & muscle & Act57B   & 29143 & 48429 & 5529.94972 & 1551.159977 & 1.832989 & 22.79012\\\\\n",
       "\tCG502324 & muscle & CG5023   &   506 &   233 &   80.14003 &    5.693488 & 3.581693 & 22.71632\\\\\n",
       "\tMlc224 & muscle & Mlc2     & 18352 & 27398 & 3332.35920 &  897.138059 & 1.891535 & 22.13618\\\\\n",
       "\tCG1137024 & muscle & CG11370  &   894 &  1032 &  194.71480 &   26.744400 & 2.811094 & 21.39976\\\\\n",
       "\tCG929724 & muscle & CG9297   &  1000 &   712 &  174.36611 &   23.538953 & 2.828974 & 21.08781\\\\\n",
       "\tUnc-8924 & muscle & Unc-89   &   965 &   739 &  173.03389 &   24.545117 & 2.759935 & 20.54282\\\\\n",
       "\tAdh24 & muscle & Adh      &  3328 &  3050 &  490.35281 &  100.088096 & 2.278207 & 20.36857\\\\\n",
       "\tEno24 & muscle & Eno      &  6334 &  8165 & 1101.18450 &  271.471284 & 2.014880 & 20.36268\\\\\n",
       "\tArgk24 & muscle & Argk     &  9835 & 14877 & 1825.55634 &  499.200202 & 1.867759 & 20.23700\\\\\n",
       "\tAld124 & muscle & Ald1     & 16321 & 27513 & 3026.70938 &  907.825437 & 1.735675 & 20.07137\\\\\n",
       "\tbeat-IIa24 & muscle & beat-IIa &   672 &  1609 &  368.71607 &   71.408369 & 2.348282 & 20.03149\\\\\n",
       "\tMlc124 & muscle & Mlc1     &  5898 &  6988 &  991.17052 &  245.206380 & 2.009265 & 20.00112\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type &lt;chr&gt; | gene &lt;chr&gt; | cell.type.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| tey24 | muscle | tey      |  1597 |  1643 |  845.04318 |   69.559791 | 3.582107 | 34.83451 |\n",
       "| Mp2024 | muscle | Mp20     |  4063 |  1966 |  649.10988 |   56.567685 | 3.495132 | 32.66040 |\n",
       "| CG4507824 | muscle | CG45078  |  1788 |   868 |  312.90491 |   24.917110 | 3.593748 | 29.80720 |\n",
       "| Mf24 | muscle | Mf       |  7225 |  5557 | 1143.41562 |  177.853242 | 2.676502 | 27.19432 |\n",
       "| Zasp6624 | muscle | Zasp66   |  1619 |   952 |  279.69292 |   28.290679 | 3.255330 | 26.47511 |\n",
       "| Gapdh124 | muscle | Gapdh1   |  4270 |  3104 |  636.48976 |   90.331424 | 2.800954 | 26.09441 |\n",
       "| Mpcp124 | muscle | Mpcp1    |   525 |   186 |   79.67202 |    5.247212 | 3.672789 | 23.26343 |\n",
       "| beat-IIb24 | muscle | beat-IIb |  1105 |  2228 |  587.96653 |  101.074582 | 2.526110 | 23.24537 |\n",
       "| Act57B24 | muscle | Act57B   | 29143 | 48429 | 5529.94972 | 1551.159977 | 1.832989 | 22.79012 |\n",
       "| CG502324 | muscle | CG5023   |   506 |   233 |   80.14003 |    5.693488 | 3.581693 | 22.71632 |\n",
       "| Mlc224 | muscle | Mlc2     | 18352 | 27398 | 3332.35920 |  897.138059 | 1.891535 | 22.13618 |\n",
       "| CG1137024 | muscle | CG11370  |   894 |  1032 |  194.71480 |   26.744400 | 2.811094 | 21.39976 |\n",
       "| CG929724 | muscle | CG9297   |  1000 |   712 |  174.36611 |   23.538953 | 2.828974 | 21.08781 |\n",
       "| Unc-8924 | muscle | Unc-89   |   965 |   739 |  173.03389 |   24.545117 | 2.759935 | 20.54282 |\n",
       "| Adh24 | muscle | Adh      |  3328 |  3050 |  490.35281 |  100.088096 | 2.278207 | 20.36857 |\n",
       "| Eno24 | muscle | Eno      |  6334 |  8165 | 1101.18450 |  271.471284 | 2.014880 | 20.36268 |\n",
       "| Argk24 | muscle | Argk     |  9835 | 14877 | 1825.55634 |  499.200202 | 1.867759 | 20.23700 |\n",
       "| Ald124 | muscle | Ald1     | 16321 | 27513 | 3026.70938 |  907.825437 | 1.735675 | 20.07137 |\n",
       "| beat-IIa24 | muscle | beat-IIa |   672 |  1609 |  368.71607 |   71.408369 | 2.348282 | 20.03149 |\n",
       "| Mlc124 | muscle | Mlc1     |  5898 |  6988 |  991.17052 |  245.206380 | 2.009265 | 20.00112 |\n",
       "\n"
      ],
      "text/plain": [
       "           cell.type gene     cell.type.n.umi other.n.umi cell.type.tpm\n",
       "tey24      muscle    tey       1597            1643        845.04318   \n",
       "Mp2024     muscle    Mp20      4063            1966        649.10988   \n",
       "CG4507824  muscle    CG45078   1788             868        312.90491   \n",
       "Mf24       muscle    Mf        7225            5557       1143.41562   \n",
       "Zasp6624   muscle    Zasp66    1619             952        279.69292   \n",
       "Gapdh124   muscle    Gapdh1    4270            3104        636.48976   \n",
       "Mpcp124    muscle    Mpcp1      525             186         79.67202   \n",
       "beat-IIb24 muscle    beat-IIb  1105            2228        587.96653   \n",
       "Act57B24   muscle    Act57B   29143           48429       5529.94972   \n",
       "CG502324   muscle    CG5023     506             233         80.14003   \n",
       "Mlc224     muscle    Mlc2     18352           27398       3332.35920   \n",
       "CG1137024  muscle    CG11370    894            1032        194.71480   \n",
       "CG929724   muscle    CG9297    1000             712        174.36611   \n",
       "Unc-8924   muscle    Unc-89     965             739        173.03389   \n",
       "Adh24      muscle    Adh       3328            3050        490.35281   \n",
       "Eno24      muscle    Eno       6334            8165       1101.18450   \n",
       "Argk24     muscle    Argk      9835           14877       1825.55634   \n",
       "Ald124     muscle    Ald1     16321           27513       3026.70938   \n",
       "beat-IIa24 muscle    beat-IIa   672            1609        368.71607   \n",
       "Mlc124     muscle    Mlc1      5898            6988        991.17052   \n",
       "           other.tpm   log2.ratio heuristic.score\n",
       "tey24        69.559791 3.582107   34.83451       \n",
       "Mp2024       56.567685 3.495132   32.66040       \n",
       "CG4507824    24.917110 3.593748   29.80720       \n",
       "Mf24        177.853242 2.676502   27.19432       \n",
       "Zasp6624     28.290679 3.255330   26.47511       \n",
       "Gapdh124     90.331424 2.800954   26.09441       \n",
       "Mpcp124       5.247212 3.672789   23.26343       \n",
       "beat-IIb24  101.074582 2.526110   23.24537       \n",
       "Act57B24   1551.159977 1.832989   22.79012       \n",
       "CG502324      5.693488 3.581693   22.71632       \n",
       "Mlc224      897.138059 1.891535   22.13618       \n",
       "CG1137024    26.744400 2.811094   21.39976       \n",
       "CG929724     23.538953 2.828974   21.08781       \n",
       "Unc-8924     24.545117 2.759935   20.54282       \n",
       "Adh24       100.088096 2.278207   20.36857       \n",
       "Eno24       271.471284 2.014880   20.36268       \n",
       "Argk24      499.200202 1.867759   20.23700       \n",
       "Ald124      907.825437 1.735675   20.07137       \n",
       "beat-IIa24   71.408369 2.348282   20.03149       \n",
       "Mlc124      245.206380 2.009265   20.00112       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.markers.sub300_no_singletons.cell.type  %>% filter(cell.type == \"muscle\")  %>%  head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1371,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type</th><th scope=col>gene</th><th scope=col>cell.type.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>bt37</th><td>somatic muscle</td><td>bt       </td><td>94422</td><td>28224</td><td>14502.3675</td><td> 855.56655</td><td>4.081579</td><td>56.42416</td></tr>\n",
       "\t<tr><th scope=row>up37</th><td>somatic muscle</td><td>up       </td><td>47990</td><td>21558</td><td> 8765.8965</td><td> 779.15485</td><td>3.490069</td><td>45.71240</td></tr>\n",
       "\t<tr><th scope=row>wupA37</th><td>somatic muscle</td><td>wupA     </td><td>32291</td><td>16004</td><td> 5760.6562</td><td> 550.86013</td><td>3.383859</td><td>42.27207</td></tr>\n",
       "\t<tr><th scope=row>Mhc37</th><td>somatic muscle</td><td>Mhc      </td><td>66939</td><td>39510</td><td>10760.8221</td><td>1235.01410</td><td>3.122021</td><td>41.81521</td></tr>\n",
       "\t<tr><th scope=row>sls37</th><td>somatic muscle</td><td>sls      </td><td>47282</td><td>25980</td><td> 6855.9405</td><td> 803.48356</td><td>3.091220</td><td>39.39249</td></tr>\n",
       "\t<tr><th scope=row>Sh37</th><td>somatic muscle</td><td>Sh       </td><td>16272</td><td> 7146</td><td> 3050.0225</td><td> 299.93213</td><td>3.341310</td><td>38.67592</td></tr>\n",
       "\t<tr><th scope=row>l(2)0128937</th><td>somatic muscle</td><td>l(2)01289</td><td> 6133</td><td> 2246</td><td> 1000.3516</td><td>  70.29999</td><td>3.810462</td><td>37.98166</td></tr>\n",
       "\t<tr><th scope=row>CG3352137</th><td>somatic muscle</td><td>CG33521  </td><td> 4961</td><td> 1947</td><td>  910.0490</td><td>  71.69011</td><td>3.646113</td><td>35.84634</td></tr>\n",
       "\t<tr><th scope=row>Mf37</th><td>somatic muscle</td><td>Mf       </td><td> 9959</td><td> 5112</td><td> 1348.4166</td><td> 122.70925</td><td>3.446241</td><td>35.83443</td></tr>\n",
       "\t<tr><th scope=row>CG888837</th><td>somatic muscle</td><td>CG8888   </td><td> 4946</td><td> 2021</td><td>  792.9674</td><td>  66.92476</td><td>3.545252</td><td>34.15119</td></tr>\n",
       "\t<tr><th scope=row>CG4389737</th><td>somatic muscle</td><td>CG43897  </td><td> 9799</td><td> 4764</td><td> 1378.5005</td><td> 149.92358</td><td>3.191210</td><td>33.28409</td></tr>\n",
       "\t<tr><th scope=row>CG167457</th><td>somatic muscle</td><td>CG1674   </td><td> 3813</td><td> 1682</td><td>  653.2129</td><td>  59.49847</td><td>3.432583</td><td>32.10706</td></tr>\n",
       "\t<tr><th scope=row>Neto37</th><td>somatic muscle</td><td>Neto     </td><td> 2542</td><td>  943</td><td>  469.0128</td><td>  39.69206</td><td>3.526808</td><td>31.30591</td></tr>\n",
       "\t<tr><th scope=row>Nlg137</th><td>somatic muscle</td><td>Nlg1     </td><td> 1565</td><td>  500</td><td>  286.3237</td><td>  20.14723</td><td>3.759107</td><td>30.69887</td></tr>\n",
       "\t<tr><th scope=row>Pkcdelta37</th><td>somatic muscle</td><td>Pkcdelta </td><td> 1593</td><td>  537</td><td>  259.2419</td><td>  18.54224</td><td>3.729631</td><td>29.92548</td></tr>\n",
       "\t<tr><th scope=row>Prm37</th><td>somatic muscle</td><td>Prm      </td><td>14784</td><td>11342</td><td> 2218.3822</td><td> 346.13961</td><td>2.675920</td><td>29.74537</td></tr>\n",
       "\t<tr><th scope=row>CG4442237</th><td>somatic muscle</td><td>CG44422  </td><td> 4974</td><td> 2853</td><td>  938.3233</td><td> 121.37908</td><td>2.938728</td><td>29.02135</td></tr>\n",
       "\t<tr><th scope=row>sals37</th><td>somatic muscle</td><td>sals     </td><td> 3555</td><td> 1892</td><td>  643.1520</td><td>  77.55719</td><td>3.033345</td><td>28.30492</td></tr>\n",
       "\t<tr><th scope=row>jp37</th><td>somatic muscle</td><td>jp       </td><td> 2483</td><td> 1369</td><td>  448.2064</td><td>  48.14144</td><td>3.189151</td><td>28.10036</td></tr>\n",
       "\t<tr><th scope=row>Strn-Mlck37</th><td>somatic muscle</td><td>Strn-Mlck</td><td> 7264</td><td> 5142</td><td> 1212.4271</td><td> 180.61630</td><td>2.738932</td><td>28.06001</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type & gene & cell.type.n.umi & other.n.umi & cell.type.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tbt37 & somatic muscle & bt        & 94422 & 28224 & 14502.3675 &  855.56655 & 4.081579 & 56.42416\\\\\n",
       "\tup37 & somatic muscle & up        & 47990 & 21558 &  8765.8965 &  779.15485 & 3.490069 & 45.71240\\\\\n",
       "\twupA37 & somatic muscle & wupA      & 32291 & 16004 &  5760.6562 &  550.86013 & 3.383859 & 42.27207\\\\\n",
       "\tMhc37 & somatic muscle & Mhc       & 66939 & 39510 & 10760.8221 & 1235.01410 & 3.122021 & 41.81521\\\\\n",
       "\tsls37 & somatic muscle & sls       & 47282 & 25980 &  6855.9405 &  803.48356 & 3.091220 & 39.39249\\\\\n",
       "\tSh37 & somatic muscle & Sh        & 16272 &  7146 &  3050.0225 &  299.93213 & 3.341310 & 38.67592\\\\\n",
       "\tl(2)0128937 & somatic muscle & l(2)01289 &  6133 &  2246 &  1000.3516 &   70.29999 & 3.810462 & 37.98166\\\\\n",
       "\tCG3352137 & somatic muscle & CG33521   &  4961 &  1947 &   910.0490 &   71.69011 & 3.646113 & 35.84634\\\\\n",
       "\tMf37 & somatic muscle & Mf        &  9959 &  5112 &  1348.4166 &  122.70925 & 3.446241 & 35.83443\\\\\n",
       "\tCG888837 & somatic muscle & CG8888    &  4946 &  2021 &   792.9674 &   66.92476 & 3.545252 & 34.15119\\\\\n",
       "\tCG4389737 & somatic muscle & CG43897   &  9799 &  4764 &  1378.5005 &  149.92358 & 3.191210 & 33.28409\\\\\n",
       "\tCG167457 & somatic muscle & CG1674    &  3813 &  1682 &   653.2129 &   59.49847 & 3.432583 & 32.10706\\\\\n",
       "\tNeto37 & somatic muscle & Neto      &  2542 &   943 &   469.0128 &   39.69206 & 3.526808 & 31.30591\\\\\n",
       "\tNlg137 & somatic muscle & Nlg1      &  1565 &   500 &   286.3237 &   20.14723 & 3.759107 & 30.69887\\\\\n",
       "\tPkcdelta37 & somatic muscle & Pkcdelta  &  1593 &   537 &   259.2419 &   18.54224 & 3.729631 & 29.92548\\\\\n",
       "\tPrm37 & somatic muscle & Prm       & 14784 & 11342 &  2218.3822 &  346.13961 & 2.675920 & 29.74537\\\\\n",
       "\tCG4442237 & somatic muscle & CG44422   &  4974 &  2853 &   938.3233 &  121.37908 & 2.938728 & 29.02135\\\\\n",
       "\tsals37 & somatic muscle & sals      &  3555 &  1892 &   643.1520 &   77.55719 & 3.033345 & 28.30492\\\\\n",
       "\tjp37 & somatic muscle & jp        &  2483 &  1369 &   448.2064 &   48.14144 & 3.189151 & 28.10036\\\\\n",
       "\tStrn-Mlck37 & somatic muscle & Strn-Mlck &  7264 &  5142 &  1212.4271 &  180.61630 & 2.738932 & 28.06001\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type &lt;chr&gt; | gene &lt;chr&gt; | cell.type.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| bt37 | somatic muscle | bt        | 94422 | 28224 | 14502.3675 |  855.56655 | 4.081579 | 56.42416 |\n",
       "| up37 | somatic muscle | up        | 47990 | 21558 |  8765.8965 |  779.15485 | 3.490069 | 45.71240 |\n",
       "| wupA37 | somatic muscle | wupA      | 32291 | 16004 |  5760.6562 |  550.86013 | 3.383859 | 42.27207 |\n",
       "| Mhc37 | somatic muscle | Mhc       | 66939 | 39510 | 10760.8221 | 1235.01410 | 3.122021 | 41.81521 |\n",
       "| sls37 | somatic muscle | sls       | 47282 | 25980 |  6855.9405 |  803.48356 | 3.091220 | 39.39249 |\n",
       "| Sh37 | somatic muscle | Sh        | 16272 |  7146 |  3050.0225 |  299.93213 | 3.341310 | 38.67592 |\n",
       "| l(2)0128937 | somatic muscle | l(2)01289 |  6133 |  2246 |  1000.3516 |   70.29999 | 3.810462 | 37.98166 |\n",
       "| CG3352137 | somatic muscle | CG33521   |  4961 |  1947 |   910.0490 |   71.69011 | 3.646113 | 35.84634 |\n",
       "| Mf37 | somatic muscle | Mf        |  9959 |  5112 |  1348.4166 |  122.70925 | 3.446241 | 35.83443 |\n",
       "| CG888837 | somatic muscle | CG8888    |  4946 |  2021 |   792.9674 |   66.92476 | 3.545252 | 34.15119 |\n",
       "| CG4389737 | somatic muscle | CG43897   |  9799 |  4764 |  1378.5005 |  149.92358 | 3.191210 | 33.28409 |\n",
       "| CG167457 | somatic muscle | CG1674    |  3813 |  1682 |   653.2129 |   59.49847 | 3.432583 | 32.10706 |\n",
       "| Neto37 | somatic muscle | Neto      |  2542 |   943 |   469.0128 |   39.69206 | 3.526808 | 31.30591 |\n",
       "| Nlg137 | somatic muscle | Nlg1      |  1565 |   500 |   286.3237 |   20.14723 | 3.759107 | 30.69887 |\n",
       "| Pkcdelta37 | somatic muscle | Pkcdelta  |  1593 |   537 |   259.2419 |   18.54224 | 3.729631 | 29.92548 |\n",
       "| Prm37 | somatic muscle | Prm       | 14784 | 11342 |  2218.3822 |  346.13961 | 2.675920 | 29.74537 |\n",
       "| CG4442237 | somatic muscle | CG44422   |  4974 |  2853 |   938.3233 |  121.37908 | 2.938728 | 29.02135 |\n",
       "| sals37 | somatic muscle | sals      |  3555 |  1892 |   643.1520 |   77.55719 | 3.033345 | 28.30492 |\n",
       "| jp37 | somatic muscle | jp        |  2483 |  1369 |   448.2064 |   48.14144 | 3.189151 | 28.10036 |\n",
       "| Strn-Mlck37 | somatic muscle | Strn-Mlck |  7264 |  5142 |  1212.4271 |  180.61630 | 2.738932 | 28.06001 |\n",
       "\n"
      ],
      "text/plain": [
       "            cell.type      gene      cell.type.n.umi other.n.umi cell.type.tpm\n",
       "bt37        somatic muscle bt        94422           28224       14502.3675   \n",
       "up37        somatic muscle up        47990           21558        8765.8965   \n",
       "wupA37      somatic muscle wupA      32291           16004        5760.6562   \n",
       "Mhc37       somatic muscle Mhc       66939           39510       10760.8221   \n",
       "sls37       somatic muscle sls       47282           25980        6855.9405   \n",
       "Sh37        somatic muscle Sh        16272            7146        3050.0225   \n",
       "l(2)0128937 somatic muscle l(2)01289  6133            2246        1000.3516   \n",
       "CG3352137   somatic muscle CG33521    4961            1947         910.0490   \n",
       "Mf37        somatic muscle Mf         9959            5112        1348.4166   \n",
       "CG888837    somatic muscle CG8888     4946            2021         792.9674   \n",
       "CG4389737   somatic muscle CG43897    9799            4764        1378.5005   \n",
       "CG167457    somatic muscle CG1674     3813            1682         653.2129   \n",
       "Neto37      somatic muscle Neto       2542             943         469.0128   \n",
       "Nlg137      somatic muscle Nlg1       1565             500         286.3237   \n",
       "Pkcdelta37  somatic muscle Pkcdelta   1593             537         259.2419   \n",
       "Prm37       somatic muscle Prm       14784           11342        2218.3822   \n",
       "CG4442237   somatic muscle CG44422    4974            2853         938.3233   \n",
       "sals37      somatic muscle sals       3555            1892         643.1520   \n",
       "jp37        somatic muscle jp         2483            1369         448.2064   \n",
       "Strn-Mlck37 somatic muscle Strn-Mlck  7264            5142        1212.4271   \n",
       "            other.tpm  log2.ratio heuristic.score\n",
       "bt37         855.56655 4.081579   56.42416       \n",
       "up37         779.15485 3.490069   45.71240       \n",
       "wupA37       550.86013 3.383859   42.27207       \n",
       "Mhc37       1235.01410 3.122021   41.81521       \n",
       "sls37        803.48356 3.091220   39.39249       \n",
       "Sh37         299.93213 3.341310   38.67592       \n",
       "l(2)0128937   70.29999 3.810462   37.98166       \n",
       "CG3352137     71.69011 3.646113   35.84634       \n",
       "Mf37         122.70925 3.446241   35.83443       \n",
       "CG888837      66.92476 3.545252   34.15119       \n",
       "CG4389737    149.92358 3.191210   33.28409       \n",
       "CG167457      59.49847 3.432583   32.10706       \n",
       "Neto37        39.69206 3.526808   31.30591       \n",
       "Nlg137        20.14723 3.759107   30.69887       \n",
       "Pkcdelta37    18.54224 3.729631   29.92548       \n",
       "Prm37        346.13961 2.675920   29.74537       \n",
       "CG4442237    121.37908 2.938728   29.02135       \n",
       "sals37        77.55719 3.033345   28.30492       \n",
       "jp37          48.14144 3.189151   28.10036       \n",
       "Strn-Mlck37  180.61630 2.738932   28.06001       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.markers.sub300.cell.type  %>% filter(cell.type == \"somatic muscle\")  %>%  head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1373,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type</th><th scope=col>gene</th><th scope=col>cell.type.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>bt36</th><td>somatic muscle</td><td>bt       </td><td>92826</td><td>27379</td><td>15359.2344</td><td> 815.02633</td><td>4.234347</td><td>58.88669</td></tr>\n",
       "\t<tr><th scope=row>up36</th><td>somatic muscle</td><td>up       </td><td>46828</td><td>20136</td><td> 9239.4016</td><td> 734.08962</td><td>3.651807</td><td>48.10796</td></tr>\n",
       "\t<tr><th scope=row>wupA36</th><td>somatic muscle</td><td>wupA     </td><td>31409</td><td>15383</td><td> 6056.9358</td><td> 527.61971</td><td>3.518286</td><td>44.20589</td></tr>\n",
       "\t<tr><th scope=row>Mhc36</th><td>somatic muscle</td><td>Mhc      </td><td>65044</td><td>37995</td><td>11311.6783</td><td>1179.65882</td><td>3.260149</td><td>43.90003</td></tr>\n",
       "\t<tr><th scope=row>sls36</th><td>somatic muscle</td><td>sls      </td><td>46054</td><td>24047</td><td> 7188.2227</td><td> 742.16007</td><td>3.273890</td><td>41.94384</td></tr>\n",
       "\t<tr><th scope=row>Sh36</th><td>somatic muscle</td><td>Sh       </td><td>15798</td><td> 7190</td><td> 3199.7939</td><td> 297.32053</td><td>3.423044</td><td>39.85865</td></tr>\n",
       "\t<tr><th scope=row>l(2)0128936</th><td>somatic muscle</td><td>l(2)01289</td><td> 5946</td><td> 2208</td><td> 1047.2543</td><td>  69.87395</td><td>3.885212</td><td>38.98334</td></tr>\n",
       "\t<tr><th scope=row>CG3352136</th><td>somatic muscle</td><td>CG33521  </td><td> 4859</td><td> 1795</td><td>  963.0629</td><td>  67.77183</td><td>3.807740</td><td>37.74607</td></tr>\n",
       "\t<tr><th scope=row>Mf36</th><td>somatic muscle</td><td>Mf       </td><td> 9606</td><td> 5191</td><td> 1407.2562</td><td> 120.62542</td><td>3.532368</td><td>36.94749</td></tr>\n",
       "\t<tr><th scope=row>CG888836</th><td>somatic muscle</td><td>CG8888   </td><td> 4735</td><td> 1872</td><td>  819.9477</td><td>  61.69747</td><td>3.709053</td><td>35.90788</td></tr>\n",
       "\t<tr><th scope=row>CG4389736</th><td>somatic muscle</td><td>CG43897  </td><td> 9584</td><td> 4544</td><td> 1450.5927</td><td> 142.05267</td><td>3.342024</td><td>35.10269</td></tr>\n",
       "\t<tr><th scope=row>CG167456</th><td>somatic muscle</td><td>CG1674   </td><td> 3758</td><td> 1584</td><td>  695.0675</td><td>  54.52930</td><td>3.645832</td><td>34.42790</td></tr>\n",
       "\t<tr><th scope=row>Neto36</th><td>somatic muscle</td><td>Neto     </td><td> 2487</td><td>  909</td><td>  495.8915</td><td>  37.21004</td><td>3.698001</td><td>33.12221</td></tr>\n",
       "\t<tr><th scope=row>Nlg136</th><td>somatic muscle</td><td>Nlg1     </td><td> 1542</td><td>  488</td><td>  304.0021</td><td>  19.79163</td><td>3.870007</td><td>31.93791</td></tr>\n",
       "\t<tr><th scope=row>Pkcdelta36</th><td>somatic muscle</td><td>Pkcdelta </td><td> 1561</td><td>  506</td><td>  274.8036</td><td>  17.65257</td><td>3.880955</td><td>31.46483</td></tr>\n",
       "\t<tr><th scope=row>sals36</th><td>somatic muscle</td><td>sals     </td><td> 3482</td><td> 1774</td><td>  679.1132</td><td>  69.27818</td><td>3.272503</td><td>30.79305</td></tr>\n",
       "\t<tr><th scope=row>Prm36</th><td>somatic muscle</td><td>Prm      </td><td>14307</td><td>11458</td><td> 2315.3785</td><td> 352.41063</td><td>2.711831</td><td>30.31191</td></tr>\n",
       "\t<tr><th scope=row>Strn-Mlck36</th><td>somatic muscle</td><td>Strn-Mlck</td><td> 7043</td><td> 4921</td><td> 1272.6820</td><td> 167.77920</td><td>2.914663</td><td>30.06414</td></tr>\n",
       "\t<tr><th scope=row>jp36</th><td>somatic muscle</td><td>jp       </td><td> 2438</td><td> 1321</td><td>  475.6961</td><td>  46.67957</td><td>3.318597</td><td>29.52531</td></tr>\n",
       "\t<tr><th scope=row>mspo36</th><td>somatic muscle</td><td>mspo     </td><td> 7772</td><td> 5270</td><td> 1543.3400</td><td> 226.68272</td><td>2.760959</td><td>29.24622</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type & gene & cell.type.n.umi & other.n.umi & cell.type.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tbt36 & somatic muscle & bt        & 92826 & 27379 & 15359.2344 &  815.02633 & 4.234347 & 58.88669\\\\\n",
       "\tup36 & somatic muscle & up        & 46828 & 20136 &  9239.4016 &  734.08962 & 3.651807 & 48.10796\\\\\n",
       "\twupA36 & somatic muscle & wupA      & 31409 & 15383 &  6056.9358 &  527.61971 & 3.518286 & 44.20589\\\\\n",
       "\tMhc36 & somatic muscle & Mhc       & 65044 & 37995 & 11311.6783 & 1179.65882 & 3.260149 & 43.90003\\\\\n",
       "\tsls36 & somatic muscle & sls       & 46054 & 24047 &  7188.2227 &  742.16007 & 3.273890 & 41.94384\\\\\n",
       "\tSh36 & somatic muscle & Sh        & 15798 &  7190 &  3199.7939 &  297.32053 & 3.423044 & 39.85865\\\\\n",
       "\tl(2)0128936 & somatic muscle & l(2)01289 &  5946 &  2208 &  1047.2543 &   69.87395 & 3.885212 & 38.98334\\\\\n",
       "\tCG3352136 & somatic muscle & CG33521   &  4859 &  1795 &   963.0629 &   67.77183 & 3.807740 & 37.74607\\\\\n",
       "\tMf36 & somatic muscle & Mf        &  9606 &  5191 &  1407.2562 &  120.62542 & 3.532368 & 36.94749\\\\\n",
       "\tCG888836 & somatic muscle & CG8888    &  4735 &  1872 &   819.9477 &   61.69747 & 3.709053 & 35.90788\\\\\n",
       "\tCG4389736 & somatic muscle & CG43897   &  9584 &  4544 &  1450.5927 &  142.05267 & 3.342024 & 35.10269\\\\\n",
       "\tCG167456 & somatic muscle & CG1674    &  3758 &  1584 &   695.0675 &   54.52930 & 3.645832 & 34.42790\\\\\n",
       "\tNeto36 & somatic muscle & Neto      &  2487 &   909 &   495.8915 &   37.21004 & 3.698001 & 33.12221\\\\\n",
       "\tNlg136 & somatic muscle & Nlg1      &  1542 &   488 &   304.0021 &   19.79163 & 3.870007 & 31.93791\\\\\n",
       "\tPkcdelta36 & somatic muscle & Pkcdelta  &  1561 &   506 &   274.8036 &   17.65257 & 3.880955 & 31.46483\\\\\n",
       "\tsals36 & somatic muscle & sals      &  3482 &  1774 &   679.1132 &   69.27818 & 3.272503 & 30.79305\\\\\n",
       "\tPrm36 & somatic muscle & Prm       & 14307 & 11458 &  2315.3785 &  352.41063 & 2.711831 & 30.31191\\\\\n",
       "\tStrn-Mlck36 & somatic muscle & Strn-Mlck &  7043 &  4921 &  1272.6820 &  167.77920 & 2.914663 & 30.06414\\\\\n",
       "\tjp36 & somatic muscle & jp        &  2438 &  1321 &   475.6961 &   46.67957 & 3.318597 & 29.52531\\\\\n",
       "\tmspo36 & somatic muscle & mspo      &  7772 &  5270 &  1543.3400 &  226.68272 & 2.760959 & 29.24622\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type &lt;chr&gt; | gene &lt;chr&gt; | cell.type.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| bt36 | somatic muscle | bt        | 92826 | 27379 | 15359.2344 |  815.02633 | 4.234347 | 58.88669 |\n",
       "| up36 | somatic muscle | up        | 46828 | 20136 |  9239.4016 |  734.08962 | 3.651807 | 48.10796 |\n",
       "| wupA36 | somatic muscle | wupA      | 31409 | 15383 |  6056.9358 |  527.61971 | 3.518286 | 44.20589 |\n",
       "| Mhc36 | somatic muscle | Mhc       | 65044 | 37995 | 11311.6783 | 1179.65882 | 3.260149 | 43.90003 |\n",
       "| sls36 | somatic muscle | sls       | 46054 | 24047 |  7188.2227 |  742.16007 | 3.273890 | 41.94384 |\n",
       "| Sh36 | somatic muscle | Sh        | 15798 |  7190 |  3199.7939 |  297.32053 | 3.423044 | 39.85865 |\n",
       "| l(2)0128936 | somatic muscle | l(2)01289 |  5946 |  2208 |  1047.2543 |   69.87395 | 3.885212 | 38.98334 |\n",
       "| CG3352136 | somatic muscle | CG33521   |  4859 |  1795 |   963.0629 |   67.77183 | 3.807740 | 37.74607 |\n",
       "| Mf36 | somatic muscle | Mf        |  9606 |  5191 |  1407.2562 |  120.62542 | 3.532368 | 36.94749 |\n",
       "| CG888836 | somatic muscle | CG8888    |  4735 |  1872 |   819.9477 |   61.69747 | 3.709053 | 35.90788 |\n",
       "| CG4389736 | somatic muscle | CG43897   |  9584 |  4544 |  1450.5927 |  142.05267 | 3.342024 | 35.10269 |\n",
       "| CG167456 | somatic muscle | CG1674    |  3758 |  1584 |   695.0675 |   54.52930 | 3.645832 | 34.42790 |\n",
       "| Neto36 | somatic muscle | Neto      |  2487 |   909 |   495.8915 |   37.21004 | 3.698001 | 33.12221 |\n",
       "| Nlg136 | somatic muscle | Nlg1      |  1542 |   488 |   304.0021 |   19.79163 | 3.870007 | 31.93791 |\n",
       "| Pkcdelta36 | somatic muscle | Pkcdelta  |  1561 |   506 |   274.8036 |   17.65257 | 3.880955 | 31.46483 |\n",
       "| sals36 | somatic muscle | sals      |  3482 |  1774 |   679.1132 |   69.27818 | 3.272503 | 30.79305 |\n",
       "| Prm36 | somatic muscle | Prm       | 14307 | 11458 |  2315.3785 |  352.41063 | 2.711831 | 30.31191 |\n",
       "| Strn-Mlck36 | somatic muscle | Strn-Mlck |  7043 |  4921 |  1272.6820 |  167.77920 | 2.914663 | 30.06414 |\n",
       "| jp36 | somatic muscle | jp        |  2438 |  1321 |   475.6961 |   46.67957 | 3.318597 | 29.52531 |\n",
       "| mspo36 | somatic muscle | mspo      |  7772 |  5270 |  1543.3400 |  226.68272 | 2.760959 | 29.24622 |\n",
       "\n"
      ],
      "text/plain": [
       "            cell.type      gene      cell.type.n.umi other.n.umi cell.type.tpm\n",
       "bt36        somatic muscle bt        92826           27379       15359.2344   \n",
       "up36        somatic muscle up        46828           20136        9239.4016   \n",
       "wupA36      somatic muscle wupA      31409           15383        6056.9358   \n",
       "Mhc36       somatic muscle Mhc       65044           37995       11311.6783   \n",
       "sls36       somatic muscle sls       46054           24047        7188.2227   \n",
       "Sh36        somatic muscle Sh        15798            7190        3199.7939   \n",
       "l(2)0128936 somatic muscle l(2)01289  5946            2208        1047.2543   \n",
       "CG3352136   somatic muscle CG33521    4859            1795         963.0629   \n",
       "Mf36        somatic muscle Mf         9606            5191        1407.2562   \n",
       "CG888836    somatic muscle CG8888     4735            1872         819.9477   \n",
       "CG4389736   somatic muscle CG43897    9584            4544        1450.5927   \n",
       "CG167456    somatic muscle CG1674     3758            1584         695.0675   \n",
       "Neto36      somatic muscle Neto       2487             909         495.8915   \n",
       "Nlg136      somatic muscle Nlg1       1542             488         304.0021   \n",
       "Pkcdelta36  somatic muscle Pkcdelta   1561             506         274.8036   \n",
       "sals36      somatic muscle sals       3482            1774         679.1132   \n",
       "Prm36       somatic muscle Prm       14307           11458        2315.3785   \n",
       "Strn-Mlck36 somatic muscle Strn-Mlck  7043            4921        1272.6820   \n",
       "jp36        somatic muscle jp         2438            1321         475.6961   \n",
       "mspo36      somatic muscle mspo       7772            5270        1543.3400   \n",
       "            other.tpm  log2.ratio heuristic.score\n",
       "bt36         815.02633 4.234347   58.88669       \n",
       "up36         734.08962 3.651807   48.10796       \n",
       "wupA36       527.61971 3.518286   44.20589       \n",
       "Mhc36       1179.65882 3.260149   43.90003       \n",
       "sls36        742.16007 3.273890   41.94384       \n",
       "Sh36         297.32053 3.423044   39.85865       \n",
       "l(2)0128936   69.87395 3.885212   38.98334       \n",
       "CG3352136     67.77183 3.807740   37.74607       \n",
       "Mf36         120.62542 3.532368   36.94749       \n",
       "CG888836      61.69747 3.709053   35.90788       \n",
       "CG4389736    142.05267 3.342024   35.10269       \n",
       "CG167456      54.52930 3.645832   34.42790       \n",
       "Neto36        37.21004 3.698001   33.12221       \n",
       "Nlg136        19.79163 3.870007   31.93791       \n",
       "Pkcdelta36    17.65257 3.880955   31.46483       \n",
       "sals36        69.27818 3.272503   30.79305       \n",
       "Prm36        352.41063 2.711831   30.31191       \n",
       "Strn-Mlck36  167.77920 2.914663   30.06414       \n",
       "jp36          46.67957 3.318597   29.52531       \n",
       "mspo36       226.68272 2.760959   29.24622       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.markers.sub300_no_singletons.cell.type  %>% filter(cell.type == \"somatic muscle\")  %>%  head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_sing, c(\"bt\"), size_pos = 0.5, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_sing, c(\"bt\"), size_pos = 0.5, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_sing, c(\"bt\"), size_pos = 0.5, size_neg = 0.06, stroke = 0.02, limit = 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub, c(\"bt\"), size_pos = 0.5, size_neg = 0.06, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub, c(\"bt\"), size_pos = 0.5, size_neg = 0.06, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub, c(\"bt\"), size_pos = 0.5, size_neg = 0.06, stroke = 0.02, limit = 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'cell.type.2.markers'</li><li>'cell.type.markers.sub300_no_singletons.cell.type'</li><li>'cell.type.markers.sub300.cell.type'</li><li>'cluster.markers'</li><li>'cluster.markers.sub300'</li><li>'cluster.markers.sub300_no_doubletons'</li><li>'cluster.markers.sub300_no_doubletons_all'</li><li>'cluster.markers.sub300_no_singletons'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'cell.type.2.markers'\n",
       "\\item 'cell.type.markers.sub300\\_no\\_singletons.cell.type'\n",
       "\\item 'cell.type.markers.sub300.cell.type'\n",
       "\\item 'cluster.markers'\n",
       "\\item 'cluster.markers.sub300'\n",
       "\\item 'cluster.markers.sub300\\_no\\_doubletons'\n",
       "\\item 'cluster.markers.sub300\\_no\\_doubletons\\_all'\n",
       "\\item 'cluster.markers.sub300\\_no\\_singletons'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'cell.type.2.markers'\n",
       "2. 'cell.type.markers.sub300_no_singletons.cell.type'\n",
       "3. 'cell.type.markers.sub300.cell.type'\n",
       "4. 'cluster.markers'\n",
       "5. 'cluster.markers.sub300'\n",
       "6. 'cluster.markers.sub300_no_doubletons'\n",
       "7. 'cluster.markers.sub300_no_doubletons_all'\n",
       "8. 'cluster.markers.sub300_no_singletons'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"cell.type.2.markers\"                             \n",
       "[2] \"cell.type.markers.sub300_no_singletons.cell.type\"\n",
       "[3] \"cell.type.markers.sub300.cell.type\"              \n",
       "[4] \"cluster.markers\"                                 \n",
       "[5] \"cluster.markers.sub300\"                          \n",
       "[6] \"cluster.markers.sub300_no_doubletons\"            \n",
       "[7] \"cluster.markers.sub300_no_doubletons_all\"        \n",
       "[8] \"cluster.markers.sub300_no_singletons\"            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ls(pattern = \"markers\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## make heatmaps of cluster with no singletons\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1375,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'cluster.markers'</li><li>'cluster.markers.sub300'</li><li>'cluster.markers.sub300_no_doubletons'</li><li>'cluster.markers.sub300_no_singletons'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'cluster.markers'\n",
       "\\item 'cluster.markers.sub300'\n",
       "\\item 'cluster.markers.sub300\\_no\\_doubletons'\n",
       "\\item 'cluster.markers.sub300\\_no\\_singletons'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'cluster.markers'\n",
       "2. 'cluster.markers.sub300'\n",
       "3. 'cluster.markers.sub300_no_doubletons'\n",
       "4. 'cluster.markers.sub300_no_singletons'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"cluster.markers\"                     \n",
       "[2] \"cluster.markers.sub300\"              \n",
       "[3] \"cluster.markers.sub300_no_doubletons\"\n",
       "[4] \"cluster.markers.sub300_no_singletons\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ls(pattern = \"clust\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get all tpm's for clusters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1402,
   "metadata": {},
   "outputs": [],
   "source": [
    "#  _all suffix means both set 1 and set 2 so results for all genes\n",
    "cluster.markers.sub300_no_doubletons_all = do.call(rbind, DEG.results.sub300_no_doubletons) %>%\n",
    "    select(\n",
    "        cluster = cluster, gene, \n",
    "        cluster.n.umi = set.1.n.umi,\n",
    "        other.n.umi = set.2.n.umi,\n",
    "        cluster.tpm = set.1.tpm,\n",
    "        other.tpm = set.2.tpm,\n",
    "        log2.ratio, heuristic.score) %>%\n",
    "    arrange(-heuristic.score)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2425212</li><li>8</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2425212\n",
       "\\item 8\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2425212\n",
       "2. 8\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2425212       8"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(cluster.markers.sub300_no_doubletons_all)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "exp_matrix_sub300_no_doubletons = cluster.markers.sub300_no_doubletons_all %>% \n",
    "select(cluster, gene, cluster.tpm )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [],
   "source": [
    "exp_matrix_sub300_no_doubletons = exp_matrix_sub300_no_doubletons %>%  \n",
    "pivot_wider(names_from = \"cluster\", values_from = \"cluster.tpm\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 175</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene</th><th scope=col>160</th><th scope=col>163</th><th scope=col>172</th><th scope=col>151</th><th scope=col>168</th><th scope=col>170</th><th scope=col>141</th><th scope=col>146</th><th scope=col>165</th><th scope=col>⋯</th><th scope=col>8</th><th scope=col>29</th><th scope=col>97</th><th scope=col>71</th><th scope=col>132</th><th scope=col>19</th><th scope=col>118</th><th scope=col>21</th><th scope=col>33</th><th scope=col>3</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Muc26B </td><td>11399.564008</td><td>  19.107997</td><td>   2.086528</td><td>   0.000000</td><td>0.000000</td><td> 0.00000</td><td>  8.585809</td><td>0.000000</td><td>4.304964</td><td>⋯</td><td>0.000000</td><td> 7.550007</td><td>7.848214</td><td> 5.677609</td><td> 8.641855</td><td> 6.529676</td><td> 4.74873</td><td> 3.8060725</td><td> 9.798139</td><td>11.985219</td></tr>\n",
       "\t<tr><td>Muc11A </td><td>    0.000000</td><td>7151.766859</td><td>   0.000000</td><td>   0.000000</td><td>7.590133</td><td> 0.00000</td><td>  0.000000</td><td>6.073342</td><td>0.000000</td><td>⋯</td><td>0.000000</td><td> 0.000000</td><td>0.652234</td><td> 2.025432</td><td> 1.459941</td><td> 1.399975</td><td> 2.39835</td><td> 0.0000000</td><td>11.207518</td><td> 2.809862</td></tr>\n",
       "\t<tr><td>CG7409 </td><td>    0.000000</td><td>   0.000000</td><td>4089.076906</td><td>   2.177805</td><td>0.000000</td><td> 0.00000</td><td> 41.150358</td><td>0.000000</td><td>0.000000</td><td>⋯</td><td>0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.928308</td><td> 0.000000</td><td> 0.000000</td><td> 0.00000</td><td> 0.7635414</td><td> 0.000000</td><td> 0.000000</td></tr>\n",
       "\t<tr><td>CG14421</td><td>    1.517765</td><td>   0.000000</td><td> 408.001412</td><td>8224.649810</td><td>0.000000</td><td>23.41646</td><td>526.892422</td><td>5.365068</td><td>0.000000</td><td>⋯</td><td>1.623706</td><td> 4.522774</td><td>1.953751</td><td> 1.548365</td><td>12.916647</td><td> 2.293638</td><td> 0.00000</td><td>16.8938476</td><td> 0.000000</td><td> 2.643439</td></tr>\n",
       "\t<tr><td>Alp8   </td><td>    0.000000</td><td>5261.148065</td><td>   0.000000</td><td>   0.000000</td><td>0.000000</td><td> 0.00000</td><td>  7.485179</td><td>4.319542</td><td>0.000000</td><td>⋯</td><td>1.398730</td><td> 0.000000</td><td>0.000000</td><td> 2.317245</td><td> 3.139688</td><td> 1.225833</td><td> 0.00000</td><td> 1.0344950</td><td> 2.556276</td><td> 3.374912</td></tr>\n",
       "\t<tr><td>CG7017 </td><td>12742.011542</td><td>   6.600423</td><td>   0.000000</td><td>   0.000000</td><td>0.000000</td><td> 0.00000</td><td>  4.109510</td><td>6.166064</td><td>8.669248</td><td>⋯</td><td>2.713134</td><td>10.096972</td><td>7.971661</td><td>16.008114</td><td>15.927200</td><td>14.990986</td><td>11.31910</td><td>15.2404849</td><td>33.954258</td><td>30.834494</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 175\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " gene & 160 & 163 & 172 & 151 & 168 & 170 & 141 & 146 & 165 & ⋯ & 8 & 29 & 97 & 71 & 132 & 19 & 118 & 21 & 33 & 3\\\\\n",
       " <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t Muc26B  & 11399.564008 &   19.107997 &    2.086528 &    0.000000 & 0.000000 &  0.00000 &   8.585809 & 0.000000 & 4.304964 & ⋯ & 0.000000 &  7.550007 & 7.848214 &  5.677609 &  8.641855 &  6.529676 &  4.74873 &  3.8060725 &  9.798139 & 11.985219\\\\\n",
       "\t Muc11A  &     0.000000 & 7151.766859 &    0.000000 &    0.000000 & 7.590133 &  0.00000 &   0.000000 & 6.073342 & 0.000000 & ⋯ & 0.000000 &  0.000000 & 0.652234 &  2.025432 &  1.459941 &  1.399975 &  2.39835 &  0.0000000 & 11.207518 &  2.809862\\\\\n",
       "\t CG7409  &     0.000000 &    0.000000 & 4089.076906 &    2.177805 & 0.000000 &  0.00000 &  41.150358 & 0.000000 & 0.000000 & ⋯ & 0.000000 &  0.000000 & 0.000000 &  0.928308 &  0.000000 &  0.000000 &  0.00000 &  0.7635414 &  0.000000 &  0.000000\\\\\n",
       "\t CG14421 &     1.517765 &    0.000000 &  408.001412 & 8224.649810 & 0.000000 & 23.41646 & 526.892422 & 5.365068 & 0.000000 & ⋯ & 1.623706 &  4.522774 & 1.953751 &  1.548365 & 12.916647 &  2.293638 &  0.00000 & 16.8938476 &  0.000000 &  2.643439\\\\\n",
       "\t Alp8    &     0.000000 & 5261.148065 &    0.000000 &    0.000000 & 0.000000 &  0.00000 &   7.485179 & 4.319542 & 0.000000 & ⋯ & 1.398730 &  0.000000 & 0.000000 &  2.317245 &  3.139688 &  1.225833 &  0.00000 &  1.0344950 &  2.556276 &  3.374912\\\\\n",
       "\t CG7017  & 12742.011542 &    6.600423 &    0.000000 &    0.000000 & 0.000000 &  0.00000 &   4.109510 & 6.166064 & 8.669248 & ⋯ & 2.713134 & 10.096972 & 7.971661 & 16.008114 & 15.927200 & 14.990986 & 11.31910 & 15.2404849 & 33.954258 & 30.834494\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 175\n",
       "\n",
       "| gene &lt;chr&gt; | 160 &lt;dbl&gt; | 163 &lt;dbl&gt; | 172 &lt;dbl&gt; | 151 &lt;dbl&gt; | 168 &lt;dbl&gt; | 170 &lt;dbl&gt; | 141 &lt;dbl&gt; | 146 &lt;dbl&gt; | 165 &lt;dbl&gt; | ⋯ ⋯ | 8 &lt;dbl&gt; | 29 &lt;dbl&gt; | 97 &lt;dbl&gt; | 71 &lt;dbl&gt; | 132 &lt;dbl&gt; | 19 &lt;dbl&gt; | 118 &lt;dbl&gt; | 21 &lt;dbl&gt; | 33 &lt;dbl&gt; | 3 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| Muc26B  | 11399.564008 |   19.107997 |    2.086528 |    0.000000 | 0.000000 |  0.00000 |   8.585809 | 0.000000 | 4.304964 | ⋯ | 0.000000 |  7.550007 | 7.848214 |  5.677609 |  8.641855 |  6.529676 |  4.74873 |  3.8060725 |  9.798139 | 11.985219 |\n",
       "| Muc11A  |     0.000000 | 7151.766859 |    0.000000 |    0.000000 | 7.590133 |  0.00000 |   0.000000 | 6.073342 | 0.000000 | ⋯ | 0.000000 |  0.000000 | 0.652234 |  2.025432 |  1.459941 |  1.399975 |  2.39835 |  0.0000000 | 11.207518 |  2.809862 |\n",
       "| CG7409  |     0.000000 |    0.000000 | 4089.076906 |    2.177805 | 0.000000 |  0.00000 |  41.150358 | 0.000000 | 0.000000 | ⋯ | 0.000000 |  0.000000 | 0.000000 |  0.928308 |  0.000000 |  0.000000 |  0.00000 |  0.7635414 |  0.000000 |  0.000000 |\n",
       "| CG14421 |     1.517765 |    0.000000 |  408.001412 | 8224.649810 | 0.000000 | 23.41646 | 526.892422 | 5.365068 | 0.000000 | ⋯ | 1.623706 |  4.522774 | 1.953751 |  1.548365 | 12.916647 |  2.293638 |  0.00000 | 16.8938476 |  0.000000 |  2.643439 |\n",
       "| Alp8    |     0.000000 | 5261.148065 |    0.000000 |    0.000000 | 0.000000 |  0.00000 |   7.485179 | 4.319542 | 0.000000 | ⋯ | 1.398730 |  0.000000 | 0.000000 |  2.317245 |  3.139688 |  1.225833 |  0.00000 |  1.0344950 |  2.556276 |  3.374912 |\n",
       "| CG7017  | 12742.011542 |    6.600423 |    0.000000 |    0.000000 | 0.000000 |  0.00000 |   4.109510 | 6.166064 | 8.669248 | ⋯ | 2.713134 | 10.096972 | 7.971661 | 16.008114 | 15.927200 | 14.990986 | 11.31910 | 15.2404849 | 33.954258 | 30.834494 |\n",
       "\n"
      ],
      "text/plain": [
       "  gene    160          163         172         151         168      170     \n",
       "1 Muc26B  11399.564008   19.107997    2.086528    0.000000 0.000000  0.00000\n",
       "2 Muc11A      0.000000 7151.766859    0.000000    0.000000 7.590133  0.00000\n",
       "3 CG7409      0.000000    0.000000 4089.076906    2.177805 0.000000  0.00000\n",
       "4 CG14421     1.517765    0.000000  408.001412 8224.649810 0.000000 23.41646\n",
       "5 Alp8        0.000000 5261.148065    0.000000    0.000000 0.000000  0.00000\n",
       "6 CG7017  12742.011542    6.600423    0.000000    0.000000 0.000000  0.00000\n",
       "  141        146      165      ⋯ 8        29        97       71       \n",
       "1   8.585809 0.000000 4.304964 ⋯ 0.000000  7.550007 7.848214  5.677609\n",
       "2   0.000000 6.073342 0.000000 ⋯ 0.000000  0.000000 0.652234  2.025432\n",
       "3  41.150358 0.000000 0.000000 ⋯ 0.000000  0.000000 0.000000  0.928308\n",
       "4 526.892422 5.365068 0.000000 ⋯ 1.623706  4.522774 1.953751  1.548365\n",
       "5   7.485179 4.319542 0.000000 ⋯ 1.398730  0.000000 0.000000  2.317245\n",
       "6   4.109510 6.166064 8.669248 ⋯ 2.713134 10.096972 7.971661 16.008114\n",
       "  132       19        118      21         33        3        \n",
       "1  8.641855  6.529676  4.74873  3.8060725  9.798139 11.985219\n",
       "2  1.459941  1.399975  2.39835  0.0000000 11.207518  2.809862\n",
       "3  0.000000  0.000000  0.00000  0.7635414  0.000000  0.000000\n",
       "4 12.916647  2.293638  0.00000 16.8938476  0.000000  2.643439\n",
       "5  3.139688  1.225833  0.00000  1.0344950  2.556276  3.374912\n",
       "6 15.927200 14.990986 11.31910 15.2404849 33.954258 30.834494"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "exp_matrix_sub300_no_doubletons  %>%  head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "#  need to get fly_peaks\n",
    "load(\"data/fly_TF_target_match.RData\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'cds_sub'</li><li>'cds_sub_no_doubletons'</li><li>'cds_sub_no_sing'</li><li>'cell_type'</li><li>'cell.type'</li><li>'cell.type.2.markers'</li><li>'cell.type.markers.sub300_no_singletons.cell.type'</li><li>'cell.type.markers.sub300.cell.type'</li><li>'cellIDs'</li><li>'cells'</li><li>'cellTypes'</li><li>'cluster.markers'</li><li>'cluster.markers.sub300'</li><li>'cluster.markers.sub300_no_doubletons'</li><li>'cluster.markers.sub300_no_doubletons_all'</li><li>'cluster.markers.sub300_no_singletons'</li><li>'df'</li><li>'df_sub_no_doubletons'</li><li>'df_sub_no_singletons'</li><li>'df_sub300'</li><li>'doubletons'</li><li>'exp_matrix'</li><li>'exp_matrix_mean'</li><li>'exp_matrix_sub300_no_doubletons'</li><li>'expresses.gene'</li><li>'fly_peaks_TF'</li><li>'gene_max_mean_entropy'</li><li>'get.gene.id'</li><li>'get.gene.id2'</li><li>'get.norm.expr.matrix'</li><li>'get.norm.expr.matrix.3'</li><li>'ids'</li><li>'k'</li><li>'k1'</li><li>'labels'</li><li>'make_df'</li><li>'missing_genes'</li><li>'missing_genes_aliases'</li><li>'nineteentons'</li><li>'nnMatches'</li><li>'pca'</li><li>'peedeef'</li><li>'plot.expr'</li><li>'plot.expr_clus'</li><li>'plot.expr_clus.br'</li><li>'plot.expr_clus.br2'</li><li>'plot.expr3'</li><li>'rann'</li><li>'res'</li><li>'res_label'</li><li>'singletons'</li><li>'two.set.differential.gene.test'</li><li>'two.set.differential.gene.test.3'</li><li>'two.set.differential.gene.test.3a'</li><li>'two.set.differential.gene.test.3b'</li><li>'vector_angle_emb_targets'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'cds\\_sub'\n",
       "\\item 'cds\\_sub\\_no\\_doubletons'\n",
       "\\item 'cds\\_sub\\_no\\_sing'\n",
       "\\item 'cell\\_type'\n",
       "\\item 'cell.type'\n",
       "\\item 'cell.type.2.markers'\n",
       "\\item 'cell.type.markers.sub300\\_no\\_singletons.cell.type'\n",
       "\\item 'cell.type.markers.sub300.cell.type'\n",
       "\\item 'cellIDs'\n",
       "\\item 'cells'\n",
       "\\item 'cellTypes'\n",
       "\\item 'cluster.markers'\n",
       "\\item 'cluster.markers.sub300'\n",
       "\\item 'cluster.markers.sub300\\_no\\_doubletons'\n",
       "\\item 'cluster.markers.sub300\\_no\\_doubletons\\_all'\n",
       "\\item 'cluster.markers.sub300\\_no\\_singletons'\n",
       "\\item 'df'\n",
       "\\item 'df\\_sub\\_no\\_doubletons'\n",
       "\\item 'df\\_sub\\_no\\_singletons'\n",
       "\\item 'df\\_sub300'\n",
       "\\item 'doubletons'\n",
       "\\item 'exp\\_matrix'\n",
       "\\item 'exp\\_matrix\\_mean'\n",
       "\\item 'exp\\_matrix\\_sub300\\_no\\_doubletons'\n",
       "\\item 'expresses.gene'\n",
       "\\item 'fly\\_peaks\\_TF'\n",
       "\\item 'gene\\_max\\_mean\\_entropy'\n",
       "\\item 'get.gene.id'\n",
       "\\item 'get.gene.id2'\n",
       "\\item 'get.norm.expr.matrix'\n",
       "\\item 'get.norm.expr.matrix.3'\n",
       "\\item 'ids'\n",
       "\\item 'k'\n",
       "\\item 'k1'\n",
       "\\item 'labels'\n",
       "\\item 'make\\_df'\n",
       "\\item 'missing\\_genes'\n",
       "\\item 'missing\\_genes\\_aliases'\n",
       "\\item 'nineteentons'\n",
       "\\item 'nnMatches'\n",
       "\\item 'pca'\n",
       "\\item 'peedeef'\n",
       "\\item 'plot.expr'\n",
       "\\item 'plot.expr\\_clus'\n",
       "\\item 'plot.expr\\_clus.br'\n",
       "\\item 'plot.expr\\_clus.br2'\n",
       "\\item 'plot.expr3'\n",
       "\\item 'rann'\n",
       "\\item 'res'\n",
       "\\item 'res\\_label'\n",
       "\\item 'singletons'\n",
       "\\item 'two.set.differential.gene.test'\n",
       "\\item 'two.set.differential.gene.test.3'\n",
       "\\item 'two.set.differential.gene.test.3a'\n",
       "\\item 'two.set.differential.gene.test.3b'\n",
       "\\item 'vector\\_angle\\_emb\\_targets'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'cds_sub'\n",
       "2. 'cds_sub_no_doubletons'\n",
       "3. 'cds_sub_no_sing'\n",
       "4. 'cell_type'\n",
       "5. 'cell.type'\n",
       "6. 'cell.type.2.markers'\n",
       "7. 'cell.type.markers.sub300_no_singletons.cell.type'\n",
       "8. 'cell.type.markers.sub300.cell.type'\n",
       "9. 'cellIDs'\n",
       "10. 'cells'\n",
       "11. 'cellTypes'\n",
       "12. 'cluster.markers'\n",
       "13. 'cluster.markers.sub300'\n",
       "14. 'cluster.markers.sub300_no_doubletons'\n",
       "15. 'cluster.markers.sub300_no_doubletons_all'\n",
       "16. 'cluster.markers.sub300_no_singletons'\n",
       "17. 'df'\n",
       "18. 'df_sub_no_doubletons'\n",
       "19. 'df_sub_no_singletons'\n",
       "20. 'df_sub300'\n",
       "21. 'doubletons'\n",
       "22. 'exp_matrix'\n",
       "23. 'exp_matrix_mean'\n",
       "24. 'exp_matrix_sub300_no_doubletons'\n",
       "25. 'expresses.gene'\n",
       "26. 'fly_peaks_TF'\n",
       "27. 'gene_max_mean_entropy'\n",
       "28. 'get.gene.id'\n",
       "29. 'get.gene.id2'\n",
       "30. 'get.norm.expr.matrix'\n",
       "31. 'get.norm.expr.matrix.3'\n",
       "32. 'ids'\n",
       "33. 'k'\n",
       "34. 'k1'\n",
       "35. 'labels'\n",
       "36. 'make_df'\n",
       "37. 'missing_genes'\n",
       "38. 'missing_genes_aliases'\n",
       "39. 'nineteentons'\n",
       "40. 'nnMatches'\n",
       "41. 'pca'\n",
       "42. 'peedeef'\n",
       "43. 'plot.expr'\n",
       "44. 'plot.expr_clus'\n",
       "45. 'plot.expr_clus.br'\n",
       "46. 'plot.expr_clus.br2'\n",
       "47. 'plot.expr3'\n",
       "48. 'rann'\n",
       "49. 'res'\n",
       "50. 'res_label'\n",
       "51. 'singletons'\n",
       "52. 'two.set.differential.gene.test'\n",
       "53. 'two.set.differential.gene.test.3'\n",
       "54. 'two.set.differential.gene.test.3a'\n",
       "55. 'two.set.differential.gene.test.3b'\n",
       "56. 'vector_angle_emb_targets'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"cds_sub\"                                         \n",
       " [2] \"cds_sub_no_doubletons\"                           \n",
       " [3] \"cds_sub_no_sing\"                                 \n",
       " [4] \"cell_type\"                                       \n",
       " [5] \"cell.type\"                                       \n",
       " [6] \"cell.type.2.markers\"                             \n",
       " [7] \"cell.type.markers.sub300_no_singletons.cell.type\"\n",
       " [8] \"cell.type.markers.sub300.cell.type\"              \n",
       " [9] \"cellIDs\"                                         \n",
       "[10] \"cells\"                                           \n",
       "[11] \"cellTypes\"                                       \n",
       "[12] \"cluster.markers\"                                 \n",
       "[13] \"cluster.markers.sub300\"                          \n",
       "[14] \"cluster.markers.sub300_no_doubletons\"            \n",
       "[15] \"cluster.markers.sub300_no_doubletons_all\"        \n",
       "[16] \"cluster.markers.sub300_no_singletons\"            \n",
       "[17] \"df\"                                              \n",
       "[18] \"df_sub_no_doubletons\"                            \n",
       "[19] \"df_sub_no_singletons\"                            \n",
       "[20] \"df_sub300\"                                       \n",
       "[21] \"doubletons\"                                      \n",
       "[22] \"exp_matrix\"                                      \n",
       "[23] \"exp_matrix_mean\"                                 \n",
       "[24] \"exp_matrix_sub300_no_doubletons\"                 \n",
       "[25] \"expresses.gene\"                                  \n",
       "[26] \"fly_peaks_TF\"                                    \n",
       "[27] \"gene_max_mean_entropy\"                           \n",
       "[28] \"get.gene.id\"                                     \n",
       "[29] \"get.gene.id2\"                                    \n",
       "[30] \"get.norm.expr.matrix\"                            \n",
       "[31] \"get.norm.expr.matrix.3\"                          \n",
       "[32] \"ids\"                                             \n",
       "[33] \"k\"                                               \n",
       "[34] \"k1\"                                              \n",
       "[35] \"labels\"                                          \n",
       "[36] \"make_df\"                                         \n",
       "[37] \"missing_genes\"                                   \n",
       "[38] \"missing_genes_aliases\"                           \n",
       "[39] \"nineteentons\"                                    \n",
       "[40] \"nnMatches\"                                       \n",
       "[41] \"pca\"                                             \n",
       "[42] \"peedeef\"                                         \n",
       "[43] \"plot.expr\"                                       \n",
       "[44] \"plot.expr_clus\"                                  \n",
       "[45] \"plot.expr_clus.br\"                               \n",
       "[46] \"plot.expr_clus.br2\"                              \n",
       "[47] \"plot.expr3\"                                      \n",
       "[48] \"rann\"                                            \n",
       "[49] \"res\"                                             \n",
       "[50] \"res_label\"                                       \n",
       "[51] \"singletons\"                                      \n",
       "[52] \"two.set.differential.gene.test\"                  \n",
       "[53] \"two.set.differential.gene.test.3\"                \n",
       "[54] \"two.set.differential.gene.test.3a\"               \n",
       "[55] \"two.set.differential.gene.test.3b\"               \n",
       "[56] \"vector_angle_emb_targets\"                        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ls()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [],
   "source": [
    "fly_TFs = c(unique(fly_peaks_TF$TF))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_exp_matrix_sub300_no_doubletons = exp_matrix_sub300_no_doubletons %>%  \n",
    "filter(gene %in% fly_TFs)  %>% \n",
    "column_to_rownames(\"gene\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_exp_matrix_sub300_no_doubletons_log = log2(TF_exp_matrix_sub300_no_doubletons + 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1416,
   "metadata": {},
   "outputs": [],
   "source": [
    "options(repr.plot.width=20, repr.plot.height=60)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in pheatmap(TF_exp_matrix_sub300_no_doubletons_log): could not find function \"pheatmap\"\n",
     "output_type": "error",
     "traceback": [
      "Error in pheatmap(TF_exp_matrix_sub300_no_doubletons_log): could not find function \"pheatmap\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "pheatmap(TF_exp_matrix_sub300_no_doubletons_log)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Try removing anything with 6 neighbors of less"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1421,
   "metadata": {},
   "outputs": [],
   "source": [
    "options(repr.plot.width=16, repr.plot.height=12)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [],
   "source": [
    "multitons = (res %>% filter(nk <= 4) )$ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1426,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "92094"
      ],
      "text/latex": [
       "92094"
      ],
      "text/markdown": [
       "92094"
      ],
      "text/plain": [
       "[1] 92094"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(multitons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "#cell.type = \"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% multitons), size = ifelse(df_sub300$cell %in% multitons, 0.1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"cells with <= 4 matches\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "cds_sub_no_multitons = cds_sub[, !(df_sub300$cell %in% multitons)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sub_no_multitons = make_df(cds_sub_no_multitons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1430,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1932 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cluster</th><th scope=col>annot</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>amnioserosa                         </td><td>2284</td></tr>\n",
       "\t<tr><td>1</td><td>brain                               </td><td>   1</td></tr>\n",
       "\t<tr><td>1</td><td>brain primordium                    </td><td>   1</td></tr>\n",
       "\t<tr><td>1</td><td>ectoderm primordium                 </td><td>   8</td></tr>\n",
       "\t<tr><td>1</td><td>epidermis primordium                </td><td>   9</td></tr>\n",
       "\t<tr><td>1</td><td>midgut primordium                   </td><td>   2</td></tr>\n",
       "\t<tr><td>1</td><td>peripheral nervous system           </td><td>   1</td></tr>\n",
       "\t<tr><td>1</td><td>plasmatocytes anlage                </td><td>   1</td></tr>\n",
       "\t<tr><td>1</td><td>procrystal cell                     </td><td>   5</td></tr>\n",
       "\t<tr><td>1</td><td>ubiquitous                          </td><td>   1</td></tr>\n",
       "\t<tr><td>1</td><td>visceral muscle primordium          </td><td>   2</td></tr>\n",
       "\t<tr><td>2</td><td>amnioserosa                         </td><td>   3</td></tr>\n",
       "\t<tr><td>2</td><td>brain                               </td><td>  15</td></tr>\n",
       "\t<tr><td>2</td><td>epidermis                           </td><td> 395</td></tr>\n",
       "\t<tr><td>2</td><td>fat body                            </td><td>   3</td></tr>\n",
       "\t<tr><td>2</td><td>foregut / hindgut / anal pad        </td><td>   1</td></tr>\n",
       "\t<tr><td>2</td><td>hindgut / anal pad                  </td><td>   1</td></tr>\n",
       "\t<tr><td>2</td><td>midgut                              </td><td> 259</td></tr>\n",
       "\t<tr><td>2</td><td>muscle                              </td><td>   2</td></tr>\n",
       "\t<tr><td>2</td><td>plasmatocytes                       </td><td>   4</td></tr>\n",
       "\t<tr><td>2</td><td>proventriculus                      </td><td> 832</td></tr>\n",
       "\t<tr><td>2</td><td>sensory nervous system              </td><td>  17</td></tr>\n",
       "\t<tr><td>2</td><td>somatic muscle                      </td><td>  46</td></tr>\n",
       "\t<tr><td>2</td><td>tracheal system                     </td><td>   7</td></tr>\n",
       "\t<tr><td>2</td><td>ubiquitous                          </td><td>   2</td></tr>\n",
       "\t<tr><td>2</td><td>ventral nerve cord                  </td><td>   1</td></tr>\n",
       "\t<tr><td>2</td><td>visceral muscle                     </td><td>   2</td></tr>\n",
       "\t<tr><td>3</td><td>brain                               </td><td> 629</td></tr>\n",
       "\t<tr><td>3</td><td>epidermis                           </td><td>   1</td></tr>\n",
       "\t<tr><td>3</td><td>midgut                              </td><td> 144</td></tr>\n",
       "\t<tr><td>3</td><td>muscle                              </td><td>   2</td></tr>\n",
       "\t<tr><td>3</td><td>ubiquitous                          </td><td>  43</td></tr>\n",
       "\t<tr><td>4</td><td>brain                               </td><td>  89</td></tr>\n",
       "\t<tr><td>4</td><td>epidermis                           </td><td>  97</td></tr>\n",
       "\t<tr><td>4</td><td>midgut                              </td><td>  43</td></tr>\n",
       "\t<tr><td>4</td><td>muscle                              </td><td>   1</td></tr>\n",
       "\t<tr><td>4</td><td>ubiquitous                          </td><td> 802</td></tr>\n",
       "\t<tr><td>5</td><td>amnioserosa                         </td><td>   1</td></tr>\n",
       "\t<tr><td>5</td><td>brain                               </td><td>  11</td></tr>\n",
       "\t<tr><td>5</td><td>epidermis                           </td><td> 146</td></tr>\n",
       "\t<tr><td>5</td><td>fat body                            </td><td>   7</td></tr>\n",
       "\t<tr><td>5</td><td>hindgut / anal pad                  </td><td>   1</td></tr>\n",
       "\t<tr><td>5</td><td>midgut                              </td><td>1615</td></tr>\n",
       "\t<tr><td>5</td><td>midgut primordium                   </td><td>  14</td></tr>\n",
       "\t<tr><td>5</td><td>plasmatocytes                       </td><td>   2</td></tr>\n",
       "\t<tr><td>5</td><td>proventriculus                      </td><td>   3</td></tr>\n",
       "\t<tr><td>5</td><td>salivary gland                      </td><td>   1</td></tr>\n",
       "\t<tr><td>5</td><td>sensory nervous system              </td><td>   1</td></tr>\n",
       "\t<tr><td>5</td><td>somatic muscle                      </td><td>  18</td></tr>\n",
       "\t<tr><td>5</td><td>tracheal system                     </td><td>   2</td></tr>\n",
       "\t<tr><td>5</td><td>ubiquitous                          </td><td>   2</td></tr>\n",
       "\t<tr><td>5</td><td>ventral nerve cord                  </td><td>   3</td></tr>\n",
       "\t<tr><td>6</td><td>brain                               </td><td>  64</td></tr>\n",
       "\t<tr><td>6</td><td>epidermis                           </td><td> 106</td></tr>\n",
       "\t<tr><td>6</td><td>fat body                            </td><td>   1</td></tr>\n",
       "\t<tr><td>6</td><td>foregut / hindgut / anal pad        </td><td>   2</td></tr>\n",
       "\t<tr><td>6</td><td>midgut                              </td><td>  47</td></tr>\n",
       "\t<tr><td>6</td><td>muscle                              </td><td>   7</td></tr>\n",
       "\t<tr><td>6</td><td>ubiquitous                          </td><td>1051</td></tr>\n",
       "\t<tr><td>7</td><td>amnioserosa                         </td><td>   3</td></tr>\n",
       "\t<tr><td>7</td><td>brain                               </td><td>  24</td></tr>\n",
       "\t<tr><td>7</td><td>brain primordium                    </td><td> 158</td></tr>\n",
       "\t<tr><td>7</td><td>ectoderm anlage in statu nascendi   </td><td>   1</td></tr>\n",
       "\t<tr><td>7</td><td>ectoderm primordium                 </td><td> 554</td></tr>\n",
       "\t<tr><td>7</td><td>endoderm primordium                 </td><td>   6</td></tr>\n",
       "\t<tr><td>7</td><td>epidermis                           </td><td>   5</td></tr>\n",
       "\t<tr><td>7</td><td>epidermis primordium                </td><td> 214</td></tr>\n",
       "\t<tr><td>7</td><td>foregut / hindgut primordium        </td><td>  38</td></tr>\n",
       "\t<tr><td>7</td><td>germ cell                           </td><td>   1</td></tr>\n",
       "\t<tr><td>7</td><td>hindgut primordium                  </td><td>   1</td></tr>\n",
       "\t<tr><td>7</td><td>mesoderm primordium                 </td><td>  85</td></tr>\n",
       "\t<tr><td>7</td><td>midgut primordium                   </td><td>   1</td></tr>\n",
       "\t<tr><td>7</td><td>peripheral nervous system           </td><td>  25</td></tr>\n",
       "\t<tr><td>7</td><td>peripheral nervous system primordium</td><td>  16</td></tr>\n",
       "\t<tr><td>7</td><td>plasmatocytes anlage                </td><td>   3</td></tr>\n",
       "\t<tr><td>7</td><td>sensory nervous system primordium   </td><td>   7</td></tr>\n",
       "\t<tr><td>7</td><td>tracheal primordium                 </td><td>   2</td></tr>\n",
       "\t<tr><td>7</td><td>ubiquitous                          </td><td>   1</td></tr>\n",
       "\t<tr><td>7</td><td>ventral nerve cord primordium       </td><td>   4</td></tr>\n",
       "\t<tr><td>7</td><td>visceral muscle primordium          </td><td>   3</td></tr>\n",
       "\t<tr><td>8</td><td>Malpighian tubule                   </td><td>   8</td></tr>\n",
       "\t<tr><td>8</td><td>brain                               </td><td> 343</td></tr>\n",
       "\t<tr><td>8</td><td>epidermis                           </td><td> 309</td></tr>\n",
       "\t<tr><td>8</td><td>fat body                            </td><td>   6</td></tr>\n",
       "\t<tr><td>8</td><td>midgut                              </td><td>  19</td></tr>\n",
       "\t<tr><td>8</td><td>muscle                              </td><td>   3</td></tr>\n",
       "\t<tr><td>8</td><td>plasmatocytes                       </td><td>   5</td></tr>\n",
       "\t<tr><td>8</td><td>proventriculus                      </td><td>   1</td></tr>\n",
       "\t<tr><td>8</td><td>sensory nervous system              </td><td> 129</td></tr>\n",
       "\t<tr><td>8</td><td>somatic muscle                      </td><td> 526</td></tr>\n",
       "\t<tr><td>8</td><td>tracheal system                     </td><td>   8</td></tr>\n",
       "\t<tr><td>8</td><td>ubiquitous                          </td><td>   3</td></tr>\n",
       "\t<tr><td>8</td><td>ventral nerve cord                  </td><td>  31</td></tr>\n",
       "\t<tr><td>8</td><td>visceral muscle                     </td><td>   8</td></tr>\n",
       "\t<tr><td>9</td><td>brain                               </td><td>   1</td></tr>\n",
       "\t<tr><td>9</td><td>epidermis                           </td><td>   1</td></tr>\n",
       "\t<tr><td>9</td><td>fat body                            </td><td>  16</td></tr>\n",
       "\t<tr><td>9</td><td>foregut / hindgut / anal pad        </td><td>   1</td></tr>\n",
       "\t<tr><td>9</td><td>muscle                              </td><td>   1</td></tr>\n",
       "\t<tr><td>9</td><td>plasmatocytes                       </td><td>   6</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>161</td><td>fat body                     </td><td>  2</td></tr>\n",
       "\t<tr><td>161</td><td>glia                         </td><td>237</td></tr>\n",
       "\t<tr><td>161</td><td>midline primordium           </td><td>  2</td></tr>\n",
       "\t<tr><td>161</td><td>muscle                       </td><td>  1</td></tr>\n",
       "\t<tr><td>161</td><td>plasmatocytes                </td><td>  1</td></tr>\n",
       "\t<tr><td>161</td><td>somatic muscle               </td><td>  2</td></tr>\n",
       "\t<tr><td>161</td><td>ubiquitous                   </td><td>  1</td></tr>\n",
       "\t<tr><td>162</td><td>brain                        </td><td>  5</td></tr>\n",
       "\t<tr><td>162</td><td>brain primordium             </td><td>119</td></tr>\n",
       "\t<tr><td>162</td><td>ectoderm primordium          </td><td>  4</td></tr>\n",
       "\t<tr><td>162</td><td>epidermis primordium         </td><td>  5</td></tr>\n",
       "\t<tr><td>162</td><td>mesoderm primordium          </td><td>  1</td></tr>\n",
       "\t<tr><td>162</td><td>midline primordium           </td><td> 61</td></tr>\n",
       "\t<tr><td>162</td><td>peripheral nervous system    </td><td>  1</td></tr>\n",
       "\t<tr><td>162</td><td>ventral nerve cord primordium</td><td> 20</td></tr>\n",
       "\t<tr><td>163</td><td>Malpighian tubule            </td><td>161</td></tr>\n",
       "\t<tr><td>163</td><td>amnioserosa                  </td><td>  1</td></tr>\n",
       "\t<tr><td>163</td><td>brain                        </td><td>  2</td></tr>\n",
       "\t<tr><td>163</td><td>epidermis                    </td><td>  3</td></tr>\n",
       "\t<tr><td>163</td><td>fat body                     </td><td>  2</td></tr>\n",
       "\t<tr><td>163</td><td>midgut                       </td><td> 46</td></tr>\n",
       "\t<tr><td>163</td><td>muscle                       </td><td>  1</td></tr>\n",
       "\t<tr><td>163</td><td>plasmatocytes                </td><td>  1</td></tr>\n",
       "\t<tr><td>163</td><td>sensory nervous system       </td><td>  1</td></tr>\n",
       "\t<tr><td>163</td><td>somatic muscle               </td><td>  1</td></tr>\n",
       "\t<tr><td>163</td><td>tracheal primordium          </td><td>  4</td></tr>\n",
       "\t<tr><td>163</td><td>tracheal system              </td><td>  1</td></tr>\n",
       "\t<tr><td>163</td><td>ubiquitous                   </td><td> 14</td></tr>\n",
       "\t<tr><td>163</td><td>unknown                      </td><td>  3</td></tr>\n",
       "\t<tr><td>164</td><td>brain                        </td><td>  9</td></tr>\n",
       "\t<tr><td>164</td><td>brain primordium             </td><td>  6</td></tr>\n",
       "\t<tr><td>164</td><td>epidermis                    </td><td>  2</td></tr>\n",
       "\t<tr><td>164</td><td>epidermis primordium         </td><td>  8</td></tr>\n",
       "\t<tr><td>164</td><td>germ cell                    </td><td> 93</td></tr>\n",
       "\t<tr><td>164</td><td>hindgut primordium           </td><td>  1</td></tr>\n",
       "\t<tr><td>164</td><td>midgut primordium            </td><td>  1</td></tr>\n",
       "\t<tr><td>164</td><td>midline primordium           </td><td>  1</td></tr>\n",
       "\t<tr><td>164</td><td>peripheral nervous system    </td><td>  5</td></tr>\n",
       "\t<tr><td>164</td><td>plasmatocytes anlage         </td><td>  3</td></tr>\n",
       "\t<tr><td>164</td><td>procrystal cell              </td><td>  6</td></tr>\n",
       "\t<tr><td>164</td><td>somatic muscle               </td><td>  2</td></tr>\n",
       "\t<tr><td>164</td><td>somatic muscle primordium    </td><td>  1</td></tr>\n",
       "\t<tr><td>164</td><td>ubiquitous                   </td><td> 15</td></tr>\n",
       "\t<tr><td>164</td><td>ventral nerve cord           </td><td>  1</td></tr>\n",
       "\t<tr><td>164</td><td>visceral muscle primordium   </td><td>  2</td></tr>\n",
       "\t<tr><td>165</td><td>amnioserosa                  </td><td>  1</td></tr>\n",
       "\t<tr><td>165</td><td>crystal cell                 </td><td>  7</td></tr>\n",
       "\t<tr><td>165</td><td>epidermis                    </td><td> 87</td></tr>\n",
       "\t<tr><td>165</td><td>procrystal cell              </td><td>114</td></tr>\n",
       "\t<tr><td>165</td><td>somatic muscle               </td><td>  1</td></tr>\n",
       "\t<tr><td>165</td><td>ubiquitous                   </td><td> 49</td></tr>\n",
       "\t<tr><td>166</td><td>brain                        </td><td> 20</td></tr>\n",
       "\t<tr><td>166</td><td>epidermis                    </td><td>  7</td></tr>\n",
       "\t<tr><td>166</td><td>midgut                       </td><td>  1</td></tr>\n",
       "\t<tr><td>166</td><td>muscle                       </td><td>  5</td></tr>\n",
       "\t<tr><td>166</td><td>ubiquitous                   </td><td>150</td></tr>\n",
       "\t<tr><td>166</td><td>yolk nuclei                  </td><td>  1</td></tr>\n",
       "\t<tr><td>167</td><td>brain                        </td><td>  2</td></tr>\n",
       "\t<tr><td>167</td><td>epidermis                    </td><td>  2</td></tr>\n",
       "\t<tr><td>167</td><td>fat body                     </td><td>141</td></tr>\n",
       "\t<tr><td>167</td><td>tracheal primordium          </td><td>  3</td></tr>\n",
       "\t<tr><td>167</td><td>tracheal system              </td><td>  1</td></tr>\n",
       "\t<tr><td>167</td><td>ubiquitous                   </td><td>  4</td></tr>\n",
       "\t<tr><td>167</td><td>unknown                      </td><td> 78</td></tr>\n",
       "\t<tr><td>168</td><td>brain                        </td><td>  1</td></tr>\n",
       "\t<tr><td>168</td><td>epidermis                    </td><td>  7</td></tr>\n",
       "\t<tr><td>168</td><td>muscle                       </td><td>  1</td></tr>\n",
       "\t<tr><td>168</td><td>somatic muscle               </td><td>  2</td></tr>\n",
       "\t<tr><td>168</td><td>tracheal primordium          </td><td> 45</td></tr>\n",
       "\t<tr><td>168</td><td>tracheal system              </td><td> 30</td></tr>\n",
       "\t<tr><td>168</td><td>ubiquitous                   </td><td>  2</td></tr>\n",
       "\t<tr><td>169</td><td>brain                        </td><td>  5</td></tr>\n",
       "\t<tr><td>169</td><td>epidermis                    </td><td>103</td></tr>\n",
       "\t<tr><td>169</td><td>foregut / hindgut / anal pad </td><td> 29</td></tr>\n",
       "\t<tr><td>169</td><td>somatic muscle               </td><td>  1</td></tr>\n",
       "\t<tr><td>169</td><td>ubiquitous                   </td><td>  8</td></tr>\n",
       "\t<tr><td>170</td><td>brain                        </td><td>  3</td></tr>\n",
       "\t<tr><td>170</td><td>epidermis                    </td><td>  1</td></tr>\n",
       "\t<tr><td>170</td><td>sensory nervous system       </td><td> 54</td></tr>\n",
       "\t<tr><td>170</td><td>ubiquitous                   </td><td>  1</td></tr>\n",
       "\t<tr><td>170</td><td>ventral nerve cord           </td><td> 33</td></tr>\n",
       "\t<tr><td>171</td><td>brain                        </td><td> 13</td></tr>\n",
       "\t<tr><td>171</td><td>epidermis                    </td><td>  2</td></tr>\n",
       "\t<tr><td>171</td><td>midgut                       </td><td>  2</td></tr>\n",
       "\t<tr><td>171</td><td>sensory nervous system       </td><td>  1</td></tr>\n",
       "\t<tr><td>171</td><td>somatic muscle               </td><td>  6</td></tr>\n",
       "\t<tr><td>171</td><td>ubiquitous                   </td><td>  1</td></tr>\n",
       "\t<tr><td>171</td><td>ventral nerve cord           </td><td> 58</td></tr>\n",
       "\t<tr><td>172</td><td>epidermis                    </td><td>  5</td></tr>\n",
       "\t<tr><td>172</td><td>midgut                       </td><td>  1</td></tr>\n",
       "\t<tr><td>172</td><td>sensory nervous system       </td><td>  2</td></tr>\n",
       "\t<tr><td>172</td><td>tracheal primordium          </td><td>  3</td></tr>\n",
       "\t<tr><td>172</td><td>ubiquitous                   </td><td>  9</td></tr>\n",
       "\t<tr><td>172</td><td>unknown                      </td><td> 25</td></tr>\n",
       "\t<tr><td>173</td><td>brain                        </td><td>  5</td></tr>\n",
       "\t<tr><td>173</td><td>epidermis                    </td><td>  3</td></tr>\n",
       "\t<tr><td>173</td><td>ubiquitous                   </td><td> 55</td></tr>\n",
       "\t<tr><td>174</td><td>brain                        </td><td>  7</td></tr>\n",
       "\t<tr><td>174</td><td>epidermis                    </td><td>  6</td></tr>\n",
       "\t<tr><td>174</td><td>ubiquitous                   </td><td> 27</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1932 × 3\n",
       "\\begin{tabular}{lll}\n",
       " cluster & annot & n\\\\\n",
       " <fct> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 & amnioserosa                          & 2284\\\\\n",
       "\t 1 & brain                                &    1\\\\\n",
       "\t 1 & brain primordium                     &    1\\\\\n",
       "\t 1 & ectoderm primordium                  &    8\\\\\n",
       "\t 1 & epidermis primordium                 &    9\\\\\n",
       "\t 1 & midgut primordium                    &    2\\\\\n",
       "\t 1 & peripheral nervous system            &    1\\\\\n",
       "\t 1 & plasmatocytes anlage                 &    1\\\\\n",
       "\t 1 & procrystal cell                      &    5\\\\\n",
       "\t 1 & ubiquitous                           &    1\\\\\n",
       "\t 1 & visceral muscle primordium           &    2\\\\\n",
       "\t 2 & amnioserosa                          &    3\\\\\n",
       "\t 2 & brain                                &   15\\\\\n",
       "\t 2 & epidermis                            &  395\\\\\n",
       "\t 2 & fat body                             &    3\\\\\n",
       "\t 2 & foregut / hindgut / anal pad         &    1\\\\\n",
       "\t 2 & hindgut / anal pad                   &    1\\\\\n",
       "\t 2 & midgut                               &  259\\\\\n",
       "\t 2 & muscle                               &    2\\\\\n",
       "\t 2 & plasmatocytes                        &    4\\\\\n",
       "\t 2 & proventriculus                       &  832\\\\\n",
       "\t 2 & sensory nervous system               &   17\\\\\n",
       "\t 2 & somatic muscle                       &   46\\\\\n",
       "\t 2 & tracheal system                      &    7\\\\\n",
       "\t 2 & ubiquitous                           &    2\\\\\n",
       "\t 2 & ventral nerve cord                   &    1\\\\\n",
       "\t 2 & visceral muscle                      &    2\\\\\n",
       "\t 3 & brain                                &  629\\\\\n",
       "\t 3 & epidermis                            &    1\\\\\n",
       "\t 3 & midgut                               &  144\\\\\n",
       "\t 3 & muscle                               &    2\\\\\n",
       "\t 3 & ubiquitous                           &   43\\\\\n",
       "\t 4 & brain                                &   89\\\\\n",
       "\t 4 & epidermis                            &   97\\\\\n",
       "\t 4 & midgut                               &   43\\\\\n",
       "\t 4 & muscle                               &    1\\\\\n",
       "\t 4 & ubiquitous                           &  802\\\\\n",
       "\t 5 & amnioserosa                          &    1\\\\\n",
       "\t 5 & brain                                &   11\\\\\n",
       "\t 5 & epidermis                            &  146\\\\\n",
       "\t 5 & fat body                             &    7\\\\\n",
       "\t 5 & hindgut / anal pad                   &    1\\\\\n",
       "\t 5 & midgut                               & 1615\\\\\n",
       "\t 5 & midgut primordium                    &   14\\\\\n",
       "\t 5 & plasmatocytes                        &    2\\\\\n",
       "\t 5 & proventriculus                       &    3\\\\\n",
       "\t 5 & salivary gland                       &    1\\\\\n",
       "\t 5 & sensory nervous system               &    1\\\\\n",
       "\t 5 & somatic muscle                       &   18\\\\\n",
       "\t 5 & tracheal system                      &    2\\\\\n",
       "\t 5 & ubiquitous                           &    2\\\\\n",
       "\t 5 & ventral nerve cord                   &    3\\\\\n",
       "\t 6 & brain                                &   64\\\\\n",
       "\t 6 & epidermis                            &  106\\\\\n",
       "\t 6 & fat body                             &    1\\\\\n",
       "\t 6 & foregut / hindgut / anal pad         &    2\\\\\n",
       "\t 6 & midgut                               &   47\\\\\n",
       "\t 6 & muscle                               &    7\\\\\n",
       "\t 6 & ubiquitous                           & 1051\\\\\n",
       "\t 7 & amnioserosa                          &    3\\\\\n",
       "\t 7 & brain                                &   24\\\\\n",
       "\t 7 & brain primordium                     &  158\\\\\n",
       "\t 7 & ectoderm anlage in statu nascendi    &    1\\\\\n",
       "\t 7 & ectoderm primordium                  &  554\\\\\n",
       "\t 7 & endoderm primordium                  &    6\\\\\n",
       "\t 7 & epidermis                            &    5\\\\\n",
       "\t 7 & epidermis primordium                 &  214\\\\\n",
       "\t 7 & foregut / hindgut primordium         &   38\\\\\n",
       "\t 7 & germ cell                            &    1\\\\\n",
       "\t 7 & hindgut primordium                   &    1\\\\\n",
       "\t 7 & mesoderm primordium                  &   85\\\\\n",
       "\t 7 & midgut primordium                    &    1\\\\\n",
       "\t 7 & peripheral nervous system            &   25\\\\\n",
       "\t 7 & peripheral nervous system primordium &   16\\\\\n",
       "\t 7 & plasmatocytes anlage                 &    3\\\\\n",
       "\t 7 & sensory nervous system primordium    &    7\\\\\n",
       "\t 7 & tracheal primordium                  &    2\\\\\n",
       "\t 7 & ubiquitous                           &    1\\\\\n",
       "\t 7 & ventral nerve cord primordium        &    4\\\\\n",
       "\t 7 & visceral muscle primordium           &    3\\\\\n",
       "\t 8 & Malpighian tubule                    &    8\\\\\n",
       "\t 8 & brain                                &  343\\\\\n",
       "\t 8 & epidermis                            &  309\\\\\n",
       "\t 8 & fat body                             &    6\\\\\n",
       "\t 8 & midgut                               &   19\\\\\n",
       "\t 8 & muscle                               &    3\\\\\n",
       "\t 8 & plasmatocytes                        &    5\\\\\n",
       "\t 8 & proventriculus                       &    1\\\\\n",
       "\t 8 & sensory nervous system               &  129\\\\\n",
       "\t 8 & somatic muscle                       &  526\\\\\n",
       "\t 8 & tracheal system                      &    8\\\\\n",
       "\t 8 & ubiquitous                           &    3\\\\\n",
       "\t 8 & ventral nerve cord                   &   31\\\\\n",
       "\t 8 & visceral muscle                      &    8\\\\\n",
       "\t 9 & brain                                &    1\\\\\n",
       "\t 9 & epidermis                            &    1\\\\\n",
       "\t 9 & fat body                             &   16\\\\\n",
       "\t 9 & foregut / hindgut / anal pad         &    1\\\\\n",
       "\t 9 & muscle                               &    1\\\\\n",
       "\t 9 & plasmatocytes                        &    6\\\\\n",
       "\t ⋮ & ⋮ & ⋮\\\\\n",
       "\t 161 & fat body                      &   2\\\\\n",
       "\t 161 & glia                          & 237\\\\\n",
       "\t 161 & midline primordium            &   2\\\\\n",
       "\t 161 & muscle                        &   1\\\\\n",
       "\t 161 & plasmatocytes                 &   1\\\\\n",
       "\t 161 & somatic muscle                &   2\\\\\n",
       "\t 161 & ubiquitous                    &   1\\\\\n",
       "\t 162 & brain                         &   5\\\\\n",
       "\t 162 & brain primordium              & 119\\\\\n",
       "\t 162 & ectoderm primordium           &   4\\\\\n",
       "\t 162 & epidermis primordium          &   5\\\\\n",
       "\t 162 & mesoderm primordium           &   1\\\\\n",
       "\t 162 & midline primordium            &  61\\\\\n",
       "\t 162 & peripheral nervous system     &   1\\\\\n",
       "\t 162 & ventral nerve cord primordium &  20\\\\\n",
       "\t 163 & Malpighian tubule             & 161\\\\\n",
       "\t 163 & amnioserosa                   &   1\\\\\n",
       "\t 163 & brain                         &   2\\\\\n",
       "\t 163 & epidermis                     &   3\\\\\n",
       "\t 163 & fat body                      &   2\\\\\n",
       "\t 163 & midgut                        &  46\\\\\n",
       "\t 163 & muscle                        &   1\\\\\n",
       "\t 163 & plasmatocytes                 &   1\\\\\n",
       "\t 163 & sensory nervous system        &   1\\\\\n",
       "\t 163 & somatic muscle                &   1\\\\\n",
       "\t 163 & tracheal primordium           &   4\\\\\n",
       "\t 163 & tracheal system               &   1\\\\\n",
       "\t 163 & ubiquitous                    &  14\\\\\n",
       "\t 163 & unknown                       &   3\\\\\n",
       "\t 164 & brain                         &   9\\\\\n",
       "\t 164 & brain primordium              &   6\\\\\n",
       "\t 164 & epidermis                     &   2\\\\\n",
       "\t 164 & epidermis primordium          &   8\\\\\n",
       "\t 164 & germ cell                     &  93\\\\\n",
       "\t 164 & hindgut primordium            &   1\\\\\n",
       "\t 164 & midgut primordium             &   1\\\\\n",
       "\t 164 & midline primordium            &   1\\\\\n",
       "\t 164 & peripheral nervous system     &   5\\\\\n",
       "\t 164 & plasmatocytes anlage          &   3\\\\\n",
       "\t 164 & procrystal cell               &   6\\\\\n",
       "\t 164 & somatic muscle                &   2\\\\\n",
       "\t 164 & somatic muscle primordium     &   1\\\\\n",
       "\t 164 & ubiquitous                    &  15\\\\\n",
       "\t 164 & ventral nerve cord            &   1\\\\\n",
       "\t 164 & visceral muscle primordium    &   2\\\\\n",
       "\t 165 & amnioserosa                   &   1\\\\\n",
       "\t 165 & crystal cell                  &   7\\\\\n",
       "\t 165 & epidermis                     &  87\\\\\n",
       "\t 165 & procrystal cell               & 114\\\\\n",
       "\t 165 & somatic muscle                &   1\\\\\n",
       "\t 165 & ubiquitous                    &  49\\\\\n",
       "\t 166 & brain                         &  20\\\\\n",
       "\t 166 & epidermis                     &   7\\\\\n",
       "\t 166 & midgut                        &   1\\\\\n",
       "\t 166 & muscle                        &   5\\\\\n",
       "\t 166 & ubiquitous                    & 150\\\\\n",
       "\t 166 & yolk nuclei                   &   1\\\\\n",
       "\t 167 & brain                         &   2\\\\\n",
       "\t 167 & epidermis                     &   2\\\\\n",
       "\t 167 & fat body                      & 141\\\\\n",
       "\t 167 & tracheal primordium           &   3\\\\\n",
       "\t 167 & tracheal system               &   1\\\\\n",
       "\t 167 & ubiquitous                    &   4\\\\\n",
       "\t 167 & unknown                       &  78\\\\\n",
       "\t 168 & brain                         &   1\\\\\n",
       "\t 168 & epidermis                     &   7\\\\\n",
       "\t 168 & muscle                        &   1\\\\\n",
       "\t 168 & somatic muscle                &   2\\\\\n",
       "\t 168 & tracheal primordium           &  45\\\\\n",
       "\t 168 & tracheal system               &  30\\\\\n",
       "\t 168 & ubiquitous                    &   2\\\\\n",
       "\t 169 & brain                         &   5\\\\\n",
       "\t 169 & epidermis                     & 103\\\\\n",
       "\t 169 & foregut / hindgut / anal pad  &  29\\\\\n",
       "\t 169 & somatic muscle                &   1\\\\\n",
       "\t 169 & ubiquitous                    &   8\\\\\n",
       "\t 170 & brain                         &   3\\\\\n",
       "\t 170 & epidermis                     &   1\\\\\n",
       "\t 170 & sensory nervous system        &  54\\\\\n",
       "\t 170 & ubiquitous                    &   1\\\\\n",
       "\t 170 & ventral nerve cord            &  33\\\\\n",
       "\t 171 & brain                         &  13\\\\\n",
       "\t 171 & epidermis                     &   2\\\\\n",
       "\t 171 & midgut                        &   2\\\\\n",
       "\t 171 & sensory nervous system        &   1\\\\\n",
       "\t 171 & somatic muscle                &   6\\\\\n",
       "\t 171 & ubiquitous                    &   1\\\\\n",
       "\t 171 & ventral nerve cord            &  58\\\\\n",
       "\t 172 & epidermis                     &   5\\\\\n",
       "\t 172 & midgut                        &   1\\\\\n",
       "\t 172 & sensory nervous system        &   2\\\\\n",
       "\t 172 & tracheal primordium           &   3\\\\\n",
       "\t 172 & ubiquitous                    &   9\\\\\n",
       "\t 172 & unknown                       &  25\\\\\n",
       "\t 173 & brain                         &   5\\\\\n",
       "\t 173 & epidermis                     &   3\\\\\n",
       "\t 173 & ubiquitous                    &  55\\\\\n",
       "\t 174 & brain                         &   7\\\\\n",
       "\t 174 & epidermis                     &   6\\\\\n",
       "\t 174 & ubiquitous                    &  27\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1932 × 3\n",
       "\n",
       "| cluster &lt;fct&gt; | annot &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| 1 | amnioserosa                          | 2284 |\n",
       "| 1 | brain                                |    1 |\n",
       "| 1 | brain primordium                     |    1 |\n",
       "| 1 | ectoderm primordium                  |    8 |\n",
       "| 1 | epidermis primordium                 |    9 |\n",
       "| 1 | midgut primordium                    |    2 |\n",
       "| 1 | peripheral nervous system            |    1 |\n",
       "| 1 | plasmatocytes anlage                 |    1 |\n",
       "| 1 | procrystal cell                      |    5 |\n",
       "| 1 | ubiquitous                           |    1 |\n",
       "| 1 | visceral muscle primordium           |    2 |\n",
       "| 2 | amnioserosa                          |    3 |\n",
       "| 2 | brain                                |   15 |\n",
       "| 2 | epidermis                            |  395 |\n",
       "| 2 | fat body                             |    3 |\n",
       "| 2 | foregut / hindgut / anal pad         |    1 |\n",
       "| 2 | hindgut / anal pad                   |    1 |\n",
       "| 2 | midgut                               |  259 |\n",
       "| 2 | muscle                               |    2 |\n",
       "| 2 | plasmatocytes                        |    4 |\n",
       "| 2 | proventriculus                       |  832 |\n",
       "| 2 | sensory nervous system               |   17 |\n",
       "| 2 | somatic muscle                       |   46 |\n",
       "| 2 | tracheal system                      |    7 |\n",
       "| 2 | ubiquitous                           |    2 |\n",
       "| 2 | ventral nerve cord                   |    1 |\n",
       "| 2 | visceral muscle                      |    2 |\n",
       "| 3 | brain                                |  629 |\n",
       "| 3 | epidermis                            |    1 |\n",
       "| 3 | midgut                               |  144 |\n",
       "| 3 | muscle                               |    2 |\n",
       "| 3 | ubiquitous                           |   43 |\n",
       "| 4 | brain                                |   89 |\n",
       "| 4 | epidermis                            |   97 |\n",
       "| 4 | midgut                               |   43 |\n",
       "| 4 | muscle                               |    1 |\n",
       "| 4 | ubiquitous                           |  802 |\n",
       "| 5 | amnioserosa                          |    1 |\n",
       "| 5 | brain                                |   11 |\n",
       "| 5 | epidermis                            |  146 |\n",
       "| 5 | fat body                             |    7 |\n",
       "| 5 | hindgut / anal pad                   |    1 |\n",
       "| 5 | midgut                               | 1615 |\n",
       "| 5 | midgut primordium                    |   14 |\n",
       "| 5 | plasmatocytes                        |    2 |\n",
       "| 5 | proventriculus                       |    3 |\n",
       "| 5 | salivary gland                       |    1 |\n",
       "| 5 | sensory nervous system               |    1 |\n",
       "| 5 | somatic muscle                       |   18 |\n",
       "| 5 | tracheal system                      |    2 |\n",
       "| 5 | ubiquitous                           |    2 |\n",
       "| 5 | ventral nerve cord                   |    3 |\n",
       "| 6 | brain                                |   64 |\n",
       "| 6 | epidermis                            |  106 |\n",
       "| 6 | fat body                             |    1 |\n",
       "| 6 | foregut / hindgut / anal pad         |    2 |\n",
       "| 6 | midgut                               |   47 |\n",
       "| 6 | muscle                               |    7 |\n",
       "| 6 | ubiquitous                           | 1051 |\n",
       "| 7 | amnioserosa                          |    3 |\n",
       "| 7 | brain                                |   24 |\n",
       "| 7 | brain primordium                     |  158 |\n",
       "| 7 | ectoderm anlage in statu nascendi    |    1 |\n",
       "| 7 | ectoderm primordium                  |  554 |\n",
       "| 7 | endoderm primordium                  |    6 |\n",
       "| 7 | epidermis                            |    5 |\n",
       "| 7 | epidermis primordium                 |  214 |\n",
       "| 7 | foregut / hindgut primordium         |   38 |\n",
       "| 7 | germ cell                            |    1 |\n",
       "| 7 | hindgut primordium                   |    1 |\n",
       "| 7 | mesoderm primordium                  |   85 |\n",
       "| 7 | midgut primordium                    |    1 |\n",
       "| 7 | peripheral nervous system            |   25 |\n",
       "| 7 | peripheral nervous system primordium |   16 |\n",
       "| 7 | plasmatocytes anlage                 |    3 |\n",
       "| 7 | sensory nervous system primordium    |    7 |\n",
       "| 7 | tracheal primordium                  |    2 |\n",
       "| 7 | ubiquitous                           |    1 |\n",
       "| 7 | ventral nerve cord primordium        |    4 |\n",
       "| 7 | visceral muscle primordium           |    3 |\n",
       "| 8 | Malpighian tubule                    |    8 |\n",
       "| 8 | brain                                |  343 |\n",
       "| 8 | epidermis                            |  309 |\n",
       "| 8 | fat body                             |    6 |\n",
       "| 8 | midgut                               |   19 |\n",
       "| 8 | muscle                               |    3 |\n",
       "| 8 | plasmatocytes                        |    5 |\n",
       "| 8 | proventriculus                       |    1 |\n",
       "| 8 | sensory nervous system               |  129 |\n",
       "| 8 | somatic muscle                       |  526 |\n",
       "| 8 | tracheal system                      |    8 |\n",
       "| 8 | ubiquitous                           |    3 |\n",
       "| 8 | ventral nerve cord                   |   31 |\n",
       "| 8 | visceral muscle                      |    8 |\n",
       "| 9 | brain                                |    1 |\n",
       "| 9 | epidermis                            |    1 |\n",
       "| 9 | fat body                             |   16 |\n",
       "| 9 | foregut / hindgut / anal pad         |    1 |\n",
       "| 9 | muscle                               |    1 |\n",
       "| 9 | plasmatocytes                        |    6 |\n",
       "| ⋮ | ⋮ | ⋮ |\n",
       "| 161 | fat body                      |   2 |\n",
       "| 161 | glia                          | 237 |\n",
       "| 161 | midline primordium            |   2 |\n",
       "| 161 | muscle                        |   1 |\n",
       "| 161 | plasmatocytes                 |   1 |\n",
       "| 161 | somatic muscle                |   2 |\n",
       "| 161 | ubiquitous                    |   1 |\n",
       "| 162 | brain                         |   5 |\n",
       "| 162 | brain primordium              | 119 |\n",
       "| 162 | ectoderm primordium           |   4 |\n",
       "| 162 | epidermis primordium          |   5 |\n",
       "| 162 | mesoderm primordium           |   1 |\n",
       "| 162 | midline primordium            |  61 |\n",
       "| 162 | peripheral nervous system     |   1 |\n",
       "| 162 | ventral nerve cord primordium |  20 |\n",
       "| 163 | Malpighian tubule             | 161 |\n",
       "| 163 | amnioserosa                   |   1 |\n",
       "| 163 | brain                         |   2 |\n",
       "| 163 | epidermis                     |   3 |\n",
       "| 163 | fat body                      |   2 |\n",
       "| 163 | midgut                        |  46 |\n",
       "| 163 | muscle                        |   1 |\n",
       "| 163 | plasmatocytes                 |   1 |\n",
       "| 163 | sensory nervous system        |   1 |\n",
       "| 163 | somatic muscle                |   1 |\n",
       "| 163 | tracheal primordium           |   4 |\n",
       "| 163 | tracheal system               |   1 |\n",
       "| 163 | ubiquitous                    |  14 |\n",
       "| 163 | unknown                       |   3 |\n",
       "| 164 | brain                         |   9 |\n",
       "| 164 | brain primordium              |   6 |\n",
       "| 164 | epidermis                     |   2 |\n",
       "| 164 | epidermis primordium          |   8 |\n",
       "| 164 | germ cell                     |  93 |\n",
       "| 164 | hindgut primordium            |   1 |\n",
       "| 164 | midgut primordium             |   1 |\n",
       "| 164 | midline primordium            |   1 |\n",
       "| 164 | peripheral nervous system     |   5 |\n",
       "| 164 | plasmatocytes anlage          |   3 |\n",
       "| 164 | procrystal cell               |   6 |\n",
       "| 164 | somatic muscle                |   2 |\n",
       "| 164 | somatic muscle primordium     |   1 |\n",
       "| 164 | ubiquitous                    |  15 |\n",
       "| 164 | ventral nerve cord            |   1 |\n",
       "| 164 | visceral muscle primordium    |   2 |\n",
       "| 165 | amnioserosa                   |   1 |\n",
       "| 165 | crystal cell                  |   7 |\n",
       "| 165 | epidermis                     |  87 |\n",
       "| 165 | procrystal cell               | 114 |\n",
       "| 165 | somatic muscle                |   1 |\n",
       "| 165 | ubiquitous                    |  49 |\n",
       "| 166 | brain                         |  20 |\n",
       "| 166 | epidermis                     |   7 |\n",
       "| 166 | midgut                        |   1 |\n",
       "| 166 | muscle                        |   5 |\n",
       "| 166 | ubiquitous                    | 150 |\n",
       "| 166 | yolk nuclei                   |   1 |\n",
       "| 167 | brain                         |   2 |\n",
       "| 167 | epidermis                     |   2 |\n",
       "| 167 | fat body                      | 141 |\n",
       "| 167 | tracheal primordium           |   3 |\n",
       "| 167 | tracheal system               |   1 |\n",
       "| 167 | ubiquitous                    |   4 |\n",
       "| 167 | unknown                       |  78 |\n",
       "| 168 | brain                         |   1 |\n",
       "| 168 | epidermis                     |   7 |\n",
       "| 168 | muscle                        |   1 |\n",
       "| 168 | somatic muscle                |   2 |\n",
       "| 168 | tracheal primordium           |  45 |\n",
       "| 168 | tracheal system               |  30 |\n",
       "| 168 | ubiquitous                    |   2 |\n",
       "| 169 | brain                         |   5 |\n",
       "| 169 | epidermis                     | 103 |\n",
       "| 169 | foregut / hindgut / anal pad  |  29 |\n",
       "| 169 | somatic muscle                |   1 |\n",
       "| 169 | ubiquitous                    |   8 |\n",
       "| 170 | brain                         |   3 |\n",
       "| 170 | epidermis                     |   1 |\n",
       "| 170 | sensory nervous system        |  54 |\n",
       "| 170 | ubiquitous                    |   1 |\n",
       "| 170 | ventral nerve cord            |  33 |\n",
       "| 171 | brain                         |  13 |\n",
       "| 171 | epidermis                     |   2 |\n",
       "| 171 | midgut                        |   2 |\n",
       "| 171 | sensory nervous system        |   1 |\n",
       "| 171 | somatic muscle                |   6 |\n",
       "| 171 | ubiquitous                    |   1 |\n",
       "| 171 | ventral nerve cord            |  58 |\n",
       "| 172 | epidermis                     |   5 |\n",
       "| 172 | midgut                        |   1 |\n",
       "| 172 | sensory nervous system        |   2 |\n",
       "| 172 | tracheal primordium           |   3 |\n",
       "| 172 | ubiquitous                    |   9 |\n",
       "| 172 | unknown                       |  25 |\n",
       "| 173 | brain                         |   5 |\n",
       "| 173 | epidermis                     |   3 |\n",
       "| 173 | ubiquitous                    |  55 |\n",
       "| 174 | brain                         |   7 |\n",
       "| 174 | epidermis                     |   6 |\n",
       "| 174 | ubiquitous                    |  27 |\n",
       "\n"
      ],
      "text/plain": [
       "     cluster annot                                n   \n",
       "1    1       amnioserosa                          2284\n",
       "2    1       brain                                   1\n",
       "3    1       brain primordium                        1\n",
       "4    1       ectoderm primordium                     8\n",
       "5    1       epidermis primordium                    9\n",
       "6    1       midgut primordium                       2\n",
       "7    1       peripheral nervous system               1\n",
       "8    1       plasmatocytes anlage                    1\n",
       "9    1       procrystal cell                         5\n",
       "10   1       ubiquitous                              1\n",
       "11   1       visceral muscle primordium              2\n",
       "12   2       amnioserosa                             3\n",
       "13   2       brain                                  15\n",
       "14   2       epidermis                             395\n",
       "15   2       fat body                                3\n",
       "16   2       foregut / hindgut / anal pad            1\n",
       "17   2       hindgut / anal pad                      1\n",
       "18   2       midgut                                259\n",
       "19   2       muscle                                  2\n",
       "20   2       plasmatocytes                           4\n",
       "21   2       proventriculus                        832\n",
       "22   2       sensory nervous system                 17\n",
       "23   2       somatic muscle                         46\n",
       "24   2       tracheal system                         7\n",
       "25   2       ubiquitous                              2\n",
       "26   2       ventral nerve cord                      1\n",
       "27   2       visceral muscle                         2\n",
       "28   3       brain                                 629\n",
       "29   3       epidermis                               1\n",
       "30   3       midgut                                144\n",
       "31   3       muscle                                  2\n",
       "32   3       ubiquitous                             43\n",
       "33   4       brain                                  89\n",
       "34   4       epidermis                              97\n",
       "35   4       midgut                                 43\n",
       "36   4       muscle                                  1\n",
       "37   4       ubiquitous                            802\n",
       "38   5       amnioserosa                             1\n",
       "39   5       brain                                  11\n",
       "40   5       epidermis                             146\n",
       "41   5       fat body                                7\n",
       "42   5       hindgut / anal pad                      1\n",
       "43   5       midgut                               1615\n",
       "44   5       midgut primordium                      14\n",
       "45   5       plasmatocytes                           2\n",
       "46   5       proventriculus                          3\n",
       "47   5       salivary gland                          1\n",
       "48   5       sensory nervous system                  1\n",
       "49   5       somatic muscle                         18\n",
       "50   5       tracheal system                         2\n",
       "51   5       ubiquitous                              2\n",
       "52   5       ventral nerve cord                      3\n",
       "53   6       brain                                  64\n",
       "54   6       epidermis                             106\n",
       "55   6       fat body                                1\n",
       "56   6       foregut / hindgut / anal pad            2\n",
       "57   6       midgut                                 47\n",
       "58   6       muscle                                  7\n",
       "59   6       ubiquitous                           1051\n",
       "60   7       amnioserosa                             3\n",
       "61   7       brain                                  24\n",
       "62   7       brain primordium                      158\n",
       "63   7       ectoderm anlage in statu nascendi       1\n",
       "64   7       ectoderm primordium                   554\n",
       "65   7       endoderm primordium                     6\n",
       "66   7       epidermis                               5\n",
       "67   7       epidermis primordium                  214\n",
       "68   7       foregut / hindgut primordium           38\n",
       "69   7       germ cell                               1\n",
       "70   7       hindgut primordium                      1\n",
       "71   7       mesoderm primordium                    85\n",
       "72   7       midgut primordium                       1\n",
       "73   7       peripheral nervous system              25\n",
       "74   7       peripheral nervous system primordium   16\n",
       "75   7       plasmatocytes anlage                    3\n",
       "76   7       sensory nervous system primordium       7\n",
       "77   7       tracheal primordium                     2\n",
       "78   7       ubiquitous                              1\n",
       "79   7       ventral nerve cord primordium           4\n",
       "80   7       visceral muscle primordium              3\n",
       "81   8       Malpighian tubule                       8\n",
       "82   8       brain                                 343\n",
       "83   8       epidermis                             309\n",
       "84   8       fat body                                6\n",
       "85   8       midgut                                 19\n",
       "86   8       muscle                                  3\n",
       "87   8       plasmatocytes                           5\n",
       "88   8       proventriculus                          1\n",
       "89   8       sensory nervous system                129\n",
       "90   8       somatic muscle                        526\n",
       "91   8       tracheal system                         8\n",
       "92   8       ubiquitous                              3\n",
       "93   8       ventral nerve cord                     31\n",
       "94   8       visceral muscle                         8\n",
       "95   9       brain                                   1\n",
       "96   9       epidermis                               1\n",
       "97   9       fat body                               16\n",
       "98   9       foregut / hindgut / anal pad            1\n",
       "99   9       muscle                                  1\n",
       "100  9       plasmatocytes                           6\n",
       "⋮    ⋮       ⋮                                    ⋮   \n",
       "1833 161     fat body                               2 \n",
       "1834 161     glia                                 237 \n",
       "1835 161     midline primordium                     2 \n",
       "1836 161     muscle                                 1 \n",
       "1837 161     plasmatocytes                          1 \n",
       "1838 161     somatic muscle                         2 \n",
       "1839 161     ubiquitous                             1 \n",
       "1840 162     brain                                  5 \n",
       "1841 162     brain primordium                     119 \n",
       "1842 162     ectoderm primordium                    4 \n",
       "1843 162     epidermis primordium                   5 \n",
       "1844 162     mesoderm primordium                    1 \n",
       "1845 162     midline primordium                    61 \n",
       "1846 162     peripheral nervous system              1 \n",
       "1847 162     ventral nerve cord primordium         20 \n",
       "1848 163     Malpighian tubule                    161 \n",
       "1849 163     amnioserosa                            1 \n",
       "1850 163     brain                                  2 \n",
       "1851 163     epidermis                              3 \n",
       "1852 163     fat body                               2 \n",
       "1853 163     midgut                                46 \n",
       "1854 163     muscle                                 1 \n",
       "1855 163     plasmatocytes                          1 \n",
       "1856 163     sensory nervous system                 1 \n",
       "1857 163     somatic muscle                         1 \n",
       "1858 163     tracheal primordium                    4 \n",
       "1859 163     tracheal system                        1 \n",
       "1860 163     ubiquitous                            14 \n",
       "1861 163     unknown                                3 \n",
       "1862 164     brain                                  9 \n",
       "1863 164     brain primordium                       6 \n",
       "1864 164     epidermis                              2 \n",
       "1865 164     epidermis primordium                   8 \n",
       "1866 164     germ cell                             93 \n",
       "1867 164     hindgut primordium                     1 \n",
       "1868 164     midgut primordium                      1 \n",
       "1869 164     midline primordium                     1 \n",
       "1870 164     peripheral nervous system              5 \n",
       "1871 164     plasmatocytes anlage                   3 \n",
       "1872 164     procrystal cell                        6 \n",
       "1873 164     somatic muscle                         2 \n",
       "1874 164     somatic muscle primordium              1 \n",
       "1875 164     ubiquitous                            15 \n",
       "1876 164     ventral nerve cord                     1 \n",
       "1877 164     visceral muscle primordium             2 \n",
       "1878 165     amnioserosa                            1 \n",
       "1879 165     crystal cell                           7 \n",
       "1880 165     epidermis                             87 \n",
       "1881 165     procrystal cell                      114 \n",
       "1882 165     somatic muscle                         1 \n",
       "1883 165     ubiquitous                            49 \n",
       "1884 166     brain                                 20 \n",
       "1885 166     epidermis                              7 \n",
       "1886 166     midgut                                 1 \n",
       "1887 166     muscle                                 5 \n",
       "1888 166     ubiquitous                           150 \n",
       "1889 166     yolk nuclei                            1 \n",
       "1890 167     brain                                  2 \n",
       "1891 167     epidermis                              2 \n",
       "1892 167     fat body                             141 \n",
       "1893 167     tracheal primordium                    3 \n",
       "1894 167     tracheal system                        1 \n",
       "1895 167     ubiquitous                             4 \n",
       "1896 167     unknown                               78 \n",
       "1897 168     brain                                  1 \n",
       "1898 168     epidermis                              7 \n",
       "1899 168     muscle                                 1 \n",
       "1900 168     somatic muscle                         2 \n",
       "1901 168     tracheal primordium                   45 \n",
       "1902 168     tracheal system                       30 \n",
       "1903 168     ubiquitous                             2 \n",
       "1904 169     brain                                  5 \n",
       "1905 169     epidermis                            103 \n",
       "1906 169     foregut / hindgut / anal pad          29 \n",
       "1907 169     somatic muscle                         1 \n",
       "1908 169     ubiquitous                             8 \n",
       "1909 170     brain                                  3 \n",
       "1910 170     epidermis                              1 \n",
       "1911 170     sensory nervous system                54 \n",
       "1912 170     ubiquitous                             1 \n",
       "1913 170     ventral nerve cord                    33 \n",
       "1914 171     brain                                 13 \n",
       "1915 171     epidermis                              2 \n",
       "1916 171     midgut                                 2 \n",
       "1917 171     sensory nervous system                 1 \n",
       "1918 171     somatic muscle                         6 \n",
       "1919 171     ubiquitous                             1 \n",
       "1920 171     ventral nerve cord                    58 \n",
       "1921 172     epidermis                              5 \n",
       "1922 172     midgut                                 1 \n",
       "1923 172     sensory nervous system                 2 \n",
       "1924 172     tracheal primordium                    3 \n",
       "1925 172     ubiquitous                             9 \n",
       "1926 172     unknown                               25 \n",
       "1927 173     brain                                  5 \n",
       "1928 173     epidermis                              3 \n",
       "1929 173     ubiquitous                            55 \n",
       "1930 174     brain                                  7 \n",
       "1931 174     epidermis                              6 \n",
       "1932 174     ubiquitous                            27 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub_no_multitons  %>% count(cluster, annot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_multitons, c(\"bt\"), size_pos = 0.5, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_multitons, c(\"bt\"), size_pos = 0.5, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_multitons, c(\"bt\"), size_pos = 0.5, size_neg = 0.06, stroke = 0.02, limit = 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1434,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sub_no_multitons$tmp.expr = counts(cds_sub_no_multitons)[\"bt\", ]/df_sub_no_multitons$Size_Factor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1436,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 33</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>orig.ident</th><th scope=col>nCount_RNA</th><th scope=col>nFeature_RNA</th><th scope=col>rt_bc</th><th scope=col>cell</th><th scope=col>time</th><th scope=col>percent.mt</th><th scope=col>percent.ribo</th><th scope=col>barn_dro</th><th scope=col>barn_hu</th><th scope=col>barn_mus</th><th scope=col>RNA_snn_res.0.8</th><th scope=col>doublet_score</th><th scope=col>predicted_doublet</th><th scope=col>doublet_cluster</th><th scope=col>seurat_clusters</th><th scope=col>doublet_subcluster</th><th scope=col>RNA_snn_res.0.08</th><th scope=col>lasso_age</th><th scope=col>NNv1_age</th><th scope=col>NNv2_age</th><th scope=col>lasso_shift</th><th scope=col>NNv1_shift</th><th scope=col>NNv2_shift</th><th scope=col>annot</th><th scope=col>ident</th><th scope=col>inferred_time_window</th><th scope=col>cell.type_time.bin</th><th scope=col>Size_Factor</th><th scope=col>cluster</th><th scope=col>umap_1</th><th scope=col>umap_2</th><th scope=col>tmp.expr</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG</th><td>exp1</td><td>292</td><td>202</td><td>AAGTAGTCAG</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG</td><td>hrs_10_14</td><td>0.3424658</td><td> 0.6849315</td><td>320</td><td>3</td><td>0</td><td>23</td><td>0.06585398</td><td>Singlet</td><td>FALSE</td><td>12</td><td>FALSE</td><td>12</td><td>12.544636</td><td>12.211529</td><td>13.83098</td><td> 0.00000000</td><td>0.00000000</td><td>0.0000000</td><td>epidermis             </td><td>12</td><td>12-14</td><td>epidermis:12-14             </td><td>0.6967951</td><td>165</td><td>-1.3580737</td><td> 0.3582040</td><td>0.000000</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA</th><td>exp1</td><td>354</td><td>272</td><td>TCGATATTGA</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA</td><td>hrs_14_18</td><td>0.5649718</td><td> 1.6949153</td><td>380</td><td>0</td><td>1</td><td>8 </td><td>0.08257165</td><td>Singlet</td><td>FALSE</td><td>11</td><td>FALSE</td><td>11</td><td>13.458162</td><td>15.126904</td><td>14.55855</td><td>-0.54183787</td><td>0.00000000</td><td>0.0000000</td><td>sensory nervous system</td><td>11</td><td>14-16</td><td>sensory nervous system:14-16</td><td>0.8306988</td><td>148</td><td>-0.6711318</td><td>-2.9462104</td><td>0.000000</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA</th><td>exp1</td><td>287</td><td>221</td><td>TGCGCCAGAA</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA</td><td>hrs_08_12</td><td>0.0000000</td><td> 3.4843206</td><td>307</td><td>0</td><td>0</td><td>6 </td><td>0.06023494</td><td>Singlet</td><td>FALSE</td><td>9 </td><td>FALSE</td><td>9 </td><td> 9.715162</td><td> 9.000631</td><td>11.35599</td><td> 0.00000000</td><td>0.00000000</td><td>0.0000000</td><td>midline primordium    </td><td>9 </td><td>8-10 </td><td>midline primordium:8-10     </td><td>0.6893560</td><td>120</td><td>-7.5437593</td><td>-0.6611074</td><td>0.000000</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA</th><td>exp1</td><td>509</td><td>376</td><td>ACGCCGTTCA</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA</td><td>hrs_08_12</td><td>0.5893910</td><td> 5.5009823</td><td>552</td><td>2</td><td>1</td><td>0 </td><td>0.08149588</td><td>Singlet</td><td>FALSE</td><td>21</td><td>FALSE</td><td>21</td><td>12.520638</td><td>11.694070</td><td>13.23313</td><td> 0.52063800</td><td>0.00000000</td><td>1.2331295</td><td>plasmatocytes         </td><td>21</td><td>10-12</td><td>plasmatocytes:10-12         </td><td>1.2076129</td><td>89 </td><td>-1.6608665</td><td> 1.2526858</td><td>0.000000</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT</th><td>exp1</td><td>438</td><td>282</td><td>GTTGAAGGAT</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT</td><td>hrs_06_10</td><td>0.6849315</td><td> 0.4566210</td><td>492</td><td>1</td><td>1</td><td>13</td><td>0.04439834</td><td>Singlet</td><td>FALSE</td><td>9 </td><td>FALSE</td><td>9 </td><td>10.045072</td><td>10.022837</td><td>10.51352</td><td> 0.04507167</td><td>0.02283669</td><td>0.5135212</td><td>brain                 </td><td>9 </td><td>10-12</td><td>brain:10-12                 </td><td>1.0017980</td><td>88 </td><td>-2.4943113</td><td>-5.3918748</td><td>0.000000</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG</th><td>exp1</td><td>313</td><td>227</td><td>TGATAGAACG</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG</td><td>hrs_14_18</td><td>0.6389776</td><td>13.4185304</td><td>344</td><td>2</td><td>0</td><td>12</td><td>0.07449210</td><td>Singlet</td><td>FALSE</td><td>11</td><td>FALSE</td><td>11</td><td>10.338754</td><td>15.219664</td><td>14.57722</td><td>-3.66124596</td><td>0.00000000</td><td>0.0000000</td><td>ubiquitous            </td><td>11</td><td>14-16</td><td>ubiquitous:14-16            </td><td>0.7637470</td><td>44 </td><td> 1.8231004</td><td> 3.9306800</td><td>1.309334</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 33\n",
       "\\begin{tabular}{r|lllllllllllllllllllllllllllllllll}\n",
       "  & orig.ident & nCount\\_RNA & nFeature\\_RNA & rt\\_bc & cell & time & percent.mt & percent.ribo & barn\\_dro & barn\\_hu & barn\\_mus & RNA\\_snn\\_res.0.8 & doublet\\_score & predicted\\_doublet & doublet\\_cluster & seurat\\_clusters & doublet\\_subcluster & RNA\\_snn\\_res.0.08 & lasso\\_age & NNv1\\_age & NNv2\\_age & lasso\\_shift & NNv1\\_shift & NNv2\\_shift & annot & ident & inferred\\_time\\_window & cell.type\\_time.bin & Size\\_Factor & cluster & umap\\_1 & umap\\_2 & tmp.expr\\\\\n",
       "  & <chr> & <dbl> & <int> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <int> & <int> & <int> & <chr> & <dbl> & <chr> & <lgl> & <fct> & <lgl> & <fct> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <fct> & <chr> & <chr> & <dbl> & <fct> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG & exp1 & 292 & 202 & AAGTAGTCAG & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG & hrs\\_10\\_14 & 0.3424658 &  0.6849315 & 320 & 3 & 0 & 23 & 0.06585398 & Singlet & FALSE & 12 & FALSE & 12 & 12.544636 & 12.211529 & 13.83098 &  0.00000000 & 0.00000000 & 0.0000000 & epidermis              & 12 & 12-14 & epidermis:12-14              & 0.6967951 & 165 & -1.3580737 &  0.3582040 & 0.000000\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA & exp1 & 354 & 272 & TCGATATTGA & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA & hrs\\_14\\_18 & 0.5649718 &  1.6949153 & 380 & 0 & 1 & 8  & 0.08257165 & Singlet & FALSE & 11 & FALSE & 11 & 13.458162 & 15.126904 & 14.55855 & -0.54183787 & 0.00000000 & 0.0000000 & sensory nervous system & 11 & 14-16 & sensory nervous system:14-16 & 0.8306988 & 148 & -0.6711318 & -2.9462104 & 0.000000\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA & exp1 & 287 & 221 & TGCGCCAGAA & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA & hrs\\_08\\_12 & 0.0000000 &  3.4843206 & 307 & 0 & 0 & 6  & 0.06023494 & Singlet & FALSE & 9  & FALSE & 9  &  9.715162 &  9.000631 & 11.35599 &  0.00000000 & 0.00000000 & 0.0000000 & midline primordium     & 9  & 8-10  & midline primordium:8-10      & 0.6893560 & 120 & -7.5437593 & -0.6611074 & 0.000000\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA & exp1 & 509 & 376 & ACGCCGTTCA & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA & hrs\\_08\\_12 & 0.5893910 &  5.5009823 & 552 & 2 & 1 & 0  & 0.08149588 & Singlet & FALSE & 21 & FALSE & 21 & 12.520638 & 11.694070 & 13.23313 &  0.52063800 & 0.00000000 & 1.2331295 & plasmatocytes          & 21 & 10-12 & plasmatocytes:10-12          & 1.2076129 & 89  & -1.6608665 &  1.2526858 & 0.000000\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT & exp1 & 438 & 282 & GTTGAAGGAT & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT & hrs\\_06\\_10 & 0.6849315 &  0.4566210 & 492 & 1 & 1 & 13 & 0.04439834 & Singlet & FALSE & 9  & FALSE & 9  & 10.045072 & 10.022837 & 10.51352 &  0.04507167 & 0.02283669 & 0.5135212 & brain                  & 9  & 10-12 & brain:10-12                  & 1.0017980 & 88  & -2.4943113 & -5.3918748 & 0.000000\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG & exp1 & 313 & 227 & TGATAGAACG & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG & hrs\\_14\\_18 & 0.6389776 & 13.4185304 & 344 & 2 & 0 & 12 & 0.07449210 & Singlet & FALSE & 11 & FALSE & 11 & 10.338754 & 15.219664 & 14.57722 & -3.66124596 & 0.00000000 & 0.0000000 & ubiquitous             & 11 & 14-16 & ubiquitous:14-16             & 0.7637470 & 44  &  1.8231004 &  3.9306800 & 1.309334\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 33\n",
       "\n",
       "| <!--/--> | orig.ident &lt;chr&gt; | nCount_RNA &lt;dbl&gt; | nFeature_RNA &lt;int&gt; | rt_bc &lt;chr&gt; | cell &lt;chr&gt; | time &lt;chr&gt; | percent.mt &lt;dbl&gt; | percent.ribo &lt;dbl&gt; | barn_dro &lt;int&gt; | barn_hu &lt;int&gt; | barn_mus &lt;int&gt; | RNA_snn_res.0.8 &lt;chr&gt; | doublet_score &lt;dbl&gt; | predicted_doublet &lt;chr&gt; | doublet_cluster &lt;lgl&gt; | seurat_clusters &lt;fct&gt; | doublet_subcluster &lt;lgl&gt; | RNA_snn_res.0.08 &lt;fct&gt; | lasso_age &lt;dbl&gt; | NNv1_age &lt;dbl&gt; | NNv2_age &lt;dbl&gt; | lasso_shift &lt;dbl&gt; | NNv1_shift &lt;dbl&gt; | NNv2_shift &lt;dbl&gt; | annot &lt;chr&gt; | ident &lt;fct&gt; | inferred_time_window &lt;chr&gt; | cell.type_time.bin &lt;chr&gt; | Size_Factor &lt;dbl&gt; | cluster &lt;fct&gt; | umap_1 &lt;dbl&gt; | umap_2 &lt;dbl&gt; | tmp.expr &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG | exp1 | 292 | 202 | AAGTAGTCAG | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG | hrs_10_14 | 0.3424658 |  0.6849315 | 320 | 3 | 0 | 23 | 0.06585398 | Singlet | FALSE | 12 | FALSE | 12 | 12.544636 | 12.211529 | 13.83098 |  0.00000000 | 0.00000000 | 0.0000000 | epidermis              | 12 | 12-14 | epidermis:12-14              | 0.6967951 | 165 | -1.3580737 |  0.3582040 | 0.000000 |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA | exp1 | 354 | 272 | TCGATATTGA | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA | hrs_14_18 | 0.5649718 |  1.6949153 | 380 | 0 | 1 | 8  | 0.08257165 | Singlet | FALSE | 11 | FALSE | 11 | 13.458162 | 15.126904 | 14.55855 | -0.54183787 | 0.00000000 | 0.0000000 | sensory nervous system | 11 | 14-16 | sensory nervous system:14-16 | 0.8306988 | 148 | -0.6711318 | -2.9462104 | 0.000000 |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA | exp1 | 287 | 221 | TGCGCCAGAA | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA | hrs_08_12 | 0.0000000 |  3.4843206 | 307 | 0 | 0 | 6  | 0.06023494 | Singlet | FALSE | 9  | FALSE | 9  |  9.715162 |  9.000631 | 11.35599 |  0.00000000 | 0.00000000 | 0.0000000 | midline primordium     | 9  | 8-10  | midline primordium:8-10      | 0.6893560 | 120 | -7.5437593 | -0.6611074 | 0.000000 |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA | exp1 | 509 | 376 | ACGCCGTTCA | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA | hrs_08_12 | 0.5893910 |  5.5009823 | 552 | 2 | 1 | 0  | 0.08149588 | Singlet | FALSE | 21 | FALSE | 21 | 12.520638 | 11.694070 | 13.23313 |  0.52063800 | 0.00000000 | 1.2331295 | plasmatocytes          | 21 | 10-12 | plasmatocytes:10-12          | 1.2076129 | 89  | -1.6608665 |  1.2526858 | 0.000000 |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT | exp1 | 438 | 282 | GTTGAAGGAT | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT | hrs_06_10 | 0.6849315 |  0.4566210 | 492 | 1 | 1 | 13 | 0.04439834 | Singlet | FALSE | 9  | FALSE | 9  | 10.045072 | 10.022837 | 10.51352 |  0.04507167 | 0.02283669 | 0.5135212 | brain                  | 9  | 10-12 | brain:10-12                  | 1.0017980 | 88  | -2.4943113 | -5.3918748 | 0.000000 |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG | exp1 | 313 | 227 | TGATAGAACG | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG | hrs_14_18 | 0.6389776 | 13.4185304 | 344 | 2 | 0 | 12 | 0.07449210 | Singlet | FALSE | 11 | FALSE | 11 | 10.338754 | 15.219664 | 14.57722 | -3.66124596 | 0.00000000 | 0.0000000 | ubiquitous             | 11 | 14-16 | ubiquitous:14-16             | 0.7637470 | 44  |  1.8231004 |  3.9306800 | 1.309334 |\n",
       "\n"
      ],
      "text/plain": [
       "                                                       orig.ident nCount_RNA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG exp1       292       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA exp1       354       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA exp1       287       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA exp1       509       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT exp1       438       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG exp1       313       \n",
       "                                                       nFeature_RNA rt_bc     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 202          AAGTAGTCAG\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 272          TCGATATTGA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 221          TGCGCCAGAA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 376          ACGCCGTTCA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 282          GTTGAAGGAT\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 227          TGATAGAACG\n",
       "                                                       cell                                                  \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG\n",
       "                                                       time      percent.mt\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG hrs_10_14 0.3424658 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA hrs_14_18 0.5649718 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA hrs_08_12 0.0000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA hrs_08_12 0.5893910 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT hrs_06_10 0.6849315 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG hrs_14_18 0.6389776 \n",
       "                                                       percent.ribo barn_dro\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG  0.6849315   320     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA  1.6949153   380     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA  3.4843206   307     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA  5.5009823   552     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT  0.4566210   492     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 13.4185304   344     \n",
       "                                                       barn_hu barn_mus\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 3       0       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 0       1       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 0       0       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 2       1       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 1       1       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 2       0       \n",
       "                                                       RNA_snn_res.0.8\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 23             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 8              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 6              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 0              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 13             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 12             \n",
       "                                                       doublet_score\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 0.06585398   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 0.08257165   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 0.06023494   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 0.08149588   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 0.04439834   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 0.07449210   \n",
       "                                                       predicted_doublet\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG Singlet          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA Singlet          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA Singlet          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA Singlet          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT Singlet          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG Singlet          \n",
       "                                                       doublet_cluster\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG FALSE          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA FALSE          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA FALSE          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA FALSE          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT FALSE          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG FALSE          \n",
       "                                                       seurat_clusters\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 11             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 9              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 21             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 9              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 11             \n",
       "                                                       doublet_subcluster\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG FALSE             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA FALSE             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA FALSE             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA FALSE             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT FALSE             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG FALSE             \n",
       "                                                       RNA_snn_res.0.08\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 11              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 9               \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 21              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 9               \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 11              \n",
       "                                                       lasso_age NNv1_age \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12.544636 12.211529\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 13.458162 15.126904\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA  9.715162  9.000631\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 12.520638 11.694070\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 10.045072 10.022837\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 10.338754 15.219664\n",
       "                                                       NNv2_age lasso_shift\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 13.83098  0.00000000\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 14.55855 -0.54183787\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 11.35599  0.00000000\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 13.23313  0.52063800\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 10.51352  0.04507167\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 14.57722 -3.66124596\n",
       "                                                       NNv1_shift NNv2_shift\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 0.00000000 0.0000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 0.00000000 0.0000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 0.00000000 0.0000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 0.00000000 1.2331295 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 0.02283669 0.5135212 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 0.00000000 0.0000000 \n",
       "                                                       annot                 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG epidermis             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA sensory nervous system\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA midline primordium    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA plasmatocytes         \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT brain                 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG ubiquitous            \n",
       "                                                       ident\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 11   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 9    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 21   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 9    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 11   \n",
       "                                                       inferred_time_window\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12-14               \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 14-16               \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 8-10                \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 10-12               \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 10-12               \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 14-16               \n",
       "                                                       cell.type_time.bin          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG epidermis:12-14             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA sensory nervous system:14-16\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA midline primordium:8-10     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA plasmatocytes:10-12         \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT brain:10-12                 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG ubiquitous:14-16            \n",
       "                                                       Size_Factor cluster\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 0.6967951   165    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 0.8306988   148    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 0.6893560   120    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 1.2076129   89     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 1.0017980   88     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 0.7637470   44     \n",
       "                                                       umap_1     umap_2    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG -1.3580737  0.3582040\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA -0.6711318 -2.9462104\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA -7.5437593 -0.6611074\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA -1.6608665  1.2526858\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT -2.4943113 -5.3918748\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG  1.8231004  3.9306800\n",
       "                                                       tmp.expr\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 0.000000\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 0.000000\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 0.000000\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 0.000000\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 0.000000\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 1.309334"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub_no_multitons  %>% head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1438,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>31783</li><li>33</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 31783\n",
       "\\item 33\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 31783\n",
       "2. 33\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 31783    33"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub_no_multitons  %>% filter(tmp.expr > 0)  %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1439,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 43 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>annot</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Malpighian tubule                </td><td>  81</td></tr>\n",
       "\t<tr><td>amnioserosa                      </td><td> 292</td></tr>\n",
       "\t<tr><td>brain                            </td><td>7045</td></tr>\n",
       "\t<tr><td>brain primordium                 </td><td>  84</td></tr>\n",
       "\t<tr><td>crystal cell                     </td><td>   1</td></tr>\n",
       "\t<tr><td>ectoderm primordium              </td><td>  20</td></tr>\n",
       "\t<tr><td>endoderm primordium              </td><td>   1</td></tr>\n",
       "\t<tr><td>epidermis                        </td><td>2935</td></tr>\n",
       "\t<tr><td>epidermis primordium             </td><td>  26</td></tr>\n",
       "\t<tr><td>fat body                         </td><td> 661</td></tr>\n",
       "\t<tr><td>foregut / hindgut / anal pad     </td><td> 363</td></tr>\n",
       "\t<tr><td>foregut / hindgut primordium     </td><td>   6</td></tr>\n",
       "\t<tr><td>germ cell                        </td><td>  40</td></tr>\n",
       "\t<tr><td>glia                             </td><td>  41</td></tr>\n",
       "\t<tr><td>hindgut / anal pad               </td><td> 195</td></tr>\n",
       "\t<tr><td>hindgut primordium               </td><td>   6</td></tr>\n",
       "\t<tr><td>midgut                           </td><td>1400</td></tr>\n",
       "\t<tr><td>midgut primordium                </td><td>   7</td></tr>\n",
       "\t<tr><td>midline primordium               </td><td>   9</td></tr>\n",
       "\t<tr><td>muscle                           </td><td>1136</td></tr>\n",
       "\t<tr><td>peripheral nervous system        </td><td>  51</td></tr>\n",
       "\t<tr><td>plasmatocytes                    </td><td> 597</td></tr>\n",
       "\t<tr><td>plasmatocytes anlage             </td><td>  18</td></tr>\n",
       "\t<tr><td>procrystal cell                  </td><td> 199</td></tr>\n",
       "\t<tr><td>proventriculus                   </td><td> 158</td></tr>\n",
       "\t<tr><td>salivary gland                   </td><td> 258</td></tr>\n",
       "\t<tr><td>salivary gland body primordium   </td><td>  10</td></tr>\n",
       "\t<tr><td>sensory nervous system           </td><td> 309</td></tr>\n",
       "\t<tr><td>sensory nervous system primordium</td><td>   8</td></tr>\n",
       "\t<tr><td>somatic muscle                   </td><td>9967</td></tr>\n",
       "\t<tr><td>somatic muscle primordium        </td><td>  43</td></tr>\n",
       "\t<tr><td>spiracle (tracheal system)       </td><td>  19</td></tr>\n",
       "\t<tr><td>spiracles (tracheal system)      </td><td>  67</td></tr>\n",
       "\t<tr><td>tracheal primordium              </td><td>  53</td></tr>\n",
       "\t<tr><td>tracheal system                  </td><td> 788</td></tr>\n",
       "\t<tr><td>trunk mesoderm primordium        </td><td> 147</td></tr>\n",
       "\t<tr><td>ubiquitous                       </td><td>3699</td></tr>\n",
       "\t<tr><td>unknown                          </td><td>   6</td></tr>\n",
       "\t<tr><td>ventral nerve cord               </td><td> 766</td></tr>\n",
       "\t<tr><td>ventral nerve cord primordium    </td><td>   3</td></tr>\n",
       "\t<tr><td>visceral muscle                  </td><td> 155</td></tr>\n",
       "\t<tr><td>visceral muscle primordium       </td><td>  99</td></tr>\n",
       "\t<tr><td>yolk nuclei                      </td><td>  14</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 43 × 2\n",
       "\\begin{tabular}{ll}\n",
       " annot & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Malpighian tubule                 &   81\\\\\n",
       "\t amnioserosa                       &  292\\\\\n",
       "\t brain                             & 7045\\\\\n",
       "\t brain primordium                  &   84\\\\\n",
       "\t crystal cell                      &    1\\\\\n",
       "\t ectoderm primordium               &   20\\\\\n",
       "\t endoderm primordium               &    1\\\\\n",
       "\t epidermis                         & 2935\\\\\n",
       "\t epidermis primordium              &   26\\\\\n",
       "\t fat body                          &  661\\\\\n",
       "\t foregut / hindgut / anal pad      &  363\\\\\n",
       "\t foregut / hindgut primordium      &    6\\\\\n",
       "\t germ cell                         &   40\\\\\n",
       "\t glia                              &   41\\\\\n",
       "\t hindgut / anal pad                &  195\\\\\n",
       "\t hindgut primordium                &    6\\\\\n",
       "\t midgut                            & 1400\\\\\n",
       "\t midgut primordium                 &    7\\\\\n",
       "\t midline primordium                &    9\\\\\n",
       "\t muscle                            & 1136\\\\\n",
       "\t peripheral nervous system         &   51\\\\\n",
       "\t plasmatocytes                     &  597\\\\\n",
       "\t plasmatocytes anlage              &   18\\\\\n",
       "\t procrystal cell                   &  199\\\\\n",
       "\t proventriculus                    &  158\\\\\n",
       "\t salivary gland                    &  258\\\\\n",
       "\t salivary gland body primordium    &   10\\\\\n",
       "\t sensory nervous system            &  309\\\\\n",
       "\t sensory nervous system primordium &    8\\\\\n",
       "\t somatic muscle                    & 9967\\\\\n",
       "\t somatic muscle primordium         &   43\\\\\n",
       "\t spiracle (tracheal system)        &   19\\\\\n",
       "\t spiracles (tracheal system)       &   67\\\\\n",
       "\t tracheal primordium               &   53\\\\\n",
       "\t tracheal system                   &  788\\\\\n",
       "\t trunk mesoderm primordium         &  147\\\\\n",
       "\t ubiquitous                        & 3699\\\\\n",
       "\t unknown                           &    6\\\\\n",
       "\t ventral nerve cord                &  766\\\\\n",
       "\t ventral nerve cord primordium     &    3\\\\\n",
       "\t visceral muscle                   &  155\\\\\n",
       "\t visceral muscle primordium        &   99\\\\\n",
       "\t yolk nuclei                       &   14\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 43 × 2\n",
       "\n",
       "| annot &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Malpighian tubule                 |   81 |\n",
       "| amnioserosa                       |  292 |\n",
       "| brain                             | 7045 |\n",
       "| brain primordium                  |   84 |\n",
       "| crystal cell                      |    1 |\n",
       "| ectoderm primordium               |   20 |\n",
       "| endoderm primordium               |    1 |\n",
       "| epidermis                         | 2935 |\n",
       "| epidermis primordium              |   26 |\n",
       "| fat body                          |  661 |\n",
       "| foregut / hindgut / anal pad      |  363 |\n",
       "| foregut / hindgut primordium      |    6 |\n",
       "| germ cell                         |   40 |\n",
       "| glia                              |   41 |\n",
       "| hindgut / anal pad                |  195 |\n",
       "| hindgut primordium                |    6 |\n",
       "| midgut                            | 1400 |\n",
       "| midgut primordium                 |    7 |\n",
       "| midline primordium                |    9 |\n",
       "| muscle                            | 1136 |\n",
       "| peripheral nervous system         |   51 |\n",
       "| plasmatocytes                     |  597 |\n",
       "| plasmatocytes anlage              |   18 |\n",
       "| procrystal cell                   |  199 |\n",
       "| proventriculus                    |  158 |\n",
       "| salivary gland                    |  258 |\n",
       "| salivary gland body primordium    |   10 |\n",
       "| sensory nervous system            |  309 |\n",
       "| sensory nervous system primordium |    8 |\n",
       "| somatic muscle                    | 9967 |\n",
       "| somatic muscle primordium         |   43 |\n",
       "| spiracle (tracheal system)        |   19 |\n",
       "| spiracles (tracheal system)       |   67 |\n",
       "| tracheal primordium               |   53 |\n",
       "| tracheal system                   |  788 |\n",
       "| trunk mesoderm primordium         |  147 |\n",
       "| ubiquitous                        | 3699 |\n",
       "| unknown                           |    6 |\n",
       "| ventral nerve cord                |  766 |\n",
       "| ventral nerve cord primordium     |    3 |\n",
       "| visceral muscle                   |  155 |\n",
       "| visceral muscle primordium        |   99 |\n",
       "| yolk nuclei                       |   14 |\n",
       "\n"
      ],
      "text/plain": [
       "   annot                             n   \n",
       "1  Malpighian tubule                   81\n",
       "2  amnioserosa                        292\n",
       "3  brain                             7045\n",
       "4  brain primordium                    84\n",
       "5  crystal cell                         1\n",
       "6  ectoderm primordium                 20\n",
       "7  endoderm primordium                  1\n",
       "8  epidermis                         2935\n",
       "9  epidermis primordium                26\n",
       "10 fat body                           661\n",
       "11 foregut / hindgut / anal pad       363\n",
       "12 foregut / hindgut primordium         6\n",
       "13 germ cell                           40\n",
       "14 glia                                41\n",
       "15 hindgut / anal pad                 195\n",
       "16 hindgut primordium                   6\n",
       "17 midgut                            1400\n",
       "18 midgut primordium                    7\n",
       "19 midline primordium                   9\n",
       "20 muscle                            1136\n",
       "21 peripheral nervous system           51\n",
       "22 plasmatocytes                      597\n",
       "23 plasmatocytes anlage                18\n",
       "24 procrystal cell                    199\n",
       "25 proventriculus                     158\n",
       "26 salivary gland                     258\n",
       "27 salivary gland body primordium      10\n",
       "28 sensory nervous system             309\n",
       "29 sensory nervous system primordium    8\n",
       "30 somatic muscle                    9967\n",
       "31 somatic muscle primordium           43\n",
       "32 spiracle (tracheal system)          19\n",
       "33 spiracles (tracheal system)         67\n",
       "34 tracheal primordium                 53\n",
       "35 tracheal system                    788\n",
       "36 trunk mesoderm primordium          147\n",
       "37 ubiquitous                        3699\n",
       "38 unknown                              6\n",
       "39 ventral nerve cord                 766\n",
       "40 ventral nerve cord primordium        3\n",
       "41 visceral muscle                    155\n",
       "42 visceral muscle primordium          99\n",
       "43 yolk nuclei                         14"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub_no_multitons  %>% filter(tmp.expr > 0)  %>% count(annot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1441,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 43 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>annot</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Malpighian tubule                </td><td>  56</td></tr>\n",
       "\t<tr><td>amnioserosa                      </td><td> 237</td></tr>\n",
       "\t<tr><td>brain                            </td><td>6182</td></tr>\n",
       "\t<tr><td>brain primordium                 </td><td>  78</td></tr>\n",
       "\t<tr><td>crystal cell                     </td><td>   1</td></tr>\n",
       "\t<tr><td>ectoderm primordium              </td><td>  18</td></tr>\n",
       "\t<tr><td>endoderm primordium              </td><td>   1</td></tr>\n",
       "\t<tr><td>epidermis                        </td><td>2255</td></tr>\n",
       "\t<tr><td>epidermis primordium             </td><td>  24</td></tr>\n",
       "\t<tr><td>fat body                         </td><td> 432</td></tr>\n",
       "\t<tr><td>foregut / hindgut / anal pad     </td><td> 266</td></tr>\n",
       "\t<tr><td>foregut / hindgut primordium     </td><td>   5</td></tr>\n",
       "\t<tr><td>germ cell                        </td><td>  38</td></tr>\n",
       "\t<tr><td>glia                             </td><td>  30</td></tr>\n",
       "\t<tr><td>hindgut / anal pad               </td><td> 123</td></tr>\n",
       "\t<tr><td>hindgut primordium               </td><td>   6</td></tr>\n",
       "\t<tr><td>midgut                           </td><td>1246</td></tr>\n",
       "\t<tr><td>midgut primordium                </td><td>   7</td></tr>\n",
       "\t<tr><td>midline primordium               </td><td>   9</td></tr>\n",
       "\t<tr><td>muscle                           </td><td> 721</td></tr>\n",
       "\t<tr><td>peripheral nervous system        </td><td>  47</td></tr>\n",
       "\t<tr><td>plasmatocytes                    </td><td> 369</td></tr>\n",
       "\t<tr><td>plasmatocytes anlage             </td><td>  18</td></tr>\n",
       "\t<tr><td>procrystal cell                  </td><td> 199</td></tr>\n",
       "\t<tr><td>proventriculus                   </td><td> 109</td></tr>\n",
       "\t<tr><td>salivary gland                   </td><td> 189</td></tr>\n",
       "\t<tr><td>salivary gland body primordium   </td><td>  10</td></tr>\n",
       "\t<tr><td>sensory nervous system           </td><td> 183</td></tr>\n",
       "\t<tr><td>sensory nervous system primordium</td><td>   7</td></tr>\n",
       "\t<tr><td>somatic muscle                   </td><td>2088</td></tr>\n",
       "\t<tr><td>somatic muscle primordium        </td><td>  36</td></tr>\n",
       "\t<tr><td>spiracle (tracheal system)       </td><td>  15</td></tr>\n",
       "\t<tr><td>spiracles (tracheal system)      </td><td>  47</td></tr>\n",
       "\t<tr><td>tracheal primordium              </td><td>  45</td></tr>\n",
       "\t<tr><td>tracheal system                  </td><td> 491</td></tr>\n",
       "\t<tr><td>trunk mesoderm primordium        </td><td>  12</td></tr>\n",
       "\t<tr><td>ubiquitous                       </td><td>3032</td></tr>\n",
       "\t<tr><td>unknown                          </td><td>   4</td></tr>\n",
       "\t<tr><td>ventral nerve cord               </td><td> 467</td></tr>\n",
       "\t<tr><td>ventral nerve cord primordium    </td><td>   2</td></tr>\n",
       "\t<tr><td>visceral muscle                  </td><td>  77</td></tr>\n",
       "\t<tr><td>visceral muscle primordium       </td><td>  82</td></tr>\n",
       "\t<tr><td>yolk nuclei                      </td><td>  10</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 43 × 2\n",
       "\\begin{tabular}{ll}\n",
       " annot & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Malpighian tubule                 &   56\\\\\n",
       "\t amnioserosa                       &  237\\\\\n",
       "\t brain                             & 6182\\\\\n",
       "\t brain primordium                  &   78\\\\\n",
       "\t crystal cell                      &    1\\\\\n",
       "\t ectoderm primordium               &   18\\\\\n",
       "\t endoderm primordium               &    1\\\\\n",
       "\t epidermis                         & 2255\\\\\n",
       "\t epidermis primordium              &   24\\\\\n",
       "\t fat body                          &  432\\\\\n",
       "\t foregut / hindgut / anal pad      &  266\\\\\n",
       "\t foregut / hindgut primordium      &    5\\\\\n",
       "\t germ cell                         &   38\\\\\n",
       "\t glia                              &   30\\\\\n",
       "\t hindgut / anal pad                &  123\\\\\n",
       "\t hindgut primordium                &    6\\\\\n",
       "\t midgut                            & 1246\\\\\n",
       "\t midgut primordium                 &    7\\\\\n",
       "\t midline primordium                &    9\\\\\n",
       "\t muscle                            &  721\\\\\n",
       "\t peripheral nervous system         &   47\\\\\n",
       "\t plasmatocytes                     &  369\\\\\n",
       "\t plasmatocytes anlage              &   18\\\\\n",
       "\t procrystal cell                   &  199\\\\\n",
       "\t proventriculus                    &  109\\\\\n",
       "\t salivary gland                    &  189\\\\\n",
       "\t salivary gland body primordium    &   10\\\\\n",
       "\t sensory nervous system            &  183\\\\\n",
       "\t sensory nervous system primordium &    7\\\\\n",
       "\t somatic muscle                    & 2088\\\\\n",
       "\t somatic muscle primordium         &   36\\\\\n",
       "\t spiracle (tracheal system)        &   15\\\\\n",
       "\t spiracles (tracheal system)       &   47\\\\\n",
       "\t tracheal primordium               &   45\\\\\n",
       "\t tracheal system                   &  491\\\\\n",
       "\t trunk mesoderm primordium         &   12\\\\\n",
       "\t ubiquitous                        & 3032\\\\\n",
       "\t unknown                           &    4\\\\\n",
       "\t ventral nerve cord                &  467\\\\\n",
       "\t ventral nerve cord primordium     &    2\\\\\n",
       "\t visceral muscle                   &   77\\\\\n",
       "\t visceral muscle primordium        &   82\\\\\n",
       "\t yolk nuclei                       &   10\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 43 × 2\n",
       "\n",
       "| annot &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Malpighian tubule                 |   56 |\n",
       "| amnioserosa                       |  237 |\n",
       "| brain                             | 6182 |\n",
       "| brain primordium                  |   78 |\n",
       "| crystal cell                      |    1 |\n",
       "| ectoderm primordium               |   18 |\n",
       "| endoderm primordium               |    1 |\n",
       "| epidermis                         | 2255 |\n",
       "| epidermis primordium              |   24 |\n",
       "| fat body                          |  432 |\n",
       "| foregut / hindgut / anal pad      |  266 |\n",
       "| foregut / hindgut primordium      |    5 |\n",
       "| germ cell                         |   38 |\n",
       "| glia                              |   30 |\n",
       "| hindgut / anal pad                |  123 |\n",
       "| hindgut primordium                |    6 |\n",
       "| midgut                            | 1246 |\n",
       "| midgut primordium                 |    7 |\n",
       "| midline primordium                |    9 |\n",
       "| muscle                            |  721 |\n",
       "| peripheral nervous system         |   47 |\n",
       "| plasmatocytes                     |  369 |\n",
       "| plasmatocytes anlage              |   18 |\n",
       "| procrystal cell                   |  199 |\n",
       "| proventriculus                    |  109 |\n",
       "| salivary gland                    |  189 |\n",
       "| salivary gland body primordium    |   10 |\n",
       "| sensory nervous system            |  183 |\n",
       "| sensory nervous system primordium |    7 |\n",
       "| somatic muscle                    | 2088 |\n",
       "| somatic muscle primordium         |   36 |\n",
       "| spiracle (tracheal system)        |   15 |\n",
       "| spiracles (tracheal system)       |   47 |\n",
       "| tracheal primordium               |   45 |\n",
       "| tracheal system                   |  491 |\n",
       "| trunk mesoderm primordium         |   12 |\n",
       "| ubiquitous                        | 3032 |\n",
       "| unknown                           |    4 |\n",
       "| ventral nerve cord                |  467 |\n",
       "| ventral nerve cord primordium     |    2 |\n",
       "| visceral muscle                   |   77 |\n",
       "| visceral muscle primordium        |   82 |\n",
       "| yolk nuclei                       |   10 |\n",
       "\n"
      ],
      "text/plain": [
       "   annot                             n   \n",
       "1  Malpighian tubule                   56\n",
       "2  amnioserosa                        237\n",
       "3  brain                             6182\n",
       "4  brain primordium                    78\n",
       "5  crystal cell                         1\n",
       "6  ectoderm primordium                 18\n",
       "7  endoderm primordium                  1\n",
       "8  epidermis                         2255\n",
       "9  epidermis primordium                24\n",
       "10 fat body                           432\n",
       "11 foregut / hindgut / anal pad       266\n",
       "12 foregut / hindgut primordium         5\n",
       "13 germ cell                           38\n",
       "14 glia                                30\n",
       "15 hindgut / anal pad                 123\n",
       "16 hindgut primordium                   6\n",
       "17 midgut                            1246\n",
       "18 midgut primordium                    7\n",
       "19 midline primordium                   9\n",
       "20 muscle                             721\n",
       "21 peripheral nervous system           47\n",
       "22 plasmatocytes                      369\n",
       "23 plasmatocytes anlage                18\n",
       "24 procrystal cell                    199\n",
       "25 proventriculus                     109\n",
       "26 salivary gland                     189\n",
       "27 salivary gland body primordium      10\n",
       "28 sensory nervous system             183\n",
       "29 sensory nervous system primordium    7\n",
       "30 somatic muscle                    2088\n",
       "31 somatic muscle primordium           36\n",
       "32 spiracle (tracheal system)          15\n",
       "33 spiracles (tracheal system)         47\n",
       "34 tracheal primordium                 45\n",
       "35 tracheal system                    491\n",
       "36 trunk mesoderm primordium           12\n",
       "37 ubiquitous                        3032\n",
       "38 unknown                              4\n",
       "39 ventral nerve cord                 467\n",
       "40 ventral nerve cord primordium        2\n",
       "41 visceral muscle                     77\n",
       "42 visceral muscle primordium          82\n",
       "43 yolk nuclei                         10"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub_no_multitons  %>% filter(tmp.expr > 0 & tmp.expr < 2)  %>% count(annot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1442,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 174 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cluster</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>50 </td><td>1126</td></tr>\n",
       "\t<tr><td>62 </td><td> 975</td></tr>\n",
       "\t<tr><td>71 </td><td> 925</td></tr>\n",
       "\t<tr><td>19 </td><td> 844</td></tr>\n",
       "\t<tr><td>3  </td><td> 571</td></tr>\n",
       "\t<tr><td>46 </td><td> 341</td></tr>\n",
       "\t<tr><td>132</td><td> 311</td></tr>\n",
       "\t<tr><td>44 </td><td> 310</td></tr>\n",
       "\t<tr><td>11 </td><td> 309</td></tr>\n",
       "\t<tr><td>18 </td><td> 293</td></tr>\n",
       "\t<tr><td>21 </td><td> 265</td></tr>\n",
       "\t<tr><td>6  </td><td> 254</td></tr>\n",
       "\t<tr><td>76 </td><td> 243</td></tr>\n",
       "\t<tr><td>33 </td><td> 241</td></tr>\n",
       "\t<tr><td>9  </td><td> 237</td></tr>\n",
       "\t<tr><td>8  </td><td> 224</td></tr>\n",
       "\t<tr><td>29 </td><td> 209</td></tr>\n",
       "\t<tr><td>4  </td><td> 206</td></tr>\n",
       "\t<tr><td>106</td><td> 206</td></tr>\n",
       "\t<tr><td>118</td><td> 202</td></tr>\n",
       "\t<tr><td>20 </td><td> 199</td></tr>\n",
       "\t<tr><td>61 </td><td> 199</td></tr>\n",
       "\t<tr><td>60 </td><td> 188</td></tr>\n",
       "\t<tr><td>53 </td><td> 184</td></tr>\n",
       "\t<tr><td>64 </td><td> 182</td></tr>\n",
       "\t<tr><td>109</td><td> 180</td></tr>\n",
       "\t<tr><td>32 </td><td> 176</td></tr>\n",
       "\t<tr><td>13 </td><td> 172</td></tr>\n",
       "\t<tr><td>38 </td><td> 172</td></tr>\n",
       "\t<tr><td>77 </td><td> 167</td></tr>\n",
       "\t<tr><td>111</td><td> 165</td></tr>\n",
       "\t<tr><td>56 </td><td> 162</td></tr>\n",
       "\t<tr><td>43 </td><td> 155</td></tr>\n",
       "\t<tr><td>58 </td><td> 155</td></tr>\n",
       "\t<tr><td>31 </td><td> 153</td></tr>\n",
       "\t<tr><td>97 </td><td> 149</td></tr>\n",
       "\t<tr><td>14 </td><td> 147</td></tr>\n",
       "\t<tr><td>65 </td><td> 145</td></tr>\n",
       "\t<tr><td>133</td><td> 142</td></tr>\n",
       "\t<tr><td>2  </td><td> 141</td></tr>\n",
       "\t<tr><td>34 </td><td> 136</td></tr>\n",
       "\t<tr><td>67 </td><td> 135</td></tr>\n",
       "\t<tr><td>39 </td><td> 132</td></tr>\n",
       "\t<tr><td>82 </td><td> 132</td></tr>\n",
       "\t<tr><td>104</td><td> 129</td></tr>\n",
       "\t<tr><td>17 </td><td> 128</td></tr>\n",
       "\t<tr><td>115</td><td> 125</td></tr>\n",
       "\t<tr><td>113</td><td> 121</td></tr>\n",
       "\t<tr><td>135</td><td> 121</td></tr>\n",
       "\t<tr><td>5  </td><td> 119</td></tr>\n",
       "\t<tr><td>117</td><td> 118</td></tr>\n",
       "\t<tr><td>40 </td><td> 117</td></tr>\n",
       "\t<tr><td>156</td><td> 113</td></tr>\n",
       "\t<tr><td>131</td><td> 111</td></tr>\n",
       "\t<tr><td>122</td><td> 109</td></tr>\n",
       "\t<tr><td>27 </td><td> 108</td></tr>\n",
       "\t<tr><td>30 </td><td> 108</td></tr>\n",
       "\t<tr><td>66 </td><td> 108</td></tr>\n",
       "\t<tr><td>138</td><td> 105</td></tr>\n",
       "\t<tr><td>25 </td><td> 103</td></tr>\n",
       "\t<tr><td>15 </td><td> 102</td></tr>\n",
       "\t<tr><td>45 </td><td> 102</td></tr>\n",
       "\t<tr><td>75 </td><td> 101</td></tr>\n",
       "\t<tr><td>98 </td><td> 101</td></tr>\n",
       "\t<tr><td>108</td><td> 101</td></tr>\n",
       "\t<tr><td>126</td><td> 101</td></tr>\n",
       "\t<tr><td>55 </td><td>  99</td></tr>\n",
       "\t<tr><td>114</td><td>  98</td></tr>\n",
       "\t<tr><td>140</td><td>  98</td></tr>\n",
       "\t<tr><td>12 </td><td>  97</td></tr>\n",
       "\t<tr><td>110</td><td>  97</td></tr>\n",
       "\t<tr><td>123</td><td>  97</td></tr>\n",
       "\t<tr><td>121</td><td>  93</td></tr>\n",
       "\t<tr><td>80 </td><td>  90</td></tr>\n",
       "\t<tr><td>54 </td><td>  89</td></tr>\n",
       "\t<tr><td>41 </td><td>  88</td></tr>\n",
       "\t<tr><td>150</td><td>  88</td></tr>\n",
       "\t<tr><td>26 </td><td>  84</td></tr>\n",
       "\t<tr><td>28 </td><td>  83</td></tr>\n",
       "\t<tr><td>125</td><td>  83</td></tr>\n",
       "\t<tr><td>73 </td><td>  82</td></tr>\n",
       "\t<tr><td>78 </td><td>  81</td></tr>\n",
       "\t<tr><td>137</td><td>  80</td></tr>\n",
       "\t<tr><td>99 </td><td>  78</td></tr>\n",
       "\t<tr><td>134</td><td>  78</td></tr>\n",
       "\t<tr><td>94 </td><td>  77</td></tr>\n",
       "\t<tr><td>96 </td><td>  74</td></tr>\n",
       "\t<tr><td>149</td><td>  72</td></tr>\n",
       "\t<tr><td>1  </td><td>  70</td></tr>\n",
       "\t<tr><td>85 </td><td>  70</td></tr>\n",
       "\t<tr><td>129</td><td>  69</td></tr>\n",
       "\t<tr><td>124</td><td>  68</td></tr>\n",
       "\t<tr><td>49 </td><td>  67</td></tr>\n",
       "\t<tr><td>37 </td><td>  66</td></tr>\n",
       "\t<tr><td>102</td><td>  64</td></tr>\n",
       "\t<tr><td>91 </td><td>  62</td></tr>\n",
       "\t<tr><td>10 </td><td>  61</td></tr>\n",
       "\t<tr><td>142</td><td>  60</td></tr>\n",
       "\t<tr><td>63 </td><td>  59</td></tr>\n",
       "\t<tr><td>47 </td><td>  58</td></tr>\n",
       "\t<tr><td>154</td><td>  58</td></tr>\n",
       "\t<tr><td>92 </td><td>  57</td></tr>\n",
       "\t<tr><td>143</td><td>  56</td></tr>\n",
       "\t<tr><td>157</td><td>  55</td></tr>\n",
       "\t<tr><td>52 </td><td>  54</td></tr>\n",
       "\t<tr><td>128</td><td>  53</td></tr>\n",
       "\t<tr><td>141</td><td>  53</td></tr>\n",
       "\t<tr><td>57 </td><td>  51</td></tr>\n",
       "\t<tr><td>84 </td><td>  51</td></tr>\n",
       "\t<tr><td>103</td><td>  49</td></tr>\n",
       "\t<tr><td>147</td><td>  48</td></tr>\n",
       "\t<tr><td>148</td><td>  48</td></tr>\n",
       "\t<tr><td>83 </td><td>  46</td></tr>\n",
       "\t<tr><td>151</td><td>  45</td></tr>\n",
       "\t<tr><td>51 </td><td>  44</td></tr>\n",
       "\t<tr><td>158</td><td>  43</td></tr>\n",
       "\t<tr><td>95 </td><td>  41</td></tr>\n",
       "\t<tr><td>130</td><td>  41</td></tr>\n",
       "\t<tr><td>144</td><td>  41</td></tr>\n",
       "\t<tr><td>22 </td><td>  39</td></tr>\n",
       "\t<tr><td>159</td><td>  38</td></tr>\n",
       "\t<tr><td>81 </td><td>  37</td></tr>\n",
       "\t<tr><td>88 </td><td>  35</td></tr>\n",
       "\t<tr><td>48 </td><td>  34</td></tr>\n",
       "\t<tr><td>163</td><td>  33</td></tr>\n",
       "\t<tr><td>139</td><td>  31</td></tr>\n",
       "\t<tr><td>36 </td><td>  29</td></tr>\n",
       "\t<tr><td>166</td><td>  28</td></tr>\n",
       "\t<tr><td>16 </td><td>  27</td></tr>\n",
       "\t<tr><td>42 </td><td>  27</td></tr>\n",
       "\t<tr><td>35 </td><td>  26</td></tr>\n",
       "\t<tr><td>70 </td><td>  26</td></tr>\n",
       "\t<tr><td>161</td><td>  26</td></tr>\n",
       "\t<tr><td>89 </td><td>  25</td></tr>\n",
       "\t<tr><td>119</td><td>  24</td></tr>\n",
       "\t<tr><td>167</td><td>  24</td></tr>\n",
       "\t<tr><td>155</td><td>  23</td></tr>\n",
       "\t<tr><td>169</td><td>  23</td></tr>\n",
       "\t<tr><td>74 </td><td>  22</td></tr>\n",
       "\t<tr><td>146</td><td>  22</td></tr>\n",
       "\t<tr><td>160</td><td>  22</td></tr>\n",
       "\t<tr><td>120</td><td>  21</td></tr>\n",
       "\t<tr><td>68 </td><td>  19</td></tr>\n",
       "\t<tr><td>72 </td><td>  19</td></tr>\n",
       "\t<tr><td>90 </td><td>  19</td></tr>\n",
       "\t<tr><td>59 </td><td>  18</td></tr>\n",
       "\t<tr><td>93 </td><td>  18</td></tr>\n",
       "\t<tr><td>7  </td><td>  17</td></tr>\n",
       "\t<tr><td>105</td><td>  16</td></tr>\n",
       "\t<tr><td>107</td><td>  16</td></tr>\n",
       "\t<tr><td>116</td><td>  16</td></tr>\n",
       "\t<tr><td>127</td><td>  16</td></tr>\n",
       "\t<tr><td>173</td><td>  16</td></tr>\n",
       "\t<tr><td>24 </td><td>  15</td></tr>\n",
       "\t<tr><td>100</td><td>  15</td></tr>\n",
       "\t<tr><td>174</td><td>  15</td></tr>\n",
       "\t<tr><td>69 </td><td>  14</td></tr>\n",
       "\t<tr><td>79 </td><td>  14</td></tr>\n",
       "\t<tr><td>86 </td><td>  14</td></tr>\n",
       "\t<tr><td>145</td><td>  14</td></tr>\n",
       "\t<tr><td>87 </td><td>  13</td></tr>\n",
       "\t<tr><td>164</td><td>  12</td></tr>\n",
       "\t<tr><td>136</td><td>  11</td></tr>\n",
       "\t<tr><td>23 </td><td>  10</td></tr>\n",
       "\t<tr><td>101</td><td>   9</td></tr>\n",
       "\t<tr><td>152</td><td>   9</td></tr>\n",
       "\t<tr><td>162</td><td>   9</td></tr>\n",
       "\t<tr><td>165</td><td>   8</td></tr>\n",
       "\t<tr><td>170</td><td>   8</td></tr>\n",
       "\t<tr><td>171</td><td>   7</td></tr>\n",
       "\t<tr><td>168</td><td>   6</td></tr>\n",
       "\t<tr><td>112</td><td>   5</td></tr>\n",
       "\t<tr><td>172</td><td>   5</td></tr>\n",
       "\t<tr><td>153</td><td>   4</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 174 × 2\n",
       "\\begin{tabular}{ll}\n",
       " cluster & n\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t 50  & 1126\\\\\n",
       "\t 62  &  975\\\\\n",
       "\t 71  &  925\\\\\n",
       "\t 19  &  844\\\\\n",
       "\t 3   &  571\\\\\n",
       "\t 46  &  341\\\\\n",
       "\t 132 &  311\\\\\n",
       "\t 44  &  310\\\\\n",
       "\t 11  &  309\\\\\n",
       "\t 18  &  293\\\\\n",
       "\t 21  &  265\\\\\n",
       "\t 6   &  254\\\\\n",
       "\t 76  &  243\\\\\n",
       "\t 33  &  241\\\\\n",
       "\t 9   &  237\\\\\n",
       "\t 8   &  224\\\\\n",
       "\t 29  &  209\\\\\n",
       "\t 4   &  206\\\\\n",
       "\t 106 &  206\\\\\n",
       "\t 118 &  202\\\\\n",
       "\t 20  &  199\\\\\n",
       "\t 61  &  199\\\\\n",
       "\t 60  &  188\\\\\n",
       "\t 53  &  184\\\\\n",
       "\t 64  &  182\\\\\n",
       "\t 109 &  180\\\\\n",
       "\t 32  &  176\\\\\n",
       "\t 13  &  172\\\\\n",
       "\t 38  &  172\\\\\n",
       "\t 77  &  167\\\\\n",
       "\t 111 &  165\\\\\n",
       "\t 56  &  162\\\\\n",
       "\t 43  &  155\\\\\n",
       "\t 58  &  155\\\\\n",
       "\t 31  &  153\\\\\n",
       "\t 97  &  149\\\\\n",
       "\t 14  &  147\\\\\n",
       "\t 65  &  145\\\\\n",
       "\t 133 &  142\\\\\n",
       "\t 2   &  141\\\\\n",
       "\t 34  &  136\\\\\n",
       "\t 67  &  135\\\\\n",
       "\t 39  &  132\\\\\n",
       "\t 82  &  132\\\\\n",
       "\t 104 &  129\\\\\n",
       "\t 17  &  128\\\\\n",
       "\t 115 &  125\\\\\n",
       "\t 113 &  121\\\\\n",
       "\t 135 &  121\\\\\n",
       "\t 5   &  119\\\\\n",
       "\t 117 &  118\\\\\n",
       "\t 40  &  117\\\\\n",
       "\t 156 &  113\\\\\n",
       "\t 131 &  111\\\\\n",
       "\t 122 &  109\\\\\n",
       "\t 27  &  108\\\\\n",
       "\t 30  &  108\\\\\n",
       "\t 66  &  108\\\\\n",
       "\t 138 &  105\\\\\n",
       "\t 25  &  103\\\\\n",
       "\t 15  &  102\\\\\n",
       "\t 45  &  102\\\\\n",
       "\t 75  &  101\\\\\n",
       "\t 98  &  101\\\\\n",
       "\t 108 &  101\\\\\n",
       "\t 126 &  101\\\\\n",
       "\t 55  &   99\\\\\n",
       "\t 114 &   98\\\\\n",
       "\t 140 &   98\\\\\n",
       "\t 12  &   97\\\\\n",
       "\t 110 &   97\\\\\n",
       "\t 123 &   97\\\\\n",
       "\t 121 &   93\\\\\n",
       "\t 80  &   90\\\\\n",
       "\t 54  &   89\\\\\n",
       "\t 41  &   88\\\\\n",
       "\t 150 &   88\\\\\n",
       "\t 26  &   84\\\\\n",
       "\t 28  &   83\\\\\n",
       "\t 125 &   83\\\\\n",
       "\t 73  &   82\\\\\n",
       "\t 78  &   81\\\\\n",
       "\t 137 &   80\\\\\n",
       "\t 99  &   78\\\\\n",
       "\t 134 &   78\\\\\n",
       "\t 94  &   77\\\\\n",
       "\t 96  &   74\\\\\n",
       "\t 149 &   72\\\\\n",
       "\t 1   &   70\\\\\n",
       "\t 85  &   70\\\\\n",
       "\t 129 &   69\\\\\n",
       "\t 124 &   68\\\\\n",
       "\t 49  &   67\\\\\n",
       "\t 37  &   66\\\\\n",
       "\t 102 &   64\\\\\n",
       "\t 91  &   62\\\\\n",
       "\t 10  &   61\\\\\n",
       "\t 142 &   60\\\\\n",
       "\t 63  &   59\\\\\n",
       "\t 47  &   58\\\\\n",
       "\t 154 &   58\\\\\n",
       "\t 92  &   57\\\\\n",
       "\t 143 &   56\\\\\n",
       "\t 157 &   55\\\\\n",
       "\t 52  &   54\\\\\n",
       "\t 128 &   53\\\\\n",
       "\t 141 &   53\\\\\n",
       "\t 57  &   51\\\\\n",
       "\t 84  &   51\\\\\n",
       "\t 103 &   49\\\\\n",
       "\t 147 &   48\\\\\n",
       "\t 148 &   48\\\\\n",
       "\t 83  &   46\\\\\n",
       "\t 151 &   45\\\\\n",
       "\t 51  &   44\\\\\n",
       "\t 158 &   43\\\\\n",
       "\t 95  &   41\\\\\n",
       "\t 130 &   41\\\\\n",
       "\t 144 &   41\\\\\n",
       "\t 22  &   39\\\\\n",
       "\t 159 &   38\\\\\n",
       "\t 81  &   37\\\\\n",
       "\t 88  &   35\\\\\n",
       "\t 48  &   34\\\\\n",
       "\t 163 &   33\\\\\n",
       "\t 139 &   31\\\\\n",
       "\t 36  &   29\\\\\n",
       "\t 166 &   28\\\\\n",
       "\t 16  &   27\\\\\n",
       "\t 42  &   27\\\\\n",
       "\t 35  &   26\\\\\n",
       "\t 70  &   26\\\\\n",
       "\t 161 &   26\\\\\n",
       "\t 89  &   25\\\\\n",
       "\t 119 &   24\\\\\n",
       "\t 167 &   24\\\\\n",
       "\t 155 &   23\\\\\n",
       "\t 169 &   23\\\\\n",
       "\t 74  &   22\\\\\n",
       "\t 146 &   22\\\\\n",
       "\t 160 &   22\\\\\n",
       "\t 120 &   21\\\\\n",
       "\t 68  &   19\\\\\n",
       "\t 72  &   19\\\\\n",
       "\t 90  &   19\\\\\n",
       "\t 59  &   18\\\\\n",
       "\t 93  &   18\\\\\n",
       "\t 7   &   17\\\\\n",
       "\t 105 &   16\\\\\n",
       "\t 107 &   16\\\\\n",
       "\t 116 &   16\\\\\n",
       "\t 127 &   16\\\\\n",
       "\t 173 &   16\\\\\n",
       "\t 24  &   15\\\\\n",
       "\t 100 &   15\\\\\n",
       "\t 174 &   15\\\\\n",
       "\t 69  &   14\\\\\n",
       "\t 79  &   14\\\\\n",
       "\t 86  &   14\\\\\n",
       "\t 145 &   14\\\\\n",
       "\t 87  &   13\\\\\n",
       "\t 164 &   12\\\\\n",
       "\t 136 &   11\\\\\n",
       "\t 23  &   10\\\\\n",
       "\t 101 &    9\\\\\n",
       "\t 152 &    9\\\\\n",
       "\t 162 &    9\\\\\n",
       "\t 165 &    8\\\\\n",
       "\t 170 &    8\\\\\n",
       "\t 171 &    7\\\\\n",
       "\t 168 &    6\\\\\n",
       "\t 112 &    5\\\\\n",
       "\t 172 &    5\\\\\n",
       "\t 153 &    4\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 174 × 2\n",
       "\n",
       "| cluster &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| 50  | 1126 |\n",
       "| 62  |  975 |\n",
       "| 71  |  925 |\n",
       "| 19  |  844 |\n",
       "| 3   |  571 |\n",
       "| 46  |  341 |\n",
       "| 132 |  311 |\n",
       "| 44  |  310 |\n",
       "| 11  |  309 |\n",
       "| 18  |  293 |\n",
       "| 21  |  265 |\n",
       "| 6   |  254 |\n",
       "| 76  |  243 |\n",
       "| 33  |  241 |\n",
       "| 9   |  237 |\n",
       "| 8   |  224 |\n",
       "| 29  |  209 |\n",
       "| 4   |  206 |\n",
       "| 106 |  206 |\n",
       "| 118 |  202 |\n",
       "| 20  |  199 |\n",
       "| 61  |  199 |\n",
       "| 60  |  188 |\n",
       "| 53  |  184 |\n",
       "| 64  |  182 |\n",
       "| 109 |  180 |\n",
       "| 32  |  176 |\n",
       "| 13  |  172 |\n",
       "| 38  |  172 |\n",
       "| 77  |  167 |\n",
       "| 111 |  165 |\n",
       "| 56  |  162 |\n",
       "| 43  |  155 |\n",
       "| 58  |  155 |\n",
       "| 31  |  153 |\n",
       "| 97  |  149 |\n",
       "| 14  |  147 |\n",
       "| 65  |  145 |\n",
       "| 133 |  142 |\n",
       "| 2   |  141 |\n",
       "| 34  |  136 |\n",
       "| 67  |  135 |\n",
       "| 39  |  132 |\n",
       "| 82  |  132 |\n",
       "| 104 |  129 |\n",
       "| 17  |  128 |\n",
       "| 115 |  125 |\n",
       "| 113 |  121 |\n",
       "| 135 |  121 |\n",
       "| 5   |  119 |\n",
       "| 117 |  118 |\n",
       "| 40  |  117 |\n",
       "| 156 |  113 |\n",
       "| 131 |  111 |\n",
       "| 122 |  109 |\n",
       "| 27  |  108 |\n",
       "| 30  |  108 |\n",
       "| 66  |  108 |\n",
       "| 138 |  105 |\n",
       "| 25  |  103 |\n",
       "| 15  |  102 |\n",
       "| 45  |  102 |\n",
       "| 75  |  101 |\n",
       "| 98  |  101 |\n",
       "| 108 |  101 |\n",
       "| 126 |  101 |\n",
       "| 55  |   99 |\n",
       "| 114 |   98 |\n",
       "| 140 |   98 |\n",
       "| 12  |   97 |\n",
       "| 110 |   97 |\n",
       "| 123 |   97 |\n",
       "| 121 |   93 |\n",
       "| 80  |   90 |\n",
       "| 54  |   89 |\n",
       "| 41  |   88 |\n",
       "| 150 |   88 |\n",
       "| 26  |   84 |\n",
       "| 28  |   83 |\n",
       "| 125 |   83 |\n",
       "| 73  |   82 |\n",
       "| 78  |   81 |\n",
       "| 137 |   80 |\n",
       "| 99  |   78 |\n",
       "| 134 |   78 |\n",
       "| 94  |   77 |\n",
       "| 96  |   74 |\n",
       "| 149 |   72 |\n",
       "| 1   |   70 |\n",
       "| 85  |   70 |\n",
       "| 129 |   69 |\n",
       "| 124 |   68 |\n",
       "| 49  |   67 |\n",
       "| 37  |   66 |\n",
       "| 102 |   64 |\n",
       "| 91  |   62 |\n",
       "| 10  |   61 |\n",
       "| 142 |   60 |\n",
       "| 63  |   59 |\n",
       "| 47  |   58 |\n",
       "| 154 |   58 |\n",
       "| 92  |   57 |\n",
       "| 143 |   56 |\n",
       "| 157 |   55 |\n",
       "| 52  |   54 |\n",
       "| 128 |   53 |\n",
       "| 141 |   53 |\n",
       "| 57  |   51 |\n",
       "| 84  |   51 |\n",
       "| 103 |   49 |\n",
       "| 147 |   48 |\n",
       "| 148 |   48 |\n",
       "| 83  |   46 |\n",
       "| 151 |   45 |\n",
       "| 51  |   44 |\n",
       "| 158 |   43 |\n",
       "| 95  |   41 |\n",
       "| 130 |   41 |\n",
       "| 144 |   41 |\n",
       "| 22  |   39 |\n",
       "| 159 |   38 |\n",
       "| 81  |   37 |\n",
       "| 88  |   35 |\n",
       "| 48  |   34 |\n",
       "| 163 |   33 |\n",
       "| 139 |   31 |\n",
       "| 36  |   29 |\n",
       "| 166 |   28 |\n",
       "| 16  |   27 |\n",
       "| 42  |   27 |\n",
       "| 35  |   26 |\n",
       "| 70  |   26 |\n",
       "| 161 |   26 |\n",
       "| 89  |   25 |\n",
       "| 119 |   24 |\n",
       "| 167 |   24 |\n",
       "| 155 |   23 |\n",
       "| 169 |   23 |\n",
       "| 74  |   22 |\n",
       "| 146 |   22 |\n",
       "| 160 |   22 |\n",
       "| 120 |   21 |\n",
       "| 68  |   19 |\n",
       "| 72  |   19 |\n",
       "| 90  |   19 |\n",
       "| 59  |   18 |\n",
       "| 93  |   18 |\n",
       "| 7   |   17 |\n",
       "| 105 |   16 |\n",
       "| 107 |   16 |\n",
       "| 116 |   16 |\n",
       "| 127 |   16 |\n",
       "| 173 |   16 |\n",
       "| 24  |   15 |\n",
       "| 100 |   15 |\n",
       "| 174 |   15 |\n",
       "| 69  |   14 |\n",
       "| 79  |   14 |\n",
       "| 86  |   14 |\n",
       "| 145 |   14 |\n",
       "| 87  |   13 |\n",
       "| 164 |   12 |\n",
       "| 136 |   11 |\n",
       "| 23  |   10 |\n",
       "| 101 |    9 |\n",
       "| 152 |    9 |\n",
       "| 162 |    9 |\n",
       "| 165 |    8 |\n",
       "| 170 |    8 |\n",
       "| 171 |    7 |\n",
       "| 168 |    6 |\n",
       "| 112 |    5 |\n",
       "| 172 |    5 |\n",
       "| 153 |    4 |\n",
       "\n"
      ],
      "text/plain": [
       "    cluster n   \n",
       "1   50      1126\n",
       "2   62       975\n",
       "3   71       925\n",
       "4   19       844\n",
       "5   3        571\n",
       "6   46       341\n",
       "7   132      311\n",
       "8   44       310\n",
       "9   11       309\n",
       "10  18       293\n",
       "11  21       265\n",
       "12  6        254\n",
       "13  76       243\n",
       "14  33       241\n",
       "15  9        237\n",
       "16  8        224\n",
       "17  29       209\n",
       "18  4        206\n",
       "19  106      206\n",
       "20  118      202\n",
       "21  20       199\n",
       "22  61       199\n",
       "23  60       188\n",
       "24  53       184\n",
       "25  64       182\n",
       "26  109      180\n",
       "27  32       176\n",
       "28  13       172\n",
       "29  38       172\n",
       "30  77       167\n",
       "31  111      165\n",
       "32  56       162\n",
       "33  43       155\n",
       "34  58       155\n",
       "35  31       153\n",
       "36  97       149\n",
       "37  14       147\n",
       "38  65       145\n",
       "39  133      142\n",
       "40  2        141\n",
       "41  34       136\n",
       "42  67       135\n",
       "43  39       132\n",
       "44  82       132\n",
       "45  104      129\n",
       "46  17       128\n",
       "47  115      125\n",
       "48  113      121\n",
       "49  135      121\n",
       "50  5        119\n",
       "51  117      118\n",
       "52  40       117\n",
       "53  156      113\n",
       "54  131      111\n",
       "55  122      109\n",
       "56  27       108\n",
       "57  30       108\n",
       "58  66       108\n",
       "59  138      105\n",
       "60  25       103\n",
       "61  15       102\n",
       "62  45       102\n",
       "63  75       101\n",
       "64  98       101\n",
       "65  108      101\n",
       "66  126      101\n",
       "67  55        99\n",
       "68  114       98\n",
       "69  140       98\n",
       "70  12        97\n",
       "71  110       97\n",
       "72  123       97\n",
       "73  121       93\n",
       "74  80        90\n",
       "75  54        89\n",
       "76  41        88\n",
       "77  150       88\n",
       "78  26        84\n",
       "79  28        83\n",
       "80  125       83\n",
       "81  73        82\n",
       "82  78        81\n",
       "83  137       80\n",
       "84  99        78\n",
       "85  134       78\n",
       "86  94        77\n",
       "87  96        74\n",
       "88  149       72\n",
       "89  1         70\n",
       "90  85        70\n",
       "91  129       69\n",
       "92  124       68\n",
       "93  49        67\n",
       "94  37        66\n",
       "95  102       64\n",
       "96  91        62\n",
       "97  10        61\n",
       "98  142       60\n",
       "99  63        59\n",
       "100 47        58\n",
       "101 154       58\n",
       "102 92        57\n",
       "103 143       56\n",
       "104 157       55\n",
       "105 52        54\n",
       "106 128       53\n",
       "107 141       53\n",
       "108 57        51\n",
       "109 84        51\n",
       "110 103       49\n",
       "111 147       48\n",
       "112 148       48\n",
       "113 83        46\n",
       "114 151       45\n",
       "115 51        44\n",
       "116 158       43\n",
       "117 95        41\n",
       "118 130       41\n",
       "119 144       41\n",
       "120 22        39\n",
       "121 159       38\n",
       "122 81        37\n",
       "123 88        35\n",
       "124 48        34\n",
       "125 163       33\n",
       "126 139       31\n",
       "127 36        29\n",
       "128 166       28\n",
       "129 16        27\n",
       "130 42        27\n",
       "131 35        26\n",
       "132 70        26\n",
       "133 161       26\n",
       "134 89        25\n",
       "135 119       24\n",
       "136 167       24\n",
       "137 155       23\n",
       "138 169       23\n",
       "139 74        22\n",
       "140 146       22\n",
       "141 160       22\n",
       "142 120       21\n",
       "143 68        19\n",
       "144 72        19\n",
       "145 90        19\n",
       "146 59        18\n",
       "147 93        18\n",
       "148 7         17\n",
       "149 105       16\n",
       "150 107       16\n",
       "151 116       16\n",
       "152 127       16\n",
       "153 173       16\n",
       "154 24        15\n",
       "155 100       15\n",
       "156 174       15\n",
       "157 69        14\n",
       "158 79        14\n",
       "159 86        14\n",
       "160 145       14\n",
       "161 87        13\n",
       "162 164       12\n",
       "163 136       11\n",
       "164 23        10\n",
       "165 101        9\n",
       "166 152        9\n",
       "167 162        9\n",
       "168 165        8\n",
       "169 170        8\n",
       "170 171        7\n",
       "171 168        6\n",
       "172 112        5\n",
       "173 172        5\n",
       "174 153        4"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub_no_multitons  %>% filter(tmp.expr > 0 & tmp.expr < 2)  %>% count(cluster)  %>% arrange(-n)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1443,
   "metadata": {},
   "outputs": [],
   "source": [
    "bt_lt2  = (df_sub_no_multitons  %>% filter(tmp.expr > 0 & tmp.expr < 2))$cell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% bt_lt2), size = ifelse(df_sub300$cell %in% bt_lt2, 0.1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in df_sub_no_multitons %>% filter(tmp.expr >= 2): could not find function \"%>%\"\n",
     "output_type": "error",
     "traceback": [
      "Error in df_sub_no_multitons %>% filter(tmp.expr >= 2): could not find function \"%>%\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "bt_gte2  = (df_sub_no_multitons  %>% filter( tmp.expr >= 2))$cell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% bt_gte2), size = ifelse(df_sub300$cell %in% bt_gte2, 0.1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1448,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 15 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>annot</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>brain                     </td><td>  35</td></tr>\n",
       "\t<tr><td>epidermis                 </td><td>   9</td></tr>\n",
       "\t<tr><td>fat body                  </td><td>   6</td></tr>\n",
       "\t<tr><td>glia                      </td><td>   1</td></tr>\n",
       "\t<tr><td>hindgut / anal pad        </td><td>   1</td></tr>\n",
       "\t<tr><td>midgut                    </td><td>   7</td></tr>\n",
       "\t<tr><td>muscle                    </td><td>   4</td></tr>\n",
       "\t<tr><td>plasmatocytes             </td><td>   6</td></tr>\n",
       "\t<tr><td>salivary gland            </td><td> 128</td></tr>\n",
       "\t<tr><td>somatic muscle            </td><td>1239</td></tr>\n",
       "\t<tr><td>tracheal system           </td><td>   3</td></tr>\n",
       "\t<tr><td>trunk mesoderm primordium </td><td>   1</td></tr>\n",
       "\t<tr><td>ubiquitous                </td><td>  16</td></tr>\n",
       "\t<tr><td>visceral muscle           </td><td>   2</td></tr>\n",
       "\t<tr><td>visceral muscle primordium</td><td>   4</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 15 × 2\n",
       "\\begin{tabular}{ll}\n",
       " annot & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t brain                      &   35\\\\\n",
       "\t epidermis                  &    9\\\\\n",
       "\t fat body                   &    6\\\\\n",
       "\t glia                       &    1\\\\\n",
       "\t hindgut / anal pad         &    1\\\\\n",
       "\t midgut                     &    7\\\\\n",
       "\t muscle                     &    4\\\\\n",
       "\t plasmatocytes              &    6\\\\\n",
       "\t salivary gland             &  128\\\\\n",
       "\t somatic muscle             & 1239\\\\\n",
       "\t tracheal system            &    3\\\\\n",
       "\t trunk mesoderm primordium  &    1\\\\\n",
       "\t ubiquitous                 &   16\\\\\n",
       "\t visceral muscle            &    2\\\\\n",
       "\t visceral muscle primordium &    4\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 15 × 2\n",
       "\n",
       "| annot &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| brain                      |   35 |\n",
       "| epidermis                  |    9 |\n",
       "| fat body                   |    6 |\n",
       "| glia                       |    1 |\n",
       "| hindgut / anal pad         |    1 |\n",
       "| midgut                     |    7 |\n",
       "| muscle                     |    4 |\n",
       "| plasmatocytes              |    6 |\n",
       "| salivary gland             |  128 |\n",
       "| somatic muscle             | 1239 |\n",
       "| tracheal system            |    3 |\n",
       "| trunk mesoderm primordium  |    1 |\n",
       "| ubiquitous                 |   16 |\n",
       "| visceral muscle            |    2 |\n",
       "| visceral muscle primordium |    4 |\n",
       "\n"
      ],
      "text/plain": [
       "   annot                      n   \n",
       "1  brain                        35\n",
       "2  epidermis                     9\n",
       "3  fat body                      6\n",
       "4  glia                          1\n",
       "5  hindgut / anal pad            1\n",
       "6  midgut                        7\n",
       "7  muscle                        4\n",
       "8  plasmatocytes                 6\n",
       "9  salivary gland              128\n",
       "10 somatic muscle             1239\n",
       "11 tracheal system               3\n",
       "12 trunk mesoderm primordium     1\n",
       "13 ubiquitous                   16\n",
       "14 visceral muscle               2\n",
       "15 visceral muscle primordium    4"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub_no_multitons  %>% filter(tmp.expr > 0 & tmp.expr < 2 &\n",
    "                               cluster %in% c(34, 40, 39, 9, 18, 106, 46, 18))  %>% count(annot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1449,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 43 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>annot</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Malpighian tubule                </td><td>  56</td></tr>\n",
       "\t<tr><td>amnioserosa                      </td><td> 237</td></tr>\n",
       "\t<tr><td>brain                            </td><td>6147</td></tr>\n",
       "\t<tr><td>brain primordium                 </td><td>  78</td></tr>\n",
       "\t<tr><td>crystal cell                     </td><td>   1</td></tr>\n",
       "\t<tr><td>ectoderm primordium              </td><td>  18</td></tr>\n",
       "\t<tr><td>endoderm primordium              </td><td>   1</td></tr>\n",
       "\t<tr><td>epidermis                        </td><td>2246</td></tr>\n",
       "\t<tr><td>epidermis primordium             </td><td>  24</td></tr>\n",
       "\t<tr><td>fat body                         </td><td> 426</td></tr>\n",
       "\t<tr><td>foregut / hindgut / anal pad     </td><td> 266</td></tr>\n",
       "\t<tr><td>foregut / hindgut primordium     </td><td>   5</td></tr>\n",
       "\t<tr><td>germ cell                        </td><td>  38</td></tr>\n",
       "\t<tr><td>glia                             </td><td>  29</td></tr>\n",
       "\t<tr><td>hindgut / anal pad               </td><td> 122</td></tr>\n",
       "\t<tr><td>hindgut primordium               </td><td>   6</td></tr>\n",
       "\t<tr><td>midgut                           </td><td>1239</td></tr>\n",
       "\t<tr><td>midgut primordium                </td><td>   7</td></tr>\n",
       "\t<tr><td>midline primordium               </td><td>   9</td></tr>\n",
       "\t<tr><td>muscle                           </td><td> 717</td></tr>\n",
       "\t<tr><td>peripheral nervous system        </td><td>  47</td></tr>\n",
       "\t<tr><td>plasmatocytes                    </td><td> 363</td></tr>\n",
       "\t<tr><td>plasmatocytes anlage             </td><td>  18</td></tr>\n",
       "\t<tr><td>procrystal cell                  </td><td> 199</td></tr>\n",
       "\t<tr><td>proventriculus                   </td><td> 109</td></tr>\n",
       "\t<tr><td>salivary gland                   </td><td>  61</td></tr>\n",
       "\t<tr><td>salivary gland body primordium   </td><td>  10</td></tr>\n",
       "\t<tr><td>sensory nervous system           </td><td> 183</td></tr>\n",
       "\t<tr><td>sensory nervous system primordium</td><td>   7</td></tr>\n",
       "\t<tr><td>somatic muscle                   </td><td> 849</td></tr>\n",
       "\t<tr><td>somatic muscle primordium        </td><td>  36</td></tr>\n",
       "\t<tr><td>spiracle (tracheal system)       </td><td>  15</td></tr>\n",
       "\t<tr><td>spiracles (tracheal system)      </td><td>  47</td></tr>\n",
       "\t<tr><td>tracheal primordium              </td><td>  45</td></tr>\n",
       "\t<tr><td>tracheal system                  </td><td> 488</td></tr>\n",
       "\t<tr><td>trunk mesoderm primordium        </td><td>  11</td></tr>\n",
       "\t<tr><td>ubiquitous                       </td><td>3016</td></tr>\n",
       "\t<tr><td>unknown                          </td><td>   4</td></tr>\n",
       "\t<tr><td>ventral nerve cord               </td><td> 467</td></tr>\n",
       "\t<tr><td>ventral nerve cord primordium    </td><td>   2</td></tr>\n",
       "\t<tr><td>visceral muscle                  </td><td>  75</td></tr>\n",
       "\t<tr><td>visceral muscle primordium       </td><td>  78</td></tr>\n",
       "\t<tr><td>yolk nuclei                      </td><td>  10</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 43 × 2\n",
       "\\begin{tabular}{ll}\n",
       " annot & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Malpighian tubule                 &   56\\\\\n",
       "\t amnioserosa                       &  237\\\\\n",
       "\t brain                             & 6147\\\\\n",
       "\t brain primordium                  &   78\\\\\n",
       "\t crystal cell                      &    1\\\\\n",
       "\t ectoderm primordium               &   18\\\\\n",
       "\t endoderm primordium               &    1\\\\\n",
       "\t epidermis                         & 2246\\\\\n",
       "\t epidermis primordium              &   24\\\\\n",
       "\t fat body                          &  426\\\\\n",
       "\t foregut / hindgut / anal pad      &  266\\\\\n",
       "\t foregut / hindgut primordium      &    5\\\\\n",
       "\t germ cell                         &   38\\\\\n",
       "\t glia                              &   29\\\\\n",
       "\t hindgut / anal pad                &  122\\\\\n",
       "\t hindgut primordium                &    6\\\\\n",
       "\t midgut                            & 1239\\\\\n",
       "\t midgut primordium                 &    7\\\\\n",
       "\t midline primordium                &    9\\\\\n",
       "\t muscle                            &  717\\\\\n",
       "\t peripheral nervous system         &   47\\\\\n",
       "\t plasmatocytes                     &  363\\\\\n",
       "\t plasmatocytes anlage              &   18\\\\\n",
       "\t procrystal cell                   &  199\\\\\n",
       "\t proventriculus                    &  109\\\\\n",
       "\t salivary gland                    &   61\\\\\n",
       "\t salivary gland body primordium    &   10\\\\\n",
       "\t sensory nervous system            &  183\\\\\n",
       "\t sensory nervous system primordium &    7\\\\\n",
       "\t somatic muscle                    &  849\\\\\n",
       "\t somatic muscle primordium         &   36\\\\\n",
       "\t spiracle (tracheal system)        &   15\\\\\n",
       "\t spiracles (tracheal system)       &   47\\\\\n",
       "\t tracheal primordium               &   45\\\\\n",
       "\t tracheal system                   &  488\\\\\n",
       "\t trunk mesoderm primordium         &   11\\\\\n",
       "\t ubiquitous                        & 3016\\\\\n",
       "\t unknown                           &    4\\\\\n",
       "\t ventral nerve cord                &  467\\\\\n",
       "\t ventral nerve cord primordium     &    2\\\\\n",
       "\t visceral muscle                   &   75\\\\\n",
       "\t visceral muscle primordium        &   78\\\\\n",
       "\t yolk nuclei                       &   10\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 43 × 2\n",
       "\n",
       "| annot &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Malpighian tubule                 |   56 |\n",
       "| amnioserosa                       |  237 |\n",
       "| brain                             | 6147 |\n",
       "| brain primordium                  |   78 |\n",
       "| crystal cell                      |    1 |\n",
       "| ectoderm primordium               |   18 |\n",
       "| endoderm primordium               |    1 |\n",
       "| epidermis                         | 2246 |\n",
       "| epidermis primordium              |   24 |\n",
       "| fat body                          |  426 |\n",
       "| foregut / hindgut / anal pad      |  266 |\n",
       "| foregut / hindgut primordium      |    5 |\n",
       "| germ cell                         |   38 |\n",
       "| glia                              |   29 |\n",
       "| hindgut / anal pad                |  122 |\n",
       "| hindgut primordium                |    6 |\n",
       "| midgut                            | 1239 |\n",
       "| midgut primordium                 |    7 |\n",
       "| midline primordium                |    9 |\n",
       "| muscle                            |  717 |\n",
       "| peripheral nervous system         |   47 |\n",
       "| plasmatocytes                     |  363 |\n",
       "| plasmatocytes anlage              |   18 |\n",
       "| procrystal cell                   |  199 |\n",
       "| proventriculus                    |  109 |\n",
       "| salivary gland                    |   61 |\n",
       "| salivary gland body primordium    |   10 |\n",
       "| sensory nervous system            |  183 |\n",
       "| sensory nervous system primordium |    7 |\n",
       "| somatic muscle                    |  849 |\n",
       "| somatic muscle primordium         |   36 |\n",
       "| spiracle (tracheal system)        |   15 |\n",
       "| spiracles (tracheal system)       |   47 |\n",
       "| tracheal primordium               |   45 |\n",
       "| tracheal system                   |  488 |\n",
       "| trunk mesoderm primordium         |   11 |\n",
       "| ubiquitous                        | 3016 |\n",
       "| unknown                           |    4 |\n",
       "| ventral nerve cord                |  467 |\n",
       "| ventral nerve cord primordium     |    2 |\n",
       "| visceral muscle                   |   75 |\n",
       "| visceral muscle primordium        |   78 |\n",
       "| yolk nuclei                       |   10 |\n",
       "\n"
      ],
      "text/plain": [
       "   annot                             n   \n",
       "1  Malpighian tubule                   56\n",
       "2  amnioserosa                        237\n",
       "3  brain                             6147\n",
       "4  brain primordium                    78\n",
       "5  crystal cell                         1\n",
       "6  ectoderm primordium                 18\n",
       "7  endoderm primordium                  1\n",
       "8  epidermis                         2246\n",
       "9  epidermis primordium                24\n",
       "10 fat body                           426\n",
       "11 foregut / hindgut / anal pad       266\n",
       "12 foregut / hindgut primordium         5\n",
       "13 germ cell                           38\n",
       "14 glia                                29\n",
       "15 hindgut / anal pad                 122\n",
       "16 hindgut primordium                   6\n",
       "17 midgut                            1239\n",
       "18 midgut primordium                    7\n",
       "19 midline primordium                   9\n",
       "20 muscle                             717\n",
       "21 peripheral nervous system           47\n",
       "22 plasmatocytes                      363\n",
       "23 plasmatocytes anlage                18\n",
       "24 procrystal cell                    199\n",
       "25 proventriculus                     109\n",
       "26 salivary gland                      61\n",
       "27 salivary gland body primordium      10\n",
       "28 sensory nervous system             183\n",
       "29 sensory nervous system primordium    7\n",
       "30 somatic muscle                     849\n",
       "31 somatic muscle primordium           36\n",
       "32 spiracle (tracheal system)          15\n",
       "33 spiracles (tracheal system)         47\n",
       "34 tracheal primordium                 45\n",
       "35 tracheal system                    488\n",
       "36 trunk mesoderm primordium           11\n",
       "37 ubiquitous                        3016\n",
       "38 unknown                              4\n",
       "39 ventral nerve cord                 467\n",
       "40 ventral nerve cord primordium        2\n",
       "41 visceral muscle                     75\n",
       "42 visceral muscle primordium          78\n",
       "43 yolk nuclei                         10"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub_no_multitons  %>% filter(tmp.expr > 0 & tmp.expr < 2 &\n",
    "                               !(cluster %in% c(34, 40, 39, 9, 18, 106, 46, 18)))  %>% count(annot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1454,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sub300$tmp.expr = counts(cds_sub)[\"bt\", ]/df_sub300$Size_Factor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1455,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 47 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>annot</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Malpighian tubule                   </td><td> 115</td></tr>\n",
       "\t<tr><td>amnioserosa                         </td><td> 290</td></tr>\n",
       "\t<tr><td>brain                               </td><td>6993</td></tr>\n",
       "\t<tr><td>brain primordium                    </td><td> 157</td></tr>\n",
       "\t<tr><td>cellular blastoderm / pole cell     </td><td>   2</td></tr>\n",
       "\t<tr><td>crystal cell                        </td><td>  14</td></tr>\n",
       "\t<tr><td>ectoderm primordium                 </td><td>  25</td></tr>\n",
       "\t<tr><td>endoderm primordium                 </td><td>   5</td></tr>\n",
       "\t<tr><td>epidermis                           </td><td>4941</td></tr>\n",
       "\t<tr><td>epidermis primordium                </td><td> 112</td></tr>\n",
       "\t<tr><td>fat body                            </td><td> 623</td></tr>\n",
       "\t<tr><td>foregut / hindgut / anal pad        </td><td>1182</td></tr>\n",
       "\t<tr><td>foregut / hindgut primordium        </td><td>  10</td></tr>\n",
       "\t<tr><td>germ cell                           </td><td>  82</td></tr>\n",
       "\t<tr><td>glia                                </td><td>  53</td></tr>\n",
       "\t<tr><td>hindgut / anal pad                  </td><td> 193</td></tr>\n",
       "\t<tr><td>hindgut primordium                  </td><td>  28</td></tr>\n",
       "\t<tr><td>mesoderm primordium                 </td><td>   1</td></tr>\n",
       "\t<tr><td>midgut                              </td><td>1830</td></tr>\n",
       "\t<tr><td>midgut primordium                   </td><td>  23</td></tr>\n",
       "\t<tr><td>midline primordium                  </td><td>  20</td></tr>\n",
       "\t<tr><td>muscle                              </td><td>1896</td></tr>\n",
       "\t<tr><td>peripheral nervous system           </td><td>  92</td></tr>\n",
       "\t<tr><td>peripheral nervous system primordium</td><td>   5</td></tr>\n",
       "\t<tr><td>plasmatocytes                       </td><td> 564</td></tr>\n",
       "\t<tr><td>plasmatocytes anlage                </td><td>  31</td></tr>\n",
       "\t<tr><td>procrystal cell                     </td><td> 230</td></tr>\n",
       "\t<tr><td>proventriculus                      </td><td> 214</td></tr>\n",
       "\t<tr><td>salivary gland                      </td><td>  85</td></tr>\n",
       "\t<tr><td>salivary gland body primordium      </td><td>  26</td></tr>\n",
       "\t<tr><td>salivary gland primordium           </td><td>   2</td></tr>\n",
       "\t<tr><td>sensory nervous system              </td><td> 544</td></tr>\n",
       "\t<tr><td>sensory nervous system primordium   </td><td>  31</td></tr>\n",
       "\t<tr><td>somatic muscle                      </td><td>1882</td></tr>\n",
       "\t<tr><td>somatic muscle primordium           </td><td> 106</td></tr>\n",
       "\t<tr><td>spiracle (tracheal system)          </td><td>  72</td></tr>\n",
       "\t<tr><td>spiracles (tracheal system)         </td><td>  76</td></tr>\n",
       "\t<tr><td>tracheal primordium                 </td><td>  96</td></tr>\n",
       "\t<tr><td>tracheal system                     </td><td> 916</td></tr>\n",
       "\t<tr><td>trunk mesoderm primordium           </td><td>  40</td></tr>\n",
       "\t<tr><td>ubiquitous                          </td><td>6029</td></tr>\n",
       "\t<tr><td>unknown                             </td><td>  40</td></tr>\n",
       "\t<tr><td>ventral nerve cord                  </td><td> 659</td></tr>\n",
       "\t<tr><td>ventral nerve cord primordium       </td><td>  12</td></tr>\n",
       "\t<tr><td>visceral muscle                     </td><td> 153</td></tr>\n",
       "\t<tr><td>visceral muscle primordium          </td><td> 317</td></tr>\n",
       "\t<tr><td>yolk nuclei                         </td><td> 123</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 47 × 2\n",
       "\\begin{tabular}{ll}\n",
       " annot & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Malpighian tubule                    &  115\\\\\n",
       "\t amnioserosa                          &  290\\\\\n",
       "\t brain                                & 6993\\\\\n",
       "\t brain primordium                     &  157\\\\\n",
       "\t cellular blastoderm / pole cell      &    2\\\\\n",
       "\t crystal cell                         &   14\\\\\n",
       "\t ectoderm primordium                  &   25\\\\\n",
       "\t endoderm primordium                  &    5\\\\\n",
       "\t epidermis                            & 4941\\\\\n",
       "\t epidermis primordium                 &  112\\\\\n",
       "\t fat body                             &  623\\\\\n",
       "\t foregut / hindgut / anal pad         & 1182\\\\\n",
       "\t foregut / hindgut primordium         &   10\\\\\n",
       "\t germ cell                            &   82\\\\\n",
       "\t glia                                 &   53\\\\\n",
       "\t hindgut / anal pad                   &  193\\\\\n",
       "\t hindgut primordium                   &   28\\\\\n",
       "\t mesoderm primordium                  &    1\\\\\n",
       "\t midgut                               & 1830\\\\\n",
       "\t midgut primordium                    &   23\\\\\n",
       "\t midline primordium                   &   20\\\\\n",
       "\t muscle                               & 1896\\\\\n",
       "\t peripheral nervous system            &   92\\\\\n",
       "\t peripheral nervous system primordium &    5\\\\\n",
       "\t plasmatocytes                        &  564\\\\\n",
       "\t plasmatocytes anlage                 &   31\\\\\n",
       "\t procrystal cell                      &  230\\\\\n",
       "\t proventriculus                       &  214\\\\\n",
       "\t salivary gland                       &   85\\\\\n",
       "\t salivary gland body primordium       &   26\\\\\n",
       "\t salivary gland primordium            &    2\\\\\n",
       "\t sensory nervous system               &  544\\\\\n",
       "\t sensory nervous system primordium    &   31\\\\\n",
       "\t somatic muscle                       & 1882\\\\\n",
       "\t somatic muscle primordium            &  106\\\\\n",
       "\t spiracle (tracheal system)           &   72\\\\\n",
       "\t spiracles (tracheal system)          &   76\\\\\n",
       "\t tracheal primordium                  &   96\\\\\n",
       "\t tracheal system                      &  916\\\\\n",
       "\t trunk mesoderm primordium            &   40\\\\\n",
       "\t ubiquitous                           & 6029\\\\\n",
       "\t unknown                              &   40\\\\\n",
       "\t ventral nerve cord                   &  659\\\\\n",
       "\t ventral nerve cord primordium        &   12\\\\\n",
       "\t visceral muscle                      &  153\\\\\n",
       "\t visceral muscle primordium           &  317\\\\\n",
       "\t yolk nuclei                          &  123\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 47 × 2\n",
       "\n",
       "| annot &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Malpighian tubule                    |  115 |\n",
       "| amnioserosa                          |  290 |\n",
       "| brain                                | 6993 |\n",
       "| brain primordium                     |  157 |\n",
       "| cellular blastoderm / pole cell      |    2 |\n",
       "| crystal cell                         |   14 |\n",
       "| ectoderm primordium                  |   25 |\n",
       "| endoderm primordium                  |    5 |\n",
       "| epidermis                            | 4941 |\n",
       "| epidermis primordium                 |  112 |\n",
       "| fat body                             |  623 |\n",
       "| foregut / hindgut / anal pad         | 1182 |\n",
       "| foregut / hindgut primordium         |   10 |\n",
       "| germ cell                            |   82 |\n",
       "| glia                                 |   53 |\n",
       "| hindgut / anal pad                   |  193 |\n",
       "| hindgut primordium                   |   28 |\n",
       "| mesoderm primordium                  |    1 |\n",
       "| midgut                               | 1830 |\n",
       "| midgut primordium                    |   23 |\n",
       "| midline primordium                   |   20 |\n",
       "| muscle                               | 1896 |\n",
       "| peripheral nervous system            |   92 |\n",
       "| peripheral nervous system primordium |    5 |\n",
       "| plasmatocytes                        |  564 |\n",
       "| plasmatocytes anlage                 |   31 |\n",
       "| procrystal cell                      |  230 |\n",
       "| proventriculus                       |  214 |\n",
       "| salivary gland                       |   85 |\n",
       "| salivary gland body primordium       |   26 |\n",
       "| salivary gland primordium            |    2 |\n",
       "| sensory nervous system               |  544 |\n",
       "| sensory nervous system primordium    |   31 |\n",
       "| somatic muscle                       | 1882 |\n",
       "| somatic muscle primordium            |  106 |\n",
       "| spiracle (tracheal system)           |   72 |\n",
       "| spiracles (tracheal system)          |   76 |\n",
       "| tracheal primordium                  |   96 |\n",
       "| tracheal system                      |  916 |\n",
       "| trunk mesoderm primordium            |   40 |\n",
       "| ubiquitous                           | 6029 |\n",
       "| unknown                              |   40 |\n",
       "| ventral nerve cord                   |  659 |\n",
       "| ventral nerve cord primordium        |   12 |\n",
       "| visceral muscle                      |  153 |\n",
       "| visceral muscle primordium           |  317 |\n",
       "| yolk nuclei                          |  123 |\n",
       "\n"
      ],
      "text/plain": [
       "   annot                                n   \n",
       "1  Malpighian tubule                     115\n",
       "2  amnioserosa                           290\n",
       "3  brain                                6993\n",
       "4  brain primordium                      157\n",
       "5  cellular blastoderm / pole cell         2\n",
       "6  crystal cell                           14\n",
       "7  ectoderm primordium                    25\n",
       "8  endoderm primordium                     5\n",
       "9  epidermis                            4941\n",
       "10 epidermis primordium                  112\n",
       "11 fat body                              623\n",
       "12 foregut / hindgut / anal pad         1182\n",
       "13 foregut / hindgut primordium           10\n",
       "14 germ cell                              82\n",
       "15 glia                                   53\n",
       "16 hindgut / anal pad                    193\n",
       "17 hindgut primordium                     28\n",
       "18 mesoderm primordium                     1\n",
       "19 midgut                               1830\n",
       "20 midgut primordium                      23\n",
       "21 midline primordium                     20\n",
       "22 muscle                               1896\n",
       "23 peripheral nervous system              92\n",
       "24 peripheral nervous system primordium    5\n",
       "25 plasmatocytes                         564\n",
       "26 plasmatocytes anlage                   31\n",
       "27 procrystal cell                       230\n",
       "28 proventriculus                        214\n",
       "29 salivary gland                         85\n",
       "30 salivary gland body primordium         26\n",
       "31 salivary gland primordium               2\n",
       "32 sensory nervous system                544\n",
       "33 sensory nervous system primordium      31\n",
       "34 somatic muscle                       1882\n",
       "35 somatic muscle primordium             106\n",
       "36 spiracle (tracheal system)             72\n",
       "37 spiracles (tracheal system)            76\n",
       "38 tracheal primordium                    96\n",
       "39 tracheal system                       916\n",
       "40 trunk mesoderm primordium              40\n",
       "41 ubiquitous                           6029\n",
       "42 unknown                                40\n",
       "43 ventral nerve cord                    659\n",
       "44 ventral nerve cord primordium          12\n",
       "45 visceral muscle                       153\n",
       "46 visceral muscle primordium            317\n",
       "47 yolk nuclei                           123"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub300  %>% filter(tmp.expr > 0 & tmp.expr < 2 &\n",
    "                               !(cluster %in% c(34, 40, 39, 9, 18, 106, 46, 18)))  %>% count(annot)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### try Muc11A"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1456,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sub_no_multitons$tmp.expr.Muc11A = counts(cds_sub_no_multitons)[\"Muc11A\", ]/df_sub_no_multitons$Size_Factor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1457,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 34</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>orig.ident</th><th scope=col>nCount_RNA</th><th scope=col>nFeature_RNA</th><th scope=col>rt_bc</th><th scope=col>cell</th><th scope=col>time</th><th scope=col>percent.mt</th><th scope=col>percent.ribo</th><th scope=col>barn_dro</th><th scope=col>barn_hu</th><th scope=col>barn_mus</th><th scope=col>RNA_snn_res.0.8</th><th scope=col>doublet_score</th><th scope=col>predicted_doublet</th><th scope=col>doublet_cluster</th><th scope=col>seurat_clusters</th><th scope=col>doublet_subcluster</th><th scope=col>RNA_snn_res.0.08</th><th scope=col>lasso_age</th><th scope=col>NNv1_age</th><th scope=col>NNv2_age</th><th scope=col>lasso_shift</th><th scope=col>NNv1_shift</th><th scope=col>NNv2_shift</th><th scope=col>annot</th><th scope=col>ident</th><th scope=col>inferred_time_window</th><th scope=col>cell.type_time.bin</th><th scope=col>Size_Factor</th><th scope=col>cluster</th><th scope=col>umap_1</th><th scope=col>umap_2</th><th scope=col>tmp.expr</th><th scope=col>tmp.expr.Muc11A</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG</th><td>exp1</td><td>292</td><td>202</td><td>AAGTAGTCAG</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG</td><td>hrs_10_14</td><td>0.3424658</td><td> 0.6849315</td><td>320</td><td>3</td><td>0</td><td>23</td><td>0.06585398</td><td>Singlet</td><td>FALSE</td><td>12</td><td>FALSE</td><td>12</td><td>12.544636</td><td>12.211529</td><td>13.83098</td><td> 0.00000000</td><td>0.00000000</td><td>0.0000000</td><td>epidermis             </td><td>12</td><td>12-14</td><td>epidermis:12-14             </td><td>0.6967951</td><td>165</td><td>-1.3580737</td><td> 0.3582040</td><td>0.000000</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA</th><td>exp1</td><td>354</td><td>272</td><td>TCGATATTGA</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA</td><td>hrs_14_18</td><td>0.5649718</td><td> 1.6949153</td><td>380</td><td>0</td><td>1</td><td>8 </td><td>0.08257165</td><td>Singlet</td><td>FALSE</td><td>11</td><td>FALSE</td><td>11</td><td>13.458162</td><td>15.126904</td><td>14.55855</td><td>-0.54183787</td><td>0.00000000</td><td>0.0000000</td><td>sensory nervous system</td><td>11</td><td>14-16</td><td>sensory nervous system:14-16</td><td>0.8306988</td><td>148</td><td>-0.6711318</td><td>-2.9462104</td><td>0.000000</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA</th><td>exp1</td><td>287</td><td>221</td><td>TGCGCCAGAA</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA</td><td>hrs_08_12</td><td>0.0000000</td><td> 3.4843206</td><td>307</td><td>0</td><td>0</td><td>6 </td><td>0.06023494</td><td>Singlet</td><td>FALSE</td><td>9 </td><td>FALSE</td><td>9 </td><td> 9.715162</td><td> 9.000631</td><td>11.35599</td><td> 0.00000000</td><td>0.00000000</td><td>0.0000000</td><td>midline primordium    </td><td>9 </td><td>8-10 </td><td>midline primordium:8-10     </td><td>0.6893560</td><td>120</td><td>-7.5437593</td><td>-0.6611074</td><td>0.000000</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA</th><td>exp1</td><td>509</td><td>376</td><td>ACGCCGTTCA</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA</td><td>hrs_08_12</td><td>0.5893910</td><td> 5.5009823</td><td>552</td><td>2</td><td>1</td><td>0 </td><td>0.08149588</td><td>Singlet</td><td>FALSE</td><td>21</td><td>FALSE</td><td>21</td><td>12.520638</td><td>11.694070</td><td>13.23313</td><td> 0.52063800</td><td>0.00000000</td><td>1.2331295</td><td>plasmatocytes         </td><td>21</td><td>10-12</td><td>plasmatocytes:10-12         </td><td>1.2076129</td><td>89 </td><td>-1.6608665</td><td> 1.2526858</td><td>0.000000</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT</th><td>exp1</td><td>438</td><td>282</td><td>GTTGAAGGAT</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT</td><td>hrs_06_10</td><td>0.6849315</td><td> 0.4566210</td><td>492</td><td>1</td><td>1</td><td>13</td><td>0.04439834</td><td>Singlet</td><td>FALSE</td><td>9 </td><td>FALSE</td><td>9 </td><td>10.045072</td><td>10.022837</td><td>10.51352</td><td> 0.04507167</td><td>0.02283669</td><td>0.5135212</td><td>brain                 </td><td>9 </td><td>10-12</td><td>brain:10-12                 </td><td>1.0017980</td><td>88 </td><td>-2.4943113</td><td>-5.3918748</td><td>0.000000</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG</th><td>exp1</td><td>313</td><td>227</td><td>TGATAGAACG</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG</td><td>hrs_14_18</td><td>0.6389776</td><td>13.4185304</td><td>344</td><td>2</td><td>0</td><td>12</td><td>0.07449210</td><td>Singlet</td><td>FALSE</td><td>11</td><td>FALSE</td><td>11</td><td>10.338754</td><td>15.219664</td><td>14.57722</td><td>-3.66124596</td><td>0.00000000</td><td>0.0000000</td><td>ubiquitous            </td><td>11</td><td>14-16</td><td>ubiquitous:14-16            </td><td>0.7637470</td><td>44 </td><td> 1.8231004</td><td> 3.9306800</td><td>1.309334</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 34\n",
       "\\begin{tabular}{r|llllllllllllllllllllllllllllllllll}\n",
       "  & orig.ident & nCount\\_RNA & nFeature\\_RNA & rt\\_bc & cell & time & percent.mt & percent.ribo & barn\\_dro & barn\\_hu & barn\\_mus & RNA\\_snn\\_res.0.8 & doublet\\_score & predicted\\_doublet & doublet\\_cluster & seurat\\_clusters & doublet\\_subcluster & RNA\\_snn\\_res.0.08 & lasso\\_age & NNv1\\_age & NNv2\\_age & lasso\\_shift & NNv1\\_shift & NNv2\\_shift & annot & ident & inferred\\_time\\_window & cell.type\\_time.bin & Size\\_Factor & cluster & umap\\_1 & umap\\_2 & tmp.expr & tmp.expr.Muc11A\\\\\n",
       "  & <chr> & <dbl> & <int> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <int> & <int> & <int> & <chr> & <dbl> & <chr> & <lgl> & <fct> & <lgl> & <fct> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <fct> & <chr> & <chr> & <dbl> & <fct> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG & exp1 & 292 & 202 & AAGTAGTCAG & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG & hrs\\_10\\_14 & 0.3424658 &  0.6849315 & 320 & 3 & 0 & 23 & 0.06585398 & Singlet & FALSE & 12 & FALSE & 12 & 12.544636 & 12.211529 & 13.83098 &  0.00000000 & 0.00000000 & 0.0000000 & epidermis              & 12 & 12-14 & epidermis:12-14              & 0.6967951 & 165 & -1.3580737 &  0.3582040 & 0.000000 & 0\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA & exp1 & 354 & 272 & TCGATATTGA & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA & hrs\\_14\\_18 & 0.5649718 &  1.6949153 & 380 & 0 & 1 & 8  & 0.08257165 & Singlet & FALSE & 11 & FALSE & 11 & 13.458162 & 15.126904 & 14.55855 & -0.54183787 & 0.00000000 & 0.0000000 & sensory nervous system & 11 & 14-16 & sensory nervous system:14-16 & 0.8306988 & 148 & -0.6711318 & -2.9462104 & 0.000000 & 0\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA & exp1 & 287 & 221 & TGCGCCAGAA & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA & hrs\\_08\\_12 & 0.0000000 &  3.4843206 & 307 & 0 & 0 & 6  & 0.06023494 & Singlet & FALSE & 9  & FALSE & 9  &  9.715162 &  9.000631 & 11.35599 &  0.00000000 & 0.00000000 & 0.0000000 & midline primordium     & 9  & 8-10  & midline primordium:8-10      & 0.6893560 & 120 & -7.5437593 & -0.6611074 & 0.000000 & 0\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA & exp1 & 509 & 376 & ACGCCGTTCA & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA & hrs\\_08\\_12 & 0.5893910 &  5.5009823 & 552 & 2 & 1 & 0  & 0.08149588 & Singlet & FALSE & 21 & FALSE & 21 & 12.520638 & 11.694070 & 13.23313 &  0.52063800 & 0.00000000 & 1.2331295 & plasmatocytes          & 21 & 10-12 & plasmatocytes:10-12          & 1.2076129 & 89  & -1.6608665 &  1.2526858 & 0.000000 & 0\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT & exp1 & 438 & 282 & GTTGAAGGAT & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT & hrs\\_06\\_10 & 0.6849315 &  0.4566210 & 492 & 1 & 1 & 13 & 0.04439834 & Singlet & FALSE & 9  & FALSE & 9  & 10.045072 & 10.022837 & 10.51352 &  0.04507167 & 0.02283669 & 0.5135212 & brain                  & 9  & 10-12 & brain:10-12                  & 1.0017980 & 88  & -2.4943113 & -5.3918748 & 0.000000 & 0\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG & exp1 & 313 & 227 & TGATAGAACG & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG & hrs\\_14\\_18 & 0.6389776 & 13.4185304 & 344 & 2 & 0 & 12 & 0.07449210 & Singlet & FALSE & 11 & FALSE & 11 & 10.338754 & 15.219664 & 14.57722 & -3.66124596 & 0.00000000 & 0.0000000 & ubiquitous             & 11 & 14-16 & ubiquitous:14-16             & 0.7637470 & 44  &  1.8231004 &  3.9306800 & 1.309334 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 34\n",
       "\n",
       "| <!--/--> | orig.ident &lt;chr&gt; | nCount_RNA &lt;dbl&gt; | nFeature_RNA &lt;int&gt; | rt_bc &lt;chr&gt; | cell &lt;chr&gt; | time &lt;chr&gt; | percent.mt &lt;dbl&gt; | percent.ribo &lt;dbl&gt; | barn_dro &lt;int&gt; | barn_hu &lt;int&gt; | barn_mus &lt;int&gt; | RNA_snn_res.0.8 &lt;chr&gt; | doublet_score &lt;dbl&gt; | predicted_doublet &lt;chr&gt; | doublet_cluster &lt;lgl&gt; | seurat_clusters &lt;fct&gt; | doublet_subcluster &lt;lgl&gt; | RNA_snn_res.0.08 &lt;fct&gt; | lasso_age &lt;dbl&gt; | NNv1_age &lt;dbl&gt; | NNv2_age &lt;dbl&gt; | lasso_shift &lt;dbl&gt; | NNv1_shift &lt;dbl&gt; | NNv2_shift &lt;dbl&gt; | annot &lt;chr&gt; | ident &lt;fct&gt; | inferred_time_window &lt;chr&gt; | cell.type_time.bin &lt;chr&gt; | Size_Factor &lt;dbl&gt; | cluster &lt;fct&gt; | umap_1 &lt;dbl&gt; | umap_2 &lt;dbl&gt; | tmp.expr &lt;dbl&gt; | tmp.expr.Muc11A &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG | exp1 | 292 | 202 | AAGTAGTCAG | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG | hrs_10_14 | 0.3424658 |  0.6849315 | 320 | 3 | 0 | 23 | 0.06585398 | Singlet | FALSE | 12 | FALSE | 12 | 12.544636 | 12.211529 | 13.83098 |  0.00000000 | 0.00000000 | 0.0000000 | epidermis              | 12 | 12-14 | epidermis:12-14              | 0.6967951 | 165 | -1.3580737 |  0.3582040 | 0.000000 | 0 |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA | exp1 | 354 | 272 | TCGATATTGA | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA | hrs_14_18 | 0.5649718 |  1.6949153 | 380 | 0 | 1 | 8  | 0.08257165 | Singlet | FALSE | 11 | FALSE | 11 | 13.458162 | 15.126904 | 14.55855 | -0.54183787 | 0.00000000 | 0.0000000 | sensory nervous system | 11 | 14-16 | sensory nervous system:14-16 | 0.8306988 | 148 | -0.6711318 | -2.9462104 | 0.000000 | 0 |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA | exp1 | 287 | 221 | TGCGCCAGAA | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA | hrs_08_12 | 0.0000000 |  3.4843206 | 307 | 0 | 0 | 6  | 0.06023494 | Singlet | FALSE | 9  | FALSE | 9  |  9.715162 |  9.000631 | 11.35599 |  0.00000000 | 0.00000000 | 0.0000000 | midline primordium     | 9  | 8-10  | midline primordium:8-10      | 0.6893560 | 120 | -7.5437593 | -0.6611074 | 0.000000 | 0 |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA | exp1 | 509 | 376 | ACGCCGTTCA | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA | hrs_08_12 | 0.5893910 |  5.5009823 | 552 | 2 | 1 | 0  | 0.08149588 | Singlet | FALSE | 21 | FALSE | 21 | 12.520638 | 11.694070 | 13.23313 |  0.52063800 | 0.00000000 | 1.2331295 | plasmatocytes          | 21 | 10-12 | plasmatocytes:10-12          | 1.2076129 | 89  | -1.6608665 |  1.2526858 | 0.000000 | 0 |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT | exp1 | 438 | 282 | GTTGAAGGAT | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT | hrs_06_10 | 0.6849315 |  0.4566210 | 492 | 1 | 1 | 13 | 0.04439834 | Singlet | FALSE | 9  | FALSE | 9  | 10.045072 | 10.022837 | 10.51352 |  0.04507167 | 0.02283669 | 0.5135212 | brain                  | 9  | 10-12 | brain:10-12                  | 1.0017980 | 88  | -2.4943113 | -5.3918748 | 0.000000 | 0 |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG | exp1 | 313 | 227 | TGATAGAACG | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG | hrs_14_18 | 0.6389776 | 13.4185304 | 344 | 2 | 0 | 12 | 0.07449210 | Singlet | FALSE | 11 | FALSE | 11 | 10.338754 | 15.219664 | 14.57722 | -3.66124596 | 0.00000000 | 0.0000000 | ubiquitous             | 11 | 14-16 | ubiquitous:14-16             | 0.7637470 | 44  |  1.8231004 |  3.9306800 | 1.309334 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "                                                       orig.ident nCount_RNA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG exp1       292       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA exp1       354       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA exp1       287       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA exp1       509       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT exp1       438       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG exp1       313       \n",
       "                                                       nFeature_RNA rt_bc     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 202          AAGTAGTCAG\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 272          TCGATATTGA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 221          TGCGCCAGAA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 376          ACGCCGTTCA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 282          GTTGAAGGAT\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 227          TGATAGAACG\n",
       "                                                       cell                                                  \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG\n",
       "                                                       time      percent.mt\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG hrs_10_14 0.3424658 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA hrs_14_18 0.5649718 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA hrs_08_12 0.0000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA hrs_08_12 0.5893910 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT hrs_06_10 0.6849315 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG hrs_14_18 0.6389776 \n",
       "                                                       percent.ribo barn_dro\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG  0.6849315   320     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA  1.6949153   380     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA  3.4843206   307     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA  5.5009823   552     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT  0.4566210   492     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 13.4185304   344     \n",
       "                                                       barn_hu barn_mus\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 3       0       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 0       1       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 0       0       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 2       1       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 1       1       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 2       0       \n",
       "                                                       RNA_snn_res.0.8\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 23             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 8              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 6              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 0              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 13             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 12             \n",
       "                                                       doublet_score\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 0.06585398   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 0.08257165   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 0.06023494   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 0.08149588   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 0.04439834   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 0.07449210   \n",
       "                                                       predicted_doublet\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG Singlet          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA Singlet          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA Singlet          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA Singlet          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT Singlet          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG Singlet          \n",
       "                                                       doublet_cluster\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG FALSE          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA FALSE          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA FALSE          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA FALSE          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT FALSE          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG FALSE          \n",
       "                                                       seurat_clusters\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 11             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 9              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 21             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 9              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 11             \n",
       "                                                       doublet_subcluster\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG FALSE             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA FALSE             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA FALSE             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA FALSE             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT FALSE             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG FALSE             \n",
       "                                                       RNA_snn_res.0.08\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 11              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 9               \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 21              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 9               \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 11              \n",
       "                                                       lasso_age NNv1_age \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12.544636 12.211529\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 13.458162 15.126904\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA  9.715162  9.000631\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 12.520638 11.694070\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 10.045072 10.022837\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 10.338754 15.219664\n",
       "                                                       NNv2_age lasso_shift\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 13.83098  0.00000000\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 14.55855 -0.54183787\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 11.35599  0.00000000\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 13.23313  0.52063800\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 10.51352  0.04507167\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 14.57722 -3.66124596\n",
       "                                                       NNv1_shift NNv2_shift\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 0.00000000 0.0000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 0.00000000 0.0000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 0.00000000 0.0000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 0.00000000 1.2331295 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 0.02283669 0.5135212 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 0.00000000 0.0000000 \n",
       "                                                       annot                 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG epidermis             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA sensory nervous system\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA midline primordium    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA plasmatocytes         \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT brain                 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG ubiquitous            \n",
       "                                                       ident\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 11   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 9    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 21   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 9    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 11   \n",
       "                                                       inferred_time_window\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12-14               \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 14-16               \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 8-10                \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 10-12               \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 10-12               \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 14-16               \n",
       "                                                       cell.type_time.bin          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG epidermis:12-14             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA sensory nervous system:14-16\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA midline primordium:8-10     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA plasmatocytes:10-12         \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT brain:10-12                 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG ubiquitous:14-16            \n",
       "                                                       Size_Factor cluster\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 0.6967951   165    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 0.8306988   148    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 0.6893560   120    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 1.2076129   89     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 1.0017980   88     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 0.7637470   44     \n",
       "                                                       umap_1     umap_2    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG -1.3580737  0.3582040\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA -0.6711318 -2.9462104\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA -7.5437593 -0.6611074\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA -1.6608665  1.2526858\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT -2.4943113 -5.3918748\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG  1.8231004  3.9306800\n",
       "                                                       tmp.expr tmp.expr.Muc11A\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 0.000000 0              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 0.000000 0              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 0.000000 0              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTACGCCGTTCA 0.000000 0              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTGTTGAAGGAT 0.000000 0              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CGAGATGCTTGATAGAACG 1.309334 0              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub_no_multitons  %>% head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1458,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>186</li><li>34</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 186\n",
       "\\item 34\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 186\n",
       "2. 34\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 186  34"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub_no_multitons  %>% filter(tmp.expr.Muc11A > 0)  %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1459,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>annot</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Malpighian tubule           </td><td>75</td></tr>\n",
       "\t<tr><td>amnioserosa                 </td><td> 1</td></tr>\n",
       "\t<tr><td>brain                       </td><td>25</td></tr>\n",
       "\t<tr><td>brain primordium            </td><td> 1</td></tr>\n",
       "\t<tr><td>ectoderm primordium         </td><td> 1</td></tr>\n",
       "\t<tr><td>epidermis                   </td><td> 1</td></tr>\n",
       "\t<tr><td>foregut / hindgut / anal pad</td><td> 1</td></tr>\n",
       "\t<tr><td>hindgut / anal pad          </td><td> 1</td></tr>\n",
       "\t<tr><td>midgut                      </td><td>41</td></tr>\n",
       "\t<tr><td>muscle                      </td><td> 3</td></tr>\n",
       "\t<tr><td>peripheral nervous system   </td><td> 1</td></tr>\n",
       "\t<tr><td>procrystal cell             </td><td> 1</td></tr>\n",
       "\t<tr><td>salivary gland              </td><td> 1</td></tr>\n",
       "\t<tr><td>sensory nervous system      </td><td> 1</td></tr>\n",
       "\t<tr><td>somatic muscle              </td><td> 1</td></tr>\n",
       "\t<tr><td>somatic muscle primordium   </td><td> 1</td></tr>\n",
       "\t<tr><td>tracheal primordium         </td><td> 2</td></tr>\n",
       "\t<tr><td>ubiquitous                  </td><td>21</td></tr>\n",
       "\t<tr><td>unknown                     </td><td> 2</td></tr>\n",
       "\t<tr><td>ventral nerve cord          </td><td> 5</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 2\n",
       "\\begin{tabular}{ll}\n",
       " annot & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Malpighian tubule            & 75\\\\\n",
       "\t amnioserosa                  &  1\\\\\n",
       "\t brain                        & 25\\\\\n",
       "\t brain primordium             &  1\\\\\n",
       "\t ectoderm primordium          &  1\\\\\n",
       "\t epidermis                    &  1\\\\\n",
       "\t foregut / hindgut / anal pad &  1\\\\\n",
       "\t hindgut / anal pad           &  1\\\\\n",
       "\t midgut                       & 41\\\\\n",
       "\t muscle                       &  3\\\\\n",
       "\t peripheral nervous system    &  1\\\\\n",
       "\t procrystal cell              &  1\\\\\n",
       "\t salivary gland               &  1\\\\\n",
       "\t sensory nervous system       &  1\\\\\n",
       "\t somatic muscle               &  1\\\\\n",
       "\t somatic muscle primordium    &  1\\\\\n",
       "\t tracheal primordium          &  2\\\\\n",
       "\t ubiquitous                   & 21\\\\\n",
       "\t unknown                      &  2\\\\\n",
       "\t ventral nerve cord           &  5\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 2\n",
       "\n",
       "| annot &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Malpighian tubule            | 75 |\n",
       "| amnioserosa                  |  1 |\n",
       "| brain                        | 25 |\n",
       "| brain primordium             |  1 |\n",
       "| ectoderm primordium          |  1 |\n",
       "| epidermis                    |  1 |\n",
       "| foregut / hindgut / anal pad |  1 |\n",
       "| hindgut / anal pad           |  1 |\n",
       "| midgut                       | 41 |\n",
       "| muscle                       |  3 |\n",
       "| peripheral nervous system    |  1 |\n",
       "| procrystal cell              |  1 |\n",
       "| salivary gland               |  1 |\n",
       "| sensory nervous system       |  1 |\n",
       "| somatic muscle               |  1 |\n",
       "| somatic muscle primordium    |  1 |\n",
       "| tracheal primordium          |  2 |\n",
       "| ubiquitous                   | 21 |\n",
       "| unknown                      |  2 |\n",
       "| ventral nerve cord           |  5 |\n",
       "\n"
      ],
      "text/plain": [
       "   annot                        n \n",
       "1  Malpighian tubule            75\n",
       "2  amnioserosa                   1\n",
       "3  brain                        25\n",
       "4  brain primordium              1\n",
       "5  ectoderm primordium           1\n",
       "6  epidermis                     1\n",
       "7  foregut / hindgut / anal pad  1\n",
       "8  hindgut / anal pad            1\n",
       "9  midgut                       41\n",
       "10 muscle                        3\n",
       "11 peripheral nervous system     1\n",
       "12 procrystal cell               1\n",
       "13 salivary gland                1\n",
       "14 sensory nervous system        1\n",
       "15 somatic muscle                1\n",
       "16 somatic muscle primordium     1\n",
       "17 tracheal primordium           2\n",
       "18 ubiquitous                   21\n",
       "19 unknown                       2\n",
       "20 ventral nerve cord            5"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub_no_multitons  %>% filter(tmp.expr.Muc11A > 0)  %>% count(annot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1461,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 13 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>annot</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Malpighian tubule           </td><td>14</td></tr>\n",
       "\t<tr><td>brain                       </td><td>22</td></tr>\n",
       "\t<tr><td>ectoderm primordium         </td><td> 1</td></tr>\n",
       "\t<tr><td>foregut / hindgut / anal pad</td><td> 1</td></tr>\n",
       "\t<tr><td>hindgut / anal pad          </td><td> 1</td></tr>\n",
       "\t<tr><td>midgut                      </td><td>23</td></tr>\n",
       "\t<tr><td>muscle                      </td><td> 3</td></tr>\n",
       "\t<tr><td>procrystal cell             </td><td> 1</td></tr>\n",
       "\t<tr><td>salivary gland              </td><td> 1</td></tr>\n",
       "\t<tr><td>somatic muscle              </td><td> 1</td></tr>\n",
       "\t<tr><td>tracheal primordium         </td><td> 1</td></tr>\n",
       "\t<tr><td>ubiquitous                  </td><td> 8</td></tr>\n",
       "\t<tr><td>ventral nerve cord          </td><td> 1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 13 × 2\n",
       "\\begin{tabular}{ll}\n",
       " annot & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Malpighian tubule            & 14\\\\\n",
       "\t brain                        & 22\\\\\n",
       "\t ectoderm primordium          &  1\\\\\n",
       "\t foregut / hindgut / anal pad &  1\\\\\n",
       "\t hindgut / anal pad           &  1\\\\\n",
       "\t midgut                       & 23\\\\\n",
       "\t muscle                       &  3\\\\\n",
       "\t procrystal cell              &  1\\\\\n",
       "\t salivary gland               &  1\\\\\n",
       "\t somatic muscle               &  1\\\\\n",
       "\t tracheal primordium          &  1\\\\\n",
       "\t ubiquitous                   &  8\\\\\n",
       "\t ventral nerve cord           &  1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 13 × 2\n",
       "\n",
       "| annot &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Malpighian tubule            | 14 |\n",
       "| brain                        | 22 |\n",
       "| ectoderm primordium          |  1 |\n",
       "| foregut / hindgut / anal pad |  1 |\n",
       "| hindgut / anal pad           |  1 |\n",
       "| midgut                       | 23 |\n",
       "| muscle                       |  3 |\n",
       "| procrystal cell              |  1 |\n",
       "| salivary gland               |  1 |\n",
       "| somatic muscle               |  1 |\n",
       "| tracheal primordium          |  1 |\n",
       "| ubiquitous                   |  8 |\n",
       "| ventral nerve cord           |  1 |\n",
       "\n"
      ],
      "text/plain": [
       "   annot                        n \n",
       "1  Malpighian tubule            14\n",
       "2  brain                        22\n",
       "3  ectoderm primordium           1\n",
       "4  foregut / hindgut / anal pad  1\n",
       "5  hindgut / anal pad            1\n",
       "6  midgut                       23\n",
       "7  muscle                        3\n",
       "8  procrystal cell               1\n",
       "9  salivary gland                1\n",
       "10 somatic muscle                1\n",
       "11 tracheal primordium           1\n",
       "12 ubiquitous                    8\n",
       "13 ventral nerve cord            1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub_no_multitons  %>% filter(tmp.expr.Muc11A > 0 & tmp.expr.Muc11A < 1)  %>% count(annot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1462,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 41 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cluster</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>163</td><td>56</td></tr>\n",
       "\t<tr><td>103</td><td> 9</td></tr>\n",
       "\t<tr><td>50 </td><td> 8</td></tr>\n",
       "\t<tr><td>62 </td><td> 5</td></tr>\n",
       "\t<tr><td>71 </td><td> 5</td></tr>\n",
       "\t<tr><td>32 </td><td> 3</td></tr>\n",
       "\t<tr><td>87 </td><td> 3</td></tr>\n",
       "\t<tr><td>3  </td><td> 2</td></tr>\n",
       "\t<tr><td>7  </td><td> 2</td></tr>\n",
       "\t<tr><td>11 </td><td> 2</td></tr>\n",
       "\t<tr><td>19 </td><td> 2</td></tr>\n",
       "\t<tr><td>23 </td><td> 2</td></tr>\n",
       "\t<tr><td>44 </td><td> 2</td></tr>\n",
       "\t<tr><td>116</td><td> 2</td></tr>\n",
       "\t<tr><td>118</td><td> 2</td></tr>\n",
       "\t<tr><td>131</td><td> 2</td></tr>\n",
       "\t<tr><td>1  </td><td> 1</td></tr>\n",
       "\t<tr><td>4  </td><td> 1</td></tr>\n",
       "\t<tr><td>10 </td><td> 1</td></tr>\n",
       "\t<tr><td>20 </td><td> 1</td></tr>\n",
       "\t<tr><td>27 </td><td> 1</td></tr>\n",
       "\t<tr><td>33 </td><td> 1</td></tr>\n",
       "\t<tr><td>34 </td><td> 1</td></tr>\n",
       "\t<tr><td>38 </td><td> 1</td></tr>\n",
       "\t<tr><td>43 </td><td> 1</td></tr>\n",
       "\t<tr><td>49 </td><td> 1</td></tr>\n",
       "\t<tr><td>53 </td><td> 1</td></tr>\n",
       "\t<tr><td>75 </td><td> 1</td></tr>\n",
       "\t<tr><td>76 </td><td> 1</td></tr>\n",
       "\t<tr><td>77 </td><td> 1</td></tr>\n",
       "\t<tr><td>80 </td><td> 1</td></tr>\n",
       "\t<tr><td>88 </td><td> 1</td></tr>\n",
       "\t<tr><td>96 </td><td> 1</td></tr>\n",
       "\t<tr><td>97 </td><td> 1</td></tr>\n",
       "\t<tr><td>98 </td><td> 1</td></tr>\n",
       "\t<tr><td>114</td><td> 1</td></tr>\n",
       "\t<tr><td>121</td><td> 1</td></tr>\n",
       "\t<tr><td>132</td><td> 1</td></tr>\n",
       "\t<tr><td>145</td><td> 1</td></tr>\n",
       "\t<tr><td>148</td><td> 1</td></tr>\n",
       "\t<tr><td>150</td><td> 1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 41 × 2\n",
       "\\begin{tabular}{ll}\n",
       " cluster & n\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t 163 & 56\\\\\n",
       "\t 103 &  9\\\\\n",
       "\t 50  &  8\\\\\n",
       "\t 62  &  5\\\\\n",
       "\t 71  &  5\\\\\n",
       "\t 32  &  3\\\\\n",
       "\t 87  &  3\\\\\n",
       "\t 3   &  2\\\\\n",
       "\t 7   &  2\\\\\n",
       "\t 11  &  2\\\\\n",
       "\t 19  &  2\\\\\n",
       "\t 23  &  2\\\\\n",
       "\t 44  &  2\\\\\n",
       "\t 116 &  2\\\\\n",
       "\t 118 &  2\\\\\n",
       "\t 131 &  2\\\\\n",
       "\t 1   &  1\\\\\n",
       "\t 4   &  1\\\\\n",
       "\t 10  &  1\\\\\n",
       "\t 20  &  1\\\\\n",
       "\t 27  &  1\\\\\n",
       "\t 33  &  1\\\\\n",
       "\t 34  &  1\\\\\n",
       "\t 38  &  1\\\\\n",
       "\t 43  &  1\\\\\n",
       "\t 49  &  1\\\\\n",
       "\t 53  &  1\\\\\n",
       "\t 75  &  1\\\\\n",
       "\t 76  &  1\\\\\n",
       "\t 77  &  1\\\\\n",
       "\t 80  &  1\\\\\n",
       "\t 88  &  1\\\\\n",
       "\t 96  &  1\\\\\n",
       "\t 97  &  1\\\\\n",
       "\t 98  &  1\\\\\n",
       "\t 114 &  1\\\\\n",
       "\t 121 &  1\\\\\n",
       "\t 132 &  1\\\\\n",
       "\t 145 &  1\\\\\n",
       "\t 148 &  1\\\\\n",
       "\t 150 &  1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 41 × 2\n",
       "\n",
       "| cluster &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| 163 | 56 |\n",
       "| 103 |  9 |\n",
       "| 50  |  8 |\n",
       "| 62  |  5 |\n",
       "| 71  |  5 |\n",
       "| 32  |  3 |\n",
       "| 87  |  3 |\n",
       "| 3   |  2 |\n",
       "| 7   |  2 |\n",
       "| 11  |  2 |\n",
       "| 19  |  2 |\n",
       "| 23  |  2 |\n",
       "| 44  |  2 |\n",
       "| 116 |  2 |\n",
       "| 118 |  2 |\n",
       "| 131 |  2 |\n",
       "| 1   |  1 |\n",
       "| 4   |  1 |\n",
       "| 10  |  1 |\n",
       "| 20  |  1 |\n",
       "| 27  |  1 |\n",
       "| 33  |  1 |\n",
       "| 34  |  1 |\n",
       "| 38  |  1 |\n",
       "| 43  |  1 |\n",
       "| 49  |  1 |\n",
       "| 53  |  1 |\n",
       "| 75  |  1 |\n",
       "| 76  |  1 |\n",
       "| 77  |  1 |\n",
       "| 80  |  1 |\n",
       "| 88  |  1 |\n",
       "| 96  |  1 |\n",
       "| 97  |  1 |\n",
       "| 98  |  1 |\n",
       "| 114 |  1 |\n",
       "| 121 |  1 |\n",
       "| 132 |  1 |\n",
       "| 145 |  1 |\n",
       "| 148 |  1 |\n",
       "| 150 |  1 |\n",
       "\n"
      ],
      "text/plain": [
       "   cluster n \n",
       "1  163     56\n",
       "2  103      9\n",
       "3  50       8\n",
       "4  62       5\n",
       "5  71       5\n",
       "6  32       3\n",
       "7  87       3\n",
       "8  3        2\n",
       "9  7        2\n",
       "10 11       2\n",
       "11 19       2\n",
       "12 23       2\n",
       "13 44       2\n",
       "14 116      2\n",
       "15 118      2\n",
       "16 131      2\n",
       "17 1        1\n",
       "18 4        1\n",
       "19 10       1\n",
       "20 20       1\n",
       "21 27       1\n",
       "22 33       1\n",
       "23 34       1\n",
       "24 38       1\n",
       "25 43       1\n",
       "26 49       1\n",
       "27 53       1\n",
       "28 75       1\n",
       "29 76       1\n",
       "30 77       1\n",
       "31 80       1\n",
       "32 88       1\n",
       "33 96       1\n",
       "34 97       1\n",
       "35 98       1\n",
       "36 114      1\n",
       "37 121      1\n",
       "38 132      1\n",
       "39 145      1\n",
       "40 148      1\n",
       "41 150      1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub_no_multitons  %>% filter(tmp.expr.Muc11A > 0 & tmp.expr.Muc11A < 2)  %>% count(cluster)  %>% arrange(-n)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1466,
   "metadata": {},
   "outputs": [],
   "source": [
    "Muc11A_lt1  = (df_sub_no_multitons  %>% filter(tmp.expr.Muc11A > 0 & tmp.expr.Muc11A < 1))$cell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% Muc11A_lt1), size = ifelse(df_sub300$cell %in% Muc11A_lt2, 2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1468,
   "metadata": {},
   "outputs": [],
   "source": [
    "Muc11A_gte1  = (df_sub_no_multitons  %>% filter( tmp.expr.Muc11A >= 1))$cell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% Muc11A_gte1), size = ifelse(df_sub300$cell %in% Muc11A_gte1, 2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1478,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>annot</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Malpighian tubule  </td><td>14</td></tr>\n",
       "\t<tr><td>midgut             </td><td>16</td></tr>\n",
       "\t<tr><td>tracheal primordium</td><td> 1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3 × 2\n",
       "\\begin{tabular}{ll}\n",
       " annot & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Malpighian tubule   & 14\\\\\n",
       "\t midgut              & 16\\\\\n",
       "\t tracheal primordium &  1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3 × 2\n",
       "\n",
       "| annot &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Malpighian tubule   | 14 |\n",
       "| midgut              | 16 |\n",
       "| tracheal primordium |  1 |\n",
       "\n"
      ],
      "text/plain": [
       "  annot               n \n",
       "1 Malpighian tubule   14\n",
       "2 midgut              16\n",
       "3 tracheal primordium  1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub_no_multitons  %>% filter(tmp.expr.Muc11A > 0 & tmp.expr.Muc11A < 1 &\n",
    "                               cluster %in% c(103, 163))  %>% count(annot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1480,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 11 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>annot</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>brain                       </td><td>22</td></tr>\n",
       "\t<tr><td>ectoderm primordium         </td><td> 1</td></tr>\n",
       "\t<tr><td>foregut / hindgut / anal pad</td><td> 1</td></tr>\n",
       "\t<tr><td>hindgut / anal pad          </td><td> 1</td></tr>\n",
       "\t<tr><td>midgut                      </td><td> 7</td></tr>\n",
       "\t<tr><td>muscle                      </td><td> 3</td></tr>\n",
       "\t<tr><td>procrystal cell             </td><td> 1</td></tr>\n",
       "\t<tr><td>salivary gland              </td><td> 1</td></tr>\n",
       "\t<tr><td>somatic muscle              </td><td> 1</td></tr>\n",
       "\t<tr><td>ubiquitous                  </td><td> 8</td></tr>\n",
       "\t<tr><td>ventral nerve cord          </td><td> 1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 11 × 2\n",
       "\\begin{tabular}{ll}\n",
       " annot & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t brain                        & 22\\\\\n",
       "\t ectoderm primordium          &  1\\\\\n",
       "\t foregut / hindgut / anal pad &  1\\\\\n",
       "\t hindgut / anal pad           &  1\\\\\n",
       "\t midgut                       &  7\\\\\n",
       "\t muscle                       &  3\\\\\n",
       "\t procrystal cell              &  1\\\\\n",
       "\t salivary gland               &  1\\\\\n",
       "\t somatic muscle               &  1\\\\\n",
       "\t ubiquitous                   &  8\\\\\n",
       "\t ventral nerve cord           &  1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 11 × 2\n",
       "\n",
       "| annot &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| brain                        | 22 |\n",
       "| ectoderm primordium          |  1 |\n",
       "| foregut / hindgut / anal pad |  1 |\n",
       "| hindgut / anal pad           |  1 |\n",
       "| midgut                       |  7 |\n",
       "| muscle                       |  3 |\n",
       "| procrystal cell              |  1 |\n",
       "| salivary gland               |  1 |\n",
       "| somatic muscle               |  1 |\n",
       "| ubiquitous                   |  8 |\n",
       "| ventral nerve cord           |  1 |\n",
       "\n"
      ],
      "text/plain": [
       "   annot                        n \n",
       "1  brain                        22\n",
       "2  ectoderm primordium           1\n",
       "3  foregut / hindgut / anal pad  1\n",
       "4  hindgut / anal pad            1\n",
       "5  midgut                        7\n",
       "6  muscle                        3\n",
       "7  procrystal cell               1\n",
       "8  salivary gland                1\n",
       "9  somatic muscle                1\n",
       "10 ubiquitous                    8\n",
       "11 ventral nerve cord            1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub_no_multitons  %>% filter(tmp.expr.Muc11A > 0 & tmp.expr.Muc11A < 1 &\n",
    "                               !(cluster %in% c(103, 163)))  %>% count(annot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1481,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sub300$tmp.expr.Muc11A = counts(cds_sub)[\"Muc11A\", ]/df_sub300$Size_Factor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1482,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 16 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>annot</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>brain                       </td><td>23</td></tr>\n",
       "\t<tr><td>brain primordium            </td><td> 1</td></tr>\n",
       "\t<tr><td>ectoderm primordium         </td><td> 2</td></tr>\n",
       "\t<tr><td>epidermis                   </td><td> 5</td></tr>\n",
       "\t<tr><td>epidermis primordium        </td><td> 1</td></tr>\n",
       "\t<tr><td>foregut / hindgut / anal pad</td><td> 1</td></tr>\n",
       "\t<tr><td>hindgut / anal pad          </td><td> 1</td></tr>\n",
       "\t<tr><td>midgut                      </td><td>10</td></tr>\n",
       "\t<tr><td>muscle                      </td><td> 7</td></tr>\n",
       "\t<tr><td>plasmatocytes anlage        </td><td> 1</td></tr>\n",
       "\t<tr><td>procrystal cell             </td><td> 2</td></tr>\n",
       "\t<tr><td>salivary gland              </td><td> 1</td></tr>\n",
       "\t<tr><td>somatic muscle              </td><td> 2</td></tr>\n",
       "\t<tr><td>tracheal primordium         </td><td> 5</td></tr>\n",
       "\t<tr><td>ubiquitous                  </td><td>16</td></tr>\n",
       "\t<tr><td>ventral nerve cord          </td><td> 1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 16 × 2\n",
       "\\begin{tabular}{ll}\n",
       " annot & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t brain                        & 23\\\\\n",
       "\t brain primordium             &  1\\\\\n",
       "\t ectoderm primordium          &  2\\\\\n",
       "\t epidermis                    &  5\\\\\n",
       "\t epidermis primordium         &  1\\\\\n",
       "\t foregut / hindgut / anal pad &  1\\\\\n",
       "\t hindgut / anal pad           &  1\\\\\n",
       "\t midgut                       & 10\\\\\n",
       "\t muscle                       &  7\\\\\n",
       "\t plasmatocytes anlage         &  1\\\\\n",
       "\t procrystal cell              &  2\\\\\n",
       "\t salivary gland               &  1\\\\\n",
       "\t somatic muscle               &  2\\\\\n",
       "\t tracheal primordium          &  5\\\\\n",
       "\t ubiquitous                   & 16\\\\\n",
       "\t ventral nerve cord           &  1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 16 × 2\n",
       "\n",
       "| annot &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| brain                        | 23 |\n",
       "| brain primordium             |  1 |\n",
       "| ectoderm primordium          |  2 |\n",
       "| epidermis                    |  5 |\n",
       "| epidermis primordium         |  1 |\n",
       "| foregut / hindgut / anal pad |  1 |\n",
       "| hindgut / anal pad           |  1 |\n",
       "| midgut                       | 10 |\n",
       "| muscle                       |  7 |\n",
       "| plasmatocytes anlage         |  1 |\n",
       "| procrystal cell              |  2 |\n",
       "| salivary gland               |  1 |\n",
       "| somatic muscle               |  2 |\n",
       "| tracheal primordium          |  5 |\n",
       "| ubiquitous                   | 16 |\n",
       "| ventral nerve cord           |  1 |\n",
       "\n"
      ],
      "text/plain": [
       "   annot                        n \n",
       "1  brain                        23\n",
       "2  brain primordium              1\n",
       "3  ectoderm primordium           2\n",
       "4  epidermis                     5\n",
       "5  epidermis primordium          1\n",
       "6  foregut / hindgut / anal pad  1\n",
       "7  hindgut / anal pad            1\n",
       "8  midgut                       10\n",
       "9  muscle                        7\n",
       "10 plasmatocytes anlage          1\n",
       "11 procrystal cell               2\n",
       "12 salivary gland                1\n",
       "13 somatic muscle                2\n",
       "14 tracheal primordium           5\n",
       "15 ubiquitous                   16\n",
       "16 ventral nerve cord            1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub300  %>% filter(tmp.expr.Muc11A > 0 & tmp.expr.Muc11A < 1 &\n",
    "                               !(cluster %in% c(103, 163)))  %>% count(annot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1483,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 7 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>annot</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Malpighian tubule     </td><td>19</td></tr>\n",
       "\t<tr><td>midgut                </td><td>22</td></tr>\n",
       "\t<tr><td>sensory nervous system</td><td> 1</td></tr>\n",
       "\t<tr><td>tracheal primordium   </td><td> 5</td></tr>\n",
       "\t<tr><td>tracheal system       </td><td> 6</td></tr>\n",
       "\t<tr><td>ubiquitous            </td><td> 1</td></tr>\n",
       "\t<tr><td>unknown               </td><td> 2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 7 × 2\n",
       "\\begin{tabular}{ll}\n",
       " annot & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Malpighian tubule      & 19\\\\\n",
       "\t midgut                 & 22\\\\\n",
       "\t sensory nervous system &  1\\\\\n",
       "\t tracheal primordium    &  5\\\\\n",
       "\t tracheal system        &  6\\\\\n",
       "\t ubiquitous             &  1\\\\\n",
       "\t unknown                &  2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 7 × 2\n",
       "\n",
       "| annot &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Malpighian tubule      | 19 |\n",
       "| midgut                 | 22 |\n",
       "| sensory nervous system |  1 |\n",
       "| tracheal primordium    |  5 |\n",
       "| tracheal system        |  6 |\n",
       "| ubiquitous             |  1 |\n",
       "| unknown                |  2 |\n",
       "\n"
      ],
      "text/plain": [
       "  annot                  n \n",
       "1 Malpighian tubule      19\n",
       "2 midgut                 22\n",
       "3 sensory nervous system  1\n",
       "4 tracheal primordium     5\n",
       "5 tracheal system         6\n",
       "6 ubiquitous              1\n",
       "7 unknown                 2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub300  %>% filter(tmp.expr.Muc11A > 0 & tmp.expr.Muc11A < 1 &\n",
    "                               (cluster %in% c(103, 163)))  %>% count(annot)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Reannotate fly cells in cds_sub (300 PCAs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 10 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cluster</th><th scope=col>gene</th><th scope=col>cluster.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cluster.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG918453</th><td>54 </td><td>CG9184 </td><td>7147</td><td>35616</td><td>14831.3446</td><td>1522.1598978</td><td>3.283510</td><td>45.49782</td></tr>\n",
       "\t<tr><th scope=row>CG1059153</th><td>54 </td><td>CG10591</td><td>5091</td><td>23244</td><td>10405.6717</td><td> 991.7346141</td><td>3.389818</td><td>45.23787</td></tr>\n",
       "\t<tr><th scope=row>CG5715149</th><td>150</td><td>CG5715 </td><td>  32</td><td>   17</td><td>  138.1496</td><td>   0.9084849</td><td>6.177659</td><td>43.98798</td></tr>\n",
       "\t<tr><th scope=row>sr143</th><td>144</td><td>sr     </td><td> 591</td><td> 3816</td><td> 2365.5387</td><td> 165.4169404</td><td>3.829295</td><td>42.92089</td></tr>\n",
       "\t<tr><th scope=row>bbg14</th><td>15 </td><td>bbg    </td><td>4838</td><td>15567</td><td> 7143.5268</td><td> 700.3065817</td><td>3.348519</td><td>42.86983</td></tr>\n",
       "\t<tr><th scope=row>scro56</th><td>57 </td><td>scro   </td><td> 598</td><td> 1464</td><td> 1232.0042</td><td>  67.3759216</td><td>4.171375</td><td>42.83152</td></tr>\n",
       "\t<tr><th scope=row>lmd99</th><td>100</td><td>lmd    </td><td> 691</td><td> 3111</td><td> 1997.0907</td><td> 132.5558449</td><td>3.902385</td><td>42.78733</td></tr>\n",
       "\t<tr><th scope=row>Desi53</th><td>54 </td><td>Desi   </td><td>  89</td><td>   86</td><td>  182.2837</td><td>   2.5356873</td><td>5.688051</td><td>42.76239</td></tr>\n",
       "\t<tr><th scope=row>ImpE253</th><td>54 </td><td>ImpE2  </td><td>3061</td><td>14790</td><td> 6225.0365</td><td> 603.3959499</td><td>3.364516</td><td>42.40670</td></tr>\n",
       "\t<tr><th scope=row>sha14</th><td>15 </td><td>sha    </td><td>1164</td><td> 2633</td><td> 1781.7802</td><td> 119.2874267</td><td>3.888762</td><td>41.99829</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 10 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cluster & gene & cluster.n.umi & other.n.umi & cluster.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <fct> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG918453 & 54  & CG9184  & 7147 & 35616 & 14831.3446 & 1522.1598978 & 3.283510 & 45.49782\\\\\n",
       "\tCG1059153 & 54  & CG10591 & 5091 & 23244 & 10405.6717 &  991.7346141 & 3.389818 & 45.23787\\\\\n",
       "\tCG5715149 & 150 & CG5715  &   32 &    17 &   138.1496 &    0.9084849 & 6.177659 & 43.98798\\\\\n",
       "\tsr143 & 144 & sr      &  591 &  3816 &  2365.5387 &  165.4169404 & 3.829295 & 42.92089\\\\\n",
       "\tbbg14 & 15  & bbg     & 4838 & 15567 &  7143.5268 &  700.3065817 & 3.348519 & 42.86983\\\\\n",
       "\tscro56 & 57  & scro    &  598 &  1464 &  1232.0042 &   67.3759216 & 4.171375 & 42.83152\\\\\n",
       "\tlmd99 & 100 & lmd     &  691 &  3111 &  1997.0907 &  132.5558449 & 3.902385 & 42.78733\\\\\n",
       "\tDesi53 & 54  & Desi    &   89 &    86 &   182.2837 &    2.5356873 & 5.688051 & 42.76239\\\\\n",
       "\tImpE253 & 54  & ImpE2   & 3061 & 14790 &  6225.0365 &  603.3959499 & 3.364516 & 42.40670\\\\\n",
       "\tsha14 & 15  & sha     & 1164 &  2633 &  1781.7802 &  119.2874267 & 3.888762 & 41.99829\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 10 × 8\n",
       "\n",
       "| <!--/--> | cluster &lt;fct&gt; | gene &lt;chr&gt; | cluster.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cluster.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG918453 | 54  | CG9184  | 7147 | 35616 | 14831.3446 | 1522.1598978 | 3.283510 | 45.49782 |\n",
       "| CG1059153 | 54  | CG10591 | 5091 | 23244 | 10405.6717 |  991.7346141 | 3.389818 | 45.23787 |\n",
       "| CG5715149 | 150 | CG5715  |   32 |    17 |   138.1496 |    0.9084849 | 6.177659 | 43.98798 |\n",
       "| sr143 | 144 | sr      |  591 |  3816 |  2365.5387 |  165.4169404 | 3.829295 | 42.92089 |\n",
       "| bbg14 | 15  | bbg     | 4838 | 15567 |  7143.5268 |  700.3065817 | 3.348519 | 42.86983 |\n",
       "| scro56 | 57  | scro    |  598 |  1464 |  1232.0042 |   67.3759216 | 4.171375 | 42.83152 |\n",
       "| lmd99 | 100 | lmd     |  691 |  3111 |  1997.0907 |  132.5558449 | 3.902385 | 42.78733 |\n",
       "| Desi53 | 54  | Desi    |   89 |    86 |   182.2837 |    2.5356873 | 5.688051 | 42.76239 |\n",
       "| ImpE253 | 54  | ImpE2   | 3061 | 14790 |  6225.0365 |  603.3959499 | 3.364516 | 42.40670 |\n",
       "| sha14 | 15  | sha     | 1164 |  2633 |  1781.7802 |  119.2874267 | 3.888762 | 41.99829 |\n",
       "\n"
      ],
      "text/plain": [
       "          cluster gene    cluster.n.umi other.n.umi cluster.tpm other.tpm   \n",
       "CG918453  54      CG9184  7147          35616       14831.3446  1522.1598978\n",
       "CG1059153 54      CG10591 5091          23244       10405.6717   991.7346141\n",
       "CG5715149 150     CG5715    32             17         138.1496     0.9084849\n",
       "sr143     144     sr       591           3816        2365.5387   165.4169404\n",
       "bbg14     15      bbg     4838          15567        7143.5268   700.3065817\n",
       "scro56    57      scro     598           1464        1232.0042    67.3759216\n",
       "lmd99     100     lmd      691           3111        1997.0907   132.5558449\n",
       "Desi53    54      Desi      89             86         182.2837     2.5356873\n",
       "ImpE253   54      ImpE2   3061          14790        6225.0365   603.3959499\n",
       "sha14     15      sha     1164           2633        1781.7802   119.2874267\n",
       "          log2.ratio heuristic.score\n",
       "CG918453  3.283510   45.49782       \n",
       "CG1059153 3.389818   45.23787       \n",
       "CG5715149 6.177659   43.98798       \n",
       "sr143     3.829295   42.92089       \n",
       "bbg14     3.348519   42.86983       \n",
       "scro56    4.171375   42.83152       \n",
       "lmd99     3.902385   42.78733       \n",
       "Desi53    5.688051   42.76239       \n",
       "ImpE253   3.364516   42.40670       \n",
       "sha14     3.888762   41.99829       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cluster.markers.sub300_no_singletons %>% filter(!(cluster %in% c(160, 163, 103, 172, 168, \n",
    "                                                              170, 14, 8,\n",
    "                                                              141, 151, 146, 165,\n",
    "                                                              85, 41, 99, 110, 31, 60, 111, 169,\n",
    "                                                              51, 95, 73, 1, 174, 2,\n",
    "                                                              171, 20, 98, 45, 96, 92, 130, 88,\n",
    "                                                              13, 55, 167, 104, 123,  25, 129,\n",
    "                                                              157, 122, 58, 28, 89, 148, 159,\n",
    "                                                              142, 154, 124, 149, 126, 113,\n",
    "                                                              81, 139, 34, 147, 166, 107,\n",
    "                                                              155, 173, 40, 80, 161, 120, 162, \n",
    "                                                               134, 156, 115, 56, 30, 128, 137, 158,\n",
    "                                                              121, 109, 94, 145, 68, 86, 114, 17, 63,\n",
    "                                                              47, 164, 166, 5, 66, 39, 9, 106,\n",
    "                                                              10, 152,  36, 127, 12, 75, 46, 133, 136, 119, 48,\n",
    "                                                              49, 64, 72, 140) ))  %>%  \n",
    "head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in h(simpleError(msg, call)): error in evaluating the argument 'x' in selecting a method for function 'head': object 'cluster_expr_sub300_no_singletons' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in h(simpleError(msg, call)): error in evaluating the argument 'x' in selecting a method for function 'head': object 'cluster_expr_sub300_no_singletons' not found\nTraceback:\n",
      "1. cluster_expr_sub300_no_singletons %>% filter(!(cluster %in% c(160, \n .     163, 103, 172, 168, 170, 14, 8, 141, 151, 146, 165, 85, 41, \n .     99, 110, 31, 60, 111, 169, 51, 95, 73, 1, 174, 2, 171, 20, \n .     98, 45, 96, 92, 130, 88, 13, 55, 167, 104, 123, 25, 129, \n .     157, 122, 58, 28, 89, 148, 159, 142, 154, 124, 149, 126, \n .     113, 81, 139, 34, 147, 166, 107, 155, 173, 40, 80, 161, 120, \n .     162, 134, 156, 115, 56, 30, 128, 137, 158, 121, 109, 94, \n .     145, 68, 86, 114, 17, 63, 47, 164, 166, 5, 66, 39, 9, 106, \n .     10, 152, 36, 127, 12, 75, 46, 133, 136, 119, 48, 49, 64, \n .     72, 140))) %>% head(10)",
      "2. head(., 10)",
      "3. filter(., !(cluster %in% c(160, 163, 103, 172, 168, 170, 14, \n .     8, 141, 151, 146, 165, 85, 41, 99, 110, 31, 60, 111, 169, \n .     51, 95, 73, 1, 174, 2, 171, 20, 98, 45, 96, 92, 130, 88, \n .     13, 55, 167, 104, 123, 25, 129, 157, 122, 58, 28, 89, 148, \n .     159, 142, 154, 124, 149, 126, 113, 81, 139, 34, 147, 166, \n .     107, 155, 173, 40, 80, 161, 120, 162, 134, 156, 115, 56, \n .     30, 128, 137, 158, 121, 109, 94, 145, 68, 86, 114, 17, 63, \n .     47, 164, 166, 5, 66, 39, 9, 106, 10, 152, 36, 127, 12, 75, \n .     46, 133, 136, 119, 48, 49, 64, 72, 140)))",
      "4. .handleSimpleError(function (cond) \n . .Internal(C_tryCatchHelper(addr, 1L, cond)), \"object 'cluster_expr_sub300_no_singletons' not found\", \n .     base::quote(filter(., !(cluster %in% c(160, 163, 103, 172, \n .         168, 170, 14, 8, 141, 151, 146, 165, 85, 41, 99, 110, \n .         31, 60, 111, 169, 51, 95, 73, 1, 174, 2, 171, 20, 98, \n .         45, 96, 92, 130, 88, 13, 55, 167, 104, 123, 25, 129, \n .         157, 122, 58, 28, 89, 148, 159, 142, 154, 124, 149, 126, \n .         113, 81, 139, 34, 147, 166, 107, 155, 173, 40, 80, 161, \n .         120, 162, 134, 156, 115, 56, 30, 128, 137, 158, 121, \n .         109, 94, 145, 68, 86, 114, 17, 63, 47, 164, 166, 5, 66, \n .         39, 9, 106, 10, 152, 36, 127, 12, 75, 46, 133, 136, 119, \n .         48, 49, 64, 72, 140)))))",
      "5. h(simpleError(msg, call))"
     ]
    }
   ],
   "source": [
    "cluster_expr_sub300_no_singletons  %>% filter(!(cluster %in% c(160, 163, 103, 172, 168, \n",
    "                                                              170, 14, 8,\n",
    "                                                              141, 151, 146, 165,\n",
    "                                                              85, 41, 99, 110, 31, 60, 111, 169,\n",
    "                                                              51, 95, 73, 1, 174, 2,\n",
    "                                                              171, 20, 98, 45, 96, 92, 130, 88,\n",
    "                                                              13, 55, 167, 104, 123,  25, 129,\n",
    "                                                              157, 122, 58, 28, 89, 148, 159,\n",
    "                                                              142, 154, 124, 149, 126, 113,\n",
    "                                                              81, 139, 34, 147, 166, 107,\n",
    "                                                              155, 173, 40, 80, 161, 120, 162, \n",
    "                                                               134, 156, 115, 56, 30, 128, 137, 158,\n",
    "                                                              121, 109, 94, 145, 68, 86, 114, 17, 63,\n",
    "                                                              47, 164, 166, 5, 66, 39, 9, 106,\n",
    "                                                              10, 152,  36, 127, 12, 75, 46, 133, 136, 119, 48,\n",
    "                                                              49, 64, 72, 140) ))  %>%  \n",
    "head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_multitons, c(\"sr\"), size_pos = 0.1, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_multitons, c(\"sr\"), size_pos = 0.1, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_multitons, c(\"sr\"), \n",
    "                   size_pos = 0.1, size_neg = 0.2, stroke = 0.02, limit = 1) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = c(160)\n",
    "b = c(163, 103)\n",
    "c = c(172)\n",
    "d = c(168)\n",
    "e = c(170, 14, 8, 143)  # 8 could be broken down further\n",
    "f = c(141, 151)\n",
    "g = c(146)\n",
    "h = c(165)\n",
    "i = c(85, 41, 99, 110, 31, 60, 111)\n",
    "j = c(169)\n",
    "k = c(174)\n",
    "l = c(51)\n",
    "m = c(95, 73)\n",
    "n = c(1)\n",
    "o = c(2) # CG43896 is only at one tip so could be broken down further\n",
    "p = c(171)\n",
    "q = c(20, 98, 45, 96, 92, 130, 88) # 20 should be split with 13, 55\n",
    "r = c(13, 55)\n",
    "s = c(167, 104, 123,  25) # has considerable scatter\n",
    "t = c(129) # possibly two branches\n",
    "u = c(157)\n",
    "v = c(122, 58, 28, 89)\n",
    "w = c(148)\n",
    "x = c(159)\n",
    "y = c(142, 154) # has spur on 154 that could be removed\n",
    "z = c(124)\n",
    "aa = c(149, 126) # 121, 109 are spurs, tributaries\n",
    "ab = c(113) # could be extension of y at least in part, but looks like much is separate\n",
    "ac = c(81, 139, 34)\n",
    "ad = c(147)\n",
    "ae = c(166, 107)  # markers shar, but distant on map; and only a small cluster of 107, which could be broken 3 ways\n",
    "af = c(173)\n",
    "ag = c(155)\n",
    "ah = c(40, 134 ) # 40 has spur that could be broken off\n",
    "ai = c(80)\n",
    "aj = c(161, 120, 162)\n",
    "ak = c(156)\n",
    "al = c(115, 56) # coalescent for 124, 128, 137\n",
    "am = c(30)\n",
    "an = c(128, 137)  # see what the markers are here\n",
    "ao = c(158) # closely related to 170, 14\n",
    "ap = c(121, 109) # Tsp74F lights up main trajectory of these two, but both could be broken further\n",
    "aq = c(94)\n",
    "ar = c(145, 68, 86)\n",
    "as = c(114)  # CG13545 is only in a single part of this larger cluster, part of brain and so might be neuron\n",
    "at = c(17)  # 49 is base and with part of 107 feeds into 12\n",
    "au = c(63) # CG7300 is only in left spur of 63, so could be broken down further; other spur feeds 128\n",
    "av = c(47, 164)\n",
    "aw = c(166) # markers also light up small part of 107, but not the main cluster.  Look for better markers of bulk of 107\n",
    "ax = c(5, 66) # Ptx1 marker also lights up 159 which is a spur of 5\n",
    "ay = c(39, 9, 106)\n",
    "az = c(10, 152, 36, 127) # basal clusters\n",
    "ba = c(12) # Lim3 also in adjacent 2, 49 is base\n",
    "bb = c(46) # CG44422 not specific but \"CG44422\", \"disco-r\" pair very specific\n",
    "bc = c(75) # separate cluster of 75 so could be broken down further tey marker also shared with 156\n",
    "bd = c(133) # Desi in small cluster of 133, part of brain, so could neuronal; shared with 121 or 54\n",
    "be = c(136, 119, 48, 100) # basal cluster\n",
    "bf = c(49) # with part of 107 base of 17, 12\n",
    "bg = c(64) # Ccp84Ac covers it but shared with 113\n",
    "bh = c(72, 57) # may be base of 124, 128, both could be broken\n",
    "bi = c(140) # very poor cluster, may be with 123\n",
    "## below done only on umap picture\n",
    "bj = c(23, 116, 105, 153)\n",
    "bk = c(52, 26, 102, 27)\n",
    "bl = c(87, 70, 35, 37, 22, 59)\n",
    "bm = c(132, 125, 108)\n",
    "bn = c(44)\n",
    "bo = c(97, 118)\n",
    "bp = c(91, 144, 54, 15, 84)\n",
    "bq = c(6, 117, 38, 43, 4)\n",
    "br = c(138, 67, 65)\n",
    "bs = c(32, 82, 131)\n",
    "bt = c(11, 76, 3, 62, 50)\n",
    "bu = c(71, 19)\n",
    "bv = c(33, 61)\n",
    "bw = c(77, 53)\n",
    "bx = c(29, 135)\n",
    "by = c(21)\n",
    "bz = c(18) # scattered cells\n",
    "ca = c(59, 74, 101)\n",
    "cb = c(112, 79, 7)\n",
    "cc = c(16, 93, 42, 78, 24)\n",
    "cd = c(69, 83)\n",
    "ce = c(90)\n",
    "cf = c(150) # very scattered cells near 12"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(150)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clus = c(a, b, c, d, e, f, g, h, i , j, k, l, m, n, o, p, q, r, s, t, u, v, w, x, y, z,\n",
    "        aa,ab, ac, ad, ae, af, ag, ah, ai, aj, ak, al, am, an, ao, ap, aq, ar, as, at, au, av, aw,\n",
    "        ax, ay, az, ba, bb, bc, bd, be, bf, bh, bi, bj, bk, bl, bm, bn, bo, bp, bq, br, bs, bt, bu, bv, \n",
    "         bw, bx, by, bz, ca, cb, cc, cd, ce)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sub300$cell.type.2 = rep(NA, ncol(cds_sub))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1747,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 35</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>orig.ident</th><th scope=col>nCount_RNA</th><th scope=col>nFeature_RNA</th><th scope=col>rt_bc</th><th scope=col>cell</th><th scope=col>time</th><th scope=col>percent.mt</th><th scope=col>percent.ribo</th><th scope=col>barn_dro</th><th scope=col>barn_hu</th><th scope=col>barn_mus</th><th scope=col>RNA_snn_res.0.8</th><th scope=col>doublet_score</th><th scope=col>predicted_doublet</th><th scope=col>doublet_cluster</th><th scope=col>seurat_clusters</th><th scope=col>doublet_subcluster</th><th scope=col>RNA_snn_res.0.08</th><th scope=col>lasso_age</th><th scope=col>NNv1_age</th><th scope=col>NNv2_age</th><th scope=col>lasso_shift</th><th scope=col>NNv1_shift</th><th scope=col>NNv2_shift</th><th scope=col>annot</th><th scope=col>ident</th><th scope=col>inferred_time_window</th><th scope=col>cell.type_time.bin</th><th scope=col>Size_Factor</th><th scope=col>cluster</th><th scope=col>umap_1</th><th scope=col>umap_2</th><th scope=col>tmp.expr</th><th scope=col>tmp.expr.Muc11A</th><th scope=col>cell.type.2</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</th><td>exp1</td><td>346</td><td>247</td><td>AAGTTGCCAT</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>hrs_06_10</td><td>1.4450867</td><td>4.3352601</td><td>371</td><td>0</td><td>0</td><td>16</td><td>0.05276064</td><td>Singlet</td><td>FALSE</td><td>21</td><td>FALSE</td><td>21</td><td> 8.402794</td><td> 7.885203</td><td> 8.119637</td><td> 0.0000000</td><td> 0.0000000</td><td> 0.000000</td><td>sensory nervous system primordium</td><td>21</td><td>6-8  </td><td>sensory nervous system primordium:6-8</td><td>0.8108612</td><td>89 </td><td>-2.0067012</td><td> 1.218446</td><td>0</td><td>0</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG</th><td>exp1</td><td>279</td><td>225</td><td>TTAATTATTG</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG</td><td>hrs_08_12</td><td>0.7168459</td><td>5.0179211</td><td>331</td><td>4</td><td>5</td><td>7 </td><td>0.07219402</td><td>Singlet</td><td>FALSE</td><td>11</td><td>FALSE</td><td>11</td><td>11.597093</td><td> 9.940381</td><td> 9.959537</td><td> 0.0000000</td><td> 0.0000000</td><td> 0.000000</td><td>visceral muscle primordium       </td><td>11</td><td>8-10 </td><td>visceral muscle primordium:8-10      </td><td>0.6571200</td><td>112</td><td>-4.6204820</td><td> 1.096802</td><td>0</td><td>0</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA</th><td>exp1</td><td>586</td><td>425</td><td>CAGGTATGGA</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA</td><td>hrs_08_12</td><td>1.0238908</td><td>6.4846416</td><td>630</td><td>0</td><td>0</td><td>4 </td><td>0.09008910</td><td>Singlet</td><td>FALSE</td><td>11</td><td>FALSE</td><td>11</td><td>11.266422</td><td>10.062633</td><td>11.156040</td><td> 0.0000000</td><td> 0.0000000</td><td> 0.000000</td><td>brain                            </td><td>11</td><td>10-12</td><td>brain:10-12                          </td><td>1.4059888</td><td>22 </td><td>-3.6831634</td><td> 2.801265</td><td>0</td><td>0</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG</th><td>exp1</td><td>292</td><td>202</td><td>AAGTAGTCAG</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG</td><td>hrs_10_14</td><td>0.3424658</td><td>0.6849315</td><td>320</td><td>3</td><td>0</td><td>23</td><td>0.06585398</td><td>Singlet</td><td>FALSE</td><td>12</td><td>FALSE</td><td>12</td><td>12.544636</td><td>12.211529</td><td>13.830982</td><td> 0.0000000</td><td> 0.0000000</td><td> 0.000000</td><td>epidermis                        </td><td>12</td><td>12-14</td><td>epidermis:12-14                      </td><td>0.6967951</td><td>165</td><td>-1.3580737</td><td> 0.358204</td><td>0</td><td>0</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA</th><td>exp1</td><td>335</td><td>243</td><td>TGAGGTAGAA</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA</td><td>hrs_14_18</td><td>4.4776119</td><td>5.6716418</td><td>386</td><td>0</td><td>0</td><td>8 </td><td>0.08478501</td><td>Singlet</td><td>FALSE</td><td>11</td><td>FALSE</td><td>11</td><td>12.542527</td><td>13.739003</td><td>12.519031</td><td>-1.4574728</td><td>-0.2609968</td><td>-1.480969</td><td>sensory nervous system           </td><td>11</td><td>12-14</td><td>sensory nervous system:12-14         </td><td>0.7811049</td><td>102</td><td>-3.8563070</td><td> 3.315827</td><td>0</td><td>0</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA</th><td>exp1</td><td>354</td><td>272</td><td>TCGATATTGA</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA</td><td>hrs_14_18</td><td>0.5649718</td><td>1.6949153</td><td>380</td><td>0</td><td>1</td><td>8 </td><td>0.08257165</td><td>Singlet</td><td>FALSE</td><td>11</td><td>FALSE</td><td>11</td><td>13.458162</td><td>15.126904</td><td>14.558554</td><td>-0.5418379</td><td> 0.0000000</td><td> 0.000000</td><td>sensory nervous system           </td><td>11</td><td>14-16</td><td>sensory nervous system:14-16         </td><td>0.8306988</td><td>148</td><td>-0.6711318</td><td>-2.946210</td><td>0</td><td>0</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 35\n",
       "\\begin{tabular}{r|lllllllllllllllllllllllllllllllllll}\n",
       "  & orig.ident & nCount\\_RNA & nFeature\\_RNA & rt\\_bc & cell & time & percent.mt & percent.ribo & barn\\_dro & barn\\_hu & barn\\_mus & RNA\\_snn\\_res.0.8 & doublet\\_score & predicted\\_doublet & doublet\\_cluster & seurat\\_clusters & doublet\\_subcluster & RNA\\_snn\\_res.0.08 & lasso\\_age & NNv1\\_age & NNv2\\_age & lasso\\_shift & NNv1\\_shift & NNv2\\_shift & annot & ident & inferred\\_time\\_window & cell.type\\_time.bin & Size\\_Factor & cluster & umap\\_1 & umap\\_2 & tmp.expr & tmp.expr.Muc11A & cell.type.2\\\\\n",
       "  & <chr> & <dbl> & <int> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <int> & <int> & <int> & <chr> & <dbl> & <chr> & <lgl> & <fct> & <lgl> & <fct> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <fct> & <chr> & <chr> & <dbl> & <fct> & <dbl> & <dbl> & <dbl> & <dbl> & <lgl>\\\\\n",
       "\\hline\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & exp1 & 346 & 247 & AAGTTGCCAT & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & hrs\\_06\\_10 & 1.4450867 & 4.3352601 & 371 & 0 & 0 & 16 & 0.05276064 & Singlet & FALSE & 21 & FALSE & 21 &  8.402794 &  7.885203 &  8.119637 &  0.0000000 &  0.0000000 &  0.000000 & sensory nervous system primordium & 21 & 6-8   & sensory nervous system primordium:6-8 & 0.8108612 & 89  & -2.0067012 &  1.218446 & 0 & 0 & NA\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG & exp1 & 279 & 225 & TTAATTATTG & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG & hrs\\_08\\_12 & 0.7168459 & 5.0179211 & 331 & 4 & 5 & 7  & 0.07219402 & Singlet & FALSE & 11 & FALSE & 11 & 11.597093 &  9.940381 &  9.959537 &  0.0000000 &  0.0000000 &  0.000000 & visceral muscle primordium        & 11 & 8-10  & visceral muscle primordium:8-10       & 0.6571200 & 112 & -4.6204820 &  1.096802 & 0 & 0 & NA\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA & exp1 & 586 & 425 & CAGGTATGGA & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA & hrs\\_08\\_12 & 1.0238908 & 6.4846416 & 630 & 0 & 0 & 4  & 0.09008910 & Singlet & FALSE & 11 & FALSE & 11 & 11.266422 & 10.062633 & 11.156040 &  0.0000000 &  0.0000000 &  0.000000 & brain                             & 11 & 10-12 & brain:10-12                           & 1.4059888 & 22  & -3.6831634 &  2.801265 & 0 & 0 & NA\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG & exp1 & 292 & 202 & AAGTAGTCAG & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG & hrs\\_10\\_14 & 0.3424658 & 0.6849315 & 320 & 3 & 0 & 23 & 0.06585398 & Singlet & FALSE & 12 & FALSE & 12 & 12.544636 & 12.211529 & 13.830982 &  0.0000000 &  0.0000000 &  0.000000 & epidermis                         & 12 & 12-14 & epidermis:12-14                       & 0.6967951 & 165 & -1.3580737 &  0.358204 & 0 & 0 & NA\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA & exp1 & 335 & 243 & TGAGGTAGAA & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA & hrs\\_14\\_18 & 4.4776119 & 5.6716418 & 386 & 0 & 0 & 8  & 0.08478501 & Singlet & FALSE & 11 & FALSE & 11 & 12.542527 & 13.739003 & 12.519031 & -1.4574728 & -0.2609968 & -1.480969 & sensory nervous system            & 11 & 12-14 & sensory nervous system:12-14          & 0.7811049 & 102 & -3.8563070 &  3.315827 & 0 & 0 & NA\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA & exp1 & 354 & 272 & TCGATATTGA & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA & hrs\\_14\\_18 & 0.5649718 & 1.6949153 & 380 & 0 & 1 & 8  & 0.08257165 & Singlet & FALSE & 11 & FALSE & 11 & 13.458162 & 15.126904 & 14.558554 & -0.5418379 &  0.0000000 &  0.000000 & sensory nervous system            & 11 & 14-16 & sensory nervous system:14-16          & 0.8306988 & 148 & -0.6711318 & -2.946210 & 0 & 0 & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 35\n",
       "\n",
       "| <!--/--> | orig.ident &lt;chr&gt; | nCount_RNA &lt;dbl&gt; | nFeature_RNA &lt;int&gt; | rt_bc &lt;chr&gt; | cell &lt;chr&gt; | time &lt;chr&gt; | percent.mt &lt;dbl&gt; | percent.ribo &lt;dbl&gt; | barn_dro &lt;int&gt; | barn_hu &lt;int&gt; | barn_mus &lt;int&gt; | RNA_snn_res.0.8 &lt;chr&gt; | doublet_score &lt;dbl&gt; | predicted_doublet &lt;chr&gt; | doublet_cluster &lt;lgl&gt; | seurat_clusters &lt;fct&gt; | doublet_subcluster &lt;lgl&gt; | RNA_snn_res.0.08 &lt;fct&gt; | lasso_age &lt;dbl&gt; | NNv1_age &lt;dbl&gt; | NNv2_age &lt;dbl&gt; | lasso_shift &lt;dbl&gt; | NNv1_shift &lt;dbl&gt; | NNv2_shift &lt;dbl&gt; | annot &lt;chr&gt; | ident &lt;fct&gt; | inferred_time_window &lt;chr&gt; | cell.type_time.bin &lt;chr&gt; | Size_Factor &lt;dbl&gt; | cluster &lt;fct&gt; | umap_1 &lt;dbl&gt; | umap_2 &lt;dbl&gt; | tmp.expr &lt;dbl&gt; | tmp.expr.Muc11A &lt;dbl&gt; | cell.type.2 &lt;lgl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | exp1 | 346 | 247 | AAGTTGCCAT | exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | hrs_06_10 | 1.4450867 | 4.3352601 | 371 | 0 | 0 | 16 | 0.05276064 | Singlet | FALSE | 21 | FALSE | 21 |  8.402794 |  7.885203 |  8.119637 |  0.0000000 |  0.0000000 |  0.000000 | sensory nervous system primordium | 21 | 6-8   | sensory nervous system primordium:6-8 | 0.8108612 | 89  | -2.0067012 |  1.218446 | 0 | 0 | NA |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG | exp1 | 279 | 225 | TTAATTATTG | exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG | hrs_08_12 | 0.7168459 | 5.0179211 | 331 | 4 | 5 | 7  | 0.07219402 | Singlet | FALSE | 11 | FALSE | 11 | 11.597093 |  9.940381 |  9.959537 |  0.0000000 |  0.0000000 |  0.000000 | visceral muscle primordium        | 11 | 8-10  | visceral muscle primordium:8-10       | 0.6571200 | 112 | -4.6204820 |  1.096802 | 0 | 0 | NA |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA | exp1 | 586 | 425 | CAGGTATGGA | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA | hrs_08_12 | 1.0238908 | 6.4846416 | 630 | 0 | 0 | 4  | 0.09008910 | Singlet | FALSE | 11 | FALSE | 11 | 11.266422 | 10.062633 | 11.156040 |  0.0000000 |  0.0000000 |  0.000000 | brain                             | 11 | 10-12 | brain:10-12                           | 1.4059888 | 22  | -3.6831634 |  2.801265 | 0 | 0 | NA |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG | exp1 | 292 | 202 | AAGTAGTCAG | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG | hrs_10_14 | 0.3424658 | 0.6849315 | 320 | 3 | 0 | 23 | 0.06585398 | Singlet | FALSE | 12 | FALSE | 12 | 12.544636 | 12.211529 | 13.830982 |  0.0000000 |  0.0000000 |  0.000000 | epidermis                         | 12 | 12-14 | epidermis:12-14                       | 0.6967951 | 165 | -1.3580737 |  0.358204 | 0 | 0 | NA |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA | exp1 | 335 | 243 | TGAGGTAGAA | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA | hrs_14_18 | 4.4776119 | 5.6716418 | 386 | 0 | 0 | 8  | 0.08478501 | Singlet | FALSE | 11 | FALSE | 11 | 12.542527 | 13.739003 | 12.519031 | -1.4574728 | -0.2609968 | -1.480969 | sensory nervous system            | 11 | 12-14 | sensory nervous system:12-14          | 0.7811049 | 102 | -3.8563070 |  3.315827 | 0 | 0 | NA |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA | exp1 | 354 | 272 | TCGATATTGA | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA | hrs_14_18 | 0.5649718 | 1.6949153 | 380 | 0 | 1 | 8  | 0.08257165 | Singlet | FALSE | 11 | FALSE | 11 | 13.458162 | 15.126904 | 14.558554 | -0.5418379 |  0.0000000 |  0.000000 | sensory nervous system            | 11 | 14-16 | sensory nervous system:14-16          | 0.8306988 | 148 | -0.6711318 | -2.946210 | 0 | 0 | NA |\n",
       "\n"
      ],
      "text/plain": [
       "                                                       orig.ident nCount_RNA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT exp1       346       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG exp1       279       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA exp1       586       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG exp1       292       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA exp1       335       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA exp1       354       \n",
       "                                                       nFeature_RNA rt_bc     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 247          AAGTTGCCAT\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 225          TTAATTATTG\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 425          CAGGTATGGA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 202          AAGTAGTCAG\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 243          TGAGGTAGAA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 272          TCGATATTGA\n",
       "                                                       cell                                                  \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA\n",
       "                                                       time      percent.mt\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT hrs_06_10 1.4450867 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG hrs_08_12 0.7168459 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA hrs_08_12 1.0238908 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG hrs_10_14 0.3424658 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA hrs_14_18 4.4776119 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA hrs_14_18 0.5649718 \n",
       "                                                       percent.ribo barn_dro\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 4.3352601    371     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 5.0179211    331     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 6.4846416    630     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 0.6849315    320     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 5.6716418    386     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 1.6949153    380     \n",
       "                                                       barn_hu barn_mus\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 0       0       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 4       5       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 0       0       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 3       0       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 0       0       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 0       1       \n",
       "                                                       RNA_snn_res.0.8\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 16             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 7              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 4              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 23             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 8              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 8              \n",
       "                                                       doublet_score\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 0.05276064   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 0.07219402   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 0.09008910   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 0.06585398   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 0.08478501   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 0.08257165   \n",
       "                                                       predicted_doublet\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT Singlet          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG Singlet          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA Singlet          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG Singlet          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA Singlet          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA Singlet          \n",
       "                                                       doublet_cluster\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT FALSE          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG FALSE          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA FALSE          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG FALSE          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA FALSE          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA FALSE          \n",
       "                                                       seurat_clusters\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 21             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 11             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 11             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 11             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 11             \n",
       "                                                       doublet_subcluster\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT FALSE             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG FALSE             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA FALSE             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG FALSE             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA FALSE             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA FALSE             \n",
       "                                                       RNA_snn_res.0.08\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 21              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 11              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 11              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 11              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 11              \n",
       "                                                       lasso_age NNv1_age \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT  8.402794  7.885203\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 11.597093  9.940381\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 11.266422 10.062633\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12.544636 12.211529\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 12.542527 13.739003\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 13.458162 15.126904\n",
       "                                                       NNv2_age  lasso_shift\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT  8.119637  0.0000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG  9.959537  0.0000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 11.156040  0.0000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 13.830982  0.0000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 12.519031 -1.4574728 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 14.558554 -0.5418379 \n",
       "                                                       NNv1_shift NNv2_shift\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT  0.0000000  0.000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG  0.0000000  0.000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA  0.0000000  0.000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG  0.0000000  0.000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA -0.2609968 -1.480969 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA  0.0000000  0.000000 \n",
       "                                                       annot                            \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT sensory nervous system primordium\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG visceral muscle primordium       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA brain                            \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG epidermis                        \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA sensory nervous system           \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA sensory nervous system           \n",
       "                                                       ident\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 21   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 11   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 11   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 11   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 11   \n",
       "                                                       inferred_time_window\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 6-8                 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 8-10                \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 10-12               \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12-14               \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 12-14               \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 14-16               \n",
       "                                                       cell.type_time.bin                   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT sensory nervous system primordium:6-8\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG visceral muscle primordium:8-10      \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA brain:10-12                          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG epidermis:12-14                      \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA sensory nervous system:12-14         \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA sensory nervous system:14-16         \n",
       "                                                       Size_Factor cluster\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 0.8108612   89     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 0.6571200   112    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 1.4059888   22     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 0.6967951   165    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 0.7811049   102    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 0.8306988   148    \n",
       "                                                       umap_1     umap_2   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT -2.0067012  1.218446\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG -4.6204820  1.096802\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA -3.6831634  2.801265\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG -1.3580737  0.358204\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA -3.8563070  3.315827\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA -0.6711318 -2.946210\n",
       "                                                       tmp.expr tmp.expr.Muc11A\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 0        0              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 0        0              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 0        0              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 0        0              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 0        0              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 0        0              \n",
       "                                                       cell.type.2\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT NA         \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG NA         \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA NA         \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG NA         \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA NA         \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA NA         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub300  %>%  head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "set.cell.type = function(data_df, cell.type, x) {\n",
    "       message(\"Assigning cells for cell type \", cell.type)\n",
    "       data_df$cell.type.2 = ifelse(x, cell.type, data_df$cell.type.2)\n",
    "       return(data_df)\n",
    "    }\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sub300_copy = df_sub300"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1786,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sub300 = df_sub300_copy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type a\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"a\", df_sub300$cluster %in% a)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type b\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"b\", df_sub300$cluster %in% b)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type c\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"c\", df_sub300$cluster %in% c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type d\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"d\", df_sub300$cluster %in% d)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type e\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"e\", df_sub300$cluster %in% e)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type f\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"f\", df_sub300$cluster %in% f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type g\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"g\", df_sub300$cluster %in% g)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type h\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"h\", df_sub300$cluster %in% h)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type i\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"i\", df_sub300$cluster %in% i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type j\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"j\", df_sub300$cluster %in% j)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type k\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"k\", df_sub300$cluster %in% k)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type l\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"l\", df_sub300$cluster %in% l)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type m\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"m\", df_sub300$cluster %in% m)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type n\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"n\", df_sub300$cluster %in% n)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type o\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"o\", df_sub300$cluster %in% o)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in parse(text = x, srcfile = src): <text>:1:6: unexpected symbol\n1: copy cells\n         ^\n",
     "output_type": "error",
     "traceback": [
      "Error in parse(text = x, srcfile = src): <text>:1:6: unexpected symbol\n1: copy cells\n         ^\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "copy cells and continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type p\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"p\", df_sub300$cluster %in% p)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type q\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"q\", df_sub300$cluster %in% q)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type r\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"r\", df_sub300$cluster %in% r)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type s\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"s\", df_sub300$cluster %in% s)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type t\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"t\", df_sub300$cluster %in% t)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type u\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"u\", df_sub300$cluster %in% u)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type v\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"v\", df_sub300$cluster %in% v)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type w\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"w\", df_sub300$cluster %in% w)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type x\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"x\", df_sub300$cluster %in% x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type y\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"y\", df_sub300$cluster %in% y)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type z\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"z\", df_sub300$cluster %in% z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type aa\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"aa\", df_sub300$cluster %in% aa)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type ab\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"ab\", df_sub300$cluster %in% ab)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type ac\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"ac\", df_sub300$cluster %in% ac)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type ad\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"ad\", df_sub300$cluster %in% ad)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## and again"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type ae\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"ae\", df_sub300$cluster %in% ae)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type af\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"af\", df_sub300$cluster %in% af)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type ag\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"ag\", df_sub300$cluster %in% ag)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type ah\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"ah\", df_sub300$cluster %in% ah)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type ai\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"ai\", df_sub300$cluster %in% ai)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type aj\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"aj\", df_sub300$cluster %in% aj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type ak\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"ak\", df_sub300$cluster %in% ak)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type al\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"al\", df_sub300$cluster %in% al)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type am\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"am\", df_sub300$cluster %in% am)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type an\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"an\", df_sub300$cluster %in% an)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type ao\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"ao\", df_sub300$cluster %in% ao)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type ap\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"ap\", df_sub300$cluster %in% ap)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type aq\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"aq\", df_sub300$cluster %in% aq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type ar\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"ar\", df_sub300$cluster %in% ar)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type as\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"as\", df_sub300$cluster %in% as)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type at\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"at\", df_sub300$cluster %in% at)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type au\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"au\", df_sub300$cluster %in% au)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type av\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"av\", df_sub300$cluster %in% av)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type aw\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"aw\", df_sub300$cluster %in% aw)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type ax\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"ax\", df_sub300$cluster %in% ax)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type ay\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"ay\", df_sub300$cluster %in% ay)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type az\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"az\", df_sub300$cluster %in% az)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type ba\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"ba\", df_sub300$cluster %in% ba)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bb\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bb\", df_sub300$cluster %in% bb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bc\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bc\", df_sub300$cluster %in% bc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bd\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bd\", df_sub300$cluster %in% bd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type be\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"be\", df_sub300$cluster %in% be)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bf\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bf\", df_sub300$cluster %in% bf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bg\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bg\", df_sub300$cluster %in% bg)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bh\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bh\", df_sub300$cluster %in% bh)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bi\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bi\", df_sub300$cluster %in% bi)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bj\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bj\", df_sub300$cluster %in% bj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bk\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bk\", df_sub300$cluster %in% bk)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bl\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bl\", df_sub300$cluster %in% bl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bm\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bm\", df_sub300$cluster %in% bm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bn\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bn\", df_sub300$cluster %in% bn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bo\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bo\", df_sub300$cluster %in% bo)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bp\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bp\", df_sub300$cluster %in% bp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bq\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bq\", df_sub300$cluster %in% bq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type br\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"br\", df_sub300$cluster %in% br)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bs\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bs\", df_sub300$cluster %in% bs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bt\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bt\", df_sub300$cluster %in% bt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bu\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bu\", df_sub300$cluster %in% bu)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bv\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bv\", df_sub300$cluster %in% bv)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bw\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bw\", df_sub300$cluster %in% bw)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bx\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bx\", df_sub300$cluster %in% bx)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type by\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"by\", df_sub300$cluster %in% by)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bz\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bz\", df_sub300$cluster %in% bz)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type ca\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"ca\", df_sub300$cluster %in% ca)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type cb\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"cb\", df_sub300$cluster %in% cb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type cc\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"cc\", df_sub300$cluster %in% cc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type cd\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"cd\", df_sub300$cluster %in% cd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type ce\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"ce\", df_sub300$cluster %in% ce)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type cf\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"cf\", df_sub300$cluster %in% cf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 174 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cell.type.2</th><th scope=col>cluster</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>a </td><td>160</td><td> 528</td></tr>\n",
       "\t<tr><td>aa</td><td>126</td><td>1081</td></tr>\n",
       "\t<tr><td>aa</td><td>149</td><td> 717</td></tr>\n",
       "\t<tr><td>ab</td><td>113</td><td>1189</td></tr>\n",
       "\t<tr><td>ac</td><td>34 </td><td>1812</td></tr>\n",
       "\t<tr><td>ac</td><td>81 </td><td>1455</td></tr>\n",
       "\t<tr><td>ac</td><td>139</td><td> 970</td></tr>\n",
       "\t<tr><td>ad</td><td>147</td><td> 850</td></tr>\n",
       "\t<tr><td>ae</td><td>107</td><td>1241</td></tr>\n",
       "\t<tr><td>af</td><td>173</td><td> 115</td></tr>\n",
       "\t<tr><td>ag</td><td>155</td><td> 636</td></tr>\n",
       "\t<tr><td>ah</td><td>40 </td><td>1766</td></tr>\n",
       "\t<tr><td>ah</td><td>134</td><td>1000</td></tr>\n",
       "\t<tr><td>ai</td><td>80 </td><td>1467</td></tr>\n",
       "\t<tr><td>aj</td><td>120</td><td>1148</td></tr>\n",
       "\t<tr><td>aj</td><td>161</td><td> 526</td></tr>\n",
       "\t<tr><td>aj</td><td>162</td><td> 485</td></tr>\n",
       "\t<tr><td>ak</td><td>156</td><td> 617</td></tr>\n",
       "\t<tr><td>al</td><td>56 </td><td>1637</td></tr>\n",
       "\t<tr><td>al</td><td>115</td><td>1184</td></tr>\n",
       "\t<tr><td>am</td><td>30 </td><td>1864</td></tr>\n",
       "\t<tr><td>an</td><td>128</td><td>1062</td></tr>\n",
       "\t<tr><td>an</td><td>137</td><td> 981</td></tr>\n",
       "\t<tr><td>ao</td><td>158</td><td> 607</td></tr>\n",
       "\t<tr><td>ap</td><td>109</td><td>1221</td></tr>\n",
       "\t<tr><td>ap</td><td>121</td><td>1146</td></tr>\n",
       "\t<tr><td>aq</td><td>94 </td><td>1355</td></tr>\n",
       "\t<tr><td>ar</td><td>68 </td><td>1557</td></tr>\n",
       "\t<tr><td>ar</td><td>86 </td><td>1406</td></tr>\n",
       "\t<tr><td>ar</td><td>145</td><td> 888</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>l</td><td>51 </td><td>1663</td></tr>\n",
       "\t<tr><td>m</td><td>73 </td><td>1528</td></tr>\n",
       "\t<tr><td>m</td><td>95 </td><td>1352</td></tr>\n",
       "\t<tr><td>n</td><td>1  </td><td>2548</td></tr>\n",
       "\t<tr><td>o</td><td>2  </td><td>2423</td></tr>\n",
       "\t<tr><td>p</td><td>171</td><td> 147</td></tr>\n",
       "\t<tr><td>q</td><td>20 </td><td>2047</td></tr>\n",
       "\t<tr><td>q</td><td>45 </td><td>1695</td></tr>\n",
       "\t<tr><td>q</td><td>88 </td><td>1402</td></tr>\n",
       "\t<tr><td>q</td><td>92 </td><td>1369</td></tr>\n",
       "\t<tr><td>q</td><td>96 </td><td>1351</td></tr>\n",
       "\t<tr><td>q</td><td>98 </td><td>1339</td></tr>\n",
       "\t<tr><td>q</td><td>130</td><td>1052</td></tr>\n",
       "\t<tr><td>r</td><td>13 </td><td>2156</td></tr>\n",
       "\t<tr><td>r</td><td>55 </td><td>1645</td></tr>\n",
       "\t<tr><td>s</td><td>25 </td><td>1977</td></tr>\n",
       "\t<tr><td>s</td><td>104</td><td>1260</td></tr>\n",
       "\t<tr><td>s</td><td>123</td><td>1125</td></tr>\n",
       "\t<tr><td>s</td><td>167</td><td> 370</td></tr>\n",
       "\t<tr><td>t</td><td>129</td><td>1056</td></tr>\n",
       "\t<tr><td>u</td><td>157</td><td> 610</td></tr>\n",
       "\t<tr><td>v</td><td>28 </td><td>1882</td></tr>\n",
       "\t<tr><td>v</td><td>58 </td><td>1621</td></tr>\n",
       "\t<tr><td>v</td><td>89 </td><td>1378</td></tr>\n",
       "\t<tr><td>v</td><td>122</td><td>1140</td></tr>\n",
       "\t<tr><td>w</td><td>148</td><td> 748</td></tr>\n",
       "\t<tr><td>x</td><td>159</td><td> 581</td></tr>\n",
       "\t<tr><td>y</td><td>142</td><td> 931</td></tr>\n",
       "\t<tr><td>y</td><td>154</td><td> 640</td></tr>\n",
       "\t<tr><td>z</td><td>124</td><td>1101</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 174 × 3\n",
       "\\begin{tabular}{lll}\n",
       " cell.type.2 & cluster & n\\\\\n",
       " <chr> & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t a  & 160 &  528\\\\\n",
       "\t aa & 126 & 1081\\\\\n",
       "\t aa & 149 &  717\\\\\n",
       "\t ab & 113 & 1189\\\\\n",
       "\t ac & 34  & 1812\\\\\n",
       "\t ac & 81  & 1455\\\\\n",
       "\t ac & 139 &  970\\\\\n",
       "\t ad & 147 &  850\\\\\n",
       "\t ae & 107 & 1241\\\\\n",
       "\t af & 173 &  115\\\\\n",
       "\t ag & 155 &  636\\\\\n",
       "\t ah & 40  & 1766\\\\\n",
       "\t ah & 134 & 1000\\\\\n",
       "\t ai & 80  & 1467\\\\\n",
       "\t aj & 120 & 1148\\\\\n",
       "\t aj & 161 &  526\\\\\n",
       "\t aj & 162 &  485\\\\\n",
       "\t ak & 156 &  617\\\\\n",
       "\t al & 56  & 1637\\\\\n",
       "\t al & 115 & 1184\\\\\n",
       "\t am & 30  & 1864\\\\\n",
       "\t an & 128 & 1062\\\\\n",
       "\t an & 137 &  981\\\\\n",
       "\t ao & 158 &  607\\\\\n",
       "\t ap & 109 & 1221\\\\\n",
       "\t ap & 121 & 1146\\\\\n",
       "\t aq & 94  & 1355\\\\\n",
       "\t ar & 68  & 1557\\\\\n",
       "\t ar & 86  & 1406\\\\\n",
       "\t ar & 145 &  888\\\\\n",
       "\t ⋮ & ⋮ & ⋮\\\\\n",
       "\t l & 51  & 1663\\\\\n",
       "\t m & 73  & 1528\\\\\n",
       "\t m & 95  & 1352\\\\\n",
       "\t n & 1   & 2548\\\\\n",
       "\t o & 2   & 2423\\\\\n",
       "\t p & 171 &  147\\\\\n",
       "\t q & 20  & 2047\\\\\n",
       "\t q & 45  & 1695\\\\\n",
       "\t q & 88  & 1402\\\\\n",
       "\t q & 92  & 1369\\\\\n",
       "\t q & 96  & 1351\\\\\n",
       "\t q & 98  & 1339\\\\\n",
       "\t q & 130 & 1052\\\\\n",
       "\t r & 13  & 2156\\\\\n",
       "\t r & 55  & 1645\\\\\n",
       "\t s & 25  & 1977\\\\\n",
       "\t s & 104 & 1260\\\\\n",
       "\t s & 123 & 1125\\\\\n",
       "\t s & 167 &  370\\\\\n",
       "\t t & 129 & 1056\\\\\n",
       "\t u & 157 &  610\\\\\n",
       "\t v & 28  & 1882\\\\\n",
       "\t v & 58  & 1621\\\\\n",
       "\t v & 89  & 1378\\\\\n",
       "\t v & 122 & 1140\\\\\n",
       "\t w & 148 &  748\\\\\n",
       "\t x & 159 &  581\\\\\n",
       "\t y & 142 &  931\\\\\n",
       "\t y & 154 &  640\\\\\n",
       "\t z & 124 & 1101\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 174 × 3\n",
       "\n",
       "| cell.type.2 &lt;chr&gt; | cluster &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| a  | 160 |  528 |\n",
       "| aa | 126 | 1081 |\n",
       "| aa | 149 |  717 |\n",
       "| ab | 113 | 1189 |\n",
       "| ac | 34  | 1812 |\n",
       "| ac | 81  | 1455 |\n",
       "| ac | 139 |  970 |\n",
       "| ad | 147 |  850 |\n",
       "| ae | 107 | 1241 |\n",
       "| af | 173 |  115 |\n",
       "| ag | 155 |  636 |\n",
       "| ah | 40  | 1766 |\n",
       "| ah | 134 | 1000 |\n",
       "| ai | 80  | 1467 |\n",
       "| aj | 120 | 1148 |\n",
       "| aj | 161 |  526 |\n",
       "| aj | 162 |  485 |\n",
       "| ak | 156 |  617 |\n",
       "| al | 56  | 1637 |\n",
       "| al | 115 | 1184 |\n",
       "| am | 30  | 1864 |\n",
       "| an | 128 | 1062 |\n",
       "| an | 137 |  981 |\n",
       "| ao | 158 |  607 |\n",
       "| ap | 109 | 1221 |\n",
       "| ap | 121 | 1146 |\n",
       "| aq | 94  | 1355 |\n",
       "| ar | 68  | 1557 |\n",
       "| ar | 86  | 1406 |\n",
       "| ar | 145 |  888 |\n",
       "| ⋮ | ⋮ | ⋮ |\n",
       "| l | 51  | 1663 |\n",
       "| m | 73  | 1528 |\n",
       "| m | 95  | 1352 |\n",
       "| n | 1   | 2548 |\n",
       "| o | 2   | 2423 |\n",
       "| p | 171 |  147 |\n",
       "| q | 20  | 2047 |\n",
       "| q | 45  | 1695 |\n",
       "| q | 88  | 1402 |\n",
       "| q | 92  | 1369 |\n",
       "| q | 96  | 1351 |\n",
       "| q | 98  | 1339 |\n",
       "| q | 130 | 1052 |\n",
       "| r | 13  | 2156 |\n",
       "| r | 55  | 1645 |\n",
       "| s | 25  | 1977 |\n",
       "| s | 104 | 1260 |\n",
       "| s | 123 | 1125 |\n",
       "| s | 167 |  370 |\n",
       "| t | 129 | 1056 |\n",
       "| u | 157 |  610 |\n",
       "| v | 28  | 1882 |\n",
       "| v | 58  | 1621 |\n",
       "| v | 89  | 1378 |\n",
       "| v | 122 | 1140 |\n",
       "| w | 148 |  748 |\n",
       "| x | 159 |  581 |\n",
       "| y | 142 |  931 |\n",
       "| y | 154 |  640 |\n",
       "| z | 124 | 1101 |\n",
       "\n"
      ],
      "text/plain": [
       "    cell.type.2 cluster n   \n",
       "1   a           160      528\n",
       "2   aa          126     1081\n",
       "3   aa          149      717\n",
       "4   ab          113     1189\n",
       "5   ac          34      1812\n",
       "6   ac          81      1455\n",
       "7   ac          139      970\n",
       "8   ad          147      850\n",
       "9   ae          107     1241\n",
       "10  af          173      115\n",
       "11  ag          155      636\n",
       "12  ah          40      1766\n",
       "13  ah          134     1000\n",
       "14  ai          80      1467\n",
       "15  aj          120     1148\n",
       "16  aj          161      526\n",
       "17  aj          162      485\n",
       "18  ak          156      617\n",
       "19  al          56      1637\n",
       "20  al          115     1184\n",
       "21  am          30      1864\n",
       "22  an          128     1062\n",
       "23  an          137      981\n",
       "24  ao          158      607\n",
       "25  ap          109     1221\n",
       "26  ap          121     1146\n",
       "27  aq          94      1355\n",
       "28  ar          68      1557\n",
       "29  ar          86      1406\n",
       "30  ar          145      888\n",
       "⋮   ⋮           ⋮       ⋮   \n",
       "145 l           51      1663\n",
       "146 m           73      1528\n",
       "147 m           95      1352\n",
       "148 n           1       2548\n",
       "149 o           2       2423\n",
       "150 p           171      147\n",
       "151 q           20      2047\n",
       "152 q           45      1695\n",
       "153 q           88      1402\n",
       "154 q           92      1369\n",
       "155 q           96      1351\n",
       "156 q           98      1339\n",
       "157 q           130     1052\n",
       "158 r           13      2156\n",
       "159 r           55      1645\n",
       "160 s           25      1977\n",
       "161 s           104     1260\n",
       "162 s           123     1125\n",
       "163 s           167      370\n",
       "164 t           129     1056\n",
       "165 u           157      610\n",
       "166 v           28      1882\n",
       "167 v           58      1621\n",
       "168 v           89      1378\n",
       "169 v           122     1140\n",
       "170 w           148      748\n",
       "171 x           159      581\n",
       "172 y           142      931\n",
       "173 y           154      640\n",
       "174 z           124     1101"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub300  %>% count(cell.type.2, cluster)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 84 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cell.type.2</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>a </td><td> 528</td></tr>\n",
       "\t<tr><td>aa</td><td>1798</td></tr>\n",
       "\t<tr><td>ab</td><td>1189</td></tr>\n",
       "\t<tr><td>ac</td><td>4237</td></tr>\n",
       "\t<tr><td>ad</td><td> 850</td></tr>\n",
       "\t<tr><td>ae</td><td>1241</td></tr>\n",
       "\t<tr><td>af</td><td> 115</td></tr>\n",
       "\t<tr><td>ag</td><td> 636</td></tr>\n",
       "\t<tr><td>ah</td><td>2766</td></tr>\n",
       "\t<tr><td>ai</td><td>1467</td></tr>\n",
       "\t<tr><td>aj</td><td>2159</td></tr>\n",
       "\t<tr><td>ak</td><td> 617</td></tr>\n",
       "\t<tr><td>al</td><td>2821</td></tr>\n",
       "\t<tr><td>am</td><td>1864</td></tr>\n",
       "\t<tr><td>an</td><td>2043</td></tr>\n",
       "\t<tr><td>ao</td><td> 607</td></tr>\n",
       "\t<tr><td>ap</td><td>2367</td></tr>\n",
       "\t<tr><td>aq</td><td>1355</td></tr>\n",
       "\t<tr><td>ar</td><td>3851</td></tr>\n",
       "\t<tr><td>as</td><td>1186</td></tr>\n",
       "\t<tr><td>at</td><td>2128</td></tr>\n",
       "\t<tr><td>au</td><td>1570</td></tr>\n",
       "\t<tr><td>av</td><td>2139</td></tr>\n",
       "\t<tr><td>aw</td><td> 371</td></tr>\n",
       "\t<tr><td>ax</td><td>3940</td></tr>\n",
       "\t<tr><td>ay</td><td>5347</td></tr>\n",
       "\t<tr><td>az</td><td>5819</td></tr>\n",
       "\t<tr><td>b </td><td>1731</td></tr>\n",
       "\t<tr><td>ba</td><td>2212</td></tr>\n",
       "\t<tr><td>bb</td><td>1693</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>c </td><td>  121</td></tr>\n",
       "\t<tr><td>ca</td><td> 4430</td></tr>\n",
       "\t<tr><td>cb</td><td> 5016</td></tr>\n",
       "\t<tr><td>cc</td><td> 8699</td></tr>\n",
       "\t<tr><td>cd</td><td> 3002</td></tr>\n",
       "\t<tr><td>ce</td><td> 1377</td></tr>\n",
       "\t<tr><td>cf</td><td>  697</td></tr>\n",
       "\t<tr><td>d </td><td>  225</td></tr>\n",
       "\t<tr><td>e </td><td> 5537</td></tr>\n",
       "\t<tr><td>f </td><td> 1629</td></tr>\n",
       "\t<tr><td>g </td><td>  887</td></tr>\n",
       "\t<tr><td>h </td><td>  432</td></tr>\n",
       "\t<tr><td>i </td><td>10353</td></tr>\n",
       "\t<tr><td>j </td><td>  184</td></tr>\n",
       "\t<tr><td>k </td><td>   59</td></tr>\n",
       "\t<tr><td>l </td><td> 1663</td></tr>\n",
       "\t<tr><td>m </td><td> 2880</td></tr>\n",
       "\t<tr><td>n </td><td> 2548</td></tr>\n",
       "\t<tr><td>o </td><td> 2423</td></tr>\n",
       "\t<tr><td>p </td><td>  147</td></tr>\n",
       "\t<tr><td>q </td><td>10255</td></tr>\n",
       "\t<tr><td>r </td><td> 3801</td></tr>\n",
       "\t<tr><td>s </td><td> 4732</td></tr>\n",
       "\t<tr><td>t </td><td> 1056</td></tr>\n",
       "\t<tr><td>u </td><td>  610</td></tr>\n",
       "\t<tr><td>v </td><td> 6021</td></tr>\n",
       "\t<tr><td>w </td><td>  748</td></tr>\n",
       "\t<tr><td>x </td><td>  581</td></tr>\n",
       "\t<tr><td>y </td><td> 1571</td></tr>\n",
       "\t<tr><td>z </td><td> 1101</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 84 × 2\n",
       "\\begin{tabular}{ll}\n",
       " cell.type.2 & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t a  &  528\\\\\n",
       "\t aa & 1798\\\\\n",
       "\t ab & 1189\\\\\n",
       "\t ac & 4237\\\\\n",
       "\t ad &  850\\\\\n",
       "\t ae & 1241\\\\\n",
       "\t af &  115\\\\\n",
       "\t ag &  636\\\\\n",
       "\t ah & 2766\\\\\n",
       "\t ai & 1467\\\\\n",
       "\t aj & 2159\\\\\n",
       "\t ak &  617\\\\\n",
       "\t al & 2821\\\\\n",
       "\t am & 1864\\\\\n",
       "\t an & 2043\\\\\n",
       "\t ao &  607\\\\\n",
       "\t ap & 2367\\\\\n",
       "\t aq & 1355\\\\\n",
       "\t ar & 3851\\\\\n",
       "\t as & 1186\\\\\n",
       "\t at & 2128\\\\\n",
       "\t au & 1570\\\\\n",
       "\t av & 2139\\\\\n",
       "\t aw &  371\\\\\n",
       "\t ax & 3940\\\\\n",
       "\t ay & 5347\\\\\n",
       "\t az & 5819\\\\\n",
       "\t b  & 1731\\\\\n",
       "\t ba & 2212\\\\\n",
       "\t bb & 1693\\\\\n",
       "\t ⋮ & ⋮\\\\\n",
       "\t c  &   121\\\\\n",
       "\t ca &  4430\\\\\n",
       "\t cb &  5016\\\\\n",
       "\t cc &  8699\\\\\n",
       "\t cd &  3002\\\\\n",
       "\t ce &  1377\\\\\n",
       "\t cf &   697\\\\\n",
       "\t d  &   225\\\\\n",
       "\t e  &  5537\\\\\n",
       "\t f  &  1629\\\\\n",
       "\t g  &   887\\\\\n",
       "\t h  &   432\\\\\n",
       "\t i  & 10353\\\\\n",
       "\t j  &   184\\\\\n",
       "\t k  &    59\\\\\n",
       "\t l  &  1663\\\\\n",
       "\t m  &  2880\\\\\n",
       "\t n  &  2548\\\\\n",
       "\t o  &  2423\\\\\n",
       "\t p  &   147\\\\\n",
       "\t q  & 10255\\\\\n",
       "\t r  &  3801\\\\\n",
       "\t s  &  4732\\\\\n",
       "\t t  &  1056\\\\\n",
       "\t u  &   610\\\\\n",
       "\t v  &  6021\\\\\n",
       "\t w  &   748\\\\\n",
       "\t x  &   581\\\\\n",
       "\t y  &  1571\\\\\n",
       "\t z  &  1101\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 84 × 2\n",
       "\n",
       "| cell.type.2 &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| a  |  528 |\n",
       "| aa | 1798 |\n",
       "| ab | 1189 |\n",
       "| ac | 4237 |\n",
       "| ad |  850 |\n",
       "| ae | 1241 |\n",
       "| af |  115 |\n",
       "| ag |  636 |\n",
       "| ah | 2766 |\n",
       "| ai | 1467 |\n",
       "| aj | 2159 |\n",
       "| ak |  617 |\n",
       "| al | 2821 |\n",
       "| am | 1864 |\n",
       "| an | 2043 |\n",
       "| ao |  607 |\n",
       "| ap | 2367 |\n",
       "| aq | 1355 |\n",
       "| ar | 3851 |\n",
       "| as | 1186 |\n",
       "| at | 2128 |\n",
       "| au | 1570 |\n",
       "| av | 2139 |\n",
       "| aw |  371 |\n",
       "| ax | 3940 |\n",
       "| ay | 5347 |\n",
       "| az | 5819 |\n",
       "| b  | 1731 |\n",
       "| ba | 2212 |\n",
       "| bb | 1693 |\n",
       "| ⋮ | ⋮ |\n",
       "| c  |   121 |\n",
       "| ca |  4430 |\n",
       "| cb |  5016 |\n",
       "| cc |  8699 |\n",
       "| cd |  3002 |\n",
       "| ce |  1377 |\n",
       "| cf |   697 |\n",
       "| d  |   225 |\n",
       "| e  |  5537 |\n",
       "| f  |  1629 |\n",
       "| g  |   887 |\n",
       "| h  |   432 |\n",
       "| i  | 10353 |\n",
       "| j  |   184 |\n",
       "| k  |    59 |\n",
       "| l  |  1663 |\n",
       "| m  |  2880 |\n",
       "| n  |  2548 |\n",
       "| o  |  2423 |\n",
       "| p  |   147 |\n",
       "| q  | 10255 |\n",
       "| r  |  3801 |\n",
       "| s  |  4732 |\n",
       "| t  |  1056 |\n",
       "| u  |   610 |\n",
       "| v  |  6021 |\n",
       "| w  |   748 |\n",
       "| x  |   581 |\n",
       "| y  |  1571 |\n",
       "| z  |  1101 |\n",
       "\n"
      ],
      "text/plain": [
       "   cell.type.2 n    \n",
       "1  a            528 \n",
       "2  aa          1798 \n",
       "3  ab          1189 \n",
       "4  ac          4237 \n",
       "5  ad           850 \n",
       "6  ae          1241 \n",
       "7  af           115 \n",
       "8  ag           636 \n",
       "9  ah          2766 \n",
       "10 ai          1467 \n",
       "11 aj          2159 \n",
       "12 ak           617 \n",
       "13 al          2821 \n",
       "14 am          1864 \n",
       "15 an          2043 \n",
       "16 ao           607 \n",
       "17 ap          2367 \n",
       "18 aq          1355 \n",
       "19 ar          3851 \n",
       "20 as          1186 \n",
       "21 at          2128 \n",
       "22 au          1570 \n",
       "23 av          2139 \n",
       "24 aw           371 \n",
       "25 ax          3940 \n",
       "26 ay          5347 \n",
       "27 az          5819 \n",
       "28 b           1731 \n",
       "29 ba          2212 \n",
       "30 bb          1693 \n",
       "⋮  ⋮           ⋮    \n",
       "55 c             121\n",
       "56 ca           4430\n",
       "57 cb           5016\n",
       "58 cc           8699\n",
       "59 cd           3002\n",
       "60 ce           1377\n",
       "61 cf            697\n",
       "62 d             225\n",
       "63 e            5537\n",
       "64 f            1629\n",
       "65 g             887\n",
       "66 h             432\n",
       "67 i           10353\n",
       "68 j             184\n",
       "69 k              59\n",
       "70 l            1663\n",
       "71 m            2880\n",
       "72 n            2548\n",
       "73 o            2423\n",
       "74 p             147\n",
       "75 q           10255\n",
       "76 r            3801\n",
       "77 s            4732\n",
       "78 t            1056\n",
       "79 u             610\n",
       "80 v            6021\n",
       "81 w             748\n",
       "82 x             581\n",
       "83 y            1571\n",
       "84 z            1101"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub300  %>% count(cell.type.2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## DE genes on new annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finding markers for cell.type a\n",
      "\n",
      "# of cells sampled in set 1: 528\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type aa\n",
      "\n",
      "# of cells sampled in set 1: 1798\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ab\n",
      "\n",
      "# of cells sampled in set 1: 1189\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ac\n",
      "\n",
      "# of cells sampled in set 1: 4237\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ad\n",
      "\n",
      "# of cells sampled in set 1: 850\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ae\n",
      "\n",
      "# of cells sampled in set 1: 1241\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type af\n",
      "\n",
      "# of cells sampled in set 1: 115\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ag\n",
      "\n",
      "# of cells sampled in set 1: 636\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ah\n",
      "\n",
      "# of cells sampled in set 1: 2766\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ai\n",
      "\n",
      "# of cells sampled in set 1: 1467\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type aj\n",
      "\n",
      "# of cells sampled in set 1: 2159\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ak\n",
      "\n",
      "# of cells sampled in set 1: 617\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type al\n",
      "\n",
      "# of cells sampled in set 1: 2821\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type am\n",
      "\n",
      "# of cells sampled in set 1: 1864\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type an\n",
      "\n",
      "# of cells sampled in set 1: 2043\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ao\n",
      "\n",
      "# of cells sampled in set 1: 607\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ap\n",
      "\n",
      "# of cells sampled in set 1: 2367\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type aq\n",
      "\n",
      "# of cells sampled in set 1: 1355\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ar\n",
      "\n",
      "# of cells sampled in set 1: 3851\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type as\n",
      "\n",
      "# of cells sampled in set 1: 1186\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type at\n",
      "\n",
      "# of cells sampled in set 1: 2128\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type au\n",
      "\n",
      "# of cells sampled in set 1: 1570\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type av\n",
      "\n",
      "# of cells sampled in set 1: 2139\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type aw\n",
      "\n",
      "# of cells sampled in set 1: 371\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ax\n",
      "\n",
      "# of cells sampled in set 1: 3940\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ay\n",
      "\n",
      "# of cells sampled in set 1: 5347\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type az\n",
      "\n",
      "# of cells sampled in set 1: 5819\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type b\n",
      "\n",
      "# of cells sampled in set 1: 1731\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ba\n",
      "\n",
      "# of cells sampled in set 1: 2212\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bb\n",
      "\n",
      "# of cells sampled in set 1: 1693\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bc\n",
      "\n",
      "# of cells sampled in set 1: 1495\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bd\n",
      "\n",
      "# of cells sampled in set 1: 1003\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type be\n",
      "\n",
      "# of cells sampled in set 1: 5139\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bf\n",
      "\n",
      "# of cells sampled in set 1: 1674\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bg\n",
      "\n",
      "# of cells sampled in set 1: 1565\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bh\n",
      "\n",
      "# of cells sampled in set 1: 3168\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bi\n",
      "\n",
      "# of cells sampled in set 1: 962\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bj\n",
      "\n",
      "# of cells sampled in set 1: 5058\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bk\n",
      "\n",
      "# of cells sampled in set 1: 6846\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bl\n",
      "\n",
      "# of cells sampled in set 1: 8547\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bm\n",
      "\n",
      "# of cells sampled in set 1: 3354\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bn\n",
      "\n",
      "# of cells sampled in set 1: 1700\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bo\n",
      "\n",
      "# of cells sampled in set 1: 2511\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bp\n",
      "\n",
      "# of cells sampled in set 1: 7476\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bq\n",
      "\n",
      "# of cells sampled in set 1: 9416\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type br\n",
      "\n",
      "# of cells sampled in set 1: 4103\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bs\n",
      "\n",
      "# of cells sampled in set 1: 4348\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bt\n",
      "\n",
      "# of cells sampled in set 1: 9401\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bu\n",
      "\n",
      "# of cells sampled in set 1: 3626\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bv\n",
      "\n",
      "# of cells sampled in set 1: 3444\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bw\n",
      "\n",
      "# of cells sampled in set 1: 3137\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bx\n",
      "\n",
      "# of cells sampled in set 1: 2876\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type by\n",
      "\n",
      "# of cells sampled in set 1: 2041\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bz\n",
      "\n",
      "# of cells sampled in set 1: 2123\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type c\n",
      "\n",
      "# of cells sampled in set 1: 121\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ca\n",
      "\n",
      "# of cells sampled in set 1: 4430\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type cb\n",
      "\n",
      "# of cells sampled in set 1: 5016\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type cc\n",
      "\n",
      "# of cells sampled in set 1: 8699\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type cd\n",
      "\n",
      "# of cells sampled in set 1: 3002\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ce\n",
      "\n",
      "# of cells sampled in set 1: 1377\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type cf\n",
      "\n",
      "# of cells sampled in set 1: 697\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type d\n",
      "\n",
      "# of cells sampled in set 1: 225\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type e\n",
      "\n",
      "# of cells sampled in set 1: 5537\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type f\n",
      "\n",
      "# of cells sampled in set 1: 1629\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type g\n",
      "\n",
      "# of cells sampled in set 1: 887\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type h\n",
      "\n",
      "# of cells sampled in set 1: 432\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type i\n",
      "\n",
      "# of cells sampled in set 1: 10353\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type j\n",
      "\n",
      "# of cells sampled in set 1: 184\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type k\n",
      "\n",
      "# of cells sampled in set 1: 59\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type l\n",
      "\n",
      "# of cells sampled in set 1: 1663\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type m\n",
      "\n",
      "# of cells sampled in set 1: 2880\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type n\n",
      "\n",
      "# of cells sampled in set 1: 2548\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type o\n",
      "\n",
      "# of cells sampled in set 1: 2423\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type p\n",
      "\n",
      "# of cells sampled in set 1: 147\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type q\n",
      "\n",
      "# of cells sampled in set 1: 10255\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type r\n",
      "\n",
      "# of cells sampled in set 1: 3801\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type s\n",
      "\n",
      "# of cells sampled in set 1: 4732\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type t\n",
      "\n",
      "# of cells sampled in set 1: 1056\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type u\n",
      "\n",
      "# of cells sampled in set 1: 610\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type v\n",
      "\n",
      "# of cells sampled in set 1: 6021\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type w\n",
      "\n",
      "# of cells sampled in set 1: 748\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type x\n",
      "\n",
      "# of cells sampled in set 1: 581\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type y\n",
      "\n",
      "# of cells sampled in set 1: 1571\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type z\n",
      "\n",
      "# of cells sampled in set 1: 1101\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    }
   ],
   "source": [
    "DEG.results = lapply(sort((unique(df_sub300$cell.type.2))),\n",
    "                     function(this.cell.type) {\n",
    "    message(\"Finding markers for cell.type \", this.cell.type)\n",
    "    cbind(\n",
    "        two.set.differential.gene.test.3a(\n",
    "            cds_sub,\n",
    "            df_sub300$cell.type.2 == this.cell.type,\n",
    "            df_sub300$cell.type.2 != this.cell.type,\n",
    "            sample.n.max = 50000),\n",
    "        data.frame(cell.type.2 = this.cell.type))\n",
    "                         })"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "metadata": {},
   "outputs": [],
   "source": [
    "cell.type.2.markers = do.call(rbind, DEG.results)  %>%\n",
    "    select(\n",
    "        cell.type.2 = cell.type.2, gene, \n",
    "        cell.type.2.n.umi = set.1.n.umi,\n",
    "        other.n.umi = set.2.n.umi,\n",
    "        cell.type.2.tpm = set.1.tpm,\n",
    "        other.tpm = set.2.tpm,\n",
    "        log2.ratio, heuristic.score) %>%\n",
    "    arrange(-heuristic.score)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Muc26B</th><td>a</td><td>Muc26B     </td><td> 3993</td><td> 111</td><td>10368.322</td><td>  4.1576480</td><td>10.973182</td><td>146.3826</td></tr>\n",
       "\t<tr><th scope=row>CG740954</th><td>c</td><td>CG7409     </td><td>  600</td><td>  24</td><td> 4121.228</td><td>  0.4739611</td><td>11.449160</td><td>137.4954</td></tr>\n",
       "\t<tr><th scope=row>CG1137061</th><td>d</td><td>CG11370    </td><td> 3708</td><td> 446</td><td>15737.252</td><td> 17.5042494</td><td> 9.732111</td><td>135.6850</td></tr>\n",
       "\t<tr><th scope=row>CG746563</th><td>f</td><td>CG7465     </td><td>23068</td><td> 761</td><td>18549.567</td><td> 23.9239377</td><td> 9.539638</td><td>135.2642</td></tr>\n",
       "\t<tr><th scope=row>CG7017</th><td>a</td><td>CG7017     </td><td> 3759</td><td> 265</td><td>11685.814</td><td> 10.6958361</td><td> 9.964547</td><td>134.6469</td></tr>\n",
       "\t<tr><th scope=row>CG1135063</th><td>f</td><td>CG11350    </td><td> 8044</td><td> 159</td><td> 6300.220</td><td>  4.2137333</td><td>10.238870</td><td>129.2290</td></tr>\n",
       "\t<tr><th scope=row>Muc96D</th><td>a</td><td>Muc96D     </td><td> 2590</td><td> 118</td><td> 6427.809</td><td>  5.1071969</td><td>10.039601</td><td>127.0043</td></tr>\n",
       "\t<tr><th scope=row>CG1313354</th><td>c</td><td>CG13133    </td><td>  405</td><td>  14</td><td> 2679.090</td><td>  0.1829864</td><td>11.145094</td><td>126.9211</td></tr>\n",
       "\t<tr><th scope=row>CG1456661</th><td>d</td><td>CG14566    </td><td>  518</td><td>  60</td><td> 3985.539</td><td>  1.6730948</td><td>10.542048</td><td>126.0926</td></tr>\n",
       "\t<tr><th scope=row>CG3260363</th><td>f</td><td>CG32603    </td><td>10610</td><td> 353</td><td> 9231.304</td><td> 11.4313571</td><td> 9.536407</td><td>125.6181</td></tr>\n",
       "\t<tr><th scope=row>CG4314954</th><td>c</td><td>CG43149    </td><td>  527</td><td>  40</td><td> 3407.622</td><td>  1.1778353</td><td>10.611655</td><td>124.5275</td></tr>\n",
       "\t<tr><th scope=row>Nplp461</th><td>d</td><td>Nplp4      </td><td>  585</td><td>  98</td><td> 3518.479</td><td>  1.4273695</td><td>10.501343</td><td>123.7179</td></tr>\n",
       "\t<tr><th scope=row>Pde1c65</th><td>h</td><td>Pde1c      </td><td> 6587</td><td>4496</td><td>42193.293</td><td>197.9561181</td><td> 7.728420</td><td>118.7453</td></tr>\n",
       "\t<tr><th scope=row>Jon99Cii64</th><td>g</td><td>Jon99Cii   </td><td> 7795</td><td> 347</td><td> 7722.641</td><td> 12.5731343</td><td> 9.152197</td><td>118.2012</td></tr>\n",
       "\t<tr><th scope=row>CG1442163</th><td>f</td><td>CG14421    </td><td> 4232</td><td>  61</td><td> 3341.966</td><td>  2.2776488</td><td> 9.993820</td><td>116.9968</td></tr>\n",
       "\t<tr><th scope=row>CG1158167</th><td>j</td><td>CG11581    </td><td>  283</td><td>  36</td><td> 2455.322</td><td>  0.9543971</td><td>10.294973</td><td>115.9449</td></tr>\n",
       "\t<tr><th scope=row>CG17826</th><td>a</td><td>CG17826    </td><td> 1107</td><td>  41</td><td> 2902.487</td><td>  1.7851269</td><td>10.025331</td><td>115.3271</td></tr>\n",
       "\t<tr><th scope=row>alphaTub85E54</th><td>c</td><td>alphaTub85E</td><td> 3378</td><td>2390</td><td>21520.691</td><td> 82.5317555</td><td> 8.009184</td><td>115.2802</td></tr>\n",
       "\t<tr><th scope=row>CG1818064</th><td>g</td><td>CG18180    </td><td> 4374</td><td> 104</td><td> 3639.959</td><td>  3.3577182</td><td> 9.706134</td><td>114.8246</td></tr>\n",
       "\t<tr><th scope=row>CG1141361</th><td>d</td><td>CG11413    </td><td>  908</td><td>  67</td><td> 3428.035</td><td>  2.9631017</td><td> 9.756536</td><td>114.5767</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tMuc26B & a & Muc26B      &  3993 &  111 & 10368.322 &   4.1576480 & 10.973182 & 146.3826\\\\\n",
       "\tCG740954 & c & CG7409      &   600 &   24 &  4121.228 &   0.4739611 & 11.449160 & 137.4954\\\\\n",
       "\tCG1137061 & d & CG11370     &  3708 &  446 & 15737.252 &  17.5042494 &  9.732111 & 135.6850\\\\\n",
       "\tCG746563 & f & CG7465      & 23068 &  761 & 18549.567 &  23.9239377 &  9.539638 & 135.2642\\\\\n",
       "\tCG7017 & a & CG7017      &  3759 &  265 & 11685.814 &  10.6958361 &  9.964547 & 134.6469\\\\\n",
       "\tCG1135063 & f & CG11350     &  8044 &  159 &  6300.220 &   4.2137333 & 10.238870 & 129.2290\\\\\n",
       "\tMuc96D & a & Muc96D      &  2590 &  118 &  6427.809 &   5.1071969 & 10.039601 & 127.0043\\\\\n",
       "\tCG1313354 & c & CG13133     &   405 &   14 &  2679.090 &   0.1829864 & 11.145094 & 126.9211\\\\\n",
       "\tCG1456661 & d & CG14566     &   518 &   60 &  3985.539 &   1.6730948 & 10.542048 & 126.0926\\\\\n",
       "\tCG3260363 & f & CG32603     & 10610 &  353 &  9231.304 &  11.4313571 &  9.536407 & 125.6181\\\\\n",
       "\tCG4314954 & c & CG43149     &   527 &   40 &  3407.622 &   1.1778353 & 10.611655 & 124.5275\\\\\n",
       "\tNplp461 & d & Nplp4       &   585 &   98 &  3518.479 &   1.4273695 & 10.501343 & 123.7179\\\\\n",
       "\tPde1c65 & h & Pde1c       &  6587 & 4496 & 42193.293 & 197.9561181 &  7.728420 & 118.7453\\\\\n",
       "\tJon99Cii64 & g & Jon99Cii    &  7795 &  347 &  7722.641 &  12.5731343 &  9.152197 & 118.2012\\\\\n",
       "\tCG1442163 & f & CG14421     &  4232 &   61 &  3341.966 &   2.2776488 &  9.993820 & 116.9968\\\\\n",
       "\tCG1158167 & j & CG11581     &   283 &   36 &  2455.322 &   0.9543971 & 10.294973 & 115.9449\\\\\n",
       "\tCG17826 & a & CG17826     &  1107 &   41 &  2902.487 &   1.7851269 & 10.025331 & 115.3271\\\\\n",
       "\talphaTub85E54 & c & alphaTub85E &  3378 & 2390 & 21520.691 &  82.5317555 &  8.009184 & 115.2802\\\\\n",
       "\tCG1818064 & g & CG18180     &  4374 &  104 &  3639.959 &   3.3577182 &  9.706134 & 114.8246\\\\\n",
       "\tCG1141361 & d & CG11413     &   908 &   67 &  3428.035 &   2.9631017 &  9.756536 & 114.5767\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Muc26B | a | Muc26B      |  3993 |  111 | 10368.322 |   4.1576480 | 10.973182 | 146.3826 |\n",
       "| CG740954 | c | CG7409      |   600 |   24 |  4121.228 |   0.4739611 | 11.449160 | 137.4954 |\n",
       "| CG1137061 | d | CG11370     |  3708 |  446 | 15737.252 |  17.5042494 |  9.732111 | 135.6850 |\n",
       "| CG746563 | f | CG7465      | 23068 |  761 | 18549.567 |  23.9239377 |  9.539638 | 135.2642 |\n",
       "| CG7017 | a | CG7017      |  3759 |  265 | 11685.814 |  10.6958361 |  9.964547 | 134.6469 |\n",
       "| CG1135063 | f | CG11350     |  8044 |  159 |  6300.220 |   4.2137333 | 10.238870 | 129.2290 |\n",
       "| Muc96D | a | Muc96D      |  2590 |  118 |  6427.809 |   5.1071969 | 10.039601 | 127.0043 |\n",
       "| CG1313354 | c | CG13133     |   405 |   14 |  2679.090 |   0.1829864 | 11.145094 | 126.9211 |\n",
       "| CG1456661 | d | CG14566     |   518 |   60 |  3985.539 |   1.6730948 | 10.542048 | 126.0926 |\n",
       "| CG3260363 | f | CG32603     | 10610 |  353 |  9231.304 |  11.4313571 |  9.536407 | 125.6181 |\n",
       "| CG4314954 | c | CG43149     |   527 |   40 |  3407.622 |   1.1778353 | 10.611655 | 124.5275 |\n",
       "| Nplp461 | d | Nplp4       |   585 |   98 |  3518.479 |   1.4273695 | 10.501343 | 123.7179 |\n",
       "| Pde1c65 | h | Pde1c       |  6587 | 4496 | 42193.293 | 197.9561181 |  7.728420 | 118.7453 |\n",
       "| Jon99Cii64 | g | Jon99Cii    |  7795 |  347 |  7722.641 |  12.5731343 |  9.152197 | 118.2012 |\n",
       "| CG1442163 | f | CG14421     |  4232 |   61 |  3341.966 |   2.2776488 |  9.993820 | 116.9968 |\n",
       "| CG1158167 | j | CG11581     |   283 |   36 |  2455.322 |   0.9543971 | 10.294973 | 115.9449 |\n",
       "| CG17826 | a | CG17826     |  1107 |   41 |  2902.487 |   1.7851269 | 10.025331 | 115.3271 |\n",
       "| alphaTub85E54 | c | alphaTub85E |  3378 | 2390 | 21520.691 |  82.5317555 |  8.009184 | 115.2802 |\n",
       "| CG1818064 | g | CG18180     |  4374 |  104 |  3639.959 |   3.3577182 |  9.706134 | 114.8246 |\n",
       "| CG1141361 | d | CG11413     |   908 |   67 |  3428.035 |   2.9631017 |  9.756536 | 114.5767 |\n",
       "\n"
      ],
      "text/plain": [
       "              cell.type.2 gene        cell.type.2.n.umi other.n.umi\n",
       "Muc26B        a           Muc26B       3993              111       \n",
       "CG740954      c           CG7409        600               24       \n",
       "CG1137061     d           CG11370      3708              446       \n",
       "CG746563      f           CG7465      23068              761       \n",
       "CG7017        a           CG7017       3759              265       \n",
       "CG1135063     f           CG11350      8044              159       \n",
       "Muc96D        a           Muc96D       2590              118       \n",
       "CG1313354     c           CG13133       405               14       \n",
       "CG1456661     d           CG14566       518               60       \n",
       "CG3260363     f           CG32603     10610              353       \n",
       "CG4314954     c           CG43149       527               40       \n",
       "Nplp461       d           Nplp4         585               98       \n",
       "Pde1c65       h           Pde1c        6587             4496       \n",
       "Jon99Cii64    g           Jon99Cii     7795              347       \n",
       "CG1442163     f           CG14421      4232               61       \n",
       "CG1158167     j           CG11581       283               36       \n",
       "CG17826       a           CG17826      1107               41       \n",
       "alphaTub85E54 c           alphaTub85E  3378             2390       \n",
       "CG1818064     g           CG18180      4374              104       \n",
       "CG1141361     d           CG11413       908               67       \n",
       "              cell.type.2.tpm other.tpm   log2.ratio heuristic.score\n",
       "Muc26B        10368.322         4.1576480 10.973182  146.3826       \n",
       "CG740954       4121.228         0.4739611 11.449160  137.4954       \n",
       "CG1137061     15737.252        17.5042494  9.732111  135.6850       \n",
       "CG746563      18549.567        23.9239377  9.539638  135.2642       \n",
       "CG7017        11685.814        10.6958361  9.964547  134.6469       \n",
       "CG1135063      6300.220         4.2137333 10.238870  129.2290       \n",
       "Muc96D         6427.809         5.1071969 10.039601  127.0043       \n",
       "CG1313354      2679.090         0.1829864 11.145094  126.9211       \n",
       "CG1456661      3985.539         1.6730948 10.542048  126.0926       \n",
       "CG3260363      9231.304        11.4313571  9.536407  125.6181       \n",
       "CG4314954      3407.622         1.1778353 10.611655  124.5275       \n",
       "Nplp461        3518.479         1.4273695 10.501343  123.7179       \n",
       "Pde1c65       42193.293       197.9561181  7.728420  118.7453       \n",
       "Jon99Cii64     7722.641        12.5731343  9.152197  118.2012       \n",
       "CG1442163      3341.966         2.2776488  9.993820  116.9968       \n",
       "CG1158167      2455.322         0.9543971 10.294973  115.9449       \n",
       "CG17826        2902.487         1.7851269 10.025331  115.3271       \n",
       "alphaTub85E54 21520.691        82.5317555  8.009184  115.2802       \n",
       "CG1818064      3639.959         3.3577182  9.706134  114.8246       \n",
       "CG1141361      3428.035         2.9631017  9.756536  114.5767       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(cell.type.2.markers, 20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>399</li><li>8</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 399\n",
       "\\item 8\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 399\n",
       "2. 8\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 399   8"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers  %>% filter(heuristic.score > 60)  %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 280,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>10</li><li>8</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 10\n",
       "\\item 8\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 10\n",
       "2. 8\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 10  8"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers  %>% filter(heuristic.score > 60 & gene %in% fly_TFs)  %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 284,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>96</li><li>8</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 96\n",
       "\\item 8\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 96\n",
       "2. 8\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 96  8"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers  %>% filter(heuristic.score > 40 & gene %in% fly_TFs)  %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'a'</li><li>'c'</li><li>'d'</li><li>'f'</li><li>'h'</li><li>'g'</li><li>'j'</li><li>'k'</li><li>'o'</li><li>'l'</li><li>'b'</li><li>'ac'</li><li>'p'</li><li>'t'</li><li>'u'</li><li>'x'</li><li>'z'</li><li>'w'</li><li>'v'</li><li>'m'</li><li>'aw'</li><li>'ad'</li><li>'ag'</li><li>'n'</li><li>'af'</li><li>'y'</li><li>'al'</li><li>'ai'</li><li>'ak'</li><li>'aa'</li><li>'as'</li><li>'ab'</li><li>'ao'</li><li>'aq'</li><li>'bc'</li><li>'ce'</li><li>'am'</li><li>'aj'</li><li>'ah'</li><li>'s'</li><li>'cd'</li><li>'ar'</li><li>'au'</li><li>'at'</li><li>'bj'</li><li>'cb'</li><li>'av'</li><li>'ca'</li><li>'ae'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'a'\n",
       "\\item 'c'\n",
       "\\item 'd'\n",
       "\\item 'f'\n",
       "\\item 'h'\n",
       "\\item 'g'\n",
       "\\item 'j'\n",
       "\\item 'k'\n",
       "\\item 'o'\n",
       "\\item 'l'\n",
       "\\item 'b'\n",
       "\\item 'ac'\n",
       "\\item 'p'\n",
       "\\item 't'\n",
       "\\item 'u'\n",
       "\\item 'x'\n",
       "\\item 'z'\n",
       "\\item 'w'\n",
       "\\item 'v'\n",
       "\\item 'm'\n",
       "\\item 'aw'\n",
       "\\item 'ad'\n",
       "\\item 'ag'\n",
       "\\item 'n'\n",
       "\\item 'af'\n",
       "\\item 'y'\n",
       "\\item 'al'\n",
       "\\item 'ai'\n",
       "\\item 'ak'\n",
       "\\item 'aa'\n",
       "\\item 'as'\n",
       "\\item 'ab'\n",
       "\\item 'ao'\n",
       "\\item 'aq'\n",
       "\\item 'bc'\n",
       "\\item 'ce'\n",
       "\\item 'am'\n",
       "\\item 'aj'\n",
       "\\item 'ah'\n",
       "\\item 's'\n",
       "\\item 'cd'\n",
       "\\item 'ar'\n",
       "\\item 'au'\n",
       "\\item 'at'\n",
       "\\item 'bj'\n",
       "\\item 'cb'\n",
       "\\item 'av'\n",
       "\\item 'ca'\n",
       "\\item 'ae'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'a'\n",
       "2. 'c'\n",
       "3. 'd'\n",
       "4. 'f'\n",
       "5. 'h'\n",
       "6. 'g'\n",
       "7. 'j'\n",
       "8. 'k'\n",
       "9. 'o'\n",
       "10. 'l'\n",
       "11. 'b'\n",
       "12. 'ac'\n",
       "13. 'p'\n",
       "14. 't'\n",
       "15. 'u'\n",
       "16. 'x'\n",
       "17. 'z'\n",
       "18. 'w'\n",
       "19. 'v'\n",
       "20. 'm'\n",
       "21. 'aw'\n",
       "22. 'ad'\n",
       "23. 'ag'\n",
       "24. 'n'\n",
       "25. 'af'\n",
       "26. 'y'\n",
       "27. 'al'\n",
       "28. 'ai'\n",
       "29. 'ak'\n",
       "30. 'aa'\n",
       "31. 'as'\n",
       "32. 'ab'\n",
       "33. 'ao'\n",
       "34. 'aq'\n",
       "35. 'bc'\n",
       "36. 'ce'\n",
       "37. 'am'\n",
       "38. 'aj'\n",
       "39. 'ah'\n",
       "40. 's'\n",
       "41. 'cd'\n",
       "42. 'ar'\n",
       "43. 'au'\n",
       "44. 'at'\n",
       "45. 'bj'\n",
       "46. 'cb'\n",
       "47. 'av'\n",
       "48. 'ca'\n",
       "49. 'ae'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"a\"  \"c\"  \"d\"  \"f\"  \"h\"  \"g\"  \"j\"  \"k\"  \"o\"  \"l\"  \"b\"  \"ac\" \"p\"  \"t\"  \"u\" \n",
       "[16] \"x\"  \"z\"  \"w\"  \"v\"  \"m\"  \"aw\" \"ad\" \"ag\" \"n\"  \"af\" \"y\"  \"al\" \"ai\" \"ak\" \"aa\"\n",
       "[31] \"as\" \"ab\" \"ao\" \"aq\" \"bc\" \"ce\" \"am\" \"aj\" \"ah\" \"s\"  \"cd\" \"ar\" \"au\" \"at\" \"bj\"\n",
       "[46] \"cb\" \"av\" \"ca\" \"ae\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique((cell.type.2.markers  %>% filter(heuristic.score > 60))$cell.type.2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "49"
      ],
      "text/latex": [
       "49"
      ],
      "text/markdown": [
       "49"
      ],
      "text/plain": [
       "[1] 49"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(unique((cell.type.2.markers  %>% filter(heuristic.score > 60))$cell.type.2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "metadata": {},
   "outputs": [],
   "source": [
    "colData(cds_sub)$cell.type.2 = rep(NA, ncol(cds_sub))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "metadata": {},
   "outputs": [],
   "source": [
    "colData(cds_sub)$cell.type.2 = df_sub300$cell.type.2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot_cells(cds_sub, color_cells_by = \"cell.type.2\", cell_size = 0.1, : could not find function \"plot_cells\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot_cells(cds_sub, color_cells_by = \"cell.type.2\", cell_size = 0.1, : could not find function \"plot_cells\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot_cells(cds_sub, color_cells_by = \"cell.type.2\", cell_size = 0.1,group_label_size = 5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot_cells(cds_sub, color_cells_by = \"cell.type.2\", cell_size = 0.1, : could not find function \"plot_cells\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot_cells(cds_sub, color_cells_by = \"cell.type.2\", cell_size = 0.1, : could not find function \"plot_cells\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot_cells(cds_sub, color_cells_by = \"cell.type.2\", cell_size = 0.1,group_label_size = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "metadata": {},
   "outputs": [],
   "source": [
    "saveRDS(cds_sub, \"data/cds_fly_sub_PCA300_older_cells_reannotated.rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {},
   "outputs": [],
   "source": [
    "saveRDS(cell.type.2.markers, \"data/fly_sub300_cell.type.2.markers\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>up25</th><td>ay</td><td>up     </td><td>24733</td><td>25245</td><td>13495.945</td><td> 986.13271</td><td>3.773138</td><td>51.76876</td></tr>\n",
       "\t<tr><th scope=row>bt25</th><td>ay</td><td>bt     </td><td>31445</td><td>40455</td><td>16680.929</td><td>1367.51442</td><td>3.607517</td><td>50.59903</td></tr>\n",
       "\t<tr><th scope=row>Sh25</th><td>ay</td><td>Sh     </td><td> 9814</td><td> 8339</td><td> 5337.145</td><td> 356.68246</td><td>3.899317</td><td>48.28182</td></tr>\n",
       "\t<tr><th scope=row>wupA25</th><td>ay</td><td>wupA   </td><td>15322</td><td>18677</td><td> 8226.581</td><td> 697.40159</td><td>3.558164</td><td>46.27838</td></tr>\n",
       "\t<tr><th scope=row>Mhc25</th><td>ay</td><td>Mhc    </td><td>25471</td><td>46980</td><td>13678.470</td><td>1561.99176</td><td>3.129525</td><td>42.99881</td></tr>\n",
       "\t<tr><th scope=row>CG3352125</th><td>ay</td><td>CG33521</td><td> 2727</td><td> 2254</td><td> 1475.741</td><td>  92.42132</td><td>3.981544</td><td>41.91849</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tup25 & ay & up      & 24733 & 25245 & 13495.945 &  986.13271 & 3.773138 & 51.76876\\\\\n",
       "\tbt25 & ay & bt      & 31445 & 40455 & 16680.929 & 1367.51442 & 3.607517 & 50.59903\\\\\n",
       "\tSh25 & ay & Sh      &  9814 &  8339 &  5337.145 &  356.68246 & 3.899317 & 48.28182\\\\\n",
       "\twupA25 & ay & wupA    & 15322 & 18677 &  8226.581 &  697.40159 & 3.558164 & 46.27838\\\\\n",
       "\tMhc25 & ay & Mhc     & 25471 & 46980 & 13678.470 & 1561.99176 & 3.129525 & 42.99881\\\\\n",
       "\tCG3352125 & ay & CG33521 &  2727 &  2254 &  1475.741 &   92.42132 & 3.981544 & 41.91849\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| up25 | ay | up      | 24733 | 25245 | 13495.945 |  986.13271 | 3.773138 | 51.76876 |\n",
       "| bt25 | ay | bt      | 31445 | 40455 | 16680.929 | 1367.51442 | 3.607517 | 50.59903 |\n",
       "| Sh25 | ay | Sh      |  9814 |  8339 |  5337.145 |  356.68246 | 3.899317 | 48.28182 |\n",
       "| wupA25 | ay | wupA    | 15322 | 18677 |  8226.581 |  697.40159 | 3.558164 | 46.27838 |\n",
       "| Mhc25 | ay | Mhc     | 25471 | 46980 | 13678.470 | 1561.99176 | 3.129525 | 42.99881 |\n",
       "| CG3352125 | ay | CG33521 |  2727 |  2254 |  1475.741 |   92.42132 | 3.981544 | 41.91849 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "up25      ay          up      24733             25245       13495.945      \n",
       "bt25      ay          bt      31445             40455       16680.929      \n",
       "Sh25      ay          Sh       9814              8339        5337.145      \n",
       "wupA25    ay          wupA    15322             18677        8226.581      \n",
       "Mhc25     ay          Mhc     25471             46980       13678.470      \n",
       "CG3352125 ay          CG33521  2727              2254        1475.741      \n",
       "          other.tpm  log2.ratio heuristic.score\n",
       "up25       986.13271 3.773138   51.76876       \n",
       "bt25      1367.51442 3.607517   50.59903       \n",
       "Sh25       356.68246 3.899317   48.28182       \n",
       "wupA25     697.40159 3.558164   46.27838       \n",
       "Mhc25     1561.99176 3.129525   42.99881       \n",
       "CG3352125   92.42132 3.981544   41.91849       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers  %>% filter(cell.type.2 == \"ay\")   %>% head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub, gene = \"up\", size_pos = 1): could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub, gene = \"up\", size_pos = 1): could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub, gene = \"up\", size_pos = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'a'</li><li>'aa'</li><li>'ab'</li><li>'ac'</li><li>'ad'</li><li>'ae'</li><li>'af'</li><li>'ag'</li><li>'ah'</li><li>'ai'</li><li>'aj'</li><li>'ak'</li><li>'al'</li><li>'am'</li><li>'an'</li><li>'ao'</li><li>'ap'</li><li>'aq'</li><li>'ar'</li><li>'as'</li><li>'at'</li><li>'au'</li><li>'av'</li><li>'aw'</li><li>'ax'</li><li>'ay'</li><li>'az'</li><li>'b'</li><li>'ba'</li><li>'bb'</li><li>'bc'</li><li>'bd'</li><li>'be'</li><li>'bf'</li><li>'bg'</li><li>'bh'</li><li>'bi'</li><li>'bj'</li><li>'bk'</li><li>'bl'</li><li>'bm'</li><li>'bn'</li><li>'bo'</li><li>'bp'</li><li>'bq'</li><li>'br'</li><li>'bs'</li><li>'bt'</li><li>'bu'</li><li>'bv'</li><li>'bw'</li><li>'bx'</li><li>'by'</li><li>'bz'</li><li>'c'</li><li>'ca'</li><li>'cb'</li><li>'cc'</li><li>'cd'</li><li>'cds_sub'</li><li>'cds_sub_no_doubletons'</li><li>'cds_sub_no_multitons'</li><li>'cds_sub_no_sing'</li><li>'ce'</li><li>'cell_type'</li><li>'cell.type'</li><li>'cell.type.2.markers'</li><li>'cell.type.markers.sub300_no_singletons.cell.type'</li><li>'cell.type.markers.sub300.cell.type'</li><li>'cellIDs'</li><li>'cells'</li><li>'cellTypes'</li><li>'cf'</li><li>'clus'</li><li>'cluster.markers'</li><li>'cluster.markers.sub300'</li><li>'cluster.markers.sub300_no_doubletons'</li><li>'cluster.markers.sub300_no_doubletons_all'</li><li>'cluster.markers.sub300_no_singletons'</li><li>'d'</li><li>'DEG.results'</li><li>'df'</li><li>'df_sub_no_doubletons'</li><li>'df_sub_no_multitons'</li><li>'df_sub_no_singletons'</li><li>'df_sub300'</li><li>'df_sub300_copy'</li><li>'doubletons'</li><li>'e'</li><li>'exp_matrix'</li><li>'exp_matrix_mean'</li><li>'exp_matrix_sub300_no_doubletons'</li><li>'expresses.gene'</li><li>'f'</li><li>'fly_peaks_TF'</li><li>'fly_TFs'</li><li>'g'</li><li>'gene_max_mean_entropy'</li><li>'get.gene.id'</li><li>'get.gene.id2'</li><li>'get.norm.expr.matrix'</li><li>'get.norm.expr.matrix.3'</li><li>'h'</li><li>'i'</li><li>'ids'</li><li>'j'</li><li>'k'</li><li>'k1'</li><li>'l'</li><li>'labels'</li><li>'m'</li><li>'make_df'</li><li>'missing_genes'</li><li>'missing_genes_aliases'</li><li>'multitons'</li><li>'n'</li><li>'nineteentons'</li><li>'nnMatches'</li><li>'o'</li><li>'p'</li><li>'pca'</li><li>'peedeef'</li><li>'plot.expr'</li><li>'plot.expr_clus'</li><li>'plot.expr_clus.br'</li><li>'plot.expr_clus.br2'</li><li>'plot.expr3'</li><li>'q'</li><li>'r'</li><li>'rann'</li><li>'res'</li><li>'res_label'</li><li>'s'</li><li>'set.cell.type'</li><li>'singletons'</li><li>'t'</li><li>'TF_exp_matrix_sub300_no_doubletons'</li><li>'TF_exp_matrix_sub300_no_doubletons_log'</li><li>'two.set.differential.gene.test'</li><li>'two.set.differential.gene.test.3'</li><li>'two.set.differential.gene.test.3a'</li><li>'two.set.differential.gene.test.3b'</li><li>'u'</li><li>'v'</li><li>'vector_angle_emb_targets'</li><li>'w'</li><li>'x'</li><li>'y'</li><li>'z'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'a'\n",
       "\\item 'aa'\n",
       "\\item 'ab'\n",
       "\\item 'ac'\n",
       "\\item 'ad'\n",
       "\\item 'ae'\n",
       "\\item 'af'\n",
       "\\item 'ag'\n",
       "\\item 'ah'\n",
       "\\item 'ai'\n",
       "\\item 'aj'\n",
       "\\item 'ak'\n",
       "\\item 'al'\n",
       "\\item 'am'\n",
       "\\item 'an'\n",
       "\\item 'ao'\n",
       "\\item 'ap'\n",
       "\\item 'aq'\n",
       "\\item 'ar'\n",
       "\\item 'as'\n",
       "\\item 'at'\n",
       "\\item 'au'\n",
       "\\item 'av'\n",
       "\\item 'aw'\n",
       "\\item 'ax'\n",
       "\\item 'ay'\n",
       "\\item 'az'\n",
       "\\item 'b'\n",
       "\\item 'ba'\n",
       "\\item 'bb'\n",
       "\\item 'bc'\n",
       "\\item 'bd'\n",
       "\\item 'be'\n",
       "\\item 'bf'\n",
       "\\item 'bg'\n",
       "\\item 'bh'\n",
       "\\item 'bi'\n",
       "\\item 'bj'\n",
       "\\item 'bk'\n",
       "\\item 'bl'\n",
       "\\item 'bm'\n",
       "\\item 'bn'\n",
       "\\item 'bo'\n",
       "\\item 'bp'\n",
       "\\item 'bq'\n",
       "\\item 'br'\n",
       "\\item 'bs'\n",
       "\\item 'bt'\n",
       "\\item 'bu'\n",
       "\\item 'bv'\n",
       "\\item 'bw'\n",
       "\\item 'bx'\n",
       "\\item 'by'\n",
       "\\item 'bz'\n",
       "\\item 'c'\n",
       "\\item 'ca'\n",
       "\\item 'cb'\n",
       "\\item 'cc'\n",
       "\\item 'cd'\n",
       "\\item 'cds\\_sub'\n",
       "\\item 'cds\\_sub\\_no\\_doubletons'\n",
       "\\item 'cds\\_sub\\_no\\_multitons'\n",
       "\\item 'cds\\_sub\\_no\\_sing'\n",
       "\\item 'ce'\n",
       "\\item 'cell\\_type'\n",
       "\\item 'cell.type'\n",
       "\\item 'cell.type.2.markers'\n",
       "\\item 'cell.type.markers.sub300\\_no\\_singletons.cell.type'\n",
       "\\item 'cell.type.markers.sub300.cell.type'\n",
       "\\item 'cellIDs'\n",
       "\\item 'cells'\n",
       "\\item 'cellTypes'\n",
       "\\item 'cf'\n",
       "\\item 'clus'\n",
       "\\item 'cluster.markers'\n",
       "\\item 'cluster.markers.sub300'\n",
       "\\item 'cluster.markers.sub300\\_no\\_doubletons'\n",
       "\\item 'cluster.markers.sub300\\_no\\_doubletons\\_all'\n",
       "\\item 'cluster.markers.sub300\\_no\\_singletons'\n",
       "\\item 'd'\n",
       "\\item 'DEG.results'\n",
       "\\item 'df'\n",
       "\\item 'df\\_sub\\_no\\_doubletons'\n",
       "\\item 'df\\_sub\\_no\\_multitons'\n",
       "\\item 'df\\_sub\\_no\\_singletons'\n",
       "\\item 'df\\_sub300'\n",
       "\\item 'df\\_sub300\\_copy'\n",
       "\\item 'doubletons'\n",
       "\\item 'e'\n",
       "\\item 'exp\\_matrix'\n",
       "\\item 'exp\\_matrix\\_mean'\n",
       "\\item 'exp\\_matrix\\_sub300\\_no\\_doubletons'\n",
       "\\item 'expresses.gene'\n",
       "\\item 'f'\n",
       "\\item 'fly\\_peaks\\_TF'\n",
       "\\item 'fly\\_TFs'\n",
       "\\item 'g'\n",
       "\\item 'gene\\_max\\_mean\\_entropy'\n",
       "\\item 'get.gene.id'\n",
       "\\item 'get.gene.id2'\n",
       "\\item 'get.norm.expr.matrix'\n",
       "\\item 'get.norm.expr.matrix.3'\n",
       "\\item 'h'\n",
       "\\item 'i'\n",
       "\\item 'ids'\n",
       "\\item 'j'\n",
       "\\item 'k'\n",
       "\\item 'k1'\n",
       "\\item 'l'\n",
       "\\item 'labels'\n",
       "\\item 'm'\n",
       "\\item 'make\\_df'\n",
       "\\item 'missing\\_genes'\n",
       "\\item 'missing\\_genes\\_aliases'\n",
       "\\item 'multitons'\n",
       "\\item 'n'\n",
       "\\item 'nineteentons'\n",
       "\\item 'nnMatches'\n",
       "\\item 'o'\n",
       "\\item 'p'\n",
       "\\item 'pca'\n",
       "\\item 'peedeef'\n",
       "\\item 'plot.expr'\n",
       "\\item 'plot.expr\\_clus'\n",
       "\\item 'plot.expr\\_clus.br'\n",
       "\\item 'plot.expr\\_clus.br2'\n",
       "\\item 'plot.expr3'\n",
       "\\item 'q'\n",
       "\\item 'r'\n",
       "\\item 'rann'\n",
       "\\item 'res'\n",
       "\\item 'res\\_label'\n",
       "\\item 's'\n",
       "\\item 'set.cell.type'\n",
       "\\item 'singletons'\n",
       "\\item 't'\n",
       "\\item 'TF\\_exp\\_matrix\\_sub300\\_no\\_doubletons'\n",
       "\\item 'TF\\_exp\\_matrix\\_sub300\\_no\\_doubletons\\_log'\n",
       "\\item 'two.set.differential.gene.test'\n",
       "\\item 'two.set.differential.gene.test.3'\n",
       "\\item 'two.set.differential.gene.test.3a'\n",
       "\\item 'two.set.differential.gene.test.3b'\n",
       "\\item 'u'\n",
       "\\item 'v'\n",
       "\\item 'vector\\_angle\\_emb\\_targets'\n",
       "\\item 'w'\n",
       "\\item 'x'\n",
       "\\item 'y'\n",
       "\\item 'z'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'a'\n",
       "2. 'aa'\n",
       "3. 'ab'\n",
       "4. 'ac'\n",
       "5. 'ad'\n",
       "6. 'ae'\n",
       "7. 'af'\n",
       "8. 'ag'\n",
       "9. 'ah'\n",
       "10. 'ai'\n",
       "11. 'aj'\n",
       "12. 'ak'\n",
       "13. 'al'\n",
       "14. 'am'\n",
       "15. 'an'\n",
       "16. 'ao'\n",
       "17. 'ap'\n",
       "18. 'aq'\n",
       "19. 'ar'\n",
       "20. 'as'\n",
       "21. 'at'\n",
       "22. 'au'\n",
       "23. 'av'\n",
       "24. 'aw'\n",
       "25. 'ax'\n",
       "26. 'ay'\n",
       "27. 'az'\n",
       "28. 'b'\n",
       "29. 'ba'\n",
       "30. 'bb'\n",
       "31. 'bc'\n",
       "32. 'bd'\n",
       "33. 'be'\n",
       "34. 'bf'\n",
       "35. 'bg'\n",
       "36. 'bh'\n",
       "37. 'bi'\n",
       "38. 'bj'\n",
       "39. 'bk'\n",
       "40. 'bl'\n",
       "41. 'bm'\n",
       "42. 'bn'\n",
       "43. 'bo'\n",
       "44. 'bp'\n",
       "45. 'bq'\n",
       "46. 'br'\n",
       "47. 'bs'\n",
       "48. 'bt'\n",
       "49. 'bu'\n",
       "50. 'bv'\n",
       "51. 'bw'\n",
       "52. 'bx'\n",
       "53. 'by'\n",
       "54. 'bz'\n",
       "55. 'c'\n",
       "56. 'ca'\n",
       "57. 'cb'\n",
       "58. 'cc'\n",
       "59. 'cd'\n",
       "60. 'cds_sub'\n",
       "61. 'cds_sub_no_doubletons'\n",
       "62. 'cds_sub_no_multitons'\n",
       "63. 'cds_sub_no_sing'\n",
       "64. 'ce'\n",
       "65. 'cell_type'\n",
       "66. 'cell.type'\n",
       "67. 'cell.type.2.markers'\n",
       "68. 'cell.type.markers.sub300_no_singletons.cell.type'\n",
       "69. 'cell.type.markers.sub300.cell.type'\n",
       "70. 'cellIDs'\n",
       "71. 'cells'\n",
       "72. 'cellTypes'\n",
       "73. 'cf'\n",
       "74. 'clus'\n",
       "75. 'cluster.markers'\n",
       "76. 'cluster.markers.sub300'\n",
       "77. 'cluster.markers.sub300_no_doubletons'\n",
       "78. 'cluster.markers.sub300_no_doubletons_all'\n",
       "79. 'cluster.markers.sub300_no_singletons'\n",
       "80. 'd'\n",
       "81. 'DEG.results'\n",
       "82. 'df'\n",
       "83. 'df_sub_no_doubletons'\n",
       "84. 'df_sub_no_multitons'\n",
       "85. 'df_sub_no_singletons'\n",
       "86. 'df_sub300'\n",
       "87. 'df_sub300_copy'\n",
       "88. 'doubletons'\n",
       "89. 'e'\n",
       "90. 'exp_matrix'\n",
       "91. 'exp_matrix_mean'\n",
       "92. 'exp_matrix_sub300_no_doubletons'\n",
       "93. 'expresses.gene'\n",
       "94. 'f'\n",
       "95. 'fly_peaks_TF'\n",
       "96. 'fly_TFs'\n",
       "97. 'g'\n",
       "98. 'gene_max_mean_entropy'\n",
       "99. 'get.gene.id'\n",
       "100. 'get.gene.id2'\n",
       "101. 'get.norm.expr.matrix'\n",
       "102. 'get.norm.expr.matrix.3'\n",
       "103. 'h'\n",
       "104. 'i'\n",
       "105. 'ids'\n",
       "106. 'j'\n",
       "107. 'k'\n",
       "108. 'k1'\n",
       "109. 'l'\n",
       "110. 'labels'\n",
       "111. 'm'\n",
       "112. 'make_df'\n",
       "113. 'missing_genes'\n",
       "114. 'missing_genes_aliases'\n",
       "115. 'multitons'\n",
       "116. 'n'\n",
       "117. 'nineteentons'\n",
       "118. 'nnMatches'\n",
       "119. 'o'\n",
       "120. 'p'\n",
       "121. 'pca'\n",
       "122. 'peedeef'\n",
       "123. 'plot.expr'\n",
       "124. 'plot.expr_clus'\n",
       "125. 'plot.expr_clus.br'\n",
       "126. 'plot.expr_clus.br2'\n",
       "127. 'plot.expr3'\n",
       "128. 'q'\n",
       "129. 'r'\n",
       "130. 'rann'\n",
       "131. 'res'\n",
       "132. 'res_label'\n",
       "133. 's'\n",
       "134. 'set.cell.type'\n",
       "135. 'singletons'\n",
       "136. 't'\n",
       "137. 'TF_exp_matrix_sub300_no_doubletons'\n",
       "138. 'TF_exp_matrix_sub300_no_doubletons_log'\n",
       "139. 'two.set.differential.gene.test'\n",
       "140. 'two.set.differential.gene.test.3'\n",
       "141. 'two.set.differential.gene.test.3a'\n",
       "142. 'two.set.differential.gene.test.3b'\n",
       "143. 'u'\n",
       "144. 'v'\n",
       "145. 'vector_angle_emb_targets'\n",
       "146. 'w'\n",
       "147. 'x'\n",
       "148. 'y'\n",
       "149. 'z'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"a\"                                               \n",
       "  [2] \"aa\"                                              \n",
       "  [3] \"ab\"                                              \n",
       "  [4] \"ac\"                                              \n",
       "  [5] \"ad\"                                              \n",
       "  [6] \"ae\"                                              \n",
       "  [7] \"af\"                                              \n",
       "  [8] \"ag\"                                              \n",
       "  [9] \"ah\"                                              \n",
       " [10] \"ai\"                                              \n",
       " [11] \"aj\"                                              \n",
       " [12] \"ak\"                                              \n",
       " [13] \"al\"                                              \n",
       " [14] \"am\"                                              \n",
       " [15] \"an\"                                              \n",
       " [16] \"ao\"                                              \n",
       " [17] \"ap\"                                              \n",
       " [18] \"aq\"                                              \n",
       " [19] \"ar\"                                              \n",
       " [20] \"as\"                                              \n",
       " [21] \"at\"                                              \n",
       " [22] \"au\"                                              \n",
       " [23] \"av\"                                              \n",
       " [24] \"aw\"                                              \n",
       " [25] \"ax\"                                              \n",
       " [26] \"ay\"                                              \n",
       " [27] \"az\"                                              \n",
       " [28] \"b\"                                               \n",
       " [29] \"ba\"                                              \n",
       " [30] \"bb\"                                              \n",
       " [31] \"bc\"                                              \n",
       " [32] \"bd\"                                              \n",
       " [33] \"be\"                                              \n",
       " [34] \"bf\"                                              \n",
       " [35] \"bg\"                                              \n",
       " [36] \"bh\"                                              \n",
       " [37] \"bi\"                                              \n",
       " [38] \"bj\"                                              \n",
       " [39] \"bk\"                                              \n",
       " [40] \"bl\"                                              \n",
       " [41] \"bm\"                                              \n",
       " [42] \"bn\"                                              \n",
       " [43] \"bo\"                                              \n",
       " [44] \"bp\"                                              \n",
       " [45] \"bq\"                                              \n",
       " [46] \"br\"                                              \n",
       " [47] \"bs\"                                              \n",
       " [48] \"bt\"                                              \n",
       " [49] \"bu\"                                              \n",
       " [50] \"bv\"                                              \n",
       " [51] \"bw\"                                              \n",
       " [52] \"bx\"                                              \n",
       " [53] \"by\"                                              \n",
       " [54] \"bz\"                                              \n",
       " [55] \"c\"                                               \n",
       " [56] \"ca\"                                              \n",
       " [57] \"cb\"                                              \n",
       " [58] \"cc\"                                              \n",
       " [59] \"cd\"                                              \n",
       " [60] \"cds_sub\"                                         \n",
       " [61] \"cds_sub_no_doubletons\"                           \n",
       " [62] \"cds_sub_no_multitons\"                            \n",
       " [63] \"cds_sub_no_sing\"                                 \n",
       " [64] \"ce\"                                              \n",
       " [65] \"cell_type\"                                       \n",
       " [66] \"cell.type\"                                       \n",
       " [67] \"cell.type.2.markers\"                             \n",
       " [68] \"cell.type.markers.sub300_no_singletons.cell.type\"\n",
       " [69] \"cell.type.markers.sub300.cell.type\"              \n",
       " [70] \"cellIDs\"                                         \n",
       " [71] \"cells\"                                           \n",
       " [72] \"cellTypes\"                                       \n",
       " [73] \"cf\"                                              \n",
       " [74] \"clus\"                                            \n",
       " [75] \"cluster.markers\"                                 \n",
       " [76] \"cluster.markers.sub300\"                          \n",
       " [77] \"cluster.markers.sub300_no_doubletons\"            \n",
       " [78] \"cluster.markers.sub300_no_doubletons_all\"        \n",
       " [79] \"cluster.markers.sub300_no_singletons\"            \n",
       " [80] \"d\"                                               \n",
       " [81] \"DEG.results\"                                     \n",
       " [82] \"df\"                                              \n",
       " [83] \"df_sub_no_doubletons\"                            \n",
       " [84] \"df_sub_no_multitons\"                             \n",
       " [85] \"df_sub_no_singletons\"                            \n",
       " [86] \"df_sub300\"                                       \n",
       " [87] \"df_sub300_copy\"                                  \n",
       " [88] \"doubletons\"                                      \n",
       " [89] \"e\"                                               \n",
       " [90] \"exp_matrix\"                                      \n",
       " [91] \"exp_matrix_mean\"                                 \n",
       " [92] \"exp_matrix_sub300_no_doubletons\"                 \n",
       " [93] \"expresses.gene\"                                  \n",
       " [94] \"f\"                                               \n",
       " [95] \"fly_peaks_TF\"                                    \n",
       " [96] \"fly_TFs\"                                         \n",
       " [97] \"g\"                                               \n",
       " [98] \"gene_max_mean_entropy\"                           \n",
       " [99] \"get.gene.id\"                                     \n",
       "[100] \"get.gene.id2\"                                    \n",
       "[101] \"get.norm.expr.matrix\"                            \n",
       "[102] \"get.norm.expr.matrix.3\"                          \n",
       "[103] \"h\"                                               \n",
       "[104] \"i\"                                               \n",
       "[105] \"ids\"                                             \n",
       "[106] \"j\"                                               \n",
       "[107] \"k\"                                               \n",
       "[108] \"k1\"                                              \n",
       "[109] \"l\"                                               \n",
       "[110] \"labels\"                                          \n",
       "[111] \"m\"                                               \n",
       "[112] \"make_df\"                                         \n",
       "[113] \"missing_genes\"                                   \n",
       "[114] \"missing_genes_aliases\"                           \n",
       "[115] \"multitons\"                                       \n",
       "[116] \"n\"                                               \n",
       "[117] \"nineteentons\"                                    \n",
       "[118] \"nnMatches\"                                       \n",
       "[119] \"o\"                                               \n",
       "[120] \"p\"                                               \n",
       "[121] \"pca\"                                             \n",
       "[122] \"peedeef\"                                         \n",
       "[123] \"plot.expr\"                                       \n",
       "[124] \"plot.expr_clus\"                                  \n",
       "[125] \"plot.expr_clus.br\"                               \n",
       "[126] \"plot.expr_clus.br2\"                              \n",
       "[127] \"plot.expr3\"                                      \n",
       "[128] \"q\"                                               \n",
       "[129] \"r\"                                               \n",
       "[130] \"rann\"                                            \n",
       "[131] \"res\"                                             \n",
       "[132] \"res_label\"                                       \n",
       "[133] \"s\"                                               \n",
       "[134] \"set.cell.type\"                                   \n",
       "[135] \"singletons\"                                      \n",
       "[136] \"t\"                                               \n",
       "[137] \"TF_exp_matrix_sub300_no_doubletons\"              \n",
       "[138] \"TF_exp_matrix_sub300_no_doubletons_log\"          \n",
       "[139] \"two.set.differential.gene.test\"                  \n",
       "[140] \"two.set.differential.gene.test.3\"                \n",
       "[141] \"two.set.differential.gene.test.3a\"               \n",
       "[142] \"two.set.differential.gene.test.3b\"               \n",
       "[143] \"u\"                                               \n",
       "[144] \"v\"                                               \n",
       "[145] \"vector_angle_emb_targets\"                        \n",
       "[146] \"w\"                                               \n",
       "[147] \"x\"                                               \n",
       "[148] \"y\"                                               \n",
       "[149] \"z\"                                               "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ls()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1907,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'cell.type.2.markers'</li><li>'cell.type.markers.sub300_no_singletons.cell.type'</li><li>'cell.type.markers.sub300.cell.type'</li><li>'cluster.markers'</li><li>'cluster.markers.sub300'</li><li>'cluster.markers.sub300_no_doubletons'</li><li>'cluster.markers.sub300_no_doubletons_all'</li><li>'cluster.markers.sub300_no_singletons'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'cell.type.2.markers'\n",
       "\\item 'cell.type.markers.sub300\\_no\\_singletons.cell.type'\n",
       "\\item 'cell.type.markers.sub300.cell.type'\n",
       "\\item 'cluster.markers'\n",
       "\\item 'cluster.markers.sub300'\n",
       "\\item 'cluster.markers.sub300\\_no\\_doubletons'\n",
       "\\item 'cluster.markers.sub300\\_no\\_doubletons\\_all'\n",
       "\\item 'cluster.markers.sub300\\_no\\_singletons'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'cell.type.2.markers'\n",
       "2. 'cell.type.markers.sub300_no_singletons.cell.type'\n",
       "3. 'cell.type.markers.sub300.cell.type'\n",
       "4. 'cluster.markers'\n",
       "5. 'cluster.markers.sub300'\n",
       "6. 'cluster.markers.sub300_no_doubletons'\n",
       "7. 'cluster.markers.sub300_no_doubletons_all'\n",
       "8. 'cluster.markers.sub300_no_singletons'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"cell.type.2.markers\"                             \n",
       "[2] \"cell.type.markers.sub300_no_singletons.cell.type\"\n",
       "[3] \"cell.type.markers.sub300.cell.type\"              \n",
       "[4] \"cluster.markers\"                                 \n",
       "[5] \"cluster.markers.sub300\"                          \n",
       "[6] \"cluster.markers.sub300_no_doubletons\"            \n",
       "[7] \"cluster.markers.sub300_no_doubletons_all\"        \n",
       "[8] \"cluster.markers.sub300_no_singletons\"            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ls(pattern = \"markers\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1909,
   "metadata": {},
   "outputs": [],
   "source": [
    "save(cell.type.2.markers, cell.type.markers.sub300_no_singletons.cell.type, \n",
    "     cell.type.markers.sub300.cell.type, cluster.markers, cluster.markers.sub300, \n",
    "     cluster.markers.sub300_no_doubletons, cluster.markers.sub300_no_doubletons_all, \n",
    "     cluster.markers.sub300_no_singletons, file = \"data/markers for fly datasets.RData\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1912,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'cds_pcg'</li><li>'cds_sub'</li><li>'cds_sub_no_doubletons'</li><li>'cds_sub_no_multitons'</li><li>'cds_sub_no_sing'</li><li>'cds_worm'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'cds\\_pcg'\n",
       "\\item 'cds\\_sub'\n",
       "\\item 'cds\\_sub\\_no\\_doubletons'\n",
       "\\item 'cds\\_sub\\_no\\_multitons'\n",
       "\\item 'cds\\_sub\\_no\\_sing'\n",
       "\\item 'cds\\_worm'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'cds_pcg'\n",
       "2. 'cds_sub'\n",
       "3. 'cds_sub_no_doubletons'\n",
       "4. 'cds_sub_no_multitons'\n",
       "5. 'cds_sub_no_sing'\n",
       "6. 'cds_worm'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"cds_pcg\"               \"cds_sub\"               \"cds_sub_no_doubletons\"\n",
       "[4] \"cds_sub_no_multitons\"  \"cds_sub_no_sing\"       \"cds_worm\"             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ls(pattern = \"cds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1913,
   "metadata": {},
   "outputs": [],
   "source": [
    "saveRDS(cds_sub_no_doubletons, \"data/cds_fly_pca300_no_doubletons,rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1914,
   "metadata": {},
   "outputs": [],
   "source": [
    "saveRDS(cds_sub_no_sing, \"data/cds_fly_pca300_no_doubletons,rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remake with new cell.type labels\n",
    "cds_sub_no_sing = cds_sub[, !(df_sub300$cell %in% doubletons)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remake with new cell.type labels\n",
    "cds_sub_no_doubletons = cds_sub_no_sing[, !(df_sub_no_singletons$cell %in% doubletons)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "metadata": {},
   "outputs": [],
   "source": [
    "saveRDS(cds_sub_no_doubletons, \"data/cds_fly_pca300_no_doubletons,rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>13938</li><li>197447</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 13938\n",
       "\\item 197447\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 13938\n",
       "2. 197447\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1]  13938 197447"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(cds_sub_no_doubletons)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## DE genes on new annotation no doubletons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remake with new cell.type labels\n",
    "df_sub_no_doubletons  = make_df(cds_sub_no_doubletons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finding markers for cell.type a\n",
      "\n",
      "# of cells sampled in set 1: 432\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type aa\n",
      "\n",
      "# of cells sampled in set 1: 1540\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ab\n",
      "\n",
      "# of cells sampled in set 1: 1048\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ac\n",
      "\n",
      "# of cells sampled in set 1: 3765\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ad\n",
      "\n",
      "# of cells sampled in set 1: 719\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ae\n",
      "\n",
      "# of cells sampled in set 1: 1019\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type af\n",
      "\n",
      "# of cells sampled in set 1: 94\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ag\n",
      "\n",
      "# of cells sampled in set 1: 495\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ah\n",
      "\n",
      "# of cells sampled in set 1: 2347\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ai\n",
      "\n",
      "# of cells sampled in set 1: 1265\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type aj\n",
      "\n",
      "# of cells sampled in set 1: 1760\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ak\n",
      "\n",
      "# of cells sampled in set 1: 517\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type al\n",
      "\n",
      "# of cells sampled in set 1: 2535\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type am\n",
      "\n",
      "# of cells sampled in set 1: 1521\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type an\n",
      "\n",
      "# of cells sampled in set 1: 1785\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ao\n",
      "\n",
      "# of cells sampled in set 1: 502\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ap\n",
      "\n",
      "# of cells sampled in set 1: 2022\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type aq\n",
      "\n",
      "# of cells sampled in set 1: 1136\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ar\n",
      "\n",
      "# of cells sampled in set 1: 3047\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type as\n",
      "\n",
      "# of cells sampled in set 1: 929\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type at\n",
      "\n",
      "# of cells sampled in set 1: 1796\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type au\n",
      "\n",
      "# of cells sampled in set 1: 1347\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type av\n",
      "\n",
      "# of cells sampled in set 1: 1689\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type aw\n",
      "\n",
      "# of cells sampled in set 1: 285\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ax\n",
      "\n",
      "# of cells sampled in set 1: 3395\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ay\n",
      "\n",
      "# of cells sampled in set 1: 4748\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type az\n",
      "\n",
      "# of cells sampled in set 1: 4763\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type b\n",
      "\n",
      "# of cells sampled in set 1: 1397\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ba\n",
      "\n",
      "# of cells sampled in set 1: 1864\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bb\n",
      "\n",
      "# of cells sampled in set 1: 1495\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bc\n",
      "\n",
      "# of cells sampled in set 1: 1216\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bd\n",
      "\n",
      "# of cells sampled in set 1: 777\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type be\n",
      "\n",
      "# of cells sampled in set 1: 3819\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bf\n",
      "\n",
      "# of cells sampled in set 1: 1412\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bg\n",
      "\n",
      "# of cells sampled in set 1: 1273\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bh\n",
      "\n",
      "# of cells sampled in set 1: 2563\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bi\n",
      "\n",
      "# of cells sampled in set 1: 840\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bj\n",
      "\n",
      "# of cells sampled in set 1: 4166\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bk\n",
      "\n",
      "# of cells sampled in set 1: 5594\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bl\n",
      "\n",
      "# of cells sampled in set 1: 6935\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bm\n",
      "\n",
      "# of cells sampled in set 1: 2842\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bn\n",
      "\n",
      "# of cells sampled in set 1: 1432\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bo\n",
      "\n",
      "# of cells sampled in set 1: 2124\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bp\n",
      "\n",
      "# of cells sampled in set 1: 6249\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bq\n",
      "\n",
      "# of cells sampled in set 1: 7495\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type br\n",
      "\n",
      "# of cells sampled in set 1: 3347\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bs\n",
      "\n",
      "# of cells sampled in set 1: 3501\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bt\n",
      "\n",
      "# of cells sampled in set 1: 7467\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bu\n",
      "\n",
      "# of cells sampled in set 1: 3012\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bv\n",
      "\n",
      "# of cells sampled in set 1: 2544\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bw\n",
      "\n",
      "# of cells sampled in set 1: 2384\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bx\n",
      "\n",
      "# of cells sampled in set 1: 2263\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type by\n",
      "\n",
      "# of cells sampled in set 1: 1629\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bz\n",
      "\n",
      "# of cells sampled in set 1: 1861\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type c\n",
      "\n",
      "# of cells sampled in set 1: 96\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ca\n",
      "\n",
      "# of cells sampled in set 1: 3298\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type cb\n",
      "\n",
      "# of cells sampled in set 1: 4006\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type cc\n",
      "\n",
      "# of cells sampled in set 1: 6735\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type cd\n",
      "\n",
      "# of cells sampled in set 1: 2267\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ce\n",
      "\n",
      "# of cells sampled in set 1: 1077\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type cf\n",
      "\n",
      "# of cells sampled in set 1: 592\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type d\n",
      "\n",
      "# of cells sampled in set 1: 178\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type e\n",
      "\n",
      "# of cells sampled in set 1: 4655\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type f\n",
      "\n",
      "# of cells sampled in set 1: 1408\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type g\n",
      "\n",
      "# of cells sampled in set 1: 667\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type h\n",
      "\n",
      "# of cells sampled in set 1: 356\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type i\n",
      "\n",
      "# of cells sampled in set 1: 8848\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type j\n",
      "\n",
      "# of cells sampled in set 1: 163\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type k\n",
      "\n",
      "# of cells sampled in set 1: 50\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type l\n",
      "\n",
      "# of cells sampled in set 1: 1566\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type m\n",
      "\n",
      "# of cells sampled in set 1: 2650\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type n\n",
      "\n",
      "# of cells sampled in set 1: 2285\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type o\n",
      "\n",
      "# of cells sampled in set 1: 2055\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type p\n",
      "\n",
      "# of cells sampled in set 1: 123\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type q\n",
      "\n",
      "# of cells sampled in set 1: 8835\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type r\n",
      "\n",
      "# of cells sampled in set 1: 3182\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type s\n",
      "\n",
      "# of cells sampled in set 1: 4153\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type t\n",
      "\n",
      "# of cells sampled in set 1: 918\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type u\n",
      "\n",
      "# of cells sampled in set 1: 505\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type v\n",
      "\n",
      "# of cells sampled in set 1: 5308\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type w\n",
      "\n",
      "# of cells sampled in set 1: 628\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type x\n",
      "\n",
      "# of cells sampled in set 1: 512\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type y\n",
      "\n",
      "# of cells sampled in set 1: 1366\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type z\n",
      "\n",
      "# of cells sampled in set 1: 963\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    }
   ],
   "source": [
    "DEG.results_no_doubletons = lapply(sort((unique(df_sub_no_doubletons$cell.type.2))),\n",
    "                     function(this.cell.type) {\n",
    "    message(\"Finding markers for cell.type \", this.cell.type)\n",
    "    cbind(\n",
    "        two.set.differential.gene.test.3a(\n",
    "            cds_sub_no_doubletons,\n",
    "            df_sub_no_doubletons$cell.type.2 == this.cell.type,\n",
    "            df_sub_no_doubletons$cell.type.2 != this.cell.type,\n",
    "            sample.n.max = 50000),\n",
    "        data.frame(cell.type.2 = this.cell.type))\n",
    "                         })"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "metadata": {},
   "outputs": [],
   "source": [
    "cell.type.2.markers_no_doubletons = do.call(rbind, DEG.results_no_doubletons)  %>%\n",
    "    select(\n",
    "        cell.type.2 = cell.type.2, gene, \n",
    "        cell.type.2.n.umi = set.1.n.umi,\n",
    "        other.n.umi = set.2.n.umi,\n",
    "        cell.type.2.tpm = set.1.tpm,\n",
    "        other.tpm = set.2.tpm,\n",
    "        log2.ratio, heuristic.score) %>%\n",
    "    arrange(-heuristic.score)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 274,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Muc26B</th><td>a</td><td>Muc26B </td><td> 3651</td><td>121</td><td>11236.035</td><td> 3.9689778</td><td>11.142896</td><td>149.9385</td></tr>\n",
       "\t<tr><th scope=row>CG7017</th><td>a</td><td>CG7017 </td><td> 3415</td><td>236</td><td>12317.772</td><td> 9.9390129</td><td>10.137043</td><td>137.7479</td></tr>\n",
       "\t<tr><th scope=row>CG740954</th><td>c</td><td>CG7409 </td><td>  458</td><td> 16</td><td> 3866.913</td><td> 0.3382439</td><td>11.496625</td><td>137.0092</td></tr>\n",
       "\t<tr><th scope=row>CG1137061</th><td>d</td><td>CG11370</td><td> 2890</td><td>826</td><td>16311.339</td><td>18.7725763</td><td> 9.688159</td><td>135.5730</td></tr>\n",
       "\t<tr><th scope=row>CG746563</th><td>f</td><td>CG7465 </td><td>20683</td><td>922</td><td>18938.232</td><td>27.8101498</td><td> 9.360509</td><td>133.0043</td></tr>\n",
       "\t<tr><th scope=row>CG3260363</th><td>f</td><td>CG32603</td><td> 9367</td><td>299</td><td> 9354.678</td><td> 8.3255405</td><td> 9.970285</td><td>131.5243</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tMuc26B & a & Muc26B  &  3651 & 121 & 11236.035 &  3.9689778 & 11.142896 & 149.9385\\\\\n",
       "\tCG7017 & a & CG7017  &  3415 & 236 & 12317.772 &  9.9390129 & 10.137043 & 137.7479\\\\\n",
       "\tCG740954 & c & CG7409  &   458 &  16 &  3866.913 &  0.3382439 & 11.496625 & 137.0092\\\\\n",
       "\tCG1137061 & d & CG11370 &  2890 & 826 & 16311.339 & 18.7725763 &  9.688159 & 135.5730\\\\\n",
       "\tCG746563 & f & CG7465  & 20683 & 922 & 18938.232 & 27.8101498 &  9.360509 & 133.0043\\\\\n",
       "\tCG3260363 & f & CG32603 &  9367 & 299 &  9354.678 &  8.3255405 &  9.970285 & 131.5243\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Muc26B | a | Muc26B  |  3651 | 121 | 11236.035 |  3.9689778 | 11.142896 | 149.9385 |\n",
       "| CG7017 | a | CG7017  |  3415 | 236 | 12317.772 |  9.9390129 | 10.137043 | 137.7479 |\n",
       "| CG740954 | c | CG7409  |   458 |  16 |  3866.913 |  0.3382439 | 11.496625 | 137.0092 |\n",
       "| CG1137061 | d | CG11370 |  2890 | 826 | 16311.339 | 18.7725763 |  9.688159 | 135.5730 |\n",
       "| CG746563 | f | CG7465  | 20683 | 922 | 18938.232 | 27.8101498 |  9.360509 | 133.0043 |\n",
       "| CG3260363 | f | CG32603 |  9367 | 299 |  9354.678 |  8.3255405 |  9.970285 | 131.5243 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "Muc26B    a           Muc26B   3651             121         11236.035      \n",
       "CG7017    a           CG7017   3415             236         12317.772      \n",
       "CG740954  c           CG7409    458              16          3866.913      \n",
       "CG1137061 d           CG11370  2890             826         16311.339      \n",
       "CG746563  f           CG7465  20683             922         18938.232      \n",
       "CG3260363 f           CG32603  9367             299          9354.678      \n",
       "          other.tpm  log2.ratio heuristic.score\n",
       "Muc26B     3.9689778 11.142896  149.9385       \n",
       "CG7017     9.9390129 10.137043  137.7479       \n",
       "CG740954   0.3382439 11.496625  137.0092       \n",
       "CG1137061 18.7725763  9.688159  135.5730       \n",
       "CG746563  27.8101498  9.360509  133.0043       \n",
       "CG3260363  8.3255405  9.970285  131.5243       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 275,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>416</li><li>8</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 416\n",
       "\\item 8\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 416\n",
       "2. 8\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 416   8"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(heuristic.score > 60)   %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 283,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2123</li><li>8</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2123\n",
       "\\item 8\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2123\n",
       "2. 8\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2123    8"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(heuristic.score > 40)   %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 285,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>10</li><li>8</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 10\n",
       "\\item 8\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 10\n",
       "2. 8\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 10  8"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(heuristic.score > 60 & gene %in% fly_TFs)  %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 286,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>113</li><li>8</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 113\n",
       "\\item 8\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 113\n",
       "2. 8\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 113   8"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(heuristic.score > 40 & gene %in% fly_TFs)  %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 288,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "51"
      ],
      "text/latex": [
       "51"
      ],
      "text/markdown": [
       "51"
      ],
      "text/plain": [
       "[1] 51"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(unique((cell.type.2.markers_no_doubletons  %>% filter(heuristic.score > 60))$cell.type.2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 455,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'a'</li><li>'aa'</li><li>'ab'</li><li>'ac'</li><li>'ad'</li><li>'ae'</li><li>'af'</li><li>'ag'</li><li>'ah'</li><li>'ai'</li><li>'aj'</li><li>'ak'</li><li>'al'</li><li>'am'</li><li>'ao'</li><li>'ap'</li><li>'aq'</li><li>'ar'</li><li>'as'</li><li>'at'</li><li>'au'</li><li>'av'</li><li>'aw'</li><li>'b'</li><li>'bc'</li><li>'bj'</li><li>'c'</li><li>'ca'</li><li>'cb'</li><li>'cc'</li><li>'cd'</li><li>'ce'</li><li>'d'</li><li>'f'</li><li>'g'</li><li>'h'</li><li>'j'</li><li>'k'</li><li>'l'</li><li>'m'</li><li>'n'</li><li>'o'</li><li>'p'</li><li>'s'</li><li>'t'</li><li>'u'</li><li>'v'</li><li>'w'</li><li>'x'</li><li>'y'</li><li>'z'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'a'\n",
       "\\item 'aa'\n",
       "\\item 'ab'\n",
       "\\item 'ac'\n",
       "\\item 'ad'\n",
       "\\item 'ae'\n",
       "\\item 'af'\n",
       "\\item 'ag'\n",
       "\\item 'ah'\n",
       "\\item 'ai'\n",
       "\\item 'aj'\n",
       "\\item 'ak'\n",
       "\\item 'al'\n",
       "\\item 'am'\n",
       "\\item 'ao'\n",
       "\\item 'ap'\n",
       "\\item 'aq'\n",
       "\\item 'ar'\n",
       "\\item 'as'\n",
       "\\item 'at'\n",
       "\\item 'au'\n",
       "\\item 'av'\n",
       "\\item 'aw'\n",
       "\\item 'b'\n",
       "\\item 'bc'\n",
       "\\item 'bj'\n",
       "\\item 'c'\n",
       "\\item 'ca'\n",
       "\\item 'cb'\n",
       "\\item 'cc'\n",
       "\\item 'cd'\n",
       "\\item 'ce'\n",
       "\\item 'd'\n",
       "\\item 'f'\n",
       "\\item 'g'\n",
       "\\item 'h'\n",
       "\\item 'j'\n",
       "\\item 'k'\n",
       "\\item 'l'\n",
       "\\item 'm'\n",
       "\\item 'n'\n",
       "\\item 'o'\n",
       "\\item 'p'\n",
       "\\item 's'\n",
       "\\item 't'\n",
       "\\item 'u'\n",
       "\\item 'v'\n",
       "\\item 'w'\n",
       "\\item 'x'\n",
       "\\item 'y'\n",
       "\\item 'z'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'a'\n",
       "2. 'aa'\n",
       "3. 'ab'\n",
       "4. 'ac'\n",
       "5. 'ad'\n",
       "6. 'ae'\n",
       "7. 'af'\n",
       "8. 'ag'\n",
       "9. 'ah'\n",
       "10. 'ai'\n",
       "11. 'aj'\n",
       "12. 'ak'\n",
       "13. 'al'\n",
       "14. 'am'\n",
       "15. 'ao'\n",
       "16. 'ap'\n",
       "17. 'aq'\n",
       "18. 'ar'\n",
       "19. 'as'\n",
       "20. 'at'\n",
       "21. 'au'\n",
       "22. 'av'\n",
       "23. 'aw'\n",
       "24. 'b'\n",
       "25. 'bc'\n",
       "26. 'bj'\n",
       "27. 'c'\n",
       "28. 'ca'\n",
       "29. 'cb'\n",
       "30. 'cc'\n",
       "31. 'cd'\n",
       "32. 'ce'\n",
       "33. 'd'\n",
       "34. 'f'\n",
       "35. 'g'\n",
       "36. 'h'\n",
       "37. 'j'\n",
       "38. 'k'\n",
       "39. 'l'\n",
       "40. 'm'\n",
       "41. 'n'\n",
       "42. 'o'\n",
       "43. 'p'\n",
       "44. 's'\n",
       "45. 't'\n",
       "46. 'u'\n",
       "47. 'v'\n",
       "48. 'w'\n",
       "49. 'x'\n",
       "50. 'y'\n",
       "51. 'z'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"a\"  \"aa\" \"ab\" \"ac\" \"ad\" \"ae\" \"af\" \"ag\" \"ah\" \"ai\" \"aj\" \"ak\" \"al\" \"am\" \"ao\"\n",
       "[16] \"ap\" \"aq\" \"ar\" \"as\" \"at\" \"au\" \"av\" \"aw\" \"b\"  \"bc\" \"bj\" \"c\"  \"ca\" \"cb\" \"cc\"\n",
       "[31] \"cd\" \"ce\" \"d\"  \"f\"  \"g\"  \"h\"  \"j\"  \"k\"  \"l\"  \"m\"  \"n\"  \"o\"  \"p\"  \"s\"  \"t\" \n",
       "[46] \"u\"  \"v\"  \"w\"  \"x\"  \"y\"  \"z\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique((cell.type.2.markers_no_doubletons  %>% filter(heuristic.score > 60))$cell.type.2)  %>% sort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 456,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'a'</li><li>'aa'</li><li>'ab'</li><li>'ac'</li><li>'ad'</li><li>'ae'</li><li>'af'</li><li>'ag'</li><li>'ah'</li><li>'ai'</li><li>'aj'</li><li>'ak'</li><li>'al'</li><li>'am'</li><li>'an'</li><li>'ao'</li><li>'ap'</li><li>'aq'</li><li>'ar'</li><li>'as'</li><li>'at'</li><li>'au'</li><li>'av'</li><li>'aw'</li><li>'ax'</li><li>'ay'</li><li>'az'</li><li>'b'</li><li>'ba'</li><li>'bb'</li><li>'bc'</li><li>'bd'</li><li>'be'</li><li>'bf'</li><li>'bg'</li><li>'bh'</li><li>'bi'</li><li>'bj'</li><li>'bk'</li><li>'bl'</li><li>'bq'</li><li>'bt'</li><li>'c'</li><li>'ca'</li><li>'cb'</li><li>'cc'</li><li>'cd'</li><li>'ce'</li><li>'cf'</li><li>'d'</li><li>'e'</li><li>'f'</li><li>'g'</li><li>'h'</li><li>'i'</li><li>'j'</li><li>'k'</li><li>'l'</li><li>'m'</li><li>'n'</li><li>'o'</li><li>'p'</li><li>'q'</li><li>'r'</li><li>'s'</li><li>'t'</li><li>'u'</li><li>'v'</li><li>'w'</li><li>'x'</li><li>'y'</li><li>'z'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'a'\n",
       "\\item 'aa'\n",
       "\\item 'ab'\n",
       "\\item 'ac'\n",
       "\\item 'ad'\n",
       "\\item 'ae'\n",
       "\\item 'af'\n",
       "\\item 'ag'\n",
       "\\item 'ah'\n",
       "\\item 'ai'\n",
       "\\item 'aj'\n",
       "\\item 'ak'\n",
       "\\item 'al'\n",
       "\\item 'am'\n",
       "\\item 'an'\n",
       "\\item 'ao'\n",
       "\\item 'ap'\n",
       "\\item 'aq'\n",
       "\\item 'ar'\n",
       "\\item 'as'\n",
       "\\item 'at'\n",
       "\\item 'au'\n",
       "\\item 'av'\n",
       "\\item 'aw'\n",
       "\\item 'ax'\n",
       "\\item 'ay'\n",
       "\\item 'az'\n",
       "\\item 'b'\n",
       "\\item 'ba'\n",
       "\\item 'bb'\n",
       "\\item 'bc'\n",
       "\\item 'bd'\n",
       "\\item 'be'\n",
       "\\item 'bf'\n",
       "\\item 'bg'\n",
       "\\item 'bh'\n",
       "\\item 'bi'\n",
       "\\item 'bj'\n",
       "\\item 'bk'\n",
       "\\item 'bl'\n",
       "\\item 'bq'\n",
       "\\item 'bt'\n",
       "\\item 'c'\n",
       "\\item 'ca'\n",
       "\\item 'cb'\n",
       "\\item 'cc'\n",
       "\\item 'cd'\n",
       "\\item 'ce'\n",
       "\\item 'cf'\n",
       "\\item 'd'\n",
       "\\item 'e'\n",
       "\\item 'f'\n",
       "\\item 'g'\n",
       "\\item 'h'\n",
       "\\item 'i'\n",
       "\\item 'j'\n",
       "\\item 'k'\n",
       "\\item 'l'\n",
       "\\item 'm'\n",
       "\\item 'n'\n",
       "\\item 'o'\n",
       "\\item 'p'\n",
       "\\item 'q'\n",
       "\\item 'r'\n",
       "\\item 's'\n",
       "\\item 't'\n",
       "\\item 'u'\n",
       "\\item 'v'\n",
       "\\item 'w'\n",
       "\\item 'x'\n",
       "\\item 'y'\n",
       "\\item 'z'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'a'\n",
       "2. 'aa'\n",
       "3. 'ab'\n",
       "4. 'ac'\n",
       "5. 'ad'\n",
       "6. 'ae'\n",
       "7. 'af'\n",
       "8. 'ag'\n",
       "9. 'ah'\n",
       "10. 'ai'\n",
       "11. 'aj'\n",
       "12. 'ak'\n",
       "13. 'al'\n",
       "14. 'am'\n",
       "15. 'an'\n",
       "16. 'ao'\n",
       "17. 'ap'\n",
       "18. 'aq'\n",
       "19. 'ar'\n",
       "20. 'as'\n",
       "21. 'at'\n",
       "22. 'au'\n",
       "23. 'av'\n",
       "24. 'aw'\n",
       "25. 'ax'\n",
       "26. 'ay'\n",
       "27. 'az'\n",
       "28. 'b'\n",
       "29. 'ba'\n",
       "30. 'bb'\n",
       "31. 'bc'\n",
       "32. 'bd'\n",
       "33. 'be'\n",
       "34. 'bf'\n",
       "35. 'bg'\n",
       "36. 'bh'\n",
       "37. 'bi'\n",
       "38. 'bj'\n",
       "39. 'bk'\n",
       "40. 'bl'\n",
       "41. 'bq'\n",
       "42. 'bt'\n",
       "43. 'c'\n",
       "44. 'ca'\n",
       "45. 'cb'\n",
       "46. 'cc'\n",
       "47. 'cd'\n",
       "48. 'ce'\n",
       "49. 'cf'\n",
       "50. 'd'\n",
       "51. 'e'\n",
       "52. 'f'\n",
       "53. 'g'\n",
       "54. 'h'\n",
       "55. 'i'\n",
       "56. 'j'\n",
       "57. 'k'\n",
       "58. 'l'\n",
       "59. 'm'\n",
       "60. 'n'\n",
       "61. 'o'\n",
       "62. 'p'\n",
       "63. 'q'\n",
       "64. 'r'\n",
       "65. 's'\n",
       "66. 't'\n",
       "67. 'u'\n",
       "68. 'v'\n",
       "69. 'w'\n",
       "70. 'x'\n",
       "71. 'y'\n",
       "72. 'z'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"a\"  \"aa\" \"ab\" \"ac\" \"ad\" \"ae\" \"af\" \"ag\" \"ah\" \"ai\" \"aj\" \"ak\" \"al\" \"am\" \"an\"\n",
       "[16] \"ao\" \"ap\" \"aq\" \"ar\" \"as\" \"at\" \"au\" \"av\" \"aw\" \"ax\" \"ay\" \"az\" \"b\"  \"ba\" \"bb\"\n",
       "[31] \"bc\" \"bd\" \"be\" \"bf\" \"bg\" \"bh\" \"bi\" \"bj\" \"bk\" \"bl\" \"bq\" \"bt\" \"c\"  \"ca\" \"cb\"\n",
       "[46] \"cc\" \"cd\" \"ce\" \"cf\" \"d\"  \"e\"  \"f\"  \"g\"  \"h\"  \"i\"  \"j\"  \"k\"  \"l\"  \"m\"  \"n\" \n",
       "[61] \"o\"  \"p\"  \"q\"  \"r\"  \"s\"  \"t\"  \"u\"  \"v\"  \"w\"  \"x\"  \"y\"  \"z\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique((cell.type.2.markers_no_doubletons  %>% filter(heuristic.score > 40))$cell.type.2)  %>% sort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 277,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>83</li><li>8</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 83\n",
       "\\item 8\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 83\n",
       "2. 8\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 83  8"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.markers.sub300.cell.type %>% filter(heuristic.score > 60)   %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 287,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>710</li><li>8</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 710\n",
       "\\item 8\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 710\n",
       "2. 8\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 710   8"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.markers.sub300.cell.type %>% filter(heuristic.score > 40)   %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 281,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>1</li><li>8</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 1\n",
       "\\item 8\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 1\n",
       "2. 8\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 1 8"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.markers.sub300.cell.type %>% filter(heuristic.score > 60 & gene %in% fly_TFs)  %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 282,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>48</li><li>8</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 48\n",
       "\\item 8\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 48\n",
       "2. 8\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 48  8"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.markers.sub300.cell.type %>% filter(heuristic.score > 40 & gene %in% fly_TFs)  %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 292,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "19"
      ],
      "text/latex": [
       "19"
      ],
      "text/markdown": [
       "19"
      ],
      "text/plain": [
       "[1] 19"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(unique((cell.type.markers.sub300.cell.type  %>% filter(heuristic.score > 60))$cell.type))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Make expression matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "metadata": {},
   "outputs": [],
   "source": [
    "exp_matrix_sub300_cell.type.2_no_doubletons = cell.type.2.markers_no_doubletons %>% \n",
    "select(cell.type.2, gene, cell.type.2.tpm )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "metadata": {},
   "outputs": [],
   "source": [
    "exp_matrix_sub300_cell.type.2_no_doubletons  = exp_matrix_sub300_cell.type.2_no_doubletons  %>%  \n",
    "pivot_wider(names_from = \"cell.type.2\", values_from = \"cell.type.2.tpm\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_exp_matrix_sub300_cell.type.2_no_doubletons = exp_matrix_sub300_cell.type.2_no_doubletons %>%  \n",
    "filter(gene %in% fly_TFs)  %>% \n",
    "column_to_rownames(\"gene\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 84</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>a</th><th scope=col>c</th><th scope=col>d</th><th scope=col>f</th><th scope=col>g</th><th scope=col>h</th><th scope=col>j</th><th scope=col>l</th><th scope=col>ac</th><th scope=col>p</th><th scope=col>⋯</th><th scope=col>bp</th><th scope=col>bs</th><th scope=col>bn</th><th scope=col>bx</th><th scope=col>bv</th><th scope=col>bu</th><th scope=col>bw</th><th scope=col>by</th><th scope=col>bm</th><th scope=col>bo</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>lz</th><td>  0.000000</td><td>  4.865328</td><td>  7.602135</td><td>  3.111284</td><td>  0.000000</td><td>2653.35239</td><td>  0.000000</td><td> 10.24233</td><td>  4.069243</td><td>  0.000000</td><td>⋯</td><td>  6.509293</td><td>  4.428309</td><td>  1.515547</td><td>  1.931052</td><td>  3.473598</td><td>  2.475727</td><td>  6.841887</td><td>  2.273606</td><td>  5.057638</td><td>  4.616199</td></tr>\n",
       "\t<tr><th scope=row>lab</th><td> 43.303446</td><td>  0.000000</td><td>  7.664362</td><td> 18.228002</td><td>314.728276</td><td>  16.78037</td><td>  0.000000</td><td> 42.32914</td><td> 28.974029</td><td>  6.755926</td><td>⋯</td><td> 52.511672</td><td> 25.319108</td><td> 35.563781</td><td> 31.744393</td><td> 30.515664</td><td> 50.839772</td><td> 24.654562</td><td> 32.160768</td><td> 47.711803</td><td> 19.879523</td></tr>\n",
       "\t<tr><th scope=row>pdm3</th><td>350.913897</td><td>119.632237</td><td>  0.000000</td><td> 16.772789</td><td> 23.640521</td><td>  68.55093</td><td>  5.234615</td><td> 25.67455</td><td> 53.504036</td><td>144.508367</td><td>⋯</td><td> 38.312221</td><td> 29.345579</td><td> 39.167407</td><td> 42.290454</td><td> 67.766081</td><td> 66.020395</td><td> 62.138596</td><td> 42.208948</td><td> 36.458530</td><td> 32.495187</td></tr>\n",
       "\t<tr><th scope=row>sim</th><td>  9.173147</td><td>  1.934386</td><td>  3.086801</td><td>  9.529305</td><td>  8.387431</td><td>   0.00000</td><td> 24.539877</td><td> 15.26463</td><td> 11.038655</td><td>  0.000000</td><td>⋯</td><td> 12.099036</td><td>  7.643226</td><td> 21.890647</td><td> 13.064890</td><td> 11.063674</td><td> 11.654416</td><td> 12.610305</td><td> 13.027923</td><td> 14.041077</td><td> 16.115924</td></tr>\n",
       "\t<tr><th scope=row>cas</th><td>118.762001</td><td> 74.381757</td><td>148.769656</td><td> 48.258788</td><td> 45.094130</td><td> 122.25675</td><td>164.447710</td><td>209.53466</td><td>131.687173</td><td>  0.000000</td><td>⋯</td><td> 96.548925</td><td>122.984584</td><td>288.878177</td><td>113.398364</td><td> 74.569828</td><td> 85.751831</td><td>110.892854</td><td>210.868084</td><td>216.678396</td><td>233.821730</td></tr>\n",
       "\t<tr><th scope=row>seq</th><td>105.169692</td><td> 78.842652</td><td> 74.145589</td><td>113.632272</td><td>324.659152</td><td> 302.86602</td><td>169.059906</td><td>225.38024</td><td>164.109962</td><td>521.302251</td><td>⋯</td><td>102.994574</td><td> 59.210993</td><td> 80.184069</td><td>100.459174</td><td>202.421599</td><td>181.147833</td><td>132.326009</td><td> 65.566818</td><td>124.914603</td><td> 91.692358</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 84\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & a & c & d & f & g & h & j & l & ac & p & ⋯ & bp & bs & bn & bx & bv & bu & bw & by & bm & bo\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tlz &   0.000000 &   4.865328 &   7.602135 &   3.111284 &   0.000000 & 2653.35239 &   0.000000 &  10.24233 &   4.069243 &   0.000000 & ⋯ &   6.509293 &   4.428309 &   1.515547 &   1.931052 &   3.473598 &   2.475727 &   6.841887 &   2.273606 &   5.057638 &   4.616199\\\\\n",
       "\tlab &  43.303446 &   0.000000 &   7.664362 &  18.228002 & 314.728276 &   16.78037 &   0.000000 &  42.32914 &  28.974029 &   6.755926 & ⋯ &  52.511672 &  25.319108 &  35.563781 &  31.744393 &  30.515664 &  50.839772 &  24.654562 &  32.160768 &  47.711803 &  19.879523\\\\\n",
       "\tpdm3 & 350.913897 & 119.632237 &   0.000000 &  16.772789 &  23.640521 &   68.55093 &   5.234615 &  25.67455 &  53.504036 & 144.508367 & ⋯ &  38.312221 &  29.345579 &  39.167407 &  42.290454 &  67.766081 &  66.020395 &  62.138596 &  42.208948 &  36.458530 &  32.495187\\\\\n",
       "\tsim &   9.173147 &   1.934386 &   3.086801 &   9.529305 &   8.387431 &    0.00000 &  24.539877 &  15.26463 &  11.038655 &   0.000000 & ⋯ &  12.099036 &   7.643226 &  21.890647 &  13.064890 &  11.063674 &  11.654416 &  12.610305 &  13.027923 &  14.041077 &  16.115924\\\\\n",
       "\tcas & 118.762001 &  74.381757 & 148.769656 &  48.258788 &  45.094130 &  122.25675 & 164.447710 & 209.53466 & 131.687173 &   0.000000 & ⋯ &  96.548925 & 122.984584 & 288.878177 & 113.398364 &  74.569828 &  85.751831 & 110.892854 & 210.868084 & 216.678396 & 233.821730\\\\\n",
       "\tseq & 105.169692 &  78.842652 &  74.145589 & 113.632272 & 324.659152 &  302.86602 & 169.059906 & 225.38024 & 164.109962 & 521.302251 & ⋯ & 102.994574 &  59.210993 &  80.184069 & 100.459174 & 202.421599 & 181.147833 & 132.326009 &  65.566818 & 124.914603 &  91.692358\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 84\n",
       "\n",
       "| <!--/--> | a &lt;dbl&gt; | c &lt;dbl&gt; | d &lt;dbl&gt; | f &lt;dbl&gt; | g &lt;dbl&gt; | h &lt;dbl&gt; | j &lt;dbl&gt; | l &lt;dbl&gt; | ac &lt;dbl&gt; | p &lt;dbl&gt; | ⋯ ⋯ | bp &lt;dbl&gt; | bs &lt;dbl&gt; | bn &lt;dbl&gt; | bx &lt;dbl&gt; | bv &lt;dbl&gt; | bu &lt;dbl&gt; | bw &lt;dbl&gt; | by &lt;dbl&gt; | bm &lt;dbl&gt; | bo &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| lz |   0.000000 |   4.865328 |   7.602135 |   3.111284 |   0.000000 | 2653.35239 |   0.000000 |  10.24233 |   4.069243 |   0.000000 | ⋯ |   6.509293 |   4.428309 |   1.515547 |   1.931052 |   3.473598 |   2.475727 |   6.841887 |   2.273606 |   5.057638 |   4.616199 |\n",
       "| lab |  43.303446 |   0.000000 |   7.664362 |  18.228002 | 314.728276 |   16.78037 |   0.000000 |  42.32914 |  28.974029 |   6.755926 | ⋯ |  52.511672 |  25.319108 |  35.563781 |  31.744393 |  30.515664 |  50.839772 |  24.654562 |  32.160768 |  47.711803 |  19.879523 |\n",
       "| pdm3 | 350.913897 | 119.632237 |   0.000000 |  16.772789 |  23.640521 |   68.55093 |   5.234615 |  25.67455 |  53.504036 | 144.508367 | ⋯ |  38.312221 |  29.345579 |  39.167407 |  42.290454 |  67.766081 |  66.020395 |  62.138596 |  42.208948 |  36.458530 |  32.495187 |\n",
       "| sim |   9.173147 |   1.934386 |   3.086801 |   9.529305 |   8.387431 |    0.00000 |  24.539877 |  15.26463 |  11.038655 |   0.000000 | ⋯ |  12.099036 |   7.643226 |  21.890647 |  13.064890 |  11.063674 |  11.654416 |  12.610305 |  13.027923 |  14.041077 |  16.115924 |\n",
       "| cas | 118.762001 |  74.381757 | 148.769656 |  48.258788 |  45.094130 |  122.25675 | 164.447710 | 209.53466 | 131.687173 |   0.000000 | ⋯ |  96.548925 | 122.984584 | 288.878177 | 113.398364 |  74.569828 |  85.751831 | 110.892854 | 210.868084 | 216.678396 | 233.821730 |\n",
       "| seq | 105.169692 |  78.842652 |  74.145589 | 113.632272 | 324.659152 |  302.86602 | 169.059906 | 225.38024 | 164.109962 | 521.302251 | ⋯ | 102.994574 |  59.210993 |  80.184069 | 100.459174 | 202.421599 | 181.147833 | 132.326009 |  65.566818 | 124.914603 |  91.692358 |\n",
       "\n"
      ],
      "text/plain": [
       "     a          c          d          f          g          h         \n",
       "lz     0.000000   4.865328   7.602135   3.111284   0.000000 2653.35239\n",
       "lab   43.303446   0.000000   7.664362  18.228002 314.728276   16.78037\n",
       "pdm3 350.913897 119.632237   0.000000  16.772789  23.640521   68.55093\n",
       "sim    9.173147   1.934386   3.086801   9.529305   8.387431    0.00000\n",
       "cas  118.762001  74.381757 148.769656  48.258788  45.094130  122.25675\n",
       "seq  105.169692  78.842652  74.145589 113.632272 324.659152  302.86602\n",
       "     j          l         ac         p          ⋯ bp         bs        \n",
       "lz     0.000000  10.24233   4.069243   0.000000 ⋯   6.509293   4.428309\n",
       "lab    0.000000  42.32914  28.974029   6.755926 ⋯  52.511672  25.319108\n",
       "pdm3   5.234615  25.67455  53.504036 144.508367 ⋯  38.312221  29.345579\n",
       "sim   24.539877  15.26463  11.038655   0.000000 ⋯  12.099036   7.643226\n",
       "cas  164.447710 209.53466 131.687173   0.000000 ⋯  96.548925 122.984584\n",
       "seq  169.059906 225.38024 164.109962 521.302251 ⋯ 102.994574  59.210993\n",
       "     bn         bx         bv         bu         bw         by        \n",
       "lz     1.515547   1.931052   3.473598   2.475727   6.841887   2.273606\n",
       "lab   35.563781  31.744393  30.515664  50.839772  24.654562  32.160768\n",
       "pdm3  39.167407  42.290454  67.766081  66.020395  62.138596  42.208948\n",
       "sim   21.890647  13.064890  11.063674  11.654416  12.610305  13.027923\n",
       "cas  288.878177 113.398364  74.569828  85.751831 110.892854 210.868084\n",
       "seq   80.184069 100.459174 202.421599 181.147833 132.326009  65.566818\n",
       "     bm         bo        \n",
       "lz     5.057638   4.616199\n",
       "lab   47.711803  19.879523\n",
       "pdm3  36.458530  32.495187\n",
       "sim   14.041077  16.115924\n",
       "cas  216.678396 233.821730\n",
       "seq  124.914603  91.692358"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "TF_exp_matrix_sub300_cell.type.2_no_doubletons  %>% head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_exp_matrix_sub300_cell.type.2_no_doubletons_log = log2(TF_exp_matrix_sub300_cell.type.2_no_doubletons + 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 461,
   "metadata": {},
   "outputs": [],
   "source": [
    "options(repr.plot.width=20, repr.plot.height = 60)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(pheatmap)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in pheatmap(TF_exp_matrix_sub300_cell.type.2_no_doubletons_log): could not find function \"pheatmap\"\n",
     "output_type": "error",
     "traceback": [
      "Error in pheatmap(TF_exp_matrix_sub300_cell.type.2_no_doubletons_log): could not find function \"pheatmap\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "pheatmap(TF_exp_matrix_sub300_cell.type.2_no_doubletons_log)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 84</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>a</th><th scope=col>c</th><th scope=col>d</th><th scope=col>f</th><th scope=col>g</th><th scope=col>h</th><th scope=col>j</th><th scope=col>l</th><th scope=col>ac</th><th scope=col>p</th><th scope=col>⋯</th><th scope=col>bp</th><th scope=col>bs</th><th scope=col>bn</th><th scope=col>bx</th><th scope=col>bv</th><th scope=col>bu</th><th scope=col>bw</th><th scope=col>by</th><th scope=col>bm</th><th scope=col>bo</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Muc26B</th><td>11236.03548</td><td>   1.934386</td><td>    0.00000</td><td>    6.085184</td><td> 0.000000</td><td>4.304964</td><td> 22.555034</td><td> 5.7737813</td><td>0.0000000</td><td>0</td><td>⋯</td><td> 1.9768935</td><td> 3.235536</td><td> 0.5368438</td><td> 8.244071</td><td>12.032071</td><td> 5.5083899</td><td> 8.627551</td><td> 3.7406520</td><td> 6.780661</td><td> 5.8641627</td></tr>\n",
       "\t<tr><th scope=row>CG7017</th><td>12317.77183</td><td>   0.000000</td><td>    0.00000</td><td>    6.260625</td><td> 5.722330</td><td>8.669248</td><td>  0.000000</td><td> 0.0000000</td><td>0.0000000</td><td>0</td><td>⋯</td><td> 0.7254060</td><td>30.514094</td><td> 6.9524386</td><td> 8.661267</td><td>25.653128</td><td>15.0861709</td><td>26.892909</td><td>17.2739601</td><td>12.820472</td><td>10.4034864</td></tr>\n",
       "\t<tr><th scope=row>CG7409</th><td>    0.00000</td><td>3866.912562</td><td>    0.00000</td><td>   23.616071</td><td> 0.000000</td><td>0.000000</td><td>  5.776807</td><td> 0.0000000</td><td>0.0000000</td><td>0</td><td>⋯</td><td> 0.0000000</td><td> 1.804183</td><td> 0.0000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.3636798</td><td> 0.000000</td><td> 0.7504173</td><td> 0.000000</td><td> 0.0000000</td></tr>\n",
       "\t<tr><th scope=row>CG11370</th><td>   44.00799</td><td>   2.179219</td><td>16311.33886</td><td>  494.561686</td><td> 9.038455</td><td>0.000000</td><td>  3.188653</td><td> 0.4046702</td><td>4.2444311</td><td>0</td><td>⋯</td><td> 0.8970721</td><td> 7.340034</td><td> 1.7951774</td><td> 9.806871</td><td> 3.573583</td><td> 9.3569042</td><td> 4.377091</td><td> 3.5630590</td><td>10.449361</td><td> 0.3818778</td></tr>\n",
       "\t<tr><th scope=row>CG7465</th><td>   29.85168</td><td>1325.683259</td><td>  151.37943</td><td>18938.231752</td><td>32.455947</td><td>5.144668</td><td>204.935374</td><td>17.6205273</td><td>5.6911549</td><td>0</td><td>⋯</td><td>10.4081381</td><td>49.504509</td><td>38.9990375</td><td>46.161684</td><td>34.820299</td><td>29.3456451</td><td>33.734044</td><td>39.3271711</td><td>40.643456</td><td>17.6810878</td></tr>\n",
       "\t<tr><th scope=row>CG32603</th><td>    0.00000</td><td> 662.552574</td><td>   33.41797</td><td> 9354.678300</td><td> 6.814617</td><td>0.000000</td><td>  0.000000</td><td>12.3753799</td><td>0.9483542</td><td>0</td><td>⋯</td><td> 4.4435928</td><td>25.183830</td><td> 8.5456464</td><td>21.146668</td><td>19.237389</td><td>18.0183938</td><td>13.889679</td><td>22.4665064</td><td>23.153994</td><td> 9.0116771</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 84\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & a & c & d & f & g & h & j & l & ac & p & ⋯ & bp & bs & bn & bx & bv & bu & bw & by & bm & bo\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tMuc26B & 11236.03548 &    1.934386 &     0.00000 &     6.085184 &  0.000000 & 4.304964 &  22.555034 &  5.7737813 & 0.0000000 & 0 & ⋯ &  1.9768935 &  3.235536 &  0.5368438 &  8.244071 & 12.032071 &  5.5083899 &  8.627551 &  3.7406520 &  6.780661 &  5.8641627\\\\\n",
       "\tCG7017 & 12317.77183 &    0.000000 &     0.00000 &     6.260625 &  5.722330 & 8.669248 &   0.000000 &  0.0000000 & 0.0000000 & 0 & ⋯ &  0.7254060 & 30.514094 &  6.9524386 &  8.661267 & 25.653128 & 15.0861709 & 26.892909 & 17.2739601 & 12.820472 & 10.4034864\\\\\n",
       "\tCG7409 &     0.00000 & 3866.912562 &     0.00000 &    23.616071 &  0.000000 & 0.000000 &   5.776807 &  0.0000000 & 0.0000000 & 0 & ⋯ &  0.0000000 &  1.804183 &  0.0000000 &  0.000000 &  0.000000 &  0.3636798 &  0.000000 &  0.7504173 &  0.000000 &  0.0000000\\\\\n",
       "\tCG11370 &    44.00799 &    2.179219 & 16311.33886 &   494.561686 &  9.038455 & 0.000000 &   3.188653 &  0.4046702 & 4.2444311 & 0 & ⋯ &  0.8970721 &  7.340034 &  1.7951774 &  9.806871 &  3.573583 &  9.3569042 &  4.377091 &  3.5630590 & 10.449361 &  0.3818778\\\\\n",
       "\tCG7465 &    29.85168 & 1325.683259 &   151.37943 & 18938.231752 & 32.455947 & 5.144668 & 204.935374 & 17.6205273 & 5.6911549 & 0 & ⋯ & 10.4081381 & 49.504509 & 38.9990375 & 46.161684 & 34.820299 & 29.3456451 & 33.734044 & 39.3271711 & 40.643456 & 17.6810878\\\\\n",
       "\tCG32603 &     0.00000 &  662.552574 &    33.41797 &  9354.678300 &  6.814617 & 0.000000 &   0.000000 & 12.3753799 & 0.9483542 & 0 & ⋯ &  4.4435928 & 25.183830 &  8.5456464 & 21.146668 & 19.237389 & 18.0183938 & 13.889679 & 22.4665064 & 23.153994 &  9.0116771\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 84\n",
       "\n",
       "| <!--/--> | a &lt;dbl&gt; | c &lt;dbl&gt; | d &lt;dbl&gt; | f &lt;dbl&gt; | g &lt;dbl&gt; | h &lt;dbl&gt; | j &lt;dbl&gt; | l &lt;dbl&gt; | ac &lt;dbl&gt; | p &lt;dbl&gt; | ⋯ ⋯ | bp &lt;dbl&gt; | bs &lt;dbl&gt; | bn &lt;dbl&gt; | bx &lt;dbl&gt; | bv &lt;dbl&gt; | bu &lt;dbl&gt; | bw &lt;dbl&gt; | by &lt;dbl&gt; | bm &lt;dbl&gt; | bo &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| Muc26B | 11236.03548 |    1.934386 |     0.00000 |     6.085184 |  0.000000 | 4.304964 |  22.555034 |  5.7737813 | 0.0000000 | 0 | ⋯ |  1.9768935 |  3.235536 |  0.5368438 |  8.244071 | 12.032071 |  5.5083899 |  8.627551 |  3.7406520 |  6.780661 |  5.8641627 |\n",
       "| CG7017 | 12317.77183 |    0.000000 |     0.00000 |     6.260625 |  5.722330 | 8.669248 |   0.000000 |  0.0000000 | 0.0000000 | 0 | ⋯ |  0.7254060 | 30.514094 |  6.9524386 |  8.661267 | 25.653128 | 15.0861709 | 26.892909 | 17.2739601 | 12.820472 | 10.4034864 |\n",
       "| CG7409 |     0.00000 | 3866.912562 |     0.00000 |    23.616071 |  0.000000 | 0.000000 |   5.776807 |  0.0000000 | 0.0000000 | 0 | ⋯ |  0.0000000 |  1.804183 |  0.0000000 |  0.000000 |  0.000000 |  0.3636798 |  0.000000 |  0.7504173 |  0.000000 |  0.0000000 |\n",
       "| CG11370 |    44.00799 |    2.179219 | 16311.33886 |   494.561686 |  9.038455 | 0.000000 |   3.188653 |  0.4046702 | 4.2444311 | 0 | ⋯ |  0.8970721 |  7.340034 |  1.7951774 |  9.806871 |  3.573583 |  9.3569042 |  4.377091 |  3.5630590 | 10.449361 |  0.3818778 |\n",
       "| CG7465 |    29.85168 | 1325.683259 |   151.37943 | 18938.231752 | 32.455947 | 5.144668 | 204.935374 | 17.6205273 | 5.6911549 | 0 | ⋯ | 10.4081381 | 49.504509 | 38.9990375 | 46.161684 | 34.820299 | 29.3456451 | 33.734044 | 39.3271711 | 40.643456 | 17.6810878 |\n",
       "| CG32603 |     0.00000 |  662.552574 |    33.41797 |  9354.678300 |  6.814617 | 0.000000 |   0.000000 | 12.3753799 | 0.9483542 | 0 | ⋯ |  4.4435928 | 25.183830 |  8.5456464 | 21.146668 | 19.237389 | 18.0183938 | 13.889679 | 22.4665064 | 23.153994 |  9.0116771 |\n",
       "\n"
      ],
      "text/plain": [
       "        a           c           d           f            g         h       \n",
       "Muc26B  11236.03548    1.934386     0.00000     6.085184  0.000000 4.304964\n",
       "CG7017  12317.77183    0.000000     0.00000     6.260625  5.722330 8.669248\n",
       "CG7409      0.00000 3866.912562     0.00000    23.616071  0.000000 0.000000\n",
       "CG11370    44.00799    2.179219 16311.33886   494.561686  9.038455 0.000000\n",
       "CG7465     29.85168 1325.683259   151.37943 18938.231752 32.455947 5.144668\n",
       "CG32603     0.00000  662.552574    33.41797  9354.678300  6.814617 0.000000\n",
       "        j          l          ac        p ⋯ bp         bs        bn        \n",
       "Muc26B   22.555034  5.7737813 0.0000000 0 ⋯  1.9768935  3.235536  0.5368438\n",
       "CG7017    0.000000  0.0000000 0.0000000 0 ⋯  0.7254060 30.514094  6.9524386\n",
       "CG7409    5.776807  0.0000000 0.0000000 0 ⋯  0.0000000  1.804183  0.0000000\n",
       "CG11370   3.188653  0.4046702 4.2444311 0 ⋯  0.8970721  7.340034  1.7951774\n",
       "CG7465  204.935374 17.6205273 5.6911549 0 ⋯ 10.4081381 49.504509 38.9990375\n",
       "CG32603   0.000000 12.3753799 0.9483542 0 ⋯  4.4435928 25.183830  8.5456464\n",
       "        bx        bv        bu         bw        by         bm       \n",
       "Muc26B   8.244071 12.032071  5.5083899  8.627551  3.7406520  6.780661\n",
       "CG7017   8.661267 25.653128 15.0861709 26.892909 17.2739601 12.820472\n",
       "CG7409   0.000000  0.000000  0.3636798  0.000000  0.7504173  0.000000\n",
       "CG11370  9.806871  3.573583  9.3569042  4.377091  3.5630590 10.449361\n",
       "CG7465  46.161684 34.820299 29.3456451 33.734044 39.3271711 40.643456\n",
       "CG32603 21.146668 19.237389 18.0183938 13.889679 22.4665064 23.153994\n",
       "        bo        \n",
       "Muc26B   5.8641627\n",
       "CG7017  10.4034864\n",
       "CG7409   0.0000000\n",
       "CG11370  0.3818778\n",
       "CG7465  17.6810878\n",
       "CG32603  9.0116771"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "(head(exp_matrix_sub300_cell.type.2_no_doubletons))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "metadata": {},
   "outputs": [],
   "source": [
    "exp_matrix_sub300_cell.type.2_no_doubletons_mean <-t(apply(exp_matrix_sub300_cell.type.2_no_doubletons,1, function(x) x - mean(x)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 6 × 84 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>a</th><th scope=col>c</th><th scope=col>d</th><th scope=col>f</th><th scope=col>g</th><th scope=col>h</th><th scope=col>j</th><th scope=col>l</th><th scope=col>ac</th><th scope=col>p</th><th scope=col>⋯</th><th scope=col>bp</th><th scope=col>bs</th><th scope=col>bn</th><th scope=col>bx</th><th scope=col>bv</th><th scope=col>bu</th><th scope=col>bw</th><th scope=col>by</th><th scope=col>bm</th><th scope=col>bo</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Muc26B</th><td>11098.28232</td><td>-135.8188</td><td> -137.75316</td><td> -131.6680</td><td>-137.75316</td><td>-133.44820</td><td>-115.19813</td><td>-131.97938</td><td>-137.75316</td><td>-137.75316</td><td>⋯</td><td>-135.77627</td><td>-134.51762</td><td>-137.21632</td><td>-129.50909</td><td>-125.72109</td><td>-132.24477</td><td>-129.12561</td><td>-134.01251</td><td>-130.97250</td><td>-131.88900</td></tr>\n",
       "\t<tr><th scope=row>CG7017</th><td>12161.74551</td><td>-156.0263</td><td> -156.02632</td><td> -149.7657</td><td>-150.30399</td><td>-147.35708</td><td>-156.02632</td><td>-156.02632</td><td>-156.02632</td><td>-156.02632</td><td>⋯</td><td>-155.30092</td><td>-125.51223</td><td>-149.07388</td><td>-147.36506</td><td>-130.37319</td><td>-140.94015</td><td>-129.13341</td><td>-138.75236</td><td>-143.20585</td><td>-145.62284</td></tr>\n",
       "\t<tr><th scope=row>CG7409</th><td>  -46.53667</td><td>3820.3759</td><td>  -46.53667</td><td>  -22.9206</td><td> -46.53667</td><td> -46.53667</td><td> -40.75986</td><td> -46.53667</td><td> -46.53667</td><td> -46.53667</td><td>⋯</td><td> -46.53667</td><td> -44.73248</td><td> -46.53667</td><td> -46.53667</td><td> -46.53667</td><td> -46.17299</td><td> -46.53667</td><td> -45.78625</td><td> -46.53667</td><td> -46.53667</td></tr>\n",
       "\t<tr><th scope=row>CG11370</th><td> -170.55460</td><td>-212.3834</td><td>16096.77626</td><td>  279.9991</td><td>-205.52414</td><td>-214.56260</td><td>-211.37394</td><td>-214.15792</td><td>-210.31816</td><td>-214.56260</td><td>⋯</td><td>-213.66552</td><td>-207.22256</td><td>-212.76742</td><td>-204.75572</td><td>-210.98901</td><td>-205.20569</td><td>-210.18550</td><td>-210.99954</td><td>-204.11323</td><td>-214.18072</td></tr>\n",
       "\t<tr><th scope=row>CG7465</th><td> -235.90670</td><td>1059.9249</td><td> -114.37894</td><td>18672.4734</td><td>-233.30243</td><td>-260.61371</td><td> -60.82300</td><td>-248.13785</td><td>-260.06722</td><td>-265.75838</td><td>⋯</td><td>-255.35024</td><td>-216.25387</td><td>-226.75934</td><td>-219.59669</td><td>-230.93808</td><td>-236.41273</td><td>-232.02433</td><td>-226.43120</td><td>-225.11492</td><td>-248.07729</td></tr>\n",
       "\t<tr><th scope=row>CG32603</th><td> -128.71377</td><td> 533.8388</td><td>  -95.29580</td><td> 9225.9645</td><td>-121.89916</td><td>-128.71377</td><td>-128.71377</td><td>-116.33839</td><td>-127.76542</td><td>-128.71377</td><td>⋯</td><td>-124.27018</td><td>-103.52994</td><td>-120.16813</td><td>-107.56710</td><td>-109.47638</td><td>-110.69538</td><td>-114.82409</td><td>-106.24727</td><td>-105.55978</td><td>-119.70210</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 6 × 84 of type dbl\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & a & c & d & f & g & h & j & l & ac & p & ⋯ & bp & bs & bn & bx & bv & bu & bw & by & bm & bo\\\\\n",
       "\\hline\n",
       "\tMuc26B & 11098.28232 & -135.8188 &  -137.75316 &  -131.6680 & -137.75316 & -133.44820 & -115.19813 & -131.97938 & -137.75316 & -137.75316 & ⋯ & -135.77627 & -134.51762 & -137.21632 & -129.50909 & -125.72109 & -132.24477 & -129.12561 & -134.01251 & -130.97250 & -131.88900\\\\\n",
       "\tCG7017 & 12161.74551 & -156.0263 &  -156.02632 &  -149.7657 & -150.30399 & -147.35708 & -156.02632 & -156.02632 & -156.02632 & -156.02632 & ⋯ & -155.30092 & -125.51223 & -149.07388 & -147.36506 & -130.37319 & -140.94015 & -129.13341 & -138.75236 & -143.20585 & -145.62284\\\\\n",
       "\tCG7409 &   -46.53667 & 3820.3759 &   -46.53667 &   -22.9206 &  -46.53667 &  -46.53667 &  -40.75986 &  -46.53667 &  -46.53667 &  -46.53667 & ⋯ &  -46.53667 &  -44.73248 &  -46.53667 &  -46.53667 &  -46.53667 &  -46.17299 &  -46.53667 &  -45.78625 &  -46.53667 &  -46.53667\\\\\n",
       "\tCG11370 &  -170.55460 & -212.3834 & 16096.77626 &   279.9991 & -205.52414 & -214.56260 & -211.37394 & -214.15792 & -210.31816 & -214.56260 & ⋯ & -213.66552 & -207.22256 & -212.76742 & -204.75572 & -210.98901 & -205.20569 & -210.18550 & -210.99954 & -204.11323 & -214.18072\\\\\n",
       "\tCG7465 &  -235.90670 & 1059.9249 &  -114.37894 & 18672.4734 & -233.30243 & -260.61371 &  -60.82300 & -248.13785 & -260.06722 & -265.75838 & ⋯ & -255.35024 & -216.25387 & -226.75934 & -219.59669 & -230.93808 & -236.41273 & -232.02433 & -226.43120 & -225.11492 & -248.07729\\\\\n",
       "\tCG32603 &  -128.71377 &  533.8388 &   -95.29580 &  9225.9645 & -121.89916 & -128.71377 & -128.71377 & -116.33839 & -127.76542 & -128.71377 & ⋯ & -124.27018 & -103.52994 & -120.16813 & -107.56710 & -109.47638 & -110.69538 & -114.82409 & -106.24727 & -105.55978 & -119.70210\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 6 × 84 of type dbl\n",
       "\n",
       "| <!--/--> | a | c | d | f | g | h | j | l | ac | p | ⋯ | bp | bs | bn | bx | bv | bu | bw | by | bm | bo |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| Muc26B | 11098.28232 | -135.8188 |  -137.75316 |  -131.6680 | -137.75316 | -133.44820 | -115.19813 | -131.97938 | -137.75316 | -137.75316 | ⋯ | -135.77627 | -134.51762 | -137.21632 | -129.50909 | -125.72109 | -132.24477 | -129.12561 | -134.01251 | -130.97250 | -131.88900 |\n",
       "| CG7017 | 12161.74551 | -156.0263 |  -156.02632 |  -149.7657 | -150.30399 | -147.35708 | -156.02632 | -156.02632 | -156.02632 | -156.02632 | ⋯ | -155.30092 | -125.51223 | -149.07388 | -147.36506 | -130.37319 | -140.94015 | -129.13341 | -138.75236 | -143.20585 | -145.62284 |\n",
       "| CG7409 |   -46.53667 | 3820.3759 |   -46.53667 |   -22.9206 |  -46.53667 |  -46.53667 |  -40.75986 |  -46.53667 |  -46.53667 |  -46.53667 | ⋯ |  -46.53667 |  -44.73248 |  -46.53667 |  -46.53667 |  -46.53667 |  -46.17299 |  -46.53667 |  -45.78625 |  -46.53667 |  -46.53667 |\n",
       "| CG11370 |  -170.55460 | -212.3834 | 16096.77626 |   279.9991 | -205.52414 | -214.56260 | -211.37394 | -214.15792 | -210.31816 | -214.56260 | ⋯ | -213.66552 | -207.22256 | -212.76742 | -204.75572 | -210.98901 | -205.20569 | -210.18550 | -210.99954 | -204.11323 | -214.18072 |\n",
       "| CG7465 |  -235.90670 | 1059.9249 |  -114.37894 | 18672.4734 | -233.30243 | -260.61371 |  -60.82300 | -248.13785 | -260.06722 | -265.75838 | ⋯ | -255.35024 | -216.25387 | -226.75934 | -219.59669 | -230.93808 | -236.41273 | -232.02433 | -226.43120 | -225.11492 | -248.07729 |\n",
       "| CG32603 |  -128.71377 |  533.8388 |   -95.29580 |  9225.9645 | -121.89916 | -128.71377 | -128.71377 | -116.33839 | -127.76542 | -128.71377 | ⋯ | -124.27018 | -103.52994 | -120.16813 | -107.56710 | -109.47638 | -110.69538 | -114.82409 | -106.24727 | -105.55978 | -119.70210 |\n",
       "\n"
      ],
      "text/plain": [
       "        a           c         d           f          g          h         \n",
       "Muc26B  11098.28232 -135.8188  -137.75316  -131.6680 -137.75316 -133.44820\n",
       "CG7017  12161.74551 -156.0263  -156.02632  -149.7657 -150.30399 -147.35708\n",
       "CG7409    -46.53667 3820.3759   -46.53667   -22.9206  -46.53667  -46.53667\n",
       "CG11370  -170.55460 -212.3834 16096.77626   279.9991 -205.52414 -214.56260\n",
       "CG7465   -235.90670 1059.9249  -114.37894 18672.4734 -233.30243 -260.61371\n",
       "CG32603  -128.71377  533.8388   -95.29580  9225.9645 -121.89916 -128.71377\n",
       "        j          l          ac         p          ⋯ bp         bs        \n",
       "Muc26B  -115.19813 -131.97938 -137.75316 -137.75316 ⋯ -135.77627 -134.51762\n",
       "CG7017  -156.02632 -156.02632 -156.02632 -156.02632 ⋯ -155.30092 -125.51223\n",
       "CG7409   -40.75986  -46.53667  -46.53667  -46.53667 ⋯  -46.53667  -44.73248\n",
       "CG11370 -211.37394 -214.15792 -210.31816 -214.56260 ⋯ -213.66552 -207.22256\n",
       "CG7465   -60.82300 -248.13785 -260.06722 -265.75838 ⋯ -255.35024 -216.25387\n",
       "CG32603 -128.71377 -116.33839 -127.76542 -128.71377 ⋯ -124.27018 -103.52994\n",
       "        bn         bx         bv         bu         bw         by        \n",
       "Muc26B  -137.21632 -129.50909 -125.72109 -132.24477 -129.12561 -134.01251\n",
       "CG7017  -149.07388 -147.36506 -130.37319 -140.94015 -129.13341 -138.75236\n",
       "CG7409   -46.53667  -46.53667  -46.53667  -46.17299  -46.53667  -45.78625\n",
       "CG11370 -212.76742 -204.75572 -210.98901 -205.20569 -210.18550 -210.99954\n",
       "CG7465  -226.75934 -219.59669 -230.93808 -236.41273 -232.02433 -226.43120\n",
       "CG32603 -120.16813 -107.56710 -109.47638 -110.69538 -114.82409 -106.24727\n",
       "        bm         bo        \n",
       "Muc26B  -130.97250 -131.88900\n",
       "CG7017  -143.20585 -145.62284\n",
       "CG7409   -46.53667  -46.53667\n",
       "CG11370 -204.11323 -214.18072\n",
       "CG7465  -225.11492 -248.07729\n",
       "CG32603 -105.55978 -119.70210"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(exp_matrix_sub300_cell.type.2_no_doubletons_mean)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "metadata": {},
   "outputs": [],
   "source": [
    "saveRDS(exp_matrix_sub300_cell.type.2_no_doubletons_mean, \n",
    "       \"data/exp_matrix_PCA300_cell.type.2_no_doubletons_mean.rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 694,
   "metadata": {},
   "outputs": [],
   "source": [
    "saveRDS(exp_matrix_sub300_cell.type.2_no_doubletons, \n",
    "       \"data/exp_matrix_PCA300_cell.type.2_no_doubletons.rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1964,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'exp_matrix'</li><li>'exp_matrix_mean'</li><li>'exp_matrix_sub300_cell.type.2_no_doubletons'</li><li>'exp_matrix_sub300_cell.type.2_no_doubletons_mean'</li><li>'exp_matrix_sub300_no_doubletons'</li><li>'exp_matrix_sub300_no_singletons'</li><li>'get.norm.expr.matrix'</li><li>'get.norm.expr.matrix.3'</li><li>'TF_exp_matrix'</li><li>'TF_exp_matrix_emb'</li><li>'TF_exp_matrix_emb_log'</li><li>'TF_exp_matrix_log'</li><li>'TF_exp_matrix_max'</li><li>'TF_exp_matrix_norm'</li><li>'TF_exp_matrix_sub300_cell.type.2_no_doubletons'</li><li>'TF_exp_matrix_sub300_cell.type.2_no_doubletons_log'</li><li>'TF_exp_matrix_sub300_no_doubletons'</li><li>'TF_exp_matrix_sub300_no_doubletons_log'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'exp\\_matrix'\n",
       "\\item 'exp\\_matrix\\_mean'\n",
       "\\item 'exp\\_matrix\\_sub300\\_cell.type.2\\_no\\_doubletons'\n",
       "\\item 'exp\\_matrix\\_sub300\\_cell.type.2\\_no\\_doubletons\\_mean'\n",
       "\\item 'exp\\_matrix\\_sub300\\_no\\_doubletons'\n",
       "\\item 'exp\\_matrix\\_sub300\\_no\\_singletons'\n",
       "\\item 'get.norm.expr.matrix'\n",
       "\\item 'get.norm.expr.matrix.3'\n",
       "\\item 'TF\\_exp\\_matrix'\n",
       "\\item 'TF\\_exp\\_matrix\\_emb'\n",
       "\\item 'TF\\_exp\\_matrix\\_emb\\_log'\n",
       "\\item 'TF\\_exp\\_matrix\\_log'\n",
       "\\item 'TF\\_exp\\_matrix\\_max'\n",
       "\\item 'TF\\_exp\\_matrix\\_norm'\n",
       "\\item 'TF\\_exp\\_matrix\\_sub300\\_cell.type.2\\_no\\_doubletons'\n",
       "\\item 'TF\\_exp\\_matrix\\_sub300\\_cell.type.2\\_no\\_doubletons\\_log'\n",
       "\\item 'TF\\_exp\\_matrix\\_sub300\\_no\\_doubletons'\n",
       "\\item 'TF\\_exp\\_matrix\\_sub300\\_no\\_doubletons\\_log'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'exp_matrix'\n",
       "2. 'exp_matrix_mean'\n",
       "3. 'exp_matrix_sub300_cell.type.2_no_doubletons'\n",
       "4. 'exp_matrix_sub300_cell.type.2_no_doubletons_mean'\n",
       "5. 'exp_matrix_sub300_no_doubletons'\n",
       "6. 'exp_matrix_sub300_no_singletons'\n",
       "7. 'get.norm.expr.matrix'\n",
       "8. 'get.norm.expr.matrix.3'\n",
       "9. 'TF_exp_matrix'\n",
       "10. 'TF_exp_matrix_emb'\n",
       "11. 'TF_exp_matrix_emb_log'\n",
       "12. 'TF_exp_matrix_log'\n",
       "13. 'TF_exp_matrix_max'\n",
       "14. 'TF_exp_matrix_norm'\n",
       "15. 'TF_exp_matrix_sub300_cell.type.2_no_doubletons'\n",
       "16. 'TF_exp_matrix_sub300_cell.type.2_no_doubletons_log'\n",
       "17. 'TF_exp_matrix_sub300_no_doubletons'\n",
       "18. 'TF_exp_matrix_sub300_no_doubletons_log'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"exp_matrix\"                                        \n",
       " [2] \"exp_matrix_mean\"                                   \n",
       " [3] \"exp_matrix_sub300_cell.type.2_no_doubletons\"       \n",
       " [4] \"exp_matrix_sub300_cell.type.2_no_doubletons_mean\"  \n",
       " [5] \"exp_matrix_sub300_no_doubletons\"                   \n",
       " [6] \"exp_matrix_sub300_no_singletons\"                   \n",
       " [7] \"get.norm.expr.matrix\"                              \n",
       " [8] \"get.norm.expr.matrix.3\"                            \n",
       " [9] \"TF_exp_matrix\"                                     \n",
       "[10] \"TF_exp_matrix_emb\"                                 \n",
       "[11] \"TF_exp_matrix_emb_log\"                             \n",
       "[12] \"TF_exp_matrix_log\"                                 \n",
       "[13] \"TF_exp_matrix_max\"                                 \n",
       "[14] \"TF_exp_matrix_norm\"                                \n",
       "[15] \"TF_exp_matrix_sub300_cell.type.2_no_doubletons\"    \n",
       "[16] \"TF_exp_matrix_sub300_cell.type.2_no_doubletons_log\"\n",
       "[17] \"TF_exp_matrix_sub300_no_doubletons\"                \n",
       "[18] \"TF_exp_matrix_sub300_no_doubletons_log\"            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ls(pattern = \"matrix\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 242,
   "metadata": {},
   "outputs": [],
   "source": [
    "exp_matrix_mean_original = exp_matrix_mean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "metadata": {},
   "outputs": [],
   "source": [
    "exp_matrix_mean = exp_matrix_sub300_cell.type.2_no_doubletons_mean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "metadata": {},
   "outputs": [],
   "source": [
    "gen = \"bs\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 463,
   "metadata": {},
   "outputs": [],
   "source": [
    "options(repr.plot.width=12, repr.plot.height=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in h(simpleError(msg, call)): error in evaluating the argument 'x' in selecting a method for function 'head': object 'fly_emb_TFs' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in h(simpleError(msg, call)): error in evaluating the argument 'x' in selecting a method for function 'head': object 'fly_emb_TFs' not found\nTraceback:\n",
      "1. head(fly_emb_TFs, 20)",
      "2. .handleSimpleError(function (cond) \n . .Internal(C_tryCatchHelper(addr, 1L, cond)), \"object 'fly_emb_TFs' not found\", \n .     base::quote(head(fly_emb_TFs, 20)))",
      "3. h(simpleError(msg, call))"
     ]
    }
   ],
   "source": [
    "head(fly_emb_TFs, 20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'gen' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'gen' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "\n",
    "tf = gen\n",
    "message(tf)\n",
    "    \n",
    "    file_name = paste(tf, \"aggreg_vs_TF_profile\",\"jpg\", sep = \".\")\n",
    "    file_name = paste(\"worm_agg_emb_graphs_emb_targets\", file_name, sep = \"/\")\n",
    "\n",
    "            \n",
    "          \n",
    "target = \"tx\"\n",
    "    \n",
    "target_values = exp_matrix_mean  %>% as.data.frame  %>% filter(rownames(exp_matrix_mean) == target)\n",
    "t_target_values = as.data.frame(t(target_values))\n",
    "t_target_values_row = rownames_to_column(t_target_values, var = \"cell.type\")\n",
    " colnames(t_target_values_row) <- c(\"cell.type\", \"expr\")\n",
    "\n",
    " \n",
    "    \n",
    "tf_values = exp_matrix_mean  %>% as.data.frame  %>% filter(rownames(exp_matrix_mean) == tf)\n",
    "t_tf_values = as.data.frame(t(tf_values))\n",
    "t_tf_values_row = rownames_to_column(t_tf_values, var = \"cell.type\")\n",
    " colnames(t_tf_values_row) <- c(\"cell.type\", \"expr\")\n",
    "\n",
    "    \n",
    "title = paste(target, \"_from_embryo\", sep = \"_\")  \n",
    "\n",
    "#agg_subtitle = paste(\"max\", target_max, \"target_mean\", target_mean, \"target_entropy\", target_entropy,  sep = \"_\" )    \n",
    "gg_agg = ggplot(t_target_values_row, aes(x = cell.type, y = expr)) + \n",
    "ggtitle(title) +\n",
    "theme_classic() +\n",
    "theme(axis.text.x= element_text(size = 0.1,angle = -90, hjust = 0)) +\n",
    " geom_bar(stat=\"identity\")\n",
    "\n",
    "    tf_title = paste(tf, \"alone\", sep = \" \")\n",
    "\n",
    "#    tf_subtitle = paste (\"max\", tf_max, \"tf_mean\", tf_mean, \"tf_entropy\", tf_entropy,  sep = \"_\")\n",
    "gg_tf = ggplot(t_tf_values_row, aes(x = cell.type, y = expr)) +\n",
    "    theme_classic() +\n",
    "theme(axis.text.x= element_text(size = 9,angle = -90, hjust = 0)) +\n",
    "  geom_bar(stat=\"identity\") + ggtitle(tf_title)\n",
    "\n",
    "#jpeg(\"test.jpg\", width =1100, height = 1100)\n",
    "    gg_both = plot_grid(gg_agg, gg_tf,              # Draw grid of plots\n",
    "          ncol = 1,\n",
    "          rel_heights = c(1,1.8),\n",
    "          align = \"v\")\n",
    "    print(gg_both)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 468,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 60 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cell.type.2</th><th scope=col>annot</th><th scope=col>n</th><th scope=col>rank</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>a </td><td>midgut                        </td><td>  78</td><td>27.0</td></tr>\n",
       "\t<tr><td>a </td><td>ubiquitous                    </td><td>  58</td><td>26.0</td></tr>\n",
       "\t<tr><td>a </td><td>epidermis                     </td><td>  49</td><td>25.0</td></tr>\n",
       "\t<tr><td>a </td><td>salivary gland                </td><td>  42</td><td>24.0</td></tr>\n",
       "\t<tr><td>aa</td><td>foregut / hindgut / anal pad  </td><td> 640</td><td>21.0</td></tr>\n",
       "\t<tr><td>aa</td><td>epidermis                     </td><td> 354</td><td>20.0</td></tr>\n",
       "\t<tr><td>aa</td><td>hindgut / anal pad            </td><td> 224</td><td>19.0</td></tr>\n",
       "\t<tr><td>aa</td><td>somatic muscle                </td><td>  99</td><td>18.0</td></tr>\n",
       "\t<tr><td>ab</td><td>epidermis                     </td><td> 516</td><td>21.0</td></tr>\n",
       "\t<tr><td>ab</td><td>foregut / hindgut / anal pad  </td><td> 417</td><td>20.0</td></tr>\n",
       "\t<tr><td>ab</td><td>brain                         </td><td>  25</td><td>19.0</td></tr>\n",
       "\t<tr><td>ab</td><td>ubiquitous                    </td><td>  23</td><td>18.0</td></tr>\n",
       "\t<tr><td>ac</td><td>salivary gland                </td><td>2725</td><td>34.0</td></tr>\n",
       "\t<tr><td>ac</td><td>salivary gland body primordium</td><td> 678</td><td>33.0</td></tr>\n",
       "\t<tr><td>ac</td><td>midgut                        </td><td> 125</td><td>32.0</td></tr>\n",
       "\t<tr><td>ac</td><td>epidermis                     </td><td>  34</td><td>31.0</td></tr>\n",
       "\t<tr><td>ad</td><td>hindgut / anal pad            </td><td> 366</td><td>22.0</td></tr>\n",
       "\t<tr><td>ad</td><td>epidermis                     </td><td> 124</td><td>21.0</td></tr>\n",
       "\t<tr><td>ad</td><td>somatic muscle                </td><td>  36</td><td>20.0</td></tr>\n",
       "\t<tr><td>ad</td><td>foregut / hindgut / anal pad  </td><td>  24</td><td>19.0</td></tr>\n",
       "\t<tr><td>ae</td><td>midgut                        </td><td> 283</td><td>39.0</td></tr>\n",
       "\t<tr><td>ae</td><td>midgut primordium             </td><td> 263</td><td>38.0</td></tr>\n",
       "\t<tr><td>ae</td><td>epidermis                     </td><td> 171</td><td>37.0</td></tr>\n",
       "\t<tr><td>ae</td><td>epidermis primordium          </td><td>  51</td><td>36.0</td></tr>\n",
       "\t<tr><td>af</td><td>ubiquitous                    </td><td>  51</td><td>12.0</td></tr>\n",
       "\t<tr><td>af</td><td>epidermis                     </td><td>  17</td><td>11.0</td></tr>\n",
       "\t<tr><td>af</td><td>brain                         </td><td>   8</td><td>10.0</td></tr>\n",
       "\t<tr><td>af</td><td>foregut / hindgut / anal pad  </td><td>   6</td><td> 9.0</td></tr>\n",
       "\t<tr><td>ag</td><td>epidermis                     </td><td> 156</td><td>36.0</td></tr>\n",
       "\t<tr><td>ag</td><td>somatic muscle                </td><td>  79</td><td>35.0</td></tr>\n",
       "\t<tr><td>ag</td><td>brain                         </td><td>  63</td><td>34.0</td></tr>\n",
       "\t<tr><td>ag</td><td>hindgut / anal pad            </td><td>  33</td><td>33.0</td></tr>\n",
       "\t<tr><td>ah</td><td>somatic muscle                </td><td>1223</td><td>23.0</td></tr>\n",
       "\t<tr><td>ah</td><td>muscle                        </td><td> 481</td><td>22.0</td></tr>\n",
       "\t<tr><td>ah</td><td>trunk mesoderm primordium     </td><td> 248</td><td>21.0</td></tr>\n",
       "\t<tr><td>ah</td><td>ubiquitous                    </td><td> 171</td><td>20.0</td></tr>\n",
       "\t<tr><td>ai</td><td>hindgut / anal pad            </td><td> 734</td><td>25.0</td></tr>\n",
       "\t<tr><td>ai</td><td>epidermis                     </td><td> 192</td><td>24.0</td></tr>\n",
       "\t<tr><td>ai</td><td>foregut / hindgut / anal pad  </td><td>  81</td><td>23.0</td></tr>\n",
       "\t<tr><td>ai</td><td>somatic muscle                </td><td>  46</td><td>22.0</td></tr>\n",
       "\t<tr><td>aj</td><td>midline primordium            </td><td> 548</td><td>43.0</td></tr>\n",
       "\t<tr><td>aj</td><td>glia                          </td><td> 486</td><td>42.0</td></tr>\n",
       "\t<tr><td>aj</td><td>brain                         </td><td> 256</td><td>41.0</td></tr>\n",
       "\t<tr><td>aj</td><td>brain primordium              </td><td> 181</td><td>40.0</td></tr>\n",
       "\t<tr><td>ak</td><td>visceral muscle               </td><td> 210</td><td>16.0</td></tr>\n",
       "\t<tr><td>ak</td><td>somatic muscle                </td><td> 144</td><td>15.0</td></tr>\n",
       "\t<tr><td>ak</td><td>muscle                        </td><td>  87</td><td>14.0</td></tr>\n",
       "\t<tr><td>ak</td><td>brain                         </td><td>  30</td><td>13.0</td></tr>\n",
       "\t<tr><td>al</td><td>tracheal system               </td><td>2173</td><td>19.0</td></tr>\n",
       "\t<tr><td>al</td><td>epidermis                     </td><td> 138</td><td>18.0</td></tr>\n",
       "\t<tr><td>al</td><td>ubiquitous                    </td><td>  66</td><td>17.0</td></tr>\n",
       "\t<tr><td>al</td><td>brain                         </td><td>  43</td><td>15.5</td></tr>\n",
       "\t<tr><td>al</td><td>muscle                        </td><td>  43</td><td>15.5</td></tr>\n",
       "\t<tr><td>am</td><td>brain                         </td><td> 390</td><td>36.0</td></tr>\n",
       "\t<tr><td>am</td><td>ventral nerve cord            </td><td> 345</td><td>35.0</td></tr>\n",
       "\t<tr><td>am</td><td>epidermis                     </td><td> 204</td><td>34.0</td></tr>\n",
       "\t<tr><td>am</td><td>somatic muscle                </td><td> 183</td><td>33.0</td></tr>\n",
       "\t<tr><td>an</td><td>tracheal system               </td><td>1288</td><td>22.0</td></tr>\n",
       "\t<tr><td>an</td><td>epidermis                     </td><td> 215</td><td>21.0</td></tr>\n",
       "\t<tr><td>an</td><td>foregut / hindgut / anal pad  </td><td>  46</td><td>20.0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 60 × 4\n",
       "\\begin{tabular}{llll}\n",
       " cell.type.2 & annot & n & rank\\\\\n",
       " <chr> & <chr> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t a  & midgut                         &   78 & 27.0\\\\\n",
       "\t a  & ubiquitous                     &   58 & 26.0\\\\\n",
       "\t a  & epidermis                      &   49 & 25.0\\\\\n",
       "\t a  & salivary gland                 &   42 & 24.0\\\\\n",
       "\t aa & foregut / hindgut / anal pad   &  640 & 21.0\\\\\n",
       "\t aa & epidermis                      &  354 & 20.0\\\\\n",
       "\t aa & hindgut / anal pad             &  224 & 19.0\\\\\n",
       "\t aa & somatic muscle                 &   99 & 18.0\\\\\n",
       "\t ab & epidermis                      &  516 & 21.0\\\\\n",
       "\t ab & foregut / hindgut / anal pad   &  417 & 20.0\\\\\n",
       "\t ab & brain                          &   25 & 19.0\\\\\n",
       "\t ab & ubiquitous                     &   23 & 18.0\\\\\n",
       "\t ac & salivary gland                 & 2725 & 34.0\\\\\n",
       "\t ac & salivary gland body primordium &  678 & 33.0\\\\\n",
       "\t ac & midgut                         &  125 & 32.0\\\\\n",
       "\t ac & epidermis                      &   34 & 31.0\\\\\n",
       "\t ad & hindgut / anal pad             &  366 & 22.0\\\\\n",
       "\t ad & epidermis                      &  124 & 21.0\\\\\n",
       "\t ad & somatic muscle                 &   36 & 20.0\\\\\n",
       "\t ad & foregut / hindgut / anal pad   &   24 & 19.0\\\\\n",
       "\t ae & midgut                         &  283 & 39.0\\\\\n",
       "\t ae & midgut primordium              &  263 & 38.0\\\\\n",
       "\t ae & epidermis                      &  171 & 37.0\\\\\n",
       "\t ae & epidermis primordium           &   51 & 36.0\\\\\n",
       "\t af & ubiquitous                     &   51 & 12.0\\\\\n",
       "\t af & epidermis                      &   17 & 11.0\\\\\n",
       "\t af & brain                          &    8 & 10.0\\\\\n",
       "\t af & foregut / hindgut / anal pad   &    6 &  9.0\\\\\n",
       "\t ag & epidermis                      &  156 & 36.0\\\\\n",
       "\t ag & somatic muscle                 &   79 & 35.0\\\\\n",
       "\t ag & brain                          &   63 & 34.0\\\\\n",
       "\t ag & hindgut / anal pad             &   33 & 33.0\\\\\n",
       "\t ah & somatic muscle                 & 1223 & 23.0\\\\\n",
       "\t ah & muscle                         &  481 & 22.0\\\\\n",
       "\t ah & trunk mesoderm primordium      &  248 & 21.0\\\\\n",
       "\t ah & ubiquitous                     &  171 & 20.0\\\\\n",
       "\t ai & hindgut / anal pad             &  734 & 25.0\\\\\n",
       "\t ai & epidermis                      &  192 & 24.0\\\\\n",
       "\t ai & foregut / hindgut / anal pad   &   81 & 23.0\\\\\n",
       "\t ai & somatic muscle                 &   46 & 22.0\\\\\n",
       "\t aj & midline primordium             &  548 & 43.0\\\\\n",
       "\t aj & glia                           &  486 & 42.0\\\\\n",
       "\t aj & brain                          &  256 & 41.0\\\\\n",
       "\t aj & brain primordium               &  181 & 40.0\\\\\n",
       "\t ak & visceral muscle                &  210 & 16.0\\\\\n",
       "\t ak & somatic muscle                 &  144 & 15.0\\\\\n",
       "\t ak & muscle                         &   87 & 14.0\\\\\n",
       "\t ak & brain                          &   30 & 13.0\\\\\n",
       "\t al & tracheal system                & 2173 & 19.0\\\\\n",
       "\t al & epidermis                      &  138 & 18.0\\\\\n",
       "\t al & ubiquitous                     &   66 & 17.0\\\\\n",
       "\t al & brain                          &   43 & 15.5\\\\\n",
       "\t al & muscle                         &   43 & 15.5\\\\\n",
       "\t am & brain                          &  390 & 36.0\\\\\n",
       "\t am & ventral nerve cord             &  345 & 35.0\\\\\n",
       "\t am & epidermis                      &  204 & 34.0\\\\\n",
       "\t am & somatic muscle                 &  183 & 33.0\\\\\n",
       "\t an & tracheal system                & 1288 & 22.0\\\\\n",
       "\t an & epidermis                      &  215 & 21.0\\\\\n",
       "\t an & foregut / hindgut / anal pad   &   46 & 20.0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 60 × 4\n",
       "\n",
       "| cell.type.2 &lt;chr&gt; | annot &lt;chr&gt; | n &lt;int&gt; | rank &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| a  | midgut                         |   78 | 27.0 |\n",
       "| a  | ubiquitous                     |   58 | 26.0 |\n",
       "| a  | epidermis                      |   49 | 25.0 |\n",
       "| a  | salivary gland                 |   42 | 24.0 |\n",
       "| aa | foregut / hindgut / anal pad   |  640 | 21.0 |\n",
       "| aa | epidermis                      |  354 | 20.0 |\n",
       "| aa | hindgut / anal pad             |  224 | 19.0 |\n",
       "| aa | somatic muscle                 |   99 | 18.0 |\n",
       "| ab | epidermis                      |  516 | 21.0 |\n",
       "| ab | foregut / hindgut / anal pad   |  417 | 20.0 |\n",
       "| ab | brain                          |   25 | 19.0 |\n",
       "| ab | ubiquitous                     |   23 | 18.0 |\n",
       "| ac | salivary gland                 | 2725 | 34.0 |\n",
       "| ac | salivary gland body primordium |  678 | 33.0 |\n",
       "| ac | midgut                         |  125 | 32.0 |\n",
       "| ac | epidermis                      |   34 | 31.0 |\n",
       "| ad | hindgut / anal pad             |  366 | 22.0 |\n",
       "| ad | epidermis                      |  124 | 21.0 |\n",
       "| ad | somatic muscle                 |   36 | 20.0 |\n",
       "| ad | foregut / hindgut / anal pad   |   24 | 19.0 |\n",
       "| ae | midgut                         |  283 | 39.0 |\n",
       "| ae | midgut primordium              |  263 | 38.0 |\n",
       "| ae | epidermis                      |  171 | 37.0 |\n",
       "| ae | epidermis primordium           |   51 | 36.0 |\n",
       "| af | ubiquitous                     |   51 | 12.0 |\n",
       "| af | epidermis                      |   17 | 11.0 |\n",
       "| af | brain                          |    8 | 10.0 |\n",
       "| af | foregut / hindgut / anal pad   |    6 |  9.0 |\n",
       "| ag | epidermis                      |  156 | 36.0 |\n",
       "| ag | somatic muscle                 |   79 | 35.0 |\n",
       "| ag | brain                          |   63 | 34.0 |\n",
       "| ag | hindgut / anal pad             |   33 | 33.0 |\n",
       "| ah | somatic muscle                 | 1223 | 23.0 |\n",
       "| ah | muscle                         |  481 | 22.0 |\n",
       "| ah | trunk mesoderm primordium      |  248 | 21.0 |\n",
       "| ah | ubiquitous                     |  171 | 20.0 |\n",
       "| ai | hindgut / anal pad             |  734 | 25.0 |\n",
       "| ai | epidermis                      |  192 | 24.0 |\n",
       "| ai | foregut / hindgut / anal pad   |   81 | 23.0 |\n",
       "| ai | somatic muscle                 |   46 | 22.0 |\n",
       "| aj | midline primordium             |  548 | 43.0 |\n",
       "| aj | glia                           |  486 | 42.0 |\n",
       "| aj | brain                          |  256 | 41.0 |\n",
       "| aj | brain primordium               |  181 | 40.0 |\n",
       "| ak | visceral muscle                |  210 | 16.0 |\n",
       "| ak | somatic muscle                 |  144 | 15.0 |\n",
       "| ak | muscle                         |   87 | 14.0 |\n",
       "| ak | brain                          |   30 | 13.0 |\n",
       "| al | tracheal system                | 2173 | 19.0 |\n",
       "| al | epidermis                      |  138 | 18.0 |\n",
       "| al | ubiquitous                     |   66 | 17.0 |\n",
       "| al | brain                          |   43 | 15.5 |\n",
       "| al | muscle                         |   43 | 15.5 |\n",
       "| am | brain                          |  390 | 36.0 |\n",
       "| am | ventral nerve cord             |  345 | 35.0 |\n",
       "| am | epidermis                      |  204 | 34.0 |\n",
       "| am | somatic muscle                 |  183 | 33.0 |\n",
       "| an | tracheal system                | 1288 | 22.0 |\n",
       "| an | epidermis                      |  215 | 21.0 |\n",
       "| an | foregut / hindgut / anal pad   |   46 | 20.0 |\n",
       "\n"
      ],
      "text/plain": [
       "   cell.type.2 annot                          n    rank\n",
       "1  a           midgut                           78 27.0\n",
       "2  a           ubiquitous                       58 26.0\n",
       "3  a           epidermis                        49 25.0\n",
       "4  a           salivary gland                   42 24.0\n",
       "5  aa          foregut / hindgut / anal pad    640 21.0\n",
       "6  aa          epidermis                       354 20.0\n",
       "7  aa          hindgut / anal pad              224 19.0\n",
       "8  aa          somatic muscle                   99 18.0\n",
       "9  ab          epidermis                       516 21.0\n",
       "10 ab          foregut / hindgut / anal pad    417 20.0\n",
       "11 ab          brain                            25 19.0\n",
       "12 ab          ubiquitous                       23 18.0\n",
       "13 ac          salivary gland                 2725 34.0\n",
       "14 ac          salivary gland body primordium  678 33.0\n",
       "15 ac          midgut                          125 32.0\n",
       "16 ac          epidermis                        34 31.0\n",
       "17 ad          hindgut / anal pad              366 22.0\n",
       "18 ad          epidermis                       124 21.0\n",
       "19 ad          somatic muscle                   36 20.0\n",
       "20 ad          foregut / hindgut / anal pad     24 19.0\n",
       "21 ae          midgut                          283 39.0\n",
       "22 ae          midgut primordium               263 38.0\n",
       "23 ae          epidermis                       171 37.0\n",
       "24 ae          epidermis primordium             51 36.0\n",
       "25 af          ubiquitous                       51 12.0\n",
       "26 af          epidermis                        17 11.0\n",
       "27 af          brain                             8 10.0\n",
       "28 af          foregut / hindgut / anal pad      6  9.0\n",
       "29 ag          epidermis                       156 36.0\n",
       "30 ag          somatic muscle                   79 35.0\n",
       "31 ag          brain                            63 34.0\n",
       "32 ag          hindgut / anal pad               33 33.0\n",
       "33 ah          somatic muscle                 1223 23.0\n",
       "34 ah          muscle                          481 22.0\n",
       "35 ah          trunk mesoderm primordium       248 21.0\n",
       "36 ah          ubiquitous                      171 20.0\n",
       "37 ai          hindgut / anal pad              734 25.0\n",
       "38 ai          epidermis                       192 24.0\n",
       "39 ai          foregut / hindgut / anal pad     81 23.0\n",
       "40 ai          somatic muscle                   46 22.0\n",
       "41 aj          midline primordium              548 43.0\n",
       "42 aj          glia                            486 42.0\n",
       "43 aj          brain                           256 41.0\n",
       "44 aj          brain primordium                181 40.0\n",
       "45 ak          visceral muscle                 210 16.0\n",
       "46 ak          somatic muscle                  144 15.0\n",
       "47 ak          muscle                           87 14.0\n",
       "48 ak          brain                            30 13.0\n",
       "49 al          tracheal system                2173 19.0\n",
       "50 al          epidermis                       138 18.0\n",
       "51 al          ubiquitous                       66 17.0\n",
       "52 al          brain                            43 15.5\n",
       "53 al          muscle                           43 15.5\n",
       "54 am          brain                           390 36.0\n",
       "55 am          ventral nerve cord              345 35.0\n",
       "56 am          epidermis                       204 34.0\n",
       "57 am          somatic muscle                  183 33.0\n",
       "58 an          tracheal system                1288 22.0\n",
       "59 an          epidermis                       215 21.0\n",
       "60 an          foregut / hindgut / anal pad     46 20.0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub_no_doubletons  %>% count(cell.type.2, annot)  %>% \n",
    "       arrange(cell.type.2, -n)  %>% group_by(cell.type.2) %>% \n",
    "    mutate(rank = rank(n))  %>%  slice_max(rank, n = 4)  %>% head(60)  %>% tail(60)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 272,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 10 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>pip3</th><td>ac</td><td>pip        </td><td>47027</td><td>6925</td><td>32792.0852</td><td>264.990019</td><td>6.945832</td><td>104.19514</td></tr>\n",
       "\t<tr><th scope=row>nur3</th><td>ac</td><td>nur        </td><td>26613</td><td>9706</td><td>19927.8068</td><td>355.439960</td><td>5.804980</td><td> 82.91002</td></tr>\n",
       "\t<tr><th scope=row>CG147563</th><td>ac</td><td>CG14756    </td><td>14481</td><td>5763</td><td>10818.1132</td><td>201.054867</td><td>5.742558</td><td> 76.95771</td></tr>\n",
       "\t<tr><th scope=row>CG131593</th><td>ac</td><td>CG13159    </td><td> 6748</td><td>2596</td><td> 4972.8202</td><td>104.269209</td><td>5.561909</td><td> 68.30101</td></tr>\n",
       "\t<tr><th scope=row>CG341903</th><td>ac</td><td>CG34190    </td><td>  914</td><td> 134</td><td>  618.1632</td><td>  4.290539</td><td>6.868429</td><td> 63.69902</td></tr>\n",
       "\t<tr><th scope=row>PH4alphaSG13</th><td>ac</td><td>PH4alphaSG1</td><td> 1719</td><td> 504</td><td> 1247.2033</td><td> 17.378873</td><td>6.084505</td><td> 62.58301</td></tr>\n",
       "\t<tr><th scope=row>PH4alphaSG23</th><td>ac</td><td>PH4alphaSG2</td><td> 1097</td><td> 293</td><td>  789.6215</td><td>  9.326634</td><td>6.256719</td><td> 60.23246</td></tr>\n",
       "\t<tr><th scope=row>CrzR3</th><td>ac</td><td>CrzR       </td><td>  952</td><td> 222</td><td>  676.5130</td><td>  9.290377</td><td>6.038750</td><td> 56.78904</td></tr>\n",
       "\t<tr><th scope=row>GlcAT-P3</th><td>ac</td><td>GlcAT-P    </td><td> 5727</td><td>4643</td><td> 4109.0061</td><td>205.663831</td><td>4.313432</td><td> 51.78242</td></tr>\n",
       "\t<tr><th scope=row>sens3</th><td>ac</td><td>sens       </td><td> 1306</td><td> 822</td><td>  944.1735</td><td> 32.547843</td><td>4.814760</td><td> 47.59118</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 10 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tpip3 & ac & pip         & 47027 & 6925 & 32792.0852 & 264.990019 & 6.945832 & 104.19514\\\\\n",
       "\tnur3 & ac & nur         & 26613 & 9706 & 19927.8068 & 355.439960 & 5.804980 &  82.91002\\\\\n",
       "\tCG147563 & ac & CG14756     & 14481 & 5763 & 10818.1132 & 201.054867 & 5.742558 &  76.95771\\\\\n",
       "\tCG131593 & ac & CG13159     &  6748 & 2596 &  4972.8202 & 104.269209 & 5.561909 &  68.30101\\\\\n",
       "\tCG341903 & ac & CG34190     &   914 &  134 &   618.1632 &   4.290539 & 6.868429 &  63.69902\\\\\n",
       "\tPH4alphaSG13 & ac & PH4alphaSG1 &  1719 &  504 &  1247.2033 &  17.378873 & 6.084505 &  62.58301\\\\\n",
       "\tPH4alphaSG23 & ac & PH4alphaSG2 &  1097 &  293 &   789.6215 &   9.326634 & 6.256719 &  60.23246\\\\\n",
       "\tCrzR3 & ac & CrzR        &   952 &  222 &   676.5130 &   9.290377 & 6.038750 &  56.78904\\\\\n",
       "\tGlcAT-P3 & ac & GlcAT-P     &  5727 & 4643 &  4109.0061 & 205.663831 & 4.313432 &  51.78242\\\\\n",
       "\tsens3 & ac & sens        &  1306 &  822 &   944.1735 &  32.547843 & 4.814760 &  47.59118\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 10 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| pip3 | ac | pip         | 47027 | 6925 | 32792.0852 | 264.990019 | 6.945832 | 104.19514 |\n",
       "| nur3 | ac | nur         | 26613 | 9706 | 19927.8068 | 355.439960 | 5.804980 |  82.91002 |\n",
       "| CG147563 | ac | CG14756     | 14481 | 5763 | 10818.1132 | 201.054867 | 5.742558 |  76.95771 |\n",
       "| CG131593 | ac | CG13159     |  6748 | 2596 |  4972.8202 | 104.269209 | 5.561909 |  68.30101 |\n",
       "| CG341903 | ac | CG34190     |   914 |  134 |   618.1632 |   4.290539 | 6.868429 |  63.69902 |\n",
       "| PH4alphaSG13 | ac | PH4alphaSG1 |  1719 |  504 |  1247.2033 |  17.378873 | 6.084505 |  62.58301 |\n",
       "| PH4alphaSG23 | ac | PH4alphaSG2 |  1097 |  293 |   789.6215 |   9.326634 | 6.256719 |  60.23246 |\n",
       "| CrzR3 | ac | CrzR        |   952 |  222 |   676.5130 |   9.290377 | 6.038750 |  56.78904 |\n",
       "| GlcAT-P3 | ac | GlcAT-P     |  5727 | 4643 |  4109.0061 | 205.663831 | 4.313432 |  51.78242 |\n",
       "| sens3 | ac | sens        |  1306 |  822 |   944.1735 |  32.547843 | 4.814760 |  47.59118 |\n",
       "\n"
      ],
      "text/plain": [
       "             cell.type.2 gene        cell.type.2.n.umi other.n.umi\n",
       "pip3         ac          pip         47027             6925       \n",
       "nur3         ac          nur         26613             9706       \n",
       "CG147563     ac          CG14756     14481             5763       \n",
       "CG131593     ac          CG13159      6748             2596       \n",
       "CG341903     ac          CG34190       914              134       \n",
       "PH4alphaSG13 ac          PH4alphaSG1  1719              504       \n",
       "PH4alphaSG23 ac          PH4alphaSG2  1097              293       \n",
       "CrzR3        ac          CrzR          952              222       \n",
       "GlcAT-P3     ac          GlcAT-P      5727             4643       \n",
       "sens3        ac          sens         1306              822       \n",
       "             cell.type.2.tpm other.tpm  log2.ratio heuristic.score\n",
       "pip3         32792.0852      264.990019 6.945832   104.19514      \n",
       "nur3         19927.8068      355.439960 5.804980    82.91002      \n",
       "CG147563     10818.1132      201.054867 5.742558    76.95771      \n",
       "CG131593      4972.8202      104.269209 5.561909    68.30101      \n",
       "CG341903       618.1632        4.290539 6.868429    63.69902      \n",
       "PH4alphaSG13  1247.2033       17.378873 6.084505    62.58301      \n",
       "PH4alphaSG23   789.6215        9.326634 6.256719    60.23246      \n",
       "CrzR3          676.5130        9.290377 6.038750    56.78904      \n",
       "GlcAT-P3      4109.0061      205.663831 4.313432    51.78242      \n",
       "sens3          944.1735       32.547843 4.814760    47.59118      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"ac\")  %>% head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 276,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type</th><th scope=col>gene</th><th scope=col>cell.type.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG92694</th><td>cellular blastoderm / pole cell</td><td>CG9269  </td><td>   19</td><td>  157</td><td> 4414.498</td><td>  5.530986</td><td>9.400733</td><td>113.82746</td></tr>\n",
       "\t<tr><th scope=row>CG50704</th><td>cellular blastoderm / pole cell</td><td>CG5070  </td><td>   17</td><td>  184</td><td> 3949.814</td><td>  6.730373</td><td>8.997031</td><td>107.49593</td></tr>\n",
       "\t<tr><th scope=row>Jon65Aiv4</th><td>cellular blastoderm / pole cell</td><td>Jon65Aiv</td><td>   16</td><td>  819</td><td> 4000.000</td><td> 16.867280</td><td>7.806536</td><td> 93.41414</td></tr>\n",
       "\t<tr><th scope=row>pip32</th><td>salivary gland                 </td><td>pip     </td><td>38398</td><td> 9911</td><td>29916.415</td><td>412.274111</td><td>6.177695</td><td> 91.85427</td></tr>\n",
       "\t<tr><th scope=row>Ppn28</th><td>plasmatocytes                  </td><td>Ppn     </td><td>73383</td><td>14116</td><td>35715.779</td><td>536.739395</td><td>6.053511</td><td> 91.55520</td></tr>\n",
       "\t<tr><th scope=row>CG43332</th><td>amnioserosa                    </td><td>CG43332 </td><td>23490</td><td> 1608</td><td> 7785.421</td><td> 56.955284</td><td>7.069691</td><td> 91.38809</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type & gene & cell.type.n.umi & other.n.umi & cell.type.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG92694 & cellular blastoderm / pole cell & CG9269   &    19 &   157 &  4414.498 &   5.530986 & 9.400733 & 113.82746\\\\\n",
       "\tCG50704 & cellular blastoderm / pole cell & CG5070   &    17 &   184 &  3949.814 &   6.730373 & 8.997031 & 107.49593\\\\\n",
       "\tJon65Aiv4 & cellular blastoderm / pole cell & Jon65Aiv &    16 &   819 &  4000.000 &  16.867280 & 7.806536 &  93.41414\\\\\n",
       "\tpip32 & salivary gland                  & pip      & 38398 &  9911 & 29916.415 & 412.274111 & 6.177695 &  91.85427\\\\\n",
       "\tPpn28 & plasmatocytes                   & Ppn      & 73383 & 14116 & 35715.779 & 536.739395 & 6.053511 &  91.55520\\\\\n",
       "\tCG43332 & amnioserosa                     & CG43332  & 23490 &  1608 &  7785.421 &  56.955284 & 7.069691 &  91.38809\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 8\n",
       "\n",
       "| <!--/--> | cell.type &lt;chr&gt; | gene &lt;chr&gt; | cell.type.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG92694 | cellular blastoderm / pole cell | CG9269   |    19 |   157 |  4414.498 |   5.530986 | 9.400733 | 113.82746 |\n",
       "| CG50704 | cellular blastoderm / pole cell | CG5070   |    17 |   184 |  3949.814 |   6.730373 | 8.997031 | 107.49593 |\n",
       "| Jon65Aiv4 | cellular blastoderm / pole cell | Jon65Aiv |    16 |   819 |  4000.000 |  16.867280 | 7.806536 |  93.41414 |\n",
       "| pip32 | salivary gland                  | pip      | 38398 |  9911 | 29916.415 | 412.274111 | 6.177695 |  91.85427 |\n",
       "| Ppn28 | plasmatocytes                   | Ppn      | 73383 | 14116 | 35715.779 | 536.739395 | 6.053511 |  91.55520 |\n",
       "| CG43332 | amnioserosa                     | CG43332  | 23490 |  1608 |  7785.421 |  56.955284 | 7.069691 |  91.38809 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type                       gene     cell.type.n.umi other.n.umi\n",
       "CG92694   cellular blastoderm / pole cell CG9269      19             157      \n",
       "CG50704   cellular blastoderm / pole cell CG5070      17             184      \n",
       "Jon65Aiv4 cellular blastoderm / pole cell Jon65Aiv    16             819      \n",
       "pip32     salivary gland                  pip      38398            9911      \n",
       "Ppn28     plasmatocytes                   Ppn      73383           14116      \n",
       "CG43332   amnioserosa                     CG43332  23490            1608      \n",
       "          cell.type.tpm other.tpm  log2.ratio heuristic.score\n",
       "CG92694    4414.498       5.530986 9.400733   113.82746      \n",
       "CG50704    3949.814       6.730373 8.997031   107.49593      \n",
       "Jon65Aiv4  4000.000      16.867280 7.806536    93.41414      \n",
       "pip32     29916.415     412.274111 6.177695    91.85427      \n",
       "Ppn28     35715.779     536.739395 6.053511    91.55520      \n",
       "CG43332    7785.421      56.955284 7.069691    91.38809      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.markers.sub300.cell.type  %>% head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 279,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'cellular blastoderm / pole cell'</li><li>'salivary gland'</li><li>'plasmatocytes'</li><li>'amnioserosa'</li><li>'maternal'</li><li>'ectoderm anlage'</li><li>'Malpighian tubule'</li><li>'tracheal primordium'</li><li>'salivary gland body primordium'</li><li>'glia'</li><li>'crystal cell'</li><li>'spiracles (tracheal system)'</li><li>'ectoderm anlage in statu nascendi'</li><li>'proventriculus'</li><li>'spiracle (tracheal system)'</li><li>'unknown'</li><li>'visceral muscle'</li><li>'tracheal system'</li><li>'fat body'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'cellular blastoderm / pole cell'\n",
       "\\item 'salivary gland'\n",
       "\\item 'plasmatocytes'\n",
       "\\item 'amnioserosa'\n",
       "\\item 'maternal'\n",
       "\\item 'ectoderm anlage'\n",
       "\\item 'Malpighian tubule'\n",
       "\\item 'tracheal primordium'\n",
       "\\item 'salivary gland body primordium'\n",
       "\\item 'glia'\n",
       "\\item 'crystal cell'\n",
       "\\item 'spiracles (tracheal system)'\n",
       "\\item 'ectoderm anlage in statu nascendi'\n",
       "\\item 'proventriculus'\n",
       "\\item 'spiracle (tracheal system)'\n",
       "\\item 'unknown'\n",
       "\\item 'visceral muscle'\n",
       "\\item 'tracheal system'\n",
       "\\item 'fat body'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'cellular blastoderm / pole cell'\n",
       "2. 'salivary gland'\n",
       "3. 'plasmatocytes'\n",
       "4. 'amnioserosa'\n",
       "5. 'maternal'\n",
       "6. 'ectoderm anlage'\n",
       "7. 'Malpighian tubule'\n",
       "8. 'tracheal primordium'\n",
       "9. 'salivary gland body primordium'\n",
       "10. 'glia'\n",
       "11. 'crystal cell'\n",
       "12. 'spiracles (tracheal system)'\n",
       "13. 'ectoderm anlage in statu nascendi'\n",
       "14. 'proventriculus'\n",
       "15. 'spiracle (tracheal system)'\n",
       "16. 'unknown'\n",
       "17. 'visceral muscle'\n",
       "18. 'tracheal system'\n",
       "19. 'fat body'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"cellular blastoderm / pole cell\"   \"salivary gland\"                   \n",
       " [3] \"plasmatocytes\"                     \"amnioserosa\"                      \n",
       " [5] \"maternal\"                          \"ectoderm anlage\"                  \n",
       " [7] \"Malpighian tubule\"                 \"tracheal primordium\"              \n",
       " [9] \"salivary gland body primordium\"    \"glia\"                             \n",
       "[11] \"crystal cell\"                      \"spiracles (tracheal system)\"      \n",
       "[13] \"ectoderm anlage in statu nascendi\" \"proventriculus\"                   \n",
       "[15] \"spiracle (tracheal system)\"        \"unknown\"                          \n",
       "[17] \"visceral muscle\"                   \"tracheal system\"                  \n",
       "[19] \"fat body\"                         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "19"
      ],
      "text/latex": [
       "19"
      ],
      "text/markdown": [
       "19"
      ],
      "text/plain": [
       "[1] 19"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique((cell.type.markers.sub300.cell.type %>% filter(heuristic.score > 60))$cell.type)\n",
    "unique((cell.type.markers.sub300.cell.type %>% filter(heuristic.score > 60))$cell.type)  %>% length"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 294,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'lz'</li><li>'lab'</li><li>'pdm3'</li><li>'sim'</li><li>'cas'</li><li>'seq'</li><li>'sv'</li><li>'Eip74EF'</li><li>'otp'</li><li>'Doc2'</li><li>'Ptx1'</li><li>'fru'</li><li>'Sp1'</li><li>'svp'</li><li>'inv'</li><li>'C15'</li><li>'CG42741'</li><li>'Lim3'</li><li>'Crg-1'</li><li>'HHEX'</li><li>'fd3F'</li><li>'disco-r'</li><li>'Clamp'</li><li>'lmd'</li><li>'slbo'</li><li>'CG34224'</li><li>'SoxN'</li><li>'ken'</li><li>'Atf6'</li><li>'Drgx'</li><li>'HLH54F'</li><li>'sens'</li><li>'tup'</li><li>'ham'</li><li>'ss'</li><li>'pros'</li><li>'dimm'</li><li>'CG4328'</li><li>'Oaz'</li><li>'ara'</li><li>'kni'</li><li>'dysf'</li><li>'ftz-f1'</li><li>'CrebA'</li><li>'Mes2'</li><li>'CG9932'</li><li>'Kah'</li><li>'Sox102F'</li><li>'Rfx'</li><li>'ush'</li><li>'pdm2'</li><li>'H2.0'</li><li>'en'</li><li>'jigr1'</li><li>'Eip93F'</li><li>'caup'</li><li>'pho'</li><li>'sna'</li><li>'sens-2'</li><li>'tx'</li><li>'dve'</li><li>'CG43347'</li><li>'klu'</li><li>'Dref'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'lz'\n",
       "\\item 'lab'\n",
       "\\item 'pdm3'\n",
       "\\item 'sim'\n",
       "\\item 'cas'\n",
       "\\item 'seq'\n",
       "\\item 'sv'\n",
       "\\item 'Eip74EF'\n",
       "\\item 'otp'\n",
       "\\item 'Doc2'\n",
       "\\item 'Ptx1'\n",
       "\\item 'fru'\n",
       "\\item 'Sp1'\n",
       "\\item 'svp'\n",
       "\\item 'inv'\n",
       "\\item 'C15'\n",
       "\\item 'CG42741'\n",
       "\\item 'Lim3'\n",
       "\\item 'Crg-1'\n",
       "\\item 'HHEX'\n",
       "\\item 'fd3F'\n",
       "\\item 'disco-r'\n",
       "\\item 'Clamp'\n",
       "\\item 'lmd'\n",
       "\\item 'slbo'\n",
       "\\item 'CG34224'\n",
       "\\item 'SoxN'\n",
       "\\item 'ken'\n",
       "\\item 'Atf6'\n",
       "\\item 'Drgx'\n",
       "\\item 'HLH54F'\n",
       "\\item 'sens'\n",
       "\\item 'tup'\n",
       "\\item 'ham'\n",
       "\\item 'ss'\n",
       "\\item 'pros'\n",
       "\\item 'dimm'\n",
       "\\item 'CG4328'\n",
       "\\item 'Oaz'\n",
       "\\item 'ara'\n",
       "\\item 'kni'\n",
       "\\item 'dysf'\n",
       "\\item 'ftz-f1'\n",
       "\\item 'CrebA'\n",
       "\\item 'Mes2'\n",
       "\\item 'CG9932'\n",
       "\\item 'Kah'\n",
       "\\item 'Sox102F'\n",
       "\\item 'Rfx'\n",
       "\\item 'ush'\n",
       "\\item 'pdm2'\n",
       "\\item 'H2.0'\n",
       "\\item 'en'\n",
       "\\item 'jigr1'\n",
       "\\item 'Eip93F'\n",
       "\\item 'caup'\n",
       "\\item 'pho'\n",
       "\\item 'sna'\n",
       "\\item 'sens-2'\n",
       "\\item 'tx'\n",
       "\\item 'dve'\n",
       "\\item 'CG43347'\n",
       "\\item 'klu'\n",
       "\\item 'Dref'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'lz'\n",
       "2. 'lab'\n",
       "3. 'pdm3'\n",
       "4. 'sim'\n",
       "5. 'cas'\n",
       "6. 'seq'\n",
       "7. 'sv'\n",
       "8. 'Eip74EF'\n",
       "9. 'otp'\n",
       "10. 'Doc2'\n",
       "11. 'Ptx1'\n",
       "12. 'fru'\n",
       "13. 'Sp1'\n",
       "14. 'svp'\n",
       "15. 'inv'\n",
       "16. 'C15'\n",
       "17. 'CG42741'\n",
       "18. 'Lim3'\n",
       "19. 'Crg-1'\n",
       "20. 'HHEX'\n",
       "21. 'fd3F'\n",
       "22. 'disco-r'\n",
       "23. 'Clamp'\n",
       "24. 'lmd'\n",
       "25. 'slbo'\n",
       "26. 'CG34224'\n",
       "27. 'SoxN'\n",
       "28. 'ken'\n",
       "29. 'Atf6'\n",
       "30. 'Drgx'\n",
       "31. 'HLH54F'\n",
       "32. 'sens'\n",
       "33. 'tup'\n",
       "34. 'ham'\n",
       "35. 'ss'\n",
       "36. 'pros'\n",
       "37. 'dimm'\n",
       "38. 'CG4328'\n",
       "39. 'Oaz'\n",
       "40. 'ara'\n",
       "41. 'kni'\n",
       "42. 'dysf'\n",
       "43. 'ftz-f1'\n",
       "44. 'CrebA'\n",
       "45. 'Mes2'\n",
       "46. 'CG9932'\n",
       "47. 'Kah'\n",
       "48. 'Sox102F'\n",
       "49. 'Rfx'\n",
       "50. 'ush'\n",
       "51. 'pdm2'\n",
       "52. 'H2.0'\n",
       "53. 'en'\n",
       "54. 'jigr1'\n",
       "55. 'Eip93F'\n",
       "56. 'caup'\n",
       "57. 'pho'\n",
       "58. 'sna'\n",
       "59. 'sens-2'\n",
       "60. 'tx'\n",
       "61. 'dve'\n",
       "62. 'CG43347'\n",
       "63. 'klu'\n",
       "64. 'Dref'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"lz\"      \"lab\"     \"pdm3\"    \"sim\"     \"cas\"     \"seq\"     \"sv\"     \n",
       " [8] \"Eip74EF\" \"otp\"     \"Doc2\"    \"Ptx1\"    \"fru\"     \"Sp1\"     \"svp\"    \n",
       "[15] \"inv\"     \"C15\"     \"CG42741\" \"Lim3\"    \"Crg-1\"   \"HHEX\"    \"fd3F\"   \n",
       "[22] \"disco-r\" \"Clamp\"   \"lmd\"     \"slbo\"    \"CG34224\" \"SoxN\"    \"ken\"    \n",
       "[29] \"Atf6\"    \"Drgx\"    \"HLH54F\"  \"sens\"    \"tup\"     \"ham\"     \"ss\"     \n",
       "[36] \"pros\"    \"dimm\"    \"CG4328\"  \"Oaz\"     \"ara\"     \"kni\"     \"dysf\"   \n",
       "[43] \"ftz-f1\"  \"CrebA\"   \"Mes2\"    \"CG9932\"  \"Kah\"     \"Sox102F\" \"Rfx\"    \n",
       "[50] \"ush\"     \"pdm2\"    \"H2.0\"    \"en\"      \"jigr1\"   \"Eip93F\"  \"caup\"   \n",
       "[57] \"pho\"     \"sna\"     \"sens-2\"  \"tx\"      \"dve\"     \"CG43347\" \"klu\"    \n",
       "[64] \"Dref\"   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique((cell.type.2.markers_no_doubletons  %>% filter(heuristic.score > 40 & gene %in% fly_TFs))$gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "lz\n",
      "\n"
     ]
    },
    {
     "ename": "ERROR",
     "evalue": "Error in exp_matrix_mean %>% as.data.frame %>% filter(rownames(exp_matrix_mean) == : could not find function \"%>%\"\n",
     "output_type": "error",
     "traceback": [
      "Error in exp_matrix_mean %>% as.data.frame %>% filter(rownames(exp_matrix_mean) == : could not find function \"%>%\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "tf = \"lz\"\n",
    "message(tf)\n",
    "    \n",
    "    file_name = paste(tf, \"aggreg_vs_TF_profile\",\"jpg\", sep = \".\")\n",
    "    file_name = paste(\"worm_agg_emb_graphs_emb_targets\", file_name, sep = \"/\")\n",
    "\n",
    "            \n",
    "          \n",
    "target = \"lab\"\n",
    "    \n",
    "target_values = exp_matrix_mean  %>% as.data.frame  %>% filter(rownames(exp_matrix_mean) == target)\n",
    "t_target_values = as.data.frame(t(target_values))\n",
    "t_target_values_row = rownames_to_column(t_target_values, var = \"cell.type\")\n",
    " colnames(t_target_values_row) <- c(\"cell.type\", \"expr\")\n",
    "\n",
    " \n",
    "    \n",
    "tf_values = exp_matrix_mean  %>% as.data.frame  %>% filter(rownames(exp_matrix_mean) == tf)\n",
    "t_tf_values = as.data.frame(t(tf_values))\n",
    "t_tf_values_row = rownames_to_column(t_tf_values, var = \"cell.type\")\n",
    " colnames(t_tf_values_row) <- c(\"cell.type\", \"expr\")\n",
    "\n",
    "    \n",
    "title = paste(target, \"_from_embryo\", sep = \"_\")  \n",
    "\n",
    "#agg_subtitle = paste(\"max\", target_max, \"target_mean\", target_mean, \"target_entropy\", target_entropy,  sep = \"_\" )    \n",
    "gg_agg = ggplot(t_target_values_row, aes(x = cell.type, y = expr)) + \n",
    "ggtitle(title) +\n",
    "theme_classic() +\n",
    "theme(axis.text.x= element_text(size = 0.1,angle = -90, hjust = 0)) +\n",
    " geom_bar(stat=\"identity\")\n",
    "\n",
    "    tf_title = paste(tf, \"alone\", sep = \" \")\n",
    "\n",
    "#    tf_subtitle = paste (\"max\", tf_max, \"tf_mean\", tf_mean, \"tf_entropy\", tf_entropy,  sep = \"_\")\n",
    "gg_tf = ggplot(t_tf_values_row, aes(x = cell.type, y = expr)) +\n",
    "    theme_classic() +\n",
    "theme(axis.text.x= element_text(size = 9,angle = -90, hjust = 0)) +\n",
    "  geom_bar(stat=\"identity\") + ggtitle(tf_title)\n",
    "\n",
    "#jpeg(\"test.jpg\", width =1100, height = 1100)\n",
    "    gg_both = plot_grid(gg_agg, gg_tf,              # Draw grid of plots\n",
    "          ncol = 1,\n",
    "          rel_heights = c(1,1.8),\n",
    "          align = \"v\")\n",
    "    print(gg_both)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub, gene = \"lz\", size_pos = 0.1, size_neg = 0.02, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub, gene = \"lz\", size_pos = 0.1, size_neg = 0.02, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub, gene = \"lz\", size_pos = 0.1, size_neg = 0.02, stroke = 0.01)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr.fly(cds_sub, gene = \"Muc11A\", size_pos = 0.3, size_neg = 0.3, : could not find function \"plot.expr.fly\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr.fly(cds_sub, gene = \"Muc11A\", size_pos = 0.3, size_neg = 0.3, : could not find function \"plot.expr.fly\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr.fly(cds_sub, gene = \"Muc11A\", size_pos = 0.3, size_neg = 0.3, stroke = 0.01)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 308,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<pre class=language-r><code>function (cds, gene_short_name, coexpr = F, ncol = NULL, x = 1, \n",
       "<span style=white-space:pre-wrap>    y = 2, size_pos = NULL, stroke = NULL, non.expr.alpha = 1/3, </span>\n",
       "<span style=white-space:pre-wrap>    color.pal = c(\"grey85\", \"red\", \"orange\", \"yellow\"), sample.n = F, </span>\n",
       "<span style=white-space:pre-wrap>    label_size = NULL, limit = NULL, size_neg = NULL) </span>\n",
       "{\n",
       "<span style=white-space:pre-wrap>    if (sample.n != F) {</span>\n",
       "<span style=white-space:pre-wrap>        set.seed(42)</span>\n",
       "<span style=white-space:pre-wrap>        cds = cds[, sample(1:ncol(cds), sample.n)]</span>\n",
       "<span style=white-space:pre-wrap>    }</span>\n",
       "<span style=white-space:pre-wrap>    S_matrix &lt;- reducedDims(cds)[[\"UMAP\"]]</span>\n",
       "<span style=white-space:pre-wrap>    data_df &lt;- data.frame(S_matrix)</span>\n",
       "<span style=white-space:pre-wrap>    colnames(data_df) &lt;- c(\"umap_1\", \"umap_2\")</span>\n",
       "<span style=white-space:pre-wrap>    data_df$cluster = as.vector(clusters(cds))</span>\n",
       "<span style=white-space:pre-wrap>    data_df$sample_name &lt;- row.names(data_df)</span>\n",
       "<span style=white-space:pre-wrap>    data_df$Size_Factor &lt;- colData(cds)$Size_Factor</span>\n",
       "<span style=white-space:pre-wrap>    data_df$lineage &lt;- colData(cds)$lineage</span>\n",
       "<span style=white-space:pre-wrap>    data_df$species &lt;- colData(cds)$species</span>\n",
       "<span style=white-space:pre-wrap>    if (length(nchar(gene_short_name)) &gt; 1) {</span>\n",
       "<span style=white-space:pre-wrap>        label_df &lt;- data_df %&gt;% group_by(cluster) %&gt;% summarize(umap_1 = median(umap_1), </span>\n",
       "<span style=white-space:pre-wrap>            umap_2 = median(umap_2))</span>\n",
       "<span style=white-space:pre-wrap>        a &lt;- as.data.frame(as.matrix(t(counts(cds)[gene_short_name, </span>\n",
       "<span style=white-space:pre-wrap>            ])))</span>\n",
       "<span style=white-space:pre-wrap>        a &lt;- a/data_df$Size_Factor</span>\n",
       "<span style=white-space:pre-wrap>        a$min &lt;- apply(a, 1, min)</span>\n",
       "<span style=white-space:pre-wrap>        data_df$min &lt;- a$min</span>\n",
       "<span style=white-space:pre-wrap>        if (is.null(size_pos)) {</span>\n",
       "<span style=white-space:pre-wrap>            size_pos = 0.1</span>\n",
       "<span style=white-space:pre-wrap>        }</span>\n",
       "<span style=white-space:pre-wrap>        if (is.null(stroke)) {</span>\n",
       "<span style=white-space:pre-wrap>            stroke = 0.4</span>\n",
       "<span style=white-space:pre-wrap>        }</span>\n",
       "<span style=white-space:pre-wrap>        if (is.null(label_size)) {</span>\n",
       "<span style=white-space:pre-wrap>            label_size = 4</span>\n",
       "<span style=white-space:pre-wrap>        }</span>\n",
       "<span style=white-space:pre-wrap>        if (is.null(limit)) {</span>\n",
       "<span style=white-space:pre-wrap>            limit = 1</span>\n",
       "<span style=white-space:pre-wrap>        }</span>\n",
       "<span style=white-space:pre-wrap>        if (is.null(size_neg)) {</span>\n",
       "<span style=white-space:pre-wrap>            size_neg = 0.1</span>\n",
       "<span style=white-space:pre-wrap>        }</span>\n",
       "<span style=white-space:pre-wrap>        plot = ggplot(data_df, alpha = min &gt; 0) + geom_point(aes(x = umap_1, </span>\n",
       "<span style=white-space:pre-wrap>            y = umap_2, color = log2(min + 1)), size = ifelse(data_df$min &gt; </span>\n",
       "<span style=white-space:pre-wrap>            limit, size_pos, size_neg), stroke = stroke) + scale_color_gradientn(colors = color.pal) + </span>\n",
       "<span style=white-space:pre-wrap>            scale_alpha_manual(values = c(non.expr.alpha, 1)) + </span>\n",
       "<span style=white-space:pre-wrap>            guides(alpha = F, color = guide_colorbar(title = \"Minimum\\nnormalized\\nlog2 expression\\nfor specified genes\")) + </span>\n",
       "<span style=white-space:pre-wrap>            geom_text(data = label_df, aes(label = cluster, x = umap_1 + </span>\n",
       "<span style=white-space:pre-wrap>                0.25, y = umap_2 + 0.25), size = label_size) + </span>\n",
       "<span style=white-space:pre-wrap>            monocle3:::monocle_theme_opts()</span>\n",
       "<span style=white-space:pre-wrap>        return(plot)</span>\n",
       "<span style=white-space:pre-wrap>    }</span>\n",
       "<span style=white-space:pre-wrap>    if (length(nchar(gene_short_name)) == 1) {</span>\n",
       "<span style=white-space:pre-wrap>        label_df &lt;- data_df %&gt;% group_by(cluster) %&gt;% summarize(umap_1 = median(umap_1), </span>\n",
       "<span style=white-space:pre-wrap>            umap_2 = median(umap_2))</span>\n",
       "<span style=white-space:pre-wrap>        data_df$tmp.expr = counts(cds)[gene_short_name, ]/data_df$Size_Factor</span>\n",
       "<span style=white-space:pre-wrap>        if (is.null(size_pos)) {</span>\n",
       "<span style=white-space:pre-wrap>            size_pos = 0</span>\n",
       "<span style=white-space:pre-wrap>        }</span>\n",
       "<span style=white-space:pre-wrap>        if (is.null(stroke)) {</span>\n",
       "<span style=white-space:pre-wrap>            stroke = 0.4</span>\n",
       "<span style=white-space:pre-wrap>        }</span>\n",
       "<span style=white-space:pre-wrap>        if (is.null(label_size)) {</span>\n",
       "<span style=white-space:pre-wrap>            label_size = 4</span>\n",
       "<span style=white-space:pre-wrap>        }</span>\n",
       "<span style=white-space:pre-wrap>        if (is.null(limit)) {</span>\n",
       "<span style=white-space:pre-wrap>            limit = 1</span>\n",
       "<span style=white-space:pre-wrap>        }</span>\n",
       "<span style=white-space:pre-wrap>        if (is.null(size_neg)) {</span>\n",
       "<span style=white-space:pre-wrap>            size_neg = 0.1</span>\n",
       "<span style=white-space:pre-wrap>        }</span>\n",
       "<span style=white-space:pre-wrap>        plot = ggplot(data_df) + geom_point(aes(x = umap_1, y = umap_2, </span>\n",
       "<span style=white-space:pre-wrap>            color = log2(tmp.expr/Size_Factor + 1), alpha = tmp.expr &gt; </span>\n",
       "<span style=white-space:pre-wrap>                0), size = ifelse(data_df$tmp.exp &gt; limit, size_pos, </span>\n",
       "<span style=white-space:pre-wrap>            size_neg), stroke = stroke) + scale_color_gradientn(colors = color.pal) + </span>\n",
       "<span style=white-space:pre-wrap>            scale_alpha_manual(values = c(non.expr.alpha, 1)) + </span>\n",
       "<span style=white-space:pre-wrap>            guides(alpha = F, color = guide_colorbar(title = paste(gene_short_name, </span>\n",
       "<span style=white-space:pre-wrap>                \"\\nnormalized\\nlog2 expression\", sep = \"\"))) + </span>\n",
       "<span style=white-space:pre-wrap>            geom_text(data = label_df, aes(label = cluster, x = umap_1 + </span>\n",
       "<span style=white-space:pre-wrap>                0.25, y = umap_2 + 0.25), size = label_size) + </span>\n",
       "<span style=white-space:pre-wrap>            monocle3:::monocle_theme_opts()</span>\n",
       "<span style=white-space:pre-wrap>        return(plot)</span>\n",
       "<span style=white-space:pre-wrap>    }</span>\n",
       "}</code></pre>"
      ],
      "text/latex": [
       "\\begin{minted}{r}\n",
       "function (cds, gene\\_short\\_name, coexpr = F, ncol = NULL, x = 1, \n",
       "    y = 2, size\\_pos = NULL, stroke = NULL, non.expr.alpha = 1/3, \n",
       "    color.pal = c(\"grey85\", \"red\", \"orange\", \"yellow\"), sample.n = F, \n",
       "    label\\_size = NULL, limit = NULL, size\\_neg = NULL) \n",
       "\\{\n",
       "    if (sample.n != F) \\{\n",
       "        set.seed(42)\n",
       "        cds = cds{[}, sample(1:ncol(cds), sample.n){]}\n",
       "    \\}\n",
       "    S\\_matrix <- reducedDims(cds){[}{[}\"UMAP\"{]}{]}\n",
       "    data\\_df <- data.frame(S\\_matrix)\n",
       "    colnames(data\\_df) <- c(\"umap\\_1\", \"umap\\_2\")\n",
       "    data\\_df\\$cluster = as.vector(clusters(cds))\n",
       "    data\\_df\\$sample\\_name <- row.names(data\\_df)\n",
       "    data\\_df\\$Size\\_Factor <- colData(cds)\\$Size\\_Factor\n",
       "    data\\_df\\$lineage <- colData(cds)\\$lineage\n",
       "    data\\_df\\$species <- colData(cds)\\$species\n",
       "    if (length(nchar(gene\\_short\\_name)) > 1) \\{\n",
       "        label\\_df <- data\\_df \\%>\\% group\\_by(cluster) \\%>\\% summarize(umap\\_1 = median(umap\\_1), \n",
       "            umap\\_2 = median(umap\\_2))\n",
       "        a <- as.data.frame(as.matrix(t(counts(cds){[}gene\\_short\\_name, \n",
       "            {]})))\n",
       "        a <- a/data\\_df\\$Size\\_Factor\n",
       "        a\\$min <- apply(a, 1, min)\n",
       "        data\\_df\\$min <- a\\$min\n",
       "        if (is.null(size\\_pos)) \\{\n",
       "            size\\_pos = 0.1\n",
       "        \\}\n",
       "        if (is.null(stroke)) \\{\n",
       "            stroke = 0.4\n",
       "        \\}\n",
       "        if (is.null(label\\_size)) \\{\n",
       "            label\\_size = 4\n",
       "        \\}\n",
       "        if (is.null(limit)) \\{\n",
       "            limit = 1\n",
       "        \\}\n",
       "        if (is.null(size\\_neg)) \\{\n",
       "            size\\_neg = 0.1\n",
       "        \\}\n",
       "        plot = ggplot(data\\_df, alpha = min > 0) + geom\\_point(aes(x = umap\\_1, \n",
       "            y = umap\\_2, color = log2(min + 1)), size = ifelse(data\\_df\\$min > \n",
       "            limit, size\\_pos, size\\_neg), stroke = stroke) + scale\\_color\\_gradientn(colors = color.pal) + \n",
       "            scale\\_alpha\\_manual(values = c(non.expr.alpha, 1)) + \n",
       "            guides(alpha = F, color = guide\\_colorbar(title = \"Minimum\\textbackslash{}nnormalized\\textbackslash{}nlog2 expression\\textbackslash{}nfor specified genes\")) + \n",
       "            geom\\_text(data = label\\_df, aes(label = cluster, x = umap\\_1 + \n",
       "                0.25, y = umap\\_2 + 0.25), size = label\\_size) + \n",
       "            monocle3:::monocle\\_theme\\_opts()\n",
       "        return(plot)\n",
       "    \\}\n",
       "    if (length(nchar(gene\\_short\\_name)) == 1) \\{\n",
       "        label\\_df <- data\\_df \\%>\\% group\\_by(cluster) \\%>\\% summarize(umap\\_1 = median(umap\\_1), \n",
       "            umap\\_2 = median(umap\\_2))\n",
       "        data\\_df\\$tmp.expr = counts(cds){[}gene\\_short\\_name, {]}/data\\_df\\$Size\\_Factor\n",
       "        if (is.null(size\\_pos)) \\{\n",
       "            size\\_pos = 0\n",
       "        \\}\n",
       "        if (is.null(stroke)) \\{\n",
       "            stroke = 0.4\n",
       "        \\}\n",
       "        if (is.null(label\\_size)) \\{\n",
       "            label\\_size = 4\n",
       "        \\}\n",
       "        if (is.null(limit)) \\{\n",
       "            limit = 1\n",
       "        \\}\n",
       "        if (is.null(size\\_neg)) \\{\n",
       "            size\\_neg = 0.1\n",
       "        \\}\n",
       "        plot = ggplot(data\\_df) + geom\\_point(aes(x = umap\\_1, y = umap\\_2, \n",
       "            color = log2(tmp.expr/Size\\_Factor + 1), alpha = tmp.expr > \n",
       "                0), size = ifelse(data\\_df\\$tmp.exp > limit, size\\_pos, \n",
       "            size\\_neg), stroke = stroke) + scale\\_color\\_gradientn(colors = color.pal) + \n",
       "            scale\\_alpha\\_manual(values = c(non.expr.alpha, 1)) + \n",
       "            guides(alpha = F, color = guide\\_colorbar(title = paste(gene\\_short\\_name, \n",
       "                \"\\textbackslash{}nnormalized\\textbackslash{}nlog2 expression\", sep = \"\"))) + \n",
       "            geom\\_text(data = label\\_df, aes(label = cluster, x = umap\\_1 + \n",
       "                0.25, y = umap\\_2 + 0.25), size = label\\_size) + \n",
       "            monocle3:::monocle\\_theme\\_opts()\n",
       "        return(plot)\n",
       "    \\}\n",
       "\\}\n",
       "\\end{minted}"
      ],
      "text/markdown": [
       "```r\n",
       "function (cds, gene_short_name, coexpr = F, ncol = NULL, x = 1, \n",
       "    y = 2, size_pos = NULL, stroke = NULL, non.expr.alpha = 1/3, \n",
       "    color.pal = c(\"grey85\", \"red\", \"orange\", \"yellow\"), sample.n = F, \n",
       "    label_size = NULL, limit = NULL, size_neg = NULL) \n",
       "{\n",
       "    if (sample.n != F) {\n",
       "        set.seed(42)\n",
       "        cds = cds[, sample(1:ncol(cds), sample.n)]\n",
       "    }\n",
       "    S_matrix <- reducedDims(cds)[[\"UMAP\"]]\n",
       "    data_df <- data.frame(S_matrix)\n",
       "    colnames(data_df) <- c(\"umap_1\", \"umap_2\")\n",
       "    data_df$cluster = as.vector(clusters(cds))\n",
       "    data_df$sample_name <- row.names(data_df)\n",
       "    data_df$Size_Factor <- colData(cds)$Size_Factor\n",
       "    data_df$lineage <- colData(cds)$lineage\n",
       "    data_df$species <- colData(cds)$species\n",
       "    if (length(nchar(gene_short_name)) > 1) {\n",
       "        label_df <- data_df %>% group_by(cluster) %>% summarize(umap_1 = median(umap_1), \n",
       "            umap_2 = median(umap_2))\n",
       "        a <- as.data.frame(as.matrix(t(counts(cds)[gene_short_name, \n",
       "            ])))\n",
       "        a <- a/data_df$Size_Factor\n",
       "        a$min <- apply(a, 1, min)\n",
       "        data_df$min <- a$min\n",
       "        if (is.null(size_pos)) {\n",
       "            size_pos = 0.1\n",
       "        }\n",
       "        if (is.null(stroke)) {\n",
       "            stroke = 0.4\n",
       "        }\n",
       "        if (is.null(label_size)) {\n",
       "            label_size = 4\n",
       "        }\n",
       "        if (is.null(limit)) {\n",
       "            limit = 1\n",
       "        }\n",
       "        if (is.null(size_neg)) {\n",
       "            size_neg = 0.1\n",
       "        }\n",
       "        plot = ggplot(data_df, alpha = min > 0) + geom_point(aes(x = umap_1, \n",
       "            y = umap_2, color = log2(min + 1)), size = ifelse(data_df$min > \n",
       "            limit, size_pos, size_neg), stroke = stroke) + scale_color_gradientn(colors = color.pal) + \n",
       "            scale_alpha_manual(values = c(non.expr.alpha, 1)) + \n",
       "            guides(alpha = F, color = guide_colorbar(title = \"Minimum\\nnormalized\\nlog2 expression\\nfor specified genes\")) + \n",
       "            geom_text(data = label_df, aes(label = cluster, x = umap_1 + \n",
       "                0.25, y = umap_2 + 0.25), size = label_size) + \n",
       "            monocle3:::monocle_theme_opts()\n",
       "        return(plot)\n",
       "    }\n",
       "    if (length(nchar(gene_short_name)) == 1) {\n",
       "        label_df <- data_df %>% group_by(cluster) %>% summarize(umap_1 = median(umap_1), \n",
       "            umap_2 = median(umap_2))\n",
       "        data_df$tmp.expr = counts(cds)[gene_short_name, ]/data_df$Size_Factor\n",
       "        if (is.null(size_pos)) {\n",
       "            size_pos = 0\n",
       "        }\n",
       "        if (is.null(stroke)) {\n",
       "            stroke = 0.4\n",
       "        }\n",
       "        if (is.null(label_size)) {\n",
       "            label_size = 4\n",
       "        }\n",
       "        if (is.null(limit)) {\n",
       "            limit = 1\n",
       "        }\n",
       "        if (is.null(size_neg)) {\n",
       "            size_neg = 0.1\n",
       "        }\n",
       "        plot = ggplot(data_df) + geom_point(aes(x = umap_1, y = umap_2, \n",
       "            color = log2(tmp.expr/Size_Factor + 1), alpha = tmp.expr > \n",
       "                0), size = ifelse(data_df$tmp.exp > limit, size_pos, \n",
       "            size_neg), stroke = stroke) + scale_color_gradientn(colors = color.pal) + \n",
       "            scale_alpha_manual(values = c(non.expr.alpha, 1)) + \n",
       "            guides(alpha = F, color = guide_colorbar(title = paste(gene_short_name, \n",
       "                \"\\nnormalized\\nlog2 expression\", sep = \"\"))) + \n",
       "            geom_text(data = label_df, aes(label = cluster, x = umap_1 + \n",
       "                0.25, y = umap_2 + 0.25), size = label_size) + \n",
       "            monocle3:::monocle_theme_opts()\n",
       "        return(plot)\n",
       "    }\n",
       "}\n",
       "```"
      ],
      "text/plain": [
       "function (cds, gene_short_name, coexpr = F, ncol = NULL, x = 1, \n",
       "    y = 2, size_pos = NULL, stroke = NULL, non.expr.alpha = 1/3, \n",
       "    color.pal = c(\"grey85\", \"red\", \"orange\", \"yellow\"), sample.n = F, \n",
       "    label_size = NULL, limit = NULL, size_neg = NULL) \n",
       "{\n",
       "    if (sample.n != F) {\n",
       "        set.seed(42)\n",
       "        cds = cds[, sample(1:ncol(cds), sample.n)]\n",
       "    }\n",
       "    S_matrix <- reducedDims(cds)[[\"UMAP\"]]\n",
       "    data_df <- data.frame(S_matrix)\n",
       "    colnames(data_df) <- c(\"umap_1\", \"umap_2\")\n",
       "    data_df$cluster = as.vector(clusters(cds))\n",
       "    data_df$sample_name <- row.names(data_df)\n",
       "    data_df$Size_Factor <- colData(cds)$Size_Factor\n",
       "    data_df$lineage <- colData(cds)$lineage\n",
       "    data_df$species <- colData(cds)$species\n",
       "    if (length(nchar(gene_short_name)) > 1) {\n",
       "        label_df <- data_df %>% group_by(cluster) %>% summarize(umap_1 = median(umap_1), \n",
       "            umap_2 = median(umap_2))\n",
       "        a <- as.data.frame(as.matrix(t(counts(cds)[gene_short_name, \n",
       "            ])))\n",
       "        a <- a/data_df$Size_Factor\n",
       "        a$min <- apply(a, 1, min)\n",
       "        data_df$min <- a$min\n",
       "        if (is.null(size_pos)) {\n",
       "            size_pos = 0.1\n",
       "        }\n",
       "        if (is.null(stroke)) {\n",
       "            stroke = 0.4\n",
       "        }\n",
       "        if (is.null(label_size)) {\n",
       "            label_size = 4\n",
       "        }\n",
       "        if (is.null(limit)) {\n",
       "            limit = 1\n",
       "        }\n",
       "        if (is.null(size_neg)) {\n",
       "            size_neg = 0.1\n",
       "        }\n",
       "        plot = ggplot(data_df, alpha = min > 0) + geom_point(aes(x = umap_1, \n",
       "            y = umap_2, color = log2(min + 1)), size = ifelse(data_df$min > \n",
       "            limit, size_pos, size_neg), stroke = stroke) + scale_color_gradientn(colors = color.pal) + \n",
       "            scale_alpha_manual(values = c(non.expr.alpha, 1)) + \n",
       "            guides(alpha = F, color = guide_colorbar(title = \"Minimum\\nnormalized\\nlog2 expression\\nfor specified genes\")) + \n",
       "            geom_text(data = label_df, aes(label = cluster, x = umap_1 + \n",
       "                0.25, y = umap_2 + 0.25), size = label_size) + \n",
       "            monocle3:::monocle_theme_opts()\n",
       "        return(plot)\n",
       "    }\n",
       "    if (length(nchar(gene_short_name)) == 1) {\n",
       "        label_df <- data_df %>% group_by(cluster) %>% summarize(umap_1 = median(umap_1), \n",
       "            umap_2 = median(umap_2))\n",
       "        data_df$tmp.expr = counts(cds)[gene_short_name, ]/data_df$Size_Factor\n",
       "        if (is.null(size_pos)) {\n",
       "            size_pos = 0\n",
       "        }\n",
       "        if (is.null(stroke)) {\n",
       "            stroke = 0.4\n",
       "        }\n",
       "        if (is.null(label_size)) {\n",
       "            label_size = 4\n",
       "        }\n",
       "        if (is.null(limit)) {\n",
       "            limit = 1\n",
       "        }\n",
       "        if (is.null(size_neg)) {\n",
       "            size_neg = 0.1\n",
       "        }\n",
       "        plot = ggplot(data_df) + geom_point(aes(x = umap_1, y = umap_2, \n",
       "            color = log2(tmp.expr/Size_Factor + 1), alpha = tmp.expr > \n",
       "                0), size = ifelse(data_df$tmp.exp > limit, size_pos, \n",
       "            size_neg), stroke = stroke) + scale_color_gradientn(colors = color.pal) + \n",
       "            scale_alpha_manual(values = c(non.expr.alpha, 1)) + \n",
       "            guides(alpha = F, color = guide_colorbar(title = paste(gene_short_name, \n",
       "                \"\\nnormalized\\nlog2 expression\", sep = \"\"))) + \n",
       "            geom_text(data = label_df, aes(label = cluster, x = umap_1 + \n",
       "                0.25, y = umap_2 + 0.25), size = label_size) + \n",
       "            monocle3:::monocle_theme_opts()\n",
       "        return(plot)\n",
       "    }\n",
       "}\n",
       "<bytecode: 0x7fa45f38d7b0>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "plot.expr_clus.br2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 313,
   "metadata": {},
   "outputs": [],
   "source": [
    "plot.expr.fly = function (cds, gene_short_name, coexpr = F, ncol = NULL, x = 1, \n",
    "    y = 2, size_pos = NULL, stroke = NULL, non.expr.alpha = 1/3, \n",
    "    color.pal = c(\"grey85\", \"red\", \"orange\", \"yellow\"), sample.n = F, \n",
    "    label_size = NULL, limit = NULL, size_neg = NULL) \n",
    "{\n",
    "    if (sample.n != F) {\n",
    "        set.seed(42)\n",
    "        cds = cds[, sample(1:ncol(cds), sample.n)]\n",
    "    }\n",
    "    S_matrix <- reducedDims(cds)[[\"UMAP\"]]\n",
    "    data_df <- data.frame(S_matrix)\n",
    "    colnames(data_df) <- c(\"umap_1\", \"umap_2\")\n",
    "    data_df$cell.type.2 = colData(cds)$cell.type.2\n",
    "    data_df$sample_name <- row.names(data_df)\n",
    "    data_df$Size_Factor <- colData(cds)$Size_Factor\n",
    "    data_df$lineage <- colData(cds)$lineage\n",
    "    data_df$species <- colData(cds)$species\n",
    "    if (length(nchar(gene_short_name)) > 1) {\n",
    "        label_df <- data_df %>% group_by(cell.type.2) %>% summarize(umap_1 = median(umap_1), \n",
    "            umap_2 = median(umap_2))\n",
    "        a <- as.data.frame(as.matrix(t(counts(cds)[gene_short_name, \n",
    "            ])))\n",
    "        a <- a/data_df$Size_Factor\n",
    "        a$min <- apply(a, 1, min)\n",
    "        data_df$min <- a$min\n",
    "        if (is.null(size_pos)) {\n",
    "            size_pos = 0.1\n",
    "        }\n",
    "        if (is.null(stroke)) {\n",
    "            stroke = 0.4\n",
    "        }\n",
    "        if (is.null(label_size)) {\n",
    "            label_size = 4\n",
    "        }\n",
    "        if (is.null(limit)) {\n",
    "            limit = 1\n",
    "        }\n",
    "        if (is.null(size_neg)) {\n",
    "            size_neg = 0.1\n",
    "        }\n",
    "        plot = ggplot(data_df, alpha = min > 0) + geom_point(aes(x = umap_1, \n",
    "            y = umap_2, color = log2(min + 1)), size = ifelse(data_df$min > \n",
    "            limit, size_pos, size_neg), stroke = stroke) + scale_color_gradientn(colors = color.pal) + \n",
    "            scale_alpha_manual(values = c(non.expr.alpha, 1)) + \n",
    "            guides(alpha = F, color = guide_colorbar(title = \"Minimum\\nnormalized\\nlog2 expression\\nfor specified genes\")) + \n",
    "            geom_text(data = label_df, aes(label = cell.type.2, x = umap_1 + \n",
    "                0.25, y = umap_2 + 0.25), size = label_size) + \n",
    "            monocle3:::monocle_theme_opts()\n",
    "        return(plot)\n",
    "    }\n",
    "    if (length(nchar(gene_short_name)) == 1) {\n",
    "        label_df <- data_df %>% group_by(cell.type.2) %>% summarize(umap_1 = median(umap_1), \n",
    "            umap_2 = median(umap_2))\n",
    "        data_df$tmp.expr = counts(cds)[gene_short_name, ]/data_df$Size_Factor\n",
    "        if (is.null(size_pos)) {\n",
    "            size_pos = 0\n",
    "        }\n",
    "        if (is.null(stroke)) {\n",
    "            stroke = 0.4\n",
    "        }\n",
    "        if (is.null(label_size)) {\n",
    "            label_size = 4\n",
    "        }\n",
    "        if (is.null(limit)) {\n",
    "            limit = 1\n",
    "        }\n",
    "        if (is.null(size_neg)) {\n",
    "            size_neg = 0.1\n",
    "        }\n",
    "        plot = ggplot(data_df) + geom_point(aes(x = umap_1, y = umap_2, \n",
    "            color = log2(tmp.expr/Size_Factor + 1), alpha = tmp.expr > \n",
    "                0), size = ifelse(data_df$tmp.exp > limit, size_pos, \n",
    "            size_neg), stroke = stroke) + scale_color_gradientn(colors = color.pal) + \n",
    "            scale_alpha_manual(values = c(non.expr.alpha, 1)) + \n",
    "            guides(alpha = F, color = guide_colorbar(title = paste(gene_short_name, \n",
    "                \"\\nnormalized\\nlog2 expression\", sep = \"\"))) + \n",
    "            geom_text(data = label_df, aes(label = cell.type.2, x = umap_1 + \n",
    "                0.25, y = umap_2 + 0.25), size = label_size) + \n",
    "            monocle3:::monocle_theme_opts()\n",
    "        return(plot)\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### distribution of annot cell types in cds_sub"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 345,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'amnioserosa'</li><li>'anlage in statu nascendi'</li><li>'brain'</li><li>'brain primordium'</li><li>'cellular blastoderm / pole cell'</li><li>'crystal cell'</li><li>'ectoderm anlage'</li><li>'ectoderm anlage in statu nascendi'</li><li>'ectoderm primordium'</li><li>'endoderm primordium'</li><li>'epidermis'</li><li>'epidermis primordium'</li><li>'fat body'</li><li>'foregut / hindgut / anal pad'</li><li>'foregut / hindgut primordium'</li><li>'germ cell'</li><li>'glia'</li><li>'hindgut / anal pad'</li><li>'hindgut primordium'</li><li>'Malpighian tubule'</li><li>'maternal'</li><li>'mesoderm primordium'</li><li>'midgut'</li><li>'midgut primordium'</li><li>'midline primordium'</li><li>'muscle'</li><li>'peripheral nervous system'</li><li>'peripheral nervous system primordium'</li><li>'plasmatocytes'</li><li>'plasmatocytes anlage'</li><li>'procrystal cell'</li><li>'proventriculus'</li><li>'salivary gland'</li><li>'salivary gland body primordium'</li><li>'salivary gland primordium'</li><li>'sensory nervous system'</li><li>'sensory nervous system primordium'</li><li>'somatic muscle'</li><li>'somatic muscle primordium'</li><li>'spiracle (tracheal system)'</li><li>'spiracles (tracheal system)'</li><li>'tracheal primordium'</li><li>'tracheal system'</li><li>'trunk mesoderm primordium'</li><li>'ubiquitous'</li><li>'unknown'</li><li>'ventral nerve cord'</li><li>'ventral nerve cord primordium'</li><li>'visceral muscle'</li><li>'visceral muscle primordium'</li><li>'yolk nuclei'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'amnioserosa'\n",
       "\\item 'anlage in statu nascendi'\n",
       "\\item 'brain'\n",
       "\\item 'brain primordium'\n",
       "\\item 'cellular blastoderm / pole cell'\n",
       "\\item 'crystal cell'\n",
       "\\item 'ectoderm anlage'\n",
       "\\item 'ectoderm anlage in statu nascendi'\n",
       "\\item 'ectoderm primordium'\n",
       "\\item 'endoderm primordium'\n",
       "\\item 'epidermis'\n",
       "\\item 'epidermis primordium'\n",
       "\\item 'fat body'\n",
       "\\item 'foregut / hindgut / anal pad'\n",
       "\\item 'foregut / hindgut primordium'\n",
       "\\item 'germ cell'\n",
       "\\item 'glia'\n",
       "\\item 'hindgut / anal pad'\n",
       "\\item 'hindgut primordium'\n",
       "\\item 'Malpighian tubule'\n",
       "\\item 'maternal'\n",
       "\\item 'mesoderm primordium'\n",
       "\\item 'midgut'\n",
       "\\item 'midgut primordium'\n",
       "\\item 'midline primordium'\n",
       "\\item 'muscle'\n",
       "\\item 'peripheral nervous system'\n",
       "\\item 'peripheral nervous system primordium'\n",
       "\\item 'plasmatocytes'\n",
       "\\item 'plasmatocytes anlage'\n",
       "\\item 'procrystal cell'\n",
       "\\item 'proventriculus'\n",
       "\\item 'salivary gland'\n",
       "\\item 'salivary gland body primordium'\n",
       "\\item 'salivary gland primordium'\n",
       "\\item 'sensory nervous system'\n",
       "\\item 'sensory nervous system primordium'\n",
       "\\item 'somatic muscle'\n",
       "\\item 'somatic muscle primordium'\n",
       "\\item 'spiracle (tracheal system)'\n",
       "\\item 'spiracles (tracheal system)'\n",
       "\\item 'tracheal primordium'\n",
       "\\item 'tracheal system'\n",
       "\\item 'trunk mesoderm primordium'\n",
       "\\item 'ubiquitous'\n",
       "\\item 'unknown'\n",
       "\\item 'ventral nerve cord'\n",
       "\\item 'ventral nerve cord primordium'\n",
       "\\item 'visceral muscle'\n",
       "\\item 'visceral muscle primordium'\n",
       "\\item 'yolk nuclei'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'amnioserosa'\n",
       "2. 'anlage in statu nascendi'\n",
       "3. 'brain'\n",
       "4. 'brain primordium'\n",
       "5. 'cellular blastoderm / pole cell'\n",
       "6. 'crystal cell'\n",
       "7. 'ectoderm anlage'\n",
       "8. 'ectoderm anlage in statu nascendi'\n",
       "9. 'ectoderm primordium'\n",
       "10. 'endoderm primordium'\n",
       "11. 'epidermis'\n",
       "12. 'epidermis primordium'\n",
       "13. 'fat body'\n",
       "14. 'foregut / hindgut / anal pad'\n",
       "15. 'foregut / hindgut primordium'\n",
       "16. 'germ cell'\n",
       "17. 'glia'\n",
       "18. 'hindgut / anal pad'\n",
       "19. 'hindgut primordium'\n",
       "20. 'Malpighian tubule'\n",
       "21. 'maternal'\n",
       "22. 'mesoderm primordium'\n",
       "23. 'midgut'\n",
       "24. 'midgut primordium'\n",
       "25. 'midline primordium'\n",
       "26. 'muscle'\n",
       "27. 'peripheral nervous system'\n",
       "28. 'peripheral nervous system primordium'\n",
       "29. 'plasmatocytes'\n",
       "30. 'plasmatocytes anlage'\n",
       "31. 'procrystal cell'\n",
       "32. 'proventriculus'\n",
       "33. 'salivary gland'\n",
       "34. 'salivary gland body primordium'\n",
       "35. 'salivary gland primordium'\n",
       "36. 'sensory nervous system'\n",
       "37. 'sensory nervous system primordium'\n",
       "38. 'somatic muscle'\n",
       "39. 'somatic muscle primordium'\n",
       "40. 'spiracle (tracheal system)'\n",
       "41. 'spiracles (tracheal system)'\n",
       "42. 'tracheal primordium'\n",
       "43. 'tracheal system'\n",
       "44. 'trunk mesoderm primordium'\n",
       "45. 'ubiquitous'\n",
       "46. 'unknown'\n",
       "47. 'ventral nerve cord'\n",
       "48. 'ventral nerve cord primordium'\n",
       "49. 'visceral muscle'\n",
       "50. 'visceral muscle primordium'\n",
       "51. 'yolk nuclei'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"amnioserosa\"                         \n",
       " [2] \"anlage in statu nascendi\"            \n",
       " [3] \"brain\"                               \n",
       " [4] \"brain primordium\"                    \n",
       " [5] \"cellular blastoderm / pole cell\"     \n",
       " [6] \"crystal cell\"                        \n",
       " [7] \"ectoderm anlage\"                     \n",
       " [8] \"ectoderm anlage in statu nascendi\"   \n",
       " [9] \"ectoderm primordium\"                 \n",
       "[10] \"endoderm primordium\"                 \n",
       "[11] \"epidermis\"                           \n",
       "[12] \"epidermis primordium\"                \n",
       "[13] \"fat body\"                            \n",
       "[14] \"foregut / hindgut / anal pad\"        \n",
       "[15] \"foregut / hindgut primordium\"        \n",
       "[16] \"germ cell\"                           \n",
       "[17] \"glia\"                                \n",
       "[18] \"hindgut / anal pad\"                  \n",
       "[19] \"hindgut primordium\"                  \n",
       "[20] \"Malpighian tubule\"                   \n",
       "[21] \"maternal\"                            \n",
       "[22] \"mesoderm primordium\"                 \n",
       "[23] \"midgut\"                              \n",
       "[24] \"midgut primordium\"                   \n",
       "[25] \"midline primordium\"                  \n",
       "[26] \"muscle\"                              \n",
       "[27] \"peripheral nervous system\"           \n",
       "[28] \"peripheral nervous system primordium\"\n",
       "[29] \"plasmatocytes\"                       \n",
       "[30] \"plasmatocytes anlage\"                \n",
       "[31] \"procrystal cell\"                     \n",
       "[32] \"proventriculus\"                      \n",
       "[33] \"salivary gland\"                      \n",
       "[34] \"salivary gland body primordium\"      \n",
       "[35] \"salivary gland primordium\"           \n",
       "[36] \"sensory nervous system\"              \n",
       "[37] \"sensory nervous system primordium\"   \n",
       "[38] \"somatic muscle\"                      \n",
       "[39] \"somatic muscle primordium\"           \n",
       "[40] \"spiracle (tracheal system)\"          \n",
       "[41] \"spiracles (tracheal system)\"         \n",
       "[42] \"tracheal primordium\"                 \n",
       "[43] \"tracheal system\"                     \n",
       "[44] \"trunk mesoderm primordium\"           \n",
       "[45] \"ubiquitous\"                          \n",
       "[46] \"unknown\"                             \n",
       "[47] \"ventral nerve cord\"                  \n",
       "[48] \"ventral nerve cord primordium\"       \n",
       "[49] \"visceral muscle\"                     \n",
       "[50] \"visceral muscle primordium\"          \n",
       "[51] \"yolk nuclei\"                         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(df_sub_no_doubletons$annot)  %>% sort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "annot_type = \"yolk nuclei\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$annot %in% annot_type), size =  ifelse(df_sub300$annot %in% annot_type, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 444,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 74 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cell.type.2</th><th scope=col>annot</th><th scope=col>n</th><th scope=col>rank</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ca</td><td>yolk nuclei</td><td>36</td><td>22.0</td></tr>\n",
       "\t<tr><td>bl</td><td>yolk nuclei</td><td>33</td><td>26.0</td></tr>\n",
       "\t<tr><td>bt</td><td>yolk nuclei</td><td>33</td><td> 6.0</td></tr>\n",
       "\t<tr><td>s </td><td>yolk nuclei</td><td>32</td><td>24.0</td></tr>\n",
       "\t<tr><td>bq</td><td>yolk nuclei</td><td>31</td><td> 3.0</td></tr>\n",
       "\t<tr><td>i </td><td>yolk nuclei</td><td>26</td><td>13.0</td></tr>\n",
       "\t<tr><td>bk</td><td>yolk nuclei</td><td>21</td><td>21.5</td></tr>\n",
       "\t<tr><td>bs</td><td>yolk nuclei</td><td>21</td><td> 5.0</td></tr>\n",
       "\t<tr><td>bx</td><td>yolk nuclei</td><td>21</td><td> 6.0</td></tr>\n",
       "\t<tr><td>br</td><td>yolk nuclei</td><td>20</td><td> 5.0</td></tr>\n",
       "\t<tr><td>cb</td><td>yolk nuclei</td><td>18</td><td>21.5</td></tr>\n",
       "\t<tr><td>bj</td><td>yolk nuclei</td><td>16</td><td>26.5</td></tr>\n",
       "\t<tr><td>cc</td><td>yolk nuclei</td><td>16</td><td>15.5</td></tr>\n",
       "\t<tr><td>bp</td><td>yolk nuclei</td><td>14</td><td>11.5</td></tr>\n",
       "\t<tr><td>bv</td><td>yolk nuclei</td><td>14</td><td> 5.0</td></tr>\n",
       "\t<tr><td>ce</td><td>yolk nuclei</td><td>13</td><td>18.5</td></tr>\n",
       "\t<tr><td>be</td><td>yolk nuclei</td><td>10</td><td>17.5</td></tr>\n",
       "\t<tr><td>bw</td><td>yolk nuclei</td><td> 9</td><td> 4.5</td></tr>\n",
       "\t<tr><td>r </td><td>yolk nuclei</td><td> 8</td><td>13.0</td></tr>\n",
       "\t<tr><td>bm</td><td>yolk nuclei</td><td> 7</td><td>13.5</td></tr>\n",
       "\t<tr><td>aq</td><td>yolk nuclei</td><td> 6</td><td> 9.5</td></tr>\n",
       "\t<tr><td>az</td><td>yolk nuclei</td><td> 6</td><td>15.0</td></tr>\n",
       "\t<tr><td>bo</td><td>yolk nuclei</td><td> 6</td><td>15.5</td></tr>\n",
       "\t<tr><td>aa</td><td>yolk nuclei</td><td> 5</td><td> 7.5</td></tr>\n",
       "\t<tr><td>ab</td><td>yolk nuclei</td><td> 5</td><td>12.0</td></tr>\n",
       "\t<tr><td>an</td><td>yolk nuclei</td><td> 5</td><td> 6.5</td></tr>\n",
       "\t<tr><td>at</td><td>yolk nuclei</td><td> 5</td><td> 8.5</td></tr>\n",
       "\t<tr><td>bc</td><td>yolk nuclei</td><td> 5</td><td>18.0</td></tr>\n",
       "\t<tr><td>bu</td><td>yolk nuclei</td><td> 5</td><td> 3.0</td></tr>\n",
       "\t<tr><td>by</td><td>yolk nuclei</td><td> 5</td><td>11.0</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>as</td><td>yolk nuclei</td><td>3</td><td> 6.5</td></tr>\n",
       "\t<tr><td>ay</td><td>yolk nuclei</td><td>3</td><td> 3.5</td></tr>\n",
       "\t<tr><td>ba</td><td>yolk nuclei</td><td>3</td><td> 7.0</td></tr>\n",
       "\t<tr><td>bf</td><td>yolk nuclei</td><td>3</td><td>10.0</td></tr>\n",
       "\t<tr><td>l </td><td>yolk nuclei</td><td>3</td><td>14.5</td></tr>\n",
       "\t<tr><td>o </td><td>yolk nuclei</td><td>3</td><td>11.0</td></tr>\n",
       "\t<tr><td>y </td><td>yolk nuclei</td><td>3</td><td> 4.0</td></tr>\n",
       "\t<tr><td>a </td><td>yolk nuclei</td><td>2</td><td> 7.0</td></tr>\n",
       "\t<tr><td>ac</td><td>yolk nuclei</td><td>2</td><td> 9.0</td></tr>\n",
       "\t<tr><td>ad</td><td>yolk nuclei</td><td>2</td><td> 3.0</td></tr>\n",
       "\t<tr><td>ag</td><td>yolk nuclei</td><td>2</td><td>12.0</td></tr>\n",
       "\t<tr><td>al</td><td>yolk nuclei</td><td>2</td><td> 6.5</td></tr>\n",
       "\t<tr><td>au</td><td>yolk nuclei</td><td>2</td><td> 4.0</td></tr>\n",
       "\t<tr><td>aw</td><td>yolk nuclei</td><td>2</td><td> 5.5</td></tr>\n",
       "\t<tr><td>ax</td><td>yolk nuclei</td><td>2</td><td> 5.5</td></tr>\n",
       "\t<tr><td>b </td><td>yolk nuclei</td><td>2</td><td> 9.0</td></tr>\n",
       "\t<tr><td>bb</td><td>yolk nuclei</td><td>2</td><td> 8.0</td></tr>\n",
       "\t<tr><td>bd</td><td>yolk nuclei</td><td>2</td><td> 2.0</td></tr>\n",
       "\t<tr><td>bz</td><td>yolk nuclei</td><td>2</td><td> 5.0</td></tr>\n",
       "\t<tr><td>f </td><td>yolk nuclei</td><td>2</td><td>12.0</td></tr>\n",
       "\t<tr><td>af</td><td>yolk nuclei</td><td>1</td><td> 3.5</td></tr>\n",
       "\t<tr><td>ao</td><td>yolk nuclei</td><td>1</td><td> 3.5</td></tr>\n",
       "\t<tr><td>bi</td><td>yolk nuclei</td><td>1</td><td> 2.5</td></tr>\n",
       "\t<tr><td>cf</td><td>yolk nuclei</td><td>1</td><td> 2.5</td></tr>\n",
       "\t<tr><td>g </td><td>yolk nuclei</td><td>1</td><td> 6.0</td></tr>\n",
       "\t<tr><td>j </td><td>yolk nuclei</td><td>1</td><td> 2.5</td></tr>\n",
       "\t<tr><td>m </td><td>yolk nuclei</td><td>1</td><td> 2.5</td></tr>\n",
       "\t<tr><td>n </td><td>yolk nuclei</td><td>1</td><td> 6.0</td></tr>\n",
       "\t<tr><td>u </td><td>yolk nuclei</td><td>1</td><td> 2.5</td></tr>\n",
       "\t<tr><td>x </td><td>yolk nuclei</td><td>1</td><td> 2.0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 74 × 4\n",
       "\\begin{tabular}{llll}\n",
       " cell.type.2 & annot & n & rank\\\\\n",
       " <chr> & <chr> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t ca & yolk nuclei & 36 & 22.0\\\\\n",
       "\t bl & yolk nuclei & 33 & 26.0\\\\\n",
       "\t bt & yolk nuclei & 33 &  6.0\\\\\n",
       "\t s  & yolk nuclei & 32 & 24.0\\\\\n",
       "\t bq & yolk nuclei & 31 &  3.0\\\\\n",
       "\t i  & yolk nuclei & 26 & 13.0\\\\\n",
       "\t bk & yolk nuclei & 21 & 21.5\\\\\n",
       "\t bs & yolk nuclei & 21 &  5.0\\\\\n",
       "\t bx & yolk nuclei & 21 &  6.0\\\\\n",
       "\t br & yolk nuclei & 20 &  5.0\\\\\n",
       "\t cb & yolk nuclei & 18 & 21.5\\\\\n",
       "\t bj & yolk nuclei & 16 & 26.5\\\\\n",
       "\t cc & yolk nuclei & 16 & 15.5\\\\\n",
       "\t bp & yolk nuclei & 14 & 11.5\\\\\n",
       "\t bv & yolk nuclei & 14 &  5.0\\\\\n",
       "\t ce & yolk nuclei & 13 & 18.5\\\\\n",
       "\t be & yolk nuclei & 10 & 17.5\\\\\n",
       "\t bw & yolk nuclei &  9 &  4.5\\\\\n",
       "\t r  & yolk nuclei &  8 & 13.0\\\\\n",
       "\t bm & yolk nuclei &  7 & 13.5\\\\\n",
       "\t aq & yolk nuclei &  6 &  9.5\\\\\n",
       "\t az & yolk nuclei &  6 & 15.0\\\\\n",
       "\t bo & yolk nuclei &  6 & 15.5\\\\\n",
       "\t aa & yolk nuclei &  5 &  7.5\\\\\n",
       "\t ab & yolk nuclei &  5 & 12.0\\\\\n",
       "\t an & yolk nuclei &  5 &  6.5\\\\\n",
       "\t at & yolk nuclei &  5 &  8.5\\\\\n",
       "\t bc & yolk nuclei &  5 & 18.0\\\\\n",
       "\t bu & yolk nuclei &  5 &  3.0\\\\\n",
       "\t by & yolk nuclei &  5 & 11.0\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t as & yolk nuclei & 3 &  6.5\\\\\n",
       "\t ay & yolk nuclei & 3 &  3.5\\\\\n",
       "\t ba & yolk nuclei & 3 &  7.0\\\\\n",
       "\t bf & yolk nuclei & 3 & 10.0\\\\\n",
       "\t l  & yolk nuclei & 3 & 14.5\\\\\n",
       "\t o  & yolk nuclei & 3 & 11.0\\\\\n",
       "\t y  & yolk nuclei & 3 &  4.0\\\\\n",
       "\t a  & yolk nuclei & 2 &  7.0\\\\\n",
       "\t ac & yolk nuclei & 2 &  9.0\\\\\n",
       "\t ad & yolk nuclei & 2 &  3.0\\\\\n",
       "\t ag & yolk nuclei & 2 & 12.0\\\\\n",
       "\t al & yolk nuclei & 2 &  6.5\\\\\n",
       "\t au & yolk nuclei & 2 &  4.0\\\\\n",
       "\t aw & yolk nuclei & 2 &  5.5\\\\\n",
       "\t ax & yolk nuclei & 2 &  5.5\\\\\n",
       "\t b  & yolk nuclei & 2 &  9.0\\\\\n",
       "\t bb & yolk nuclei & 2 &  8.0\\\\\n",
       "\t bd & yolk nuclei & 2 &  2.0\\\\\n",
       "\t bz & yolk nuclei & 2 &  5.0\\\\\n",
       "\t f  & yolk nuclei & 2 & 12.0\\\\\n",
       "\t af & yolk nuclei & 1 &  3.5\\\\\n",
       "\t ao & yolk nuclei & 1 &  3.5\\\\\n",
       "\t bi & yolk nuclei & 1 &  2.5\\\\\n",
       "\t cf & yolk nuclei & 1 &  2.5\\\\\n",
       "\t g  & yolk nuclei & 1 &  6.0\\\\\n",
       "\t j  & yolk nuclei & 1 &  2.5\\\\\n",
       "\t m  & yolk nuclei & 1 &  2.5\\\\\n",
       "\t n  & yolk nuclei & 1 &  6.0\\\\\n",
       "\t u  & yolk nuclei & 1 &  2.5\\\\\n",
       "\t x  & yolk nuclei & 1 &  2.0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 74 × 4\n",
       "\n",
       "| cell.type.2 &lt;chr&gt; | annot &lt;chr&gt; | n &lt;int&gt; | rank &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| ca | yolk nuclei | 36 | 22.0 |\n",
       "| bl | yolk nuclei | 33 | 26.0 |\n",
       "| bt | yolk nuclei | 33 |  6.0 |\n",
       "| s  | yolk nuclei | 32 | 24.0 |\n",
       "| bq | yolk nuclei | 31 |  3.0 |\n",
       "| i  | yolk nuclei | 26 | 13.0 |\n",
       "| bk | yolk nuclei | 21 | 21.5 |\n",
       "| bs | yolk nuclei | 21 |  5.0 |\n",
       "| bx | yolk nuclei | 21 |  6.0 |\n",
       "| br | yolk nuclei | 20 |  5.0 |\n",
       "| cb | yolk nuclei | 18 | 21.5 |\n",
       "| bj | yolk nuclei | 16 | 26.5 |\n",
       "| cc | yolk nuclei | 16 | 15.5 |\n",
       "| bp | yolk nuclei | 14 | 11.5 |\n",
       "| bv | yolk nuclei | 14 |  5.0 |\n",
       "| ce | yolk nuclei | 13 | 18.5 |\n",
       "| be | yolk nuclei | 10 | 17.5 |\n",
       "| bw | yolk nuclei |  9 |  4.5 |\n",
       "| r  | yolk nuclei |  8 | 13.0 |\n",
       "| bm | yolk nuclei |  7 | 13.5 |\n",
       "| aq | yolk nuclei |  6 |  9.5 |\n",
       "| az | yolk nuclei |  6 | 15.0 |\n",
       "| bo | yolk nuclei |  6 | 15.5 |\n",
       "| aa | yolk nuclei |  5 |  7.5 |\n",
       "| ab | yolk nuclei |  5 | 12.0 |\n",
       "| an | yolk nuclei |  5 |  6.5 |\n",
       "| at | yolk nuclei |  5 |  8.5 |\n",
       "| bc | yolk nuclei |  5 | 18.0 |\n",
       "| bu | yolk nuclei |  5 |  3.0 |\n",
       "| by | yolk nuclei |  5 | 11.0 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| as | yolk nuclei | 3 |  6.5 |\n",
       "| ay | yolk nuclei | 3 |  3.5 |\n",
       "| ba | yolk nuclei | 3 |  7.0 |\n",
       "| bf | yolk nuclei | 3 | 10.0 |\n",
       "| l  | yolk nuclei | 3 | 14.5 |\n",
       "| o  | yolk nuclei | 3 | 11.0 |\n",
       "| y  | yolk nuclei | 3 |  4.0 |\n",
       "| a  | yolk nuclei | 2 |  7.0 |\n",
       "| ac | yolk nuclei | 2 |  9.0 |\n",
       "| ad | yolk nuclei | 2 |  3.0 |\n",
       "| ag | yolk nuclei | 2 | 12.0 |\n",
       "| al | yolk nuclei | 2 |  6.5 |\n",
       "| au | yolk nuclei | 2 |  4.0 |\n",
       "| aw | yolk nuclei | 2 |  5.5 |\n",
       "| ax | yolk nuclei | 2 |  5.5 |\n",
       "| b  | yolk nuclei | 2 |  9.0 |\n",
       "| bb | yolk nuclei | 2 |  8.0 |\n",
       "| bd | yolk nuclei | 2 |  2.0 |\n",
       "| bz | yolk nuclei | 2 |  5.0 |\n",
       "| f  | yolk nuclei | 2 | 12.0 |\n",
       "| af | yolk nuclei | 1 |  3.5 |\n",
       "| ao | yolk nuclei | 1 |  3.5 |\n",
       "| bi | yolk nuclei | 1 |  2.5 |\n",
       "| cf | yolk nuclei | 1 |  2.5 |\n",
       "| g  | yolk nuclei | 1 |  6.0 |\n",
       "| j  | yolk nuclei | 1 |  2.5 |\n",
       "| m  | yolk nuclei | 1 |  2.5 |\n",
       "| n  | yolk nuclei | 1 |  6.0 |\n",
       "| u  | yolk nuclei | 1 |  2.5 |\n",
       "| x  | yolk nuclei | 1 |  2.0 |\n",
       "\n"
      ],
      "text/plain": [
       "   cell.type.2 annot       n  rank\n",
       "1  ca          yolk nuclei 36 22.0\n",
       "2  bl          yolk nuclei 33 26.0\n",
       "3  bt          yolk nuclei 33  6.0\n",
       "4  s           yolk nuclei 32 24.0\n",
       "5  bq          yolk nuclei 31  3.0\n",
       "6  i           yolk nuclei 26 13.0\n",
       "7  bk          yolk nuclei 21 21.5\n",
       "8  bs          yolk nuclei 21  5.0\n",
       "9  bx          yolk nuclei 21  6.0\n",
       "10 br          yolk nuclei 20  5.0\n",
       "11 cb          yolk nuclei 18 21.5\n",
       "12 bj          yolk nuclei 16 26.5\n",
       "13 cc          yolk nuclei 16 15.5\n",
       "14 bp          yolk nuclei 14 11.5\n",
       "15 bv          yolk nuclei 14  5.0\n",
       "16 ce          yolk nuclei 13 18.5\n",
       "17 be          yolk nuclei 10 17.5\n",
       "18 bw          yolk nuclei  9  4.5\n",
       "19 r           yolk nuclei  8 13.0\n",
       "20 bm          yolk nuclei  7 13.5\n",
       "21 aq          yolk nuclei  6  9.5\n",
       "22 az          yolk nuclei  6 15.0\n",
       "23 bo          yolk nuclei  6 15.5\n",
       "24 aa          yolk nuclei  5  7.5\n",
       "25 ab          yolk nuclei  5 12.0\n",
       "26 an          yolk nuclei  5  6.5\n",
       "27 at          yolk nuclei  5  8.5\n",
       "28 bc          yolk nuclei  5 18.0\n",
       "29 bu          yolk nuclei  5  3.0\n",
       "30 by          yolk nuclei  5 11.0\n",
       "⋮  ⋮           ⋮           ⋮  ⋮   \n",
       "45 as          yolk nuclei 3   6.5\n",
       "46 ay          yolk nuclei 3   3.5\n",
       "47 ba          yolk nuclei 3   7.0\n",
       "48 bf          yolk nuclei 3  10.0\n",
       "49 l           yolk nuclei 3  14.5\n",
       "50 o           yolk nuclei 3  11.0\n",
       "51 y           yolk nuclei 3   4.0\n",
       "52 a           yolk nuclei 2   7.0\n",
       "53 ac          yolk nuclei 2   9.0\n",
       "54 ad          yolk nuclei 2   3.0\n",
       "55 ag          yolk nuclei 2  12.0\n",
       "56 al          yolk nuclei 2   6.5\n",
       "57 au          yolk nuclei 2   4.0\n",
       "58 aw          yolk nuclei 2   5.5\n",
       "59 ax          yolk nuclei 2   5.5\n",
       "60 b           yolk nuclei 2   9.0\n",
       "61 bb          yolk nuclei 2   8.0\n",
       "62 bd          yolk nuclei 2   2.0\n",
       "63 bz          yolk nuclei 2   5.0\n",
       "64 f           yolk nuclei 2  12.0\n",
       "65 af          yolk nuclei 1   3.5\n",
       "66 ao          yolk nuclei 1   3.5\n",
       "67 bi          yolk nuclei 1   2.5\n",
       "68 cf          yolk nuclei 1   2.5\n",
       "69 g           yolk nuclei 1   6.0\n",
       "70 j           yolk nuclei 1   2.5\n",
       "71 m           yolk nuclei 1   2.5\n",
       "72 n           yolk nuclei 1   6.0\n",
       "73 u           yolk nuclei 1   2.5\n",
       "74 x           yolk nuclei 1   2.0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub_no_doubletons  %>% count(cell.type.2, annot)  %>% \n",
    "       arrange(cell.type.2, -n)  %>% group_by(cell.type.2) %>% \n",
    "    mutate(rank = rank(n))  %>%  filter(annot == annot_type)  %>% arrange(-n)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 453,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 24 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cell.type.2</th><th scope=col>annot</th><th scope=col>n</th><th scope=col>rank</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>x</td><td>midgut                      </td><td>381</td><td>24.0</td></tr>\n",
       "\t<tr><td>x</td><td>epidermis                   </td><td> 38</td><td>23.0</td></tr>\n",
       "\t<tr><td>x</td><td>somatic muscle              </td><td> 17</td><td>22.0</td></tr>\n",
       "\t<tr><td>x</td><td>proventriculus              </td><td> 14</td><td>21.0</td></tr>\n",
       "\t<tr><td>x</td><td>brain                       </td><td>  7</td><td>20.0</td></tr>\n",
       "\t<tr><td>x</td><td>fat body                    </td><td>  6</td><td>19.0</td></tr>\n",
       "\t<tr><td>x</td><td>tracheal system             </td><td>  5</td><td>17.5</td></tr>\n",
       "\t<tr><td>x</td><td>ventral nerve cord          </td><td>  5</td><td>17.5</td></tr>\n",
       "\t<tr><td>x</td><td>foregut / hindgut / anal pad</td><td>  4</td><td>15.0</td></tr>\n",
       "\t<tr><td>x</td><td>hindgut / anal pad          </td><td>  4</td><td>15.0</td></tr>\n",
       "\t<tr><td>x</td><td>plasmatocytes               </td><td>  4</td><td>15.0</td></tr>\n",
       "\t<tr><td>x</td><td>salivary gland              </td><td>  3</td><td>11.5</td></tr>\n",
       "\t<tr><td>x</td><td>sensory nervous system      </td><td>  3</td><td>11.5</td></tr>\n",
       "\t<tr><td>x</td><td>spiracle (tracheal system)  </td><td>  3</td><td>11.5</td></tr>\n",
       "\t<tr><td>x</td><td>visceral muscle             </td><td>  3</td><td>11.5</td></tr>\n",
       "\t<tr><td>x</td><td>amnioserosa                 </td><td>  2</td><td> 6.5</td></tr>\n",
       "\t<tr><td>x</td><td>germ cell                   </td><td>  2</td><td> 6.5</td></tr>\n",
       "\t<tr><td>x</td><td>midgut primordium           </td><td>  2</td><td> 6.5</td></tr>\n",
       "\t<tr><td>x</td><td>muscle                      </td><td>  2</td><td> 6.5</td></tr>\n",
       "\t<tr><td>x</td><td>spiracles (tracheal system) </td><td>  2</td><td> 6.5</td></tr>\n",
       "\t<tr><td>x</td><td>ubiquitous                  </td><td>  2</td><td> 6.5</td></tr>\n",
       "\t<tr><td>x</td><td>crystal cell                </td><td>  1</td><td> 2.0</td></tr>\n",
       "\t<tr><td>x</td><td>glia                        </td><td>  1</td><td> 2.0</td></tr>\n",
       "\t<tr><td>x</td><td>yolk nuclei                 </td><td>  1</td><td> 2.0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 24 × 4\n",
       "\\begin{tabular}{llll}\n",
       " cell.type.2 & annot & n & rank\\\\\n",
       " <chr> & <chr> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t x & midgut                       & 381 & 24.0\\\\\n",
       "\t x & epidermis                    &  38 & 23.0\\\\\n",
       "\t x & somatic muscle               &  17 & 22.0\\\\\n",
       "\t x & proventriculus               &  14 & 21.0\\\\\n",
       "\t x & brain                        &   7 & 20.0\\\\\n",
       "\t x & fat body                     &   6 & 19.0\\\\\n",
       "\t x & tracheal system              &   5 & 17.5\\\\\n",
       "\t x & ventral nerve cord           &   5 & 17.5\\\\\n",
       "\t x & foregut / hindgut / anal pad &   4 & 15.0\\\\\n",
       "\t x & hindgut / anal pad           &   4 & 15.0\\\\\n",
       "\t x & plasmatocytes                &   4 & 15.0\\\\\n",
       "\t x & salivary gland               &   3 & 11.5\\\\\n",
       "\t x & sensory nervous system       &   3 & 11.5\\\\\n",
       "\t x & spiracle (tracheal system)   &   3 & 11.5\\\\\n",
       "\t x & visceral muscle              &   3 & 11.5\\\\\n",
       "\t x & amnioserosa                  &   2 &  6.5\\\\\n",
       "\t x & germ cell                    &   2 &  6.5\\\\\n",
       "\t x & midgut primordium            &   2 &  6.5\\\\\n",
       "\t x & muscle                       &   2 &  6.5\\\\\n",
       "\t x & spiracles (tracheal system)  &   2 &  6.5\\\\\n",
       "\t x & ubiquitous                   &   2 &  6.5\\\\\n",
       "\t x & crystal cell                 &   1 &  2.0\\\\\n",
       "\t x & glia                         &   1 &  2.0\\\\\n",
       "\t x & yolk nuclei                  &   1 &  2.0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 24 × 4\n",
       "\n",
       "| cell.type.2 &lt;chr&gt; | annot &lt;chr&gt; | n &lt;int&gt; | rank &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| x | midgut                       | 381 | 24.0 |\n",
       "| x | epidermis                    |  38 | 23.0 |\n",
       "| x | somatic muscle               |  17 | 22.0 |\n",
       "| x | proventriculus               |  14 | 21.0 |\n",
       "| x | brain                        |   7 | 20.0 |\n",
       "| x | fat body                     |   6 | 19.0 |\n",
       "| x | tracheal system              |   5 | 17.5 |\n",
       "| x | ventral nerve cord           |   5 | 17.5 |\n",
       "| x | foregut / hindgut / anal pad |   4 | 15.0 |\n",
       "| x | hindgut / anal pad           |   4 | 15.0 |\n",
       "| x | plasmatocytes                |   4 | 15.0 |\n",
       "| x | salivary gland               |   3 | 11.5 |\n",
       "| x | sensory nervous system       |   3 | 11.5 |\n",
       "| x | spiracle (tracheal system)   |   3 | 11.5 |\n",
       "| x | visceral muscle              |   3 | 11.5 |\n",
       "| x | amnioserosa                  |   2 |  6.5 |\n",
       "| x | germ cell                    |   2 |  6.5 |\n",
       "| x | midgut primordium            |   2 |  6.5 |\n",
       "| x | muscle                       |   2 |  6.5 |\n",
       "| x | spiracles (tracheal system)  |   2 |  6.5 |\n",
       "| x | ubiquitous                   |   2 |  6.5 |\n",
       "| x | crystal cell                 |   1 |  2.0 |\n",
       "| x | glia                         |   1 |  2.0 |\n",
       "| x | yolk nuclei                  |   1 |  2.0 |\n",
       "\n"
      ],
      "text/plain": [
       "   cell.type.2 annot                        n   rank\n",
       "1  x           midgut                       381 24.0\n",
       "2  x           epidermis                     38 23.0\n",
       "3  x           somatic muscle                17 22.0\n",
       "4  x           proventriculus                14 21.0\n",
       "5  x           brain                          7 20.0\n",
       "6  x           fat body                       6 19.0\n",
       "7  x           tracheal system                5 17.5\n",
       "8  x           ventral nerve cord             5 17.5\n",
       "9  x           foregut / hindgut / anal pad   4 15.0\n",
       "10 x           hindgut / anal pad             4 15.0\n",
       "11 x           plasmatocytes                  4 15.0\n",
       "12 x           salivary gland                 3 11.5\n",
       "13 x           sensory nervous system         3 11.5\n",
       "14 x           spiracle (tracheal system)     3 11.5\n",
       "15 x           visceral muscle                3 11.5\n",
       "16 x           amnioserosa                    2  6.5\n",
       "17 x           germ cell                      2  6.5\n",
       "18 x           midgut primordium              2  6.5\n",
       "19 x           muscle                         2  6.5\n",
       "20 x           spiracles (tracheal system)    2  6.5\n",
       "21 x           ubiquitous                     2  6.5\n",
       "22 x           crystal cell                   1  2.0\n",
       "23 x           glia                           1  2.0\n",
       "24 x           yolk nuclei                    1  2.0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub_no_doubletons  %>% count(cell.type.2, annot)  %>% \n",
    "       arrange(cell.type.2, -n)  %>% group_by(cell.type.2) %>% \n",
    "    mutate(rank = rank(n))  %>%  filter(cell.type.2 == \"x\") %>% head(120)  %>% tail(60)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 470,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'cell.type.2.markers'</li><li>'cell.type.2.markers_no_doubletons'</li><li>'cell.type.markers.sub300_no_singletons.cell.type'</li><li>'cell.type.markers.sub300.cell.type'</li><li>'cluster.markers'</li><li>'cluster.markers.sub300'</li><li>'cluster.markers.sub300_no_doubletons'</li><li>'cluster.markers.sub300_no_doubletons_all'</li><li>'cluster.markers.sub300_no_singletons'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'cell.type.2.markers'\n",
       "\\item 'cell.type.2.markers\\_no\\_doubletons'\n",
       "\\item 'cell.type.markers.sub300\\_no\\_singletons.cell.type'\n",
       "\\item 'cell.type.markers.sub300.cell.type'\n",
       "\\item 'cluster.markers'\n",
       "\\item 'cluster.markers.sub300'\n",
       "\\item 'cluster.markers.sub300\\_no\\_doubletons'\n",
       "\\item 'cluster.markers.sub300\\_no\\_doubletons\\_all'\n",
       "\\item 'cluster.markers.sub300\\_no\\_singletons'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'cell.type.2.markers'\n",
       "2. 'cell.type.2.markers_no_doubletons'\n",
       "3. 'cell.type.markers.sub300_no_singletons.cell.type'\n",
       "4. 'cell.type.markers.sub300.cell.type'\n",
       "5. 'cluster.markers'\n",
       "6. 'cluster.markers.sub300'\n",
       "7. 'cluster.markers.sub300_no_doubletons'\n",
       "8. 'cluster.markers.sub300_no_doubletons_all'\n",
       "9. 'cluster.markers.sub300_no_singletons'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"cell.type.2.markers\"                             \n",
       "[2] \"cell.type.2.markers_no_doubletons\"               \n",
       "[3] \"cell.type.markers.sub300_no_singletons.cell.type\"\n",
       "[4] \"cell.type.markers.sub300.cell.type\"              \n",
       "[5] \"cluster.markers\"                                 \n",
       "[6] \"cluster.markers.sub300\"                          \n",
       "[7] \"cluster.markers.sub300_no_doubletons\"            \n",
       "[8] \"cluster.markers.sub300_no_doubletons_all\"        \n",
       "[9] \"cluster.markers.sub300_no_singletons\"            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ls(pattern = \"marker\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## get markers for Sue."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 473,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 60 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>a </td><td>Muc26B     </td><td> 3651</td><td>  121</td><td>11236.035477</td><td>   3.9689778</td><td>11.142896</td><td>149.93852</td></tr>\n",
       "\t<tr><td>a </td><td>CG7017     </td><td> 3415</td><td>  236</td><td>12317.771831</td><td>   9.9390129</td><td>10.137043</td><td>137.74793</td></tr>\n",
       "\t<tr><td>a </td><td>Muc96D     </td><td> 2335</td><td>  122</td><td> 6924.704567</td><td>   5.8653887</td><td> 9.978195</td><td>127.29927</td></tr>\n",
       "\t<tr><td>a </td><td>CG17826    </td><td>  970</td><td>   33</td><td> 2991.398860</td><td>   1.5259765</td><td>10.209763</td><td>117.89302</td></tr>\n",
       "\t<tr><td>a </td><td>CG11300    </td><td>  477</td><td>   17</td><td> 1102.911259</td><td>   0.6713437</td><td> 9.366093</td><td> 94.67629</td></tr>\n",
       "\t<tr><td>a </td><td>EAChm      </td><td>  257</td><td>  117</td><td> 1599.503427</td><td>   3.7442647</td><td> 8.397224</td><td> 89.38265</td></tr>\n",
       "\t<tr><td>a </td><td>CG18649    </td><td>   97</td><td>    6</td><td>  530.483940</td><td>   0.2496856</td><td> 8.729600</td><td> 79.03677</td></tr>\n",
       "\t<tr><td>a </td><td>CG32198    </td><td>  211</td><td>   11</td><td>  540.493251</td><td>   0.3955226</td><td> 8.597327</td><td> 78.07061</td></tr>\n",
       "\t<tr><td>a </td><td>CG32453    </td><td>   83</td><td>   16</td><td>  549.659426</td><td>   0.4831467</td><td> 8.533733</td><td> 77.69978</td></tr>\n",
       "\t<tr><td>a </td><td>CG11672    </td><td>  151</td><td>   34</td><td>  694.884816</td><td>   1.4618825</td><td> 8.140868</td><td> 76.87181</td></tr>\n",
       "\t<tr><td>aa</td><td>TwdlX      </td><td> 1401</td><td>  415</td><td> 1986.601172</td><td>  13.5208150</td><td> 7.096036</td><td> 77.74994</td></tr>\n",
       "\t<tr><td>aa</td><td>CG12920    </td><td> 1056</td><td>  395</td><td> 1942.942484</td><td>  17.0922640</td><td> 6.746726</td><td> 73.70643</td></tr>\n",
       "\t<tr><td>aa</td><td>Cyp49a1    </td><td> 1544</td><td>  819</td><td> 2722.951408</td><td>  34.0529600</td><td> 6.279491</td><td> 71.65832</td></tr>\n",
       "\t<tr><td>aa</td><td>CG6409     </td><td> 1251</td><td>  860</td><td> 1986.822597</td><td>  31.4038439</td><td> 5.938154</td><td> 65.06420</td></tr>\n",
       "\t<tr><td>aa</td><td>CG14949    </td><td>  232</td><td>   44</td><td>  341.610053</td><td>   1.1380942</td><td> 7.319881</td><td> 61.63650</td></tr>\n",
       "\t<tr><td>aa</td><td>CG42296    </td><td>  800</td><td>  915</td><td> 1422.554357</td><td>  37.6468204</td><td> 5.201990</td><td> 54.49231</td></tr>\n",
       "\t<tr><td>aa</td><td>CG32564    </td><td>  795</td><td> 1494</td><td> 1193.287003</td><td>  43.7231524</td><td> 4.737775</td><td> 48.42923</td></tr>\n",
       "\t<tr><td>aa</td><td>CG6125     </td><td>  202</td><td>  145</td><td>  304.848222</td><td>   4.2440423</td><td> 5.861268</td><td> 48.39457</td></tr>\n",
       "\t<tr><td>aa</td><td>NKCC       </td><td>  258</td><td>  202</td><td>  389.712875</td><td>   7.4734855</td><td> 5.523312</td><td> 47.55553</td></tr>\n",
       "\t<tr><td>aa</td><td>CG44098    </td><td>  272</td><td>  181</td><td>  371.403132</td><td>   7.3418310</td><td> 5.476478</td><td> 46.77307</td></tr>\n",
       "\t<tr><td>ab</td><td>TwdlC      </td><td> 1284</td><td>  866</td><td> 3105.376180</td><td>  28.9784505</td><td> 6.694698</td><td> 77.66531</td></tr>\n",
       "\t<tr><td>ab</td><td>CG12523    </td><td>  826</td><td>  414</td><td> 1943.215939</td><td>  13.5321645</td><td> 7.063053</td><td> 77.16366</td></tr>\n",
       "\t<tr><td>ab</td><td>TwdlE      </td><td> 1509</td><td> 1199</td><td> 3559.543421</td><td>  42.2346251</td><td> 6.363361</td><td> 75.07418</td></tr>\n",
       "\t<tr><td>ab</td><td>CG2150     </td><td> 3308</td><td> 3453</td><td> 7091.485057</td><td> 124.0199569</td><td> 5.825857</td><td> 74.52481</td></tr>\n",
       "\t<tr><td>ab</td><td>CG13041    </td><td>  296</td><td>  121</td><td>  717.557079</td><td>   3.9442471</td><td> 7.181199</td><td> 68.14210</td></tr>\n",
       "\t<tr><td>ab</td><td>Cpr56F     </td><td>  826</td><td>  806</td><td> 1955.138928</td><td>  25.9217876</td><td> 6.182353</td><td> 67.59657</td></tr>\n",
       "\t<tr><td>ab</td><td>CG13059    </td><td>  566</td><td>  463</td><td> 1466.470210</td><td>  16.1343253</td><td> 6.419315</td><td> 67.52551</td></tr>\n",
       "\t<tr><td>ab</td><td>Cpr62Bc    </td><td>  774</td><td>  805</td><td> 1912.485003</td><td>  27.1771622</td><td> 6.084778</td><td> 66.33617</td></tr>\n",
       "\t<tr><td>ab</td><td>Cpr50Cb    </td><td>  258</td><td>   92</td><td>  649.595090</td><td>   3.8905488</td><td> 7.053401</td><td> 65.91837</td></tr>\n",
       "\t<tr><td>ab</td><td>CG34461    </td><td>  868</td><td> 1139</td><td> 2090.446577</td><td>  36.1193300</td><td> 5.815497</td><td> 64.14659</td></tr>\n",
       "\t<tr><td>ac</td><td>pip        </td><td>47027</td><td> 6925</td><td>32792.085193</td><td> 264.9900195</td><td> 6.945832</td><td>104.19514</td></tr>\n",
       "\t<tr><td>ac</td><td>nur        </td><td>26613</td><td> 9706</td><td>19927.806794</td><td> 355.4399598</td><td> 5.804980</td><td> 82.91002</td></tr>\n",
       "\t<tr><td>ac</td><td>CG14756    </td><td>14481</td><td> 5763</td><td>10818.113243</td><td> 201.0548668</td><td> 5.742558</td><td> 76.95771</td></tr>\n",
       "\t<tr><td>ac</td><td>CG13159    </td><td> 6748</td><td> 2596</td><td> 4972.820166</td><td> 104.2692095</td><td> 5.561909</td><td> 68.30101</td></tr>\n",
       "\t<tr><td>ac</td><td>CG34190    </td><td>  914</td><td>  134</td><td>  618.163155</td><td>   4.2905393</td><td> 6.868429</td><td> 63.69902</td></tr>\n",
       "\t<tr><td>ac</td><td>PH4alphaSG1</td><td> 1719</td><td>  504</td><td> 1247.203350</td><td>  17.3788729</td><td> 6.084505</td><td> 62.58301</td></tr>\n",
       "\t<tr><td>ac</td><td>PH4alphaSG2</td><td> 1097</td><td>  293</td><td>  789.621457</td><td>   9.3266336</td><td> 6.256719</td><td> 60.23246</td></tr>\n",
       "\t<tr><td>ac</td><td>CrzR       </td><td>  952</td><td>  222</td><td>  676.513021</td><td>   9.2903769</td><td> 6.038750</td><td> 56.78904</td></tr>\n",
       "\t<tr><td>ac</td><td>GlcAT-P    </td><td> 5727</td><td> 4643</td><td> 4109.006145</td><td> 205.6638306</td><td> 4.313432</td><td> 51.78242</td></tr>\n",
       "\t<tr><td>ac</td><td>sens       </td><td> 1306</td><td>  822</td><td>  944.173495</td><td>  32.5478429</td><td> 4.814760</td><td> 47.59118</td></tr>\n",
       "\t<tr><td>ad</td><td>CG17782    </td><td>  340</td><td>   90</td><td> 1323.369396</td><td>   3.3047155</td><td> 8.264082</td><td> 85.70754</td></tr>\n",
       "\t<tr><td>ad</td><td>CG5768     </td><td>  211</td><td>   33</td><td>  809.593243</td><td>   1.2391984</td><td> 8.498071</td><td> 82.11545</td></tr>\n",
       "\t<tr><td>ad</td><td>CG13615    </td><td>  191</td><td>   27</td><td>  737.411449</td><td>   1.0560434</td><td> 8.486455</td><td> 80.86133</td></tr>\n",
       "\t<tr><td>ad</td><td>CG33003    </td><td>  632</td><td>  745</td><td> 2356.796314</td><td>  34.0934208</td><td> 6.069483</td><td> 67.99777</td></tr>\n",
       "\t<tr><td>ad</td><td>CG17784    </td><td>  101</td><td>   14</td><td>  373.182140</td><td>   0.5367777</td><td> 7.923828</td><td> 67.72968</td></tr>\n",
       "\t<tr><td>ad</td><td>CG34444    </td><td>  113</td><td>   29</td><td>  421.850247</td><td>   1.0059735</td><td> 7.716285</td><td> 67.31689</td></tr>\n",
       "\t<tr><td>ad</td><td>CG6785     </td><td>  361</td><td>  443</td><td> 1392.631600</td><td>  17.8752324</td><td> 6.205175</td><td> 64.81078</td></tr>\n",
       "\t<tr><td>ad</td><td>CG34442    </td><td>  110</td><td>   37</td><td>  428.023985</td><td>   1.6051126</td><td> 7.360202</td><td> 64.36434</td></tr>\n",
       "\t<tr><td>ad</td><td>CG17780    </td><td>  305</td><td>  342</td><td> 1159.157611</td><td>  15.0652428</td><td> 6.172990</td><td> 62.84169</td></tr>\n",
       "\t<tr><td>ad</td><td>CG13613    </td><td>   57</td><td>    4</td><td>  236.376236</td><td>   0.1169610</td><td> 7.725362</td><td> 60.96108</td></tr>\n",
       "\t<tr><td>ae</td><td>Osi20      </td><td>    0</td><td> 7617</td><td>    0.000000</td><td> 307.8304388</td><td> 8.265992</td><td> 68.36530</td></tr>\n",
       "\t<tr><td>ae</td><td>Cubn       </td><td>  634</td><td>  843</td><td> 1629.779138</td><td>  35.5365007</td><td> 5.479194</td><td> 58.47038</td></tr>\n",
       "\t<tr><td>ae</td><td>CG42255    </td><td>  266</td><td>  278</td><td>  684.742771</td><td>  11.2769586</td><td> 5.801537</td><td> 54.65932</td></tr>\n",
       "\t<tr><td>ae</td><td>Osi19      </td><td>    0</td><td> 3898</td><td>    0.000000</td><td> 157.9350323</td><td> 7.303187</td><td> 53.40305</td></tr>\n",
       "\t<tr><td>ae</td><td>TwdlM      </td><td>    0</td><td> 3772</td><td>    0.000000</td><td> 129.3287412</td><td> 7.014899</td><td> 49.28676</td></tr>\n",
       "\t<tr><td>ae</td><td>Muc91C     </td><td>    0</td><td> 3164</td><td>    0.000000</td><td> 121.2018765</td><td> 6.921268</td><td> 47.98600</td></tr>\n",
       "\t<tr><td>ae</td><td>Ipod       </td><td>    4</td><td>10953</td><td>    9.611553</td><td> 394.8978670</td><td> 5.217772</td><td> 45.02407</td></tr>\n",
       "\t<tr><td>ae</td><td>CG2962     </td><td>   17</td><td>31194</td><td>   52.087003</td><td>1113.6244100</td><td> 4.390760</td><td> 44.44478</td></tr>\n",
       "\t<tr><td>ae</td><td>Amnionless </td><td>  156</td><td>  288</td><td>  391.584189</td><td>  10.8204319</td><td> 5.049968</td><td> 43.51486</td></tr>\n",
       "\t<tr><td>ae</td><td>CG12723    </td><td>    0</td><td> 2379</td><td>    0.000000</td><td>  95.5038780</td><td> 6.577487</td><td> 43.36218</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 60 × 8\n",
       "\\begin{tabular}{llllllll}\n",
       " cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       " <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t a  & Muc26B      &  3651 &   121 & 11236.035477 &    3.9689778 & 11.142896 & 149.93852\\\\\n",
       "\t a  & CG7017      &  3415 &   236 & 12317.771831 &    9.9390129 & 10.137043 & 137.74793\\\\\n",
       "\t a  & Muc96D      &  2335 &   122 &  6924.704567 &    5.8653887 &  9.978195 & 127.29927\\\\\n",
       "\t a  & CG17826     &   970 &    33 &  2991.398860 &    1.5259765 & 10.209763 & 117.89302\\\\\n",
       "\t a  & CG11300     &   477 &    17 &  1102.911259 &    0.6713437 &  9.366093 &  94.67629\\\\\n",
       "\t a  & EAChm       &   257 &   117 &  1599.503427 &    3.7442647 &  8.397224 &  89.38265\\\\\n",
       "\t a  & CG18649     &    97 &     6 &   530.483940 &    0.2496856 &  8.729600 &  79.03677\\\\\n",
       "\t a  & CG32198     &   211 &    11 &   540.493251 &    0.3955226 &  8.597327 &  78.07061\\\\\n",
       "\t a  & CG32453     &    83 &    16 &   549.659426 &    0.4831467 &  8.533733 &  77.69978\\\\\n",
       "\t a  & CG11672     &   151 &    34 &   694.884816 &    1.4618825 &  8.140868 &  76.87181\\\\\n",
       "\t aa & TwdlX       &  1401 &   415 &  1986.601172 &   13.5208150 &  7.096036 &  77.74994\\\\\n",
       "\t aa & CG12920     &  1056 &   395 &  1942.942484 &   17.0922640 &  6.746726 &  73.70643\\\\\n",
       "\t aa & Cyp49a1     &  1544 &   819 &  2722.951408 &   34.0529600 &  6.279491 &  71.65832\\\\\n",
       "\t aa & CG6409      &  1251 &   860 &  1986.822597 &   31.4038439 &  5.938154 &  65.06420\\\\\n",
       "\t aa & CG14949     &   232 &    44 &   341.610053 &    1.1380942 &  7.319881 &  61.63650\\\\\n",
       "\t aa & CG42296     &   800 &   915 &  1422.554357 &   37.6468204 &  5.201990 &  54.49231\\\\\n",
       "\t aa & CG32564     &   795 &  1494 &  1193.287003 &   43.7231524 &  4.737775 &  48.42923\\\\\n",
       "\t aa & CG6125      &   202 &   145 &   304.848222 &    4.2440423 &  5.861268 &  48.39457\\\\\n",
       "\t aa & NKCC        &   258 &   202 &   389.712875 &    7.4734855 &  5.523312 &  47.55553\\\\\n",
       "\t aa & CG44098     &   272 &   181 &   371.403132 &    7.3418310 &  5.476478 &  46.77307\\\\\n",
       "\t ab & TwdlC       &  1284 &   866 &  3105.376180 &   28.9784505 &  6.694698 &  77.66531\\\\\n",
       "\t ab & CG12523     &   826 &   414 &  1943.215939 &   13.5321645 &  7.063053 &  77.16366\\\\\n",
       "\t ab & TwdlE       &  1509 &  1199 &  3559.543421 &   42.2346251 &  6.363361 &  75.07418\\\\\n",
       "\t ab & CG2150      &  3308 &  3453 &  7091.485057 &  124.0199569 &  5.825857 &  74.52481\\\\\n",
       "\t ab & CG13041     &   296 &   121 &   717.557079 &    3.9442471 &  7.181199 &  68.14210\\\\\n",
       "\t ab & Cpr56F      &   826 &   806 &  1955.138928 &   25.9217876 &  6.182353 &  67.59657\\\\\n",
       "\t ab & CG13059     &   566 &   463 &  1466.470210 &   16.1343253 &  6.419315 &  67.52551\\\\\n",
       "\t ab & Cpr62Bc     &   774 &   805 &  1912.485003 &   27.1771622 &  6.084778 &  66.33617\\\\\n",
       "\t ab & Cpr50Cb     &   258 &    92 &   649.595090 &    3.8905488 &  7.053401 &  65.91837\\\\\n",
       "\t ab & CG34461     &   868 &  1139 &  2090.446577 &   36.1193300 &  5.815497 &  64.14659\\\\\n",
       "\t ac & pip         & 47027 &  6925 & 32792.085193 &  264.9900195 &  6.945832 & 104.19514\\\\\n",
       "\t ac & nur         & 26613 &  9706 & 19927.806794 &  355.4399598 &  5.804980 &  82.91002\\\\\n",
       "\t ac & CG14756     & 14481 &  5763 & 10818.113243 &  201.0548668 &  5.742558 &  76.95771\\\\\n",
       "\t ac & CG13159     &  6748 &  2596 &  4972.820166 &  104.2692095 &  5.561909 &  68.30101\\\\\n",
       "\t ac & CG34190     &   914 &   134 &   618.163155 &    4.2905393 &  6.868429 &  63.69902\\\\\n",
       "\t ac & PH4alphaSG1 &  1719 &   504 &  1247.203350 &   17.3788729 &  6.084505 &  62.58301\\\\\n",
       "\t ac & PH4alphaSG2 &  1097 &   293 &   789.621457 &    9.3266336 &  6.256719 &  60.23246\\\\\n",
       "\t ac & CrzR        &   952 &   222 &   676.513021 &    9.2903769 &  6.038750 &  56.78904\\\\\n",
       "\t ac & GlcAT-P     &  5727 &  4643 &  4109.006145 &  205.6638306 &  4.313432 &  51.78242\\\\\n",
       "\t ac & sens        &  1306 &   822 &   944.173495 &   32.5478429 &  4.814760 &  47.59118\\\\\n",
       "\t ad & CG17782     &   340 &    90 &  1323.369396 &    3.3047155 &  8.264082 &  85.70754\\\\\n",
       "\t ad & CG5768      &   211 &    33 &   809.593243 &    1.2391984 &  8.498071 &  82.11545\\\\\n",
       "\t ad & CG13615     &   191 &    27 &   737.411449 &    1.0560434 &  8.486455 &  80.86133\\\\\n",
       "\t ad & CG33003     &   632 &   745 &  2356.796314 &   34.0934208 &  6.069483 &  67.99777\\\\\n",
       "\t ad & CG17784     &   101 &    14 &   373.182140 &    0.5367777 &  7.923828 &  67.72968\\\\\n",
       "\t ad & CG34444     &   113 &    29 &   421.850247 &    1.0059735 &  7.716285 &  67.31689\\\\\n",
       "\t ad & CG6785      &   361 &   443 &  1392.631600 &   17.8752324 &  6.205175 &  64.81078\\\\\n",
       "\t ad & CG34442     &   110 &    37 &   428.023985 &    1.6051126 &  7.360202 &  64.36434\\\\\n",
       "\t ad & CG17780     &   305 &   342 &  1159.157611 &   15.0652428 &  6.172990 &  62.84169\\\\\n",
       "\t ad & CG13613     &    57 &     4 &   236.376236 &    0.1169610 &  7.725362 &  60.96108\\\\\n",
       "\t ae & Osi20       &     0 &  7617 &     0.000000 &  307.8304388 &  8.265992 &  68.36530\\\\\n",
       "\t ae & Cubn        &   634 &   843 &  1629.779138 &   35.5365007 &  5.479194 &  58.47038\\\\\n",
       "\t ae & CG42255     &   266 &   278 &   684.742771 &   11.2769586 &  5.801537 &  54.65932\\\\\n",
       "\t ae & Osi19       &     0 &  3898 &     0.000000 &  157.9350323 &  7.303187 &  53.40305\\\\\n",
       "\t ae & TwdlM       &     0 &  3772 &     0.000000 &  129.3287412 &  7.014899 &  49.28676\\\\\n",
       "\t ae & Muc91C      &     0 &  3164 &     0.000000 &  121.2018765 &  6.921268 &  47.98600\\\\\n",
       "\t ae & Ipod        &     4 & 10953 &     9.611553 &  394.8978670 &  5.217772 &  45.02407\\\\\n",
       "\t ae & CG2962      &    17 & 31194 &    52.087003 & 1113.6244100 &  4.390760 &  44.44478\\\\\n",
       "\t ae & Amnionless  &   156 &   288 &   391.584189 &   10.8204319 &  5.049968 &  43.51486\\\\\n",
       "\t ae & CG12723     &     0 &  2379 &     0.000000 &   95.5038780 &  6.577487 &  43.36218\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 60 × 8\n",
       "\n",
       "| cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| a  | Muc26B      |  3651 |   121 | 11236.035477 |    3.9689778 | 11.142896 | 149.93852 |\n",
       "| a  | CG7017      |  3415 |   236 | 12317.771831 |    9.9390129 | 10.137043 | 137.74793 |\n",
       "| a  | Muc96D      |  2335 |   122 |  6924.704567 |    5.8653887 |  9.978195 | 127.29927 |\n",
       "| a  | CG17826     |   970 |    33 |  2991.398860 |    1.5259765 | 10.209763 | 117.89302 |\n",
       "| a  | CG11300     |   477 |    17 |  1102.911259 |    0.6713437 |  9.366093 |  94.67629 |\n",
       "| a  | EAChm       |   257 |   117 |  1599.503427 |    3.7442647 |  8.397224 |  89.38265 |\n",
       "| a  | CG18649     |    97 |     6 |   530.483940 |    0.2496856 |  8.729600 |  79.03677 |\n",
       "| a  | CG32198     |   211 |    11 |   540.493251 |    0.3955226 |  8.597327 |  78.07061 |\n",
       "| a  | CG32453     |    83 |    16 |   549.659426 |    0.4831467 |  8.533733 |  77.69978 |\n",
       "| a  | CG11672     |   151 |    34 |   694.884816 |    1.4618825 |  8.140868 |  76.87181 |\n",
       "| aa | TwdlX       |  1401 |   415 |  1986.601172 |   13.5208150 |  7.096036 |  77.74994 |\n",
       "| aa | CG12920     |  1056 |   395 |  1942.942484 |   17.0922640 |  6.746726 |  73.70643 |\n",
       "| aa | Cyp49a1     |  1544 |   819 |  2722.951408 |   34.0529600 |  6.279491 |  71.65832 |\n",
       "| aa | CG6409      |  1251 |   860 |  1986.822597 |   31.4038439 |  5.938154 |  65.06420 |\n",
       "| aa | CG14949     |   232 |    44 |   341.610053 |    1.1380942 |  7.319881 |  61.63650 |\n",
       "| aa | CG42296     |   800 |   915 |  1422.554357 |   37.6468204 |  5.201990 |  54.49231 |\n",
       "| aa | CG32564     |   795 |  1494 |  1193.287003 |   43.7231524 |  4.737775 |  48.42923 |\n",
       "| aa | CG6125      |   202 |   145 |   304.848222 |    4.2440423 |  5.861268 |  48.39457 |\n",
       "| aa | NKCC        |   258 |   202 |   389.712875 |    7.4734855 |  5.523312 |  47.55553 |\n",
       "| aa | CG44098     |   272 |   181 |   371.403132 |    7.3418310 |  5.476478 |  46.77307 |\n",
       "| ab | TwdlC       |  1284 |   866 |  3105.376180 |   28.9784505 |  6.694698 |  77.66531 |\n",
       "| ab | CG12523     |   826 |   414 |  1943.215939 |   13.5321645 |  7.063053 |  77.16366 |\n",
       "| ab | TwdlE       |  1509 |  1199 |  3559.543421 |   42.2346251 |  6.363361 |  75.07418 |\n",
       "| ab | CG2150      |  3308 |  3453 |  7091.485057 |  124.0199569 |  5.825857 |  74.52481 |\n",
       "| ab | CG13041     |   296 |   121 |   717.557079 |    3.9442471 |  7.181199 |  68.14210 |\n",
       "| ab | Cpr56F      |   826 |   806 |  1955.138928 |   25.9217876 |  6.182353 |  67.59657 |\n",
       "| ab | CG13059     |   566 |   463 |  1466.470210 |   16.1343253 |  6.419315 |  67.52551 |\n",
       "| ab | Cpr62Bc     |   774 |   805 |  1912.485003 |   27.1771622 |  6.084778 |  66.33617 |\n",
       "| ab | Cpr50Cb     |   258 |    92 |   649.595090 |    3.8905488 |  7.053401 |  65.91837 |\n",
       "| ab | CG34461     |   868 |  1139 |  2090.446577 |   36.1193300 |  5.815497 |  64.14659 |\n",
       "| ac | pip         | 47027 |  6925 | 32792.085193 |  264.9900195 |  6.945832 | 104.19514 |\n",
       "| ac | nur         | 26613 |  9706 | 19927.806794 |  355.4399598 |  5.804980 |  82.91002 |\n",
       "| ac | CG14756     | 14481 |  5763 | 10818.113243 |  201.0548668 |  5.742558 |  76.95771 |\n",
       "| ac | CG13159     |  6748 |  2596 |  4972.820166 |  104.2692095 |  5.561909 |  68.30101 |\n",
       "| ac | CG34190     |   914 |   134 |   618.163155 |    4.2905393 |  6.868429 |  63.69902 |\n",
       "| ac | PH4alphaSG1 |  1719 |   504 |  1247.203350 |   17.3788729 |  6.084505 |  62.58301 |\n",
       "| ac | PH4alphaSG2 |  1097 |   293 |   789.621457 |    9.3266336 |  6.256719 |  60.23246 |\n",
       "| ac | CrzR        |   952 |   222 |   676.513021 |    9.2903769 |  6.038750 |  56.78904 |\n",
       "| ac | GlcAT-P     |  5727 |  4643 |  4109.006145 |  205.6638306 |  4.313432 |  51.78242 |\n",
       "| ac | sens        |  1306 |   822 |   944.173495 |   32.5478429 |  4.814760 |  47.59118 |\n",
       "| ad | CG17782     |   340 |    90 |  1323.369396 |    3.3047155 |  8.264082 |  85.70754 |\n",
       "| ad | CG5768      |   211 |    33 |   809.593243 |    1.2391984 |  8.498071 |  82.11545 |\n",
       "| ad | CG13615     |   191 |    27 |   737.411449 |    1.0560434 |  8.486455 |  80.86133 |\n",
       "| ad | CG33003     |   632 |   745 |  2356.796314 |   34.0934208 |  6.069483 |  67.99777 |\n",
       "| ad | CG17784     |   101 |    14 |   373.182140 |    0.5367777 |  7.923828 |  67.72968 |\n",
       "| ad | CG34444     |   113 |    29 |   421.850247 |    1.0059735 |  7.716285 |  67.31689 |\n",
       "| ad | CG6785      |   361 |   443 |  1392.631600 |   17.8752324 |  6.205175 |  64.81078 |\n",
       "| ad | CG34442     |   110 |    37 |   428.023985 |    1.6051126 |  7.360202 |  64.36434 |\n",
       "| ad | CG17780     |   305 |   342 |  1159.157611 |   15.0652428 |  6.172990 |  62.84169 |\n",
       "| ad | CG13613     |    57 |     4 |   236.376236 |    0.1169610 |  7.725362 |  60.96108 |\n",
       "| ae | Osi20       |     0 |  7617 |     0.000000 |  307.8304388 |  8.265992 |  68.36530 |\n",
       "| ae | Cubn        |   634 |   843 |  1629.779138 |   35.5365007 |  5.479194 |  58.47038 |\n",
       "| ae | CG42255     |   266 |   278 |   684.742771 |   11.2769586 |  5.801537 |  54.65932 |\n",
       "| ae | Osi19       |     0 |  3898 |     0.000000 |  157.9350323 |  7.303187 |  53.40305 |\n",
       "| ae | TwdlM       |     0 |  3772 |     0.000000 |  129.3287412 |  7.014899 |  49.28676 |\n",
       "| ae | Muc91C      |     0 |  3164 |     0.000000 |  121.2018765 |  6.921268 |  47.98600 |\n",
       "| ae | Ipod        |     4 | 10953 |     9.611553 |  394.8978670 |  5.217772 |  45.02407 |\n",
       "| ae | CG2962      |    17 | 31194 |    52.087003 | 1113.6244100 |  4.390760 |  44.44478 |\n",
       "| ae | Amnionless  |   156 |   288 |   391.584189 |   10.8204319 |  5.049968 |  43.51486 |\n",
       "| ae | CG12723     |     0 |  2379 |     0.000000 |   95.5038780 |  6.577487 |  43.36218 |\n",
       "\n"
      ],
      "text/plain": [
       "   cell.type.2 gene        cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "1  a           Muc26B       3651               121       11236.035477   \n",
       "2  a           CG7017       3415               236       12317.771831   \n",
       "3  a           Muc96D       2335               122        6924.704567   \n",
       "4  a           CG17826       970                33        2991.398860   \n",
       "5  a           CG11300       477                17        1102.911259   \n",
       "6  a           EAChm         257               117        1599.503427   \n",
       "7  a           CG18649        97                 6         530.483940   \n",
       "8  a           CG32198       211                11         540.493251   \n",
       "9  a           CG32453        83                16         549.659426   \n",
       "10 a           CG11672       151                34         694.884816   \n",
       "11 aa          TwdlX        1401               415        1986.601172   \n",
       "12 aa          CG12920      1056               395        1942.942484   \n",
       "13 aa          Cyp49a1      1544               819        2722.951408   \n",
       "14 aa          CG6409       1251               860        1986.822597   \n",
       "15 aa          CG14949       232                44         341.610053   \n",
       "16 aa          CG42296       800               915        1422.554357   \n",
       "17 aa          CG32564       795              1494        1193.287003   \n",
       "18 aa          CG6125        202               145         304.848222   \n",
       "19 aa          NKCC          258               202         389.712875   \n",
       "20 aa          CG44098       272               181         371.403132   \n",
       "21 ab          TwdlC        1284               866        3105.376180   \n",
       "22 ab          CG12523       826               414        1943.215939   \n",
       "23 ab          TwdlE        1509              1199        3559.543421   \n",
       "24 ab          CG2150       3308              3453        7091.485057   \n",
       "25 ab          CG13041       296               121         717.557079   \n",
       "26 ab          Cpr56F        826               806        1955.138928   \n",
       "27 ab          CG13059       566               463        1466.470210   \n",
       "28 ab          Cpr62Bc       774               805        1912.485003   \n",
       "29 ab          Cpr50Cb       258                92         649.595090   \n",
       "30 ab          CG34461       868              1139        2090.446577   \n",
       "31 ac          pip         47027              6925       32792.085193   \n",
       "32 ac          nur         26613              9706       19927.806794   \n",
       "33 ac          CG14756     14481              5763       10818.113243   \n",
       "34 ac          CG13159      6748              2596        4972.820166   \n",
       "35 ac          CG34190       914               134         618.163155   \n",
       "36 ac          PH4alphaSG1  1719               504        1247.203350   \n",
       "37 ac          PH4alphaSG2  1097               293         789.621457   \n",
       "38 ac          CrzR          952               222         676.513021   \n",
       "39 ac          GlcAT-P      5727              4643        4109.006145   \n",
       "40 ac          sens         1306               822         944.173495   \n",
       "41 ad          CG17782       340                90        1323.369396   \n",
       "42 ad          CG5768        211                33         809.593243   \n",
       "43 ad          CG13615       191                27         737.411449   \n",
       "44 ad          CG33003       632               745        2356.796314   \n",
       "45 ad          CG17784       101                14         373.182140   \n",
       "46 ad          CG34444       113                29         421.850247   \n",
       "47 ad          CG6785        361               443        1392.631600   \n",
       "48 ad          CG34442       110                37         428.023985   \n",
       "49 ad          CG17780       305               342        1159.157611   \n",
       "50 ad          CG13613        57                 4         236.376236   \n",
       "51 ae          Osi20           0              7617           0.000000   \n",
       "52 ae          Cubn          634               843        1629.779138   \n",
       "53 ae          CG42255       266               278         684.742771   \n",
       "54 ae          Osi19           0              3898           0.000000   \n",
       "55 ae          TwdlM           0              3772           0.000000   \n",
       "56 ae          Muc91C          0              3164           0.000000   \n",
       "57 ae          Ipod            4             10953           9.611553   \n",
       "58 ae          CG2962         17             31194          52.087003   \n",
       "59 ae          Amnionless    156               288         391.584189   \n",
       "60 ae          CG12723         0              2379           0.000000   \n",
       "   other.tpm    log2.ratio heuristic.score\n",
       "1     3.9689778 11.142896  149.93852      \n",
       "2     9.9390129 10.137043  137.74793      \n",
       "3     5.8653887  9.978195  127.29927      \n",
       "4     1.5259765 10.209763  117.89302      \n",
       "5     0.6713437  9.366093   94.67629      \n",
       "6     3.7442647  8.397224   89.38265      \n",
       "7     0.2496856  8.729600   79.03677      \n",
       "8     0.3955226  8.597327   78.07061      \n",
       "9     0.4831467  8.533733   77.69978      \n",
       "10    1.4618825  8.140868   76.87181      \n",
       "11   13.5208150  7.096036   77.74994      \n",
       "12   17.0922640  6.746726   73.70643      \n",
       "13   34.0529600  6.279491   71.65832      \n",
       "14   31.4038439  5.938154   65.06420      \n",
       "15    1.1380942  7.319881   61.63650      \n",
       "16   37.6468204  5.201990   54.49231      \n",
       "17   43.7231524  4.737775   48.42923      \n",
       "18    4.2440423  5.861268   48.39457      \n",
       "19    7.4734855  5.523312   47.55553      \n",
       "20    7.3418310  5.476478   46.77307      \n",
       "21   28.9784505  6.694698   77.66531      \n",
       "22   13.5321645  7.063053   77.16366      \n",
       "23   42.2346251  6.363361   75.07418      \n",
       "24  124.0199569  5.825857   74.52481      \n",
       "25    3.9442471  7.181199   68.14210      \n",
       "26   25.9217876  6.182353   67.59657      \n",
       "27   16.1343253  6.419315   67.52551      \n",
       "28   27.1771622  6.084778   66.33617      \n",
       "29    3.8905488  7.053401   65.91837      \n",
       "30   36.1193300  5.815497   64.14659      \n",
       "31  264.9900195  6.945832  104.19514      \n",
       "32  355.4399598  5.804980   82.91002      \n",
       "33  201.0548668  5.742558   76.95771      \n",
       "34  104.2692095  5.561909   68.30101      \n",
       "35    4.2905393  6.868429   63.69902      \n",
       "36   17.3788729  6.084505   62.58301      \n",
       "37    9.3266336  6.256719   60.23246      \n",
       "38    9.2903769  6.038750   56.78904      \n",
       "39  205.6638306  4.313432   51.78242      \n",
       "40   32.5478429  4.814760   47.59118      \n",
       "41    3.3047155  8.264082   85.70754      \n",
       "42    1.2391984  8.498071   82.11545      \n",
       "43    1.0560434  8.486455   80.86133      \n",
       "44   34.0934208  6.069483   67.99777      \n",
       "45    0.5367777  7.923828   67.72968      \n",
       "46    1.0059735  7.716285   67.31689      \n",
       "47   17.8752324  6.205175   64.81078      \n",
       "48    1.6051126  7.360202   64.36434      \n",
       "49   15.0652428  6.172990   62.84169      \n",
       "50    0.1169610  7.725362   60.96108      \n",
       "51  307.8304388  8.265992   68.36530      \n",
       "52   35.5365007  5.479194   58.47038      \n",
       "53   11.2769586  5.801537   54.65932      \n",
       "54  157.9350323  7.303187   53.40305      \n",
       "55  129.3287412  7.014899   49.28676      \n",
       "56  121.2018765  6.921268   47.98600      \n",
       "57  394.8978670  5.217772   45.02407      \n",
       "58 1113.6244100  4.390760   44.44478      \n",
       "59   10.8204319  5.049968   43.51486      \n",
       "60   95.5038780  6.577487   43.36218      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% arrange(-heuristic.score)  %>% group_by(cell.type.2) %>% \n",
    "    slice_max(heuristic.score, n = 10)  %>% head(60)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 640,
   "metadata": {},
   "outputs": [],
   "source": [
    "fly_top_ten_markers.20 = cell.type.2.markers_no_doubletons  %>% filter(cell.type.2.tpm > other.tpm)  %>% arrange(-heuristic.score)  %>% group_by(cell.type.2) %>% \n",
    "    slice_max(heuristic.score, n = 20) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 641,
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(fly_top_ten_markers.20, \"data/fly_top_ten_markers_new_annotation.20.tsv\", quote = FALSE, sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 642,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 84 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cell.type.2</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>a </td><td> 432</td></tr>\n",
       "\t<tr><td>aa</td><td>1540</td></tr>\n",
       "\t<tr><td>ab</td><td>1048</td></tr>\n",
       "\t<tr><td>ac</td><td>3765</td></tr>\n",
       "\t<tr><td>ad</td><td> 719</td></tr>\n",
       "\t<tr><td>ae</td><td>1019</td></tr>\n",
       "\t<tr><td>af</td><td>  94</td></tr>\n",
       "\t<tr><td>ag</td><td> 495</td></tr>\n",
       "\t<tr><td>ah</td><td>2347</td></tr>\n",
       "\t<tr><td>ai</td><td>1265</td></tr>\n",
       "\t<tr><td>aj</td><td>1760</td></tr>\n",
       "\t<tr><td>ak</td><td> 517</td></tr>\n",
       "\t<tr><td>al</td><td>2535</td></tr>\n",
       "\t<tr><td>am</td><td>1521</td></tr>\n",
       "\t<tr><td>an</td><td>1785</td></tr>\n",
       "\t<tr><td>ao</td><td> 502</td></tr>\n",
       "\t<tr><td>ap</td><td>2022</td></tr>\n",
       "\t<tr><td>aq</td><td>1136</td></tr>\n",
       "\t<tr><td>ar</td><td>3047</td></tr>\n",
       "\t<tr><td>as</td><td> 929</td></tr>\n",
       "\t<tr><td>at</td><td>1796</td></tr>\n",
       "\t<tr><td>au</td><td>1347</td></tr>\n",
       "\t<tr><td>av</td><td>1689</td></tr>\n",
       "\t<tr><td>aw</td><td> 285</td></tr>\n",
       "\t<tr><td>ax</td><td>3395</td></tr>\n",
       "\t<tr><td>ay</td><td>4748</td></tr>\n",
       "\t<tr><td>az</td><td>4763</td></tr>\n",
       "\t<tr><td>b </td><td>1397</td></tr>\n",
       "\t<tr><td>ba</td><td>1864</td></tr>\n",
       "\t<tr><td>bb</td><td>1495</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>c </td><td>  96</td></tr>\n",
       "\t<tr><td>ca</td><td>3298</td></tr>\n",
       "\t<tr><td>cb</td><td>4006</td></tr>\n",
       "\t<tr><td>cc</td><td>6735</td></tr>\n",
       "\t<tr><td>cd</td><td>2267</td></tr>\n",
       "\t<tr><td>ce</td><td>1077</td></tr>\n",
       "\t<tr><td>cf</td><td> 592</td></tr>\n",
       "\t<tr><td>d </td><td> 178</td></tr>\n",
       "\t<tr><td>e </td><td>4655</td></tr>\n",
       "\t<tr><td>f </td><td>1408</td></tr>\n",
       "\t<tr><td>g </td><td> 667</td></tr>\n",
       "\t<tr><td>h </td><td> 356</td></tr>\n",
       "\t<tr><td>i </td><td>8848</td></tr>\n",
       "\t<tr><td>j </td><td> 163</td></tr>\n",
       "\t<tr><td>k </td><td>  50</td></tr>\n",
       "\t<tr><td>l </td><td>1566</td></tr>\n",
       "\t<tr><td>m </td><td>2650</td></tr>\n",
       "\t<tr><td>n </td><td>2285</td></tr>\n",
       "\t<tr><td>o </td><td>2055</td></tr>\n",
       "\t<tr><td>p </td><td> 123</td></tr>\n",
       "\t<tr><td>q </td><td>8835</td></tr>\n",
       "\t<tr><td>r </td><td>3182</td></tr>\n",
       "\t<tr><td>s </td><td>4153</td></tr>\n",
       "\t<tr><td>t </td><td> 918</td></tr>\n",
       "\t<tr><td>u </td><td> 505</td></tr>\n",
       "\t<tr><td>v </td><td>5308</td></tr>\n",
       "\t<tr><td>w </td><td> 628</td></tr>\n",
       "\t<tr><td>x </td><td> 512</td></tr>\n",
       "\t<tr><td>y </td><td>1366</td></tr>\n",
       "\t<tr><td>z </td><td> 963</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 84 × 2\n",
       "\\begin{tabular}{ll}\n",
       " cell.type.2 & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t a  &  432\\\\\n",
       "\t aa & 1540\\\\\n",
       "\t ab & 1048\\\\\n",
       "\t ac & 3765\\\\\n",
       "\t ad &  719\\\\\n",
       "\t ae & 1019\\\\\n",
       "\t af &   94\\\\\n",
       "\t ag &  495\\\\\n",
       "\t ah & 2347\\\\\n",
       "\t ai & 1265\\\\\n",
       "\t aj & 1760\\\\\n",
       "\t ak &  517\\\\\n",
       "\t al & 2535\\\\\n",
       "\t am & 1521\\\\\n",
       "\t an & 1785\\\\\n",
       "\t ao &  502\\\\\n",
       "\t ap & 2022\\\\\n",
       "\t aq & 1136\\\\\n",
       "\t ar & 3047\\\\\n",
       "\t as &  929\\\\\n",
       "\t at & 1796\\\\\n",
       "\t au & 1347\\\\\n",
       "\t av & 1689\\\\\n",
       "\t aw &  285\\\\\n",
       "\t ax & 3395\\\\\n",
       "\t ay & 4748\\\\\n",
       "\t az & 4763\\\\\n",
       "\t b  & 1397\\\\\n",
       "\t ba & 1864\\\\\n",
       "\t bb & 1495\\\\\n",
       "\t ⋮ & ⋮\\\\\n",
       "\t c  &   96\\\\\n",
       "\t ca & 3298\\\\\n",
       "\t cb & 4006\\\\\n",
       "\t cc & 6735\\\\\n",
       "\t cd & 2267\\\\\n",
       "\t ce & 1077\\\\\n",
       "\t cf &  592\\\\\n",
       "\t d  &  178\\\\\n",
       "\t e  & 4655\\\\\n",
       "\t f  & 1408\\\\\n",
       "\t g  &  667\\\\\n",
       "\t h  &  356\\\\\n",
       "\t i  & 8848\\\\\n",
       "\t j  &  163\\\\\n",
       "\t k  &   50\\\\\n",
       "\t l  & 1566\\\\\n",
       "\t m  & 2650\\\\\n",
       "\t n  & 2285\\\\\n",
       "\t o  & 2055\\\\\n",
       "\t p  &  123\\\\\n",
       "\t q  & 8835\\\\\n",
       "\t r  & 3182\\\\\n",
       "\t s  & 4153\\\\\n",
       "\t t  &  918\\\\\n",
       "\t u  &  505\\\\\n",
       "\t v  & 5308\\\\\n",
       "\t w  &  628\\\\\n",
       "\t x  &  512\\\\\n",
       "\t y  & 1366\\\\\n",
       "\t z  &  963\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 84 × 2\n",
       "\n",
       "| cell.type.2 &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| a  |  432 |\n",
       "| aa | 1540 |\n",
       "| ab | 1048 |\n",
       "| ac | 3765 |\n",
       "| ad |  719 |\n",
       "| ae | 1019 |\n",
       "| af |   94 |\n",
       "| ag |  495 |\n",
       "| ah | 2347 |\n",
       "| ai | 1265 |\n",
       "| aj | 1760 |\n",
       "| ak |  517 |\n",
       "| al | 2535 |\n",
       "| am | 1521 |\n",
       "| an | 1785 |\n",
       "| ao |  502 |\n",
       "| ap | 2022 |\n",
       "| aq | 1136 |\n",
       "| ar | 3047 |\n",
       "| as |  929 |\n",
       "| at | 1796 |\n",
       "| au | 1347 |\n",
       "| av | 1689 |\n",
       "| aw |  285 |\n",
       "| ax | 3395 |\n",
       "| ay | 4748 |\n",
       "| az | 4763 |\n",
       "| b  | 1397 |\n",
       "| ba | 1864 |\n",
       "| bb | 1495 |\n",
       "| ⋮ | ⋮ |\n",
       "| c  |   96 |\n",
       "| ca | 3298 |\n",
       "| cb | 4006 |\n",
       "| cc | 6735 |\n",
       "| cd | 2267 |\n",
       "| ce | 1077 |\n",
       "| cf |  592 |\n",
       "| d  |  178 |\n",
       "| e  | 4655 |\n",
       "| f  | 1408 |\n",
       "| g  |  667 |\n",
       "| h  |  356 |\n",
       "| i  | 8848 |\n",
       "| j  |  163 |\n",
       "| k  |   50 |\n",
       "| l  | 1566 |\n",
       "| m  | 2650 |\n",
       "| n  | 2285 |\n",
       "| o  | 2055 |\n",
       "| p  |  123 |\n",
       "| q  | 8835 |\n",
       "| r  | 3182 |\n",
       "| s  | 4153 |\n",
       "| t  |  918 |\n",
       "| u  |  505 |\n",
       "| v  | 5308 |\n",
       "| w  |  628 |\n",
       "| x  |  512 |\n",
       "| y  | 1366 |\n",
       "| z  |  963 |\n",
       "\n"
      ],
      "text/plain": [
       "   cell.type.2 n   \n",
       "1  a            432\n",
       "2  aa          1540\n",
       "3  ab          1048\n",
       "4  ac          3765\n",
       "5  ad           719\n",
       "6  ae          1019\n",
       "7  af            94\n",
       "8  ag           495\n",
       "9  ah          2347\n",
       "10 ai          1265\n",
       "11 aj          1760\n",
       "12 ak           517\n",
       "13 al          2535\n",
       "14 am          1521\n",
       "15 an          1785\n",
       "16 ao           502\n",
       "17 ap          2022\n",
       "18 aq          1136\n",
       "19 ar          3047\n",
       "20 as           929\n",
       "21 at          1796\n",
       "22 au          1347\n",
       "23 av          1689\n",
       "24 aw           285\n",
       "25 ax          3395\n",
       "26 ay          4748\n",
       "27 az          4763\n",
       "28 b           1397\n",
       "29 ba          1864\n",
       "30 bb          1495\n",
       "⋮  ⋮           ⋮   \n",
       "55 c             96\n",
       "56 ca          3298\n",
       "57 cb          4006\n",
       "58 cc          6735\n",
       "59 cd          2267\n",
       "60 ce          1077\n",
       "61 cf           592\n",
       "62 d            178\n",
       "63 e           4655\n",
       "64 f           1408\n",
       "65 g            667\n",
       "66 h            356\n",
       "67 i           8848\n",
       "68 j            163\n",
       "69 k             50\n",
       "70 l           1566\n",
       "71 m           2650\n",
       "72 n           2285\n",
       "73 o           2055\n",
       "74 p            123\n",
       "75 q           8835\n",
       "76 r           3182\n",
       "77 s           4153\n",
       "78 t            918\n",
       "79 u            505\n",
       "80 v           5308\n",
       "81 w            628\n",
       "82 x            512\n",
       "83 y           1366\n",
       "84 z            963"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub_no_doubletons  %>% count(cell.type.2)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 644,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 0 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 0 × 8\n",
       "\\begin{tabular}{llllllll}\n",
       " cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       " <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 0 × 8\n",
       "\n",
       "| cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "\n"
      ],
      "text/plain": [
       "     cell.type.2 gene cell.type.2.n.umi other.n.umi cell.type.2.tpm other.tpm\n",
       "     log2.ratio heuristic.score"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fly_top_ten_markers.20   %>% filter(gene == \"ad\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 488,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'cell.type.2'</li><li>'gene'</li><li>'cell.type.2.n.umi'</li><li>'other.n.umi'</li><li>'cell.type.2.tpm'</li><li>'other.tpm'</li><li>'log2.ratio'</li><li>'heuristic.score'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'cell.type.2'\n",
       "\\item 'gene'\n",
       "\\item 'cell.type.2.n.umi'\n",
       "\\item 'other.n.umi'\n",
       "\\item 'cell.type.2.tpm'\n",
       "\\item 'other.tpm'\n",
       "\\item 'log2.ratio'\n",
       "\\item 'heuristic.score'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'cell.type.2'\n",
       "2. 'gene'\n",
       "3. 'cell.type.2.n.umi'\n",
       "4. 'other.n.umi'\n",
       "5. 'cell.type.2.tpm'\n",
       "6. 'other.tpm'\n",
       "7. 'log2.ratio'\n",
       "8. 'heuristic.score'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"cell.type.2\"       \"gene\"              \"cell.type.2.n.umi\"\n",
       "[4] \"other.n.umi\"       \"cell.type.2.tpm\"   \"other.tpm\"        \n",
       "[7] \"log2.ratio\"        \"heuristic.score\"  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "colnames(fly_top_ten_markers)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get TF targets for Sue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 645,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'a'</li><li>'aa'</li><li>'ab'</li><li>'ac'</li><li>'ad'</li><li>'ae'</li><li>'af'</li><li>'ag'</li><li>'ah'</li><li>'ai'</li><li>'aj'</li><li>'ak'</li><li>'al'</li><li>'am'</li><li>'an'</li><li>'annot_type'</li><li>'ao'</li><li>'ap'</li><li>'aq'</li><li>'ar'</li><li>'as'</li><li>'at'</li><li>'au'</li><li>'av'</li><li>'aw'</li><li>'ax'</li><li>'ay'</li><li>'az'</li><li>'b'</li><li>'ba'</li><li>'bb'</li><li>'bc'</li><li>'bd'</li><li>'be'</li><li>'bf'</li><li>'bg'</li><li>'bh'</li><li>'bi'</li><li>'bj'</li><li>'bk'</li><li>'bl'</li><li>'bm'</li><li>'bn'</li><li>'bo'</li><li>'bp'</li><li>'bq'</li><li>'br'</li><li>'bs'</li><li>'bt'</li><li>'bu'</li><li>'bv'</li><li>'bw'</li><li>'bx'</li><li>'by'</li><li>'bz'</li><li>'c'</li><li>'ca'</li><li>'cb'</li><li>'cc'</li><li>'cd'</li><li>'cds_sub'</li><li>'cds_sub_no_doubletons'</li><li>'cds_sub_no_multitons'</li><li>'cds_sub_no_sing'</li><li>'ce'</li><li>'cell_type'</li><li>'cell.type'</li><li>'cell.type.2.markers'</li><li>'cell.type.2.markers_no_doubletons'</li><li>'cell.type.markers.sub300_no_singletons.cell.type'</li><li>'cell.type.markers.sub300.cell.type'</li><li>'cellIDs'</li><li>'cells'</li><li>'cellTypes'</li><li>'cf'</li><li>'Cholinergic_1'</li><li>'clus'</li><li>'cluster.markers'</li><li>'cluster.markers.sub300'</li><li>'cluster.markers.sub300_no_doubletons'</li><li>'cluster.markers.sub300_no_doubletons_all'</li><li>'cluster.markers.sub300_no_singletons'</li><li>'d'</li><li>'DEG.results'</li><li>'DEG.results_no_doubletons'</li><li>'df'</li><li>'df_sub_no_doubletons'</li><li>'df_sub_no_multitons'</li><li>'df_sub_no_singletons'</li><li>'df_sub300'</li><li>'df_sub300_copy'</li><li>'doubletons'</li><li>'e'</li><li>'exp_matrix'</li><li>'exp_matrix_mean'</li><li>'exp_matrix_mean_original'</li><li>'exp_matrix_sub300_cell.type.2_no_doubletons'</li><li>'exp_matrix_sub300_cell.type.2_no_doubletons_mean'</li><li>'exp_matrix_sub300_no_doubletons'</li><li>'expresses.gene'</li><li>'f'</li><li>'file_name'</li><li>'fly_peaks_TF'</li><li>'fly_TFs'</li><li>'fly_top_ten_markers'</li><li>'fly_top_ten_markers.20'</li><li>'g'</li><li>'gen'</li><li>'gene_max_mean_entropy'</li><li>'get.gene.id'</li><li>'get.gene.id2'</li><li>'get.norm.expr.matrix'</li><li>'get.norm.expr.matrix.3'</li><li>'gg_agg'</li><li>'gg_both'</li><li>'gg_tf'</li><li>'h'</li><li>'i'</li><li>'ids'</li><li>'j'</li><li>'k'</li><li>'k1'</li><li>'k2'</li><li>'l'</li><li>'labels'</li><li>'m'</li><li>'make_df'</li><li>'missing_genes'</li><li>'missing_genes_aliases'</li><li>'multitons'</li><li>'n'</li><li>'names_expanded'</li><li>'neuronal_subcluster'</li><li>'neuronal_subsubcluster'</li><li>'neuronal_subsubcluster2'</li><li>'nineteentons'</li><li>'nnMatches'</li><li>'o'</li><li>'p'</li><li>'pca'</li><li>'peedeef'</li><li>'plot.expr'</li><li>'plot.expr_clus'</li><li>'plot.expr_clus.br'</li><li>'plot.expr_clus.br2'</li><li>'plot.expr.fly'</li><li>'plot.expr3'</li><li>'q'</li><li>'r'</li><li>'rann'</li><li>'res'</li><li>'res_label'</li><li>'s'</li><li>'set.cell.type'</li><li>'singletons'</li><li>'t'</li><li>'t_target_values'</li><li>'t_target_values_row'</li><li>'t_tf_values'</li><li>'t_tf_values_row'</li><li>'target'</li><li>'target_values'</li><li>'tf'</li><li>'TF_exp_matrix_sub300_cell.type.2_no_doubletons'</li><li>'TF_exp_matrix_sub300_cell.type.2_no_doubletons_log'</li><li>'TF_exp_matrix_sub300_no_doubletons'</li><li>'TF_exp_matrix_sub300_no_doubletons_log'</li><li>'tf_title'</li><li>'tf_values'</li><li>'title'</li><li>'twentytons'</li><li>'two.set.differential.gene.test'</li><li>'two.set.differential.gene.test.3'</li><li>'two.set.differential.gene.test.3a'</li><li>'two.set.differential.gene.test.3b'</li><li>'u'</li><li>'v'</li><li>'vector_angle_emb_targets'</li><li>'w'</li><li>'x'</li><li>'y'</li><li>'z'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'a'\n",
       "\\item 'aa'\n",
       "\\item 'ab'\n",
       "\\item 'ac'\n",
       "\\item 'ad'\n",
       "\\item 'ae'\n",
       "\\item 'af'\n",
       "\\item 'ag'\n",
       "\\item 'ah'\n",
       "\\item 'ai'\n",
       "\\item 'aj'\n",
       "\\item 'ak'\n",
       "\\item 'al'\n",
       "\\item 'am'\n",
       "\\item 'an'\n",
       "\\item 'annot\\_type'\n",
       "\\item 'ao'\n",
       "\\item 'ap'\n",
       "\\item 'aq'\n",
       "\\item 'ar'\n",
       "\\item 'as'\n",
       "\\item 'at'\n",
       "\\item 'au'\n",
       "\\item 'av'\n",
       "\\item 'aw'\n",
       "\\item 'ax'\n",
       "\\item 'ay'\n",
       "\\item 'az'\n",
       "\\item 'b'\n",
       "\\item 'ba'\n",
       "\\item 'bb'\n",
       "\\item 'bc'\n",
       "\\item 'bd'\n",
       "\\item 'be'\n",
       "\\item 'bf'\n",
       "\\item 'bg'\n",
       "\\item 'bh'\n",
       "\\item 'bi'\n",
       "\\item 'bj'\n",
       "\\item 'bk'\n",
       "\\item 'bl'\n",
       "\\item 'bm'\n",
       "\\item 'bn'\n",
       "\\item 'bo'\n",
       "\\item 'bp'\n",
       "\\item 'bq'\n",
       "\\item 'br'\n",
       "\\item 'bs'\n",
       "\\item 'bt'\n",
       "\\item 'bu'\n",
       "\\item 'bv'\n",
       "\\item 'bw'\n",
       "\\item 'bx'\n",
       "\\item 'by'\n",
       "\\item 'bz'\n",
       "\\item 'c'\n",
       "\\item 'ca'\n",
       "\\item 'cb'\n",
       "\\item 'cc'\n",
       "\\item 'cd'\n",
       "\\item 'cds\\_sub'\n",
       "\\item 'cds\\_sub\\_no\\_doubletons'\n",
       "\\item 'cds\\_sub\\_no\\_multitons'\n",
       "\\item 'cds\\_sub\\_no\\_sing'\n",
       "\\item 'ce'\n",
       "\\item 'cell\\_type'\n",
       "\\item 'cell.type'\n",
       "\\item 'cell.type.2.markers'\n",
       "\\item 'cell.type.2.markers\\_no\\_doubletons'\n",
       "\\item 'cell.type.markers.sub300\\_no\\_singletons.cell.type'\n",
       "\\item 'cell.type.markers.sub300.cell.type'\n",
       "\\item 'cellIDs'\n",
       "\\item 'cells'\n",
       "\\item 'cellTypes'\n",
       "\\item 'cf'\n",
       "\\item 'Cholinergic\\_1'\n",
       "\\item 'clus'\n",
       "\\item 'cluster.markers'\n",
       "\\item 'cluster.markers.sub300'\n",
       "\\item 'cluster.markers.sub300\\_no\\_doubletons'\n",
       "\\item 'cluster.markers.sub300\\_no\\_doubletons\\_all'\n",
       "\\item 'cluster.markers.sub300\\_no\\_singletons'\n",
       "\\item 'd'\n",
       "\\item 'DEG.results'\n",
       "\\item 'DEG.results\\_no\\_doubletons'\n",
       "\\item 'df'\n",
       "\\item 'df\\_sub\\_no\\_doubletons'\n",
       "\\item 'df\\_sub\\_no\\_multitons'\n",
       "\\item 'df\\_sub\\_no\\_singletons'\n",
       "\\item 'df\\_sub300'\n",
       "\\item 'df\\_sub300\\_copy'\n",
       "\\item 'doubletons'\n",
       "\\item 'e'\n",
       "\\item 'exp\\_matrix'\n",
       "\\item 'exp\\_matrix\\_mean'\n",
       "\\item 'exp\\_matrix\\_mean\\_original'\n",
       "\\item 'exp\\_matrix\\_sub300\\_cell.type.2\\_no\\_doubletons'\n",
       "\\item 'exp\\_matrix\\_sub300\\_cell.type.2\\_no\\_doubletons\\_mean'\n",
       "\\item 'exp\\_matrix\\_sub300\\_no\\_doubletons'\n",
       "\\item 'expresses.gene'\n",
       "\\item 'f'\n",
       "\\item 'file\\_name'\n",
       "\\item 'fly\\_peaks\\_TF'\n",
       "\\item 'fly\\_TFs'\n",
       "\\item 'fly\\_top\\_ten\\_markers'\n",
       "\\item 'fly\\_top\\_ten\\_markers.20'\n",
       "\\item 'g'\n",
       "\\item 'gen'\n",
       "\\item 'gene\\_max\\_mean\\_entropy'\n",
       "\\item 'get.gene.id'\n",
       "\\item 'get.gene.id2'\n",
       "\\item 'get.norm.expr.matrix'\n",
       "\\item 'get.norm.expr.matrix.3'\n",
       "\\item 'gg\\_agg'\n",
       "\\item 'gg\\_both'\n",
       "\\item 'gg\\_tf'\n",
       "\\item 'h'\n",
       "\\item 'i'\n",
       "\\item 'ids'\n",
       "\\item 'j'\n",
       "\\item 'k'\n",
       "\\item 'k1'\n",
       "\\item 'k2'\n",
       "\\item 'l'\n",
       "\\item 'labels'\n",
       "\\item 'm'\n",
       "\\item 'make\\_df'\n",
       "\\item 'missing\\_genes'\n",
       "\\item 'missing\\_genes\\_aliases'\n",
       "\\item 'multitons'\n",
       "\\item 'n'\n",
       "\\item 'names\\_expanded'\n",
       "\\item 'neuronal\\_subcluster'\n",
       "\\item 'neuronal\\_subsubcluster'\n",
       "\\item 'neuronal\\_subsubcluster2'\n",
       "\\item 'nineteentons'\n",
       "\\item 'nnMatches'\n",
       "\\item 'o'\n",
       "\\item 'p'\n",
       "\\item 'pca'\n",
       "\\item 'peedeef'\n",
       "\\item 'plot.expr'\n",
       "\\item 'plot.expr\\_clus'\n",
       "\\item 'plot.expr\\_clus.br'\n",
       "\\item 'plot.expr\\_clus.br2'\n",
       "\\item 'plot.expr.fly'\n",
       "\\item 'plot.expr3'\n",
       "\\item 'q'\n",
       "\\item 'r'\n",
       "\\item 'rann'\n",
       "\\item 'res'\n",
       "\\item 'res\\_label'\n",
       "\\item 's'\n",
       "\\item 'set.cell.type'\n",
       "\\item 'singletons'\n",
       "\\item 't'\n",
       "\\item 't\\_target\\_values'\n",
       "\\item 't\\_target\\_values\\_row'\n",
       "\\item 't\\_tf\\_values'\n",
       "\\item 't\\_tf\\_values\\_row'\n",
       "\\item 'target'\n",
       "\\item 'target\\_values'\n",
       "\\item 'tf'\n",
       "\\item 'TF\\_exp\\_matrix\\_sub300\\_cell.type.2\\_no\\_doubletons'\n",
       "\\item 'TF\\_exp\\_matrix\\_sub300\\_cell.type.2\\_no\\_doubletons\\_log'\n",
       "\\item 'TF\\_exp\\_matrix\\_sub300\\_no\\_doubletons'\n",
       "\\item 'TF\\_exp\\_matrix\\_sub300\\_no\\_doubletons\\_log'\n",
       "\\item 'tf\\_title'\n",
       "\\item 'tf\\_values'\n",
       "\\item 'title'\n",
       "\\item 'twentytons'\n",
       "\\item 'two.set.differential.gene.test'\n",
       "\\item 'two.set.differential.gene.test.3'\n",
       "\\item 'two.set.differential.gene.test.3a'\n",
       "\\item 'two.set.differential.gene.test.3b'\n",
       "\\item 'u'\n",
       "\\item 'v'\n",
       "\\item 'vector\\_angle\\_emb\\_targets'\n",
       "\\item 'w'\n",
       "\\item 'x'\n",
       "\\item 'y'\n",
       "\\item 'z'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'a'\n",
       "2. 'aa'\n",
       "3. 'ab'\n",
       "4. 'ac'\n",
       "5. 'ad'\n",
       "6. 'ae'\n",
       "7. 'af'\n",
       "8. 'ag'\n",
       "9. 'ah'\n",
       "10. 'ai'\n",
       "11. 'aj'\n",
       "12. 'ak'\n",
       "13. 'al'\n",
       "14. 'am'\n",
       "15. 'an'\n",
       "16. 'annot_type'\n",
       "17. 'ao'\n",
       "18. 'ap'\n",
       "19. 'aq'\n",
       "20. 'ar'\n",
       "21. 'as'\n",
       "22. 'at'\n",
       "23. 'au'\n",
       "24. 'av'\n",
       "25. 'aw'\n",
       "26. 'ax'\n",
       "27. 'ay'\n",
       "28. 'az'\n",
       "29. 'b'\n",
       "30. 'ba'\n",
       "31. 'bb'\n",
       "32. 'bc'\n",
       "33. 'bd'\n",
       "34. 'be'\n",
       "35. 'bf'\n",
       "36. 'bg'\n",
       "37. 'bh'\n",
       "38. 'bi'\n",
       "39. 'bj'\n",
       "40. 'bk'\n",
       "41. 'bl'\n",
       "42. 'bm'\n",
       "43. 'bn'\n",
       "44. 'bo'\n",
       "45. 'bp'\n",
       "46. 'bq'\n",
       "47. 'br'\n",
       "48. 'bs'\n",
       "49. 'bt'\n",
       "50. 'bu'\n",
       "51. 'bv'\n",
       "52. 'bw'\n",
       "53. 'bx'\n",
       "54. 'by'\n",
       "55. 'bz'\n",
       "56. 'c'\n",
       "57. 'ca'\n",
       "58. 'cb'\n",
       "59. 'cc'\n",
       "60. 'cd'\n",
       "61. 'cds_sub'\n",
       "62. 'cds_sub_no_doubletons'\n",
       "63. 'cds_sub_no_multitons'\n",
       "64. 'cds_sub_no_sing'\n",
       "65. 'ce'\n",
       "66. 'cell_type'\n",
       "67. 'cell.type'\n",
       "68. 'cell.type.2.markers'\n",
       "69. 'cell.type.2.markers_no_doubletons'\n",
       "70. 'cell.type.markers.sub300_no_singletons.cell.type'\n",
       "71. 'cell.type.markers.sub300.cell.type'\n",
       "72. 'cellIDs'\n",
       "73. 'cells'\n",
       "74. 'cellTypes'\n",
       "75. 'cf'\n",
       "76. 'Cholinergic_1'\n",
       "77. 'clus'\n",
       "78. 'cluster.markers'\n",
       "79. 'cluster.markers.sub300'\n",
       "80. 'cluster.markers.sub300_no_doubletons'\n",
       "81. 'cluster.markers.sub300_no_doubletons_all'\n",
       "82. 'cluster.markers.sub300_no_singletons'\n",
       "83. 'd'\n",
       "84. 'DEG.results'\n",
       "85. 'DEG.results_no_doubletons'\n",
       "86. 'df'\n",
       "87. 'df_sub_no_doubletons'\n",
       "88. 'df_sub_no_multitons'\n",
       "89. 'df_sub_no_singletons'\n",
       "90. 'df_sub300'\n",
       "91. 'df_sub300_copy'\n",
       "92. 'doubletons'\n",
       "93. 'e'\n",
       "94. 'exp_matrix'\n",
       "95. 'exp_matrix_mean'\n",
       "96. 'exp_matrix_mean_original'\n",
       "97. 'exp_matrix_sub300_cell.type.2_no_doubletons'\n",
       "98. 'exp_matrix_sub300_cell.type.2_no_doubletons_mean'\n",
       "99. 'exp_matrix_sub300_no_doubletons'\n",
       "100. 'expresses.gene'\n",
       "101. 'f'\n",
       "102. 'file_name'\n",
       "103. 'fly_peaks_TF'\n",
       "104. 'fly_TFs'\n",
       "105. 'fly_top_ten_markers'\n",
       "106. 'fly_top_ten_markers.20'\n",
       "107. 'g'\n",
       "108. 'gen'\n",
       "109. 'gene_max_mean_entropy'\n",
       "110. 'get.gene.id'\n",
       "111. 'get.gene.id2'\n",
       "112. 'get.norm.expr.matrix'\n",
       "113. 'get.norm.expr.matrix.3'\n",
       "114. 'gg_agg'\n",
       "115. 'gg_both'\n",
       "116. 'gg_tf'\n",
       "117. 'h'\n",
       "118. 'i'\n",
       "119. 'ids'\n",
       "120. 'j'\n",
       "121. 'k'\n",
       "122. 'k1'\n",
       "123. 'k2'\n",
       "124. 'l'\n",
       "125. 'labels'\n",
       "126. 'm'\n",
       "127. 'make_df'\n",
       "128. 'missing_genes'\n",
       "129. 'missing_genes_aliases'\n",
       "130. 'multitons'\n",
       "131. 'n'\n",
       "132. 'names_expanded'\n",
       "133. 'neuronal_subcluster'\n",
       "134. 'neuronal_subsubcluster'\n",
       "135. 'neuronal_subsubcluster2'\n",
       "136. 'nineteentons'\n",
       "137. 'nnMatches'\n",
       "138. 'o'\n",
       "139. 'p'\n",
       "140. 'pca'\n",
       "141. 'peedeef'\n",
       "142. 'plot.expr'\n",
       "143. 'plot.expr_clus'\n",
       "144. 'plot.expr_clus.br'\n",
       "145. 'plot.expr_clus.br2'\n",
       "146. 'plot.expr.fly'\n",
       "147. 'plot.expr3'\n",
       "148. 'q'\n",
       "149. 'r'\n",
       "150. 'rann'\n",
       "151. 'res'\n",
       "152. 'res_label'\n",
       "153. 's'\n",
       "154. 'set.cell.type'\n",
       "155. 'singletons'\n",
       "156. 't'\n",
       "157. 't_target_values'\n",
       "158. 't_target_values_row'\n",
       "159. 't_tf_values'\n",
       "160. 't_tf_values_row'\n",
       "161. 'target'\n",
       "162. 'target_values'\n",
       "163. 'tf'\n",
       "164. 'TF_exp_matrix_sub300_cell.type.2_no_doubletons'\n",
       "165. 'TF_exp_matrix_sub300_cell.type.2_no_doubletons_log'\n",
       "166. 'TF_exp_matrix_sub300_no_doubletons'\n",
       "167. 'TF_exp_matrix_sub300_no_doubletons_log'\n",
       "168. 'tf_title'\n",
       "169. 'tf_values'\n",
       "170. 'title'\n",
       "171. 'twentytons'\n",
       "172. 'two.set.differential.gene.test'\n",
       "173. 'two.set.differential.gene.test.3'\n",
       "174. 'two.set.differential.gene.test.3a'\n",
       "175. 'two.set.differential.gene.test.3b'\n",
       "176. 'u'\n",
       "177. 'v'\n",
       "178. 'vector_angle_emb_targets'\n",
       "179. 'w'\n",
       "180. 'x'\n",
       "181. 'y'\n",
       "182. 'z'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"a\"                                                 \n",
       "  [2] \"aa\"                                                \n",
       "  [3] \"ab\"                                                \n",
       "  [4] \"ac\"                                                \n",
       "  [5] \"ad\"                                                \n",
       "  [6] \"ae\"                                                \n",
       "  [7] \"af\"                                                \n",
       "  [8] \"ag\"                                                \n",
       "  [9] \"ah\"                                                \n",
       " [10] \"ai\"                                                \n",
       " [11] \"aj\"                                                \n",
       " [12] \"ak\"                                                \n",
       " [13] \"al\"                                                \n",
       " [14] \"am\"                                                \n",
       " [15] \"an\"                                                \n",
       " [16] \"annot_type\"                                        \n",
       " [17] \"ao\"                                                \n",
       " [18] \"ap\"                                                \n",
       " [19] \"aq\"                                                \n",
       " [20] \"ar\"                                                \n",
       " [21] \"as\"                                                \n",
       " [22] \"at\"                                                \n",
       " [23] \"au\"                                                \n",
       " [24] \"av\"                                                \n",
       " [25] \"aw\"                                                \n",
       " [26] \"ax\"                                                \n",
       " [27] \"ay\"                                                \n",
       " [28] \"az\"                                                \n",
       " [29] \"b\"                                                 \n",
       " [30] \"ba\"                                                \n",
       " [31] \"bb\"                                                \n",
       " [32] \"bc\"                                                \n",
       " [33] \"bd\"                                                \n",
       " [34] \"be\"                                                \n",
       " [35] \"bf\"                                                \n",
       " [36] \"bg\"                                                \n",
       " [37] \"bh\"                                                \n",
       " [38] \"bi\"                                                \n",
       " [39] \"bj\"                                                \n",
       " [40] \"bk\"                                                \n",
       " [41] \"bl\"                                                \n",
       " [42] \"bm\"                                                \n",
       " [43] \"bn\"                                                \n",
       " [44] \"bo\"                                                \n",
       " [45] \"bp\"                                                \n",
       " [46] \"bq\"                                                \n",
       " [47] \"br\"                                                \n",
       " [48] \"bs\"                                                \n",
       " [49] \"bt\"                                                \n",
       " [50] \"bu\"                                                \n",
       " [51] \"bv\"                                                \n",
       " [52] \"bw\"                                                \n",
       " [53] \"bx\"                                                \n",
       " [54] \"by\"                                                \n",
       " [55] \"bz\"                                                \n",
       " [56] \"c\"                                                 \n",
       " [57] \"ca\"                                                \n",
       " [58] \"cb\"                                                \n",
       " [59] \"cc\"                                                \n",
       " [60] \"cd\"                                                \n",
       " [61] \"cds_sub\"                                           \n",
       " [62] \"cds_sub_no_doubletons\"                             \n",
       " [63] \"cds_sub_no_multitons\"                              \n",
       " [64] \"cds_sub_no_sing\"                                   \n",
       " [65] \"ce\"                                                \n",
       " [66] \"cell_type\"                                         \n",
       " [67] \"cell.type\"                                         \n",
       " [68] \"cell.type.2.markers\"                               \n",
       " [69] \"cell.type.2.markers_no_doubletons\"                 \n",
       " [70] \"cell.type.markers.sub300_no_singletons.cell.type\"  \n",
       " [71] \"cell.type.markers.sub300.cell.type\"                \n",
       " [72] \"cellIDs\"                                           \n",
       " [73] \"cells\"                                             \n",
       " [74] \"cellTypes\"                                         \n",
       " [75] \"cf\"                                                \n",
       " [76] \"Cholinergic_1\"                                     \n",
       " [77] \"clus\"                                              \n",
       " [78] \"cluster.markers\"                                   \n",
       " [79] \"cluster.markers.sub300\"                            \n",
       " [80] \"cluster.markers.sub300_no_doubletons\"              \n",
       " [81] \"cluster.markers.sub300_no_doubletons_all\"          \n",
       " [82] \"cluster.markers.sub300_no_singletons\"              \n",
       " [83] \"d\"                                                 \n",
       " [84] \"DEG.results\"                                       \n",
       " [85] \"DEG.results_no_doubletons\"                         \n",
       " [86] \"df\"                                                \n",
       " [87] \"df_sub_no_doubletons\"                              \n",
       " [88] \"df_sub_no_multitons\"                               \n",
       " [89] \"df_sub_no_singletons\"                              \n",
       " [90] \"df_sub300\"                                         \n",
       " [91] \"df_sub300_copy\"                                    \n",
       " [92] \"doubletons\"                                        \n",
       " [93] \"e\"                                                 \n",
       " [94] \"exp_matrix\"                                        \n",
       " [95] \"exp_matrix_mean\"                                   \n",
       " [96] \"exp_matrix_mean_original\"                          \n",
       " [97] \"exp_matrix_sub300_cell.type.2_no_doubletons\"       \n",
       " [98] \"exp_matrix_sub300_cell.type.2_no_doubletons_mean\"  \n",
       " [99] \"exp_matrix_sub300_no_doubletons\"                   \n",
       "[100] \"expresses.gene\"                                    \n",
       "[101] \"f\"                                                 \n",
       "[102] \"file_name\"                                         \n",
       "[103] \"fly_peaks_TF\"                                      \n",
       "[104] \"fly_TFs\"                                           \n",
       "[105] \"fly_top_ten_markers\"                               \n",
       "[106] \"fly_top_ten_markers.20\"                            \n",
       "[107] \"g\"                                                 \n",
       "[108] \"gen\"                                               \n",
       "[109] \"gene_max_mean_entropy\"                             \n",
       "[110] \"get.gene.id\"                                       \n",
       "[111] \"get.gene.id2\"                                      \n",
       "[112] \"get.norm.expr.matrix\"                              \n",
       "[113] \"get.norm.expr.matrix.3\"                            \n",
       "[114] \"gg_agg\"                                            \n",
       "[115] \"gg_both\"                                           \n",
       "[116] \"gg_tf\"                                             \n",
       "[117] \"h\"                                                 \n",
       "[118] \"i\"                                                 \n",
       "[119] \"ids\"                                               \n",
       "[120] \"j\"                                                 \n",
       "[121] \"k\"                                                 \n",
       "[122] \"k1\"                                                \n",
       "[123] \"k2\"                                                \n",
       "[124] \"l\"                                                 \n",
       "[125] \"labels\"                                            \n",
       "[126] \"m\"                                                 \n",
       "[127] \"make_df\"                                           \n",
       "[128] \"missing_genes\"                                     \n",
       "[129] \"missing_genes_aliases\"                             \n",
       "[130] \"multitons\"                                         \n",
       "[131] \"n\"                                                 \n",
       "[132] \"names_expanded\"                                    \n",
       "[133] \"neuronal_subcluster\"                               \n",
       "[134] \"neuronal_subsubcluster\"                            \n",
       "[135] \"neuronal_subsubcluster2\"                           \n",
       "[136] \"nineteentons\"                                      \n",
       "[137] \"nnMatches\"                                         \n",
       "[138] \"o\"                                                 \n",
       "[139] \"p\"                                                 \n",
       "[140] \"pca\"                                               \n",
       "[141] \"peedeef\"                                           \n",
       "[142] \"plot.expr\"                                         \n",
       "[143] \"plot.expr_clus\"                                    \n",
       "[144] \"plot.expr_clus.br\"                                 \n",
       "[145] \"plot.expr_clus.br2\"                                \n",
       "[146] \"plot.expr.fly\"                                     \n",
       "[147] \"plot.expr3\"                                        \n",
       "[148] \"q\"                                                 \n",
       "[149] \"r\"                                                 \n",
       "[150] \"rann\"                                              \n",
       "[151] \"res\"                                               \n",
       "[152] \"res_label\"                                         \n",
       "[153] \"s\"                                                 \n",
       "[154] \"set.cell.type\"                                     \n",
       "[155] \"singletons\"                                        \n",
       "[156] \"t\"                                                 \n",
       "[157] \"t_target_values\"                                   \n",
       "[158] \"t_target_values_row\"                               \n",
       "[159] \"t_tf_values\"                                       \n",
       "[160] \"t_tf_values_row\"                                   \n",
       "[161] \"target\"                                            \n",
       "[162] \"target_values\"                                     \n",
       "[163] \"tf\"                                                \n",
       "[164] \"TF_exp_matrix_sub300_cell.type.2_no_doubletons\"    \n",
       "[165] \"TF_exp_matrix_sub300_cell.type.2_no_doubletons_log\"\n",
       "[166] \"TF_exp_matrix_sub300_no_doubletons\"                \n",
       "[167] \"TF_exp_matrix_sub300_no_doubletons_log\"            \n",
       "[168] \"tf_title\"                                          \n",
       "[169] \"tf_values\"                                         \n",
       "[170] \"title\"                                             \n",
       "[171] \"twentytons\"                                        \n",
       "[172] \"two.set.differential.gene.test\"                    \n",
       "[173] \"two.set.differential.gene.test.3\"                  \n",
       "[174] \"two.set.differential.gene.test.3a\"                 \n",
       "[175] \"two.set.differential.gene.test.3b\"                 \n",
       "[176] \"u\"                                                 \n",
       "[177] \"v\"                                                 \n",
       "[178] \"vector_angle_emb_targets\"                          \n",
       "[179] \"w\"                                                 \n",
       "[180] \"x\"                                                 \n",
       "[181] \"y\"                                                 \n",
       "[182] \"z\"                                                 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ls()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 647,
   "metadata": {},
   "outputs": [],
   "source": [
    "fly_jsd_tf_target = readRDS(\"data/Fly_JSD_ChiSquare_ttest.rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 648,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>TF</th><th scope=col>nTargets</th><th scope=col>JSD</th><th scope=col>wilcoxP</th><th scope=col>chiP</th><th scope=col>t_test_p</th><th scope=col>targetMean</th><th scope=col>randomMean</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ab   </td><td> 135</td><td>0.2587306</td><td>3.316526e-01</td><td>1.081180e-04</td><td>1.457897e-01</td><td>0.007761771</td><td>-0.034110661</td></tr>\n",
       "\t<tr><td>abd-A</td><td>2877</td><td>0.1841270</td><td>1.619032e-77</td><td>1.289780e-65</td><td>1.048386e-75</td><td>0.046724300</td><td>-0.054141999</td></tr>\n",
       "\t<tr><td>Abd-B</td><td>1522</td><td>0.1162042</td><td>1.142477e-11</td><td>1.274070e-10</td><td>2.915755e-14</td><td>0.002285970</td><td>-0.038557669</td></tr>\n",
       "\t<tr><td>ac   </td><td> 799</td><td>0.2061522</td><td>7.235478e-15</td><td>1.076058e-13</td><td>2.997855e-17</td><td>0.107429744</td><td> 0.004356283</td></tr>\n",
       "\t<tr><td>Adf1 </td><td>1546</td><td>0.1626163</td><td>9.429591e-18</td><td>8.331135e-27</td><td>1.093153e-12</td><td>0.018633070</td><td> 0.099185161</td></tr>\n",
       "\t<tr><td>Aef1 </td><td>1083</td><td>0.1370532</td><td>1.211261e-08</td><td>3.250265e-13</td><td>6.120103e-04</td><td>0.053187933</td><td> 0.098447248</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 8\n",
       "\\begin{tabular}{llllllll}\n",
       " TF & nTargets & JSD & wilcoxP & chiP & t\\_test\\_p & targetMean & randomMean\\\\\n",
       " <chr> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t ab    &  135 & 0.2587306 & 3.316526e-01 & 1.081180e-04 & 1.457897e-01 & 0.007761771 & -0.034110661\\\\\n",
       "\t abd-A & 2877 & 0.1841270 & 1.619032e-77 & 1.289780e-65 & 1.048386e-75 & 0.046724300 & -0.054141999\\\\\n",
       "\t Abd-B & 1522 & 0.1162042 & 1.142477e-11 & 1.274070e-10 & 2.915755e-14 & 0.002285970 & -0.038557669\\\\\n",
       "\t ac    &  799 & 0.2061522 & 7.235478e-15 & 1.076058e-13 & 2.997855e-17 & 0.107429744 &  0.004356283\\\\\n",
       "\t Adf1  & 1546 & 0.1626163 & 9.429591e-18 & 8.331135e-27 & 1.093153e-12 & 0.018633070 &  0.099185161\\\\\n",
       "\t Aef1  & 1083 & 0.1370532 & 1.211261e-08 & 3.250265e-13 & 6.120103e-04 & 0.053187933 &  0.098447248\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 8\n",
       "\n",
       "| TF &lt;chr&gt; | nTargets &lt;int&gt; | JSD &lt;dbl&gt; | wilcoxP &lt;dbl&gt; | chiP &lt;dbl&gt; | t_test_p &lt;dbl&gt; | targetMean &lt;dbl&gt; | randomMean &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| ab    |  135 | 0.2587306 | 3.316526e-01 | 1.081180e-04 | 1.457897e-01 | 0.007761771 | -0.034110661 |\n",
       "| abd-A | 2877 | 0.1841270 | 1.619032e-77 | 1.289780e-65 | 1.048386e-75 | 0.046724300 | -0.054141999 |\n",
       "| Abd-B | 1522 | 0.1162042 | 1.142477e-11 | 1.274070e-10 | 2.915755e-14 | 0.002285970 | -0.038557669 |\n",
       "| ac    |  799 | 0.2061522 | 7.235478e-15 | 1.076058e-13 | 2.997855e-17 | 0.107429744 |  0.004356283 |\n",
       "| Adf1  | 1546 | 0.1626163 | 9.429591e-18 | 8.331135e-27 | 1.093153e-12 | 0.018633070 |  0.099185161 |\n",
       "| Aef1  | 1083 | 0.1370532 | 1.211261e-08 | 3.250265e-13 | 6.120103e-04 | 0.053187933 |  0.098447248 |\n",
       "\n"
      ],
      "text/plain": [
       "  TF    nTargets JSD       wilcoxP      chiP         t_test_p     targetMean \n",
       "1 ab     135     0.2587306 3.316526e-01 1.081180e-04 1.457897e-01 0.007761771\n",
       "2 abd-A 2877     0.1841270 1.619032e-77 1.289780e-65 1.048386e-75 0.046724300\n",
       "3 Abd-B 1522     0.1162042 1.142477e-11 1.274070e-10 2.915755e-14 0.002285970\n",
       "4 ac     799     0.2061522 7.235478e-15 1.076058e-13 2.997855e-17 0.107429744\n",
       "5 Adf1  1546     0.1626163 9.429591e-18 8.331135e-27 1.093153e-12 0.018633070\n",
       "6 Aef1  1083     0.1370532 1.211261e-08 3.250265e-13 6.120103e-04 0.053187933\n",
       "  randomMean  \n",
       "1 -0.034110661\n",
       "2 -0.054141999\n",
       "3 -0.038557669\n",
       "4  0.004356283\n",
       "5  0.099185161\n",
       "6  0.098447248"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fly_jsd_tf_target  %>% head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 655,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>66</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 1 × 1\n",
       "\\begin{tabular}{l}\n",
       " n\\\\\n",
       " <int>\\\\\n",
       "\\hline\n",
       "\t 66\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 1 × 1\n",
       "\n",
       "| n &lt;int&gt; |\n",
       "|---|\n",
       "| 66 |\n",
       "\n"
      ],
      "text/plain": [
       "  n \n",
       "1 66"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fly_jsd_tf_target  %>% filter(wilcoxP < 0.00001 & JSD > 0.18)  %>% count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 658,
   "metadata": {},
   "outputs": [],
   "source": [
    "options(repr.matrix.max.rows = 100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 659,
   "metadata": {},
   "outputs": [],
   "source": [
    "fly_jsd_tf_target_trim = fly_jsd_tf_target  %>% filter(wilcoxP < 0.00001 & JSD > 0.18)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 660,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 66 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>TF</th><th scope=col>nTargets</th><th scope=col>JSD</th><th scope=col>wilcoxP</th><th scope=col>chiP</th><th scope=col>t_test_p</th><th scope=col>targetMean</th><th scope=col>randomMean</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>abd-A      </td><td>2877</td><td>0.1841270</td><td> 1.619032e-77</td><td> 1.289780e-65</td><td> 1.048386e-75</td><td> 0.046724300</td><td>-0.054141999</td></tr>\n",
       "\t<tr><td>ac         </td><td> 799</td><td>0.2061522</td><td> 7.235478e-15</td><td> 1.076058e-13</td><td> 2.997855e-17</td><td> 0.107429744</td><td> 0.004356283</td></tr>\n",
       "\t<tr><td>Asciz      </td><td>1015</td><td>0.2666737</td><td> 1.381913e-58</td><td> 1.012269e-47</td><td> 6.875121e-57</td><td>-0.059485335</td><td> 0.139650653</td></tr>\n",
       "\t<tr><td>ATbp       </td><td>1094</td><td>0.2460150</td><td> 1.277078e-52</td><td> 7.181448e-44</td><td> 1.770003e-49</td><td>-0.069432889</td><td> 0.120991008</td></tr>\n",
       "\t<tr><td>bcd        </td><td> 966</td><td>0.1976081</td><td> 2.120398e-25</td><td> 4.919029e-25</td><td> 2.333386e-16</td><td> 0.016560839</td><td> 0.135763915</td></tr>\n",
       "\t<tr><td>BtbVII     </td><td> 620</td><td>0.2092590</td><td> 6.663234e-19</td><td> 4.253111e-12</td><td> 3.976728e-18</td><td>-0.047787715</td><td> 0.124351316</td></tr>\n",
       "\t<tr><td>CG10209    </td><td>1056</td><td>0.1938530</td><td> 1.188169e-33</td><td> 8.973868e-25</td><td> 1.125802e-29</td><td>-0.014933450</td><td> 0.131503296</td></tr>\n",
       "\t<tr><td>CG10274    </td><td> 298</td><td>0.1940550</td><td> 1.103532e-07</td><td> 1.652799e-03</td><td> 1.172761e-07</td><td> 0.006879823</td><td> 0.136545159</td></tr>\n",
       "\t<tr><td>CG10949    </td><td>2101</td><td>0.1910954</td><td> 1.981784e-37</td><td> 2.235072e-55</td><td> 2.206041e-41</td><td> 0.282186925</td><td> 0.138537611</td></tr>\n",
       "\t<tr><td>CG10959    </td><td> 823</td><td>0.1889121</td><td> 2.234092e-22</td><td> 2.180663e-15</td><td> 2.365133e-22</td><td>-0.005531773</td><td> 0.127055950</td></tr>\n",
       "\t<tr><td>CG11723    </td><td> 410</td><td>0.2789488</td><td> 9.729101e-14</td><td> 1.918151e-16</td><td> 1.846671e-10</td><td>-0.002630109</td><td> 0.166492063</td></tr>\n",
       "\t<tr><td>CG11966    </td><td>1448</td><td>0.2014731</td><td> 8.135759e-46</td><td> 6.757597e-42</td><td> 7.780729e-35</td><td>-0.142548642</td><td>-0.036389337</td></tr>\n",
       "\t<tr><td>CG12219    </td><td> 861</td><td>0.2691150</td><td> 3.135447e-48</td><td> 2.924184e-40</td><td> 3.236165e-47</td><td>-0.053150829</td><td> 0.163211143</td></tr>\n",
       "\t<tr><td>CG12299    </td><td> 313</td><td>0.1930630</td><td> 5.479753e-06</td><td> 8.728920e-05</td><td> 2.260266e-04</td><td> 0.024638065</td><td> 0.134959828</td></tr>\n",
       "\t<tr><td>CG14667    </td><td> 701</td><td>0.1858104</td><td> 3.732674e-19</td><td> 1.759347e-14</td><td> 2.125609e-16</td><td> 0.000103919</td><td> 0.120560790</td></tr>\n",
       "\t<tr><td>CG1529     </td><td> 754</td><td>0.2276228</td><td> 4.285129e-28</td><td> 1.960199e-23</td><td> 8.142205e-31</td><td>-0.045024506</td><td> 0.132851213</td></tr>\n",
       "\t<tr><td>CG15436    </td><td>1657</td><td>0.2285434</td><td> 2.243861e-47</td><td> 2.063290e-63</td><td> 6.950563e-53</td><td> 0.314150233</td><td> 0.130652885</td></tr>\n",
       "\t<tr><td>CG17186    </td><td> 866</td><td>0.3019150</td><td> 2.901028e-43</td><td> 4.255408e-54</td><td> 2.481502e-50</td><td> 0.388168429</td><td> 0.132508860</td></tr>\n",
       "\t<tr><td>CG2116     </td><td>2076</td><td>0.2375196</td><td> 6.239468e-58</td><td> 1.079665e-89</td><td> 2.787552e-66</td><td> 0.364874401</td><td> 0.159416953</td></tr>\n",
       "\t<tr><td>CG2129     </td><td> 911</td><td>0.2909086</td><td> 6.823515e-37</td><td> 1.014683e-52</td><td> 5.201714e-41</td><td> 0.349181462</td><td> 0.136017447</td></tr>\n",
       "\t<tr><td>CG30020    </td><td>2107</td><td>0.1988688</td><td> 6.883364e-46</td><td> 2.412924e-58</td><td> 7.979029e-48</td><td> 0.264267970</td><td> 0.094139720</td></tr>\n",
       "\t<tr><td>CG31365    </td><td> 572</td><td>0.1953316</td><td> 3.581825e-13</td><td> 7.796664e-09</td><td> 2.123240e-12</td><td>-0.042095954</td><td> 0.097634141</td></tr>\n",
       "\t<tr><td>CG3163     </td><td>1166</td><td>0.1900361</td><td> 3.582387e-23</td><td> 1.233733e-24</td><td> 2.356179e-19</td><td> 0.003897010</td><td> 0.095635076</td></tr>\n",
       "\t<tr><td>CG3407     </td><td> 714</td><td>0.2066034</td><td> 9.758215e-21</td><td> 1.075283e-17</td><td> 2.403734e-22</td><td>-0.061084982</td><td> 0.086095655</td></tr>\n",
       "\t<tr><td>CG3995     </td><td>1222</td><td>0.1876980</td><td> 5.994396e-17</td><td> 7.523293e-25</td><td> 1.337816e-19</td><td> 0.257616012</td><td> 0.128783560</td></tr>\n",
       "\t<tr><td>CG4404     </td><td> 850</td><td>0.2012464</td><td> 4.886377e-27</td><td> 2.408808e-21</td><td> 1.203296e-19</td><td>-0.003248097</td><td> 0.135493897</td></tr>\n",
       "\t<tr><td>CG4617     </td><td>1635</td><td>0.3281141</td><td>5.827115e-102</td><td>4.658995e-129</td><td>5.676966e-114</td><td> 0.410943420</td><td> 0.140817924</td></tr>\n",
       "\t<tr><td>CG4707     </td><td> 775</td><td>0.2457735</td><td> 1.867235e-36</td><td> 4.222885e-26</td><td> 9.408617e-35</td><td>-0.054662286</td><td> 0.128831051</td></tr>\n",
       "\t<tr><td>CG4936     </td><td>3157</td><td>0.1997852</td><td> 6.037059e-57</td><td> 3.603209e-90</td><td> 3.954210e-63</td><td> 0.298630783</td><td> 0.135342182</td></tr>\n",
       "\t<tr><td>CG5180     </td><td>1260</td><td>0.2559252</td><td> 3.131889e-68</td><td> 1.222765e-58</td><td> 1.038801e-59</td><td>-0.051976675</td><td> 0.154004358</td></tr>\n",
       "\t<tr><td>CG6254     </td><td> 523</td><td>0.1980461</td><td> 4.685226e-12</td><td> 4.697466e-10</td><td> 6.518909e-12</td><td>-0.005881226</td><td> 0.115780457</td></tr>\n",
       "\t<tr><td>CG6654     </td><td> 247</td><td>0.1920782</td><td> 1.204304e-06</td><td> 1.896209e-03</td><td> 1.905793e-07</td><td>-0.038850057</td><td> 0.098295459</td></tr>\n",
       "\t<tr><td>CG7745     </td><td> 892</td><td>0.2116448</td><td> 4.153458e-32</td><td> 4.280538e-24</td><td> 2.106722e-29</td><td>-0.005219479</td><td> 0.142559974</td></tr>\n",
       "\t<tr><td>CG8281     </td><td> 992</td><td>0.1816777</td><td> 2.078335e-26</td><td> 1.436929e-18</td><td> 8.086626e-28</td><td>-0.037952966</td><td> 0.079578555</td></tr>\n",
       "\t<tr><td>CG8478     </td><td>1526</td><td>0.1942710</td><td> 1.147688e-35</td><td> 1.036276e-38</td><td> 1.018631e-37</td><td> 0.314816307</td><td> 0.128128927</td></tr>\n",
       "\t<tr><td>CG8765     </td><td> 524</td><td>0.1955157</td><td> 8.319339e-11</td><td> 4.298056e-09</td><td> 1.042086e-08</td><td> 0.015232145</td><td> 0.126163685</td></tr>\n",
       "\t<tr><td>CG8944     </td><td> 487</td><td>0.1861204</td><td> 2.241822e-07</td><td> 9.052383e-09</td><td> 1.620600e-04</td><td> 0.042045603</td><td> 0.127275647</td></tr>\n",
       "\t<tr><td>CHES-1-like</td><td> 711</td><td>0.2391004</td><td> 2.288620e-15</td><td> 6.827325e-24</td><td> 5.364112e-12</td><td>-0.188372659</td><td>-0.090068067</td></tr>\n",
       "\t<tr><td>CTPsyn     </td><td> 312</td><td>0.2318203</td><td> 2.742279e-12</td><td> 8.330241e-07</td><td> 8.920054e-13</td><td>-0.085334266</td><td> 0.102692781</td></tr>\n",
       "\t<tr><td>D19B       </td><td> 807</td><td>0.2215437</td><td> 1.287444e-29</td><td> 1.743521e-29</td><td> 1.414499e-27</td><td>-0.053320479</td><td> 0.108614210</td></tr>\n",
       "\t<tr><td>danr       </td><td> 382</td><td>0.1921022</td><td> 5.268404e-08</td><td> 1.272998e-06</td><td> 6.421957e-05</td><td>-0.030648457</td><td> 0.042900721</td></tr>\n",
       "\t<tr><td>Dip2       </td><td> 212</td><td>0.2056627</td><td> 7.136490e-06</td><td> 5.301688e-03</td><td> 3.883121e-06</td><td>-0.022129256</td><td> 0.116468751</td></tr>\n",
       "\t<tr><td>E(var)3-9  </td><td> 344</td><td>0.1952326</td><td> 2.010919e-07</td><td> 1.308109e-05</td><td> 3.135337e-05</td><td> 0.039284371</td><td> 0.152220458</td></tr>\n",
       "\t<tr><td>Ets97D     </td><td>1904</td><td>0.2078508</td><td> 2.487780e-53</td><td> 4.750789e-57</td><td> 1.987603e-57</td><td> 0.257307178</td><td> 0.103982722</td></tr>\n",
       "\t<tr><td>FoxK       </td><td>1414</td><td>0.1992621</td><td> 1.957893e-32</td><td> 2.806645e-37</td><td> 4.772897e-32</td><td> 0.216887384</td><td> 0.088927254</td></tr>\n",
       "\t<tr><td>GATAe      </td><td> 993</td><td>0.2358533</td><td> 3.672830e-37</td><td> 6.251177e-33</td><td> 1.406419e-41</td><td> 0.122367126</td><td>-0.017804265</td></tr>\n",
       "\t<tr><td>gfzf       </td><td>3128</td><td>0.2026743</td><td> 1.364488e-70</td><td> 1.589082e-92</td><td> 4.307257e-74</td><td> 0.276108130</td><td> 0.115810225</td></tr>\n",
       "\t<tr><td>hng2       </td><td> 243</td><td>0.2386692</td><td> 2.696054e-11</td><td> 2.054171e-09</td><td> 2.225340e-12</td><td>-0.053696666</td><td> 0.103576661</td></tr>\n",
       "\t<tr><td>jing       </td><td>1312</td><td>0.2064318</td><td> 3.404742e-41</td><td> 1.081720e-34</td><td> 6.534174e-46</td><td> 0.055137131</td><td>-0.072292912</td></tr>\n",
       "\t<tr><td>M1BP       </td><td>1896</td><td>0.2617553</td><td> 7.750881e-89</td><td> 1.938715e-95</td><td> 3.736714e-94</td><td> 0.337684218</td><td> 0.114418283</td></tr>\n",
       "\t<tr><td>Max        </td><td> 779</td><td>0.1909587</td><td> 2.063142e-11</td><td> 2.722002e-15</td><td> 5.607111e-11</td><td> 0.145708600</td><td> 0.078762031</td></tr>\n",
       "\t<tr><td>MBD-R2     </td><td>3159</td><td>0.2182654</td><td> 7.116922e-91</td><td>2.843666e-116</td><td> 9.544123e-97</td><td> 0.290387950</td><td> 0.111080281</td></tr>\n",
       "\t<tr><td>MESR4      </td><td>3631</td><td>0.2120284</td><td> 2.019840e-76</td><td>1.452051e-121</td><td> 1.374121e-81</td><td> 0.262356287</td><td> 0.094648848</td></tr>\n",
       "\t<tr><td>Nf-YA      </td><td> 749</td><td>0.1951598</td><td> 3.985163e-20</td><td> 5.037662e-18</td><td> 4.354238e-17</td><td>-0.001219586</td><td> 0.137559101</td></tr>\n",
       "\t<tr><td>org-1      </td><td> 572</td><td>0.2032877</td><td> 1.578571e-19</td><td> 1.021813e-14</td><td> 4.660215e-18</td><td> 0.032154298</td><td>-0.025203887</td></tr>\n",
       "\t<tr><td>peb        </td><td> 380</td><td>0.1841528</td><td> 1.631842e-07</td><td> 7.444145e-06</td><td> 5.264600e-08</td><td> 0.043673019</td><td>-0.026753805</td></tr>\n",
       "\t<tr><td>Spps       </td><td> 609</td><td>0.2142526</td><td> 5.817278e-18</td><td> 1.354387e-15</td><td> 4.859203e-14</td><td> 0.003398894</td><td> 0.151296263</td></tr>\n",
       "\t<tr><td>Sry-delta  </td><td>3202</td><td>0.1912862</td><td> 1.861836e-37</td><td> 1.987275e-86</td><td> 2.312160e-41</td><td> 0.242036210</td><td> 0.118336356</td></tr>\n",
       "\t<tr><td>stc        </td><td>1256</td><td>0.2105694</td><td> 1.777988e-39</td><td> 2.369474e-36</td><td> 2.164370e-33</td><td>-0.025897792</td><td> 0.144356954</td></tr>\n",
       "\t<tr><td>TFAM       </td><td> 119</td><td>0.3389546</td><td> 8.579793e-10</td><td> 4.963642e-05</td><td> 2.588923e-09</td><td>-0.087369909</td><td> 0.146208847</td></tr>\n",
       "\t<tr><td>trem       </td><td>2391</td><td>0.1864756</td><td> 3.407150e-52</td><td> 2.052565e-56</td><td> 8.775081e-57</td><td> 0.288086469</td><td> 0.134467274</td></tr>\n",
       "\t<tr><td>usp        </td><td>  75</td><td>0.4063670</td><td> 6.913466e-07</td><td> 3.167060e-03</td><td> 1.317686e-05</td><td>-0.071510691</td><td> 0.167494995</td></tr>\n",
       "\t<tr><td>wdn        </td><td> 574</td><td>0.1940326</td><td> 1.705572e-13</td><td> 1.269292e-09</td><td> 2.986834e-10</td><td> 0.001969243</td><td> 0.137094629</td></tr>\n",
       "\t<tr><td>wek        </td><td>1073</td><td>0.1813718</td><td> 3.122239e-12</td><td> 2.557064e-23</td><td> 4.831707e-09</td><td> 0.046114494</td><td> 0.132690137</td></tr>\n",
       "\t<tr><td>z          </td><td>1524</td><td>0.1956860</td><td> 8.325318e-44</td><td> 1.994418e-38</td><td> 5.166293e-41</td><td>-0.027386484</td><td> 0.137941197</td></tr>\n",
       "\t<tr><td>ZIPIC      </td><td> 311</td><td>0.2665339</td><td> 2.540120e-10</td><td> 2.961706e-11</td><td> 6.507265e-12</td><td> 0.340614145</td><td> 0.133778131</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 66 × 8\n",
       "\\begin{tabular}{llllllll}\n",
       " TF & nTargets & JSD & wilcoxP & chiP & t\\_test\\_p & targetMean & randomMean\\\\\n",
       " <chr> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t abd-A       & 2877 & 0.1841270 &  1.619032e-77 &  1.289780e-65 &  1.048386e-75 &  0.046724300 & -0.054141999\\\\\n",
       "\t ac          &  799 & 0.2061522 &  7.235478e-15 &  1.076058e-13 &  2.997855e-17 &  0.107429744 &  0.004356283\\\\\n",
       "\t Asciz       & 1015 & 0.2666737 &  1.381913e-58 &  1.012269e-47 &  6.875121e-57 & -0.059485335 &  0.139650653\\\\\n",
       "\t ATbp        & 1094 & 0.2460150 &  1.277078e-52 &  7.181448e-44 &  1.770003e-49 & -0.069432889 &  0.120991008\\\\\n",
       "\t bcd         &  966 & 0.1976081 &  2.120398e-25 &  4.919029e-25 &  2.333386e-16 &  0.016560839 &  0.135763915\\\\\n",
       "\t BtbVII      &  620 & 0.2092590 &  6.663234e-19 &  4.253111e-12 &  3.976728e-18 & -0.047787715 &  0.124351316\\\\\n",
       "\t CG10209     & 1056 & 0.1938530 &  1.188169e-33 &  8.973868e-25 &  1.125802e-29 & -0.014933450 &  0.131503296\\\\\n",
       "\t CG10274     &  298 & 0.1940550 &  1.103532e-07 &  1.652799e-03 &  1.172761e-07 &  0.006879823 &  0.136545159\\\\\n",
       "\t CG10949     & 2101 & 0.1910954 &  1.981784e-37 &  2.235072e-55 &  2.206041e-41 &  0.282186925 &  0.138537611\\\\\n",
       "\t CG10959     &  823 & 0.1889121 &  2.234092e-22 &  2.180663e-15 &  2.365133e-22 & -0.005531773 &  0.127055950\\\\\n",
       "\t CG11723     &  410 & 0.2789488 &  9.729101e-14 &  1.918151e-16 &  1.846671e-10 & -0.002630109 &  0.166492063\\\\\n",
       "\t CG11966     & 1448 & 0.2014731 &  8.135759e-46 &  6.757597e-42 &  7.780729e-35 & -0.142548642 & -0.036389337\\\\\n",
       "\t CG12219     &  861 & 0.2691150 &  3.135447e-48 &  2.924184e-40 &  3.236165e-47 & -0.053150829 &  0.163211143\\\\\n",
       "\t CG12299     &  313 & 0.1930630 &  5.479753e-06 &  8.728920e-05 &  2.260266e-04 &  0.024638065 &  0.134959828\\\\\n",
       "\t CG14667     &  701 & 0.1858104 &  3.732674e-19 &  1.759347e-14 &  2.125609e-16 &  0.000103919 &  0.120560790\\\\\n",
       "\t CG1529      &  754 & 0.2276228 &  4.285129e-28 &  1.960199e-23 &  8.142205e-31 & -0.045024506 &  0.132851213\\\\\n",
       "\t CG15436     & 1657 & 0.2285434 &  2.243861e-47 &  2.063290e-63 &  6.950563e-53 &  0.314150233 &  0.130652885\\\\\n",
       "\t CG17186     &  866 & 0.3019150 &  2.901028e-43 &  4.255408e-54 &  2.481502e-50 &  0.388168429 &  0.132508860\\\\\n",
       "\t CG2116      & 2076 & 0.2375196 &  6.239468e-58 &  1.079665e-89 &  2.787552e-66 &  0.364874401 &  0.159416953\\\\\n",
       "\t CG2129      &  911 & 0.2909086 &  6.823515e-37 &  1.014683e-52 &  5.201714e-41 &  0.349181462 &  0.136017447\\\\\n",
       "\t CG30020     & 2107 & 0.1988688 &  6.883364e-46 &  2.412924e-58 &  7.979029e-48 &  0.264267970 &  0.094139720\\\\\n",
       "\t CG31365     &  572 & 0.1953316 &  3.581825e-13 &  7.796664e-09 &  2.123240e-12 & -0.042095954 &  0.097634141\\\\\n",
       "\t CG3163      & 1166 & 0.1900361 &  3.582387e-23 &  1.233733e-24 &  2.356179e-19 &  0.003897010 &  0.095635076\\\\\n",
       "\t CG3407      &  714 & 0.2066034 &  9.758215e-21 &  1.075283e-17 &  2.403734e-22 & -0.061084982 &  0.086095655\\\\\n",
       "\t CG3995      & 1222 & 0.1876980 &  5.994396e-17 &  7.523293e-25 &  1.337816e-19 &  0.257616012 &  0.128783560\\\\\n",
       "\t CG4404      &  850 & 0.2012464 &  4.886377e-27 &  2.408808e-21 &  1.203296e-19 & -0.003248097 &  0.135493897\\\\\n",
       "\t CG4617      & 1635 & 0.3281141 & 5.827115e-102 & 4.658995e-129 & 5.676966e-114 &  0.410943420 &  0.140817924\\\\\n",
       "\t CG4707      &  775 & 0.2457735 &  1.867235e-36 &  4.222885e-26 &  9.408617e-35 & -0.054662286 &  0.128831051\\\\\n",
       "\t CG4936      & 3157 & 0.1997852 &  6.037059e-57 &  3.603209e-90 &  3.954210e-63 &  0.298630783 &  0.135342182\\\\\n",
       "\t CG5180      & 1260 & 0.2559252 &  3.131889e-68 &  1.222765e-58 &  1.038801e-59 & -0.051976675 &  0.154004358\\\\\n",
       "\t CG6254      &  523 & 0.1980461 &  4.685226e-12 &  4.697466e-10 &  6.518909e-12 & -0.005881226 &  0.115780457\\\\\n",
       "\t CG6654      &  247 & 0.1920782 &  1.204304e-06 &  1.896209e-03 &  1.905793e-07 & -0.038850057 &  0.098295459\\\\\n",
       "\t CG7745      &  892 & 0.2116448 &  4.153458e-32 &  4.280538e-24 &  2.106722e-29 & -0.005219479 &  0.142559974\\\\\n",
       "\t CG8281      &  992 & 0.1816777 &  2.078335e-26 &  1.436929e-18 &  8.086626e-28 & -0.037952966 &  0.079578555\\\\\n",
       "\t CG8478      & 1526 & 0.1942710 &  1.147688e-35 &  1.036276e-38 &  1.018631e-37 &  0.314816307 &  0.128128927\\\\\n",
       "\t CG8765      &  524 & 0.1955157 &  8.319339e-11 &  4.298056e-09 &  1.042086e-08 &  0.015232145 &  0.126163685\\\\\n",
       "\t CG8944      &  487 & 0.1861204 &  2.241822e-07 &  9.052383e-09 &  1.620600e-04 &  0.042045603 &  0.127275647\\\\\n",
       "\t CHES-1-like &  711 & 0.2391004 &  2.288620e-15 &  6.827325e-24 &  5.364112e-12 & -0.188372659 & -0.090068067\\\\\n",
       "\t CTPsyn      &  312 & 0.2318203 &  2.742279e-12 &  8.330241e-07 &  8.920054e-13 & -0.085334266 &  0.102692781\\\\\n",
       "\t D19B        &  807 & 0.2215437 &  1.287444e-29 &  1.743521e-29 &  1.414499e-27 & -0.053320479 &  0.108614210\\\\\n",
       "\t danr        &  382 & 0.1921022 &  5.268404e-08 &  1.272998e-06 &  6.421957e-05 & -0.030648457 &  0.042900721\\\\\n",
       "\t Dip2        &  212 & 0.2056627 &  7.136490e-06 &  5.301688e-03 &  3.883121e-06 & -0.022129256 &  0.116468751\\\\\n",
       "\t E(var)3-9   &  344 & 0.1952326 &  2.010919e-07 &  1.308109e-05 &  3.135337e-05 &  0.039284371 &  0.152220458\\\\\n",
       "\t Ets97D      & 1904 & 0.2078508 &  2.487780e-53 &  4.750789e-57 &  1.987603e-57 &  0.257307178 &  0.103982722\\\\\n",
       "\t FoxK        & 1414 & 0.1992621 &  1.957893e-32 &  2.806645e-37 &  4.772897e-32 &  0.216887384 &  0.088927254\\\\\n",
       "\t GATAe       &  993 & 0.2358533 &  3.672830e-37 &  6.251177e-33 &  1.406419e-41 &  0.122367126 & -0.017804265\\\\\n",
       "\t gfzf        & 3128 & 0.2026743 &  1.364488e-70 &  1.589082e-92 &  4.307257e-74 &  0.276108130 &  0.115810225\\\\\n",
       "\t hng2        &  243 & 0.2386692 &  2.696054e-11 &  2.054171e-09 &  2.225340e-12 & -0.053696666 &  0.103576661\\\\\n",
       "\t jing        & 1312 & 0.2064318 &  3.404742e-41 &  1.081720e-34 &  6.534174e-46 &  0.055137131 & -0.072292912\\\\\n",
       "\t M1BP        & 1896 & 0.2617553 &  7.750881e-89 &  1.938715e-95 &  3.736714e-94 &  0.337684218 &  0.114418283\\\\\n",
       "\t Max         &  779 & 0.1909587 &  2.063142e-11 &  2.722002e-15 &  5.607111e-11 &  0.145708600 &  0.078762031\\\\\n",
       "\t MBD-R2      & 3159 & 0.2182654 &  7.116922e-91 & 2.843666e-116 &  9.544123e-97 &  0.290387950 &  0.111080281\\\\\n",
       "\t MESR4       & 3631 & 0.2120284 &  2.019840e-76 & 1.452051e-121 &  1.374121e-81 &  0.262356287 &  0.094648848\\\\\n",
       "\t Nf-YA       &  749 & 0.1951598 &  3.985163e-20 &  5.037662e-18 &  4.354238e-17 & -0.001219586 &  0.137559101\\\\\n",
       "\t org-1       &  572 & 0.2032877 &  1.578571e-19 &  1.021813e-14 &  4.660215e-18 &  0.032154298 & -0.025203887\\\\\n",
       "\t peb         &  380 & 0.1841528 &  1.631842e-07 &  7.444145e-06 &  5.264600e-08 &  0.043673019 & -0.026753805\\\\\n",
       "\t Spps        &  609 & 0.2142526 &  5.817278e-18 &  1.354387e-15 &  4.859203e-14 &  0.003398894 &  0.151296263\\\\\n",
       "\t Sry-delta   & 3202 & 0.1912862 &  1.861836e-37 &  1.987275e-86 &  2.312160e-41 &  0.242036210 &  0.118336356\\\\\n",
       "\t stc         & 1256 & 0.2105694 &  1.777988e-39 &  2.369474e-36 &  2.164370e-33 & -0.025897792 &  0.144356954\\\\\n",
       "\t TFAM        &  119 & 0.3389546 &  8.579793e-10 &  4.963642e-05 &  2.588923e-09 & -0.087369909 &  0.146208847\\\\\n",
       "\t trem        & 2391 & 0.1864756 &  3.407150e-52 &  2.052565e-56 &  8.775081e-57 &  0.288086469 &  0.134467274\\\\\n",
       "\t usp         &   75 & 0.4063670 &  6.913466e-07 &  3.167060e-03 &  1.317686e-05 & -0.071510691 &  0.167494995\\\\\n",
       "\t wdn         &  574 & 0.1940326 &  1.705572e-13 &  1.269292e-09 &  2.986834e-10 &  0.001969243 &  0.137094629\\\\\n",
       "\t wek         & 1073 & 0.1813718 &  3.122239e-12 &  2.557064e-23 &  4.831707e-09 &  0.046114494 &  0.132690137\\\\\n",
       "\t z           & 1524 & 0.1956860 &  8.325318e-44 &  1.994418e-38 &  5.166293e-41 & -0.027386484 &  0.137941197\\\\\n",
       "\t ZIPIC       &  311 & 0.2665339 &  2.540120e-10 &  2.961706e-11 &  6.507265e-12 &  0.340614145 &  0.133778131\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 66 × 8\n",
       "\n",
       "| TF &lt;chr&gt; | nTargets &lt;int&gt; | JSD &lt;dbl&gt; | wilcoxP &lt;dbl&gt; | chiP &lt;dbl&gt; | t_test_p &lt;dbl&gt; | targetMean &lt;dbl&gt; | randomMean &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| abd-A       | 2877 | 0.1841270 |  1.619032e-77 |  1.289780e-65 |  1.048386e-75 |  0.046724300 | -0.054141999 |\n",
       "| ac          |  799 | 0.2061522 |  7.235478e-15 |  1.076058e-13 |  2.997855e-17 |  0.107429744 |  0.004356283 |\n",
       "| Asciz       | 1015 | 0.2666737 |  1.381913e-58 |  1.012269e-47 |  6.875121e-57 | -0.059485335 |  0.139650653 |\n",
       "| ATbp        | 1094 | 0.2460150 |  1.277078e-52 |  7.181448e-44 |  1.770003e-49 | -0.069432889 |  0.120991008 |\n",
       "| bcd         |  966 | 0.1976081 |  2.120398e-25 |  4.919029e-25 |  2.333386e-16 |  0.016560839 |  0.135763915 |\n",
       "| BtbVII      |  620 | 0.2092590 |  6.663234e-19 |  4.253111e-12 |  3.976728e-18 | -0.047787715 |  0.124351316 |\n",
       "| CG10209     | 1056 | 0.1938530 |  1.188169e-33 |  8.973868e-25 |  1.125802e-29 | -0.014933450 |  0.131503296 |\n",
       "| CG10274     |  298 | 0.1940550 |  1.103532e-07 |  1.652799e-03 |  1.172761e-07 |  0.006879823 |  0.136545159 |\n",
       "| CG10949     | 2101 | 0.1910954 |  1.981784e-37 |  2.235072e-55 |  2.206041e-41 |  0.282186925 |  0.138537611 |\n",
       "| CG10959     |  823 | 0.1889121 |  2.234092e-22 |  2.180663e-15 |  2.365133e-22 | -0.005531773 |  0.127055950 |\n",
       "| CG11723     |  410 | 0.2789488 |  9.729101e-14 |  1.918151e-16 |  1.846671e-10 | -0.002630109 |  0.166492063 |\n",
       "| CG11966     | 1448 | 0.2014731 |  8.135759e-46 |  6.757597e-42 |  7.780729e-35 | -0.142548642 | -0.036389337 |\n",
       "| CG12219     |  861 | 0.2691150 |  3.135447e-48 |  2.924184e-40 |  3.236165e-47 | -0.053150829 |  0.163211143 |\n",
       "| CG12299     |  313 | 0.1930630 |  5.479753e-06 |  8.728920e-05 |  2.260266e-04 |  0.024638065 |  0.134959828 |\n",
       "| CG14667     |  701 | 0.1858104 |  3.732674e-19 |  1.759347e-14 |  2.125609e-16 |  0.000103919 |  0.120560790 |\n",
       "| CG1529      |  754 | 0.2276228 |  4.285129e-28 |  1.960199e-23 |  8.142205e-31 | -0.045024506 |  0.132851213 |\n",
       "| CG15436     | 1657 | 0.2285434 |  2.243861e-47 |  2.063290e-63 |  6.950563e-53 |  0.314150233 |  0.130652885 |\n",
       "| CG17186     |  866 | 0.3019150 |  2.901028e-43 |  4.255408e-54 |  2.481502e-50 |  0.388168429 |  0.132508860 |\n",
       "| CG2116      | 2076 | 0.2375196 |  6.239468e-58 |  1.079665e-89 |  2.787552e-66 |  0.364874401 |  0.159416953 |\n",
       "| CG2129      |  911 | 0.2909086 |  6.823515e-37 |  1.014683e-52 |  5.201714e-41 |  0.349181462 |  0.136017447 |\n",
       "| CG30020     | 2107 | 0.1988688 |  6.883364e-46 |  2.412924e-58 |  7.979029e-48 |  0.264267970 |  0.094139720 |\n",
       "| CG31365     |  572 | 0.1953316 |  3.581825e-13 |  7.796664e-09 |  2.123240e-12 | -0.042095954 |  0.097634141 |\n",
       "| CG3163      | 1166 | 0.1900361 |  3.582387e-23 |  1.233733e-24 |  2.356179e-19 |  0.003897010 |  0.095635076 |\n",
       "| CG3407      |  714 | 0.2066034 |  9.758215e-21 |  1.075283e-17 |  2.403734e-22 | -0.061084982 |  0.086095655 |\n",
       "| CG3995      | 1222 | 0.1876980 |  5.994396e-17 |  7.523293e-25 |  1.337816e-19 |  0.257616012 |  0.128783560 |\n",
       "| CG4404      |  850 | 0.2012464 |  4.886377e-27 |  2.408808e-21 |  1.203296e-19 | -0.003248097 |  0.135493897 |\n",
       "| CG4617      | 1635 | 0.3281141 | 5.827115e-102 | 4.658995e-129 | 5.676966e-114 |  0.410943420 |  0.140817924 |\n",
       "| CG4707      |  775 | 0.2457735 |  1.867235e-36 |  4.222885e-26 |  9.408617e-35 | -0.054662286 |  0.128831051 |\n",
       "| CG4936      | 3157 | 0.1997852 |  6.037059e-57 |  3.603209e-90 |  3.954210e-63 |  0.298630783 |  0.135342182 |\n",
       "| CG5180      | 1260 | 0.2559252 |  3.131889e-68 |  1.222765e-58 |  1.038801e-59 | -0.051976675 |  0.154004358 |\n",
       "| CG6254      |  523 | 0.1980461 |  4.685226e-12 |  4.697466e-10 |  6.518909e-12 | -0.005881226 |  0.115780457 |\n",
       "| CG6654      |  247 | 0.1920782 |  1.204304e-06 |  1.896209e-03 |  1.905793e-07 | -0.038850057 |  0.098295459 |\n",
       "| CG7745      |  892 | 0.2116448 |  4.153458e-32 |  4.280538e-24 |  2.106722e-29 | -0.005219479 |  0.142559974 |\n",
       "| CG8281      |  992 | 0.1816777 |  2.078335e-26 |  1.436929e-18 |  8.086626e-28 | -0.037952966 |  0.079578555 |\n",
       "| CG8478      | 1526 | 0.1942710 |  1.147688e-35 |  1.036276e-38 |  1.018631e-37 |  0.314816307 |  0.128128927 |\n",
       "| CG8765      |  524 | 0.1955157 |  8.319339e-11 |  4.298056e-09 |  1.042086e-08 |  0.015232145 |  0.126163685 |\n",
       "| CG8944      |  487 | 0.1861204 |  2.241822e-07 |  9.052383e-09 |  1.620600e-04 |  0.042045603 |  0.127275647 |\n",
       "| CHES-1-like |  711 | 0.2391004 |  2.288620e-15 |  6.827325e-24 |  5.364112e-12 | -0.188372659 | -0.090068067 |\n",
       "| CTPsyn      |  312 | 0.2318203 |  2.742279e-12 |  8.330241e-07 |  8.920054e-13 | -0.085334266 |  0.102692781 |\n",
       "| D19B        |  807 | 0.2215437 |  1.287444e-29 |  1.743521e-29 |  1.414499e-27 | -0.053320479 |  0.108614210 |\n",
       "| danr        |  382 | 0.1921022 |  5.268404e-08 |  1.272998e-06 |  6.421957e-05 | -0.030648457 |  0.042900721 |\n",
       "| Dip2        |  212 | 0.2056627 |  7.136490e-06 |  5.301688e-03 |  3.883121e-06 | -0.022129256 |  0.116468751 |\n",
       "| E(var)3-9   |  344 | 0.1952326 |  2.010919e-07 |  1.308109e-05 |  3.135337e-05 |  0.039284371 |  0.152220458 |\n",
       "| Ets97D      | 1904 | 0.2078508 |  2.487780e-53 |  4.750789e-57 |  1.987603e-57 |  0.257307178 |  0.103982722 |\n",
       "| FoxK        | 1414 | 0.1992621 |  1.957893e-32 |  2.806645e-37 |  4.772897e-32 |  0.216887384 |  0.088927254 |\n",
       "| GATAe       |  993 | 0.2358533 |  3.672830e-37 |  6.251177e-33 |  1.406419e-41 |  0.122367126 | -0.017804265 |\n",
       "| gfzf        | 3128 | 0.2026743 |  1.364488e-70 |  1.589082e-92 |  4.307257e-74 |  0.276108130 |  0.115810225 |\n",
       "| hng2        |  243 | 0.2386692 |  2.696054e-11 |  2.054171e-09 |  2.225340e-12 | -0.053696666 |  0.103576661 |\n",
       "| jing        | 1312 | 0.2064318 |  3.404742e-41 |  1.081720e-34 |  6.534174e-46 |  0.055137131 | -0.072292912 |\n",
       "| M1BP        | 1896 | 0.2617553 |  7.750881e-89 |  1.938715e-95 |  3.736714e-94 |  0.337684218 |  0.114418283 |\n",
       "| Max         |  779 | 0.1909587 |  2.063142e-11 |  2.722002e-15 |  5.607111e-11 |  0.145708600 |  0.078762031 |\n",
       "| MBD-R2      | 3159 | 0.2182654 |  7.116922e-91 | 2.843666e-116 |  9.544123e-97 |  0.290387950 |  0.111080281 |\n",
       "| MESR4       | 3631 | 0.2120284 |  2.019840e-76 | 1.452051e-121 |  1.374121e-81 |  0.262356287 |  0.094648848 |\n",
       "| Nf-YA       |  749 | 0.1951598 |  3.985163e-20 |  5.037662e-18 |  4.354238e-17 | -0.001219586 |  0.137559101 |\n",
       "| org-1       |  572 | 0.2032877 |  1.578571e-19 |  1.021813e-14 |  4.660215e-18 |  0.032154298 | -0.025203887 |\n",
       "| peb         |  380 | 0.1841528 |  1.631842e-07 |  7.444145e-06 |  5.264600e-08 |  0.043673019 | -0.026753805 |\n",
       "| Spps        |  609 | 0.2142526 |  5.817278e-18 |  1.354387e-15 |  4.859203e-14 |  0.003398894 |  0.151296263 |\n",
       "| Sry-delta   | 3202 | 0.1912862 |  1.861836e-37 |  1.987275e-86 |  2.312160e-41 |  0.242036210 |  0.118336356 |\n",
       "| stc         | 1256 | 0.2105694 |  1.777988e-39 |  2.369474e-36 |  2.164370e-33 | -0.025897792 |  0.144356954 |\n",
       "| TFAM        |  119 | 0.3389546 |  8.579793e-10 |  4.963642e-05 |  2.588923e-09 | -0.087369909 |  0.146208847 |\n",
       "| trem        | 2391 | 0.1864756 |  3.407150e-52 |  2.052565e-56 |  8.775081e-57 |  0.288086469 |  0.134467274 |\n",
       "| usp         |   75 | 0.4063670 |  6.913466e-07 |  3.167060e-03 |  1.317686e-05 | -0.071510691 |  0.167494995 |\n",
       "| wdn         |  574 | 0.1940326 |  1.705572e-13 |  1.269292e-09 |  2.986834e-10 |  0.001969243 |  0.137094629 |\n",
       "| wek         | 1073 | 0.1813718 |  3.122239e-12 |  2.557064e-23 |  4.831707e-09 |  0.046114494 |  0.132690137 |\n",
       "| z           | 1524 | 0.1956860 |  8.325318e-44 |  1.994418e-38 |  5.166293e-41 | -0.027386484 |  0.137941197 |\n",
       "| ZIPIC       |  311 | 0.2665339 |  2.540120e-10 |  2.961706e-11 |  6.507265e-12 |  0.340614145 |  0.133778131 |\n",
       "\n"
      ],
      "text/plain": [
       "   TF          nTargets JSD       wilcoxP       chiP          t_test_p     \n",
       "1  abd-A       2877     0.1841270  1.619032e-77  1.289780e-65  1.048386e-75\n",
       "2  ac           799     0.2061522  7.235478e-15  1.076058e-13  2.997855e-17\n",
       "3  Asciz       1015     0.2666737  1.381913e-58  1.012269e-47  6.875121e-57\n",
       "4  ATbp        1094     0.2460150  1.277078e-52  7.181448e-44  1.770003e-49\n",
       "5  bcd          966     0.1976081  2.120398e-25  4.919029e-25  2.333386e-16\n",
       "6  BtbVII       620     0.2092590  6.663234e-19  4.253111e-12  3.976728e-18\n",
       "7  CG10209     1056     0.1938530  1.188169e-33  8.973868e-25  1.125802e-29\n",
       "8  CG10274      298     0.1940550  1.103532e-07  1.652799e-03  1.172761e-07\n",
       "9  CG10949     2101     0.1910954  1.981784e-37  2.235072e-55  2.206041e-41\n",
       "10 CG10959      823     0.1889121  2.234092e-22  2.180663e-15  2.365133e-22\n",
       "11 CG11723      410     0.2789488  9.729101e-14  1.918151e-16  1.846671e-10\n",
       "12 CG11966     1448     0.2014731  8.135759e-46  6.757597e-42  7.780729e-35\n",
       "13 CG12219      861     0.2691150  3.135447e-48  2.924184e-40  3.236165e-47\n",
       "14 CG12299      313     0.1930630  5.479753e-06  8.728920e-05  2.260266e-04\n",
       "15 CG14667      701     0.1858104  3.732674e-19  1.759347e-14  2.125609e-16\n",
       "16 CG1529       754     0.2276228  4.285129e-28  1.960199e-23  8.142205e-31\n",
       "17 CG15436     1657     0.2285434  2.243861e-47  2.063290e-63  6.950563e-53\n",
       "18 CG17186      866     0.3019150  2.901028e-43  4.255408e-54  2.481502e-50\n",
       "19 CG2116      2076     0.2375196  6.239468e-58  1.079665e-89  2.787552e-66\n",
       "20 CG2129       911     0.2909086  6.823515e-37  1.014683e-52  5.201714e-41\n",
       "21 CG30020     2107     0.1988688  6.883364e-46  2.412924e-58  7.979029e-48\n",
       "22 CG31365      572     0.1953316  3.581825e-13  7.796664e-09  2.123240e-12\n",
       "23 CG3163      1166     0.1900361  3.582387e-23  1.233733e-24  2.356179e-19\n",
       "24 CG3407       714     0.2066034  9.758215e-21  1.075283e-17  2.403734e-22\n",
       "25 CG3995      1222     0.1876980  5.994396e-17  7.523293e-25  1.337816e-19\n",
       "26 CG4404       850     0.2012464  4.886377e-27  2.408808e-21  1.203296e-19\n",
       "27 CG4617      1635     0.3281141 5.827115e-102 4.658995e-129 5.676966e-114\n",
       "28 CG4707       775     0.2457735  1.867235e-36  4.222885e-26  9.408617e-35\n",
       "29 CG4936      3157     0.1997852  6.037059e-57  3.603209e-90  3.954210e-63\n",
       "30 CG5180      1260     0.2559252  3.131889e-68  1.222765e-58  1.038801e-59\n",
       "31 CG6254       523     0.1980461  4.685226e-12  4.697466e-10  6.518909e-12\n",
       "32 CG6654       247     0.1920782  1.204304e-06  1.896209e-03  1.905793e-07\n",
       "33 CG7745       892     0.2116448  4.153458e-32  4.280538e-24  2.106722e-29\n",
       "34 CG8281       992     0.1816777  2.078335e-26  1.436929e-18  8.086626e-28\n",
       "35 CG8478      1526     0.1942710  1.147688e-35  1.036276e-38  1.018631e-37\n",
       "36 CG8765       524     0.1955157  8.319339e-11  4.298056e-09  1.042086e-08\n",
       "37 CG8944       487     0.1861204  2.241822e-07  9.052383e-09  1.620600e-04\n",
       "38 CHES-1-like  711     0.2391004  2.288620e-15  6.827325e-24  5.364112e-12\n",
       "39 CTPsyn       312     0.2318203  2.742279e-12  8.330241e-07  8.920054e-13\n",
       "40 D19B         807     0.2215437  1.287444e-29  1.743521e-29  1.414499e-27\n",
       "41 danr         382     0.1921022  5.268404e-08  1.272998e-06  6.421957e-05\n",
       "42 Dip2         212     0.2056627  7.136490e-06  5.301688e-03  3.883121e-06\n",
       "43 E(var)3-9    344     0.1952326  2.010919e-07  1.308109e-05  3.135337e-05\n",
       "44 Ets97D      1904     0.2078508  2.487780e-53  4.750789e-57  1.987603e-57\n",
       "45 FoxK        1414     0.1992621  1.957893e-32  2.806645e-37  4.772897e-32\n",
       "46 GATAe        993     0.2358533  3.672830e-37  6.251177e-33  1.406419e-41\n",
       "47 gfzf        3128     0.2026743  1.364488e-70  1.589082e-92  4.307257e-74\n",
       "48 hng2         243     0.2386692  2.696054e-11  2.054171e-09  2.225340e-12\n",
       "49 jing        1312     0.2064318  3.404742e-41  1.081720e-34  6.534174e-46\n",
       "50 M1BP        1896     0.2617553  7.750881e-89  1.938715e-95  3.736714e-94\n",
       "51 Max          779     0.1909587  2.063142e-11  2.722002e-15  5.607111e-11\n",
       "52 MBD-R2      3159     0.2182654  7.116922e-91 2.843666e-116  9.544123e-97\n",
       "53 MESR4       3631     0.2120284  2.019840e-76 1.452051e-121  1.374121e-81\n",
       "54 Nf-YA        749     0.1951598  3.985163e-20  5.037662e-18  4.354238e-17\n",
       "55 org-1        572     0.2032877  1.578571e-19  1.021813e-14  4.660215e-18\n",
       "56 peb          380     0.1841528  1.631842e-07  7.444145e-06  5.264600e-08\n",
       "57 Spps         609     0.2142526  5.817278e-18  1.354387e-15  4.859203e-14\n",
       "58 Sry-delta   3202     0.1912862  1.861836e-37  1.987275e-86  2.312160e-41\n",
       "59 stc         1256     0.2105694  1.777988e-39  2.369474e-36  2.164370e-33\n",
       "60 TFAM         119     0.3389546  8.579793e-10  4.963642e-05  2.588923e-09\n",
       "61 trem        2391     0.1864756  3.407150e-52  2.052565e-56  8.775081e-57\n",
       "62 usp           75     0.4063670  6.913466e-07  3.167060e-03  1.317686e-05\n",
       "63 wdn          574     0.1940326  1.705572e-13  1.269292e-09  2.986834e-10\n",
       "64 wek         1073     0.1813718  3.122239e-12  2.557064e-23  4.831707e-09\n",
       "65 z           1524     0.1956860  8.325318e-44  1.994418e-38  5.166293e-41\n",
       "66 ZIPIC        311     0.2665339  2.540120e-10  2.961706e-11  6.507265e-12\n",
       "   targetMean   randomMean  \n",
       "1   0.046724300 -0.054141999\n",
       "2   0.107429744  0.004356283\n",
       "3  -0.059485335  0.139650653\n",
       "4  -0.069432889  0.120991008\n",
       "5   0.016560839  0.135763915\n",
       "6  -0.047787715  0.124351316\n",
       "7  -0.014933450  0.131503296\n",
       "8   0.006879823  0.136545159\n",
       "9   0.282186925  0.138537611\n",
       "10 -0.005531773  0.127055950\n",
       "11 -0.002630109  0.166492063\n",
       "12 -0.142548642 -0.036389337\n",
       "13 -0.053150829  0.163211143\n",
       "14  0.024638065  0.134959828\n",
       "15  0.000103919  0.120560790\n",
       "16 -0.045024506  0.132851213\n",
       "17  0.314150233  0.130652885\n",
       "18  0.388168429  0.132508860\n",
       "19  0.364874401  0.159416953\n",
       "20  0.349181462  0.136017447\n",
       "21  0.264267970  0.094139720\n",
       "22 -0.042095954  0.097634141\n",
       "23  0.003897010  0.095635076\n",
       "24 -0.061084982  0.086095655\n",
       "25  0.257616012  0.128783560\n",
       "26 -0.003248097  0.135493897\n",
       "27  0.410943420  0.140817924\n",
       "28 -0.054662286  0.128831051\n",
       "29  0.298630783  0.135342182\n",
       "30 -0.051976675  0.154004358\n",
       "31 -0.005881226  0.115780457\n",
       "32 -0.038850057  0.098295459\n",
       "33 -0.005219479  0.142559974\n",
       "34 -0.037952966  0.079578555\n",
       "35  0.314816307  0.128128927\n",
       "36  0.015232145  0.126163685\n",
       "37  0.042045603  0.127275647\n",
       "38 -0.188372659 -0.090068067\n",
       "39 -0.085334266  0.102692781\n",
       "40 -0.053320479  0.108614210\n",
       "41 -0.030648457  0.042900721\n",
       "42 -0.022129256  0.116468751\n",
       "43  0.039284371  0.152220458\n",
       "44  0.257307178  0.103982722\n",
       "45  0.216887384  0.088927254\n",
       "46  0.122367126 -0.017804265\n",
       "47  0.276108130  0.115810225\n",
       "48 -0.053696666  0.103576661\n",
       "49  0.055137131 -0.072292912\n",
       "50  0.337684218  0.114418283\n",
       "51  0.145708600  0.078762031\n",
       "52  0.290387950  0.111080281\n",
       "53  0.262356287  0.094648848\n",
       "54 -0.001219586  0.137559101\n",
       "55  0.032154298 -0.025203887\n",
       "56  0.043673019 -0.026753805\n",
       "57  0.003398894  0.151296263\n",
       "58  0.242036210  0.118336356\n",
       "59 -0.025897792  0.144356954\n",
       "60 -0.087369909  0.146208847\n",
       "61  0.288086469  0.134467274\n",
       "62 -0.071510691  0.167494995\n",
       "63  0.001969243  0.137094629\n",
       "64  0.046114494  0.132690137\n",
       "65 -0.027386484  0.137941197\n",
       "66  0.340614145  0.133778131"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fly_jsd_tf_target_trim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 661,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1mRows: \u001b[22m\u001b[34m699829\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m7\u001b[39m\n",
      "\u001b[36m──\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \"\\t\"\n",
      "\u001b[31mchr\u001b[39m (4): peakID, TF, Gene, lifeStage\n",
      "\u001b[32mdbl\u001b[39m (3): CosineAngle, nPeaks, targetDist\n",
      "\n",
      "\u001b[36mℹ\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36mℹ\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n"
     ]
    }
   ],
   "source": [
    "fly_targets_new_annot = read_tsv(\"fly_TF_TargetAngles_2_277_no_doubletons.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 662,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1mRows: \u001b[22m\u001b[34m671260\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m3\u001b[39m\n",
      "\u001b[36m──\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \"\\t\"\n",
      "\u001b[31mchr\u001b[39m (2): TF, Gene\n",
      "\u001b[32mdbl\u001b[39m (1): CosineAngle\n",
      "\n",
      "\u001b[36mℹ\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36mℹ\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n"
     ]
    }
   ],
   "source": [
    "fly_random_new_annot = read_tsv(\"fly_TF_RandomAngles_2_277_no_doubletons.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 663,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>CosineAngle</th><th scope=col>peakID</th><th scope=col>TF</th><th scope=col>Gene</th><th scope=col>nPeaks</th><th scope=col>targetDist</th><th scope=col>lifeStage</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 0.14595045</td><td>ab_ab-GFP_embryonic_1chrX17176552:17176952 </td><td>ab</td><td>baz   </td><td>251</td><td>16746</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.38162215</td><td>ab_ab-GFP_embryonic_1chrX13994830:13995230 </td><td>ab</td><td>ben   </td><td>168</td><td>-1315</td><td>embryonic</td></tr>\n",
       "\t<tr><td> 0.08301382</td><td>ab_ab-GFP_embryonic_1chr2R10227892:10228292</td><td>ab</td><td>Hr3   </td><td>212</td><td> -135</td><td>embryonic</td></tr>\n",
       "\t<tr><td> 0.10863988</td><td>ab_ab-GFP_embryonic_1chr3L17949012:17949412</td><td>ab</td><td>Eip75B</td><td>175</td><td>12602</td><td>embryonic</td></tr>\n",
       "\t<tr><td> 0.15421326</td><td>ab_ab-GFP_embryonic_1chr3L12468026:12468426</td><td>ab</td><td>eyg   </td><td>262</td><td> -205</td><td>embryonic</td></tr>\n",
       "\t<tr><td> 0.38344353</td><td>ab_ab-GFP_embryonic_1chr2L18327114:18327514</td><td>ab</td><td>Fas3  </td><td>220</td><td> 7212</td><td>embryonic</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 7\n",
       "\\begin{tabular}{lllllll}\n",
       " CosineAngle & peakID & TF & Gene & nPeaks & targetDist & lifeStage\\\\\n",
       " <dbl> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t  0.14595045 & ab\\_ab-GFP\\_embryonic\\_1chrX17176552:17176952  & ab & baz    & 251 & 16746 & embryonic\\\\\n",
       "\t -0.38162215 & ab\\_ab-GFP\\_embryonic\\_1chrX13994830:13995230  & ab & ben    & 168 & -1315 & embryonic\\\\\n",
       "\t  0.08301382 & ab\\_ab-GFP\\_embryonic\\_1chr2R10227892:10228292 & ab & Hr3    & 212 &  -135 & embryonic\\\\\n",
       "\t  0.10863988 & ab\\_ab-GFP\\_embryonic\\_1chr3L17949012:17949412 & ab & Eip75B & 175 & 12602 & embryonic\\\\\n",
       "\t  0.15421326 & ab\\_ab-GFP\\_embryonic\\_1chr3L12468026:12468426 & ab & eyg    & 262 &  -205 & embryonic\\\\\n",
       "\t  0.38344353 & ab\\_ab-GFP\\_embryonic\\_1chr2L18327114:18327514 & ab & Fas3   & 220 &  7212 & embryonic\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 7\n",
       "\n",
       "| CosineAngle &lt;dbl&gt; | peakID &lt;chr&gt; | TF &lt;chr&gt; | Gene &lt;chr&gt; | nPeaks &lt;dbl&gt; | targetDist &lt;dbl&gt; | lifeStage &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "|  0.14595045 | ab_ab-GFP_embryonic_1chrX17176552:17176952  | ab | baz    | 251 | 16746 | embryonic |\n",
       "| -0.38162215 | ab_ab-GFP_embryonic_1chrX13994830:13995230  | ab | ben    | 168 | -1315 | embryonic |\n",
       "|  0.08301382 | ab_ab-GFP_embryonic_1chr2R10227892:10228292 | ab | Hr3    | 212 |  -135 | embryonic |\n",
       "|  0.10863988 | ab_ab-GFP_embryonic_1chr3L17949012:17949412 | ab | Eip75B | 175 | 12602 | embryonic |\n",
       "|  0.15421326 | ab_ab-GFP_embryonic_1chr3L12468026:12468426 | ab | eyg    | 262 |  -205 | embryonic |\n",
       "|  0.38344353 | ab_ab-GFP_embryonic_1chr2L18327114:18327514 | ab | Fas3   | 220 |  7212 | embryonic |\n",
       "\n"
      ],
      "text/plain": [
       "  CosineAngle peakID                                      TF Gene   nPeaks\n",
       "1  0.14595045 ab_ab-GFP_embryonic_1chrX17176552:17176952  ab baz    251   \n",
       "2 -0.38162215 ab_ab-GFP_embryonic_1chrX13994830:13995230  ab ben    168   \n",
       "3  0.08301382 ab_ab-GFP_embryonic_1chr2R10227892:10228292 ab Hr3    212   \n",
       "4  0.10863988 ab_ab-GFP_embryonic_1chr3L17949012:17949412 ab Eip75B 175   \n",
       "5  0.15421326 ab_ab-GFP_embryonic_1chr3L12468026:12468426 ab eyg    262   \n",
       "6  0.38344353 ab_ab-GFP_embryonic_1chr2L18327114:18327514 ab Fas3   220   \n",
       "  targetDist lifeStage\n",
       "1 16746      embryonic\n",
       "2 -1315      embryonic\n",
       "3  -135      embryonic\n",
       "4 12602      embryonic\n",
       "5  -205      embryonic\n",
       "6  7212      embryonic"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fly_targets_new_annot  %>%  head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 670,
   "metadata": {},
   "outputs": [],
   "source": [
    "fly_top_TFs = unique(fly_jsd_tf_target_trim$TF)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 671,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "66"
      ],
      "text/latex": [
       "66"
      ],
      "text/markdown": [
       "66"
      ],
      "text/plain": [
       "[1] 66"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fly_top_TFs   %>% length"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 672,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 660 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>CosineAngle</th><th scope=col>peakID</th><th scope=col>TF</th><th scope=col>Gene</th><th scope=col>nPeaks</th><th scope=col>targetDist</th><th scope=col>lifeStage</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>0.6189138</td><td>ATbp_ATbp-GFP_embryonic_1chr2L6028600:6029220        </td><td>ATbp   </td><td>CG9107      </td><td> 63</td><td> -4371</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6096739</td><td>ATbp_ATbp-GFP_embryonic_1chrX18839131:18839751       </td><td>ATbp   </td><td>htk         </td><td>105</td><td> -4222</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5729203</td><td>ATbp_ATbp-GFP_embryonic_1chr3L14780103:14780723      </td><td>ATbp   </td><td>mop         </td><td>143</td><td> -4284</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5657947</td><td>ATbp_ATbp-GFP_embryonic_1chr2L19126804:19127424      </td><td>ATbp   </td><td>l(2)37Cb    </td><td>230</td><td>  -138</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5621825</td><td>ATbp_ATbp-GFP_embryonic_1chrX14242925:14243545       </td><td>ATbp   </td><td>mRNA-cap    </td><td>256</td><td>  -413</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5530836</td><td>ATbp_ATbp-GFP_embryonic_1chr3R5746732:5746967        </td><td>ATbp   </td><td>Madm        </td><td> 48</td><td>  4324</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5304849</td><td>ATbp_ATbp-GFP_embryonic_1chr3R22530151:22530771      </td><td>ATbp   </td><td>CG33099     </td><td>215</td><td>   111</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5222559</td><td>ATbp_ATbp-GFP_embryonic_1chr3R21263270:21263491      </td><td>ATbp   </td><td>Rab1        </td><td>184</td><td>  5689</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5117363</td><td>ATbp_ATbp-GFP_embryonic_1chr3L21464358:21464622      </td><td>ATbp   </td><td>croc        </td><td>246</td><td> 11983</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.4935757</td><td>ATbp_ATbp-GFP_embryonic_1chr2R23681001:23681621      </td><td>ATbp   </td><td>St1         </td><td>123</td><td>  -312</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5660885</td><td>Asciz_CG14962-GFP_embryonic_1chr2L7401457:7402157    </td><td>Asciz  </td><td>CG5171      </td><td>274</td><td>   472</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5388026</td><td>Asciz_CG14962-GFP_embryonic_1chr3L21204554:21205254  </td><td>Asciz  </td><td>CG10512     </td><td>202</td><td>   117</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5359796</td><td>Asciz_CG14962-GFP_embryonic_1chr3L23095104:23095804  </td><td>Asciz  </td><td>RpL10       </td><td>211</td><td>    17</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5156593</td><td>Asciz_CG14962-GFP_embryonic_1chr3R23134565:23135265  </td><td>Asciz  </td><td>CG31457     </td><td> 56</td><td> -6934</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5150351</td><td>Asciz_CG14962-GFP_embryonic_1chr3L19932819:19933519  </td><td>Asciz  </td><td>CG46462     </td><td>184</td><td>  1478</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.4925116</td><td>Asciz_CG14962-GFP_embryonic_1chr3L16010047:16010747  </td><td>Asciz  </td><td>mib1        </td><td> 26</td><td>   -88</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.4853068</td><td>Asciz_CG14962-GFP_embryonic_1chr3R17067241:17067941  </td><td>Asciz  </td><td>Ns1         </td><td> 61</td><td>  5084</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.4462886</td><td>Asciz_CG14962-GFP_embryonic_1chr2L7386781:7387481    </td><td>Asciz  </td><td>CG5149      </td><td> 41</td><td>  1122</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.4323968</td><td>Asciz_CG14962-GFP_embryonic_1chr3L16749585:16750285  </td><td>Asciz  </td><td>CG42852     </td><td>113</td><td> -5554</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.4308037</td><td>Asciz_CG14962-GFP_embryonic_1chr2L8423636:8424336    </td><td>Asciz  </td><td>Akap200     </td><td>154</td><td> -2787</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6810581</td><td>BtbVII_BtbVII-GFP_embryonic_1chr3R8694118:8694458    </td><td>BtbVII </td><td>hb          </td><td>107</td><td>   303</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6055603</td><td>BtbVII_BtbVII-GFP_embryonic_1chr3L15347886:15348670  </td><td>BtbVII </td><td>Toll-6      </td><td>262</td><td> 11586</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5840373</td><td>BtbVII_BtbVII-GFP_embryonic_1chr3R24319821:24320605  </td><td>BtbVII </td><td>BRWD3       </td><td>134</td><td>  1213</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5788026</td><td>BtbVII_BtbVII-GFP_embryonic_1chr3L2825207:2825991    </td><td>BtbVII </td><td>Tet         </td><td>232</td><td> -4496</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5718206</td><td>BtbVII_BtbVII-GFP_embryonic_1chr2L8029494:8030278    </td><td>BtbVII </td><td>baf         </td><td>113</td><td>   424</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5630155</td><td>BtbVII_BtbVII-GFP_embryonic_1chr2L14474785:14475569  </td><td>BtbVII </td><td>noc         </td><td>221</td><td>-15682</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5621026</td><td>BtbVII_BtbVII-GFP_embryonic_1chr2L21218652:21219436  </td><td>BtbVII </td><td>CG8679      </td><td>232</td><td>  -500</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5499432</td><td>BtbVII_BtbVII-GFP_embryonic_1chr3R30800074:30800858  </td><td>BtbVII </td><td>wts         </td><td>158</td><td>  6153</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5480910</td><td>BtbVII_BtbVII-GFP_embryonic_1chrX13207891:13208675   </td><td>BtbVII </td><td>sno         </td><td>127</td><td>  3391</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5426162</td><td>BtbVII_BtbVII-GFP_embryonic_1chr3L8440510:8441294    </td><td>BtbVII </td><td>Unr         </td><td>225</td><td>   805</td><td>embryonic</td></tr>\n",
       "\t<tr><td>1.0000000</td><td>CG10209_CG10209-GFP_embryonic_1chr2R14848187:14848564</td><td>CG10209</td><td>CG10209     </td><td> 75</td><td>     4</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6515456</td><td>CG10209_CG10209-GFP_embryonic_1chr2L10987232:10987836</td><td>CG10209</td><td>piwi        </td><td>265</td><td>  -114</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6473721</td><td>CG10209_CG10209-GFP_embryonic_1chr3R19877336:19877940</td><td>CG10209</td><td>Arc42       </td><td>157</td><td>   -99</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5799880</td><td>CG10209_CG10209-GFP_embryonic_1chr3L5819163:5819767  </td><td>CG10209</td><td>mad2        </td><td>170</td><td> -6372</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5614826</td><td>CG10209_CG10209-GFP_embryonic_1chrX1380263:1380867   </td><td>CG10209</td><td>CG3719      </td><td>261</td><td>   -85</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5596084</td><td>CG10209_CG10209-GFP_embryonic_1chr3L6885588:6886192  </td><td>CG10209</td><td>Prat2       </td><td>116</td><td> 28074</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5569965</td><td>CG10209_CG10209-GFP_embryonic_1chr3R22530196:22530800</td><td>CG10209</td><td>CG33099     </td><td>215</td><td>    74</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5404655</td><td>CG10209_CG10209-GFP_embryonic_1chr3R21263107:21263711</td><td>CG10209</td><td>Rab1        </td><td>184</td><td>  5651</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5234731</td><td>CG10209_CG10209-GFP_embryonic_1chr3L14780114:14780718</td><td>CG10209</td><td>mop         </td><td>143</td><td> -4287</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5180713</td><td>CG10209_CG10209-GFP_embryonic_1chr2L11188976:11189580</td><td>CG10209</td><td>Boh2        </td><td>249</td><td> 10246</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6919174</td><td>CG10274_CG10274-GFP_embryonic_1chr2L1156453:1157017  </td><td>CG10274</td><td>capt        </td><td> 26</td><td>  -424</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6579180</td><td>CG10274_CG10274-GFP_embryonic_1chrX15786208:15786622 </td><td>CG10274</td><td>mRpS30      </td><td>112</td><td>   393</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6501383</td><td>CG10274_CG10274-GFP_embryonic_1chr2R7917372:7917711  </td><td>CG10274</td><td>Cul1        </td><td> 44</td><td>   367</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6393406</td><td>CG10274_CG10274-GFP_embryonic_1chr3R17020878:17021442</td><td>CG10274</td><td>AdSL        </td><td>275</td><td>   -40</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6164763</td><td>CG10274_CG10274-GFP_embryonic_1chr3R11392892:11393098</td><td>CG10274</td><td>mRpL40      </td><td>230</td><td>-14934</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6075494</td><td>CG10274_CG10274-GFP_embryonic_1chr3R11822127:11822691</td><td>CG10274</td><td>Kyat        </td><td>167</td><td>    11</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5341254</td><td>CG10274_CG10274-GFP_embryonic_1chrX14948750:14949314 </td><td>CG10274</td><td>pdgy        </td><td> 92</td><td>  1717</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5322367</td><td>CG10274_CG10274-GFP_embryonic_1chr2L8000486:8001029  </td><td>CG10274</td><td>Sirup       </td><td>264</td><td>  1881</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5177671</td><td>CG10274_CG10274-GFP_embryonic_1chr3L9712112:9712676  </td><td>CG10274</td><td>SH3PX1      </td><td> 42</td><td>  1433</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.4960898</td><td>CG10274_CG10274-GFP_embryonic_1chrX1922637:1923201   </td><td>CG10274</td><td>eIF2Bepsilon</td><td>160</td><td>   -45</td><td>embryonic</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>0.8207679</td><td>trem_trem_embryonic_1chrX2727348:2728008     </td><td>trem</td><td>dwg        </td><td>101</td><td>   -20</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.7929026</td><td>trem_trem_embryonic_1chr3R16639784:16640444  </td><td>trem</td><td>nonA-l     </td><td>151</td><td>   -91</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.7677257</td><td>trem_trem_embryonic_1chr3L22071911:22072571  </td><td>trem</td><td>CG7139     </td><td>248</td><td>  -393</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.7533282</td><td>trem_trem_embryonic_1chr3R24058076:24058397  </td><td>trem</td><td>SMC1       </td><td>226</td><td>   -62</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.7509802</td><td>trem_trem_embryonic_1chr2R13132922:13133256  </td><td>trem</td><td>Cap-G      </td><td>158</td><td>   -86</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.7496516</td><td>trem_trem_embryonic_1chr2R7962019:7962679    </td><td>trem</td><td>nito       </td><td>208</td><td>  -114</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.7379604</td><td>trem_trem_embryonic_1chr2R24158988:24159464  </td><td>trem</td><td>spag       </td><td>246</td><td>   -32</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.7352163</td><td>trem_trem_embryonic_1chrX3936212:3936872     </td><td>trem</td><td>Glos       </td><td>141</td><td>    25</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.7310491</td><td>trem_trem_embryonic_1chr3L9838351:9839011    </td><td>trem</td><td>Cdk8       </td><td>131</td><td>   -11</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.7279310</td><td>trem_trem_embryonic_1chr3L18901445:18901697  </td><td>trem</td><td>CG6843     </td><td>164</td><td>   -18</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6729249</td><td>usp_usp-GFP_embryonic_1chr3L21261221:21261305</td><td>usp </td><td>CG43218    </td><td> 80</td><td>  3243</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5878926</td><td>usp_usp-GFP_embryonic_1chr3L20405430:20405514</td><td>usp </td><td>trbl       </td><td> 23</td><td> -3863</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.5116316</td><td>usp_usp-GFP_embryonic_1chr3L15545117:15545201</td><td>usp </td><td>AGO2       </td><td>129</td><td> -8954</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.4485843</td><td>usp_usp-GFP_embryonic_1chr2R9548145:9548229  </td><td>usp </td><td>CG1888     </td><td>107</td><td>   -52</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.4352487</td><td>usp_usp-GFP_embryonic_1chr2R21707509:21707593</td><td>usp </td><td>HmgZ       </td><td> 28</td><td>   310</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.4196480</td><td>usp_usp-GFP_embryonic_1chrX647042:647126     </td><td>usp </td><td>Roc1a      </td><td>193</td><td>  -220</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.3443677</td><td>usp_usp-GFP_embryonic_1chr3R13790064:13790148</td><td>usp </td><td>Dip-B      </td><td> 27</td><td>  7341</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.3379946</td><td>usp_usp-GFP_embryonic_1chr3L645998:646082    </td><td>usp </td><td>Gale       </td><td> 44</td><td> -2515</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.2819830</td><td>usp_usp-GFP_embryonic_1chrX12012557:12012641 </td><td>usp </td><td>regucalcin </td><td>147</td><td>   222</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.2472901</td><td>usp_usp-GFP_embryonic_1chr3L3978406:3978490  </td><td>usp </td><td>scrt       </td><td> 96</td><td> -5448</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.7488630</td><td>wdn_wdn-GFP_embryonic_1chr2R21701778:21702082</td><td>wdn </td><td>HmgD       </td><td> 42</td><td>   472</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.7385835</td><td>wdn_wdn-GFP_embryonic_1chr2R22640149:22640453</td><td>wdn </td><td>CG3732     </td><td>207</td><td>  -100</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6968361</td><td>wdn_wdn-GFP_embryonic_1chr2L13366120:13366424</td><td>wdn </td><td>Nnp-1      </td><td>210</td><td>   -83</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6831170</td><td>wdn_wdn-GFP_embryonic_1chr3L16087389:16087693</td><td>wdn </td><td>Tasp1      </td><td>240</td><td>   -16</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6790940</td><td>wdn_wdn-GFP_embryonic_1chr3L4626924:4627228  </td><td>wdn </td><td>HDAC1      </td><td>101</td><td>   342</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6747096</td><td>wdn_wdn-GFP_embryonic_1chr2L18136614:18136918</td><td>wdn </td><td>kel        </td><td>149</td><td>  1317</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6718097</td><td>wdn_wdn-GFP_embryonic_1chr3R9713295:9713599  </td><td>wdn </td><td>dmt        </td><td>  3</td><td>  2305</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6576688</td><td>wdn_wdn-GFP_embryonic_1chr3L15616918:15617222</td><td>wdn </td><td>Phs        </td><td> 27</td><td> -2831</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6498783</td><td>wdn_wdn-GFP_embryonic_1chr3R24829897:24830201</td><td>wdn </td><td>Kaz        </td><td>180</td><td>   -90</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6364874</td><td>wdn_wdn-GFP_embryonic_1chr2R14410812:14411116</td><td>wdn </td><td>CG17385    </td><td>  2</td><td>   608</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.7198855</td><td>wek_wek-GFP_embryonic_1chr3L22065629:22066119</td><td>wek </td><td>CG14561    </td><td>218</td><td>   -33</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6980301</td><td>wek_wek-GFP_embryonic_1chr2R22806706:22807196</td><td>wek </td><td>CycB       </td><td>245</td><td>    -3</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6966879</td><td>wek_wek-GFP_embryonic_1chr2R9116356:9116846  </td><td>wek </td><td>Su(var)2-10</td><td>234</td><td>   -17</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6735059</td><td>wek_wek-GFP_embryonic_1chr2R21720467:21720957</td><td>wek </td><td>LBR        </td><td>153</td><td>   -22</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6538962</td><td>wek_wek-GFP_embryonic_1chr3R24753535:24753961</td><td>wek </td><td>asp        </td><td>155</td><td>   -77</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6421066</td><td>wek_wek-GFP_embryonic_1chr3R12985836:12986223</td><td>wek </td><td>CG31342    </td><td>210</td><td>    -6</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6376551</td><td>wek_wek-GFP_embryonic_1chr3R24870674:24871157</td><td>wek </td><td>CycB3      </td><td>174</td><td>  -104</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6364278</td><td>wek_wek-GFP_embryonic_1chr2R22637483:22637973</td><td>wek </td><td>Cdk9       </td><td>120</td><td>    -1</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6187072</td><td>wek_wek-GFP_embryonic_1chr3L8175765:8176255  </td><td>wek </td><td>RecQ4      </td><td> 92</td><td>   -32</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.6058364</td><td>wek_wek-GFP_embryonic_1chr2L21630459:21630949</td><td>wek </td><td>Df31       </td><td>  5</td><td>  -772</td><td>embryonic</td></tr>\n",
       "\t<tr><td>1.0000000</td><td>z_z-GFP_embryonic_1chrX2447768:2448519       </td><td>z   </td><td>z          </td><td>193</td><td>   276</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.8797171</td><td>z_z-GFP_embryonic_1chr2R11894739:11895135    </td><td>z   </td><td>eEF1alpha1 </td><td> 11</td><td>  1395</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.8579696</td><td>z_z-GFP_embryonic_1chr3R25032723:25033119    </td><td>z   </td><td>Smg6       </td><td> 82</td><td>   544</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.8510130</td><td>z_z-GFP_embryonic_1chr2L6049899:6050295      </td><td>z   </td><td>CG9135     </td><td>  7</td><td>  -703</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.8137604</td><td>z_z-GFP_embryonic_1chr3R17066775:17067171    </td><td>z   </td><td>Ns1        </td><td> 26</td><td>  5702</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.8109701</td><td>z_z-GFP_embryonic_1chr3R9503561:9503957      </td><td>z   </td><td>CG8149     </td><td> 52</td><td>  -897</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.8047936</td><td>z_z-GFP_embryonic_1chr2L1082625:1082980      </td><td>z   </td><td>Atg4a      </td><td> 28</td><td>  1913</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.7998836</td><td>z_z-GFP_embryonic_1chr2R6696542:6696938      </td><td>z   </td><td>Rab2       </td><td>270</td><td>     1</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.7972664</td><td>z_z-GFP_embryonic_1chrX763570:763966         </td><td>z   </td><td>RpL22      </td><td>  7</td><td>  1362</td><td>embryonic</td></tr>\n",
       "\t<tr><td>0.7848290</td><td>z_z-GFP_embryonic_1chr2R12904388:12904784    </td><td>z   </td><td>NAT1       </td><td>114</td><td>-11408</td><td>embryonic</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 660 × 7\n",
       "\\begin{tabular}{lllllll}\n",
       " CosineAngle & peakID & TF & Gene & nPeaks & targetDist & lifeStage\\\\\n",
       " <dbl> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t 0.6189138 & ATbp\\_ATbp-GFP\\_embryonic\\_1chr2L6028600:6029220         & ATbp    & CG9107       &  63 &  -4371 & embryonic\\\\\n",
       "\t 0.6096739 & ATbp\\_ATbp-GFP\\_embryonic\\_1chrX18839131:18839751        & ATbp    & htk          & 105 &  -4222 & embryonic\\\\\n",
       "\t 0.5729203 & ATbp\\_ATbp-GFP\\_embryonic\\_1chr3L14780103:14780723       & ATbp    & mop          & 143 &  -4284 & embryonic\\\\\n",
       "\t 0.5657947 & ATbp\\_ATbp-GFP\\_embryonic\\_1chr2L19126804:19127424       & ATbp    & l(2)37Cb     & 230 &   -138 & embryonic\\\\\n",
       "\t 0.5621825 & ATbp\\_ATbp-GFP\\_embryonic\\_1chrX14242925:14243545        & ATbp    & mRNA-cap     & 256 &   -413 & embryonic\\\\\n",
       "\t 0.5530836 & ATbp\\_ATbp-GFP\\_embryonic\\_1chr3R5746732:5746967         & ATbp    & Madm         &  48 &   4324 & embryonic\\\\\n",
       "\t 0.5304849 & ATbp\\_ATbp-GFP\\_embryonic\\_1chr3R22530151:22530771       & ATbp    & CG33099      & 215 &    111 & embryonic\\\\\n",
       "\t 0.5222559 & ATbp\\_ATbp-GFP\\_embryonic\\_1chr3R21263270:21263491       & ATbp    & Rab1         & 184 &   5689 & embryonic\\\\\n",
       "\t 0.5117363 & ATbp\\_ATbp-GFP\\_embryonic\\_1chr3L21464358:21464622       & ATbp    & croc         & 246 &  11983 & embryonic\\\\\n",
       "\t 0.4935757 & ATbp\\_ATbp-GFP\\_embryonic\\_1chr2R23681001:23681621       & ATbp    & St1          & 123 &   -312 & embryonic\\\\\n",
       "\t 0.5660885 & Asciz\\_CG14962-GFP\\_embryonic\\_1chr2L7401457:7402157     & Asciz   & CG5171       & 274 &    472 & embryonic\\\\\n",
       "\t 0.5388026 & Asciz\\_CG14962-GFP\\_embryonic\\_1chr3L21204554:21205254   & Asciz   & CG10512      & 202 &    117 & embryonic\\\\\n",
       "\t 0.5359796 & Asciz\\_CG14962-GFP\\_embryonic\\_1chr3L23095104:23095804   & Asciz   & RpL10        & 211 &     17 & embryonic\\\\\n",
       "\t 0.5156593 & Asciz\\_CG14962-GFP\\_embryonic\\_1chr3R23134565:23135265   & Asciz   & CG31457      &  56 &  -6934 & embryonic\\\\\n",
       "\t 0.5150351 & Asciz\\_CG14962-GFP\\_embryonic\\_1chr3L19932819:19933519   & Asciz   & CG46462      & 184 &   1478 & embryonic\\\\\n",
       "\t 0.4925116 & Asciz\\_CG14962-GFP\\_embryonic\\_1chr3L16010047:16010747   & Asciz   & mib1         &  26 &    -88 & embryonic\\\\\n",
       "\t 0.4853068 & Asciz\\_CG14962-GFP\\_embryonic\\_1chr3R17067241:17067941   & Asciz   & Ns1          &  61 &   5084 & embryonic\\\\\n",
       "\t 0.4462886 & Asciz\\_CG14962-GFP\\_embryonic\\_1chr2L7386781:7387481     & Asciz   & CG5149       &  41 &   1122 & embryonic\\\\\n",
       "\t 0.4323968 & Asciz\\_CG14962-GFP\\_embryonic\\_1chr3L16749585:16750285   & Asciz   & CG42852      & 113 &  -5554 & embryonic\\\\\n",
       "\t 0.4308037 & Asciz\\_CG14962-GFP\\_embryonic\\_1chr2L8423636:8424336     & Asciz   & Akap200      & 154 &  -2787 & embryonic\\\\\n",
       "\t 0.6810581 & BtbVII\\_BtbVII-GFP\\_embryonic\\_1chr3R8694118:8694458     & BtbVII  & hb           & 107 &    303 & embryonic\\\\\n",
       "\t 0.6055603 & BtbVII\\_BtbVII-GFP\\_embryonic\\_1chr3L15347886:15348670   & BtbVII  & Toll-6       & 262 &  11586 & embryonic\\\\\n",
       "\t 0.5840373 & BtbVII\\_BtbVII-GFP\\_embryonic\\_1chr3R24319821:24320605   & BtbVII  & BRWD3        & 134 &   1213 & embryonic\\\\\n",
       "\t 0.5788026 & BtbVII\\_BtbVII-GFP\\_embryonic\\_1chr3L2825207:2825991     & BtbVII  & Tet          & 232 &  -4496 & embryonic\\\\\n",
       "\t 0.5718206 & BtbVII\\_BtbVII-GFP\\_embryonic\\_1chr2L8029494:8030278     & BtbVII  & baf          & 113 &    424 & embryonic\\\\\n",
       "\t 0.5630155 & BtbVII\\_BtbVII-GFP\\_embryonic\\_1chr2L14474785:14475569   & BtbVII  & noc          & 221 & -15682 & embryonic\\\\\n",
       "\t 0.5621026 & BtbVII\\_BtbVII-GFP\\_embryonic\\_1chr2L21218652:21219436   & BtbVII  & CG8679       & 232 &   -500 & embryonic\\\\\n",
       "\t 0.5499432 & BtbVII\\_BtbVII-GFP\\_embryonic\\_1chr3R30800074:30800858   & BtbVII  & wts          & 158 &   6153 & embryonic\\\\\n",
       "\t 0.5480910 & BtbVII\\_BtbVII-GFP\\_embryonic\\_1chrX13207891:13208675    & BtbVII  & sno          & 127 &   3391 & embryonic\\\\\n",
       "\t 0.5426162 & BtbVII\\_BtbVII-GFP\\_embryonic\\_1chr3L8440510:8441294     & BtbVII  & Unr          & 225 &    805 & embryonic\\\\\n",
       "\t 1.0000000 & CG10209\\_CG10209-GFP\\_embryonic\\_1chr2R14848187:14848564 & CG10209 & CG10209      &  75 &      4 & embryonic\\\\\n",
       "\t 0.6515456 & CG10209\\_CG10209-GFP\\_embryonic\\_1chr2L10987232:10987836 & CG10209 & piwi         & 265 &   -114 & embryonic\\\\\n",
       "\t 0.6473721 & CG10209\\_CG10209-GFP\\_embryonic\\_1chr3R19877336:19877940 & CG10209 & Arc42        & 157 &    -99 & embryonic\\\\\n",
       "\t 0.5799880 & CG10209\\_CG10209-GFP\\_embryonic\\_1chr3L5819163:5819767   & CG10209 & mad2         & 170 &  -6372 & embryonic\\\\\n",
       "\t 0.5614826 & CG10209\\_CG10209-GFP\\_embryonic\\_1chrX1380263:1380867    & CG10209 & CG3719       & 261 &    -85 & embryonic\\\\\n",
       "\t 0.5596084 & CG10209\\_CG10209-GFP\\_embryonic\\_1chr3L6885588:6886192   & CG10209 & Prat2        & 116 &  28074 & embryonic\\\\\n",
       "\t 0.5569965 & CG10209\\_CG10209-GFP\\_embryonic\\_1chr3R22530196:22530800 & CG10209 & CG33099      & 215 &     74 & embryonic\\\\\n",
       "\t 0.5404655 & CG10209\\_CG10209-GFP\\_embryonic\\_1chr3R21263107:21263711 & CG10209 & Rab1         & 184 &   5651 & embryonic\\\\\n",
       "\t 0.5234731 & CG10209\\_CG10209-GFP\\_embryonic\\_1chr3L14780114:14780718 & CG10209 & mop          & 143 &  -4287 & embryonic\\\\\n",
       "\t 0.5180713 & CG10209\\_CG10209-GFP\\_embryonic\\_1chr2L11188976:11189580 & CG10209 & Boh2         & 249 &  10246 & embryonic\\\\\n",
       "\t 0.6919174 & CG10274\\_CG10274-GFP\\_embryonic\\_1chr2L1156453:1157017   & CG10274 & capt         &  26 &   -424 & embryonic\\\\\n",
       "\t 0.6579180 & CG10274\\_CG10274-GFP\\_embryonic\\_1chrX15786208:15786622  & CG10274 & mRpS30       & 112 &    393 & embryonic\\\\\n",
       "\t 0.6501383 & CG10274\\_CG10274-GFP\\_embryonic\\_1chr2R7917372:7917711   & CG10274 & Cul1         &  44 &    367 & embryonic\\\\\n",
       "\t 0.6393406 & CG10274\\_CG10274-GFP\\_embryonic\\_1chr3R17020878:17021442 & CG10274 & AdSL         & 275 &    -40 & embryonic\\\\\n",
       "\t 0.6164763 & CG10274\\_CG10274-GFP\\_embryonic\\_1chr3R11392892:11393098 & CG10274 & mRpL40       & 230 & -14934 & embryonic\\\\\n",
       "\t 0.6075494 & CG10274\\_CG10274-GFP\\_embryonic\\_1chr3R11822127:11822691 & CG10274 & Kyat         & 167 &     11 & embryonic\\\\\n",
       "\t 0.5341254 & CG10274\\_CG10274-GFP\\_embryonic\\_1chrX14948750:14949314  & CG10274 & pdgy         &  92 &   1717 & embryonic\\\\\n",
       "\t 0.5322367 & CG10274\\_CG10274-GFP\\_embryonic\\_1chr2L8000486:8001029   & CG10274 & Sirup        & 264 &   1881 & embryonic\\\\\n",
       "\t 0.5177671 & CG10274\\_CG10274-GFP\\_embryonic\\_1chr3L9712112:9712676   & CG10274 & SH3PX1       &  42 &   1433 & embryonic\\\\\n",
       "\t 0.4960898 & CG10274\\_CG10274-GFP\\_embryonic\\_1chrX1922637:1923201    & CG10274 & eIF2Bepsilon & 160 &    -45 & embryonic\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t 0.8207679 & trem\\_trem\\_embryonic\\_1chrX2727348:2728008      & trem & dwg         & 101 &    -20 & embryonic\\\\\n",
       "\t 0.7929026 & trem\\_trem\\_embryonic\\_1chr3R16639784:16640444   & trem & nonA-l      & 151 &    -91 & embryonic\\\\\n",
       "\t 0.7677257 & trem\\_trem\\_embryonic\\_1chr3L22071911:22072571   & trem & CG7139      & 248 &   -393 & embryonic\\\\\n",
       "\t 0.7533282 & trem\\_trem\\_embryonic\\_1chr3R24058076:24058397   & trem & SMC1        & 226 &    -62 & embryonic\\\\\n",
       "\t 0.7509802 & trem\\_trem\\_embryonic\\_1chr2R13132922:13133256   & trem & Cap-G       & 158 &    -86 & embryonic\\\\\n",
       "\t 0.7496516 & trem\\_trem\\_embryonic\\_1chr2R7962019:7962679     & trem & nito        & 208 &   -114 & embryonic\\\\\n",
       "\t 0.7379604 & trem\\_trem\\_embryonic\\_1chr2R24158988:24159464   & trem & spag        & 246 &    -32 & embryonic\\\\\n",
       "\t 0.7352163 & trem\\_trem\\_embryonic\\_1chrX3936212:3936872      & trem & Glos        & 141 &     25 & embryonic\\\\\n",
       "\t 0.7310491 & trem\\_trem\\_embryonic\\_1chr3L9838351:9839011     & trem & Cdk8        & 131 &    -11 & embryonic\\\\\n",
       "\t 0.7279310 & trem\\_trem\\_embryonic\\_1chr3L18901445:18901697   & trem & CG6843      & 164 &    -18 & embryonic\\\\\n",
       "\t 0.6729249 & usp\\_usp-GFP\\_embryonic\\_1chr3L21261221:21261305 & usp  & CG43218     &  80 &   3243 & embryonic\\\\\n",
       "\t 0.5878926 & usp\\_usp-GFP\\_embryonic\\_1chr3L20405430:20405514 & usp  & trbl        &  23 &  -3863 & embryonic\\\\\n",
       "\t 0.5116316 & usp\\_usp-GFP\\_embryonic\\_1chr3L15545117:15545201 & usp  & AGO2        & 129 &  -8954 & embryonic\\\\\n",
       "\t 0.4485843 & usp\\_usp-GFP\\_embryonic\\_1chr2R9548145:9548229   & usp  & CG1888      & 107 &    -52 & embryonic\\\\\n",
       "\t 0.4352487 & usp\\_usp-GFP\\_embryonic\\_1chr2R21707509:21707593 & usp  & HmgZ        &  28 &    310 & embryonic\\\\\n",
       "\t 0.4196480 & usp\\_usp-GFP\\_embryonic\\_1chrX647042:647126      & usp  & Roc1a       & 193 &   -220 & embryonic\\\\\n",
       "\t 0.3443677 & usp\\_usp-GFP\\_embryonic\\_1chr3R13790064:13790148 & usp  & Dip-B       &  27 &   7341 & embryonic\\\\\n",
       "\t 0.3379946 & usp\\_usp-GFP\\_embryonic\\_1chr3L645998:646082     & usp  & Gale        &  44 &  -2515 & embryonic\\\\\n",
       "\t 0.2819830 & usp\\_usp-GFP\\_embryonic\\_1chrX12012557:12012641  & usp  & regucalcin  & 147 &    222 & embryonic\\\\\n",
       "\t 0.2472901 & usp\\_usp-GFP\\_embryonic\\_1chr3L3978406:3978490   & usp  & scrt        &  96 &  -5448 & embryonic\\\\\n",
       "\t 0.7488630 & wdn\\_wdn-GFP\\_embryonic\\_1chr2R21701778:21702082 & wdn  & HmgD        &  42 &    472 & embryonic\\\\\n",
       "\t 0.7385835 & wdn\\_wdn-GFP\\_embryonic\\_1chr2R22640149:22640453 & wdn  & CG3732      & 207 &   -100 & embryonic\\\\\n",
       "\t 0.6968361 & wdn\\_wdn-GFP\\_embryonic\\_1chr2L13366120:13366424 & wdn  & Nnp-1       & 210 &    -83 & embryonic\\\\\n",
       "\t 0.6831170 & wdn\\_wdn-GFP\\_embryonic\\_1chr3L16087389:16087693 & wdn  & Tasp1       & 240 &    -16 & embryonic\\\\\n",
       "\t 0.6790940 & wdn\\_wdn-GFP\\_embryonic\\_1chr3L4626924:4627228   & wdn  & HDAC1       & 101 &    342 & embryonic\\\\\n",
       "\t 0.6747096 & wdn\\_wdn-GFP\\_embryonic\\_1chr2L18136614:18136918 & wdn  & kel         & 149 &   1317 & embryonic\\\\\n",
       "\t 0.6718097 & wdn\\_wdn-GFP\\_embryonic\\_1chr3R9713295:9713599   & wdn  & dmt         &   3 &   2305 & embryonic\\\\\n",
       "\t 0.6576688 & wdn\\_wdn-GFP\\_embryonic\\_1chr3L15616918:15617222 & wdn  & Phs         &  27 &  -2831 & embryonic\\\\\n",
       "\t 0.6498783 & wdn\\_wdn-GFP\\_embryonic\\_1chr3R24829897:24830201 & wdn  & Kaz         & 180 &    -90 & embryonic\\\\\n",
       "\t 0.6364874 & wdn\\_wdn-GFP\\_embryonic\\_1chr2R14410812:14411116 & wdn  & CG17385     &   2 &    608 & embryonic\\\\\n",
       "\t 0.7198855 & wek\\_wek-GFP\\_embryonic\\_1chr3L22065629:22066119 & wek  & CG14561     & 218 &    -33 & embryonic\\\\\n",
       "\t 0.6980301 & wek\\_wek-GFP\\_embryonic\\_1chr2R22806706:22807196 & wek  & CycB        & 245 &     -3 & embryonic\\\\\n",
       "\t 0.6966879 & wek\\_wek-GFP\\_embryonic\\_1chr2R9116356:9116846   & wek  & Su(var)2-10 & 234 &    -17 & embryonic\\\\\n",
       "\t 0.6735059 & wek\\_wek-GFP\\_embryonic\\_1chr2R21720467:21720957 & wek  & LBR         & 153 &    -22 & embryonic\\\\\n",
       "\t 0.6538962 & wek\\_wek-GFP\\_embryonic\\_1chr3R24753535:24753961 & wek  & asp         & 155 &    -77 & embryonic\\\\\n",
       "\t 0.6421066 & wek\\_wek-GFP\\_embryonic\\_1chr3R12985836:12986223 & wek  & CG31342     & 210 &     -6 & embryonic\\\\\n",
       "\t 0.6376551 & wek\\_wek-GFP\\_embryonic\\_1chr3R24870674:24871157 & wek  & CycB3       & 174 &   -104 & embryonic\\\\\n",
       "\t 0.6364278 & wek\\_wek-GFP\\_embryonic\\_1chr2R22637483:22637973 & wek  & Cdk9        & 120 &     -1 & embryonic\\\\\n",
       "\t 0.6187072 & wek\\_wek-GFP\\_embryonic\\_1chr3L8175765:8176255   & wek  & RecQ4       &  92 &    -32 & embryonic\\\\\n",
       "\t 0.6058364 & wek\\_wek-GFP\\_embryonic\\_1chr2L21630459:21630949 & wek  & Df31        &   5 &   -772 & embryonic\\\\\n",
       "\t 1.0000000 & z\\_z-GFP\\_embryonic\\_1chrX2447768:2448519        & z    & z           & 193 &    276 & embryonic\\\\\n",
       "\t 0.8797171 & z\\_z-GFP\\_embryonic\\_1chr2R11894739:11895135     & z    & eEF1alpha1  &  11 &   1395 & embryonic\\\\\n",
       "\t 0.8579696 & z\\_z-GFP\\_embryonic\\_1chr3R25032723:25033119     & z    & Smg6        &  82 &    544 & embryonic\\\\\n",
       "\t 0.8510130 & z\\_z-GFP\\_embryonic\\_1chr2L6049899:6050295       & z    & CG9135      &   7 &   -703 & embryonic\\\\\n",
       "\t 0.8137604 & z\\_z-GFP\\_embryonic\\_1chr3R17066775:17067171     & z    & Ns1         &  26 &   5702 & embryonic\\\\\n",
       "\t 0.8109701 & z\\_z-GFP\\_embryonic\\_1chr3R9503561:9503957       & z    & CG8149      &  52 &   -897 & embryonic\\\\\n",
       "\t 0.8047936 & z\\_z-GFP\\_embryonic\\_1chr2L1082625:1082980       & z    & Atg4a       &  28 &   1913 & embryonic\\\\\n",
       "\t 0.7998836 & z\\_z-GFP\\_embryonic\\_1chr2R6696542:6696938       & z    & Rab2        & 270 &      1 & embryonic\\\\\n",
       "\t 0.7972664 & z\\_z-GFP\\_embryonic\\_1chrX763570:763966          & z    & RpL22       &   7 &   1362 & embryonic\\\\\n",
       "\t 0.7848290 & z\\_z-GFP\\_embryonic\\_1chr2R12904388:12904784     & z    & NAT1        & 114 & -11408 & embryonic\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 660 × 7\n",
       "\n",
       "| CosineAngle &lt;dbl&gt; | peakID &lt;chr&gt; | TF &lt;chr&gt; | Gene &lt;chr&gt; | nPeaks &lt;dbl&gt; | targetDist &lt;dbl&gt; | lifeStage &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 0.6189138 | ATbp_ATbp-GFP_embryonic_1chr2L6028600:6029220         | ATbp    | CG9107       |  63 |  -4371 | embryonic |\n",
       "| 0.6096739 | ATbp_ATbp-GFP_embryonic_1chrX18839131:18839751        | ATbp    | htk          | 105 |  -4222 | embryonic |\n",
       "| 0.5729203 | ATbp_ATbp-GFP_embryonic_1chr3L14780103:14780723       | ATbp    | mop          | 143 |  -4284 | embryonic |\n",
       "| 0.5657947 | ATbp_ATbp-GFP_embryonic_1chr2L19126804:19127424       | ATbp    | l(2)37Cb     | 230 |   -138 | embryonic |\n",
       "| 0.5621825 | ATbp_ATbp-GFP_embryonic_1chrX14242925:14243545        | ATbp    | mRNA-cap     | 256 |   -413 | embryonic |\n",
       "| 0.5530836 | ATbp_ATbp-GFP_embryonic_1chr3R5746732:5746967         | ATbp    | Madm         |  48 |   4324 | embryonic |\n",
       "| 0.5304849 | ATbp_ATbp-GFP_embryonic_1chr3R22530151:22530771       | ATbp    | CG33099      | 215 |    111 | embryonic |\n",
       "| 0.5222559 | ATbp_ATbp-GFP_embryonic_1chr3R21263270:21263491       | ATbp    | Rab1         | 184 |   5689 | embryonic |\n",
       "| 0.5117363 | ATbp_ATbp-GFP_embryonic_1chr3L21464358:21464622       | ATbp    | croc         | 246 |  11983 | embryonic |\n",
       "| 0.4935757 | ATbp_ATbp-GFP_embryonic_1chr2R23681001:23681621       | ATbp    | St1          | 123 |   -312 | embryonic |\n",
       "| 0.5660885 | Asciz_CG14962-GFP_embryonic_1chr2L7401457:7402157     | Asciz   | CG5171       | 274 |    472 | embryonic |\n",
       "| 0.5388026 | Asciz_CG14962-GFP_embryonic_1chr3L21204554:21205254   | Asciz   | CG10512      | 202 |    117 | embryonic |\n",
       "| 0.5359796 | Asciz_CG14962-GFP_embryonic_1chr3L23095104:23095804   | Asciz   | RpL10        | 211 |     17 | embryonic |\n",
       "| 0.5156593 | Asciz_CG14962-GFP_embryonic_1chr3R23134565:23135265   | Asciz   | CG31457      |  56 |  -6934 | embryonic |\n",
       "| 0.5150351 | Asciz_CG14962-GFP_embryonic_1chr3L19932819:19933519   | Asciz   | CG46462      | 184 |   1478 | embryonic |\n",
       "| 0.4925116 | Asciz_CG14962-GFP_embryonic_1chr3L16010047:16010747   | Asciz   | mib1         |  26 |    -88 | embryonic |\n",
       "| 0.4853068 | Asciz_CG14962-GFP_embryonic_1chr3R17067241:17067941   | Asciz   | Ns1          |  61 |   5084 | embryonic |\n",
       "| 0.4462886 | Asciz_CG14962-GFP_embryonic_1chr2L7386781:7387481     | Asciz   | CG5149       |  41 |   1122 | embryonic |\n",
       "| 0.4323968 | Asciz_CG14962-GFP_embryonic_1chr3L16749585:16750285   | Asciz   | CG42852      | 113 |  -5554 | embryonic |\n",
       "| 0.4308037 | Asciz_CG14962-GFP_embryonic_1chr2L8423636:8424336     | Asciz   | Akap200      | 154 |  -2787 | embryonic |\n",
       "| 0.6810581 | BtbVII_BtbVII-GFP_embryonic_1chr3R8694118:8694458     | BtbVII  | hb           | 107 |    303 | embryonic |\n",
       "| 0.6055603 | BtbVII_BtbVII-GFP_embryonic_1chr3L15347886:15348670   | BtbVII  | Toll-6       | 262 |  11586 | embryonic |\n",
       "| 0.5840373 | BtbVII_BtbVII-GFP_embryonic_1chr3R24319821:24320605   | BtbVII  | BRWD3        | 134 |   1213 | embryonic |\n",
       "| 0.5788026 | BtbVII_BtbVII-GFP_embryonic_1chr3L2825207:2825991     | BtbVII  | Tet          | 232 |  -4496 | embryonic |\n",
       "| 0.5718206 | BtbVII_BtbVII-GFP_embryonic_1chr2L8029494:8030278     | BtbVII  | baf          | 113 |    424 | embryonic |\n",
       "| 0.5630155 | BtbVII_BtbVII-GFP_embryonic_1chr2L14474785:14475569   | BtbVII  | noc          | 221 | -15682 | embryonic |\n",
       "| 0.5621026 | BtbVII_BtbVII-GFP_embryonic_1chr2L21218652:21219436   | BtbVII  | CG8679       | 232 |   -500 | embryonic |\n",
       "| 0.5499432 | BtbVII_BtbVII-GFP_embryonic_1chr3R30800074:30800858   | BtbVII  | wts          | 158 |   6153 | embryonic |\n",
       "| 0.5480910 | BtbVII_BtbVII-GFP_embryonic_1chrX13207891:13208675    | BtbVII  | sno          | 127 |   3391 | embryonic |\n",
       "| 0.5426162 | BtbVII_BtbVII-GFP_embryonic_1chr3L8440510:8441294     | BtbVII  | Unr          | 225 |    805 | embryonic |\n",
       "| 1.0000000 | CG10209_CG10209-GFP_embryonic_1chr2R14848187:14848564 | CG10209 | CG10209      |  75 |      4 | embryonic |\n",
       "| 0.6515456 | CG10209_CG10209-GFP_embryonic_1chr2L10987232:10987836 | CG10209 | piwi         | 265 |   -114 | embryonic |\n",
       "| 0.6473721 | CG10209_CG10209-GFP_embryonic_1chr3R19877336:19877940 | CG10209 | Arc42        | 157 |    -99 | embryonic |\n",
       "| 0.5799880 | CG10209_CG10209-GFP_embryonic_1chr3L5819163:5819767   | CG10209 | mad2         | 170 |  -6372 | embryonic |\n",
       "| 0.5614826 | CG10209_CG10209-GFP_embryonic_1chrX1380263:1380867    | CG10209 | CG3719       | 261 |    -85 | embryonic |\n",
       "| 0.5596084 | CG10209_CG10209-GFP_embryonic_1chr3L6885588:6886192   | CG10209 | Prat2        | 116 |  28074 | embryonic |\n",
       "| 0.5569965 | CG10209_CG10209-GFP_embryonic_1chr3R22530196:22530800 | CG10209 | CG33099      | 215 |     74 | embryonic |\n",
       "| 0.5404655 | CG10209_CG10209-GFP_embryonic_1chr3R21263107:21263711 | CG10209 | Rab1         | 184 |   5651 | embryonic |\n",
       "| 0.5234731 | CG10209_CG10209-GFP_embryonic_1chr3L14780114:14780718 | CG10209 | mop          | 143 |  -4287 | embryonic |\n",
       "| 0.5180713 | CG10209_CG10209-GFP_embryonic_1chr2L11188976:11189580 | CG10209 | Boh2         | 249 |  10246 | embryonic |\n",
       "| 0.6919174 | CG10274_CG10274-GFP_embryonic_1chr2L1156453:1157017   | CG10274 | capt         |  26 |   -424 | embryonic |\n",
       "| 0.6579180 | CG10274_CG10274-GFP_embryonic_1chrX15786208:15786622  | CG10274 | mRpS30       | 112 |    393 | embryonic |\n",
       "| 0.6501383 | CG10274_CG10274-GFP_embryonic_1chr2R7917372:7917711   | CG10274 | Cul1         |  44 |    367 | embryonic |\n",
       "| 0.6393406 | CG10274_CG10274-GFP_embryonic_1chr3R17020878:17021442 | CG10274 | AdSL         | 275 |    -40 | embryonic |\n",
       "| 0.6164763 | CG10274_CG10274-GFP_embryonic_1chr3R11392892:11393098 | CG10274 | mRpL40       | 230 | -14934 | embryonic |\n",
       "| 0.6075494 | CG10274_CG10274-GFP_embryonic_1chr3R11822127:11822691 | CG10274 | Kyat         | 167 |     11 | embryonic |\n",
       "| 0.5341254 | CG10274_CG10274-GFP_embryonic_1chrX14948750:14949314  | CG10274 | pdgy         |  92 |   1717 | embryonic |\n",
       "| 0.5322367 | CG10274_CG10274-GFP_embryonic_1chr2L8000486:8001029   | CG10274 | Sirup        | 264 |   1881 | embryonic |\n",
       "| 0.5177671 | CG10274_CG10274-GFP_embryonic_1chr3L9712112:9712676   | CG10274 | SH3PX1       |  42 |   1433 | embryonic |\n",
       "| 0.4960898 | CG10274_CG10274-GFP_embryonic_1chrX1922637:1923201    | CG10274 | eIF2Bepsilon | 160 |    -45 | embryonic |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 0.8207679 | trem_trem_embryonic_1chrX2727348:2728008      | trem | dwg         | 101 |    -20 | embryonic |\n",
       "| 0.7929026 | trem_trem_embryonic_1chr3R16639784:16640444   | trem | nonA-l      | 151 |    -91 | embryonic |\n",
       "| 0.7677257 | trem_trem_embryonic_1chr3L22071911:22072571   | trem | CG7139      | 248 |   -393 | embryonic |\n",
       "| 0.7533282 | trem_trem_embryonic_1chr3R24058076:24058397   | trem | SMC1        | 226 |    -62 | embryonic |\n",
       "| 0.7509802 | trem_trem_embryonic_1chr2R13132922:13133256   | trem | Cap-G       | 158 |    -86 | embryonic |\n",
       "| 0.7496516 | trem_trem_embryonic_1chr2R7962019:7962679     | trem | nito        | 208 |   -114 | embryonic |\n",
       "| 0.7379604 | trem_trem_embryonic_1chr2R24158988:24159464   | trem | spag        | 246 |    -32 | embryonic |\n",
       "| 0.7352163 | trem_trem_embryonic_1chrX3936212:3936872      | trem | Glos        | 141 |     25 | embryonic |\n",
       "| 0.7310491 | trem_trem_embryonic_1chr3L9838351:9839011     | trem | Cdk8        | 131 |    -11 | embryonic |\n",
       "| 0.7279310 | trem_trem_embryonic_1chr3L18901445:18901697   | trem | CG6843      | 164 |    -18 | embryonic |\n",
       "| 0.6729249 | usp_usp-GFP_embryonic_1chr3L21261221:21261305 | usp  | CG43218     |  80 |   3243 | embryonic |\n",
       "| 0.5878926 | usp_usp-GFP_embryonic_1chr3L20405430:20405514 | usp  | trbl        |  23 |  -3863 | embryonic |\n",
       "| 0.5116316 | usp_usp-GFP_embryonic_1chr3L15545117:15545201 | usp  | AGO2        | 129 |  -8954 | embryonic |\n",
       "| 0.4485843 | usp_usp-GFP_embryonic_1chr2R9548145:9548229   | usp  | CG1888      | 107 |    -52 | embryonic |\n",
       "| 0.4352487 | usp_usp-GFP_embryonic_1chr2R21707509:21707593 | usp  | HmgZ        |  28 |    310 | embryonic |\n",
       "| 0.4196480 | usp_usp-GFP_embryonic_1chrX647042:647126      | usp  | Roc1a       | 193 |   -220 | embryonic |\n",
       "| 0.3443677 | usp_usp-GFP_embryonic_1chr3R13790064:13790148 | usp  | Dip-B       |  27 |   7341 | embryonic |\n",
       "| 0.3379946 | usp_usp-GFP_embryonic_1chr3L645998:646082     | usp  | Gale        |  44 |  -2515 | embryonic |\n",
       "| 0.2819830 | usp_usp-GFP_embryonic_1chrX12012557:12012641  | usp  | regucalcin  | 147 |    222 | embryonic |\n",
       "| 0.2472901 | usp_usp-GFP_embryonic_1chr3L3978406:3978490   | usp  | scrt        |  96 |  -5448 | embryonic |\n",
       "| 0.7488630 | wdn_wdn-GFP_embryonic_1chr2R21701778:21702082 | wdn  | HmgD        |  42 |    472 | embryonic |\n",
       "| 0.7385835 | wdn_wdn-GFP_embryonic_1chr2R22640149:22640453 | wdn  | CG3732      | 207 |   -100 | embryonic |\n",
       "| 0.6968361 | wdn_wdn-GFP_embryonic_1chr2L13366120:13366424 | wdn  | Nnp-1       | 210 |    -83 | embryonic |\n",
       "| 0.6831170 | wdn_wdn-GFP_embryonic_1chr3L16087389:16087693 | wdn  | Tasp1       | 240 |    -16 | embryonic |\n",
       "| 0.6790940 | wdn_wdn-GFP_embryonic_1chr3L4626924:4627228   | wdn  | HDAC1       | 101 |    342 | embryonic |\n",
       "| 0.6747096 | wdn_wdn-GFP_embryonic_1chr2L18136614:18136918 | wdn  | kel         | 149 |   1317 | embryonic |\n",
       "| 0.6718097 | wdn_wdn-GFP_embryonic_1chr3R9713295:9713599   | wdn  | dmt         |   3 |   2305 | embryonic |\n",
       "| 0.6576688 | wdn_wdn-GFP_embryonic_1chr3L15616918:15617222 | wdn  | Phs         |  27 |  -2831 | embryonic |\n",
       "| 0.6498783 | wdn_wdn-GFP_embryonic_1chr3R24829897:24830201 | wdn  | Kaz         | 180 |    -90 | embryonic |\n",
       "| 0.6364874 | wdn_wdn-GFP_embryonic_1chr2R14410812:14411116 | wdn  | CG17385     |   2 |    608 | embryonic |\n",
       "| 0.7198855 | wek_wek-GFP_embryonic_1chr3L22065629:22066119 | wek  | CG14561     | 218 |    -33 | embryonic |\n",
       "| 0.6980301 | wek_wek-GFP_embryonic_1chr2R22806706:22807196 | wek  | CycB        | 245 |     -3 | embryonic |\n",
       "| 0.6966879 | wek_wek-GFP_embryonic_1chr2R9116356:9116846   | wek  | Su(var)2-10 | 234 |    -17 | embryonic |\n",
       "| 0.6735059 | wek_wek-GFP_embryonic_1chr2R21720467:21720957 | wek  | LBR         | 153 |    -22 | embryonic |\n",
       "| 0.6538962 | wek_wek-GFP_embryonic_1chr3R24753535:24753961 | wek  | asp         | 155 |    -77 | embryonic |\n",
       "| 0.6421066 | wek_wek-GFP_embryonic_1chr3R12985836:12986223 | wek  | CG31342     | 210 |     -6 | embryonic |\n",
       "| 0.6376551 | wek_wek-GFP_embryonic_1chr3R24870674:24871157 | wek  | CycB3       | 174 |   -104 | embryonic |\n",
       "| 0.6364278 | wek_wek-GFP_embryonic_1chr2R22637483:22637973 | wek  | Cdk9        | 120 |     -1 | embryonic |\n",
       "| 0.6187072 | wek_wek-GFP_embryonic_1chr3L8175765:8176255   | wek  | RecQ4       |  92 |    -32 | embryonic |\n",
       "| 0.6058364 | wek_wek-GFP_embryonic_1chr2L21630459:21630949 | wek  | Df31        |   5 |   -772 | embryonic |\n",
       "| 1.0000000 | z_z-GFP_embryonic_1chrX2447768:2448519        | z    | z           | 193 |    276 | embryonic |\n",
       "| 0.8797171 | z_z-GFP_embryonic_1chr2R11894739:11895135     | z    | eEF1alpha1  |  11 |   1395 | embryonic |\n",
       "| 0.8579696 | z_z-GFP_embryonic_1chr3R25032723:25033119     | z    | Smg6        |  82 |    544 | embryonic |\n",
       "| 0.8510130 | z_z-GFP_embryonic_1chr2L6049899:6050295       | z    | CG9135      |   7 |   -703 | embryonic |\n",
       "| 0.8137604 | z_z-GFP_embryonic_1chr3R17066775:17067171     | z    | Ns1         |  26 |   5702 | embryonic |\n",
       "| 0.8109701 | z_z-GFP_embryonic_1chr3R9503561:9503957       | z    | CG8149      |  52 |   -897 | embryonic |\n",
       "| 0.8047936 | z_z-GFP_embryonic_1chr2L1082625:1082980       | z    | Atg4a       |  28 |   1913 | embryonic |\n",
       "| 0.7998836 | z_z-GFP_embryonic_1chr2R6696542:6696938       | z    | Rab2        | 270 |      1 | embryonic |\n",
       "| 0.7972664 | z_z-GFP_embryonic_1chrX763570:763966          | z    | RpL22       |   7 |   1362 | embryonic |\n",
       "| 0.7848290 | z_z-GFP_embryonic_1chr2R12904388:12904784     | z    | NAT1        | 114 | -11408 | embryonic |\n",
       "\n"
      ],
      "text/plain": [
       "    CosineAngle peakID                                                TF     \n",
       "1   0.6189138   ATbp_ATbp-GFP_embryonic_1chr2L6028600:6029220         ATbp   \n",
       "2   0.6096739   ATbp_ATbp-GFP_embryonic_1chrX18839131:18839751        ATbp   \n",
       "3   0.5729203   ATbp_ATbp-GFP_embryonic_1chr3L14780103:14780723       ATbp   \n",
       "4   0.5657947   ATbp_ATbp-GFP_embryonic_1chr2L19126804:19127424       ATbp   \n",
       "5   0.5621825   ATbp_ATbp-GFP_embryonic_1chrX14242925:14243545        ATbp   \n",
       "6   0.5530836   ATbp_ATbp-GFP_embryonic_1chr3R5746732:5746967         ATbp   \n",
       "7   0.5304849   ATbp_ATbp-GFP_embryonic_1chr3R22530151:22530771       ATbp   \n",
       "8   0.5222559   ATbp_ATbp-GFP_embryonic_1chr3R21263270:21263491       ATbp   \n",
       "9   0.5117363   ATbp_ATbp-GFP_embryonic_1chr3L21464358:21464622       ATbp   \n",
       "10  0.4935757   ATbp_ATbp-GFP_embryonic_1chr2R23681001:23681621       ATbp   \n",
       "11  0.5660885   Asciz_CG14962-GFP_embryonic_1chr2L7401457:7402157     Asciz  \n",
       "12  0.5388026   Asciz_CG14962-GFP_embryonic_1chr3L21204554:21205254   Asciz  \n",
       "13  0.5359796   Asciz_CG14962-GFP_embryonic_1chr3L23095104:23095804   Asciz  \n",
       "14  0.5156593   Asciz_CG14962-GFP_embryonic_1chr3R23134565:23135265   Asciz  \n",
       "15  0.5150351   Asciz_CG14962-GFP_embryonic_1chr3L19932819:19933519   Asciz  \n",
       "16  0.4925116   Asciz_CG14962-GFP_embryonic_1chr3L16010047:16010747   Asciz  \n",
       "17  0.4853068   Asciz_CG14962-GFP_embryonic_1chr3R17067241:17067941   Asciz  \n",
       "18  0.4462886   Asciz_CG14962-GFP_embryonic_1chr2L7386781:7387481     Asciz  \n",
       "19  0.4323968   Asciz_CG14962-GFP_embryonic_1chr3L16749585:16750285   Asciz  \n",
       "20  0.4308037   Asciz_CG14962-GFP_embryonic_1chr2L8423636:8424336     Asciz  \n",
       "21  0.6810581   BtbVII_BtbVII-GFP_embryonic_1chr3R8694118:8694458     BtbVII \n",
       "22  0.6055603   BtbVII_BtbVII-GFP_embryonic_1chr3L15347886:15348670   BtbVII \n",
       "23  0.5840373   BtbVII_BtbVII-GFP_embryonic_1chr3R24319821:24320605   BtbVII \n",
       "24  0.5788026   BtbVII_BtbVII-GFP_embryonic_1chr3L2825207:2825991     BtbVII \n",
       "25  0.5718206   BtbVII_BtbVII-GFP_embryonic_1chr2L8029494:8030278     BtbVII \n",
       "26  0.5630155   BtbVII_BtbVII-GFP_embryonic_1chr2L14474785:14475569   BtbVII \n",
       "27  0.5621026   BtbVII_BtbVII-GFP_embryonic_1chr2L21218652:21219436   BtbVII \n",
       "28  0.5499432   BtbVII_BtbVII-GFP_embryonic_1chr3R30800074:30800858   BtbVII \n",
       "29  0.5480910   BtbVII_BtbVII-GFP_embryonic_1chrX13207891:13208675    BtbVII \n",
       "30  0.5426162   BtbVII_BtbVII-GFP_embryonic_1chr3L8440510:8441294     BtbVII \n",
       "31  1.0000000   CG10209_CG10209-GFP_embryonic_1chr2R14848187:14848564 CG10209\n",
       "32  0.6515456   CG10209_CG10209-GFP_embryonic_1chr2L10987232:10987836 CG10209\n",
       "33  0.6473721   CG10209_CG10209-GFP_embryonic_1chr3R19877336:19877940 CG10209\n",
       "34  0.5799880   CG10209_CG10209-GFP_embryonic_1chr3L5819163:5819767   CG10209\n",
       "35  0.5614826   CG10209_CG10209-GFP_embryonic_1chrX1380263:1380867    CG10209\n",
       "36  0.5596084   CG10209_CG10209-GFP_embryonic_1chr3L6885588:6886192   CG10209\n",
       "37  0.5569965   CG10209_CG10209-GFP_embryonic_1chr3R22530196:22530800 CG10209\n",
       "38  0.5404655   CG10209_CG10209-GFP_embryonic_1chr3R21263107:21263711 CG10209\n",
       "39  0.5234731   CG10209_CG10209-GFP_embryonic_1chr3L14780114:14780718 CG10209\n",
       "40  0.5180713   CG10209_CG10209-GFP_embryonic_1chr2L11188976:11189580 CG10209\n",
       "41  0.6919174   CG10274_CG10274-GFP_embryonic_1chr2L1156453:1157017   CG10274\n",
       "42  0.6579180   CG10274_CG10274-GFP_embryonic_1chrX15786208:15786622  CG10274\n",
       "43  0.6501383   CG10274_CG10274-GFP_embryonic_1chr2R7917372:7917711   CG10274\n",
       "44  0.6393406   CG10274_CG10274-GFP_embryonic_1chr3R17020878:17021442 CG10274\n",
       "45  0.6164763   CG10274_CG10274-GFP_embryonic_1chr3R11392892:11393098 CG10274\n",
       "46  0.6075494   CG10274_CG10274-GFP_embryonic_1chr3R11822127:11822691 CG10274\n",
       "47  0.5341254   CG10274_CG10274-GFP_embryonic_1chrX14948750:14949314  CG10274\n",
       "48  0.5322367   CG10274_CG10274-GFP_embryonic_1chr2L8000486:8001029   CG10274\n",
       "49  0.5177671   CG10274_CG10274-GFP_embryonic_1chr3L9712112:9712676   CG10274\n",
       "50  0.4960898   CG10274_CG10274-GFP_embryonic_1chrX1922637:1923201    CG10274\n",
       "⋮   ⋮           ⋮                                                     ⋮      \n",
       "611 0.8207679   trem_trem_embryonic_1chrX2727348:2728008              trem   \n",
       "612 0.7929026   trem_trem_embryonic_1chr3R16639784:16640444           trem   \n",
       "613 0.7677257   trem_trem_embryonic_1chr3L22071911:22072571           trem   \n",
       "614 0.7533282   trem_trem_embryonic_1chr3R24058076:24058397           trem   \n",
       "615 0.7509802   trem_trem_embryonic_1chr2R13132922:13133256           trem   \n",
       "616 0.7496516   trem_trem_embryonic_1chr2R7962019:7962679             trem   \n",
       "617 0.7379604   trem_trem_embryonic_1chr2R24158988:24159464           trem   \n",
       "618 0.7352163   trem_trem_embryonic_1chrX3936212:3936872              trem   \n",
       "619 0.7310491   trem_trem_embryonic_1chr3L9838351:9839011             trem   \n",
       "620 0.7279310   trem_trem_embryonic_1chr3L18901445:18901697           trem   \n",
       "621 0.6729249   usp_usp-GFP_embryonic_1chr3L21261221:21261305         usp    \n",
       "622 0.5878926   usp_usp-GFP_embryonic_1chr3L20405430:20405514         usp    \n",
       "623 0.5116316   usp_usp-GFP_embryonic_1chr3L15545117:15545201         usp    \n",
       "624 0.4485843   usp_usp-GFP_embryonic_1chr2R9548145:9548229           usp    \n",
       "625 0.4352487   usp_usp-GFP_embryonic_1chr2R21707509:21707593         usp    \n",
       "626 0.4196480   usp_usp-GFP_embryonic_1chrX647042:647126              usp    \n",
       "627 0.3443677   usp_usp-GFP_embryonic_1chr3R13790064:13790148         usp    \n",
       "628 0.3379946   usp_usp-GFP_embryonic_1chr3L645998:646082             usp    \n",
       "629 0.2819830   usp_usp-GFP_embryonic_1chrX12012557:12012641          usp    \n",
       "630 0.2472901   usp_usp-GFP_embryonic_1chr3L3978406:3978490           usp    \n",
       "631 0.7488630   wdn_wdn-GFP_embryonic_1chr2R21701778:21702082         wdn    \n",
       "632 0.7385835   wdn_wdn-GFP_embryonic_1chr2R22640149:22640453         wdn    \n",
       "633 0.6968361   wdn_wdn-GFP_embryonic_1chr2L13366120:13366424         wdn    \n",
       "634 0.6831170   wdn_wdn-GFP_embryonic_1chr3L16087389:16087693         wdn    \n",
       "635 0.6790940   wdn_wdn-GFP_embryonic_1chr3L4626924:4627228           wdn    \n",
       "636 0.6747096   wdn_wdn-GFP_embryonic_1chr2L18136614:18136918         wdn    \n",
       "637 0.6718097   wdn_wdn-GFP_embryonic_1chr3R9713295:9713599           wdn    \n",
       "638 0.6576688   wdn_wdn-GFP_embryonic_1chr3L15616918:15617222         wdn    \n",
       "639 0.6498783   wdn_wdn-GFP_embryonic_1chr3R24829897:24830201         wdn    \n",
       "640 0.6364874   wdn_wdn-GFP_embryonic_1chr2R14410812:14411116         wdn    \n",
       "641 0.7198855   wek_wek-GFP_embryonic_1chr3L22065629:22066119         wek    \n",
       "642 0.6980301   wek_wek-GFP_embryonic_1chr2R22806706:22807196         wek    \n",
       "643 0.6966879   wek_wek-GFP_embryonic_1chr2R9116356:9116846           wek    \n",
       "644 0.6735059   wek_wek-GFP_embryonic_1chr2R21720467:21720957         wek    \n",
       "645 0.6538962   wek_wek-GFP_embryonic_1chr3R24753535:24753961         wek    \n",
       "646 0.6421066   wek_wek-GFP_embryonic_1chr3R12985836:12986223         wek    \n",
       "647 0.6376551   wek_wek-GFP_embryonic_1chr3R24870674:24871157         wek    \n",
       "648 0.6364278   wek_wek-GFP_embryonic_1chr2R22637483:22637973         wek    \n",
       "649 0.6187072   wek_wek-GFP_embryonic_1chr3L8175765:8176255           wek    \n",
       "650 0.6058364   wek_wek-GFP_embryonic_1chr2L21630459:21630949         wek    \n",
       "651 1.0000000   z_z-GFP_embryonic_1chrX2447768:2448519                z      \n",
       "652 0.8797171   z_z-GFP_embryonic_1chr2R11894739:11895135             z      \n",
       "653 0.8579696   z_z-GFP_embryonic_1chr3R25032723:25033119             z      \n",
       "654 0.8510130   z_z-GFP_embryonic_1chr2L6049899:6050295               z      \n",
       "655 0.8137604   z_z-GFP_embryonic_1chr3R17066775:17067171             z      \n",
       "656 0.8109701   z_z-GFP_embryonic_1chr3R9503561:9503957               z      \n",
       "657 0.8047936   z_z-GFP_embryonic_1chr2L1082625:1082980               z      \n",
       "658 0.7998836   z_z-GFP_embryonic_1chr2R6696542:6696938               z      \n",
       "659 0.7972664   z_z-GFP_embryonic_1chrX763570:763966                  z      \n",
       "660 0.7848290   z_z-GFP_embryonic_1chr2R12904388:12904784             z      \n",
       "    Gene         nPeaks targetDist lifeStage\n",
       "1   CG9107        63     -4371     embryonic\n",
       "2   htk          105     -4222     embryonic\n",
       "3   mop          143     -4284     embryonic\n",
       "4   l(2)37Cb     230      -138     embryonic\n",
       "5   mRNA-cap     256      -413     embryonic\n",
       "6   Madm          48      4324     embryonic\n",
       "7   CG33099      215       111     embryonic\n",
       "8   Rab1         184      5689     embryonic\n",
       "9   croc         246     11983     embryonic\n",
       "10  St1          123      -312     embryonic\n",
       "11  CG5171       274       472     embryonic\n",
       "12  CG10512      202       117     embryonic\n",
       "13  RpL10        211        17     embryonic\n",
       "14  CG31457       56     -6934     embryonic\n",
       "15  CG46462      184      1478     embryonic\n",
       "16  mib1          26       -88     embryonic\n",
       "17  Ns1           61      5084     embryonic\n",
       "18  CG5149        41      1122     embryonic\n",
       "19  CG42852      113     -5554     embryonic\n",
       "20  Akap200      154     -2787     embryonic\n",
       "21  hb           107       303     embryonic\n",
       "22  Toll-6       262     11586     embryonic\n",
       "23  BRWD3        134      1213     embryonic\n",
       "24  Tet          232     -4496     embryonic\n",
       "25  baf          113       424     embryonic\n",
       "26  noc          221    -15682     embryonic\n",
       "27  CG8679       232      -500     embryonic\n",
       "28  wts          158      6153     embryonic\n",
       "29  sno          127      3391     embryonic\n",
       "30  Unr          225       805     embryonic\n",
       "31  CG10209       75         4     embryonic\n",
       "32  piwi         265      -114     embryonic\n",
       "33  Arc42        157       -99     embryonic\n",
       "34  mad2         170     -6372     embryonic\n",
       "35  CG3719       261       -85     embryonic\n",
       "36  Prat2        116     28074     embryonic\n",
       "37  CG33099      215        74     embryonic\n",
       "38  Rab1         184      5651     embryonic\n",
       "39  mop          143     -4287     embryonic\n",
       "40  Boh2         249     10246     embryonic\n",
       "41  capt          26      -424     embryonic\n",
       "42  mRpS30       112       393     embryonic\n",
       "43  Cul1          44       367     embryonic\n",
       "44  AdSL         275       -40     embryonic\n",
       "45  mRpL40       230    -14934     embryonic\n",
       "46  Kyat         167        11     embryonic\n",
       "47  pdgy          92      1717     embryonic\n",
       "48  Sirup        264      1881     embryonic\n",
       "49  SH3PX1        42      1433     embryonic\n",
       "50  eIF2Bepsilon 160       -45     embryonic\n",
       "⋮   ⋮            ⋮      ⋮          ⋮        \n",
       "611 dwg          101       -20     embryonic\n",
       "612 nonA-l       151       -91     embryonic\n",
       "613 CG7139       248      -393     embryonic\n",
       "614 SMC1         226       -62     embryonic\n",
       "615 Cap-G        158       -86     embryonic\n",
       "616 nito         208      -114     embryonic\n",
       "617 spag         246       -32     embryonic\n",
       "618 Glos         141        25     embryonic\n",
       "619 Cdk8         131       -11     embryonic\n",
       "620 CG6843       164       -18     embryonic\n",
       "621 CG43218       80      3243     embryonic\n",
       "622 trbl          23     -3863     embryonic\n",
       "623 AGO2         129     -8954     embryonic\n",
       "624 CG1888       107       -52     embryonic\n",
       "625 HmgZ          28       310     embryonic\n",
       "626 Roc1a        193      -220     embryonic\n",
       "627 Dip-B         27      7341     embryonic\n",
       "628 Gale          44     -2515     embryonic\n",
       "629 regucalcin   147       222     embryonic\n",
       "630 scrt          96     -5448     embryonic\n",
       "631 HmgD          42       472     embryonic\n",
       "632 CG3732       207      -100     embryonic\n",
       "633 Nnp-1        210       -83     embryonic\n",
       "634 Tasp1        240       -16     embryonic\n",
       "635 HDAC1        101       342     embryonic\n",
       "636 kel          149      1317     embryonic\n",
       "637 dmt            3      2305     embryonic\n",
       "638 Phs           27     -2831     embryonic\n",
       "639 Kaz          180       -90     embryonic\n",
       "640 CG17385        2       608     embryonic\n",
       "641 CG14561      218       -33     embryonic\n",
       "642 CycB         245        -3     embryonic\n",
       "643 Su(var)2-10  234       -17     embryonic\n",
       "644 LBR          153       -22     embryonic\n",
       "645 asp          155       -77     embryonic\n",
       "646 CG31342      210        -6     embryonic\n",
       "647 CycB3        174      -104     embryonic\n",
       "648 Cdk9         120        -1     embryonic\n",
       "649 RecQ4         92       -32     embryonic\n",
       "650 Df31           5      -772     embryonic\n",
       "651 z            193       276     embryonic\n",
       "652 eEF1alpha1    11      1395     embryonic\n",
       "653 Smg6          82       544     embryonic\n",
       "654 CG9135         7      -703     embryonic\n",
       "655 Ns1           26      5702     embryonic\n",
       "656 CG8149        52      -897     embryonic\n",
       "657 Atg4a         28      1913     embryonic\n",
       "658 Rab2         270         1     embryonic\n",
       "659 RpL22          7      1362     embryonic\n",
       "660 NAT1         114    -11408     embryonic"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fly_targets_new_annot %>% arrange(-CosineAngle)  %>% group_by(TF) %>% \n",
    "    slice_max(CosineAngle, n = 10)  %>% filter(TF %in% fly_top_TFs) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 675,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 660 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>CosineAngle</th><th scope=col>peakID</th><th scope=col>TF</th><th scope=col>Gene</th><th scope=col>nPeaks</th><th scope=col>targetDist</th><th scope=col>lifeStage</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>-0.4859642</td><td>ATbp_ATbp-GFP_embryonic_1chr3L440263:440570          </td><td>ATbp   </td><td>klar     </td><td>202</td><td>   666</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4460530</td><td>ATbp_ATbp-GFP_embryonic_1chrX9349102:9349722         </td><td>ATbp   </td><td>Zdhhc8   </td><td>247</td><td>  6815</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4363592</td><td>ATbp_ATbp-GFP_embryonic_1chr2L12462886:12463506      </td><td>ATbp   </td><td>bun      </td><td>139</td><td> -1940</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4319920</td><td>ATbp_ATbp-GFP_embryonic_1chr2L17387950:17388570      </td><td>ATbp   </td><td>CLIP-190 </td><td> 72</td><td> -1081</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4313847</td><td>ATbp_ATbp-GFP_embryonic_1chr2L2904931:2905551        </td><td>ATbp   </td><td>lilli    </td><td>241</td><td> 15070</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4272610</td><td>ATbp_ATbp-GFP_embryonic_1chr2L8097305:8097925        </td><td>ATbp   </td><td>Bsg      </td><td>234</td><td>  8512</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4179265</td><td>ATbp_ATbp-GFP_embryonic_1chr3R21330422:21331042      </td><td>ATbp   </td><td>aus      </td><td>237</td><td> 18869</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4082246</td><td>ATbp_ATbp-GFP_embryonic_1chr3R21594255:21594875      </td><td>ATbp   </td><td>InR      </td><td>178</td><td> -2444</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4081062</td><td>ATbp_ATbp-GFP_embryonic_1chrX3704290:3704910         </td><td>ATbp   </td><td>Rala     </td><td>221</td><td> 15035</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4033101</td><td>ATbp_ATbp-GFP_embryonic_1chr3L16056116:16056736      </td><td>ATbp   </td><td>Mbs      </td><td> 94</td><td>  4467</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3826688</td><td>Asciz_CG14962-GFP_embryonic_1chr3L3788576:3789276    </td><td>Asciz  </td><td>CG32264  </td><td>209</td><td>  -960</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3761955</td><td>Asciz_CG14962-GFP_embryonic_1chrX16089526:16090226   </td><td>Asciz  </td><td>Tob      </td><td> 97</td><td>  -509</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3599308</td><td>Asciz_CG14962-GFP_embryonic_1chr3R29040410:29041110  </td><td>Asciz  </td><td>Pkc98E   </td><td>202</td><td>   930</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3580836</td><td>Asciz_CG14962-GFP_embryonic_1chr3L440061:440761      </td><td>Asciz  </td><td>klar     </td><td>202</td><td>   685</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3543654</td><td>Asciz_CG14962-GFP_embryonic_1chr2L17450647:17451347  </td><td>Asciz  </td><td>dl       </td><td>265</td><td>  -633</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3377280</td><td>Asciz_CG14962-GFP_embryonic_1chr3R13999598:13999855  </td><td>Asciz  </td><td>rdx      </td><td>245</td><td>-16369</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3366867</td><td>Asciz_CG14962-GFP_embryonic_1chrX9651844:9652544     </td><td>Asciz  </td><td>Ptpmeg2  </td><td>246</td><td> -1418</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3358785</td><td>Asciz_CG14962-GFP_embryonic_1chr2L12462809:12463509  </td><td>Asciz  </td><td>bun      </td><td>139</td><td> -1903</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3293083</td><td>Asciz_CG14962-GFP_embryonic_1chr2L7940487:7941187    </td><td>Asciz  </td><td>Snoo     </td><td>153</td><td> 36228</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3292932</td><td>Asciz_CG14962-GFP_embryonic_1chr41053626:1054326     </td><td>Asciz  </td><td>CaMKII   </td><td>158</td><td>  -273</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.5697019</td><td>BtbVII_BtbVII-GFP_embryonic_1chr3L16969432:16970216  </td><td>BtbVII </td><td>Exn      </td><td>195</td><td>  -189</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.5331014</td><td>BtbVII_BtbVII-GFP_embryonic_1chr2R14265400:14266184  </td><td>BtbVII </td><td>CG8547   </td><td> 98</td><td>  1705</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4981587</td><td>BtbVII_BtbVII-GFP_embryonic_1chrX1844900:1845684     </td><td>BtbVII </td><td>prage    </td><td>219</td><td>  1946</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4905457</td><td>BtbVII_BtbVII-GFP_embryonic_1chr2L6082679:6083463    </td><td>BtbVII </td><td>Kr-h1    </td><td>261</td><td>   468</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4903502</td><td>BtbVII_BtbVII-GFP_embryonic_1chr3L4288414:4289198    </td><td>BtbVII </td><td>Hexo1    </td><td>188</td><td>  -139</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4841757</td><td>BtbVII_BtbVII-GFP_embryonic_1chr2L22143962:22144746  </td><td>BtbVII </td><td>CG42748  </td><td>115</td><td>  4233</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4795594</td><td>BtbVII_BtbVII-GFP_embryonic_1chr3L8065636:8066420    </td><td>BtbVII </td><td>Sarm     </td><td>199</td><td>  2154</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4736077</td><td>BtbVII_BtbVII-GFP_embryonic_1chr3L11278232:11279016  </td><td>BtbVII </td><td>Dyro     </td><td>149</td><td> 16164</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4559693</td><td>BtbVII_BtbVII-GFP_embryonic_1chr2L6784686:6785470    </td><td>BtbVII </td><td>nrv1     </td><td> 21</td><td>  1783</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4428619</td><td>BtbVII_BtbVII-GFP_embryonic_1chr2R7727321:7728105    </td><td>BtbVII </td><td>Gdap2    </td><td>109</td><td>  -203</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3728905</td><td>CG10209_CG10209-GFP_embryonic_1chrX20503070:20503674 </td><td>CG10209</td><td>RhoGAP19D</td><td>260</td><td>  7322</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3683510</td><td>CG10209_CG10209-GFP_embryonic_1chrX9349105:9349709   </td><td>CG10209</td><td>Zdhhc8   </td><td>247</td><td>  6820</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3430521</td><td>CG10209_CG10209-GFP_embryonic_1chrX15296796:15297400 </td><td>CG10209</td><td>gce      </td><td>229</td><td>  -668</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3416768</td><td>CG10209_CG10209-GFP_embryonic_1chr3R21594264:21594868</td><td>CG10209</td><td>InR      </td><td>178</td><td> -2445</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3412808</td><td>CG10209_CG10209-GFP_embryonic_1chr2R13908824:13909428</td><td>CG10209</td><td>shot     </td><td> 54</td><td>  1820</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3347060</td><td>CG10209_CG10209-GFP_embryonic_1chr2L12465613:12466217</td><td>CG10209</td><td>bun      </td><td>118</td><td> -4659</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3313975</td><td>CG10209_CG10209-GFP_embryonic_1chr3L3518769:3519373  </td><td>CG10209</td><td>Eip63E   </td><td>161</td><td>  3320</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3303142</td><td>CG10209_CG10209-GFP_embryonic_1chr2L2898142:2898746  </td><td>CG10209</td><td>lilli    </td><td> 19</td><td>  8273</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3216505</td><td>CG10209_CG10209-GFP_embryonic_1chr3L15591075:15591679</td><td>CG10209</td><td>RhoGAP71E</td><td>162</td><td>  2517</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3194778</td><td>CG10209_CG10209-GFP_embryonic_1chrX3739121:3739725   </td><td>CG10209</td><td>Tlk      </td><td>164</td><td>  -474</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4938688</td><td>CG10274_CG10274-GFP_embryonic_1chr3R30062492:30062820</td><td>CG10274</td><td>sima     </td><td> 76</td><td>  -202</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4644055</td><td>CG10274_CG10274-GFP_embryonic_1chr2L3100748:3101312  </td><td>CG10274</td><td>toc      </td><td>178</td><td> -6335</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4460244</td><td>CG10274_CG10274-GFP_embryonic_1chr3L16060760:16061324</td><td>CG10274</td><td>Mbs      </td><td>219</td><td>  9083</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4396431</td><td>CG10274_CG10274-GFP_embryonic_1chr3R14019803:14020367</td><td>CG10274</td><td>rdx      </td><td>101</td><td>  5877</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4242030</td><td>CG10274_CG10274-GFP_embryonic_1chr2L20884493:20884691</td><td>CG10274</td><td>sky      </td><td> 76</td><td>  1397</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4192172</td><td>CG10274_CG10274-GFP_embryonic_1chrX20503146:20503710 </td><td>CG10274</td><td>RhoGAP19D</td><td>260</td><td>  7266</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4012545</td><td>CG10274_CG10274-GFP_embryonic_1chr3R26540382:26540946</td><td>CG10274</td><td>scrib    </td><td>101</td><td>  4314</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3874745</td><td>CG10274_CG10274-GFP_embryonic_1chrX13325261:13325825 </td><td>CG10274</td><td>IP3K2    </td><td> 49</td><td>  4241</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3828380</td><td>CG10274_CG10274-GFP_embryonic_1chr2R8614890:8615454  </td><td>CG10274</td><td>Cirl     </td><td>226</td><td>   450</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3798026</td><td>CG10274_CG10274-GFP_embryonic_1chr2R17285195:17285759</td><td>CG10274</td><td>mbl      </td><td>  4</td><td> 10067</td><td>embryonic</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>-0.4686251</td><td>trem_trem_embryonic_1chr2L7983862:7984522    </td><td>trem</td><td>Snoo    </td><td>127</td><td>     4</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4267233</td><td>trem_trem_embryonic_1chr2R2848173:2848833    </td><td>trem</td><td>CG41378 </td><td> 66</td><td> -1795</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4036791</td><td>trem_trem_embryonic_1chr2L17473149:17473497  </td><td>trem</td><td>BicD    </td><td>261</td><td>  -112</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3981968</td><td>trem_trem_embryonic_1chrX7323717:7324074     </td><td>trem</td><td>Atg5    </td><td>233</td><td>  -140</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3709291</td><td>trem_trem_embryonic_1chr3L27278901:27279561  </td><td>trem</td><td>Dbp80   </td><td> 58</td><td>   -85</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3562638</td><td>trem_trem_embryonic_1chr3R21594135:21594795  </td><td>trem</td><td>InR     </td><td>178</td><td> -2344</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3542753</td><td>trem_trem_embryonic_1chr3R14025931:14026591  </td><td>trem</td><td>rdx     </td><td>196</td><td>  -299</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3517334</td><td>trem_trem_embryonic_1chr2L12487144:12487804  </td><td>trem</td><td>bun     </td><td>149</td><td> -3885</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3470893</td><td>trem_trem_embryonic_1chr2L8740351:8741011    </td><td>trem</td><td>raw     </td><td>241</td><td>   -21</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3433986</td><td>trem_trem_embryonic_1chrX11369294:11369954   </td><td>trem</td><td>dlg1    </td><td>112</td><td>   -41</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4190771</td><td>usp_usp-GFP_embryonic_1chr2L6077596:6077680  </td><td>usp </td><td>Kr-h1   </td><td> 32</td><td> -4965</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3524770</td><td>usp_usp-GFP_embryonic_1chr3L4350313:4350397  </td><td>usp </td><td>Cip4    </td><td> 17</td><td> -4858</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3363832</td><td>usp_usp-GFP_embryonic_1chr3R9886101:9886185  </td><td>usp </td><td>Glut4EF </td><td> 86</td><td> 11462</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3340628</td><td>usp_usp-GFP_embryonic_1chr3L2579965:2580049  </td><td>usp </td><td>msn     </td><td> 99</td><td>  6247</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3245632</td><td>usp_usp-GFP_embryonic_1chr3R14690908:14690992</td><td>usp </td><td>Meltrin </td><td>  8</td><td>  6342</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3061045</td><td>usp_usp-GFP_embryonic_1chr3R19072227:19072311</td><td>usp </td><td>cdi     </td><td> 11</td><td> 22001</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.2991443</td><td>usp_usp-GFP_embryonic_1chrX1095265:1095349   </td><td>usp </td><td>CG3655  </td><td>  3</td><td>-12868</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.2839525</td><td>usp_usp-GFP_embryonic_1chr3R10612564:10612648</td><td>usp </td><td>hth     </td><td> 19</td><td>  1670</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.2743099</td><td>usp_usp-GFP_embryonic_1chr3R17680490:17680574</td><td>usp </td><td>alt     </td><td>101</td><td>  -656</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.2715323</td><td>usp_usp-GFP_embryonic_1chr2L1982107:1982191  </td><td>usp </td><td>CG7289  </td><td>145</td><td>   -68</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4658849</td><td>wdn_wdn-GFP_embryonic_1chr3R9891834:9892138  </td><td>wdn </td><td>Glut4EF </td><td> 89</td><td> 17305</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4547252</td><td>wdn_wdn-GFP_embryonic_1chr2L12459952:12460256</td><td>wdn </td><td>bun     </td><td> 53</td><td>  1152</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4382633</td><td>wdn_wdn-GFP_embryonic_1chr3R23915049:23915353</td><td>wdn </td><td>sba     </td><td>244</td><td>  7772</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4176897</td><td>wdn_wdn-GFP_embryonic_1chrX13320671:13320975 </td><td>wdn </td><td>IP3K2   </td><td>188</td><td>  -479</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4169000</td><td>wdn_wdn-GFP_embryonic_1chrX11370814:11371118 </td><td>wdn </td><td>dlg1    </td><td>147</td><td>  1301</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4019413</td><td>wdn_wdn-GFP_embryonic_1chr3L16160085:16160389</td><td>wdn </td><td>CG5151  </td><td> 89</td><td>  4464</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3951208</td><td>wdn_wdn-GFP_embryonic_1chrX11621940:11622244 </td><td>wdn </td><td>Ptp10D  </td><td>257</td><td>    77</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3945413</td><td>wdn_wdn-GFP_embryonic_1chrX13273482:13273786 </td><td>wdn </td><td>HDAC4   </td><td> 53</td><td>  -890</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3861245</td><td>wdn_wdn-GFP_embryonic_1chr2L8714063:8714367  </td><td>wdn </td><td>raw     </td><td> 23</td><td>   368</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3835956</td><td>wdn_wdn-GFP_embryonic_1chr3L3787899:3788203  </td><td>wdn </td><td>CG32264 </td><td>250</td><td>   -85</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4193265</td><td>wek_wek-GFP_embryonic_1chr2R16609894:16610384</td><td>wek </td><td>CG5065  </td><td>150</td><td>  1011</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4191661</td><td>wek_wek-GFP_embryonic_1chr2R14267708:14268198</td><td>wek </td><td>CG8547  </td><td>276</td><td>   -34</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.4122494</td><td>wek_wek-GFP_embryonic_1chr2R8065348:8065838  </td><td>wek </td><td>CG12769 </td><td>222</td><td>   -39</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3930019</td><td>wek_wek-GFP_embryonic_1chr2R14769057:14769547</td><td>wek </td><td>Spred   </td><td>211</td><td>   121</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3699499</td><td>wek_wek-GFP_embryonic_1chr2L17472883:17473373</td><td>wek </td><td>BicD    </td><td>261</td><td>    87</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3590291</td><td>wek_wek-GFP_embryonic_1chr3R23803764:23803976</td><td>wek </td><td>CG34355 </td><td> 97</td><td>    17</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3573415</td><td>wek_wek-GFP_embryonic_1chr3R20949747:20950237</td><td>wek </td><td>Atpalpha</td><td>193</td><td>  1212</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3440037</td><td>wek_wek-GFP_embryonic_1chrX11621814:11622304 </td><td>wek </td><td>Ptp10D  </td><td>257</td><td>    44</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3309026</td><td>wek_wek-GFP_embryonic_1chr3R23962858:23963348</td><td>wek </td><td>CG31140 </td><td>166</td><td>   -77</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.3296127</td><td>wek_wek-GFP_embryonic_1chr3L2144231:2144721  </td><td>wek </td><td>zormin  </td><td> 29</td><td>  7224</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.7890103</td><td>z_z-GFP_embryonic_1chr3R30090355:30090641    </td><td>z   </td><td>sima    </td><td> 76</td><td> -9011</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.7089874</td><td>z_z-GFP_embryonic_1chr2L12459929:12460325    </td><td>z   </td><td>bun     </td><td> 53</td><td>  1129</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.7062584</td><td>z_z-GFP_embryonic_1chr2R2495616:2496012      </td><td>z   </td><td>MFS17   </td><td>  9</td><td> 69571</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.7013925</td><td>z_z-GFP_embryonic_1chr3R13973796:13974192    </td><td>z   </td><td>rdx     </td><td>  6</td><td> -2194</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.6471482</td><td>z_z-GFP_embryonic_1chr3L3787819:3788215      </td><td>z   </td><td>CG32264 </td><td>250</td><td>   -51</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.6462794</td><td>z_z-GFP_embryonic_1chr2R19467149:19467545    </td><td>z   </td><td>par-1   </td><td>121</td><td>  -119</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.6421429</td><td>z_z-GFP_embryonic_1chr2L2905161:2905557      </td><td>z   </td><td>lilli   </td><td>241</td><td> 15188</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.6401059</td><td>z_z-GFP_embryonic_1chr3L13226436:13226832    </td><td>z   </td><td>caps    </td><td>210</td><td> -2058</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.6387908</td><td>z_z-GFP_embryonic_1chr3L489544:489940        </td><td>z   </td><td>klar    </td><td>226</td><td>  1081</td><td>embryonic</td></tr>\n",
       "\t<tr><td>-0.6347419</td><td>z_z-GFP_embryonic_1chrX11397214:11397610     </td><td>z   </td><td>dlg1    </td><td>  4</td><td>  7040</td><td>embryonic</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 660 × 7\n",
       "\\begin{tabular}{lllllll}\n",
       " CosineAngle & peakID & TF & Gene & nPeaks & targetDist & lifeStage\\\\\n",
       " <dbl> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t -0.4859642 & ATbp\\_ATbp-GFP\\_embryonic\\_1chr3L440263:440570           & ATbp    & klar      & 202 &    666 & embryonic\\\\\n",
       "\t -0.4460530 & ATbp\\_ATbp-GFP\\_embryonic\\_1chrX9349102:9349722          & ATbp    & Zdhhc8    & 247 &   6815 & embryonic\\\\\n",
       "\t -0.4363592 & ATbp\\_ATbp-GFP\\_embryonic\\_1chr2L12462886:12463506       & ATbp    & bun       & 139 &  -1940 & embryonic\\\\\n",
       "\t -0.4319920 & ATbp\\_ATbp-GFP\\_embryonic\\_1chr2L17387950:17388570       & ATbp    & CLIP-190  &  72 &  -1081 & embryonic\\\\\n",
       "\t -0.4313847 & ATbp\\_ATbp-GFP\\_embryonic\\_1chr2L2904931:2905551         & ATbp    & lilli     & 241 &  15070 & embryonic\\\\\n",
       "\t -0.4272610 & ATbp\\_ATbp-GFP\\_embryonic\\_1chr2L8097305:8097925         & ATbp    & Bsg       & 234 &   8512 & embryonic\\\\\n",
       "\t -0.4179265 & ATbp\\_ATbp-GFP\\_embryonic\\_1chr3R21330422:21331042       & ATbp    & aus       & 237 &  18869 & embryonic\\\\\n",
       "\t -0.4082246 & ATbp\\_ATbp-GFP\\_embryonic\\_1chr3R21594255:21594875       & ATbp    & InR       & 178 &  -2444 & embryonic\\\\\n",
       "\t -0.4081062 & ATbp\\_ATbp-GFP\\_embryonic\\_1chrX3704290:3704910          & ATbp    & Rala      & 221 &  15035 & embryonic\\\\\n",
       "\t -0.4033101 & ATbp\\_ATbp-GFP\\_embryonic\\_1chr3L16056116:16056736       & ATbp    & Mbs       &  94 &   4467 & embryonic\\\\\n",
       "\t -0.3826688 & Asciz\\_CG14962-GFP\\_embryonic\\_1chr3L3788576:3789276     & Asciz   & CG32264   & 209 &   -960 & embryonic\\\\\n",
       "\t -0.3761955 & Asciz\\_CG14962-GFP\\_embryonic\\_1chrX16089526:16090226    & Asciz   & Tob       &  97 &   -509 & embryonic\\\\\n",
       "\t -0.3599308 & Asciz\\_CG14962-GFP\\_embryonic\\_1chr3R29040410:29041110   & Asciz   & Pkc98E    & 202 &    930 & embryonic\\\\\n",
       "\t -0.3580836 & Asciz\\_CG14962-GFP\\_embryonic\\_1chr3L440061:440761       & Asciz   & klar      & 202 &    685 & embryonic\\\\\n",
       "\t -0.3543654 & Asciz\\_CG14962-GFP\\_embryonic\\_1chr2L17450647:17451347   & Asciz   & dl        & 265 &   -633 & embryonic\\\\\n",
       "\t -0.3377280 & Asciz\\_CG14962-GFP\\_embryonic\\_1chr3R13999598:13999855   & Asciz   & rdx       & 245 & -16369 & embryonic\\\\\n",
       "\t -0.3366867 & Asciz\\_CG14962-GFP\\_embryonic\\_1chrX9651844:9652544      & Asciz   & Ptpmeg2   & 246 &  -1418 & embryonic\\\\\n",
       "\t -0.3358785 & Asciz\\_CG14962-GFP\\_embryonic\\_1chr2L12462809:12463509   & Asciz   & bun       & 139 &  -1903 & embryonic\\\\\n",
       "\t -0.3293083 & Asciz\\_CG14962-GFP\\_embryonic\\_1chr2L7940487:7941187     & Asciz   & Snoo      & 153 &  36228 & embryonic\\\\\n",
       "\t -0.3292932 & Asciz\\_CG14962-GFP\\_embryonic\\_1chr41053626:1054326      & Asciz   & CaMKII    & 158 &   -273 & embryonic\\\\\n",
       "\t -0.5697019 & BtbVII\\_BtbVII-GFP\\_embryonic\\_1chr3L16969432:16970216   & BtbVII  & Exn       & 195 &   -189 & embryonic\\\\\n",
       "\t -0.5331014 & BtbVII\\_BtbVII-GFP\\_embryonic\\_1chr2R14265400:14266184   & BtbVII  & CG8547    &  98 &   1705 & embryonic\\\\\n",
       "\t -0.4981587 & BtbVII\\_BtbVII-GFP\\_embryonic\\_1chrX1844900:1845684      & BtbVII  & prage     & 219 &   1946 & embryonic\\\\\n",
       "\t -0.4905457 & BtbVII\\_BtbVII-GFP\\_embryonic\\_1chr2L6082679:6083463     & BtbVII  & Kr-h1     & 261 &    468 & embryonic\\\\\n",
       "\t -0.4903502 & BtbVII\\_BtbVII-GFP\\_embryonic\\_1chr3L4288414:4289198     & BtbVII  & Hexo1     & 188 &   -139 & embryonic\\\\\n",
       "\t -0.4841757 & BtbVII\\_BtbVII-GFP\\_embryonic\\_1chr2L22143962:22144746   & BtbVII  & CG42748   & 115 &   4233 & embryonic\\\\\n",
       "\t -0.4795594 & BtbVII\\_BtbVII-GFP\\_embryonic\\_1chr3L8065636:8066420     & BtbVII  & Sarm      & 199 &   2154 & embryonic\\\\\n",
       "\t -0.4736077 & BtbVII\\_BtbVII-GFP\\_embryonic\\_1chr3L11278232:11279016   & BtbVII  & Dyro      & 149 &  16164 & embryonic\\\\\n",
       "\t -0.4559693 & BtbVII\\_BtbVII-GFP\\_embryonic\\_1chr2L6784686:6785470     & BtbVII  & nrv1      &  21 &   1783 & embryonic\\\\\n",
       "\t -0.4428619 & BtbVII\\_BtbVII-GFP\\_embryonic\\_1chr2R7727321:7728105     & BtbVII  & Gdap2     & 109 &   -203 & embryonic\\\\\n",
       "\t -0.3728905 & CG10209\\_CG10209-GFP\\_embryonic\\_1chrX20503070:20503674  & CG10209 & RhoGAP19D & 260 &   7322 & embryonic\\\\\n",
       "\t -0.3683510 & CG10209\\_CG10209-GFP\\_embryonic\\_1chrX9349105:9349709    & CG10209 & Zdhhc8    & 247 &   6820 & embryonic\\\\\n",
       "\t -0.3430521 & CG10209\\_CG10209-GFP\\_embryonic\\_1chrX15296796:15297400  & CG10209 & gce       & 229 &   -668 & embryonic\\\\\n",
       "\t -0.3416768 & CG10209\\_CG10209-GFP\\_embryonic\\_1chr3R21594264:21594868 & CG10209 & InR       & 178 &  -2445 & embryonic\\\\\n",
       "\t -0.3412808 & CG10209\\_CG10209-GFP\\_embryonic\\_1chr2R13908824:13909428 & CG10209 & shot      &  54 &   1820 & embryonic\\\\\n",
       "\t -0.3347060 & CG10209\\_CG10209-GFP\\_embryonic\\_1chr2L12465613:12466217 & CG10209 & bun       & 118 &  -4659 & embryonic\\\\\n",
       "\t -0.3313975 & CG10209\\_CG10209-GFP\\_embryonic\\_1chr3L3518769:3519373   & CG10209 & Eip63E    & 161 &   3320 & embryonic\\\\\n",
       "\t -0.3303142 & CG10209\\_CG10209-GFP\\_embryonic\\_1chr2L2898142:2898746   & CG10209 & lilli     &  19 &   8273 & embryonic\\\\\n",
       "\t -0.3216505 & CG10209\\_CG10209-GFP\\_embryonic\\_1chr3L15591075:15591679 & CG10209 & RhoGAP71E & 162 &   2517 & embryonic\\\\\n",
       "\t -0.3194778 & CG10209\\_CG10209-GFP\\_embryonic\\_1chrX3739121:3739725    & CG10209 & Tlk       & 164 &   -474 & embryonic\\\\\n",
       "\t -0.4938688 & CG10274\\_CG10274-GFP\\_embryonic\\_1chr3R30062492:30062820 & CG10274 & sima      &  76 &   -202 & embryonic\\\\\n",
       "\t -0.4644055 & CG10274\\_CG10274-GFP\\_embryonic\\_1chr2L3100748:3101312   & CG10274 & toc       & 178 &  -6335 & embryonic\\\\\n",
       "\t -0.4460244 & CG10274\\_CG10274-GFP\\_embryonic\\_1chr3L16060760:16061324 & CG10274 & Mbs       & 219 &   9083 & embryonic\\\\\n",
       "\t -0.4396431 & CG10274\\_CG10274-GFP\\_embryonic\\_1chr3R14019803:14020367 & CG10274 & rdx       & 101 &   5877 & embryonic\\\\\n",
       "\t -0.4242030 & CG10274\\_CG10274-GFP\\_embryonic\\_1chr2L20884493:20884691 & CG10274 & sky       &  76 &   1397 & embryonic\\\\\n",
       "\t -0.4192172 & CG10274\\_CG10274-GFP\\_embryonic\\_1chrX20503146:20503710  & CG10274 & RhoGAP19D & 260 &   7266 & embryonic\\\\\n",
       "\t -0.4012545 & CG10274\\_CG10274-GFP\\_embryonic\\_1chr3R26540382:26540946 & CG10274 & scrib     & 101 &   4314 & embryonic\\\\\n",
       "\t -0.3874745 & CG10274\\_CG10274-GFP\\_embryonic\\_1chrX13325261:13325825  & CG10274 & IP3K2     &  49 &   4241 & embryonic\\\\\n",
       "\t -0.3828380 & CG10274\\_CG10274-GFP\\_embryonic\\_1chr2R8614890:8615454   & CG10274 & Cirl      & 226 &    450 & embryonic\\\\\n",
       "\t -0.3798026 & CG10274\\_CG10274-GFP\\_embryonic\\_1chr2R17285195:17285759 & CG10274 & mbl       &   4 &  10067 & embryonic\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t -0.4686251 & trem\\_trem\\_embryonic\\_1chr2L7983862:7984522     & trem & Snoo     & 127 &      4 & embryonic\\\\\n",
       "\t -0.4267233 & trem\\_trem\\_embryonic\\_1chr2R2848173:2848833     & trem & CG41378  &  66 &  -1795 & embryonic\\\\\n",
       "\t -0.4036791 & trem\\_trem\\_embryonic\\_1chr2L17473149:17473497   & trem & BicD     & 261 &   -112 & embryonic\\\\\n",
       "\t -0.3981968 & trem\\_trem\\_embryonic\\_1chrX7323717:7324074      & trem & Atg5     & 233 &   -140 & embryonic\\\\\n",
       "\t -0.3709291 & trem\\_trem\\_embryonic\\_1chr3L27278901:27279561   & trem & Dbp80    &  58 &    -85 & embryonic\\\\\n",
       "\t -0.3562638 & trem\\_trem\\_embryonic\\_1chr3R21594135:21594795   & trem & InR      & 178 &  -2344 & embryonic\\\\\n",
       "\t -0.3542753 & trem\\_trem\\_embryonic\\_1chr3R14025931:14026591   & trem & rdx      & 196 &   -299 & embryonic\\\\\n",
       "\t -0.3517334 & trem\\_trem\\_embryonic\\_1chr2L12487144:12487804   & trem & bun      & 149 &  -3885 & embryonic\\\\\n",
       "\t -0.3470893 & trem\\_trem\\_embryonic\\_1chr2L8740351:8741011     & trem & raw      & 241 &    -21 & embryonic\\\\\n",
       "\t -0.3433986 & trem\\_trem\\_embryonic\\_1chrX11369294:11369954    & trem & dlg1     & 112 &    -41 & embryonic\\\\\n",
       "\t -0.4190771 & usp\\_usp-GFP\\_embryonic\\_1chr2L6077596:6077680   & usp  & Kr-h1    &  32 &  -4965 & embryonic\\\\\n",
       "\t -0.3524770 & usp\\_usp-GFP\\_embryonic\\_1chr3L4350313:4350397   & usp  & Cip4     &  17 &  -4858 & embryonic\\\\\n",
       "\t -0.3363832 & usp\\_usp-GFP\\_embryonic\\_1chr3R9886101:9886185   & usp  & Glut4EF  &  86 &  11462 & embryonic\\\\\n",
       "\t -0.3340628 & usp\\_usp-GFP\\_embryonic\\_1chr3L2579965:2580049   & usp  & msn      &  99 &   6247 & embryonic\\\\\n",
       "\t -0.3245632 & usp\\_usp-GFP\\_embryonic\\_1chr3R14690908:14690992 & usp  & Meltrin  &   8 &   6342 & embryonic\\\\\n",
       "\t -0.3061045 & usp\\_usp-GFP\\_embryonic\\_1chr3R19072227:19072311 & usp  & cdi      &  11 &  22001 & embryonic\\\\\n",
       "\t -0.2991443 & usp\\_usp-GFP\\_embryonic\\_1chrX1095265:1095349    & usp  & CG3655   &   3 & -12868 & embryonic\\\\\n",
       "\t -0.2839525 & usp\\_usp-GFP\\_embryonic\\_1chr3R10612564:10612648 & usp  & hth      &  19 &   1670 & embryonic\\\\\n",
       "\t -0.2743099 & usp\\_usp-GFP\\_embryonic\\_1chr3R17680490:17680574 & usp  & alt      & 101 &   -656 & embryonic\\\\\n",
       "\t -0.2715323 & usp\\_usp-GFP\\_embryonic\\_1chr2L1982107:1982191   & usp  & CG7289   & 145 &    -68 & embryonic\\\\\n",
       "\t -0.4658849 & wdn\\_wdn-GFP\\_embryonic\\_1chr3R9891834:9892138   & wdn  & Glut4EF  &  89 &  17305 & embryonic\\\\\n",
       "\t -0.4547252 & wdn\\_wdn-GFP\\_embryonic\\_1chr2L12459952:12460256 & wdn  & bun      &  53 &   1152 & embryonic\\\\\n",
       "\t -0.4382633 & wdn\\_wdn-GFP\\_embryonic\\_1chr3R23915049:23915353 & wdn  & sba      & 244 &   7772 & embryonic\\\\\n",
       "\t -0.4176897 & wdn\\_wdn-GFP\\_embryonic\\_1chrX13320671:13320975  & wdn  & IP3K2    & 188 &   -479 & embryonic\\\\\n",
       "\t -0.4169000 & wdn\\_wdn-GFP\\_embryonic\\_1chrX11370814:11371118  & wdn  & dlg1     & 147 &   1301 & embryonic\\\\\n",
       "\t -0.4019413 & wdn\\_wdn-GFP\\_embryonic\\_1chr3L16160085:16160389 & wdn  & CG5151   &  89 &   4464 & embryonic\\\\\n",
       "\t -0.3951208 & wdn\\_wdn-GFP\\_embryonic\\_1chrX11621940:11622244  & wdn  & Ptp10D   & 257 &     77 & embryonic\\\\\n",
       "\t -0.3945413 & wdn\\_wdn-GFP\\_embryonic\\_1chrX13273482:13273786  & wdn  & HDAC4    &  53 &   -890 & embryonic\\\\\n",
       "\t -0.3861245 & wdn\\_wdn-GFP\\_embryonic\\_1chr2L8714063:8714367   & wdn  & raw      &  23 &    368 & embryonic\\\\\n",
       "\t -0.3835956 & wdn\\_wdn-GFP\\_embryonic\\_1chr3L3787899:3788203   & wdn  & CG32264  & 250 &    -85 & embryonic\\\\\n",
       "\t -0.4193265 & wek\\_wek-GFP\\_embryonic\\_1chr2R16609894:16610384 & wek  & CG5065   & 150 &   1011 & embryonic\\\\\n",
       "\t -0.4191661 & wek\\_wek-GFP\\_embryonic\\_1chr2R14267708:14268198 & wek  & CG8547   & 276 &    -34 & embryonic\\\\\n",
       "\t -0.4122494 & wek\\_wek-GFP\\_embryonic\\_1chr2R8065348:8065838   & wek  & CG12769  & 222 &    -39 & embryonic\\\\\n",
       "\t -0.3930019 & wek\\_wek-GFP\\_embryonic\\_1chr2R14769057:14769547 & wek  & Spred    & 211 &    121 & embryonic\\\\\n",
       "\t -0.3699499 & wek\\_wek-GFP\\_embryonic\\_1chr2L17472883:17473373 & wek  & BicD     & 261 &     87 & embryonic\\\\\n",
       "\t -0.3590291 & wek\\_wek-GFP\\_embryonic\\_1chr3R23803764:23803976 & wek  & CG34355  &  97 &     17 & embryonic\\\\\n",
       "\t -0.3573415 & wek\\_wek-GFP\\_embryonic\\_1chr3R20949747:20950237 & wek  & Atpalpha & 193 &   1212 & embryonic\\\\\n",
       "\t -0.3440037 & wek\\_wek-GFP\\_embryonic\\_1chrX11621814:11622304  & wek  & Ptp10D   & 257 &     44 & embryonic\\\\\n",
       "\t -0.3309026 & wek\\_wek-GFP\\_embryonic\\_1chr3R23962858:23963348 & wek  & CG31140  & 166 &    -77 & embryonic\\\\\n",
       "\t -0.3296127 & wek\\_wek-GFP\\_embryonic\\_1chr3L2144231:2144721   & wek  & zormin   &  29 &   7224 & embryonic\\\\\n",
       "\t -0.7890103 & z\\_z-GFP\\_embryonic\\_1chr3R30090355:30090641     & z    & sima     &  76 &  -9011 & embryonic\\\\\n",
       "\t -0.7089874 & z\\_z-GFP\\_embryonic\\_1chr2L12459929:12460325     & z    & bun      &  53 &   1129 & embryonic\\\\\n",
       "\t -0.7062584 & z\\_z-GFP\\_embryonic\\_1chr2R2495616:2496012       & z    & MFS17    &   9 &  69571 & embryonic\\\\\n",
       "\t -0.7013925 & z\\_z-GFP\\_embryonic\\_1chr3R13973796:13974192     & z    & rdx      &   6 &  -2194 & embryonic\\\\\n",
       "\t -0.6471482 & z\\_z-GFP\\_embryonic\\_1chr3L3787819:3788215       & z    & CG32264  & 250 &    -51 & embryonic\\\\\n",
       "\t -0.6462794 & z\\_z-GFP\\_embryonic\\_1chr2R19467149:19467545     & z    & par-1    & 121 &   -119 & embryonic\\\\\n",
       "\t -0.6421429 & z\\_z-GFP\\_embryonic\\_1chr2L2905161:2905557       & z    & lilli    & 241 &  15188 & embryonic\\\\\n",
       "\t -0.6401059 & z\\_z-GFP\\_embryonic\\_1chr3L13226436:13226832     & z    & caps     & 210 &  -2058 & embryonic\\\\\n",
       "\t -0.6387908 & z\\_z-GFP\\_embryonic\\_1chr3L489544:489940         & z    & klar     & 226 &   1081 & embryonic\\\\\n",
       "\t -0.6347419 & z\\_z-GFP\\_embryonic\\_1chrX11397214:11397610      & z    & dlg1     &   4 &   7040 & embryonic\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 660 × 7\n",
       "\n",
       "| CosineAngle &lt;dbl&gt; | peakID &lt;chr&gt; | TF &lt;chr&gt; | Gene &lt;chr&gt; | nPeaks &lt;dbl&gt; | targetDist &lt;dbl&gt; | lifeStage &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| -0.4859642 | ATbp_ATbp-GFP_embryonic_1chr3L440263:440570           | ATbp    | klar      | 202 |    666 | embryonic |\n",
       "| -0.4460530 | ATbp_ATbp-GFP_embryonic_1chrX9349102:9349722          | ATbp    | Zdhhc8    | 247 |   6815 | embryonic |\n",
       "| -0.4363592 | ATbp_ATbp-GFP_embryonic_1chr2L12462886:12463506       | ATbp    | bun       | 139 |  -1940 | embryonic |\n",
       "| -0.4319920 | ATbp_ATbp-GFP_embryonic_1chr2L17387950:17388570       | ATbp    | CLIP-190  |  72 |  -1081 | embryonic |\n",
       "| -0.4313847 | ATbp_ATbp-GFP_embryonic_1chr2L2904931:2905551         | ATbp    | lilli     | 241 |  15070 | embryonic |\n",
       "| -0.4272610 | ATbp_ATbp-GFP_embryonic_1chr2L8097305:8097925         | ATbp    | Bsg       | 234 |   8512 | embryonic |\n",
       "| -0.4179265 | ATbp_ATbp-GFP_embryonic_1chr3R21330422:21331042       | ATbp    | aus       | 237 |  18869 | embryonic |\n",
       "| -0.4082246 | ATbp_ATbp-GFP_embryonic_1chr3R21594255:21594875       | ATbp    | InR       | 178 |  -2444 | embryonic |\n",
       "| -0.4081062 | ATbp_ATbp-GFP_embryonic_1chrX3704290:3704910          | ATbp    | Rala      | 221 |  15035 | embryonic |\n",
       "| -0.4033101 | ATbp_ATbp-GFP_embryonic_1chr3L16056116:16056736       | ATbp    | Mbs       |  94 |   4467 | embryonic |\n",
       "| -0.3826688 | Asciz_CG14962-GFP_embryonic_1chr3L3788576:3789276     | Asciz   | CG32264   | 209 |   -960 | embryonic |\n",
       "| -0.3761955 | Asciz_CG14962-GFP_embryonic_1chrX16089526:16090226    | Asciz   | Tob       |  97 |   -509 | embryonic |\n",
       "| -0.3599308 | Asciz_CG14962-GFP_embryonic_1chr3R29040410:29041110   | Asciz   | Pkc98E    | 202 |    930 | embryonic |\n",
       "| -0.3580836 | Asciz_CG14962-GFP_embryonic_1chr3L440061:440761       | Asciz   | klar      | 202 |    685 | embryonic |\n",
       "| -0.3543654 | Asciz_CG14962-GFP_embryonic_1chr2L17450647:17451347   | Asciz   | dl        | 265 |   -633 | embryonic |\n",
       "| -0.3377280 | Asciz_CG14962-GFP_embryonic_1chr3R13999598:13999855   | Asciz   | rdx       | 245 | -16369 | embryonic |\n",
       "| -0.3366867 | Asciz_CG14962-GFP_embryonic_1chrX9651844:9652544      | Asciz   | Ptpmeg2   | 246 |  -1418 | embryonic |\n",
       "| -0.3358785 | Asciz_CG14962-GFP_embryonic_1chr2L12462809:12463509   | Asciz   | bun       | 139 |  -1903 | embryonic |\n",
       "| -0.3293083 | Asciz_CG14962-GFP_embryonic_1chr2L7940487:7941187     | Asciz   | Snoo      | 153 |  36228 | embryonic |\n",
       "| -0.3292932 | Asciz_CG14962-GFP_embryonic_1chr41053626:1054326      | Asciz   | CaMKII    | 158 |   -273 | embryonic |\n",
       "| -0.5697019 | BtbVII_BtbVII-GFP_embryonic_1chr3L16969432:16970216   | BtbVII  | Exn       | 195 |   -189 | embryonic |\n",
       "| -0.5331014 | BtbVII_BtbVII-GFP_embryonic_1chr2R14265400:14266184   | BtbVII  | CG8547    |  98 |   1705 | embryonic |\n",
       "| -0.4981587 | BtbVII_BtbVII-GFP_embryonic_1chrX1844900:1845684      | BtbVII  | prage     | 219 |   1946 | embryonic |\n",
       "| -0.4905457 | BtbVII_BtbVII-GFP_embryonic_1chr2L6082679:6083463     | BtbVII  | Kr-h1     | 261 |    468 | embryonic |\n",
       "| -0.4903502 | BtbVII_BtbVII-GFP_embryonic_1chr3L4288414:4289198     | BtbVII  | Hexo1     | 188 |   -139 | embryonic |\n",
       "| -0.4841757 | BtbVII_BtbVII-GFP_embryonic_1chr2L22143962:22144746   | BtbVII  | CG42748   | 115 |   4233 | embryonic |\n",
       "| -0.4795594 | BtbVII_BtbVII-GFP_embryonic_1chr3L8065636:8066420     | BtbVII  | Sarm      | 199 |   2154 | embryonic |\n",
       "| -0.4736077 | BtbVII_BtbVII-GFP_embryonic_1chr3L11278232:11279016   | BtbVII  | Dyro      | 149 |  16164 | embryonic |\n",
       "| -0.4559693 | BtbVII_BtbVII-GFP_embryonic_1chr2L6784686:6785470     | BtbVII  | nrv1      |  21 |   1783 | embryonic |\n",
       "| -0.4428619 | BtbVII_BtbVII-GFP_embryonic_1chr2R7727321:7728105     | BtbVII  | Gdap2     | 109 |   -203 | embryonic |\n",
       "| -0.3728905 | CG10209_CG10209-GFP_embryonic_1chrX20503070:20503674  | CG10209 | RhoGAP19D | 260 |   7322 | embryonic |\n",
       "| -0.3683510 | CG10209_CG10209-GFP_embryonic_1chrX9349105:9349709    | CG10209 | Zdhhc8    | 247 |   6820 | embryonic |\n",
       "| -0.3430521 | CG10209_CG10209-GFP_embryonic_1chrX15296796:15297400  | CG10209 | gce       | 229 |   -668 | embryonic |\n",
       "| -0.3416768 | CG10209_CG10209-GFP_embryonic_1chr3R21594264:21594868 | CG10209 | InR       | 178 |  -2445 | embryonic |\n",
       "| -0.3412808 | CG10209_CG10209-GFP_embryonic_1chr2R13908824:13909428 | CG10209 | shot      |  54 |   1820 | embryonic |\n",
       "| -0.3347060 | CG10209_CG10209-GFP_embryonic_1chr2L12465613:12466217 | CG10209 | bun       | 118 |  -4659 | embryonic |\n",
       "| -0.3313975 | CG10209_CG10209-GFP_embryonic_1chr3L3518769:3519373   | CG10209 | Eip63E    | 161 |   3320 | embryonic |\n",
       "| -0.3303142 | CG10209_CG10209-GFP_embryonic_1chr2L2898142:2898746   | CG10209 | lilli     |  19 |   8273 | embryonic |\n",
       "| -0.3216505 | CG10209_CG10209-GFP_embryonic_1chr3L15591075:15591679 | CG10209 | RhoGAP71E | 162 |   2517 | embryonic |\n",
       "| -0.3194778 | CG10209_CG10209-GFP_embryonic_1chrX3739121:3739725    | CG10209 | Tlk       | 164 |   -474 | embryonic |\n",
       "| -0.4938688 | CG10274_CG10274-GFP_embryonic_1chr3R30062492:30062820 | CG10274 | sima      |  76 |   -202 | embryonic |\n",
       "| -0.4644055 | CG10274_CG10274-GFP_embryonic_1chr2L3100748:3101312   | CG10274 | toc       | 178 |  -6335 | embryonic |\n",
       "| -0.4460244 | CG10274_CG10274-GFP_embryonic_1chr3L16060760:16061324 | CG10274 | Mbs       | 219 |   9083 | embryonic |\n",
       "| -0.4396431 | CG10274_CG10274-GFP_embryonic_1chr3R14019803:14020367 | CG10274 | rdx       | 101 |   5877 | embryonic |\n",
       "| -0.4242030 | CG10274_CG10274-GFP_embryonic_1chr2L20884493:20884691 | CG10274 | sky       |  76 |   1397 | embryonic |\n",
       "| -0.4192172 | CG10274_CG10274-GFP_embryonic_1chrX20503146:20503710  | CG10274 | RhoGAP19D | 260 |   7266 | embryonic |\n",
       "| -0.4012545 | CG10274_CG10274-GFP_embryonic_1chr3R26540382:26540946 | CG10274 | scrib     | 101 |   4314 | embryonic |\n",
       "| -0.3874745 | CG10274_CG10274-GFP_embryonic_1chrX13325261:13325825  | CG10274 | IP3K2     |  49 |   4241 | embryonic |\n",
       "| -0.3828380 | CG10274_CG10274-GFP_embryonic_1chr2R8614890:8615454   | CG10274 | Cirl      | 226 |    450 | embryonic |\n",
       "| -0.3798026 | CG10274_CG10274-GFP_embryonic_1chr2R17285195:17285759 | CG10274 | mbl       |   4 |  10067 | embryonic |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| -0.4686251 | trem_trem_embryonic_1chr2L7983862:7984522     | trem | Snoo     | 127 |      4 | embryonic |\n",
       "| -0.4267233 | trem_trem_embryonic_1chr2R2848173:2848833     | trem | CG41378  |  66 |  -1795 | embryonic |\n",
       "| -0.4036791 | trem_trem_embryonic_1chr2L17473149:17473497   | trem | BicD     | 261 |   -112 | embryonic |\n",
       "| -0.3981968 | trem_trem_embryonic_1chrX7323717:7324074      | trem | Atg5     | 233 |   -140 | embryonic |\n",
       "| -0.3709291 | trem_trem_embryonic_1chr3L27278901:27279561   | trem | Dbp80    |  58 |    -85 | embryonic |\n",
       "| -0.3562638 | trem_trem_embryonic_1chr3R21594135:21594795   | trem | InR      | 178 |  -2344 | embryonic |\n",
       "| -0.3542753 | trem_trem_embryonic_1chr3R14025931:14026591   | trem | rdx      | 196 |   -299 | embryonic |\n",
       "| -0.3517334 | trem_trem_embryonic_1chr2L12487144:12487804   | trem | bun      | 149 |  -3885 | embryonic |\n",
       "| -0.3470893 | trem_trem_embryonic_1chr2L8740351:8741011     | trem | raw      | 241 |    -21 | embryonic |\n",
       "| -0.3433986 | trem_trem_embryonic_1chrX11369294:11369954    | trem | dlg1     | 112 |    -41 | embryonic |\n",
       "| -0.4190771 | usp_usp-GFP_embryonic_1chr2L6077596:6077680   | usp  | Kr-h1    |  32 |  -4965 | embryonic |\n",
       "| -0.3524770 | usp_usp-GFP_embryonic_1chr3L4350313:4350397   | usp  | Cip4     |  17 |  -4858 | embryonic |\n",
       "| -0.3363832 | usp_usp-GFP_embryonic_1chr3R9886101:9886185   | usp  | Glut4EF  |  86 |  11462 | embryonic |\n",
       "| -0.3340628 | usp_usp-GFP_embryonic_1chr3L2579965:2580049   | usp  | msn      |  99 |   6247 | embryonic |\n",
       "| -0.3245632 | usp_usp-GFP_embryonic_1chr3R14690908:14690992 | usp  | Meltrin  |   8 |   6342 | embryonic |\n",
       "| -0.3061045 | usp_usp-GFP_embryonic_1chr3R19072227:19072311 | usp  | cdi      |  11 |  22001 | embryonic |\n",
       "| -0.2991443 | usp_usp-GFP_embryonic_1chrX1095265:1095349    | usp  | CG3655   |   3 | -12868 | embryonic |\n",
       "| -0.2839525 | usp_usp-GFP_embryonic_1chr3R10612564:10612648 | usp  | hth      |  19 |   1670 | embryonic |\n",
       "| -0.2743099 | usp_usp-GFP_embryonic_1chr3R17680490:17680574 | usp  | alt      | 101 |   -656 | embryonic |\n",
       "| -0.2715323 | usp_usp-GFP_embryonic_1chr2L1982107:1982191   | usp  | CG7289   | 145 |    -68 | embryonic |\n",
       "| -0.4658849 | wdn_wdn-GFP_embryonic_1chr3R9891834:9892138   | wdn  | Glut4EF  |  89 |  17305 | embryonic |\n",
       "| -0.4547252 | wdn_wdn-GFP_embryonic_1chr2L12459952:12460256 | wdn  | bun      |  53 |   1152 | embryonic |\n",
       "| -0.4382633 | wdn_wdn-GFP_embryonic_1chr3R23915049:23915353 | wdn  | sba      | 244 |   7772 | embryonic |\n",
       "| -0.4176897 | wdn_wdn-GFP_embryonic_1chrX13320671:13320975  | wdn  | IP3K2    | 188 |   -479 | embryonic |\n",
       "| -0.4169000 | wdn_wdn-GFP_embryonic_1chrX11370814:11371118  | wdn  | dlg1     | 147 |   1301 | embryonic |\n",
       "| -0.4019413 | wdn_wdn-GFP_embryonic_1chr3L16160085:16160389 | wdn  | CG5151   |  89 |   4464 | embryonic |\n",
       "| -0.3951208 | wdn_wdn-GFP_embryonic_1chrX11621940:11622244  | wdn  | Ptp10D   | 257 |     77 | embryonic |\n",
       "| -0.3945413 | wdn_wdn-GFP_embryonic_1chrX13273482:13273786  | wdn  | HDAC4    |  53 |   -890 | embryonic |\n",
       "| -0.3861245 | wdn_wdn-GFP_embryonic_1chr2L8714063:8714367   | wdn  | raw      |  23 |    368 | embryonic |\n",
       "| -0.3835956 | wdn_wdn-GFP_embryonic_1chr3L3787899:3788203   | wdn  | CG32264  | 250 |    -85 | embryonic |\n",
       "| -0.4193265 | wek_wek-GFP_embryonic_1chr2R16609894:16610384 | wek  | CG5065   | 150 |   1011 | embryonic |\n",
       "| -0.4191661 | wek_wek-GFP_embryonic_1chr2R14267708:14268198 | wek  | CG8547   | 276 |    -34 | embryonic |\n",
       "| -0.4122494 | wek_wek-GFP_embryonic_1chr2R8065348:8065838   | wek  | CG12769  | 222 |    -39 | embryonic |\n",
       "| -0.3930019 | wek_wek-GFP_embryonic_1chr2R14769057:14769547 | wek  | Spred    | 211 |    121 | embryonic |\n",
       "| -0.3699499 | wek_wek-GFP_embryonic_1chr2L17472883:17473373 | wek  | BicD     | 261 |     87 | embryonic |\n",
       "| -0.3590291 | wek_wek-GFP_embryonic_1chr3R23803764:23803976 | wek  | CG34355  |  97 |     17 | embryonic |\n",
       "| -0.3573415 | wek_wek-GFP_embryonic_1chr3R20949747:20950237 | wek  | Atpalpha | 193 |   1212 | embryonic |\n",
       "| -0.3440037 | wek_wek-GFP_embryonic_1chrX11621814:11622304  | wek  | Ptp10D   | 257 |     44 | embryonic |\n",
       "| -0.3309026 | wek_wek-GFP_embryonic_1chr3R23962858:23963348 | wek  | CG31140  | 166 |    -77 | embryonic |\n",
       "| -0.3296127 | wek_wek-GFP_embryonic_1chr3L2144231:2144721   | wek  | zormin   |  29 |   7224 | embryonic |\n",
       "| -0.7890103 | z_z-GFP_embryonic_1chr3R30090355:30090641     | z    | sima     |  76 |  -9011 | embryonic |\n",
       "| -0.7089874 | z_z-GFP_embryonic_1chr2L12459929:12460325     | z    | bun      |  53 |   1129 | embryonic |\n",
       "| -0.7062584 | z_z-GFP_embryonic_1chr2R2495616:2496012       | z    | MFS17    |   9 |  69571 | embryonic |\n",
       "| -0.7013925 | z_z-GFP_embryonic_1chr3R13973796:13974192     | z    | rdx      |   6 |  -2194 | embryonic |\n",
       "| -0.6471482 | z_z-GFP_embryonic_1chr3L3787819:3788215       | z    | CG32264  | 250 |    -51 | embryonic |\n",
       "| -0.6462794 | z_z-GFP_embryonic_1chr2R19467149:19467545     | z    | par-1    | 121 |   -119 | embryonic |\n",
       "| -0.6421429 | z_z-GFP_embryonic_1chr2L2905161:2905557       | z    | lilli    | 241 |  15188 | embryonic |\n",
       "| -0.6401059 | z_z-GFP_embryonic_1chr3L13226436:13226832     | z    | caps     | 210 |  -2058 | embryonic |\n",
       "| -0.6387908 | z_z-GFP_embryonic_1chr3L489544:489940         | z    | klar     | 226 |   1081 | embryonic |\n",
       "| -0.6347419 | z_z-GFP_embryonic_1chrX11397214:11397610      | z    | dlg1     |   4 |   7040 | embryonic |\n",
       "\n"
      ],
      "text/plain": [
       "    CosineAngle peakID                                                TF     \n",
       "1   -0.4859642  ATbp_ATbp-GFP_embryonic_1chr3L440263:440570           ATbp   \n",
       "2   -0.4460530  ATbp_ATbp-GFP_embryonic_1chrX9349102:9349722          ATbp   \n",
       "3   -0.4363592  ATbp_ATbp-GFP_embryonic_1chr2L12462886:12463506       ATbp   \n",
       "4   -0.4319920  ATbp_ATbp-GFP_embryonic_1chr2L17387950:17388570       ATbp   \n",
       "5   -0.4313847  ATbp_ATbp-GFP_embryonic_1chr2L2904931:2905551         ATbp   \n",
       "6   -0.4272610  ATbp_ATbp-GFP_embryonic_1chr2L8097305:8097925         ATbp   \n",
       "7   -0.4179265  ATbp_ATbp-GFP_embryonic_1chr3R21330422:21331042       ATbp   \n",
       "8   -0.4082246  ATbp_ATbp-GFP_embryonic_1chr3R21594255:21594875       ATbp   \n",
       "9   -0.4081062  ATbp_ATbp-GFP_embryonic_1chrX3704290:3704910          ATbp   \n",
       "10  -0.4033101  ATbp_ATbp-GFP_embryonic_1chr3L16056116:16056736       ATbp   \n",
       "11  -0.3826688  Asciz_CG14962-GFP_embryonic_1chr3L3788576:3789276     Asciz  \n",
       "12  -0.3761955  Asciz_CG14962-GFP_embryonic_1chrX16089526:16090226    Asciz  \n",
       "13  -0.3599308  Asciz_CG14962-GFP_embryonic_1chr3R29040410:29041110   Asciz  \n",
       "14  -0.3580836  Asciz_CG14962-GFP_embryonic_1chr3L440061:440761       Asciz  \n",
       "15  -0.3543654  Asciz_CG14962-GFP_embryonic_1chr2L17450647:17451347   Asciz  \n",
       "16  -0.3377280  Asciz_CG14962-GFP_embryonic_1chr3R13999598:13999855   Asciz  \n",
       "17  -0.3366867  Asciz_CG14962-GFP_embryonic_1chrX9651844:9652544      Asciz  \n",
       "18  -0.3358785  Asciz_CG14962-GFP_embryonic_1chr2L12462809:12463509   Asciz  \n",
       "19  -0.3293083  Asciz_CG14962-GFP_embryonic_1chr2L7940487:7941187     Asciz  \n",
       "20  -0.3292932  Asciz_CG14962-GFP_embryonic_1chr41053626:1054326      Asciz  \n",
       "21  -0.5697019  BtbVII_BtbVII-GFP_embryonic_1chr3L16969432:16970216   BtbVII \n",
       "22  -0.5331014  BtbVII_BtbVII-GFP_embryonic_1chr2R14265400:14266184   BtbVII \n",
       "23  -0.4981587  BtbVII_BtbVII-GFP_embryonic_1chrX1844900:1845684      BtbVII \n",
       "24  -0.4905457  BtbVII_BtbVII-GFP_embryonic_1chr2L6082679:6083463     BtbVII \n",
       "25  -0.4903502  BtbVII_BtbVII-GFP_embryonic_1chr3L4288414:4289198     BtbVII \n",
       "26  -0.4841757  BtbVII_BtbVII-GFP_embryonic_1chr2L22143962:22144746   BtbVII \n",
       "27  -0.4795594  BtbVII_BtbVII-GFP_embryonic_1chr3L8065636:8066420     BtbVII \n",
       "28  -0.4736077  BtbVII_BtbVII-GFP_embryonic_1chr3L11278232:11279016   BtbVII \n",
       "29  -0.4559693  BtbVII_BtbVII-GFP_embryonic_1chr2L6784686:6785470     BtbVII \n",
       "30  -0.4428619  BtbVII_BtbVII-GFP_embryonic_1chr2R7727321:7728105     BtbVII \n",
       "31  -0.3728905  CG10209_CG10209-GFP_embryonic_1chrX20503070:20503674  CG10209\n",
       "32  -0.3683510  CG10209_CG10209-GFP_embryonic_1chrX9349105:9349709    CG10209\n",
       "33  -0.3430521  CG10209_CG10209-GFP_embryonic_1chrX15296796:15297400  CG10209\n",
       "34  -0.3416768  CG10209_CG10209-GFP_embryonic_1chr3R21594264:21594868 CG10209\n",
       "35  -0.3412808  CG10209_CG10209-GFP_embryonic_1chr2R13908824:13909428 CG10209\n",
       "36  -0.3347060  CG10209_CG10209-GFP_embryonic_1chr2L12465613:12466217 CG10209\n",
       "37  -0.3313975  CG10209_CG10209-GFP_embryonic_1chr3L3518769:3519373   CG10209\n",
       "38  -0.3303142  CG10209_CG10209-GFP_embryonic_1chr2L2898142:2898746   CG10209\n",
       "39  -0.3216505  CG10209_CG10209-GFP_embryonic_1chr3L15591075:15591679 CG10209\n",
       "40  -0.3194778  CG10209_CG10209-GFP_embryonic_1chrX3739121:3739725    CG10209\n",
       "41  -0.4938688  CG10274_CG10274-GFP_embryonic_1chr3R30062492:30062820 CG10274\n",
       "42  -0.4644055  CG10274_CG10274-GFP_embryonic_1chr2L3100748:3101312   CG10274\n",
       "43  -0.4460244  CG10274_CG10274-GFP_embryonic_1chr3L16060760:16061324 CG10274\n",
       "44  -0.4396431  CG10274_CG10274-GFP_embryonic_1chr3R14019803:14020367 CG10274\n",
       "45  -0.4242030  CG10274_CG10274-GFP_embryonic_1chr2L20884493:20884691 CG10274\n",
       "46  -0.4192172  CG10274_CG10274-GFP_embryonic_1chrX20503146:20503710  CG10274\n",
       "47  -0.4012545  CG10274_CG10274-GFP_embryonic_1chr3R26540382:26540946 CG10274\n",
       "48  -0.3874745  CG10274_CG10274-GFP_embryonic_1chrX13325261:13325825  CG10274\n",
       "49  -0.3828380  CG10274_CG10274-GFP_embryonic_1chr2R8614890:8615454   CG10274\n",
       "50  -0.3798026  CG10274_CG10274-GFP_embryonic_1chr2R17285195:17285759 CG10274\n",
       "⋮   ⋮           ⋮                                                     ⋮      \n",
       "611 -0.4686251  trem_trem_embryonic_1chr2L7983862:7984522             trem   \n",
       "612 -0.4267233  trem_trem_embryonic_1chr2R2848173:2848833             trem   \n",
       "613 -0.4036791  trem_trem_embryonic_1chr2L17473149:17473497           trem   \n",
       "614 -0.3981968  trem_trem_embryonic_1chrX7323717:7324074              trem   \n",
       "615 -0.3709291  trem_trem_embryonic_1chr3L27278901:27279561           trem   \n",
       "616 -0.3562638  trem_trem_embryonic_1chr3R21594135:21594795           trem   \n",
       "617 -0.3542753  trem_trem_embryonic_1chr3R14025931:14026591           trem   \n",
       "618 -0.3517334  trem_trem_embryonic_1chr2L12487144:12487804           trem   \n",
       "619 -0.3470893  trem_trem_embryonic_1chr2L8740351:8741011             trem   \n",
       "620 -0.3433986  trem_trem_embryonic_1chrX11369294:11369954            trem   \n",
       "621 -0.4190771  usp_usp-GFP_embryonic_1chr2L6077596:6077680           usp    \n",
       "622 -0.3524770  usp_usp-GFP_embryonic_1chr3L4350313:4350397           usp    \n",
       "623 -0.3363832  usp_usp-GFP_embryonic_1chr3R9886101:9886185           usp    \n",
       "624 -0.3340628  usp_usp-GFP_embryonic_1chr3L2579965:2580049           usp    \n",
       "625 -0.3245632  usp_usp-GFP_embryonic_1chr3R14690908:14690992         usp    \n",
       "626 -0.3061045  usp_usp-GFP_embryonic_1chr3R19072227:19072311         usp    \n",
       "627 -0.2991443  usp_usp-GFP_embryonic_1chrX1095265:1095349            usp    \n",
       "628 -0.2839525  usp_usp-GFP_embryonic_1chr3R10612564:10612648         usp    \n",
       "629 -0.2743099  usp_usp-GFP_embryonic_1chr3R17680490:17680574         usp    \n",
       "630 -0.2715323  usp_usp-GFP_embryonic_1chr2L1982107:1982191           usp    \n",
       "631 -0.4658849  wdn_wdn-GFP_embryonic_1chr3R9891834:9892138           wdn    \n",
       "632 -0.4547252  wdn_wdn-GFP_embryonic_1chr2L12459952:12460256         wdn    \n",
       "633 -0.4382633  wdn_wdn-GFP_embryonic_1chr3R23915049:23915353         wdn    \n",
       "634 -0.4176897  wdn_wdn-GFP_embryonic_1chrX13320671:13320975          wdn    \n",
       "635 -0.4169000  wdn_wdn-GFP_embryonic_1chrX11370814:11371118          wdn    \n",
       "636 -0.4019413  wdn_wdn-GFP_embryonic_1chr3L16160085:16160389         wdn    \n",
       "637 -0.3951208  wdn_wdn-GFP_embryonic_1chrX11621940:11622244          wdn    \n",
       "638 -0.3945413  wdn_wdn-GFP_embryonic_1chrX13273482:13273786          wdn    \n",
       "639 -0.3861245  wdn_wdn-GFP_embryonic_1chr2L8714063:8714367           wdn    \n",
       "640 -0.3835956  wdn_wdn-GFP_embryonic_1chr3L3787899:3788203           wdn    \n",
       "641 -0.4193265  wek_wek-GFP_embryonic_1chr2R16609894:16610384         wek    \n",
       "642 -0.4191661  wek_wek-GFP_embryonic_1chr2R14267708:14268198         wek    \n",
       "643 -0.4122494  wek_wek-GFP_embryonic_1chr2R8065348:8065838           wek    \n",
       "644 -0.3930019  wek_wek-GFP_embryonic_1chr2R14769057:14769547         wek    \n",
       "645 -0.3699499  wek_wek-GFP_embryonic_1chr2L17472883:17473373         wek    \n",
       "646 -0.3590291  wek_wek-GFP_embryonic_1chr3R23803764:23803976         wek    \n",
       "647 -0.3573415  wek_wek-GFP_embryonic_1chr3R20949747:20950237         wek    \n",
       "648 -0.3440037  wek_wek-GFP_embryonic_1chrX11621814:11622304          wek    \n",
       "649 -0.3309026  wek_wek-GFP_embryonic_1chr3R23962858:23963348         wek    \n",
       "650 -0.3296127  wek_wek-GFP_embryonic_1chr3L2144231:2144721           wek    \n",
       "651 -0.7890103  z_z-GFP_embryonic_1chr3R30090355:30090641             z      \n",
       "652 -0.7089874  z_z-GFP_embryonic_1chr2L12459929:12460325             z      \n",
       "653 -0.7062584  z_z-GFP_embryonic_1chr2R2495616:2496012               z      \n",
       "654 -0.7013925  z_z-GFP_embryonic_1chr3R13973796:13974192             z      \n",
       "655 -0.6471482  z_z-GFP_embryonic_1chr3L3787819:3788215               z      \n",
       "656 -0.6462794  z_z-GFP_embryonic_1chr2R19467149:19467545             z      \n",
       "657 -0.6421429  z_z-GFP_embryonic_1chr2L2905161:2905557               z      \n",
       "658 -0.6401059  z_z-GFP_embryonic_1chr3L13226436:13226832             z      \n",
       "659 -0.6387908  z_z-GFP_embryonic_1chr3L489544:489940                 z      \n",
       "660 -0.6347419  z_z-GFP_embryonic_1chrX11397214:11397610              z      \n",
       "    Gene      nPeaks targetDist lifeStage\n",
       "1   klar      202       666     embryonic\n",
       "2   Zdhhc8    247      6815     embryonic\n",
       "3   bun       139     -1940     embryonic\n",
       "4   CLIP-190   72     -1081     embryonic\n",
       "5   lilli     241     15070     embryonic\n",
       "6   Bsg       234      8512     embryonic\n",
       "7   aus       237     18869     embryonic\n",
       "8   InR       178     -2444     embryonic\n",
       "9   Rala      221     15035     embryonic\n",
       "10  Mbs        94      4467     embryonic\n",
       "11  CG32264   209      -960     embryonic\n",
       "12  Tob        97      -509     embryonic\n",
       "13  Pkc98E    202       930     embryonic\n",
       "14  klar      202       685     embryonic\n",
       "15  dl        265      -633     embryonic\n",
       "16  rdx       245    -16369     embryonic\n",
       "17  Ptpmeg2   246     -1418     embryonic\n",
       "18  bun       139     -1903     embryonic\n",
       "19  Snoo      153     36228     embryonic\n",
       "20  CaMKII    158      -273     embryonic\n",
       "21  Exn       195      -189     embryonic\n",
       "22  CG8547     98      1705     embryonic\n",
       "23  prage     219      1946     embryonic\n",
       "24  Kr-h1     261       468     embryonic\n",
       "25  Hexo1     188      -139     embryonic\n",
       "26  CG42748   115      4233     embryonic\n",
       "27  Sarm      199      2154     embryonic\n",
       "28  Dyro      149     16164     embryonic\n",
       "29  nrv1       21      1783     embryonic\n",
       "30  Gdap2     109      -203     embryonic\n",
       "31  RhoGAP19D 260      7322     embryonic\n",
       "32  Zdhhc8    247      6820     embryonic\n",
       "33  gce       229      -668     embryonic\n",
       "34  InR       178     -2445     embryonic\n",
       "35  shot       54      1820     embryonic\n",
       "36  bun       118     -4659     embryonic\n",
       "37  Eip63E    161      3320     embryonic\n",
       "38  lilli      19      8273     embryonic\n",
       "39  RhoGAP71E 162      2517     embryonic\n",
       "40  Tlk       164      -474     embryonic\n",
       "41  sima       76      -202     embryonic\n",
       "42  toc       178     -6335     embryonic\n",
       "43  Mbs       219      9083     embryonic\n",
       "44  rdx       101      5877     embryonic\n",
       "45  sky        76      1397     embryonic\n",
       "46  RhoGAP19D 260      7266     embryonic\n",
       "47  scrib     101      4314     embryonic\n",
       "48  IP3K2      49      4241     embryonic\n",
       "49  Cirl      226       450     embryonic\n",
       "50  mbl         4     10067     embryonic\n",
       "⋮   ⋮         ⋮      ⋮          ⋮        \n",
       "611 Snoo      127         4     embryonic\n",
       "612 CG41378    66     -1795     embryonic\n",
       "613 BicD      261      -112     embryonic\n",
       "614 Atg5      233      -140     embryonic\n",
       "615 Dbp80      58       -85     embryonic\n",
       "616 InR       178     -2344     embryonic\n",
       "617 rdx       196      -299     embryonic\n",
       "618 bun       149     -3885     embryonic\n",
       "619 raw       241       -21     embryonic\n",
       "620 dlg1      112       -41     embryonic\n",
       "621 Kr-h1      32     -4965     embryonic\n",
       "622 Cip4       17     -4858     embryonic\n",
       "623 Glut4EF    86     11462     embryonic\n",
       "624 msn        99      6247     embryonic\n",
       "625 Meltrin     8      6342     embryonic\n",
       "626 cdi        11     22001     embryonic\n",
       "627 CG3655      3    -12868     embryonic\n",
       "628 hth        19      1670     embryonic\n",
       "629 alt       101      -656     embryonic\n",
       "630 CG7289    145       -68     embryonic\n",
       "631 Glut4EF    89     17305     embryonic\n",
       "632 bun        53      1152     embryonic\n",
       "633 sba       244      7772     embryonic\n",
       "634 IP3K2     188      -479     embryonic\n",
       "635 dlg1      147      1301     embryonic\n",
       "636 CG5151     89      4464     embryonic\n",
       "637 Ptp10D    257        77     embryonic\n",
       "638 HDAC4      53      -890     embryonic\n",
       "639 raw        23       368     embryonic\n",
       "640 CG32264   250       -85     embryonic\n",
       "641 CG5065    150      1011     embryonic\n",
       "642 CG8547    276       -34     embryonic\n",
       "643 CG12769   222       -39     embryonic\n",
       "644 Spred     211       121     embryonic\n",
       "645 BicD      261        87     embryonic\n",
       "646 CG34355    97        17     embryonic\n",
       "647 Atpalpha  193      1212     embryonic\n",
       "648 Ptp10D    257        44     embryonic\n",
       "649 CG31140   166       -77     embryonic\n",
       "650 zormin     29      7224     embryonic\n",
       "651 sima       76     -9011     embryonic\n",
       "652 bun        53      1129     embryonic\n",
       "653 MFS17       9     69571     embryonic\n",
       "654 rdx         6     -2194     embryonic\n",
       "655 CG32264   250       -51     embryonic\n",
       "656 par-1     121      -119     embryonic\n",
       "657 lilli     241     15188     embryonic\n",
       "658 caps      210     -2058     embryonic\n",
       "659 klar      226      1081     embryonic\n",
       "660 dlg1        4      7040     embryonic"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fly_targets_new_annot %>% arrange(CosineAngle)  %>% group_by(TF) %>% \n",
    "    slice_min(CosineAngle, n = 10)  %>% filter(TF %in% fly_top_TFs) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 673,
   "metadata": {},
   "outputs": [],
   "source": [
    "fly_targets_new_annot_top10 = fly_targets_new_annot %>% arrange(-CosineAngle)  %>% group_by(TF) %>% \n",
    "    slice_max(CosineAngle, n = 10)  %>% filter(TF %in% fly_top_TFs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 676,
   "metadata": {},
   "outputs": [],
   "source": [
    "fly_targets_new_annot_bot10 = fly_targets_new_annot %>% arrange(CosineAngle)  %>% group_by(TF) %>% \n",
    "    slice_min(CosineAngle, n = 10)  %>% filter(TF %in% fly_top_TFs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 677,
   "metadata": {},
   "outputs": [],
   "source": [
    "write_tsv(fly_targets_new_annot_top10, \"data/fly_targets_new_annot_top10.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 678,
   "metadata": {},
   "outputs": [],
   "source": [
    "write_tsv(fly_targets_new_annot_bot10, \"data/fly_targets_new_annot_bot10.tsv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Redo nearest neighbor to get all 20 neighbors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "cells <- colData(cds_sub)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "cellTypes <- cells$annot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "cellIDs <- cells$cell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "pca <- cds_sub@int_colData$reducedDims$PCA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(RANN)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "nnMatches <- function(ids,labels,df,k=10){\n",
    "   k1 <- k + 1  # need one more since always finds itself\n",
    "   rann <- RANN::nn2(df,query=df,k1)\n",
    "   df <- as.data.frame(rann$nn.idx)\n",
    "   df <- mutate(df,ID=ids,.before=1)  # add the ids to the df\n",
    "   df <- mutate(df,Label=labels[V1],.before=2) %>% select(-V1)  # put the labels into the df\n",
    "   df <- pivot_longer(df,cols=3:k1,names_to = \"NN\")\n",
    "   df <- mutate(df,nnLabel=labels[value])  # label the nearest neighbors\n",
    "   df <- select(df,-NN,-value)\n",
    "   res <- group_by(df,ID) %>% summarise(nk=sum (Label==nnLabel)  )  # count the matches in the nearest neighors\n",
    "   return(res)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Do individual steps of Lou's function above"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "k = 20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "k1 <- k + 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pca"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "rann <- RANN::nn2(df,query=df,k1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "saveRDS(rann, \"data/fly_rann_PCA300.rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 492,
   "metadata": {},
   "outputs": [],
   "source": [
    "ids = cellIDs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 493,
   "metadata": {},
   "outputs": [],
   "source": [
    "df <- as.data.frame(rann$nn.idx)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 494,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 21</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V4</th><th scope=col>V5</th><th scope=col>V6</th><th scope=col>V7</th><th scope=col>V8</th><th scope=col>V9</th><th scope=col>V10</th><th scope=col>⋯</th><th scope=col>V12</th><th scope=col>V13</th><th scope=col>V14</th><th scope=col>V15</th><th scope=col>V16</th><th scope=col>V17</th><th scope=col>V18</th><th scope=col>V19</th><th scope=col>V20</th><th scope=col>V21</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>  1084</td><td>117474</td><td> 33297</td><td>  2934</td><td> 2355</td><td>   854</td><td>  2437</td><td>  4221</td><td> 63518</td><td>⋯</td><td>116766</td><td> 48132</td><td> 20116</td><td>115992</td><td> 34544</td><td> 31818</td><td> 50700</td><td>143608</td><td>   571</td><td>  3626</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>2</td><td> 51177</td><td> 19874</td><td> 69268</td><td> 44378</td><td>67519</td><td>212700</td><td>189174</td><td>131569</td><td> 24279</td><td>⋯</td><td> 20273</td><td>130972</td><td>  9391</td><td>  3473</td><td>109102</td><td>132457</td><td>130068</td><td> 46597</td><td> 53857</td><td>183430</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>3</td><td>137166</td><td> 35175</td><td>  4381</td><td>123690</td><td>99621</td><td> 35423</td><td>123344</td><td>108376</td><td>126736</td><td>⋯</td><td>123710</td><td> 47008</td><td>134782</td><td>136306</td><td>137709</td><td>123045</td><td> 33837</td><td> 32261</td><td>136304</td><td>135305</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>4</td><td> 68440</td><td> 71253</td><td> 67238</td><td> 73868</td><td>67236</td><td> 71880</td><td> 64019</td><td> 64022</td><td> 66768</td><td>⋯</td><td> 64612</td><td> 63786</td><td> 70436</td><td> 65552</td><td>  2853</td><td> 65551</td><td> 67168</td><td> 39930</td><td> 28538</td><td> 74114</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>5</td><td>137728</td><td> 19780</td><td>120926</td><td> 91533</td><td> 5730</td><td>144152</td><td>143823</td><td> 35446</td><td> 54161</td><td>⋯</td><td> 51952</td><td>  4980</td><td>134914</td><td> 19964</td><td>132236</td><td> 52185</td><td> 34199</td><td>214925</td><td>117952</td><td> 68538</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>6</td><td> 36849</td><td>  1387</td><td>  2385</td><td> 63108</td><td>36476</td><td>  2887</td><td> 67810</td><td>  3884</td><td> 35089</td><td>⋯</td><td>  5792</td><td> 75285</td><td>  2730</td><td> 19767</td><td> 34789</td><td> 35981</td><td>  1882</td><td> 64621</td><td> 49619</td><td>   781</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 21\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & V1 & V2 & V3 & V4 & V5 & V6 & V7 & V8 & V9 & V10 & ⋯ & V12 & V13 & V14 & V15 & V16 & V17 & V18 & V19 & V20 & V21\\\\\n",
       "  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 1 &   1084 & 117474 &  33297 &   2934 &  2355 &    854 &   2437 &   4221 &  63518 & ⋯ & 116766 &  48132 &  20116 & 115992 &  34544 &  31818 &  50700 & 143608 &    571 &   3626\\\\\n",
       "\t2 & 2 &  51177 &  19874 &  69268 &  44378 & 67519 & 212700 & 189174 & 131569 &  24279 & ⋯ &  20273 & 130972 &   9391 &   3473 & 109102 & 132457 & 130068 &  46597 &  53857 & 183430\\\\\n",
       "\t3 & 3 & 137166 &  35175 &   4381 & 123690 & 99621 &  35423 & 123344 & 108376 & 126736 & ⋯ & 123710 &  47008 & 134782 & 136306 & 137709 & 123045 &  33837 &  32261 & 136304 & 135305\\\\\n",
       "\t4 & 4 &  68440 &  71253 &  67238 &  73868 & 67236 &  71880 &  64019 &  64022 &  66768 & ⋯ &  64612 &  63786 &  70436 &  65552 &   2853 &  65551 &  67168 &  39930 &  28538 &  74114\\\\\n",
       "\t5 & 5 & 137728 &  19780 & 120926 &  91533 &  5730 & 144152 & 143823 &  35446 &  54161 & ⋯ &  51952 &   4980 & 134914 &  19964 & 132236 &  52185 &  34199 & 214925 & 117952 &  68538\\\\\n",
       "\t6 & 6 &  36849 &   1387 &   2385 &  63108 & 36476 &   2887 &  67810 &   3884 &  35089 & ⋯ &   5792 &  75285 &   2730 &  19767 &  34789 &  35981 &   1882 &  64621 &  49619 &    781\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 21\n",
       "\n",
       "| <!--/--> | V1 &lt;int&gt; | V2 &lt;int&gt; | V3 &lt;int&gt; | V4 &lt;int&gt; | V5 &lt;int&gt; | V6 &lt;int&gt; | V7 &lt;int&gt; | V8 &lt;int&gt; | V9 &lt;int&gt; | V10 &lt;int&gt; | ⋯ ⋯ | V12 &lt;int&gt; | V13 &lt;int&gt; | V14 &lt;int&gt; | V15 &lt;int&gt; | V16 &lt;int&gt; | V17 &lt;int&gt; | V18 &lt;int&gt; | V19 &lt;int&gt; | V20 &lt;int&gt; | V21 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 |   1084 | 117474 |  33297 |   2934 |  2355 |    854 |   2437 |   4221 |  63518 | ⋯ | 116766 |  48132 |  20116 | 115992 |  34544 |  31818 |  50700 | 143608 |    571 |   3626 |\n",
       "| 2 | 2 |  51177 |  19874 |  69268 |  44378 | 67519 | 212700 | 189174 | 131569 |  24279 | ⋯ |  20273 | 130972 |   9391 |   3473 | 109102 | 132457 | 130068 |  46597 |  53857 | 183430 |\n",
       "| 3 | 3 | 137166 |  35175 |   4381 | 123690 | 99621 |  35423 | 123344 | 108376 | 126736 | ⋯ | 123710 |  47008 | 134782 | 136306 | 137709 | 123045 |  33837 |  32261 | 136304 | 135305 |\n",
       "| 4 | 4 |  68440 |  71253 |  67238 |  73868 | 67236 |  71880 |  64019 |  64022 |  66768 | ⋯ |  64612 |  63786 |  70436 |  65552 |   2853 |  65551 |  67168 |  39930 |  28538 |  74114 |\n",
       "| 5 | 5 | 137728 |  19780 | 120926 |  91533 |  5730 | 144152 | 143823 |  35446 |  54161 | ⋯ |  51952 |   4980 | 134914 |  19964 | 132236 |  52185 |  34199 | 214925 | 117952 |  68538 |\n",
       "| 6 | 6 |  36849 |   1387 |   2385 |  63108 | 36476 |   2887 |  67810 |   3884 |  35089 | ⋯ |   5792 |  75285 |   2730 |  19767 |  34789 |  35981 |   1882 |  64621 |  49619 |    781 |\n",
       "\n"
      ],
      "text/plain": [
       "  V1 V2     V3     V4     V5     V6    V7     V8     V9     V10    ⋯ V12   \n",
       "1 1    1084 117474  33297   2934  2355    854   2437   4221  63518 ⋯ 116766\n",
       "2 2   51177  19874  69268  44378 67519 212700 189174 131569  24279 ⋯  20273\n",
       "3 3  137166  35175   4381 123690 99621  35423 123344 108376 126736 ⋯ 123710\n",
       "4 4   68440  71253  67238  73868 67236  71880  64019  64022  66768 ⋯  64612\n",
       "5 5  137728  19780 120926  91533  5730 144152 143823  35446  54161 ⋯  51952\n",
       "6 6   36849   1387   2385  63108 36476   2887  67810   3884  35089 ⋯   5792\n",
       "  V13    V14    V15    V16    V17    V18    V19    V20    V21   \n",
       "1  48132  20116 115992  34544  31818  50700 143608    571   3626\n",
       "2 130972   9391   3473 109102 132457 130068  46597  53857 183430\n",
       "3  47008 134782 136306 137709 123045  33837  32261 136304 135305\n",
       "4  63786  70436  65552   2853  65551  67168  39930  28538  74114\n",
       "5   4980 134914  19964 132236  52185  34199 214925 117952  68538\n",
       "6  75285   2730  19767  34789  35981   1882  64621  49619    781"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 495,
   "metadata": {},
   "outputs": [],
   "source": [
    "df <- mutate(df,ID=ids,.before=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 496,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 22</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ID</th><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V4</th><th scope=col>V5</th><th scope=col>V6</th><th scope=col>V7</th><th scope=col>V8</th><th scope=col>V9</th><th scope=col>⋯</th><th scope=col>V12</th><th scope=col>V13</th><th scope=col>V14</th><th scope=col>V15</th><th scope=col>V16</th><th scope=col>V17</th><th scope=col>V18</th><th scope=col>V19</th><th scope=col>V20</th><th scope=col>V21</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>1</td><td>  1084</td><td>117474</td><td> 33297</td><td>  2934</td><td> 2355</td><td>   854</td><td>  2437</td><td>  4221</td><td>⋯</td><td>116766</td><td> 48132</td><td> 20116</td><td>115992</td><td> 34544</td><td> 31818</td><td> 50700</td><td>143608</td><td>   571</td><td>  3626</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG</td><td>2</td><td> 51177</td><td> 19874</td><td> 69268</td><td> 44378</td><td>67519</td><td>212700</td><td>189174</td><td>131569</td><td>⋯</td><td> 20273</td><td>130972</td><td>  9391</td><td>  3473</td><td>109102</td><td>132457</td><td>130068</td><td> 46597</td><td> 53857</td><td>183430</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA</td><td>3</td><td>137166</td><td> 35175</td><td>  4381</td><td>123690</td><td>99621</td><td> 35423</td><td>123344</td><td>108376</td><td>⋯</td><td>123710</td><td> 47008</td><td>134782</td><td>136306</td><td>137709</td><td>123045</td><td> 33837</td><td> 32261</td><td>136304</td><td>135305</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG</td><td>4</td><td> 68440</td><td> 71253</td><td> 67238</td><td> 73868</td><td>67236</td><td> 71880</td><td> 64019</td><td> 64022</td><td>⋯</td><td> 64612</td><td> 63786</td><td> 70436</td><td> 65552</td><td>  2853</td><td> 65551</td><td> 67168</td><td> 39930</td><td> 28538</td><td> 74114</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA</td><td>5</td><td>137728</td><td> 19780</td><td>120926</td><td> 91533</td><td> 5730</td><td>144152</td><td>143823</td><td> 35446</td><td>⋯</td><td> 51952</td><td>  4980</td><td>134914</td><td> 19964</td><td>132236</td><td> 52185</td><td> 34199</td><td>214925</td><td>117952</td><td> 68538</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA</td><td>6</td><td> 36849</td><td>  1387</td><td>  2385</td><td> 63108</td><td>36476</td><td>  2887</td><td> 67810</td><td>  3884</td><td>⋯</td><td>  5792</td><td> 75285</td><td>  2730</td><td> 19767</td><td> 34789</td><td> 35981</td><td>  1882</td><td> 64621</td><td> 49619</td><td>   781</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 22\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & ID & V1 & V2 & V3 & V4 & V5 & V6 & V7 & V8 & V9 & ⋯ & V12 & V13 & V14 & V15 & V16 & V17 & V18 & V19 & V20 & V21\\\\\n",
       "  & <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & 1 &   1084 & 117474 &  33297 &   2934 &  2355 &    854 &   2437 &   4221 & ⋯ & 116766 &  48132 &  20116 & 115992 &  34544 &  31818 &  50700 & 143608 &    571 &   3626\\\\\n",
       "\t2 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG & 2 &  51177 &  19874 &  69268 &  44378 & 67519 & 212700 & 189174 & 131569 & ⋯ &  20273 & 130972 &   9391 &   3473 & 109102 & 132457 & 130068 &  46597 &  53857 & 183430\\\\\n",
       "\t3 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA & 3 & 137166 &  35175 &   4381 & 123690 & 99621 &  35423 & 123344 & 108376 & ⋯ & 123710 &  47008 & 134782 & 136306 & 137709 & 123045 &  33837 &  32261 & 136304 & 135305\\\\\n",
       "\t4 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG & 4 &  68440 &  71253 &  67238 &  73868 & 67236 &  71880 &  64019 &  64022 & ⋯ &  64612 &  63786 &  70436 &  65552 &   2853 &  65551 &  67168 &  39930 &  28538 &  74114\\\\\n",
       "\t5 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA & 5 & 137728 &  19780 & 120926 &  91533 &  5730 & 144152 & 143823 &  35446 & ⋯ &  51952 &   4980 & 134914 &  19964 & 132236 &  52185 &  34199 & 214925 & 117952 &  68538\\\\\n",
       "\t6 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA & 6 &  36849 &   1387 &   2385 &  63108 & 36476 &   2887 &  67810 &   3884 & ⋯ &   5792 &  75285 &   2730 &  19767 &  34789 &  35981 &   1882 &  64621 &  49619 &    781\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 22\n",
       "\n",
       "| <!--/--> | ID &lt;chr&gt; | V1 &lt;int&gt; | V2 &lt;int&gt; | V3 &lt;int&gt; | V4 &lt;int&gt; | V5 &lt;int&gt; | V6 &lt;int&gt; | V7 &lt;int&gt; | V8 &lt;int&gt; | V9 &lt;int&gt; | ⋯ ⋯ | V12 &lt;int&gt; | V13 &lt;int&gt; | V14 &lt;int&gt; | V15 &lt;int&gt; | V16 &lt;int&gt; | V17 &lt;int&gt; | V18 &lt;int&gt; | V19 &lt;int&gt; | V20 &lt;int&gt; | V21 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | 1 |   1084 | 117474 |  33297 |   2934 |  2355 |    854 |   2437 |   4221 | ⋯ | 116766 |  48132 |  20116 | 115992 |  34544 |  31818 |  50700 | 143608 |    571 |   3626 |\n",
       "| 2 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG | 2 |  51177 |  19874 |  69268 |  44378 | 67519 | 212700 | 189174 | 131569 | ⋯ |  20273 | 130972 |   9391 |   3473 | 109102 | 132457 | 130068 |  46597 |  53857 | 183430 |\n",
       "| 3 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA | 3 | 137166 |  35175 |   4381 | 123690 | 99621 |  35423 | 123344 | 108376 | ⋯ | 123710 |  47008 | 134782 | 136306 | 137709 | 123045 |  33837 |  32261 | 136304 | 135305 |\n",
       "| 4 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG | 4 |  68440 |  71253 |  67238 |  73868 | 67236 |  71880 |  64019 |  64022 | ⋯ |  64612 |  63786 |  70436 |  65552 |   2853 |  65551 |  67168 |  39930 |  28538 |  74114 |\n",
       "| 5 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA | 5 | 137728 |  19780 | 120926 |  91533 |  5730 | 144152 | 143823 |  35446 | ⋯ |  51952 |   4980 | 134914 |  19964 | 132236 |  52185 |  34199 | 214925 | 117952 |  68538 |\n",
       "| 6 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA | 6 |  36849 |   1387 |   2385 |  63108 | 36476 |   2887 |  67810 |   3884 | ⋯ |   5792 |  75285 |   2730 |  19767 |  34789 |  35981 |   1882 |  64621 |  49619 |    781 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID                                                     V1 V2     V3    \n",
       "1 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 1    1084 117474\n",
       "2 exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 2   51177  19874\n",
       "3 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 3  137166  35175\n",
       "4 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 4   68440  71253\n",
       "5 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 5  137728  19780\n",
       "6 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 6   36849   1387\n",
       "  V4     V5     V6    V7     V8     V9     ⋯ V12    V13    V14    V15    V16   \n",
       "1  33297   2934  2355    854   2437   4221 ⋯ 116766  48132  20116 115992  34544\n",
       "2  69268  44378 67519 212700 189174 131569 ⋯  20273 130972   9391   3473 109102\n",
       "3   4381 123690 99621  35423 123344 108376 ⋯ 123710  47008 134782 136306 137709\n",
       "4  67238  73868 67236  71880  64019  64022 ⋯  64612  63786  70436  65552   2853\n",
       "5 120926  91533  5730 144152 143823  35446 ⋯  51952   4980 134914  19964 132236\n",
       "6   2385  63108 36476   2887  67810   3884 ⋯   5792  75285   2730  19767  34789\n",
       "  V17    V18    V19    V20    V21   \n",
       "1  31818  50700 143608    571   3626\n",
       "2 132457 130068  46597  53857 183430\n",
       "3 123045  33837  32261 136304 135305\n",
       "4  65551  67168  39930  28538  74114\n",
       "5  52185  34199 214925 117952  68538\n",
       "6  35981   1882  64621  49619    781"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 497,
   "metadata": {},
   "outputs": [],
   "source": [
    "labels = cells$annot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 498,
   "metadata": {},
   "outputs": [],
   "source": [
    "df <- mutate(df,Label=labels[V1],.before=2) %>% select(-V1) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 499,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 22</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ID</th><th scope=col>Label</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V4</th><th scope=col>V5</th><th scope=col>V6</th><th scope=col>V7</th><th scope=col>V8</th><th scope=col>V9</th><th scope=col>⋯</th><th scope=col>V12</th><th scope=col>V13</th><th scope=col>V14</th><th scope=col>V15</th><th scope=col>V16</th><th scope=col>V17</th><th scope=col>V18</th><th scope=col>V19</th><th scope=col>V20</th><th scope=col>V21</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>  1084</td><td>117474</td><td> 33297</td><td>  2934</td><td> 2355</td><td>   854</td><td>  2437</td><td>  4221</td><td>⋯</td><td>116766</td><td> 48132</td><td> 20116</td><td>115992</td><td> 34544</td><td> 31818</td><td> 50700</td><td>143608</td><td>   571</td><td>  3626</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG</td><td>visceral muscle primordium       </td><td> 51177</td><td> 19874</td><td> 69268</td><td> 44378</td><td>67519</td><td>212700</td><td>189174</td><td>131569</td><td>⋯</td><td> 20273</td><td>130972</td><td>  9391</td><td>  3473</td><td>109102</td><td>132457</td><td>130068</td><td> 46597</td><td> 53857</td><td>183430</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA</td><td>brain                            </td><td>137166</td><td> 35175</td><td>  4381</td><td>123690</td><td>99621</td><td> 35423</td><td>123344</td><td>108376</td><td>⋯</td><td>123710</td><td> 47008</td><td>134782</td><td>136306</td><td>137709</td><td>123045</td><td> 33837</td><td> 32261</td><td>136304</td><td>135305</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG</td><td>epidermis                        </td><td> 68440</td><td> 71253</td><td> 67238</td><td> 73868</td><td>67236</td><td> 71880</td><td> 64019</td><td> 64022</td><td>⋯</td><td> 64612</td><td> 63786</td><td> 70436</td><td> 65552</td><td>  2853</td><td> 65551</td><td> 67168</td><td> 39930</td><td> 28538</td><td> 74114</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA</td><td>sensory nervous system           </td><td>137728</td><td> 19780</td><td>120926</td><td> 91533</td><td> 5730</td><td>144152</td><td>143823</td><td> 35446</td><td>⋯</td><td> 51952</td><td>  4980</td><td>134914</td><td> 19964</td><td>132236</td><td> 52185</td><td> 34199</td><td>214925</td><td>117952</td><td> 68538</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA</td><td>sensory nervous system           </td><td> 36849</td><td>  1387</td><td>  2385</td><td> 63108</td><td>36476</td><td>  2887</td><td> 67810</td><td>  3884</td><td>⋯</td><td>  5792</td><td> 75285</td><td>  2730</td><td> 19767</td><td> 34789</td><td> 35981</td><td>  1882</td><td> 64621</td><td> 49619</td><td>   781</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 22\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & ID & Label & V2 & V3 & V4 & V5 & V6 & V7 & V8 & V9 & ⋯ & V12 & V13 & V14 & V15 & V16 & V17 & V18 & V19 & V20 & V21\\\\\n",
       "  & <chr> & <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium &   1084 & 117474 &  33297 &   2934 &  2355 &    854 &   2437 &   4221 & ⋯ & 116766 &  48132 &  20116 & 115992 &  34544 &  31818 &  50700 & 143608 &    571 &   3626\\\\\n",
       "\t2 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG & visceral muscle primordium        &  51177 &  19874 &  69268 &  44378 & 67519 & 212700 & 189174 & 131569 & ⋯ &  20273 & 130972 &   9391 &   3473 & 109102 & 132457 & 130068 &  46597 &  53857 & 183430\\\\\n",
       "\t3 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA & brain                             & 137166 &  35175 &   4381 & 123690 & 99621 &  35423 & 123344 & 108376 & ⋯ & 123710 &  47008 & 134782 & 136306 & 137709 & 123045 &  33837 &  32261 & 136304 & 135305\\\\\n",
       "\t4 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG & epidermis                         &  68440 &  71253 &  67238 &  73868 & 67236 &  71880 &  64019 &  64022 & ⋯ &  64612 &  63786 &  70436 &  65552 &   2853 &  65551 &  67168 &  39930 &  28538 &  74114\\\\\n",
       "\t5 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA & sensory nervous system            & 137728 &  19780 & 120926 &  91533 &  5730 & 144152 & 143823 &  35446 & ⋯ &  51952 &   4980 & 134914 &  19964 & 132236 &  52185 &  34199 & 214925 & 117952 &  68538\\\\\n",
       "\t6 & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA & sensory nervous system            &  36849 &   1387 &   2385 &  63108 & 36476 &   2887 &  67810 &   3884 & ⋯ &   5792 &  75285 &   2730 &  19767 &  34789 &  35981 &   1882 &  64621 &  49619 &    781\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 22\n",
       "\n",
       "| <!--/--> | ID &lt;chr&gt; | Label &lt;chr&gt; | V2 &lt;int&gt; | V3 &lt;int&gt; | V4 &lt;int&gt; | V5 &lt;int&gt; | V6 &lt;int&gt; | V7 &lt;int&gt; | V8 &lt;int&gt; | V9 &lt;int&gt; | ⋯ ⋯ | V12 &lt;int&gt; | V13 &lt;int&gt; | V14 &lt;int&gt; | V15 &lt;int&gt; | V16 &lt;int&gt; | V17 &lt;int&gt; | V18 &lt;int&gt; | V19 &lt;int&gt; | V20 &lt;int&gt; | V21 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium |   1084 | 117474 |  33297 |   2934 |  2355 |    854 |   2437 |   4221 | ⋯ | 116766 |  48132 |  20116 | 115992 |  34544 |  31818 |  50700 | 143608 |    571 |   3626 |\n",
       "| 2 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG | visceral muscle primordium        |  51177 |  19874 |  69268 |  44378 | 67519 | 212700 | 189174 | 131569 | ⋯ |  20273 | 130972 |   9391 |   3473 | 109102 | 132457 | 130068 |  46597 |  53857 | 183430 |\n",
       "| 3 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA | brain                             | 137166 |  35175 |   4381 | 123690 | 99621 |  35423 | 123344 | 108376 | ⋯ | 123710 |  47008 | 134782 | 136306 | 137709 | 123045 |  33837 |  32261 | 136304 | 135305 |\n",
       "| 4 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG | epidermis                         |  68440 |  71253 |  67238 |  73868 | 67236 |  71880 |  64019 |  64022 | ⋯ |  64612 |  63786 |  70436 |  65552 |   2853 |  65551 |  67168 |  39930 |  28538 |  74114 |\n",
       "| 5 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA | sensory nervous system            | 137728 |  19780 | 120926 |  91533 |  5730 | 144152 | 143823 |  35446 | ⋯ |  51952 |   4980 | 134914 |  19964 | 132236 |  52185 |  34199 | 214925 | 117952 |  68538 |\n",
       "| 6 | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA | sensory nervous system            |  36849 |   1387 |   2385 |  63108 | 36476 |   2887 |  67810 |   3884 | ⋯ |   5792 |  75285 |   2730 |  19767 |  34789 |  35981 |   1882 |  64621 |  49619 |    781 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID                                                    \n",
       "1 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "2 exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG\n",
       "3 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA\n",
       "4 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG\n",
       "5 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA\n",
       "6 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA\n",
       "  Label                             V2     V3     V4     V5     V6    V7    \n",
       "1 sensory nervous system primordium   1084 117474  33297   2934  2355    854\n",
       "2 visceral muscle primordium         51177  19874  69268  44378 67519 212700\n",
       "3 brain                             137166  35175   4381 123690 99621  35423\n",
       "4 epidermis                          68440  71253  67238  73868 67236  71880\n",
       "5 sensory nervous system            137728  19780 120926  91533  5730 144152\n",
       "6 sensory nervous system             36849   1387   2385  63108 36476   2887\n",
       "  V8     V9     ⋯ V12    V13    V14    V15    V16    V17    V18    V19   \n",
       "1   2437   4221 ⋯ 116766  48132  20116 115992  34544  31818  50700 143608\n",
       "2 189174 131569 ⋯  20273 130972   9391   3473 109102 132457 130068  46597\n",
       "3 123344 108376 ⋯ 123710  47008 134782 136306 137709 123045  33837  32261\n",
       "4  64019  64022 ⋯  64612  63786  70436  65552   2853  65551  67168  39930\n",
       "5 143823  35446 ⋯  51952   4980 134914  19964 132236  52185  34199 214925\n",
       "6  67810   3884 ⋯   5792  75285   2730  19767  34789  35981   1882  64621\n",
       "  V20    V21   \n",
       "1    571   3626\n",
       "2  53857 183430\n",
       "3 136304 135305\n",
       "4  28538  74114\n",
       "5 117952  68538\n",
       "6  49619    781"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 500,
   "metadata": {},
   "outputs": [],
   "source": [
    "k2 = k1 + 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 501,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“\u001b[1m\u001b[22mUsing an external vector in selections was deprecated in tidyselect 1.1.0.\n",
      "\u001b[36mℹ\u001b[39m Please use `all_of()` or `any_of()` instead.\n",
      "  # Was:\n",
      "  data %>% select(k2)\n",
      "\n",
      "  # Now:\n",
      "  data %>% select(all_of(k2))\n",
      "\n",
      "See <https://tidyselect.r-lib.org/reference/faq-external-vector.html>.”\n"
     ]
    }
   ],
   "source": [
    "df <- pivot_longer(df,cols=3:k2,names_to = \"NN\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 502,
   "metadata": {},
   "outputs": [],
   "source": [
    "df <- mutate(df,nnLabel=labels[value])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 503,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>Label</th><th scope=col>NN</th><th scope=col>value</th><th scope=col>nnLabel</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>V2</td><td>  1084</td><td>plasmatocytes anlage             </td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>V3</td><td>117474</td><td>plasmatocytes anlage             </td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>V4</td><td> 33297</td><td>sensory nervous system primordium</td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>V5</td><td>  2934</td><td>plasmatocytes anlage             </td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>V6</td><td>  2355</td><td>plasmatocytes anlage             </td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>V7</td><td>   854</td><td>plasmatocytes anlage             </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " ID & Label & NN & value & nnLabel\\\\\n",
       " <chr> & <chr> & <chr> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & V2 &   1084 & plasmatocytes anlage             \\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & V3 & 117474 & plasmatocytes anlage             \\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & V4 &  33297 & sensory nervous system primordium\\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & V5 &   2934 & plasmatocytes anlage             \\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & V6 &   2355 & plasmatocytes anlage             \\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & V7 &    854 & plasmatocytes anlage             \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 5\n",
       "\n",
       "| ID &lt;chr&gt; | Label &lt;chr&gt; | NN &lt;chr&gt; | value &lt;int&gt; | nnLabel &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | V2 |   1084 | plasmatocytes anlage              |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | V3 | 117474 | plasmatocytes anlage              |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | V4 |  33297 | sensory nervous system primordium |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | V5 |   2934 | plasmatocytes anlage              |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | V6 |   2355 | plasmatocytes anlage              |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | V7 |    854 | plasmatocytes anlage              |\n",
       "\n"
      ],
      "text/plain": [
       "  ID                                                    \n",
       "1 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "2 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "3 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "4 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "5 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "6 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "  Label                             NN value  nnLabel                          \n",
       "1 sensory nervous system primordium V2   1084 plasmatocytes anlage             \n",
       "2 sensory nervous system primordium V3 117474 plasmatocytes anlage             \n",
       "3 sensory nervous system primordium V4  33297 sensory nervous system primordium\n",
       "4 sensory nervous system primordium V5   2934 plasmatocytes anlage             \n",
       "5 sensory nervous system primordium V6   2355 plasmatocytes anlage             \n",
       "6 sensory nervous system primordium V7    854 plasmatocytes anlage             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df, 6)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 504,
   "metadata": {},
   "outputs": [],
   "source": [
    "df <- select(df,-NN,-value)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 505,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>Label</th><th scope=col>nnLabel</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>plasmatocytes anlage             </td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>plasmatocytes anlage             </td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>sensory nervous system primordium</td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>plasmatocytes anlage             </td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>plasmatocytes anlage             </td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>sensory nervous system primordium</td><td>plasmatocytes anlage             </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 3\n",
       "\\begin{tabular}{lll}\n",
       " ID & Label & nnLabel\\\\\n",
       " <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & plasmatocytes anlage             \\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & plasmatocytes anlage             \\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & sensory nervous system primordium\\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & plasmatocytes anlage             \\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & plasmatocytes anlage             \\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & sensory nervous system primordium & plasmatocytes anlage             \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 3\n",
       "\n",
       "| ID &lt;chr&gt; | Label &lt;chr&gt; | nnLabel &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | plasmatocytes anlage              |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | plasmatocytes anlage              |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | sensory nervous system primordium |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | plasmatocytes anlage              |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | plasmatocytes anlage              |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | sensory nervous system primordium | plasmatocytes anlage              |\n",
       "\n"
      ],
      "text/plain": [
       "  ID                                                    \n",
       "1 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "2 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "3 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "4 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "5 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "6 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "  Label                             nnLabel                          \n",
       "1 sensory nervous system primordium plasmatocytes anlage             \n",
       "2 sensory nervous system primordium plasmatocytes anlage             \n",
       "3 sensory nervous system primordium sensory nervous system primordium\n",
       "4 sensory nervous system primordium plasmatocytes anlage             \n",
       "5 sensory nervous system primordium plasmatocytes anlage             \n",
       "6 sensory nervous system primordium plasmatocytes anlage             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 506,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>4768900</li><li>3</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 4768900\n",
       "\\item 3\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 4768900\n",
       "2. 3\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 4768900       3"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 507,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 50 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Label</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Malpighian tubule                   </td><td>  6759</td></tr>\n",
       "\t<tr><td>amnioserosa                         </td><td>123495</td></tr>\n",
       "\t<tr><td>anlage in statu nascendi            </td><td>    37</td></tr>\n",
       "\t<tr><td>brain                               </td><td>241543</td></tr>\n",
       "\t<tr><td>brain primordium                    </td><td> 55562</td></tr>\n",
       "\t<tr><td>cellular blastoderm / pole cell     </td><td>     7</td></tr>\n",
       "\t<tr><td>crystal cell                        </td><td>   165</td></tr>\n",
       "\t<tr><td>ectoderm anlage in statu nascendi   </td><td>    77</td></tr>\n",
       "\t<tr><td>ectoderm primordium                 </td><td> 10200</td></tr>\n",
       "\t<tr><td>endoderm primordium                 </td><td>  1441</td></tr>\n",
       "\t<tr><td>epidermis                           </td><td>332567</td></tr>\n",
       "\t<tr><td>epidermis primordium                </td><td> 28797</td></tr>\n",
       "\t<tr><td>fat body                            </td><td> 70270</td></tr>\n",
       "\t<tr><td>foregut / hindgut / anal pad        </td><td> 34498</td></tr>\n",
       "\t<tr><td>foregut / hindgut primordium        </td><td>  6712</td></tr>\n",
       "\t<tr><td>germ cell                           </td><td> 10130</td></tr>\n",
       "\t<tr><td>glia                                </td><td>  5314</td></tr>\n",
       "\t<tr><td>hindgut / anal pad                  </td><td> 22097</td></tr>\n",
       "\t<tr><td>hindgut primordium                  </td><td>  8248</td></tr>\n",
       "\t<tr><td>maternal                            </td><td>    42</td></tr>\n",
       "\t<tr><td>mesoderm primordium                 </td><td>  1302</td></tr>\n",
       "\t<tr><td>midgut                              </td><td>107327</td></tr>\n",
       "\t<tr><td>midgut primordium                   </td><td>  9303</td></tr>\n",
       "\t<tr><td>midline primordium                  </td><td>  8101</td></tr>\n",
       "\t<tr><td>muscle                              </td><td> 34927</td></tr>\n",
       "\t<tr><td>peripheral nervous system           </td><td> 23698</td></tr>\n",
       "\t<tr><td>peripheral nervous system primordium</td><td>  2004</td></tr>\n",
       "\t<tr><td>plasmatocytes                       </td><td> 68219</td></tr>\n",
       "\t<tr><td>plasmatocytes anlage                </td><td> 12517</td></tr>\n",
       "\t<tr><td>procrystal cell                     </td><td> 13080</td></tr>\n",
       "\t<tr><td>proventriculus                      </td><td> 17741</td></tr>\n",
       "\t<tr><td>salivary gland                      </td><td> 50492</td></tr>\n",
       "\t<tr><td>salivary gland body primordium      </td><td>  8494</td></tr>\n",
       "\t<tr><td>salivary gland primordium           </td><td>    36</td></tr>\n",
       "\t<tr><td>sensory nervous system              </td><td> 26404</td></tr>\n",
       "\t<tr><td>sensory nervous system primordium   </td><td>  5284</td></tr>\n",
       "\t<tr><td>somatic muscle                      </td><td>176700</td></tr>\n",
       "\t<tr><td>somatic muscle primordium           </td><td>  6801</td></tr>\n",
       "\t<tr><td>spiracle (tracheal system)          </td><td>  2186</td></tr>\n",
       "\t<tr><td>spiracles (tracheal system)         </td><td>  5777</td></tr>\n",
       "\t<tr><td>tracheal primordium                 </td><td> 12204</td></tr>\n",
       "\t<tr><td>tracheal system                     </td><td> 86670</td></tr>\n",
       "\t<tr><td>trunk mesoderm primordium           </td><td>  2310</td></tr>\n",
       "\t<tr><td>ubiquitous                          </td><td>273840</td></tr>\n",
       "\t<tr><td>unknown                             </td><td>  3043</td></tr>\n",
       "\t<tr><td>ventral nerve cord                  </td><td> 73216</td></tr>\n",
       "\t<tr><td>ventral nerve cord primordium       </td><td>  1859</td></tr>\n",
       "\t<tr><td>visceral muscle                     </td><td>  2934</td></tr>\n",
       "\t<tr><td>visceral muscle primordium          </td><td> 15393</td></tr>\n",
       "\t<tr><td>yolk nuclei                         </td><td>   878</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 50 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Label & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Malpighian tubule                    &   6759\\\\\n",
       "\t amnioserosa                          & 123495\\\\\n",
       "\t anlage in statu nascendi             &     37\\\\\n",
       "\t brain                                & 241543\\\\\n",
       "\t brain primordium                     &  55562\\\\\n",
       "\t cellular blastoderm / pole cell      &      7\\\\\n",
       "\t crystal cell                         &    165\\\\\n",
       "\t ectoderm anlage in statu nascendi    &     77\\\\\n",
       "\t ectoderm primordium                  &  10200\\\\\n",
       "\t endoderm primordium                  &   1441\\\\\n",
       "\t epidermis                            & 332567\\\\\n",
       "\t epidermis primordium                 &  28797\\\\\n",
       "\t fat body                             &  70270\\\\\n",
       "\t foregut / hindgut / anal pad         &  34498\\\\\n",
       "\t foregut / hindgut primordium         &   6712\\\\\n",
       "\t germ cell                            &  10130\\\\\n",
       "\t glia                                 &   5314\\\\\n",
       "\t hindgut / anal pad                   &  22097\\\\\n",
       "\t hindgut primordium                   &   8248\\\\\n",
       "\t maternal                             &     42\\\\\n",
       "\t mesoderm primordium                  &   1302\\\\\n",
       "\t midgut                               & 107327\\\\\n",
       "\t midgut primordium                    &   9303\\\\\n",
       "\t midline primordium                   &   8101\\\\\n",
       "\t muscle                               &  34927\\\\\n",
       "\t peripheral nervous system            &  23698\\\\\n",
       "\t peripheral nervous system primordium &   2004\\\\\n",
       "\t plasmatocytes                        &  68219\\\\\n",
       "\t plasmatocytes anlage                 &  12517\\\\\n",
       "\t procrystal cell                      &  13080\\\\\n",
       "\t proventriculus                       &  17741\\\\\n",
       "\t salivary gland                       &  50492\\\\\n",
       "\t salivary gland body primordium       &   8494\\\\\n",
       "\t salivary gland primordium            &     36\\\\\n",
       "\t sensory nervous system               &  26404\\\\\n",
       "\t sensory nervous system primordium    &   5284\\\\\n",
       "\t somatic muscle                       & 176700\\\\\n",
       "\t somatic muscle primordium            &   6801\\\\\n",
       "\t spiracle (tracheal system)           &   2186\\\\\n",
       "\t spiracles (tracheal system)          &   5777\\\\\n",
       "\t tracheal primordium                  &  12204\\\\\n",
       "\t tracheal system                      &  86670\\\\\n",
       "\t trunk mesoderm primordium            &   2310\\\\\n",
       "\t ubiquitous                           & 273840\\\\\n",
       "\t unknown                              &   3043\\\\\n",
       "\t ventral nerve cord                   &  73216\\\\\n",
       "\t ventral nerve cord primordium        &   1859\\\\\n",
       "\t visceral muscle                      &   2934\\\\\n",
       "\t visceral muscle primordium           &  15393\\\\\n",
       "\t yolk nuclei                          &    878\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 50 × 2\n",
       "\n",
       "| Label &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Malpighian tubule                    |   6759 |\n",
       "| amnioserosa                          | 123495 |\n",
       "| anlage in statu nascendi             |     37 |\n",
       "| brain                                | 241543 |\n",
       "| brain primordium                     |  55562 |\n",
       "| cellular blastoderm / pole cell      |      7 |\n",
       "| crystal cell                         |    165 |\n",
       "| ectoderm anlage in statu nascendi    |     77 |\n",
       "| ectoderm primordium                  |  10200 |\n",
       "| endoderm primordium                  |   1441 |\n",
       "| epidermis                            | 332567 |\n",
       "| epidermis primordium                 |  28797 |\n",
       "| fat body                             |  70270 |\n",
       "| foregut / hindgut / anal pad         |  34498 |\n",
       "| foregut / hindgut primordium         |   6712 |\n",
       "| germ cell                            |  10130 |\n",
       "| glia                                 |   5314 |\n",
       "| hindgut / anal pad                   |  22097 |\n",
       "| hindgut primordium                   |   8248 |\n",
       "| maternal                             |     42 |\n",
       "| mesoderm primordium                  |   1302 |\n",
       "| midgut                               | 107327 |\n",
       "| midgut primordium                    |   9303 |\n",
       "| midline primordium                   |   8101 |\n",
       "| muscle                               |  34927 |\n",
       "| peripheral nervous system            |  23698 |\n",
       "| peripheral nervous system primordium |   2004 |\n",
       "| plasmatocytes                        |  68219 |\n",
       "| plasmatocytes anlage                 |  12517 |\n",
       "| procrystal cell                      |  13080 |\n",
       "| proventriculus                       |  17741 |\n",
       "| salivary gland                       |  50492 |\n",
       "| salivary gland body primordium       |   8494 |\n",
       "| salivary gland primordium            |     36 |\n",
       "| sensory nervous system               |  26404 |\n",
       "| sensory nervous system primordium    |   5284 |\n",
       "| somatic muscle                       | 176700 |\n",
       "| somatic muscle primordium            |   6801 |\n",
       "| spiracle (tracheal system)           |   2186 |\n",
       "| spiracles (tracheal system)          |   5777 |\n",
       "| tracheal primordium                  |  12204 |\n",
       "| tracheal system                      |  86670 |\n",
       "| trunk mesoderm primordium            |   2310 |\n",
       "| ubiquitous                           | 273840 |\n",
       "| unknown                              |   3043 |\n",
       "| ventral nerve cord                   |  73216 |\n",
       "| ventral nerve cord primordium        |   1859 |\n",
       "| visceral muscle                      |   2934 |\n",
       "| visceral muscle primordium           |  15393 |\n",
       "| yolk nuclei                          |    878 |\n",
       "\n"
      ],
      "text/plain": [
       "   Label                                n     \n",
       "1  Malpighian tubule                      6759\n",
       "2  amnioserosa                          123495\n",
       "3  anlage in statu nascendi                 37\n",
       "4  brain                                241543\n",
       "5  brain primordium                      55562\n",
       "6  cellular blastoderm / pole cell           7\n",
       "7  crystal cell                            165\n",
       "8  ectoderm anlage in statu nascendi        77\n",
       "9  ectoderm primordium                   10200\n",
       "10 endoderm primordium                    1441\n",
       "11 epidermis                            332567\n",
       "12 epidermis primordium                  28797\n",
       "13 fat body                              70270\n",
       "14 foregut / hindgut / anal pad          34498\n",
       "15 foregut / hindgut primordium           6712\n",
       "16 germ cell                             10130\n",
       "17 glia                                   5314\n",
       "18 hindgut / anal pad                    22097\n",
       "19 hindgut primordium                     8248\n",
       "20 maternal                                 42\n",
       "21 mesoderm primordium                    1302\n",
       "22 midgut                               107327\n",
       "23 midgut primordium                      9303\n",
       "24 midline primordium                     8101\n",
       "25 muscle                                34927\n",
       "26 peripheral nervous system             23698\n",
       "27 peripheral nervous system primordium   2004\n",
       "28 plasmatocytes                         68219\n",
       "29 plasmatocytes anlage                  12517\n",
       "30 procrystal cell                       13080\n",
       "31 proventriculus                        17741\n",
       "32 salivary gland                        50492\n",
       "33 salivary gland body primordium         8494\n",
       "34 salivary gland primordium                36\n",
       "35 sensory nervous system                26404\n",
       "36 sensory nervous system primordium      5284\n",
       "37 somatic muscle                       176700\n",
       "38 somatic muscle primordium              6801\n",
       "39 spiracle (tracheal system)             2186\n",
       "40 spiracles (tracheal system)            5777\n",
       "41 tracheal primordium                   12204\n",
       "42 tracheal system                       86670\n",
       "43 trunk mesoderm primordium              2310\n",
       "44 ubiquitous                           273840\n",
       "45 unknown                                3043\n",
       "46 ventral nerve cord                    73216\n",
       "47 ventral nerve cord primordium          1859\n",
       "48 visceral muscle                        2934\n",
       "49 visceral muscle primordium            15393\n",
       "50 yolk nuclei                             878"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df  %>% filter(Label==nnLabel)  %>% count(Label)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 508,
   "metadata": {},
   "outputs": [],
   "source": [
    "res <- group_by(df,ID) %>% summarise(nk=sum (Label==nnLabel)  )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 509,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>nk</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAACTCCAATCGTCCGTCCT</td><td> 6</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AACTAGTTGTAACGGACCAA</td><td> 5</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATAACTAGGCGC </td><td> 5</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATCCTAGCGCCT </td><td>15</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG </td><td> 1</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td> 0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " ID & nk\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAACTCCAATCGTCCGTCCT &  6\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AACTAGTTGTAACGGACCAA &  5\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGACGATAACTAGGCGC  &  5\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGACGATCCTAGCGCCT  & 15\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG  &  1\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA &  0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 2\n",
       "\n",
       "| ID &lt;chr&gt; | nk &lt;int&gt; |\n",
       "|---|---|\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAACTCCAATCGTCCGTCCT |  6 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AACTAGTTGTAACGGACCAA |  5 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATAACTAGGCGC  |  5 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATCCTAGCGCCT  | 15 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG  |  1 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA |  0 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID                                                       nk\n",
       "1 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAACTCCAATCGTCCGTCCT  6\n",
       "2 exp10_idx1AATACGATAA_idx2AACTACCGAC.AACTAGTTGTAACGGACCAA  5\n",
       "3 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATAACTAGGCGC   5\n",
       "4 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATCCTAGCGCCT  15\n",
       "5 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG   1\n",
       "6 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA  0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(res)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 510,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>238445</li><li>2</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 238445\n",
       "\\item 2\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 238445\n",
       "2. 2\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 238445      2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(res)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 511,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 21 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>nk</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 0</td><td>25996</td></tr>\n",
       "\t<tr><td> 1</td><td>19488</td></tr>\n",
       "\t<tr><td> 2</td><td>16298</td></tr>\n",
       "\t<tr><td> 3</td><td>14433</td></tr>\n",
       "\t<tr><td> 4</td><td>13027</td></tr>\n",
       "\t<tr><td> 5</td><td>11691</td></tr>\n",
       "\t<tr><td> 6</td><td>10684</td></tr>\n",
       "\t<tr><td> 7</td><td>10048</td></tr>\n",
       "\t<tr><td> 8</td><td> 9463</td></tr>\n",
       "\t<tr><td> 9</td><td> 9250</td></tr>\n",
       "\t<tr><td>10</td><td> 9093</td></tr>\n",
       "\t<tr><td>11</td><td> 8652</td></tr>\n",
       "\t<tr><td>12</td><td> 8226</td></tr>\n",
       "\t<tr><td>13</td><td> 7944</td></tr>\n",
       "\t<tr><td>14</td><td> 7596</td></tr>\n",
       "\t<tr><td>15</td><td> 7448</td></tr>\n",
       "\t<tr><td>16</td><td> 7345</td></tr>\n",
       "\t<tr><td>17</td><td> 7416</td></tr>\n",
       "\t<tr><td>18</td><td> 7918</td></tr>\n",
       "\t<tr><td>19</td><td> 9485</td></tr>\n",
       "\t<tr><td>20</td><td>16944</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 21 × 2\n",
       "\\begin{tabular}{ll}\n",
       " nk & n\\\\\n",
       " <int> & <int>\\\\\n",
       "\\hline\n",
       "\t  0 & 25996\\\\\n",
       "\t  1 & 19488\\\\\n",
       "\t  2 & 16298\\\\\n",
       "\t  3 & 14433\\\\\n",
       "\t  4 & 13027\\\\\n",
       "\t  5 & 11691\\\\\n",
       "\t  6 & 10684\\\\\n",
       "\t  7 & 10048\\\\\n",
       "\t  8 &  9463\\\\\n",
       "\t  9 &  9250\\\\\n",
       "\t 10 &  9093\\\\\n",
       "\t 11 &  8652\\\\\n",
       "\t 12 &  8226\\\\\n",
       "\t 13 &  7944\\\\\n",
       "\t 14 &  7596\\\\\n",
       "\t 15 &  7448\\\\\n",
       "\t 16 &  7345\\\\\n",
       "\t 17 &  7416\\\\\n",
       "\t 18 &  7918\\\\\n",
       "\t 19 &  9485\\\\\n",
       "\t 20 & 16944\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 21 × 2\n",
       "\n",
       "| nk &lt;int&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "|  0 | 25996 |\n",
       "|  1 | 19488 |\n",
       "|  2 | 16298 |\n",
       "|  3 | 14433 |\n",
       "|  4 | 13027 |\n",
       "|  5 | 11691 |\n",
       "|  6 | 10684 |\n",
       "|  7 | 10048 |\n",
       "|  8 |  9463 |\n",
       "|  9 |  9250 |\n",
       "| 10 |  9093 |\n",
       "| 11 |  8652 |\n",
       "| 12 |  8226 |\n",
       "| 13 |  7944 |\n",
       "| 14 |  7596 |\n",
       "| 15 |  7448 |\n",
       "| 16 |  7345 |\n",
       "| 17 |  7416 |\n",
       "| 18 |  7918 |\n",
       "| 19 |  9485 |\n",
       "| 20 | 16944 |\n",
       "\n"
      ],
      "text/plain": [
       "   nk n    \n",
       "1   0 25996\n",
       "2   1 19488\n",
       "3   2 16298\n",
       "4   3 14433\n",
       "5   4 13027\n",
       "6   5 11691\n",
       "7   6 10684\n",
       "8   7 10048\n",
       "9   8  9463\n",
       "10  9  9250\n",
       "11 10  9093\n",
       "12 11  8652\n",
       "13 12  8226\n",
       "14 13  7944\n",
       "15 14  7596\n",
       "16 15  7448\n",
       "17 16  7345\n",
       "18 17  7416\n",
       "19 18  7918\n",
       "20 19  9485\n",
       "21 20 16944"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "res  %>% count(nk)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 512,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>nk</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG</td><td>1</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.CAAGCAACTGCTCTCGCCT</td><td>1</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.CAGCAAGTTGCCGCAACGA</td><td>1</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.CTGAGCATTACTTAACCTT</td><td>1</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.CTGAGCATTTATATACTAA</td><td>1</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.GACCAGGATCCTAGCGCCT</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " ID & nk\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG & 1\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.CAAGCAACTGCTCTCGCCT & 1\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.CAGCAAGTTGCCGCAACGA & 1\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.CTGAGCATTACTTAACCTT & 1\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.CTGAGCATTTATATACTAA & 1\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.GACCAGGATCCTAGCGCCT & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 2\n",
       "\n",
       "| ID &lt;chr&gt; | nk &lt;int&gt; |\n",
       "|---|---|\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG | 1 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.CAAGCAACTGCTCTCGCCT | 1 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.CAGCAAGTTGCCGCAACGA | 1 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.CTGAGCATTACTTAACCTT | 1 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.CTGAGCATTTATATACTAA | 1 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.GACCAGGATCCTAGCGCCT | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID                                                      nk\n",
       "1 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG 1 \n",
       "2 exp10_idx1AATACGATAA_idx2AACTACCGAC.CAAGCAACTGCTCTCGCCT 1 \n",
       "3 exp10_idx1AATACGATAA_idx2AACTACCGAC.CAGCAAGTTGCCGCAACGA 1 \n",
       "4 exp10_idx1AATACGATAA_idx2AACTACCGAC.CTGAGCATTACTTAACCTT 1 \n",
       "5 exp10_idx1AATACGATAA_idx2AACTACCGAC.CTGAGCATTTATATACTAA 1 \n",
       "6 exp10_idx1AATACGATAA_idx2AACTACCGAC.GACCAGGATCCTAGCGCCT 1 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "res %>% filter(nk == 1)  %>% head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 513,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 20 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>Label</th><th scope=col>nnLabel</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>midgut</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>midgut</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>brain </td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>midgut</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle</td><td>midgut</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 20 × 3\n",
       "\\begin{tabular}{lll}\n",
       " ID & Label & nnLabel\\\\\n",
       " <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & midgut\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & midgut\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & brain \\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & midgut\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle & midgut\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 20 × 3\n",
       "\n",
       "| ID &lt;chr&gt; | Label &lt;chr&gt; | nnLabel &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | midgut |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | midgut |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | brain  |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | midgut |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle | midgut |\n",
       "\n"
      ],
      "text/plain": [
       "   ID                                                       Label  nnLabel\n",
       "1  exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle brain  \n",
       "2  exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle brain  \n",
       "3  exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle brain  \n",
       "4  exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle brain  \n",
       "5  exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle brain  \n",
       "6  exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle brain  \n",
       "7  exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle midgut \n",
       "8  exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle brain  \n",
       "9  exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle brain  \n",
       "10 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle brain  \n",
       "11 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle brain  \n",
       "12 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle midgut \n",
       "13 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle brain  \n",
       "14 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle brain  \n",
       "15 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle brain  \n",
       "16 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle brain  \n",
       "17 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle brain  \n",
       "18 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle brain  \n",
       "19 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle midgut \n",
       "20 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle midgut "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df  %>% filter(ID == \"exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 514,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'ID'. You can override using the `.groups` argument.\n"
     ]
    }
   ],
   "source": [
    "res_label <- group_by(df,ID, Label) %>% summarise(nk=sum (Label==nnLabel)  )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 515,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>Label</th><th scope=col>nk</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAACTCCAATCGTCCGTCCT</td><td>ubiquitous</td><td> 6</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AACTAGTTGTAACGGACCAA</td><td>midgut    </td><td> 5</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATAACTAGGCGC </td><td>muscle    </td><td> 5</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATCCTAGCGCCT </td><td>brain     </td><td>15</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG </td><td>epidermis </td><td> 1</td></tr>\n",
       "\t<tr><td>exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA</td><td>muscle    </td><td> 0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 6 × 3\n",
       "\\begin{tabular}{lll}\n",
       " ID & Label & nk\\\\\n",
       " <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAACTCCAATCGTCCGTCCT & ubiquitous &  6\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AACTAGTTGTAACGGACCAA & midgut     &  5\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGACGATAACTAGGCGC  & muscle     &  5\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGACGATCCTAGCGCCT  & brain      & 15\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG  & epidermis  &  1\\\\\n",
       "\t exp10\\_idx1AATACGATAA\\_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA & muscle     &  0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 6 × 3\n",
       "\n",
       "| ID &lt;chr&gt; | Label &lt;chr&gt; | nk &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAACTCCAATCGTCCGTCCT | ubiquitous |  6 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AACTAGTTGTAACGGACCAA | midgut     |  5 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATAACTAGGCGC  | muscle     |  5 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATCCTAGCGCCT  | brain      | 15 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG  | epidermis  |  1 |\n",
       "| exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA | muscle     |  0 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID                                                       Label      nk\n",
       "1 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAACTCCAATCGTCCGTCCT ubiquitous  6\n",
       "2 exp10_idx1AATACGATAA_idx2AACTACCGAC.AACTAGTTGTAACGGACCAA midgut      5\n",
       "3 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATAACTAGGCGC  muscle      5\n",
       "4 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATCCTAGCGCCT  brain      15\n",
       "5 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGACGATGGTCCAGGAG  epidermis   1\n",
       "6 exp10_idx1AATACGATAA_idx2AACTACCGAC.AAGGATATGTCTGATTAAGA muscle      0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(res_label)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 516,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 51 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Label</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Malpighian tubule                   </td><td> 248</td></tr>\n",
       "\t<tr><td>amnioserosa                         </td><td> 473</td></tr>\n",
       "\t<tr><td>anlage in statu nascendi            </td><td>  20</td></tr>\n",
       "\t<tr><td>brain                               </td><td> 864</td></tr>\n",
       "\t<tr><td>brain primordium                    </td><td> 841</td></tr>\n",
       "\t<tr><td>cellular blastoderm / pole cell     </td><td>  12</td></tr>\n",
       "\t<tr><td>crystal cell                        </td><td> 109</td></tr>\n",
       "\t<tr><td>ectoderm anlage                     </td><td>   5</td></tr>\n",
       "\t<tr><td>ectoderm anlage in statu nascendi   </td><td>  18</td></tr>\n",
       "\t<tr><td>ectoderm primordium                 </td><td> 142</td></tr>\n",
       "\t<tr><td>endoderm primordium                 </td><td>  99</td></tr>\n",
       "\t<tr><td>epidermis                           </td><td>2420</td></tr>\n",
       "\t<tr><td>epidermis primordium                </td><td> 697</td></tr>\n",
       "\t<tr><td>fat body                            </td><td> 466</td></tr>\n",
       "\t<tr><td>foregut / hindgut / anal pad        </td><td>1751</td></tr>\n",
       "\t<tr><td>foregut / hindgut primordium        </td><td> 217</td></tr>\n",
       "\t<tr><td>germ cell                           </td><td> 277</td></tr>\n",
       "\t<tr><td>glia                                </td><td> 232</td></tr>\n",
       "\t<tr><td>hindgut / anal pad                  </td><td> 367</td></tr>\n",
       "\t<tr><td>hindgut primordium                  </td><td> 380</td></tr>\n",
       "\t<tr><td>maternal                            </td><td>   5</td></tr>\n",
       "\t<tr><td>mesoderm primordium                 </td><td> 100</td></tr>\n",
       "\t<tr><td>midgut                              </td><td>1197</td></tr>\n",
       "\t<tr><td>midgut primordium                   </td><td> 404</td></tr>\n",
       "\t<tr><td>midline primordium                  </td><td> 226</td></tr>\n",
       "\t<tr><td>muscle                              </td><td>1524</td></tr>\n",
       "\t<tr><td>peripheral nervous system           </td><td> 386</td></tr>\n",
       "\t<tr><td>peripheral nervous system primordium</td><td> 135</td></tr>\n",
       "\t<tr><td>plasmatocytes                       </td><td> 676</td></tr>\n",
       "\t<tr><td>plasmatocytes anlage                </td><td> 245</td></tr>\n",
       "\t<tr><td>procrystal cell                     </td><td> 241</td></tr>\n",
       "\t<tr><td>proventriculus                      </td><td> 389</td></tr>\n",
       "\t<tr><td>salivary gland                      </td><td> 242</td></tr>\n",
       "\t<tr><td>salivary gland body primordium      </td><td> 139</td></tr>\n",
       "\t<tr><td>salivary gland primordium           </td><td>  93</td></tr>\n",
       "\t<tr><td>sensory nervous system              </td><td>1066</td></tr>\n",
       "\t<tr><td>sensory nervous system primordium   </td><td> 368</td></tr>\n",
       "\t<tr><td>somatic muscle                      </td><td> 968</td></tr>\n",
       "\t<tr><td>somatic muscle primordium           </td><td> 372</td></tr>\n",
       "\t<tr><td>spiracle (tracheal system)          </td><td> 336</td></tr>\n",
       "\t<tr><td>spiracles (tracheal system)         </td><td> 134</td></tr>\n",
       "\t<tr><td>tracheal primordium                 </td><td> 514</td></tr>\n",
       "\t<tr><td>tracheal system                     </td><td>1115</td></tr>\n",
       "\t<tr><td>trunk mesoderm primordium           </td><td> 302</td></tr>\n",
       "\t<tr><td>ubiquitous                          </td><td>2658</td></tr>\n",
       "\t<tr><td>unknown                             </td><td> 552</td></tr>\n",
       "\t<tr><td>ventral nerve cord                  </td><td> 461</td></tr>\n",
       "\t<tr><td>ventral nerve cord primordium       </td><td> 142</td></tr>\n",
       "\t<tr><td>visceral muscle                     </td><td> 282</td></tr>\n",
       "\t<tr><td>visceral muscle primordium          </td><td> 624</td></tr>\n",
       "\t<tr><td>yolk nuclei                         </td><td> 462</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 51 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Label & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Malpighian tubule                    &  248\\\\\n",
       "\t amnioserosa                          &  473\\\\\n",
       "\t anlage in statu nascendi             &   20\\\\\n",
       "\t brain                                &  864\\\\\n",
       "\t brain primordium                     &  841\\\\\n",
       "\t cellular blastoderm / pole cell      &   12\\\\\n",
       "\t crystal cell                         &  109\\\\\n",
       "\t ectoderm anlage                      &    5\\\\\n",
       "\t ectoderm anlage in statu nascendi    &   18\\\\\n",
       "\t ectoderm primordium                  &  142\\\\\n",
       "\t endoderm primordium                  &   99\\\\\n",
       "\t epidermis                            & 2420\\\\\n",
       "\t epidermis primordium                 &  697\\\\\n",
       "\t fat body                             &  466\\\\\n",
       "\t foregut / hindgut / anal pad         & 1751\\\\\n",
       "\t foregut / hindgut primordium         &  217\\\\\n",
       "\t germ cell                            &  277\\\\\n",
       "\t glia                                 &  232\\\\\n",
       "\t hindgut / anal pad                   &  367\\\\\n",
       "\t hindgut primordium                   &  380\\\\\n",
       "\t maternal                             &    5\\\\\n",
       "\t mesoderm primordium                  &  100\\\\\n",
       "\t midgut                               & 1197\\\\\n",
       "\t midgut primordium                    &  404\\\\\n",
       "\t midline primordium                   &  226\\\\\n",
       "\t muscle                               & 1524\\\\\n",
       "\t peripheral nervous system            &  386\\\\\n",
       "\t peripheral nervous system primordium &  135\\\\\n",
       "\t plasmatocytes                        &  676\\\\\n",
       "\t plasmatocytes anlage                 &  245\\\\\n",
       "\t procrystal cell                      &  241\\\\\n",
       "\t proventriculus                       &  389\\\\\n",
       "\t salivary gland                       &  242\\\\\n",
       "\t salivary gland body primordium       &  139\\\\\n",
       "\t salivary gland primordium            &   93\\\\\n",
       "\t sensory nervous system               & 1066\\\\\n",
       "\t sensory nervous system primordium    &  368\\\\\n",
       "\t somatic muscle                       &  968\\\\\n",
       "\t somatic muscle primordium            &  372\\\\\n",
       "\t spiracle (tracheal system)           &  336\\\\\n",
       "\t spiracles (tracheal system)          &  134\\\\\n",
       "\t tracheal primordium                  &  514\\\\\n",
       "\t tracheal system                      & 1115\\\\\n",
       "\t trunk mesoderm primordium            &  302\\\\\n",
       "\t ubiquitous                           & 2658\\\\\n",
       "\t unknown                              &  552\\\\\n",
       "\t ventral nerve cord                   &  461\\\\\n",
       "\t ventral nerve cord primordium        &  142\\\\\n",
       "\t visceral muscle                      &  282\\\\\n",
       "\t visceral muscle primordium           &  624\\\\\n",
       "\t yolk nuclei                          &  462\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 51 × 2\n",
       "\n",
       "| Label &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Malpighian tubule                    |  248 |\n",
       "| amnioserosa                          |  473 |\n",
       "| anlage in statu nascendi             |   20 |\n",
       "| brain                                |  864 |\n",
       "| brain primordium                     |  841 |\n",
       "| cellular blastoderm / pole cell      |   12 |\n",
       "| crystal cell                         |  109 |\n",
       "| ectoderm anlage                      |    5 |\n",
       "| ectoderm anlage in statu nascendi    |   18 |\n",
       "| ectoderm primordium                  |  142 |\n",
       "| endoderm primordium                  |   99 |\n",
       "| epidermis                            | 2420 |\n",
       "| epidermis primordium                 |  697 |\n",
       "| fat body                             |  466 |\n",
       "| foregut / hindgut / anal pad         | 1751 |\n",
       "| foregut / hindgut primordium         |  217 |\n",
       "| germ cell                            |  277 |\n",
       "| glia                                 |  232 |\n",
       "| hindgut / anal pad                   |  367 |\n",
       "| hindgut primordium                   |  380 |\n",
       "| maternal                             |    5 |\n",
       "| mesoderm primordium                  |  100 |\n",
       "| midgut                               | 1197 |\n",
       "| midgut primordium                    |  404 |\n",
       "| midline primordium                   |  226 |\n",
       "| muscle                               | 1524 |\n",
       "| peripheral nervous system            |  386 |\n",
       "| peripheral nervous system primordium |  135 |\n",
       "| plasmatocytes                        |  676 |\n",
       "| plasmatocytes anlage                 |  245 |\n",
       "| procrystal cell                      |  241 |\n",
       "| proventriculus                       |  389 |\n",
       "| salivary gland                       |  242 |\n",
       "| salivary gland body primordium       |  139 |\n",
       "| salivary gland primordium            |   93 |\n",
       "| sensory nervous system               | 1066 |\n",
       "| sensory nervous system primordium    |  368 |\n",
       "| somatic muscle                       |  968 |\n",
       "| somatic muscle primordium            |  372 |\n",
       "| spiracle (tracheal system)           |  336 |\n",
       "| spiracles (tracheal system)          |  134 |\n",
       "| tracheal primordium                  |  514 |\n",
       "| tracheal system                      | 1115 |\n",
       "| trunk mesoderm primordium            |  302 |\n",
       "| ubiquitous                           | 2658 |\n",
       "| unknown                              |  552 |\n",
       "| ventral nerve cord                   |  461 |\n",
       "| ventral nerve cord primordium        |  142 |\n",
       "| visceral muscle                      |  282 |\n",
       "| visceral muscle primordium           |  624 |\n",
       "| yolk nuclei                          |  462 |\n",
       "\n"
      ],
      "text/plain": [
       "   Label                                n   \n",
       "1  Malpighian tubule                     248\n",
       "2  amnioserosa                           473\n",
       "3  anlage in statu nascendi               20\n",
       "4  brain                                 864\n",
       "5  brain primordium                      841\n",
       "6  cellular blastoderm / pole cell        12\n",
       "7  crystal cell                          109\n",
       "8  ectoderm anlage                         5\n",
       "9  ectoderm anlage in statu nascendi      18\n",
       "10 ectoderm primordium                   142\n",
       "11 endoderm primordium                    99\n",
       "12 epidermis                            2420\n",
       "13 epidermis primordium                  697\n",
       "14 fat body                              466\n",
       "15 foregut / hindgut / anal pad         1751\n",
       "16 foregut / hindgut primordium          217\n",
       "17 germ cell                             277\n",
       "18 glia                                  232\n",
       "19 hindgut / anal pad                    367\n",
       "20 hindgut primordium                    380\n",
       "21 maternal                                5\n",
       "22 mesoderm primordium                   100\n",
       "23 midgut                               1197\n",
       "24 midgut primordium                     404\n",
       "25 midline primordium                    226\n",
       "26 muscle                               1524\n",
       "27 peripheral nervous system             386\n",
       "28 peripheral nervous system primordium  135\n",
       "29 plasmatocytes                         676\n",
       "30 plasmatocytes anlage                  245\n",
       "31 procrystal cell                       241\n",
       "32 proventriculus                        389\n",
       "33 salivary gland                        242\n",
       "34 salivary gland body primordium        139\n",
       "35 salivary gland primordium              93\n",
       "36 sensory nervous system               1066\n",
       "37 sensory nervous system primordium     368\n",
       "38 somatic muscle                        968\n",
       "39 somatic muscle primordium             372\n",
       "40 spiracle (tracheal system)            336\n",
       "41 spiracles (tracheal system)           134\n",
       "42 tracheal primordium                   514\n",
       "43 tracheal system                      1115\n",
       "44 trunk mesoderm primordium             302\n",
       "45 ubiquitous                           2658\n",
       "46 unknown                               552\n",
       "47 ventral nerve cord                    461\n",
       "48 ventral nerve cord primordium         142\n",
       "49 visceral muscle                       282\n",
       "50 visceral muscle primordium            624\n",
       "51 yolk nuclei                           462"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "res_label  %>% filter(nk == 0)  %>% ungroup  %>% count(Label)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 51 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>annot</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Malpighian tubule                   </td><td> 1090</td></tr>\n",
       "\t<tr><td>amnioserosa                         </td><td> 7610</td></tr>\n",
       "\t<tr><td>anlage in statu nascendi            </td><td>   40</td></tr>\n",
       "\t<tr><td>brain                               </td><td>26617</td></tr>\n",
       "\t<tr><td>brain primordium                    </td><td> 9816</td></tr>\n",
       "\t<tr><td>cellular blastoderm / pole cell     </td><td>   16</td></tr>\n",
       "\t<tr><td>crystal cell                        </td><td>  177</td></tr>\n",
       "\t<tr><td>ectoderm anlage                     </td><td>    5</td></tr>\n",
       "\t<tr><td>ectoderm anlage in statu nascendi   </td><td>   51</td></tr>\n",
       "\t<tr><td>ectoderm primordium                 </td><td> 1601</td></tr>\n",
       "\t<tr><td>endoderm primordium                 </td><td>  400</td></tr>\n",
       "\t<tr><td>epidermis                           </td><td>36935</td></tr>\n",
       "\t<tr><td>epidermis primordium                </td><td> 6976</td></tr>\n",
       "\t<tr><td>fat body                            </td><td> 5550</td></tr>\n",
       "\t<tr><td>foregut / hindgut / anal pad        </td><td> 7035</td></tr>\n",
       "\t<tr><td>foregut / hindgut primordium        </td><td> 1218</td></tr>\n",
       "\t<tr><td>germ cell                           </td><td> 1309</td></tr>\n",
       "\t<tr><td>glia                                </td><td>  899</td></tr>\n",
       "\t<tr><td>hindgut / anal pad                  </td><td> 2662</td></tr>\n",
       "\t<tr><td>hindgut primordium                  </td><td> 1916</td></tr>\n",
       "\t<tr><td>maternal                            </td><td>   17</td></tr>\n",
       "\t<tr><td>mesoderm primordium                 </td><td>  384</td></tr>\n",
       "\t<tr><td>midgut                              </td><td>11649</td></tr>\n",
       "\t<tr><td>midgut primordium                   </td><td> 1848</td></tr>\n",
       "\t<tr><td>midline primordium                  </td><td>  982</td></tr>\n",
       "\t<tr><td>muscle                              </td><td> 6184</td></tr>\n",
       "\t<tr><td>peripheral nervous system           </td><td> 3591</td></tr>\n",
       "\t<tr><td>peripheral nervous system primordium</td><td>  559</td></tr>\n",
       "\t<tr><td>plasmatocytes                       </td><td> 5575</td></tr>\n",
       "\t<tr><td>plasmatocytes anlage                </td><td> 1658</td></tr>\n",
       "\t<tr><td>procrystal cell                     </td><td> 1556</td></tr>\n",
       "\t<tr><td>proventriculus                      </td><td> 2480</td></tr>\n",
       "\t<tr><td>salivary gland                      </td><td> 3487</td></tr>\n",
       "\t<tr><td>salivary gland body primordium      </td><td> 1042</td></tr>\n",
       "\t<tr><td>salivary gland primordium           </td><td>  115</td></tr>\n",
       "\t<tr><td>sensory nervous system              </td><td> 4816</td></tr>\n",
       "\t<tr><td>sensory nervous system primordium   </td><td> 1627</td></tr>\n",
       "\t<tr><td>somatic muscle                      </td><td>14715</td></tr>\n",
       "\t<tr><td>somatic muscle primordium           </td><td> 1674</td></tr>\n",
       "\t<tr><td>spiracle (tracheal system)          </td><td>  986</td></tr>\n",
       "\t<tr><td>spiracles (tracheal system)         </td><td>  751</td></tr>\n",
       "\t<tr><td>tracheal primordium                 </td><td> 2526</td></tr>\n",
       "\t<tr><td>tracheal system                     </td><td> 8052</td></tr>\n",
       "\t<tr><td>trunk mesoderm primordium           </td><td>  880</td></tr>\n",
       "\t<tr><td>ubiquitous                          </td><td>34589</td></tr>\n",
       "\t<tr><td>unknown                             </td><td> 1444</td></tr>\n",
       "\t<tr><td>ventral nerve cord                  </td><td> 7542</td></tr>\n",
       "\t<tr><td>ventral nerve cord primordium       </td><td>  671</td></tr>\n",
       "\t<tr><td>visceral muscle                     </td><td>  799</td></tr>\n",
       "\t<tr><td>visceral muscle primordium          </td><td> 3546</td></tr>\n",
       "\t<tr><td>yolk nuclei                         </td><td>  777</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 51 × 2\n",
       "\\begin{tabular}{ll}\n",
       " annot & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Malpighian tubule                    &  1090\\\\\n",
       "\t amnioserosa                          &  7610\\\\\n",
       "\t anlage in statu nascendi             &    40\\\\\n",
       "\t brain                                & 26617\\\\\n",
       "\t brain primordium                     &  9816\\\\\n",
       "\t cellular blastoderm / pole cell      &    16\\\\\n",
       "\t crystal cell                         &   177\\\\\n",
       "\t ectoderm anlage                      &     5\\\\\n",
       "\t ectoderm anlage in statu nascendi    &    51\\\\\n",
       "\t ectoderm primordium                  &  1601\\\\\n",
       "\t endoderm primordium                  &   400\\\\\n",
       "\t epidermis                            & 36935\\\\\n",
       "\t epidermis primordium                 &  6976\\\\\n",
       "\t fat body                             &  5550\\\\\n",
       "\t foregut / hindgut / anal pad         &  7035\\\\\n",
       "\t foregut / hindgut primordium         &  1218\\\\\n",
       "\t germ cell                            &  1309\\\\\n",
       "\t glia                                 &   899\\\\\n",
       "\t hindgut / anal pad                   &  2662\\\\\n",
       "\t hindgut primordium                   &  1916\\\\\n",
       "\t maternal                             &    17\\\\\n",
       "\t mesoderm primordium                  &   384\\\\\n",
       "\t midgut                               & 11649\\\\\n",
       "\t midgut primordium                    &  1848\\\\\n",
       "\t midline primordium                   &   982\\\\\n",
       "\t muscle                               &  6184\\\\\n",
       "\t peripheral nervous system            &  3591\\\\\n",
       "\t peripheral nervous system primordium &   559\\\\\n",
       "\t plasmatocytes                        &  5575\\\\\n",
       "\t plasmatocytes anlage                 &  1658\\\\\n",
       "\t procrystal cell                      &  1556\\\\\n",
       "\t proventriculus                       &  2480\\\\\n",
       "\t salivary gland                       &  3487\\\\\n",
       "\t salivary gland body primordium       &  1042\\\\\n",
       "\t salivary gland primordium            &   115\\\\\n",
       "\t sensory nervous system               &  4816\\\\\n",
       "\t sensory nervous system primordium    &  1627\\\\\n",
       "\t somatic muscle                       & 14715\\\\\n",
       "\t somatic muscle primordium            &  1674\\\\\n",
       "\t spiracle (tracheal system)           &   986\\\\\n",
       "\t spiracles (tracheal system)          &   751\\\\\n",
       "\t tracheal primordium                  &  2526\\\\\n",
       "\t tracheal system                      &  8052\\\\\n",
       "\t trunk mesoderm primordium            &   880\\\\\n",
       "\t ubiquitous                           & 34589\\\\\n",
       "\t unknown                              &  1444\\\\\n",
       "\t ventral nerve cord                   &  7542\\\\\n",
       "\t ventral nerve cord primordium        &   671\\\\\n",
       "\t visceral muscle                      &   799\\\\\n",
       "\t visceral muscle primordium           &  3546\\\\\n",
       "\t yolk nuclei                          &   777\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 51 × 2\n",
       "\n",
       "| annot &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Malpighian tubule                    |  1090 |\n",
       "| amnioserosa                          |  7610 |\n",
       "| anlage in statu nascendi             |    40 |\n",
       "| brain                                | 26617 |\n",
       "| brain primordium                     |  9816 |\n",
       "| cellular blastoderm / pole cell      |    16 |\n",
       "| crystal cell                         |   177 |\n",
       "| ectoderm anlage                      |     5 |\n",
       "| ectoderm anlage in statu nascendi    |    51 |\n",
       "| ectoderm primordium                  |  1601 |\n",
       "| endoderm primordium                  |   400 |\n",
       "| epidermis                            | 36935 |\n",
       "| epidermis primordium                 |  6976 |\n",
       "| fat body                             |  5550 |\n",
       "| foregut / hindgut / anal pad         |  7035 |\n",
       "| foregut / hindgut primordium         |  1218 |\n",
       "| germ cell                            |  1309 |\n",
       "| glia                                 |   899 |\n",
       "| hindgut / anal pad                   |  2662 |\n",
       "| hindgut primordium                   |  1916 |\n",
       "| maternal                             |    17 |\n",
       "| mesoderm primordium                  |   384 |\n",
       "| midgut                               | 11649 |\n",
       "| midgut primordium                    |  1848 |\n",
       "| midline primordium                   |   982 |\n",
       "| muscle                               |  6184 |\n",
       "| peripheral nervous system            |  3591 |\n",
       "| peripheral nervous system primordium |   559 |\n",
       "| plasmatocytes                        |  5575 |\n",
       "| plasmatocytes anlage                 |  1658 |\n",
       "| procrystal cell                      |  1556 |\n",
       "| proventriculus                       |  2480 |\n",
       "| salivary gland                       |  3487 |\n",
       "| salivary gland body primordium       |  1042 |\n",
       "| salivary gland primordium            |   115 |\n",
       "| sensory nervous system               |  4816 |\n",
       "| sensory nervous system primordium    |  1627 |\n",
       "| somatic muscle                       | 14715 |\n",
       "| somatic muscle primordium            |  1674 |\n",
       "| spiracle (tracheal system)           |   986 |\n",
       "| spiracles (tracheal system)          |   751 |\n",
       "| tracheal primordium                  |  2526 |\n",
       "| tracheal system                      |  8052 |\n",
       "| trunk mesoderm primordium            |   880 |\n",
       "| ubiquitous                           | 34589 |\n",
       "| unknown                              |  1444 |\n",
       "| ventral nerve cord                   |  7542 |\n",
       "| ventral nerve cord primordium        |   671 |\n",
       "| visceral muscle                      |   799 |\n",
       "| visceral muscle primordium           |  3546 |\n",
       "| yolk nuclei                          |   777 |\n",
       "\n"
      ],
      "text/plain": [
       "   annot                                n    \n",
       "1  Malpighian tubule                     1090\n",
       "2  amnioserosa                           7610\n",
       "3  anlage in statu nascendi                40\n",
       "4  brain                                26617\n",
       "5  brain primordium                      9816\n",
       "6  cellular blastoderm / pole cell         16\n",
       "7  crystal cell                           177\n",
       "8  ectoderm anlage                          5\n",
       "9  ectoderm anlage in statu nascendi       51\n",
       "10 ectoderm primordium                   1601\n",
       "11 endoderm primordium                    400\n",
       "12 epidermis                            36935\n",
       "13 epidermis primordium                  6976\n",
       "14 fat body                              5550\n",
       "15 foregut / hindgut / anal pad          7035\n",
       "16 foregut / hindgut primordium          1218\n",
       "17 germ cell                             1309\n",
       "18 glia                                   899\n",
       "19 hindgut / anal pad                    2662\n",
       "20 hindgut primordium                    1916\n",
       "21 maternal                                17\n",
       "22 mesoderm primordium                    384\n",
       "23 midgut                               11649\n",
       "24 midgut primordium                     1848\n",
       "25 midline primordium                     982\n",
       "26 muscle                                6184\n",
       "27 peripheral nervous system             3591\n",
       "28 peripheral nervous system primordium   559\n",
       "29 plasmatocytes                         5575\n",
       "30 plasmatocytes anlage                  1658\n",
       "31 procrystal cell                       1556\n",
       "32 proventriculus                        2480\n",
       "33 salivary gland                        3487\n",
       "34 salivary gland body primordium        1042\n",
       "35 salivary gland primordium              115\n",
       "36 sensory nervous system                4816\n",
       "37 sensory nervous system primordium     1627\n",
       "38 somatic muscle                       14715\n",
       "39 somatic muscle primordium             1674\n",
       "40 spiracle (tracheal system)             986\n",
       "41 spiracles (tracheal system)            751\n",
       "42 tracheal primordium                   2526\n",
       "43 tracheal system                       8052\n",
       "44 trunk mesoderm primordium              880\n",
       "45 ubiquitous                           34589\n",
       "46 unknown                               1444\n",
       "47 ventral nerve cord                    7542\n",
       "48 ventral nerve cord primordium          671\n",
       "49 visceral muscle                        799\n",
       "50 visceral muscle primordium            3546\n",
       "51 yolk nuclei                            777"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub300  %>% count(annot)  %>% arrange(annot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 517,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 50 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Label</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Malpighian tubule                   </td><td>  87</td></tr>\n",
       "\t<tr><td>amnioserosa                         </td><td> 278</td></tr>\n",
       "\t<tr><td>anlage in statu nascendi            </td><td>  12</td></tr>\n",
       "\t<tr><td>brain                               </td><td>1277</td></tr>\n",
       "\t<tr><td>brain primordium                    </td><td>1096</td></tr>\n",
       "\t<tr><td>cellular blastoderm / pole cell     </td><td>   2</td></tr>\n",
       "\t<tr><td>crystal cell                        </td><td>  23</td></tr>\n",
       "\t<tr><td>ectoderm anlage in statu nascendi   </td><td>  14</td></tr>\n",
       "\t<tr><td>ectoderm primordium                 </td><td> 158</td></tr>\n",
       "\t<tr><td>endoderm primordium                 </td><td>  71</td></tr>\n",
       "\t<tr><td>epidermis                           </td><td>2453</td></tr>\n",
       "\t<tr><td>epidermis primordium                </td><td> 818</td></tr>\n",
       "\t<tr><td>fat body                            </td><td> 288</td></tr>\n",
       "\t<tr><td>foregut / hindgut / anal pad        </td><td>1266</td></tr>\n",
       "\t<tr><td>foregut / hindgut primordium        </td><td> 138</td></tr>\n",
       "\t<tr><td>germ cell                           </td><td> 129</td></tr>\n",
       "\t<tr><td>glia                                </td><td>  73</td></tr>\n",
       "\t<tr><td>hindgut / anal pad                  </td><td> 212</td></tr>\n",
       "\t<tr><td>hindgut primordium                  </td><td> 323</td></tr>\n",
       "\t<tr><td>maternal                            </td><td>   3</td></tr>\n",
       "\t<tr><td>mesoderm primordium                 </td><td>  68</td></tr>\n",
       "\t<tr><td>midgut                              </td><td> 854</td></tr>\n",
       "\t<tr><td>midgut primordium                   </td><td> 285</td></tr>\n",
       "\t<tr><td>midline primordium                  </td><td>  71</td></tr>\n",
       "\t<tr><td>muscle                              </td><td> 938</td></tr>\n",
       "\t<tr><td>peripheral nervous system           </td><td> 381</td></tr>\n",
       "\t<tr><td>peripheral nervous system primordium</td><td>  86</td></tr>\n",
       "\t<tr><td>plasmatocytes                       </td><td> 363</td></tr>\n",
       "\t<tr><td>plasmatocytes anlage                </td><td> 150</td></tr>\n",
       "\t<tr><td>procrystal cell                     </td><td> 142</td></tr>\n",
       "\t<tr><td>proventriculus                      </td><td> 206</td></tr>\n",
       "\t<tr><td>salivary gland                      </td><td> 116</td></tr>\n",
       "\t<tr><td>salivary gland body primordium      </td><td>  68</td></tr>\n",
       "\t<tr><td>salivary gland primordium           </td><td>  12</td></tr>\n",
       "\t<tr><td>sensory nervous system              </td><td> 635</td></tr>\n",
       "\t<tr><td>sensory nervous system primordium   </td><td> 278</td></tr>\n",
       "\t<tr><td>somatic muscle                      </td><td> 753</td></tr>\n",
       "\t<tr><td>somatic muscle primordium           </td><td> 291</td></tr>\n",
       "\t<tr><td>spiracle (tracheal system)          </td><td> 202</td></tr>\n",
       "\t<tr><td>spiracles (tracheal system)         </td><td>  35</td></tr>\n",
       "\t<tr><td>tracheal primordium                 </td><td> 364</td></tr>\n",
       "\t<tr><td>tracheal system                     </td><td> 617</td></tr>\n",
       "\t<tr><td>trunk mesoderm primordium           </td><td> 138</td></tr>\n",
       "\t<tr><td>ubiquitous                          </td><td>2086</td></tr>\n",
       "\t<tr><td>unknown                             </td><td> 321</td></tr>\n",
       "\t<tr><td>ventral nerve cord                  </td><td> 362</td></tr>\n",
       "\t<tr><td>ventral nerve cord primordium       </td><td> 128</td></tr>\n",
       "\t<tr><td>visceral muscle                     </td><td> 103</td></tr>\n",
       "\t<tr><td>visceral muscle primordium          </td><td> 577</td></tr>\n",
       "\t<tr><td>yolk nuclei                         </td><td> 137</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 50 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Label & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Malpighian tubule                    &   87\\\\\n",
       "\t amnioserosa                          &  278\\\\\n",
       "\t anlage in statu nascendi             &   12\\\\\n",
       "\t brain                                & 1277\\\\\n",
       "\t brain primordium                     & 1096\\\\\n",
       "\t cellular blastoderm / pole cell      &    2\\\\\n",
       "\t crystal cell                         &   23\\\\\n",
       "\t ectoderm anlage in statu nascendi    &   14\\\\\n",
       "\t ectoderm primordium                  &  158\\\\\n",
       "\t endoderm primordium                  &   71\\\\\n",
       "\t epidermis                            & 2453\\\\\n",
       "\t epidermis primordium                 &  818\\\\\n",
       "\t fat body                             &  288\\\\\n",
       "\t foregut / hindgut / anal pad         & 1266\\\\\n",
       "\t foregut / hindgut primordium         &  138\\\\\n",
       "\t germ cell                            &  129\\\\\n",
       "\t glia                                 &   73\\\\\n",
       "\t hindgut / anal pad                   &  212\\\\\n",
       "\t hindgut primordium                   &  323\\\\\n",
       "\t maternal                             &    3\\\\\n",
       "\t mesoderm primordium                  &   68\\\\\n",
       "\t midgut                               &  854\\\\\n",
       "\t midgut primordium                    &  285\\\\\n",
       "\t midline primordium                   &   71\\\\\n",
       "\t muscle                               &  938\\\\\n",
       "\t peripheral nervous system            &  381\\\\\n",
       "\t peripheral nervous system primordium &   86\\\\\n",
       "\t plasmatocytes                        &  363\\\\\n",
       "\t plasmatocytes anlage                 &  150\\\\\n",
       "\t procrystal cell                      &  142\\\\\n",
       "\t proventriculus                       &  206\\\\\n",
       "\t salivary gland                       &  116\\\\\n",
       "\t salivary gland body primordium       &   68\\\\\n",
       "\t salivary gland primordium            &   12\\\\\n",
       "\t sensory nervous system               &  635\\\\\n",
       "\t sensory nervous system primordium    &  278\\\\\n",
       "\t somatic muscle                       &  753\\\\\n",
       "\t somatic muscle primordium            &  291\\\\\n",
       "\t spiracle (tracheal system)           &  202\\\\\n",
       "\t spiracles (tracheal system)          &   35\\\\\n",
       "\t tracheal primordium                  &  364\\\\\n",
       "\t tracheal system                      &  617\\\\\n",
       "\t trunk mesoderm primordium            &  138\\\\\n",
       "\t ubiquitous                           & 2086\\\\\n",
       "\t unknown                              &  321\\\\\n",
       "\t ventral nerve cord                   &  362\\\\\n",
       "\t ventral nerve cord primordium        &  128\\\\\n",
       "\t visceral muscle                      &  103\\\\\n",
       "\t visceral muscle primordium           &  577\\\\\n",
       "\t yolk nuclei                          &  137\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 50 × 2\n",
       "\n",
       "| Label &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Malpighian tubule                    |   87 |\n",
       "| amnioserosa                          |  278 |\n",
       "| anlage in statu nascendi             |   12 |\n",
       "| brain                                | 1277 |\n",
       "| brain primordium                     | 1096 |\n",
       "| cellular blastoderm / pole cell      |    2 |\n",
       "| crystal cell                         |   23 |\n",
       "| ectoderm anlage in statu nascendi    |   14 |\n",
       "| ectoderm primordium                  |  158 |\n",
       "| endoderm primordium                  |   71 |\n",
       "| epidermis                            | 2453 |\n",
       "| epidermis primordium                 |  818 |\n",
       "| fat body                             |  288 |\n",
       "| foregut / hindgut / anal pad         | 1266 |\n",
       "| foregut / hindgut primordium         |  138 |\n",
       "| germ cell                            |  129 |\n",
       "| glia                                 |   73 |\n",
       "| hindgut / anal pad                   |  212 |\n",
       "| hindgut primordium                   |  323 |\n",
       "| maternal                             |    3 |\n",
       "| mesoderm primordium                  |   68 |\n",
       "| midgut                               |  854 |\n",
       "| midgut primordium                    |  285 |\n",
       "| midline primordium                   |   71 |\n",
       "| muscle                               |  938 |\n",
       "| peripheral nervous system            |  381 |\n",
       "| peripheral nervous system primordium |   86 |\n",
       "| plasmatocytes                        |  363 |\n",
       "| plasmatocytes anlage                 |  150 |\n",
       "| procrystal cell                      |  142 |\n",
       "| proventriculus                       |  206 |\n",
       "| salivary gland                       |  116 |\n",
       "| salivary gland body primordium       |   68 |\n",
       "| salivary gland primordium            |   12 |\n",
       "| sensory nervous system               |  635 |\n",
       "| sensory nervous system primordium    |  278 |\n",
       "| somatic muscle                       |  753 |\n",
       "| somatic muscle primordium            |  291 |\n",
       "| spiracle (tracheal system)           |  202 |\n",
       "| spiracles (tracheal system)          |   35 |\n",
       "| tracheal primordium                  |  364 |\n",
       "| tracheal system                      |  617 |\n",
       "| trunk mesoderm primordium            |  138 |\n",
       "| ubiquitous                           | 2086 |\n",
       "| unknown                              |  321 |\n",
       "| ventral nerve cord                   |  362 |\n",
       "| ventral nerve cord primordium        |  128 |\n",
       "| visceral muscle                      |  103 |\n",
       "| visceral muscle primordium           |  577 |\n",
       "| yolk nuclei                          |  137 |\n",
       "\n"
      ],
      "text/plain": [
       "   Label                                n   \n",
       "1  Malpighian tubule                      87\n",
       "2  amnioserosa                           278\n",
       "3  anlage in statu nascendi               12\n",
       "4  brain                                1277\n",
       "5  brain primordium                     1096\n",
       "6  cellular blastoderm / pole cell         2\n",
       "7  crystal cell                           23\n",
       "8  ectoderm anlage in statu nascendi      14\n",
       "9  ectoderm primordium                   158\n",
       "10 endoderm primordium                    71\n",
       "11 epidermis                            2453\n",
       "12 epidermis primordium                  818\n",
       "13 fat body                              288\n",
       "14 foregut / hindgut / anal pad         1266\n",
       "15 foregut / hindgut primordium          138\n",
       "16 germ cell                             129\n",
       "17 glia                                   73\n",
       "18 hindgut / anal pad                    212\n",
       "19 hindgut primordium                    323\n",
       "20 maternal                                3\n",
       "21 mesoderm primordium                    68\n",
       "22 midgut                                854\n",
       "23 midgut primordium                     285\n",
       "24 midline primordium                     71\n",
       "25 muscle                                938\n",
       "26 peripheral nervous system             381\n",
       "27 peripheral nervous system primordium   86\n",
       "28 plasmatocytes                         363\n",
       "29 plasmatocytes anlage                  150\n",
       "30 procrystal cell                       142\n",
       "31 proventriculus                        206\n",
       "32 salivary gland                        116\n",
       "33 salivary gland body primordium         68\n",
       "34 salivary gland primordium              12\n",
       "35 sensory nervous system                635\n",
       "36 sensory nervous system primordium     278\n",
       "37 somatic muscle                        753\n",
       "38 somatic muscle primordium             291\n",
       "39 spiracle (tracheal system)            202\n",
       "40 spiracles (tracheal system)            35\n",
       "41 tracheal primordium                   364\n",
       "42 tracheal system                       617\n",
       "43 trunk mesoderm primordium             138\n",
       "44 ubiquitous                           2086\n",
       "45 unknown                               321\n",
       "46 ventral nerve cord                    362\n",
       "47 ventral nerve cord primordium         128\n",
       "48 visceral muscle                       103\n",
       "49 visceral muscle primordium            577\n",
       "50 yolk nuclei                           137"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "res_label  %>% filter(nk == 1)  %>% ungroup  %>% count(Label)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 518,
   "metadata": {},
   "outputs": [],
   "source": [
    "singletons = (res %>% filter(nk == 0) )$ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 519,
   "metadata": {},
   "outputs": [],
   "source": [
    "doubletons = (res %>% filter(nk == 1) )$ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# create the dataframe from the cds from 300 PCAs\n",
    "df_sub300 = make_df(cds_sub)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 520,
   "metadata": {},
   "outputs": [],
   "source": [
    "options(repr.plot.width=14, repr.plot.height=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "# cell.type = \"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% singletons), size = ifelse(df_sub300$cell %in% singletons, 0.2, 0.05), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"singletons\") + \n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "#cell.type = \"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% singletons & df_sub300$annot == \"Malpighian tubule\"), \n",
    "           size = ifelse( df_sub300$cell %in% singletons & df_sub300$annot == \"Malpighian tubule\", 2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Malpighian tubule singletons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "#cell.type = \"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% singletons & df_sub300$annot == \"Malpighian tubule\"), \n",
    "           size = ifelse( df_sub300$cell %in% singletons & df_sub300$annot == \"Malpighian tubule\", 2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Malpighian tubule singletons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "#cell.type = \"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = !(df_sub300$cell %in% singletons) & df_sub300$annot == \"Malpighian tubule\"), \n",
    "           size = ifelse( !(df_sub300$cell %in% singletons) & df_sub300$annot == \"Malpighian tubule\", 2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Malpighian tubule not singletons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "#cell.type = \"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% doubletons & df_sub300$annot == \"Malpighian tubule\"), \n",
    "           size = ifelse( df_sub300$cell %in% doubletons & df_sub300$annot == \"Malpighian tubule\", 2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Malpighian tubule doubletons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "#cell.type = \"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = !(df_sub300$cell %in% doubletons) & df_sub300$annot == \"Malpighian tubule\" ), \n",
    "           size = ifelse( !(df_sub300$cell %in% doubletons) & df_sub300$annot == \"Malpighian tubule\", 2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Malpighian tubule not doubletons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "#cell.type = \"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% doubletons), size = ifelse(df_sub300$cell %in% doubletons, 0.1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"doubletons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 528,
   "metadata": {},
   "outputs": [],
   "source": [
    "nineteentons = (res %>% filter(nk == 19) )$ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "#cell.type = \"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% nineteentons), size = ifelse(df_sub300$cell %in% nineteentons, 0.1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"nineteentons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 540,
   "metadata": {},
   "outputs": [],
   "source": [
    "twentytons = (res %>% filter(nk == 20) )$ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "#cell.type = \"visceral muscle\"\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% twentytons), \n",
    "           size = ifelse(df_sub300$cell %in% twentytons, 0.1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"nineteentons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 682,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 33</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>orig.ident</th><th scope=col>nCount_RNA</th><th scope=col>nFeature_RNA</th><th scope=col>rt_bc</th><th scope=col>cell</th><th scope=col>time</th><th scope=col>percent.mt</th><th scope=col>percent.ribo</th><th scope=col>barn_dro</th><th scope=col>barn_hu</th><th scope=col>⋯</th><th scope=col>NNv2_shift</th><th scope=col>annot</th><th scope=col>ident</th><th scope=col>inferred_time_window</th><th scope=col>cell.type_time.bin</th><th scope=col>Size_Factor</th><th scope=col>cluster</th><th scope=col>umap_1</th><th scope=col>umap_2</th><th scope=col>cell.type.2</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</th><td>exp1</td><td>346</td><td>247</td><td>AAGTTGCCAT</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>hrs_06_10</td><td>1.4450867</td><td>4.3352601</td><td>371</td><td>0</td><td>⋯</td><td> 0.000000</td><td>sensory nervous system primordium</td><td>21</td><td>6-8  </td><td>sensory nervous system primordium:6-8</td><td>0.8108612</td><td>89 </td><td>-2.0067012</td><td> 1.218446</td><td>v </td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG</th><td>exp1</td><td>279</td><td>225</td><td>TTAATTATTG</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG</td><td>hrs_08_12</td><td>0.7168459</td><td>5.0179211</td><td>331</td><td>4</td><td>⋯</td><td> 0.000000</td><td>visceral muscle primordium       </td><td>11</td><td>8-10 </td><td>visceral muscle primordium:8-10      </td><td>0.6571200</td><td>112</td><td>-4.6204820</td><td> 1.096802</td><td>cb</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA</th><td>exp1</td><td>586</td><td>425</td><td>CAGGTATGGA</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA</td><td>hrs_08_12</td><td>1.0238908</td><td>6.4846416</td><td>630</td><td>0</td><td>⋯</td><td> 0.000000</td><td>brain                            </td><td>11</td><td>10-12</td><td>brain:10-12                          </td><td>1.4059888</td><td>22 </td><td>-3.6831634</td><td> 2.801265</td><td>bl</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG</th><td>exp1</td><td>292</td><td>202</td><td>AAGTAGTCAG</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG</td><td>hrs_10_14</td><td>0.3424658</td><td>0.6849315</td><td>320</td><td>3</td><td>⋯</td><td> 0.000000</td><td>epidermis                        </td><td>12</td><td>12-14</td><td>epidermis:12-14                      </td><td>0.6967951</td><td>165</td><td>-1.3580737</td><td> 0.358204</td><td>h </td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA</th><td>exp1</td><td>335</td><td>243</td><td>TGAGGTAGAA</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA</td><td>hrs_14_18</td><td>4.4776119</td><td>5.6716418</td><td>386</td><td>0</td><td>⋯</td><td>-1.480969</td><td>sensory nervous system           </td><td>11</td><td>12-14</td><td>sensory nervous system:12-14         </td><td>0.7811049</td><td>102</td><td>-3.8563070</td><td> 3.315827</td><td>bk</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA</th><td>exp1</td><td>354</td><td>272</td><td>TCGATATTGA</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA</td><td>hrs_14_18</td><td>0.5649718</td><td>1.6949153</td><td>380</td><td>0</td><td>⋯</td><td> 0.000000</td><td>sensory nervous system           </td><td>11</td><td>14-16</td><td>sensory nervous system:14-16         </td><td>0.8306988</td><td>148</td><td>-0.6711318</td><td>-2.946210</td><td>w </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 33\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & orig.ident & nCount\\_RNA & nFeature\\_RNA & rt\\_bc & cell & time & percent.mt & percent.ribo & barn\\_dro & barn\\_hu & ⋯ & NNv2\\_shift & annot & ident & inferred\\_time\\_window & cell.type\\_time.bin & Size\\_Factor & cluster & umap\\_1 & umap\\_2 & cell.type.2\\\\\n",
       "  & <chr> & <dbl> & <int> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <int> & <int> & ⋯ & <dbl> & <chr> & <fct> & <chr> & <chr> & <dbl> & <fct> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & exp1 & 346 & 247 & AAGTTGCCAT & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & hrs\\_06\\_10 & 1.4450867 & 4.3352601 & 371 & 0 & ⋯ &  0.000000 & sensory nervous system primordium & 21 & 6-8   & sensory nervous system primordium:6-8 & 0.8108612 & 89  & -2.0067012 &  1.218446 & v \\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG & exp1 & 279 & 225 & TTAATTATTG & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG & hrs\\_08\\_12 & 0.7168459 & 5.0179211 & 331 & 4 & ⋯ &  0.000000 & visceral muscle primordium        & 11 & 8-10  & visceral muscle primordium:8-10       & 0.6571200 & 112 & -4.6204820 &  1.096802 & cb\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA & exp1 & 586 & 425 & CAGGTATGGA & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA & hrs\\_08\\_12 & 1.0238908 & 6.4846416 & 630 & 0 & ⋯ &  0.000000 & brain                             & 11 & 10-12 & brain:10-12                           & 1.4059888 & 22  & -3.6831634 &  2.801265 & bl\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG & exp1 & 292 & 202 & AAGTAGTCAG & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG & hrs\\_10\\_14 & 0.3424658 & 0.6849315 & 320 & 3 & ⋯ &  0.000000 & epidermis                         & 12 & 12-14 & epidermis:12-14                       & 0.6967951 & 165 & -1.3580737 &  0.358204 & h \\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA & exp1 & 335 & 243 & TGAGGTAGAA & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA & hrs\\_14\\_18 & 4.4776119 & 5.6716418 & 386 & 0 & ⋯ & -1.480969 & sensory nervous system            & 11 & 12-14 & sensory nervous system:12-14          & 0.7811049 & 102 & -3.8563070 &  3.315827 & bk\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA & exp1 & 354 & 272 & TCGATATTGA & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA & hrs\\_14\\_18 & 0.5649718 & 1.6949153 & 380 & 0 & ⋯ &  0.000000 & sensory nervous system            & 11 & 14-16 & sensory nervous system:14-16          & 0.8306988 & 148 & -0.6711318 & -2.946210 & w \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 33\n",
       "\n",
       "| <!--/--> | orig.ident &lt;chr&gt; | nCount_RNA &lt;dbl&gt; | nFeature_RNA &lt;int&gt; | rt_bc &lt;chr&gt; | cell &lt;chr&gt; | time &lt;chr&gt; | percent.mt &lt;dbl&gt; | percent.ribo &lt;dbl&gt; | barn_dro &lt;int&gt; | barn_hu &lt;int&gt; | ⋯ ⋯ | NNv2_shift &lt;dbl&gt; | annot &lt;chr&gt; | ident &lt;fct&gt; | inferred_time_window &lt;chr&gt; | cell.type_time.bin &lt;chr&gt; | Size_Factor &lt;dbl&gt; | cluster &lt;fct&gt; | umap_1 &lt;dbl&gt; | umap_2 &lt;dbl&gt; | cell.type.2 &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | exp1 | 346 | 247 | AAGTTGCCAT | exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | hrs_06_10 | 1.4450867 | 4.3352601 | 371 | 0 | ⋯ |  0.000000 | sensory nervous system primordium | 21 | 6-8   | sensory nervous system primordium:6-8 | 0.8108612 | 89  | -2.0067012 |  1.218446 | v  |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG | exp1 | 279 | 225 | TTAATTATTG | exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG | hrs_08_12 | 0.7168459 | 5.0179211 | 331 | 4 | ⋯ |  0.000000 | visceral muscle primordium        | 11 | 8-10  | visceral muscle primordium:8-10       | 0.6571200 | 112 | -4.6204820 |  1.096802 | cb |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA | exp1 | 586 | 425 | CAGGTATGGA | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA | hrs_08_12 | 1.0238908 | 6.4846416 | 630 | 0 | ⋯ |  0.000000 | brain                             | 11 | 10-12 | brain:10-12                           | 1.4059888 | 22  | -3.6831634 |  2.801265 | bl |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG | exp1 | 292 | 202 | AAGTAGTCAG | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG | hrs_10_14 | 0.3424658 | 0.6849315 | 320 | 3 | ⋯ |  0.000000 | epidermis                         | 12 | 12-14 | epidermis:12-14                       | 0.6967951 | 165 | -1.3580737 |  0.358204 | h  |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA | exp1 | 335 | 243 | TGAGGTAGAA | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA | hrs_14_18 | 4.4776119 | 5.6716418 | 386 | 0 | ⋯ | -1.480969 | sensory nervous system            | 11 | 12-14 | sensory nervous system:12-14          | 0.7811049 | 102 | -3.8563070 |  3.315827 | bk |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA | exp1 | 354 | 272 | TCGATATTGA | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA | hrs_14_18 | 0.5649718 | 1.6949153 | 380 | 0 | ⋯ |  0.000000 | sensory nervous system            | 11 | 14-16 | sensory nervous system:14-16          | 0.8306988 | 148 | -0.6711318 | -2.946210 | w  |\n",
       "\n"
      ],
      "text/plain": [
       "                                                       orig.ident nCount_RNA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT exp1       346       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG exp1       279       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA exp1       586       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG exp1       292       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA exp1       335       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA exp1       354       \n",
       "                                                       nFeature_RNA rt_bc     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 247          AAGTTGCCAT\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 225          TTAATTATTG\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 425          CAGGTATGGA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 202          AAGTAGTCAG\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 243          TGAGGTAGAA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 272          TCGATATTGA\n",
       "                                                       cell                                                  \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA\n",
       "                                                       time      percent.mt\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT hrs_06_10 1.4450867 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG hrs_08_12 0.7168459 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA hrs_08_12 1.0238908 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG hrs_10_14 0.3424658 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA hrs_14_18 4.4776119 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA hrs_14_18 0.5649718 \n",
       "                                                       percent.ribo barn_dro\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 4.3352601    371     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 5.0179211    331     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 6.4846416    630     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 0.6849315    320     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 5.6716418    386     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 1.6949153    380     \n",
       "                                                       barn_hu ⋯ NNv2_shift\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 0       ⋯  0.000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 4       ⋯  0.000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 0       ⋯  0.000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 3       ⋯  0.000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 0       ⋯ -1.480969 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 0       ⋯  0.000000 \n",
       "                                                       annot                            \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT sensory nervous system primordium\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG visceral muscle primordium       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA brain                            \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG epidermis                        \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA sensory nervous system           \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA sensory nervous system           \n",
       "                                                       ident\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 21   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 11   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 11   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 11   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 11   \n",
       "                                                       inferred_time_window\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 6-8                 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 8-10                \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 10-12               \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12-14               \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 12-14               \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 14-16               \n",
       "                                                       cell.type_time.bin                   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT sensory nervous system primordium:6-8\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG visceral muscle primordium:8-10      \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA brain:10-12                          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG epidermis:12-14                      \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA sensory nervous system:12-14         \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA sensory nervous system:14-16         \n",
       "                                                       Size_Factor cluster\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 0.8108612   89     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 0.6571200   112    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 1.4059888   22     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 0.6967951   165    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 0.7811049   102    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 0.8306988   148    \n",
       "                                                       umap_1     umap_2   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT -2.0067012  1.218446\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG -4.6204820  1.096802\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA -3.6831634  2.801265\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG -1.3580737  0.358204\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA -3.8563070  3.315827\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA -0.6711318 -2.946210\n",
       "                                                       cell.type.2\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT v          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG cb         \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA bl         \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG h          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA bk         \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA w          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub300  %>%  head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "# plot of GATAe positive clusters\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell.type.2 %in% c(\"at\", \"ax\", \"b\", \"ba\", \"bf\", \"ac\")), \n",
    "               size = ifelse(df_sub300$cell.type.2 %in% c(\"at\", \"ax\", \"b\", \"ba\", \"bf\", \"ac\"), 0.1, 0.02), \n",
    "               stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"nineteentons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "# plot of common negative clusters in high angle positives andhigh negative angles\n",
    "# and \n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell.type.2 %in% c(\"af\", \"aw\", \"as\", \"bj\", \"bk\", \"bl\", \"bm\", \"bn\", \"bo\",\n",
    "                                                   \"bq\", \"br\", \"bs\", \"bt\", \"bv\", \"bw\", \"bx\",'by' )), \n",
    "               size = ifelse(df_sub300$cell.type.2 %in% c(\"bj\", \"bk\", \"bl\", \"bm\", \"bn\", \"bo\",\n",
    "                                                   \"bq\", \"br\", \"bs\", \"bt\", \"bv\", \"bx\", \"bw\", 'by'), 0.1, 0.02), \n",
    "               stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"nineteentons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Subset on singletons and also on doubletons using the nn21 result for Lou\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'cds_sub'</li><li>'cell_type'</li><li>'cell.type'</li><li>'cellIDs'</li><li>'cells'</li><li>'cellTypes'</li><li>'df'</li><li>'df_sub300'</li><li>'doubletons'</li><li>'expresses.gene'</li><li>'get.gene.id'</li><li>'get.gene.id2'</li><li>'get.norm.expr.matrix'</li><li>'get.norm.expr.matrix.3'</li><li>'ids'</li><li>'k'</li><li>'k1'</li><li>'labels'</li><li>'make_df'</li><li>'nineteentons'</li><li>'nnMatches'</li><li>'pca'</li><li>'peedeef'</li><li>'plot.expr'</li><li>'plot.expr_clus'</li><li>'plot.expr_clus.br'</li><li>'plot.expr_clus.br2'</li><li>'plot.expr3'</li><li>'rann'</li><li>'res'</li><li>'res_label'</li><li>'singletons'</li><li>'two.set.differential.gene.test'</li><li>'two.set.differential.gene.test.3'</li><li>'two.set.differential.gene.test.3a'</li><li>'two.set.differential.gene.test.3b'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'cds\\_sub'\n",
       "\\item 'cell\\_type'\n",
       "\\item 'cell.type'\n",
       "\\item 'cellIDs'\n",
       "\\item 'cells'\n",
       "\\item 'cellTypes'\n",
       "\\item 'df'\n",
       "\\item 'df\\_sub300'\n",
       "\\item 'doubletons'\n",
       "\\item 'expresses.gene'\n",
       "\\item 'get.gene.id'\n",
       "\\item 'get.gene.id2'\n",
       "\\item 'get.norm.expr.matrix'\n",
       "\\item 'get.norm.expr.matrix.3'\n",
       "\\item 'ids'\n",
       "\\item 'k'\n",
       "\\item 'k1'\n",
       "\\item 'labels'\n",
       "\\item 'make\\_df'\n",
       "\\item 'nineteentons'\n",
       "\\item 'nnMatches'\n",
       "\\item 'pca'\n",
       "\\item 'peedeef'\n",
       "\\item 'plot.expr'\n",
       "\\item 'plot.expr\\_clus'\n",
       "\\item 'plot.expr\\_clus.br'\n",
       "\\item 'plot.expr\\_clus.br2'\n",
       "\\item 'plot.expr3'\n",
       "\\item 'rann'\n",
       "\\item 'res'\n",
       "\\item 'res\\_label'\n",
       "\\item 'singletons'\n",
       "\\item 'two.set.differential.gene.test'\n",
       "\\item 'two.set.differential.gene.test.3'\n",
       "\\item 'two.set.differential.gene.test.3a'\n",
       "\\item 'two.set.differential.gene.test.3b'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'cds_sub'\n",
       "2. 'cell_type'\n",
       "3. 'cell.type'\n",
       "4. 'cellIDs'\n",
       "5. 'cells'\n",
       "6. 'cellTypes'\n",
       "7. 'df'\n",
       "8. 'df_sub300'\n",
       "9. 'doubletons'\n",
       "10. 'expresses.gene'\n",
       "11. 'get.gene.id'\n",
       "12. 'get.gene.id2'\n",
       "13. 'get.norm.expr.matrix'\n",
       "14. 'get.norm.expr.matrix.3'\n",
       "15. 'ids'\n",
       "16. 'k'\n",
       "17. 'k1'\n",
       "18. 'labels'\n",
       "19. 'make_df'\n",
       "20. 'nineteentons'\n",
       "21. 'nnMatches'\n",
       "22. 'pca'\n",
       "23. 'peedeef'\n",
       "24. 'plot.expr'\n",
       "25. 'plot.expr_clus'\n",
       "26. 'plot.expr_clus.br'\n",
       "27. 'plot.expr_clus.br2'\n",
       "28. 'plot.expr3'\n",
       "29. 'rann'\n",
       "30. 'res'\n",
       "31. 'res_label'\n",
       "32. 'singletons'\n",
       "33. 'two.set.differential.gene.test'\n",
       "34. 'two.set.differential.gene.test.3'\n",
       "35. 'two.set.differential.gene.test.3a'\n",
       "36. 'two.set.differential.gene.test.3b'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"cds_sub\"                           \"cell_type\"                        \n",
       " [3] \"cell.type\"                         \"cellIDs\"                          \n",
       " [5] \"cells\"                             \"cellTypes\"                        \n",
       " [7] \"df\"                                \"df_sub300\"                        \n",
       " [9] \"doubletons\"                        \"expresses.gene\"                   \n",
       "[11] \"get.gene.id\"                       \"get.gene.id2\"                     \n",
       "[13] \"get.norm.expr.matrix\"              \"get.norm.expr.matrix.3\"           \n",
       "[15] \"ids\"                               \"k\"                                \n",
       "[17] \"k1\"                                \"labels\"                           \n",
       "[19] \"make_df\"                           \"nineteentons\"                     \n",
       "[21] \"nnMatches\"                         \"pca\"                              \n",
       "[23] \"peedeef\"                           \"plot.expr\"                        \n",
       "[25] \"plot.expr_clus\"                    \"plot.expr_clus.br\"                \n",
       "[27] \"plot.expr_clus.br2\"                \"plot.expr3\"                       \n",
       "[29] \"rann\"                              \"res\"                              \n",
       "[31] \"res_label\"                         \"singletons\"                       \n",
       "[33] \"two.set.differential.gene.test\"    \"two.set.differential.gene.test.3\" \n",
       "[35] \"two.set.differential.gene.test.3a\" \"two.set.differential.gene.test.3b\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ls()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 530,
   "metadata": {},
   "outputs": [],
   "source": [
    "cds_sub_no_sing = cds_sub[, !(df_sub300$cell %in% singletons)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 531,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>212449</li><li>33</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 212449\n",
       "\\item 33\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 212449\n",
       "2. 33\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 212449     33"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub300  %>% filter(!(cell %in% singletons))  %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 532,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>13938</li><li>212449</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 13938\n",
       "\\item 212449\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 13938\n",
       "2. 212449\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1]  13938 212449"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(cds_sub_no_sing)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 533,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sub_no_singletons = make_df(cds_sub_no_sing)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 535,
   "metadata": {},
   "outputs": [],
   "source": [
    "## continue to remove doubletons\n",
    "cds_sub_no_doubletons = cds_sub[, !(df_sub_no_singletons$cell %in% doubletons)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 536,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>13938</li><li>216570</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 13938\n",
       "\\item 216570\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 13938\n",
       "2. 216570\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1]  13938 216570"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(cds_sub_no_doubletons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 538,
   "metadata": {},
   "outputs": [],
   "source": [
    "saveRDS(cds_sub_no_doubletons, \"data/cds_fly_pca300_no_doubletons.nn21.rds\") "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###  Check neuronal subsubcluster -- Lou fixed  names of cell column in this subcluster so they match the old"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 595,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1mRows: \u001b[22m\u001b[34m6703\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m3\u001b[39m\n",
      "\u001b[36m──\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \"\\t\"\n",
      "\u001b[31mchr\u001b[39m (2): cell_name, annotation\n",
      "\u001b[32mdbl\u001b[39m (1): seurat_clusters\n",
      "\n",
      "\u001b[36mℹ\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36mℹ\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n"
     ]
    }
   ],
   "source": [
    "neuronal_subsubcluster = read_tsv(\"neuronal_subsubcluster_annotations.modified.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 596,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cell_name</th><th scope=col>annotation</th><th scope=col>seurat_clusters</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exp1_idx1GTAGGCTGCA_idx2AACGGACCAA.GTCCTGCATTTAGGATCTA  </td><td>Chordotonal glia   </td><td>9</td></tr>\n",
       "\t<tr><td>exp1_idx1GTAGGCTGCA_idx2TGAGGTAGAA.CCGTCTCATTGATGGCCTT  </td><td>Chordotonal glia   </td><td>9</td></tr>\n",
       "\t<tr><td>exp1_idx1TAGGTAACTT_idx2CCGTCGGAGG.TCCGGTCATGGCTATGACT  </td><td>Chordotonal glia   </td><td>9</td></tr>\n",
       "\t<tr><td>exp1_idx1TAGTAAGCCG_idx2CCAAGCTGAA.GCAACGTTTCAAGTACTGG  </td><td>Chordotonal glia   </td><td>9</td></tr>\n",
       "\t<tr><td>exp10_idx1TCTTGAGGTT_idx2CGACTACCAA.TGAATCTGATGCTCTCGCCT</td><td>Chordotonal glia   </td><td>9</td></tr>\n",
       "\t<tr><td>exp1_idx1CCTCAGCTCC_idx2TTGAGTATCA.TCCGTATGTTGGAACTAGA  </td><td>Chordotonal sensory</td><td>6</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 3\n",
       "\\begin{tabular}{lll}\n",
       " cell\\_name & annotation & seurat\\_clusters\\\\\n",
       " <chr> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t exp1\\_idx1GTAGGCTGCA\\_idx2AACGGACCAA.GTCCTGCATTTAGGATCTA   & Chordotonal glia    & 9\\\\\n",
       "\t exp1\\_idx1GTAGGCTGCA\\_idx2TGAGGTAGAA.CCGTCTCATTGATGGCCTT   & Chordotonal glia    & 9\\\\\n",
       "\t exp1\\_idx1TAGGTAACTT\\_idx2CCGTCGGAGG.TCCGGTCATGGCTATGACT   & Chordotonal glia    & 9\\\\\n",
       "\t exp1\\_idx1TAGTAAGCCG\\_idx2CCAAGCTGAA.GCAACGTTTCAAGTACTGG   & Chordotonal glia    & 9\\\\\n",
       "\t exp10\\_idx1TCTTGAGGTT\\_idx2CGACTACCAA.TGAATCTGATGCTCTCGCCT & Chordotonal glia    & 9\\\\\n",
       "\t exp1\\_idx1CCTCAGCTCC\\_idx2TTGAGTATCA.TCCGTATGTTGGAACTAGA   & Chordotonal sensory & 6\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 3\n",
       "\n",
       "| cell_name &lt;chr&gt; | annotation &lt;chr&gt; | seurat_clusters &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| exp1_idx1GTAGGCTGCA_idx2AACGGACCAA.GTCCTGCATTTAGGATCTA   | Chordotonal glia    | 9 |\n",
       "| exp1_idx1GTAGGCTGCA_idx2TGAGGTAGAA.CCGTCTCATTGATGGCCTT   | Chordotonal glia    | 9 |\n",
       "| exp1_idx1TAGGTAACTT_idx2CCGTCGGAGG.TCCGGTCATGGCTATGACT   | Chordotonal glia    | 9 |\n",
       "| exp1_idx1TAGTAAGCCG_idx2CCAAGCTGAA.GCAACGTTTCAAGTACTGG   | Chordotonal glia    | 9 |\n",
       "| exp10_idx1TCTTGAGGTT_idx2CGACTACCAA.TGAATCTGATGCTCTCGCCT | Chordotonal glia    | 9 |\n",
       "| exp1_idx1CCTCAGCTCC_idx2TTGAGTATCA.TCCGTATGTTGGAACTAGA   | Chordotonal sensory | 6 |\n",
       "\n"
      ],
      "text/plain": [
       "  cell_name                                                annotation         \n",
       "1 exp1_idx1GTAGGCTGCA_idx2AACGGACCAA.GTCCTGCATTTAGGATCTA   Chordotonal glia   \n",
       "2 exp1_idx1GTAGGCTGCA_idx2TGAGGTAGAA.CCGTCTCATTGATGGCCTT   Chordotonal glia   \n",
       "3 exp1_idx1TAGGTAACTT_idx2CCGTCGGAGG.TCCGGTCATGGCTATGACT   Chordotonal glia   \n",
       "4 exp1_idx1TAGTAAGCCG_idx2CCAAGCTGAA.GCAACGTTTCAAGTACTGG   Chordotonal glia   \n",
       "5 exp10_idx1TCTTGAGGTT_idx2CGACTACCAA.TGAATCTGATGCTCTCGCCT Chordotonal glia   \n",
       "6 exp1_idx1CCTCAGCTCC_idx2TTGAGTATCA.TCCGTATGTTGGAACTAGA   Chordotonal sensory\n",
       "  seurat_clusters\n",
       "1 9              \n",
       "2 9              \n",
       "3 9              \n",
       "4 9              \n",
       "5 9              \n",
       "6 6              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "neuronal_subsubcluster  %>%  head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "str_split(fruits, \" and \", n = 2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 559,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol>\n",
       "\t<li><style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'16'</li><li>'18'</li><li>'exp1_idx1GTAGGCTGCA_idx2AACGGACCAA.GTCCTGCATTTAGGATCTA'</li></ol>\n",
       "</li>\n",
       "\t<li><style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'16'</li><li>'18'</li><li>'exp1_idx1GTAGGCTGCA_idx2TGAGGTAGAA.CCGTCTCATTGATGGCCTT'</li></ol>\n",
       "</li>\n",
       "\t<li><style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'16'</li><li>'18'</li><li>'exp1_idx1TAGGTAACTT_idx2CCGTCGGAGG.TCCGGTCATGGCTATGACT'</li></ol>\n",
       "</li>\n",
       "\t<li><style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'16'</li><li>'18'</li><li>'exp1_idx1TAGTAAGCCG_idx2CCAAGCTGAA.GCAACGTTTCAAGTACTGG'</li></ol>\n",
       "</li>\n",
       "\t<li><style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'16'</li><li>'18'</li><li>'exp10_idx1TCTTGAGGTT_idx2CGACTACCAA.TGAATCTGATGCTCTCGCCT'</li></ol>\n",
       "</li>\n",
       "\t<li><style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'16'</li><li>'18'</li><li>'exp1_idx1CCTCAGCTCC_idx2TTGAGTATCA.TCCGTATGTTGGAACTAGA'</li></ol>\n",
       "</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate}\n",
       "\\item \\begin{enumerate*}\n",
       "\\item '16'\n",
       "\\item '18'\n",
       "\\item 'exp1\\_idx1GTAGGCTGCA\\_idx2AACGGACCAA.GTCCTGCATTTAGGATCTA'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item \\begin{enumerate*}\n",
       "\\item '16'\n",
       "\\item '18'\n",
       "\\item 'exp1\\_idx1GTAGGCTGCA\\_idx2TGAGGTAGAA.CCGTCTCATTGATGGCCTT'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item \\begin{enumerate*}\n",
       "\\item '16'\n",
       "\\item '18'\n",
       "\\item 'exp1\\_idx1TAGGTAACTT\\_idx2CCGTCGGAGG.TCCGGTCATGGCTATGACT'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item \\begin{enumerate*}\n",
       "\\item '16'\n",
       "\\item '18'\n",
       "\\item 'exp1\\_idx1TAGTAAGCCG\\_idx2CCAAGCTGAA.GCAACGTTTCAAGTACTGG'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item \\begin{enumerate*}\n",
       "\\item '16'\n",
       "\\item '18'\n",
       "\\item 'exp10\\_idx1TCTTGAGGTT\\_idx2CGACTACCAA.TGAATCTGATGCTCTCGCCT'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item \\begin{enumerate*}\n",
       "\\item '16'\n",
       "\\item '18'\n",
       "\\item 'exp1\\_idx1CCTCAGCTCC\\_idx2TTGAGTATCA.TCCGTATGTTGGAACTAGA'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\end{enumerate}\n"
      ],
      "text/markdown": [
       "1. 1. '16'\n",
       "2. '18'\n",
       "3. 'exp1_idx1GTAGGCTGCA_idx2AACGGACCAA.GTCCTGCATTTAGGATCTA'\n",
       "\n",
       "\n",
       "\n",
       "2. 1. '16'\n",
       "2. '18'\n",
       "3. 'exp1_idx1GTAGGCTGCA_idx2TGAGGTAGAA.CCGTCTCATTGATGGCCTT'\n",
       "\n",
       "\n",
       "\n",
       "3. 1. '16'\n",
       "2. '18'\n",
       "3. 'exp1_idx1TAGGTAACTT_idx2CCGTCGGAGG.TCCGGTCATGGCTATGACT'\n",
       "\n",
       "\n",
       "\n",
       "4. 1. '16'\n",
       "2. '18'\n",
       "3. 'exp1_idx1TAGTAAGCCG_idx2CCAAGCTGAA.GCAACGTTTCAAGTACTGG'\n",
       "\n",
       "\n",
       "\n",
       "5. 1. '16'\n",
       "2. '18'\n",
       "3. 'exp10_idx1TCTTGAGGTT_idx2CGACTACCAA.TGAATCTGATGCTCTCGCCT'\n",
       "\n",
       "\n",
       "\n",
       "6. 1. '16'\n",
       "2. '18'\n",
       "3. 'exp1_idx1CCTCAGCTCC_idx2TTGAGTATCA.TCCGTATGTTGGAACTAGA'\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[[1]]\n",
       "[1] \"16\"                                                    \n",
       "[2] \"18\"                                                    \n",
       "[3] \"exp1_idx1GTAGGCTGCA_idx2AACGGACCAA.GTCCTGCATTTAGGATCTA\"\n",
       "\n",
       "[[2]]\n",
       "[1] \"16\"                                                    \n",
       "[2] \"18\"                                                    \n",
       "[3] \"exp1_idx1GTAGGCTGCA_idx2TGAGGTAGAA.CCGTCTCATTGATGGCCTT\"\n",
       "\n",
       "[[3]]\n",
       "[1] \"16\"                                                    \n",
       "[2] \"18\"                                                    \n",
       "[3] \"exp1_idx1TAGGTAACTT_idx2CCGTCGGAGG.TCCGGTCATGGCTATGACT\"\n",
       "\n",
       "[[4]]\n",
       "[1] \"16\"                                                    \n",
       "[2] \"18\"                                                    \n",
       "[3] \"exp1_idx1TAGTAAGCCG_idx2CCAAGCTGAA.GCAACGTTTCAAGTACTGG\"\n",
       "\n",
       "[[5]]\n",
       "[1] \"16\"                                                      \n",
       "[2] \"18\"                                                      \n",
       "[3] \"exp10_idx1TCTTGAGGTT_idx2CGACTACCAA.TGAATCTGATGCTCTCGCCT\"\n",
       "\n",
       "[[6]]\n",
       "[1] \"16\"                                                    \n",
       "[2] \"18\"                                                    \n",
       "[3] \"exp1_idx1CCTCAGCTCC_idx2TTGAGTATCA.TCCGTATGTTGGAACTAGA\"\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "neuronal_subsubcluster$cell_name  %>% str_split(\"_\", n=3)   %>% head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 594,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "\u001b[1m\u001b[33mError\u001b[39m in `separate_wider_delim()`:\u001b[22m\n\u001b[1m\u001b[22m\u001b[33m!\u001b[39m Expected 5 pieces in each element of `cell_name`.\n\u001b[33m!\u001b[39m 4 values were too short.\n\u001b[36mℹ\u001b[39m Use `too_few = \"debug\"` to diagnose the problem.\n\u001b[36mℹ\u001b[39m Use `too_few = \"align_start\"/\"align_end\"` to silence this message.\n",
     "output_type": "error",
     "traceback": [
      "\u001b[1m\u001b[33mError\u001b[39m in `separate_wider_delim()`:\u001b[22m\n\u001b[1m\u001b[22m\u001b[33m!\u001b[39m Expected 5 pieces in each element of `cell_name`.\n\u001b[33m!\u001b[39m 4 values were too short.\n\u001b[36mℹ\u001b[39m Use `too_few = \"debug\"` to diagnose the problem.\n\u001b[36mℹ\u001b[39m Use `too_few = \"align_start\"/\"align_end\"` to silence this message.\nTraceback:\n",
      "1. neuronal_subsubcluster %>% separate_wider_delim(cell_name, delim = \"_\", \n .     names = c(\"t1\", \"t2\", \"cell\", too_many = \"merge\", too_few = \"align_start\"))",
      "2. separate_wider_delim(., cell_name, delim = \"_\", names = c(\"t1\", \n .     \"t2\", \"cell\", too_many = \"merge\", too_few = \"align_start\"))",
      "3. map_unpack(data, {\n .     {\n .         cols\n .     }\n . }, function(x, col) str_separate_wider_delim(x, col, names = names, \n .     delim = delim, names_sep = names_sep, too_few = too_few, \n .     too_many = too_many, cols_remove = cols_remove, error_call = error_call), \n .     names_sep = names_sep, names_repair = names_repair)",
      "4. fun(data[[col]], col)",
      "5. str_separate_wider_delim(x, col, names = names, delim = delim, \n .     names_sep = names_sep, too_few = too_few, too_many = too_many, \n .     cols_remove = cols_remove, error_call = error_call)",
      "6. check_df_alignment(col, p, \"pieces\", n_pieces, too_few = too_few, \n .     too_many = too_many, advice_short = c(i = \"Use `too_few = \\\"debug\\\"` to diagnose the problem.\", \n .         i = \"Use `too_few = \\\"align_start\\\"/\\\"align_end\\\"` to silence this message.\"), \n .     advice_long = c(i = \"Use `too_many = \\\"debug\\\"` to diagnose the problem.\", \n .         i = \"Use `too_many = \\\"drop\\\"/\\\"merge\\\"` to silence this message.\"), \n .     call = error_call)",
      "7. cli::cli_abort(c(\"Expected {p} {type} in each element of {.var {col}}.\", \n .     `!` = if (error_short) \"{n_short} value{?s} {?was/were} too short.\", \n .     if (error_short) advice_short, `!` = if (error_long) \"{n_long} value{?s} {?was/were} too long.\", \n .     if (error_long) advice_long), call = call)",
      "8. rlang::abort(message, ..., call = call, use_cli_format = TRUE, \n .     .frame = .frame)",
      "9. signal_abort(cnd, .file)"
     ]
    }
   ],
   "source": [
    "names_expanded = neuronal_subsubcluster  %>%  separate_wider_delim(cell_name, delim = \"_\", \n",
    "                                                 names = c(\"t1\", \"t2\", \"cell\", \n",
    "                                                           too_many = \"merge\", too_few = \"align_start\")) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 590,
   "metadata": {},
   "outputs": [],
   "source": [
    "neuronal_subsubcluster2 = names_expanded  %>% mutate(cell_name = paste(cell, merge, align_start, sep = \"_\"), .before=-0)  %>%  \n",
    "select(cell_name, annotation, seurat_clusters)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 598,
   "metadata": {},
   "outputs": [],
   "source": [
    "Cholinergic_1 = (neuronal_subsubcluster  %>% filter(annotation == \"Cholinergic 1\"))$cell_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 599,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "1562"
      ],
      "text/latex": [
       "1562"
      ],
      "text/markdown": [
       "1562"
      ],
      "text/plain": [
       "[1] 1562"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(Cholinergic_1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \"Cholinergic 1\"))$cell_name), size = ifelse(df_sub300$cell %in% Cholinergic_1, 0.1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Cholinergic_1\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \"Cholinergic 2\"))$cell_name), \n",
    "           size = ifelse(df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \"Cholinergic 2\"))$cell_name, 0.1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Cholinergic_2\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \"Chordotonal glia\"))$cell_name), \n",
    "           size = ifelse(df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \"Chordotonal glia\"))$cell_name, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Chordotonal glia\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \"Chordotonal sensory\"))$cell_name),\n",
    "           size = ifelse(df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \"Chordotonal sensory\"))$cell_name, 0.1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Chordotonal sensory\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \n",
    "                                                       \"External sensory\"))$cell_name), \n",
    "           size = ifelse(df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \n",
    "                                                       \"External sensory\"))$cell_name, 0.1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"External sensory\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \n",
    "                                                       \"Glutamatergic\"))$cell_name), \n",
    "           size = ifelse(df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \n",
    "                                                       \"Glutamatergic\"))$cell_name, 0.1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Glutamatergic\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \n",
    "                                                       \"Monoaminergic\"))$cell_name), \n",
    "           size = ifelse(df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \n",
    "                                                       \"Monoaminergic\"))$cell_name, 0.1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Monoaminergic\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \n",
    "                                                       \"Multidendritic sensory\"))$cell_name), \n",
    "           size = ifelse(df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \n",
    "                                                       \"Multidendritic sensory\"))$cell_name, 0.1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Multidendritic sensory\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 597,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A spec_tbl_df: 11 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>annotation</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Cholinergic 1         </td><td>1562</td></tr>\n",
       "\t<tr><td>Cholinergic 2         </td><td> 997</td></tr>\n",
       "\t<tr><td>Chordotonal glia      </td><td>  70</td></tr>\n",
       "\t<tr><td>Chordotonal sensory   </td><td> 353</td></tr>\n",
       "\t<tr><td>External sensory      </td><td> 447</td></tr>\n",
       "\t<tr><td>Glutamatergic         </td><td> 564</td></tr>\n",
       "\t<tr><td>Monoaminergic         </td><td> 247</td></tr>\n",
       "\t<tr><td>Multidendritic sensory</td><td> 656</td></tr>\n",
       "\t<tr><td>Peptidergic           </td><td>  49</td></tr>\n",
       "\t<tr><td>Undiff. 1             </td><td>1514</td></tr>\n",
       "\t<tr><td>Undiff. 2             </td><td> 244</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A spec\\_tbl\\_df: 11 × 2\n",
       "\\begin{tabular}{ll}\n",
       " annotation & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Cholinergic 1          & 1562\\\\\n",
       "\t Cholinergic 2          &  997\\\\\n",
       "\t Chordotonal glia       &   70\\\\\n",
       "\t Chordotonal sensory    &  353\\\\\n",
       "\t External sensory       &  447\\\\\n",
       "\t Glutamatergic          &  564\\\\\n",
       "\t Monoaminergic          &  247\\\\\n",
       "\t Multidendritic sensory &  656\\\\\n",
       "\t Peptidergic            &   49\\\\\n",
       "\t Undiff. 1              & 1514\\\\\n",
       "\t Undiff. 2              &  244\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A spec_tbl_df: 11 × 2\n",
       "\n",
       "| annotation &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Cholinergic 1          | 1562 |\n",
       "| Cholinergic 2          |  997 |\n",
       "| Chordotonal glia       |   70 |\n",
       "| Chordotonal sensory    |  353 |\n",
       "| External sensory       |  447 |\n",
       "| Glutamatergic          |  564 |\n",
       "| Monoaminergic          |  247 |\n",
       "| Multidendritic sensory |  656 |\n",
       "| Peptidergic            |   49 |\n",
       "| Undiff. 1              | 1514 |\n",
       "| Undiff. 2              |  244 |\n",
       "\n"
      ],
      "text/plain": [
       "   annotation             n   \n",
       "1  Cholinergic 1          1562\n",
       "2  Cholinergic 2           997\n",
       "3  Chordotonal glia         70\n",
       "4  Chordotonal sensory     353\n",
       "5  External sensory        447\n",
       "6  Glutamatergic           564\n",
       "7  Monoaminergic           247\n",
       "8  Multidendritic sensory  656\n",
       "9  Peptidergic              49\n",
       "10 Undiff. 1              1514\n",
       "11 Undiff. 2               244"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "neuronal_subsubcluster  %>% count(annotation)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \"Peptidergic\"))$cell_name), size = \n",
    "           ifelse(df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \"Peptidergic\"))$cell_name, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Peptidergic\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \"Undiff. 1\"))$cell_name), \n",
    "           size = ifelse(df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \"Undiff. 1\"))$cell_name, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Undiff. 1\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \"Undiff. 2\"))$cell_name), \n",
    "           size = ifelse(df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \"Undiff. 2\"))$cell_name, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Undiff. 2\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check the larger subset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 629,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1mRows: \u001b[22m\u001b[34m59012\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m3\u001b[39m\n",
      "\u001b[36m──\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \"\\t\"\n",
      "\u001b[31mchr\u001b[39m (2): cell_name, annotation\n",
      "\u001b[32mdbl\u001b[39m (1): seurat_clusters\n",
      "\n",
      "\u001b[36mℹ\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36mℹ\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n"
     ]
    }
   ],
   "source": [
    "neuronal_subcluster = read_tsv(\"neuronal_subcluster_annotations.modified.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% (neuronal_subcluster   \n",
    "                                            %>% filter(annotation == \"Head epid. /muscle\"))$cell_name), size = \n",
    "           ifelse(df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \"Head epid. /muscle\"))$cell_name, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Head epid. /muscle\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% (neuronal_subcluster   \n",
    "                                            %>% filter(annotation == \"Immature neurons\"))$cell_name), size = \n",
    "           ifelse(df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \"Immature neurons\"))$cell_name, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Immature neurons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% (neuronal_subcluster   \n",
    "                                            %>% filter(annotation == \"Mature neurons\"))$cell_name), size = \n",
    "           ifelse(df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \"Mature neurons\"))$cell_name, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Mature neurons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% (neuronal_subcluster   \n",
    "                                            %>% filter(annotation == \"Immature neurons\"))$cell_name), size = \n",
    "           ifelse(df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \"Immature neurons\"))$cell_name, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Immature neurons\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% (neuronal_subcluster   \n",
    "                                            %>% filter(annotation == \"Neural progenitors\"))$cell_name), size = \n",
    "           ifelse(df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \"Neural progenitors\"))$cell_name, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Neural progenitors\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 630,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A spec_tbl_df: 8 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>annotation</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Germline           </td><td>   43</td></tr>\n",
       "\t<tr><td>Head epid. /muscle </td><td> 4486</td></tr>\n",
       "\t<tr><td>Immature neurons   </td><td>15602</td></tr>\n",
       "\t<tr><td>Mature neurons     </td><td> 6703</td></tr>\n",
       "\t<tr><td>Neural progenitors </td><td>24825</td></tr>\n",
       "\t<tr><td>Oenocytes          </td><td>  140</td></tr>\n",
       "\t<tr><td>Sensory progenitors</td><td> 5149</td></tr>\n",
       "\t<tr><td>Ventral midline    </td><td> 2064</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A spec\\_tbl\\_df: 8 × 2\n",
       "\\begin{tabular}{ll}\n",
       " annotation & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Germline            &    43\\\\\n",
       "\t Head epid. /muscle  &  4486\\\\\n",
       "\t Immature neurons    & 15602\\\\\n",
       "\t Mature neurons      &  6703\\\\\n",
       "\t Neural progenitors  & 24825\\\\\n",
       "\t Oenocytes           &   140\\\\\n",
       "\t Sensory progenitors &  5149\\\\\n",
       "\t Ventral midline     &  2064\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A spec_tbl_df: 8 × 2\n",
       "\n",
       "| annotation &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Germline            |    43 |\n",
       "| Head epid. /muscle  |  4486 |\n",
       "| Immature neurons    | 15602 |\n",
       "| Mature neurons      |  6703 |\n",
       "| Neural progenitors  | 24825 |\n",
       "| Oenocytes           |   140 |\n",
       "| Sensory progenitors |  5149 |\n",
       "| Ventral midline     |  2064 |\n",
       "\n"
      ],
      "text/plain": [
       "  annotation          n    \n",
       "1 Germline               43\n",
       "2 Head epid. /muscle   4486\n",
       "3 Immature neurons    15602\n",
       "4 Mature neurons       6703\n",
       "5 Neural progenitors  24825\n",
       "6 Oenocytes             140\n",
       "7 Sensory progenitors  5149\n",
       "8 Ventral midline      2064"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "neuronal_subcluster  %>% count(annotation)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% (neuronal_subcluster   \n",
    "                                            %>% filter(annotation == \"Oenocytes\"))$cell_name), size = \n",
    "           ifelse(df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \"Oenocytes\"))$cell_name, 2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Oenocytes\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% (neuronal_subcluster   \n",
    "                                            %>% filter(annotation == \"Sensory progenitors\"))$cell_name), size = \n",
    "           ifelse(df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \"Sensory progenitors\"))$cell_name, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Sensory progenitors\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cell %in% (neuronal_subcluster   \n",
    "                                            %>% filter(annotation == \"Ventral midline\"))$cell_name), size = \n",
    "           ifelse(df_sub300$cell %in% (neuronal_subsubcluster  \n",
    "                                            %>% filter(annotation == \"Ventral midline\"))$cell_name, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = \"Ventral midline\") +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 695,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'cds_sub'</li><li>'cds_sub_no_doubletons'</li><li>'cds_sub_no_multitons'</li><li>'cds_sub_no_sing'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'cds\\_sub'\n",
       "\\item 'cds\\_sub\\_no\\_doubletons'\n",
       "\\item 'cds\\_sub\\_no\\_multitons'\n",
       "\\item 'cds\\_sub\\_no\\_sing'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'cds_sub'\n",
       "2. 'cds_sub_no_doubletons'\n",
       "3. 'cds_sub_no_multitons'\n",
       "4. 'cds_sub_no_sing'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"cds_sub\"               \"cds_sub_no_doubletons\" \"cds_sub_no_multitons\" \n",
       "[4] \"cds_sub_no_sing\"      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ls(pattern = \"cds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 696,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene_short_name</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>gfzf</th><td>gfzf  </td></tr>\n",
       "\t<tr><th scope=row>Osi24</th><td>Osi24 </td></tr>\n",
       "\t<tr><th scope=row>msps</th><td>msps  </td></tr>\n",
       "\t<tr><th scope=row>CG6013</th><td>CG6013</td></tr>\n",
       "\t<tr><th scope=row>Dsk</th><td>Dsk   </td></tr>\n",
       "\t<tr><th scope=row>CG8043</th><td>CG8043</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & gene\\_short\\_name\\\\\n",
       "  & <chr>\\\\\n",
       "\\hline\n",
       "\tgfzf & gfzf  \\\\\n",
       "\tOsi24 & Osi24 \\\\\n",
       "\tmsps & msps  \\\\\n",
       "\tCG6013 & CG6013\\\\\n",
       "\tDsk & Dsk   \\\\\n",
       "\tCG8043 & CG8043\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 1\n",
       "\n",
       "| <!--/--> | gene_short_name &lt;chr&gt; |\n",
       "|---|---|\n",
       "| gfzf | gfzf   |\n",
       "| Osi24 | Osi24  |\n",
       "| msps | msps   |\n",
       "| CG6013 | CG6013 |\n",
       "| Dsk | Dsk    |\n",
       "| CG8043 | CG8043 |\n",
       "\n"
      ],
      "text/plain": [
       "       gene_short_name\n",
       "gfzf   gfzf           \n",
       "Osi24  Osi24          \n",
       "msps   msps           \n",
       "CG6013 CG6013         \n",
       "Dsk    Dsk            \n",
       "CG8043 CG8043         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "as.data.frame(rowData(cds_sub))  %>%  head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 697,
   "metadata": {},
   "outputs": [],
   "source": [
    "tf = \"GATAe\"\n",
    "ranklb = 0\n",
    "rankub = 1\n",
    "## set the cluster size range\n",
    "lb = 2\n",
    "ub = 277\n",
    "target_set = unique((fly_peaks_TF  %>% \n",
    "                            filter(TF == tf & \n",
    "                                   rank >= ranklb & rank <=rankub & \n",
    "                                   nPeaks > lb & nPeaks <= ub\n",
    "                                  & Gene != tf \n",
    "                                   & grepl(\"emb\", exp)\n",
    "                                  ))$Gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 698,
   "metadata": {},
   "outputs": [],
   "source": [
    "target_set_GATAe = target_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 699,
   "metadata": {},
   "outputs": [],
   "source": [
    "tf = \"crol\"\n",
    "ranklb = 0\n",
    "rankub = 1\n",
    "## set the cluster size range\n",
    "lb = 2\n",
    "ub = 277\n",
    "target_set = unique((fly_peaks_TF  %>% \n",
    "                            filter(TF == tf & \n",
    "                                   rank >= ranklb & rank <=rankub & \n",
    "                                   nPeaks > lb & nPeaks <= ub\n",
    "                                  & Gene != tf \n",
    "                                   & grepl(\"emb\", exp)\n",
    "                                  ))$Gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 700,
   "metadata": {},
   "outputs": [],
   "source": [
    "target_set_crol = target_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 701,
   "metadata": {},
   "outputs": [],
   "source": [
    "tf = \"Camta\"\n",
    "ranklb = 0\n",
    "rankub = 1\n",
    "## set the cluster size range\n",
    "lb = 2\n",
    "ub = 277\n",
    "target_set = unique((fly_peaks_TF  %>% \n",
    "                            filter(TF == tf & \n",
    "                                   rank >= ranklb & rank <=rankub & \n",
    "                                   nPeaks > lb & nPeaks <= ub\n",
    "                                  & Gene != tf \n",
    "                                   & grepl(\"emb\", exp)\n",
    "                                  ))$Gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 702,
   "metadata": {},
   "outputs": [],
   "source": [
    "target_set_Camta = target_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 711,
   "metadata": {},
   "outputs": [],
   "source": [
    "tf = \"sens\"\n",
    "ranklb = 0\n",
    "rankub = 1\n",
    "## set the cluster size range\n",
    "lb = 2\n",
    "ub = 277\n",
    "target_set = unique((fly_peaks_TF  %>% \n",
    "                            filter(TF == tf & \n",
    "                                   rank >= ranklb & rank <=rankub & \n",
    "                                   nPeaks > lb & nPeaks <= ub\n",
    "                                  & Gene != tf \n",
    "                                   & grepl(\"emb\", exp)\n",
    "                                  ))$Gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 712,
   "metadata": {},
   "outputs": [],
   "source": [
    "target_set_sens = target_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 713,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "1002"
      ],
      "text/latex": [
       "1002"
      ],
      "text/markdown": [
       "1002"
      ],
      "text/plain": [
       "[1] 1002"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "493"
      ],
      "text/latex": [
       "493"
      ],
      "text/markdown": [
       "493"
      ],
      "text/plain": [
       "[1] 493"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "2852"
      ],
      "text/latex": [
       "2852"
      ],
      "text/markdown": [
       "2852"
      ],
      "text/plain": [
       "[1] 2852"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "407"
      ],
      "text/latex": [
       "407"
      ],
      "text/markdown": [
       "407"
      ],
      "text/plain": [
       "[1] 407"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(target_set_GATAe)\n",
    "length(target_set_Camta)\n",
    "length(target_set_crol)\n",
    "length(target_set_sens)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 709,
   "metadata": {},
   "outputs": [],
   "source": [
    "options(repr.matrix.max.cols = 100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 710,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 33</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>peakID</th><th scope=col>absDist</th><th scope=col>chromo.x</th><th scope=col>start.x</th><th scope=col>end.x</th><th scope=col>exp</th><th scope=col>score</th><th scope=col>strand.x</th><th scope=col>apexS.x</th><th scope=col>apexE.x</th><th scope=col>color.x</th><th scope=col>cluster</th><th scope=col>target</th><th scope=col>targetDist</th><th scope=col>chromo.y</th><th scope=col>start.y</th><th scope=col>end.y</th><th scope=col>nPeaks</th><th scope=col>strand.y</th><th scope=col>apexS.y</th><th scope=col>apexE.y</th><th scope=col>color.y</th><th scope=col>TranscriptCoord</th><th scope=col>Transcript</th><th scope=col>TranscriptLoc</th><th scope=col>Gene</th><th scope=col>GeneLoc</th><th scope=col>GeneExons</th><th scope=col>TranscriptNo</th><th scope=col>UniqueGeneTSSs</th><th scope=col>GeneTranscripts</th><th scope=col>TF</th><th scope=col>rank</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>GATAe_GATAe-GFP_embryonic_1chr2L10003641:10004001</td><td>  11</td><td>chr2L</td><td>10003641</td><td>10004001</td><td>GATAe_GATAe-GFP_embryonic_1</td><td> 856</td><td>.</td><td>10003819</td><td>10003823</td><td>0,255,0</td><td>39_narrowPeak1435</td><td>Ndf-RA::5P   </td><td> -11</td><td>chr2L</td><td>10003248</td><td>10004300</td><td>487</td><td>.</td><td>10003611</td><td>10003615</td><td>0,255,0</td><td>2L:10003832:10008889:+</td><td>Ndf-RA   </td><td>5P</td><td>Ndf   </td><td>5P</td><td> 9</td><td>1</td><td>1</td><td>3</td><td>GATAe</td><td>0.2848704</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>GATAe_GATAe-GFP_embryonic_1chr2L10010030:10010390</td><td> 144</td><td>chr2L</td><td>10010030</td><td>10010390</td><td>GATAe_GATAe-GFP_embryonic_1</td><td>1000</td><td>.</td><td>10010208</td><td>10010212</td><td>0,0,255</td><td>42_narrowPeak1395</td><td>Trp1-RB::I1  </td><td>-144</td><td>chr2L</td><td>10009842</td><td>10010733</td><td>552</td><td>.</td><td>10010160</td><td>10010164</td><td>0,0,255</td><td>2L:10010354:10013170:+</td><td>Trp1-RB  </td><td>5P</td><td>Trp1  </td><td>I1</td><td> 5</td><td>2</td><td>2</td><td>4</td><td>GATAe</td><td>0.8562521</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>GATAe_GATAe-GFP_embryonic_1chr2L10200888:10201248</td><td>  32</td><td>chr2L</td><td>10200888</td><td>10201248</td><td>GATAe_GATAe-GFP_embryonic_1</td><td>1000</td><td>.</td><td>10201066</td><td>10201070</td><td>0,255,0</td><td>7_narrowPeak2571 </td><td>RpL7-RA::E1  </td><td> -32</td><td>chr2L</td><td>10200412</td><td>10201648</td><td>657</td><td>.</td><td>10200836</td><td>10200840</td><td>0,255,0</td><td>2L:10201100:10202145:+</td><td>RpL7-RA  </td><td>5P</td><td>RpL7  </td><td>E1</td><td> 3</td><td>2</td><td>2</td><td>2</td><td>GATAe</td><td>0.8099770</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>GATAe_GATAe-GFP_embryonic_1chr2L10334905:10335056</td><td>1109</td><td>chr2L</td><td>10334905</td><td>10335056</td><td>GATAe_GATAe-GFP_embryonic_1</td><td>1000</td><td>.</td><td>10335021</td><td>10335025</td><td>0,255,0</td><td>13_narrowPeak2152</td><td>GATAd-RC::I1 </td><td>1109</td><td>chr2L</td><td>10334370</td><td>10335803</td><td>391</td><td>.</td><td>10334963</td><td>10334967</td><td>0,255,0</td><td>2L:10333914:10339185:+</td><td>GATAd-RC </td><td>I1</td><td>GATAd </td><td>I1</td><td> 8</td><td>2</td><td>2</td><td>3</td><td>GATAe</td><td>0.9271415</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>GATAe_GATAe-GFP_embryonic_1chr2L10381942:10382274</td><td> 144</td><td>chr2L</td><td>10381942</td><td>10382274</td><td>GATAe_GATAe-GFP_embryonic_1</td><td>1000</td><td>.</td><td>10382117</td><td>10382121</td><td>0,0,255</td><td>2_narrowPeak4097 </td><td>CG5096-RA::5P</td><td>-144</td><td>chr2L</td><td>10381670</td><td>10383136</td><td>429</td><td>.</td><td>10382092</td><td>10382096</td><td>0,0,255</td><td>2L:10382263:10384723:+</td><td>CG5096-RA</td><td>5P</td><td>CG5096</td><td>5P</td><td> 6</td><td>1</td><td>1</td><td>1</td><td>GATAe</td><td>0.9977027</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>GATAe_GATAe-GFP_embryonic_1chr2L10474484:10474622</td><td> 165</td><td>chr2L</td><td>10474484</td><td>10474622</td><td>GATAe_GATAe-GFP_embryonic_1</td><td>1000</td><td>.</td><td>10474512</td><td>10474516</td><td>255,0,0</td><td>2_narrowPeak4138 </td><td>RluA-1-RA::5P</td><td>-165</td><td>chr2L</td><td>10473891</td><td>10475100</td><td>626</td><td>.</td><td>10474516</td><td>10474520</td><td>255,0,0</td><td>2L:10463340:10474349:-</td><td>RluA-1-RA</td><td>5P</td><td>RluA-1</td><td>5P</td><td>11</td><td>1</td><td>1</td><td>4</td><td>GATAe</td><td>0.9606170</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 33\n",
       "\\begin{tabular}{r|lllllllllllllllllllllllllllllllll}\n",
       "  & peakID & absDist & chromo.x & start.x & end.x & exp & score & strand.x & apexS.x & apexE.x & color.x & cluster & target & targetDist & chromo.y & start.y & end.y & nPeaks & strand.y & apexS.y & apexE.y & color.y & TranscriptCoord & Transcript & TranscriptLoc & Gene & GeneLoc & GeneExons & TranscriptNo & UniqueGeneTSSs & GeneTranscripts & TF & rank\\\\\n",
       "  & <chr> & <dbl> & <chr> & <dbl> & <dbl> & <chr> & <dbl> & <chr> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <dbl> & <chr> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & GATAe\\_GATAe-GFP\\_embryonic\\_1chr2L10003641:10004001 &   11 & chr2L & 10003641 & 10004001 & GATAe\\_GATAe-GFP\\_embryonic\\_1 &  856 & . & 10003819 & 10003823 & 0,255,0 & 39\\_narrowPeak1435 & Ndf-RA::5P    &  -11 & chr2L & 10003248 & 10004300 & 487 & . & 10003611 & 10003615 & 0,255,0 & 2L:10003832:10008889:+ & Ndf-RA    & 5P & Ndf    & 5P &  9 & 1 & 1 & 3 & GATAe & 0.2848704\\\\\n",
       "\t2 & GATAe\\_GATAe-GFP\\_embryonic\\_1chr2L10010030:10010390 &  144 & chr2L & 10010030 & 10010390 & GATAe\\_GATAe-GFP\\_embryonic\\_1 & 1000 & . & 10010208 & 10010212 & 0,0,255 & 42\\_narrowPeak1395 & Trp1-RB::I1   & -144 & chr2L & 10009842 & 10010733 & 552 & . & 10010160 & 10010164 & 0,0,255 & 2L:10010354:10013170:+ & Trp1-RB   & 5P & Trp1   & I1 &  5 & 2 & 2 & 4 & GATAe & 0.8562521\\\\\n",
       "\t3 & GATAe\\_GATAe-GFP\\_embryonic\\_1chr2L10200888:10201248 &   32 & chr2L & 10200888 & 10201248 & GATAe\\_GATAe-GFP\\_embryonic\\_1 & 1000 & . & 10201066 & 10201070 & 0,255,0 & 7\\_narrowPeak2571  & RpL7-RA::E1   &  -32 & chr2L & 10200412 & 10201648 & 657 & . & 10200836 & 10200840 & 0,255,0 & 2L:10201100:10202145:+ & RpL7-RA   & 5P & RpL7   & E1 &  3 & 2 & 2 & 2 & GATAe & 0.8099770\\\\\n",
       "\t4 & GATAe\\_GATAe-GFP\\_embryonic\\_1chr2L10334905:10335056 & 1109 & chr2L & 10334905 & 10335056 & GATAe\\_GATAe-GFP\\_embryonic\\_1 & 1000 & . & 10335021 & 10335025 & 0,255,0 & 13\\_narrowPeak2152 & GATAd-RC::I1  & 1109 & chr2L & 10334370 & 10335803 & 391 & . & 10334963 & 10334967 & 0,255,0 & 2L:10333914:10339185:+ & GATAd-RC  & I1 & GATAd  & I1 &  8 & 2 & 2 & 3 & GATAe & 0.9271415\\\\\n",
       "\t5 & GATAe\\_GATAe-GFP\\_embryonic\\_1chr2L10381942:10382274 &  144 & chr2L & 10381942 & 10382274 & GATAe\\_GATAe-GFP\\_embryonic\\_1 & 1000 & . & 10382117 & 10382121 & 0,0,255 & 2\\_narrowPeak4097  & CG5096-RA::5P & -144 & chr2L & 10381670 & 10383136 & 429 & . & 10382092 & 10382096 & 0,0,255 & 2L:10382263:10384723:+ & CG5096-RA & 5P & CG5096 & 5P &  6 & 1 & 1 & 1 & GATAe & 0.9977027\\\\\n",
       "\t6 & GATAe\\_GATAe-GFP\\_embryonic\\_1chr2L10474484:10474622 &  165 & chr2L & 10474484 & 10474622 & GATAe\\_GATAe-GFP\\_embryonic\\_1 & 1000 & . & 10474512 & 10474516 & 255,0,0 & 2\\_narrowPeak4138  & RluA-1-RA::5P & -165 & chr2L & 10473891 & 10475100 & 626 & . & 10474516 & 10474520 & 255,0,0 & 2L:10463340:10474349:- & RluA-1-RA & 5P & RluA-1 & 5P & 11 & 1 & 1 & 4 & GATAe & 0.9606170\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 33\n",
       "\n",
       "| <!--/--> | peakID &lt;chr&gt; | absDist &lt;dbl&gt; | chromo.x &lt;chr&gt; | start.x &lt;dbl&gt; | end.x &lt;dbl&gt; | exp &lt;chr&gt; | score &lt;dbl&gt; | strand.x &lt;chr&gt; | apexS.x &lt;dbl&gt; | apexE.x &lt;dbl&gt; | color.x &lt;chr&gt; | cluster &lt;chr&gt; | target &lt;chr&gt; | targetDist &lt;dbl&gt; | chromo.y &lt;chr&gt; | start.y &lt;dbl&gt; | end.y &lt;dbl&gt; | nPeaks &lt;dbl&gt; | strand.y &lt;chr&gt; | apexS.y &lt;dbl&gt; | apexE.y &lt;dbl&gt; | color.y &lt;chr&gt; | TranscriptCoord &lt;chr&gt; | Transcript &lt;chr&gt; | TranscriptLoc &lt;chr&gt; | Gene &lt;chr&gt; | GeneLoc &lt;chr&gt; | GeneExons &lt;dbl&gt; | TranscriptNo &lt;dbl&gt; | UniqueGeneTSSs &lt;dbl&gt; | GeneTranscripts &lt;dbl&gt; | TF &lt;chr&gt; | rank &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | GATAe_GATAe-GFP_embryonic_1chr2L10003641:10004001 |   11 | chr2L | 10003641 | 10004001 | GATAe_GATAe-GFP_embryonic_1 |  856 | . | 10003819 | 10003823 | 0,255,0 | 39_narrowPeak1435 | Ndf-RA::5P    |  -11 | chr2L | 10003248 | 10004300 | 487 | . | 10003611 | 10003615 | 0,255,0 | 2L:10003832:10008889:+ | Ndf-RA    | 5P | Ndf    | 5P |  9 | 1 | 1 | 3 | GATAe | 0.2848704 |\n",
       "| 2 | GATAe_GATAe-GFP_embryonic_1chr2L10010030:10010390 |  144 | chr2L | 10010030 | 10010390 | GATAe_GATAe-GFP_embryonic_1 | 1000 | . | 10010208 | 10010212 | 0,0,255 | 42_narrowPeak1395 | Trp1-RB::I1   | -144 | chr2L | 10009842 | 10010733 | 552 | . | 10010160 | 10010164 | 0,0,255 | 2L:10010354:10013170:+ | Trp1-RB   | 5P | Trp1   | I1 |  5 | 2 | 2 | 4 | GATAe | 0.8562521 |\n",
       "| 3 | GATAe_GATAe-GFP_embryonic_1chr2L10200888:10201248 |   32 | chr2L | 10200888 | 10201248 | GATAe_GATAe-GFP_embryonic_1 | 1000 | . | 10201066 | 10201070 | 0,255,0 | 7_narrowPeak2571  | RpL7-RA::E1   |  -32 | chr2L | 10200412 | 10201648 | 657 | . | 10200836 | 10200840 | 0,255,0 | 2L:10201100:10202145:+ | RpL7-RA   | 5P | RpL7   | E1 |  3 | 2 | 2 | 2 | GATAe | 0.8099770 |\n",
       "| 4 | GATAe_GATAe-GFP_embryonic_1chr2L10334905:10335056 | 1109 | chr2L | 10334905 | 10335056 | GATAe_GATAe-GFP_embryonic_1 | 1000 | . | 10335021 | 10335025 | 0,255,0 | 13_narrowPeak2152 | GATAd-RC::I1  | 1109 | chr2L | 10334370 | 10335803 | 391 | . | 10334963 | 10334967 | 0,255,0 | 2L:10333914:10339185:+ | GATAd-RC  | I1 | GATAd  | I1 |  8 | 2 | 2 | 3 | GATAe | 0.9271415 |\n",
       "| 5 | GATAe_GATAe-GFP_embryonic_1chr2L10381942:10382274 |  144 | chr2L | 10381942 | 10382274 | GATAe_GATAe-GFP_embryonic_1 | 1000 | . | 10382117 | 10382121 | 0,0,255 | 2_narrowPeak4097  | CG5096-RA::5P | -144 | chr2L | 10381670 | 10383136 | 429 | . | 10382092 | 10382096 | 0,0,255 | 2L:10382263:10384723:+ | CG5096-RA | 5P | CG5096 | 5P |  6 | 1 | 1 | 1 | GATAe | 0.9977027 |\n",
       "| 6 | GATAe_GATAe-GFP_embryonic_1chr2L10474484:10474622 |  165 | chr2L | 10474484 | 10474622 | GATAe_GATAe-GFP_embryonic_1 | 1000 | . | 10474512 | 10474516 | 255,0,0 | 2_narrowPeak4138  | RluA-1-RA::5P | -165 | chr2L | 10473891 | 10475100 | 626 | . | 10474516 | 10474520 | 255,0,0 | 2L:10463340:10474349:- | RluA-1-RA | 5P | RluA-1 | 5P | 11 | 1 | 1 | 4 | GATAe | 0.9606170 |\n",
       "\n"
      ],
      "text/plain": [
       "  peakID                                            absDist chromo.x start.x \n",
       "1 GATAe_GATAe-GFP_embryonic_1chr2L10003641:10004001   11    chr2L    10003641\n",
       "2 GATAe_GATAe-GFP_embryonic_1chr2L10010030:10010390  144    chr2L    10010030\n",
       "3 GATAe_GATAe-GFP_embryonic_1chr2L10200888:10201248   32    chr2L    10200888\n",
       "4 GATAe_GATAe-GFP_embryonic_1chr2L10334905:10335056 1109    chr2L    10334905\n",
       "5 GATAe_GATAe-GFP_embryonic_1chr2L10381942:10382274  144    chr2L    10381942\n",
       "6 GATAe_GATAe-GFP_embryonic_1chr2L10474484:10474622  165    chr2L    10474484\n",
       "  end.x    exp                         score strand.x apexS.x  apexE.x  color.x\n",
       "1 10004001 GATAe_GATAe-GFP_embryonic_1  856  .        10003819 10003823 0,255,0\n",
       "2 10010390 GATAe_GATAe-GFP_embryonic_1 1000  .        10010208 10010212 0,0,255\n",
       "3 10201248 GATAe_GATAe-GFP_embryonic_1 1000  .        10201066 10201070 0,255,0\n",
       "4 10335056 GATAe_GATAe-GFP_embryonic_1 1000  .        10335021 10335025 0,255,0\n",
       "5 10382274 GATAe_GATAe-GFP_embryonic_1 1000  .        10382117 10382121 0,0,255\n",
       "6 10474622 GATAe_GATAe-GFP_embryonic_1 1000  .        10474512 10474516 255,0,0\n",
       "  cluster           target        targetDist chromo.y start.y  end.y    nPeaks\n",
       "1 39_narrowPeak1435 Ndf-RA::5P     -11       chr2L    10003248 10004300 487   \n",
       "2 42_narrowPeak1395 Trp1-RB::I1   -144       chr2L    10009842 10010733 552   \n",
       "3 7_narrowPeak2571  RpL7-RA::E1    -32       chr2L    10200412 10201648 657   \n",
       "4 13_narrowPeak2152 GATAd-RC::I1  1109       chr2L    10334370 10335803 391   \n",
       "5 2_narrowPeak4097  CG5096-RA::5P -144       chr2L    10381670 10383136 429   \n",
       "6 2_narrowPeak4138  RluA-1-RA::5P -165       chr2L    10473891 10475100 626   \n",
       "  strand.y apexS.y  apexE.y  color.y TranscriptCoord        Transcript\n",
       "1 .        10003611 10003615 0,255,0 2L:10003832:10008889:+ Ndf-RA    \n",
       "2 .        10010160 10010164 0,0,255 2L:10010354:10013170:+ Trp1-RB   \n",
       "3 .        10200836 10200840 0,255,0 2L:10201100:10202145:+ RpL7-RA   \n",
       "4 .        10334963 10334967 0,255,0 2L:10333914:10339185:+ GATAd-RC  \n",
       "5 .        10382092 10382096 0,0,255 2L:10382263:10384723:+ CG5096-RA \n",
       "6 .        10474516 10474520 255,0,0 2L:10463340:10474349:- RluA-1-RA \n",
       "  TranscriptLoc Gene   GeneLoc GeneExons TranscriptNo UniqueGeneTSSs\n",
       "1 5P            Ndf    5P       9        1            1             \n",
       "2 5P            Trp1   I1       5        2            2             \n",
       "3 5P            RpL7   E1       3        2            2             \n",
       "4 I1            GATAd  I1       8        2            2             \n",
       "5 5P            CG5096 5P       6        1            1             \n",
       "6 5P            RluA-1 5P      11        1            1             \n",
       "  GeneTranscripts TF    rank     \n",
       "1 3               GATAe 0.2848704\n",
       "2 4               GATAe 0.8562521\n",
       "3 2               GATAe 0.8099770\n",
       "4 3               GATAe 0.9271415\n",
       "5 1               GATAe 0.9977027\n",
       "6 4               GATAe 0.9606170"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fly_peaks_TF  %>% filter(nPeaks >= 277 & TF == \"GATAe\")  %>% head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Reannotate fly cells using same bins but reame with Sue's names"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Also add in subdivisions from neural annotations from Shendure paper."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 717,
   "metadata": {},
   "outputs": [],
   "source": [
    "load(\"data/markers for fly datasets.RData\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 718,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'cluster.markers'</li><li>'cluster.markers.sub300'</li><li>'cluster.markers.sub300_no_doubletons'</li><li>'cluster.markers.sub300_no_doubletons_all'</li><li>'cluster.markers.sub300_no_singletons'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'cluster.markers'\n",
       "\\item 'cluster.markers.sub300'\n",
       "\\item 'cluster.markers.sub300\\_no\\_doubletons'\n",
       "\\item 'cluster.markers.sub300\\_no\\_doubletons\\_all'\n",
       "\\item 'cluster.markers.sub300\\_no\\_singletons'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'cluster.markers'\n",
       "2. 'cluster.markers.sub300'\n",
       "3. 'cluster.markers.sub300_no_doubletons'\n",
       "4. 'cluster.markers.sub300_no_doubletons_all'\n",
       "5. 'cluster.markers.sub300_no_singletons'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"cluster.markers\"                         \n",
       "[2] \"cluster.markers.sub300\"                  \n",
       "[3] \"cluster.markers.sub300_no_doubletons\"    \n",
       "[4] \"cluster.markers.sub300_no_doubletons_all\"\n",
       "[5] \"cluster.markers.sub300_no_singletons\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ls(pattern = \"cluster.markers\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 10 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cluster</th><th scope=col>gene</th><th scope=col>cluster.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cluster.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG918453</th><td>54 </td><td>CG9184 </td><td>7147</td><td>35616</td><td>14831.3446</td><td>1522.1598978</td><td>3.283510</td><td>45.49782</td></tr>\n",
       "\t<tr><th scope=row>CG1059153</th><td>54 </td><td>CG10591</td><td>5091</td><td>23244</td><td>10405.6717</td><td> 991.7346141</td><td>3.389818</td><td>45.23787</td></tr>\n",
       "\t<tr><th scope=row>CG5715149</th><td>150</td><td>CG5715 </td><td>  32</td><td>   17</td><td>  138.1496</td><td>   0.9084849</td><td>6.177659</td><td>43.98798</td></tr>\n",
       "\t<tr><th scope=row>sr143</th><td>144</td><td>sr     </td><td> 591</td><td> 3816</td><td> 2365.5387</td><td> 165.4169404</td><td>3.829295</td><td>42.92089</td></tr>\n",
       "\t<tr><th scope=row>bbg14</th><td>15 </td><td>bbg    </td><td>4838</td><td>15567</td><td> 7143.5268</td><td> 700.3065817</td><td>3.348519</td><td>42.86983</td></tr>\n",
       "\t<tr><th scope=row>scro56</th><td>57 </td><td>scro   </td><td> 598</td><td> 1464</td><td> 1232.0042</td><td>  67.3759216</td><td>4.171375</td><td>42.83152</td></tr>\n",
       "\t<tr><th scope=row>lmd99</th><td>100</td><td>lmd    </td><td> 691</td><td> 3111</td><td> 1997.0907</td><td> 132.5558449</td><td>3.902385</td><td>42.78733</td></tr>\n",
       "\t<tr><th scope=row>Desi53</th><td>54 </td><td>Desi   </td><td>  89</td><td>   86</td><td>  182.2837</td><td>   2.5356873</td><td>5.688051</td><td>42.76239</td></tr>\n",
       "\t<tr><th scope=row>ImpE253</th><td>54 </td><td>ImpE2  </td><td>3061</td><td>14790</td><td> 6225.0365</td><td> 603.3959499</td><td>3.364516</td><td>42.40670</td></tr>\n",
       "\t<tr><th scope=row>sha14</th><td>15 </td><td>sha    </td><td>1164</td><td> 2633</td><td> 1781.7802</td><td> 119.2874267</td><td>3.888762</td><td>41.99829</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 10 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cluster & gene & cluster.n.umi & other.n.umi & cluster.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <fct> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG918453 & 54  & CG9184  & 7147 & 35616 & 14831.3446 & 1522.1598978 & 3.283510 & 45.49782\\\\\n",
       "\tCG1059153 & 54  & CG10591 & 5091 & 23244 & 10405.6717 &  991.7346141 & 3.389818 & 45.23787\\\\\n",
       "\tCG5715149 & 150 & CG5715  &   32 &    17 &   138.1496 &    0.9084849 & 6.177659 & 43.98798\\\\\n",
       "\tsr143 & 144 & sr      &  591 &  3816 &  2365.5387 &  165.4169404 & 3.829295 & 42.92089\\\\\n",
       "\tbbg14 & 15  & bbg     & 4838 & 15567 &  7143.5268 &  700.3065817 & 3.348519 & 42.86983\\\\\n",
       "\tscro56 & 57  & scro    &  598 &  1464 &  1232.0042 &   67.3759216 & 4.171375 & 42.83152\\\\\n",
       "\tlmd99 & 100 & lmd     &  691 &  3111 &  1997.0907 &  132.5558449 & 3.902385 & 42.78733\\\\\n",
       "\tDesi53 & 54  & Desi    &   89 &    86 &   182.2837 &    2.5356873 & 5.688051 & 42.76239\\\\\n",
       "\tImpE253 & 54  & ImpE2   & 3061 & 14790 &  6225.0365 &  603.3959499 & 3.364516 & 42.40670\\\\\n",
       "\tsha14 & 15  & sha     & 1164 &  2633 &  1781.7802 &  119.2874267 & 3.888762 & 41.99829\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 10 × 8\n",
       "\n",
       "| <!--/--> | cluster &lt;fct&gt; | gene &lt;chr&gt; | cluster.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cluster.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG918453 | 54  | CG9184  | 7147 | 35616 | 14831.3446 | 1522.1598978 | 3.283510 | 45.49782 |\n",
       "| CG1059153 | 54  | CG10591 | 5091 | 23244 | 10405.6717 |  991.7346141 | 3.389818 | 45.23787 |\n",
       "| CG5715149 | 150 | CG5715  |   32 |    17 |   138.1496 |    0.9084849 | 6.177659 | 43.98798 |\n",
       "| sr143 | 144 | sr      |  591 |  3816 |  2365.5387 |  165.4169404 | 3.829295 | 42.92089 |\n",
       "| bbg14 | 15  | bbg     | 4838 | 15567 |  7143.5268 |  700.3065817 | 3.348519 | 42.86983 |\n",
       "| scro56 | 57  | scro    |  598 |  1464 |  1232.0042 |   67.3759216 | 4.171375 | 42.83152 |\n",
       "| lmd99 | 100 | lmd     |  691 |  3111 |  1997.0907 |  132.5558449 | 3.902385 | 42.78733 |\n",
       "| Desi53 | 54  | Desi    |   89 |    86 |   182.2837 |    2.5356873 | 5.688051 | 42.76239 |\n",
       "| ImpE253 | 54  | ImpE2   | 3061 | 14790 |  6225.0365 |  603.3959499 | 3.364516 | 42.40670 |\n",
       "| sha14 | 15  | sha     | 1164 |  2633 |  1781.7802 |  119.2874267 | 3.888762 | 41.99829 |\n",
       "\n"
      ],
      "text/plain": [
       "          cluster gene    cluster.n.umi other.n.umi cluster.tpm other.tpm   \n",
       "CG918453  54      CG9184  7147          35616       14831.3446  1522.1598978\n",
       "CG1059153 54      CG10591 5091          23244       10405.6717   991.7346141\n",
       "CG5715149 150     CG5715    32             17         138.1496     0.9084849\n",
       "sr143     144     sr       591           3816        2365.5387   165.4169404\n",
       "bbg14     15      bbg     4838          15567        7143.5268   700.3065817\n",
       "scro56    57      scro     598           1464        1232.0042    67.3759216\n",
       "lmd99     100     lmd      691           3111        1997.0907   132.5558449\n",
       "Desi53    54      Desi      89             86         182.2837     2.5356873\n",
       "ImpE253   54      ImpE2   3061          14790        6225.0365   603.3959499\n",
       "sha14     15      sha     1164           2633        1781.7802   119.2874267\n",
       "          log2.ratio heuristic.score\n",
       "CG918453  3.283510   45.49782       \n",
       "CG1059153 3.389818   45.23787       \n",
       "CG5715149 6.177659   43.98798       \n",
       "sr143     3.829295   42.92089       \n",
       "bbg14     3.348519   42.86983       \n",
       "scro56    4.171375   42.83152       \n",
       "lmd99     3.902385   42.78733       \n",
       "Desi53    5.688051   42.76239       \n",
       "ImpE253   3.364516   42.40670       \n",
       "sha14     3.888762   41.99829       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cluster.markers.sub300_no_singletons %>% filter(!(cluster %in% c(160, 163, 103, 172, 168, \n",
    "                                                              170, 14, 8,\n",
    "                                                              141, 151, 146, 165,\n",
    "                                                              85, 41, 99, 110, 31, 60, 111, 169,\n",
    "                                                              51, 95, 73, 1, 174, 2,\n",
    "                                                              171, 20, 98, 45, 96, 92, 130, 88,\n",
    "                                                              13, 55, 167, 104, 123,  25, 129,\n",
    "                                                              157, 122, 58, 28, 89, 148, 159,\n",
    "                                                              142, 154, 124, 149, 126, 113,\n",
    "                                                              81, 139, 34, 147, 166, 107,\n",
    "                                                              155, 173, 40, 80, 161, 120, 162, \n",
    "                                                               134, 156, 115, 56, 30, 128, 137, 158,\n",
    "                                                              121, 109, 94, 145, 68, 86, 114, 17, 63,\n",
    "                                                              47, 164, 166, 5, 66, 39, 9, 106,\n",
    "                                                              10, 152,  36, 127, 12, 75, 46, 133, 136, 119, 48,\n",
    "                                                              49, 64, 72, 140) ))  %>%  \n",
    "head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 721,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 10 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cluster</th><th scope=col>gene</th><th scope=col>cluster.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cluster.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG918453</th><td>54 </td><td>CG9184 </td><td>7147</td><td>35616</td><td>14831.3446</td><td>1522.1598978</td><td>3.283510</td><td>45.49782</td></tr>\n",
       "\t<tr><th scope=row>CG1059153</th><td>54 </td><td>CG10591</td><td>5091</td><td>23244</td><td>10405.6717</td><td> 991.7346141</td><td>3.389818</td><td>45.23787</td></tr>\n",
       "\t<tr><th scope=row>CG5715149</th><td>150</td><td>CG5715 </td><td>  32</td><td>   17</td><td>  138.1496</td><td>   0.9084849</td><td>6.177659</td><td>43.98798</td></tr>\n",
       "\t<tr><th scope=row>sr143</th><td>144</td><td>sr     </td><td> 591</td><td> 3816</td><td> 2365.5387</td><td> 165.4169404</td><td>3.829295</td><td>42.92089</td></tr>\n",
       "\t<tr><th scope=row>bbg14</th><td>15 </td><td>bbg    </td><td>4838</td><td>15567</td><td> 7143.5268</td><td> 700.3065817</td><td>3.348519</td><td>42.86983</td></tr>\n",
       "\t<tr><th scope=row>scro56</th><td>57 </td><td>scro   </td><td> 598</td><td> 1464</td><td> 1232.0042</td><td>  67.3759216</td><td>4.171375</td><td>42.83152</td></tr>\n",
       "\t<tr><th scope=row>lmd99</th><td>100</td><td>lmd    </td><td> 691</td><td> 3111</td><td> 1997.0907</td><td> 132.5558449</td><td>3.902385</td><td>42.78733</td></tr>\n",
       "\t<tr><th scope=row>Desi53</th><td>54 </td><td>Desi   </td><td>  89</td><td>   86</td><td>  182.2837</td><td>   2.5356873</td><td>5.688051</td><td>42.76239</td></tr>\n",
       "\t<tr><th scope=row>ImpE253</th><td>54 </td><td>ImpE2  </td><td>3061</td><td>14790</td><td> 6225.0365</td><td> 603.3959499</td><td>3.364516</td><td>42.40670</td></tr>\n",
       "\t<tr><th scope=row>sha14</th><td>15 </td><td>sha    </td><td>1164</td><td> 2633</td><td> 1781.7802</td><td> 119.2874267</td><td>3.888762</td><td>41.99829</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 10 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cluster & gene & cluster.n.umi & other.n.umi & cluster.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <fct> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG918453 & 54  & CG9184  & 7147 & 35616 & 14831.3446 & 1522.1598978 & 3.283510 & 45.49782\\\\\n",
       "\tCG1059153 & 54  & CG10591 & 5091 & 23244 & 10405.6717 &  991.7346141 & 3.389818 & 45.23787\\\\\n",
       "\tCG5715149 & 150 & CG5715  &   32 &    17 &   138.1496 &    0.9084849 & 6.177659 & 43.98798\\\\\n",
       "\tsr143 & 144 & sr      &  591 &  3816 &  2365.5387 &  165.4169404 & 3.829295 & 42.92089\\\\\n",
       "\tbbg14 & 15  & bbg     & 4838 & 15567 &  7143.5268 &  700.3065817 & 3.348519 & 42.86983\\\\\n",
       "\tscro56 & 57  & scro    &  598 &  1464 &  1232.0042 &   67.3759216 & 4.171375 & 42.83152\\\\\n",
       "\tlmd99 & 100 & lmd     &  691 &  3111 &  1997.0907 &  132.5558449 & 3.902385 & 42.78733\\\\\n",
       "\tDesi53 & 54  & Desi    &   89 &    86 &   182.2837 &    2.5356873 & 5.688051 & 42.76239\\\\\n",
       "\tImpE253 & 54  & ImpE2   & 3061 & 14790 &  6225.0365 &  603.3959499 & 3.364516 & 42.40670\\\\\n",
       "\tsha14 & 15  & sha     & 1164 &  2633 &  1781.7802 &  119.2874267 & 3.888762 & 41.99829\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 10 × 8\n",
       "\n",
       "| <!--/--> | cluster &lt;fct&gt; | gene &lt;chr&gt; | cluster.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cluster.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG918453 | 54  | CG9184  | 7147 | 35616 | 14831.3446 | 1522.1598978 | 3.283510 | 45.49782 |\n",
       "| CG1059153 | 54  | CG10591 | 5091 | 23244 | 10405.6717 |  991.7346141 | 3.389818 | 45.23787 |\n",
       "| CG5715149 | 150 | CG5715  |   32 |    17 |   138.1496 |    0.9084849 | 6.177659 | 43.98798 |\n",
       "| sr143 | 144 | sr      |  591 |  3816 |  2365.5387 |  165.4169404 | 3.829295 | 42.92089 |\n",
       "| bbg14 | 15  | bbg     | 4838 | 15567 |  7143.5268 |  700.3065817 | 3.348519 | 42.86983 |\n",
       "| scro56 | 57  | scro    |  598 |  1464 |  1232.0042 |   67.3759216 | 4.171375 | 42.83152 |\n",
       "| lmd99 | 100 | lmd     |  691 |  3111 |  1997.0907 |  132.5558449 | 3.902385 | 42.78733 |\n",
       "| Desi53 | 54  | Desi    |   89 |    86 |   182.2837 |    2.5356873 | 5.688051 | 42.76239 |\n",
       "| ImpE253 | 54  | ImpE2   | 3061 | 14790 |  6225.0365 |  603.3959499 | 3.364516 | 42.40670 |\n",
       "| sha14 | 15  | sha     | 1164 |  2633 |  1781.7802 |  119.2874267 | 3.888762 | 41.99829 |\n",
       "\n"
      ],
      "text/plain": [
       "          cluster gene    cluster.n.umi other.n.umi cluster.tpm other.tpm   \n",
       "CG918453  54      CG9184  7147          35616       14831.3446  1522.1598978\n",
       "CG1059153 54      CG10591 5091          23244       10405.6717   991.7346141\n",
       "CG5715149 150     CG5715    32             17         138.1496     0.9084849\n",
       "sr143     144     sr       591           3816        2365.5387   165.4169404\n",
       "bbg14     15      bbg     4838          15567        7143.5268   700.3065817\n",
       "scro56    57      scro     598           1464        1232.0042    67.3759216\n",
       "lmd99     100     lmd      691           3111        1997.0907   132.5558449\n",
       "Desi53    54      Desi      89             86         182.2837     2.5356873\n",
       "ImpE253   54      ImpE2   3061          14790        6225.0365   603.3959499\n",
       "sha14     15      sha     1164           2633        1781.7802   119.2874267\n",
       "          log2.ratio heuristic.score\n",
       "CG918453  3.283510   45.49782       \n",
       "CG1059153 3.389818   45.23787       \n",
       "CG5715149 6.177659   43.98798       \n",
       "sr143     3.829295   42.92089       \n",
       "bbg14     3.348519   42.86983       \n",
       "scro56    4.171375   42.83152       \n",
       "lmd99     3.902385   42.78733       \n",
       "Desi53    5.688051   42.76239       \n",
       "ImpE253   3.364516   42.40670       \n",
       "sha14     3.888762   41.99829       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cluster.markers.sub300_no_singletons  %>% filter(!(cluster %in% c(160, 163, 103, 172, 168, \n",
    "                                                              170, 14, 8,\n",
    "                                                              141, 151, 146, 165,\n",
    "                                                              85, 41, 99, 110, 31, 60, 111, 169,\n",
    "                                                              51, 95, 73, 1, 174, 2,\n",
    "                                                              171, 20, 98, 45, 96, 92, 130, 88,\n",
    "                                                              13, 55, 167, 104, 123,  25, 129,\n",
    "                                                              157, 122, 58, 28, 89, 148, 159,\n",
    "                                                              142, 154, 124, 149, 126, 113,\n",
    "                                                              81, 139, 34, 147, 166, 107,\n",
    "                                                              155, 173, 40, 80, 161, 120, 162, \n",
    "                                                               134, 156, 115, 56, 30, 128, 137, 158,\n",
    "                                                              121, 109, 94, 145, 68, 86, 114, 17, 63,\n",
    "                                                              47, 164, 166, 5, 66, 39, 9, 106,\n",
    "                                                              10, 152,  36, 127, 12, 75, 46, 133, 136, 119, 48,\n",
    "                                                              49, 64, 72, 140) ))  %>%  \n",
    "head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_multitons, c(\"sr\"), size_pos = 0.1, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_multitons, c(\"sr\"), size_pos = 0.1, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_multitons, c(\"sr\"), \n",
    "                   size_pos = 0.1, size_neg = 0.2, stroke = 0.02, limit = 1) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = c(160)\n",
    "b = c(163, 103)\n",
    "c = c(172)\n",
    "d = c(168)\n",
    "e = c(170, 14, 8, 143)  # 8 could be broken down further\n",
    "f = c(141, 151)\n",
    "g = c(146)\n",
    "h = c(165)\n",
    "i = c(85, 41, 99, 110, 31, 60, 111)\n",
    "j = c(169)\n",
    "k = c(174)\n",
    "l = c(51)\n",
    "m = c(95, 73)\n",
    "n = c(1)\n",
    "o = c(2) # CG43896 is only at one tip so could be broken down further\n",
    "p = c(171)\n",
    "q = c(20, 98, 45, 96, 92, 130, 88) # 20 should be split with 13, 55\n",
    "r = c(13, 55)\n",
    "s = c(167, 104, 123,  25) # has considerable scatter\n",
    "t = c(129) # possibly two branches\n",
    "u = c(157)\n",
    "v = c(122, 58, 28, 89)\n",
    "w = c(148)\n",
    "x = c(159)\n",
    "y = c(142, 154) # has spur on 154 that could be removed\n",
    "z = c(124)\n",
    "aa = c(149, 126) # 121, 109 are spurs, tributaries\n",
    "ab = c(113) # could be extension of y at least in part, but looks like much is separate\n",
    "ac = c(81, 139, 34)\n",
    "ad = c(147)\n",
    "ae = c(166, 107)  # markers shar, but distant on map; and only a small cluster of 107, which could be broken 3 ways\n",
    "af = c(173)\n",
    "ag = c(155)\n",
    "ah = c(40, 134 ) # 40 has spur that could be broken off\n",
    "ai = c(80)\n",
    "aj = c(161, 120, 162)\n",
    "ak = c(156)\n",
    "al = c(115, 56) # coalescent for 124, 128, 137\n",
    "am = c(30)\n",
    "an = c(128, 137)  # see what the markers are here\n",
    "ao = c(158) # closely related to 170, 14\n",
    "ap = c(121, 109) # Tsp74F lights up main trajectory of these two, but both could be broken further\n",
    "aq = c(94)\n",
    "ar = c(145, 68, 86)\n",
    "as = c(114)  # CG13545 is only in a single part of this larger cluster, part of brain and so might be neuron\n",
    "at = c(17)  # 49 is base and with part of 107 feeds into 12\n",
    "au = c(63) # CG7300 is only in left spur of 63, so could be broken down further; other spur feeds 128\n",
    "av = c(47, 164)\n",
    "aw = c(166) # markers also light up small part of 107, but not the main cluster.  Look for better markers of bulk of 107\n",
    "ax = c(5, 66) # Ptx1 marker also lights up 159 which is a spur of 5\n",
    "ay = c(39, 9, 106)\n",
    "az = c(10, 152, 36, 127) # basal clusters\n",
    "ba = c(12) # Lim3 also in adjacent 2, 49 is base\n",
    "bb = c(46) # CG44422 not specific but \"CG44422\", \"disco-r\" pair very specific\n",
    "bc = c(75) # separate cluster of 75 so could be broken down further tey marker also shared with 156\n",
    "bd = c(133) # Desi in small cluster of 133, part of brain, so could neuronal; shared with 121 or 54\n",
    "be = c(136, 119, 48, 100) # basal cluster\n",
    "bf = c(49) # with part of 107 base of 17, 12\n",
    "bg = c(64) # Ccp84Ac covers it but shared with 113\n",
    "bh = c(72, 57) # may be base of 124, 128, both could be broken\n",
    "bi = c(140) # very poor cluster, may be with 123\n",
    "## below done only on umap picture\n",
    "bj = c(23, 116, 105, 153)\n",
    "bk = c(52, 26, 102, 27)\n",
    "bl = c(87, 70, 35, 37, 22, 59)\n",
    "bm = c(132, 125, 108)\n",
    "bn = c(44)\n",
    "bo = c(97, 118)\n",
    "bp = c(91, 144, 54, 15, 84)\n",
    "bq = c(6, 117, 38, 43, 4)\n",
    "br = c(138, 67, 65)\n",
    "bs = c(32, 82, 131)\n",
    "bt = c(11, 76, 3, 62, 50)\n",
    "bu = c(71, 19)\n",
    "bv = c(33, 61)\n",
    "bw = c(77, 53)\n",
    "bx = c(29, 135)\n",
    "by = c(21)\n",
    "bz = c(18) # scattered cells\n",
    "ca = c(59, 74, 101)\n",
    "cb = c(112, 79, 7)\n",
    "cc = c(16, 93, 42, 78, 24)\n",
    "cd = c(69, 83)\n",
    "ce = c(90)\n",
    "cf = c(150) # very scattered cells near 12"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(150)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'a' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'a' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(a, b, c, d, e, f, g, h, i , j, k, l, m, n, o, p, q, r, s, t, u, v, w, x, y, z,\n",
    "        aa,ab, ac, ad, ae, af, ag, ah, ai, aj, ak, al, am, an, ao, ap, aq, ar, as, at, au, av, aw,\n",
    "        ax, ay, az, ba, bb, bc, bd, be, bf, bh, bi, bj, bk, bl, bm, bn, bo, bp, bq, br, bs, bt, bu, bv, \n",
    "         bw, bx, by, bz, ca, cb, cc, cd, ce)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ncol(cds_sub): object 'cds_sub' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in ncol(cds_sub): object 'cds_sub' not found\nTraceback:\n",
      "1. ncol(cds_sub)"
     ]
    }
   ],
   "source": [
    "df_sub300$cell.type.3 = rep(NA, ncol(cds_sub))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 724,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 34</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>orig.ident</th><th scope=col>nCount_RNA</th><th scope=col>nFeature_RNA</th><th scope=col>rt_bc</th><th scope=col>cell</th><th scope=col>time</th><th scope=col>percent.mt</th><th scope=col>percent.ribo</th><th scope=col>barn_dro</th><th scope=col>barn_hu</th><th scope=col>barn_mus</th><th scope=col>RNA_snn_res.0.8</th><th scope=col>doublet_score</th><th scope=col>predicted_doublet</th><th scope=col>doublet_cluster</th><th scope=col>seurat_clusters</th><th scope=col>doublet_subcluster</th><th scope=col>RNA_snn_res.0.08</th><th scope=col>lasso_age</th><th scope=col>NNv1_age</th><th scope=col>NNv2_age</th><th scope=col>lasso_shift</th><th scope=col>NNv1_shift</th><th scope=col>NNv2_shift</th><th scope=col>annot</th><th scope=col>ident</th><th scope=col>inferred_time_window</th><th scope=col>cell.type_time.bin</th><th scope=col>Size_Factor</th><th scope=col>cluster</th><th scope=col>umap_1</th><th scope=col>umap_2</th><th scope=col>cell.type.2</th><th scope=col>cell.type.3</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</th><td>exp1</td><td>346</td><td>247</td><td>AAGTTGCCAT</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>hrs_06_10</td><td>1.4450867</td><td>4.3352601</td><td>371</td><td>0</td><td>0</td><td>16</td><td>0.05276064</td><td>Singlet</td><td>FALSE</td><td>21</td><td>FALSE</td><td>21</td><td> 8.402794</td><td> 7.885203</td><td> 8.119637</td><td> 0.0000000</td><td> 0.0000000</td><td> 0.000000</td><td>sensory nervous system primordium</td><td>21</td><td>6-8  </td><td>sensory nervous system primordium:6-8</td><td>0.8108612</td><td>89 </td><td>-2.0067012</td><td> 1.218446</td><td>v </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG</th><td>exp1</td><td>279</td><td>225</td><td>TTAATTATTG</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG</td><td>hrs_08_12</td><td>0.7168459</td><td>5.0179211</td><td>331</td><td>4</td><td>5</td><td>7 </td><td>0.07219402</td><td>Singlet</td><td>FALSE</td><td>11</td><td>FALSE</td><td>11</td><td>11.597093</td><td> 9.940381</td><td> 9.959537</td><td> 0.0000000</td><td> 0.0000000</td><td> 0.000000</td><td>visceral muscle primordium       </td><td>11</td><td>8-10 </td><td>visceral muscle primordium:8-10      </td><td>0.6571200</td><td>112</td><td>-4.6204820</td><td> 1.096802</td><td>cb</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA</th><td>exp1</td><td>586</td><td>425</td><td>CAGGTATGGA</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA</td><td>hrs_08_12</td><td>1.0238908</td><td>6.4846416</td><td>630</td><td>0</td><td>0</td><td>4 </td><td>0.09008910</td><td>Singlet</td><td>FALSE</td><td>11</td><td>FALSE</td><td>11</td><td>11.266422</td><td>10.062633</td><td>11.156040</td><td> 0.0000000</td><td> 0.0000000</td><td> 0.000000</td><td>brain                            </td><td>11</td><td>10-12</td><td>brain:10-12                          </td><td>1.4059888</td><td>22 </td><td>-3.6831634</td><td> 2.801265</td><td>bl</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG</th><td>exp1</td><td>292</td><td>202</td><td>AAGTAGTCAG</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG</td><td>hrs_10_14</td><td>0.3424658</td><td>0.6849315</td><td>320</td><td>3</td><td>0</td><td>23</td><td>0.06585398</td><td>Singlet</td><td>FALSE</td><td>12</td><td>FALSE</td><td>12</td><td>12.544636</td><td>12.211529</td><td>13.830982</td><td> 0.0000000</td><td> 0.0000000</td><td> 0.000000</td><td>epidermis                        </td><td>12</td><td>12-14</td><td>epidermis:12-14                      </td><td>0.6967951</td><td>165</td><td>-1.3580737</td><td> 0.358204</td><td>h </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA</th><td>exp1</td><td>335</td><td>243</td><td>TGAGGTAGAA</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA</td><td>hrs_14_18</td><td>4.4776119</td><td>5.6716418</td><td>386</td><td>0</td><td>0</td><td>8 </td><td>0.08478501</td><td>Singlet</td><td>FALSE</td><td>11</td><td>FALSE</td><td>11</td><td>12.542527</td><td>13.739003</td><td>12.519031</td><td>-1.4574728</td><td>-0.2609968</td><td>-1.480969</td><td>sensory nervous system           </td><td>11</td><td>12-14</td><td>sensory nervous system:12-14         </td><td>0.7811049</td><td>102</td><td>-3.8563070</td><td> 3.315827</td><td>bk</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA</th><td>exp1</td><td>354</td><td>272</td><td>TCGATATTGA</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA</td><td>hrs_14_18</td><td>0.5649718</td><td>1.6949153</td><td>380</td><td>0</td><td>1</td><td>8 </td><td>0.08257165</td><td>Singlet</td><td>FALSE</td><td>11</td><td>FALSE</td><td>11</td><td>13.458162</td><td>15.126904</td><td>14.558554</td><td>-0.5418379</td><td> 0.0000000</td><td> 0.000000</td><td>sensory nervous system           </td><td>11</td><td>14-16</td><td>sensory nervous system:14-16         </td><td>0.8306988</td><td>148</td><td>-0.6711318</td><td>-2.946210</td><td>w </td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 34\n",
       "\\begin{tabular}{r|llllllllllllllllllllllllllllllllll}\n",
       "  & orig.ident & nCount\\_RNA & nFeature\\_RNA & rt\\_bc & cell & time & percent.mt & percent.ribo & barn\\_dro & barn\\_hu & barn\\_mus & RNA\\_snn\\_res.0.8 & doublet\\_score & predicted\\_doublet & doublet\\_cluster & seurat\\_clusters & doublet\\_subcluster & RNA\\_snn\\_res.0.08 & lasso\\_age & NNv1\\_age & NNv2\\_age & lasso\\_shift & NNv1\\_shift & NNv2\\_shift & annot & ident & inferred\\_time\\_window & cell.type\\_time.bin & Size\\_Factor & cluster & umap\\_1 & umap\\_2 & cell.type.2 & cell.type.3\\\\\n",
       "  & <chr> & <dbl> & <int> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <int> & <int> & <int> & <chr> & <dbl> & <chr> & <lgl> & <fct> & <lgl> & <fct> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <fct> & <chr> & <chr> & <dbl> & <fct> & <dbl> & <dbl> & <chr> & <lgl>\\\\\n",
       "\\hline\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & exp1 & 346 & 247 & AAGTTGCCAT & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & hrs\\_06\\_10 & 1.4450867 & 4.3352601 & 371 & 0 & 0 & 16 & 0.05276064 & Singlet & FALSE & 21 & FALSE & 21 &  8.402794 &  7.885203 &  8.119637 &  0.0000000 &  0.0000000 &  0.000000 & sensory nervous system primordium & 21 & 6-8   & sensory nervous system primordium:6-8 & 0.8108612 & 89  & -2.0067012 &  1.218446 & v  & NA\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG & exp1 & 279 & 225 & TTAATTATTG & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG & hrs\\_08\\_12 & 0.7168459 & 5.0179211 & 331 & 4 & 5 & 7  & 0.07219402 & Singlet & FALSE & 11 & FALSE & 11 & 11.597093 &  9.940381 &  9.959537 &  0.0000000 &  0.0000000 &  0.000000 & visceral muscle primordium        & 11 & 8-10  & visceral muscle primordium:8-10       & 0.6571200 & 112 & -4.6204820 &  1.096802 & cb & NA\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA & exp1 & 586 & 425 & CAGGTATGGA & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA & hrs\\_08\\_12 & 1.0238908 & 6.4846416 & 630 & 0 & 0 & 4  & 0.09008910 & Singlet & FALSE & 11 & FALSE & 11 & 11.266422 & 10.062633 & 11.156040 &  0.0000000 &  0.0000000 &  0.000000 & brain                             & 11 & 10-12 & brain:10-12                           & 1.4059888 & 22  & -3.6831634 &  2.801265 & bl & NA\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG & exp1 & 292 & 202 & AAGTAGTCAG & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG & hrs\\_10\\_14 & 0.3424658 & 0.6849315 & 320 & 3 & 0 & 23 & 0.06585398 & Singlet & FALSE & 12 & FALSE & 12 & 12.544636 & 12.211529 & 13.830982 &  0.0000000 &  0.0000000 &  0.000000 & epidermis                         & 12 & 12-14 & epidermis:12-14                       & 0.6967951 & 165 & -1.3580737 &  0.358204 & h  & NA\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA & exp1 & 335 & 243 & TGAGGTAGAA & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA & hrs\\_14\\_18 & 4.4776119 & 5.6716418 & 386 & 0 & 0 & 8  & 0.08478501 & Singlet & FALSE & 11 & FALSE & 11 & 12.542527 & 13.739003 & 12.519031 & -1.4574728 & -0.2609968 & -1.480969 & sensory nervous system            & 11 & 12-14 & sensory nervous system:12-14          & 0.7811049 & 102 & -3.8563070 &  3.315827 & bk & NA\\\\\n",
       "\texp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA & exp1 & 354 & 272 & TCGATATTGA & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA & hrs\\_14\\_18 & 0.5649718 & 1.6949153 & 380 & 0 & 1 & 8  & 0.08257165 & Singlet & FALSE & 11 & FALSE & 11 & 13.458162 & 15.126904 & 14.558554 & -0.5418379 &  0.0000000 &  0.000000 & sensory nervous system            & 11 & 14-16 & sensory nervous system:14-16          & 0.8306988 & 148 & -0.6711318 & -2.946210 & w  & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 34\n",
       "\n",
       "| <!--/--> | orig.ident &lt;chr&gt; | nCount_RNA &lt;dbl&gt; | nFeature_RNA &lt;int&gt; | rt_bc &lt;chr&gt; | cell &lt;chr&gt; | time &lt;chr&gt; | percent.mt &lt;dbl&gt; | percent.ribo &lt;dbl&gt; | barn_dro &lt;int&gt; | barn_hu &lt;int&gt; | barn_mus &lt;int&gt; | RNA_snn_res.0.8 &lt;chr&gt; | doublet_score &lt;dbl&gt; | predicted_doublet &lt;chr&gt; | doublet_cluster &lt;lgl&gt; | seurat_clusters &lt;fct&gt; | doublet_subcluster &lt;lgl&gt; | RNA_snn_res.0.08 &lt;fct&gt; | lasso_age &lt;dbl&gt; | NNv1_age &lt;dbl&gt; | NNv2_age &lt;dbl&gt; | lasso_shift &lt;dbl&gt; | NNv1_shift &lt;dbl&gt; | NNv2_shift &lt;dbl&gt; | annot &lt;chr&gt; | ident &lt;fct&gt; | inferred_time_window &lt;chr&gt; | cell.type_time.bin &lt;chr&gt; | Size_Factor &lt;dbl&gt; | cluster &lt;fct&gt; | umap_1 &lt;dbl&gt; | umap_2 &lt;dbl&gt; | cell.type.2 &lt;chr&gt; | cell.type.3 &lt;lgl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | exp1 | 346 | 247 | AAGTTGCCAT | exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | hrs_06_10 | 1.4450867 | 4.3352601 | 371 | 0 | 0 | 16 | 0.05276064 | Singlet | FALSE | 21 | FALSE | 21 |  8.402794 |  7.885203 |  8.119637 |  0.0000000 |  0.0000000 |  0.000000 | sensory nervous system primordium | 21 | 6-8   | sensory nervous system primordium:6-8 | 0.8108612 | 89  | -2.0067012 |  1.218446 | v  | NA |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG | exp1 | 279 | 225 | TTAATTATTG | exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG | hrs_08_12 | 0.7168459 | 5.0179211 | 331 | 4 | 5 | 7  | 0.07219402 | Singlet | FALSE | 11 | FALSE | 11 | 11.597093 |  9.940381 |  9.959537 |  0.0000000 |  0.0000000 |  0.000000 | visceral muscle primordium        | 11 | 8-10  | visceral muscle primordium:8-10       | 0.6571200 | 112 | -4.6204820 |  1.096802 | cb | NA |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA | exp1 | 586 | 425 | CAGGTATGGA | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA | hrs_08_12 | 1.0238908 | 6.4846416 | 630 | 0 | 0 | 4  | 0.09008910 | Singlet | FALSE | 11 | FALSE | 11 | 11.266422 | 10.062633 | 11.156040 |  0.0000000 |  0.0000000 |  0.000000 | brain                             | 11 | 10-12 | brain:10-12                           | 1.4059888 | 22  | -3.6831634 |  2.801265 | bl | NA |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG | exp1 | 292 | 202 | AAGTAGTCAG | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG | hrs_10_14 | 0.3424658 | 0.6849315 | 320 | 3 | 0 | 23 | 0.06585398 | Singlet | FALSE | 12 | FALSE | 12 | 12.544636 | 12.211529 | 13.830982 |  0.0000000 |  0.0000000 |  0.000000 | epidermis                         | 12 | 12-14 | epidermis:12-14                       | 0.6967951 | 165 | -1.3580737 |  0.358204 | h  | NA |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA | exp1 | 335 | 243 | TGAGGTAGAA | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA | hrs_14_18 | 4.4776119 | 5.6716418 | 386 | 0 | 0 | 8  | 0.08478501 | Singlet | FALSE | 11 | FALSE | 11 | 12.542527 | 13.739003 | 12.519031 | -1.4574728 | -0.2609968 | -1.480969 | sensory nervous system            | 11 | 12-14 | sensory nervous system:12-14          | 0.7811049 | 102 | -3.8563070 |  3.315827 | bk | NA |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA | exp1 | 354 | 272 | TCGATATTGA | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA | hrs_14_18 | 0.5649718 | 1.6949153 | 380 | 0 | 1 | 8  | 0.08257165 | Singlet | FALSE | 11 | FALSE | 11 | 13.458162 | 15.126904 | 14.558554 | -0.5418379 |  0.0000000 |  0.000000 | sensory nervous system            | 11 | 14-16 | sensory nervous system:14-16          | 0.8306988 | 148 | -0.6711318 | -2.946210 | w  | NA |\n",
       "\n"
      ],
      "text/plain": [
       "                                                       orig.ident nCount_RNA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT exp1       346       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG exp1       279       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA exp1       586       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG exp1       292       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA exp1       335       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA exp1       354       \n",
       "                                                       nFeature_RNA rt_bc     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 247          AAGTTGCCAT\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 225          TTAATTATTG\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 425          CAGGTATGGA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 202          AAGTAGTCAG\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 243          TGAGGTAGAA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 272          TCGATATTGA\n",
       "                                                       cell                                                  \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA\n",
       "                                                       time      percent.mt\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT hrs_06_10 1.4450867 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG hrs_08_12 0.7168459 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA hrs_08_12 1.0238908 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG hrs_10_14 0.3424658 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA hrs_14_18 4.4776119 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA hrs_14_18 0.5649718 \n",
       "                                                       percent.ribo barn_dro\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 4.3352601    371     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 5.0179211    331     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 6.4846416    630     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 0.6849315    320     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 5.6716418    386     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 1.6949153    380     \n",
       "                                                       barn_hu barn_mus\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 0       0       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 4       5       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 0       0       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 3       0       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 0       0       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 0       1       \n",
       "                                                       RNA_snn_res.0.8\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 16             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 7              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 4              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 23             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 8              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 8              \n",
       "                                                       doublet_score\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 0.05276064   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 0.07219402   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 0.09008910   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 0.06585398   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 0.08478501   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 0.08257165   \n",
       "                                                       predicted_doublet\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT Singlet          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG Singlet          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA Singlet          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG Singlet          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA Singlet          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA Singlet          \n",
       "                                                       doublet_cluster\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT FALSE          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG FALSE          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA FALSE          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG FALSE          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA FALSE          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA FALSE          \n",
       "                                                       seurat_clusters\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 21             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 11             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 11             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 11             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 11             \n",
       "                                                       doublet_subcluster\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT FALSE             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG FALSE             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA FALSE             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG FALSE             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA FALSE             \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA FALSE             \n",
       "                                                       RNA_snn_res.0.08\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 21              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 11              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 11              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 11              \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 11              \n",
       "                                                       lasso_age NNv1_age \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT  8.402794  7.885203\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 11.597093  9.940381\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 11.266422 10.062633\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12.544636 12.211529\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 12.542527 13.739003\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 13.458162 15.126904\n",
       "                                                       NNv2_age  lasso_shift\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT  8.119637  0.0000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG  9.959537  0.0000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 11.156040  0.0000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 13.830982  0.0000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 12.519031 -1.4574728 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 14.558554 -0.5418379 \n",
       "                                                       NNv1_shift NNv2_shift\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT  0.0000000  0.000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG  0.0000000  0.000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA  0.0000000  0.000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG  0.0000000  0.000000 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA -0.2609968 -1.480969 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA  0.0000000  0.000000 \n",
       "                                                       annot                            \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT sensory nervous system primordium\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG visceral muscle primordium       \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA brain                            \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG epidermis                        \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA sensory nervous system           \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA sensory nervous system           \n",
       "                                                       ident\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 21   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 11   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 11   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 11   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 11   \n",
       "                                                       inferred_time_window\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 6-8                 \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 8-10                \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 10-12               \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 12-14               \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 12-14               \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 14-16               \n",
       "                                                       cell.type_time.bin                   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT sensory nervous system primordium:6-8\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG visceral muscle primordium:8-10      \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA brain:10-12                          \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG epidermis:12-14                      \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA sensory nervous system:12-14         \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA sensory nervous system:14-16         \n",
       "                                                       Size_Factor cluster\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT 0.8108612   89     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG 0.6571200   112    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 1.4059888   22     \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG 0.6967951   165    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA 0.7811049   102    \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 0.8306988   148    \n",
       "                                                       umap_1     umap_2   \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT -2.0067012  1.218446\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG -4.6204820  1.096802\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA -3.6831634  2.801265\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG -1.3580737  0.358204\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA -3.8563070  3.315827\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA -0.6711318 -2.946210\n",
       "                                                       cell.type.2 cell.type.3\n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT v           NA         \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.ACGAGGTTTTTAATTATTG cb          NA         \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA bl          NA         \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG h           NA         \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA bk          NA         \n",
       "exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA w           NA         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub300  %>%  head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Start the reannotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 725,
   "metadata": {},
   "outputs": [],
   "source": [
    "set.cell.type = function(data_df, cell.type, x) {\n",
    "       message(\"Assigning cells for cell type \", cell.type)\n",
    "       data_df$cell.type.3 = ifelse(x, cell.type, data_df$cell.type.3)\n",
    "       return(data_df)\n",
    "    }\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 726,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sub300_copy = df_sub300"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1786,
   "metadata": {},
   "outputs": [],
   "source": [
    "# not used, from prior\n",
    "df_sub300 = df_sub300_copy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 787,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type proventriculus\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"proventriculus\", df_sub300$cluster %in% a)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1781,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sub300$cell.type.3 <- \n",
    "gsub(\"proventriculus\", NA, df_sub300$cell.type.3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1785,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type unknown_4\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"unknown_4\", df_sub300$cluster %in% a &\n",
    "                          df_sub300$umap_2 < 1.8*df_sub300$umap_1 -4.9\n",
    "                         )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1788,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Proventriculus_older\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Proventriculus_older\", df_sub300$cluster %in% a &\n",
    "                         df_sub300$umap_2 < 1.8*df_sub300$umap_1 -4.4 &\n",
    "              df_sub300$umap_2 >= 1.8*df_sub300$umap_1 -4.9)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1789,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type unknown_5\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"unknown_5\", df_sub300$cluster %in% a &\n",
    "                         df_sub300$umap_2 >= 1.8*df_sub300$umap_1 -4.4 &\n",
    "              df_sub300$umap_2 < 1.8*df_sub300$umap_1 -3.95)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1790,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type unknown_6\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"unknown_6\", df_sub300$cluster %in% a &\n",
    "                         df_sub300$umap_2 >= 1.8*df_sub300$umap_1 -3.95)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'a' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'a' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(a)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clus = c(a)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in cell.type.2.markers_no_doubletons %>% filter(cell.type.2 == \"a\" & : could not find function \"%>%\"\n",
     "output_type": "error",
     "traceback": [
      "Error in cell.type.2.markers_no_doubletons %>% filter(cell.type.2 == \"a\" & : could not find function \"%>%\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"a\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Muc26B\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Muc26B\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Muc26B\"), \n",
    "                   size_pos = 4, size_neg = 2, stroke = 0.8, limit = 1) +\n",
    "xlim(2, 6) + ylim(0,4)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG7017\"), \n",
    "                   size_pos = 4, size_neg = 2, stroke = 0.8, limit = 1) +\n",
    "xlim(2, 6) + ylim(0,4)\n",
    "# flybase says expressed in proventriculus\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Muc96D\"), \n",
    "                   size_pos = 4, size_neg = 2, stroke = 0.9, limit = 1) +\n",
    "xlim(2, 6) + ylim(0,4)\n",
    "# flybase says expressed in proventriculus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1723,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 528\n",
      "\n",
      "# of cells sampled in set 2: 20000\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Muc26B</th><td>Muc26B </td><td>3993</td><td> 41</td><td>10368.322</td><td>3.0813262</td><td>Set 1</td><td>11.310857</td><td>150.88721</td></tr>\n",
       "\t<tr><th scope=row>CG7017</th><td>CG7017 </td><td>3759</td><td>102</td><td>11685.814</td><td>9.1726831</td><td>Set 1</td><td>10.165842</td><td>137.36690</td></tr>\n",
       "\t<tr><th scope=row>Muc96D</th><td>Muc96D </td><td>2590</td><td> 53</td><td> 6427.809</td><td>4.5846032</td><td>Set 1</td><td>10.168657</td><td>128.63692</td></tr>\n",
       "\t<tr><th scope=row>CG17826</th><td>CG17826</td><td>1107</td><td>  9</td><td> 2902.487</td><td>0.8681700</td><td>Set 1</td><td>10.601448</td><td>121.95451</td></tr>\n",
       "\t<tr><th scope=row>CG11300</th><td>CG11300</td><td> 523</td><td>  5</td><td> 1123.508</td><td>0.4703003</td><td>Set 1</td><td> 9.577683</td><td> 97.07057</td></tr>\n",
       "\t<tr><th scope=row>EAChm</th><td>EAChm  </td><td> 327</td><td> 53</td><td> 1601.716</td><td>4.7526624</td><td>Set 1</td><td> 8.121173</td><td> 86.46047</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tMuc26B & Muc26B  & 3993 &  41 & 10368.322 & 3.0813262 & Set 1 & 11.310857 & 150.88721\\\\\n",
       "\tCG7017 & CG7017  & 3759 & 102 & 11685.814 & 9.1726831 & Set 1 & 10.165842 & 137.36690\\\\\n",
       "\tMuc96D & Muc96D  & 2590 &  53 &  6427.809 & 4.5846032 & Set 1 & 10.168657 & 128.63692\\\\\n",
       "\tCG17826 & CG17826 & 1107 &   9 &  2902.487 & 0.8681700 & Set 1 & 10.601448 & 121.95451\\\\\n",
       "\tCG11300 & CG11300 &  523 &   5 &  1123.508 & 0.4703003 & Set 1 &  9.577683 &  97.07057\\\\\n",
       "\tEAChm & EAChm   &  327 &  53 &  1601.716 & 4.7526624 & Set 1 &  8.121173 &  86.46047\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Muc26B | Muc26B  | 3993 |  41 | 10368.322 | 3.0813262 | Set 1 | 11.310857 | 150.88721 |\n",
       "| CG7017 | CG7017  | 3759 | 102 | 11685.814 | 9.1726831 | Set 1 | 10.165842 | 137.36690 |\n",
       "| Muc96D | Muc96D  | 2590 |  53 |  6427.809 | 4.5846032 | Set 1 | 10.168657 | 128.63692 |\n",
       "| CG17826 | CG17826 | 1107 |   9 |  2902.487 | 0.8681700 | Set 1 | 10.601448 | 121.95451 |\n",
       "| CG11300 | CG11300 |  523 |   5 |  1123.508 | 0.4703003 | Set 1 |  9.577683 |  97.07057 |\n",
       "| EAChm | EAChm   |  327 |  53 |  1601.716 | 4.7526624 | Set 1 |  8.121173 |  86.46047 |\n",
       "\n"
      ],
      "text/plain": [
       "        gene    set.1.n.umi set.2.n.umi set.1.tpm set.2.tpm higher.expr\n",
       "Muc26B  Muc26B  3993         41         10368.322 3.0813262 Set 1      \n",
       "CG7017  CG7017  3759        102         11685.814 9.1726831 Set 1      \n",
       "Muc96D  Muc96D  2590         53          6427.809 4.5846032 Set 1      \n",
       "CG17826 CG17826 1107          9          2902.487 0.8681700 Set 1      \n",
       "CG11300 CG11300  523          5          1123.508 0.4703003 Set 1      \n",
       "EAChm   EAChm    327         53          1601.716 4.7526624 Set 1      \n",
       "        log2.ratio heuristic.score\n",
       "Muc26B  11.310857  150.88721      \n",
       "CG7017  10.165842  137.36690      \n",
       "Muc96D  10.168657  128.63692      \n",
       "CG17826 10.601448  121.95451      \n",
       "CG11300  9.577683   97.07057      \n",
       "EAChm    8.121173   86.46047      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "two.set.differential.gene.test.3a(cds_sub, \n",
    "                                  df_sub300$cluster == a, \n",
    "                                 !(df_sub300$cluster == a),\n",
    "                                 sample = 20000)  %>% head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'a' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'a' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(a)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 < 1.8*umap_1 -4.9), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 4, 0.2), stroke = 0.5) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 1.8, intercept = -4.4) +\n",
    "geom_abline(slope = 1.8, intercept = -4.9) +\n",
    "geom_abline(slope = 1.8, intercept = -5.8) +\n",
    "xlim(2, 6) + ylim(0,4)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1752,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 156\n",
      "\n",
      "# of cells sampled in set 2: 20000\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>EAChm</th><td>EAChm  </td><td> 326</td><td>  57</td><td> 5419.5933</td><td>  4.63990761</td><td>Set 1</td><td> 9.908297</td><td>122.90485</td></tr>\n",
       "\t<tr><th scope=row>CG32453</th><td>CG32453</td><td> 107</td><td>   7</td><td> 1866.6010</td><td>  0.49633844</td><td>Set 1</td><td>10.284761</td><td>111.76420</td></tr>\n",
       "\t<tr><th scope=row>CG18649</th><td>CG18649</td><td>  97</td><td>  11</td><td> 1688.4494</td><td>  0.86471948</td><td>Set 1</td><td> 9.822525</td><td>105.32042</td></tr>\n",
       "\t<tr><th scope=row>CG5402</th><td>CG5402 </td><td>  60</td><td>   3</td><td> 1067.1943</td><td>  0.32416603</td><td>Set 1</td><td> 9.654523</td><td> 97.13376</td></tr>\n",
       "\t<tr><th scope=row>Eip74EF</th><td>Eip74EF</td><td> 488</td><td> 547</td><td> 8605.4569</td><td> 55.99706117</td><td>Set 1</td><td> 7.238220</td><td> 94.61225</td></tr>\n",
       "\t<tr><th scope=row>CG34290</th><td>CG34290</td><td>  52</td><td>   3</td><td>  817.0746</td><td>  0.23087190</td><td>Set 1</td><td> 9.374643</td><td> 90.70988</td></tr>\n",
       "\t<tr><th scope=row>CG14453</th><td>CG14453</td><td>  57</td><td>   9</td><td>  960.0928</td><td>  0.87437564</td><td>Set 1</td><td> 9.000620</td><td> 89.18293</td></tr>\n",
       "\t<tr><th scope=row>CG14454</th><td>CG14454</td><td>  47</td><td>   4</td><td>  772.9593</td><td>  0.48061288</td><td>Set 1</td><td> 9.028054</td><td> 86.63424</td></tr>\n",
       "\t<tr><th scope=row>CG13738</th><td>CG13738</td><td>  32</td><td>   2</td><td>  516.4712</td><td>  0.09613526</td><td>Set 1</td><td> 8.880118</td><td> 80.05724</td></tr>\n",
       "\t<tr><th scope=row>CG30411</th><td>CG30411</td><td>  60</td><td>  57</td><td> 1088.1170</td><td>  4.81219807</td><td>Set 1</td><td> 7.548534</td><td> 76.15673</td></tr>\n",
       "\t<tr><th scope=row>CG43139</th><td>CG43139</td><td>  30</td><td>  14</td><td>  589.4781</td><td>  1.70373658</td><td>Set 1</td><td> 7.768340</td><td> 71.51331</td></tr>\n",
       "\t<tr><th scope=row>CG10175</th><td>CG10175</td><td> 112</td><td> 208</td><td> 2169.7916</td><td> 24.04060412</td><td>Set 1</td><td> 6.437143</td><td> 71.34933</td></tr>\n",
       "\t<tr><th scope=row>CG9812</th><td>CG9812 </td><td> 109</td><td> 241</td><td> 2081.2639</td><td> 22.94896611</td><td>Set 1</td><td> 6.441353</td><td> 71.00907</td></tr>\n",
       "\t<tr><th scope=row>nur</th><td>nur    </td><td>1085</td><td>6111</td><td>19856.4982</td><td>689.07176797</td><td>Set 1</td><td> 4.846721</td><td> 69.19856</td></tr>\n",
       "\t<tr><th scope=row>CG33160</th><td>CG33160</td><td>  16</td><td>   2</td><td>  326.3448</td><td>  0.10993181</td><td>Set 1</td><td> 8.199782</td><td> 68.50645</td></tr>\n",
       "\t<tr><th scope=row>CG30371</th><td>CG30371</td><td>  36</td><td>  45</td><td>  588.6692</td><td>  2.71898524</td><td>Set 1</td><td> 7.306404</td><td> 67.24641</td></tr>\n",
       "\t<tr><th scope=row>CG14636</th><td>CG14636</td><td> 110</td><td> 304</td><td> 1979.1657</td><td> 29.00472013</td><td>Set 1</td><td> 6.043559</td><td> 66.18547</td></tr>\n",
       "\t<tr><th scope=row>pip</th><td>pip    </td><td> 962</td><td>7211</td><td>18120.7886</td><td>879.03924164</td><td>Set 1</td><td> 4.363934</td><td> 61.72976</td></tr>\n",
       "\t<tr><th scope=row>CG43181</th><td>CG43181</td><td>  17</td><td>   7</td><td>  279.9461</td><td>  0.58988272</td><td>Set 1</td><td> 7.460085</td><td> 60.68145</td></tr>\n",
       "\t<tr><th scope=row>CG1986</th><td>CG1986 </td><td>  73</td><td> 233</td><td> 1315.0394</td><td> 24.88324796</td><td>Set 1</td><td> 5.666944</td><td> 58.72080</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tEAChm & EAChm   &  326 &   57 &  5419.5933 &   4.63990761 & Set 1 &  9.908297 & 122.90485\\\\\n",
       "\tCG32453 & CG32453 &  107 &    7 &  1866.6010 &   0.49633844 & Set 1 & 10.284761 & 111.76420\\\\\n",
       "\tCG18649 & CG18649 &   97 &   11 &  1688.4494 &   0.86471948 & Set 1 &  9.822525 & 105.32042\\\\\n",
       "\tCG5402 & CG5402  &   60 &    3 &  1067.1943 &   0.32416603 & Set 1 &  9.654523 &  97.13376\\\\\n",
       "\tEip74EF & Eip74EF &  488 &  547 &  8605.4569 &  55.99706117 & Set 1 &  7.238220 &  94.61225\\\\\n",
       "\tCG34290 & CG34290 &   52 &    3 &   817.0746 &   0.23087190 & Set 1 &  9.374643 &  90.70988\\\\\n",
       "\tCG14453 & CG14453 &   57 &    9 &   960.0928 &   0.87437564 & Set 1 &  9.000620 &  89.18293\\\\\n",
       "\tCG14454 & CG14454 &   47 &    4 &   772.9593 &   0.48061288 & Set 1 &  9.028054 &  86.63424\\\\\n",
       "\tCG13738 & CG13738 &   32 &    2 &   516.4712 &   0.09613526 & Set 1 &  8.880118 &  80.05724\\\\\n",
       "\tCG30411 & CG30411 &   60 &   57 &  1088.1170 &   4.81219807 & Set 1 &  7.548534 &  76.15673\\\\\n",
       "\tCG43139 & CG43139 &   30 &   14 &   589.4781 &   1.70373658 & Set 1 &  7.768340 &  71.51331\\\\\n",
       "\tCG10175 & CG10175 &  112 &  208 &  2169.7916 &  24.04060412 & Set 1 &  6.437143 &  71.34933\\\\\n",
       "\tCG9812 & CG9812  &  109 &  241 &  2081.2639 &  22.94896611 & Set 1 &  6.441353 &  71.00907\\\\\n",
       "\tnur & nur     & 1085 & 6111 & 19856.4982 & 689.07176797 & Set 1 &  4.846721 &  69.19856\\\\\n",
       "\tCG33160 & CG33160 &   16 &    2 &   326.3448 &   0.10993181 & Set 1 &  8.199782 &  68.50645\\\\\n",
       "\tCG30371 & CG30371 &   36 &   45 &   588.6692 &   2.71898524 & Set 1 &  7.306404 &  67.24641\\\\\n",
       "\tCG14636 & CG14636 &  110 &  304 &  1979.1657 &  29.00472013 & Set 1 &  6.043559 &  66.18547\\\\\n",
       "\tpip & pip     &  962 & 7211 & 18120.7886 & 879.03924164 & Set 1 &  4.363934 &  61.72976\\\\\n",
       "\tCG43181 & CG43181 &   17 &    7 &   279.9461 &   0.58988272 & Set 1 &  7.460085 &  60.68145\\\\\n",
       "\tCG1986 & CG1986  &   73 &  233 &  1315.0394 &  24.88324796 & Set 1 &  5.666944 &  58.72080\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| EAChm | EAChm   |  326 |   57 |  5419.5933 |   4.63990761 | Set 1 |  9.908297 | 122.90485 |\n",
       "| CG32453 | CG32453 |  107 |    7 |  1866.6010 |   0.49633844 | Set 1 | 10.284761 | 111.76420 |\n",
       "| CG18649 | CG18649 |   97 |   11 |  1688.4494 |   0.86471948 | Set 1 |  9.822525 | 105.32042 |\n",
       "| CG5402 | CG5402  |   60 |    3 |  1067.1943 |   0.32416603 | Set 1 |  9.654523 |  97.13376 |\n",
       "| Eip74EF | Eip74EF |  488 |  547 |  8605.4569 |  55.99706117 | Set 1 |  7.238220 |  94.61225 |\n",
       "| CG34290 | CG34290 |   52 |    3 |   817.0746 |   0.23087190 | Set 1 |  9.374643 |  90.70988 |\n",
       "| CG14453 | CG14453 |   57 |    9 |   960.0928 |   0.87437564 | Set 1 |  9.000620 |  89.18293 |\n",
       "| CG14454 | CG14454 |   47 |    4 |   772.9593 |   0.48061288 | Set 1 |  9.028054 |  86.63424 |\n",
       "| CG13738 | CG13738 |   32 |    2 |   516.4712 |   0.09613526 | Set 1 |  8.880118 |  80.05724 |\n",
       "| CG30411 | CG30411 |   60 |   57 |  1088.1170 |   4.81219807 | Set 1 |  7.548534 |  76.15673 |\n",
       "| CG43139 | CG43139 |   30 |   14 |   589.4781 |   1.70373658 | Set 1 |  7.768340 |  71.51331 |\n",
       "| CG10175 | CG10175 |  112 |  208 |  2169.7916 |  24.04060412 | Set 1 |  6.437143 |  71.34933 |\n",
       "| CG9812 | CG9812  |  109 |  241 |  2081.2639 |  22.94896611 | Set 1 |  6.441353 |  71.00907 |\n",
       "| nur | nur     | 1085 | 6111 | 19856.4982 | 689.07176797 | Set 1 |  4.846721 |  69.19856 |\n",
       "| CG33160 | CG33160 |   16 |    2 |   326.3448 |   0.10993181 | Set 1 |  8.199782 |  68.50645 |\n",
       "| CG30371 | CG30371 |   36 |   45 |   588.6692 |   2.71898524 | Set 1 |  7.306404 |  67.24641 |\n",
       "| CG14636 | CG14636 |  110 |  304 |  1979.1657 |  29.00472013 | Set 1 |  6.043559 |  66.18547 |\n",
       "| pip | pip     |  962 | 7211 | 18120.7886 | 879.03924164 | Set 1 |  4.363934 |  61.72976 |\n",
       "| CG43181 | CG43181 |   17 |    7 |   279.9461 |   0.58988272 | Set 1 |  7.460085 |  60.68145 |\n",
       "| CG1986 | CG1986  |   73 |  233 |  1315.0394 |  24.88324796 | Set 1 |  5.666944 |  58.72080 |\n",
       "\n"
      ],
      "text/plain": [
       "        gene    set.1.n.umi set.2.n.umi set.1.tpm  set.2.tpm    higher.expr\n",
       "EAChm   EAChm    326          57         5419.5933   4.63990761 Set 1      \n",
       "CG32453 CG32453  107           7         1866.6010   0.49633844 Set 1      \n",
       "CG18649 CG18649   97          11         1688.4494   0.86471948 Set 1      \n",
       "CG5402  CG5402    60           3         1067.1943   0.32416603 Set 1      \n",
       "Eip74EF Eip74EF  488         547         8605.4569  55.99706117 Set 1      \n",
       "CG34290 CG34290   52           3          817.0746   0.23087190 Set 1      \n",
       "CG14453 CG14453   57           9          960.0928   0.87437564 Set 1      \n",
       "CG14454 CG14454   47           4          772.9593   0.48061288 Set 1      \n",
       "CG13738 CG13738   32           2          516.4712   0.09613526 Set 1      \n",
       "CG30411 CG30411   60          57         1088.1170   4.81219807 Set 1      \n",
       "CG43139 CG43139   30          14          589.4781   1.70373658 Set 1      \n",
       "CG10175 CG10175  112         208         2169.7916  24.04060412 Set 1      \n",
       "CG9812  CG9812   109         241         2081.2639  22.94896611 Set 1      \n",
       "nur     nur     1085        6111        19856.4982 689.07176797 Set 1      \n",
       "CG33160 CG33160   16           2          326.3448   0.10993181 Set 1      \n",
       "CG30371 CG30371   36          45          588.6692   2.71898524 Set 1      \n",
       "CG14636 CG14636  110         304         1979.1657  29.00472013 Set 1      \n",
       "pip     pip      962        7211        18120.7886 879.03924164 Set 1      \n",
       "CG43181 CG43181   17           7          279.9461   0.58988272 Set 1      \n",
       "CG1986  CG1986    73         233         1315.0394  24.88324796 Set 1      \n",
       "        log2.ratio heuristic.score\n",
       "EAChm    9.908297  122.90485      \n",
       "CG32453 10.284761  111.76420      \n",
       "CG18649  9.822525  105.32042      \n",
       "CG5402   9.654523   97.13376      \n",
       "Eip74EF  7.238220   94.61225      \n",
       "CG34290  9.374643   90.70988      \n",
       "CG14453  9.000620   89.18293      \n",
       "CG14454  9.028054   86.63424      \n",
       "CG13738  8.880118   80.05724      \n",
       "CG30411  7.548534   76.15673      \n",
       "CG43139  7.768340   71.51331      \n",
       "CG10175  6.437143   71.34933      \n",
       "CG9812   6.441353   71.00907      \n",
       "nur      4.846721   69.19856      \n",
       "CG33160  8.199782   68.50645      \n",
       "CG30371  7.306404   67.24641      \n",
       "CG14636  6.043559   66.18547      \n",
       "pip      4.363934   61.72976      \n",
       "CG43181  7.460085   60.68145      \n",
       "CG1986   5.666944   58.72080      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "two.set.differential.gene.test.3a(cds_sub, \n",
    "                                  df_sub300$cluster == a &\n",
    "                                  df_sub300$umap_2 < 1.8*df_sub300$umap_1 -4.9, \n",
    "                                 !(df_sub300$cluster == a &\n",
    "                                  df_sub300$umap_2 < 1.8*df_sub300$umap_1 -4.9) ,\n",
    "                                 sample = 20000)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"EAChm\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"EAChm\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"EAChm\"), \n",
    "                   size_pos = 4, size_neg = 2, stroke = 0.8, limit = 1) +\n",
    "xlim(2, 6) + ylim(0,4)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG32453\"), \n",
    "                   size_pos = 4, size_neg = 2, stroke = 0.8, limit = 1) +\n",
    "xlim(2, 6) + ylim(0,4)\n",
    "\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Eip74EF\"), \n",
    "                   size_pos = 4, size_neg = 2, stroke = 0.9, limit = 1) +\n",
    "xlim(2, 6) + ylim(0,4)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'a' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'a' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(a, as)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "              df_sub300$umap_2 < 1.8*umap_1 -4.4 &\n",
    "              df_sub300$umap_2 >= 1.8*umap_1 -4.9), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 4, 0.2), stroke = 0.5) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 1.8, intercept = -4.4) +\n",
    "geom_abline(slope = 1.8, intercept = -4.9) +\n",
    "geom_abline(slope = 1.8, intercept = -5.8) +\n",
    "xlim(2, 6) + ylim(0,4)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1755,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 113\n",
      "\n",
      "# of cells sampled in set 2: 20000\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Muc26B</th><td>Muc26B         </td><td>3991</td><td>  45</td><td>48387.9156</td><td> 4.57977773</td><td>Set 1</td><td>13.082152</td><td>203.58953</td></tr>\n",
       "\t<tr><th scope=row>CG7017</th><td>CG7017         </td><td>3750</td><td>  86</td><td>54290.7803</td><td> 7.18114684</td><td>Set 1</td><td>12.696117</td><td>199.69018</td></tr>\n",
       "\t<tr><th scope=row>Muc96D</th><td>Muc96D         </td><td>2590</td><td>  54</td><td>30034.3652</td><td> 5.63011511</td><td>Set 1</td><td>12.145293</td><td>180.65363</td></tr>\n",
       "\t<tr><th scope=row>CG17826</th><td>CG17826        </td><td>1107</td><td>  11</td><td>13562.0611</td><td> 0.64822638</td><td>Set 1</td><td>13.006374</td><td>178.54364</td></tr>\n",
       "\t<tr><th scope=row>CG10725</th><td>CG10725        </td><td> 144</td><td>   3</td><td> 2748.7208</td><td> 0.25184319</td><td>Set 1</td><td>11.100491</td><td>126.82388</td></tr>\n",
       "\t<tr><th scope=row>CG12934</th><td>CG12934        </td><td> 209</td><td>  21</td><td> 4031.9803</td><td> 1.77571514</td><td>Set 1</td><td>10.504413</td><td>125.81798</td></tr>\n",
       "\t<tr><th scope=row>CG34282</th><td>CG34282        </td><td> 168</td><td>   7</td><td> 2052.5797</td><td> 0.11675219</td><td>Set 1</td><td>10.843913</td><td>119.32561</td></tr>\n",
       "\t<tr><th scope=row>CG11672</th><td>CG11672        </td><td> 184</td><td>  23</td><td> 3202.8374</td><td> 1.90727691</td><td>Set 1</td><td>10.105466</td><td>117.68407</td></tr>\n",
       "\t<tr><th scope=row>CG7715</th><td>CG7715         </td><td>1006</td><td>1737</td><td>13228.6826</td><td>60.84266392</td><td>Set 1</td><td> 7.740851</td><td>105.98379</td></tr>\n",
       "\t<tr><th scope=row>CG10154</th><td>CG10154        </td><td>  58</td><td>   0</td><td> 1171.6675</td><td> 0.00000000</td><td>Set 1</td><td>10.194348</td><td>103.93727</td></tr>\n",
       "\t<tr><th scope=row>obst-J</th><td>obst-J         </td><td>  64</td><td>  15</td><td> 1411.9971</td><td> 0.73871012</td><td>Set 1</td><td> 9.665504</td><td>101.14508</td></tr>\n",
       "\t<tr><th scope=row>CG7714</th><td>CG7714         </td><td> 200</td><td> 140</td><td> 4729.7403</td><td>15.23335362</td><td>Set 1</td><td> 8.186656</td><td> 99.94147</td></tr>\n",
       "\t<tr><th scope=row>CG6933</th><td>CG6933         </td><td> 201</td><td> 185</td><td> 4557.8000</td><td>17.79963193</td><td>Set 1</td><td> 7.921489</td><td> 96.28125</td></tr>\n",
       "\t<tr><th scope=row>Pgant4</th><td>Pgant4         </td><td> 148</td><td>  40</td><td> 2077.9205</td><td> 4.53325251</td><td>Set 1</td><td> 8.552797</td><td> 94.26567</td></tr>\n",
       "\t<tr><th scope=row>Peritrophin-15a</th><td>Peritrophin-15a</td><td>  55</td><td>   5</td><td>  979.2066</td><td> 0.46590879</td><td>Set 1</td><td> 9.383674</td><td> 93.24503</td></tr>\n",
       "\t<tr><th scope=row>Kaz-m1</th><td>Kaz-m1         </td><td> 255</td><td> 138</td><td> 3750.5225</td><td>16.12834951</td><td>Set 1</td><td> 7.774562</td><td> 92.30940</td></tr>\n",
       "\t<tr><th scope=row>CG9988</th><td>CG9988         </td><td>  66</td><td>   1</td><td>  754.1769</td><td> 0.05720824</td><td>Set 1</td><td> 9.478500</td><td> 90.62082</td></tr>\n",
       "\t<tr><th scope=row>CG5084</th><td>CG5084         </td><td>  40</td><td>  18</td><td>  938.2140</td><td> 1.68228169</td><td>Set 1</td><td> 8.450312</td><td> 83.44946</td></tr>\n",
       "\t<tr><th scope=row>obst-G</th><td>obst-G         </td><td>  28</td><td>   0</td><td>  537.5586</td><td> 0.00000000</td><td>Set 1</td><td> 9.070278</td><td> 82.29427</td></tr>\n",
       "\t<tr><th scope=row>obst-H</th><td>obst-H         </td><td>  82</td><td>  45</td><td> 1398.4546</td><td> 5.57612444</td><td>Set 1</td><td> 7.732380</td><td> 80.80839</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tMuc26B & Muc26B          & 3991 &   45 & 48387.9156 &  4.57977773 & Set 1 & 13.082152 & 203.58953\\\\\n",
       "\tCG7017 & CG7017          & 3750 &   86 & 54290.7803 &  7.18114684 & Set 1 & 12.696117 & 199.69018\\\\\n",
       "\tMuc96D & Muc96D          & 2590 &   54 & 30034.3652 &  5.63011511 & Set 1 & 12.145293 & 180.65363\\\\\n",
       "\tCG17826 & CG17826         & 1107 &   11 & 13562.0611 &  0.64822638 & Set 1 & 13.006374 & 178.54364\\\\\n",
       "\tCG10725 & CG10725         &  144 &    3 &  2748.7208 &  0.25184319 & Set 1 & 11.100491 & 126.82388\\\\\n",
       "\tCG12934 & CG12934         &  209 &   21 &  4031.9803 &  1.77571514 & Set 1 & 10.504413 & 125.81798\\\\\n",
       "\tCG34282 & CG34282         &  168 &    7 &  2052.5797 &  0.11675219 & Set 1 & 10.843913 & 119.32561\\\\\n",
       "\tCG11672 & CG11672         &  184 &   23 &  3202.8374 &  1.90727691 & Set 1 & 10.105466 & 117.68407\\\\\n",
       "\tCG7715 & CG7715          & 1006 & 1737 & 13228.6826 & 60.84266392 & Set 1 &  7.740851 & 105.98379\\\\\n",
       "\tCG10154 & CG10154         &   58 &    0 &  1171.6675 &  0.00000000 & Set 1 & 10.194348 & 103.93727\\\\\n",
       "\tobst-J & obst-J          &   64 &   15 &  1411.9971 &  0.73871012 & Set 1 &  9.665504 & 101.14508\\\\\n",
       "\tCG7714 & CG7714          &  200 &  140 &  4729.7403 & 15.23335362 & Set 1 &  8.186656 &  99.94147\\\\\n",
       "\tCG6933 & CG6933          &  201 &  185 &  4557.8000 & 17.79963193 & Set 1 &  7.921489 &  96.28125\\\\\n",
       "\tPgant4 & Pgant4          &  148 &   40 &  2077.9205 &  4.53325251 & Set 1 &  8.552797 &  94.26567\\\\\n",
       "\tPeritrophin-15a & Peritrophin-15a &   55 &    5 &   979.2066 &  0.46590879 & Set 1 &  9.383674 &  93.24503\\\\\n",
       "\tKaz-m1 & Kaz-m1          &  255 &  138 &  3750.5225 & 16.12834951 & Set 1 &  7.774562 &  92.30940\\\\\n",
       "\tCG9988 & CG9988          &   66 &    1 &   754.1769 &  0.05720824 & Set 1 &  9.478500 &  90.62082\\\\\n",
       "\tCG5084 & CG5084          &   40 &   18 &   938.2140 &  1.68228169 & Set 1 &  8.450312 &  83.44946\\\\\n",
       "\tobst-G & obst-G          &   28 &    0 &   537.5586 &  0.00000000 & Set 1 &  9.070278 &  82.29427\\\\\n",
       "\tobst-H & obst-H          &   82 &   45 &  1398.4546 &  5.57612444 & Set 1 &  7.732380 &  80.80839\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Muc26B | Muc26B          | 3991 |   45 | 48387.9156 |  4.57977773 | Set 1 | 13.082152 | 203.58953 |\n",
       "| CG7017 | CG7017          | 3750 |   86 | 54290.7803 |  7.18114684 | Set 1 | 12.696117 | 199.69018 |\n",
       "| Muc96D | Muc96D          | 2590 |   54 | 30034.3652 |  5.63011511 | Set 1 | 12.145293 | 180.65363 |\n",
       "| CG17826 | CG17826         | 1107 |   11 | 13562.0611 |  0.64822638 | Set 1 | 13.006374 | 178.54364 |\n",
       "| CG10725 | CG10725         |  144 |    3 |  2748.7208 |  0.25184319 | Set 1 | 11.100491 | 126.82388 |\n",
       "| CG12934 | CG12934         |  209 |   21 |  4031.9803 |  1.77571514 | Set 1 | 10.504413 | 125.81798 |\n",
       "| CG34282 | CG34282         |  168 |    7 |  2052.5797 |  0.11675219 | Set 1 | 10.843913 | 119.32561 |\n",
       "| CG11672 | CG11672         |  184 |   23 |  3202.8374 |  1.90727691 | Set 1 | 10.105466 | 117.68407 |\n",
       "| CG7715 | CG7715          | 1006 | 1737 | 13228.6826 | 60.84266392 | Set 1 |  7.740851 | 105.98379 |\n",
       "| CG10154 | CG10154         |   58 |    0 |  1171.6675 |  0.00000000 | Set 1 | 10.194348 | 103.93727 |\n",
       "| obst-J | obst-J          |   64 |   15 |  1411.9971 |  0.73871012 | Set 1 |  9.665504 | 101.14508 |\n",
       "| CG7714 | CG7714          |  200 |  140 |  4729.7403 | 15.23335362 | Set 1 |  8.186656 |  99.94147 |\n",
       "| CG6933 | CG6933          |  201 |  185 |  4557.8000 | 17.79963193 | Set 1 |  7.921489 |  96.28125 |\n",
       "| Pgant4 | Pgant4          |  148 |   40 |  2077.9205 |  4.53325251 | Set 1 |  8.552797 |  94.26567 |\n",
       "| Peritrophin-15a | Peritrophin-15a |   55 |    5 |   979.2066 |  0.46590879 | Set 1 |  9.383674 |  93.24503 |\n",
       "| Kaz-m1 | Kaz-m1          |  255 |  138 |  3750.5225 | 16.12834951 | Set 1 |  7.774562 |  92.30940 |\n",
       "| CG9988 | CG9988          |   66 |    1 |   754.1769 |  0.05720824 | Set 1 |  9.478500 |  90.62082 |\n",
       "| CG5084 | CG5084          |   40 |   18 |   938.2140 |  1.68228169 | Set 1 |  8.450312 |  83.44946 |\n",
       "| obst-G | obst-G          |   28 |    0 |   537.5586 |  0.00000000 | Set 1 |  9.070278 |  82.29427 |\n",
       "| obst-H | obst-H          |   82 |   45 |  1398.4546 |  5.57612444 | Set 1 |  7.732380 |  80.80839 |\n",
       "\n"
      ],
      "text/plain": [
       "                gene            set.1.n.umi set.2.n.umi set.1.tpm  set.2.tpm  \n",
       "Muc26B          Muc26B          3991          45        48387.9156  4.57977773\n",
       "CG7017          CG7017          3750          86        54290.7803  7.18114684\n",
       "Muc96D          Muc96D          2590          54        30034.3652  5.63011511\n",
       "CG17826         CG17826         1107          11        13562.0611  0.64822638\n",
       "CG10725         CG10725          144           3         2748.7208  0.25184319\n",
       "CG12934         CG12934          209          21         4031.9803  1.77571514\n",
       "CG34282         CG34282          168           7         2052.5797  0.11675219\n",
       "CG11672         CG11672          184          23         3202.8374  1.90727691\n",
       "CG7715          CG7715          1006        1737        13228.6826 60.84266392\n",
       "CG10154         CG10154           58           0         1171.6675  0.00000000\n",
       "obst-J          obst-J            64          15         1411.9971  0.73871012\n",
       "CG7714          CG7714           200         140         4729.7403 15.23335362\n",
       "CG6933          CG6933           201         185         4557.8000 17.79963193\n",
       "Pgant4          Pgant4           148          40         2077.9205  4.53325251\n",
       "Peritrophin-15a Peritrophin-15a   55           5          979.2066  0.46590879\n",
       "Kaz-m1          Kaz-m1           255         138         3750.5225 16.12834951\n",
       "CG9988          CG9988            66           1          754.1769  0.05720824\n",
       "CG5084          CG5084            40          18          938.2140  1.68228169\n",
       "obst-G          obst-G            28           0          537.5586  0.00000000\n",
       "obst-H          obst-H            82          45         1398.4546  5.57612444\n",
       "                higher.expr log2.ratio heuristic.score\n",
       "Muc26B          Set 1       13.082152  203.58953      \n",
       "CG7017          Set 1       12.696117  199.69018      \n",
       "Muc96D          Set 1       12.145293  180.65363      \n",
       "CG17826         Set 1       13.006374  178.54364      \n",
       "CG10725         Set 1       11.100491  126.82388      \n",
       "CG12934         Set 1       10.504413  125.81798      \n",
       "CG34282         Set 1       10.843913  119.32561      \n",
       "CG11672         Set 1       10.105466  117.68407      \n",
       "CG7715          Set 1        7.740851  105.98379      \n",
       "CG10154         Set 1       10.194348  103.93727      \n",
       "obst-J          Set 1        9.665504  101.14508      \n",
       "CG7714          Set 1        8.186656   99.94147      \n",
       "CG6933          Set 1        7.921489   96.28125      \n",
       "Pgant4          Set 1        8.552797   94.26567      \n",
       "Peritrophin-15a Set 1        9.383674   93.24503      \n",
       "Kaz-m1          Set 1        7.774562   92.30940      \n",
       "CG9988          Set 1        9.478500   90.62082      \n",
       "CG5084          Set 1        8.450312   83.44946      \n",
       "obst-G          Set 1        9.070278   82.29427      \n",
       "obst-H          Set 1        7.732380   80.80839      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "two.set.differential.gene.test.3a(cds_sub, \n",
    "                                  df_sub300$cluster == a &\n",
    "                                  df_sub300$umap_2 < 1.8*df_sub300$umap_1 -4.4 &\n",
    "                                  df_sub300$umap_2 >= 1.8*df_sub300$umap_1 -4.9, \n",
    "                                 !(df_sub300$cluster == a &\n",
    "                                   df_sub300$umap_2 < 1.8*df_sub300$umap_1 -4.4 &\n",
    "                                  df_sub300$umap_2 >= 1.8*df_sub300$umap_1 -4.9) ,\n",
    "                                 sample = 20000)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Muc26B\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Muc26B\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Muc26B\"), \n",
    "                   size_pos = 4, size_neg = 2, stroke = 0.8, limit = 1) \n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG7017\"), \n",
    "                   size_pos = 4, size_neg = 2, stroke = 0.8, limit = 1) \n",
    "\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"\"), \n",
    "                   size_pos = 4, size_neg = 2, stroke = 0.9, limit = 1) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'a' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'a' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(a)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "              df_sub300$umap_2 >= 1.8*umap_1 -4.4 &\n",
    "              df_sub300$umap_2 < 1.8*umap_1 -3.95\n",
    "              ), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 4, 0.2), stroke = 0.5) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 1.8, intercept = -4.4) +\n",
    "geom_abline(slope = 1.8, intercept = -4.9) +\n",
    "geom_abline(slope = 1.8, intercept = -3.95) +\n",
    "xlim(2, 6) + ylim(0,4)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1773,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 127\n",
      "\n",
      "# of cells sampled in set 2: 20000\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Nepl15</th><td>Nepl15     </td><td>377</td><td> 145</td><td>6282.1009</td><td> 14.8561718</td><td>Set 1</td><td>8.630059</td><td>108.88770</td></tr>\n",
       "\t<tr><th scope=row>Mdr65</th><td>Mdr65      </td><td> 73</td><td>  39</td><td>1456.0449</td><td>  3.8260605</td><td>Set 1</td><td>8.236993</td><td> 86.56116</td></tr>\n",
       "\t<tr><th scope=row>CG32032</th><td>CG32032    </td><td>139</td><td> 143</td><td>2615.0025</td><td> 15.3889816</td><td>Set 1</td><td>7.317942</td><td> 83.08168</td></tr>\n",
       "\t<tr><th scope=row>CG5687</th><td>CG5687     </td><td> 41</td><td>  64</td><td> 814.1171</td><td>  5.4370560</td><td>Set 1</td><td>6.982691</td><td> 67.52866</td></tr>\n",
       "\t<tr><th scope=row>CG43332</th><td>CG43332    </td><td>  0</td><td>2401</td><td>   0.0000</td><td>281.3025212</td><td>Set 2</td><td>8.135979</td><td> 66.23580</td></tr>\n",
       "\t<tr><th scope=row>nAChRalpha6</th><td>nAChRalpha6</td><td>  0</td><td>2343</td><td>   0.0000</td><td>279.8162283</td><td>Set 2</td><td>8.128336</td><td> 66.11168</td></tr>\n",
       "\t<tr><th scope=row>Spn43Ad</th><td>Spn43Ad    </td><td> 21</td><td>  15</td><td> 382.7064</td><td>  1.1310144</td><td>Set 1</td><td>7.488554</td><td> 64.28069</td></tr>\n",
       "\t<tr><th scope=row>Pxn</th><td>Pxn        </td><td>  0</td><td>2064</td><td>   0.0000</td><td>250.5413048</td><td>Set 2</td><td>7.968905</td><td> 63.54924</td></tr>\n",
       "\t<tr><th scope=row>CG15203</th><td>CG15203    </td><td> 45</td><td>  52</td><td> 601.0952</td><td>  4.4261253</td><td>Set 1</td><td>6.791527</td><td> 62.71193</td></tr>\n",
       "\t<tr><th scope=row>CG42724</th><td>CG42724    </td><td>  0</td><td>2045</td><td>   0.0000</td><td>230.8473489</td><td>Set 2</td><td>7.850795</td><td> 61.68395</td></tr>\n",
       "\t<tr><th scope=row>Maf1</th><td>Maf1       </td><td>  0</td><td>1901</td><td>   0.0000</td><td>216.3518891</td><td>Set 2</td><td>7.757236</td><td> 60.22632</td></tr>\n",
       "\t<tr><th scope=row>bdl</th><td>bdl        </td><td> 31</td><td>  50</td><td> 608.7306</td><td>  5.8567406</td><td>Set 1</td><td>6.472137</td><td> 59.88039</td></tr>\n",
       "\t<tr><th scope=row>Cyp9h1</th><td>Cyp9h1     </td><td> 12</td><td>   5</td><td> 236.6051</td><td>  0.4035306</td><td>Set 1</td><td>7.397277</td><td> 58.38243</td></tr>\n",
       "\t<tr><th scope=row>CG9336</th><td>CG9336     </td><td>129</td><td> 574</td><td>2425.1081</td><td> 65.6706018</td><td>Set 1</td><td>5.184854</td><td> 58.30072</td></tr>\n",
       "\t<tr><th scope=row>Notum</th><td>Notum      </td><td>  0</td><td>1735</td><td>   0.0000</td><td>186.8421458</td><td>Set 2</td><td>7.545676</td><td> 56.99534</td></tr>\n",
       "\t<tr><th scope=row>prc</th><td>prc        </td><td>176</td><td>1003</td><td>3015.3745</td><td>101.3198296</td><td>Set 1</td><td>4.881180</td><td> 56.41960</td></tr>\n",
       "\t<tr><th scope=row>CG10226</th><td>CG10226    </td><td> 11</td><td>  22</td><td> 252.9464</td><td>  1.1307857</td><td>Set 1</td><td>6.891303</td><td> 55.05035</td></tr>\n",
       "\t<tr><th scope=row>CG3168</th><td>CG3168     </td><td>135</td><td>1451</td><td>2209.8482</td><td> 71.4040953</td><td>Set 1</td><td>4.931732</td><td> 54.79344</td></tr>\n",
       "\t<tr><th scope=row>bib</th><td>bib        </td><td>  0</td><td>1589</td><td>   0.0000</td><td>163.8646473</td><td>Set 2</td><td>7.356361</td><td> 54.18061</td></tr>\n",
       "\t<tr><th scope=row>srp</th><td>srp        </td><td>  0</td><td>1487</td><td>   0.0000</td><td>159.5312918</td><td>Set 2</td><td>7.317696</td><td> 53.61464</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tNepl15 & Nepl15      & 377 &  145 & 6282.1009 &  14.8561718 & Set 1 & 8.630059 & 108.88770\\\\\n",
       "\tMdr65 & Mdr65       &  73 &   39 & 1456.0449 &   3.8260605 & Set 1 & 8.236993 &  86.56116\\\\\n",
       "\tCG32032 & CG32032     & 139 &  143 & 2615.0025 &  15.3889816 & Set 1 & 7.317942 &  83.08168\\\\\n",
       "\tCG5687 & CG5687      &  41 &   64 &  814.1171 &   5.4370560 & Set 1 & 6.982691 &  67.52866\\\\\n",
       "\tCG43332 & CG43332     &   0 & 2401 &    0.0000 & 281.3025212 & Set 2 & 8.135979 &  66.23580\\\\\n",
       "\tnAChRalpha6 & nAChRalpha6 &   0 & 2343 &    0.0000 & 279.8162283 & Set 2 & 8.128336 &  66.11168\\\\\n",
       "\tSpn43Ad & Spn43Ad     &  21 &   15 &  382.7064 &   1.1310144 & Set 1 & 7.488554 &  64.28069\\\\\n",
       "\tPxn & Pxn         &   0 & 2064 &    0.0000 & 250.5413048 & Set 2 & 7.968905 &  63.54924\\\\\n",
       "\tCG15203 & CG15203     &  45 &   52 &  601.0952 &   4.4261253 & Set 1 & 6.791527 &  62.71193\\\\\n",
       "\tCG42724 & CG42724     &   0 & 2045 &    0.0000 & 230.8473489 & Set 2 & 7.850795 &  61.68395\\\\\n",
       "\tMaf1 & Maf1        &   0 & 1901 &    0.0000 & 216.3518891 & Set 2 & 7.757236 &  60.22632\\\\\n",
       "\tbdl & bdl         &  31 &   50 &  608.7306 &   5.8567406 & Set 1 & 6.472137 &  59.88039\\\\\n",
       "\tCyp9h1 & Cyp9h1      &  12 &    5 &  236.6051 &   0.4035306 & Set 1 & 7.397277 &  58.38243\\\\\n",
       "\tCG9336 & CG9336      & 129 &  574 & 2425.1081 &  65.6706018 & Set 1 & 5.184854 &  58.30072\\\\\n",
       "\tNotum & Notum       &   0 & 1735 &    0.0000 & 186.8421458 & Set 2 & 7.545676 &  56.99534\\\\\n",
       "\tprc & prc         & 176 & 1003 & 3015.3745 & 101.3198296 & Set 1 & 4.881180 &  56.41960\\\\\n",
       "\tCG10226 & CG10226     &  11 &   22 &  252.9464 &   1.1307857 & Set 1 & 6.891303 &  55.05035\\\\\n",
       "\tCG3168 & CG3168      & 135 & 1451 & 2209.8482 &  71.4040953 & Set 1 & 4.931732 &  54.79344\\\\\n",
       "\tbib & bib         &   0 & 1589 &    0.0000 & 163.8646473 & Set 2 & 7.356361 &  54.18061\\\\\n",
       "\tsrp & srp         &   0 & 1487 &    0.0000 & 159.5312918 & Set 2 & 7.317696 &  53.61464\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Nepl15 | Nepl15      | 377 |  145 | 6282.1009 |  14.8561718 | Set 1 | 8.630059 | 108.88770 |\n",
       "| Mdr65 | Mdr65       |  73 |   39 | 1456.0449 |   3.8260605 | Set 1 | 8.236993 |  86.56116 |\n",
       "| CG32032 | CG32032     | 139 |  143 | 2615.0025 |  15.3889816 | Set 1 | 7.317942 |  83.08168 |\n",
       "| CG5687 | CG5687      |  41 |   64 |  814.1171 |   5.4370560 | Set 1 | 6.982691 |  67.52866 |\n",
       "| CG43332 | CG43332     |   0 | 2401 |    0.0000 | 281.3025212 | Set 2 | 8.135979 |  66.23580 |\n",
       "| nAChRalpha6 | nAChRalpha6 |   0 | 2343 |    0.0000 | 279.8162283 | Set 2 | 8.128336 |  66.11168 |\n",
       "| Spn43Ad | Spn43Ad     |  21 |   15 |  382.7064 |   1.1310144 | Set 1 | 7.488554 |  64.28069 |\n",
       "| Pxn | Pxn         |   0 | 2064 |    0.0000 | 250.5413048 | Set 2 | 7.968905 |  63.54924 |\n",
       "| CG15203 | CG15203     |  45 |   52 |  601.0952 |   4.4261253 | Set 1 | 6.791527 |  62.71193 |\n",
       "| CG42724 | CG42724     |   0 | 2045 |    0.0000 | 230.8473489 | Set 2 | 7.850795 |  61.68395 |\n",
       "| Maf1 | Maf1        |   0 | 1901 |    0.0000 | 216.3518891 | Set 2 | 7.757236 |  60.22632 |\n",
       "| bdl | bdl         |  31 |   50 |  608.7306 |   5.8567406 | Set 1 | 6.472137 |  59.88039 |\n",
       "| Cyp9h1 | Cyp9h1      |  12 |    5 |  236.6051 |   0.4035306 | Set 1 | 7.397277 |  58.38243 |\n",
       "| CG9336 | CG9336      | 129 |  574 | 2425.1081 |  65.6706018 | Set 1 | 5.184854 |  58.30072 |\n",
       "| Notum | Notum       |   0 | 1735 |    0.0000 | 186.8421458 | Set 2 | 7.545676 |  56.99534 |\n",
       "| prc | prc         | 176 | 1003 | 3015.3745 | 101.3198296 | Set 1 | 4.881180 |  56.41960 |\n",
       "| CG10226 | CG10226     |  11 |   22 |  252.9464 |   1.1307857 | Set 1 | 6.891303 |  55.05035 |\n",
       "| CG3168 | CG3168      | 135 | 1451 | 2209.8482 |  71.4040953 | Set 1 | 4.931732 |  54.79344 |\n",
       "| bib | bib         |   0 | 1589 |    0.0000 | 163.8646473 | Set 2 | 7.356361 |  54.18061 |\n",
       "| srp | srp         |   0 | 1487 |    0.0000 | 159.5312918 | Set 2 | 7.317696 |  53.61464 |\n",
       "\n"
      ],
      "text/plain": [
       "            gene        set.1.n.umi set.2.n.umi set.1.tpm set.2.tpm  \n",
       "Nepl15      Nepl15      377          145        6282.1009  14.8561718\n",
       "Mdr65       Mdr65        73           39        1456.0449   3.8260605\n",
       "CG32032     CG32032     139          143        2615.0025  15.3889816\n",
       "CG5687      CG5687       41           64         814.1171   5.4370560\n",
       "CG43332     CG43332       0         2401           0.0000 281.3025212\n",
       "nAChRalpha6 nAChRalpha6   0         2343           0.0000 279.8162283\n",
       "Spn43Ad     Spn43Ad      21           15         382.7064   1.1310144\n",
       "Pxn         Pxn           0         2064           0.0000 250.5413048\n",
       "CG15203     CG15203      45           52         601.0952   4.4261253\n",
       "CG42724     CG42724       0         2045           0.0000 230.8473489\n",
       "Maf1        Maf1          0         1901           0.0000 216.3518891\n",
       "bdl         bdl          31           50         608.7306   5.8567406\n",
       "Cyp9h1      Cyp9h1       12            5         236.6051   0.4035306\n",
       "CG9336      CG9336      129          574        2425.1081  65.6706018\n",
       "Notum       Notum         0         1735           0.0000 186.8421458\n",
       "prc         prc         176         1003        3015.3745 101.3198296\n",
       "CG10226     CG10226      11           22         252.9464   1.1307857\n",
       "CG3168      CG3168      135         1451        2209.8482  71.4040953\n",
       "bib         bib           0         1589           0.0000 163.8646473\n",
       "srp         srp           0         1487           0.0000 159.5312918\n",
       "            higher.expr log2.ratio heuristic.score\n",
       "Nepl15      Set 1       8.630059   108.88770      \n",
       "Mdr65       Set 1       8.236993    86.56116      \n",
       "CG32032     Set 1       7.317942    83.08168      \n",
       "CG5687      Set 1       6.982691    67.52866      \n",
       "CG43332     Set 2       8.135979    66.23580      \n",
       "nAChRalpha6 Set 2       8.128336    66.11168      \n",
       "Spn43Ad     Set 1       7.488554    64.28069      \n",
       "Pxn         Set 2       7.968905    63.54924      \n",
       "CG15203     Set 1       6.791527    62.71193      \n",
       "CG42724     Set 2       7.850795    61.68395      \n",
       "Maf1        Set 2       7.757236    60.22632      \n",
       "bdl         Set 1       6.472137    59.88039      \n",
       "Cyp9h1      Set 1       7.397277    58.38243      \n",
       "CG9336      Set 1       5.184854    58.30072      \n",
       "Notum       Set 2       7.545676    56.99534      \n",
       "prc         Set 1       4.881180    56.41960      \n",
       "CG10226     Set 1       6.891303    55.05035      \n",
       "CG3168      Set 1       4.931732    54.79344      \n",
       "bib         Set 2       7.356361    54.18061      \n",
       "srp         Set 2       7.317696    53.61464      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "two.set.differential.gene.test.3a(cds_sub, \n",
    "                                  df_sub300$cluster == a &\n",
    "                                  df_sub300$umap_2 >= 1.8*df_sub300$umap_1 -4.4 &\n",
    "              df_sub300$umap_2 < 1.8*df_sub300$umap_1 -3.99, \n",
    "                                 !(df_sub300$cluster == a &\n",
    "                                   df_sub300$umap_2 >= 1.8*df_sub300$umap_1 -4.4 &\n",
    "              df_sub300$umap_2 < 1.8*df_sub300$umap_1 -3.99)  ,\n",
    "                                 sample = 20000)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Mdr65\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Mdr65\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Mdr65\"), \n",
    "                   size_pos = 4, size_neg = 2, stroke = 0.8, limit = 1) +\n",
    "xlim(2, 6) + ylim(0,4)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG32198\"), \n",
    "                   size_pos = 4, size_neg = 2, stroke = 0.8, limit = 1) +\n",
    "xlim(2, 6) + ylim(0,4)\n",
    "\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG32032\"), \n",
    "                   size_pos = 4, size_neg = 2, stroke = 0.9, limit = 1) +\n",
    "xlim(2, 6) + ylim(0,4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'a' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'a' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(a)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "              df_sub300$umap_2 >= 1.8*umap_1 -3.95\n",
    "              ), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 4, 0.2), stroke = 0.5) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 1.8, intercept = -4.4) +\n",
    "geom_abline(slope = 1.8, intercept = -4.9) +\n",
    "geom_abline(slope = 1.8, intercept = -3.95) +\n",
    "xlim(2, 6) + ylim(0,4)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1775,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 107\n",
      "\n",
      "# of cells sampled in set 2: 20000\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG11300</th><td>CG11300</td><td>474</td><td> 9</td><td>5447.1744</td><td>0.62539446</td><td>Set 1</td><td>11.710502</td><td>145.34557</td></tr>\n",
       "\t<tr><th scope=row>CG32198</th><td>CG32198</td><td>217</td><td> 4</td><td>2784.8835</td><td>0.49615488</td><td>Set 1</td><td>10.862142</td><td>124.30547</td></tr>\n",
       "\t<tr><th scope=row>CG13460</th><td>CG13460</td><td>258</td><td> 3</td><td>1958.5329</td><td>0.31157878</td><td>Set 1</td><td>10.544253</td><td>115.31505</td></tr>\n",
       "\t<tr><th scope=row>CG12310</th><td>CG12310</td><td>140</td><td> 1</td><td>1642.5407</td><td>0.03958828</td><td>Set 1</td><td>10.625701</td><td>113.51003</td></tr>\n",
       "\t<tr><th scope=row>CG13135</th><td>CG13135</td><td> 89</td><td> 8</td><td>2140.4677</td><td>0.93511941</td><td>Set 1</td><td>10.111288</td><td>111.87517</td></tr>\n",
       "\t<tr><th scope=row>CG17134</th><td>CG17134</td><td>183</td><td>52</td><td>2826.4112</td><td>4.01105118</td><td>Set 1</td><td> 9.139642</td><td>104.78843</td></tr>\n",
       "\t<tr><th scope=row>CG32189</th><td>CG32189</td><td>111</td><td> 0</td><td>1163.3159</td><td>0.00000000</td><td>Set 1</td><td>10.184027</td><td>103.72703</td></tr>\n",
       "\t<tr><th scope=row>CG10918</th><td>CG10918</td><td>171</td><td>10</td><td>1372.4396</td><td>0.78699221</td><td>Set 1</td><td> 9.584994</td><td> 99.90993</td></tr>\n",
       "\t<tr><th scope=row>CG13461</th><td>CG13461</td><td>105</td><td> 0</td><td> 896.1248</td><td>0.00000000</td><td>Set 1</td><td> 9.807556</td><td> 96.20393</td></tr>\n",
       "\t<tr><th scope=row>CG13560</th><td>CG13560</td><td>114</td><td> 3</td><td> 954.7878</td><td>0.37312977</td><td>Set 1</td><td> 9.441568</td><td> 93.47669</td></tr>\n",
       "\t<tr><th scope=row>CG42815</th><td>CG42815</td><td> 63</td><td> 4</td><td> 969.3107</td><td>0.51417805</td><td>Set 1</td><td> 9.322280</td><td> 92.49849</td></tr>\n",
       "\t<tr><th scope=row>CG14265</th><td>CG14265</td><td> 82</td><td> 5</td><td> 884.5920</td><td>0.27246501</td><td>Set 1</td><td> 9.441242</td><td> 92.43447</td></tr>\n",
       "\t<tr><th scope=row>CG14852</th><td>CG14852</td><td> 46</td><td> 2</td><td> 831.1797</td><td>0.27835850</td><td>Set 1</td><td> 9.344724</td><td> 90.65085</td></tr>\n",
       "\t<tr><th scope=row>lcs</th><td>lcs    </td><td> 35</td><td> 0</td><td> 662.9758</td><td>0.00000000</td><td>Set 1</td><td> 9.372812</td><td> 87.86999</td></tr>\n",
       "\t<tr><th scope=row>Muc68Ca</th><td>Muc68Ca</td><td> 66</td><td>11</td><td> 919.3447</td><td>1.01535163</td><td>Set 1</td><td> 8.833431</td><td> 86.97423</td></tr>\n",
       "\t<tr><th scope=row>Ag5r</th><td>Ag5r   </td><td>137</td><td>45</td><td>1448.9853</td><td>4.88921353</td><td>Set 1</td><td> 7.942752</td><td> 83.41338</td></tr>\n",
       "\t<tr><th scope=row>CG13947</th><td>CG13947</td><td> 56</td><td>16</td><td> 816.6269</td><td>1.67407187</td><td>Set 1</td><td> 8.254495</td><td> 79.86471</td></tr>\n",
       "\t<tr><th scope=row>CG31496</th><td>CG31496</td><td> 46</td><td> 6</td><td> 626.7147</td><td>0.63359132</td><td>Set 1</td><td> 8.583618</td><td> 79.77584</td></tr>\n",
       "\t<tr><th scope=row>CG32188</th><td>CG32188</td><td> 38</td><td> 0</td><td> 428.8767</td><td>0.00000000</td><td>Set 1</td><td> 8.744419</td><td> 76.49425</td></tr>\n",
       "\t<tr><th scope=row>CG9040</th><td>CG9040 </td><td> 60</td><td> 1</td><td> 415.7094</td><td>0.04016064</td><td>Set 1</td><td> 8.642625</td><td> 75.21589</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG11300 & CG11300 & 474 &  9 & 5447.1744 & 0.62539446 & Set 1 & 11.710502 & 145.34557\\\\\n",
       "\tCG32198 & CG32198 & 217 &  4 & 2784.8835 & 0.49615488 & Set 1 & 10.862142 & 124.30547\\\\\n",
       "\tCG13460 & CG13460 & 258 &  3 & 1958.5329 & 0.31157878 & Set 1 & 10.544253 & 115.31505\\\\\n",
       "\tCG12310 & CG12310 & 140 &  1 & 1642.5407 & 0.03958828 & Set 1 & 10.625701 & 113.51003\\\\\n",
       "\tCG13135 & CG13135 &  89 &  8 & 2140.4677 & 0.93511941 & Set 1 & 10.111288 & 111.87517\\\\\n",
       "\tCG17134 & CG17134 & 183 & 52 & 2826.4112 & 4.01105118 & Set 1 &  9.139642 & 104.78843\\\\\n",
       "\tCG32189 & CG32189 & 111 &  0 & 1163.3159 & 0.00000000 & Set 1 & 10.184027 & 103.72703\\\\\n",
       "\tCG10918 & CG10918 & 171 & 10 & 1372.4396 & 0.78699221 & Set 1 &  9.584994 &  99.90993\\\\\n",
       "\tCG13461 & CG13461 & 105 &  0 &  896.1248 & 0.00000000 & Set 1 &  9.807556 &  96.20393\\\\\n",
       "\tCG13560 & CG13560 & 114 &  3 &  954.7878 & 0.37312977 & Set 1 &  9.441568 &  93.47669\\\\\n",
       "\tCG42815 & CG42815 &  63 &  4 &  969.3107 & 0.51417805 & Set 1 &  9.322280 &  92.49849\\\\\n",
       "\tCG14265 & CG14265 &  82 &  5 &  884.5920 & 0.27246501 & Set 1 &  9.441242 &  92.43447\\\\\n",
       "\tCG14852 & CG14852 &  46 &  2 &  831.1797 & 0.27835850 & Set 1 &  9.344724 &  90.65085\\\\\n",
       "\tlcs & lcs     &  35 &  0 &  662.9758 & 0.00000000 & Set 1 &  9.372812 &  87.86999\\\\\n",
       "\tMuc68Ca & Muc68Ca &  66 & 11 &  919.3447 & 1.01535163 & Set 1 &  8.833431 &  86.97423\\\\\n",
       "\tAg5r & Ag5r    & 137 & 45 & 1448.9853 & 4.88921353 & Set 1 &  7.942752 &  83.41338\\\\\n",
       "\tCG13947 & CG13947 &  56 & 16 &  816.6269 & 1.67407187 & Set 1 &  8.254495 &  79.86471\\\\\n",
       "\tCG31496 & CG31496 &  46 &  6 &  626.7147 & 0.63359132 & Set 1 &  8.583618 &  79.77584\\\\\n",
       "\tCG32188 & CG32188 &  38 &  0 &  428.8767 & 0.00000000 & Set 1 &  8.744419 &  76.49425\\\\\n",
       "\tCG9040 & CG9040  &  60 &  1 &  415.7094 & 0.04016064 & Set 1 &  8.642625 &  75.21589\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG11300 | CG11300 | 474 |  9 | 5447.1744 | 0.62539446 | Set 1 | 11.710502 | 145.34557 |\n",
       "| CG32198 | CG32198 | 217 |  4 | 2784.8835 | 0.49615488 | Set 1 | 10.862142 | 124.30547 |\n",
       "| CG13460 | CG13460 | 258 |  3 | 1958.5329 | 0.31157878 | Set 1 | 10.544253 | 115.31505 |\n",
       "| CG12310 | CG12310 | 140 |  1 | 1642.5407 | 0.03958828 | Set 1 | 10.625701 | 113.51003 |\n",
       "| CG13135 | CG13135 |  89 |  8 | 2140.4677 | 0.93511941 | Set 1 | 10.111288 | 111.87517 |\n",
       "| CG17134 | CG17134 | 183 | 52 | 2826.4112 | 4.01105118 | Set 1 |  9.139642 | 104.78843 |\n",
       "| CG32189 | CG32189 | 111 |  0 | 1163.3159 | 0.00000000 | Set 1 | 10.184027 | 103.72703 |\n",
       "| CG10918 | CG10918 | 171 | 10 | 1372.4396 | 0.78699221 | Set 1 |  9.584994 |  99.90993 |\n",
       "| CG13461 | CG13461 | 105 |  0 |  896.1248 | 0.00000000 | Set 1 |  9.807556 |  96.20393 |\n",
       "| CG13560 | CG13560 | 114 |  3 |  954.7878 | 0.37312977 | Set 1 |  9.441568 |  93.47669 |\n",
       "| CG42815 | CG42815 |  63 |  4 |  969.3107 | 0.51417805 | Set 1 |  9.322280 |  92.49849 |\n",
       "| CG14265 | CG14265 |  82 |  5 |  884.5920 | 0.27246501 | Set 1 |  9.441242 |  92.43447 |\n",
       "| CG14852 | CG14852 |  46 |  2 |  831.1797 | 0.27835850 | Set 1 |  9.344724 |  90.65085 |\n",
       "| lcs | lcs     |  35 |  0 |  662.9758 | 0.00000000 | Set 1 |  9.372812 |  87.86999 |\n",
       "| Muc68Ca | Muc68Ca |  66 | 11 |  919.3447 | 1.01535163 | Set 1 |  8.833431 |  86.97423 |\n",
       "| Ag5r | Ag5r    | 137 | 45 | 1448.9853 | 4.88921353 | Set 1 |  7.942752 |  83.41338 |\n",
       "| CG13947 | CG13947 |  56 | 16 |  816.6269 | 1.67407187 | Set 1 |  8.254495 |  79.86471 |\n",
       "| CG31496 | CG31496 |  46 |  6 |  626.7147 | 0.63359132 | Set 1 |  8.583618 |  79.77584 |\n",
       "| CG32188 | CG32188 |  38 |  0 |  428.8767 | 0.00000000 | Set 1 |  8.744419 |  76.49425 |\n",
       "| CG9040 | CG9040  |  60 |  1 |  415.7094 | 0.04016064 | Set 1 |  8.642625 |  75.21589 |\n",
       "\n"
      ],
      "text/plain": [
       "        gene    set.1.n.umi set.2.n.umi set.1.tpm set.2.tpm  higher.expr\n",
       "CG11300 CG11300 474          9          5447.1744 0.62539446 Set 1      \n",
       "CG32198 CG32198 217          4          2784.8835 0.49615488 Set 1      \n",
       "CG13460 CG13460 258          3          1958.5329 0.31157878 Set 1      \n",
       "CG12310 CG12310 140          1          1642.5407 0.03958828 Set 1      \n",
       "CG13135 CG13135  89          8          2140.4677 0.93511941 Set 1      \n",
       "CG17134 CG17134 183         52          2826.4112 4.01105118 Set 1      \n",
       "CG32189 CG32189 111          0          1163.3159 0.00000000 Set 1      \n",
       "CG10918 CG10918 171         10          1372.4396 0.78699221 Set 1      \n",
       "CG13461 CG13461 105          0           896.1248 0.00000000 Set 1      \n",
       "CG13560 CG13560 114          3           954.7878 0.37312977 Set 1      \n",
       "CG42815 CG42815  63          4           969.3107 0.51417805 Set 1      \n",
       "CG14265 CG14265  82          5           884.5920 0.27246501 Set 1      \n",
       "CG14852 CG14852  46          2           831.1797 0.27835850 Set 1      \n",
       "lcs     lcs      35          0           662.9758 0.00000000 Set 1      \n",
       "Muc68Ca Muc68Ca  66         11           919.3447 1.01535163 Set 1      \n",
       "Ag5r    Ag5r    137         45          1448.9853 4.88921353 Set 1      \n",
       "CG13947 CG13947  56         16           816.6269 1.67407187 Set 1      \n",
       "CG31496 CG31496  46          6           626.7147 0.63359132 Set 1      \n",
       "CG32188 CG32188  38          0           428.8767 0.00000000 Set 1      \n",
       "CG9040  CG9040   60          1           415.7094 0.04016064 Set 1      \n",
       "        log2.ratio heuristic.score\n",
       "CG11300 11.710502  145.34557      \n",
       "CG32198 10.862142  124.30547      \n",
       "CG13460 10.544253  115.31505      \n",
       "CG12310 10.625701  113.51003      \n",
       "CG13135 10.111288  111.87517      \n",
       "CG17134  9.139642  104.78843      \n",
       "CG32189 10.184027  103.72703      \n",
       "CG10918  9.584994   99.90993      \n",
       "CG13461  9.807556   96.20393      \n",
       "CG13560  9.441568   93.47669      \n",
       "CG42815  9.322280   92.49849      \n",
       "CG14265  9.441242   92.43447      \n",
       "CG14852  9.344724   90.65085      \n",
       "lcs      9.372812   87.86999      \n",
       "Muc68Ca  8.833431   86.97423      \n",
       "Ag5r     7.942752   83.41338      \n",
       "CG13947  8.254495   79.86471      \n",
       "CG31496  8.583618   79.77584      \n",
       "CG32188  8.744419   76.49425      \n",
       "CG9040   8.642625   75.21589      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "two.set.differential.gene.test.3a(cds_sub, \n",
    "                                  df_sub300$cluster == a &\n",
    "                                  df_sub300$umap_2 >= 1.8*df_sub300$umap_1 -3.95 , \n",
    "                                 !(df_sub300$cluster == a &\n",
    "                                   df_sub300$umap_2 >= 1.8*df_sub300$umap_1 -3.95 )  ,\n",
    "                                 sample = 20000)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### df_sub300 = set.cell.type(df_sub300, \"Malpighian tubule\", df_sub300$cluster %in% b)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 789,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Chordotonal_organ\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# α-Tubulin at 85E (αTub85E) encodes the major α-Tubulin constituent of accessory and \n",
    "# attachment cells of chordotonal organs. [Date last reviewed: 2019-03-21] (FlyBase Gene Snapshot)\n",
    "# 141 (f) also has chordotonal signal from alphaTub85E, CG7409\n",
    "df_sub300 = set.cell.type(df_sub300, \"Chordotonal_organ\", df_sub300$cluster %in% c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(c)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 732,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG740954</th><td>c</td><td>CG7409     </td><td> 458</td><td>  16</td><td> 3866.9126</td><td>  0.3382439</td><td>11.496625</td><td>137.00919</td></tr>\n",
       "\t<tr><th scope=row>CG4314954</th><td>c</td><td>CG43149    </td><td> 436</td><td>  27</td><td> 3464.9822</td><td>  1.1330986</td><td>10.665682</td><td>125.41827</td></tr>\n",
       "\t<tr><th scope=row>CG1313354</th><td>c</td><td>CG13133    </td><td> 316</td><td>  16</td><td> 2682.3791</td><td>  0.6332884</td><td>10.681518</td><td>121.66073</td></tr>\n",
       "\t<tr><th scope=row>CG1444554</th><td>c</td><td>CG14445    </td><td> 395</td><td>  65</td><td> 2888.5918</td><td>  1.5446626</td><td>10.148676</td><td>116.67578</td></tr>\n",
       "\t<tr><th scope=row>alphaTub85E54</th><td>c</td><td>alphaTub85E</td><td>2602</td><td>2401</td><td>21206.5698</td><td> 80.7327894</td><td> 8.019381</td><td>115.25688</td></tr>\n",
       "\t<tr><th scope=row>CG3134554</th><td>c</td><td>CG31345    </td><td> 154</td><td>  76</td><td> 1216.6799</td><td>  2.8782176</td><td> 8.293340</td><td> 85.00607</td></tr>\n",
       "\t<tr><th scope=row>sosie54</th><td>c</td><td>sosie      </td><td> 127</td><td> 118</td><td>  973.5354</td><td>  4.2438966</td><td> 7.536450</td><td> 74.82618</td></tr>\n",
       "\t<tr><th scope=row>Tektin-C54</th><td>c</td><td>Tektin-C   </td><td> 177</td><td> 141</td><td> 1197.0941</td><td>  6.6733289</td><td> 7.285468</td><td> 74.50503</td></tr>\n",
       "\t<tr><th scope=row>epsilonTry54</th><td>c</td><td>epsilonTry </td><td>  41</td><td> 371</td><td> 1166.9506</td><td>  8.3214057</td><td> 6.967980</td><td> 71.00207</td></tr>\n",
       "\t<tr><th scope=row>CAP54</th><td>c</td><td>CAP        </td><td> 801</td><td>5252</td><td> 8475.0857</td><td>205.3194991</td><td> 5.360276</td><td> 69.94722</td></tr>\n",
       "\t<tr><th scope=row>CG605554</th><td>c</td><td>CG6055     </td><td> 196</td><td> 540</td><td> 1846.7878</td><td> 21.7197133</td><td> 6.344930</td><td> 68.85253</td></tr>\n",
       "\t<tr><th scope=row>Adgf-C54</th><td>c</td><td>Adgf-C     </td><td>  24</td><td>  18</td><td>  373.9197</td><td>  0.4402711</td><td> 8.020244</td><td> 68.57660</td></tr>\n",
       "\t<tr><th scope=row>Root54</th><td>c</td><td>Root       </td><td> 204</td><td> 446</td><td> 1536.4017</td><td> 16.8977175</td><td> 6.423636</td><td> 68.00240</td></tr>\n",
       "\t<tr><th scope=row>CG1006254</th><td>c</td><td>CG10062    </td><td>  61</td><td>  52</td><td>  523.7765</td><td>  2.0544652</td><td> 7.421888</td><td> 67.06091</td></tr>\n",
       "\t<tr><th scope=row>betaTry54</th><td>c</td><td>betaTry    </td><td>  23</td><td> 285</td><td>  633.5182</td><td>  5.0075317</td><td> 6.720470</td><td> 62.56433</td></tr>\n",
       "\t<tr><th scope=row>kn54</th><td>c</td><td>kn         </td><td>   0</td><td>5184</td><td>    0.0000</td><td>221.1771521</td><td> 7.789059</td><td> 60.72013</td></tr>\n",
       "\t<tr><th scope=row>CG1190554</th><td>c</td><td>CG11905    </td><td>   0</td><td>4928</td><td>    0.0000</td><td>212.0292709</td><td> 7.728120</td><td> 59.77629</td></tr>\n",
       "\t<tr><th scope=row>bdl54</th><td>c</td><td>bdl        </td><td>  57</td><td> 159</td><td>  632.4113</td><td>  6.5253570</td><td> 6.392959</td><td> 59.49926</td></tr>\n",
       "\t<tr><th scope=row>Ptr54</th><td>c</td><td>Ptr        </td><td>   0</td><td>5045</td><td>    0.0000</td><td>207.6105212</td><td> 7.697736</td><td> 59.30850</td></tr>\n",
       "\t<tr><th scope=row>CG1320354</th><td>c</td><td>CG13203    </td><td>  44</td><td>  95</td><td>  444.1414</td><td>  3.2921534</td><td> 6.693174</td><td> 58.88734</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG740954 & c & CG7409      &  458 &   16 &  3866.9126 &   0.3382439 & 11.496625 & 137.00919\\\\\n",
       "\tCG4314954 & c & CG43149     &  436 &   27 &  3464.9822 &   1.1330986 & 10.665682 & 125.41827\\\\\n",
       "\tCG1313354 & c & CG13133     &  316 &   16 &  2682.3791 &   0.6332884 & 10.681518 & 121.66073\\\\\n",
       "\tCG1444554 & c & CG14445     &  395 &   65 &  2888.5918 &   1.5446626 & 10.148676 & 116.67578\\\\\n",
       "\talphaTub85E54 & c & alphaTub85E & 2602 & 2401 & 21206.5698 &  80.7327894 &  8.019381 & 115.25688\\\\\n",
       "\tCG3134554 & c & CG31345     &  154 &   76 &  1216.6799 &   2.8782176 &  8.293340 &  85.00607\\\\\n",
       "\tsosie54 & c & sosie       &  127 &  118 &   973.5354 &   4.2438966 &  7.536450 &  74.82618\\\\\n",
       "\tTektin-C54 & c & Tektin-C    &  177 &  141 &  1197.0941 &   6.6733289 &  7.285468 &  74.50503\\\\\n",
       "\tepsilonTry54 & c & epsilonTry  &   41 &  371 &  1166.9506 &   8.3214057 &  6.967980 &  71.00207\\\\\n",
       "\tCAP54 & c & CAP         &  801 & 5252 &  8475.0857 & 205.3194991 &  5.360276 &  69.94722\\\\\n",
       "\tCG605554 & c & CG6055      &  196 &  540 &  1846.7878 &  21.7197133 &  6.344930 &  68.85253\\\\\n",
       "\tAdgf-C54 & c & Adgf-C      &   24 &   18 &   373.9197 &   0.4402711 &  8.020244 &  68.57660\\\\\n",
       "\tRoot54 & c & Root        &  204 &  446 &  1536.4017 &  16.8977175 &  6.423636 &  68.00240\\\\\n",
       "\tCG1006254 & c & CG10062     &   61 &   52 &   523.7765 &   2.0544652 &  7.421888 &  67.06091\\\\\n",
       "\tbetaTry54 & c & betaTry     &   23 &  285 &   633.5182 &   5.0075317 &  6.720470 &  62.56433\\\\\n",
       "\tkn54 & c & kn          &    0 & 5184 &     0.0000 & 221.1771521 &  7.789059 &  60.72013\\\\\n",
       "\tCG1190554 & c & CG11905     &    0 & 4928 &     0.0000 & 212.0292709 &  7.728120 &  59.77629\\\\\n",
       "\tbdl54 & c & bdl         &   57 &  159 &   632.4113 &   6.5253570 &  6.392959 &  59.49926\\\\\n",
       "\tPtr54 & c & Ptr         &    0 & 5045 &     0.0000 & 207.6105212 &  7.697736 &  59.30850\\\\\n",
       "\tCG1320354 & c & CG13203     &   44 &   95 &   444.1414 &   3.2921534 &  6.693174 &  58.88734\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG740954 | c | CG7409      |  458 |   16 |  3866.9126 |   0.3382439 | 11.496625 | 137.00919 |\n",
       "| CG4314954 | c | CG43149     |  436 |   27 |  3464.9822 |   1.1330986 | 10.665682 | 125.41827 |\n",
       "| CG1313354 | c | CG13133     |  316 |   16 |  2682.3791 |   0.6332884 | 10.681518 | 121.66073 |\n",
       "| CG1444554 | c | CG14445     |  395 |   65 |  2888.5918 |   1.5446626 | 10.148676 | 116.67578 |\n",
       "| alphaTub85E54 | c | alphaTub85E | 2602 | 2401 | 21206.5698 |  80.7327894 |  8.019381 | 115.25688 |\n",
       "| CG3134554 | c | CG31345     |  154 |   76 |  1216.6799 |   2.8782176 |  8.293340 |  85.00607 |\n",
       "| sosie54 | c | sosie       |  127 |  118 |   973.5354 |   4.2438966 |  7.536450 |  74.82618 |\n",
       "| Tektin-C54 | c | Tektin-C    |  177 |  141 |  1197.0941 |   6.6733289 |  7.285468 |  74.50503 |\n",
       "| epsilonTry54 | c | epsilonTry  |   41 |  371 |  1166.9506 |   8.3214057 |  6.967980 |  71.00207 |\n",
       "| CAP54 | c | CAP         |  801 | 5252 |  8475.0857 | 205.3194991 |  5.360276 |  69.94722 |\n",
       "| CG605554 | c | CG6055      |  196 |  540 |  1846.7878 |  21.7197133 |  6.344930 |  68.85253 |\n",
       "| Adgf-C54 | c | Adgf-C      |   24 |   18 |   373.9197 |   0.4402711 |  8.020244 |  68.57660 |\n",
       "| Root54 | c | Root        |  204 |  446 |  1536.4017 |  16.8977175 |  6.423636 |  68.00240 |\n",
       "| CG1006254 | c | CG10062     |   61 |   52 |   523.7765 |   2.0544652 |  7.421888 |  67.06091 |\n",
       "| betaTry54 | c | betaTry     |   23 |  285 |   633.5182 |   5.0075317 |  6.720470 |  62.56433 |\n",
       "| kn54 | c | kn          |    0 | 5184 |     0.0000 | 221.1771521 |  7.789059 |  60.72013 |\n",
       "| CG1190554 | c | CG11905     |    0 | 4928 |     0.0000 | 212.0292709 |  7.728120 |  59.77629 |\n",
       "| bdl54 | c | bdl         |   57 |  159 |   632.4113 |   6.5253570 |  6.392959 |  59.49926 |\n",
       "| Ptr54 | c | Ptr         |    0 | 5045 |     0.0000 | 207.6105212 |  7.697736 |  59.30850 |\n",
       "| CG1320354 | c | CG13203     |   44 |   95 |   444.1414 |   3.2921534 |  6.693174 |  58.88734 |\n",
       "\n"
      ],
      "text/plain": [
       "              cell.type.2 gene        cell.type.2.n.umi other.n.umi\n",
       "CG740954      c           CG7409       458                16       \n",
       "CG4314954     c           CG43149      436                27       \n",
       "CG1313354     c           CG13133      316                16       \n",
       "CG1444554     c           CG14445      395                65       \n",
       "alphaTub85E54 c           alphaTub85E 2602              2401       \n",
       "CG3134554     c           CG31345      154                76       \n",
       "sosie54       c           sosie        127               118       \n",
       "Tektin-C54    c           Tektin-C     177               141       \n",
       "epsilonTry54  c           epsilonTry    41               371       \n",
       "CAP54         c           CAP          801              5252       \n",
       "CG605554      c           CG6055       196               540       \n",
       "Adgf-C54      c           Adgf-C        24                18       \n",
       "Root54        c           Root         204               446       \n",
       "CG1006254     c           CG10062       61                52       \n",
       "betaTry54     c           betaTry       23               285       \n",
       "kn54          c           kn             0              5184       \n",
       "CG1190554     c           CG11905        0              4928       \n",
       "bdl54         c           bdl           57               159       \n",
       "Ptr54         c           Ptr            0              5045       \n",
       "CG1320354     c           CG13203       44                95       \n",
       "              cell.type.2.tpm other.tpm   log2.ratio heuristic.score\n",
       "CG740954       3866.9126        0.3382439 11.496625  137.00919      \n",
       "CG4314954      3464.9822        1.1330986 10.665682  125.41827      \n",
       "CG1313354      2682.3791        0.6332884 10.681518  121.66073      \n",
       "CG1444554      2888.5918        1.5446626 10.148676  116.67578      \n",
       "alphaTub85E54 21206.5698       80.7327894  8.019381  115.25688      \n",
       "CG3134554      1216.6799        2.8782176  8.293340   85.00607      \n",
       "sosie54         973.5354        4.2438966  7.536450   74.82618      \n",
       "Tektin-C54     1197.0941        6.6733289  7.285468   74.50503      \n",
       "epsilonTry54   1166.9506        8.3214057  6.967980   71.00207      \n",
       "CAP54          8475.0857      205.3194991  5.360276   69.94722      \n",
       "CG605554       1846.7878       21.7197133  6.344930   68.85253      \n",
       "Adgf-C54        373.9197        0.4402711  8.020244   68.57660      \n",
       "Root54         1536.4017       16.8977175  6.423636   68.00240      \n",
       "CG1006254       523.7765        2.0544652  7.421888   67.06091      \n",
       "betaTry54       633.5182        5.0075317  6.720470   62.56433      \n",
       "kn54              0.0000      221.1771521  7.789059   60.72013      \n",
       "CG1190554         0.0000      212.0292709  7.728120   59.77629      \n",
       "bdl54           632.4113        6.5253570  6.392959   59.49926      \n",
       "Ptr54             0.0000      207.6105212  7.697736   59.30850      \n",
       "CG1320354       444.1414        3.2921534  6.693174   58.88734      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"c\")  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"alphaTub85E\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"alphaTub85E\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"alphaTub85E\"), \n",
    "                   size_pos = 4, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG7409\"), \n",
    "                   size_pos = 3, size_neg = 0.1, stroke = 0.02, limit = 1)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG13133\"), \n",
    "                   size_pos = 3, size_neg = 0.2, stroke = 0.02, limit = 0.01)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1719,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Neuron_unknown_2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Neuron_unknown_2\", df_sub300$cluster %in% d)\n",
    "# based on presence of Nplp4 a neuropeptide."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1718,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 70 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cell.type.3</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Amnioserosa                               </td><td> 7091</td></tr>\n",
       "\t<tr><td>Analpad                                   </td><td>  850</td></tr>\n",
       "\t<tr><td>Analpad_large_intestine                   </td><td>  734</td></tr>\n",
       "\t<tr><td>Antennal_sense_organs                     </td><td>  748</td></tr>\n",
       "\t<tr><td>Bolwig_organ                              </td><td>  115</td></tr>\n",
       "\t<tr><td>Central_brain_ventral_nerve_cord_1        </td><td> 6846</td></tr>\n",
       "\t<tr><td>Central_brain_ventral_nerve_cord_2        </td><td>10161</td></tr>\n",
       "\t<tr><td>Central_brain_ventral_nerve_cord_early    </td><td> 5058</td></tr>\n",
       "\t<tr><td>Central_brain_ventral_nerve_cord_primorium</td><td> 5819</td></tr>\n",
       "\t<tr><td>Cholinergic                               </td><td> 8949</td></tr>\n",
       "\t<tr><td>Chordotonal_organ                         </td><td>  121</td></tr>\n",
       "\t<tr><td>Chordotonal_sensory                       </td><td>  750</td></tr>\n",
       "\t<tr><td>Copper_cells                              </td><td> 4327</td></tr>\n",
       "\t<tr><td>Crystal_cell                              </td><td>  432</td></tr>\n",
       "\t<tr><td>Cuprophilic_cells                         </td><td>  581</td></tr>\n",
       "\t<tr><td>Digestive_system                          </td><td> 2212</td></tr>\n",
       "\t<tr><td>Embryonic_body_wall                       </td><td>42392</td></tr>\n",
       "\t<tr><td>Epidermis                                 </td><td>17230</td></tr>\n",
       "\t<tr><td>Epidermis_late                            </td><td>  184</td></tr>\n",
       "\t<tr><td>External_sensory                          </td><td>  647</td></tr>\n",
       "\t<tr><td>Fat_body                                  </td><td> 5694</td></tr>\n",
       "\t<tr><td>Garland_cell_nephrocyte                   </td><td>  536</td></tr>\n",
       "\t<tr><td>Germ_cells_ovaries_query                  </td><td> 2139</td></tr>\n",
       "\t<tr><td>Glutamatergic                             </td><td> 5107</td></tr>\n",
       "\t<tr><td>Late_cells                                </td><td> 1565</td></tr>\n",
       "\t<tr><td>Longitudinal_visceral_muscle              </td><td>  617</td></tr>\n",
       "\t<tr><td>Malpighian tubule                         </td><td> 1731</td></tr>\n",
       "\t<tr><td>Maxillary_sensory_complex                 </td><td>  533</td></tr>\n",
       "\t<tr><td>Midgut                                    </td><td> 1674</td></tr>\n",
       "\t<tr><td>Midgut_4th_chamber                        </td><td> 2128</td></tr>\n",
       "\t<tr><td>Midgut_primordium                         </td><td>  689</td></tr>\n",
       "\t<tr><td>Monoaminergic                             </td><td>  454</td></tr>\n",
       "\t<tr><td>Multidendritic_sensory                    </td><td> 2069</td></tr>\n",
       "\t<tr><td>Muscle_system                             </td><td> 5347</td></tr>\n",
       "\t<tr><td>Neuron_unknown_1                          </td><td>  602</td></tr>\n",
       "\t<tr><td>Oenocytes                                 </td><td>  142</td></tr>\n",
       "\t<tr><td>Oenocytes_older                           </td><td>  250</td></tr>\n",
       "\t<tr><td>Olfactory_neurons                         </td><td> 1864</td></tr>\n",
       "\t<tr><td>Peptidergic_neuron                        </td><td>  147</td></tr>\n",
       "\t<tr><td>Peripheral_nervous_system_early           </td><td> 3851</td></tr>\n",
       "\t<tr><td>Pharynx                                   </td><td> 1704</td></tr>\n",
       "\t<tr><td>Pharynx_2                                 </td><td>  975</td></tr>\n",
       "\t<tr><td>Pharynx_primordium_query                  </td><td> 1990</td></tr>\n",
       "\t<tr><td>Plasmatocytes                             </td><td> 6021</td></tr>\n",
       "\t<tr><td>Posterior_spiracle                        </td><td> 1436</td></tr>\n",
       "\t<tr><td>Proventriculus                            </td><td> 2423</td></tr>\n",
       "\t<tr><td>Proventriculus_older                      </td><td>  253</td></tr>\n",
       "\t<tr><td>Salivary_gland                            </td><td> 4237</td></tr>\n",
       "\t<tr><td>Somatic_muscle                            </td><td> 2766</td></tr>\n",
       "\t<tr><td>Somatic_muscle_primordium                 </td><td> 5139</td></tr>\n",
       "\t<tr><td>Tracheal_system_1                         </td><td> 1101</td></tr>\n",
       "\t<tr><td>Tracheal_system_2                         </td><td> 2642</td></tr>\n",
       "\t<tr><td>Tracheal_system_early                     </td><td> 1797</td></tr>\n",
       "\t<tr><td>Tracheal_system_older                     </td><td> 2821</td></tr>\n",
       "\t<tr><td>Undifferentiated_1                        </td><td> 3354</td></tr>\n",
       "\t<tr><td>Undifferentiated_2                        </td><td> 1700</td></tr>\n",
       "\t<tr><td>Undifferentiated_3                        </td><td> 2511</td></tr>\n",
       "\t<tr><td>Ventral_midline                           </td><td> 2159</td></tr>\n",
       "\t<tr><td>Visceral_mesoderm                         </td><td> 1495</td></tr>\n",
       "\t<tr><td>Visceral_muscle                           </td><td> 2308</td></tr>\n",
       "\t<tr><td>Visceral_muscle_2                         </td><td> 1508</td></tr>\n",
       "\t<tr><td>gut_query                                 </td><td>  887</td></tr>\n",
       "\t<tr><td>large_intestine_hindgut_analpad           </td><td> 2226</td></tr>\n",
       "\t<tr><td>large_intestine_hindgut_analpad_2         </td><td> 1467</td></tr>\n",
       "\t<tr><td>lateral_cord_glia                         </td><td>  636</td></tr>\n",
       "\t<tr><td>proventriculus                            </td><td>  528</td></tr>\n",
       "\t<tr><td>uncertain_1                               </td><td>   59</td></tr>\n",
       "\t<tr><td>unknown_2                                 </td><td> 1629</td></tr>\n",
       "\t<tr><td>unknown_3                                 </td><td>  180</td></tr>\n",
       "\t<tr><td>NA                                        </td><td>28107</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 70 × 2\n",
       "\\begin{tabular}{ll}\n",
       " cell.type.3 & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Amnioserosa                                &  7091\\\\\n",
       "\t Analpad                                    &   850\\\\\n",
       "\t Analpad\\_large\\_intestine                    &   734\\\\\n",
       "\t Antennal\\_sense\\_organs                      &   748\\\\\n",
       "\t Bolwig\\_organ                               &   115\\\\\n",
       "\t Central\\_brain\\_ventral\\_nerve\\_cord\\_1         &  6846\\\\\n",
       "\t Central\\_brain\\_ventral\\_nerve\\_cord\\_2         & 10161\\\\\n",
       "\t Central\\_brain\\_ventral\\_nerve\\_cord\\_early     &  5058\\\\\n",
       "\t Central\\_brain\\_ventral\\_nerve\\_cord\\_primorium &  5819\\\\\n",
       "\t Cholinergic                                &  8949\\\\\n",
       "\t Chordotonal\\_organ                          &   121\\\\\n",
       "\t Chordotonal\\_sensory                        &   750\\\\\n",
       "\t Copper\\_cells                               &  4327\\\\\n",
       "\t Crystal\\_cell                               &   432\\\\\n",
       "\t Cuprophilic\\_cells                          &   581\\\\\n",
       "\t Digestive\\_system                           &  2212\\\\\n",
       "\t Embryonic\\_body\\_wall                        & 42392\\\\\n",
       "\t Epidermis                                  & 17230\\\\\n",
       "\t Epidermis\\_late                             &   184\\\\\n",
       "\t External\\_sensory                           &   647\\\\\n",
       "\t Fat\\_body                                   &  5694\\\\\n",
       "\t Garland\\_cell\\_nephrocyte                    &   536\\\\\n",
       "\t Germ\\_cells\\_ovaries\\_query                   &  2139\\\\\n",
       "\t Glutamatergic                              &  5107\\\\\n",
       "\t Late\\_cells                                 &  1565\\\\\n",
       "\t Longitudinal\\_visceral\\_muscle               &   617\\\\\n",
       "\t Malpighian tubule                          &  1731\\\\\n",
       "\t Maxillary\\_sensory\\_complex                  &   533\\\\\n",
       "\t Midgut                                     &  1674\\\\\n",
       "\t Midgut\\_4th\\_chamber                         &  2128\\\\\n",
       "\t Midgut\\_primordium                          &   689\\\\\n",
       "\t Monoaminergic                              &   454\\\\\n",
       "\t Multidendritic\\_sensory                     &  2069\\\\\n",
       "\t Muscle\\_system                              &  5347\\\\\n",
       "\t Neuron\\_unknown\\_1                           &   602\\\\\n",
       "\t Oenocytes                                  &   142\\\\\n",
       "\t Oenocytes\\_older                            &   250\\\\\n",
       "\t Olfactory\\_neurons                          &  1864\\\\\n",
       "\t Peptidergic\\_neuron                         &   147\\\\\n",
       "\t Peripheral\\_nervous\\_system\\_early            &  3851\\\\\n",
       "\t Pharynx                                    &  1704\\\\\n",
       "\t Pharynx\\_2                                  &   975\\\\\n",
       "\t Pharynx\\_primordium\\_query                   &  1990\\\\\n",
       "\t Plasmatocytes                              &  6021\\\\\n",
       "\t Posterior\\_spiracle                         &  1436\\\\\n",
       "\t Proventriculus                             &  2423\\\\\n",
       "\t Proventriculus\\_older                       &   253\\\\\n",
       "\t Salivary\\_gland                             &  4237\\\\\n",
       "\t Somatic\\_muscle                             &  2766\\\\\n",
       "\t Somatic\\_muscle\\_primordium                  &  5139\\\\\n",
       "\t Tracheal\\_system\\_1                          &  1101\\\\\n",
       "\t Tracheal\\_system\\_2                          &  2642\\\\\n",
       "\t Tracheal\\_system\\_early                      &  1797\\\\\n",
       "\t Tracheal\\_system\\_older                      &  2821\\\\\n",
       "\t Undifferentiated\\_1                         &  3354\\\\\n",
       "\t Undifferentiated\\_2                         &  1700\\\\\n",
       "\t Undifferentiated\\_3                         &  2511\\\\\n",
       "\t Ventral\\_midline                            &  2159\\\\\n",
       "\t Visceral\\_mesoderm                          &  1495\\\\\n",
       "\t Visceral\\_muscle                            &  2308\\\\\n",
       "\t Visceral\\_muscle\\_2                          &  1508\\\\\n",
       "\t gut\\_query                                  &   887\\\\\n",
       "\t large\\_intestine\\_hindgut\\_analpad            &  2226\\\\\n",
       "\t large\\_intestine\\_hindgut\\_analpad\\_2          &  1467\\\\\n",
       "\t lateral\\_cord\\_glia                          &   636\\\\\n",
       "\t proventriculus                             &   528\\\\\n",
       "\t uncertain\\_1                                &    59\\\\\n",
       "\t unknown\\_2                                  &  1629\\\\\n",
       "\t unknown\\_3                                  &   180\\\\\n",
       "\t NA                                         & 28107\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 70 × 2\n",
       "\n",
       "| cell.type.3 &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Amnioserosa                                |  7091 |\n",
       "| Analpad                                    |   850 |\n",
       "| Analpad_large_intestine                    |   734 |\n",
       "| Antennal_sense_organs                      |   748 |\n",
       "| Bolwig_organ                               |   115 |\n",
       "| Central_brain_ventral_nerve_cord_1         |  6846 |\n",
       "| Central_brain_ventral_nerve_cord_2         | 10161 |\n",
       "| Central_brain_ventral_nerve_cord_early     |  5058 |\n",
       "| Central_brain_ventral_nerve_cord_primorium |  5819 |\n",
       "| Cholinergic                                |  8949 |\n",
       "| Chordotonal_organ                          |   121 |\n",
       "| Chordotonal_sensory                        |   750 |\n",
       "| Copper_cells                               |  4327 |\n",
       "| Crystal_cell                               |   432 |\n",
       "| Cuprophilic_cells                          |   581 |\n",
       "| Digestive_system                           |  2212 |\n",
       "| Embryonic_body_wall                        | 42392 |\n",
       "| Epidermis                                  | 17230 |\n",
       "| Epidermis_late                             |   184 |\n",
       "| External_sensory                           |   647 |\n",
       "| Fat_body                                   |  5694 |\n",
       "| Garland_cell_nephrocyte                    |   536 |\n",
       "| Germ_cells_ovaries_query                   |  2139 |\n",
       "| Glutamatergic                              |  5107 |\n",
       "| Late_cells                                 |  1565 |\n",
       "| Longitudinal_visceral_muscle               |   617 |\n",
       "| Malpighian tubule                          |  1731 |\n",
       "| Maxillary_sensory_complex                  |   533 |\n",
       "| Midgut                                     |  1674 |\n",
       "| Midgut_4th_chamber                         |  2128 |\n",
       "| Midgut_primordium                          |   689 |\n",
       "| Monoaminergic                              |   454 |\n",
       "| Multidendritic_sensory                     |  2069 |\n",
       "| Muscle_system                              |  5347 |\n",
       "| Neuron_unknown_1                           |   602 |\n",
       "| Oenocytes                                  |   142 |\n",
       "| Oenocytes_older                            |   250 |\n",
       "| Olfactory_neurons                          |  1864 |\n",
       "| Peptidergic_neuron                         |   147 |\n",
       "| Peripheral_nervous_system_early            |  3851 |\n",
       "| Pharynx                                    |  1704 |\n",
       "| Pharynx_2                                  |   975 |\n",
       "| Pharynx_primordium_query                   |  1990 |\n",
       "| Plasmatocytes                              |  6021 |\n",
       "| Posterior_spiracle                         |  1436 |\n",
       "| Proventriculus                             |  2423 |\n",
       "| Proventriculus_older                       |   253 |\n",
       "| Salivary_gland                             |  4237 |\n",
       "| Somatic_muscle                             |  2766 |\n",
       "| Somatic_muscle_primordium                  |  5139 |\n",
       "| Tracheal_system_1                          |  1101 |\n",
       "| Tracheal_system_2                          |  2642 |\n",
       "| Tracheal_system_early                      |  1797 |\n",
       "| Tracheal_system_older                      |  2821 |\n",
       "| Undifferentiated_1                         |  3354 |\n",
       "| Undifferentiated_2                         |  1700 |\n",
       "| Undifferentiated_3                         |  2511 |\n",
       "| Ventral_midline                            |  2159 |\n",
       "| Visceral_mesoderm                          |  1495 |\n",
       "| Visceral_muscle                            |  2308 |\n",
       "| Visceral_muscle_2                          |  1508 |\n",
       "| gut_query                                  |   887 |\n",
       "| large_intestine_hindgut_analpad            |  2226 |\n",
       "| large_intestine_hindgut_analpad_2          |  1467 |\n",
       "| lateral_cord_glia                          |   636 |\n",
       "| proventriculus                             |   528 |\n",
       "| uncertain_1                                |    59 |\n",
       "| unknown_2                                  |  1629 |\n",
       "| unknown_3                                  |   180 |\n",
       "| NA                                         | 28107 |\n",
       "\n"
      ],
      "text/plain": [
       "   cell.type.3                                n    \n",
       "1  Amnioserosa                                 7091\n",
       "2  Analpad                                      850\n",
       "3  Analpad_large_intestine                      734\n",
       "4  Antennal_sense_organs                        748\n",
       "5  Bolwig_organ                                 115\n",
       "6  Central_brain_ventral_nerve_cord_1          6846\n",
       "7  Central_brain_ventral_nerve_cord_2         10161\n",
       "8  Central_brain_ventral_nerve_cord_early      5058\n",
       "9  Central_brain_ventral_nerve_cord_primorium  5819\n",
       "10 Cholinergic                                 8949\n",
       "11 Chordotonal_organ                            121\n",
       "12 Chordotonal_sensory                          750\n",
       "13 Copper_cells                                4327\n",
       "14 Crystal_cell                                 432\n",
       "15 Cuprophilic_cells                            581\n",
       "16 Digestive_system                            2212\n",
       "17 Embryonic_body_wall                        42392\n",
       "18 Epidermis                                  17230\n",
       "19 Epidermis_late                               184\n",
       "20 External_sensory                             647\n",
       "21 Fat_body                                    5694\n",
       "22 Garland_cell_nephrocyte                      536\n",
       "23 Germ_cells_ovaries_query                    2139\n",
       "24 Glutamatergic                               5107\n",
       "25 Late_cells                                  1565\n",
       "26 Longitudinal_visceral_muscle                 617\n",
       "27 Malpighian tubule                           1731\n",
       "28 Maxillary_sensory_complex                    533\n",
       "29 Midgut                                      1674\n",
       "30 Midgut_4th_chamber                          2128\n",
       "31 Midgut_primordium                            689\n",
       "32 Monoaminergic                                454\n",
       "33 Multidendritic_sensory                      2069\n",
       "34 Muscle_system                               5347\n",
       "35 Neuron_unknown_1                             602\n",
       "36 Oenocytes                                    142\n",
       "37 Oenocytes_older                              250\n",
       "38 Olfactory_neurons                           1864\n",
       "39 Peptidergic_neuron                           147\n",
       "40 Peripheral_nervous_system_early             3851\n",
       "41 Pharynx                                     1704\n",
       "42 Pharynx_2                                    975\n",
       "43 Pharynx_primordium_query                    1990\n",
       "44 Plasmatocytes                               6021\n",
       "45 Posterior_spiracle                          1436\n",
       "46 Proventriculus                              2423\n",
       "47 Proventriculus_older                         253\n",
       "48 Salivary_gland                              4237\n",
       "49 Somatic_muscle                              2766\n",
       "50 Somatic_muscle_primordium                   5139\n",
       "51 Tracheal_system_1                           1101\n",
       "52 Tracheal_system_2                           2642\n",
       "53 Tracheal_system_early                       1797\n",
       "54 Tracheal_system_older                       2821\n",
       "55 Undifferentiated_1                          3354\n",
       "56 Undifferentiated_2                          1700\n",
       "57 Undifferentiated_3                          2511\n",
       "58 Ventral_midline                             2159\n",
       "59 Visceral_mesoderm                           1495\n",
       "60 Visceral_muscle                             2308\n",
       "61 Visceral_muscle_2                           1508\n",
       "62 gut_query                                    887\n",
       "63 large_intestine_hindgut_analpad             2226\n",
       "64 large_intestine_hindgut_analpad_2           1467\n",
       "65 lateral_cord_glia                            636\n",
       "66 proventriculus                               528\n",
       "67 uncertain_1                                   59\n",
       "68 unknown_2                                   1629\n",
       "69 unknown_3                                    180\n",
       "70 NA                                         28107"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub300  %>% count(cell.type.3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'd' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'd' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(d)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1715,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Muc26B</th><td>a</td><td>Muc26B </td><td> 3651</td><td>121</td><td>11236.035</td><td> 3.9689778</td><td>11.142896</td><td>149.9385</td></tr>\n",
       "\t<tr><th scope=row>CG7017</th><td>a</td><td>CG7017 </td><td> 3415</td><td>236</td><td>12317.772</td><td> 9.9390129</td><td>10.137043</td><td>137.7479</td></tr>\n",
       "\t<tr><th scope=row>CG740954</th><td>c</td><td>CG7409 </td><td>  458</td><td> 16</td><td> 3866.913</td><td> 0.3382439</td><td>11.496625</td><td>137.0092</td></tr>\n",
       "\t<tr><th scope=row>CG1137061</th><td>d</td><td>CG11370</td><td> 2890</td><td>826</td><td>16311.339</td><td>18.7725763</td><td> 9.688159</td><td>135.5730</td></tr>\n",
       "\t<tr><th scope=row>CG746563</th><td>f</td><td>CG7465 </td><td>20683</td><td>922</td><td>18938.232</td><td>27.8101498</td><td> 9.360509</td><td>133.0043</td></tr>\n",
       "\t<tr><th scope=row>CG3260363</th><td>f</td><td>CG32603</td><td> 9367</td><td>299</td><td> 9354.678</td><td> 8.3255405</td><td> 9.970285</td><td>131.5243</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tMuc26B & a & Muc26B  &  3651 & 121 & 11236.035 &  3.9689778 & 11.142896 & 149.9385\\\\\n",
       "\tCG7017 & a & CG7017  &  3415 & 236 & 12317.772 &  9.9390129 & 10.137043 & 137.7479\\\\\n",
       "\tCG740954 & c & CG7409  &   458 &  16 &  3866.913 &  0.3382439 & 11.496625 & 137.0092\\\\\n",
       "\tCG1137061 & d & CG11370 &  2890 & 826 & 16311.339 & 18.7725763 &  9.688159 & 135.5730\\\\\n",
       "\tCG746563 & f & CG7465  & 20683 & 922 & 18938.232 & 27.8101498 &  9.360509 & 133.0043\\\\\n",
       "\tCG3260363 & f & CG32603 &  9367 & 299 &  9354.678 &  8.3255405 &  9.970285 & 131.5243\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Muc26B | a | Muc26B  |  3651 | 121 | 11236.035 |  3.9689778 | 11.142896 | 149.9385 |\n",
       "| CG7017 | a | CG7017  |  3415 | 236 | 12317.772 |  9.9390129 | 10.137043 | 137.7479 |\n",
       "| CG740954 | c | CG7409  |   458 |  16 |  3866.913 |  0.3382439 | 11.496625 | 137.0092 |\n",
       "| CG1137061 | d | CG11370 |  2890 | 826 | 16311.339 | 18.7725763 |  9.688159 | 135.5730 |\n",
       "| CG746563 | f | CG7465  | 20683 | 922 | 18938.232 | 27.8101498 |  9.360509 | 133.0043 |\n",
       "| CG3260363 | f | CG32603 |  9367 | 299 |  9354.678 |  8.3255405 |  9.970285 | 131.5243 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "Muc26B    a           Muc26B   3651             121         11236.035      \n",
       "CG7017    a           CG7017   3415             236         12317.772      \n",
       "CG740954  c           CG7409    458              16          3866.913      \n",
       "CG1137061 d           CG11370  2890             826         16311.339      \n",
       "CG746563  f           CG7465  20683             922         18938.232      \n",
       "CG3260363 f           CG32603  9367             299          9354.678      \n",
       "          other.tpm  log2.ratio heuristic.score\n",
       "Muc26B     3.9689778 11.142896  149.9385       \n",
       "CG7017     9.9390129 10.137043  137.7479       \n",
       "CG740954   0.3382439 11.496625  137.0092       \n",
       "CG1137061 18.7725763  9.688159  135.5730       \n",
       "CG746563  27.8101498  9.360509  133.0043       \n",
       "CG3260363  8.3255405  9.970285  131.5243       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1716,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG1137061</th><td>d</td><td>CG11370</td><td>2890</td><td>826</td><td>16311.3389</td><td>18.7725763</td><td> 9.688159</td><td>135.57295</td></tr>\n",
       "\t<tr><th scope=row>Nplp461</th><td>d</td><td>Nplp4  </td><td> 494</td><td> 28</td><td> 3733.8508</td><td> 0.9382501</td><td>10.911694</td><td>129.48727</td></tr>\n",
       "\t<tr><th scope=row>CG1456661</th><td>d</td><td>CG14566</td><td> 407</td><td> 62</td><td> 4152.7877</td><td> 2.2952246</td><td>10.299488</td><td>123.80202</td></tr>\n",
       "\t<tr><th scope=row>CG1141361</th><td>d</td><td>CG11413</td><td> 660</td><td> 87</td><td> 3416.5312</td><td> 3.5819705</td><td> 9.542348</td><td>112.01514</td></tr>\n",
       "\t<tr><th scope=row>CG801261</th><td>d</td><td>CG8012 </td><td> 150</td><td> 11</td><td> 1085.4801</td><td> 0.1985319</td><td> 9.822849</td><td> 99.06782</td></tr>\n",
       "\t<tr><th scope=row>CG1306761</th><td>d</td><td>CG13067</td><td> 353</td><td>125</td><td> 2263.7980</td><td> 3.9548949</td><td> 8.835675</td><td> 98.47507</td></tr>\n",
       "\t<tr><th scope=row>CG1367861</th><td>d</td><td>CG13678</td><td> 280</td><td>177</td><td> 2273.6973</td><td> 6.6786228</td><td> 8.209977</td><td> 91.55322</td></tr>\n",
       "\t<tr><th scope=row>CG1304461</th><td>d</td><td>CG13044</td><td> 516</td><td>318</td><td> 2717.6486</td><td>10.3132017</td><td> 7.908208</td><td> 90.22216</td></tr>\n",
       "\t<tr><th scope=row>CG863061</th><td>d</td><td>CG8630 </td><td> 381</td><td>300</td><td> 2484.6331</td><td> 9.0119569</td><td> 7.955165</td><td> 89.72947</td></tr>\n",
       "\t<tr><th scope=row>CG1456961</th><td>d</td><td>CG14569</td><td> 230</td><td> 81</td><td> 1127.2510</td><td> 2.6591611</td><td> 8.267080</td><td> 83.82714</td></tr>\n",
       "\t<tr><th scope=row>Cpr49Ag61</th><td>d</td><td>Cpr49Ag</td><td> 182</td><td>180</td><td> 1543.2851</td><td> 6.1895321</td><td> 7.745891</td><td> 82.05008</td></tr>\n",
       "\t<tr><th scope=row>CG1457261</th><td>d</td><td>CG14572</td><td>  90</td><td> 86</td><td>  660.2793</td><td> 2.4496515</td><td> 7.580482</td><td> 71.02241</td></tr>\n",
       "\t<tr><th scope=row>CG1185261</th><td>d</td><td>CG11852</td><td> 121</td><td>154</td><td>  865.6844</td><td> 4.6218260</td><td> 7.266659</td><td> 70.91796</td></tr>\n",
       "\t<tr><th scope=row>CAH961</th><td>d</td><td>CAH9   </td><td>  70</td><td> 52</td><td>  557.8493</td><td> 1.9003399</td><td> 7.587510</td><td> 69.24601</td></tr>\n",
       "\t<tr><th scope=row>CG516261</th><td>d</td><td>CG5162 </td><td>  48</td><td> 16</td><td>  345.1249</td><td> 0.5076798</td><td> 7.838644</td><td> 66.12013</td></tr>\n",
       "\t<tr><th scope=row>CG981261</th><td>d</td><td>CG9812 </td><td> 179</td><td>688</td><td> 1206.2896</td><td>27.6027239</td><td> 5.398280</td><td> 55.26519</td></tr>\n",
       "\t<tr><th scope=row>Cpr12A61</th><td>d</td><td>Cpr12A </td><td>  18</td><td>  7</td><td>  181.9422</td><td> 0.3152279</td><td> 7.112024</td><td> 53.44860</td></tr>\n",
       "\t<tr><th scope=row>CAH261</th><td>d</td><td>CAH2   </td><td> 119</td><td>904</td><td> 1078.6168</td><td>31.2954136</td><td> 5.061709</td><td> 51.00332</td></tr>\n",
       "\t<tr><th scope=row>CG1456561</th><td>d</td><td>CG14565</td><td>  80</td><td>128</td><td>  309.2638</td><td> 3.5324118</td><td> 6.092415</td><td> 50.42906</td></tr>\n",
       "\t<tr><th scope=row>CG1719061</th><td>d</td><td>CG17190</td><td>  11</td><td> 13</td><td>  151.6676</td><td> 0.2860525</td><td> 6.881819</td><td> 49.92243</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG1137061 & d & CG11370 & 2890 & 826 & 16311.3389 & 18.7725763 &  9.688159 & 135.57295\\\\\n",
       "\tNplp461 & d & Nplp4   &  494 &  28 &  3733.8508 &  0.9382501 & 10.911694 & 129.48727\\\\\n",
       "\tCG1456661 & d & CG14566 &  407 &  62 &  4152.7877 &  2.2952246 & 10.299488 & 123.80202\\\\\n",
       "\tCG1141361 & d & CG11413 &  660 &  87 &  3416.5312 &  3.5819705 &  9.542348 & 112.01514\\\\\n",
       "\tCG801261 & d & CG8012  &  150 &  11 &  1085.4801 &  0.1985319 &  9.822849 &  99.06782\\\\\n",
       "\tCG1306761 & d & CG13067 &  353 & 125 &  2263.7980 &  3.9548949 &  8.835675 &  98.47507\\\\\n",
       "\tCG1367861 & d & CG13678 &  280 & 177 &  2273.6973 &  6.6786228 &  8.209977 &  91.55322\\\\\n",
       "\tCG1304461 & d & CG13044 &  516 & 318 &  2717.6486 & 10.3132017 &  7.908208 &  90.22216\\\\\n",
       "\tCG863061 & d & CG8630  &  381 & 300 &  2484.6331 &  9.0119569 &  7.955165 &  89.72947\\\\\n",
       "\tCG1456961 & d & CG14569 &  230 &  81 &  1127.2510 &  2.6591611 &  8.267080 &  83.82714\\\\\n",
       "\tCpr49Ag61 & d & Cpr49Ag &  182 & 180 &  1543.2851 &  6.1895321 &  7.745891 &  82.05008\\\\\n",
       "\tCG1457261 & d & CG14572 &   90 &  86 &   660.2793 &  2.4496515 &  7.580482 &  71.02241\\\\\n",
       "\tCG1185261 & d & CG11852 &  121 & 154 &   865.6844 &  4.6218260 &  7.266659 &  70.91796\\\\\n",
       "\tCAH961 & d & CAH9    &   70 &  52 &   557.8493 &  1.9003399 &  7.587510 &  69.24601\\\\\n",
       "\tCG516261 & d & CG5162  &   48 &  16 &   345.1249 &  0.5076798 &  7.838644 &  66.12013\\\\\n",
       "\tCG981261 & d & CG9812  &  179 & 688 &  1206.2896 & 27.6027239 &  5.398280 &  55.26519\\\\\n",
       "\tCpr12A61 & d & Cpr12A  &   18 &   7 &   181.9422 &  0.3152279 &  7.112024 &  53.44860\\\\\n",
       "\tCAH261 & d & CAH2    &  119 & 904 &  1078.6168 & 31.2954136 &  5.061709 &  51.00332\\\\\n",
       "\tCG1456561 & d & CG14565 &   80 & 128 &   309.2638 &  3.5324118 &  6.092415 &  50.42906\\\\\n",
       "\tCG1719061 & d & CG17190 &   11 &  13 &   151.6676 &  0.2860525 &  6.881819 &  49.92243\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG1137061 | d | CG11370 | 2890 | 826 | 16311.3389 | 18.7725763 |  9.688159 | 135.57295 |\n",
       "| Nplp461 | d | Nplp4   |  494 |  28 |  3733.8508 |  0.9382501 | 10.911694 | 129.48727 |\n",
       "| CG1456661 | d | CG14566 |  407 |  62 |  4152.7877 |  2.2952246 | 10.299488 | 123.80202 |\n",
       "| CG1141361 | d | CG11413 |  660 |  87 |  3416.5312 |  3.5819705 |  9.542348 | 112.01514 |\n",
       "| CG801261 | d | CG8012  |  150 |  11 |  1085.4801 |  0.1985319 |  9.822849 |  99.06782 |\n",
       "| CG1306761 | d | CG13067 |  353 | 125 |  2263.7980 |  3.9548949 |  8.835675 |  98.47507 |\n",
       "| CG1367861 | d | CG13678 |  280 | 177 |  2273.6973 |  6.6786228 |  8.209977 |  91.55322 |\n",
       "| CG1304461 | d | CG13044 |  516 | 318 |  2717.6486 | 10.3132017 |  7.908208 |  90.22216 |\n",
       "| CG863061 | d | CG8630  |  381 | 300 |  2484.6331 |  9.0119569 |  7.955165 |  89.72947 |\n",
       "| CG1456961 | d | CG14569 |  230 |  81 |  1127.2510 |  2.6591611 |  8.267080 |  83.82714 |\n",
       "| Cpr49Ag61 | d | Cpr49Ag |  182 | 180 |  1543.2851 |  6.1895321 |  7.745891 |  82.05008 |\n",
       "| CG1457261 | d | CG14572 |   90 |  86 |   660.2793 |  2.4496515 |  7.580482 |  71.02241 |\n",
       "| CG1185261 | d | CG11852 |  121 | 154 |   865.6844 |  4.6218260 |  7.266659 |  70.91796 |\n",
       "| CAH961 | d | CAH9    |   70 |  52 |   557.8493 |  1.9003399 |  7.587510 |  69.24601 |\n",
       "| CG516261 | d | CG5162  |   48 |  16 |   345.1249 |  0.5076798 |  7.838644 |  66.12013 |\n",
       "| CG981261 | d | CG9812  |  179 | 688 |  1206.2896 | 27.6027239 |  5.398280 |  55.26519 |\n",
       "| Cpr12A61 | d | Cpr12A  |   18 |   7 |   181.9422 |  0.3152279 |  7.112024 |  53.44860 |\n",
       "| CAH261 | d | CAH2    |  119 | 904 |  1078.6168 | 31.2954136 |  5.061709 |  51.00332 |\n",
       "| CG1456561 | d | CG14565 |   80 | 128 |   309.2638 |  3.5324118 |  6.092415 |  50.42906 |\n",
       "| CG1719061 | d | CG17190 |   11 |  13 |   151.6676 |  0.2860525 |  6.881819 |  49.92243 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "CG1137061 d           CG11370 2890              826         16311.3389     \n",
       "Nplp461   d           Nplp4    494               28          3733.8508     \n",
       "CG1456661 d           CG14566  407               62          4152.7877     \n",
       "CG1141361 d           CG11413  660               87          3416.5312     \n",
       "CG801261  d           CG8012   150               11          1085.4801     \n",
       "CG1306761 d           CG13067  353              125          2263.7980     \n",
       "CG1367861 d           CG13678  280              177          2273.6973     \n",
       "CG1304461 d           CG13044  516              318          2717.6486     \n",
       "CG863061  d           CG8630   381              300          2484.6331     \n",
       "CG1456961 d           CG14569  230               81          1127.2510     \n",
       "Cpr49Ag61 d           Cpr49Ag  182              180          1543.2851     \n",
       "CG1457261 d           CG14572   90               86           660.2793     \n",
       "CG1185261 d           CG11852  121              154           865.6844     \n",
       "CAH961    d           CAH9      70               52           557.8493     \n",
       "CG516261  d           CG5162    48               16           345.1249     \n",
       "CG981261  d           CG9812   179              688          1206.2896     \n",
       "Cpr12A61  d           Cpr12A    18                7           181.9422     \n",
       "CAH261    d           CAH2     119              904          1078.6168     \n",
       "CG1456561 d           CG14565   80              128           309.2638     \n",
       "CG1719061 d           CG17190   11               13           151.6676     \n",
       "          other.tpm  log2.ratio heuristic.score\n",
       "CG1137061 18.7725763  9.688159  135.57295      \n",
       "Nplp461    0.9382501 10.911694  129.48727      \n",
       "CG1456661  2.2952246 10.299488  123.80202      \n",
       "CG1141361  3.5819705  9.542348  112.01514      \n",
       "CG801261   0.1985319  9.822849   99.06782      \n",
       "CG1306761  3.9548949  8.835675   98.47507      \n",
       "CG1367861  6.6786228  8.209977   91.55322      \n",
       "CG1304461 10.3132017  7.908208   90.22216      \n",
       "CG863061   9.0119569  7.955165   89.72947      \n",
       "CG1456961  2.6591611  8.267080   83.82714      \n",
       "Cpr49Ag61  6.1895321  7.745891   82.05008      \n",
       "CG1457261  2.4496515  7.580482   71.02241      \n",
       "CG1185261  4.6218260  7.266659   70.91796      \n",
       "CAH961     1.9003399  7.587510   69.24601      \n",
       "CG516261   0.5076798  7.838644   66.12013      \n",
       "CG981261  27.6027239  5.398280   55.26519      \n",
       "Cpr12A61   0.3152279  7.112024   53.44860      \n",
       "CAH261    31.2954136  5.061709   51.00332      \n",
       "CG1456561  3.5324118  6.092415   50.42906      \n",
       "CG1719061  0.2860525  6.881819   49.92243      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"d\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG11370\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG11370\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG11370\"), \n",
    "                   size_pos = 4, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Nplp4\"), \n",
    "                   size_pos = 3, size_neg = 0.1, stroke = 0.02, limit = 1)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG14566\"), \n",
    "                   size_pos = 3, size_neg = 0.2, stroke = 0.02, limit = 0.01)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"syt\"), \n",
    "                   size_pos = 4, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Nplp4\"), \n",
    "                   size_pos = 3, size_neg = 0.1, stroke = 0.02, limit = 1)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG14566\"), \n",
    "                   size_pos = 3, size_neg = 0.2, stroke = 0.02, limit = 0.01)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### e cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 790,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Multidendritic_sensory\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Multidendritic_sensory\", df_sub300$cluster %in% e &\n",
    "                        df_sub300$umap_2 < -1*df_sub300$umap_1 -6.7 & \n",
    "                          df_sub300$umap_2 > -1* df_sub300$umap_1 -7.2 & df_sub300$umap_1 > -1 )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 791,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type External_sensory\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"External_sensory\", df_sub300$cluster %in% e &\n",
    "                        df_sub300$umap_2 < -1*df_sub300$umap_1 -6.3 & \n",
    "                          df_sub300$umap_2 > -1* df_sub300$umap_1 -6.7 & df_sub300$umap_1 > 0 )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 792,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Chordotonal_sensory\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Chordotonal_sensory\", df_sub300$cluster %in% c(ao,e) &\n",
    "                        df_sub300$umap_2 > -1*df_sub300$umap_1 -6.1 & \n",
    "                           df_sub300$umap_1 > 0.3 )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'e' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'e' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(e)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = -1, intercept = -7.2) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(-4, 2) + ylim(-9, -3)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'e' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'e' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(e)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &\n",
    "              umap_2 < -1*umap_1 -6.7 & umap_2 > -1* umap_1 -7.2 & umap_1 > -1), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = -1, intercept = -7.2) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(-4, 2) + ylim(-9, -3)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ao' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ao' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ao, e)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = -1, intercept = -7.2) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "geom_abline(slope = -1., intercept = -6.3) +\n",
    "xlim(-4, 2) + ylim(-9, -3)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ao' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ao' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ao, e)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &\n",
    "              umap_2 > -1*umap_1 -6.1 & umap_1 > 0.3), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = -1, intercept = -7.2) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(-4, 2) + ylim(-9, -3)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 804,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>pHCl-162</th><td>e</td><td>pHCl-1   </td><td>2151</td><td>  815</td><td>1166.77111</td><td>  33.206456</td><td>5.092109</td><td>51.88626</td></tr>\n",
       "\t<tr><th scope=row>para62</th><td>e</td><td>para     </td><td>5159</td><td> 4049</td><td>2547.94885</td><td> 165.041233</td><td>3.939723</td><td>44.58067</td></tr>\n",
       "\t<tr><th scope=row>ppk392</th><td>e</td><td>ppk      </td><td> 376</td><td>   56</td><td> 139.05923</td><td>   2.071240</td><td>5.500734</td><td>39.21965</td></tr>\n",
       "\t<tr><th scope=row>Gfrl62</th><td>e</td><td>Gfrl     </td><td>1056</td><td>  837</td><td> 569.79301</td><td>  35.666195</td><td>3.957915</td><td>36.24193</td></tr>\n",
       "\t<tr><th scope=row>unc-13-4A62</th><td>e</td><td>unc-13-4A</td><td>1096</td><td>  930</td><td> 584.54631</td><td>  37.067356</td><td>3.940691</td><td>36.22929</td></tr>\n",
       "\t<tr><th scope=row>ppk2662</th><td>e</td><td>ppk26    </td><td> 256</td><td>   40</td><td>  92.57590</td><td>   1.082549</td><td>5.474214</td><td>35.84551</td></tr>\n",
       "\t<tr><th scope=row>cac62</th><td>e</td><td>cac      </td><td>6189</td><td>10432</td><td>3502.51877</td><td> 479.896483</td><td>2.864595</td><td>33.72942</td></tr>\n",
       "\t<tr><th scope=row>eys62</th><td>e</td><td>eys      </td><td>1023</td><td> 1070</td><td> 599.82083</td><td>  46.664328</td><td>3.653550</td><td>33.72515</td></tr>\n",
       "\t<tr><th scope=row>igl62</th><td>e</td><td>igl      </td><td> 542</td><td>  383</td><td> 269.91856</td><td>  14.058442</td><td>4.163880</td><td>33.65129</td></tr>\n",
       "\t<tr><th scope=row>klg62</th><td>e</td><td>klg      </td><td>3967</td><td> 6684</td><td>2379.96804</td><td> 299.714115</td><td>2.984478</td><td>33.47788</td></tr>\n",
       "\t<tr><th scope=row>dysc62</th><td>e</td><td>dysc     </td><td>2411</td><td> 3230</td><td>1254.66140</td><td> 137.139423</td><td>3.183101</td><td>32.76758</td></tr>\n",
       "\t<tr><th scope=row>CG3434262</th><td>e</td><td>CG34342  </td><td> 113</td><td>   38</td><td>  73.49032</td><td>   1.038493</td><td>5.171979</td><td>32.16444</td></tr>\n",
       "\t<tr><th scope=row>CG1201162</th><td>e</td><td>CG12011  </td><td> 376</td><td>35259</td><td> 206.35135</td><td>1620.507229</td><td>2.966296</td><td>31.62997</td></tr>\n",
       "\t<tr><th scope=row>Frq162</th><td>e</td><td>Frq1     </td><td>1369</td><td> 1988</td><td> 721.68570</td><td>  77.088828</td><td>3.208183</td><td>30.46883</td></tr>\n",
       "\t<tr><th scope=row>Tsp47F62</th><td>e</td><td>Tsp47F   </td><td> 156</td><td>   54</td><td>  83.30321</td><td>   2.175700</td><td>4.713226</td><td>30.15293</td></tr>\n",
       "\t<tr><th scope=row>dpr862</th><td>e</td><td>dpr8     </td><td>1665</td><td> 2628</td><td> 951.39367</td><td> 116.982473</td><td>3.011470</td><td>29.79974</td></tr>\n",
       "\t<tr><th scope=row>sosie62</th><td>e</td><td>sosie    </td><td> 202</td><td>   92</td><td>  81.14301</td><td>   2.232755</td><td>4.649631</td><td>29.57196</td></tr>\n",
       "\t<tr><th scope=row>ss62</th><td>e</td><td>ss       </td><td>1283</td><td> 2167</td><td> 780.32392</td><td>  95.400212</td><td>3.016965</td><td>28.99236</td></tr>\n",
       "\t<tr><th scope=row>GABA-B-R162</th><td>e</td><td>GABA-B-R1</td><td> 604</td><td>  705</td><td> 322.82879</td><td>  30.678733</td><td>3.349183</td><td>27.92913</td></tr>\n",
       "\t<tr><th scope=row>Sema2b62</th><td>e</td><td>Sema2b   </td><td>3039</td><td> 6357</td><td>1733.12792</td><td> 286.225915</td><td>2.593120</td><td>27.90196</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tpHCl-162 & e & pHCl-1    & 2151 &   815 & 1166.77111 &   33.206456 & 5.092109 & 51.88626\\\\\n",
       "\tpara62 & e & para      & 5159 &  4049 & 2547.94885 &  165.041233 & 3.939723 & 44.58067\\\\\n",
       "\tppk392 & e & ppk       &  376 &    56 &  139.05923 &    2.071240 & 5.500734 & 39.21965\\\\\n",
       "\tGfrl62 & e & Gfrl      & 1056 &   837 &  569.79301 &   35.666195 & 3.957915 & 36.24193\\\\\n",
       "\tunc-13-4A62 & e & unc-13-4A & 1096 &   930 &  584.54631 &   37.067356 & 3.940691 & 36.22929\\\\\n",
       "\tppk2662 & e & ppk26     &  256 &    40 &   92.57590 &    1.082549 & 5.474214 & 35.84551\\\\\n",
       "\tcac62 & e & cac       & 6189 & 10432 & 3502.51877 &  479.896483 & 2.864595 & 33.72942\\\\\n",
       "\teys62 & e & eys       & 1023 &  1070 &  599.82083 &   46.664328 & 3.653550 & 33.72515\\\\\n",
       "\tigl62 & e & igl       &  542 &   383 &  269.91856 &   14.058442 & 4.163880 & 33.65129\\\\\n",
       "\tklg62 & e & klg       & 3967 &  6684 & 2379.96804 &  299.714115 & 2.984478 & 33.47788\\\\\n",
       "\tdysc62 & e & dysc      & 2411 &  3230 & 1254.66140 &  137.139423 & 3.183101 & 32.76758\\\\\n",
       "\tCG3434262 & e & CG34342   &  113 &    38 &   73.49032 &    1.038493 & 5.171979 & 32.16444\\\\\n",
       "\tCG1201162 & e & CG12011   &  376 & 35259 &  206.35135 & 1620.507229 & 2.966296 & 31.62997\\\\\n",
       "\tFrq162 & e & Frq1      & 1369 &  1988 &  721.68570 &   77.088828 & 3.208183 & 30.46883\\\\\n",
       "\tTsp47F62 & e & Tsp47F    &  156 &    54 &   83.30321 &    2.175700 & 4.713226 & 30.15293\\\\\n",
       "\tdpr862 & e & dpr8      & 1665 &  2628 &  951.39367 &  116.982473 & 3.011470 & 29.79974\\\\\n",
       "\tsosie62 & e & sosie     &  202 &    92 &   81.14301 &    2.232755 & 4.649631 & 29.57196\\\\\n",
       "\tss62 & e & ss        & 1283 &  2167 &  780.32392 &   95.400212 & 3.016965 & 28.99236\\\\\n",
       "\tGABA-B-R162 & e & GABA-B-R1 &  604 &   705 &  322.82879 &   30.678733 & 3.349183 & 27.92913\\\\\n",
       "\tSema2b62 & e & Sema2b    & 3039 &  6357 & 1733.12792 &  286.225915 & 2.593120 & 27.90196\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| pHCl-162 | e | pHCl-1    | 2151 |   815 | 1166.77111 |   33.206456 | 5.092109 | 51.88626 |\n",
       "| para62 | e | para      | 5159 |  4049 | 2547.94885 |  165.041233 | 3.939723 | 44.58067 |\n",
       "| ppk392 | e | ppk       |  376 |    56 |  139.05923 |    2.071240 | 5.500734 | 39.21965 |\n",
       "| Gfrl62 | e | Gfrl      | 1056 |   837 |  569.79301 |   35.666195 | 3.957915 | 36.24193 |\n",
       "| unc-13-4A62 | e | unc-13-4A | 1096 |   930 |  584.54631 |   37.067356 | 3.940691 | 36.22929 |\n",
       "| ppk2662 | e | ppk26     |  256 |    40 |   92.57590 |    1.082549 | 5.474214 | 35.84551 |\n",
       "| cac62 | e | cac       | 6189 | 10432 | 3502.51877 |  479.896483 | 2.864595 | 33.72942 |\n",
       "| eys62 | e | eys       | 1023 |  1070 |  599.82083 |   46.664328 | 3.653550 | 33.72515 |\n",
       "| igl62 | e | igl       |  542 |   383 |  269.91856 |   14.058442 | 4.163880 | 33.65129 |\n",
       "| klg62 | e | klg       | 3967 |  6684 | 2379.96804 |  299.714115 | 2.984478 | 33.47788 |\n",
       "| dysc62 | e | dysc      | 2411 |  3230 | 1254.66140 |  137.139423 | 3.183101 | 32.76758 |\n",
       "| CG3434262 | e | CG34342   |  113 |    38 |   73.49032 |    1.038493 | 5.171979 | 32.16444 |\n",
       "| CG1201162 | e | CG12011   |  376 | 35259 |  206.35135 | 1620.507229 | 2.966296 | 31.62997 |\n",
       "| Frq162 | e | Frq1      | 1369 |  1988 |  721.68570 |   77.088828 | 3.208183 | 30.46883 |\n",
       "| Tsp47F62 | e | Tsp47F    |  156 |    54 |   83.30321 |    2.175700 | 4.713226 | 30.15293 |\n",
       "| dpr862 | e | dpr8      | 1665 |  2628 |  951.39367 |  116.982473 | 3.011470 | 29.79974 |\n",
       "| sosie62 | e | sosie     |  202 |    92 |   81.14301 |    2.232755 | 4.649631 | 29.57196 |\n",
       "| ss62 | e | ss        | 1283 |  2167 |  780.32392 |   95.400212 | 3.016965 | 28.99236 |\n",
       "| GABA-B-R162 | e | GABA-B-R1 |  604 |   705 |  322.82879 |   30.678733 | 3.349183 | 27.92913 |\n",
       "| Sema2b62 | e | Sema2b    | 3039 |  6357 | 1733.12792 |  286.225915 | 2.593120 | 27.90196 |\n",
       "\n"
      ],
      "text/plain": [
       "            cell.type.2 gene      cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "pHCl-162    e           pHCl-1    2151                815       1166.77111     \n",
       "para62      e           para      5159               4049       2547.94885     \n",
       "ppk392      e           ppk        376                 56        139.05923     \n",
       "Gfrl62      e           Gfrl      1056                837        569.79301     \n",
       "unc-13-4A62 e           unc-13-4A 1096                930        584.54631     \n",
       "ppk2662     e           ppk26      256                 40         92.57590     \n",
       "cac62       e           cac       6189              10432       3502.51877     \n",
       "eys62       e           eys       1023               1070        599.82083     \n",
       "igl62       e           igl        542                383        269.91856     \n",
       "klg62       e           klg       3967               6684       2379.96804     \n",
       "dysc62      e           dysc      2411               3230       1254.66140     \n",
       "CG3434262   e           CG34342    113                 38         73.49032     \n",
       "CG1201162   e           CG12011    376              35259        206.35135     \n",
       "Frq162      e           Frq1      1369               1988        721.68570     \n",
       "Tsp47F62    e           Tsp47F     156                 54         83.30321     \n",
       "dpr862      e           dpr8      1665               2628        951.39367     \n",
       "sosie62     e           sosie      202                 92         81.14301     \n",
       "ss62        e           ss        1283               2167        780.32392     \n",
       "GABA-B-R162 e           GABA-B-R1  604                705        322.82879     \n",
       "Sema2b62    e           Sema2b    3039               6357       1733.12792     \n",
       "            other.tpm   log2.ratio heuristic.score\n",
       "pHCl-162      33.206456 5.092109   51.88626       \n",
       "para62       165.041233 3.939723   44.58067       \n",
       "ppk392         2.071240 5.500734   39.21965       \n",
       "Gfrl62        35.666195 3.957915   36.24193       \n",
       "unc-13-4A62   37.067356 3.940691   36.22929       \n",
       "ppk2662        1.082549 5.474214   35.84551       \n",
       "cac62        479.896483 2.864595   33.72942       \n",
       "eys62         46.664328 3.653550   33.72515       \n",
       "igl62         14.058442 4.163880   33.65129       \n",
       "klg62        299.714115 2.984478   33.47788       \n",
       "dysc62       137.139423 3.183101   32.76758       \n",
       "CG3434262      1.038493 5.171979   32.16444       \n",
       "CG1201162   1620.507229 2.966296   31.62997       \n",
       "Frq162        77.088828 3.208183   30.46883       \n",
       "Tsp47F62       2.175700 4.713226   30.15293       \n",
       "dpr862       116.982473 3.011470   29.79974       \n",
       "sosie62        2.232755 4.649631   29.57196       \n",
       "ss62          95.400212 3.016965   28.99236       \n",
       "GABA-B-R162   30.678733 3.349183   27.92913       \n",
       "Sema2b62     286.225915 2.593120   27.90196       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"e\")  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"pHCl-1\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"pHCl-1\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"pHCl-1\"), \n",
    "                   size_pos = 4, size_neg = 0.05, stroke = 0.008, limit = 3)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"para\"), \n",
    "                   size_pos = 2, size_neg = 0.1, stroke = 0.02, limit = 3)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"ppk\"), \n",
    "                   size_pos = 3, size_neg = 0.2, stroke = 0.02, limit = 0.01)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 812,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type unknown_2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"unknown_2\", df_sub300$cluster %in% f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'f' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'f' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(f)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 802,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG746563</th><td>f</td><td>CG7465  </td><td>20683</td><td> 922</td><td>18938.2318</td><td> 27.8101498</td><td> 9.360509</td><td>133.00431</td></tr>\n",
       "\t<tr><th scope=row>CG3260363</th><td>f</td><td>CG32603 </td><td> 9367</td><td> 299</td><td> 9354.6783</td><td>  8.3255405</td><td> 9.970285</td><td>131.52428</td></tr>\n",
       "\t<tr><th scope=row>CG1135063</th><td>f</td><td>CG11350 </td><td> 7226</td><td> 169</td><td> 6348.3690</td><td>  4.8244082</td><td>10.090059</td><td>127.46163</td></tr>\n",
       "\t<tr><th scope=row>CG1442163</th><td>f</td><td>CG14421 </td><td> 3808</td><td>  49</td><td> 3552.7510</td><td>  1.5882504</td><td>10.422744</td><td>122.93758</td></tr>\n",
       "\t<tr><th scope=row>CG1442463</th><td>f</td><td>CG14424 </td><td> 3717</td><td>  73</td><td> 3555.1975</td><td>  2.4802559</td><td> 9.996521</td><td>117.92015</td></tr>\n",
       "\t<tr><th scope=row>CG3224166</th><td>f</td><td>CG32241 </td><td> 4339</td><td> 150</td><td> 3941.1991</td><td>  3.9336622</td><td> 9.641760</td><td>115.16875</td></tr>\n",
       "\t<tr><th scope=row>CG729963</th><td>f</td><td>CG7299  </td><td> 9476</td><td> 705</td><td> 8369.3150</td><td> 24.0240067</td><td> 8.385653</td><td>109.27400</td></tr>\n",
       "\t<tr><th scope=row>CG1442263</th><td>f</td><td>CG14422 </td><td> 1973</td><td>  23</td><td> 1519.9472</td><td>  0.7092652</td><td> 9.796429</td><td>103.55565</td></tr>\n",
       "\t<tr><th scope=row>CG3045763</th><td>f</td><td>CG30457 </td><td> 8344</td><td> 834</td><td> 7163.4002</td><td> 26.3604659</td><td> 8.032408</td><td>102.86808</td></tr>\n",
       "\t<tr><th scope=row>CG354663</th><td>f</td><td>CG3546  </td><td> 1744</td><td>  50</td><td> 1688.9212</td><td>  1.6200228</td><td> 9.332307</td><td>100.06790</td></tr>\n",
       "\t<tr><th scope=row>CG3279363</th><td>f</td><td>CG32793 </td><td> 1975</td><td>  69</td><td> 2015.4409</td><td>  2.9268640</td><td> 9.003502</td><td> 98.83680</td></tr>\n",
       "\t<tr><th scope=row>CG4366663</th><td>f</td><td>CG43666 </td><td> 1624</td><td>  42</td><td> 1517.2581</td><td>  1.5830946</td><td> 9.198150</td><td> 97.20791</td></tr>\n",
       "\t<tr><th scope=row>Lcp65Ag363</th><td>f</td><td>Lcp65Ag3</td><td> 5841</td><td> 648</td><td> 4576.5318</td><td> 19.4396589</td><td> 7.806740</td><td> 94.93272</td></tr>\n",
       "\t<tr><th scope=row>CG1370563</th><td>f</td><td>CG13705 </td><td> 4173</td><td> 476</td><td> 3982.0456</td><td> 17.5951122</td><td> 7.742442</td><td> 92.59695</td></tr>\n",
       "\t<tr><th scope=row>CG3420264</th><td>f</td><td>CG34202 </td><td> 1371</td><td>  53</td><td> 1332.2848</td><td>  1.7758037</td><td> 8.906781</td><td> 92.45924</td></tr>\n",
       "\t<tr><th scope=row>CG1248063</th><td>f</td><td>CG12480 </td><td> 2278</td><td> 171</td><td> 2040.6156</td><td>  5.8945495</td><td> 8.209332</td><td> 90.26568</td></tr>\n",
       "\t<tr><th scope=row>CG1522563</th><td>f</td><td>CG15225 </td><td>  985</td><td>  34</td><td>  994.2739</td><td>  1.1052924</td><td> 8.883479</td><td> 88.47012</td></tr>\n",
       "\t<tr><th scope=row>Mur89F63</th><td>f</td><td>Mur89F  </td><td>  859</td><td>  96</td><td> 1166.2783</td><td>  2.1089253</td><td> 8.551280</td><td> 87.12842</td></tr>\n",
       "\t<tr><th scope=row>CG136874</th><td>f</td><td>CG1368  </td><td> 6688</td><td>1465</td><td> 6170.7766</td><td> 52.7552770</td><td> 6.842902</td><td> 86.16219</td></tr>\n",
       "\t<tr><th scope=row>CG1373163</th><td>f</td><td>CG13731 </td><td>10121</td><td>3097</td><td> 9352.3157</td><td>104.7186247</td><td> 6.467022</td><td> 85.30818</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG746563 & f & CG7465   & 20683 &  922 & 18938.2318 &  27.8101498 &  9.360509 & 133.00431\\\\\n",
       "\tCG3260363 & f & CG32603  &  9367 &  299 &  9354.6783 &   8.3255405 &  9.970285 & 131.52428\\\\\n",
       "\tCG1135063 & f & CG11350  &  7226 &  169 &  6348.3690 &   4.8244082 & 10.090059 & 127.46163\\\\\n",
       "\tCG1442163 & f & CG14421  &  3808 &   49 &  3552.7510 &   1.5882504 & 10.422744 & 122.93758\\\\\n",
       "\tCG1442463 & f & CG14424  &  3717 &   73 &  3555.1975 &   2.4802559 &  9.996521 & 117.92015\\\\\n",
       "\tCG3224166 & f & CG32241  &  4339 &  150 &  3941.1991 &   3.9336622 &  9.641760 & 115.16875\\\\\n",
       "\tCG729963 & f & CG7299   &  9476 &  705 &  8369.3150 &  24.0240067 &  8.385653 & 109.27400\\\\\n",
       "\tCG1442263 & f & CG14422  &  1973 &   23 &  1519.9472 &   0.7092652 &  9.796429 & 103.55565\\\\\n",
       "\tCG3045763 & f & CG30457  &  8344 &  834 &  7163.4002 &  26.3604659 &  8.032408 & 102.86808\\\\\n",
       "\tCG354663 & f & CG3546   &  1744 &   50 &  1688.9212 &   1.6200228 &  9.332307 & 100.06790\\\\\n",
       "\tCG3279363 & f & CG32793  &  1975 &   69 &  2015.4409 &   2.9268640 &  9.003502 &  98.83680\\\\\n",
       "\tCG4366663 & f & CG43666  &  1624 &   42 &  1517.2581 &   1.5830946 &  9.198150 &  97.20791\\\\\n",
       "\tLcp65Ag363 & f & Lcp65Ag3 &  5841 &  648 &  4576.5318 &  19.4396589 &  7.806740 &  94.93272\\\\\n",
       "\tCG1370563 & f & CG13705  &  4173 &  476 &  3982.0456 &  17.5951122 &  7.742442 &  92.59695\\\\\n",
       "\tCG3420264 & f & CG34202  &  1371 &   53 &  1332.2848 &   1.7758037 &  8.906781 &  92.45924\\\\\n",
       "\tCG1248063 & f & CG12480  &  2278 &  171 &  2040.6156 &   5.8945495 &  8.209332 &  90.26568\\\\\n",
       "\tCG1522563 & f & CG15225  &   985 &   34 &   994.2739 &   1.1052924 &  8.883479 &  88.47012\\\\\n",
       "\tMur89F63 & f & Mur89F   &   859 &   96 &  1166.2783 &   2.1089253 &  8.551280 &  87.12842\\\\\n",
       "\tCG136874 & f & CG1368   &  6688 & 1465 &  6170.7766 &  52.7552770 &  6.842902 &  86.16219\\\\\n",
       "\tCG1373163 & f & CG13731  & 10121 & 3097 &  9352.3157 & 104.7186247 &  6.467022 &  85.30818\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG746563 | f | CG7465   | 20683 |  922 | 18938.2318 |  27.8101498 |  9.360509 | 133.00431 |\n",
       "| CG3260363 | f | CG32603  |  9367 |  299 |  9354.6783 |   8.3255405 |  9.970285 | 131.52428 |\n",
       "| CG1135063 | f | CG11350  |  7226 |  169 |  6348.3690 |   4.8244082 | 10.090059 | 127.46163 |\n",
       "| CG1442163 | f | CG14421  |  3808 |   49 |  3552.7510 |   1.5882504 | 10.422744 | 122.93758 |\n",
       "| CG1442463 | f | CG14424  |  3717 |   73 |  3555.1975 |   2.4802559 |  9.996521 | 117.92015 |\n",
       "| CG3224166 | f | CG32241  |  4339 |  150 |  3941.1991 |   3.9336622 |  9.641760 | 115.16875 |\n",
       "| CG729963 | f | CG7299   |  9476 |  705 |  8369.3150 |  24.0240067 |  8.385653 | 109.27400 |\n",
       "| CG1442263 | f | CG14422  |  1973 |   23 |  1519.9472 |   0.7092652 |  9.796429 | 103.55565 |\n",
       "| CG3045763 | f | CG30457  |  8344 |  834 |  7163.4002 |  26.3604659 |  8.032408 | 102.86808 |\n",
       "| CG354663 | f | CG3546   |  1744 |   50 |  1688.9212 |   1.6200228 |  9.332307 | 100.06790 |\n",
       "| CG3279363 | f | CG32793  |  1975 |   69 |  2015.4409 |   2.9268640 |  9.003502 |  98.83680 |\n",
       "| CG4366663 | f | CG43666  |  1624 |   42 |  1517.2581 |   1.5830946 |  9.198150 |  97.20791 |\n",
       "| Lcp65Ag363 | f | Lcp65Ag3 |  5841 |  648 |  4576.5318 |  19.4396589 |  7.806740 |  94.93272 |\n",
       "| CG1370563 | f | CG13705  |  4173 |  476 |  3982.0456 |  17.5951122 |  7.742442 |  92.59695 |\n",
       "| CG3420264 | f | CG34202  |  1371 |   53 |  1332.2848 |   1.7758037 |  8.906781 |  92.45924 |\n",
       "| CG1248063 | f | CG12480  |  2278 |  171 |  2040.6156 |   5.8945495 |  8.209332 |  90.26568 |\n",
       "| CG1522563 | f | CG15225  |   985 |   34 |   994.2739 |   1.1052924 |  8.883479 |  88.47012 |\n",
       "| Mur89F63 | f | Mur89F   |   859 |   96 |  1166.2783 |   2.1089253 |  8.551280 |  87.12842 |\n",
       "| CG136874 | f | CG1368   |  6688 | 1465 |  6170.7766 |  52.7552770 |  6.842902 |  86.16219 |\n",
       "| CG1373163 | f | CG13731  | 10121 | 3097 |  9352.3157 | 104.7186247 |  6.467022 |  85.30818 |\n",
       "\n"
      ],
      "text/plain": [
       "           cell.type.2 gene     cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "CG746563   f           CG7465   20683              922        18938.2318     \n",
       "CG3260363  f           CG32603   9367              299         9354.6783     \n",
       "CG1135063  f           CG11350   7226              169         6348.3690     \n",
       "CG1442163  f           CG14421   3808               49         3552.7510     \n",
       "CG1442463  f           CG14424   3717               73         3555.1975     \n",
       "CG3224166  f           CG32241   4339              150         3941.1991     \n",
       "CG729963   f           CG7299    9476              705         8369.3150     \n",
       "CG1442263  f           CG14422   1973               23         1519.9472     \n",
       "CG3045763  f           CG30457   8344              834         7163.4002     \n",
       "CG354663   f           CG3546    1744               50         1688.9212     \n",
       "CG3279363  f           CG32793   1975               69         2015.4409     \n",
       "CG4366663  f           CG43666   1624               42         1517.2581     \n",
       "Lcp65Ag363 f           Lcp65Ag3  5841              648         4576.5318     \n",
       "CG1370563  f           CG13705   4173              476         3982.0456     \n",
       "CG3420264  f           CG34202   1371               53         1332.2848     \n",
       "CG1248063  f           CG12480   2278              171         2040.6156     \n",
       "CG1522563  f           CG15225    985               34          994.2739     \n",
       "Mur89F63   f           Mur89F     859               96         1166.2783     \n",
       "CG136874   f           CG1368    6688             1465         6170.7766     \n",
       "CG1373163  f           CG13731  10121             3097         9352.3157     \n",
       "           other.tpm   log2.ratio heuristic.score\n",
       "CG746563    27.8101498  9.360509  133.00431      \n",
       "CG3260363    8.3255405  9.970285  131.52428      \n",
       "CG1135063    4.8244082 10.090059  127.46163      \n",
       "CG1442163    1.5882504 10.422744  122.93758      \n",
       "CG1442463    2.4802559  9.996521  117.92015      \n",
       "CG3224166    3.9336622  9.641760  115.16875      \n",
       "CG729963    24.0240067  8.385653  109.27400      \n",
       "CG1442263    0.7092652  9.796429  103.55565      \n",
       "CG3045763   26.3604659  8.032408  102.86808      \n",
       "CG354663     1.6200228  9.332307  100.06790      \n",
       "CG3279363    2.9268640  9.003502   98.83680      \n",
       "CG4366663    1.5830946  9.198150   97.20791      \n",
       "Lcp65Ag363  19.4396589  7.806740   94.93272      \n",
       "CG1370563   17.5951122  7.742442   92.59695      \n",
       "CG3420264    1.7758037  8.906781   92.45924      \n",
       "CG1248063    5.8945495  8.209332   90.26568      \n",
       "CG1522563    1.1052924  8.883479   88.47012      \n",
       "Mur89F63     2.1089253  8.551280   87.12842      \n",
       "CG136874    52.7552770  6.842902   86.16219      \n",
       "CG1373163  104.7186247  6.467022   85.30818      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"f\")  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG7465\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG7465\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG7465\"), \n",
    "                   size_pos = 2, size_neg = 0.5, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG32603\"), \n",
    "                   size_pos = 2, size_neg = 0.1, stroke = 0.02, limit = 3)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG11350\"), \n",
    "                   size_pos = 3, size_neg = 0.2, stroke = 0.02, limit = 1)\n",
    "## all label both clusters -- throughout.  \n",
    "## CG11350 and CG32603 \"Predicted to be part of RNA polymerase II transcription regulator complex.\"\n",
    "## shares markers with epidermis (i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 817,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type gut_query\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"gut_query\", df_sub300$cluster %in% g)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'g' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'g' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(g)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 814,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Jon99Cii64</th><td>g</td><td>Jon99Cii </td><td>6060</td><td>300</td><td>7851.6008</td><td> 9.6160613</td><td> 9.530594</td><td>123.31593</td></tr>\n",
       "\t<tr><th scope=row>Jon99Ciii64</th><td>g</td><td>Jon99Ciii</td><td>4035</td><td>217</td><td>5562.8124</td><td> 7.4670202</td><td> 9.359744</td><td>116.45261</td></tr>\n",
       "\t<tr><th scope=row>CG1818064</th><td>g</td><td>CG18180  </td><td>3716</td><td>103</td><td>3973.0577</td><td> 3.8846047</td><td> 9.667792</td><td>115.59196</td></tr>\n",
       "\t<tr><th scope=row>CG729064</th><td>g</td><td>CG7290   </td><td>1400</td><td> 64</td><td>2807.3954</td><td> 1.7413531</td><td>10.000128</td><td>114.55677</td></tr>\n",
       "\t<tr><th scope=row>Jon99Fii64</th><td>g</td><td>Jon99Fii </td><td>2876</td><td>109</td><td>3492.1304</td><td> 3.7020013</td><td> 9.536617</td><td>112.24889</td></tr>\n",
       "\t<tr><th scope=row>CG1091064</th><td>g</td><td>CG10910  </td><td> 766</td><td> 22</td><td>1898.3644</td><td> 0.6184414</td><td>10.195936</td><td>111.04701</td></tr>\n",
       "\t<tr><th scope=row>CG968264</th><td>g</td><td>CG9682   </td><td> 600</td><td> 39</td><td>2150.8132</td><td> 1.3179139</td><td> 9.857840</td><td>109.13947</td></tr>\n",
       "\t<tr><th scope=row>Jon65Aiv64</th><td>g</td><td>Jon65Aiv </td><td>1764</td><td> 99</td><td>2960.9984</td><td> 3.4616372</td><td> 9.374295</td><td>108.10770</td></tr>\n",
       "\t<tr><th scope=row>Jon25Biii64</th><td>g</td><td>Jon25Biii</td><td>2186</td><td>117</td><td>3357.9222</td><td> 4.8876112</td><td> 9.155671</td><td>107.24754</td></tr>\n",
       "\t<tr><th scope=row>Jon65Ai64</th><td>g</td><td>Jon65Ai  </td><td>1660</td><td> 43</td><td>1993.1683</td><td> 1.2900127</td><td> 9.765492</td><td>107.04514</td></tr>\n",
       "\t<tr><th scope=row>Jon99Fi64</th><td>g</td><td>Jon99Fi  </td><td>1609</td><td>110</td><td>2686.3841</td><td> 3.6408327</td><td> 9.177066</td><td>104.54502</td></tr>\n",
       "\t<tr><th scope=row>CG1763364</th><td>g</td><td>CG17633  </td><td>2053</td><td> 50</td><td>1901.6767</td><td> 1.8719914</td><td> 9.371005</td><td>102.08599</td></tr>\n",
       "\t<tr><th scope=row>yip764</th><td>g</td><td>yip7     </td><td> 843</td><td> 30</td><td>1285.6859</td><td> 0.7131014</td><td> 9.551712</td><td> 98.66388</td></tr>\n",
       "\t<tr><th scope=row>Jon65Aiii64</th><td>g</td><td>Jon65Aiii</td><td>1125</td><td> 62</td><td>1814.7886</td><td> 2.6601121</td><td> 8.953698</td><td> 96.93614</td></tr>\n",
       "\t<tr><th scope=row>CG856064</th><td>g</td><td>CG8560   </td><td>2354</td><td>199</td><td>2525.7773</td><td> 8.6935119</td><td> 8.025492</td><td> 90.71280</td></tr>\n",
       "\t<tr><th scope=row>CG729864</th><td>g</td><td>CG7298   </td><td>1099</td><td>278</td><td>2833.1884</td><td>10.8267283</td><td> 7.904232</td><td> 90.65142</td></tr>\n",
       "\t<tr><th scope=row>CG1091164</th><td>g</td><td>CG10911  </td><td> 444</td><td> 27</td><td> 954.2742</td><td> 0.9365480</td><td> 8.944773</td><td> 88.55120</td></tr>\n",
       "\t<tr><th scope=row>CG3312864</th><td>g</td><td>CG33128  </td><td> 675</td><td>175</td><td>2144.3972</td><td> 7.4882799</td><td> 7.980884</td><td> 88.32468</td></tr>\n",
       "\t<tr><th scope=row>Jon25Bi64</th><td>g</td><td>Jon25Bi  </td><td> 505</td><td> 57</td><td>1158.1755</td><td> 2.3216346</td><td> 8.445745</td><td> 85.96825</td></tr>\n",
       "\t<tr><th scope=row>Muc18B64</th><td>g</td><td>Muc18B   </td><td> 709</td><td> 47</td><td> 950.1543</td><td> 1.5948864</td><td> 8.516347</td><td> 84.25678</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tJon99Cii64 & g & Jon99Cii  & 6060 & 300 & 7851.6008 &  9.6160613 &  9.530594 & 123.31593\\\\\n",
       "\tJon99Ciii64 & g & Jon99Ciii & 4035 & 217 & 5562.8124 &  7.4670202 &  9.359744 & 116.45261\\\\\n",
       "\tCG1818064 & g & CG18180   & 3716 & 103 & 3973.0577 &  3.8846047 &  9.667792 & 115.59196\\\\\n",
       "\tCG729064 & g & CG7290    & 1400 &  64 & 2807.3954 &  1.7413531 & 10.000128 & 114.55677\\\\\n",
       "\tJon99Fii64 & g & Jon99Fii  & 2876 & 109 & 3492.1304 &  3.7020013 &  9.536617 & 112.24889\\\\\n",
       "\tCG1091064 & g & CG10910   &  766 &  22 & 1898.3644 &  0.6184414 & 10.195936 & 111.04701\\\\\n",
       "\tCG968264 & g & CG9682    &  600 &  39 & 2150.8132 &  1.3179139 &  9.857840 & 109.13947\\\\\n",
       "\tJon65Aiv64 & g & Jon65Aiv  & 1764 &  99 & 2960.9984 &  3.4616372 &  9.374295 & 108.10770\\\\\n",
       "\tJon25Biii64 & g & Jon25Biii & 2186 & 117 & 3357.9222 &  4.8876112 &  9.155671 & 107.24754\\\\\n",
       "\tJon65Ai64 & g & Jon65Ai   & 1660 &  43 & 1993.1683 &  1.2900127 &  9.765492 & 107.04514\\\\\n",
       "\tJon99Fi64 & g & Jon99Fi   & 1609 & 110 & 2686.3841 &  3.6408327 &  9.177066 & 104.54502\\\\\n",
       "\tCG1763364 & g & CG17633   & 2053 &  50 & 1901.6767 &  1.8719914 &  9.371005 & 102.08599\\\\\n",
       "\tyip764 & g & yip7      &  843 &  30 & 1285.6859 &  0.7131014 &  9.551712 &  98.66388\\\\\n",
       "\tJon65Aiii64 & g & Jon65Aiii & 1125 &  62 & 1814.7886 &  2.6601121 &  8.953698 &  96.93614\\\\\n",
       "\tCG856064 & g & CG8560    & 2354 & 199 & 2525.7773 &  8.6935119 &  8.025492 &  90.71280\\\\\n",
       "\tCG729864 & g & CG7298    & 1099 & 278 & 2833.1884 & 10.8267283 &  7.904232 &  90.65142\\\\\n",
       "\tCG1091164 & g & CG10911   &  444 &  27 &  954.2742 &  0.9365480 &  8.944773 &  88.55120\\\\\n",
       "\tCG3312864 & g & CG33128   &  675 & 175 & 2144.3972 &  7.4882799 &  7.980884 &  88.32468\\\\\n",
       "\tJon25Bi64 & g & Jon25Bi   &  505 &  57 & 1158.1755 &  2.3216346 &  8.445745 &  85.96825\\\\\n",
       "\tMuc18B64 & g & Muc18B    &  709 &  47 &  950.1543 &  1.5948864 &  8.516347 &  84.25678\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Jon99Cii64 | g | Jon99Cii  | 6060 | 300 | 7851.6008 |  9.6160613 |  9.530594 | 123.31593 |\n",
       "| Jon99Ciii64 | g | Jon99Ciii | 4035 | 217 | 5562.8124 |  7.4670202 |  9.359744 | 116.45261 |\n",
       "| CG1818064 | g | CG18180   | 3716 | 103 | 3973.0577 |  3.8846047 |  9.667792 | 115.59196 |\n",
       "| CG729064 | g | CG7290    | 1400 |  64 | 2807.3954 |  1.7413531 | 10.000128 | 114.55677 |\n",
       "| Jon99Fii64 | g | Jon99Fii  | 2876 | 109 | 3492.1304 |  3.7020013 |  9.536617 | 112.24889 |\n",
       "| CG1091064 | g | CG10910   |  766 |  22 | 1898.3644 |  0.6184414 | 10.195936 | 111.04701 |\n",
       "| CG968264 | g | CG9682    |  600 |  39 | 2150.8132 |  1.3179139 |  9.857840 | 109.13947 |\n",
       "| Jon65Aiv64 | g | Jon65Aiv  | 1764 |  99 | 2960.9984 |  3.4616372 |  9.374295 | 108.10770 |\n",
       "| Jon25Biii64 | g | Jon25Biii | 2186 | 117 | 3357.9222 |  4.8876112 |  9.155671 | 107.24754 |\n",
       "| Jon65Ai64 | g | Jon65Ai   | 1660 |  43 | 1993.1683 |  1.2900127 |  9.765492 | 107.04514 |\n",
       "| Jon99Fi64 | g | Jon99Fi   | 1609 | 110 | 2686.3841 |  3.6408327 |  9.177066 | 104.54502 |\n",
       "| CG1763364 | g | CG17633   | 2053 |  50 | 1901.6767 |  1.8719914 |  9.371005 | 102.08599 |\n",
       "| yip764 | g | yip7      |  843 |  30 | 1285.6859 |  0.7131014 |  9.551712 |  98.66388 |\n",
       "| Jon65Aiii64 | g | Jon65Aiii | 1125 |  62 | 1814.7886 |  2.6601121 |  8.953698 |  96.93614 |\n",
       "| CG856064 | g | CG8560    | 2354 | 199 | 2525.7773 |  8.6935119 |  8.025492 |  90.71280 |\n",
       "| CG729864 | g | CG7298    | 1099 | 278 | 2833.1884 | 10.8267283 |  7.904232 |  90.65142 |\n",
       "| CG1091164 | g | CG10911   |  444 |  27 |  954.2742 |  0.9365480 |  8.944773 |  88.55120 |\n",
       "| CG3312864 | g | CG33128   |  675 | 175 | 2144.3972 |  7.4882799 |  7.980884 |  88.32468 |\n",
       "| Jon25Bi64 | g | Jon25Bi   |  505 |  57 | 1158.1755 |  2.3216346 |  8.445745 |  85.96825 |\n",
       "| Muc18B64 | g | Muc18B    |  709 |  47 |  950.1543 |  1.5948864 |  8.516347 |  84.25678 |\n",
       "\n"
      ],
      "text/plain": [
       "            cell.type.2 gene      cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "Jon99Cii64  g           Jon99Cii  6060              300         7851.6008      \n",
       "Jon99Ciii64 g           Jon99Ciii 4035              217         5562.8124      \n",
       "CG1818064   g           CG18180   3716              103         3973.0577      \n",
       "CG729064    g           CG7290    1400               64         2807.3954      \n",
       "Jon99Fii64  g           Jon99Fii  2876              109         3492.1304      \n",
       "CG1091064   g           CG10910    766               22         1898.3644      \n",
       "CG968264    g           CG9682     600               39         2150.8132      \n",
       "Jon65Aiv64  g           Jon65Aiv  1764               99         2960.9984      \n",
       "Jon25Biii64 g           Jon25Biii 2186              117         3357.9222      \n",
       "Jon65Ai64   g           Jon65Ai   1660               43         1993.1683      \n",
       "Jon99Fi64   g           Jon99Fi   1609              110         2686.3841      \n",
       "CG1763364   g           CG17633   2053               50         1901.6767      \n",
       "yip764      g           yip7       843               30         1285.6859      \n",
       "Jon65Aiii64 g           Jon65Aiii 1125               62         1814.7886      \n",
       "CG856064    g           CG8560    2354              199         2525.7773      \n",
       "CG729864    g           CG7298    1099              278         2833.1884      \n",
       "CG1091164   g           CG10911    444               27          954.2742      \n",
       "CG3312864   g           CG33128    675              175         2144.3972      \n",
       "Jon25Bi64   g           Jon25Bi    505               57         1158.1755      \n",
       "Muc18B64    g           Muc18B     709               47          950.1543      \n",
       "            other.tpm  log2.ratio heuristic.score\n",
       "Jon99Cii64   9.6160613  9.530594  123.31593      \n",
       "Jon99Ciii64  7.4670202  9.359744  116.45261      \n",
       "CG1818064    3.8846047  9.667792  115.59196      \n",
       "CG729064     1.7413531 10.000128  114.55677      \n",
       "Jon99Fii64   3.7020013  9.536617  112.24889      \n",
       "CG1091064    0.6184414 10.195936  111.04701      \n",
       "CG968264     1.3179139  9.857840  109.13947      \n",
       "Jon65Aiv64   3.4616372  9.374295  108.10770      \n",
       "Jon25Biii64  4.8876112  9.155671  107.24754      \n",
       "Jon65Ai64    1.2900127  9.765492  107.04514      \n",
       "Jon99Fi64    3.6408327  9.177066  104.54502      \n",
       "CG1763364    1.8719914  9.371005  102.08599      \n",
       "yip764       0.7131014  9.551712   98.66388      \n",
       "Jon65Aiii64  2.6601121  8.953698   96.93614      \n",
       "CG856064     8.6935119  8.025492   90.71280      \n",
       "CG729864    10.8267283  7.904232   90.65142      \n",
       "CG1091164    0.9365480  8.944773   88.55120      \n",
       "CG3312864    7.4882799  7.980884   88.32468      \n",
       "Jon25Bi64    2.3216346  8.445745   85.96825      \n",
       "Muc18B64     1.5948864  8.516347   84.25678      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"g\")  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Jon99Cii\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Jon99Cii\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Jon99Cii\"), \n",
    "                   size_pos = 4, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Jon99Ciii\"), \n",
    "                   size_pos = 3, size_neg = 0.1, stroke = 0.02, limit = 3)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Jon65Aiv\"), \n",
    "                   size_pos = 3, size_neg = 0.2, stroke = 0.02, limit = 2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 823,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Crystal_cell\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Crystal_cell\", df_sub300$cluster %in% h)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'h' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'h' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(h)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 819,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Pde1c65</th><td>h</td><td>Pde1c  </td><td>5618</td><td>4675</td><td>44234.440995</td><td>206.433746</td><td>7.736376</td><td>119.39483</td></tr>\n",
       "\t<tr><th scope=row>lz65</th><td>h</td><td>lz     </td><td> 331</td><td> 131</td><td> 2653.352391</td><td>  5.413617</td><td>8.692462</td><td> 98.86932</td></tr>\n",
       "\t<tr><th scope=row>PPO265</th><td>h</td><td>PPO2   </td><td> 908</td><td>2330</td><td> 7240.858646</td><td> 80.280150</td><td>6.477114</td><td> 83.05049</td></tr>\n",
       "\t<tr><th scope=row>PPO165</th><td>h</td><td>PPO1   </td><td> 682</td><td>1477</td><td> 5267.829417</td><td> 53.895262</td><td>6.584383</td><td> 81.40449</td></tr>\n",
       "\t<tr><th scope=row>CG4236965</th><td>h</td><td>CG42369</td><td> 167</td><td> 214</td><td> 1411.183520</td><td>  8.616885</td><td>7.197120</td><td> 75.30859</td></tr>\n",
       "\t<tr><th scope=row>CG1374365</th><td>h</td><td>CG13743</td><td> 286</td><td> 570</td><td> 2193.693999</td><td> 20.336551</td><td>6.683892</td><td> 74.18989</td></tr>\n",
       "\t<tr><th scope=row>CG3132465</th><td>h</td><td>CG31324</td><td> 506</td><td>1989</td><td> 3917.574868</td><td> 89.337219</td><td>5.438497</td><td> 64.91451</td></tr>\n",
       "\t<tr><th scope=row>para65</th><td>h</td><td>para   </td><td>   0</td><td>5394</td><td>    0.000000</td><td>221.106880</td><td>7.788600</td><td> 60.71300</td></tr>\n",
       "\t<tr><th scope=row>qless65</th><td>h</td><td>qless  </td><td>   0</td><td>5385</td><td>    0.000000</td><td>209.657456</td><td>7.711890</td><td> 59.52619</td></tr>\n",
       "\t<tr><th scope=row>CG1710965</th><td>h</td><td>CG17109</td><td>  89</td><td> 277</td><td>  740.115192</td><td>  9.419068</td><td>6.150452</td><td> 58.63566</td></tr>\n",
       "\t<tr><th scope=row>CG1573165</th><td>h</td><td>CG15731</td><td>   0</td><td>5594</td><td>    0.000000</td><td>188.729537</td><td>7.560176</td><td> 57.21391</td></tr>\n",
       "\t<tr><th scope=row>CG1373165</th><td>h</td><td>CG13731</td><td>   0</td><td>5535</td><td>    0.000000</td><td>166.159682</td><td>7.376427</td><td> 54.47552</td></tr>\n",
       "\t<tr><th scope=row>CG541865</th><td>h</td><td>CG5418 </td><td>  38</td><td>  88</td><td>  339.758799</td><td>  3.045545</td><td>6.392033</td><td> 53.77366</td></tr>\n",
       "\t<tr><th scope=row>CG974765</th><td>h</td><td>CG9747 </td><td>   1</td><td>7160</td><td>    2.211802</td><td>274.437571</td><td>6.416951</td><td> 52.01312</td></tr>\n",
       "\t<tr><th scope=row>CG1060265</th><td>h</td><td>CG10602</td><td> 104</td><td> 742</td><td>  789.395246</td><td> 23.787681</td><td>4.993053</td><td> 48.06527</td></tr>\n",
       "\t<tr><th scope=row>CG673365</th><td>h</td><td>CG6733 </td><td>  61</td><td> 310</td><td>  463.967892</td><td> 10.173906</td><td>5.375820</td><td> 47.63507</td></tr>\n",
       "\t<tr><th scope=row>mthl1065</th><td>h</td><td>mthl10 </td><td> 119</td><td> 903</td><td>  944.985923</td><td> 36.702292</td><td>4.647569</td><td> 45.94435</td></tr>\n",
       "\t<tr><th scope=row>crq65</th><td>h</td><td>crq    </td><td> 207</td><td>2034</td><td> 1635.041462</td><td> 84.076824</td><td>4.264417</td><td> 45.52689</td></tr>\n",
       "\t<tr><th scope=row>CG911965</th><td>h</td><td>CG9119 </td><td> 138</td><td>1407</td><td> 1085.530596</td><td> 48.150638</td><td>4.465046</td><td> 45.03228</td></tr>\n",
       "\t<tr><th scope=row>Pxn65</th><td>h</td><td>Pxn    </td><td> 405</td><td>5183</td><td> 3275.454148</td><td>240.537422</td><td>3.761376</td><td> 43.92505</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tPde1c65 & h & Pde1c   & 5618 & 4675 & 44234.440995 & 206.433746 & 7.736376 & 119.39483\\\\\n",
       "\tlz65 & h & lz      &  331 &  131 &  2653.352391 &   5.413617 & 8.692462 &  98.86932\\\\\n",
       "\tPPO265 & h & PPO2    &  908 & 2330 &  7240.858646 &  80.280150 & 6.477114 &  83.05049\\\\\n",
       "\tPPO165 & h & PPO1    &  682 & 1477 &  5267.829417 &  53.895262 & 6.584383 &  81.40449\\\\\n",
       "\tCG4236965 & h & CG42369 &  167 &  214 &  1411.183520 &   8.616885 & 7.197120 &  75.30859\\\\\n",
       "\tCG1374365 & h & CG13743 &  286 &  570 &  2193.693999 &  20.336551 & 6.683892 &  74.18989\\\\\n",
       "\tCG3132465 & h & CG31324 &  506 & 1989 &  3917.574868 &  89.337219 & 5.438497 &  64.91451\\\\\n",
       "\tpara65 & h & para    &    0 & 5394 &     0.000000 & 221.106880 & 7.788600 &  60.71300\\\\\n",
       "\tqless65 & h & qless   &    0 & 5385 &     0.000000 & 209.657456 & 7.711890 &  59.52619\\\\\n",
       "\tCG1710965 & h & CG17109 &   89 &  277 &   740.115192 &   9.419068 & 6.150452 &  58.63566\\\\\n",
       "\tCG1573165 & h & CG15731 &    0 & 5594 &     0.000000 & 188.729537 & 7.560176 &  57.21391\\\\\n",
       "\tCG1373165 & h & CG13731 &    0 & 5535 &     0.000000 & 166.159682 & 7.376427 &  54.47552\\\\\n",
       "\tCG541865 & h & CG5418  &   38 &   88 &   339.758799 &   3.045545 & 6.392033 &  53.77366\\\\\n",
       "\tCG974765 & h & CG9747  &    1 & 7160 &     2.211802 & 274.437571 & 6.416951 &  52.01312\\\\\n",
       "\tCG1060265 & h & CG10602 &  104 &  742 &   789.395246 &  23.787681 & 4.993053 &  48.06527\\\\\n",
       "\tCG673365 & h & CG6733  &   61 &  310 &   463.967892 &  10.173906 & 5.375820 &  47.63507\\\\\n",
       "\tmthl1065 & h & mthl10  &  119 &  903 &   944.985923 &  36.702292 & 4.647569 &  45.94435\\\\\n",
       "\tcrq65 & h & crq     &  207 & 2034 &  1635.041462 &  84.076824 & 4.264417 &  45.52689\\\\\n",
       "\tCG911965 & h & CG9119  &  138 & 1407 &  1085.530596 &  48.150638 & 4.465046 &  45.03228\\\\\n",
       "\tPxn65 & h & Pxn     &  405 & 5183 &  3275.454148 & 240.537422 & 3.761376 &  43.92505\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Pde1c65 | h | Pde1c   | 5618 | 4675 | 44234.440995 | 206.433746 | 7.736376 | 119.39483 |\n",
       "| lz65 | h | lz      |  331 |  131 |  2653.352391 |   5.413617 | 8.692462 |  98.86932 |\n",
       "| PPO265 | h | PPO2    |  908 | 2330 |  7240.858646 |  80.280150 | 6.477114 |  83.05049 |\n",
       "| PPO165 | h | PPO1    |  682 | 1477 |  5267.829417 |  53.895262 | 6.584383 |  81.40449 |\n",
       "| CG4236965 | h | CG42369 |  167 |  214 |  1411.183520 |   8.616885 | 7.197120 |  75.30859 |\n",
       "| CG1374365 | h | CG13743 |  286 |  570 |  2193.693999 |  20.336551 | 6.683892 |  74.18989 |\n",
       "| CG3132465 | h | CG31324 |  506 | 1989 |  3917.574868 |  89.337219 | 5.438497 |  64.91451 |\n",
       "| para65 | h | para    |    0 | 5394 |     0.000000 | 221.106880 | 7.788600 |  60.71300 |\n",
       "| qless65 | h | qless   |    0 | 5385 |     0.000000 | 209.657456 | 7.711890 |  59.52619 |\n",
       "| CG1710965 | h | CG17109 |   89 |  277 |   740.115192 |   9.419068 | 6.150452 |  58.63566 |\n",
       "| CG1573165 | h | CG15731 |    0 | 5594 |     0.000000 | 188.729537 | 7.560176 |  57.21391 |\n",
       "| CG1373165 | h | CG13731 |    0 | 5535 |     0.000000 | 166.159682 | 7.376427 |  54.47552 |\n",
       "| CG541865 | h | CG5418  |   38 |   88 |   339.758799 |   3.045545 | 6.392033 |  53.77366 |\n",
       "| CG974765 | h | CG9747  |    1 | 7160 |     2.211802 | 274.437571 | 6.416951 |  52.01312 |\n",
       "| CG1060265 | h | CG10602 |  104 |  742 |   789.395246 |  23.787681 | 4.993053 |  48.06527 |\n",
       "| CG673365 | h | CG6733  |   61 |  310 |   463.967892 |  10.173906 | 5.375820 |  47.63507 |\n",
       "| mthl1065 | h | mthl10  |  119 |  903 |   944.985923 |  36.702292 | 4.647569 |  45.94435 |\n",
       "| crq65 | h | crq     |  207 | 2034 |  1635.041462 |  84.076824 | 4.264417 |  45.52689 |\n",
       "| CG911965 | h | CG9119  |  138 | 1407 |  1085.530596 |  48.150638 | 4.465046 |  45.03228 |\n",
       "| Pxn65 | h | Pxn     |  405 | 5183 |  3275.454148 | 240.537422 | 3.761376 |  43.92505 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "Pde1c65   h           Pde1c   5618              4675        44234.440995   \n",
       "lz65      h           lz       331               131         2653.352391   \n",
       "PPO265    h           PPO2     908              2330         7240.858646   \n",
       "PPO165    h           PPO1     682              1477         5267.829417   \n",
       "CG4236965 h           CG42369  167               214         1411.183520   \n",
       "CG1374365 h           CG13743  286               570         2193.693999   \n",
       "CG3132465 h           CG31324  506              1989         3917.574868   \n",
       "para65    h           para       0              5394            0.000000   \n",
       "qless65   h           qless      0              5385            0.000000   \n",
       "CG1710965 h           CG17109   89               277          740.115192   \n",
       "CG1573165 h           CG15731    0              5594            0.000000   \n",
       "CG1373165 h           CG13731    0              5535            0.000000   \n",
       "CG541865  h           CG5418    38                88          339.758799   \n",
       "CG974765  h           CG9747     1              7160            2.211802   \n",
       "CG1060265 h           CG10602  104               742          789.395246   \n",
       "CG673365  h           CG6733    61               310          463.967892   \n",
       "mthl1065  h           mthl10   119               903          944.985923   \n",
       "crq65     h           crq      207              2034         1635.041462   \n",
       "CG911965  h           CG9119   138              1407         1085.530596   \n",
       "Pxn65     h           Pxn      405              5183         3275.454148   \n",
       "          other.tpm  log2.ratio heuristic.score\n",
       "Pde1c65   206.433746 7.736376   119.39483      \n",
       "lz65        5.413617 8.692462    98.86932      \n",
       "PPO265     80.280150 6.477114    83.05049      \n",
       "PPO165     53.895262 6.584383    81.40449      \n",
       "CG4236965   8.616885 7.197120    75.30859      \n",
       "CG1374365  20.336551 6.683892    74.18989      \n",
       "CG3132465  89.337219 5.438497    64.91451      \n",
       "para65    221.106880 7.788600    60.71300      \n",
       "qless65   209.657456 7.711890    59.52619      \n",
       "CG1710965   9.419068 6.150452    58.63566      \n",
       "CG1573165 188.729537 7.560176    57.21391      \n",
       "CG1373165 166.159682 7.376427    54.47552      \n",
       "CG541865    3.045545 6.392033    53.77366      \n",
       "CG974765  274.437571 6.416951    52.01312      \n",
       "CG1060265  23.787681 4.993053    48.06527      \n",
       "CG673365   10.173906 5.375820    47.63507      \n",
       "mthl1065   36.702292 4.647569    45.94435      \n",
       "crq65      84.076824 4.264417    45.52689      \n",
       "CG911965   48.150638 4.465046    45.03228      \n",
       "Pxn65     240.537422 3.761376    43.92505      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"h\")  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Pde1c\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Pde1c\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Pde1c\"), \n",
    "                   size_pos = 4, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"lz\"), \n",
    "                   size_pos = 3, size_neg = 0.1, stroke = 0.02, limit = 3)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"PPO2\"), \n",
    "                   size_pos = 3, size_neg = 0.2, stroke = 0.02, limit = 8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 856,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Epidermis\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Epidermis\", df_sub300$cluster %in% c(i, ap) &\n",
    "                         (df_sub300$umap_2 < 0.5*df_sub300$umap_1 -4.2 | df_sub300$umap_2 < 0.9* df_sub300$umap_1 -5.4 & df_sub300$umap_1 > -1)  &\n",
    "              df_sub300$umap_2 > 0.9*df_sub300$umap_1 -6.7)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'i' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'i' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(i)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 825,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>ELOVL66</th><td>i</td><td>ELOVL  </td><td> 7058</td><td> 4681</td><td> 2239.4818</td><td> 168.48327</td><td>3.723950</td><td>41.44605</td></tr>\n",
       "\t<tr><th scope=row>Osi666</th><td>i</td><td>Osi6   </td><td>63628</td><td>81136</td><td>20941.4140</td><td>3251.02531</td><td>2.686948</td><td>38.56883</td></tr>\n",
       "\t<tr><th scope=row>frm66</th><td>i</td><td>frm    </td><td> 4034</td><td> 2856</td><td> 1200.3364</td><td>  97.60995</td><td>3.605562</td><td>36.88643</td></tr>\n",
       "\t<tr><th scope=row>CG3269466</th><td>i</td><td>CG32694</td><td> 3694</td><td> 2363</td><td> 1070.9823</td><td>  90.18310</td><td>3.554024</td><td>35.77504</td></tr>\n",
       "\t<tr><th scope=row>Cht666</th><td>i</td><td>Cht6   </td><td> 4982</td><td> 3946</td><td> 1552.5921</td><td> 148.92165</td><td>3.372398</td><td>35.75212</td></tr>\n",
       "\t<tr><th scope=row>dyl66</th><td>i</td><td>dyl    </td><td>10725</td><td>11990</td><td> 3456.0333</td><td> 455.42317</td><td>2.920673</td><td>34.33344</td></tr>\n",
       "\t<tr><th scope=row>CG532666</th><td>i</td><td>CG5326 </td><td> 1800</td><td> 1053</td><td>  498.1302</td><td>  34.38688</td><td>3.815236</td><td>34.19700</td></tr>\n",
       "\t<tr><th scope=row>CG1021166</th><td>i</td><td>CG10211</td><td> 7358</td><td> 7622</td><td> 2395.9431</td><td> 290.61937</td><td>3.038435</td><td>34.11245</td></tr>\n",
       "\t<tr><th scope=row>CG941166</th><td>i</td><td>CG9411 </td><td>10035</td><td>12260</td><td> 3191.8854</td><td> 437.69971</td><td>2.863103</td><td>33.32837</td></tr>\n",
       "\t<tr><th scope=row>Cpr51A66</th><td>i</td><td>Cpr51A </td><td> 3098</td><td> 2241</td><td>  903.3172</td><td>  86.47175</td><td>3.368344</td><td>33.07944</td></tr>\n",
       "\t<tr><th scope=row>CG1093666</th><td>i</td><td>CG10936</td><td> 6005</td><td> 5800</td><td> 1907.5139</td><td> 247.70906</td><td>2.939163</td><td>32.03168</td></tr>\n",
       "\t<tr><th scope=row>stw66</th><td>i</td><td>stw    </td><td> 2902</td><td> 2454</td><td>  934.2348</td><td>  99.93148</td><td>3.210409</td><td>31.68412</td></tr>\n",
       "\t<tr><th scope=row>TwdlB66</th><td>i</td><td>TwdlB  </td><td> 1995</td><td> 1399</td><td>  494.0137</td><td>  41.62806</td><td>3.534676</td><td>31.64003</td></tr>\n",
       "\t<tr><th scope=row>f66</th><td>i</td><td>f      </td><td> 2113</td><td> 1532</td><td>  675.8753</td><td>  65.28195</td><td>3.350069</td><td>31.49985</td></tr>\n",
       "\t<tr><th scope=row>Cda466</th><td>i</td><td>Cda4   </td><td> 5741</td><td> 6214</td><td> 1809.5542</td><td> 240.17688</td><td>2.907471</td><td>31.46528</td></tr>\n",
       "\t<tr><th scope=row>CG3397866</th><td>i</td><td>CG33978</td><td> 6278</td><td> 6989</td><td> 1966.2247</td><td> 270.49123</td><td>2.856451</td><td>31.25513</td></tr>\n",
       "\t<tr><th scope=row>CG4333366</th><td>i</td><td>CG43333</td><td> 4802</td><td> 5448</td><td> 1546.3962</td><td> 201.60579</td><td>2.932163</td><td>31.06810</td></tr>\n",
       "\t<tr><th scope=row>CG575666</th><td>i</td><td>CG5756 </td><td> 3160</td><td> 3450</td><td> 1036.4229</td><td> 120.41618</td><td>3.093580</td><td>30.99393</td></tr>\n",
       "\t<tr><th scope=row>TwdlM66</th><td>i</td><td>TwdlM  </td><td> 3538</td><td> 3196</td><td>  861.4892</td><td>  97.27835</td><td>3.131887</td><td>30.54330</td></tr>\n",
       "\t<tr><th scope=row>CG1425766</th><td>i</td><td>CG14257</td><td> 2330</td><td> 2038</td><td>  751.4888</td><td>  81.08390</td><td>3.194581</td><td>30.52590</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tELOVL66 & i & ELOVL   &  7058 &  4681 &  2239.4818 &  168.48327 & 3.723950 & 41.44605\\\\\n",
       "\tOsi666 & i & Osi6    & 63628 & 81136 & 20941.4140 & 3251.02531 & 2.686948 & 38.56883\\\\\n",
       "\tfrm66 & i & frm     &  4034 &  2856 &  1200.3364 &   97.60995 & 3.605562 & 36.88643\\\\\n",
       "\tCG3269466 & i & CG32694 &  3694 &  2363 &  1070.9823 &   90.18310 & 3.554024 & 35.77504\\\\\n",
       "\tCht666 & i & Cht6    &  4982 &  3946 &  1552.5921 &  148.92165 & 3.372398 & 35.75212\\\\\n",
       "\tdyl66 & i & dyl     & 10725 & 11990 &  3456.0333 &  455.42317 & 2.920673 & 34.33344\\\\\n",
       "\tCG532666 & i & CG5326  &  1800 &  1053 &   498.1302 &   34.38688 & 3.815236 & 34.19700\\\\\n",
       "\tCG1021166 & i & CG10211 &  7358 &  7622 &  2395.9431 &  290.61937 & 3.038435 & 34.11245\\\\\n",
       "\tCG941166 & i & CG9411  & 10035 & 12260 &  3191.8854 &  437.69971 & 2.863103 & 33.32837\\\\\n",
       "\tCpr51A66 & i & Cpr51A  &  3098 &  2241 &   903.3172 &   86.47175 & 3.368344 & 33.07944\\\\\n",
       "\tCG1093666 & i & CG10936 &  6005 &  5800 &  1907.5139 &  247.70906 & 2.939163 & 32.03168\\\\\n",
       "\tstw66 & i & stw     &  2902 &  2454 &   934.2348 &   99.93148 & 3.210409 & 31.68412\\\\\n",
       "\tTwdlB66 & i & TwdlB   &  1995 &  1399 &   494.0137 &   41.62806 & 3.534676 & 31.64003\\\\\n",
       "\tf66 & i & f       &  2113 &  1532 &   675.8753 &   65.28195 & 3.350069 & 31.49985\\\\\n",
       "\tCda466 & i & Cda4    &  5741 &  6214 &  1809.5542 &  240.17688 & 2.907471 & 31.46528\\\\\n",
       "\tCG3397866 & i & CG33978 &  6278 &  6989 &  1966.2247 &  270.49123 & 2.856451 & 31.25513\\\\\n",
       "\tCG4333366 & i & CG43333 &  4802 &  5448 &  1546.3962 &  201.60579 & 2.932163 & 31.06810\\\\\n",
       "\tCG575666 & i & CG5756  &  3160 &  3450 &  1036.4229 &  120.41618 & 3.093580 & 30.99393\\\\\n",
       "\tTwdlM66 & i & TwdlM   &  3538 &  3196 &   861.4892 &   97.27835 & 3.131887 & 30.54330\\\\\n",
       "\tCG1425766 & i & CG14257 &  2330 &  2038 &   751.4888 &   81.08390 & 3.194581 & 30.52590\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| ELOVL66 | i | ELOVL   |  7058 |  4681 |  2239.4818 |  168.48327 | 3.723950 | 41.44605 |\n",
       "| Osi666 | i | Osi6    | 63628 | 81136 | 20941.4140 | 3251.02531 | 2.686948 | 38.56883 |\n",
       "| frm66 | i | frm     |  4034 |  2856 |  1200.3364 |   97.60995 | 3.605562 | 36.88643 |\n",
       "| CG3269466 | i | CG32694 |  3694 |  2363 |  1070.9823 |   90.18310 | 3.554024 | 35.77504 |\n",
       "| Cht666 | i | Cht6    |  4982 |  3946 |  1552.5921 |  148.92165 | 3.372398 | 35.75212 |\n",
       "| dyl66 | i | dyl     | 10725 | 11990 |  3456.0333 |  455.42317 | 2.920673 | 34.33344 |\n",
       "| CG532666 | i | CG5326  |  1800 |  1053 |   498.1302 |   34.38688 | 3.815236 | 34.19700 |\n",
       "| CG1021166 | i | CG10211 |  7358 |  7622 |  2395.9431 |  290.61937 | 3.038435 | 34.11245 |\n",
       "| CG941166 | i | CG9411  | 10035 | 12260 |  3191.8854 |  437.69971 | 2.863103 | 33.32837 |\n",
       "| Cpr51A66 | i | Cpr51A  |  3098 |  2241 |   903.3172 |   86.47175 | 3.368344 | 33.07944 |\n",
       "| CG1093666 | i | CG10936 |  6005 |  5800 |  1907.5139 |  247.70906 | 2.939163 | 32.03168 |\n",
       "| stw66 | i | stw     |  2902 |  2454 |   934.2348 |   99.93148 | 3.210409 | 31.68412 |\n",
       "| TwdlB66 | i | TwdlB   |  1995 |  1399 |   494.0137 |   41.62806 | 3.534676 | 31.64003 |\n",
       "| f66 | i | f       |  2113 |  1532 |   675.8753 |   65.28195 | 3.350069 | 31.49985 |\n",
       "| Cda466 | i | Cda4    |  5741 |  6214 |  1809.5542 |  240.17688 | 2.907471 | 31.46528 |\n",
       "| CG3397866 | i | CG33978 |  6278 |  6989 |  1966.2247 |  270.49123 | 2.856451 | 31.25513 |\n",
       "| CG4333366 | i | CG43333 |  4802 |  5448 |  1546.3962 |  201.60579 | 2.932163 | 31.06810 |\n",
       "| CG575666 | i | CG5756  |  3160 |  3450 |  1036.4229 |  120.41618 | 3.093580 | 30.99393 |\n",
       "| TwdlM66 | i | TwdlM   |  3538 |  3196 |   861.4892 |   97.27835 | 3.131887 | 30.54330 |\n",
       "| CG1425766 | i | CG14257 |  2330 |  2038 |   751.4888 |   81.08390 | 3.194581 | 30.52590 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "ELOVL66   i           ELOVL    7058              4681        2239.4818     \n",
       "Osi666    i           Osi6    63628             81136       20941.4140     \n",
       "frm66     i           frm      4034              2856        1200.3364     \n",
       "CG3269466 i           CG32694  3694              2363        1070.9823     \n",
       "Cht666    i           Cht6     4982              3946        1552.5921     \n",
       "dyl66     i           dyl     10725             11990        3456.0333     \n",
       "CG532666  i           CG5326   1800              1053         498.1302     \n",
       "CG1021166 i           CG10211  7358              7622        2395.9431     \n",
       "CG941166  i           CG9411  10035             12260        3191.8854     \n",
       "Cpr51A66  i           Cpr51A   3098              2241         903.3172     \n",
       "CG1093666 i           CG10936  6005              5800        1907.5139     \n",
       "stw66     i           stw      2902              2454         934.2348     \n",
       "TwdlB66   i           TwdlB    1995              1399         494.0137     \n",
       "f66       i           f        2113              1532         675.8753     \n",
       "Cda466    i           Cda4     5741              6214        1809.5542     \n",
       "CG3397866 i           CG33978  6278              6989        1966.2247     \n",
       "CG4333366 i           CG43333  4802              5448        1546.3962     \n",
       "CG575666  i           CG5756   3160              3450        1036.4229     \n",
       "TwdlM66   i           TwdlM    3538              3196         861.4892     \n",
       "CG1425766 i           CG14257  2330              2038         751.4888     \n",
       "          other.tpm  log2.ratio heuristic.score\n",
       "ELOVL66    168.48327 3.723950   41.44605       \n",
       "Osi666    3251.02531 2.686948   38.56883       \n",
       "frm66       97.60995 3.605562   36.88643       \n",
       "CG3269466   90.18310 3.554024   35.77504       \n",
       "Cht666     148.92165 3.372398   35.75212       \n",
       "dyl66      455.42317 2.920673   34.33344       \n",
       "CG532666    34.38688 3.815236   34.19700       \n",
       "CG1021166  290.61937 3.038435   34.11245       \n",
       "CG941166   437.69971 2.863103   33.32837       \n",
       "Cpr51A66    86.47175 3.368344   33.07944       \n",
       "CG1093666  247.70906 2.939163   32.03168       \n",
       "stw66       99.93148 3.210409   31.68412       \n",
       "TwdlB66     41.62806 3.534676   31.64003       \n",
       "f66         65.28195 3.350069   31.49985       \n",
       "Cda466     240.17688 2.907471   31.46528       \n",
       "CG3397866  270.49123 2.856451   31.25513       \n",
       "CG4333366  201.60579 2.932163   31.06810       \n",
       "CG575666   120.41618 3.093580   30.99393       \n",
       "TwdlM66     97.27835 3.131887   30.54330       \n",
       "CG1425766   81.08390 3.194581   30.52590       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"i\")  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"ELOVL\"), size_pos = 3, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"ELOVL\"), size_pos = 3, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"ELOVL\"), \n",
    "                   size_pos = 3, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Cht6\"), \n",
    "                   size_pos = 3, size_neg = 0.1, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"frm\"), \n",
    "                   size_pos = 3, size_neg = 0.2, stroke = 0.02, limit = 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'i' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'i' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(i, ap)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.5, intercept = -4.2) +\n",
    "geom_abline(slope = 0.9, intercept = -6.7) +\n",
    "geom_abline(slope = 0.9, intercept = -5.4) # +\n",
    "# xlim(-4, 2) + ylim(-9, -3)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'i' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'i' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(i, ap)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &\n",
    "              (umap_2 < 0.5*umap_1 -4.2 | umap_2 < 0.9* umap_1 -5.4 & umap_1 > -1)  &\n",
    "              umap_2 > 0.9*umap_1 -6.7), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = -1, intercept = -7.2) +\n",
    "geom_abline(slope = -1., intercept = -6.7) # +\n",
    "# xlim(-4, 2) + ylim(-9, -3)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### j cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 862,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Epidermis_late\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Epidermis_late\", df_sub300$cluster %in% j)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'j' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'j' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(j)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 860,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG1158167</th><td>j</td><td>CG11581 </td><td> 257</td><td>   44</td><td> 2564.3011</td><td>   1.1160558</td><td>10.242972</td><td>116.00076</td></tr>\n",
       "\t<tr><th scope=row>Tb67</th><td>j</td><td>Tb      </td><td> 814</td><td>  491</td><td> 8391.4464</td><td>  19.2476737</td><td> 8.695020</td><td>113.33850</td></tr>\n",
       "\t<tr><th scope=row>CG975767</th><td>j</td><td>CG9757  </td><td> 390</td><td>  524</td><td> 4011.9120</td><td>  12.9308165</td><td> 8.169866</td><td> 97.79684</td></tr>\n",
       "\t<tr><th scope=row>CG335587</th><td>j</td><td>CG3355  </td><td> 282</td><td>  228</td><td> 2712.3344</td><td>   6.7587166</td><td> 8.449501</td><td> 96.37375</td></tr>\n",
       "\t<tr><th scope=row>TwdlR67</th><td>j</td><td>TwdlR   </td><td>  89</td><td>    9</td><td> 1051.6910</td><td>   0.3729667</td><td> 9.581199</td><td> 96.19395</td></tr>\n",
       "\t<tr><th scope=row>Cpr47Ec67</th><td>j</td><td>Cpr47Ec </td><td>  99</td><td>   46</td><td> 1186.7214</td><td>   1.5841392</td><td> 8.843082</td><td> 90.32307</td></tr>\n",
       "\t<tr><th scope=row>CG3162667</th><td>j</td><td>CG31626 </td><td>1007</td><td> 2460</td><td> 9432.7732</td><td>  86.1951177</td><td> 6.757291</td><td> 89.22069</td></tr>\n",
       "\t<tr><th scope=row>TwdlQ67</th><td>j</td><td>TwdlQ   </td><td> 114</td><td>   86</td><td> 1332.2345</td><td>   3.1993661</td><td> 8.309461</td><td> 86.25814</td></tr>\n",
       "\t<tr><th scope=row>CG1158467</th><td>j</td><td>CG11584 </td><td> 412</td><td> 1078</td><td> 3697.6084</td><td>  38.4811689</td><td> 6.549284</td><td> 77.62714</td></tr>\n",
       "\t<tr><th scope=row>TwdlF67</th><td>j</td><td>TwdlF   </td><td>2100</td><td>13904</td><td>19457.6713</td><td> 511.5772333</td><td> 5.246426</td><td> 74.75173</td></tr>\n",
       "\t<tr><th scope=row>CG1574067</th><td>j</td><td>CG15740 </td><td> 690</td><td> 2832</td><td> 6338.2398</td><td> 104.5155089</td><td> 5.908555</td><td> 74.62561</td></tr>\n",
       "\t<tr><th scope=row>TwdlH67</th><td>j</td><td>TwdlH   </td><td>  78</td><td>   88</td><td>  868.8420</td><td>   3.5621377</td><td> 7.573240</td><td> 73.94973</td></tr>\n",
       "\t<tr><th scope=row>zye67</th><td>j</td><td>zye     </td><td> 171</td><td>  412</td><td> 1802.9146</td><td>  15.7982111</td><td> 6.745880</td><td> 72.96961</td></tr>\n",
       "\t<tr><th scope=row>TwdlS67</th><td>j</td><td>TwdlS   </td><td>  30</td><td>    4</td><td>  392.0188</td><td>   0.1809149</td><td> 8.374874</td><td> 72.17847</td></tr>\n",
       "\t<tr><th scope=row>CG1573167</th><td>j</td><td>CG15731 </td><td> 825</td><td> 5506</td><td> 7853.2288</td><td> 183.1030068</td><td> 5.414701</td><td> 70.06219</td></tr>\n",
       "\t<tr><th scope=row>CG729967</th><td>j</td><td>CG7299  </td><td> 423</td><td> 3089</td><td> 3738.9729</td><td>  77.9581934</td><td> 5.565409</td><td> 66.05480</td></tr>\n",
       "\t<tr><th scope=row>CG3047167</th><td>j</td><td>CG30471 </td><td>  35</td><td>   19</td><td>  340.9280</td><td>   0.8553598</td><td> 7.521624</td><td> 63.31364</td></tr>\n",
       "\t<tr><th scope=row>Cpr67Fa167</th><td>j</td><td>Cpr67Fa1</td><td>  51</td><td>  155</td><td>  671.9780</td><td>   6.0538185</td><td> 6.573866</td><td> 61.75763</td></tr>\n",
       "\t<tr><th scope=row>CG740667</th><td>j</td><td>CG7406  </td><td> 151</td><td>  684</td><td> 1456.3888</td><td>  24.4234510</td><td> 5.840092</td><td> 61.37452</td></tr>\n",
       "\t<tr><th scope=row>CG296267</th><td>j</td><td>CG2962  </td><td>2109</td><td>30464</td><td>20712.5480</td><td>1096.9101674</td><td> 4.237673</td><td> 60.76097</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG1158167 & j & CG11581  &  257 &    44 &  2564.3011 &    1.1160558 & 10.242972 & 116.00076\\\\\n",
       "\tTb67 & j & Tb       &  814 &   491 &  8391.4464 &   19.2476737 &  8.695020 & 113.33850\\\\\n",
       "\tCG975767 & j & CG9757   &  390 &   524 &  4011.9120 &   12.9308165 &  8.169866 &  97.79684\\\\\n",
       "\tCG335587 & j & CG3355   &  282 &   228 &  2712.3344 &    6.7587166 &  8.449501 &  96.37375\\\\\n",
       "\tTwdlR67 & j & TwdlR    &   89 &     9 &  1051.6910 &    0.3729667 &  9.581199 &  96.19395\\\\\n",
       "\tCpr47Ec67 & j & Cpr47Ec  &   99 &    46 &  1186.7214 &    1.5841392 &  8.843082 &  90.32307\\\\\n",
       "\tCG3162667 & j & CG31626  & 1007 &  2460 &  9432.7732 &   86.1951177 &  6.757291 &  89.22069\\\\\n",
       "\tTwdlQ67 & j & TwdlQ    &  114 &    86 &  1332.2345 &    3.1993661 &  8.309461 &  86.25814\\\\\n",
       "\tCG1158467 & j & CG11584  &  412 &  1078 &  3697.6084 &   38.4811689 &  6.549284 &  77.62714\\\\\n",
       "\tTwdlF67 & j & TwdlF    & 2100 & 13904 & 19457.6713 &  511.5772333 &  5.246426 &  74.75173\\\\\n",
       "\tCG1574067 & j & CG15740  &  690 &  2832 &  6338.2398 &  104.5155089 &  5.908555 &  74.62561\\\\\n",
       "\tTwdlH67 & j & TwdlH    &   78 &    88 &   868.8420 &    3.5621377 &  7.573240 &  73.94973\\\\\n",
       "\tzye67 & j & zye      &  171 &   412 &  1802.9146 &   15.7982111 &  6.745880 &  72.96961\\\\\n",
       "\tTwdlS67 & j & TwdlS    &   30 &     4 &   392.0188 &    0.1809149 &  8.374874 &  72.17847\\\\\n",
       "\tCG1573167 & j & CG15731  &  825 &  5506 &  7853.2288 &  183.1030068 &  5.414701 &  70.06219\\\\\n",
       "\tCG729967 & j & CG7299   &  423 &  3089 &  3738.9729 &   77.9581934 &  5.565409 &  66.05480\\\\\n",
       "\tCG3047167 & j & CG30471  &   35 &    19 &   340.9280 &    0.8553598 &  7.521624 &  63.31364\\\\\n",
       "\tCpr67Fa167 & j & Cpr67Fa1 &   51 &   155 &   671.9780 &    6.0538185 &  6.573866 &  61.75763\\\\\n",
       "\tCG740667 & j & CG7406   &  151 &   684 &  1456.3888 &   24.4234510 &  5.840092 &  61.37452\\\\\n",
       "\tCG296267 & j & CG2962   & 2109 & 30464 & 20712.5480 & 1096.9101674 &  4.237673 &  60.76097\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG1158167 | j | CG11581  |  257 |    44 |  2564.3011 |    1.1160558 | 10.242972 | 116.00076 |\n",
       "| Tb67 | j | Tb       |  814 |   491 |  8391.4464 |   19.2476737 |  8.695020 | 113.33850 |\n",
       "| CG975767 | j | CG9757   |  390 |   524 |  4011.9120 |   12.9308165 |  8.169866 |  97.79684 |\n",
       "| CG335587 | j | CG3355   |  282 |   228 |  2712.3344 |    6.7587166 |  8.449501 |  96.37375 |\n",
       "| TwdlR67 | j | TwdlR    |   89 |     9 |  1051.6910 |    0.3729667 |  9.581199 |  96.19395 |\n",
       "| Cpr47Ec67 | j | Cpr47Ec  |   99 |    46 |  1186.7214 |    1.5841392 |  8.843082 |  90.32307 |\n",
       "| CG3162667 | j | CG31626  | 1007 |  2460 |  9432.7732 |   86.1951177 |  6.757291 |  89.22069 |\n",
       "| TwdlQ67 | j | TwdlQ    |  114 |    86 |  1332.2345 |    3.1993661 |  8.309461 |  86.25814 |\n",
       "| CG1158467 | j | CG11584  |  412 |  1078 |  3697.6084 |   38.4811689 |  6.549284 |  77.62714 |\n",
       "| TwdlF67 | j | TwdlF    | 2100 | 13904 | 19457.6713 |  511.5772333 |  5.246426 |  74.75173 |\n",
       "| CG1574067 | j | CG15740  |  690 |  2832 |  6338.2398 |  104.5155089 |  5.908555 |  74.62561 |\n",
       "| TwdlH67 | j | TwdlH    |   78 |    88 |   868.8420 |    3.5621377 |  7.573240 |  73.94973 |\n",
       "| zye67 | j | zye      |  171 |   412 |  1802.9146 |   15.7982111 |  6.745880 |  72.96961 |\n",
       "| TwdlS67 | j | TwdlS    |   30 |     4 |   392.0188 |    0.1809149 |  8.374874 |  72.17847 |\n",
       "| CG1573167 | j | CG15731  |  825 |  5506 |  7853.2288 |  183.1030068 |  5.414701 |  70.06219 |\n",
       "| CG729967 | j | CG7299   |  423 |  3089 |  3738.9729 |   77.9581934 |  5.565409 |  66.05480 |\n",
       "| CG3047167 | j | CG30471  |   35 |    19 |   340.9280 |    0.8553598 |  7.521624 |  63.31364 |\n",
       "| Cpr67Fa167 | j | Cpr67Fa1 |   51 |   155 |   671.9780 |    6.0538185 |  6.573866 |  61.75763 |\n",
       "| CG740667 | j | CG7406   |  151 |   684 |  1456.3888 |   24.4234510 |  5.840092 |  61.37452 |\n",
       "| CG296267 | j | CG2962   | 2109 | 30464 | 20712.5480 | 1096.9101674 |  4.237673 |  60.76097 |\n",
       "\n"
      ],
      "text/plain": [
       "           cell.type.2 gene     cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "CG1158167  j           CG11581   257                 44        2564.3011     \n",
       "Tb67       j           Tb        814                491        8391.4464     \n",
       "CG975767   j           CG9757    390                524        4011.9120     \n",
       "CG335587   j           CG3355    282                228        2712.3344     \n",
       "TwdlR67    j           TwdlR      89                  9        1051.6910     \n",
       "Cpr47Ec67  j           Cpr47Ec    99                 46        1186.7214     \n",
       "CG3162667  j           CG31626  1007               2460        9432.7732     \n",
       "TwdlQ67    j           TwdlQ     114                 86        1332.2345     \n",
       "CG1158467  j           CG11584   412               1078        3697.6084     \n",
       "TwdlF67    j           TwdlF    2100              13904       19457.6713     \n",
       "CG1574067  j           CG15740   690               2832        6338.2398     \n",
       "TwdlH67    j           TwdlH      78                 88         868.8420     \n",
       "zye67      j           zye       171                412        1802.9146     \n",
       "TwdlS67    j           TwdlS      30                  4         392.0188     \n",
       "CG1573167  j           CG15731   825               5506        7853.2288     \n",
       "CG729967   j           CG7299    423               3089        3738.9729     \n",
       "CG3047167  j           CG30471    35                 19         340.9280     \n",
       "Cpr67Fa167 j           Cpr67Fa1   51                155         671.9780     \n",
       "CG740667   j           CG7406    151                684        1456.3888     \n",
       "CG296267   j           CG2962   2109              30464       20712.5480     \n",
       "           other.tpm    log2.ratio heuristic.score\n",
       "CG1158167     1.1160558 10.242972  116.00076      \n",
       "Tb67         19.2476737  8.695020  113.33850      \n",
       "CG975767     12.9308165  8.169866   97.79684      \n",
       "CG335587      6.7587166  8.449501   96.37375      \n",
       "TwdlR67       0.3729667  9.581199   96.19395      \n",
       "Cpr47Ec67     1.5841392  8.843082   90.32307      \n",
       "CG3162667    86.1951177  6.757291   89.22069      \n",
       "TwdlQ67       3.1993661  8.309461   86.25814      \n",
       "CG1158467    38.4811689  6.549284   77.62714      \n",
       "TwdlF67     511.5772333  5.246426   74.75173      \n",
       "CG1574067   104.5155089  5.908555   74.62561      \n",
       "TwdlH67       3.5621377  7.573240   73.94973      \n",
       "zye67        15.7982111  6.745880   72.96961      \n",
       "TwdlS67       0.1809149  8.374874   72.17847      \n",
       "CG1573167   183.1030068  5.414701   70.06219      \n",
       "CG729967     77.9581934  5.565409   66.05480      \n",
       "CG3047167     0.8553598  7.521624   63.31364      \n",
       "Cpr67Fa167    6.0538185  6.573866   61.75763      \n",
       "CG740667     24.4234510  5.840092   61.37452      \n",
       "CG296267   1096.9101674  4.237673   60.76097      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"j\")  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Tb\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Tb\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Tb\"), \n",
    "                   size_pos = 4, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"TwdlR\"), \n",
    "                   size_pos = 3, size_neg = 0.1, stroke = 0.02, limit = 1)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"TwdlQ\"), \n",
    "                   size_pos = 3, size_neg = 0.2, stroke = 0.02, limit = 0.01)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 865,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type uncertain_1\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"uncertain_1\", df_sub300$cluster %in% k)\n",
    "## very small; markers specific but very low, might be artifact"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'k' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'k' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(k)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 864,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG3195468</th><td>k</td><td>CG31954 </td><td>28</td><td>   11</td><td>1255.6939</td><td>  0.3485554</td><td>9.862854</td><td>101.54220</td></tr>\n",
       "\t<tr><th scope=row>CG3444768</th><td>k</td><td>CG34447 </td><td>27</td><td>    6</td><td>1141.3501</td><td>  0.1320903</td><td>9.977537</td><td>101.34971</td></tr>\n",
       "\t<tr><th scope=row>CG525568</th><td>k</td><td>CG5255  </td><td>20</td><td>   13</td><td>1073.4703</td><td>  0.4351591</td><td>9.546856</td><td> 96.13120</td></tr>\n",
       "\t<tr><th scope=row>CG401768</th><td>k</td><td>CG4017  </td><td>12</td><td>    6</td><td> 613.0370</td><td>  0.2387753</td><td>8.950916</td><td> 82.90501</td></tr>\n",
       "\t<tr><th scope=row>fng68</th><td>k</td><td>fng     </td><td> 0</td><td>11165</td><td>   0.0000</td><td>524.1208112</td><td>9.033756</td><td> 81.63358</td></tr>\n",
       "\t<tr><th scope=row>CG4333268</th><td>k</td><td>CG43332 </td><td> 0</td><td> 7448</td><td>   0.0000</td><td>354.2582008</td><td>8.468657</td><td> 71.75260</td></tr>\n",
       "\t<tr><th scope=row>Nipped-A68</th><td>k</td><td>Nipped-A</td><td> 0</td><td> 8224</td><td>   0.0000</td><td>322.5693846</td><td>8.333466</td><td> 69.48386</td></tr>\n",
       "\t<tr><th scope=row>Ser99</th><td>k</td><td>Ser     </td><td> 0</td><td> 6688</td><td>   0.0000</td><td>304.5744894</td><td>8.250651</td><td> 68.11226</td></tr>\n",
       "\t<tr><th scope=row>MYPT-75D68</th><td>k</td><td>MYPT-75D</td><td> 0</td><td> 6518</td><td>   0.0000</td><td>291.5150520</td><td>8.187427</td><td> 67.07440</td></tr>\n",
       "\t<tr><th scope=row>CG772068</th><td>k</td><td>CG7720  </td><td> 0</td><td> 4716</td><td>   0.0000</td><td>222.6207734</td><td>7.798444</td><td> 60.86616</td></tr>\n",
       "\t<tr><th scope=row>Hmgcr68</th><td>k</td><td>Hmgcr   </td><td> 0</td><td> 4744</td><td>   0.0000</td><td>220.4553360</td><td>7.784343</td><td> 60.64682</td></tr>\n",
       "\t<tr><th scope=row>CG588568</th><td>k</td><td>CG5885  </td><td> 0</td><td> 4492</td><td>   0.0000</td><td>206.8538025</td><td>7.692468</td><td> 59.22758</td></tr>\n",
       "\t<tr><th scope=row>CG1094868</th><td>k</td><td>CG10948 </td><td> 0</td><td> 4608</td><td>   0.0000</td><td>204.9914719</td><td>7.679420</td><td> 59.02741</td></tr>\n",
       "\t<tr><th scope=row>pcs68</th><td>k</td><td>pcs     </td><td> 0</td><td> 4725</td><td>   0.0000</td><td>193.7074763</td><td>7.597736</td><td> 57.78203</td></tr>\n",
       "\t<tr><th scope=row>Scamp68</th><td>k</td><td>Scamp   </td><td> 0</td><td> 4671</td><td>   0.0000</td><td>184.6481637</td><td>7.528635</td><td> 56.73901</td></tr>\n",
       "\t<tr><th scope=row>mei-W6868</th><td>k</td><td>mei-W68 </td><td> 0</td><td> 4029</td><td>   0.0000</td><td>182.2124950</td><td>7.509478</td><td> 56.45156</td></tr>\n",
       "\t<tr><th scope=row>CG980468</th><td>k</td><td>CG9804  </td><td> 7</td><td>  170</td><td> 535.9008</td><td>  6.2338415</td><td>6.211060</td><td> 56.32507</td></tr>\n",
       "\t<tr><th scope=row>Notum68</th><td>k</td><td>Notum   </td><td> 0</td><td> 4153</td><td>   0.0000</td><td>180.3219519</td><td>7.494431</td><td> 56.22629</td></tr>\n",
       "\t<tr><th scope=row>AdamTS-A68</th><td>k</td><td>AdamTS-A</td><td> 0</td><td> 3680</td><td>   0.0000</td><td>170.3103195</td><td>7.412022</td><td> 55.00067</td></tr>\n",
       "\t<tr><th scope=row>CG1534768</th><td>k</td><td>CG15347 </td><td> 0</td><td> 3586</td><td>   0.0000</td><td>169.7153933</td><td>7.406974</td><td> 54.92604</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG3195468 & k & CG31954  & 28 &    11 & 1255.6939 &   0.3485554 & 9.862854 & 101.54220\\\\\n",
       "\tCG3444768 & k & CG34447  & 27 &     6 & 1141.3501 &   0.1320903 & 9.977537 & 101.34971\\\\\n",
       "\tCG525568 & k & CG5255   & 20 &    13 & 1073.4703 &   0.4351591 & 9.546856 &  96.13120\\\\\n",
       "\tCG401768 & k & CG4017   & 12 &     6 &  613.0370 &   0.2387753 & 8.950916 &  82.90501\\\\\n",
       "\tfng68 & k & fng      &  0 & 11165 &    0.0000 & 524.1208112 & 9.033756 &  81.63358\\\\\n",
       "\tCG4333268 & k & CG43332  &  0 &  7448 &    0.0000 & 354.2582008 & 8.468657 &  71.75260\\\\\n",
       "\tNipped-A68 & k & Nipped-A &  0 &  8224 &    0.0000 & 322.5693846 & 8.333466 &  69.48386\\\\\n",
       "\tSer99 & k & Ser      &  0 &  6688 &    0.0000 & 304.5744894 & 8.250651 &  68.11226\\\\\n",
       "\tMYPT-75D68 & k & MYPT-75D &  0 &  6518 &    0.0000 & 291.5150520 & 8.187427 &  67.07440\\\\\n",
       "\tCG772068 & k & CG7720   &  0 &  4716 &    0.0000 & 222.6207734 & 7.798444 &  60.86616\\\\\n",
       "\tHmgcr68 & k & Hmgcr    &  0 &  4744 &    0.0000 & 220.4553360 & 7.784343 &  60.64682\\\\\n",
       "\tCG588568 & k & CG5885   &  0 &  4492 &    0.0000 & 206.8538025 & 7.692468 &  59.22758\\\\\n",
       "\tCG1094868 & k & CG10948  &  0 &  4608 &    0.0000 & 204.9914719 & 7.679420 &  59.02741\\\\\n",
       "\tpcs68 & k & pcs      &  0 &  4725 &    0.0000 & 193.7074763 & 7.597736 &  57.78203\\\\\n",
       "\tScamp68 & k & Scamp    &  0 &  4671 &    0.0000 & 184.6481637 & 7.528635 &  56.73901\\\\\n",
       "\tmei-W6868 & k & mei-W68  &  0 &  4029 &    0.0000 & 182.2124950 & 7.509478 &  56.45156\\\\\n",
       "\tCG980468 & k & CG9804   &  7 &   170 &  535.9008 &   6.2338415 & 6.211060 &  56.32507\\\\\n",
       "\tNotum68 & k & Notum    &  0 &  4153 &    0.0000 & 180.3219519 & 7.494431 &  56.22629\\\\\n",
       "\tAdamTS-A68 & k & AdamTS-A &  0 &  3680 &    0.0000 & 170.3103195 & 7.412022 &  55.00067\\\\\n",
       "\tCG1534768 & k & CG15347  &  0 &  3586 &    0.0000 & 169.7153933 & 7.406974 &  54.92604\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG3195468 | k | CG31954  | 28 |    11 | 1255.6939 |   0.3485554 | 9.862854 | 101.54220 |\n",
       "| CG3444768 | k | CG34447  | 27 |     6 | 1141.3501 |   0.1320903 | 9.977537 | 101.34971 |\n",
       "| CG525568 | k | CG5255   | 20 |    13 | 1073.4703 |   0.4351591 | 9.546856 |  96.13120 |\n",
       "| CG401768 | k | CG4017   | 12 |     6 |  613.0370 |   0.2387753 | 8.950916 |  82.90501 |\n",
       "| fng68 | k | fng      |  0 | 11165 |    0.0000 | 524.1208112 | 9.033756 |  81.63358 |\n",
       "| CG4333268 | k | CG43332  |  0 |  7448 |    0.0000 | 354.2582008 | 8.468657 |  71.75260 |\n",
       "| Nipped-A68 | k | Nipped-A |  0 |  8224 |    0.0000 | 322.5693846 | 8.333466 |  69.48386 |\n",
       "| Ser99 | k | Ser      |  0 |  6688 |    0.0000 | 304.5744894 | 8.250651 |  68.11226 |\n",
       "| MYPT-75D68 | k | MYPT-75D |  0 |  6518 |    0.0000 | 291.5150520 | 8.187427 |  67.07440 |\n",
       "| CG772068 | k | CG7720   |  0 |  4716 |    0.0000 | 222.6207734 | 7.798444 |  60.86616 |\n",
       "| Hmgcr68 | k | Hmgcr    |  0 |  4744 |    0.0000 | 220.4553360 | 7.784343 |  60.64682 |\n",
       "| CG588568 | k | CG5885   |  0 |  4492 |    0.0000 | 206.8538025 | 7.692468 |  59.22758 |\n",
       "| CG1094868 | k | CG10948  |  0 |  4608 |    0.0000 | 204.9914719 | 7.679420 |  59.02741 |\n",
       "| pcs68 | k | pcs      |  0 |  4725 |    0.0000 | 193.7074763 | 7.597736 |  57.78203 |\n",
       "| Scamp68 | k | Scamp    |  0 |  4671 |    0.0000 | 184.6481637 | 7.528635 |  56.73901 |\n",
       "| mei-W6868 | k | mei-W68  |  0 |  4029 |    0.0000 | 182.2124950 | 7.509478 |  56.45156 |\n",
       "| CG980468 | k | CG9804   |  7 |   170 |  535.9008 |   6.2338415 | 6.211060 |  56.32507 |\n",
       "| Notum68 | k | Notum    |  0 |  4153 |    0.0000 | 180.3219519 | 7.494431 |  56.22629 |\n",
       "| AdamTS-A68 | k | AdamTS-A |  0 |  3680 |    0.0000 | 170.3103195 | 7.412022 |  55.00067 |\n",
       "| CG1534768 | k | CG15347  |  0 |  3586 |    0.0000 | 169.7153933 | 7.406974 |  54.92604 |\n",
       "\n"
      ],
      "text/plain": [
       "           cell.type.2 gene     cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "CG3195468  k           CG31954  28                   11       1255.6939      \n",
       "CG3444768  k           CG34447  27                    6       1141.3501      \n",
       "CG525568   k           CG5255   20                   13       1073.4703      \n",
       "CG401768   k           CG4017   12                    6        613.0370      \n",
       "fng68      k           fng       0                11165          0.0000      \n",
       "CG4333268  k           CG43332   0                 7448          0.0000      \n",
       "Nipped-A68 k           Nipped-A  0                 8224          0.0000      \n",
       "Ser99      k           Ser       0                 6688          0.0000      \n",
       "MYPT-75D68 k           MYPT-75D  0                 6518          0.0000      \n",
       "CG772068   k           CG7720    0                 4716          0.0000      \n",
       "Hmgcr68    k           Hmgcr     0                 4744          0.0000      \n",
       "CG588568   k           CG5885    0                 4492          0.0000      \n",
       "CG1094868  k           CG10948   0                 4608          0.0000      \n",
       "pcs68      k           pcs       0                 4725          0.0000      \n",
       "Scamp68    k           Scamp     0                 4671          0.0000      \n",
       "mei-W6868  k           mei-W68   0                 4029          0.0000      \n",
       "CG980468   k           CG9804    7                  170        535.9008      \n",
       "Notum68    k           Notum     0                 4153          0.0000      \n",
       "AdamTS-A68 k           AdamTS-A  0                 3680          0.0000      \n",
       "CG1534768  k           CG15347   0                 3586          0.0000      \n",
       "           other.tpm   log2.ratio heuristic.score\n",
       "CG3195468    0.3485554 9.862854   101.54220      \n",
       "CG3444768    0.1320903 9.977537   101.34971      \n",
       "CG525568     0.4351591 9.546856    96.13120      \n",
       "CG401768     0.2387753 8.950916    82.90501      \n",
       "fng68      524.1208112 9.033756    81.63358      \n",
       "CG4333268  354.2582008 8.468657    71.75260      \n",
       "Nipped-A68 322.5693846 8.333466    69.48386      \n",
       "Ser99      304.5744894 8.250651    68.11226      \n",
       "MYPT-75D68 291.5150520 8.187427    67.07440      \n",
       "CG772068   222.6207734 7.798444    60.86616      \n",
       "Hmgcr68    220.4553360 7.784343    60.64682      \n",
       "CG588568   206.8538025 7.692468    59.22758      \n",
       "CG1094868  204.9914719 7.679420    59.02741      \n",
       "pcs68      193.7074763 7.597736    57.78203      \n",
       "Scamp68    184.6481637 7.528635    56.73901      \n",
       "mei-W6868  182.2124950 7.509478    56.45156      \n",
       "CG980468     6.2338415 6.211060    56.32507      \n",
       "Notum68    180.3219519 7.494431    56.22629      \n",
       "AdamTS-A68 170.3103195 7.412022    55.00067      \n",
       "CG1534768  169.7153933 7.406974    54.92604      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"k\")  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG31954\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG31954\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG31954\"), \n",
    "                   size_pos = 4, size_neg = 0.05, stroke = 0.008, limit = 1)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG34447\"), \n",
    "                   size_pos = 3, size_neg = 0.1, stroke = 0.02, limit = 1)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG5255\"), \n",
    "                   size_pos = 3, size_neg = 0.2, stroke = 0.02, limit = 0.01)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 894,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Amnioserosa\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Amnioserosa\", df_sub300$cluster %in% c(l, m, n))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'l' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'l' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(l, m,n)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 868,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG1201169</th><td>l</td><td>CG12011  </td><td>45157</td><td>23686</td><td>8.132876e+04</td><td> 938.912102</td><td>6.435096</td><td>104.96604</td></tr>\n",
       "\t<tr><th scope=row>CG4333269</th><td>l</td><td>CG43332  </td><td> 9720</td><td> 4780</td><td>1.654825e+04</td><td> 208.926590</td><td>6.300650</td><td> 88.30033</td></tr>\n",
       "\t<tr><th scope=row>CG1808869</th><td>l</td><td>CG18088  </td><td> 2885</td><td> 1749</td><td>4.975342e+03</td><td>  73.930836</td><td>6.053092</td><td> 74.33724</td></tr>\n",
       "\t<tr><th scope=row>CG799769</th><td>l</td><td>CG7997   </td><td> 2571</td><td> 2881</td><td>4.458669e+03</td><td> 121.222249</td><td>5.189034</td><td> 62.90521</td></tr>\n",
       "\t<tr><th scope=row>Muc30E69</th><td>l</td><td>Muc30E   </td><td> 3517</td><td> 4295</td><td>5.627461e+03</td><td> 184.366425</td><td>4.924032</td><td> 61.34618</td></tr>\n",
       "\t<tr><th scope=row>CG1573169</th><td>l</td><td>CG15731  </td><td>    0</td><td> 5739</td><td>0.000000e+00</td><td> 193.252562</td><td>7.594344</td><td> 57.73060</td></tr>\n",
       "\t<tr><th scope=row>CG1373169</th><td>l</td><td>CG13731  </td><td>    0</td><td> 5381</td><td>0.000000e+00</td><td> 165.166416</td><td>7.367777</td><td> 54.34829</td></tr>\n",
       "\t<tr><th scope=row>CG296269</th><td>l</td><td>CG2962   </td><td>   16</td><td>31237</td><td>2.663353e+01</td><td>1109.593103</td><td>5.327467</td><td> 53.89860</td></tr>\n",
       "\t<tr><th scope=row>Osi1569</th><td>l</td><td>Osi15    </td><td>   10</td><td>20217</td><td>1.856681e+01</td><td> 804.191644</td><td>5.361059</td><td> 51.75131</td></tr>\n",
       "\t<tr><th scope=row>Vajk469</th><td>l</td><td>Vajk4    </td><td>    0</td><td> 4186</td><td>0.000000e+00</td><td> 141.762728</td><td>7.147334</td><td> 51.15687</td></tr>\n",
       "\t<tr><th scope=row>CG3422469</th><td>l</td><td>CG34224  </td><td>  302</td><td>  226</td><td>5.035308e+02</td><td>   9.327145</td><td>5.607567</td><td> 50.34921</td></tr>\n",
       "\t<tr><th scope=row>CG908369</th><td>l</td><td>CG9083   </td><td>    5</td><td>11867</td><td>7.649006e+00</td><td> 440.938397</td><td>5.671899</td><td> 49.84296</td></tr>\n",
       "\t<tr><th scope=row>MFS1569</th><td>l</td><td>MFS15    </td><td>  942</td><td> 1688</td><td>1.656653e+03</td><td>  67.486499</td><td>4.596308</td><td> 49.15717</td></tr>\n",
       "\t<tr><th scope=row>Shab69</th><td>l</td><td>Shab     </td><td>    1</td><td> 4924</td><td>2.001786e+00</td><td> 225.976174</td><td>6.234206</td><td> 48.79137</td></tr>\n",
       "\t<tr><th scope=row>TwdlF69</th><td>l</td><td>TwdlF    </td><td>    6</td><td>14661</td><td>1.221650e+01</td><td> 532.491380</td><td>5.332346</td><td> 48.30743</td></tr>\n",
       "\t<tr><th scope=row>TwdlD69</th><td>l</td><td>TwdlD    </td><td>    0</td><td> 3412</td><td>0.000000e+00</td><td> 120.150708</td><td>6.908701</td><td> 47.81277</td></tr>\n",
       "\t<tr><th scope=row>C1GalTA69</th><td>l</td><td>C1GalTA  </td><td> 1450</td><td> 2966</td><td>2.381000e+03</td><td> 126.180297</td><td>4.226620</td><td> 47.41405</td></tr>\n",
       "\t<tr><th scope=row>Muc91C69</th><td>l</td><td>Muc91C   </td><td>    0</td><td> 2883</td><td>0.000000e+00</td><td> 108.698048</td><td>6.764182</td><td> 45.84353</td></tr>\n",
       "\t<tr><th scope=row>l(2)0128969</th><td>l</td><td>l(2)01289</td><td>    1</td><td> 3601</td><td>5.577027e-01</td><td> 135.006864</td><td>6.437469</td><td> 45.62579</td></tr>\n",
       "\t<tr><th scope=row>CG1425769</th><td>l</td><td>CG14257  </td><td>    0</td><td> 2497</td><td>0.000000e+00</td><td> 107.191797</td><td>6.744051</td><td> 45.57257</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG1201169 & l & CG12011   & 45157 & 23686 & 8.132876e+04 &  938.912102 & 6.435096 & 104.96604\\\\\n",
       "\tCG4333269 & l & CG43332   &  9720 &  4780 & 1.654825e+04 &  208.926590 & 6.300650 &  88.30033\\\\\n",
       "\tCG1808869 & l & CG18088   &  2885 &  1749 & 4.975342e+03 &   73.930836 & 6.053092 &  74.33724\\\\\n",
       "\tCG799769 & l & CG7997    &  2571 &  2881 & 4.458669e+03 &  121.222249 & 5.189034 &  62.90521\\\\\n",
       "\tMuc30E69 & l & Muc30E    &  3517 &  4295 & 5.627461e+03 &  184.366425 & 4.924032 &  61.34618\\\\\n",
       "\tCG1573169 & l & CG15731   &     0 &  5739 & 0.000000e+00 &  193.252562 & 7.594344 &  57.73060\\\\\n",
       "\tCG1373169 & l & CG13731   &     0 &  5381 & 0.000000e+00 &  165.166416 & 7.367777 &  54.34829\\\\\n",
       "\tCG296269 & l & CG2962    &    16 & 31237 & 2.663353e+01 & 1109.593103 & 5.327467 &  53.89860\\\\\n",
       "\tOsi1569 & l & Osi15     &    10 & 20217 & 1.856681e+01 &  804.191644 & 5.361059 &  51.75131\\\\\n",
       "\tVajk469 & l & Vajk4     &     0 &  4186 & 0.000000e+00 &  141.762728 & 7.147334 &  51.15687\\\\\n",
       "\tCG3422469 & l & CG34224   &   302 &   226 & 5.035308e+02 &    9.327145 & 5.607567 &  50.34921\\\\\n",
       "\tCG908369 & l & CG9083    &     5 & 11867 & 7.649006e+00 &  440.938397 & 5.671899 &  49.84296\\\\\n",
       "\tMFS1569 & l & MFS15     &   942 &  1688 & 1.656653e+03 &   67.486499 & 4.596308 &  49.15717\\\\\n",
       "\tShab69 & l & Shab      &     1 &  4924 & 2.001786e+00 &  225.976174 & 6.234206 &  48.79137\\\\\n",
       "\tTwdlF69 & l & TwdlF     &     6 & 14661 & 1.221650e+01 &  532.491380 & 5.332346 &  48.30743\\\\\n",
       "\tTwdlD69 & l & TwdlD     &     0 &  3412 & 0.000000e+00 &  120.150708 & 6.908701 &  47.81277\\\\\n",
       "\tC1GalTA69 & l & C1GalTA   &  1450 &  2966 & 2.381000e+03 &  126.180297 & 4.226620 &  47.41405\\\\\n",
       "\tMuc91C69 & l & Muc91C    &     0 &  2883 & 0.000000e+00 &  108.698048 & 6.764182 &  45.84353\\\\\n",
       "\tl(2)0128969 & l & l(2)01289 &     1 &  3601 & 5.577027e-01 &  135.006864 & 6.437469 &  45.62579\\\\\n",
       "\tCG1425769 & l & CG14257   &     0 &  2497 & 0.000000e+00 &  107.191797 & 6.744051 &  45.57257\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG1201169 | l | CG12011   | 45157 | 23686 | 8.132876e+04 |  938.912102 | 6.435096 | 104.96604 |\n",
       "| CG4333269 | l | CG43332   |  9720 |  4780 | 1.654825e+04 |  208.926590 | 6.300650 |  88.30033 |\n",
       "| CG1808869 | l | CG18088   |  2885 |  1749 | 4.975342e+03 |   73.930836 | 6.053092 |  74.33724 |\n",
       "| CG799769 | l | CG7997    |  2571 |  2881 | 4.458669e+03 |  121.222249 | 5.189034 |  62.90521 |\n",
       "| Muc30E69 | l | Muc30E    |  3517 |  4295 | 5.627461e+03 |  184.366425 | 4.924032 |  61.34618 |\n",
       "| CG1573169 | l | CG15731   |     0 |  5739 | 0.000000e+00 |  193.252562 | 7.594344 |  57.73060 |\n",
       "| CG1373169 | l | CG13731   |     0 |  5381 | 0.000000e+00 |  165.166416 | 7.367777 |  54.34829 |\n",
       "| CG296269 | l | CG2962    |    16 | 31237 | 2.663353e+01 | 1109.593103 | 5.327467 |  53.89860 |\n",
       "| Osi1569 | l | Osi15     |    10 | 20217 | 1.856681e+01 |  804.191644 | 5.361059 |  51.75131 |\n",
       "| Vajk469 | l | Vajk4     |     0 |  4186 | 0.000000e+00 |  141.762728 | 7.147334 |  51.15687 |\n",
       "| CG3422469 | l | CG34224   |   302 |   226 | 5.035308e+02 |    9.327145 | 5.607567 |  50.34921 |\n",
       "| CG908369 | l | CG9083    |     5 | 11867 | 7.649006e+00 |  440.938397 | 5.671899 |  49.84296 |\n",
       "| MFS1569 | l | MFS15     |   942 |  1688 | 1.656653e+03 |   67.486499 | 4.596308 |  49.15717 |\n",
       "| Shab69 | l | Shab      |     1 |  4924 | 2.001786e+00 |  225.976174 | 6.234206 |  48.79137 |\n",
       "| TwdlF69 | l | TwdlF     |     6 | 14661 | 1.221650e+01 |  532.491380 | 5.332346 |  48.30743 |\n",
       "| TwdlD69 | l | TwdlD     |     0 |  3412 | 0.000000e+00 |  120.150708 | 6.908701 |  47.81277 |\n",
       "| C1GalTA69 | l | C1GalTA   |  1450 |  2966 | 2.381000e+03 |  126.180297 | 4.226620 |  47.41405 |\n",
       "| Muc91C69 | l | Muc91C    |     0 |  2883 | 0.000000e+00 |  108.698048 | 6.764182 |  45.84353 |\n",
       "| l(2)0128969 | l | l(2)01289 |     1 |  3601 | 5.577027e-01 |  135.006864 | 6.437469 |  45.62579 |\n",
       "| CG1425769 | l | CG14257   |     0 |  2497 | 0.000000e+00 |  107.191797 | 6.744051 |  45.57257 |\n",
       "\n"
      ],
      "text/plain": [
       "            cell.type.2 gene      cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "CG1201169   l           CG12011   45157             23686       8.132876e+04   \n",
       "CG4333269   l           CG43332    9720              4780       1.654825e+04   \n",
       "CG1808869   l           CG18088    2885              1749       4.975342e+03   \n",
       "CG799769    l           CG7997     2571              2881       4.458669e+03   \n",
       "Muc30E69    l           Muc30E     3517              4295       5.627461e+03   \n",
       "CG1573169   l           CG15731       0              5739       0.000000e+00   \n",
       "CG1373169   l           CG13731       0              5381       0.000000e+00   \n",
       "CG296269    l           CG2962       16             31237       2.663353e+01   \n",
       "Osi1569     l           Osi15        10             20217       1.856681e+01   \n",
       "Vajk469     l           Vajk4         0              4186       0.000000e+00   \n",
       "CG3422469   l           CG34224     302               226       5.035308e+02   \n",
       "CG908369    l           CG9083        5             11867       7.649006e+00   \n",
       "MFS1569     l           MFS15       942              1688       1.656653e+03   \n",
       "Shab69      l           Shab          1              4924       2.001786e+00   \n",
       "TwdlF69     l           TwdlF         6             14661       1.221650e+01   \n",
       "TwdlD69     l           TwdlD         0              3412       0.000000e+00   \n",
       "C1GalTA69   l           C1GalTA    1450              2966       2.381000e+03   \n",
       "Muc91C69    l           Muc91C        0              2883       0.000000e+00   \n",
       "l(2)0128969 l           l(2)01289     1              3601       5.577027e-01   \n",
       "CG1425769   l           CG14257       0              2497       0.000000e+00   \n",
       "            other.tpm   log2.ratio heuristic.score\n",
       "CG1201169    938.912102 6.435096   104.96604      \n",
       "CG4333269    208.926590 6.300650    88.30033      \n",
       "CG1808869     73.930836 6.053092    74.33724      \n",
       "CG799769     121.222249 5.189034    62.90521      \n",
       "Muc30E69     184.366425 4.924032    61.34618      \n",
       "CG1573169    193.252562 7.594344    57.73060      \n",
       "CG1373169    165.166416 7.367777    54.34829      \n",
       "CG296269    1109.593103 5.327467    53.89860      \n",
       "Osi1569      804.191644 5.361059    51.75131      \n",
       "Vajk469      141.762728 7.147334    51.15687      \n",
       "CG3422469      9.327145 5.607567    50.34921      \n",
       "CG908369     440.938397 5.671899    49.84296      \n",
       "MFS1569       67.486499 4.596308    49.15717      \n",
       "Shab69       225.976174 6.234206    48.79137      \n",
       "TwdlF69      532.491380 5.332346    48.30743      \n",
       "TwdlD69      120.150708 6.908701    47.81277      \n",
       "C1GalTA69    126.180297 4.226620    47.41405      \n",
       "Muc91C69     108.698048 6.764182    45.84353      \n",
       "l(2)0128969  135.006864 6.437469    45.62579      \n",
       "CG1425769    107.191797 6.744051    45.57257      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"l\")  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG43332\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG43332\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG43332\"), \n",
    "                   size_pos = 4, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG18088\"), \n",
    "                   size_pos = 3, size_neg = 0.1, stroke = 0.02, limit = 2)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG7997\"), \n",
    "                   size_pos = 3, size_neg = 0.2, stroke = 0.02, limit = 2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 883,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sub300_no_doubletons = make_df(cds_sub_no_doubletons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 888,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 1562\n",
      "\n",
      "# of cells sampled in set 2: 2664\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>fipi</th><td>fipi     </td><td>  177</td><td> 4823</td><td>  282.94543</td><td> 4931.2569</td><td>Set 2</td><td>4.118270</td><td>50.52307</td></tr>\n",
       "\t<tr><th scope=row>CG43861</th><td>CG43861  </td><td>   15</td><td>  548</td><td>   22.67978</td><td>  551.8351</td><td>Set 2</td><td>4.542509</td><td>41.38547</td></tr>\n",
       "\t<tr><th scope=row>Orcokinin</th><td>Orcokinin</td><td>    7</td><td>  307</td><td>   12.35747</td><td>  353.6630</td><td>Set 2</td><td>4.726656</td><td>40.03622</td></tr>\n",
       "\t<tr><th scope=row>CG31145</th><td>CG31145  </td><td>   79</td><td> 1430</td><td>  108.17657</td><td> 1438.6929</td><td>Set 2</td><td>3.720023</td><td>39.02879</td></tr>\n",
       "\t<tr><th scope=row>a</th><td>a        </td><td>   45</td><td>  958</td><td>   76.79683</td><td> 1008.1222</td><td>Set 2</td><td>3.695815</td><td>36.88012</td></tr>\n",
       "\t<tr><th scope=row>CG12011</th><td>CG12011  </td><td>44881</td><td>18632</td><td>80845.62026</td><td>18782.3481</td><td>Set 1</td><td>2.105715</td><td>34.32927</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tfipi & fipi      &   177 &  4823 &   282.94543 &  4931.2569 & Set 2 & 4.118270 & 50.52307\\\\\n",
       "\tCG43861 & CG43861   &    15 &   548 &    22.67978 &   551.8351 & Set 2 & 4.542509 & 41.38547\\\\\n",
       "\tOrcokinin & Orcokinin &     7 &   307 &    12.35747 &   353.6630 & Set 2 & 4.726656 & 40.03622\\\\\n",
       "\tCG31145 & CG31145   &    79 &  1430 &   108.17657 &  1438.6929 & Set 2 & 3.720023 & 39.02879\\\\\n",
       "\ta & a         &    45 &   958 &    76.79683 &  1008.1222 & Set 2 & 3.695815 & 36.88012\\\\\n",
       "\tCG12011 & CG12011   & 44881 & 18632 & 80845.62026 & 18782.3481 & Set 1 & 2.105715 & 34.32927\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| fipi | fipi      |   177 |  4823 |   282.94543 |  4931.2569 | Set 2 | 4.118270 | 50.52307 |\n",
       "| CG43861 | CG43861   |    15 |   548 |    22.67978 |   551.8351 | Set 2 | 4.542509 | 41.38547 |\n",
       "| Orcokinin | Orcokinin |     7 |   307 |    12.35747 |   353.6630 | Set 2 | 4.726656 | 40.03622 |\n",
       "| CG31145 | CG31145   |    79 |  1430 |   108.17657 |  1438.6929 | Set 2 | 3.720023 | 39.02879 |\n",
       "| a | a         |    45 |   958 |    76.79683 |  1008.1222 | Set 2 | 3.695815 | 36.88012 |\n",
       "| CG12011 | CG12011   | 44881 | 18632 | 80845.62026 | 18782.3481 | Set 1 | 2.105715 | 34.32927 |\n",
       "\n"
      ],
      "text/plain": [
       "          gene      set.1.n.umi set.2.n.umi set.1.tpm   set.2.tpm  higher.expr\n",
       "fipi      fipi        177        4823         282.94543  4931.2569 Set 2      \n",
       "CG43861   CG43861      15         548          22.67978   551.8351 Set 2      \n",
       "Orcokinin Orcokinin     7         307          12.35747   353.6630 Set 2      \n",
       "CG31145   CG31145      79        1430         108.17657  1438.6929 Set 2      \n",
       "a         a            45         958          76.79683  1008.1222 Set 2      \n",
       "CG12011   CG12011   44881       18632       80845.62026 18782.3481 Set 1      \n",
       "          log2.ratio heuristic.score\n",
       "fipi      4.118270   50.52307       \n",
       "CG43861   4.542509   41.38547       \n",
       "Orcokinin 4.726656   40.03622       \n",
       "CG31145   3.720023   39.02879       \n",
       "a         3.695815   36.88012       \n",
       "CG12011   2.105715   34.32927       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "two.set.differential.gene.test.3a(cds_sub_no_doubletons, df_sub300_no_doubletons$cell.type.2 == \"l\", \n",
    "                                 df_sub300_no_doubletons$cell.type.2 == \"m\")  %>% head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"fipi\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"fipi\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"fipi\"), \n",
    "                   size_pos = 4, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG43861\"), \n",
    "                   size_pos = 3, size_neg = 0.1, stroke = 0.02, limit = 2)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Orcokinin\"), \n",
    "                   size_pos = 3, size_neg = 0.2, stroke = 0.02, limit = 2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### l cluster not big enough differences below to warrant separating them!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 889,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 1562\n",
      "\n",
      "# of cells sampled in set 2: 2664\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG12011</th><td>CG12011</td><td>44881</td><td>18632</td><td>80845.62026</td><td>18782.34808</td><td>Set 1</td><td>2.105715</td><td>34.32927</td></tr>\n",
       "\t<tr><th scope=row>CG6753</th><td>CG6753 </td><td>   77</td><td>   14</td><td>  142.72037</td><td>   10.01414</td><td>Set 1</td><td>3.695763</td><td>26.48798</td></tr>\n",
       "\t<tr><th scope=row>Sr-CII</th><td>Sr-CII </td><td>   17</td><td>    0</td><td>   33.16107</td><td>    0.00000</td><td>Set 1</td><td>5.051419</td><td>25.73335</td></tr>\n",
       "\t<tr><th scope=row>bnb</th><td>bnb    </td><td> 5124</td><td> 2795</td><td> 8638.59911</td><td> 2572.09225</td><td>Set 1</td><td>1.747294</td><td>22.84893</td></tr>\n",
       "\t<tr><th scope=row>CG18417</th><td>CG18417</td><td>   26</td><td>    2</td><td>   46.44776</td><td>    1.92734</td><td>Set 1</td><td>3.987947</td><td>22.20596</td></tr>\n",
       "\t<tr><th scope=row>link</th><td>link   </td><td>  271</td><td>   86</td><td>  421.45603</td><td>   71.55207</td><td>Set 1</td><td>2.538293</td><td>22.14066</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG12011 & CG12011 & 44881 & 18632 & 80845.62026 & 18782.34808 & Set 1 & 2.105715 & 34.32927\\\\\n",
       "\tCG6753 & CG6753  &    77 &    14 &   142.72037 &    10.01414 & Set 1 & 3.695763 & 26.48798\\\\\n",
       "\tSr-CII & Sr-CII  &    17 &     0 &    33.16107 &     0.00000 & Set 1 & 5.051419 & 25.73335\\\\\n",
       "\tbnb & bnb     &  5124 &  2795 &  8638.59911 &  2572.09225 & Set 1 & 1.747294 & 22.84893\\\\\n",
       "\tCG18417 & CG18417 &    26 &     2 &    46.44776 &     1.92734 & Set 1 & 3.987947 & 22.20596\\\\\n",
       "\tlink & link    &   271 &    86 &   421.45603 &    71.55207 & Set 1 & 2.538293 & 22.14066\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG12011 | CG12011 | 44881 | 18632 | 80845.62026 | 18782.34808 | Set 1 | 2.105715 | 34.32927 |\n",
       "| CG6753 | CG6753  |    77 |    14 |   142.72037 |    10.01414 | Set 1 | 3.695763 | 26.48798 |\n",
       "| Sr-CII | Sr-CII  |    17 |     0 |    33.16107 |     0.00000 | Set 1 | 5.051419 | 25.73335 |\n",
       "| bnb | bnb     |  5124 |  2795 |  8638.59911 |  2572.09225 | Set 1 | 1.747294 | 22.84893 |\n",
       "| CG18417 | CG18417 |    26 |     2 |    46.44776 |     1.92734 | Set 1 | 3.987947 | 22.20596 |\n",
       "| link | link    |   271 |    86 |   421.45603 |    71.55207 | Set 1 | 2.538293 | 22.14066 |\n",
       "\n"
      ],
      "text/plain": [
       "        gene    set.1.n.umi set.2.n.umi set.1.tpm   set.2.tpm   higher.expr\n",
       "CG12011 CG12011 44881       18632       80845.62026 18782.34808 Set 1      \n",
       "CG6753  CG6753     77          14         142.72037    10.01414 Set 1      \n",
       "Sr-CII  Sr-CII     17           0          33.16107     0.00000 Set 1      \n",
       "bnb     bnb      5124        2795        8638.59911  2572.09225 Set 1      \n",
       "CG18417 CG18417    26           2          46.44776     1.92734 Set 1      \n",
       "link    link      271          86         421.45603    71.55207 Set 1      \n",
       "        log2.ratio heuristic.score\n",
       "CG12011 2.105715   34.32927       \n",
       "CG6753  3.695763   26.48798       \n",
       "Sr-CII  5.051419   25.73335       \n",
       "bnb     1.747294   22.84893       \n",
       "CG18417 3.987947   22.20596       \n",
       "link    2.538293   22.14066       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "two.set.differential.gene.test.3a(cds_sub_no_doubletons, df_sub300_no_doubletons$cell.type.2 == \"l\", \n",
    "                                 df_sub300_no_doubletons$cell.type.2 == \"m\")  %>% \n",
    "filter(higher.expr == \"Set 1\")  %>% head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG6753\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG6753\"), size_pos = 4, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG6753\"), \n",
    "                   size_pos = 4, size_neg = 0.05, stroke = 0.008, limit = 1)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Sr-CII\"), \n",
    "                   size_pos = 3, size_neg = 0.1, stroke = 0.02, limit = 2)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"link\"), \n",
    "                   size_pos = 3, size_neg = 0.2, stroke = 0.02, limit = 2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"m\", df_sub300$cluster %in% m)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"n\", df_sub300$cluster %in% n)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 899,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Proventriculus\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Proventriculus\", df_sub300$cluster %in% o)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'o' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'o' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(o)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 896,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG4389672</th><td>o</td><td>CG43896</td><td>5171</td><td> 116</td><td>2550.7945</td><td>  4.2726933</td><td>8.918191</td><td>100.92981</td></tr>\n",
       "\t<tr><th scope=row>CG771572</th><td>o</td><td>CG7715 </td><td>5047</td><td> 648</td><td>2729.6245</td><td> 20.1938883</td><td>7.008910</td><td> 80.00682</td></tr>\n",
       "\t<tr><th scope=row>CG1380672</th><td>o</td><td>CG13806</td><td>1300</td><td> 780</td><td>1851.1054</td><td> 30.5982426</td><td>5.872399</td><td> 63.74460</td></tr>\n",
       "\t<tr><th scope=row>Cda972</th><td>o</td><td>Cda9   </td><td> 523</td><td>  15</td><td> 268.3877</td><td>  0.5763477</td><td>7.411589</td><td> 59.83776</td></tr>\n",
       "\t<tr><th scope=row>obst-F72</th><td>o</td><td>obst-F </td><td> 412</td><td>  12</td><td> 206.2760</td><td>  0.4334067</td><td>7.168984</td><td> 55.16827</td></tr>\n",
       "\t<tr><th scope=row>CG1581872</th><td>o</td><td>CG15818</td><td> 665</td><td> 609</td><td> 991.2771</td><td> 25.0539270</td><td>5.249716</td><td> 52.25882</td></tr>\n",
       "\t<tr><th scope=row>Kaz-m172</th><td>o</td><td>Kaz-m1 </td><td> 443</td><td> 313</td><td> 620.6277</td><td> 13.0898354</td><td>5.461001</td><td> 50.67758</td></tr>\n",
       "\t<tr><th scope=row>Lim372</th><td>o</td><td>Lim3   </td><td>1878</td><td>2662</td><td>2678.8615</td><td>122.1671318</td><td>4.442931</td><td> 50.59584</td></tr>\n",
       "\t<tr><th scope=row>CG3041372</th><td>o</td><td>CG30413</td><td> 595</td><td> 492</td><td> 826.1740</td><td> 21.6057364</td><td>5.191685</td><td> 50.31805</td></tr>\n",
       "\t<tr><th scope=row>Lip172</th><td>o</td><td>Lip1   </td><td> 938</td><td>1198</td><td>1342.1341</td><td> 47.0035731</td><td>4.805243</td><td> 49.93314</td></tr>\n",
       "\t<tr><th scope=row>CG756772</th><td>o</td><td>CG7567 </td><td> 359</td><td>  31</td><td> 183.5246</td><td>  1.5643612</td><td>6.161230</td><td> 46.37970</td></tr>\n",
       "\t<tr><th scope=row>ara72</th><td>o</td><td>ara    </td><td>1527</td><td>3007</td><td>2164.3578</td><td>129.0005879</td><td>4.057349</td><td> 44.95701</td></tr>\n",
       "\t<tr><th scope=row>CG1381072</th><td>o</td><td>CG13810</td><td> 193</td><td> 129</td><td> 267.9930</td><td>  4.6667940</td><td>5.563519</td><td> 44.90552</td></tr>\n",
       "\t<tr><th scope=row>CG799772</th><td>o</td><td>CG7997 </td><td>1276</td><td>3042</td><td>1813.3591</td><td>128.8836535</td><td>3.803373</td><td> 41.17244</td></tr>\n",
       "\t<tr><th scope=row>caup72</th><td>o</td><td>caup   </td><td>1285</td><td>3264</td><td>1816.0627</td><td>132.2666722</td><td>3.768426</td><td> 40.80223</td></tr>\n",
       "\t<tr><th scope=row>dve72</th><td>o</td><td>dve    </td><td>1376</td><td>3417</td><td>1889.5560</td><td>145.6833299</td><td>3.687270</td><td> 40.13445</td></tr>\n",
       "\t<tr><th scope=row>CG1704772</th><td>o</td><td>CG17047</td><td> 433</td><td> 767</td><td> 619.5877</td><td> 30.5675669</td><td>4.294794</td><td> 39.84491</td></tr>\n",
       "\t<tr><th scope=row>tgy72</th><td>o</td><td>tgy    </td><td> 142</td><td> 116</td><td> 196.3252</td><td>  4.3787627</td><td>5.189827</td><td> 39.56948</td></tr>\n",
       "\t<tr><th scope=row>CG892572</th><td>o</td><td>CG8925 </td><td> 179</td><td> 162</td><td> 252.6631</td><td>  7.3400917</td><td>4.921008</td><td> 39.30296</td></tr>\n",
       "\t<tr><th scope=row>al72</th><td>o</td><td>al     </td><td> 504</td><td> 953</td><td> 724.0565</td><td> 42.0602668</td><td>4.071673</td><td> 38.68883</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG4389672 & o & CG43896 & 5171 &  116 & 2550.7945 &   4.2726933 & 8.918191 & 100.92981\\\\\n",
       "\tCG771572 & o & CG7715  & 5047 &  648 & 2729.6245 &  20.1938883 & 7.008910 &  80.00682\\\\\n",
       "\tCG1380672 & o & CG13806 & 1300 &  780 & 1851.1054 &  30.5982426 & 5.872399 &  63.74460\\\\\n",
       "\tCda972 & o & Cda9    &  523 &   15 &  268.3877 &   0.5763477 & 7.411589 &  59.83776\\\\\n",
       "\tobst-F72 & o & obst-F  &  412 &   12 &  206.2760 &   0.4334067 & 7.168984 &  55.16827\\\\\n",
       "\tCG1581872 & o & CG15818 &  665 &  609 &  991.2771 &  25.0539270 & 5.249716 &  52.25882\\\\\n",
       "\tKaz-m172 & o & Kaz-m1  &  443 &  313 &  620.6277 &  13.0898354 & 5.461001 &  50.67758\\\\\n",
       "\tLim372 & o & Lim3    & 1878 & 2662 & 2678.8615 & 122.1671318 & 4.442931 &  50.59584\\\\\n",
       "\tCG3041372 & o & CG30413 &  595 &  492 &  826.1740 &  21.6057364 & 5.191685 &  50.31805\\\\\n",
       "\tLip172 & o & Lip1    &  938 & 1198 & 1342.1341 &  47.0035731 & 4.805243 &  49.93314\\\\\n",
       "\tCG756772 & o & CG7567  &  359 &   31 &  183.5246 &   1.5643612 & 6.161230 &  46.37970\\\\\n",
       "\tara72 & o & ara     & 1527 & 3007 & 2164.3578 & 129.0005879 & 4.057349 &  44.95701\\\\\n",
       "\tCG1381072 & o & CG13810 &  193 &  129 &  267.9930 &   4.6667940 & 5.563519 &  44.90552\\\\\n",
       "\tCG799772 & o & CG7997  & 1276 & 3042 & 1813.3591 & 128.8836535 & 3.803373 &  41.17244\\\\\n",
       "\tcaup72 & o & caup    & 1285 & 3264 & 1816.0627 & 132.2666722 & 3.768426 &  40.80223\\\\\n",
       "\tdve72 & o & dve     & 1376 & 3417 & 1889.5560 & 145.6833299 & 3.687270 &  40.13445\\\\\n",
       "\tCG1704772 & o & CG17047 &  433 &  767 &  619.5877 &  30.5675669 & 4.294794 &  39.84491\\\\\n",
       "\ttgy72 & o & tgy     &  142 &  116 &  196.3252 &   4.3787627 & 5.189827 &  39.56948\\\\\n",
       "\tCG892572 & o & CG8925  &  179 &  162 &  252.6631 &   7.3400917 & 4.921008 &  39.30296\\\\\n",
       "\tal72 & o & al      &  504 &  953 &  724.0565 &  42.0602668 & 4.071673 &  38.68883\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG4389672 | o | CG43896 | 5171 |  116 | 2550.7945 |   4.2726933 | 8.918191 | 100.92981 |\n",
       "| CG771572 | o | CG7715  | 5047 |  648 | 2729.6245 |  20.1938883 | 7.008910 |  80.00682 |\n",
       "| CG1380672 | o | CG13806 | 1300 |  780 | 1851.1054 |  30.5982426 | 5.872399 |  63.74460 |\n",
       "| Cda972 | o | Cda9    |  523 |   15 |  268.3877 |   0.5763477 | 7.411589 |  59.83776 |\n",
       "| obst-F72 | o | obst-F  |  412 |   12 |  206.2760 |   0.4334067 | 7.168984 |  55.16827 |\n",
       "| CG1581872 | o | CG15818 |  665 |  609 |  991.2771 |  25.0539270 | 5.249716 |  52.25882 |\n",
       "| Kaz-m172 | o | Kaz-m1  |  443 |  313 |  620.6277 |  13.0898354 | 5.461001 |  50.67758 |\n",
       "| Lim372 | o | Lim3    | 1878 | 2662 | 2678.8615 | 122.1671318 | 4.442931 |  50.59584 |\n",
       "| CG3041372 | o | CG30413 |  595 |  492 |  826.1740 |  21.6057364 | 5.191685 |  50.31805 |\n",
       "| Lip172 | o | Lip1    |  938 | 1198 | 1342.1341 |  47.0035731 | 4.805243 |  49.93314 |\n",
       "| CG756772 | o | CG7567  |  359 |   31 |  183.5246 |   1.5643612 | 6.161230 |  46.37970 |\n",
       "| ara72 | o | ara     | 1527 | 3007 | 2164.3578 | 129.0005879 | 4.057349 |  44.95701 |\n",
       "| CG1381072 | o | CG13810 |  193 |  129 |  267.9930 |   4.6667940 | 5.563519 |  44.90552 |\n",
       "| CG799772 | o | CG7997  | 1276 | 3042 | 1813.3591 | 128.8836535 | 3.803373 |  41.17244 |\n",
       "| caup72 | o | caup    | 1285 | 3264 | 1816.0627 | 132.2666722 | 3.768426 |  40.80223 |\n",
       "| dve72 | o | dve     | 1376 | 3417 | 1889.5560 | 145.6833299 | 3.687270 |  40.13445 |\n",
       "| CG1704772 | o | CG17047 |  433 |  767 |  619.5877 |  30.5675669 | 4.294794 |  39.84491 |\n",
       "| tgy72 | o | tgy     |  142 |  116 |  196.3252 |   4.3787627 | 5.189827 |  39.56948 |\n",
       "| CG892572 | o | CG8925  |  179 |  162 |  252.6631 |   7.3400917 | 4.921008 |  39.30296 |\n",
       "| al72 | o | al      |  504 |  953 |  724.0565 |  42.0602668 | 4.071673 |  38.68883 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "CG4389672 o           CG43896 5171               116        2550.7945      \n",
       "CG771572  o           CG7715  5047               648        2729.6245      \n",
       "CG1380672 o           CG13806 1300               780        1851.1054      \n",
       "Cda972    o           Cda9     523                15         268.3877      \n",
       "obst-F72  o           obst-F   412                12         206.2760      \n",
       "CG1581872 o           CG15818  665               609         991.2771      \n",
       "Kaz-m172  o           Kaz-m1   443               313         620.6277      \n",
       "Lim372    o           Lim3    1878              2662        2678.8615      \n",
       "CG3041372 o           CG30413  595               492         826.1740      \n",
       "Lip172    o           Lip1     938              1198        1342.1341      \n",
       "CG756772  o           CG7567   359                31         183.5246      \n",
       "ara72     o           ara     1527              3007        2164.3578      \n",
       "CG1381072 o           CG13810  193               129         267.9930      \n",
       "CG799772  o           CG7997  1276              3042        1813.3591      \n",
       "caup72    o           caup    1285              3264        1816.0627      \n",
       "dve72     o           dve     1376              3417        1889.5560      \n",
       "CG1704772 o           CG17047  433               767         619.5877      \n",
       "tgy72     o           tgy      142               116         196.3252      \n",
       "CG892572  o           CG8925   179               162         252.6631      \n",
       "al72      o           al       504               953         724.0565      \n",
       "          other.tpm   log2.ratio heuristic.score\n",
       "CG4389672   4.2726933 8.918191   100.92981      \n",
       "CG771572   20.1938883 7.008910    80.00682      \n",
       "CG1380672  30.5982426 5.872399    63.74460      \n",
       "Cda972      0.5763477 7.411589    59.83776      \n",
       "obst-F72    0.4334067 7.168984    55.16827      \n",
       "CG1581872  25.0539270 5.249716    52.25882      \n",
       "Kaz-m172   13.0898354 5.461001    50.67758      \n",
       "Lim372    122.1671318 4.442931    50.59584      \n",
       "CG3041372  21.6057364 5.191685    50.31805      \n",
       "Lip172     47.0035731 4.805243    49.93314      \n",
       "CG756772    1.5643612 6.161230    46.37970      \n",
       "ara72     129.0005879 4.057349    44.95701      \n",
       "CG1381072   4.6667940 5.563519    44.90552      \n",
       "CG799772  128.8836535 3.803373    41.17244      \n",
       "caup72    132.2666722 3.768426    40.80223      \n",
       "dve72     145.6833299 3.687270    40.13445      \n",
       "CG1704772  30.5675669 4.294794    39.84491      \n",
       "tgy72       4.3787627 5.189827    39.56948      \n",
       "CG892572    7.3400917 4.921008    39.30296      \n",
       "al72       42.0602668 4.071673    38.68883      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"o\")  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG43896\"), size_pos = 3, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG43896\"), size_pos = 3, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG43896\"), \n",
    "                   size_pos = 3, size_neg = 0.05, stroke = 0.008, limit = 1)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG7715\"), \n",
    "                   size_pos = 3, size_neg = 0.1, stroke = 0.02, limit = 3)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Cda9\"), \n",
    "                   size_pos = 3, size_neg = 0.2, stroke = 0.02, limit = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 905,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Peptidergic_neuron\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Peptidergic_neuron\", df_sub300$cluster %in% p) \n",
    "# from neuron subset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'p' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'p' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(p)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 900,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG457773</th><td>p</td><td>CG4577 </td><td>146</td><td>  117</td><td> 2994.3931</td><td>  5.1710612</td><td>8.922529</td><td>103.04209</td></tr>\n",
       "\t<tr><th scope=row>CG4254173</th><td>p</td><td>CG42541</td><td>318</td><td> 1414</td><td> 7257.9265</td><td> 61.4327884</td><td>6.861110</td><td> 87.99744</td></tr>\n",
       "\t<tr><th scope=row>AstC73</th><td>p</td><td>AstC   </td><td> 62</td><td>   42</td><td>  667.1167</td><td>  0.8393817</td><td>8.502574</td><td> 79.78779</td></tr>\n",
       "\t<tr><th scope=row>CG4333273</th><td>p</td><td>CG43332</td><td>  0</td><td> 7374</td><td>    0.0000</td><td>354.7325993</td><td>8.470588</td><td> 71.78526</td></tr>\n",
       "\t<tr><th scope=row>CG3438673</th><td>p</td><td>CG34386</td><td> 14</td><td>   16</td><td>  443.0765</td><td>  0.7154520</td><td>8.012823</td><td> 70.47009</td></tr>\n",
       "\t<tr><th scope=row>Nplp173</th><td>p</td><td>Nplp1  </td><td> 43</td><td>  236</td><td> 1266.2370</td><td> 10.5808130</td><td>6.772667</td><td> 69.80907</td></tr>\n",
       "\t<tr><th scope=row>Mip73</th><td>p</td><td>Mip    </td><td> 36</td><td>  223</td><td>  999.5280</td><td>  7.8848033</td><td>6.813763</td><td> 67.90968</td></tr>\n",
       "\t<tr><th scope=row>MsR173</th><td>p</td><td>MsR1   </td><td> 39</td><td>  165</td><td>  962.1831</td><td>  7.3577280</td><td>6.847057</td><td> 67.86574</td></tr>\n",
       "\t<tr><th scope=row>cas73</th><td>p</td><td>cas    </td><td>  0</td><td> 5896</td><td>    0.0000</td><td>272.4327174</td><td>8.089756</td><td> 65.48692</td></tr>\n",
       "\t<tr><th scope=row>CycB83</th><td>p</td><td>CycB   </td><td>  0</td><td> 7938</td><td>    0.0000</td><td>271.1318191</td><td>8.082851</td><td> 65.37540</td></tr>\n",
       "\t<tr><th scope=row>amon73</th><td>p</td><td>amon   </td><td> 52</td><td>  351</td><td> 1172.3573</td><td> 13.9914465</td><td>6.289129</td><td> 64.12664</td></tr>\n",
       "\t<tr><th scope=row>Ms73</th><td>p</td><td>Ms     </td><td> 12</td><td>   19</td><td>  330.0292</td><td>  0.7257723</td><td>7.579208</td><td> 63.44414</td></tr>\n",
       "\t<tr><th scope=row>otp73</th><td>p</td><td>otp    </td><td>  0</td><td> 5092</td><td>    0.0000</td><td>242.3849194</td><td>7.921156</td><td> 62.79176</td></tr>\n",
       "\t<tr><th scope=row>Mmp273</th><td>p</td><td>Mmp2   </td><td>476</td><td>10047</td><td>10602.0184</td><td>467.1636791</td><td>4.501182</td><td> 60.19065</td></tr>\n",
       "\t<tr><th scope=row>Muc30E73</th><td>p</td><td>Muc30E </td><td>  0</td><td> 4940</td><td>    0.0000</td><td>213.6206937</td><td>7.738908</td><td> 59.94283</td></tr>\n",
       "\t<tr><th scope=row>AstA73</th><td>p</td><td>AstA   </td><td> 27</td><td>  182</td><td>  632.6334</td><td>  6.4103377</td><td>6.415687</td><td> 59.71403</td></tr>\n",
       "\t<tr><th scope=row>CG1553773</th><td>p</td><td>CG15537</td><td> 25</td><td>   65</td><td>  437.7598</td><td>  2.9291768</td><td>6.799769</td><td> 59.68352</td></tr>\n",
       "\t<tr><th scope=row>Pal273</th><td>p</td><td>Pal2   </td><td> 49</td><td>  182</td><td>  709.3928</td><td>  8.1755014</td><td>6.272654</td><td> 59.41755</td></tr>\n",
       "\t<tr><th scope=row>Sp173</th><td>p</td><td>Sp1    </td><td>  0</td><td> 4441</td><td>    0.0000</td><td>192.5813413</td><td>7.589324</td><td> 57.65455</td></tr>\n",
       "\t<tr><th scope=row>svp73</th><td>p</td><td>svp    </td><td>  0</td><td> 4451</td><td>    0.0000</td><td>191.1235515</td><td>7.578362</td><td> 57.48862</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG457773 & p & CG4577  & 146 &   117 &  2994.3931 &   5.1710612 & 8.922529 & 103.04209\\\\\n",
       "\tCG4254173 & p & CG42541 & 318 &  1414 &  7257.9265 &  61.4327884 & 6.861110 &  87.99744\\\\\n",
       "\tAstC73 & p & AstC    &  62 &    42 &   667.1167 &   0.8393817 & 8.502574 &  79.78779\\\\\n",
       "\tCG4333273 & p & CG43332 &   0 &  7374 &     0.0000 & 354.7325993 & 8.470588 &  71.78526\\\\\n",
       "\tCG3438673 & p & CG34386 &  14 &    16 &   443.0765 &   0.7154520 & 8.012823 &  70.47009\\\\\n",
       "\tNplp173 & p & Nplp1   &  43 &   236 &  1266.2370 &  10.5808130 & 6.772667 &  69.80907\\\\\n",
       "\tMip73 & p & Mip     &  36 &   223 &   999.5280 &   7.8848033 & 6.813763 &  67.90968\\\\\n",
       "\tMsR173 & p & MsR1    &  39 &   165 &   962.1831 &   7.3577280 & 6.847057 &  67.86574\\\\\n",
       "\tcas73 & p & cas     &   0 &  5896 &     0.0000 & 272.4327174 & 8.089756 &  65.48692\\\\\n",
       "\tCycB83 & p & CycB    &   0 &  7938 &     0.0000 & 271.1318191 & 8.082851 &  65.37540\\\\\n",
       "\tamon73 & p & amon    &  52 &   351 &  1172.3573 &  13.9914465 & 6.289129 &  64.12664\\\\\n",
       "\tMs73 & p & Ms      &  12 &    19 &   330.0292 &   0.7257723 & 7.579208 &  63.44414\\\\\n",
       "\totp73 & p & otp     &   0 &  5092 &     0.0000 & 242.3849194 & 7.921156 &  62.79176\\\\\n",
       "\tMmp273 & p & Mmp2    & 476 & 10047 & 10602.0184 & 467.1636791 & 4.501182 &  60.19065\\\\\n",
       "\tMuc30E73 & p & Muc30E  &   0 &  4940 &     0.0000 & 213.6206937 & 7.738908 &  59.94283\\\\\n",
       "\tAstA73 & p & AstA    &  27 &   182 &   632.6334 &   6.4103377 & 6.415687 &  59.71403\\\\\n",
       "\tCG1553773 & p & CG15537 &  25 &    65 &   437.7598 &   2.9291768 & 6.799769 &  59.68352\\\\\n",
       "\tPal273 & p & Pal2    &  49 &   182 &   709.3928 &   8.1755014 & 6.272654 &  59.41755\\\\\n",
       "\tSp173 & p & Sp1     &   0 &  4441 &     0.0000 & 192.5813413 & 7.589324 &  57.65455\\\\\n",
       "\tsvp73 & p & svp     &   0 &  4451 &     0.0000 & 191.1235515 & 7.578362 &  57.48862\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG457773 | p | CG4577  | 146 |   117 |  2994.3931 |   5.1710612 | 8.922529 | 103.04209 |\n",
       "| CG4254173 | p | CG42541 | 318 |  1414 |  7257.9265 |  61.4327884 | 6.861110 |  87.99744 |\n",
       "| AstC73 | p | AstC    |  62 |    42 |   667.1167 |   0.8393817 | 8.502574 |  79.78779 |\n",
       "| CG4333273 | p | CG43332 |   0 |  7374 |     0.0000 | 354.7325993 | 8.470588 |  71.78526 |\n",
       "| CG3438673 | p | CG34386 |  14 |    16 |   443.0765 |   0.7154520 | 8.012823 |  70.47009 |\n",
       "| Nplp173 | p | Nplp1   |  43 |   236 |  1266.2370 |  10.5808130 | 6.772667 |  69.80907 |\n",
       "| Mip73 | p | Mip     |  36 |   223 |   999.5280 |   7.8848033 | 6.813763 |  67.90968 |\n",
       "| MsR173 | p | MsR1    |  39 |   165 |   962.1831 |   7.3577280 | 6.847057 |  67.86574 |\n",
       "| cas73 | p | cas     |   0 |  5896 |     0.0000 | 272.4327174 | 8.089756 |  65.48692 |\n",
       "| CycB83 | p | CycB    |   0 |  7938 |     0.0000 | 271.1318191 | 8.082851 |  65.37540 |\n",
       "| amon73 | p | amon    |  52 |   351 |  1172.3573 |  13.9914465 | 6.289129 |  64.12664 |\n",
       "| Ms73 | p | Ms      |  12 |    19 |   330.0292 |   0.7257723 | 7.579208 |  63.44414 |\n",
       "| otp73 | p | otp     |   0 |  5092 |     0.0000 | 242.3849194 | 7.921156 |  62.79176 |\n",
       "| Mmp273 | p | Mmp2    | 476 | 10047 | 10602.0184 | 467.1636791 | 4.501182 |  60.19065 |\n",
       "| Muc30E73 | p | Muc30E  |   0 |  4940 |     0.0000 | 213.6206937 | 7.738908 |  59.94283 |\n",
       "| AstA73 | p | AstA    |  27 |   182 |   632.6334 |   6.4103377 | 6.415687 |  59.71403 |\n",
       "| CG1553773 | p | CG15537 |  25 |    65 |   437.7598 |   2.9291768 | 6.799769 |  59.68352 |\n",
       "| Pal273 | p | Pal2    |  49 |   182 |   709.3928 |   8.1755014 | 6.272654 |  59.41755 |\n",
       "| Sp173 | p | Sp1     |   0 |  4441 |     0.0000 | 192.5813413 | 7.589324 |  57.65455 |\n",
       "| svp73 | p | svp     |   0 |  4451 |     0.0000 | 191.1235515 | 7.578362 |  57.48862 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "CG457773  p           CG4577  146                 117        2994.3931     \n",
       "CG4254173 p           CG42541 318                1414        7257.9265     \n",
       "AstC73    p           AstC     62                  42         667.1167     \n",
       "CG4333273 p           CG43332   0                7374           0.0000     \n",
       "CG3438673 p           CG34386  14                  16         443.0765     \n",
       "Nplp173   p           Nplp1    43                 236        1266.2370     \n",
       "Mip73     p           Mip      36                 223         999.5280     \n",
       "MsR173    p           MsR1     39                 165         962.1831     \n",
       "cas73     p           cas       0                5896           0.0000     \n",
       "CycB83    p           CycB      0                7938           0.0000     \n",
       "amon73    p           amon     52                 351        1172.3573     \n",
       "Ms73      p           Ms       12                  19         330.0292     \n",
       "otp73     p           otp       0                5092           0.0000     \n",
       "Mmp273    p           Mmp2    476               10047       10602.0184     \n",
       "Muc30E73  p           Muc30E    0                4940           0.0000     \n",
       "AstA73    p           AstA     27                 182         632.6334     \n",
       "CG1553773 p           CG15537  25                  65         437.7598     \n",
       "Pal273    p           Pal2     49                 182         709.3928     \n",
       "Sp173     p           Sp1       0                4441           0.0000     \n",
       "svp73     p           svp       0                4451           0.0000     \n",
       "          other.tpm   log2.ratio heuristic.score\n",
       "CG457773    5.1710612 8.922529   103.04209      \n",
       "CG4254173  61.4327884 6.861110    87.99744      \n",
       "AstC73      0.8393817 8.502574    79.78779      \n",
       "CG4333273 354.7325993 8.470588    71.78526      \n",
       "CG3438673   0.7154520 8.012823    70.47009      \n",
       "Nplp173    10.5808130 6.772667    69.80907      \n",
       "Mip73       7.8848033 6.813763    67.90968      \n",
       "MsR173      7.3577280 6.847057    67.86574      \n",
       "cas73     272.4327174 8.089756    65.48692      \n",
       "CycB83    271.1318191 8.082851    65.37540      \n",
       "amon73     13.9914465 6.289129    64.12664      \n",
       "Ms73        0.7257723 7.579208    63.44414      \n",
       "otp73     242.3849194 7.921156    62.79176      \n",
       "Mmp273    467.1636791 4.501182    60.19065      \n",
       "Muc30E73  213.6206937 7.738908    59.94283      \n",
       "AstA73      6.4103377 6.415687    59.71403      \n",
       "CG1553773   2.9291768 6.799769    59.68352      \n",
       "Pal273      8.1755014 6.272654    59.41755      \n",
       "Sp173     192.5813413 7.589324    57.65455      \n",
       "svp73     191.1235515 7.578362    57.48862      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"p\")  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG4577\"), size_pos = 3, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG4577\"), size_pos = 3, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG4577\"), \n",
    "                   size_pos = 3, size_neg = 0.05, stroke = 0.008, limit = 1)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG42541\"), \n",
    "                   size_pos = 3, size_neg = 0.1, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"AstC\"), \n",
    "                   size_pos = 3, size_neg = 0.2, stroke = 0.02, limit = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 925,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Cholinergic\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Cholinergic\", df_sub300$cluster %in% c(q, r) &\n",
    "                          df_sub300$umap_2 < -1.3*df_sub300$umap_1 -8.7\n",
    "                         )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 926,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Glutamatergic\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Glutamatergic\", df_sub300$cluster %in% c(q, r) &\n",
    "                          df_sub300$umap_2 >= -1.3*df_sub300$umap_1 -8.7\n",
    "                         )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(q)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 906,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>nAChRalpha674</th><td>q</td><td>nAChRalpha6</td><td> 9146</td><td> 3318</td><td>2902.6524</td><td>122.78563</td><td>4.551456</td><td>52.35837</td></tr>\n",
       "\t<tr><th scope=row>Rdl74</th><td>q</td><td>Rdl        </td><td> 6730</td><td> 2886</td><td>2103.3193</td><td> 99.88408</td><td>4.381897</td><td>48.37237</td></tr>\n",
       "\t<tr><th scope=row>Cngl74</th><td>q</td><td>Cngl       </td><td> 4977</td><td> 1781</td><td>1546.7907</td><td> 64.83379</td><td>4.554306</td><td>48.25740</td></tr>\n",
       "\t<tr><th scope=row>Nckx30C74</th><td>q</td><td>Nckx30C    </td><td> 8208</td><td> 3813</td><td>2576.2272</td><td>138.91031</td><td>4.202686</td><td>47.62317</td></tr>\n",
       "\t<tr><th scope=row>Syt194</th><td>q</td><td>Syt1       </td><td> 8071</td><td> 4526</td><td>2546.2870</td><td>175.92800</td><td>3.847161</td><td>43.52965</td></tr>\n",
       "\t<tr><th scope=row>Rbp674</th><td>q</td><td>Rbp6       </td><td>15601</td><td>11186</td><td>4905.8670</td><td>418.31458</td><td>3.548403</td><td>43.50550</td></tr>\n",
       "\t<tr><th scope=row>mmd74</th><td>q</td><td>mmd        </td><td> 4766</td><td> 2269</td><td>1504.2977</td><td> 86.22446</td><td>4.108213</td><td>43.36562</td></tr>\n",
       "\t<tr><th scope=row>cac74</th><td>q</td><td>cac        </td><td>13342</td><td> 8632</td><td>4243.3721</td><td>372.97304</td><td>3.504205</td><td>42.23035</td></tr>\n",
       "\t<tr><th scope=row>GluClalpha74</th><td>q</td><td>GluClalpha </td><td> 3455</td><td> 1495</td><td>1077.0762</td><td> 59.66785</td><td>4.150044</td><td>41.80856</td></tr>\n",
       "\t<tr><th scope=row>nAChRalpha174</th><td>q</td><td>nAChRalpha1</td><td> 3669</td><td> 1803</td><td>1158.7698</td><td> 66.72205</td><td>4.096825</td><td>41.70413</td></tr>\n",
       "\t<tr><th scope=row>rad75</th><td>q</td><td>rad        </td><td> 2875</td><td> 1280</td><td> 911.8342</td><td> 48.65994</td><td>4.198617</td><td>41.29008</td></tr>\n",
       "\t<tr><th scope=row>Dop2R74</th><td>q</td><td>Dop2R      </td><td> 3154</td><td> 1399</td><td> 983.1482</td><td> 55.19638</td><td>4.128860</td><td>41.05215</td></tr>\n",
       "\t<tr><th scope=row>side-II74</th><td>q</td><td>side-II    </td><td> 1872</td><td>  683</td><td> 591.5713</td><td> 26.44296</td><td>4.430044</td><td>40.80445</td></tr>\n",
       "\t<tr><th scope=row>mtt74</th><td>q</td><td>mtt        </td><td> 1726</td><td>  668</td><td> 538.1925</td><td> 24.33261</td><td>4.409055</td><td>40.01066</td></tr>\n",
       "\t<tr><th scope=row>nAChRalpha574</th><td>q</td><td>nAChRalpha5</td><td> 3171</td><td> 1560</td><td> 995.2929</td><td> 61.32346</td><td>3.997274</td><td>39.81455</td></tr>\n",
       "\t<tr><th scope=row>CG4500274</th><td>q</td><td>CG45002    </td><td> 2460</td><td> 1179</td><td> 780.9768</td><td> 43.71215</td><td>4.126541</td><td>39.66011</td></tr>\n",
       "\t<tr><th scope=row>br74</th><td>q</td><td>br         </td><td> 8471</td><td> 6330</td><td>2755.0500</td><td>249.47998</td><td>3.459311</td><td>39.53435</td></tr>\n",
       "\t<tr><th scope=row>side-III74</th><td>q</td><td>side-III   </td><td> 1983</td><td>  860</td><td> 605.6024</td><td> 31.07390</td><td>4.238899</td><td>39.18696</td></tr>\n",
       "\t<tr><th scope=row>CG3437074</th><td>q</td><td>CG34370    </td><td> 1744</td><td>  775</td><td> 558.6540</td><td> 29.12917</td><td>4.212722</td><td>38.45538</td></tr>\n",
       "\t<tr><th scope=row>DIP-beta74</th><td>q</td><td>DIP-beta   </td><td> 1776</td><td>  764</td><td> 567.8443</td><td> 30.32163</td><td>4.180264</td><td>38.25732</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tnAChRalpha674 & q & nAChRalpha6 &  9146 &  3318 & 2902.6524 & 122.78563 & 4.551456 & 52.35837\\\\\n",
       "\tRdl74 & q & Rdl         &  6730 &  2886 & 2103.3193 &  99.88408 & 4.381897 & 48.37237\\\\\n",
       "\tCngl74 & q & Cngl        &  4977 &  1781 & 1546.7907 &  64.83379 & 4.554306 & 48.25740\\\\\n",
       "\tNckx30C74 & q & Nckx30C     &  8208 &  3813 & 2576.2272 & 138.91031 & 4.202686 & 47.62317\\\\\n",
       "\tSyt194 & q & Syt1        &  8071 &  4526 & 2546.2870 & 175.92800 & 3.847161 & 43.52965\\\\\n",
       "\tRbp674 & q & Rbp6        & 15601 & 11186 & 4905.8670 & 418.31458 & 3.548403 & 43.50550\\\\\n",
       "\tmmd74 & q & mmd         &  4766 &  2269 & 1504.2977 &  86.22446 & 4.108213 & 43.36562\\\\\n",
       "\tcac74 & q & cac         & 13342 &  8632 & 4243.3721 & 372.97304 & 3.504205 & 42.23035\\\\\n",
       "\tGluClalpha74 & q & GluClalpha  &  3455 &  1495 & 1077.0762 &  59.66785 & 4.150044 & 41.80856\\\\\n",
       "\tnAChRalpha174 & q & nAChRalpha1 &  3669 &  1803 & 1158.7698 &  66.72205 & 4.096825 & 41.70413\\\\\n",
       "\trad75 & q & rad         &  2875 &  1280 &  911.8342 &  48.65994 & 4.198617 & 41.29008\\\\\n",
       "\tDop2R74 & q & Dop2R       &  3154 &  1399 &  983.1482 &  55.19638 & 4.128860 & 41.05215\\\\\n",
       "\tside-II74 & q & side-II     &  1872 &   683 &  591.5713 &  26.44296 & 4.430044 & 40.80445\\\\\n",
       "\tmtt74 & q & mtt         &  1726 &   668 &  538.1925 &  24.33261 & 4.409055 & 40.01066\\\\\n",
       "\tnAChRalpha574 & q & nAChRalpha5 &  3171 &  1560 &  995.2929 &  61.32346 & 3.997274 & 39.81455\\\\\n",
       "\tCG4500274 & q & CG45002     &  2460 &  1179 &  780.9768 &  43.71215 & 4.126541 & 39.66011\\\\\n",
       "\tbr74 & q & br          &  8471 &  6330 & 2755.0500 & 249.47998 & 3.459311 & 39.53435\\\\\n",
       "\tside-III74 & q & side-III    &  1983 &   860 &  605.6024 &  31.07390 & 4.238899 & 39.18696\\\\\n",
       "\tCG3437074 & q & CG34370     &  1744 &   775 &  558.6540 &  29.12917 & 4.212722 & 38.45538\\\\\n",
       "\tDIP-beta74 & q & DIP-beta    &  1776 &   764 &  567.8443 &  30.32163 & 4.180264 & 38.25732\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| nAChRalpha674 | q | nAChRalpha6 |  9146 |  3318 | 2902.6524 | 122.78563 | 4.551456 | 52.35837 |\n",
       "| Rdl74 | q | Rdl         |  6730 |  2886 | 2103.3193 |  99.88408 | 4.381897 | 48.37237 |\n",
       "| Cngl74 | q | Cngl        |  4977 |  1781 | 1546.7907 |  64.83379 | 4.554306 | 48.25740 |\n",
       "| Nckx30C74 | q | Nckx30C     |  8208 |  3813 | 2576.2272 | 138.91031 | 4.202686 | 47.62317 |\n",
       "| Syt194 | q | Syt1        |  8071 |  4526 | 2546.2870 | 175.92800 | 3.847161 | 43.52965 |\n",
       "| Rbp674 | q | Rbp6        | 15601 | 11186 | 4905.8670 | 418.31458 | 3.548403 | 43.50550 |\n",
       "| mmd74 | q | mmd         |  4766 |  2269 | 1504.2977 |  86.22446 | 4.108213 | 43.36562 |\n",
       "| cac74 | q | cac         | 13342 |  8632 | 4243.3721 | 372.97304 | 3.504205 | 42.23035 |\n",
       "| GluClalpha74 | q | GluClalpha  |  3455 |  1495 | 1077.0762 |  59.66785 | 4.150044 | 41.80856 |\n",
       "| nAChRalpha174 | q | nAChRalpha1 |  3669 |  1803 | 1158.7698 |  66.72205 | 4.096825 | 41.70413 |\n",
       "| rad75 | q | rad         |  2875 |  1280 |  911.8342 |  48.65994 | 4.198617 | 41.29008 |\n",
       "| Dop2R74 | q | Dop2R       |  3154 |  1399 |  983.1482 |  55.19638 | 4.128860 | 41.05215 |\n",
       "| side-II74 | q | side-II     |  1872 |   683 |  591.5713 |  26.44296 | 4.430044 | 40.80445 |\n",
       "| mtt74 | q | mtt         |  1726 |   668 |  538.1925 |  24.33261 | 4.409055 | 40.01066 |\n",
       "| nAChRalpha574 | q | nAChRalpha5 |  3171 |  1560 |  995.2929 |  61.32346 | 3.997274 | 39.81455 |\n",
       "| CG4500274 | q | CG45002     |  2460 |  1179 |  780.9768 |  43.71215 | 4.126541 | 39.66011 |\n",
       "| br74 | q | br          |  8471 |  6330 | 2755.0500 | 249.47998 | 3.459311 | 39.53435 |\n",
       "| side-III74 | q | side-III    |  1983 |   860 |  605.6024 |  31.07390 | 4.238899 | 39.18696 |\n",
       "| CG3437074 | q | CG34370     |  1744 |   775 |  558.6540 |  29.12917 | 4.212722 | 38.45538 |\n",
       "| DIP-beta74 | q | DIP-beta    |  1776 |   764 |  567.8443 |  30.32163 | 4.180264 | 38.25732 |\n",
       "\n"
      ],
      "text/plain": [
       "              cell.type.2 gene        cell.type.2.n.umi other.n.umi\n",
       "nAChRalpha674 q           nAChRalpha6  9146              3318      \n",
       "Rdl74         q           Rdl          6730              2886      \n",
       "Cngl74        q           Cngl         4977              1781      \n",
       "Nckx30C74     q           Nckx30C      8208              3813      \n",
       "Syt194        q           Syt1         8071              4526      \n",
       "Rbp674        q           Rbp6        15601             11186      \n",
       "mmd74         q           mmd          4766              2269      \n",
       "cac74         q           cac         13342              8632      \n",
       "GluClalpha74  q           GluClalpha   3455              1495      \n",
       "nAChRalpha174 q           nAChRalpha1  3669              1803      \n",
       "rad75         q           rad          2875              1280      \n",
       "Dop2R74       q           Dop2R        3154              1399      \n",
       "side-II74     q           side-II      1872               683      \n",
       "mtt74         q           mtt          1726               668      \n",
       "nAChRalpha574 q           nAChRalpha5  3171              1560      \n",
       "CG4500274     q           CG45002      2460              1179      \n",
       "br74          q           br           8471              6330      \n",
       "side-III74    q           side-III     1983               860      \n",
       "CG3437074     q           CG34370      1744               775      \n",
       "DIP-beta74    q           DIP-beta     1776               764      \n",
       "              cell.type.2.tpm other.tpm log2.ratio heuristic.score\n",
       "nAChRalpha674 2902.6524       122.78563 4.551456   52.35837       \n",
       "Rdl74         2103.3193        99.88408 4.381897   48.37237       \n",
       "Cngl74        1546.7907        64.83379 4.554306   48.25740       \n",
       "Nckx30C74     2576.2272       138.91031 4.202686   47.62317       \n",
       "Syt194        2546.2870       175.92800 3.847161   43.52965       \n",
       "Rbp674        4905.8670       418.31458 3.548403   43.50550       \n",
       "mmd74         1504.2977        86.22446 4.108213   43.36562       \n",
       "cac74         4243.3721       372.97304 3.504205   42.23035       \n",
       "GluClalpha74  1077.0762        59.66785 4.150044   41.80856       \n",
       "nAChRalpha174 1158.7698        66.72205 4.096825   41.70413       \n",
       "rad75          911.8342        48.65994 4.198617   41.29008       \n",
       "Dop2R74        983.1482        55.19638 4.128860   41.05215       \n",
       "side-II74      591.5713        26.44296 4.430044   40.80445       \n",
       "mtt74          538.1925        24.33261 4.409055   40.01066       \n",
       "nAChRalpha574  995.2929        61.32346 3.997274   39.81455       \n",
       "CG4500274      780.9768        43.71215 4.126541   39.66011       \n",
       "br74          2755.0500       249.47998 3.459311   39.53435       \n",
       "side-III74     605.6024        31.07390 4.238899   39.18696       \n",
       "CG3437074      558.6540        29.12917 4.212722   38.45538       \n",
       "DIP-beta74     567.8443        30.32163 4.180264   38.25732       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"q\")  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"nAChRalpha6\"), size_pos = 3, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"nAChRalpha6\"), size_pos = 3, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"nAChRalpha6\"), \n",
    "                   size_pos = 3, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Rdl\"), \n",
    "                   size_pos = 3, size_neg = 0.1, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Cngl\"), \n",
    "                   size_pos = 3, size_neg = 0.2, stroke = 0.02, limit = 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'r' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'r' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(r,q)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = -1.3, intercept = -8.7) +\n",
    "geom_abline(slope = 0.9, intercept = -6.7) +\n",
    "geom_abline(slope = 0.9, intercept = -5.4) +\n",
    " xlim(-4, 2) + ylim(-9, -3)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &\n",
    "              df_sub300$umap_2 < -1.3*umap_1 -8.7), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = -1.3, intercept = -8.7) +\n",
    "geom_abline(slope = 0.9, intercept = -6.7) +\n",
    "geom_abline(slope = 0.9, intercept = -5.4) +\n",
    " xlim(-4, 2) + ylim(-9, -3)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &\n",
    "              df_sub300$umap_2 >= -1.3*umap_1 -8.7), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = -1.3, intercept = -8.7) +\n",
    "geom_abline(slope = 0.9, intercept = -6.7) +\n",
    "geom_abline(slope = 0.9, intercept = -5.4) +\n",
    " xlim(-4, 2) + ylim(-9, -3)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type r\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"r\", df_sub300$cluster %in% r)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### s cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 933,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Fat_body\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Fat_body\", df_sub300$cluster %in% s)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 's' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 's' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(s)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 930,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>fon76</th><td>s</td><td>fon        </td><td> 3297</td><td>  647</td><td> 1916.1924</td><td>  21.780420</td><td>6.394304</td><td>69.72848</td></tr>\n",
       "\t<tr><th scope=row>apolpp76</th><td>s</td><td>apolpp     </td><td>52336</td><td>66912</td><td>34062.8642</td><td>2123.758505</td><td>4.002829</td><td>60.26641</td></tr>\n",
       "\t<tr><th scope=row>CG223376</th><td>s</td><td>CG2233     </td><td> 1722</td><td>  326</td><td>  915.3464</td><td>  12.929236</td><td>6.038130</td><td>59.41368</td></tr>\n",
       "\t<tr><th scope=row>NimB276</th><td>s</td><td>NimB2      </td><td> 2470</td><td> 1051</td><td> 1639.1646</td><td>  39.603412</td><td>5.335216</td><td>56.97811</td></tr>\n",
       "\t<tr><th scope=row>CG273676</th><td>s</td><td>CG2736     </td><td> 1281</td><td>  379</td><td>  839.0123</td><td>  14.842270</td><td>5.726841</td><td>55.63206</td></tr>\n",
       "\t<tr><th scope=row>Glt76</th><td>s</td><td>Glt        </td><td> 5086</td><td> 3177</td><td> 3462.6335</td><td> 135.959769</td><td>4.660046</td><td>54.79315</td></tr>\n",
       "\t<tr><th scope=row>CG722776</th><td>s</td><td>CG7227     </td><td> 3191</td><td> 1839</td><td> 2214.8709</td><td>  81.024817</td><td>4.755018</td><td>52.84565</td></tr>\n",
       "\t<tr><th scope=row>Idgf276</th><td>s</td><td>Idgf2      </td><td> 2374</td><td> 1458</td><td> 1643.7176</td><td>  56.854241</td><td>4.828396</td><td>51.58477</td></tr>\n",
       "\t<tr><th scope=row>teq76</th><td>s</td><td>teq        </td><td> 1727</td><td>  856</td><td> 1167.4463</td><td>  35.371768</td><td>5.004393</td><td>50.99665</td></tr>\n",
       "\t<tr><th scope=row>CG816076</th><td>s</td><td>CG8160     </td><td>  703</td><td>  173</td><td>  383.8797</td><td>   6.107334</td><td>5.755202</td><td>49.42719</td></tr>\n",
       "\t<tr><th scope=row>Gbp276</th><td>s</td><td>Gbp2       </td><td>  918</td><td>  461</td><td>  623.7060</td><td>  18.168402</td><td>5.024064</td><td>46.65865</td></tr>\n",
       "\t<tr><th scope=row>Cpr49Ac76</th><td>s</td><td>Cpr49Ac    </td><td>  618</td><td>  175</td><td>  364.3981</td><td>   7.435924</td><td>5.432825</td><td>46.25141</td></tr>\n",
       "\t<tr><th scope=row>CG3252176</th><td>s</td><td>CG32521    </td><td> 3751</td><td> 3589</td><td> 2489.1181</td><td> 149.962961</td><td>4.043368</td><td>45.61727</td></tr>\n",
       "\t<tr><th scope=row>Idgf476</th><td>s</td><td>Idgf4      </td><td> 5272</td><td> 6633</td><td> 3533.7785</td><td> 242.835552</td><td>3.857231</td><td>45.46674</td></tr>\n",
       "\t<tr><th scope=row>Apoltp76</th><td>s</td><td>Apoltp     </td><td> 2814</td><td> 2908</td><td> 1883.5861</td><td> 104.436146</td><td>4.159041</td><td>45.25049</td></tr>\n",
       "\t<tr><th scope=row>fat-spondin76</th><td>s</td><td>fat-spondin</td><td> 2799</td><td> 2683</td><td> 1925.5965</td><td> 118.820347</td><td>4.006361</td><td>43.71676</td></tr>\n",
       "\t<tr><th scope=row>CG324695</th><td>s</td><td>CG3246     </td><td> 1900</td><td> 1791</td><td> 1306.1738</td><td>  70.428346</td><td>4.192706</td><td>43.40388</td></tr>\n",
       "\t<tr><th scope=row>CG993276</th><td>s</td><td>CG9932     </td><td> 5744</td><td> 8240</td><td> 3968.3370</td><td> 319.884931</td><td>3.628407</td><td>43.37645</td></tr>\n",
       "\t<tr><th scope=row>CG858676</th><td>s</td><td>CG8586     </td><td>  486</td><td>  242</td><td>  334.0196</td><td>   8.465704</td><td>5.141079</td><td>43.12389</td></tr>\n",
       "\t<tr><th scope=row>Hayan76</th><td>s</td><td>Hayan      </td><td>  858</td><td>  517</td><td>  597.8652</td><td>  22.715031</td><td>4.655946</td><td>42.95617</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tfon76 & s & fon         &  3297 &   647 &  1916.1924 &   21.780420 & 6.394304 & 69.72848\\\\\n",
       "\tapolpp76 & s & apolpp      & 52336 & 66912 & 34062.8642 & 2123.758505 & 4.002829 & 60.26641\\\\\n",
       "\tCG223376 & s & CG2233      &  1722 &   326 &   915.3464 &   12.929236 & 6.038130 & 59.41368\\\\\n",
       "\tNimB276 & s & NimB2       &  2470 &  1051 &  1639.1646 &   39.603412 & 5.335216 & 56.97811\\\\\n",
       "\tCG273676 & s & CG2736      &  1281 &   379 &   839.0123 &   14.842270 & 5.726841 & 55.63206\\\\\n",
       "\tGlt76 & s & Glt         &  5086 &  3177 &  3462.6335 &  135.959769 & 4.660046 & 54.79315\\\\\n",
       "\tCG722776 & s & CG7227      &  3191 &  1839 &  2214.8709 &   81.024817 & 4.755018 & 52.84565\\\\\n",
       "\tIdgf276 & s & Idgf2       &  2374 &  1458 &  1643.7176 &   56.854241 & 4.828396 & 51.58477\\\\\n",
       "\tteq76 & s & teq         &  1727 &   856 &  1167.4463 &   35.371768 & 5.004393 & 50.99665\\\\\n",
       "\tCG816076 & s & CG8160      &   703 &   173 &   383.8797 &    6.107334 & 5.755202 & 49.42719\\\\\n",
       "\tGbp276 & s & Gbp2        &   918 &   461 &   623.7060 &   18.168402 & 5.024064 & 46.65865\\\\\n",
       "\tCpr49Ac76 & s & Cpr49Ac     &   618 &   175 &   364.3981 &    7.435924 & 5.432825 & 46.25141\\\\\n",
       "\tCG3252176 & s & CG32521     &  3751 &  3589 &  2489.1181 &  149.962961 & 4.043368 & 45.61727\\\\\n",
       "\tIdgf476 & s & Idgf4       &  5272 &  6633 &  3533.7785 &  242.835552 & 3.857231 & 45.46674\\\\\n",
       "\tApoltp76 & s & Apoltp      &  2814 &  2908 &  1883.5861 &  104.436146 & 4.159041 & 45.25049\\\\\n",
       "\tfat-spondin76 & s & fat-spondin &  2799 &  2683 &  1925.5965 &  118.820347 & 4.006361 & 43.71676\\\\\n",
       "\tCG324695 & s & CG3246      &  1900 &  1791 &  1306.1738 &   70.428346 & 4.192706 & 43.40388\\\\\n",
       "\tCG993276 & s & CG9932      &  5744 &  8240 &  3968.3370 &  319.884931 & 3.628407 & 43.37645\\\\\n",
       "\tCG858676 & s & CG8586      &   486 &   242 &   334.0196 &    8.465704 & 5.141079 & 43.12389\\\\\n",
       "\tHayan76 & s & Hayan       &   858 &   517 &   597.8652 &   22.715031 & 4.655946 & 42.95617\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| fon76 | s | fon         |  3297 |   647 |  1916.1924 |   21.780420 | 6.394304 | 69.72848 |\n",
       "| apolpp76 | s | apolpp      | 52336 | 66912 | 34062.8642 | 2123.758505 | 4.002829 | 60.26641 |\n",
       "| CG223376 | s | CG2233      |  1722 |   326 |   915.3464 |   12.929236 | 6.038130 | 59.41368 |\n",
       "| NimB276 | s | NimB2       |  2470 |  1051 |  1639.1646 |   39.603412 | 5.335216 | 56.97811 |\n",
       "| CG273676 | s | CG2736      |  1281 |   379 |   839.0123 |   14.842270 | 5.726841 | 55.63206 |\n",
       "| Glt76 | s | Glt         |  5086 |  3177 |  3462.6335 |  135.959769 | 4.660046 | 54.79315 |\n",
       "| CG722776 | s | CG7227      |  3191 |  1839 |  2214.8709 |   81.024817 | 4.755018 | 52.84565 |\n",
       "| Idgf276 | s | Idgf2       |  2374 |  1458 |  1643.7176 |   56.854241 | 4.828396 | 51.58477 |\n",
       "| teq76 | s | teq         |  1727 |   856 |  1167.4463 |   35.371768 | 5.004393 | 50.99665 |\n",
       "| CG816076 | s | CG8160      |   703 |   173 |   383.8797 |    6.107334 | 5.755202 | 49.42719 |\n",
       "| Gbp276 | s | Gbp2        |   918 |   461 |   623.7060 |   18.168402 | 5.024064 | 46.65865 |\n",
       "| Cpr49Ac76 | s | Cpr49Ac     |   618 |   175 |   364.3981 |    7.435924 | 5.432825 | 46.25141 |\n",
       "| CG3252176 | s | CG32521     |  3751 |  3589 |  2489.1181 |  149.962961 | 4.043368 | 45.61727 |\n",
       "| Idgf476 | s | Idgf4       |  5272 |  6633 |  3533.7785 |  242.835552 | 3.857231 | 45.46674 |\n",
       "| Apoltp76 | s | Apoltp      |  2814 |  2908 |  1883.5861 |  104.436146 | 4.159041 | 45.25049 |\n",
       "| fat-spondin76 | s | fat-spondin |  2799 |  2683 |  1925.5965 |  118.820347 | 4.006361 | 43.71676 |\n",
       "| CG324695 | s | CG3246      |  1900 |  1791 |  1306.1738 |   70.428346 | 4.192706 | 43.40388 |\n",
       "| CG993276 | s | CG9932      |  5744 |  8240 |  3968.3370 |  319.884931 | 3.628407 | 43.37645 |\n",
       "| CG858676 | s | CG8586      |   486 |   242 |   334.0196 |    8.465704 | 5.141079 | 43.12389 |\n",
       "| Hayan76 | s | Hayan       |   858 |   517 |   597.8652 |   22.715031 | 4.655946 | 42.95617 |\n",
       "\n"
      ],
      "text/plain": [
       "              cell.type.2 gene        cell.type.2.n.umi other.n.umi\n",
       "fon76         s           fon          3297               647      \n",
       "apolpp76      s           apolpp      52336             66912      \n",
       "CG223376      s           CG2233       1722               326      \n",
       "NimB276       s           NimB2        2470              1051      \n",
       "CG273676      s           CG2736       1281               379      \n",
       "Glt76         s           Glt          5086              3177      \n",
       "CG722776      s           CG7227       3191              1839      \n",
       "Idgf276       s           Idgf2        2374              1458      \n",
       "teq76         s           teq          1727               856      \n",
       "CG816076      s           CG8160        703               173      \n",
       "Gbp276        s           Gbp2          918               461      \n",
       "Cpr49Ac76     s           Cpr49Ac       618               175      \n",
       "CG3252176     s           CG32521      3751              3589      \n",
       "Idgf476       s           Idgf4        5272              6633      \n",
       "Apoltp76      s           Apoltp       2814              2908      \n",
       "fat-spondin76 s           fat-spondin  2799              2683      \n",
       "CG324695      s           CG3246       1900              1791      \n",
       "CG993276      s           CG9932       5744              8240      \n",
       "CG858676      s           CG8586        486               242      \n",
       "Hayan76       s           Hayan         858               517      \n",
       "              cell.type.2.tpm other.tpm   log2.ratio heuristic.score\n",
       "fon76          1916.1924        21.780420 6.394304   69.72848       \n",
       "apolpp76      34062.8642      2123.758505 4.002829   60.26641       \n",
       "CG223376        915.3464        12.929236 6.038130   59.41368       \n",
       "NimB276        1639.1646        39.603412 5.335216   56.97811       \n",
       "CG273676        839.0123        14.842270 5.726841   55.63206       \n",
       "Glt76          3462.6335       135.959769 4.660046   54.79315       \n",
       "CG722776       2214.8709        81.024817 4.755018   52.84565       \n",
       "Idgf276        1643.7176        56.854241 4.828396   51.58477       \n",
       "teq76          1167.4463        35.371768 5.004393   50.99665       \n",
       "CG816076        383.8797         6.107334 5.755202   49.42719       \n",
       "Gbp276          623.7060        18.168402 5.024064   46.65865       \n",
       "Cpr49Ac76       364.3981         7.435924 5.432825   46.25141       \n",
       "CG3252176      2489.1181       149.962961 4.043368   45.61727       \n",
       "Idgf476        3533.7785       242.835552 3.857231   45.46674       \n",
       "Apoltp76       1883.5861       104.436146 4.159041   45.25049       \n",
       "fat-spondin76  1925.5965       118.820347 4.006361   43.71676       \n",
       "CG324695       1306.1738        70.428346 4.192706   43.40388       \n",
       "CG993276       3968.3370       319.884931 3.628407   43.37645       \n",
       "CG858676        334.0196         8.465704 5.141079   43.12389       \n",
       "Hayan76         597.8652        22.715031 4.655946   42.95617       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"s\")  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"fon\"), size_pos = 3, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"fon\"), size_pos = 3, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"fon\"), \n",
    "                   size_pos = 3, size_neg = 0.05, stroke = 0.008, limit = 3)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG2233\"), \n",
    "                   size_pos = 3, size_neg = 0.1, stroke = 0.02, limit = 3)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"NimB2\"), \n",
    "                   size_pos = 3, size_neg = 0.2, stroke = 0.02, limit = 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 950,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Monoaminergic\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Monoaminergic\", df_sub300$cluster %in% t &\n",
    "                          df_sub300$umap_2 < -1*df_sub300$umap_1 -9.8\n",
    "                         )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 952,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Neuron_unknown_1\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Neuron_unknown_1\", df_sub300$cluster %in% t &\n",
    "                          df_sub300$umap_2 >= -1*df_sub300$umap_1 -9.8\n",
    "                         )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 953,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 21 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cell.type.3</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Amnioserosa           </td><td>  7091</td></tr>\n",
       "\t<tr><td>Cholinergic           </td><td>  8949</td></tr>\n",
       "\t<tr><td>Chordotonal_organ     </td><td>   121</td></tr>\n",
       "\t<tr><td>Chordotonal_sensory   </td><td>   750</td></tr>\n",
       "\t<tr><td>Crystal_cell          </td><td>   432</td></tr>\n",
       "\t<tr><td>Epidermis             </td><td> 10373</td></tr>\n",
       "\t<tr><td>Epidermis_late        </td><td>   184</td></tr>\n",
       "\t<tr><td>External_sensory      </td><td>   647</td></tr>\n",
       "\t<tr><td>Fat_body              </td><td>  4732</td></tr>\n",
       "\t<tr><td>Glutamatergic         </td><td>  5107</td></tr>\n",
       "\t<tr><td>Malpighian tubule     </td><td>  1731</td></tr>\n",
       "\t<tr><td>Monoaminergic         </td><td>   454</td></tr>\n",
       "\t<tr><td>Multidendritic_sensory</td><td>  2069</td></tr>\n",
       "\t<tr><td>Neuron_unknown_1      </td><td>   602</td></tr>\n",
       "\t<tr><td>Peptidergic_neuron    </td><td>   147</td></tr>\n",
       "\t<tr><td>Proventriculus        </td><td>  2423</td></tr>\n",
       "\t<tr><td>gut_query             </td><td>   887</td></tr>\n",
       "\t<tr><td>proventriculus        </td><td>   528</td></tr>\n",
       "\t<tr><td>uncertain_1           </td><td>    59</td></tr>\n",
       "\t<tr><td>unknown_2             </td><td>  1629</td></tr>\n",
       "\t<tr><td>NA                    </td><td>189530</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 21 × 2\n",
       "\\begin{tabular}{ll}\n",
       " cell.type.3 & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Amnioserosa            &   7091\\\\\n",
       "\t Cholinergic            &   8949\\\\\n",
       "\t Chordotonal\\_organ      &    121\\\\\n",
       "\t Chordotonal\\_sensory    &    750\\\\\n",
       "\t Crystal\\_cell           &    432\\\\\n",
       "\t Epidermis              &  10373\\\\\n",
       "\t Epidermis\\_late         &    184\\\\\n",
       "\t External\\_sensory       &    647\\\\\n",
       "\t Fat\\_body               &   4732\\\\\n",
       "\t Glutamatergic          &   5107\\\\\n",
       "\t Malpighian tubule      &   1731\\\\\n",
       "\t Monoaminergic          &    454\\\\\n",
       "\t Multidendritic\\_sensory &   2069\\\\\n",
       "\t Neuron\\_unknown\\_1       &    602\\\\\n",
       "\t Peptidergic\\_neuron     &    147\\\\\n",
       "\t Proventriculus         &   2423\\\\\n",
       "\t gut\\_query              &    887\\\\\n",
       "\t proventriculus         &    528\\\\\n",
       "\t uncertain\\_1            &     59\\\\\n",
       "\t unknown\\_2              &   1629\\\\\n",
       "\t NA                     & 189530\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 21 × 2\n",
       "\n",
       "| cell.type.3 &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Amnioserosa            |   7091 |\n",
       "| Cholinergic            |   8949 |\n",
       "| Chordotonal_organ      |    121 |\n",
       "| Chordotonal_sensory    |    750 |\n",
       "| Crystal_cell           |    432 |\n",
       "| Epidermis              |  10373 |\n",
       "| Epidermis_late         |    184 |\n",
       "| External_sensory       |    647 |\n",
       "| Fat_body               |   4732 |\n",
       "| Glutamatergic          |   5107 |\n",
       "| Malpighian tubule      |   1731 |\n",
       "| Monoaminergic          |    454 |\n",
       "| Multidendritic_sensory |   2069 |\n",
       "| Neuron_unknown_1       |    602 |\n",
       "| Peptidergic_neuron     |    147 |\n",
       "| Proventriculus         |   2423 |\n",
       "| gut_query              |    887 |\n",
       "| proventriculus         |    528 |\n",
       "| uncertain_1            |     59 |\n",
       "| unknown_2              |   1629 |\n",
       "| NA                     | 189530 |\n",
       "\n"
      ],
      "text/plain": [
       "   cell.type.3            n     \n",
       "1  Amnioserosa              7091\n",
       "2  Cholinergic              8949\n",
       "3  Chordotonal_organ         121\n",
       "4  Chordotonal_sensory       750\n",
       "5  Crystal_cell              432\n",
       "6  Epidermis               10373\n",
       "7  Epidermis_late            184\n",
       "8  External_sensory          647\n",
       "9  Fat_body                 4732\n",
       "10 Glutamatergic            5107\n",
       "11 Malpighian tubule        1731\n",
       "12 Monoaminergic             454\n",
       "13 Multidendritic_sensory   2069\n",
       "14 Neuron_unknown_1          602\n",
       "15 Peptidergic_neuron        147\n",
       "16 Proventriculus           2423\n",
       "17 gut_query                 887\n",
       "18 proventriculus            528\n",
       "19 uncertain_1                59\n",
       "20 unknown_2                1629\n",
       "21 NA                     189530"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub300  %>% count(cell.type.3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(t)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 935,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Vmat77</th><td>t</td><td>Vmat       </td><td>3511</td><td>1741</td><td>10124.6978</td><td> 60.350716</td><td>7.366583</td><td>98.01779</td></tr>\n",
       "\t<tr><th scope=row>Tbh77</th><td>t</td><td>Tbh        </td><td> 582</td><td> 179</td><td> 1643.1032</td><td>  6.388548</td><td>7.796917</td><td>83.29512</td></tr>\n",
       "\t<tr><th scope=row>Tdc277</th><td>t</td><td>Tdc2       </td><td> 203</td><td> 119</td><td>  603.9666</td><td>  4.077435</td><td>6.894225</td><td>63.70754</td></tr>\n",
       "\t<tr><th scope=row>CG3119184</th><td>t</td><td>CG31191    </td><td> 224</td><td> 213</td><td>  679.0681</td><td>  8.786192</td><td>6.116665</td><td>57.55498</td></tr>\n",
       "\t<tr><th scope=row>CG226977</th><td>t</td><td>CG2269     </td><td> 252</td><td> 239</td><td>  715.7692</td><td> 10.506789</td><td>5.958937</td><td>56.52269</td></tr>\n",
       "\t<tr><th scope=row>CG4500277</th><td>t</td><td>CG45002    </td><td> 725</td><td>1555</td><td> 2033.4604</td><td> 69.040869</td><td>4.859596</td><td>53.40905</td></tr>\n",
       "\t<tr><th scope=row>SLO277</th><td>t</td><td>SLO2       </td><td> 504</td><td>1064</td><td> 1383.0233</td><td> 45.940672</td><td>4.880843</td><td>50.92990</td></tr>\n",
       "\t<tr><th scope=row>GluRIB77</th><td>t</td><td>GluRIB     </td><td> 330</td><td> 631</td><td>  936.9447</td><td> 27.792948</td><td>5.024176</td><td>49.60550</td></tr>\n",
       "\t<tr><th scope=row>DAT77</th><td>t</td><td>DAT        </td><td> 126</td><td> 184</td><td>  399.6778</td><td>  6.840168</td><td>5.671809</td><td>49.04016</td></tr>\n",
       "\t<tr><th scope=row>cpx77</th><td>t</td><td>cpx        </td><td> 845</td><td>2931</td><td> 2458.3679</td><td>124.621367</td><td>4.290547</td><td>48.32903</td></tr>\n",
       "\t<tr><th scope=row>DIP-eta77</th><td>t</td><td>DIP-eta    </td><td> 208</td><td> 300</td><td>  543.3513</td><td> 12.685593</td><td>5.311155</td><td>48.26987</td></tr>\n",
       "\t<tr><th scope=row>bma77</th><td>t</td><td>bma        </td><td> 167</td><td> 252</td><td>  475.3713</td><td> 10.288676</td><td>5.396107</td><td>48.00346</td></tr>\n",
       "\t<tr><th scope=row>dpr2177</th><td>t</td><td>dpr21      </td><td> 365</td><td> 849</td><td> 1017.8617</td><td> 38.792918</td><td>4.676886</td><td>46.73492</td></tr>\n",
       "\t<tr><th scope=row>jus77</th><td>t</td><td>jus        </td><td> 553</td><td>2091</td><td> 1595.0323</td><td> 79.579916</td><td>4.307022</td><td>45.82789</td></tr>\n",
       "\t<tr><th scope=row>IA-277</th><td>t</td><td>IA-2       </td><td> 544</td><td>2181</td><td> 1505.9409</td><td> 74.187341</td><td>4.324032</td><td>45.65056</td></tr>\n",
       "\t<tr><th scope=row>nAChRalpha677</th><td>t</td><td>nAChRalpha6</td><td>1243</td><td>5082</td><td> 3430.8495</td><td>233.213158</td><td>3.872672</td><td>45.48364</td></tr>\n",
       "\t<tr><th scope=row>CG432878</th><td>t</td><td>CG4328     </td><td> 104</td><td> 136</td><td>  292.6234</td><td>  5.276942</td><td>5.542839</td><td>45.43921</td></tr>\n",
       "\t<tr><th scope=row>5-HT1A77</th><td>t</td><td>5-HT1A     </td><td> 302</td><td> 745</td><td>  830.2874</td><td> 32.897463</td><td>4.614362</td><td>44.75563</td></tr>\n",
       "\t<tr><th scope=row>dpr1377</th><td>t</td><td>dpr13      </td><td> 367</td><td>1038</td><td> 1019.1029</td><td> 45.986889</td><td>4.438898</td><td>44.36456</td></tr>\n",
       "\t<tr><th scope=row>SerT77</th><td>t</td><td>SerT       </td><td>  81</td><td>  94</td><td>  240.4180</td><td>  3.991988</td><td>5.589786</td><td>44.24533</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tVmat77 & t & Vmat        & 3511 & 1741 & 10124.6978 &  60.350716 & 7.366583 & 98.01779\\\\\n",
       "\tTbh77 & t & Tbh         &  582 &  179 &  1643.1032 &   6.388548 & 7.796917 & 83.29512\\\\\n",
       "\tTdc277 & t & Tdc2        &  203 &  119 &   603.9666 &   4.077435 & 6.894225 & 63.70754\\\\\n",
       "\tCG3119184 & t & CG31191     &  224 &  213 &   679.0681 &   8.786192 & 6.116665 & 57.55498\\\\\n",
       "\tCG226977 & t & CG2269      &  252 &  239 &   715.7692 &  10.506789 & 5.958937 & 56.52269\\\\\n",
       "\tCG4500277 & t & CG45002     &  725 & 1555 &  2033.4604 &  69.040869 & 4.859596 & 53.40905\\\\\n",
       "\tSLO277 & t & SLO2        &  504 & 1064 &  1383.0233 &  45.940672 & 4.880843 & 50.92990\\\\\n",
       "\tGluRIB77 & t & GluRIB      &  330 &  631 &   936.9447 &  27.792948 & 5.024176 & 49.60550\\\\\n",
       "\tDAT77 & t & DAT         &  126 &  184 &   399.6778 &   6.840168 & 5.671809 & 49.04016\\\\\n",
       "\tcpx77 & t & cpx         &  845 & 2931 &  2458.3679 & 124.621367 & 4.290547 & 48.32903\\\\\n",
       "\tDIP-eta77 & t & DIP-eta     &  208 &  300 &   543.3513 &  12.685593 & 5.311155 & 48.26987\\\\\n",
       "\tbma77 & t & bma         &  167 &  252 &   475.3713 &  10.288676 & 5.396107 & 48.00346\\\\\n",
       "\tdpr2177 & t & dpr21       &  365 &  849 &  1017.8617 &  38.792918 & 4.676886 & 46.73492\\\\\n",
       "\tjus77 & t & jus         &  553 & 2091 &  1595.0323 &  79.579916 & 4.307022 & 45.82789\\\\\n",
       "\tIA-277 & t & IA-2        &  544 & 2181 &  1505.9409 &  74.187341 & 4.324032 & 45.65056\\\\\n",
       "\tnAChRalpha677 & t & nAChRalpha6 & 1243 & 5082 &  3430.8495 & 233.213158 & 3.872672 & 45.48364\\\\\n",
       "\tCG432878 & t & CG4328      &  104 &  136 &   292.6234 &   5.276942 & 5.542839 & 45.43921\\\\\n",
       "\t5-HT1A77 & t & 5-HT1A      &  302 &  745 &   830.2874 &  32.897463 & 4.614362 & 44.75563\\\\\n",
       "\tdpr1377 & t & dpr13       &  367 & 1038 &  1019.1029 &  45.986889 & 4.438898 & 44.36456\\\\\n",
       "\tSerT77 & t & SerT        &   81 &   94 &   240.4180 &   3.991988 & 5.589786 & 44.24533\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Vmat77 | t | Vmat        | 3511 | 1741 | 10124.6978 |  60.350716 | 7.366583 | 98.01779 |\n",
       "| Tbh77 | t | Tbh         |  582 |  179 |  1643.1032 |   6.388548 | 7.796917 | 83.29512 |\n",
       "| Tdc277 | t | Tdc2        |  203 |  119 |   603.9666 |   4.077435 | 6.894225 | 63.70754 |\n",
       "| CG3119184 | t | CG31191     |  224 |  213 |   679.0681 |   8.786192 | 6.116665 | 57.55498 |\n",
       "| CG226977 | t | CG2269      |  252 |  239 |   715.7692 |  10.506789 | 5.958937 | 56.52269 |\n",
       "| CG4500277 | t | CG45002     |  725 | 1555 |  2033.4604 |  69.040869 | 4.859596 | 53.40905 |\n",
       "| SLO277 | t | SLO2        |  504 | 1064 |  1383.0233 |  45.940672 | 4.880843 | 50.92990 |\n",
       "| GluRIB77 | t | GluRIB      |  330 |  631 |   936.9447 |  27.792948 | 5.024176 | 49.60550 |\n",
       "| DAT77 | t | DAT         |  126 |  184 |   399.6778 |   6.840168 | 5.671809 | 49.04016 |\n",
       "| cpx77 | t | cpx         |  845 | 2931 |  2458.3679 | 124.621367 | 4.290547 | 48.32903 |\n",
       "| DIP-eta77 | t | DIP-eta     |  208 |  300 |   543.3513 |  12.685593 | 5.311155 | 48.26987 |\n",
       "| bma77 | t | bma         |  167 |  252 |   475.3713 |  10.288676 | 5.396107 | 48.00346 |\n",
       "| dpr2177 | t | dpr21       |  365 |  849 |  1017.8617 |  38.792918 | 4.676886 | 46.73492 |\n",
       "| jus77 | t | jus         |  553 | 2091 |  1595.0323 |  79.579916 | 4.307022 | 45.82789 |\n",
       "| IA-277 | t | IA-2        |  544 | 2181 |  1505.9409 |  74.187341 | 4.324032 | 45.65056 |\n",
       "| nAChRalpha677 | t | nAChRalpha6 | 1243 | 5082 |  3430.8495 | 233.213158 | 3.872672 | 45.48364 |\n",
       "| CG432878 | t | CG4328      |  104 |  136 |   292.6234 |   5.276942 | 5.542839 | 45.43921 |\n",
       "| 5-HT1A77 | t | 5-HT1A      |  302 |  745 |   830.2874 |  32.897463 | 4.614362 | 44.75563 |\n",
       "| dpr1377 | t | dpr13       |  367 | 1038 |  1019.1029 |  45.986889 | 4.438898 | 44.36456 |\n",
       "| SerT77 | t | SerT        |   81 |   94 |   240.4180 |   3.991988 | 5.589786 | 44.24533 |\n",
       "\n"
      ],
      "text/plain": [
       "              cell.type.2 gene        cell.type.2.n.umi other.n.umi\n",
       "Vmat77        t           Vmat        3511              1741       \n",
       "Tbh77         t           Tbh          582               179       \n",
       "Tdc277        t           Tdc2         203               119       \n",
       "CG3119184     t           CG31191      224               213       \n",
       "CG226977      t           CG2269       252               239       \n",
       "CG4500277     t           CG45002      725              1555       \n",
       "SLO277        t           SLO2         504              1064       \n",
       "GluRIB77      t           GluRIB       330               631       \n",
       "DAT77         t           DAT          126               184       \n",
       "cpx77         t           cpx          845              2931       \n",
       "DIP-eta77     t           DIP-eta      208               300       \n",
       "bma77         t           bma          167               252       \n",
       "dpr2177       t           dpr21        365               849       \n",
       "jus77         t           jus          553              2091       \n",
       "IA-277        t           IA-2         544              2181       \n",
       "nAChRalpha677 t           nAChRalpha6 1243              5082       \n",
       "CG432878      t           CG4328       104               136       \n",
       "5-HT1A77      t           5-HT1A       302               745       \n",
       "dpr1377       t           dpr13        367              1038       \n",
       "SerT77        t           SerT          81                94       \n",
       "              cell.type.2.tpm other.tpm  log2.ratio heuristic.score\n",
       "Vmat77        10124.6978       60.350716 7.366583   98.01779       \n",
       "Tbh77          1643.1032        6.388548 7.796917   83.29512       \n",
       "Tdc277          603.9666        4.077435 6.894225   63.70754       \n",
       "CG3119184       679.0681        8.786192 6.116665   57.55498       \n",
       "CG226977        715.7692       10.506789 5.958937   56.52269       \n",
       "CG4500277      2033.4604       69.040869 4.859596   53.40905       \n",
       "SLO277         1383.0233       45.940672 4.880843   50.92990       \n",
       "GluRIB77        936.9447       27.792948 5.024176   49.60550       \n",
       "DAT77           399.6778        6.840168 5.671809   49.04016       \n",
       "cpx77          2458.3679      124.621367 4.290547   48.32903       \n",
       "DIP-eta77       543.3513       12.685593 5.311155   48.26987       \n",
       "bma77           475.3713       10.288676 5.396107   48.00346       \n",
       "dpr2177        1017.8617       38.792918 4.676886   46.73492       \n",
       "jus77          1595.0323       79.579916 4.307022   45.82789       \n",
       "IA-277         1505.9409       74.187341 4.324032   45.65056       \n",
       "nAChRalpha677  3430.8495      233.213158 3.872672   45.48364       \n",
       "CG432878        292.6234        5.276942 5.542839   45.43921       \n",
       "5-HT1A77        830.2874       32.897463 4.614362   44.75563       \n",
       "dpr1377        1019.1029       45.986889 4.438898   44.36456       \n",
       "SerT77          240.4180        3.991988 5.589786   44.24533       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"t\")  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Vmat\"), size_pos = 1, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Vmat\"), size_pos = 1, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Vmat\"), \n",
    "                   size_pos = 1, size_neg = 0.05, stroke = 0.008, limit = 4)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Tbh\"), \n",
    "                   size_pos = 1, size_neg = 0.1, stroke = 0.02, limit = 4)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Tdc2\"), \n",
    "                   size_pos = 3, size_neg = 0.2, stroke = 0.02, limit = 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(t)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = -1, intercept = -9.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(-4, 2) + ylim(-9, -3)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(t)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus & \n",
    "              df_sub300$umap_2 >= -1*umap_1 -9.8), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = -1, intercept = -9.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(-4, 2) + ylim(-9, -3)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(t)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus & \n",
    "              df_sub300$umap_2 < -1*umap_1 -9.8), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = -1, intercept = -9.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(-4, 2) + ylim(-9, -3)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 943,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 414\n",
      "\n",
      "# of cells sampled in set 2: 547\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG14459</th><td>CG14459</td><td>  0</td><td> 34</td><td>   0.000000</td><td> 144.18660</td><td>Set 2</td><td>7.171793</td><td>51.50613</td></tr>\n",
       "\t<tr><th scope=row>Tbh</th><td>Tbh    </td><td>555</td><td> 43</td><td>3433.693516</td><td> 216.95094</td><td>Set 1</td><td>3.977686</td><td>46.72176</td></tr>\n",
       "\t<tr><th scope=row>DIP-eta</th><td>DIP-eta</td><td>  6</td><td>209</td><td>  39.463102</td><td> 945.39880</td><td>Set 2</td><td>4.546244</td><td>44.94555</td></tr>\n",
       "\t<tr><th scope=row>Tdc2</th><td>Tdc2   </td><td>199</td><td> 12</td><td>1313.603998</td><td>  65.46006</td><td>Set 1</td><td>4.304899</td><td>44.60053</td></tr>\n",
       "\t<tr><th scope=row>CG1688</th><td>CG1688 </td><td>  2</td><td> 98</td><td>  18.724488</td><td> 491.28494</td><td>Set 2</td><td>4.638500</td><td>41.48373</td></tr>\n",
       "\t<tr><th scope=row>odd</th><td>odd    </td><td>  0</td><td> 16</td><td>   0.000000</td><td>  84.05808</td><td>Set 2</td><td>6.393315</td><td>40.98355</td></tr>\n",
       "\t<tr><th scope=row>CG6154</th><td>CG6154 </td><td>  1</td><td> 33</td><td>   2.181986</td><td> 149.58077</td><td>Set 2</td><td>5.554853</td><td>40.18600</td></tr>\n",
       "\t<tr><th scope=row>TwdlW</th><td>TwdlW  </td><td> 11</td><td>  0</td><td>  77.280046</td><td>   0.00000</td><td>Set 1</td><td>6.272024</td><td>39.45462</td></tr>\n",
       "\t<tr><th scope=row>Hmr</th><td>Hmr    </td><td>  0</td><td> 13</td><td>   0.000000</td><td>  69.88530</td><td>Set 2</td><td>6.126917</td><td>37.66470</td></tr>\n",
       "\t<tr><th scope=row>SLO2</th><td>SLO2   </td><td> 44</td><td>527</td><td> 248.030610</td><td>2443.48937</td><td>Set 2</td><td>3.294548</td><td>37.08118</td></tr>\n",
       "\t<tr><th scope=row>Lip4</th><td>Lip4   </td><td>  0</td><td> 15</td><td>   0.000000</td><td>  65.69895</td><td>Set 2</td><td>6.037798</td><td>36.58660</td></tr>\n",
       "\t<tr><th scope=row>dpr17</th><td>dpr17  </td><td>  0</td><td> 14</td><td>   0.000000</td><td>  63.04611</td><td>Set 2</td><td>5.978335</td><td>35.87622</td></tr>\n",
       "\t<tr><th scope=row>plh</th><td>plh    </td><td>  0</td><td> 13</td><td>   0.000000</td><td>  62.66436</td><td>Set 2</td><td>5.969573</td><td>35.77215</td></tr>\n",
       "\t<tr><th scope=row>Als2</th><td>Als2   </td><td>  0</td><td> 12</td><td>   0.000000</td><td>  61.53575</td><td>Set 2</td><td>5.943353</td><td>35.46167</td></tr>\n",
       "\t<tr><th scope=row>zpg</th><td>zpg    </td><td>  0</td><td> 11</td><td>   0.000000</td><td>  59.33279</td><td>Set 2</td><td>5.890758</td><td>34.84307</td></tr>\n",
       "\t<tr><th scope=row>Proc</th><td>Proc   </td><td>  2</td><td> 41</td><td>   9.362244</td><td> 226.22446</td><td>Set 2</td><td>4.448347</td><td>34.82154</td></tr>\n",
       "\t<tr><th scope=row>CG33958</th><td>CG33958</td><td>  0</td><td> 10</td><td>   0.000000</td><td>  58.88103</td><td>Set 2</td><td>5.879731</td><td>34.71409</td></tr>\n",
       "\t<tr><th scope=row>Syb</th><td>Syb    </td><td>  0</td><td> 12</td><td>   0.000000</td><td>  58.27078</td><td>Set 2</td><td>5.864701</td><td>34.53868</td></tr>\n",
       "\t<tr><th scope=row>CG10483</th><td>CG10483</td><td>  0</td><td> 10</td><td>   0.000000</td><td>  57.29031</td><td>Set 2</td><td>5.840219</td><td>34.25396</td></tr>\n",
       "\t<tr><th scope=row>CG4375</th><td>CG4375 </td><td>  0</td><td> 12</td><td>   0.000000</td><td>  56.18891</td><td>Set 2</td><td>5.812214</td><td>33.92975</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG14459 & CG14459 &   0 &  34 &    0.000000 &  144.18660 & Set 2 & 7.171793 & 51.50613\\\\\n",
       "\tTbh & Tbh     & 555 &  43 & 3433.693516 &  216.95094 & Set 1 & 3.977686 & 46.72176\\\\\n",
       "\tDIP-eta & DIP-eta &   6 & 209 &   39.463102 &  945.39880 & Set 2 & 4.546244 & 44.94555\\\\\n",
       "\tTdc2 & Tdc2    & 199 &  12 & 1313.603998 &   65.46006 & Set 1 & 4.304899 & 44.60053\\\\\n",
       "\tCG1688 & CG1688  &   2 &  98 &   18.724488 &  491.28494 & Set 2 & 4.638500 & 41.48373\\\\\n",
       "\todd & odd     &   0 &  16 &    0.000000 &   84.05808 & Set 2 & 6.393315 & 40.98355\\\\\n",
       "\tCG6154 & CG6154  &   1 &  33 &    2.181986 &  149.58077 & Set 2 & 5.554853 & 40.18600\\\\\n",
       "\tTwdlW & TwdlW   &  11 &   0 &   77.280046 &    0.00000 & Set 1 & 6.272024 & 39.45462\\\\\n",
       "\tHmr & Hmr     &   0 &  13 &    0.000000 &   69.88530 & Set 2 & 6.126917 & 37.66470\\\\\n",
       "\tSLO2 & SLO2    &  44 & 527 &  248.030610 & 2443.48937 & Set 2 & 3.294548 & 37.08118\\\\\n",
       "\tLip4 & Lip4    &   0 &  15 &    0.000000 &   65.69895 & Set 2 & 6.037798 & 36.58660\\\\\n",
       "\tdpr17 & dpr17   &   0 &  14 &    0.000000 &   63.04611 & Set 2 & 5.978335 & 35.87622\\\\\n",
       "\tplh & plh     &   0 &  13 &    0.000000 &   62.66436 & Set 2 & 5.969573 & 35.77215\\\\\n",
       "\tAls2 & Als2    &   0 &  12 &    0.000000 &   61.53575 & Set 2 & 5.943353 & 35.46167\\\\\n",
       "\tzpg & zpg     &   0 &  11 &    0.000000 &   59.33279 & Set 2 & 5.890758 & 34.84307\\\\\n",
       "\tProc & Proc    &   2 &  41 &    9.362244 &  226.22446 & Set 2 & 4.448347 & 34.82154\\\\\n",
       "\tCG33958 & CG33958 &   0 &  10 &    0.000000 &   58.88103 & Set 2 & 5.879731 & 34.71409\\\\\n",
       "\tSyb & Syb     &   0 &  12 &    0.000000 &   58.27078 & Set 2 & 5.864701 & 34.53868\\\\\n",
       "\tCG10483 & CG10483 &   0 &  10 &    0.000000 &   57.29031 & Set 2 & 5.840219 & 34.25396\\\\\n",
       "\tCG4375 & CG4375  &   0 &  12 &    0.000000 &   56.18891 & Set 2 & 5.812214 & 33.92975\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG14459 | CG14459 |   0 |  34 |    0.000000 |  144.18660 | Set 2 | 7.171793 | 51.50613 |\n",
       "| Tbh | Tbh     | 555 |  43 | 3433.693516 |  216.95094 | Set 1 | 3.977686 | 46.72176 |\n",
       "| DIP-eta | DIP-eta |   6 | 209 |   39.463102 |  945.39880 | Set 2 | 4.546244 | 44.94555 |\n",
       "| Tdc2 | Tdc2    | 199 |  12 | 1313.603998 |   65.46006 | Set 1 | 4.304899 | 44.60053 |\n",
       "| CG1688 | CG1688  |   2 |  98 |   18.724488 |  491.28494 | Set 2 | 4.638500 | 41.48373 |\n",
       "| odd | odd     |   0 |  16 |    0.000000 |   84.05808 | Set 2 | 6.393315 | 40.98355 |\n",
       "| CG6154 | CG6154  |   1 |  33 |    2.181986 |  149.58077 | Set 2 | 5.554853 | 40.18600 |\n",
       "| TwdlW | TwdlW   |  11 |   0 |   77.280046 |    0.00000 | Set 1 | 6.272024 | 39.45462 |\n",
       "| Hmr | Hmr     |   0 |  13 |    0.000000 |   69.88530 | Set 2 | 6.126917 | 37.66470 |\n",
       "| SLO2 | SLO2    |  44 | 527 |  248.030610 | 2443.48937 | Set 2 | 3.294548 | 37.08118 |\n",
       "| Lip4 | Lip4    |   0 |  15 |    0.000000 |   65.69895 | Set 2 | 6.037798 | 36.58660 |\n",
       "| dpr17 | dpr17   |   0 |  14 |    0.000000 |   63.04611 | Set 2 | 5.978335 | 35.87622 |\n",
       "| plh | plh     |   0 |  13 |    0.000000 |   62.66436 | Set 2 | 5.969573 | 35.77215 |\n",
       "| Als2 | Als2    |   0 |  12 |    0.000000 |   61.53575 | Set 2 | 5.943353 | 35.46167 |\n",
       "| zpg | zpg     |   0 |  11 |    0.000000 |   59.33279 | Set 2 | 5.890758 | 34.84307 |\n",
       "| Proc | Proc    |   2 |  41 |    9.362244 |  226.22446 | Set 2 | 4.448347 | 34.82154 |\n",
       "| CG33958 | CG33958 |   0 |  10 |    0.000000 |   58.88103 | Set 2 | 5.879731 | 34.71409 |\n",
       "| Syb | Syb     |   0 |  12 |    0.000000 |   58.27078 | Set 2 | 5.864701 | 34.53868 |\n",
       "| CG10483 | CG10483 |   0 |  10 |    0.000000 |   57.29031 | Set 2 | 5.840219 | 34.25396 |\n",
       "| CG4375 | CG4375  |   0 |  12 |    0.000000 |   56.18891 | Set 2 | 5.812214 | 33.92975 |\n",
       "\n"
      ],
      "text/plain": [
       "        gene    set.1.n.umi set.2.n.umi set.1.tpm   set.2.tpm  higher.expr\n",
       "CG14459 CG14459   0          34            0.000000  144.18660 Set 2      \n",
       "Tbh     Tbh     555          43         3433.693516  216.95094 Set 1      \n",
       "DIP-eta DIP-eta   6         209           39.463102  945.39880 Set 2      \n",
       "Tdc2    Tdc2    199          12         1313.603998   65.46006 Set 1      \n",
       "CG1688  CG1688    2          98           18.724488  491.28494 Set 2      \n",
       "odd     odd       0          16            0.000000   84.05808 Set 2      \n",
       "CG6154  CG6154    1          33            2.181986  149.58077 Set 2      \n",
       "TwdlW   TwdlW    11           0           77.280046    0.00000 Set 1      \n",
       "Hmr     Hmr       0          13            0.000000   69.88530 Set 2      \n",
       "SLO2    SLO2     44         527          248.030610 2443.48937 Set 2      \n",
       "Lip4    Lip4      0          15            0.000000   65.69895 Set 2      \n",
       "dpr17   dpr17     0          14            0.000000   63.04611 Set 2      \n",
       "plh     plh       0          13            0.000000   62.66436 Set 2      \n",
       "Als2    Als2      0          12            0.000000   61.53575 Set 2      \n",
       "zpg     zpg       0          11            0.000000   59.33279 Set 2      \n",
       "Proc    Proc      2          41            9.362244  226.22446 Set 2      \n",
       "CG33958 CG33958   0          10            0.000000   58.88103 Set 2      \n",
       "Syb     Syb       0          12            0.000000   58.27078 Set 2      \n",
       "CG10483 CG10483   0          10            0.000000   57.29031 Set 2      \n",
       "CG4375  CG4375    0          12            0.000000   56.18891 Set 2      \n",
       "        log2.ratio heuristic.score\n",
       "CG14459 7.171793   51.50613       \n",
       "Tbh     3.977686   46.72176       \n",
       "DIP-eta 4.546244   44.94555       \n",
       "Tdc2    4.304899   44.60053       \n",
       "CG1688  4.638500   41.48373       \n",
       "odd     6.393315   40.98355       \n",
       "CG6154  5.554853   40.18600       \n",
       "TwdlW   6.272024   39.45462       \n",
       "Hmr     6.126917   37.66470       \n",
       "SLO2    3.294548   37.08118       \n",
       "Lip4    6.037798   36.58660       \n",
       "dpr17   5.978335   35.87622       \n",
       "plh     5.969573   35.77215       \n",
       "Als2    5.943353   35.46167       \n",
       "zpg     5.890758   34.84307       \n",
       "Proc    4.448347   34.82154       \n",
       "CG33958 5.879731   34.71409       \n",
       "Syb     5.864701   34.53868       \n",
       "CG10483 5.840219   34.25396       \n",
       "CG4375  5.812214   33.92975       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "two.set.differential.gene.test.3a(cds_sub_no_doubletons, \n",
    "                                  df_sub300_no_doubletons$cell.type.2 == \"t\" & \n",
    "                                  df_sub300_no_doubletons$umap_2 < -1*df_sub300_no_doubletons$umap_1 -9.8 , \n",
    "                                 df_sub300_no_doubletons$cell.type.2 == \"t\" & \n",
    "                                  df_sub300_no_doubletons$umap_2 >= -1*df_sub300_no_doubletons$umap_1 -9.8)  %>% \n",
    "head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"DIP-eta\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"DIP-eta\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"DIP-eta\"),  # Is expressed in optic lobe intrinsic columnar neurons\n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 4)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG1688\"), \n",
    "                   size_pos = 2, size_neg = 0.1, stroke = 0.02, limit = 4)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"SLO2\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type u\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"u\", df_sub300$cluster %in% u)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'u' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'u' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(u)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 961,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cluster</th><th scope=col>cell.type.2</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>157</td><td>u</td><td>610</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 3\n",
       "\\begin{tabular}{lll}\n",
       " cluster & cell.type.2 & n\\\\\n",
       " <fct> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t 157 & u & 610\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 3\n",
       "\n",
       "| cluster &lt;fct&gt; | cell.type.2 &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| 157 | u | 610 |\n",
       "\n"
      ],
      "text/plain": [
       "  cluster cell.type.2 n  \n",
       "1 157     u           610"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub300  %>% count(cluster, cell.type.2)  %>% filter(cell.type.2 == \"u\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 957,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG1409578</th><td>u</td><td>CG14095</td><td>678</td><td>363</td><td>3563.2732</td><td>13.9285223</td><td>7.898988</td><td>93.20326</td></tr>\n",
       "\t<tr><th scope=row>Ccp84Ae78</th><td>u</td><td>Ccp84Ae</td><td>302</td><td> 93</td><td>1612.0340</td><td> 3.1663426</td><td>8.595885</td><td>91.59398</td></tr>\n",
       "\t<tr><th scope=row>Irk278</th><td>u</td><td>Irk2   </td><td>258</td><td> 36</td><td> 915.8146</td><td> 1.0524391</td><td>8.801572</td><td>86.61175</td></tr>\n",
       "\t<tr><th scope=row>CG854378</th><td>u</td><td>CG8543 </td><td>191</td><td> 10</td><td> 770.4581</td><td> 0.4802334</td><td>9.023748</td><td>86.55077</td></tr>\n",
       "\t<tr><th scope=row>CG854178</th><td>u</td><td>CG8541 </td><td>150</td><td> 18</td><td> 586.4125</td><td> 0.6465243</td><td>8.476348</td><td>77.96740</td></tr>\n",
       "\t<tr><th scope=row>CG1367478</th><td>u</td><td>CG13674</td><td>253</td><td>180</td><td>1334.0015</td><td> 6.5432938</td><td>7.466350</td><td>77.52032</td></tr>\n",
       "\t<tr><th scope=row>CG1305178</th><td>u</td><td>CG13051</td><td>196</td><td> 93</td><td> 930.2568</td><td> 3.1252837</td><td>7.816992</td><td>77.09927</td></tr>\n",
       "\t<tr><th scope=row>CG507078</th><td>u</td><td>CG5070 </td><td>212</td><td>120</td><td> 912.3066</td><td> 4.5339374</td><td>7.365069</td><td>72.43512</td></tr>\n",
       "\t<tr><th scope=row>CG1829478</th><td>u</td><td>CG18294</td><td>120</td><td> 65</td><td> 605.7769</td><td> 2.0218477</td><td>7.647212</td><td>70.69864</td></tr>\n",
       "\t<tr><th scope=row>CG1457078</th><td>u</td><td>CG14570</td><td> 93</td><td> 27</td><td> 446.9857</td><td> 0.7295196</td><td>8.013714</td><td>70.57925</td></tr>\n",
       "\t<tr><th scope=row>CG1367878</th><td>u</td><td>CG13678</td><td>191</td><td>179</td><td> 942.5535</td><td> 5.7751733</td><td>7.120173</td><td>70.36127</td></tr>\n",
       "\t<tr><th scope=row>CG926978</th><td>u</td><td>CG9269 </td><td>179</td><td> 89</td><td> 762.0696</td><td> 3.7803183</td><td>7.316672</td><td>70.06204</td></tr>\n",
       "\t<tr><th scope=row>CG1306978</th><td>u</td><td>CG13069</td><td>100</td><td> 35</td><td> 509.4775</td><td> 1.3337787</td><td>7.770207</td><td>69.89848</td></tr>\n",
       "\t<tr><th scope=row>CG1304778</th><td>u</td><td>CG13047</td><td>449</td><td>745</td><td>2139.0407</td><td>26.3725992</td><td>6.288088</td><td>69.56777</td></tr>\n",
       "\t<tr><th scope=row>CG3221478</th><td>u</td><td>CG32214</td><td>115</td><td> 69</td><td> 582.7057</td><td> 2.3294431</td><td>7.451343</td><td>68.47111</td></tr>\n",
       "\t<tr><th scope=row>CG1409678</th><td>u</td><td>CG14096</td><td>116</td><td>112</td><td> 628.2733</td><td> 3.7265997</td><td>7.054446</td><td>65.58901</td></tr>\n",
       "\t<tr><th scope=row>CG1306678</th><td>u</td><td>CG13066</td><td> 82</td><td> 51</td><td> 423.3523</td><td> 1.3186739</td><td>7.512415</td><td>65.57676</td></tr>\n",
       "\t<tr><th scope=row>CG1306878</th><td>u</td><td>CG13068</td><td> 80</td><td> 31</td><td> 387.7446</td><td> 1.1929848</td><td>7.466067</td><td>64.22818</td></tr>\n",
       "\t<tr><th scope=row>CG3223778</th><td>u</td><td>CG32237</td><td>127</td><td> 99</td><td> 525.2722</td><td> 3.0048336</td><td>7.035179</td><td>63.59566</td></tr>\n",
       "\t<tr><th scope=row>CG1367978</th><td>u</td><td>CG13679</td><td> 84</td><td> 73</td><td> 477.2075</td><td> 2.4616439</td><td>7.107016</td><td>63.26305</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG1409578 & u & CG14095 & 678 & 363 & 3563.2732 & 13.9285223 & 7.898988 & 93.20326\\\\\n",
       "\tCcp84Ae78 & u & Ccp84Ae & 302 &  93 & 1612.0340 &  3.1663426 & 8.595885 & 91.59398\\\\\n",
       "\tIrk278 & u & Irk2    & 258 &  36 &  915.8146 &  1.0524391 & 8.801572 & 86.61175\\\\\n",
       "\tCG854378 & u & CG8543  & 191 &  10 &  770.4581 &  0.4802334 & 9.023748 & 86.55077\\\\\n",
       "\tCG854178 & u & CG8541  & 150 &  18 &  586.4125 &  0.6465243 & 8.476348 & 77.96740\\\\\n",
       "\tCG1367478 & u & CG13674 & 253 & 180 & 1334.0015 &  6.5432938 & 7.466350 & 77.52032\\\\\n",
       "\tCG1305178 & u & CG13051 & 196 &  93 &  930.2568 &  3.1252837 & 7.816992 & 77.09927\\\\\n",
       "\tCG507078 & u & CG5070  & 212 & 120 &  912.3066 &  4.5339374 & 7.365069 & 72.43512\\\\\n",
       "\tCG1829478 & u & CG18294 & 120 &  65 &  605.7769 &  2.0218477 & 7.647212 & 70.69864\\\\\n",
       "\tCG1457078 & u & CG14570 &  93 &  27 &  446.9857 &  0.7295196 & 8.013714 & 70.57925\\\\\n",
       "\tCG1367878 & u & CG13678 & 191 & 179 &  942.5535 &  5.7751733 & 7.120173 & 70.36127\\\\\n",
       "\tCG926978 & u & CG9269  & 179 &  89 &  762.0696 &  3.7803183 & 7.316672 & 70.06204\\\\\n",
       "\tCG1306978 & u & CG13069 & 100 &  35 &  509.4775 &  1.3337787 & 7.770207 & 69.89848\\\\\n",
       "\tCG1304778 & u & CG13047 & 449 & 745 & 2139.0407 & 26.3725992 & 6.288088 & 69.56777\\\\\n",
       "\tCG3221478 & u & CG32214 & 115 &  69 &  582.7057 &  2.3294431 & 7.451343 & 68.47111\\\\\n",
       "\tCG1409678 & u & CG14096 & 116 & 112 &  628.2733 &  3.7265997 & 7.054446 & 65.58901\\\\\n",
       "\tCG1306678 & u & CG13066 &  82 &  51 &  423.3523 &  1.3186739 & 7.512415 & 65.57676\\\\\n",
       "\tCG1306878 & u & CG13068 &  80 &  31 &  387.7446 &  1.1929848 & 7.466067 & 64.22818\\\\\n",
       "\tCG3223778 & u & CG32237 & 127 &  99 &  525.2722 &  3.0048336 & 7.035179 & 63.59566\\\\\n",
       "\tCG1367978 & u & CG13679 &  84 &  73 &  477.2075 &  2.4616439 & 7.107016 & 63.26305\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG1409578 | u | CG14095 | 678 | 363 | 3563.2732 | 13.9285223 | 7.898988 | 93.20326 |\n",
       "| Ccp84Ae78 | u | Ccp84Ae | 302 |  93 | 1612.0340 |  3.1663426 | 8.595885 | 91.59398 |\n",
       "| Irk278 | u | Irk2    | 258 |  36 |  915.8146 |  1.0524391 | 8.801572 | 86.61175 |\n",
       "| CG854378 | u | CG8543  | 191 |  10 |  770.4581 |  0.4802334 | 9.023748 | 86.55077 |\n",
       "| CG854178 | u | CG8541  | 150 |  18 |  586.4125 |  0.6465243 | 8.476348 | 77.96740 |\n",
       "| CG1367478 | u | CG13674 | 253 | 180 | 1334.0015 |  6.5432938 | 7.466350 | 77.52032 |\n",
       "| CG1305178 | u | CG13051 | 196 |  93 |  930.2568 |  3.1252837 | 7.816992 | 77.09927 |\n",
       "| CG507078 | u | CG5070  | 212 | 120 |  912.3066 |  4.5339374 | 7.365069 | 72.43512 |\n",
       "| CG1829478 | u | CG18294 | 120 |  65 |  605.7769 |  2.0218477 | 7.647212 | 70.69864 |\n",
       "| CG1457078 | u | CG14570 |  93 |  27 |  446.9857 |  0.7295196 | 8.013714 | 70.57925 |\n",
       "| CG1367878 | u | CG13678 | 191 | 179 |  942.5535 |  5.7751733 | 7.120173 | 70.36127 |\n",
       "| CG926978 | u | CG9269  | 179 |  89 |  762.0696 |  3.7803183 | 7.316672 | 70.06204 |\n",
       "| CG1306978 | u | CG13069 | 100 |  35 |  509.4775 |  1.3337787 | 7.770207 | 69.89848 |\n",
       "| CG1304778 | u | CG13047 | 449 | 745 | 2139.0407 | 26.3725992 | 6.288088 | 69.56777 |\n",
       "| CG3221478 | u | CG32214 | 115 |  69 |  582.7057 |  2.3294431 | 7.451343 | 68.47111 |\n",
       "| CG1409678 | u | CG14096 | 116 | 112 |  628.2733 |  3.7265997 | 7.054446 | 65.58901 |\n",
       "| CG1306678 | u | CG13066 |  82 |  51 |  423.3523 |  1.3186739 | 7.512415 | 65.57676 |\n",
       "| CG1306878 | u | CG13068 |  80 |  31 |  387.7446 |  1.1929848 | 7.466067 | 64.22818 |\n",
       "| CG3223778 | u | CG32237 | 127 |  99 |  525.2722 |  3.0048336 | 7.035179 | 63.59566 |\n",
       "| CG1367978 | u | CG13679 |  84 |  73 |  477.2075 |  2.4616439 | 7.107016 | 63.26305 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "CG1409578 u           CG14095 678               363         3563.2732      \n",
       "Ccp84Ae78 u           Ccp84Ae 302                93         1612.0340      \n",
       "Irk278    u           Irk2    258                36          915.8146      \n",
       "CG854378  u           CG8543  191                10          770.4581      \n",
       "CG854178  u           CG8541  150                18          586.4125      \n",
       "CG1367478 u           CG13674 253               180         1334.0015      \n",
       "CG1305178 u           CG13051 196                93          930.2568      \n",
       "CG507078  u           CG5070  212               120          912.3066      \n",
       "CG1829478 u           CG18294 120                65          605.7769      \n",
       "CG1457078 u           CG14570  93                27          446.9857      \n",
       "CG1367878 u           CG13678 191               179          942.5535      \n",
       "CG926978  u           CG9269  179                89          762.0696      \n",
       "CG1306978 u           CG13069 100                35          509.4775      \n",
       "CG1304778 u           CG13047 449               745         2139.0407      \n",
       "CG3221478 u           CG32214 115                69          582.7057      \n",
       "CG1409678 u           CG14096 116               112          628.2733      \n",
       "CG1306678 u           CG13066  82                51          423.3523      \n",
       "CG1306878 u           CG13068  80                31          387.7446      \n",
       "CG3223778 u           CG32237 127                99          525.2722      \n",
       "CG1367978 u           CG13679  84                73          477.2075      \n",
       "          other.tpm  log2.ratio heuristic.score\n",
       "CG1409578 13.9285223 7.898988   93.20326       \n",
       "Ccp84Ae78  3.1663426 8.595885   91.59398       \n",
       "Irk278     1.0524391 8.801572   86.61175       \n",
       "CG854378   0.4802334 9.023748   86.55077       \n",
       "CG854178   0.6465243 8.476348   77.96740       \n",
       "CG1367478  6.5432938 7.466350   77.52032       \n",
       "CG1305178  3.1252837 7.816992   77.09927       \n",
       "CG507078   4.5339374 7.365069   72.43512       \n",
       "CG1829478  2.0218477 7.647212   70.69864       \n",
       "CG1457078  0.7295196 8.013714   70.57925       \n",
       "CG1367878  5.7751733 7.120173   70.36127       \n",
       "CG926978   3.7803183 7.316672   70.06204       \n",
       "CG1306978  1.3337787 7.770207   69.89848       \n",
       "CG1304778 26.3725992 6.288088   69.56777       \n",
       "CG3221478  2.3294431 7.451343   68.47111       \n",
       "CG1409678  3.7265997 7.054446   65.58901       \n",
       "CG1306678  1.3186739 7.512415   65.57676       \n",
       "CG1306878  1.1929848 7.466067   64.22818       \n",
       "CG3223778  3.0048336 7.035179   63.59566       \n",
       "CG1367978  2.4616439 7.107016   63.26305       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"u\")  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG14095\"), size_pos = 3, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG14095\"), size_pos = 3, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG14095\"), \n",
    "                   size_pos = 3, size_neg = 0.05, stroke = 0.008, limit = 3)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Ccp84Ae\"), \n",
    "                   size_pos = 3, size_neg = 0.1, stroke = 0.02, limit = 3)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Irk2\"), \n",
    "                   size_pos = 3, size_neg = 0.2, stroke = 0.02, limit = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 965,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Plasmatocytes\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Plasmatocytes\", df_sub300$cluster %in% v)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'v' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'v' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(v)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 963,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Ppn79</th><td>v</td><td>Ppn        </td><td>70207</td><td>14001</td><td>35737.5046</td><td>517.641752</td><td>6.106557</td><td>92.36284</td></tr>\n",
       "\t<tr><th scope=row>Pxn79</th><td>v</td><td>Pxn        </td><td>12373</td><td> 2288</td><td> 6346.7825</td><td> 85.559596</td><td>6.196188</td><td>78.27047</td></tr>\n",
       "\t<tr><th scope=row>Col4a179</th><td>v</td><td>Col4a1     </td><td>32538</td><td>12541</td><td>16628.8867</td><td>493.331004</td><td>5.072070</td><td>71.11799</td></tr>\n",
       "\t<tr><th scope=row>vkg79</th><td>v</td><td>vkg        </td><td>20725</td><td> 8194</td><td>10547.7162</td><td>322.180377</td><td>5.028447</td><td>67.20409</td></tr>\n",
       "\t<tr><th scope=row>NtR79</th><td>v</td><td>NtR        </td><td> 1936</td><td>  444</td><td> 1010.1096</td><td> 17.426035</td><td>5.776622</td><td>57.66065</td></tr>\n",
       "\t<tr><th scope=row>NimC479</th><td>v</td><td>NimC4      </td><td> 5948</td><td> 2861</td><td> 3117.9010</td><td>105.681643</td><td>4.869191</td><td>56.51584</td></tr>\n",
       "\t<tr><th scope=row>Ance-579</th><td>v</td><td>Ance-5     </td><td>  920</td><td>  230</td><td>  478.2397</td><td>  8.224957</td><td>5.696048</td><td>50.72105</td></tr>\n",
       "\t<tr><th scope=row>Plod79</th><td>v</td><td>Plod       </td><td>14578</td><td>11757</td><td> 7469.3525</td><td>495.350830</td><td>3.911551</td><td>50.32977</td></tr>\n",
       "\t<tr><th scope=row>Karl79</th><td>v</td><td>Karl       </td><td>  845</td><td>  194</td><td>  434.3417</td><td>  7.252419</td><td>5.717870</td><td>50.12287</td></tr>\n",
       "\t<tr><th scope=row>Sr-CI79</th><td>v</td><td>Sr-CI      </td><td>  877</td><td>  233</td><td>  457.9558</td><td>  8.018820</td><td>5.666126</td><td>50.10108</td></tr>\n",
       "\t<tr><th scope=row>CG109289</th><td>v</td><td>CG1092     </td><td> 1202</td><td>  348</td><td>  624.0653</td><td> 14.026172</td><td>5.376148</td><td>49.93292</td></tr>\n",
       "\t<tr><th scope=row>firl79</th><td>v</td><td>firl       </td><td> 1434</td><td>  500</td><td>  747.2188</td><td> 19.866686</td><td>5.162257</td><td>49.28570</td></tr>\n",
       "\t<tr><th scope=row>CG631079</th><td>v</td><td>CG6310     </td><td> 1674</td><td>  621</td><td>  867.0516</td><td> 26.384512</td><td>4.984686</td><td>48.65869</td></tr>\n",
       "\t<tr><th scope=row>CG3105179</th><td>v</td><td>CG31051    </td><td> 1042</td><td>  290</td><td>  544.0158</td><td> 12.362197</td><td>5.347419</td><td>48.60887</td></tr>\n",
       "\t<tr><th scope=row>nAChRbeta379</th><td>v</td><td>nAChRbeta3 </td><td>  955</td><td>  298</td><td>  505.5809</td><td> 10.949695</td><td>5.402896</td><td>48.54313</td></tr>\n",
       "\t<tr><th scope=row>SPARC79</th><td>v</td><td>SPARC      </td><td> 8547</td><td> 7319</td><td> 4365.6122</td><td>303.784610</td><td>3.840322</td><td>46.43832</td></tr>\n",
       "\t<tr><th scope=row>fat-spondin79</th><td>v</td><td>fat-spondin</td><td> 3706</td><td> 2465</td><td> 1907.2278</td><td>106.383992</td><td>4.150626</td><td>45.23360</td></tr>\n",
       "\t<tr><th scope=row>Tig79</th><td>v</td><td>Tig        </td><td> 9686</td><td> 9502</td><td> 4974.9129</td><td>392.292585</td><td>3.660996</td><td>44.95976</td></tr>\n",
       "\t<tr><th scope=row>Hexo279</th><td>v</td><td>Hexo2      </td><td> 1182</td><td>  548</td><td>  607.9545</td><td> 21.609180</td><td>4.748983</td><td>43.92900</td></tr>\n",
       "\t<tr><th scope=row>CG4379979</th><td>v</td><td>CG43799    </td><td>  388</td><td>   75</td><td>  203.9512</td><td>  3.312434</td><td>5.563578</td><td>42.72348</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tPpn79 & v & Ppn         & 70207 & 14001 & 35737.5046 & 517.641752 & 6.106557 & 92.36284\\\\\n",
       "\tPxn79 & v & Pxn         & 12373 &  2288 &  6346.7825 &  85.559596 & 6.196188 & 78.27047\\\\\n",
       "\tCol4a179 & v & Col4a1      & 32538 & 12541 & 16628.8867 & 493.331004 & 5.072070 & 71.11799\\\\\n",
       "\tvkg79 & v & vkg         & 20725 &  8194 & 10547.7162 & 322.180377 & 5.028447 & 67.20409\\\\\n",
       "\tNtR79 & v & NtR         &  1936 &   444 &  1010.1096 &  17.426035 & 5.776622 & 57.66065\\\\\n",
       "\tNimC479 & v & NimC4       &  5948 &  2861 &  3117.9010 & 105.681643 & 4.869191 & 56.51584\\\\\n",
       "\tAnce-579 & v & Ance-5      &   920 &   230 &   478.2397 &   8.224957 & 5.696048 & 50.72105\\\\\n",
       "\tPlod79 & v & Plod        & 14578 & 11757 &  7469.3525 & 495.350830 & 3.911551 & 50.32977\\\\\n",
       "\tKarl79 & v & Karl        &   845 &   194 &   434.3417 &   7.252419 & 5.717870 & 50.12287\\\\\n",
       "\tSr-CI79 & v & Sr-CI       &   877 &   233 &   457.9558 &   8.018820 & 5.666126 & 50.10108\\\\\n",
       "\tCG109289 & v & CG1092      &  1202 &   348 &   624.0653 &  14.026172 & 5.376148 & 49.93292\\\\\n",
       "\tfirl79 & v & firl        &  1434 &   500 &   747.2188 &  19.866686 & 5.162257 & 49.28570\\\\\n",
       "\tCG631079 & v & CG6310      &  1674 &   621 &   867.0516 &  26.384512 & 4.984686 & 48.65869\\\\\n",
       "\tCG3105179 & v & CG31051     &  1042 &   290 &   544.0158 &  12.362197 & 5.347419 & 48.60887\\\\\n",
       "\tnAChRbeta379 & v & nAChRbeta3  &   955 &   298 &   505.5809 &  10.949695 & 5.402896 & 48.54313\\\\\n",
       "\tSPARC79 & v & SPARC       &  8547 &  7319 &  4365.6122 & 303.784610 & 3.840322 & 46.43832\\\\\n",
       "\tfat-spondin79 & v & fat-spondin &  3706 &  2465 &  1907.2278 & 106.383992 & 4.150626 & 45.23360\\\\\n",
       "\tTig79 & v & Tig         &  9686 &  9502 &  4974.9129 & 392.292585 & 3.660996 & 44.95976\\\\\n",
       "\tHexo279 & v & Hexo2       &  1182 &   548 &   607.9545 &  21.609180 & 4.748983 & 43.92900\\\\\n",
       "\tCG4379979 & v & CG43799     &   388 &    75 &   203.9512 &   3.312434 & 5.563578 & 42.72348\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Ppn79 | v | Ppn         | 70207 | 14001 | 35737.5046 | 517.641752 | 6.106557 | 92.36284 |\n",
       "| Pxn79 | v | Pxn         | 12373 |  2288 |  6346.7825 |  85.559596 | 6.196188 | 78.27047 |\n",
       "| Col4a179 | v | Col4a1      | 32538 | 12541 | 16628.8867 | 493.331004 | 5.072070 | 71.11799 |\n",
       "| vkg79 | v | vkg         | 20725 |  8194 | 10547.7162 | 322.180377 | 5.028447 | 67.20409 |\n",
       "| NtR79 | v | NtR         |  1936 |   444 |  1010.1096 |  17.426035 | 5.776622 | 57.66065 |\n",
       "| NimC479 | v | NimC4       |  5948 |  2861 |  3117.9010 | 105.681643 | 4.869191 | 56.51584 |\n",
       "| Ance-579 | v | Ance-5      |   920 |   230 |   478.2397 |   8.224957 | 5.696048 | 50.72105 |\n",
       "| Plod79 | v | Plod        | 14578 | 11757 |  7469.3525 | 495.350830 | 3.911551 | 50.32977 |\n",
       "| Karl79 | v | Karl        |   845 |   194 |   434.3417 |   7.252419 | 5.717870 | 50.12287 |\n",
       "| Sr-CI79 | v | Sr-CI       |   877 |   233 |   457.9558 |   8.018820 | 5.666126 | 50.10108 |\n",
       "| CG109289 | v | CG1092      |  1202 |   348 |   624.0653 |  14.026172 | 5.376148 | 49.93292 |\n",
       "| firl79 | v | firl        |  1434 |   500 |   747.2188 |  19.866686 | 5.162257 | 49.28570 |\n",
       "| CG631079 | v | CG6310      |  1674 |   621 |   867.0516 |  26.384512 | 4.984686 | 48.65869 |\n",
       "| CG3105179 | v | CG31051     |  1042 |   290 |   544.0158 |  12.362197 | 5.347419 | 48.60887 |\n",
       "| nAChRbeta379 | v | nAChRbeta3  |   955 |   298 |   505.5809 |  10.949695 | 5.402896 | 48.54313 |\n",
       "| SPARC79 | v | SPARC       |  8547 |  7319 |  4365.6122 | 303.784610 | 3.840322 | 46.43832 |\n",
       "| fat-spondin79 | v | fat-spondin |  3706 |  2465 |  1907.2278 | 106.383992 | 4.150626 | 45.23360 |\n",
       "| Tig79 | v | Tig         |  9686 |  9502 |  4974.9129 | 392.292585 | 3.660996 | 44.95976 |\n",
       "| Hexo279 | v | Hexo2       |  1182 |   548 |   607.9545 |  21.609180 | 4.748983 | 43.92900 |\n",
       "| CG4379979 | v | CG43799     |   388 |    75 |   203.9512 |   3.312434 | 5.563578 | 42.72348 |\n",
       "\n"
      ],
      "text/plain": [
       "              cell.type.2 gene        cell.type.2.n.umi other.n.umi\n",
       "Ppn79         v           Ppn         70207             14001      \n",
       "Pxn79         v           Pxn         12373              2288      \n",
       "Col4a179      v           Col4a1      32538             12541      \n",
       "vkg79         v           vkg         20725              8194      \n",
       "NtR79         v           NtR          1936               444      \n",
       "NimC479       v           NimC4        5948              2861      \n",
       "Ance-579      v           Ance-5        920               230      \n",
       "Plod79        v           Plod        14578             11757      \n",
       "Karl79        v           Karl          845               194      \n",
       "Sr-CI79       v           Sr-CI         877               233      \n",
       "CG109289      v           CG1092       1202               348      \n",
       "firl79        v           firl         1434               500      \n",
       "CG631079      v           CG6310       1674               621      \n",
       "CG3105179     v           CG31051      1042               290      \n",
       "nAChRbeta379  v           nAChRbeta3    955               298      \n",
       "SPARC79       v           SPARC        8547              7319      \n",
       "fat-spondin79 v           fat-spondin  3706              2465      \n",
       "Tig79         v           Tig          9686              9502      \n",
       "Hexo279       v           Hexo2        1182               548      \n",
       "CG4379979     v           CG43799       388                75      \n",
       "              cell.type.2.tpm other.tpm  log2.ratio heuristic.score\n",
       "Ppn79         35737.5046      517.641752 6.106557   92.36284       \n",
       "Pxn79          6346.7825       85.559596 6.196188   78.27047       \n",
       "Col4a179      16628.8867      493.331004 5.072070   71.11799       \n",
       "vkg79         10547.7162      322.180377 5.028447   67.20409       \n",
       "NtR79          1010.1096       17.426035 5.776622   57.66065       \n",
       "NimC479        3117.9010      105.681643 4.869191   56.51584       \n",
       "Ance-579        478.2397        8.224957 5.696048   50.72105       \n",
       "Plod79         7469.3525      495.350830 3.911551   50.32977       \n",
       "Karl79          434.3417        7.252419 5.717870   50.12287       \n",
       "Sr-CI79         457.9558        8.018820 5.666126   50.10108       \n",
       "CG109289        624.0653       14.026172 5.376148   49.93292       \n",
       "firl79          747.2188       19.866686 5.162257   49.28570       \n",
       "CG631079        867.0516       26.384512 4.984686   48.65869       \n",
       "CG3105179       544.0158       12.362197 5.347419   48.60887       \n",
       "nAChRbeta379    505.5809       10.949695 5.402896   48.54313       \n",
       "SPARC79        4365.6122      303.784610 3.840322   46.43832       \n",
       "fat-spondin79  1907.2278      106.383992 4.150626   45.23360       \n",
       "Tig79          4974.9129      392.292585 3.660996   44.95976       \n",
       "Hexo279         607.9545       21.609180 4.748983   43.92900       \n",
       "CG4379979       203.9512        3.312434 5.563578   42.72348       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"v\")  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Ppn\"), size_pos = 1, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Ppn\"), size_pos = 1, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Ppn\"), \n",
    "                   size_pos = 1, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Pxn\"), \n",
    "                   size_pos = 1, size_neg = 0.1, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Col4a1\"), \n",
    "                   size_pos = 1, size_neg = 0.2, stroke = 0.02, limit = 8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 969,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Antennal_sense_organs\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Antennal_sense_organs\", df_sub300$cluster %in% w)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'w' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'w' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(w)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 966,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>nompA80</th><td>w</td><td>nompA  </td><td>1553</td><td> 1351</td><td> 7555.3387</td><td>  53.6086022</td><td>7.112224</td><td>91.63014</td></tr>\n",
       "\t<tr><th scope=row>CG1291180</th><td>w</td><td>CG12911</td><td> 825</td><td>  597</td><td> 3894.4554</td><td>  23.3972386</td><td>7.318560</td><td>87.29268</td></tr>\n",
       "\t<tr><th scope=row>CG1331880</th><td>w</td><td>CG13318</td><td> 193</td><td>   94</td><td>  928.5511</td><td>   3.5219592</td><td>7.681890</td><td>75.74643</td></tr>\n",
       "\t<tr><th scope=row>CG1286980</th><td>w</td><td>CG12869</td><td> 198</td><td>  168</td><td>  958.5543</td><td>   6.5469492</td><td>6.988823</td><td>69.23282</td></tr>\n",
       "\t<tr><th scope=row>CG1250780</th><td>w</td><td>CG12507</td><td> 158</td><td>  123</td><td>  759.2947</td><td>   4.6364366</td><td>7.073733</td><td>67.69856</td></tr>\n",
       "\t<tr><th scope=row>CG1427780</th><td>w</td><td>CG14277</td><td>  94</td><td>  105</td><td>  477.8394</td><td>   2.1491981</td><td>7.245398</td><td>64.50866</td></tr>\n",
       "\t<tr><th scope=row>sv80</th><td>w</td><td>sv     </td><td> 910</td><td> 2515</td><td> 4167.8641</td><td> 108.4272849</td><td>5.251264</td><td>63.14875</td></tr>\n",
       "\t<tr><th scope=row>CG1330580</th><td>w</td><td>CG13305</td><td>  81</td><td>  141</td><td>  411.1573</td><td>   5.3392267</td><td>6.019240</td><td>52.28945</td></tr>\n",
       "\t<tr><th scope=row>CG1445680</th><td>w</td><td>CG14456</td><td>  44</td><td>  131</td><td>  226.0664</td><td>   1.9646874</td><td>6.252723</td><td>48.93988</td></tr>\n",
       "\t<tr><th scope=row>pros80</th><td>w</td><td>pros   </td><td> 793</td><td> 6474</td><td> 3595.7738</td><td> 239.0715519</td><td>3.904766</td><td>46.12500</td></tr>\n",
       "\t<tr><th scope=row>cpo80</th><td>w</td><td>cpo    </td><td>3501</td><td>39315</td><td>16297.4284</td><td>1681.1780158</td><td>3.276242</td><td>45.84264</td></tr>\n",
       "\t<tr><th scope=row>jv80</th><td>w</td><td>jv     </td><td> 206</td><td>  966</td><td>  963.4303</td><td>  40.9841698</td><td>4.520263</td><td>44.81178</td></tr>\n",
       "\t<tr><th scope=row>CAP80</th><td>w</td><td>CAP    </td><td> 622</td><td> 5256</td><td> 2942.3193</td><td> 201.5984004</td><td>3.860259</td><td>44.48265</td></tr>\n",
       "\t<tr><th scope=row>CG3236580</th><td>w</td><td>CG32365</td><td> 256</td><td> 1625</td><td> 1188.0722</td><td>  72.4723655</td><td>4.015277</td><td>41.01855</td></tr>\n",
       "\t<tr><th scope=row>CG3260380</th><td>w</td><td>CG32603</td><td>   0</td><td> 2752</td><td>    0.0000</td><td>  79.1172777</td><td>6.305921</td><td>39.87891</td></tr>\n",
       "\t<tr><th scope=row>CG3226080</th><td>w</td><td>CG32260</td><td>  24</td><td>   27</td><td>  107.7366</td><td>   0.9404929</td><td>5.794942</td><td>39.20101</td></tr>\n",
       "\t<tr><th scope=row>CG1531280</th><td>w</td><td>CG15312</td><td> 194</td><td> 1322</td><td>  969.1433</td><td>  63.0535240</td><td>3.919360</td><td>38.88810</td></tr>\n",
       "\t<tr><th scope=row>Rpt280</th><td>w</td><td>Rpt2   </td><td>   0</td><td> 2361</td><td>    0.0000</td><td>  74.5750092</td><td>6.220620</td><td>38.81566</td></tr>\n",
       "\t<tr><th scope=row>CG981580</th><td>w</td><td>CG9815 </td><td>  34</td><td>   79</td><td>  161.7811</td><td>   3.4611047</td><td>5.180498</td><td>38.06003</td></tr>\n",
       "\t<tr><th scope=row>CG1201180</th><td>w</td><td>CG12011</td><td>  32</td><td>36314</td><td>  142.3954</td><td>1664.9756980</td><td>3.537430</td><td>37.85811</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tnompA80 & w & nompA   & 1553 &  1351 &  7555.3387 &   53.6086022 & 7.112224 & 91.63014\\\\\n",
       "\tCG1291180 & w & CG12911 &  825 &   597 &  3894.4554 &   23.3972386 & 7.318560 & 87.29268\\\\\n",
       "\tCG1331880 & w & CG13318 &  193 &    94 &   928.5511 &    3.5219592 & 7.681890 & 75.74643\\\\\n",
       "\tCG1286980 & w & CG12869 &  198 &   168 &   958.5543 &    6.5469492 & 6.988823 & 69.23282\\\\\n",
       "\tCG1250780 & w & CG12507 &  158 &   123 &   759.2947 &    4.6364366 & 7.073733 & 67.69856\\\\\n",
       "\tCG1427780 & w & CG14277 &   94 &   105 &   477.8394 &    2.1491981 & 7.245398 & 64.50866\\\\\n",
       "\tsv80 & w & sv      &  910 &  2515 &  4167.8641 &  108.4272849 & 5.251264 & 63.14875\\\\\n",
       "\tCG1330580 & w & CG13305 &   81 &   141 &   411.1573 &    5.3392267 & 6.019240 & 52.28945\\\\\n",
       "\tCG1445680 & w & CG14456 &   44 &   131 &   226.0664 &    1.9646874 & 6.252723 & 48.93988\\\\\n",
       "\tpros80 & w & pros    &  793 &  6474 &  3595.7738 &  239.0715519 & 3.904766 & 46.12500\\\\\n",
       "\tcpo80 & w & cpo     & 3501 & 39315 & 16297.4284 & 1681.1780158 & 3.276242 & 45.84264\\\\\n",
       "\tjv80 & w & jv      &  206 &   966 &   963.4303 &   40.9841698 & 4.520263 & 44.81178\\\\\n",
       "\tCAP80 & w & CAP     &  622 &  5256 &  2942.3193 &  201.5984004 & 3.860259 & 44.48265\\\\\n",
       "\tCG3236580 & w & CG32365 &  256 &  1625 &  1188.0722 &   72.4723655 & 4.015277 & 41.01855\\\\\n",
       "\tCG3260380 & w & CG32603 &    0 &  2752 &     0.0000 &   79.1172777 & 6.305921 & 39.87891\\\\\n",
       "\tCG3226080 & w & CG32260 &   24 &    27 &   107.7366 &    0.9404929 & 5.794942 & 39.20101\\\\\n",
       "\tCG1531280 & w & CG15312 &  194 &  1322 &   969.1433 &   63.0535240 & 3.919360 & 38.88810\\\\\n",
       "\tRpt280 & w & Rpt2    &    0 &  2361 &     0.0000 &   74.5750092 & 6.220620 & 38.81566\\\\\n",
       "\tCG981580 & w & CG9815  &   34 &    79 &   161.7811 &    3.4611047 & 5.180498 & 38.06003\\\\\n",
       "\tCG1201180 & w & CG12011 &   32 & 36314 &   142.3954 & 1664.9756980 & 3.537430 & 37.85811\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| nompA80 | w | nompA   | 1553 |  1351 |  7555.3387 |   53.6086022 | 7.112224 | 91.63014 |\n",
       "| CG1291180 | w | CG12911 |  825 |   597 |  3894.4554 |   23.3972386 | 7.318560 | 87.29268 |\n",
       "| CG1331880 | w | CG13318 |  193 |    94 |   928.5511 |    3.5219592 | 7.681890 | 75.74643 |\n",
       "| CG1286980 | w | CG12869 |  198 |   168 |   958.5543 |    6.5469492 | 6.988823 | 69.23282 |\n",
       "| CG1250780 | w | CG12507 |  158 |   123 |   759.2947 |    4.6364366 | 7.073733 | 67.69856 |\n",
       "| CG1427780 | w | CG14277 |   94 |   105 |   477.8394 |    2.1491981 | 7.245398 | 64.50866 |\n",
       "| sv80 | w | sv      |  910 |  2515 |  4167.8641 |  108.4272849 | 5.251264 | 63.14875 |\n",
       "| CG1330580 | w | CG13305 |   81 |   141 |   411.1573 |    5.3392267 | 6.019240 | 52.28945 |\n",
       "| CG1445680 | w | CG14456 |   44 |   131 |   226.0664 |    1.9646874 | 6.252723 | 48.93988 |\n",
       "| pros80 | w | pros    |  793 |  6474 |  3595.7738 |  239.0715519 | 3.904766 | 46.12500 |\n",
       "| cpo80 | w | cpo     | 3501 | 39315 | 16297.4284 | 1681.1780158 | 3.276242 | 45.84264 |\n",
       "| jv80 | w | jv      |  206 |   966 |   963.4303 |   40.9841698 | 4.520263 | 44.81178 |\n",
       "| CAP80 | w | CAP     |  622 |  5256 |  2942.3193 |  201.5984004 | 3.860259 | 44.48265 |\n",
       "| CG3236580 | w | CG32365 |  256 |  1625 |  1188.0722 |   72.4723655 | 4.015277 | 41.01855 |\n",
       "| CG3260380 | w | CG32603 |    0 |  2752 |     0.0000 |   79.1172777 | 6.305921 | 39.87891 |\n",
       "| CG3226080 | w | CG32260 |   24 |    27 |   107.7366 |    0.9404929 | 5.794942 | 39.20101 |\n",
       "| CG1531280 | w | CG15312 |  194 |  1322 |   969.1433 |   63.0535240 | 3.919360 | 38.88810 |\n",
       "| Rpt280 | w | Rpt2    |    0 |  2361 |     0.0000 |   74.5750092 | 6.220620 | 38.81566 |\n",
       "| CG981580 | w | CG9815  |   34 |    79 |   161.7811 |    3.4611047 | 5.180498 | 38.06003 |\n",
       "| CG1201180 | w | CG12011 |   32 | 36314 |   142.3954 | 1664.9756980 | 3.537430 | 37.85811 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "nompA80   w           nompA   1553               1351        7555.3387     \n",
       "CG1291180 w           CG12911  825                597        3894.4554     \n",
       "CG1331880 w           CG13318  193                 94         928.5511     \n",
       "CG1286980 w           CG12869  198                168         958.5543     \n",
       "CG1250780 w           CG12507  158                123         759.2947     \n",
       "CG1427780 w           CG14277   94                105         477.8394     \n",
       "sv80      w           sv       910               2515        4167.8641     \n",
       "CG1330580 w           CG13305   81                141         411.1573     \n",
       "CG1445680 w           CG14456   44                131         226.0664     \n",
       "pros80    w           pros     793               6474        3595.7738     \n",
       "cpo80     w           cpo     3501              39315       16297.4284     \n",
       "jv80      w           jv       206                966         963.4303     \n",
       "CAP80     w           CAP      622               5256        2942.3193     \n",
       "CG3236580 w           CG32365  256               1625        1188.0722     \n",
       "CG3260380 w           CG32603    0               2752           0.0000     \n",
       "CG3226080 w           CG32260   24                 27         107.7366     \n",
       "CG1531280 w           CG15312  194               1322         969.1433     \n",
       "Rpt280    w           Rpt2       0               2361           0.0000     \n",
       "CG981580  w           CG9815    34                 79         161.7811     \n",
       "CG1201180 w           CG12011   32              36314         142.3954     \n",
       "          other.tpm    log2.ratio heuristic.score\n",
       "nompA80     53.6086022 7.112224   91.63014       \n",
       "CG1291180   23.3972386 7.318560   87.29268       \n",
       "CG1331880    3.5219592 7.681890   75.74643       \n",
       "CG1286980    6.5469492 6.988823   69.23282       \n",
       "CG1250780    4.6364366 7.073733   67.69856       \n",
       "CG1427780    2.1491981 7.245398   64.50866       \n",
       "sv80       108.4272849 5.251264   63.14875       \n",
       "CG1330580    5.3392267 6.019240   52.28945       \n",
       "CG1445680    1.9646874 6.252723   48.93988       \n",
       "pros80     239.0715519 3.904766   46.12500       \n",
       "cpo80     1681.1780158 3.276242   45.84264       \n",
       "jv80        40.9841698 4.520263   44.81178       \n",
       "CAP80      201.5984004 3.860259   44.48265       \n",
       "CG3236580   72.4723655 4.015277   41.01855       \n",
       "CG3260380   79.1172777 6.305921   39.87891       \n",
       "CG3226080    0.9404929 5.794942   39.20101       \n",
       "CG1531280   63.0535240 3.919360   38.88810       \n",
       "Rpt280      74.5750092 6.220620   38.81566       \n",
       "CG981580     3.4611047 5.180498   38.06003       \n",
       "CG1201180 1664.9756980 3.537430   37.85811       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"w\")  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"nompA\"), size_pos = 1, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"nompA\"), size_pos = 1, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"nompA\"), \n",
    "                   size_pos = 1, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG12911\"), \n",
    "                   size_pos = 1, size_neg = 0.1, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"sv\"), \n",
    "                   size_pos = 3, size_neg = 0.2, stroke = 0.02, limit = 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 976,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Cuprophilic_cells\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Cuprophilic_cells\", df_sub300$cluster %in% x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'x' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'x' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(x)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 971,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Vha100-481</th><td>x</td><td>Vha100-4</td><td> 616</td><td>  289</td><td>3170.249334</td><td> 12.3098817</td><td>7.895955</td><td>91.83655</td></tr>\n",
       "\t<tr><th scope=row>lab81</th><td>x</td><td>lab     </td><td>1815</td><td> 2588</td><td>9129.744553</td><td>119.9039820</td><td>6.238641</td><td>82.07878</td></tr>\n",
       "\t<tr><th scope=row>CG3144681</th><td>x</td><td>CG31446 </td><td> 157</td><td>   34</td><td> 876.870940</td><td>  1.7843818</td><td>8.298864</td><td>81.14517</td></tr>\n",
       "\t<tr><th scope=row>CG746581</th><td>x</td><td>CG7465  </td><td>   0</td><td> 5957</td><td>   0.000000</td><td>159.9967739</td><td>7.321899</td><td>53.67602</td></tr>\n",
       "\t<tr><th scope=row>Pgcl81</th><td>x</td><td>Pgcl    </td><td>  59</td><td>  137</td><td> 304.959681</td><td>  5.2740117</td><td>5.603086</td><td>46.26579</td></tr>\n",
       "\t<tr><th scope=row>Ugt307A181</th><td>x</td><td>Ugt307A1</td><td>  18</td><td>    3</td><td> 110.561474</td><td>  0.1240624</td><td>6.619983</td><td>45.02710</td></tr>\n",
       "\t<tr><th scope=row>CG577081</th><td>x</td><td>CG5770  </td><td>  23</td><td>   17</td><td> 126.194380</td><td>  0.6870172</td><td>6.225029</td><td>43.51850</td></tr>\n",
       "\t<tr><th scope=row>CG1368101</th><td>x</td><td>CG1368  </td><td>   0</td><td> 3034</td><td>   0.000000</td><td> 96.1787791</td><td>6.587647</td><td>43.49539</td></tr>\n",
       "\t<tr><th scope=row>elB81</th><td>x</td><td>elB     </td><td>1000</td><td>10092</td><td>5006.104342</td><td>444.5248346</td><td>3.490111</td><td>42.89262</td></tr>\n",
       "\t<tr><th scope=row>Ugt317A181</th><td>x</td><td>Ugt317A1</td><td>   0</td><td> 2138</td><td>   0.000000</td><td> 92.8049194</td><td>6.536129</td><td>42.82205</td></tr>\n",
       "\t<tr><th scope=row>CG3108781</th><td>x</td><td>CG31087 </td><td>  33</td><td>   50</td><td> 167.200268</td><td>  2.1057276</td><td>5.750502</td><td>42.51942</td></tr>\n",
       "\t<tr><th scope=row>CG215081</th><td>x</td><td>CG2150  </td><td>   1</td><td> 4482</td><td>   2.507221</td><td>161.6464815</td><td>5.526370</td><td>40.59448</td></tr>\n",
       "\t<tr><th scope=row>meru81</th><td>x</td><td>meru    </td><td>   0</td><td> 1855</td><td>   0.000000</td><td> 80.9733138</td><td>6.339375</td><td>40.29993</td></tr>\n",
       "\t<tr><th scope=row>CG1095381</th><td>x</td><td>CG10953 </td><td>   0</td><td> 2395</td><td>   0.000000</td><td> 80.2361170</td><td>6.326180</td><td>40.13360</td></tr>\n",
       "\t<tr><th scope=row>hb81</th><td>x</td><td>hb      </td><td>   0</td><td> 2237</td><td>   0.000000</td><td> 79.4359807</td><td>6.311721</td><td>39.95173</td></tr>\n",
       "\t<tr><th scope=row>CG3045781</th><td>x</td><td>CG30457 </td><td>   0</td><td> 2701</td><td>   0.000000</td><td> 72.3001278</td><td>6.175926</td><td>38.26446</td></tr>\n",
       "\t<tr><th scope=row>out81</th><td>x</td><td>out     </td><td> 223</td><td> 1912</td><td>1119.945664</td><td> 81.6614376</td><td>3.760070</td><td>38.09140</td></tr>\n",
       "\t<tr><th scope=row>Rrp181</th><td>x</td><td>Rrp1    </td><td>   0</td><td> 2025</td><td>   0.000000</td><td> 70.6651013</td><td>6.142926</td><td>37.86007</td></tr>\n",
       "\t<tr><th scope=row>5-HT781</th><td>x</td><td>5-HT7   </td><td> 219</td><td> 1607</td><td>1050.970700</td><td> 76.4394021</td><td>3.762511</td><td>37.77139</td></tr>\n",
       "\t<tr><th scope=row>Tsp42Ec81</th><td>x</td><td>Tsp42Ec </td><td>  26</td><td>   77</td><td> 135.686857</td><td>  2.8019943</td><td>5.157381</td><td>36.59023</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tVha100-481 & x & Vha100-4 &  616 &   289 & 3170.249334 &  12.3098817 & 7.895955 & 91.83655\\\\\n",
       "\tlab81 & x & lab      & 1815 &  2588 & 9129.744553 & 119.9039820 & 6.238641 & 82.07878\\\\\n",
       "\tCG3144681 & x & CG31446  &  157 &    34 &  876.870940 &   1.7843818 & 8.298864 & 81.14517\\\\\n",
       "\tCG746581 & x & CG7465   &    0 &  5957 &    0.000000 & 159.9967739 & 7.321899 & 53.67602\\\\\n",
       "\tPgcl81 & x & Pgcl     &   59 &   137 &  304.959681 &   5.2740117 & 5.603086 & 46.26579\\\\\n",
       "\tUgt307A181 & x & Ugt307A1 &   18 &     3 &  110.561474 &   0.1240624 & 6.619983 & 45.02710\\\\\n",
       "\tCG577081 & x & CG5770   &   23 &    17 &  126.194380 &   0.6870172 & 6.225029 & 43.51850\\\\\n",
       "\tCG1368101 & x & CG1368   &    0 &  3034 &    0.000000 &  96.1787791 & 6.587647 & 43.49539\\\\\n",
       "\telB81 & x & elB      & 1000 & 10092 & 5006.104342 & 444.5248346 & 3.490111 & 42.89262\\\\\n",
       "\tUgt317A181 & x & Ugt317A1 &    0 &  2138 &    0.000000 &  92.8049194 & 6.536129 & 42.82205\\\\\n",
       "\tCG3108781 & x & CG31087  &   33 &    50 &  167.200268 &   2.1057276 & 5.750502 & 42.51942\\\\\n",
       "\tCG215081 & x & CG2150   &    1 &  4482 &    2.507221 & 161.6464815 & 5.526370 & 40.59448\\\\\n",
       "\tmeru81 & x & meru     &    0 &  1855 &    0.000000 &  80.9733138 & 6.339375 & 40.29993\\\\\n",
       "\tCG1095381 & x & CG10953  &    0 &  2395 &    0.000000 &  80.2361170 & 6.326180 & 40.13360\\\\\n",
       "\thb81 & x & hb       &    0 &  2237 &    0.000000 &  79.4359807 & 6.311721 & 39.95173\\\\\n",
       "\tCG3045781 & x & CG30457  &    0 &  2701 &    0.000000 &  72.3001278 & 6.175926 & 38.26446\\\\\n",
       "\tout81 & x & out      &  223 &  1912 & 1119.945664 &  81.6614376 & 3.760070 & 38.09140\\\\\n",
       "\tRrp181 & x & Rrp1     &    0 &  2025 &    0.000000 &  70.6651013 & 6.142926 & 37.86007\\\\\n",
       "\t5-HT781 & x & 5-HT7    &  219 &  1607 & 1050.970700 &  76.4394021 & 3.762511 & 37.77139\\\\\n",
       "\tTsp42Ec81 & x & Tsp42Ec  &   26 &    77 &  135.686857 &   2.8019943 & 5.157381 & 36.59023\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Vha100-481 | x | Vha100-4 |  616 |   289 | 3170.249334 |  12.3098817 | 7.895955 | 91.83655 |\n",
       "| lab81 | x | lab      | 1815 |  2588 | 9129.744553 | 119.9039820 | 6.238641 | 82.07878 |\n",
       "| CG3144681 | x | CG31446  |  157 |    34 |  876.870940 |   1.7843818 | 8.298864 | 81.14517 |\n",
       "| CG746581 | x | CG7465   |    0 |  5957 |    0.000000 | 159.9967739 | 7.321899 | 53.67602 |\n",
       "| Pgcl81 | x | Pgcl     |   59 |   137 |  304.959681 |   5.2740117 | 5.603086 | 46.26579 |\n",
       "| Ugt307A181 | x | Ugt307A1 |   18 |     3 |  110.561474 |   0.1240624 | 6.619983 | 45.02710 |\n",
       "| CG577081 | x | CG5770   |   23 |    17 |  126.194380 |   0.6870172 | 6.225029 | 43.51850 |\n",
       "| CG1368101 | x | CG1368   |    0 |  3034 |    0.000000 |  96.1787791 | 6.587647 | 43.49539 |\n",
       "| elB81 | x | elB      | 1000 | 10092 | 5006.104342 | 444.5248346 | 3.490111 | 42.89262 |\n",
       "| Ugt317A181 | x | Ugt317A1 |    0 |  2138 |    0.000000 |  92.8049194 | 6.536129 | 42.82205 |\n",
       "| CG3108781 | x | CG31087  |   33 |    50 |  167.200268 |   2.1057276 | 5.750502 | 42.51942 |\n",
       "| CG215081 | x | CG2150   |    1 |  4482 |    2.507221 | 161.6464815 | 5.526370 | 40.59448 |\n",
       "| meru81 | x | meru     |    0 |  1855 |    0.000000 |  80.9733138 | 6.339375 | 40.29993 |\n",
       "| CG1095381 | x | CG10953  |    0 |  2395 |    0.000000 |  80.2361170 | 6.326180 | 40.13360 |\n",
       "| hb81 | x | hb       |    0 |  2237 |    0.000000 |  79.4359807 | 6.311721 | 39.95173 |\n",
       "| CG3045781 | x | CG30457  |    0 |  2701 |    0.000000 |  72.3001278 | 6.175926 | 38.26446 |\n",
       "| out81 | x | out      |  223 |  1912 | 1119.945664 |  81.6614376 | 3.760070 | 38.09140 |\n",
       "| Rrp181 | x | Rrp1     |    0 |  2025 |    0.000000 |  70.6651013 | 6.142926 | 37.86007 |\n",
       "| 5-HT781 | x | 5-HT7    |  219 |  1607 | 1050.970700 |  76.4394021 | 3.762511 | 37.77139 |\n",
       "| Tsp42Ec81 | x | Tsp42Ec  |   26 |    77 |  135.686857 |   2.8019943 | 5.157381 | 36.59023 |\n",
       "\n"
      ],
      "text/plain": [
       "           cell.type.2 gene     cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "Vha100-481 x           Vha100-4  616                289       3170.249334    \n",
       "lab81      x           lab      1815               2588       9129.744553    \n",
       "CG3144681  x           CG31446   157                 34        876.870940    \n",
       "CG746581   x           CG7465      0               5957          0.000000    \n",
       "Pgcl81     x           Pgcl       59                137        304.959681    \n",
       "Ugt307A181 x           Ugt307A1   18                  3        110.561474    \n",
       "CG577081   x           CG5770     23                 17        126.194380    \n",
       "CG1368101  x           CG1368      0               3034          0.000000    \n",
       "elB81      x           elB      1000              10092       5006.104342    \n",
       "Ugt317A181 x           Ugt317A1    0               2138          0.000000    \n",
       "CG3108781  x           CG31087    33                 50        167.200268    \n",
       "CG215081   x           CG2150      1               4482          2.507221    \n",
       "meru81     x           meru        0               1855          0.000000    \n",
       "CG1095381  x           CG10953     0               2395          0.000000    \n",
       "hb81       x           hb          0               2237          0.000000    \n",
       "CG3045781  x           CG30457     0               2701          0.000000    \n",
       "out81      x           out       223               1912       1119.945664    \n",
       "Rrp181     x           Rrp1        0               2025          0.000000    \n",
       "5-HT781    x           5-HT7     219               1607       1050.970700    \n",
       "Tsp42Ec81  x           Tsp42Ec    26                 77        135.686857    \n",
       "           other.tpm   log2.ratio heuristic.score\n",
       "Vha100-481  12.3098817 7.895955   91.83655       \n",
       "lab81      119.9039820 6.238641   82.07878       \n",
       "CG3144681    1.7843818 8.298864   81.14517       \n",
       "CG746581   159.9967739 7.321899   53.67602       \n",
       "Pgcl81       5.2740117 5.603086   46.26579       \n",
       "Ugt307A181   0.1240624 6.619983   45.02710       \n",
       "CG577081     0.6870172 6.225029   43.51850       \n",
       "CG1368101   96.1787791 6.587647   43.49539       \n",
       "elB81      444.5248346 3.490111   42.89262       \n",
       "Ugt317A181  92.8049194 6.536129   42.82205       \n",
       "CG3108781    2.1057276 5.750502   42.51942       \n",
       "CG215081   161.6464815 5.526370   40.59448       \n",
       "meru81      80.9733138 6.339375   40.29993       \n",
       "CG1095381   80.2361170 6.326180   40.13360       \n",
       "hb81        79.4359807 6.311721   39.95173       \n",
       "CG3045781   72.3001278 6.175926   38.26446       \n",
       "out81       81.6614376 3.760070   38.09140       \n",
       "Rrp181      70.6651013 6.142926   37.86007       \n",
       "5-HT781     76.4394021 3.762511   37.77139       \n",
       "Tsp42Ec81    2.8019943 5.157381   36.59023       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"x\")  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Vha100-4\"), size_pos = 1, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Vha100-4\"), size_pos = 1, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Vha100-4\"), \n",
    "                   size_pos = 1, size_neg = 0.05, stroke = 0.008, limit = 3)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"lab\"), \n",
    "                   size_pos = 3, size_neg = 0.1, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG31446\"), \n",
    "                   size_pos = 1, size_neg = 0.2, stroke = 0.02, limit = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1005,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Pharynx\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Pharynx\", df_sub300$cluster %in% y & \n",
    "                         df_sub300$umap_2 >= 1*df_sub300$umap_1 -8.1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'y' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'y' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(y)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 4, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "xlim(2, 6) + ylim(-6, -2)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'y' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'y' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(y)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus & \n",
    "              df_sub300$umap_2 >= 1*umap_1 -8.1), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 1, intercept = -8.1) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(1, 5) + ylim(-6, -2)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 978,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>dy82</th><td>y</td><td>dy     </td><td>3438</td><td> 1838</td><td>6602.9940</td><td> 81.644936</td><td>6.320050</td><td>80.19589</td></tr>\n",
       "\t<tr><th scope=row>TwdlT82</th><td>y</td><td>TwdlT  </td><td>1425</td><td> 1350</td><td>2800.1017</td><td> 48.255267</td><td>5.829057</td><td>66.75307</td></tr>\n",
       "\t<tr><th scope=row>CG856882</th><td>y</td><td>CG8568 </td><td> 283</td><td>   93</td><td> 500.8593</td><td>  3.358475</td><td>6.844438</td><td>61.40241</td></tr>\n",
       "\t<tr><th scope=row>CAH682</th><td>y</td><td>CAH6   </td><td> 200</td><td>   37</td><td> 386.8225</td><td>  1.840266</td><td>7.089502</td><td>60.96442</td></tr>\n",
       "\t<tr><th scope=row>CG4233182</th><td>y</td><td>CG42331</td><td>1854</td><td> 2499</td><td>3570.6358</td><td>116.127642</td><td>4.930028</td><td>58.18601</td></tr>\n",
       "\t<tr><th scope=row>jv82</th><td>y</td><td>jv     </td><td> 832</td><td>  844</td><td>1550.9703</td><td> 36.343657</td><td>5.376164</td><td>56.98672</td></tr>\n",
       "\t<tr><th scope=row>Muc91C82</th><td>y</td><td>Muc91C </td><td>1434</td><td> 2484</td><td>2686.0894</td><td> 94.865583</td><td>4.808351</td><td>54.77591</td></tr>\n",
       "\t<tr><th scope=row>CG3117682</th><td>y</td><td>CG31176</td><td>1750</td><td> 2747</td><td>3210.0636</td><td>123.954414</td><td>4.683128</td><td>54.55299</td></tr>\n",
       "\t<tr><th scope=row>ImpE182</th><td>y</td><td>ImpE1  </td><td>4556</td><td>11123</td><td>8564.7561</td><td>511.768632</td><td>4.062032</td><td>53.06787</td></tr>\n",
       "\t<tr><th scope=row>Cpr23B82</th><td>y</td><td>Cpr23B </td><td> 179</td><td>   65</td><td> 274.7207</td><td>  2.253912</td><td>6.399647</td><td>51.88234</td></tr>\n",
       "\t<tr><th scope=row>CG1488082</th><td>y</td><td>CG14880</td><td> 336</td><td>  328</td><td> 623.2257</td><td> 13.441886</td><td>5.431424</td><td>50.43579</td></tr>\n",
       "\t<tr><th scope=row>fng82</th><td>y</td><td>fng    </td><td>2829</td><td>10173</td><td>5391.5846</td><td>476.356658</td><td>3.497570</td><td>43.35854</td></tr>\n",
       "\t<tr><th scope=row>Drip82</th><td>y</td><td>Drip   </td><td> 470</td><td>  995</td><td> 897.9784</td><td> 41.859368</td><td>4.388998</td><td>43.06548</td></tr>\n",
       "\t<tr><th scope=row>neo82</th><td>y</td><td>neo    </td><td>1558</td><td> 6491</td><td>3036.6992</td><td>284.876336</td><td>3.409041</td><td>39.43839</td></tr>\n",
       "\t<tr><th scope=row>Cpr50Cb82</th><td>y</td><td>Cpr50Cb</td><td> 112</td><td>  143</td><td> 233.9383</td><td>  6.450689</td><td>4.972610</td><td>39.16497</td></tr>\n",
       "\t<tr><th scope=row>bbg82</th><td>y</td><td>bbg    </td><td>3337</td><td>15654</td><td>6061.8845</td><td>702.251458</td><td>3.107654</td><td>39.05012</td></tr>\n",
       "\t<tr><th scope=row>CG3235482</th><td>y</td><td>CG32354</td><td> 494</td><td> 1462</td><td> 973.0331</td><td> 63.776832</td><td>3.908939</td><td>38.80727</td></tr>\n",
       "\t<tr><th scope=row>nyo82</th><td>y</td><td>nyo    </td><td>1111</td><td> 4099</td><td>2110.7496</td><td>185.502167</td><td>3.500491</td><td>38.66021</td></tr>\n",
       "\t<tr><th scope=row>CG1767282</th><td>y</td><td>CG17672</td><td> 972</td><td> 3470</td><td>1851.6299</td><td>156.227889</td><td>3.557867</td><td>38.62192</td></tr>\n",
       "\t<tr><th scope=row>CG1550682</th><td>y</td><td>CG15506</td><td> 425</td><td> 1320</td><td> 843.5728</td><td> 55.262546</td><td>3.906266</td><td>37.97702</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tdy82 & y & dy      & 3438 &  1838 & 6602.9940 &  81.644936 & 6.320050 & 80.19589\\\\\n",
       "\tTwdlT82 & y & TwdlT   & 1425 &  1350 & 2800.1017 &  48.255267 & 5.829057 & 66.75307\\\\\n",
       "\tCG856882 & y & CG8568  &  283 &    93 &  500.8593 &   3.358475 & 6.844438 & 61.40241\\\\\n",
       "\tCAH682 & y & CAH6    &  200 &    37 &  386.8225 &   1.840266 & 7.089502 & 60.96442\\\\\n",
       "\tCG4233182 & y & CG42331 & 1854 &  2499 & 3570.6358 & 116.127642 & 4.930028 & 58.18601\\\\\n",
       "\tjv82 & y & jv      &  832 &   844 & 1550.9703 &  36.343657 & 5.376164 & 56.98672\\\\\n",
       "\tMuc91C82 & y & Muc91C  & 1434 &  2484 & 2686.0894 &  94.865583 & 4.808351 & 54.77591\\\\\n",
       "\tCG3117682 & y & CG31176 & 1750 &  2747 & 3210.0636 & 123.954414 & 4.683128 & 54.55299\\\\\n",
       "\tImpE182 & y & ImpE1   & 4556 & 11123 & 8564.7561 & 511.768632 & 4.062032 & 53.06787\\\\\n",
       "\tCpr23B82 & y & Cpr23B  &  179 &    65 &  274.7207 &   2.253912 & 6.399647 & 51.88234\\\\\n",
       "\tCG1488082 & y & CG14880 &  336 &   328 &  623.2257 &  13.441886 & 5.431424 & 50.43579\\\\\n",
       "\tfng82 & y & fng     & 2829 & 10173 & 5391.5846 & 476.356658 & 3.497570 & 43.35854\\\\\n",
       "\tDrip82 & y & Drip    &  470 &   995 &  897.9784 &  41.859368 & 4.388998 & 43.06548\\\\\n",
       "\tneo82 & y & neo     & 1558 &  6491 & 3036.6992 & 284.876336 & 3.409041 & 39.43839\\\\\n",
       "\tCpr50Cb82 & y & Cpr50Cb &  112 &   143 &  233.9383 &   6.450689 & 4.972610 & 39.16497\\\\\n",
       "\tbbg82 & y & bbg     & 3337 & 15654 & 6061.8845 & 702.251458 & 3.107654 & 39.05012\\\\\n",
       "\tCG3235482 & y & CG32354 &  494 &  1462 &  973.0331 &  63.776832 & 3.908939 & 38.80727\\\\\n",
       "\tnyo82 & y & nyo     & 1111 &  4099 & 2110.7496 & 185.502167 & 3.500491 & 38.66021\\\\\n",
       "\tCG1767282 & y & CG17672 &  972 &  3470 & 1851.6299 & 156.227889 & 3.557867 & 38.62192\\\\\n",
       "\tCG1550682 & y & CG15506 &  425 &  1320 &  843.5728 &  55.262546 & 3.906266 & 37.97702\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| dy82 | y | dy      | 3438 |  1838 | 6602.9940 |  81.644936 | 6.320050 | 80.19589 |\n",
       "| TwdlT82 | y | TwdlT   | 1425 |  1350 | 2800.1017 |  48.255267 | 5.829057 | 66.75307 |\n",
       "| CG856882 | y | CG8568  |  283 |    93 |  500.8593 |   3.358475 | 6.844438 | 61.40241 |\n",
       "| CAH682 | y | CAH6    |  200 |    37 |  386.8225 |   1.840266 | 7.089502 | 60.96442 |\n",
       "| CG4233182 | y | CG42331 | 1854 |  2499 | 3570.6358 | 116.127642 | 4.930028 | 58.18601 |\n",
       "| jv82 | y | jv      |  832 |   844 | 1550.9703 |  36.343657 | 5.376164 | 56.98672 |\n",
       "| Muc91C82 | y | Muc91C  | 1434 |  2484 | 2686.0894 |  94.865583 | 4.808351 | 54.77591 |\n",
       "| CG3117682 | y | CG31176 | 1750 |  2747 | 3210.0636 | 123.954414 | 4.683128 | 54.55299 |\n",
       "| ImpE182 | y | ImpE1   | 4556 | 11123 | 8564.7561 | 511.768632 | 4.062032 | 53.06787 |\n",
       "| Cpr23B82 | y | Cpr23B  |  179 |    65 |  274.7207 |   2.253912 | 6.399647 | 51.88234 |\n",
       "| CG1488082 | y | CG14880 |  336 |   328 |  623.2257 |  13.441886 | 5.431424 | 50.43579 |\n",
       "| fng82 | y | fng     | 2829 | 10173 | 5391.5846 | 476.356658 | 3.497570 | 43.35854 |\n",
       "| Drip82 | y | Drip    |  470 |   995 |  897.9784 |  41.859368 | 4.388998 | 43.06548 |\n",
       "| neo82 | y | neo     | 1558 |  6491 | 3036.6992 | 284.876336 | 3.409041 | 39.43839 |\n",
       "| Cpr50Cb82 | y | Cpr50Cb |  112 |   143 |  233.9383 |   6.450689 | 4.972610 | 39.16497 |\n",
       "| bbg82 | y | bbg     | 3337 | 15654 | 6061.8845 | 702.251458 | 3.107654 | 39.05012 |\n",
       "| CG3235482 | y | CG32354 |  494 |  1462 |  973.0331 |  63.776832 | 3.908939 | 38.80727 |\n",
       "| nyo82 | y | nyo     | 1111 |  4099 | 2110.7496 | 185.502167 | 3.500491 | 38.66021 |\n",
       "| CG1767282 | y | CG17672 |  972 |  3470 | 1851.6299 | 156.227889 | 3.557867 | 38.62192 |\n",
       "| CG1550682 | y | CG15506 |  425 |  1320 |  843.5728 |  55.262546 | 3.906266 | 37.97702 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "dy82      y           dy      3438               1838       6602.9940      \n",
       "TwdlT82   y           TwdlT   1425               1350       2800.1017      \n",
       "CG856882  y           CG8568   283                 93        500.8593      \n",
       "CAH682    y           CAH6     200                 37        386.8225      \n",
       "CG4233182 y           CG42331 1854               2499       3570.6358      \n",
       "jv82      y           jv       832                844       1550.9703      \n",
       "Muc91C82  y           Muc91C  1434               2484       2686.0894      \n",
       "CG3117682 y           CG31176 1750               2747       3210.0636      \n",
       "ImpE182   y           ImpE1   4556              11123       8564.7561      \n",
       "Cpr23B82  y           Cpr23B   179                 65        274.7207      \n",
       "CG1488082 y           CG14880  336                328        623.2257      \n",
       "fng82     y           fng     2829              10173       5391.5846      \n",
       "Drip82    y           Drip     470                995        897.9784      \n",
       "neo82     y           neo     1558               6491       3036.6992      \n",
       "Cpr50Cb82 y           Cpr50Cb  112                143        233.9383      \n",
       "bbg82     y           bbg     3337              15654       6061.8845      \n",
       "CG3235482 y           CG32354  494               1462        973.0331      \n",
       "nyo82     y           nyo     1111               4099       2110.7496      \n",
       "CG1767282 y           CG17672  972               3470       1851.6299      \n",
       "CG1550682 y           CG15506  425               1320        843.5728      \n",
       "          other.tpm  log2.ratio heuristic.score\n",
       "dy82       81.644936 6.320050   80.19589       \n",
       "TwdlT82    48.255267 5.829057   66.75307       \n",
       "CG856882    3.358475 6.844438   61.40241       \n",
       "CAH682      1.840266 7.089502   60.96442       \n",
       "CG4233182 116.127642 4.930028   58.18601       \n",
       "jv82       36.343657 5.376164   56.98672       \n",
       "Muc91C82   94.865583 4.808351   54.77591       \n",
       "CG3117682 123.954414 4.683128   54.55299       \n",
       "ImpE182   511.768632 4.062032   53.06787       \n",
       "Cpr23B82    2.253912 6.399647   51.88234       \n",
       "CG1488082  13.441886 5.431424   50.43579       \n",
       "fng82     476.356658 3.497570   43.35854       \n",
       "Drip82     41.859368 4.388998   43.06548       \n",
       "neo82     284.876336 3.409041   39.43839       \n",
       "Cpr50Cb82   6.450689 4.972610   39.16497       \n",
       "bbg82     702.251458 3.107654   39.05012       \n",
       "CG3235482  63.776832 3.908939   38.80727       \n",
       "nyo82     185.502167 3.500491   38.66021       \n",
       "CG1767282 156.227889 3.557867   38.62192       \n",
       "CG1550682  55.262546 3.906266   37.97702       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"y\")  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"dy\"), size_pos = 1, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"dy\"), size_pos = 1, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"dy\"), \n",
    "                   size_pos = 1, size_neg = 0.05, stroke = 0.008, limit = 3)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"TwdlT\"), \n",
    "                   size_pos = 1, size_neg = 0.1, stroke = 0.02, limit = 3)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CAH6\"), \n",
    "                   size_pos = 1, size_neg = 0.2, stroke = 0.02, limit = 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1587,
   "metadata": {},
   "outputs": [],
   "source": [
    "saveRDS(df_sub300, \"data/df_cell.type.3_reannotation.rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1087,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Tracheal_system_1\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Tracheal_system_1\", df_sub300$cluster %in% z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'z' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'z' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(z)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1014,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>rdo83</th><td>z</td><td>rdo    </td><td>2479</td><td> 1147</td><td>7369.2398</td><td> 50.271102</td><td>7.167226</td><td>92.08091</td></tr>\n",
       "\t<tr><th scope=row>wdp83</th><td>z</td><td>wdp    </td><td> 498</td><td> 1738</td><td>1502.3538</td><td> 71.338587</td><td>4.376315</td><td>46.18750</td></tr>\n",
       "\t<tr><th scope=row>Osi1583</th><td>z</td><td>Osi15  </td><td>2805</td><td>19199</td><td>8206.9273</td><td>747.288410</td><td>3.455176</td><td>44.92697</td></tr>\n",
       "\t<tr><th scope=row>Gasp83</th><td>z</td><td>Gasp   </td><td>2714</td><td>18108</td><td>7960.3058</td><td>727.413577</td><td>3.449994</td><td>44.70775</td></tr>\n",
       "\t<tr><th scope=row>stumps83</th><td>z</td><td>stumps </td><td>1092</td><td> 5538</td><td>3235.5433</td><td>226.520022</td><td>3.829943</td><td>44.65804</td></tr>\n",
       "\t<tr><th scope=row>CG583583</th><td>z</td><td>CG5835 </td><td> 245</td><td>  722</td><td> 743.7968</td><td> 30.093228</td><td>4.580236</td><td>43.69867</td></tr>\n",
       "\t<tr><th scope=row>apn83</th><td>z</td><td>apn    </td><td> 478</td><td> 2319</td><td>1415.4810</td><td> 88.213053</td><td>3.987894</td><td>41.74565</td></tr>\n",
       "\t<tr><th scope=row>CG1178683</th><td>z</td><td>CG11786</td><td> 186</td><td>  582</td><td> 569.8222</td><td> 27.995683</td><td>4.296602</td><td>39.34354</td></tr>\n",
       "\t<tr><th scope=row>sano83</th><td>z</td><td>sano   </td><td>1441</td><td> 9710</td><td>4116.7097</td><td>428.608211</td><td>3.260398</td><td>39.14964</td></tr>\n",
       "\t<tr><th scope=row>CG751283</th><td>z</td><td>CG7512 </td><td>  68</td><td>  133</td><td> 206.4401</td><td>  5.205659</td><td>5.055995</td><td>38.91372</td></tr>\n",
       "\t<tr><th scope=row>Spn43Aa83</th><td>z</td><td>Spn43Aa</td><td>1033</td><td> 7548</td><td>3072.1801</td><td>327.582802</td><td>3.224934</td><td>37.36253</td></tr>\n",
       "\t<tr><th scope=row>knrl83</th><td>z</td><td>knrl   </td><td> 764</td><td> 5177</td><td>2140.9823</td><td>216.223622</td><td>3.301020</td><td>36.52490</td></tr>\n",
       "\t<tr><th scope=row>raskol83</th><td>z</td><td>raskol </td><td>1479</td><td>13039</td><td>4306.3208</td><td>566.690446</td><td>2.923279</td><td>35.29151</td></tr>\n",
       "\t<tr><th scope=row>bs83</th><td>z</td><td>bs     </td><td> 236</td><td> 1301</td><td> 677.1236</td><td> 50.457583</td><td>3.717964</td><td>34.96895</td></tr>\n",
       "\t<tr><th scope=row>form383</th><td>z</td><td>form3  </td><td> 342</td><td> 1969</td><td> 990.4829</td><td> 86.371565</td><td>3.502896</td><td>34.86588</td></tr>\n",
       "\t<tr><th scope=row>uif83</th><td>z</td><td>uif    </td><td>1991</td><td>20412</td><td>5693.3714</td><td>849.486214</td><td>2.742923</td><td>34.21885</td></tr>\n",
       "\t<tr><th scope=row>CG950383</th><td>z</td><td>CG9503 </td><td> 230</td><td> 1488</td><td> 692.8069</td><td> 59.259454</td><td>3.523194</td><td>33.25328</td></tr>\n",
       "\t<tr><th scope=row>Cpr51A83</th><td>z</td><td>Cpr51A </td><td> 356</td><td> 2764</td><td>1055.2861</td><td>112.736392</td><td>3.213868</td><td>32.28262</td></tr>\n",
       "\t<tr><th scope=row>CG1220683</th><td>z</td><td>CG12206</td><td> 214</td><td> 1283</td><td> 620.5933</td><td> 55.496864</td><td>3.457405</td><td>32.08412</td></tr>\n",
       "\t<tr><th scope=row>Buffy83</th><td>z</td><td>Buffy  </td><td>  65</td><td>  195</td><td> 177.0541</td><td>  8.637933</td><td>4.199323</td><td>31.39486</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\trdo83 & z & rdo     & 2479 &  1147 & 7369.2398 &  50.271102 & 7.167226 & 92.08091\\\\\n",
       "\twdp83 & z & wdp     &  498 &  1738 & 1502.3538 &  71.338587 & 4.376315 & 46.18750\\\\\n",
       "\tOsi1583 & z & Osi15   & 2805 & 19199 & 8206.9273 & 747.288410 & 3.455176 & 44.92697\\\\\n",
       "\tGasp83 & z & Gasp    & 2714 & 18108 & 7960.3058 & 727.413577 & 3.449994 & 44.70775\\\\\n",
       "\tstumps83 & z & stumps  & 1092 &  5538 & 3235.5433 & 226.520022 & 3.829943 & 44.65804\\\\\n",
       "\tCG583583 & z & CG5835  &  245 &   722 &  743.7968 &  30.093228 & 4.580236 & 43.69867\\\\\n",
       "\tapn83 & z & apn     &  478 &  2319 & 1415.4810 &  88.213053 & 3.987894 & 41.74565\\\\\n",
       "\tCG1178683 & z & CG11786 &  186 &   582 &  569.8222 &  27.995683 & 4.296602 & 39.34354\\\\\n",
       "\tsano83 & z & sano    & 1441 &  9710 & 4116.7097 & 428.608211 & 3.260398 & 39.14964\\\\\n",
       "\tCG751283 & z & CG7512  &   68 &   133 &  206.4401 &   5.205659 & 5.055995 & 38.91372\\\\\n",
       "\tSpn43Aa83 & z & Spn43Aa & 1033 &  7548 & 3072.1801 & 327.582802 & 3.224934 & 37.36253\\\\\n",
       "\tknrl83 & z & knrl    &  764 &  5177 & 2140.9823 & 216.223622 & 3.301020 & 36.52490\\\\\n",
       "\traskol83 & z & raskol  & 1479 & 13039 & 4306.3208 & 566.690446 & 2.923279 & 35.29151\\\\\n",
       "\tbs83 & z & bs      &  236 &  1301 &  677.1236 &  50.457583 & 3.717964 & 34.96895\\\\\n",
       "\tform383 & z & form3   &  342 &  1969 &  990.4829 &  86.371565 & 3.502896 & 34.86588\\\\\n",
       "\tuif83 & z & uif     & 1991 & 20412 & 5693.3714 & 849.486214 & 2.742923 & 34.21885\\\\\n",
       "\tCG950383 & z & CG9503  &  230 &  1488 &  692.8069 &  59.259454 & 3.523194 & 33.25328\\\\\n",
       "\tCpr51A83 & z & Cpr51A  &  356 &  2764 & 1055.2861 & 112.736392 & 3.213868 & 32.28262\\\\\n",
       "\tCG1220683 & z & CG12206 &  214 &  1283 &  620.5933 &  55.496864 & 3.457405 & 32.08412\\\\\n",
       "\tBuffy83 & z & Buffy   &   65 &   195 &  177.0541 &   8.637933 & 4.199323 & 31.39486\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| rdo83 | z | rdo     | 2479 |  1147 | 7369.2398 |  50.271102 | 7.167226 | 92.08091 |\n",
       "| wdp83 | z | wdp     |  498 |  1738 | 1502.3538 |  71.338587 | 4.376315 | 46.18750 |\n",
       "| Osi1583 | z | Osi15   | 2805 | 19199 | 8206.9273 | 747.288410 | 3.455176 | 44.92697 |\n",
       "| Gasp83 | z | Gasp    | 2714 | 18108 | 7960.3058 | 727.413577 | 3.449994 | 44.70775 |\n",
       "| stumps83 | z | stumps  | 1092 |  5538 | 3235.5433 | 226.520022 | 3.829943 | 44.65804 |\n",
       "| CG583583 | z | CG5835  |  245 |   722 |  743.7968 |  30.093228 | 4.580236 | 43.69867 |\n",
       "| apn83 | z | apn     |  478 |  2319 | 1415.4810 |  88.213053 | 3.987894 | 41.74565 |\n",
       "| CG1178683 | z | CG11786 |  186 |   582 |  569.8222 |  27.995683 | 4.296602 | 39.34354 |\n",
       "| sano83 | z | sano    | 1441 |  9710 | 4116.7097 | 428.608211 | 3.260398 | 39.14964 |\n",
       "| CG751283 | z | CG7512  |   68 |   133 |  206.4401 |   5.205659 | 5.055995 | 38.91372 |\n",
       "| Spn43Aa83 | z | Spn43Aa | 1033 |  7548 | 3072.1801 | 327.582802 | 3.224934 | 37.36253 |\n",
       "| knrl83 | z | knrl    |  764 |  5177 | 2140.9823 | 216.223622 | 3.301020 | 36.52490 |\n",
       "| raskol83 | z | raskol  | 1479 | 13039 | 4306.3208 | 566.690446 | 2.923279 | 35.29151 |\n",
       "| bs83 | z | bs      |  236 |  1301 |  677.1236 |  50.457583 | 3.717964 | 34.96895 |\n",
       "| form383 | z | form3   |  342 |  1969 |  990.4829 |  86.371565 | 3.502896 | 34.86588 |\n",
       "| uif83 | z | uif     | 1991 | 20412 | 5693.3714 | 849.486214 | 2.742923 | 34.21885 |\n",
       "| CG950383 | z | CG9503  |  230 |  1488 |  692.8069 |  59.259454 | 3.523194 | 33.25328 |\n",
       "| Cpr51A83 | z | Cpr51A  |  356 |  2764 | 1055.2861 | 112.736392 | 3.213868 | 32.28262 |\n",
       "| CG1220683 | z | CG12206 |  214 |  1283 |  620.5933 |  55.496864 | 3.457405 | 32.08412 |\n",
       "| Buffy83 | z | Buffy   |   65 |   195 |  177.0541 |   8.637933 | 4.199323 | 31.39486 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "rdo83     z           rdo     2479               1147       7369.2398      \n",
       "wdp83     z           wdp      498               1738       1502.3538      \n",
       "Osi1583   z           Osi15   2805              19199       8206.9273      \n",
       "Gasp83    z           Gasp    2714              18108       7960.3058      \n",
       "stumps83  z           stumps  1092               5538       3235.5433      \n",
       "CG583583  z           CG5835   245                722        743.7968      \n",
       "apn83     z           apn      478               2319       1415.4810      \n",
       "CG1178683 z           CG11786  186                582        569.8222      \n",
       "sano83    z           sano    1441               9710       4116.7097      \n",
       "CG751283  z           CG7512    68                133        206.4401      \n",
       "Spn43Aa83 z           Spn43Aa 1033               7548       3072.1801      \n",
       "knrl83    z           knrl     764               5177       2140.9823      \n",
       "raskol83  z           raskol  1479              13039       4306.3208      \n",
       "bs83      z           bs       236               1301        677.1236      \n",
       "form383   z           form3    342               1969        990.4829      \n",
       "uif83     z           uif     1991              20412       5693.3714      \n",
       "CG950383  z           CG9503   230               1488        692.8069      \n",
       "Cpr51A83  z           Cpr51A   356               2764       1055.2861      \n",
       "CG1220683 z           CG12206  214               1283        620.5933      \n",
       "Buffy83   z           Buffy     65                195        177.0541      \n",
       "          other.tpm  log2.ratio heuristic.score\n",
       "rdo83      50.271102 7.167226   92.08091       \n",
       "wdp83      71.338587 4.376315   46.18750       \n",
       "Osi1583   747.288410 3.455176   44.92697       \n",
       "Gasp83    727.413577 3.449994   44.70775       \n",
       "stumps83  226.520022 3.829943   44.65804       \n",
       "CG583583   30.093228 4.580236   43.69867       \n",
       "apn83      88.213053 3.987894   41.74565       \n",
       "CG1178683  27.995683 4.296602   39.34354       \n",
       "sano83    428.608211 3.260398   39.14964       \n",
       "CG751283    5.205659 5.055995   38.91372       \n",
       "Spn43Aa83 327.582802 3.224934   37.36253       \n",
       "knrl83    216.223622 3.301020   36.52490       \n",
       "raskol83  566.690446 2.923279   35.29151       \n",
       "bs83       50.457583 3.717964   34.96895       \n",
       "form383    86.371565 3.502896   34.86588       \n",
       "uif83     849.486214 2.742923   34.21885       \n",
       "CG950383   59.259454 3.523194   33.25328       \n",
       "Cpr51A83  112.736392 3.213868   32.28262       \n",
       "CG1220683  55.496864 3.457405   32.08412       \n",
       "Buffy83     8.637933 4.199323   31.39486       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"z\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"rdo\"), size_pos = 1, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"rdo\"), size_pos = 1, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"rdo\"), \n",
    "                   size_pos = 1, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"wdp\"), \n",
    "                   size_pos = 1, size_neg = 0.1, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"stumps\"), \n",
    "                   size_pos = 1, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"apn\"), size_pos = 1, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"apn\"), size_pos = 1, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"apn\"), \n",
    "                   size_pos = 1, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"sano\"), \n",
    "                   size_pos = 1, size_neg = 0.1, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG11786\"), \n",
    "                   size_pos = 1, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1112,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type large_intestine_hindgut_analpad\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"large_intestine_hindgut_analpad\", df_sub300$cluster %in% aa | \n",
    "                         df_sub300$cluster %in% c(109) &  \n",
    "              df_sub300$umap_2 < 0.7*df_sub300$umap_1 -4 &\n",
    "              df_sub300$umap_2 >= 0.5*df_sub300$umap_1 -3.8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'aa' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'aa' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(aa)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(109)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1089,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>TwdlX1</th><td>aa</td><td>TwdlX  </td><td>1401</td><td> 415</td><td>1986.60117</td><td> 13.5208150</td><td>7.096036</td><td>77.74994</td></tr>\n",
       "\t<tr><th scope=row>CG129201</th><td>aa</td><td>CG12920</td><td>1056</td><td> 395</td><td>1942.94248</td><td> 17.0922640</td><td>6.746726</td><td>73.70643</td></tr>\n",
       "\t<tr><th scope=row>Cyp49a11</th><td>aa</td><td>Cyp49a1</td><td>1544</td><td> 819</td><td>2722.95141</td><td> 34.0529600</td><td>6.279491</td><td>71.65832</td></tr>\n",
       "\t<tr><th scope=row>CG64091</th><td>aa</td><td>CG6409 </td><td>1251</td><td> 860</td><td>1986.82260</td><td> 31.4038439</td><td>5.938154</td><td>65.06420</td></tr>\n",
       "\t<tr><th scope=row>CG149491</th><td>aa</td><td>CG14949</td><td> 232</td><td>  44</td><td> 341.61005</td><td>  1.1380942</td><td>7.319881</td><td>61.63650</td></tr>\n",
       "\t<tr><th scope=row>CG422961</th><td>aa</td><td>CG42296</td><td> 800</td><td> 915</td><td>1422.55436</td><td> 37.6468204</td><td>5.201990</td><td>54.49231</td></tr>\n",
       "\t<tr><th scope=row>CG325641</th><td>aa</td><td>CG32564</td><td> 795</td><td>1494</td><td>1193.28700</td><td> 43.7231524</td><td>4.737775</td><td>48.42923</td></tr>\n",
       "\t<tr><th scope=row>CG61251</th><td>aa</td><td>CG6125 </td><td> 202</td><td> 145</td><td> 304.84822</td><td>  4.2440423</td><td>5.861268</td><td>48.39457</td></tr>\n",
       "\t<tr><th scope=row>NKCC1</th><td>aa</td><td>NKCC   </td><td> 258</td><td> 202</td><td> 389.71288</td><td>  7.4734855</td><td>5.523312</td><td>47.55553</td></tr>\n",
       "\t<tr><th scope=row>CG440981</th><td>aa</td><td>CG44098</td><td> 272</td><td> 181</td><td> 371.40313</td><td>  7.3418310</td><td>5.476478</td><td>46.77307</td></tr>\n",
       "\t<tr><th scope=row>CG120171</th><td>aa</td><td>CG12017</td><td>1668</td><td>3849</td><td>2957.78568</td><td>183.1041412</td><td>4.005924</td><td>46.19146</td></tr>\n",
       "\t<tr><th scope=row>CG97021</th><td>aa</td><td>CG9702 </td><td> 160</td><td> 105</td><td> 267.17400</td><td>  4.0944801</td><td>5.712701</td><td>46.08450</td></tr>\n",
       "\t<tr><th scope=row>CG77781</th><td>aa</td><td>CG7778 </td><td> 154</td><td>  94</td><td> 220.94700</td><td>  2.8001490</td><td>5.861501</td><td>45.68495</td></tr>\n",
       "\t<tr><th scope=row>CG99931</th><td>aa</td><td>CG9993 </td><td> 188</td><td>  85</td><td> 222.76993</td><td>  3.1123957</td><td>5.759432</td><td>44.95739</td></tr>\n",
       "\t<tr><th scope=row>Cyp4aa11</th><td>aa</td><td>Cyp4aa1</td><td> 168</td><td> 146</td><td> 283.31485</td><td>  5.7449458</td><td>5.392456</td><td>43.95577</td></tr>\n",
       "\t<tr><th scope=row>CG44621</th><td>aa</td><td>CG4462 </td><td> 118</td><td>  59</td><td> 165.23537</td><td>  2.4812478</td><td>5.568774</td><td>41.08131</td></tr>\n",
       "\t<tr><th scope=row>Nha11</th><td>aa</td><td>Nha1   </td><td> 171</td><td> 182</td><td> 260.17460</td><td>  6.8344054</td><td>5.053513</td><td>40.57400</td></tr>\n",
       "\t<tr><th scope=row>CG87851</th><td>aa</td><td>CG8785 </td><td> 195</td><td> 242</td><td> 324.34917</td><td> 10.2162138</td><td>4.853890</td><td>40.50981</td></tr>\n",
       "\t<tr><th scope=row>CG156151</th><td>aa</td><td>CG15615</td><td> 577</td><td>1408</td><td> 940.96848</td><td> 54.5916622</td><td>4.081206</td><td>40.32042</td></tr>\n",
       "\t<tr><th scope=row>CG132151</th><td>aa</td><td>CG13215</td><td>  50</td><td>   4</td><td>  81.33123</td><td>  0.1169645</td><td>6.186154</td><td>39.36478</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tTwdlX1 & aa & TwdlX   & 1401 &  415 & 1986.60117 &  13.5208150 & 7.096036 & 77.74994\\\\\n",
       "\tCG129201 & aa & CG12920 & 1056 &  395 & 1942.94248 &  17.0922640 & 6.746726 & 73.70643\\\\\n",
       "\tCyp49a11 & aa & Cyp49a1 & 1544 &  819 & 2722.95141 &  34.0529600 & 6.279491 & 71.65832\\\\\n",
       "\tCG64091 & aa & CG6409  & 1251 &  860 & 1986.82260 &  31.4038439 & 5.938154 & 65.06420\\\\\n",
       "\tCG149491 & aa & CG14949 &  232 &   44 &  341.61005 &   1.1380942 & 7.319881 & 61.63650\\\\\n",
       "\tCG422961 & aa & CG42296 &  800 &  915 & 1422.55436 &  37.6468204 & 5.201990 & 54.49231\\\\\n",
       "\tCG325641 & aa & CG32564 &  795 & 1494 & 1193.28700 &  43.7231524 & 4.737775 & 48.42923\\\\\n",
       "\tCG61251 & aa & CG6125  &  202 &  145 &  304.84822 &   4.2440423 & 5.861268 & 48.39457\\\\\n",
       "\tNKCC1 & aa & NKCC    &  258 &  202 &  389.71288 &   7.4734855 & 5.523312 & 47.55553\\\\\n",
       "\tCG440981 & aa & CG44098 &  272 &  181 &  371.40313 &   7.3418310 & 5.476478 & 46.77307\\\\\n",
       "\tCG120171 & aa & CG12017 & 1668 & 3849 & 2957.78568 & 183.1041412 & 4.005924 & 46.19146\\\\\n",
       "\tCG97021 & aa & CG9702  &  160 &  105 &  267.17400 &   4.0944801 & 5.712701 & 46.08450\\\\\n",
       "\tCG77781 & aa & CG7778  &  154 &   94 &  220.94700 &   2.8001490 & 5.861501 & 45.68495\\\\\n",
       "\tCG99931 & aa & CG9993  &  188 &   85 &  222.76993 &   3.1123957 & 5.759432 & 44.95739\\\\\n",
       "\tCyp4aa11 & aa & Cyp4aa1 &  168 &  146 &  283.31485 &   5.7449458 & 5.392456 & 43.95577\\\\\n",
       "\tCG44621 & aa & CG4462  &  118 &   59 &  165.23537 &   2.4812478 & 5.568774 & 41.08131\\\\\n",
       "\tNha11 & aa & Nha1    &  171 &  182 &  260.17460 &   6.8344054 & 5.053513 & 40.57400\\\\\n",
       "\tCG87851 & aa & CG8785  &  195 &  242 &  324.34917 &  10.2162138 & 4.853890 & 40.50981\\\\\n",
       "\tCG156151 & aa & CG15615 &  577 & 1408 &  940.96848 &  54.5916622 & 4.081206 & 40.32042\\\\\n",
       "\tCG132151 & aa & CG13215 &   50 &    4 &   81.33123 &   0.1169645 & 6.186154 & 39.36478\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| TwdlX1 | aa | TwdlX   | 1401 |  415 | 1986.60117 |  13.5208150 | 7.096036 | 77.74994 |\n",
       "| CG129201 | aa | CG12920 | 1056 |  395 | 1942.94248 |  17.0922640 | 6.746726 | 73.70643 |\n",
       "| Cyp49a11 | aa | Cyp49a1 | 1544 |  819 | 2722.95141 |  34.0529600 | 6.279491 | 71.65832 |\n",
       "| CG64091 | aa | CG6409  | 1251 |  860 | 1986.82260 |  31.4038439 | 5.938154 | 65.06420 |\n",
       "| CG149491 | aa | CG14949 |  232 |   44 |  341.61005 |   1.1380942 | 7.319881 | 61.63650 |\n",
       "| CG422961 | aa | CG42296 |  800 |  915 | 1422.55436 |  37.6468204 | 5.201990 | 54.49231 |\n",
       "| CG325641 | aa | CG32564 |  795 | 1494 | 1193.28700 |  43.7231524 | 4.737775 | 48.42923 |\n",
       "| CG61251 | aa | CG6125  |  202 |  145 |  304.84822 |   4.2440423 | 5.861268 | 48.39457 |\n",
       "| NKCC1 | aa | NKCC    |  258 |  202 |  389.71288 |   7.4734855 | 5.523312 | 47.55553 |\n",
       "| CG440981 | aa | CG44098 |  272 |  181 |  371.40313 |   7.3418310 | 5.476478 | 46.77307 |\n",
       "| CG120171 | aa | CG12017 | 1668 | 3849 | 2957.78568 | 183.1041412 | 4.005924 | 46.19146 |\n",
       "| CG97021 | aa | CG9702  |  160 |  105 |  267.17400 |   4.0944801 | 5.712701 | 46.08450 |\n",
       "| CG77781 | aa | CG7778  |  154 |   94 |  220.94700 |   2.8001490 | 5.861501 | 45.68495 |\n",
       "| CG99931 | aa | CG9993  |  188 |   85 |  222.76993 |   3.1123957 | 5.759432 | 44.95739 |\n",
       "| Cyp4aa11 | aa | Cyp4aa1 |  168 |  146 |  283.31485 |   5.7449458 | 5.392456 | 43.95577 |\n",
       "| CG44621 | aa | CG4462  |  118 |   59 |  165.23537 |   2.4812478 | 5.568774 | 41.08131 |\n",
       "| Nha11 | aa | Nha1    |  171 |  182 |  260.17460 |   6.8344054 | 5.053513 | 40.57400 |\n",
       "| CG87851 | aa | CG8785  |  195 |  242 |  324.34917 |  10.2162138 | 4.853890 | 40.50981 |\n",
       "| CG156151 | aa | CG15615 |  577 | 1408 |  940.96848 |  54.5916622 | 4.081206 | 40.32042 |\n",
       "| CG132151 | aa | CG13215 |   50 |    4 |   81.33123 |   0.1169645 | 6.186154 | 39.36478 |\n",
       "\n"
      ],
      "text/plain": [
       "         cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "TwdlX1   aa          TwdlX   1401               415        1986.60117     \n",
       "CG129201 aa          CG12920 1056               395        1942.94248     \n",
       "Cyp49a11 aa          Cyp49a1 1544               819        2722.95141     \n",
       "CG64091  aa          CG6409  1251               860        1986.82260     \n",
       "CG149491 aa          CG14949  232                44         341.61005     \n",
       "CG422961 aa          CG42296  800               915        1422.55436     \n",
       "CG325641 aa          CG32564  795              1494        1193.28700     \n",
       "CG61251  aa          CG6125   202               145         304.84822     \n",
       "NKCC1    aa          NKCC     258               202         389.71288     \n",
       "CG440981 aa          CG44098  272               181         371.40313     \n",
       "CG120171 aa          CG12017 1668              3849        2957.78568     \n",
       "CG97021  aa          CG9702   160               105         267.17400     \n",
       "CG77781  aa          CG7778   154                94         220.94700     \n",
       "CG99931  aa          CG9993   188                85         222.76993     \n",
       "Cyp4aa11 aa          Cyp4aa1  168               146         283.31485     \n",
       "CG44621  aa          CG4462   118                59         165.23537     \n",
       "Nha11    aa          Nha1     171               182         260.17460     \n",
       "CG87851  aa          CG8785   195               242         324.34917     \n",
       "CG156151 aa          CG15615  577              1408         940.96848     \n",
       "CG132151 aa          CG13215   50                 4          81.33123     \n",
       "         other.tpm   log2.ratio heuristic.score\n",
       "TwdlX1    13.5208150 7.096036   77.74994       \n",
       "CG129201  17.0922640 6.746726   73.70643       \n",
       "Cyp49a11  34.0529600 6.279491   71.65832       \n",
       "CG64091   31.4038439 5.938154   65.06420       \n",
       "CG149491   1.1380942 7.319881   61.63650       \n",
       "CG422961  37.6468204 5.201990   54.49231       \n",
       "CG325641  43.7231524 4.737775   48.42923       \n",
       "CG61251    4.2440423 5.861268   48.39457       \n",
       "NKCC1      7.4734855 5.523312   47.55553       \n",
       "CG440981   7.3418310 5.476478   46.77307       \n",
       "CG120171 183.1041412 4.005924   46.19146       \n",
       "CG97021    4.0944801 5.712701   46.08450       \n",
       "CG77781    2.8001490 5.861501   45.68495       \n",
       "CG99931    3.1123957 5.759432   44.95739       \n",
       "Cyp4aa11   5.7449458 5.392456   43.95577       \n",
       "CG44621    2.4812478 5.568774   41.08131       \n",
       "Nha11      6.8344054 5.053513   40.57400       \n",
       "CG87851   10.2162138 4.853890   40.50981       \n",
       "CG156151  54.5916622 4.081206   40.32042       \n",
       "CG132151   0.1169645 6.186154   39.36478       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"aa\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"TwdlX\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"TwdlX\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"TwdlX\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG12920\"), \n",
    "                   size_pos = 2, size_neg = 0.1, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Cyp49a1\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(109)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "              df_sub300$umap_2 < 0.7*umap_1 -4 &\n",
    "              df_sub300$umap_2 >= 0.5*umap_1 -3.8), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -4) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(1, 5) + ylim(-3, 1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1131,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Pharynx_2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Pharynx_2\", df_sub300$cluster %in% ab &\n",
    "                         df_sub300$umap_2 >= 0.7*df_sub300$umap_1 -6.3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1132,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Pharynx\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Pharynx\", df_sub300$cluster %in% ab &\n",
    "                         df_sub300$umap_2 < 0.7*df_sub300$umap_1 -6.3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ab' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ab' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(113)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1116,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>TwdlC2</th><td>ab</td><td>TwdlC  </td><td>1284</td><td> 866</td><td>3105.3762</td><td> 28.978450</td><td>6.694698</td><td>77.66531</td></tr>\n",
       "\t<tr><th scope=row>CG125232</th><td>ab</td><td>CG12523</td><td> 826</td><td> 414</td><td>1943.2159</td><td> 13.532165</td><td>7.063053</td><td>77.16366</td></tr>\n",
       "\t<tr><th scope=row>TwdlE2</th><td>ab</td><td>TwdlE  </td><td>1509</td><td>1199</td><td>3559.5434</td><td> 42.234625</td><td>6.363361</td><td>75.07418</td></tr>\n",
       "\t<tr><th scope=row>CG21502</th><td>ab</td><td>CG2150 </td><td>3308</td><td>3453</td><td>7091.4851</td><td>124.019957</td><td>5.825857</td><td>74.52481</td></tr>\n",
       "\t<tr><th scope=row>CG130412</th><td>ab</td><td>CG13041</td><td> 296</td><td> 121</td><td> 717.5571</td><td>  3.944247</td><td>7.181199</td><td>68.14210</td></tr>\n",
       "\t<tr><th scope=row>Cpr56F2</th><td>ab</td><td>Cpr56F </td><td> 826</td><td> 806</td><td>1955.1389</td><td> 25.921788</td><td>6.182353</td><td>67.59657</td></tr>\n",
       "\t<tr><th scope=row>CG130592</th><td>ab</td><td>CG13059</td><td> 566</td><td> 463</td><td>1466.4702</td><td> 16.134325</td><td>6.419315</td><td>67.52551</td></tr>\n",
       "\t<tr><th scope=row>Cpr62Bc2</th><td>ab</td><td>Cpr62Bc</td><td> 774</td><td> 805</td><td>1912.4850</td><td> 27.177162</td><td>6.084778</td><td>66.33617</td></tr>\n",
       "\t<tr><th scope=row>Cpr50Cb2</th><td>ab</td><td>Cpr50Cb</td><td> 258</td><td>  92</td><td> 649.5951</td><td>  3.890549</td><td>7.053401</td><td>65.91837</td></tr>\n",
       "\t<tr><th scope=row>CG344612</th><td>ab</td><td>CG34461</td><td> 868</td><td>1139</td><td>2090.4466</td><td> 36.119330</td><td>5.815497</td><td>64.14659</td></tr>\n",
       "\t<tr><th scope=row>Muc91C2</th><td>ab</td><td>Muc91C </td><td>1506</td><td>2588</td><td>3697.1308</td><td> 92.246957</td><td>5.309206</td><td>62.92779</td></tr>\n",
       "\t<tr><th scope=row>Cpr66Cb2</th><td>ab</td><td>Cpr66Cb</td><td> 743</td><td> 545</td><td>1370.9431</td><td> 20.047113</td><td>6.025403</td><td>62.79678</td></tr>\n",
       "\t<tr><th scope=row>CG130602</th><td>ab</td><td>CG13060</td><td> 246</td><td> 150</td><td> 627.2689</td><td>  5.226858</td><td>6.654436</td><td>61.85457</td></tr>\n",
       "\t<tr><th scope=row>Cpr65Av2</th><td>ab</td><td>Cpr65Av</td><td> 158</td><td>  38</td><td> 341.1285</td><td>  1.226574</td><td>7.259346</td><td>61.11204</td></tr>\n",
       "\t<tr><th scope=row>CG136752</th><td>ab</td><td>CG13675</td><td> 200</td><td>  82</td><td> 454.7549</td><td>  3.370995</td><td>6.700984</td><td>59.18385</td></tr>\n",
       "\t<tr><th scope=row>Ccp84Ac2</th><td>ab</td><td>Ccp84Ac</td><td> 591</td><td>1019</td><td>1559.8422</td><td> 35.995950</td><td>5.397889</td><td>57.26139</td></tr>\n",
       "\t<tr><th scope=row>TwdlV2</th><td>ab</td><td>TwdlV  </td><td> 418</td><td> 491</td><td> 950.9060</td><td> 17.492830</td><td>5.684265</td><td>56.24395</td></tr>\n",
       "\t<tr><th scope=row>Cpr65Eb2</th><td>ab</td><td>Cpr65Eb</td><td> 819</td><td>1496</td><td>1907.6521</td><td> 55.947416</td><td>5.066024</td><td>55.21124</td></tr>\n",
       "\t<tr><th scope=row>CG130432</th><td>ab</td><td>CG13043</td><td> 226</td><td> 254</td><td> 587.9498</td><td>  8.530352</td><td>5.947020</td><td>54.72448</td></tr>\n",
       "\t<tr><th scope=row>Vajk12</th><td>ab</td><td>Vajk1  </td><td>1314</td><td>2549</td><td>2634.7168</td><td> 93.141487</td><td>4.806673</td><td>54.62294</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tTwdlC2 & ab & TwdlC   & 1284 &  866 & 3105.3762 &  28.978450 & 6.694698 & 77.66531\\\\\n",
       "\tCG125232 & ab & CG12523 &  826 &  414 & 1943.2159 &  13.532165 & 7.063053 & 77.16366\\\\\n",
       "\tTwdlE2 & ab & TwdlE   & 1509 & 1199 & 3559.5434 &  42.234625 & 6.363361 & 75.07418\\\\\n",
       "\tCG21502 & ab & CG2150  & 3308 & 3453 & 7091.4851 & 124.019957 & 5.825857 & 74.52481\\\\\n",
       "\tCG130412 & ab & CG13041 &  296 &  121 &  717.5571 &   3.944247 & 7.181199 & 68.14210\\\\\n",
       "\tCpr56F2 & ab & Cpr56F  &  826 &  806 & 1955.1389 &  25.921788 & 6.182353 & 67.59657\\\\\n",
       "\tCG130592 & ab & CG13059 &  566 &  463 & 1466.4702 &  16.134325 & 6.419315 & 67.52551\\\\\n",
       "\tCpr62Bc2 & ab & Cpr62Bc &  774 &  805 & 1912.4850 &  27.177162 & 6.084778 & 66.33617\\\\\n",
       "\tCpr50Cb2 & ab & Cpr50Cb &  258 &   92 &  649.5951 &   3.890549 & 7.053401 & 65.91837\\\\\n",
       "\tCG344612 & ab & CG34461 &  868 & 1139 & 2090.4466 &  36.119330 & 5.815497 & 64.14659\\\\\n",
       "\tMuc91C2 & ab & Muc91C  & 1506 & 2588 & 3697.1308 &  92.246957 & 5.309206 & 62.92779\\\\\n",
       "\tCpr66Cb2 & ab & Cpr66Cb &  743 &  545 & 1370.9431 &  20.047113 & 6.025403 & 62.79678\\\\\n",
       "\tCG130602 & ab & CG13060 &  246 &  150 &  627.2689 &   5.226858 & 6.654436 & 61.85457\\\\\n",
       "\tCpr65Av2 & ab & Cpr65Av &  158 &   38 &  341.1285 &   1.226574 & 7.259346 & 61.11204\\\\\n",
       "\tCG136752 & ab & CG13675 &  200 &   82 &  454.7549 &   3.370995 & 6.700984 & 59.18385\\\\\n",
       "\tCcp84Ac2 & ab & Ccp84Ac &  591 & 1019 & 1559.8422 &  35.995950 & 5.397889 & 57.26139\\\\\n",
       "\tTwdlV2 & ab & TwdlV   &  418 &  491 &  950.9060 &  17.492830 & 5.684265 & 56.24395\\\\\n",
       "\tCpr65Eb2 & ab & Cpr65Eb &  819 & 1496 & 1907.6521 &  55.947416 & 5.066024 & 55.21124\\\\\n",
       "\tCG130432 & ab & CG13043 &  226 &  254 &  587.9498 &   8.530352 & 5.947020 & 54.72448\\\\\n",
       "\tVajk12 & ab & Vajk1   & 1314 & 2549 & 2634.7168 &  93.141487 & 4.806673 & 54.62294\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| TwdlC2 | ab | TwdlC   | 1284 |  866 | 3105.3762 |  28.978450 | 6.694698 | 77.66531 |\n",
       "| CG125232 | ab | CG12523 |  826 |  414 | 1943.2159 |  13.532165 | 7.063053 | 77.16366 |\n",
       "| TwdlE2 | ab | TwdlE   | 1509 | 1199 | 3559.5434 |  42.234625 | 6.363361 | 75.07418 |\n",
       "| CG21502 | ab | CG2150  | 3308 | 3453 | 7091.4851 | 124.019957 | 5.825857 | 74.52481 |\n",
       "| CG130412 | ab | CG13041 |  296 |  121 |  717.5571 |   3.944247 | 7.181199 | 68.14210 |\n",
       "| Cpr56F2 | ab | Cpr56F  |  826 |  806 | 1955.1389 |  25.921788 | 6.182353 | 67.59657 |\n",
       "| CG130592 | ab | CG13059 |  566 |  463 | 1466.4702 |  16.134325 | 6.419315 | 67.52551 |\n",
       "| Cpr62Bc2 | ab | Cpr62Bc |  774 |  805 | 1912.4850 |  27.177162 | 6.084778 | 66.33617 |\n",
       "| Cpr50Cb2 | ab | Cpr50Cb |  258 |   92 |  649.5951 |   3.890549 | 7.053401 | 65.91837 |\n",
       "| CG344612 | ab | CG34461 |  868 | 1139 | 2090.4466 |  36.119330 | 5.815497 | 64.14659 |\n",
       "| Muc91C2 | ab | Muc91C  | 1506 | 2588 | 3697.1308 |  92.246957 | 5.309206 | 62.92779 |\n",
       "| Cpr66Cb2 | ab | Cpr66Cb |  743 |  545 | 1370.9431 |  20.047113 | 6.025403 | 62.79678 |\n",
       "| CG130602 | ab | CG13060 |  246 |  150 |  627.2689 |   5.226858 | 6.654436 | 61.85457 |\n",
       "| Cpr65Av2 | ab | Cpr65Av |  158 |   38 |  341.1285 |   1.226574 | 7.259346 | 61.11204 |\n",
       "| CG136752 | ab | CG13675 |  200 |   82 |  454.7549 |   3.370995 | 6.700984 | 59.18385 |\n",
       "| Ccp84Ac2 | ab | Ccp84Ac |  591 | 1019 | 1559.8422 |  35.995950 | 5.397889 | 57.26139 |\n",
       "| TwdlV2 | ab | TwdlV   |  418 |  491 |  950.9060 |  17.492830 | 5.684265 | 56.24395 |\n",
       "| Cpr65Eb2 | ab | Cpr65Eb |  819 | 1496 | 1907.6521 |  55.947416 | 5.066024 | 55.21124 |\n",
       "| CG130432 | ab | CG13043 |  226 |  254 |  587.9498 |   8.530352 | 5.947020 | 54.72448 |\n",
       "| Vajk12 | ab | Vajk1   | 1314 | 2549 | 2634.7168 |  93.141487 | 4.806673 | 54.62294 |\n",
       "\n"
      ],
      "text/plain": [
       "         cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "TwdlC2   ab          TwdlC   1284               866        3105.3762      \n",
       "CG125232 ab          CG12523  826               414        1943.2159      \n",
       "TwdlE2   ab          TwdlE   1509              1199        3559.5434      \n",
       "CG21502  ab          CG2150  3308              3453        7091.4851      \n",
       "CG130412 ab          CG13041  296               121         717.5571      \n",
       "Cpr56F2  ab          Cpr56F   826               806        1955.1389      \n",
       "CG130592 ab          CG13059  566               463        1466.4702      \n",
       "Cpr62Bc2 ab          Cpr62Bc  774               805        1912.4850      \n",
       "Cpr50Cb2 ab          Cpr50Cb  258                92         649.5951      \n",
       "CG344612 ab          CG34461  868              1139        2090.4466      \n",
       "Muc91C2  ab          Muc91C  1506              2588        3697.1308      \n",
       "Cpr66Cb2 ab          Cpr66Cb  743               545        1370.9431      \n",
       "CG130602 ab          CG13060  246               150         627.2689      \n",
       "Cpr65Av2 ab          Cpr65Av  158                38         341.1285      \n",
       "CG136752 ab          CG13675  200                82         454.7549      \n",
       "Ccp84Ac2 ab          Ccp84Ac  591              1019        1559.8422      \n",
       "TwdlV2   ab          TwdlV    418               491         950.9060      \n",
       "Cpr65Eb2 ab          Cpr65Eb  819              1496        1907.6521      \n",
       "CG130432 ab          CG13043  226               254         587.9498      \n",
       "Vajk12   ab          Vajk1   1314              2549        2634.7168      \n",
       "         other.tpm  log2.ratio heuristic.score\n",
       "TwdlC2    28.978450 6.694698   77.66531       \n",
       "CG125232  13.532165 7.063053   77.16366       \n",
       "TwdlE2    42.234625 6.363361   75.07418       \n",
       "CG21502  124.019957 5.825857   74.52481       \n",
       "CG130412   3.944247 7.181199   68.14210       \n",
       "Cpr56F2   25.921788 6.182353   67.59657       \n",
       "CG130592  16.134325 6.419315   67.52551       \n",
       "Cpr62Bc2  27.177162 6.084778   66.33617       \n",
       "Cpr50Cb2   3.890549 7.053401   65.91837       \n",
       "CG344612  36.119330 5.815497   64.14659       \n",
       "Muc91C2   92.246957 5.309206   62.92779       \n",
       "Cpr66Cb2  20.047113 6.025403   62.79678       \n",
       "CG130602   5.226858 6.654436   61.85457       \n",
       "Cpr65Av2   1.226574 7.259346   61.11204       \n",
       "CG136752   3.370995 6.700984   59.18385       \n",
       "Ccp84Ac2  35.995950 5.397889   57.26139       \n",
       "TwdlV2    17.492830 5.684265   56.24395       \n",
       "Cpr65Eb2  55.947416 5.066024   55.21124       \n",
       "CG130432   8.530352 5.947020   54.72448       \n",
       "Vajk12    93.141487 4.806673   54.62294       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"ab\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"TwdlC\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"TwdlC\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"TwdlC\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"TwdlE\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Cpr56F\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ab' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ab' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### ac cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1139,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Salivary_gland\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Salivary_gland\", df_sub300$cluster %in% ac)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ac' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ac' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ac)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(113)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1136,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>pip3</th><td>ac</td><td>pip        </td><td>47027</td><td> 6925</td><td>32792.0852</td><td> 264.990019</td><td>6.945832</td><td>104.19514</td></tr>\n",
       "\t<tr><th scope=row>nur3</th><td>ac</td><td>nur        </td><td>26613</td><td> 9706</td><td>19927.8068</td><td> 355.439960</td><td>5.804980</td><td> 82.91002</td></tr>\n",
       "\t<tr><th scope=row>CG147563</th><td>ac</td><td>CG14756    </td><td>14481</td><td> 5763</td><td>10818.1132</td><td> 201.054867</td><td>5.742558</td><td> 76.95771</td></tr>\n",
       "\t<tr><th scope=row>CG131593</th><td>ac</td><td>CG13159    </td><td> 6748</td><td> 2596</td><td> 4972.8202</td><td> 104.269209</td><td>5.561909</td><td> 68.30101</td></tr>\n",
       "\t<tr><th scope=row>CG341903</th><td>ac</td><td>CG34190    </td><td>  914</td><td>  134</td><td>  618.1632</td><td>   4.290539</td><td>6.868429</td><td> 63.69902</td></tr>\n",
       "\t<tr><th scope=row>PH4alphaSG13</th><td>ac</td><td>PH4alphaSG1</td><td> 1719</td><td>  504</td><td> 1247.2033</td><td>  17.378873</td><td>6.084505</td><td> 62.58301</td></tr>\n",
       "\t<tr><th scope=row>PH4alphaSG23</th><td>ac</td><td>PH4alphaSG2</td><td> 1097</td><td>  293</td><td>  789.6215</td><td>   9.326634</td><td>6.256719</td><td> 60.23246</td></tr>\n",
       "\t<tr><th scope=row>CrzR3</th><td>ac</td><td>CrzR       </td><td>  952</td><td>  222</td><td>  676.5130</td><td>   9.290377</td><td>6.038750</td><td> 56.78904</td></tr>\n",
       "\t<tr><th scope=row>GlcAT-P3</th><td>ac</td><td>GlcAT-P    </td><td> 5727</td><td> 4643</td><td> 4109.0061</td><td> 205.663831</td><td>4.313432</td><td> 51.78242</td></tr>\n",
       "\t<tr><th scope=row>sens3</th><td>ac</td><td>sens       </td><td> 1306</td><td>  822</td><td>  944.1735</td><td>  32.547843</td><td>4.814760</td><td> 47.59118</td></tr>\n",
       "\t<tr><th scope=row>CG87083</th><td>ac</td><td>CG8708     </td><td>  790</td><td>  442</td><td>  579.3617</td><td>  14.945111</td><td>5.183278</td><td> 47.58669</td></tr>\n",
       "\t<tr><th scope=row>Gmap3</th><td>ac</td><td>Gmap       </td><td> 9509</td><td>11758</td><td> 6880.6581</td><td> 544.069728</td><td>3.658034</td><td> 46.63459</td></tr>\n",
       "\t<tr><th scope=row>CG153123</th><td>ac</td><td>CG15312    </td><td> 1513</td><td> 1072</td><td> 1106.4770</td><td>  47.536391</td><td>4.510763</td><td> 45.61762</td></tr>\n",
       "\t<tr><th scope=row>CrebA3</th><td>ac</td><td>CrebA      </td><td>15475</td><td>26146</td><td>10924.0562</td><td>1132.625155</td><td>3.268493</td><td> 43.84799</td></tr>\n",
       "\t<tr><th scope=row>sll3</th><td>ac</td><td>sll        </td><td> 1125</td><td>  890</td><td>  810.4889</td><td>  35.191709</td><td>4.485061</td><td> 43.34555</td></tr>\n",
       "\t<tr><th scope=row>Tpst3</th><td>ac</td><td>Tpst       </td><td> 1820</td><td> 1922</td><td> 1297.8764</td><td>  75.051992</td><td>4.093023</td><td> 42.33434</td></tr>\n",
       "\t<tr><th scope=row>CG139503</th><td>ac</td><td>CG13950    </td><td>  370</td><td>  133</td><td>  275.1557</td><td>   6.634145</td><td>5.171638</td><td> 41.93856</td></tr>\n",
       "\t<tr><th scope=row>ome3</th><td>ac</td><td>ome        </td><td> 4581</td><td> 5999</td><td> 3236.9477</td><td> 271.891477</td><td>3.568235</td><td> 41.60870</td></tr>\n",
       "\t<tr><th scope=row>Papss3</th><td>ac</td><td>Papss      </td><td> 2028</td><td> 2942</td><td> 1473.9783</td><td> 102.321933</td><td>3.834497</td><td> 40.36375</td></tr>\n",
       "\t<tr><th scope=row>Pde93</th><td>ac</td><td>Pde9       </td><td> 5927</td><td>10370</td><td> 4189.3378</td><td> 486.253705</td><td>3.103977</td><td> 37.34969</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tpip3 & ac & pip         & 47027 &  6925 & 32792.0852 &  264.990019 & 6.945832 & 104.19514\\\\\n",
       "\tnur3 & ac & nur         & 26613 &  9706 & 19927.8068 &  355.439960 & 5.804980 &  82.91002\\\\\n",
       "\tCG147563 & ac & CG14756     & 14481 &  5763 & 10818.1132 &  201.054867 & 5.742558 &  76.95771\\\\\n",
       "\tCG131593 & ac & CG13159     &  6748 &  2596 &  4972.8202 &  104.269209 & 5.561909 &  68.30101\\\\\n",
       "\tCG341903 & ac & CG34190     &   914 &   134 &   618.1632 &    4.290539 & 6.868429 &  63.69902\\\\\n",
       "\tPH4alphaSG13 & ac & PH4alphaSG1 &  1719 &   504 &  1247.2033 &   17.378873 & 6.084505 &  62.58301\\\\\n",
       "\tPH4alphaSG23 & ac & PH4alphaSG2 &  1097 &   293 &   789.6215 &    9.326634 & 6.256719 &  60.23246\\\\\n",
       "\tCrzR3 & ac & CrzR        &   952 &   222 &   676.5130 &    9.290377 & 6.038750 &  56.78904\\\\\n",
       "\tGlcAT-P3 & ac & GlcAT-P     &  5727 &  4643 &  4109.0061 &  205.663831 & 4.313432 &  51.78242\\\\\n",
       "\tsens3 & ac & sens        &  1306 &   822 &   944.1735 &   32.547843 & 4.814760 &  47.59118\\\\\n",
       "\tCG87083 & ac & CG8708      &   790 &   442 &   579.3617 &   14.945111 & 5.183278 &  47.58669\\\\\n",
       "\tGmap3 & ac & Gmap        &  9509 & 11758 &  6880.6581 &  544.069728 & 3.658034 &  46.63459\\\\\n",
       "\tCG153123 & ac & CG15312     &  1513 &  1072 &  1106.4770 &   47.536391 & 4.510763 &  45.61762\\\\\n",
       "\tCrebA3 & ac & CrebA       & 15475 & 26146 & 10924.0562 & 1132.625155 & 3.268493 &  43.84799\\\\\n",
       "\tsll3 & ac & sll         &  1125 &   890 &   810.4889 &   35.191709 & 4.485061 &  43.34555\\\\\n",
       "\tTpst3 & ac & Tpst        &  1820 &  1922 &  1297.8764 &   75.051992 & 4.093023 &  42.33434\\\\\n",
       "\tCG139503 & ac & CG13950     &   370 &   133 &   275.1557 &    6.634145 & 5.171638 &  41.93856\\\\\n",
       "\tome3 & ac & ome         &  4581 &  5999 &  3236.9477 &  271.891477 & 3.568235 &  41.60870\\\\\n",
       "\tPapss3 & ac & Papss       &  2028 &  2942 &  1473.9783 &  102.321933 & 3.834497 &  40.36375\\\\\n",
       "\tPde93 & ac & Pde9        &  5927 & 10370 &  4189.3378 &  486.253705 & 3.103977 &  37.34969\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| pip3 | ac | pip         | 47027 |  6925 | 32792.0852 |  264.990019 | 6.945832 | 104.19514 |\n",
       "| nur3 | ac | nur         | 26613 |  9706 | 19927.8068 |  355.439960 | 5.804980 |  82.91002 |\n",
       "| CG147563 | ac | CG14756     | 14481 |  5763 | 10818.1132 |  201.054867 | 5.742558 |  76.95771 |\n",
       "| CG131593 | ac | CG13159     |  6748 |  2596 |  4972.8202 |  104.269209 | 5.561909 |  68.30101 |\n",
       "| CG341903 | ac | CG34190     |   914 |   134 |   618.1632 |    4.290539 | 6.868429 |  63.69902 |\n",
       "| PH4alphaSG13 | ac | PH4alphaSG1 |  1719 |   504 |  1247.2033 |   17.378873 | 6.084505 |  62.58301 |\n",
       "| PH4alphaSG23 | ac | PH4alphaSG2 |  1097 |   293 |   789.6215 |    9.326634 | 6.256719 |  60.23246 |\n",
       "| CrzR3 | ac | CrzR        |   952 |   222 |   676.5130 |    9.290377 | 6.038750 |  56.78904 |\n",
       "| GlcAT-P3 | ac | GlcAT-P     |  5727 |  4643 |  4109.0061 |  205.663831 | 4.313432 |  51.78242 |\n",
       "| sens3 | ac | sens        |  1306 |   822 |   944.1735 |   32.547843 | 4.814760 |  47.59118 |\n",
       "| CG87083 | ac | CG8708      |   790 |   442 |   579.3617 |   14.945111 | 5.183278 |  47.58669 |\n",
       "| Gmap3 | ac | Gmap        |  9509 | 11758 |  6880.6581 |  544.069728 | 3.658034 |  46.63459 |\n",
       "| CG153123 | ac | CG15312     |  1513 |  1072 |  1106.4770 |   47.536391 | 4.510763 |  45.61762 |\n",
       "| CrebA3 | ac | CrebA       | 15475 | 26146 | 10924.0562 | 1132.625155 | 3.268493 |  43.84799 |\n",
       "| sll3 | ac | sll         |  1125 |   890 |   810.4889 |   35.191709 | 4.485061 |  43.34555 |\n",
       "| Tpst3 | ac | Tpst        |  1820 |  1922 |  1297.8764 |   75.051992 | 4.093023 |  42.33434 |\n",
       "| CG139503 | ac | CG13950     |   370 |   133 |   275.1557 |    6.634145 | 5.171638 |  41.93856 |\n",
       "| ome3 | ac | ome         |  4581 |  5999 |  3236.9477 |  271.891477 | 3.568235 |  41.60870 |\n",
       "| Papss3 | ac | Papss       |  2028 |  2942 |  1473.9783 |  102.321933 | 3.834497 |  40.36375 |\n",
       "| Pde93 | ac | Pde9        |  5927 | 10370 |  4189.3378 |  486.253705 | 3.103977 |  37.34969 |\n",
       "\n"
      ],
      "text/plain": [
       "             cell.type.2 gene        cell.type.2.n.umi other.n.umi\n",
       "pip3         ac          pip         47027              6925      \n",
       "nur3         ac          nur         26613              9706      \n",
       "CG147563     ac          CG14756     14481              5763      \n",
       "CG131593     ac          CG13159      6748              2596      \n",
       "CG341903     ac          CG34190       914               134      \n",
       "PH4alphaSG13 ac          PH4alphaSG1  1719               504      \n",
       "PH4alphaSG23 ac          PH4alphaSG2  1097               293      \n",
       "CrzR3        ac          CrzR          952               222      \n",
       "GlcAT-P3     ac          GlcAT-P      5727              4643      \n",
       "sens3        ac          sens         1306               822      \n",
       "CG87083      ac          CG8708        790               442      \n",
       "Gmap3        ac          Gmap         9509             11758      \n",
       "CG153123     ac          CG15312      1513              1072      \n",
       "CrebA3       ac          CrebA       15475             26146      \n",
       "sll3         ac          sll          1125               890      \n",
       "Tpst3        ac          Tpst         1820              1922      \n",
       "CG139503     ac          CG13950       370               133      \n",
       "ome3         ac          ome          4581              5999      \n",
       "Papss3       ac          Papss        2028              2942      \n",
       "Pde93        ac          Pde9         5927             10370      \n",
       "             cell.type.2.tpm other.tpm   log2.ratio heuristic.score\n",
       "pip3         32792.0852       264.990019 6.945832   104.19514      \n",
       "nur3         19927.8068       355.439960 5.804980    82.91002      \n",
       "CG147563     10818.1132       201.054867 5.742558    76.95771      \n",
       "CG131593      4972.8202       104.269209 5.561909    68.30101      \n",
       "CG341903       618.1632         4.290539 6.868429    63.69902      \n",
       "PH4alphaSG13  1247.2033        17.378873 6.084505    62.58301      \n",
       "PH4alphaSG23   789.6215         9.326634 6.256719    60.23246      \n",
       "CrzR3          676.5130         9.290377 6.038750    56.78904      \n",
       "GlcAT-P3      4109.0061       205.663831 4.313432    51.78242      \n",
       "sens3          944.1735        32.547843 4.814760    47.59118      \n",
       "CG87083        579.3617        14.945111 5.183278    47.58669      \n",
       "Gmap3         6880.6581       544.069728 3.658034    46.63459      \n",
       "CG153123      1106.4770        47.536391 4.510763    45.61762      \n",
       "CrebA3       10924.0562      1132.625155 3.268493    43.84799      \n",
       "sll3           810.4889        35.191709 4.485061    43.34555      \n",
       "Tpst3         1297.8764        75.051992 4.093023    42.33434      \n",
       "CG139503       275.1557         6.634145 5.171638    41.93856      \n",
       "ome3          3236.9477       271.891477 3.568235    41.60870      \n",
       "Papss3        1473.9783       102.321933 3.834497    40.36375      \n",
       "Pde93         4189.3378       486.253705 3.103977    37.34969      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"ac\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"pip\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"pip\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"pip\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"nur\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"PH4alphaSG1\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1143,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Analpad\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Analpad\", df_sub300$cluster %in% ad)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ad' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ad' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ad)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(113)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1141,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG177824</th><td>ad</td><td>CG17782</td><td> 340</td><td>  90</td><td>1323.3694</td><td>  3.3047155</td><td>8.264082</td><td>85.70754</td></tr>\n",
       "\t<tr><th scope=row>CG57684</th><td>ad</td><td>CG5768 </td><td> 211</td><td>  33</td><td> 809.5932</td><td>  1.2391984</td><td>8.498071</td><td>82.11545</td></tr>\n",
       "\t<tr><th scope=row>CG136154</th><td>ad</td><td>CG13615</td><td> 191</td><td>  27</td><td> 737.4114</td><td>  1.0560434</td><td>8.486455</td><td>80.86133</td></tr>\n",
       "\t<tr><th scope=row>CG330034</th><td>ad</td><td>CG33003</td><td> 632</td><td> 745</td><td>2356.7963</td><td> 34.0934208</td><td>6.069483</td><td>67.99777</td></tr>\n",
       "\t<tr><th scope=row>CG177844</th><td>ad</td><td>CG17784</td><td> 101</td><td>  14</td><td> 373.1821</td><td>  0.5367777</td><td>7.923828</td><td>67.72968</td></tr>\n",
       "\t<tr><th scope=row>CG344444</th><td>ad</td><td>CG34444</td><td> 113</td><td>  29</td><td> 421.8502</td><td>  1.0059735</td><td>7.716285</td><td>67.31689</td></tr>\n",
       "\t<tr><th scope=row>CG67854</th><td>ad</td><td>CG6785 </td><td> 361</td><td> 443</td><td>1392.6316</td><td> 17.8752324</td><td>6.205175</td><td>64.81078</td></tr>\n",
       "\t<tr><th scope=row>CG344424</th><td>ad</td><td>CG34442</td><td> 110</td><td>  37</td><td> 428.0240</td><td>  1.6051126</td><td>7.360202</td><td>64.36434</td></tr>\n",
       "\t<tr><th scope=row>CG177804</th><td>ad</td><td>CG17780</td><td> 305</td><td> 342</td><td>1159.1576</td><td> 15.0652428</td><td>6.172990</td><td>62.84169</td></tr>\n",
       "\t<tr><th scope=row>CG136134</th><td>ad</td><td>CG13613</td><td>  57</td><td>   4</td><td> 236.3762</td><td>  0.1169610</td><td>7.725362</td><td>60.96108</td></tr>\n",
       "\t<tr><th scope=row>CG333404</th><td>ad</td><td>CG33340</td><td>  62</td><td>  13</td><td> 262.5149</td><td>  0.3912239</td><td>7.559901</td><td>60.79476</td></tr>\n",
       "\t<tr><th scope=row>CG333394</th><td>ad</td><td>CG33339</td><td>  69</td><td>  24</td><td> 267.9977</td><td>  0.8353557</td><td>7.190017</td><td>58.03386</td></tr>\n",
       "\t<tr><th scope=row>CG179994</th><td>ad</td><td>CG17999</td><td>  85</td><td>  38</td><td> 323.9834</td><td>  1.8909338</td><td>6.808240</td><td>56.80947</td></tr>\n",
       "\t<tr><th scope=row>Lgr14</th><td>ad</td><td>Lgr1   </td><td> 188</td><td> 296</td><td> 718.4721</td><td> 13.7366895</td><td>5.607448</td><td>53.21914</td></tr>\n",
       "\t<tr><th scope=row>otp4</th><td>ad</td><td>otp    </td><td>1186</td><td>4789</td><td>4598.9373</td><td>226.7808372</td><td>4.335582</td><td>52.75276</td></tr>\n",
       "\t<tr><th scope=row>List4</th><td>ad</td><td>List   </td><td>  66</td><td>  47</td><td> 235.2981</td><td>  1.9653946</td><td>6.310122</td><td>49.75193</td></tr>\n",
       "\t<tr><th scope=row>CG99934</th><td>ad</td><td>CG9993 </td><td>  57</td><td>  81</td><td> 230.2043</td><td>  3.4157128</td><td>5.704125</td><td>44.79463</td></tr>\n",
       "\t<tr><th scope=row>CG344284</th><td>ad</td><td>CG34428</td><td>  32</td><td>  10</td><td> 122.4980</td><td>  0.4121325</td><td>6.438738</td><td>44.73856</td></tr>\n",
       "\t<tr><th scope=row>dysf4</th><td>ad</td><td>dysf   </td><td> 252</td><td> 951</td><td> 972.9676</td><td> 43.6469470</td><td>4.445758</td><td>44.13629</td></tr>\n",
       "\t<tr><th scope=row>CG113184</th><td>ad</td><td>CG11318</td><td>  52</td><td>  70</td><td> 212.4839</td><td>  3.3279471</td><td>5.617527</td><td>43.46833</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG177824 & ad & CG17782 &  340 &   90 & 1323.3694 &   3.3047155 & 8.264082 & 85.70754\\\\\n",
       "\tCG57684 & ad & CG5768  &  211 &   33 &  809.5932 &   1.2391984 & 8.498071 & 82.11545\\\\\n",
       "\tCG136154 & ad & CG13615 &  191 &   27 &  737.4114 &   1.0560434 & 8.486455 & 80.86133\\\\\n",
       "\tCG330034 & ad & CG33003 &  632 &  745 & 2356.7963 &  34.0934208 & 6.069483 & 67.99777\\\\\n",
       "\tCG177844 & ad & CG17784 &  101 &   14 &  373.1821 &   0.5367777 & 7.923828 & 67.72968\\\\\n",
       "\tCG344444 & ad & CG34444 &  113 &   29 &  421.8502 &   1.0059735 & 7.716285 & 67.31689\\\\\n",
       "\tCG67854 & ad & CG6785  &  361 &  443 & 1392.6316 &  17.8752324 & 6.205175 & 64.81078\\\\\n",
       "\tCG344424 & ad & CG34442 &  110 &   37 &  428.0240 &   1.6051126 & 7.360202 & 64.36434\\\\\n",
       "\tCG177804 & ad & CG17780 &  305 &  342 & 1159.1576 &  15.0652428 & 6.172990 & 62.84169\\\\\n",
       "\tCG136134 & ad & CG13613 &   57 &    4 &  236.3762 &   0.1169610 & 7.725362 & 60.96108\\\\\n",
       "\tCG333404 & ad & CG33340 &   62 &   13 &  262.5149 &   0.3912239 & 7.559901 & 60.79476\\\\\n",
       "\tCG333394 & ad & CG33339 &   69 &   24 &  267.9977 &   0.8353557 & 7.190017 & 58.03386\\\\\n",
       "\tCG179994 & ad & CG17999 &   85 &   38 &  323.9834 &   1.8909338 & 6.808240 & 56.80947\\\\\n",
       "\tLgr14 & ad & Lgr1    &  188 &  296 &  718.4721 &  13.7366895 & 5.607448 & 53.21914\\\\\n",
       "\totp4 & ad & otp     & 1186 & 4789 & 4598.9373 & 226.7808372 & 4.335582 & 52.75276\\\\\n",
       "\tList4 & ad & List    &   66 &   47 &  235.2981 &   1.9653946 & 6.310122 & 49.75193\\\\\n",
       "\tCG99934 & ad & CG9993  &   57 &   81 &  230.2043 &   3.4157128 & 5.704125 & 44.79463\\\\\n",
       "\tCG344284 & ad & CG34428 &   32 &   10 &  122.4980 &   0.4121325 & 6.438738 & 44.73856\\\\\n",
       "\tdysf4 & ad & dysf    &  252 &  951 &  972.9676 &  43.6469470 & 4.445758 & 44.13629\\\\\n",
       "\tCG113184 & ad & CG11318 &   52 &   70 &  212.4839 &   3.3279471 & 5.617527 & 43.46833\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG177824 | ad | CG17782 |  340 |   90 | 1323.3694 |   3.3047155 | 8.264082 | 85.70754 |\n",
       "| CG57684 | ad | CG5768  |  211 |   33 |  809.5932 |   1.2391984 | 8.498071 | 82.11545 |\n",
       "| CG136154 | ad | CG13615 |  191 |   27 |  737.4114 |   1.0560434 | 8.486455 | 80.86133 |\n",
       "| CG330034 | ad | CG33003 |  632 |  745 | 2356.7963 |  34.0934208 | 6.069483 | 67.99777 |\n",
       "| CG177844 | ad | CG17784 |  101 |   14 |  373.1821 |   0.5367777 | 7.923828 | 67.72968 |\n",
       "| CG344444 | ad | CG34444 |  113 |   29 |  421.8502 |   1.0059735 | 7.716285 | 67.31689 |\n",
       "| CG67854 | ad | CG6785  |  361 |  443 | 1392.6316 |  17.8752324 | 6.205175 | 64.81078 |\n",
       "| CG344424 | ad | CG34442 |  110 |   37 |  428.0240 |   1.6051126 | 7.360202 | 64.36434 |\n",
       "| CG177804 | ad | CG17780 |  305 |  342 | 1159.1576 |  15.0652428 | 6.172990 | 62.84169 |\n",
       "| CG136134 | ad | CG13613 |   57 |    4 |  236.3762 |   0.1169610 | 7.725362 | 60.96108 |\n",
       "| CG333404 | ad | CG33340 |   62 |   13 |  262.5149 |   0.3912239 | 7.559901 | 60.79476 |\n",
       "| CG333394 | ad | CG33339 |   69 |   24 |  267.9977 |   0.8353557 | 7.190017 | 58.03386 |\n",
       "| CG179994 | ad | CG17999 |   85 |   38 |  323.9834 |   1.8909338 | 6.808240 | 56.80947 |\n",
       "| Lgr14 | ad | Lgr1    |  188 |  296 |  718.4721 |  13.7366895 | 5.607448 | 53.21914 |\n",
       "| otp4 | ad | otp     | 1186 | 4789 | 4598.9373 | 226.7808372 | 4.335582 | 52.75276 |\n",
       "| List4 | ad | List    |   66 |   47 |  235.2981 |   1.9653946 | 6.310122 | 49.75193 |\n",
       "| CG99934 | ad | CG9993  |   57 |   81 |  230.2043 |   3.4157128 | 5.704125 | 44.79463 |\n",
       "| CG344284 | ad | CG34428 |   32 |   10 |  122.4980 |   0.4121325 | 6.438738 | 44.73856 |\n",
       "| dysf4 | ad | dysf    |  252 |  951 |  972.9676 |  43.6469470 | 4.445758 | 44.13629 |\n",
       "| CG113184 | ad | CG11318 |   52 |   70 |  212.4839 |   3.3279471 | 5.617527 | 43.46833 |\n",
       "\n"
      ],
      "text/plain": [
       "         cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "CG177824 ad          CG17782  340                90        1323.3694      \n",
       "CG57684  ad          CG5768   211                33         809.5932      \n",
       "CG136154 ad          CG13615  191                27         737.4114      \n",
       "CG330034 ad          CG33003  632               745        2356.7963      \n",
       "CG177844 ad          CG17784  101                14         373.1821      \n",
       "CG344444 ad          CG34444  113                29         421.8502      \n",
       "CG67854  ad          CG6785   361               443        1392.6316      \n",
       "CG344424 ad          CG34442  110                37         428.0240      \n",
       "CG177804 ad          CG17780  305               342        1159.1576      \n",
       "CG136134 ad          CG13613   57                 4         236.3762      \n",
       "CG333404 ad          CG33340   62                13         262.5149      \n",
       "CG333394 ad          CG33339   69                24         267.9977      \n",
       "CG179994 ad          CG17999   85                38         323.9834      \n",
       "Lgr14    ad          Lgr1     188               296         718.4721      \n",
       "otp4     ad          otp     1186              4789        4598.9373      \n",
       "List4    ad          List      66                47         235.2981      \n",
       "CG99934  ad          CG9993    57                81         230.2043      \n",
       "CG344284 ad          CG34428   32                10         122.4980      \n",
       "dysf4    ad          dysf     252               951         972.9676      \n",
       "CG113184 ad          CG11318   52                70         212.4839      \n",
       "         other.tpm   log2.ratio heuristic.score\n",
       "CG177824   3.3047155 8.264082   85.70754       \n",
       "CG57684    1.2391984 8.498071   82.11545       \n",
       "CG136154   1.0560434 8.486455   80.86133       \n",
       "CG330034  34.0934208 6.069483   67.99777       \n",
       "CG177844   0.5367777 7.923828   67.72968       \n",
       "CG344444   1.0059735 7.716285   67.31689       \n",
       "CG67854   17.8752324 6.205175   64.81078       \n",
       "CG344424   1.6051126 7.360202   64.36434       \n",
       "CG177804  15.0652428 6.172990   62.84169       \n",
       "CG136134   0.1169610 7.725362   60.96108       \n",
       "CG333404   0.3912239 7.559901   60.79476       \n",
       "CG333394   0.8353557 7.190017   58.03386       \n",
       "CG179994   1.8909338 6.808240   56.80947       \n",
       "Lgr14     13.7366895 5.607448   53.21914       \n",
       "otp4     226.7808372 4.335582   52.75276       \n",
       "List4      1.9653946 6.310122   49.75193       \n",
       "CG99934    3.4157128 5.704125   44.79463       \n",
       "CG344284   0.4121325 6.438738   44.73856       \n",
       "dysf4     43.6469470 4.445758   44.13629       \n",
       "CG113184   3.3279471 5.617527   43.46833       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"ad\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG17782\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG17782\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG17782\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG5768\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG13615\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1602,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sub300_copy = df_sub300"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1603,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Garland_cell_nephrocyte\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Garland_cell_nephrocyte\", df_sub300$cluster %in% ae &\n",
    "                         df_sub300$umap_2 >= -0.1*df_sub300$umap_1 -0.7)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1186,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Midgut_primordium\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Midgut_primordium\", df_sub300$cluster %in% 107 &  \n",
    "              df_sub300$umap_2 > -0.1*df_sub300$umap_1 -1.3 &\n",
    "              df_sub300$umap_2 < -0.1*df_sub300$umap_1 -0.7)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1200,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Copper_cells\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Copper_cells\", df_sub300$cluster %in% 107 & \n",
    "                          df_sub300$umap_2 <= -0.1*df_sub300$umap_1 -1.3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ae' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ae' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ae)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(166)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c( 107)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1151,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Cubn5</th><td>ae</td><td>Cubn      </td><td> 634</td><td>  843</td><td> 1629.77914</td><td>  35.53650</td><td>5.479194</td><td>58.47038</td></tr>\n",
       "\t<tr><th scope=row>CG422555</th><td>ae</td><td>CG42255   </td><td> 266</td><td>  278</td><td>  684.74277</td><td>  11.27696</td><td>5.801537</td><td>54.65932</td></tr>\n",
       "\t<tr><th scope=row>Amnionless5</th><td>ae</td><td>Amnionless</td><td> 156</td><td>  288</td><td>  391.58419</td><td>  10.82043</td><td>5.049968</td><td>43.51486</td></tr>\n",
       "\t<tr><th scope=row>CG343475</th><td>ae</td><td>CG34347   </td><td>1911</td><td>10156</td><td> 5039.30873</td><td> 484.25921</td><td>3.376398</td><td>41.52732</td></tr>\n",
       "\t<tr><th scope=row>HHEX5</th><td>ae</td><td>HHEX      </td><td> 140</td><td>  495</td><td>  373.62416</td><td>  22.94760</td><td>3.963635</td><td>33.88631</td></tr>\n",
       "\t<tr><th scope=row>wb5</th><td>ae</td><td>wb        </td><td>1183</td><td>10438</td><td> 3307.68454</td><td> 465.56124</td><td>2.825683</td><td>33.03801</td></tr>\n",
       "\t<tr><th scope=row>Bx5</th><td>ae</td><td>Bx        </td><td> 442</td><td> 3359</td><td> 1193.80577</td><td> 149.31455</td><td>2.989512</td><td>30.56046</td></tr>\n",
       "\t<tr><th scope=row>Ptx15</th><td>ae</td><td>Ptx1      </td><td> 434</td><td> 3232</td><td> 1149.58829</td><td> 142.96691</td><td>2.997308</td><td>30.47710</td></tr>\n",
       "\t<tr><th scope=row>grn5</th><td>ae</td><td>grn       </td><td>1572</td><td>17793</td><td> 4405.42469</td><td> 788.64217</td><td>2.480010</td><td>30.02150</td></tr>\n",
       "\t<tr><th scope=row>Pdp15</th><td>ae</td><td>Pdp1      </td><td>4250</td><td>57404</td><td>11678.68300</td><td>2504.35268</td><td>2.220792</td><td>30.00670</td></tr>\n",
       "\t<tr><th scope=row>Clk5</th><td>ae</td><td>Clk       </td><td>  23</td><td>   27</td><td>   64.75099</td><td>   1.09229</td><td>4.951747</td><td>29.90331</td></tr>\n",
       "\t<tr><th scope=row>bru25</th><td>ae</td><td>bru2      </td><td>1277</td><td>13549</td><td> 3567.25115</td><td> 624.12018</td><td>2.512607</td><td>29.65128</td></tr>\n",
       "\t<tr><th scope=row>Myc5</th><td>ae</td><td>Myc       </td><td>2155</td><td>29690</td><td> 5959.94821</td><td>1199.83286</td><td>2.311264</td><td>28.98632</td></tr>\n",
       "\t<tr><th scope=row>Sox21a5</th><td>ae</td><td>Sox21a    </td><td>  67</td><td>  399</td><td>  216.86324</td><td>  15.74107</td><td>3.695322</td><td>28.70260</td></tr>\n",
       "\t<tr><th scope=row>Zasp525</th><td>ae</td><td>Zasp52    </td><td> 766</td><td> 8598</td><td> 2132.77890</td><td> 367.05244</td><td>2.534751</td><td>28.03231</td></tr>\n",
       "\t<tr><th scope=row>CG425945</th><td>ae</td><td>CG42594   </td><td> 212</td><td> 1734</td><td>  621.77358</td><td>  79.64054</td><td>2.946812</td><td>27.35397</td></tr>\n",
       "\t<tr><th scope=row>CG178345</th><td>ae</td><td>CG17834   </td><td> 193</td><td> 1523</td><td>  555.87165</td><td>  71.65942</td><td>2.935530</td><td>26.77556</td></tr>\n",
       "\t<tr><th scope=row>lab5</th><td>ae</td><td>lab       </td><td> 339</td><td> 2963</td><td>  901.38059</td><td> 136.18647</td><td>2.715998</td><td>26.66456</td></tr>\n",
       "\t<tr><th scope=row>Trim95</th><td>ae</td><td>Trim9     </td><td> 915</td><td>11336</td><td> 2543.38654</td><td> 511.64168</td><td>2.310728</td><td>26.14150</td></tr>\n",
       "\t<tr><th scope=row>Oatp33Ea5</th><td>ae</td><td>Oatp33Ea  </td><td>  99</td><td>  669</td><td>  268.05304</td><td>  28.55585</td><td>3.181003</td><td>25.67625</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCubn5 & ae & Cubn       &  634 &   843 &  1629.77914 &   35.53650 & 5.479194 & 58.47038\\\\\n",
       "\tCG422555 & ae & CG42255    &  266 &   278 &   684.74277 &   11.27696 & 5.801537 & 54.65932\\\\\n",
       "\tAmnionless5 & ae & Amnionless &  156 &   288 &   391.58419 &   10.82043 & 5.049968 & 43.51486\\\\\n",
       "\tCG343475 & ae & CG34347    & 1911 & 10156 &  5039.30873 &  484.25921 & 3.376398 & 41.52732\\\\\n",
       "\tHHEX5 & ae & HHEX       &  140 &   495 &   373.62416 &   22.94760 & 3.963635 & 33.88631\\\\\n",
       "\twb5 & ae & wb         & 1183 & 10438 &  3307.68454 &  465.56124 & 2.825683 & 33.03801\\\\\n",
       "\tBx5 & ae & Bx         &  442 &  3359 &  1193.80577 &  149.31455 & 2.989512 & 30.56046\\\\\n",
       "\tPtx15 & ae & Ptx1       &  434 &  3232 &  1149.58829 &  142.96691 & 2.997308 & 30.47710\\\\\n",
       "\tgrn5 & ae & grn        & 1572 & 17793 &  4405.42469 &  788.64217 & 2.480010 & 30.02150\\\\\n",
       "\tPdp15 & ae & Pdp1       & 4250 & 57404 & 11678.68300 & 2504.35268 & 2.220792 & 30.00670\\\\\n",
       "\tClk5 & ae & Clk        &   23 &    27 &    64.75099 &    1.09229 & 4.951747 & 29.90331\\\\\n",
       "\tbru25 & ae & bru2       & 1277 & 13549 &  3567.25115 &  624.12018 & 2.512607 & 29.65128\\\\\n",
       "\tMyc5 & ae & Myc        & 2155 & 29690 &  5959.94821 & 1199.83286 & 2.311264 & 28.98632\\\\\n",
       "\tSox21a5 & ae & Sox21a     &   67 &   399 &   216.86324 &   15.74107 & 3.695322 & 28.70260\\\\\n",
       "\tZasp525 & ae & Zasp52     &  766 &  8598 &  2132.77890 &  367.05244 & 2.534751 & 28.03231\\\\\n",
       "\tCG425945 & ae & CG42594    &  212 &  1734 &   621.77358 &   79.64054 & 2.946812 & 27.35397\\\\\n",
       "\tCG178345 & ae & CG17834    &  193 &  1523 &   555.87165 &   71.65942 & 2.935530 & 26.77556\\\\\n",
       "\tlab5 & ae & lab        &  339 &  2963 &   901.38059 &  136.18647 & 2.715998 & 26.66456\\\\\n",
       "\tTrim95 & ae & Trim9      &  915 & 11336 &  2543.38654 &  511.64168 & 2.310728 & 26.14150\\\\\n",
       "\tOatp33Ea5 & ae & Oatp33Ea   &   99 &   669 &   268.05304 &   28.55585 & 3.181003 & 25.67625\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Cubn5 | ae | Cubn       |  634 |   843 |  1629.77914 |   35.53650 | 5.479194 | 58.47038 |\n",
       "| CG422555 | ae | CG42255    |  266 |   278 |   684.74277 |   11.27696 | 5.801537 | 54.65932 |\n",
       "| Amnionless5 | ae | Amnionless |  156 |   288 |   391.58419 |   10.82043 | 5.049968 | 43.51486 |\n",
       "| CG343475 | ae | CG34347    | 1911 | 10156 |  5039.30873 |  484.25921 | 3.376398 | 41.52732 |\n",
       "| HHEX5 | ae | HHEX       |  140 |   495 |   373.62416 |   22.94760 | 3.963635 | 33.88631 |\n",
       "| wb5 | ae | wb         | 1183 | 10438 |  3307.68454 |  465.56124 | 2.825683 | 33.03801 |\n",
       "| Bx5 | ae | Bx         |  442 |  3359 |  1193.80577 |  149.31455 | 2.989512 | 30.56046 |\n",
       "| Ptx15 | ae | Ptx1       |  434 |  3232 |  1149.58829 |  142.96691 | 2.997308 | 30.47710 |\n",
       "| grn5 | ae | grn        | 1572 | 17793 |  4405.42469 |  788.64217 | 2.480010 | 30.02150 |\n",
       "| Pdp15 | ae | Pdp1       | 4250 | 57404 | 11678.68300 | 2504.35268 | 2.220792 | 30.00670 |\n",
       "| Clk5 | ae | Clk        |   23 |    27 |    64.75099 |    1.09229 | 4.951747 | 29.90331 |\n",
       "| bru25 | ae | bru2       | 1277 | 13549 |  3567.25115 |  624.12018 | 2.512607 | 29.65128 |\n",
       "| Myc5 | ae | Myc        | 2155 | 29690 |  5959.94821 | 1199.83286 | 2.311264 | 28.98632 |\n",
       "| Sox21a5 | ae | Sox21a     |   67 |   399 |   216.86324 |   15.74107 | 3.695322 | 28.70260 |\n",
       "| Zasp525 | ae | Zasp52     |  766 |  8598 |  2132.77890 |  367.05244 | 2.534751 | 28.03231 |\n",
       "| CG425945 | ae | CG42594    |  212 |  1734 |   621.77358 |   79.64054 | 2.946812 | 27.35397 |\n",
       "| CG178345 | ae | CG17834    |  193 |  1523 |   555.87165 |   71.65942 | 2.935530 | 26.77556 |\n",
       "| lab5 | ae | lab        |  339 |  2963 |   901.38059 |  136.18647 | 2.715998 | 26.66456 |\n",
       "| Trim95 | ae | Trim9      |  915 | 11336 |  2543.38654 |  511.64168 | 2.310728 | 26.14150 |\n",
       "| Oatp33Ea5 | ae | Oatp33Ea   |   99 |   669 |   268.05304 |   28.55585 | 3.181003 | 25.67625 |\n",
       "\n"
      ],
      "text/plain": [
       "            cell.type.2 gene       cell.type.2.n.umi other.n.umi\n",
       "Cubn5       ae          Cubn        634                843      \n",
       "CG422555    ae          CG42255     266                278      \n",
       "Amnionless5 ae          Amnionless  156                288      \n",
       "CG343475    ae          CG34347    1911              10156      \n",
       "HHEX5       ae          HHEX        140                495      \n",
       "wb5         ae          wb         1183              10438      \n",
       "Bx5         ae          Bx          442               3359      \n",
       "Ptx15       ae          Ptx1        434               3232      \n",
       "grn5        ae          grn        1572              17793      \n",
       "Pdp15       ae          Pdp1       4250              57404      \n",
       "Clk5        ae          Clk          23                 27      \n",
       "bru25       ae          bru2       1277              13549      \n",
       "Myc5        ae          Myc        2155              29690      \n",
       "Sox21a5     ae          Sox21a       67                399      \n",
       "Zasp525     ae          Zasp52      766               8598      \n",
       "CG425945    ae          CG42594     212               1734      \n",
       "CG178345    ae          CG17834     193               1523      \n",
       "lab5        ae          lab         339               2963      \n",
       "Trim95      ae          Trim9       915              11336      \n",
       "Oatp33Ea5   ae          Oatp33Ea     99                669      \n",
       "            cell.type.2.tpm other.tpm  log2.ratio heuristic.score\n",
       "Cubn5        1629.77914       35.53650 5.479194   58.47038       \n",
       "CG422555      684.74277       11.27696 5.801537   54.65932       \n",
       "Amnionless5   391.58419       10.82043 5.049968   43.51486       \n",
       "CG343475     5039.30873      484.25921 3.376398   41.52732       \n",
       "HHEX5         373.62416       22.94760 3.963635   33.88631       \n",
       "wb5          3307.68454      465.56124 2.825683   33.03801       \n",
       "Bx5          1193.80577      149.31455 2.989512   30.56046       \n",
       "Ptx15        1149.58829      142.96691 2.997308   30.47710       \n",
       "grn5         4405.42469      788.64217 2.480010   30.02150       \n",
       "Pdp15       11678.68300     2504.35268 2.220792   30.00670       \n",
       "Clk5           64.75099        1.09229 4.951747   29.90331       \n",
       "bru25        3567.25115      624.12018 2.512607   29.65128       \n",
       "Myc5         5959.94821     1199.83286 2.311264   28.98632       \n",
       "Sox21a5       216.86324       15.74107 3.695322   28.70260       \n",
       "Zasp525      2132.77890      367.05244 2.534751   28.03231       \n",
       "CG425945      621.77358       79.64054 2.946812   27.35397       \n",
       "CG178345      555.87165       71.65942 2.935530   26.77556       \n",
       "lab5          901.38059      136.18647 2.715998   26.66456       \n",
       "Trim95       2543.38654      511.64168 2.310728   26.14150       \n",
       "Oatp33Ea5     268.05304       28.55585 3.181003   25.67625       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"ae\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Cubn\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Cubn\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Cubn\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG42255\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Amnionless\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "clus = c(107)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "              df_sub300$umap_2 <= -0.1*umap_1 -1.3, #&\n",
    "             # df_sub300$umap_2 < -0.1*umap_1 -0.7\n",
    "              ), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 0.5, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = -0.1, intercept = -1.3) +\n",
    "geom_abline(slope = -0.1, intercept = -0.7) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(-4, 0) + ylim(-3, 1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clus = c(107)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= -0.1*umap_1 -0.7), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 0.5, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = -0.1, intercept = -0.7) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(-4, 0) + ylim(-3, 1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1170,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 165\n",
      "\n",
      "# of cells sampled in set 2: 371\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Osi9</th><td>Osi9     </td><td> 0</td><td>59</td><td>  0.0000</td><td>430.8151</td><td>Set 2</td><td>8.750925</td><td>76.60796</td></tr>\n",
       "\t<tr><th scope=row>GlyP</th><td>GlyP     </td><td> 0</td><td>48</td><td>  0.0000</td><td>339.6609</td><td>Set 2</td><td>8.407951</td><td>70.72931</td></tr>\n",
       "\t<tr><th scope=row>Muc91C</th><td>Muc91C   </td><td> 0</td><td>49</td><td>  0.0000</td><td>329.5218</td><td>Set 2</td><td>8.364230</td><td>69.99691</td></tr>\n",
       "\t<tr><th scope=row>TwdlD</th><td>TwdlD    </td><td> 0</td><td>44</td><td>  0.0000</td><td>328.2649</td><td>Set 2</td><td>8.358716</td><td>69.90482</td></tr>\n",
       "\t<tr><th scope=row>Pgk</th><td>Pgk      </td><td> 0</td><td>50</td><td>  0.0000</td><td>311.7071</td><td>Set 2</td><td>8.284047</td><td>68.66372</td></tr>\n",
       "\t<tr><th scope=row>Vajk4</th><td>Vajk4    </td><td> 0</td><td>37</td><td>  0.0000</td><td>289.8846</td><td>Set 2</td><td>8.179335</td><td>66.94216</td></tr>\n",
       "\t<tr><th scope=row>CG12723</th><td>CG12723  </td><td> 0</td><td>38</td><td>  0.0000</td><td>277.4163</td><td>Set 2</td><td>8.115909</td><td>65.91010</td></tr>\n",
       "\t<tr><th scope=row>Prx2540-1</th><td>Prx2540-1</td><td> 0</td><td>37</td><td>  0.0000</td><td>263.1085</td><td>Set 2</td><td>8.039514</td><td>64.67779</td></tr>\n",
       "\t<tr><th scope=row>Pdss2</th><td>Pdss2    </td><td> 0</td><td>38</td><td>  0.0000</td><td>258.1015</td><td>Set 2</td><td>8.011795</td><td>64.23355</td></tr>\n",
       "\t<tr><th scope=row>CG31626</th><td>CG31626  </td><td> 0</td><td>34</td><td>  0.0000</td><td>256.8293</td><td>Set 2</td><td>8.004666</td><td>64.11955</td></tr>\n",
       "\t<tr><th scope=row>Pvf2</th><td>Pvf2     </td><td>13</td><td> 0</td><td>249.2623</td><td>  0.0000</td><td>Set 1</td><td>7.961521</td><td>63.43180</td></tr>\n",
       "\t<tr><th scope=row>TwdlW</th><td>TwdlW    </td><td> 0</td><td>34</td><td>  0.0000</td><td>248.9366</td><td>Set 2</td><td>7.959635</td><td>63.40182</td></tr>\n",
       "\t<tr><th scope=row>CG17574</th><td>CG17574  </td><td>11</td><td> 0</td><td>230.8214</td><td>  0.0000</td><td>Set 1</td><td>7.850633</td><td>61.68140</td></tr>\n",
       "\t<tr><th scope=row>Mtpbeta</th><td>Mtpbeta  </td><td> 0</td><td>34</td><td>  0.0000</td><td>224.7553</td><td>Set 2</td><td>7.812211</td><td>61.08068</td></tr>\n",
       "\t<tr><th scope=row>CG3777</th><td>CG3777   </td><td> 0</td><td>31</td><td>  0.0000</td><td>222.0532</td><td>Set 2</td><td>7.794761</td><td>60.80883</td></tr>\n",
       "\t<tr><th scope=row>Osi19</th><td>Osi19    </td><td> 0</td><td>30</td><td>  0.0000</td><td>211.4205</td><td>Set 2</td><td>7.723972</td><td>59.71232</td></tr>\n",
       "\t<tr><th scope=row>Rcc1</th><td>Rcc1     </td><td> 0</td><td>32</td><td>  0.0000</td><td>207.4916</td><td>Set 2</td><td>7.696909</td><td>59.29579</td></tr>\n",
       "\t<tr><th scope=row>Sms</th><td>Sms      </td><td> 0</td><td>29</td><td>  0.0000</td><td>204.8185</td><td>Set 2</td><td>7.678202</td><td>59.00874</td></tr>\n",
       "\t<tr><th scope=row>TwdlG</th><td>TwdlG    </td><td> 0</td><td>28</td><td>  0.0000</td><td>202.4118</td><td>Set 2</td><td>7.661149</td><td>58.74768</td></tr>\n",
       "\t<tr><th scope=row>Oat</th><td>Oat      </td><td> 0</td><td>30</td><td>  0.0000</td><td>201.1781</td><td>Set 2</td><td>7.652330</td><td>58.61289</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tOsi9 & Osi9      &  0 & 59 &   0.0000 & 430.8151 & Set 2 & 8.750925 & 76.60796\\\\\n",
       "\tGlyP & GlyP      &  0 & 48 &   0.0000 & 339.6609 & Set 2 & 8.407951 & 70.72931\\\\\n",
       "\tMuc91C & Muc91C    &  0 & 49 &   0.0000 & 329.5218 & Set 2 & 8.364230 & 69.99691\\\\\n",
       "\tTwdlD & TwdlD     &  0 & 44 &   0.0000 & 328.2649 & Set 2 & 8.358716 & 69.90482\\\\\n",
       "\tPgk & Pgk       &  0 & 50 &   0.0000 & 311.7071 & Set 2 & 8.284047 & 68.66372\\\\\n",
       "\tVajk4 & Vajk4     &  0 & 37 &   0.0000 & 289.8846 & Set 2 & 8.179335 & 66.94216\\\\\n",
       "\tCG12723 & CG12723   &  0 & 38 &   0.0000 & 277.4163 & Set 2 & 8.115909 & 65.91010\\\\\n",
       "\tPrx2540-1 & Prx2540-1 &  0 & 37 &   0.0000 & 263.1085 & Set 2 & 8.039514 & 64.67779\\\\\n",
       "\tPdss2 & Pdss2     &  0 & 38 &   0.0000 & 258.1015 & Set 2 & 8.011795 & 64.23355\\\\\n",
       "\tCG31626 & CG31626   &  0 & 34 &   0.0000 & 256.8293 & Set 2 & 8.004666 & 64.11955\\\\\n",
       "\tPvf2 & Pvf2      & 13 &  0 & 249.2623 &   0.0000 & Set 1 & 7.961521 & 63.43180\\\\\n",
       "\tTwdlW & TwdlW     &  0 & 34 &   0.0000 & 248.9366 & Set 2 & 7.959635 & 63.40182\\\\\n",
       "\tCG17574 & CG17574   & 11 &  0 & 230.8214 &   0.0000 & Set 1 & 7.850633 & 61.68140\\\\\n",
       "\tMtpbeta & Mtpbeta   &  0 & 34 &   0.0000 & 224.7553 & Set 2 & 7.812211 & 61.08068\\\\\n",
       "\tCG3777 & CG3777    &  0 & 31 &   0.0000 & 222.0532 & Set 2 & 7.794761 & 60.80883\\\\\n",
       "\tOsi19 & Osi19     &  0 & 30 &   0.0000 & 211.4205 & Set 2 & 7.723972 & 59.71232\\\\\n",
       "\tRcc1 & Rcc1      &  0 & 32 &   0.0000 & 207.4916 & Set 2 & 7.696909 & 59.29579\\\\\n",
       "\tSms & Sms       &  0 & 29 &   0.0000 & 204.8185 & Set 2 & 7.678202 & 59.00874\\\\\n",
       "\tTwdlG & TwdlG     &  0 & 28 &   0.0000 & 202.4118 & Set 2 & 7.661149 & 58.74768\\\\\n",
       "\tOat & Oat       &  0 & 30 &   0.0000 & 201.1781 & Set 2 & 7.652330 & 58.61289\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Osi9 | Osi9      |  0 | 59 |   0.0000 | 430.8151 | Set 2 | 8.750925 | 76.60796 |\n",
       "| GlyP | GlyP      |  0 | 48 |   0.0000 | 339.6609 | Set 2 | 8.407951 | 70.72931 |\n",
       "| Muc91C | Muc91C    |  0 | 49 |   0.0000 | 329.5218 | Set 2 | 8.364230 | 69.99691 |\n",
       "| TwdlD | TwdlD     |  0 | 44 |   0.0000 | 328.2649 | Set 2 | 8.358716 | 69.90482 |\n",
       "| Pgk | Pgk       |  0 | 50 |   0.0000 | 311.7071 | Set 2 | 8.284047 | 68.66372 |\n",
       "| Vajk4 | Vajk4     |  0 | 37 |   0.0000 | 289.8846 | Set 2 | 8.179335 | 66.94216 |\n",
       "| CG12723 | CG12723   |  0 | 38 |   0.0000 | 277.4163 | Set 2 | 8.115909 | 65.91010 |\n",
       "| Prx2540-1 | Prx2540-1 |  0 | 37 |   0.0000 | 263.1085 | Set 2 | 8.039514 | 64.67779 |\n",
       "| Pdss2 | Pdss2     |  0 | 38 |   0.0000 | 258.1015 | Set 2 | 8.011795 | 64.23355 |\n",
       "| CG31626 | CG31626   |  0 | 34 |   0.0000 | 256.8293 | Set 2 | 8.004666 | 64.11955 |\n",
       "| Pvf2 | Pvf2      | 13 |  0 | 249.2623 |   0.0000 | Set 1 | 7.961521 | 63.43180 |\n",
       "| TwdlW | TwdlW     |  0 | 34 |   0.0000 | 248.9366 | Set 2 | 7.959635 | 63.40182 |\n",
       "| CG17574 | CG17574   | 11 |  0 | 230.8214 |   0.0000 | Set 1 | 7.850633 | 61.68140 |\n",
       "| Mtpbeta | Mtpbeta   |  0 | 34 |   0.0000 | 224.7553 | Set 2 | 7.812211 | 61.08068 |\n",
       "| CG3777 | CG3777    |  0 | 31 |   0.0000 | 222.0532 | Set 2 | 7.794761 | 60.80883 |\n",
       "| Osi19 | Osi19     |  0 | 30 |   0.0000 | 211.4205 | Set 2 | 7.723972 | 59.71232 |\n",
       "| Rcc1 | Rcc1      |  0 | 32 |   0.0000 | 207.4916 | Set 2 | 7.696909 | 59.29579 |\n",
       "| Sms | Sms       |  0 | 29 |   0.0000 | 204.8185 | Set 2 | 7.678202 | 59.00874 |\n",
       "| TwdlG | TwdlG     |  0 | 28 |   0.0000 | 202.4118 | Set 2 | 7.661149 | 58.74768 |\n",
       "| Oat | Oat       |  0 | 30 |   0.0000 | 201.1781 | Set 2 | 7.652330 | 58.61289 |\n",
       "\n"
      ],
      "text/plain": [
       "          gene      set.1.n.umi set.2.n.umi set.1.tpm set.2.tpm higher.expr\n",
       "Osi9      Osi9       0          59            0.0000  430.8151  Set 2      \n",
       "GlyP      GlyP       0          48            0.0000  339.6609  Set 2      \n",
       "Muc91C    Muc91C     0          49            0.0000  329.5218  Set 2      \n",
       "TwdlD     TwdlD      0          44            0.0000  328.2649  Set 2      \n",
       "Pgk       Pgk        0          50            0.0000  311.7071  Set 2      \n",
       "Vajk4     Vajk4      0          37            0.0000  289.8846  Set 2      \n",
       "CG12723   CG12723    0          38            0.0000  277.4163  Set 2      \n",
       "Prx2540-1 Prx2540-1  0          37            0.0000  263.1085  Set 2      \n",
       "Pdss2     Pdss2      0          38            0.0000  258.1015  Set 2      \n",
       "CG31626   CG31626    0          34            0.0000  256.8293  Set 2      \n",
       "Pvf2      Pvf2      13           0          249.2623    0.0000  Set 1      \n",
       "TwdlW     TwdlW      0          34            0.0000  248.9366  Set 2      \n",
       "CG17574   CG17574   11           0          230.8214    0.0000  Set 1      \n",
       "Mtpbeta   Mtpbeta    0          34            0.0000  224.7553  Set 2      \n",
       "CG3777    CG3777     0          31            0.0000  222.0532  Set 2      \n",
       "Osi19     Osi19      0          30            0.0000  211.4205  Set 2      \n",
       "Rcc1      Rcc1       0          32            0.0000  207.4916  Set 2      \n",
       "Sms       Sms        0          29            0.0000  204.8185  Set 2      \n",
       "TwdlG     TwdlG      0          28            0.0000  202.4118  Set 2      \n",
       "Oat       Oat        0          30            0.0000  201.1781  Set 2      \n",
       "          log2.ratio heuristic.score\n",
       "Osi9      8.750925   76.60796       \n",
       "GlyP      8.407951   70.72931       \n",
       "Muc91C    8.364230   69.99691       \n",
       "TwdlD     8.358716   69.90482       \n",
       "Pgk       8.284047   68.66372       \n",
       "Vajk4     8.179335   66.94216       \n",
       "CG12723   8.115909   65.91010       \n",
       "Prx2540-1 8.039514   64.67779       \n",
       "Pdss2     8.011795   64.23355       \n",
       "CG31626   8.004666   64.11955       \n",
       "Pvf2      7.961521   63.43180       \n",
       "TwdlW     7.959635   63.40182       \n",
       "CG17574   7.850633   61.68140       \n",
       "Mtpbeta   7.812211   61.08068       \n",
       "CG3777    7.794761   60.80883       \n",
       "Osi19     7.723972   59.71232       \n",
       "Rcc1      7.696909   59.29579       \n",
       "Sms       7.678202   59.00874       \n",
       "TwdlG     7.661149   58.74768       \n",
       "Oat       7.652330   58.61289       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "two.set.differential.gene.test.3a(cds_sub, \n",
    "                                  df_sub300$cluster == 107 & \n",
    "                                  df_sub300$umap_2 >= -0.1*df_sub300$umap_1 -0.7 , \n",
    "                                 df_sub300$cluster == 166  \n",
    "                                )  %>% \n",
    "head(20)\n",
    "## these are all background genes so no real differences between this part of 107 and 166; call one cell type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Osi9\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Osi9\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Osi9\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"GlyP\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Pvf2\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1206,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Bolwig_organ\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Bolwig_organ\", df_sub300$cluster %in% af)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'af' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'af' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(af)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(113)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1204,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Arr16</th><td>af</td><td>Arr1    </td><td>24</td><td>  8</td><td>577.94757</td><td>0.22720383</td><td>8.879420</td><td>81.48900</td></tr>\n",
       "\t<tr><th scope=row>Gbeta76C6</th><td>af</td><td>Gbeta76C</td><td>21</td><td>  9</td><td>521.41372</td><td>0.29412247</td><td>8.654311</td><td>78.14019</td></tr>\n",
       "\t<tr><th scope=row>Rh66</th><td>af</td><td>Rh6     </td><td>20</td><td> 24</td><td>567.42528</td><td>0.62850855</td><td>8.444735</td><td>77.27631</td></tr>\n",
       "\t<tr><th scope=row>ninaB6</th><td>af</td><td>ninaB   </td><td>11</td><td> 12</td><td>357.48074</td><td>0.37504383</td><td>8.022244</td><td>68.07477</td></tr>\n",
       "\t<tr><th scope=row>trpl6</th><td>af</td><td>trpl    </td><td>11</td><td> 10</td><td>293.60030</td><td>0.60410093</td><td>7.515945</td><td>61.65040</td></tr>\n",
       "\t<tr><th scope=row>LysD6</th><td>af</td><td>LysD    </td><td> 2</td><td>  4</td><td> 83.76612</td><td>0.15613995</td><td>6.178979</td><td>39.57893</td></tr>\n",
       "\t<tr><th scope=row>CG134036</th><td>af</td><td>CG13403 </td><td> 3</td><td> 26</td><td>111.98208</td><td>1.38586363</td><td>5.552613</td><td>37.86854</td></tr>\n",
       "\t<tr><th scope=row>ninaA6</th><td>af</td><td>ninaA   </td><td> 3</td><td>  5</td><td> 76.31583</td><td>0.21632321</td><td>5.971384</td><td>37.45665</td></tr>\n",
       "\t<tr><th scope=row>elfless6</th><td>af</td><td>elfless </td><td> 2</td><td>  3</td><td> 71.30338</td><td>0.07666857</td><td>6.049324</td><td>37.36057</td></tr>\n",
       "\t<tr><th scope=row>Mst84Dc6</th><td>af</td><td>Mst84Dc </td><td> 3</td><td>  0</td><td> 61.37480</td><td>0.00000000</td><td>5.939574</td><td>35.41704</td></tr>\n",
       "\t<tr><th scope=row>CG304146</th><td>af</td><td>CG30414 </td><td> 5</td><td> 69</td><td>117.17397</td><td>2.76223639</td><td>4.960918</td><td>34.15477</td></tr>\n",
       "\t<tr><th scope=row>chp6</th><td>af</td><td>chp     </td><td> 3</td><td> 39</td><td> 99.95318</td><td>1.88965570</td><td>5.112283</td><td>34.03524</td></tr>\n",
       "\t<tr><th scope=row>CG149066</th><td>af</td><td>CG14906 </td><td> 3</td><td>118</td><td>125.49137</td><td>3.98278077</td><td>4.654493</td><td>32.50184</td></tr>\n",
       "\t<tr><th scope=row>CG177396</th><td>af</td><td>CG17739 </td><td> 2</td><td> 16</td><td> 64.33979</td><td>0.67756741</td><td>5.261269</td><td>31.72487</td></tr>\n",
       "\t<tr><th scope=row>RpL37b6</th><td>af</td><td>RpL37b  </td><td> 2</td><td>  0</td><td> 46.65920</td><td>0.00000000</td><td>5.544090</td><td>30.90654</td></tr>\n",
       "\t<tr><th scope=row>CG52456</th><td>af</td><td>CG5245  </td><td> 2</td><td> 30</td><td> 67.11860</td><td>1.00454009</td><td>5.065369</td><td>30.84798</td></tr>\n",
       "\t<tr><th scope=row>CG133766</th><td>af</td><td>CG13376 </td><td> 2</td><td> 29</td><td> 67.11860</td><td>1.10938940</td><td>4.991815</td><td>30.40004</td></tr>\n",
       "\t<tr><th scope=row>CG128686</th><td>af</td><td>CG12868 </td><td> 3</td><td> 51</td><td> 78.99726</td><td>1.83541436</td><td>4.800171</td><td>30.34610</td></tr>\n",
       "\t<tr><th scope=row>Mst36Fb6</th><td>af</td><td>Mst36Fb </td><td> 3</td><td> 48</td><td> 83.67547</td><td>2.22933229</td><td>4.695497</td><td>30.06936</td></tr>\n",
       "\t<tr><th scope=row>Cad88C6</th><td>af</td><td>Cad88C  </td><td> 4</td><td>136</td><td>130.85116</td><td>6.19091307</td><td>4.185608</td><td>29.47826</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tArr16 & af & Arr1     & 24 &   8 & 577.94757 & 0.22720383 & 8.879420 & 81.48900\\\\\n",
       "\tGbeta76C6 & af & Gbeta76C & 21 &   9 & 521.41372 & 0.29412247 & 8.654311 & 78.14019\\\\\n",
       "\tRh66 & af & Rh6      & 20 &  24 & 567.42528 & 0.62850855 & 8.444735 & 77.27631\\\\\n",
       "\tninaB6 & af & ninaB    & 11 &  12 & 357.48074 & 0.37504383 & 8.022244 & 68.07477\\\\\n",
       "\ttrpl6 & af & trpl     & 11 &  10 & 293.60030 & 0.60410093 & 7.515945 & 61.65040\\\\\n",
       "\tLysD6 & af & LysD     &  2 &   4 &  83.76612 & 0.15613995 & 6.178979 & 39.57893\\\\\n",
       "\tCG134036 & af & CG13403  &  3 &  26 & 111.98208 & 1.38586363 & 5.552613 & 37.86854\\\\\n",
       "\tninaA6 & af & ninaA    &  3 &   5 &  76.31583 & 0.21632321 & 5.971384 & 37.45665\\\\\n",
       "\telfless6 & af & elfless  &  2 &   3 &  71.30338 & 0.07666857 & 6.049324 & 37.36057\\\\\n",
       "\tMst84Dc6 & af & Mst84Dc  &  3 &   0 &  61.37480 & 0.00000000 & 5.939574 & 35.41704\\\\\n",
       "\tCG304146 & af & CG30414  &  5 &  69 & 117.17397 & 2.76223639 & 4.960918 & 34.15477\\\\\n",
       "\tchp6 & af & chp      &  3 &  39 &  99.95318 & 1.88965570 & 5.112283 & 34.03524\\\\\n",
       "\tCG149066 & af & CG14906  &  3 & 118 & 125.49137 & 3.98278077 & 4.654493 & 32.50184\\\\\n",
       "\tCG177396 & af & CG17739  &  2 &  16 &  64.33979 & 0.67756741 & 5.261269 & 31.72487\\\\\n",
       "\tRpL37b6 & af & RpL37b   &  2 &   0 &  46.65920 & 0.00000000 & 5.544090 & 30.90654\\\\\n",
       "\tCG52456 & af & CG5245   &  2 &  30 &  67.11860 & 1.00454009 & 5.065369 & 30.84798\\\\\n",
       "\tCG133766 & af & CG13376  &  2 &  29 &  67.11860 & 1.10938940 & 4.991815 & 30.40004\\\\\n",
       "\tCG128686 & af & CG12868  &  3 &  51 &  78.99726 & 1.83541436 & 4.800171 & 30.34610\\\\\n",
       "\tMst36Fb6 & af & Mst36Fb  &  3 &  48 &  83.67547 & 2.22933229 & 4.695497 & 30.06936\\\\\n",
       "\tCad88C6 & af & Cad88C   &  4 & 136 & 130.85116 & 6.19091307 & 4.185608 & 29.47826\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Arr16 | af | Arr1     | 24 |   8 | 577.94757 | 0.22720383 | 8.879420 | 81.48900 |\n",
       "| Gbeta76C6 | af | Gbeta76C | 21 |   9 | 521.41372 | 0.29412247 | 8.654311 | 78.14019 |\n",
       "| Rh66 | af | Rh6      | 20 |  24 | 567.42528 | 0.62850855 | 8.444735 | 77.27631 |\n",
       "| ninaB6 | af | ninaB    | 11 |  12 | 357.48074 | 0.37504383 | 8.022244 | 68.07477 |\n",
       "| trpl6 | af | trpl     | 11 |  10 | 293.60030 | 0.60410093 | 7.515945 | 61.65040 |\n",
       "| LysD6 | af | LysD     |  2 |   4 |  83.76612 | 0.15613995 | 6.178979 | 39.57893 |\n",
       "| CG134036 | af | CG13403  |  3 |  26 | 111.98208 | 1.38586363 | 5.552613 | 37.86854 |\n",
       "| ninaA6 | af | ninaA    |  3 |   5 |  76.31583 | 0.21632321 | 5.971384 | 37.45665 |\n",
       "| elfless6 | af | elfless  |  2 |   3 |  71.30338 | 0.07666857 | 6.049324 | 37.36057 |\n",
       "| Mst84Dc6 | af | Mst84Dc  |  3 |   0 |  61.37480 | 0.00000000 | 5.939574 | 35.41704 |\n",
       "| CG304146 | af | CG30414  |  5 |  69 | 117.17397 | 2.76223639 | 4.960918 | 34.15477 |\n",
       "| chp6 | af | chp      |  3 |  39 |  99.95318 | 1.88965570 | 5.112283 | 34.03524 |\n",
       "| CG149066 | af | CG14906  |  3 | 118 | 125.49137 | 3.98278077 | 4.654493 | 32.50184 |\n",
       "| CG177396 | af | CG17739  |  2 |  16 |  64.33979 | 0.67756741 | 5.261269 | 31.72487 |\n",
       "| RpL37b6 | af | RpL37b   |  2 |   0 |  46.65920 | 0.00000000 | 5.544090 | 30.90654 |\n",
       "| CG52456 | af | CG5245   |  2 |  30 |  67.11860 | 1.00454009 | 5.065369 | 30.84798 |\n",
       "| CG133766 | af | CG13376  |  2 |  29 |  67.11860 | 1.10938940 | 4.991815 | 30.40004 |\n",
       "| CG128686 | af | CG12868  |  3 |  51 |  78.99726 | 1.83541436 | 4.800171 | 30.34610 |\n",
       "| Mst36Fb6 | af | Mst36Fb  |  3 |  48 |  83.67547 | 2.22933229 | 4.695497 | 30.06936 |\n",
       "| Cad88C6 | af | Cad88C   |  4 | 136 | 130.85116 | 6.19091307 | 4.185608 | 29.47826 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene     cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "Arr16     af          Arr1     24                  8         577.94757      \n",
       "Gbeta76C6 af          Gbeta76C 21                  9         521.41372      \n",
       "Rh66      af          Rh6      20                 24         567.42528      \n",
       "ninaB6    af          ninaB    11                 12         357.48074      \n",
       "trpl6     af          trpl     11                 10         293.60030      \n",
       "LysD6     af          LysD      2                  4          83.76612      \n",
       "CG134036  af          CG13403   3                 26         111.98208      \n",
       "ninaA6    af          ninaA     3                  5          76.31583      \n",
       "elfless6  af          elfless   2                  3          71.30338      \n",
       "Mst84Dc6  af          Mst84Dc   3                  0          61.37480      \n",
       "CG304146  af          CG30414   5                 69         117.17397      \n",
       "chp6      af          chp       3                 39          99.95318      \n",
       "CG149066  af          CG14906   3                118         125.49137      \n",
       "CG177396  af          CG17739   2                 16          64.33979      \n",
       "RpL37b6   af          RpL37b    2                  0          46.65920      \n",
       "CG52456   af          CG5245    2                 30          67.11860      \n",
       "CG133766  af          CG13376   2                 29          67.11860      \n",
       "CG128686  af          CG12868   3                 51          78.99726      \n",
       "Mst36Fb6  af          Mst36Fb   3                 48          83.67547      \n",
       "Cad88C6   af          Cad88C    4                136         130.85116      \n",
       "          other.tpm  log2.ratio heuristic.score\n",
       "Arr16     0.22720383 8.879420   81.48900       \n",
       "Gbeta76C6 0.29412247 8.654311   78.14019       \n",
       "Rh66      0.62850855 8.444735   77.27631       \n",
       "ninaB6    0.37504383 8.022244   68.07477       \n",
       "trpl6     0.60410093 7.515945   61.65040       \n",
       "LysD6     0.15613995 6.178979   39.57893       \n",
       "CG134036  1.38586363 5.552613   37.86854       \n",
       "ninaA6    0.21632321 5.971384   37.45665       \n",
       "elfless6  0.07666857 6.049324   37.36057       \n",
       "Mst84Dc6  0.00000000 5.939574   35.41704       \n",
       "CG304146  2.76223639 4.960918   34.15477       \n",
       "chp6      1.88965570 5.112283   34.03524       \n",
       "CG149066  3.98278077 4.654493   32.50184       \n",
       "CG177396  0.67756741 5.261269   31.72487       \n",
       "RpL37b6   0.00000000 5.544090   30.90654       \n",
       "CG52456   1.00454009 5.065369   30.84798       \n",
       "CG133766  1.10938940 4.991815   30.40004       \n",
       "CG128686  1.83541436 4.800171   30.34610       \n",
       "Mst36Fb6  2.22933229 4.695497   30.06936       \n",
       "Cad88C6   6.19091307 4.185608   29.47826       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"af\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Arr1\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Arr1\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Arr1\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Gbeta76C\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Rh6\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1211,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type lateral_cord_glia\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"lateral_cord_glia\", df_sub300$cluster %in% ag)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ag' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ag' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ag)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(113)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1208,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>naz7</th><td>ag</td><td>naz      </td><td>323</td><td>  240</td><td>1938.9989</td><td>  8.535027</td><td>7.667859</td><td>83.74713</td></tr>\n",
       "\t<tr><th scope=row>CG404707</th><td>ag</td><td>CG40470  </td><td>432</td><td>  388</td><td>2515.9861</td><td> 14.455873</td><td>7.346825</td><td>83.00062</td></tr>\n",
       "\t<tr><th scope=row>Gat7</th><td>ag</td><td>Gat      </td><td>328</td><td>  471</td><td>1927.7464</td><td> 18.637423</td><td>6.617166</td><td>72.21609</td></tr>\n",
       "\t<tr><th scope=row>CG112417</th><td>ag</td><td>CG1124   </td><td>133</td><td>  312</td><td> 777.3491</td><td> 12.969334</td><td>5.798228</td><td>55.68776</td></tr>\n",
       "\t<tr><th scope=row>zyd7</th><td>ag</td><td>zyd      </td><td>308</td><td> 1156</td><td>1780.0444</td><td> 49.574808</td><td>5.137351</td><td>55.47572</td></tr>\n",
       "\t<tr><th scope=row>CG316637</th><td>ag</td><td>CG31663  </td><td>153</td><td>  396</td><td> 843.6520</td><td> 15.985689</td><td>5.634256</td><td>54.77744</td></tr>\n",
       "\t<tr><th scope=row>CG427417</th><td>ag</td><td>CG42741  </td><td>393</td><td> 1699</td><td>2248.2818</td><td> 75.792200</td><td>4.871719</td><td>54.24781</td></tr>\n",
       "\t<tr><th scope=row>Cyp4g157</th><td>ag</td><td>Cyp4g15  </td><td> 52</td><td>   69</td><td> 311.3520</td><td>  2.613807</td><td>6.428883</td><td>53.27634</td></tr>\n",
       "\t<tr><th scope=row>alrm7</th><td>ag</td><td>alrm     </td><td>181</td><td>  705</td><td>1030.1346</td><td> 25.257338</td><td>5.293968</td><td>52.99271</td></tr>\n",
       "\t<tr><th scope=row>CG44099</th><td>ag</td><td>CG4409   </td><td> 52</td><td>   81</td><td> 320.2987</td><td>  2.892233</td><td>6.362676</td><td>52.98691</td></tr>\n",
       "\t<tr><th scope=row>axo7</th><td>ag</td><td>axo      </td><td>246</td><td>  940</td><td>1411.2267</td><td> 42.055880</td><td>5.034596</td><td>52.68078</td></tr>\n",
       "\t<tr><th scope=row>CG57587</th><td>ag</td><td>CG5758   </td><td> 86</td><td>  218</td><td> 452.3581</td><td>  8.860033</td><td>5.519729</td><td>48.70889</td></tr>\n",
       "\t<tr><th scope=row>CG67237</th><td>ag</td><td>CG6723   </td><td> 26</td><td>   36</td><td> 162.4139</td><td>  1.337491</td><td>6.118570</td><td>44.98609</td></tr>\n",
       "\t<tr><th scope=row>Eaat17</th><td>ag</td><td>Eaat1    </td><td> 51</td><td>  168</td><td> 313.0830</td><td>  6.792326</td><td>5.328347</td><td>44.19865</td></tr>\n",
       "\t<tr><th scope=row>Octbeta2R7</th><td>ag</td><td>Octbeta2R</td><td>226</td><td> 1447</td><td>1223.8591</td><td> 65.067524</td><td>4.211352</td><td>43.20174</td></tr>\n",
       "\t<tr><th scope=row>NimA7</th><td>ag</td><td>NimA     </td><td> 29</td><td>   32</td><td> 149.2869</td><td>  1.543351</td><td>5.875214</td><td>42.48705</td></tr>\n",
       "\t<tr><th scope=row>CG426137</th><td>ag</td><td>CG42613  </td><td>252</td><td> 2355</td><td>1570.9590</td><td>103.839486</td><td>3.905391</td><td>41.46880</td></tr>\n",
       "\t<tr><th scope=row>Nep47</th><td>ag</td><td>Nep4     </td><td> 54</td><td>  294</td><td> 349.9479</td><td> 11.589991</td><td>4.796791</td><td>40.55741</td></tr>\n",
       "\t<tr><th scope=row>Pde97</th><td>ag</td><td>Pde9     </td><td>922</td><td>11363</td><td>5227.8258</td><td>536.106192</td><td>3.282932</td><td>40.55166</td></tr>\n",
       "\t<tr><th scope=row>Ace7</th><td>ag</td><td>Ace      </td><td>163</td><td> 1450</td><td> 967.7010</td><td> 58.186170</td><td>4.031229</td><td>39.98942</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tnaz7 & ag & naz       & 323 &   240 & 1938.9989 &   8.535027 & 7.667859 & 83.74713\\\\\n",
       "\tCG404707 & ag & CG40470   & 432 &   388 & 2515.9861 &  14.455873 & 7.346825 & 83.00062\\\\\n",
       "\tGat7 & ag & Gat       & 328 &   471 & 1927.7464 &  18.637423 & 6.617166 & 72.21609\\\\\n",
       "\tCG112417 & ag & CG1124    & 133 &   312 &  777.3491 &  12.969334 & 5.798228 & 55.68776\\\\\n",
       "\tzyd7 & ag & zyd       & 308 &  1156 & 1780.0444 &  49.574808 & 5.137351 & 55.47572\\\\\n",
       "\tCG316637 & ag & CG31663   & 153 &   396 &  843.6520 &  15.985689 & 5.634256 & 54.77744\\\\\n",
       "\tCG427417 & ag & CG42741   & 393 &  1699 & 2248.2818 &  75.792200 & 4.871719 & 54.24781\\\\\n",
       "\tCyp4g157 & ag & Cyp4g15   &  52 &    69 &  311.3520 &   2.613807 & 6.428883 & 53.27634\\\\\n",
       "\talrm7 & ag & alrm      & 181 &   705 & 1030.1346 &  25.257338 & 5.293968 & 52.99271\\\\\n",
       "\tCG44099 & ag & CG4409    &  52 &    81 &  320.2987 &   2.892233 & 6.362676 & 52.98691\\\\\n",
       "\taxo7 & ag & axo       & 246 &   940 & 1411.2267 &  42.055880 & 5.034596 & 52.68078\\\\\n",
       "\tCG57587 & ag & CG5758    &  86 &   218 &  452.3581 &   8.860033 & 5.519729 & 48.70889\\\\\n",
       "\tCG67237 & ag & CG6723    &  26 &    36 &  162.4139 &   1.337491 & 6.118570 & 44.98609\\\\\n",
       "\tEaat17 & ag & Eaat1     &  51 &   168 &  313.0830 &   6.792326 & 5.328347 & 44.19865\\\\\n",
       "\tOctbeta2R7 & ag & Octbeta2R & 226 &  1447 & 1223.8591 &  65.067524 & 4.211352 & 43.20174\\\\\n",
       "\tNimA7 & ag & NimA      &  29 &    32 &  149.2869 &   1.543351 & 5.875214 & 42.48705\\\\\n",
       "\tCG426137 & ag & CG42613   & 252 &  2355 & 1570.9590 & 103.839486 & 3.905391 & 41.46880\\\\\n",
       "\tNep47 & ag & Nep4      &  54 &   294 &  349.9479 &  11.589991 & 4.796791 & 40.55741\\\\\n",
       "\tPde97 & ag & Pde9      & 922 & 11363 & 5227.8258 & 536.106192 & 3.282932 & 40.55166\\\\\n",
       "\tAce7 & ag & Ace       & 163 &  1450 &  967.7010 &  58.186170 & 4.031229 & 39.98942\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| naz7 | ag | naz       | 323 |   240 | 1938.9989 |   8.535027 | 7.667859 | 83.74713 |\n",
       "| CG404707 | ag | CG40470   | 432 |   388 | 2515.9861 |  14.455873 | 7.346825 | 83.00062 |\n",
       "| Gat7 | ag | Gat       | 328 |   471 | 1927.7464 |  18.637423 | 6.617166 | 72.21609 |\n",
       "| CG112417 | ag | CG1124    | 133 |   312 |  777.3491 |  12.969334 | 5.798228 | 55.68776 |\n",
       "| zyd7 | ag | zyd       | 308 |  1156 | 1780.0444 |  49.574808 | 5.137351 | 55.47572 |\n",
       "| CG316637 | ag | CG31663   | 153 |   396 |  843.6520 |  15.985689 | 5.634256 | 54.77744 |\n",
       "| CG427417 | ag | CG42741   | 393 |  1699 | 2248.2818 |  75.792200 | 4.871719 | 54.24781 |\n",
       "| Cyp4g157 | ag | Cyp4g15   |  52 |    69 |  311.3520 |   2.613807 | 6.428883 | 53.27634 |\n",
       "| alrm7 | ag | alrm      | 181 |   705 | 1030.1346 |  25.257338 | 5.293968 | 52.99271 |\n",
       "| CG44099 | ag | CG4409    |  52 |    81 |  320.2987 |   2.892233 | 6.362676 | 52.98691 |\n",
       "| axo7 | ag | axo       | 246 |   940 | 1411.2267 |  42.055880 | 5.034596 | 52.68078 |\n",
       "| CG57587 | ag | CG5758    |  86 |   218 |  452.3581 |   8.860033 | 5.519729 | 48.70889 |\n",
       "| CG67237 | ag | CG6723    |  26 |    36 |  162.4139 |   1.337491 | 6.118570 | 44.98609 |\n",
       "| Eaat17 | ag | Eaat1     |  51 |   168 |  313.0830 |   6.792326 | 5.328347 | 44.19865 |\n",
       "| Octbeta2R7 | ag | Octbeta2R | 226 |  1447 | 1223.8591 |  65.067524 | 4.211352 | 43.20174 |\n",
       "| NimA7 | ag | NimA      |  29 |    32 |  149.2869 |   1.543351 | 5.875214 | 42.48705 |\n",
       "| CG426137 | ag | CG42613   | 252 |  2355 | 1570.9590 | 103.839486 | 3.905391 | 41.46880 |\n",
       "| Nep47 | ag | Nep4      |  54 |   294 |  349.9479 |  11.589991 | 4.796791 | 40.55741 |\n",
       "| Pde97 | ag | Pde9      | 922 | 11363 | 5227.8258 | 536.106192 | 3.282932 | 40.55166 |\n",
       "| Ace7 | ag | Ace       | 163 |  1450 |  967.7010 |  58.186170 | 4.031229 | 39.98942 |\n",
       "\n"
      ],
      "text/plain": [
       "           cell.type.2 gene      cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "naz7       ag          naz       323                 240       1938.9989      \n",
       "CG404707   ag          CG40470   432                 388       2515.9861      \n",
       "Gat7       ag          Gat       328                 471       1927.7464      \n",
       "CG112417   ag          CG1124    133                 312        777.3491      \n",
       "zyd7       ag          zyd       308                1156       1780.0444      \n",
       "CG316637   ag          CG31663   153                 396        843.6520      \n",
       "CG427417   ag          CG42741   393                1699       2248.2818      \n",
       "Cyp4g157   ag          Cyp4g15    52                  69        311.3520      \n",
       "alrm7      ag          alrm      181                 705       1030.1346      \n",
       "CG44099    ag          CG4409     52                  81        320.2987      \n",
       "axo7       ag          axo       246                 940       1411.2267      \n",
       "CG57587    ag          CG5758     86                 218        452.3581      \n",
       "CG67237    ag          CG6723     26                  36        162.4139      \n",
       "Eaat17     ag          Eaat1      51                 168        313.0830      \n",
       "Octbeta2R7 ag          Octbeta2R 226                1447       1223.8591      \n",
       "NimA7      ag          NimA       29                  32        149.2869      \n",
       "CG426137   ag          CG42613   252                2355       1570.9590      \n",
       "Nep47      ag          Nep4       54                 294        349.9479      \n",
       "Pde97      ag          Pde9      922               11363       5227.8258      \n",
       "Ace7       ag          Ace       163                1450        967.7010      \n",
       "           other.tpm  log2.ratio heuristic.score\n",
       "naz7         8.535027 7.667859   83.74713       \n",
       "CG404707    14.455873 7.346825   83.00062       \n",
       "Gat7        18.637423 6.617166   72.21609       \n",
       "CG112417    12.969334 5.798228   55.68776       \n",
       "zyd7        49.574808 5.137351   55.47572       \n",
       "CG316637    15.985689 5.634256   54.77744       \n",
       "CG427417    75.792200 4.871719   54.24781       \n",
       "Cyp4g157     2.613807 6.428883   53.27634       \n",
       "alrm7       25.257338 5.293968   52.99271       \n",
       "CG44099      2.892233 6.362676   52.98691       \n",
       "axo7        42.055880 5.034596   52.68078       \n",
       "CG57587      8.860033 5.519729   48.70889       \n",
       "CG67237      1.337491 6.118570   44.98609       \n",
       "Eaat17       6.792326 5.328347   44.19865       \n",
       "Octbeta2R7  65.067524 4.211352   43.20174       \n",
       "NimA7        1.543351 5.875214   42.48705       \n",
       "CG426137   103.839486 3.905391   41.46880       \n",
       "Nep47       11.589991 4.796791   40.55741       \n",
       "Pde97      536.106192 3.282932   40.55166       \n",
       "Ace7        58.186170 4.031229   39.98942       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"ag\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"naz\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"naz\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"naz\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG40470\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Gat\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1215,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Somatic_muscle\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Somatic_muscle\", df_sub300$cluster %in% ah)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ah' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ah' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ah)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(113)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1213,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>bt8</th><td>ah</td><td>bt       </td><td>62060</td><td>32330</td><td>35937.1751</td><td>1342.014822</td><td>4.741930</td><td>71.76071</td></tr>\n",
       "\t<tr><th scope=row>Mp208</th><td>ah</td><td>Mp20     </td><td> 7028</td><td> 1021</td><td> 2651.2943</td><td>  36.565323</td><td>6.141152</td><td>69.84347</td></tr>\n",
       "\t<tr><th scope=row>Zasp668</th><td>ah</td><td>Zasp66   </td><td> 3442</td><td>  434</td><td> 1621.1014</td><td>  16.385339</td><td>6.542959</td><td>69.77182</td></tr>\n",
       "\t<tr><th scope=row>Mf8</th><td>ah</td><td>Mf       </td><td>14532</td><td> 3325</td><td> 6044.8988</td><td> 128.154749</td><td>5.548546</td><td>69.69939</td></tr>\n",
       "\t<tr><th scope=row>CG450788</th><td>ah</td><td>CG45078  </td><td> 3351</td><td>  404</td><td> 1392.4439</td><td>  15.018880</td><td>6.441702</td><td>67.27997</td></tr>\n",
       "\t<tr><th scope=row>sls8</th><td>ah</td><td>sls      </td><td>35880</td><td>25607</td><td>21292.0738</td><td> 925.880292</td><td>4.521790</td><td>65.01473</td></tr>\n",
       "\t<tr><th scope=row>CG50238</th><td>ah</td><td>CG5023   </td><td> 1101</td><td>   68</td><td>  456.0565</td><td>   2.330025</td><td>7.097536</td><td>62.71545</td></tr>\n",
       "\t<tr><th scope=row>CG438978</th><td>ah</td><td>CG43897  </td><td> 9334</td><td> 4324</td><td> 5307.7064</td><td> 163.281674</td><td>5.013845</td><td>62.04204</td></tr>\n",
       "\t<tr><th scope=row>CG92978</th><td>ah</td><td>CG9297   </td><td> 2277</td><td>  359</td><td> 1068.6102</td><td>  14.595883</td><td>6.098427</td><td>61.36767</td></tr>\n",
       "\t<tr><th scope=row>Unc-898</th><td>ah</td><td>Unc-89   </td><td> 2151</td><td>  373</td><td> 1044.6286</td><td>  14.264607</td><td>6.096656</td><td>61.15040</td></tr>\n",
       "\t<tr><th scope=row>Mhc8</th><td>ah</td><td>Mhc      </td><td>48630</td><td>40391</td><td>22215.1392</td><td>1610.846218</td><td>3.784757</td><td>54.64932</td></tr>\n",
       "\t<tr><th scope=row>CG450768</th><td>ah</td><td>CG45076  </td><td> 2002</td><td>  599</td><td> 1068.7123</td><td>  24.447911</td><td>5.392182</td><td>54.26157</td></tr>\n",
       "\t<tr><th scope=row>Mpcp18</th><td>ah</td><td>Mpcp1    </td><td>  873</td><td>   98</td><td>  351.8164</td><td>   3.141948</td><td>6.408369</td><td>54.23258</td></tr>\n",
       "\t<tr><th scope=row>Gapdh18</th><td>ah</td><td>Gapdh1   </td><td> 6509</td><td> 2354</td><td> 2232.4084</td><td>  75.664667</td><td>4.863895</td><td>54.11099</td></tr>\n",
       "\t<tr><th scope=row>Prm8</th><td>ah</td><td>Prm      </td><td>14851</td><td>10397</td><td> 7003.7905</td><td> 386.759047</td><td>4.174904</td><td>53.33075</td></tr>\n",
       "\t<tr><th scope=row>l(2)012898</th><td>ah</td><td>l(2)01289</td><td> 4687</td><td> 2346</td><td> 2453.6054</td><td> 103.775342</td><td>4.549532</td><td>51.23353</td></tr>\n",
       "\t<tr><th scope=row>Mlc28</th><td>ah</td><td>Mlc2     </td><td>27893</td><td>24240</td><td>10865.7972</td><td> 817.458471</td><td>3.730741</td><td>50.02043</td></tr>\n",
       "\t<tr><th scope=row>Mlc18</th><td>ah</td><td>Mlc1     </td><td> 9821</td><td> 5599</td><td> 3756.8414</td><td> 209.312138</td><td>4.158916</td><td>49.38999</td></tr>\n",
       "\t<tr><th scope=row>Act57B8</th><td>ah</td><td>Act57B   </td><td>42031</td><td>44427</td><td>16747.9129</td><td>1460.985409</td><td>3.517980</td><td>49.36353</td></tr>\n",
       "\t<tr><th scope=row>CG88888</th><td>ah</td><td>CG8888   </td><td> 3003</td><td> 2170</td><td> 1956.8788</td><td>  90.314663</td><td>4.421564</td><td>48.35014</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tbt8 & ah & bt        & 62060 & 32330 & 35937.1751 & 1342.014822 & 4.741930 & 71.76071\\\\\n",
       "\tMp208 & ah & Mp20      &  7028 &  1021 &  2651.2943 &   36.565323 & 6.141152 & 69.84347\\\\\n",
       "\tZasp668 & ah & Zasp66    &  3442 &   434 &  1621.1014 &   16.385339 & 6.542959 & 69.77182\\\\\n",
       "\tMf8 & ah & Mf        & 14532 &  3325 &  6044.8988 &  128.154749 & 5.548546 & 69.69939\\\\\n",
       "\tCG450788 & ah & CG45078   &  3351 &   404 &  1392.4439 &   15.018880 & 6.441702 & 67.27997\\\\\n",
       "\tsls8 & ah & sls       & 35880 & 25607 & 21292.0738 &  925.880292 & 4.521790 & 65.01473\\\\\n",
       "\tCG50238 & ah & CG5023    &  1101 &    68 &   456.0565 &    2.330025 & 7.097536 & 62.71545\\\\\n",
       "\tCG438978 & ah & CG43897   &  9334 &  4324 &  5307.7064 &  163.281674 & 5.013845 & 62.04204\\\\\n",
       "\tCG92978 & ah & CG9297    &  2277 &   359 &  1068.6102 &   14.595883 & 6.098427 & 61.36767\\\\\n",
       "\tUnc-898 & ah & Unc-89    &  2151 &   373 &  1044.6286 &   14.264607 & 6.096656 & 61.15040\\\\\n",
       "\tMhc8 & ah & Mhc       & 48630 & 40391 & 22215.1392 & 1610.846218 & 3.784757 & 54.64932\\\\\n",
       "\tCG450768 & ah & CG45076   &  2002 &   599 &  1068.7123 &   24.447911 & 5.392182 & 54.26157\\\\\n",
       "\tMpcp18 & ah & Mpcp1     &   873 &    98 &   351.8164 &    3.141948 & 6.408369 & 54.23258\\\\\n",
       "\tGapdh18 & ah & Gapdh1    &  6509 &  2354 &  2232.4084 &   75.664667 & 4.863895 & 54.11099\\\\\n",
       "\tPrm8 & ah & Prm       & 14851 & 10397 &  7003.7905 &  386.759047 & 4.174904 & 53.33075\\\\\n",
       "\tl(2)012898 & ah & l(2)01289 &  4687 &  2346 &  2453.6054 &  103.775342 & 4.549532 & 51.23353\\\\\n",
       "\tMlc28 & ah & Mlc2      & 27893 & 24240 & 10865.7972 &  817.458471 & 3.730741 & 50.02043\\\\\n",
       "\tMlc18 & ah & Mlc1      &  9821 &  5599 &  3756.8414 &  209.312138 & 4.158916 & 49.38999\\\\\n",
       "\tAct57B8 & ah & Act57B    & 42031 & 44427 & 16747.9129 & 1460.985409 & 3.517980 & 49.36353\\\\\n",
       "\tCG88888 & ah & CG8888    &  3003 &  2170 &  1956.8788 &   90.314663 & 4.421564 & 48.35014\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| bt8 | ah | bt        | 62060 | 32330 | 35937.1751 | 1342.014822 | 4.741930 | 71.76071 |\n",
       "| Mp208 | ah | Mp20      |  7028 |  1021 |  2651.2943 |   36.565323 | 6.141152 | 69.84347 |\n",
       "| Zasp668 | ah | Zasp66    |  3442 |   434 |  1621.1014 |   16.385339 | 6.542959 | 69.77182 |\n",
       "| Mf8 | ah | Mf        | 14532 |  3325 |  6044.8988 |  128.154749 | 5.548546 | 69.69939 |\n",
       "| CG450788 | ah | CG45078   |  3351 |   404 |  1392.4439 |   15.018880 | 6.441702 | 67.27997 |\n",
       "| sls8 | ah | sls       | 35880 | 25607 | 21292.0738 |  925.880292 | 4.521790 | 65.01473 |\n",
       "| CG50238 | ah | CG5023    |  1101 |    68 |   456.0565 |    2.330025 | 7.097536 | 62.71545 |\n",
       "| CG438978 | ah | CG43897   |  9334 |  4324 |  5307.7064 |  163.281674 | 5.013845 | 62.04204 |\n",
       "| CG92978 | ah | CG9297    |  2277 |   359 |  1068.6102 |   14.595883 | 6.098427 | 61.36767 |\n",
       "| Unc-898 | ah | Unc-89    |  2151 |   373 |  1044.6286 |   14.264607 | 6.096656 | 61.15040 |\n",
       "| Mhc8 | ah | Mhc       | 48630 | 40391 | 22215.1392 | 1610.846218 | 3.784757 | 54.64932 |\n",
       "| CG450768 | ah | CG45076   |  2002 |   599 |  1068.7123 |   24.447911 | 5.392182 | 54.26157 |\n",
       "| Mpcp18 | ah | Mpcp1     |   873 |    98 |   351.8164 |    3.141948 | 6.408369 | 54.23258 |\n",
       "| Gapdh18 | ah | Gapdh1    |  6509 |  2354 |  2232.4084 |   75.664667 | 4.863895 | 54.11099 |\n",
       "| Prm8 | ah | Prm       | 14851 | 10397 |  7003.7905 |  386.759047 | 4.174904 | 53.33075 |\n",
       "| l(2)012898 | ah | l(2)01289 |  4687 |  2346 |  2453.6054 |  103.775342 | 4.549532 | 51.23353 |\n",
       "| Mlc28 | ah | Mlc2      | 27893 | 24240 | 10865.7972 |  817.458471 | 3.730741 | 50.02043 |\n",
       "| Mlc18 | ah | Mlc1      |  9821 |  5599 |  3756.8414 |  209.312138 | 4.158916 | 49.38999 |\n",
       "| Act57B8 | ah | Act57B    | 42031 | 44427 | 16747.9129 | 1460.985409 | 3.517980 | 49.36353 |\n",
       "| CG88888 | ah | CG8888    |  3003 |  2170 |  1956.8788 |   90.314663 | 4.421564 | 48.35014 |\n",
       "\n"
      ],
      "text/plain": [
       "           cell.type.2 gene      cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "bt8        ah          bt        62060             32330       35937.1751     \n",
       "Mp208      ah          Mp20       7028              1021        2651.2943     \n",
       "Zasp668    ah          Zasp66     3442               434        1621.1014     \n",
       "Mf8        ah          Mf        14532              3325        6044.8988     \n",
       "CG450788   ah          CG45078    3351               404        1392.4439     \n",
       "sls8       ah          sls       35880             25607       21292.0738     \n",
       "CG50238    ah          CG5023     1101                68         456.0565     \n",
       "CG438978   ah          CG43897    9334              4324        5307.7064     \n",
       "CG92978    ah          CG9297     2277               359        1068.6102     \n",
       "Unc-898    ah          Unc-89     2151               373        1044.6286     \n",
       "Mhc8       ah          Mhc       48630             40391       22215.1392     \n",
       "CG450768   ah          CG45076    2002               599        1068.7123     \n",
       "Mpcp18     ah          Mpcp1       873                98         351.8164     \n",
       "Gapdh18    ah          Gapdh1     6509              2354        2232.4084     \n",
       "Prm8       ah          Prm       14851             10397        7003.7905     \n",
       "l(2)012898 ah          l(2)01289  4687              2346        2453.6054     \n",
       "Mlc28      ah          Mlc2      27893             24240       10865.7972     \n",
       "Mlc18      ah          Mlc1       9821              5599        3756.8414     \n",
       "Act57B8    ah          Act57B    42031             44427       16747.9129     \n",
       "CG88888    ah          CG8888     3003              2170        1956.8788     \n",
       "           other.tpm   log2.ratio heuristic.score\n",
       "bt8        1342.014822 4.741930   71.76071       \n",
       "Mp208        36.565323 6.141152   69.84347       \n",
       "Zasp668      16.385339 6.542959   69.77182       \n",
       "Mf8         128.154749 5.548546   69.69939       \n",
       "CG450788     15.018880 6.441702   67.27997       \n",
       "sls8        925.880292 4.521790   65.01473       \n",
       "CG50238       2.330025 7.097536   62.71545       \n",
       "CG438978    163.281674 5.013845   62.04204       \n",
       "CG92978      14.595883 6.098427   61.36767       \n",
       "Unc-898      14.264607 6.096656   61.15040       \n",
       "Mhc8       1610.846218 3.784757   54.64932       \n",
       "CG450768     24.447911 5.392182   54.26157       \n",
       "Mpcp18        3.141948 6.408369   54.23258       \n",
       "Gapdh18      75.664667 4.863895   54.11099       \n",
       "Prm8        386.759047 4.174904   53.33075       \n",
       "l(2)012898  103.775342 4.549532   51.23353       \n",
       "Mlc28       817.458471 3.730741   50.02043       \n",
       "Mlc18       209.312138 4.158916   49.38999       \n",
       "Act57B8    1460.985409 3.517980   49.36353       \n",
       "CG88888      90.314663 4.421564   48.35014       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"ah\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"bt\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"bt\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"bt\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Mp20\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Mf\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1222,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type large_intestine_hindgut_analpad_2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"large_intestine_hindgut_analpad_2\", df_sub300$cluster %in% ai)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ai' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ai' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ai)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(113)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1217,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG177819</th><td>ai</td><td>CG17781</td><td> 662</td><td>  172</td><td>1549.72237</td><td>  7.1383187</td><td>7.573063</td><td>80.26481</td></tr>\n",
       "\t<tr><th scope=row>CG330039</th><td>ai</td><td>CG33003</td><td> 962</td><td>  649</td><td>2191.35950</td><td> 28.1453376</td><td>6.232417</td><td>69.16904</td></tr>\n",
       "\t<tr><th scope=row>CG130829</th><td>ai</td><td>CG13082</td><td>1653</td><td> 2094</td><td>3855.55731</td><td> 89.6252499</td><td>5.410883</td><td>64.46037</td></tr>\n",
       "\t<tr><th scope=row>CG177809</th><td>ai</td><td>CG17780</td><td> 409</td><td>  330</td><td> 952.78345</td><td> 14.7350480</td><td>5.920095</td><td>58.59425</td></tr>\n",
       "\t<tr><th scope=row>CG136169</th><td>ai</td><td>CG13616</td><td> 170</td><td>   83</td><td> 398.11669</td><td>  2.9610686</td><td>6.651158</td><td>57.47044</td></tr>\n",
       "\t<tr><th scope=row>CG67859</th><td>ai</td><td>CG6785 </td><td> 449</td><td>  451</td><td> 997.81458</td><td> 19.9419961</td><td>5.574301</td><td>55.54274</td></tr>\n",
       "\t<tr><th scope=row>CG147209</th><td>ai</td><td>CG14720</td><td> 137</td><td>   62</td><td> 310.64695</td><td>  2.5760936</td><td>6.440748</td><td>53.35366</td></tr>\n",
       "\t<tr><th scope=row>ine9</th><td>ai</td><td>ine    </td><td> 606</td><td> 1002</td><td>1350.15923</td><td> 41.7728514</td><td>4.980290</td><td>51.79493</td></tr>\n",
       "\t<tr><th scope=row>otp9</th><td>ai</td><td>otp    </td><td>1875</td><td> 4849</td><td>4288.71618</td><td>222.8642733</td><td>4.259850</td><td>51.40218</td></tr>\n",
       "\t<tr><th scope=row>CG77209</th><td>ai</td><td>CG7720 </td><td>1625</td><td> 4546</td><td>3631.73769</td><td>209.9621473</td><td>4.105604</td><td>48.55633</td></tr>\n",
       "\t<tr><th scope=row>Ac78C9</th><td>ai</td><td>Ac78C  </td><td> 288</td><td>  477</td><td> 654.90987</td><td> 20.4519237</td><td>4.932117</td><td>46.15157</td></tr>\n",
       "\t<tr><th scope=row>CG136149</th><td>ai</td><td>CG13614</td><td>  62</td><td>   25</td><td> 154.19308</td><td>  0.9322095</td><td>6.318343</td><td>45.98439</td></tr>\n",
       "\t<tr><th scope=row>CG133139</th><td>ai</td><td>CG13313</td><td> 134</td><td>  149</td><td> 313.65944</td><td>  6.1276992</td><td>5.459619</td><td>45.30199</td></tr>\n",
       "\t<tr><th scope=row>Sox21b9</th><td>ai</td><td>Sox21b </td><td> 509</td><td> 1529</td><td>1180.19641</td><td> 60.3829509</td><td>4.265045</td><td>43.52919</td></tr>\n",
       "\t<tr><th scope=row>dysf9</th><td>ai</td><td>dysf   </td><td> 346</td><td>  843</td><td> 802.80386</td><td> 37.5375673</td><td>4.380710</td><td>42.27692</td></tr>\n",
       "\t<tr><th scope=row>CG148269</th><td>ai</td><td>CG14826</td><td>  55</td><td>   46</td><td> 135.35528</td><td>  1.6502044</td><td>5.674504</td><td>40.23919</td></tr>\n",
       "\t<tr><th scope=row>CG344439</th><td>ai</td><td>CG34443</td><td>  42</td><td>   15</td><td> 102.86477</td><td>  0.6741551</td><td>5.941172</td><td>39.79731</td></tr>\n",
       "\t<tr><th scope=row>CG47029</th><td>ai</td><td>CG4702 </td><td>2386</td><td>13973</td><td>5371.37356</td><td>615.2078650</td><td>3.123802</td><td>38.70811</td></tr>\n",
       "\t<tr><th scope=row>CG340319</th><td>ai</td><td>CG34031</td><td>  54</td><td>   57</td><td> 122.45543</td><td>  2.4692483</td><td>5.141490</td><td>35.72228</td></tr>\n",
       "\t<tr><th scope=row>CG148299</th><td>ai</td><td>CG14829</td><td>  36</td><td>   23</td><td>  82.41555</td><td>  0.7297718</td><td>5.574263</td><td>35.57631</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG177819 & ai & CG17781 &  662 &   172 & 1549.72237 &   7.1383187 & 7.573063 & 80.26481\\\\\n",
       "\tCG330039 & ai & CG33003 &  962 &   649 & 2191.35950 &  28.1453376 & 6.232417 & 69.16904\\\\\n",
       "\tCG130829 & ai & CG13082 & 1653 &  2094 & 3855.55731 &  89.6252499 & 5.410883 & 64.46037\\\\\n",
       "\tCG177809 & ai & CG17780 &  409 &   330 &  952.78345 &  14.7350480 & 5.920095 & 58.59425\\\\\n",
       "\tCG136169 & ai & CG13616 &  170 &    83 &  398.11669 &   2.9610686 & 6.651158 & 57.47044\\\\\n",
       "\tCG67859 & ai & CG6785  &  449 &   451 &  997.81458 &  19.9419961 & 5.574301 & 55.54274\\\\\n",
       "\tCG147209 & ai & CG14720 &  137 &    62 &  310.64695 &   2.5760936 & 6.440748 & 53.35366\\\\\n",
       "\tine9 & ai & ine     &  606 &  1002 & 1350.15923 &  41.7728514 & 4.980290 & 51.79493\\\\\n",
       "\totp9 & ai & otp     & 1875 &  4849 & 4288.71618 & 222.8642733 & 4.259850 & 51.40218\\\\\n",
       "\tCG77209 & ai & CG7720  & 1625 &  4546 & 3631.73769 & 209.9621473 & 4.105604 & 48.55633\\\\\n",
       "\tAc78C9 & ai & Ac78C   &  288 &   477 &  654.90987 &  20.4519237 & 4.932117 & 46.15157\\\\\n",
       "\tCG136149 & ai & CG13614 &   62 &    25 &  154.19308 &   0.9322095 & 6.318343 & 45.98439\\\\\n",
       "\tCG133139 & ai & CG13313 &  134 &   149 &  313.65944 &   6.1276992 & 5.459619 & 45.30199\\\\\n",
       "\tSox21b9 & ai & Sox21b  &  509 &  1529 & 1180.19641 &  60.3829509 & 4.265045 & 43.52919\\\\\n",
       "\tdysf9 & ai & dysf    &  346 &   843 &  802.80386 &  37.5375673 & 4.380710 & 42.27692\\\\\n",
       "\tCG148269 & ai & CG14826 &   55 &    46 &  135.35528 &   1.6502044 & 5.674504 & 40.23919\\\\\n",
       "\tCG344439 & ai & CG34443 &   42 &    15 &  102.86477 &   0.6741551 & 5.941172 & 39.79731\\\\\n",
       "\tCG47029 & ai & CG4702  & 2386 & 13973 & 5371.37356 & 615.2078650 & 3.123802 & 38.70811\\\\\n",
       "\tCG340319 & ai & CG34031 &   54 &    57 &  122.45543 &   2.4692483 & 5.141490 & 35.72228\\\\\n",
       "\tCG148299 & ai & CG14829 &   36 &    23 &   82.41555 &   0.7297718 & 5.574263 & 35.57631\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG177819 | ai | CG17781 |  662 |   172 | 1549.72237 |   7.1383187 | 7.573063 | 80.26481 |\n",
       "| CG330039 | ai | CG33003 |  962 |   649 | 2191.35950 |  28.1453376 | 6.232417 | 69.16904 |\n",
       "| CG130829 | ai | CG13082 | 1653 |  2094 | 3855.55731 |  89.6252499 | 5.410883 | 64.46037 |\n",
       "| CG177809 | ai | CG17780 |  409 |   330 |  952.78345 |  14.7350480 | 5.920095 | 58.59425 |\n",
       "| CG136169 | ai | CG13616 |  170 |    83 |  398.11669 |   2.9610686 | 6.651158 | 57.47044 |\n",
       "| CG67859 | ai | CG6785  |  449 |   451 |  997.81458 |  19.9419961 | 5.574301 | 55.54274 |\n",
       "| CG147209 | ai | CG14720 |  137 |    62 |  310.64695 |   2.5760936 | 6.440748 | 53.35366 |\n",
       "| ine9 | ai | ine     |  606 |  1002 | 1350.15923 |  41.7728514 | 4.980290 | 51.79493 |\n",
       "| otp9 | ai | otp     | 1875 |  4849 | 4288.71618 | 222.8642733 | 4.259850 | 51.40218 |\n",
       "| CG77209 | ai | CG7720  | 1625 |  4546 | 3631.73769 | 209.9621473 | 4.105604 | 48.55633 |\n",
       "| Ac78C9 | ai | Ac78C   |  288 |   477 |  654.90987 |  20.4519237 | 4.932117 | 46.15157 |\n",
       "| CG136149 | ai | CG13614 |   62 |    25 |  154.19308 |   0.9322095 | 6.318343 | 45.98439 |\n",
       "| CG133139 | ai | CG13313 |  134 |   149 |  313.65944 |   6.1276992 | 5.459619 | 45.30199 |\n",
       "| Sox21b9 | ai | Sox21b  |  509 |  1529 | 1180.19641 |  60.3829509 | 4.265045 | 43.52919 |\n",
       "| dysf9 | ai | dysf    |  346 |   843 |  802.80386 |  37.5375673 | 4.380710 | 42.27692 |\n",
       "| CG148269 | ai | CG14826 |   55 |    46 |  135.35528 |   1.6502044 | 5.674504 | 40.23919 |\n",
       "| CG344439 | ai | CG34443 |   42 |    15 |  102.86477 |   0.6741551 | 5.941172 | 39.79731 |\n",
       "| CG47029 | ai | CG4702  | 2386 | 13973 | 5371.37356 | 615.2078650 | 3.123802 | 38.70811 |\n",
       "| CG340319 | ai | CG34031 |   54 |    57 |  122.45543 |   2.4692483 | 5.141490 | 35.72228 |\n",
       "| CG148299 | ai | CG14829 |   36 |    23 |   82.41555 |   0.7297718 | 5.574263 | 35.57631 |\n",
       "\n"
      ],
      "text/plain": [
       "         cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "CG177819 ai          CG17781  662                172       1549.72237     \n",
       "CG330039 ai          CG33003  962                649       2191.35950     \n",
       "CG130829 ai          CG13082 1653               2094       3855.55731     \n",
       "CG177809 ai          CG17780  409                330        952.78345     \n",
       "CG136169 ai          CG13616  170                 83        398.11669     \n",
       "CG67859  ai          CG6785   449                451        997.81458     \n",
       "CG147209 ai          CG14720  137                 62        310.64695     \n",
       "ine9     ai          ine      606               1002       1350.15923     \n",
       "otp9     ai          otp     1875               4849       4288.71618     \n",
       "CG77209  ai          CG7720  1625               4546       3631.73769     \n",
       "Ac78C9   ai          Ac78C    288                477        654.90987     \n",
       "CG136149 ai          CG13614   62                 25        154.19308     \n",
       "CG133139 ai          CG13313  134                149        313.65944     \n",
       "Sox21b9  ai          Sox21b   509               1529       1180.19641     \n",
       "dysf9    ai          dysf     346                843        802.80386     \n",
       "CG148269 ai          CG14826   55                 46        135.35528     \n",
       "CG344439 ai          CG34443   42                 15        102.86477     \n",
       "CG47029  ai          CG4702  2386              13973       5371.37356     \n",
       "CG340319 ai          CG34031   54                 57        122.45543     \n",
       "CG148299 ai          CG14829   36                 23         82.41555     \n",
       "         other.tpm   log2.ratio heuristic.score\n",
       "CG177819   7.1383187 7.573063   80.26481       \n",
       "CG330039  28.1453376 6.232417   69.16904       \n",
       "CG130829  89.6252499 5.410883   64.46037       \n",
       "CG177809  14.7350480 5.920095   58.59425       \n",
       "CG136169   2.9610686 6.651158   57.47044       \n",
       "CG67859   19.9419961 5.574301   55.54274       \n",
       "CG147209   2.5760936 6.440748   53.35366       \n",
       "ine9      41.7728514 4.980290   51.79493       \n",
       "otp9     222.8642733 4.259850   51.40218       \n",
       "CG77209  209.9621473 4.105604   48.55633       \n",
       "Ac78C9    20.4519237 4.932117   46.15157       \n",
       "CG136149   0.9322095 6.318343   45.98439       \n",
       "CG133139   6.1276992 5.459619   45.30199       \n",
       "Sox21b9   60.3829509 4.265045   43.52919       \n",
       "dysf9     37.5375673 4.380710   42.27692       \n",
       "CG148269   1.6502044 5.674504   40.23919       \n",
       "CG344439   0.6741551 5.941172   39.79731       \n",
       "CG47029  615.2078650 3.123802   38.70811       \n",
       "CG340319   2.4692483 5.141490   35.72228       \n",
       "CG148299   0.7297718 5.574263   35.57631       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"ai\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG17781\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG17781\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG17781\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG33003\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"ine\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1228,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Ventral_midline\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Ventral_midline\", df_sub300$cluster %in% aj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'aj' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'aj' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(aj)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(113)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1224,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>wrapper10</th><td>aj</td><td>wrapper</td><td>1811</td><td>  797</td><td> 2624.7089</td><td>  31.255734</td><td>6.346458</td><td>72.08619</td></tr>\n",
       "\t<tr><th scope=row>sim10</th><td>aj</td><td>sim    </td><td>1298</td><td>  623</td><td> 1799.1418</td><td>  25.258318</td><td>6.098391</td><td>65.94735</td></tr>\n",
       "\t<tr><th scope=row>Oatp26F10</th><td>aj</td><td>Oatp26F</td><td>1311</td><td>  737</td><td> 1839.9327</td><td>  29.395739</td><td>5.919640</td><td>64.20573</td></tr>\n",
       "\t<tr><th scope=row>tbc10</th><td>aj</td><td>tbc    </td><td> 840</td><td>  476</td><td> 1223.6856</td><td>  20.034026</td><td>5.862364</td><td>60.13728</td></tr>\n",
       "\t<tr><th scope=row>Ect310</th><td>aj</td><td>Ect3   </td><td>1297</td><td> 1301</td><td> 1930.5687</td><td>  52.079066</td><td>5.184739</td><td>56.59432</td></tr>\n",
       "\t<tr><th scope=row>NetB10</th><td>aj</td><td>NetB   </td><td>9778</td><td>19799</td><td>13787.3998</td><td> 838.776003</td><td>4.037202</td><td>55.51624</td></tr>\n",
       "\t<tr><th scope=row>Epac10</th><td>aj</td><td>Epac   </td><td>1647</td><td> 2254</td><td> 2384.6223</td><td>  90.432179</td><td>4.704915</td><td>52.78985</td></tr>\n",
       "\t<tr><th scope=row>hbs10</th><td>aj</td><td>hbs    </td><td>3840</td><td> 8003</td><td> 5332.0549</td><td> 343.383250</td><td>3.952605</td><td>48.93620</td></tr>\n",
       "\t<tr><th scope=row>CG3114510</th><td>aj</td><td>CG31145</td><td>5738</td><td>13541</td><td> 7752.9059</td><td> 631.753902</td><td>3.615021</td><td>46.70863</td></tr>\n",
       "\t<tr><th scope=row>CG4526310</th><td>aj</td><td>CG45263</td><td>7294</td><td>21983</td><td>10071.1472</td><td> 963.640610</td><td>3.384093</td><td>45.00195</td></tr>\n",
       "\t<tr><th scope=row>sli10</th><td>aj</td><td>sli    </td><td>8460</td><td>28936</td><td>11898.0217</td><td>1255.023136</td><td>3.243787</td><td>43.91619</td></tr>\n",
       "\t<tr><th scope=row>NetA10</th><td>aj</td><td>NetA   </td><td>2681</td><td> 7359</td><td> 3736.5960</td><td> 319.980817</td><td>3.541166</td><td>42.02618</td></tr>\n",
       "\t<tr><th scope=row>CG1315710</th><td>aj</td><td>CG13157</td><td> 127</td><td>   80</td><td>  174.6199</td><td>   2.966085</td><td>5.460359</td><td>40.71414</td></tr>\n",
       "\t<tr><th scope=row>shep10</th><td>aj</td><td>shep   </td><td>8749</td><td>35922</td><td>11957.8983</td><td>1528.249140</td><td>2.967068</td><td>40.19131</td></tr>\n",
       "\t<tr><th scope=row>oc10</th><td>aj</td><td>oc     </td><td>1450</td><td> 4544</td><td> 2029.4439</td><td> 178.176175</td><td>3.501634</td><td>38.47448</td></tr>\n",
       "\t<tr><th scope=row>dpr1310</th><td>aj</td><td>dpr13  </td><td> 517</td><td> 1001</td><td>  734.4911</td><td>  45.473872</td><td>3.982253</td><td>37.92126</td></tr>\n",
       "\t<tr><th scope=row>vir-110</th><td>aj</td><td>vir-1  </td><td> 687</td><td> 1801</td><td>  986.6289</td><td>  72.291517</td><td>3.750789</td><td>37.31220</td></tr>\n",
       "\t<tr><th scope=row>Pvf310</th><td>aj</td><td>Pvf3   </td><td>3030</td><td>12586</td><td> 4119.4431</td><td> 542.406933</td><td>2.922344</td><td>35.09322</td></tr>\n",
       "\t<tr><th scope=row>Tsp5D10</th><td>aj</td><td>Tsp5D  </td><td> 314</td><td>  713</td><td>  460.8116</td><td>  30.117315</td><td>3.888387</td><td>34.41674</td></tr>\n",
       "\t<tr><th scope=row>hog10</th><td>aj</td><td>hog    </td><td> 117</td><td>  141</td><td>  169.6039</td><td>   6.269954</td><td>4.544080</td><td>33.69211</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\twrapper10 & aj & wrapper & 1811 &   797 &  2624.7089 &   31.255734 & 6.346458 & 72.08619\\\\\n",
       "\tsim10 & aj & sim     & 1298 &   623 &  1799.1418 &   25.258318 & 6.098391 & 65.94735\\\\\n",
       "\tOatp26F10 & aj & Oatp26F & 1311 &   737 &  1839.9327 &   29.395739 & 5.919640 & 64.20573\\\\\n",
       "\ttbc10 & aj & tbc     &  840 &   476 &  1223.6856 &   20.034026 & 5.862364 & 60.13728\\\\\n",
       "\tEct310 & aj & Ect3    & 1297 &  1301 &  1930.5687 &   52.079066 & 5.184739 & 56.59432\\\\\n",
       "\tNetB10 & aj & NetB    & 9778 & 19799 & 13787.3998 &  838.776003 & 4.037202 & 55.51624\\\\\n",
       "\tEpac10 & aj & Epac    & 1647 &  2254 &  2384.6223 &   90.432179 & 4.704915 & 52.78985\\\\\n",
       "\thbs10 & aj & hbs     & 3840 &  8003 &  5332.0549 &  343.383250 & 3.952605 & 48.93620\\\\\n",
       "\tCG3114510 & aj & CG31145 & 5738 & 13541 &  7752.9059 &  631.753902 & 3.615021 & 46.70863\\\\\n",
       "\tCG4526310 & aj & CG45263 & 7294 & 21983 & 10071.1472 &  963.640610 & 3.384093 & 45.00195\\\\\n",
       "\tsli10 & aj & sli     & 8460 & 28936 & 11898.0217 & 1255.023136 & 3.243787 & 43.91619\\\\\n",
       "\tNetA10 & aj & NetA    & 2681 &  7359 &  3736.5960 &  319.980817 & 3.541166 & 42.02618\\\\\n",
       "\tCG1315710 & aj & CG13157 &  127 &    80 &   174.6199 &    2.966085 & 5.460359 & 40.71414\\\\\n",
       "\tshep10 & aj & shep    & 8749 & 35922 & 11957.8983 & 1528.249140 & 2.967068 & 40.19131\\\\\n",
       "\toc10 & aj & oc      & 1450 &  4544 &  2029.4439 &  178.176175 & 3.501634 & 38.47448\\\\\n",
       "\tdpr1310 & aj & dpr13   &  517 &  1001 &   734.4911 &   45.473872 & 3.982253 & 37.92126\\\\\n",
       "\tvir-110 & aj & vir-1   &  687 &  1801 &   986.6289 &   72.291517 & 3.750789 & 37.31220\\\\\n",
       "\tPvf310 & aj & Pvf3    & 3030 & 12586 &  4119.4431 &  542.406933 & 2.922344 & 35.09322\\\\\n",
       "\tTsp5D10 & aj & Tsp5D   &  314 &   713 &   460.8116 &   30.117315 & 3.888387 & 34.41674\\\\\n",
       "\thog10 & aj & hog     &  117 &   141 &   169.6039 &    6.269954 & 4.544080 & 33.69211\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| wrapper10 | aj | wrapper | 1811 |   797 |  2624.7089 |   31.255734 | 6.346458 | 72.08619 |\n",
       "| sim10 | aj | sim     | 1298 |   623 |  1799.1418 |   25.258318 | 6.098391 | 65.94735 |\n",
       "| Oatp26F10 | aj | Oatp26F | 1311 |   737 |  1839.9327 |   29.395739 | 5.919640 | 64.20573 |\n",
       "| tbc10 | aj | tbc     |  840 |   476 |  1223.6856 |   20.034026 | 5.862364 | 60.13728 |\n",
       "| Ect310 | aj | Ect3    | 1297 |  1301 |  1930.5687 |   52.079066 | 5.184739 | 56.59432 |\n",
       "| NetB10 | aj | NetB    | 9778 | 19799 | 13787.3998 |  838.776003 | 4.037202 | 55.51624 |\n",
       "| Epac10 | aj | Epac    | 1647 |  2254 |  2384.6223 |   90.432179 | 4.704915 | 52.78985 |\n",
       "| hbs10 | aj | hbs     | 3840 |  8003 |  5332.0549 |  343.383250 | 3.952605 | 48.93620 |\n",
       "| CG3114510 | aj | CG31145 | 5738 | 13541 |  7752.9059 |  631.753902 | 3.615021 | 46.70863 |\n",
       "| CG4526310 | aj | CG45263 | 7294 | 21983 | 10071.1472 |  963.640610 | 3.384093 | 45.00195 |\n",
       "| sli10 | aj | sli     | 8460 | 28936 | 11898.0217 | 1255.023136 | 3.243787 | 43.91619 |\n",
       "| NetA10 | aj | NetA    | 2681 |  7359 |  3736.5960 |  319.980817 | 3.541166 | 42.02618 |\n",
       "| CG1315710 | aj | CG13157 |  127 |    80 |   174.6199 |    2.966085 | 5.460359 | 40.71414 |\n",
       "| shep10 | aj | shep    | 8749 | 35922 | 11957.8983 | 1528.249140 | 2.967068 | 40.19131 |\n",
       "| oc10 | aj | oc      | 1450 |  4544 |  2029.4439 |  178.176175 | 3.501634 | 38.47448 |\n",
       "| dpr1310 | aj | dpr13   |  517 |  1001 |   734.4911 |   45.473872 | 3.982253 | 37.92126 |\n",
       "| vir-110 | aj | vir-1   |  687 |  1801 |   986.6289 |   72.291517 | 3.750789 | 37.31220 |\n",
       "| Pvf310 | aj | Pvf3    | 3030 | 12586 |  4119.4431 |  542.406933 | 2.922344 | 35.09322 |\n",
       "| Tsp5D10 | aj | Tsp5D   |  314 |   713 |   460.8116 |   30.117315 | 3.888387 | 34.41674 |\n",
       "| hog10 | aj | hog     |  117 |   141 |   169.6039 |    6.269954 | 4.544080 | 33.69211 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "wrapper10 aj          wrapper 1811                797        2624.7089     \n",
       "sim10     aj          sim     1298                623        1799.1418     \n",
       "Oatp26F10 aj          Oatp26F 1311                737        1839.9327     \n",
       "tbc10     aj          tbc      840                476        1223.6856     \n",
       "Ect310    aj          Ect3    1297               1301        1930.5687     \n",
       "NetB10    aj          NetB    9778              19799       13787.3998     \n",
       "Epac10    aj          Epac    1647               2254        2384.6223     \n",
       "hbs10     aj          hbs     3840               8003        5332.0549     \n",
       "CG3114510 aj          CG31145 5738              13541        7752.9059     \n",
       "CG4526310 aj          CG45263 7294              21983       10071.1472     \n",
       "sli10     aj          sli     8460              28936       11898.0217     \n",
       "NetA10    aj          NetA    2681               7359        3736.5960     \n",
       "CG1315710 aj          CG13157  127                 80         174.6199     \n",
       "shep10    aj          shep    8749              35922       11957.8983     \n",
       "oc10      aj          oc      1450               4544        2029.4439     \n",
       "dpr1310   aj          dpr13    517               1001         734.4911     \n",
       "vir-110   aj          vir-1    687               1801         986.6289     \n",
       "Pvf310    aj          Pvf3    3030              12586        4119.4431     \n",
       "Tsp5D10   aj          Tsp5D    314                713         460.8116     \n",
       "hog10     aj          hog      117                141         169.6039     \n",
       "          other.tpm   log2.ratio heuristic.score\n",
       "wrapper10   31.255734 6.346458   72.08619       \n",
       "sim10       25.258318 6.098391   65.94735       \n",
       "Oatp26F10   29.395739 5.919640   64.20573       \n",
       "tbc10       20.034026 5.862364   60.13728       \n",
       "Ect310      52.079066 5.184739   56.59432       \n",
       "NetB10     838.776003 4.037202   55.51624       \n",
       "Epac10      90.432179 4.704915   52.78985       \n",
       "hbs10      343.383250 3.952605   48.93620       \n",
       "CG3114510  631.753902 3.615021   46.70863       \n",
       "CG4526310  963.640610 3.384093   45.00195       \n",
       "sli10     1255.023136 3.243787   43.91619       \n",
       "NetA10     319.980817 3.541166   42.02618       \n",
       "CG1315710    2.966085 5.460359   40.71414       \n",
       "shep10    1528.249140 2.967068   40.19131       \n",
       "oc10       178.176175 3.501634   38.47448       \n",
       "dpr1310     45.473872 3.982253   37.92126       \n",
       "vir-110     72.291517 3.750789   37.31220       \n",
       "Pvf310     542.406933 2.922344   35.09322       \n",
       "Tsp5D10     30.117315 3.888387   34.41674       \n",
       "hog10        6.269954 4.544080   33.69211       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"aj\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"sim\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"sim\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"sim\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"rho\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"wrapper\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1271,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Longitudinal_visceral_muscle\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#also same genes in lower part of 75/bc added to this cell type.\n",
    "df_sub300 = set.cell.type(df_sub300, \"Longitudinal_visceral_muscle\", df_sub300$cluster %in% ak)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ak' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ak' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ak)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bc' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bc' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bc)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1230,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>tey11</th><td>ak</td><td>tey     </td><td>1065</td><td> 1710</td><td> 6114.6290</td><td>  71.99142</td><td>6.388394</td><td>80.35504</td></tr>\n",
       "\t<tr><th scope=row>AdamTS-A11</th><td>ak</td><td>AdamTS-A</td><td> 912</td><td> 3531</td><td> 5263.4964</td><td> 158.82552</td><td>5.041452</td><td>62.32283</td></tr>\n",
       "\t<tr><th scope=row>beat-IIb11</th><td>ak</td><td>beat-IIb</td><td> 643</td><td> 2318</td><td> 3694.9865</td><td> 106.34885</td><td>5.105190</td><td>60.50541</td></tr>\n",
       "\t<tr><th scope=row>Syn211</th><td>ak</td><td>Syn2    </td><td> 166</td><td>  429</td><td>  973.9816</td><td>  17.11328</td><td>5.748775</td><td>57.08091</td></tr>\n",
       "\t<tr><th scope=row>CG1518611</th><td>ak</td><td>CG15186 </td><td> 169</td><td>  590</td><td>  979.2745</td><td>  25.50536</td><td>5.207358</td><td>51.74573</td></tr>\n",
       "\t<tr><th scope=row>OtopLa11</th><td>ak</td><td>OtopLa  </td><td> 974</td><td> 7605</td><td> 5450.5802</td><td> 335.41754</td><td>4.018085</td><td>49.87431</td></tr>\n",
       "\t<tr><th scope=row>Shaw11</th><td>ak</td><td>Shaw    </td><td> 202</td><td>  926</td><td> 1176.6363</td><td>  40.63772</td><td>4.820634</td><td>49.17855</td></tr>\n",
       "\t<tr><th scope=row>HLH54F11</th><td>ak</td><td>HLH54F  </td><td> 104</td><td>  394</td><td>  607.0405</td><td>  16.11352</td><td>5.148585</td><td>47.61423</td></tr>\n",
       "\t<tr><th scope=row>beat-IIa11</th><td>ak</td><td>beat-IIa</td><td> 254</td><td> 1656</td><td> 1433.7905</td><td>  73.57266</td><td>4.265044</td><td>44.72591</td></tr>\n",
       "\t<tr><th scope=row>CG3011611</th><td>ak</td><td>CG30116 </td><td> 178</td><td> 1136</td><td>  979.0871</td><td>  50.09711</td><td>4.260124</td><td>42.33185</td></tr>\n",
       "\t<tr><th scope=row>trol11</th><td>ak</td><td>trol    </td><td>3116</td><td>47092</td><td>17916.2486</td><td>2246.06191</td><td>2.995157</td><td>42.31875</td></tr>\n",
       "\t<tr><th scope=row>H2.011</th><td>ak</td><td>H2.0    </td><td> 101</td><td>  568</td><td>  580.5796</td><td>  22.96913</td><td>4.598244</td><td>42.22951</td></tr>\n",
       "\t<tr><th scope=row>pyx11</th><td>ak</td><td>pyx     </td><td>  72</td><td>  289</td><td>  388.5335</td><td>  12.92488</td><td>4.802302</td><td>41.32671</td></tr>\n",
       "\t<tr><th scope=row>disco-r11</th><td>ak</td><td>disco-r </td><td> 623</td><td> 7480</td><td> 3517.8613</td><td> 332.77288</td><td>3.397760</td><td>40.02865</td></tr>\n",
       "\t<tr><th scope=row>org-111</th><td>ak</td><td>org-1   </td><td> 155</td><td> 1120</td><td>  847.2127</td><td>  50.46593</td><td>4.041034</td><td>39.31232</td></tr>\n",
       "\t<tr><th scope=row>sals11</th><td>ak</td><td>sals    </td><td> 229</td><td> 2583</td><td> 1252.3009</td><td> 110.12948</td><td>3.494268</td><td>35.96132</td></tr>\n",
       "\t<tr><th scope=row>shakB11</th><td>ak</td><td>shakB   </td><td> 583</td><td> 9613</td><td> 3257.2239</td><td> 433.28646</td><td>2.906924</td><td>33.92343</td></tr>\n",
       "\t<tr><th scope=row>CG1350611</th><td>ak</td><td>CG13506 </td><td> 226</td><td> 3025</td><td> 1319.7441</td><td> 137.03422</td><td>3.257160</td><td>33.76742</td></tr>\n",
       "\t<tr><th scope=row>bru311</th><td>ak</td><td>bru3    </td><td>1919</td><td>42922</td><td>10787.4963</td><td>1904.97366</td><td>2.500760</td><td>33.50320</td></tr>\n",
       "\t<tr><th scope=row>sqa11</th><td>ak</td><td>sqa     </td><td> 133</td><td> 1574</td><td>  754.4927</td><td>  67.26239</td><td>3.466344</td><td>33.14267</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\ttey11 & ak & tey      & 1065 &  1710 &  6114.6290 &   71.99142 & 6.388394 & 80.35504\\\\\n",
       "\tAdamTS-A11 & ak & AdamTS-A &  912 &  3531 &  5263.4964 &  158.82552 & 5.041452 & 62.32283\\\\\n",
       "\tbeat-IIb11 & ak & beat-IIb &  643 &  2318 &  3694.9865 &  106.34885 & 5.105190 & 60.50541\\\\\n",
       "\tSyn211 & ak & Syn2     &  166 &   429 &   973.9816 &   17.11328 & 5.748775 & 57.08091\\\\\n",
       "\tCG1518611 & ak & CG15186  &  169 &   590 &   979.2745 &   25.50536 & 5.207358 & 51.74573\\\\\n",
       "\tOtopLa11 & ak & OtopLa   &  974 &  7605 &  5450.5802 &  335.41754 & 4.018085 & 49.87431\\\\\n",
       "\tShaw11 & ak & Shaw     &  202 &   926 &  1176.6363 &   40.63772 & 4.820634 & 49.17855\\\\\n",
       "\tHLH54F11 & ak & HLH54F   &  104 &   394 &   607.0405 &   16.11352 & 5.148585 & 47.61423\\\\\n",
       "\tbeat-IIa11 & ak & beat-IIa &  254 &  1656 &  1433.7905 &   73.57266 & 4.265044 & 44.72591\\\\\n",
       "\tCG3011611 & ak & CG30116  &  178 &  1136 &   979.0871 &   50.09711 & 4.260124 & 42.33185\\\\\n",
       "\ttrol11 & ak & trol     & 3116 & 47092 & 17916.2486 & 2246.06191 & 2.995157 & 42.31875\\\\\n",
       "\tH2.011 & ak & H2.0     &  101 &   568 &   580.5796 &   22.96913 & 4.598244 & 42.22951\\\\\n",
       "\tpyx11 & ak & pyx      &   72 &   289 &   388.5335 &   12.92488 & 4.802302 & 41.32671\\\\\n",
       "\tdisco-r11 & ak & disco-r  &  623 &  7480 &  3517.8613 &  332.77288 & 3.397760 & 40.02865\\\\\n",
       "\torg-111 & ak & org-1    &  155 &  1120 &   847.2127 &   50.46593 & 4.041034 & 39.31232\\\\\n",
       "\tsals11 & ak & sals     &  229 &  2583 &  1252.3009 &  110.12948 & 3.494268 & 35.96132\\\\\n",
       "\tshakB11 & ak & shakB    &  583 &  9613 &  3257.2239 &  433.28646 & 2.906924 & 33.92343\\\\\n",
       "\tCG1350611 & ak & CG13506  &  226 &  3025 &  1319.7441 &  137.03422 & 3.257160 & 33.76742\\\\\n",
       "\tbru311 & ak & bru3     & 1919 & 42922 & 10787.4963 & 1904.97366 & 2.500760 & 33.50320\\\\\n",
       "\tsqa11 & ak & sqa      &  133 &  1574 &   754.4927 &   67.26239 & 3.466344 & 33.14267\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| tey11 | ak | tey      | 1065 |  1710 |  6114.6290 |   71.99142 | 6.388394 | 80.35504 |\n",
       "| AdamTS-A11 | ak | AdamTS-A |  912 |  3531 |  5263.4964 |  158.82552 | 5.041452 | 62.32283 |\n",
       "| beat-IIb11 | ak | beat-IIb |  643 |  2318 |  3694.9865 |  106.34885 | 5.105190 | 60.50541 |\n",
       "| Syn211 | ak | Syn2     |  166 |   429 |   973.9816 |   17.11328 | 5.748775 | 57.08091 |\n",
       "| CG1518611 | ak | CG15186  |  169 |   590 |   979.2745 |   25.50536 | 5.207358 | 51.74573 |\n",
       "| OtopLa11 | ak | OtopLa   |  974 |  7605 |  5450.5802 |  335.41754 | 4.018085 | 49.87431 |\n",
       "| Shaw11 | ak | Shaw     |  202 |   926 |  1176.6363 |   40.63772 | 4.820634 | 49.17855 |\n",
       "| HLH54F11 | ak | HLH54F   |  104 |   394 |   607.0405 |   16.11352 | 5.148585 | 47.61423 |\n",
       "| beat-IIa11 | ak | beat-IIa |  254 |  1656 |  1433.7905 |   73.57266 | 4.265044 | 44.72591 |\n",
       "| CG3011611 | ak | CG30116  |  178 |  1136 |   979.0871 |   50.09711 | 4.260124 | 42.33185 |\n",
       "| trol11 | ak | trol     | 3116 | 47092 | 17916.2486 | 2246.06191 | 2.995157 | 42.31875 |\n",
       "| H2.011 | ak | H2.0     |  101 |   568 |   580.5796 |   22.96913 | 4.598244 | 42.22951 |\n",
       "| pyx11 | ak | pyx      |   72 |   289 |   388.5335 |   12.92488 | 4.802302 | 41.32671 |\n",
       "| disco-r11 | ak | disco-r  |  623 |  7480 |  3517.8613 |  332.77288 | 3.397760 | 40.02865 |\n",
       "| org-111 | ak | org-1    |  155 |  1120 |   847.2127 |   50.46593 | 4.041034 | 39.31232 |\n",
       "| sals11 | ak | sals     |  229 |  2583 |  1252.3009 |  110.12948 | 3.494268 | 35.96132 |\n",
       "| shakB11 | ak | shakB    |  583 |  9613 |  3257.2239 |  433.28646 | 2.906924 | 33.92343 |\n",
       "| CG1350611 | ak | CG13506  |  226 |  3025 |  1319.7441 |  137.03422 | 3.257160 | 33.76742 |\n",
       "| bru311 | ak | bru3     | 1919 | 42922 | 10787.4963 | 1904.97366 | 2.500760 | 33.50320 |\n",
       "| sqa11 | ak | sqa      |  133 |  1574 |   754.4927 |   67.26239 | 3.466344 | 33.14267 |\n",
       "\n"
      ],
      "text/plain": [
       "           cell.type.2 gene     cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "tey11      ak          tey      1065               1710        6114.6290     \n",
       "AdamTS-A11 ak          AdamTS-A  912               3531        5263.4964     \n",
       "beat-IIb11 ak          beat-IIb  643               2318        3694.9865     \n",
       "Syn211     ak          Syn2      166                429         973.9816     \n",
       "CG1518611  ak          CG15186   169                590         979.2745     \n",
       "OtopLa11   ak          OtopLa    974               7605        5450.5802     \n",
       "Shaw11     ak          Shaw      202                926        1176.6363     \n",
       "HLH54F11   ak          HLH54F    104                394         607.0405     \n",
       "beat-IIa11 ak          beat-IIa  254               1656        1433.7905     \n",
       "CG3011611  ak          CG30116   178               1136         979.0871     \n",
       "trol11     ak          trol     3116              47092       17916.2486     \n",
       "H2.011     ak          H2.0      101                568         580.5796     \n",
       "pyx11      ak          pyx        72                289         388.5335     \n",
       "disco-r11  ak          disco-r   623               7480        3517.8613     \n",
       "org-111    ak          org-1     155               1120         847.2127     \n",
       "sals11     ak          sals      229               2583        1252.3009     \n",
       "shakB11    ak          shakB     583               9613        3257.2239     \n",
       "CG1350611  ak          CG13506   226               3025        1319.7441     \n",
       "bru311     ak          bru3     1919              42922       10787.4963     \n",
       "sqa11      ak          sqa       133               1574         754.4927     \n",
       "           other.tpm  log2.ratio heuristic.score\n",
       "tey11        71.99142 6.388394   80.35504       \n",
       "AdamTS-A11  158.82552 5.041452   62.32283       \n",
       "beat-IIb11  106.34885 5.105190   60.50541       \n",
       "Syn211       17.11328 5.748775   57.08091       \n",
       "CG1518611    25.50536 5.207358   51.74573       \n",
       "OtopLa11    335.41754 4.018085   49.87431       \n",
       "Shaw11       40.63772 4.820634   49.17855       \n",
       "HLH54F11     16.11352 5.148585   47.61423       \n",
       "beat-IIa11   73.57266 4.265044   44.72591       \n",
       "CG3011611    50.09711 4.260124   42.33185       \n",
       "trol11     2246.06191 2.995157   42.31875       \n",
       "H2.011       22.96913 4.598244   42.22951       \n",
       "pyx11        12.92488 4.802302   41.32671       \n",
       "disco-r11   332.77288 3.397760   40.02865       \n",
       "org-111      50.46593 4.041034   39.31232       \n",
       "sals11      110.12948 3.494268   35.96132       \n",
       "shakB11     433.28646 2.906924   33.92343       \n",
       "CG1350611   137.03422 3.257160   33.76742       \n",
       "bru311     1904.97366 2.500760   33.50320       \n",
       "sqa11        67.26239 3.466344   33.14267       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"ak\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"tey\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"tey\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"tey\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"AdamTS-A\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"beat-IIb\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ab' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ab' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### al cell type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1279,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Tracheal_system_older\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Tracheal_system_older\", df_sub300$cluster %in% al)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'al' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'al' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(al)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bc' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bc' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bc)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1277,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Osi1512</th><td>al</td><td>Osi15   </td><td>22898</td><td>13527</td><td>23154.0409</td><td>494.415436</td><td>5.546481</td><td>80.41865</td></tr>\n",
       "\t<tr><th scope=row>Osi912</th><td>al</td><td>Osi9    </td><td>10784</td><td> 5385</td><td>11256.0215</td><td>199.378375</td><td>5.811826</td><td>78.21868</td></tr>\n",
       "\t<tr><th scope=row>Osi2012</th><td>al</td><td>Osi20   </td><td> 9176</td><td> 5191</td><td> 9648.3967</td><td>180.703142</td><td>5.730634</td><td>75.85195</td></tr>\n",
       "\t<tr><th scope=row>CG1272312</th><td>al</td><td>CG12723 </td><td> 3357</td><td> 1306</td><td> 3528.6900</td><td> 39.895201</td><td>6.431057</td><td>75.79211</td></tr>\n",
       "\t<tr><th scope=row>Osi1912</th><td>al</td><td>Osi19   </td><td> 5121</td><td> 2634</td><td> 5270.5818</td><td> 88.219825</td><td>5.884454</td><td>72.75551</td></tr>\n",
       "\t<tr><th scope=row>Osi1812</th><td>al</td><td>Osi18   </td><td> 4771</td><td> 2927</td><td> 4940.4450</td><td>104.274374</td><td>5.552415</td><td>68.13211</td></tr>\n",
       "\t<tr><th scope=row>CG4274912</th><td>al</td><td>CG42749 </td><td> 1220</td><td>  387</td><td> 1354.3149</td><td> 16.345314</td><td>6.286873</td><td>65.41122</td></tr>\n",
       "\t<tr><th scope=row>CG3254812</th><td>al</td><td>CG32548 </td><td> 1580</td><td>  740</td><td> 1607.3770</td><td> 23.476831</td><td>6.037148</td><td>64.30401</td></tr>\n",
       "\t<tr><th scope=row>CG1200912</th><td>al</td><td>CG12009 </td><td> 5579</td><td> 4749</td><td> 5940.4636</td><td>175.985507</td><td>5.068872</td><td>63.54644</td></tr>\n",
       "\t<tr><th scope=row>CG720112</th><td>al</td><td>CG7201  </td><td>  777</td><td>  222</td><td>  811.7920</td><td>  7.794014</td><td>6.528445</td><td>63.10879</td></tr>\n",
       "\t<tr><th scope=row>Mal-B112</th><td>al</td><td>Mal-B1  </td><td>  273</td><td>   59</td><td>  315.5729</td><td>  1.943188</td><td>6.744450</td><td>56.02206</td></tr>\n",
       "\t<tr><th scope=row>CG1327212</th><td>al</td><td>CG13272 </td><td> 1370</td><td> 1006</td><td> 1489.1380</td><td> 39.745363</td><td>5.191698</td><td>54.72688</td></tr>\n",
       "\t<tr><th scope=row>CG1304912</th><td>al</td><td>CG13049 </td><td> 1491</td><td> 1496</td><td> 1583.0508</td><td> 52.214841</td><td>4.894735</td><td>52.02811</td></tr>\n",
       "\t<tr><th scope=row>wat12</th><td>al</td><td>wat     </td><td> 1847</td><td> 1588</td><td> 1896.8104</td><td> 69.631771</td><td>4.747114</td><td>51.69665</td></tr>\n",
       "\t<tr><th scope=row>CG1178612</th><td>al</td><td>CG11786 </td><td>  794</td><td>  428</td><td>  834.0502</td><td> 20.512054</td><td>5.276917</td><td>51.21627</td></tr>\n",
       "\t<tr><th scope=row>CG1304812</th><td>al</td><td>CG13048 </td><td>  459</td><td>  300</td><td>  480.2652</td><td>  9.573110</td><td>5.505359</td><td>49.05654</td></tr>\n",
       "\t<tr><th scope=row>apn12</th><td>al</td><td>apn     </td><td> 1726</td><td> 1961</td><td> 1677.5698</td><td> 76.018999</td><td>4.445015</td><td>47.61952</td></tr>\n",
       "\t<tr><th scope=row>CG1300312</th><td>al</td><td>CG13003 </td><td> 1623</td><td> 1947</td><td> 1787.8613</td><td> 83.609581</td><td>4.401270</td><td>47.55496</td></tr>\n",
       "\t<tr><th scope=row>Twdlbeta12</th><td>al</td><td>Twdlbeta</td><td>  709</td><td>  647</td><td>  721.0234</td><td> 22.946668</td><td>4.912149</td><td>46.64529</td></tr>\n",
       "\t<tr><th scope=row>CG1435612</th><td>al</td><td>CG14356 </td><td>  184</td><td>   55</td><td>  187.7935</td><td>  1.803246</td><td>6.065905</td><td>45.86227</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tOsi1512 & al & Osi15    & 22898 & 13527 & 23154.0409 & 494.415436 & 5.546481 & 80.41865\\\\\n",
       "\tOsi912 & al & Osi9     & 10784 &  5385 & 11256.0215 & 199.378375 & 5.811826 & 78.21868\\\\\n",
       "\tOsi2012 & al & Osi20    &  9176 &  5191 &  9648.3967 & 180.703142 & 5.730634 & 75.85195\\\\\n",
       "\tCG1272312 & al & CG12723  &  3357 &  1306 &  3528.6900 &  39.895201 & 6.431057 & 75.79211\\\\\n",
       "\tOsi1912 & al & Osi19    &  5121 &  2634 &  5270.5818 &  88.219825 & 5.884454 & 72.75551\\\\\n",
       "\tOsi1812 & al & Osi18    &  4771 &  2927 &  4940.4450 & 104.274374 & 5.552415 & 68.13211\\\\\n",
       "\tCG4274912 & al & CG42749  &  1220 &   387 &  1354.3149 &  16.345314 & 6.286873 & 65.41122\\\\\n",
       "\tCG3254812 & al & CG32548  &  1580 &   740 &  1607.3770 &  23.476831 & 6.037148 & 64.30401\\\\\n",
       "\tCG1200912 & al & CG12009  &  5579 &  4749 &  5940.4636 & 175.985507 & 5.068872 & 63.54644\\\\\n",
       "\tCG720112 & al & CG7201   &   777 &   222 &   811.7920 &   7.794014 & 6.528445 & 63.10879\\\\\n",
       "\tMal-B112 & al & Mal-B1   &   273 &    59 &   315.5729 &   1.943188 & 6.744450 & 56.02206\\\\\n",
       "\tCG1327212 & al & CG13272  &  1370 &  1006 &  1489.1380 &  39.745363 & 5.191698 & 54.72688\\\\\n",
       "\tCG1304912 & al & CG13049  &  1491 &  1496 &  1583.0508 &  52.214841 & 4.894735 & 52.02811\\\\\n",
       "\twat12 & al & wat      &  1847 &  1588 &  1896.8104 &  69.631771 & 4.747114 & 51.69665\\\\\n",
       "\tCG1178612 & al & CG11786  &   794 &   428 &   834.0502 &  20.512054 & 5.276917 & 51.21627\\\\\n",
       "\tCG1304812 & al & CG13048  &   459 &   300 &   480.2652 &   9.573110 & 5.505359 & 49.05654\\\\\n",
       "\tapn12 & al & apn      &  1726 &  1961 &  1677.5698 &  76.018999 & 4.445015 & 47.61952\\\\\n",
       "\tCG1300312 & al & CG13003  &  1623 &  1947 &  1787.8613 &  83.609581 & 4.401270 & 47.55496\\\\\n",
       "\tTwdlbeta12 & al & Twdlbeta &   709 &   647 &   721.0234 &  22.946668 & 4.912149 & 46.64529\\\\\n",
       "\tCG1435612 & al & CG14356  &   184 &    55 &   187.7935 &   1.803246 & 6.065905 & 45.86227\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Osi1512 | al | Osi15    | 22898 | 13527 | 23154.0409 | 494.415436 | 5.546481 | 80.41865 |\n",
       "| Osi912 | al | Osi9     | 10784 |  5385 | 11256.0215 | 199.378375 | 5.811826 | 78.21868 |\n",
       "| Osi2012 | al | Osi20    |  9176 |  5191 |  9648.3967 | 180.703142 | 5.730634 | 75.85195 |\n",
       "| CG1272312 | al | CG12723  |  3357 |  1306 |  3528.6900 |  39.895201 | 6.431057 | 75.79211 |\n",
       "| Osi1912 | al | Osi19    |  5121 |  2634 |  5270.5818 |  88.219825 | 5.884454 | 72.75551 |\n",
       "| Osi1812 | al | Osi18    |  4771 |  2927 |  4940.4450 | 104.274374 | 5.552415 | 68.13211 |\n",
       "| CG4274912 | al | CG42749  |  1220 |   387 |  1354.3149 |  16.345314 | 6.286873 | 65.41122 |\n",
       "| CG3254812 | al | CG32548  |  1580 |   740 |  1607.3770 |  23.476831 | 6.037148 | 64.30401 |\n",
       "| CG1200912 | al | CG12009  |  5579 |  4749 |  5940.4636 | 175.985507 | 5.068872 | 63.54644 |\n",
       "| CG720112 | al | CG7201   |   777 |   222 |   811.7920 |   7.794014 | 6.528445 | 63.10879 |\n",
       "| Mal-B112 | al | Mal-B1   |   273 |    59 |   315.5729 |   1.943188 | 6.744450 | 56.02206 |\n",
       "| CG1327212 | al | CG13272  |  1370 |  1006 |  1489.1380 |  39.745363 | 5.191698 | 54.72688 |\n",
       "| CG1304912 | al | CG13049  |  1491 |  1496 |  1583.0508 |  52.214841 | 4.894735 | 52.02811 |\n",
       "| wat12 | al | wat      |  1847 |  1588 |  1896.8104 |  69.631771 | 4.747114 | 51.69665 |\n",
       "| CG1178612 | al | CG11786  |   794 |   428 |   834.0502 |  20.512054 | 5.276917 | 51.21627 |\n",
       "| CG1304812 | al | CG13048  |   459 |   300 |   480.2652 |   9.573110 | 5.505359 | 49.05654 |\n",
       "| apn12 | al | apn      |  1726 |  1961 |  1677.5698 |  76.018999 | 4.445015 | 47.61952 |\n",
       "| CG1300312 | al | CG13003  |  1623 |  1947 |  1787.8613 |  83.609581 | 4.401270 | 47.55496 |\n",
       "| Twdlbeta12 | al | Twdlbeta |   709 |   647 |   721.0234 |  22.946668 | 4.912149 | 46.64529 |\n",
       "| CG1435612 | al | CG14356  |   184 |    55 |   187.7935 |   1.803246 | 6.065905 | 45.86227 |\n",
       "\n"
      ],
      "text/plain": [
       "           cell.type.2 gene     cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "Osi1512    al          Osi15    22898             13527       23154.0409     \n",
       "Osi912     al          Osi9     10784              5385       11256.0215     \n",
       "Osi2012    al          Osi20     9176              5191        9648.3967     \n",
       "CG1272312  al          CG12723   3357              1306        3528.6900     \n",
       "Osi1912    al          Osi19     5121              2634        5270.5818     \n",
       "Osi1812    al          Osi18     4771              2927        4940.4450     \n",
       "CG4274912  al          CG42749   1220               387        1354.3149     \n",
       "CG3254812  al          CG32548   1580               740        1607.3770     \n",
       "CG1200912  al          CG12009   5579              4749        5940.4636     \n",
       "CG720112   al          CG7201     777               222         811.7920     \n",
       "Mal-B112   al          Mal-B1     273                59         315.5729     \n",
       "CG1327212  al          CG13272   1370              1006        1489.1380     \n",
       "CG1304912  al          CG13049   1491              1496        1583.0508     \n",
       "wat12      al          wat       1847              1588        1896.8104     \n",
       "CG1178612  al          CG11786    794               428         834.0502     \n",
       "CG1304812  al          CG13048    459               300         480.2652     \n",
       "apn12      al          apn       1726              1961        1677.5698     \n",
       "CG1300312  al          CG13003   1623              1947        1787.8613     \n",
       "Twdlbeta12 al          Twdlbeta   709               647         721.0234     \n",
       "CG1435612  al          CG14356    184                55         187.7935     \n",
       "           other.tpm  log2.ratio heuristic.score\n",
       "Osi1512    494.415436 5.546481   80.41865       \n",
       "Osi912     199.378375 5.811826   78.21868       \n",
       "Osi2012    180.703142 5.730634   75.85195       \n",
       "CG1272312   39.895201 6.431057   75.79211       \n",
       "Osi1912     88.219825 5.884454   72.75551       \n",
       "Osi1812    104.274374 5.552415   68.13211       \n",
       "CG4274912   16.345314 6.286873   65.41122       \n",
       "CG3254812   23.476831 6.037148   64.30401       \n",
       "CG1200912  175.985507 5.068872   63.54644       \n",
       "CG720112     7.794014 6.528445   63.10879       \n",
       "Mal-B112     1.943188 6.744450   56.02206       \n",
       "CG1327212   39.745363 5.191698   54.72688       \n",
       "CG1304912   52.214841 4.894735   52.02811       \n",
       "wat12       69.631771 4.747114   51.69665       \n",
       "CG1178612   20.512054 5.276917   51.21627       \n",
       "CG1304812    9.573110 5.505359   49.05654       \n",
       "apn12       76.018999 4.445015   47.61952       \n",
       "CG1300312   83.609581 4.401270   47.55496       \n",
       "Twdlbeta12  22.946668 4.912149   46.64529       \n",
       "CG1435612    1.803246 6.065905   45.86227       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"al\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Osi15\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Osi15\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Osi15\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Osi9\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Osi20\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1284,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Olfactory_neurons\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Olfactory_neurons\", df_sub300$cluster %in% am)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "flybase says pdm3 \"encodes a POU domain transcription factor that acts in odor receptor gene expression and axon targeting of olfactory neurons\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'am' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'am' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(am)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bc' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bc' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bc)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1281,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>pdm313</th><td>am</td><td>pdm3   </td><td>2526</td><td> 1368</td><td> 4691.1372</td><td>  57.753709</td><td>6.319114</td><td>77.06810</td></tr>\n",
       "\t<tr><th scope=row>CG1163513</th><td>am</td><td>CG11635</td><td> 186</td><td>   58</td><td>  346.1046</td><td>   2.234541</td><td>6.741503</td><td>56.89307</td></tr>\n",
       "\t<tr><th scope=row>Drgx13</th><td>am</td><td>Drgx   </td><td> 473</td><td>  592</td><td>  872.6917</td><td>  26.292682</td><td>4.998886</td><td>48.84402</td></tr>\n",
       "\t<tr><th scope=row>fipi13</th><td>am</td><td>fipi   </td><td>1147</td><td> 2284</td><td> 1994.3340</td><td> 106.570141</td><td>4.212557</td><td>46.17980</td></tr>\n",
       "\t<tr><th scope=row>CG4377813</th><td>am</td><td>CG43778</td><td> 241</td><td>  244</td><td>  352.6458</td><td>   9.573613</td><td>5.059680</td><td>42.83606</td></tr>\n",
       "\t<tr><th scope=row>CG1167013</th><td>am</td><td>CG11670</td><td> 115</td><td>  109</td><td>  205.2060</td><td>   3.800027</td><td>5.417886</td><td>41.65240</td></tr>\n",
       "\t<tr><th scope=row>CG3203213</th><td>am</td><td>CG32032</td><td> 213</td><td>  326</td><td>  386.1673</td><td>  13.007378</td><td>4.784967</td><td>41.13547</td></tr>\n",
       "\t<tr><th scope=row>prc13</th><td>am</td><td>prc    </td><td> 589</td><td> 2155</td><td> 1127.2325</td><td>  81.560291</td><td>3.771193</td><td>38.23933</td></tr>\n",
       "\t<tr><th scope=row>zyd13</th><td>am</td><td>zyd    </td><td> 411</td><td> 1135</td><td>  760.0021</td><td>  48.369396</td><td>3.944315</td><td>37.75402</td></tr>\n",
       "\t<tr><th scope=row>moody13</th><td>am</td><td>moody  </td><td> 546</td><td> 1783</td><td>  984.8754</td><td>  74.513632</td><td>3.705132</td><td>36.84851</td></tr>\n",
       "\t<tr><th scope=row>luna13</th><td>am</td><td>luna   </td><td>5442</td><td>33246</td><td> 9893.1554</td><td>1486.685411</td><td>2.733361</td><td>36.27816</td></tr>\n",
       "\t<tr><th scope=row>CG933613</th><td>am</td><td>CG9336 </td><td> 445</td><td> 1469</td><td>  836.8392</td><td>  62.200421</td><td>3.726944</td><td>36.19060</td></tr>\n",
       "\t<tr><th scope=row>Ntan113</th><td>am</td><td>Ntan1  </td><td> 863</td><td> 3386</td><td> 1589.8455</td><td> 150.050682</td><td>3.395782</td><td>36.11610</td></tr>\n",
       "\t<tr><th scope=row>Timp13</th><td>am</td><td>Timp   </td><td> 306</td><td>  886</td><td>  572.9080</td><td>  37.695194</td><td>3.888077</td><td>35.63297</td></tr>\n",
       "\t<tr><th scope=row>CG316825</th><td>am</td><td>CG3168 </td><td> 408</td><td> 2370</td><td>  768.6671</td><td>  61.242660</td><td>3.626383</td><td>34.77009</td></tr>\n",
       "\t<tr><th scope=row>CG1070213</th><td>am</td><td>CG10702</td><td> 163</td><td>  414</td><td>  306.9642</td><td>  15.728294</td><td>4.197708</td><td>34.70085</td></tr>\n",
       "\t<tr><th scope=row>Mdr6513</th><td>am</td><td>Mdr65  </td><td>  67</td><td>  103</td><td>  117.5403</td><td>   3.412272</td><td>4.735490</td><td>32.62390</td></tr>\n",
       "\t<tr><th scope=row>Sur13</th><td>am</td><td>Sur    </td><td> 195</td><td>  558</td><td>  360.9832</td><td>  25.053336</td><td>3.792392</td><td>32.23449</td></tr>\n",
       "\t<tr><th scope=row>trol13</th><td>am</td><td>trol   </td><td>6099</td><td>45540</td><td>11289.6513</td><td>2162.113029</td><td>2.383820</td><td>32.09299</td></tr>\n",
       "\t<tr><th scope=row>sev13</th><td>am</td><td>sev    </td><td>  73</td><td>  133</td><td>  134.7271</td><td>   4.832070</td><td>4.529888</td><td>32.09228</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tpdm313 & am & pdm3    & 2526 &  1368 &  4691.1372 &   57.753709 & 6.319114 & 77.06810\\\\\n",
       "\tCG1163513 & am & CG11635 &  186 &    58 &   346.1046 &    2.234541 & 6.741503 & 56.89307\\\\\n",
       "\tDrgx13 & am & Drgx    &  473 &   592 &   872.6917 &   26.292682 & 4.998886 & 48.84402\\\\\n",
       "\tfipi13 & am & fipi    & 1147 &  2284 &  1994.3340 &  106.570141 & 4.212557 & 46.17980\\\\\n",
       "\tCG4377813 & am & CG43778 &  241 &   244 &   352.6458 &    9.573613 & 5.059680 & 42.83606\\\\\n",
       "\tCG1167013 & am & CG11670 &  115 &   109 &   205.2060 &    3.800027 & 5.417886 & 41.65240\\\\\n",
       "\tCG3203213 & am & CG32032 &  213 &   326 &   386.1673 &   13.007378 & 4.784967 & 41.13547\\\\\n",
       "\tprc13 & am & prc     &  589 &  2155 &  1127.2325 &   81.560291 & 3.771193 & 38.23933\\\\\n",
       "\tzyd13 & am & zyd     &  411 &  1135 &   760.0021 &   48.369396 & 3.944315 & 37.75402\\\\\n",
       "\tmoody13 & am & moody   &  546 &  1783 &   984.8754 &   74.513632 & 3.705132 & 36.84851\\\\\n",
       "\tluna13 & am & luna    & 5442 & 33246 &  9893.1554 & 1486.685411 & 2.733361 & 36.27816\\\\\n",
       "\tCG933613 & am & CG9336  &  445 &  1469 &   836.8392 &   62.200421 & 3.726944 & 36.19060\\\\\n",
       "\tNtan113 & am & Ntan1   &  863 &  3386 &  1589.8455 &  150.050682 & 3.395782 & 36.11610\\\\\n",
       "\tTimp13 & am & Timp    &  306 &   886 &   572.9080 &   37.695194 & 3.888077 & 35.63297\\\\\n",
       "\tCG316825 & am & CG3168  &  408 &  2370 &   768.6671 &   61.242660 & 3.626383 & 34.77009\\\\\n",
       "\tCG1070213 & am & CG10702 &  163 &   414 &   306.9642 &   15.728294 & 4.197708 & 34.70085\\\\\n",
       "\tMdr6513 & am & Mdr65   &   67 &   103 &   117.5403 &    3.412272 & 4.735490 & 32.62390\\\\\n",
       "\tSur13 & am & Sur     &  195 &   558 &   360.9832 &   25.053336 & 3.792392 & 32.23449\\\\\n",
       "\ttrol13 & am & trol    & 6099 & 45540 & 11289.6513 & 2162.113029 & 2.383820 & 32.09299\\\\\n",
       "\tsev13 & am & sev     &   73 &   133 &   134.7271 &    4.832070 & 4.529888 & 32.09228\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| pdm313 | am | pdm3    | 2526 |  1368 |  4691.1372 |   57.753709 | 6.319114 | 77.06810 |\n",
       "| CG1163513 | am | CG11635 |  186 |    58 |   346.1046 |    2.234541 | 6.741503 | 56.89307 |\n",
       "| Drgx13 | am | Drgx    |  473 |   592 |   872.6917 |   26.292682 | 4.998886 | 48.84402 |\n",
       "| fipi13 | am | fipi    | 1147 |  2284 |  1994.3340 |  106.570141 | 4.212557 | 46.17980 |\n",
       "| CG4377813 | am | CG43778 |  241 |   244 |   352.6458 |    9.573613 | 5.059680 | 42.83606 |\n",
       "| CG1167013 | am | CG11670 |  115 |   109 |   205.2060 |    3.800027 | 5.417886 | 41.65240 |\n",
       "| CG3203213 | am | CG32032 |  213 |   326 |   386.1673 |   13.007378 | 4.784967 | 41.13547 |\n",
       "| prc13 | am | prc     |  589 |  2155 |  1127.2325 |   81.560291 | 3.771193 | 38.23933 |\n",
       "| zyd13 | am | zyd     |  411 |  1135 |   760.0021 |   48.369396 | 3.944315 | 37.75402 |\n",
       "| moody13 | am | moody   |  546 |  1783 |   984.8754 |   74.513632 | 3.705132 | 36.84851 |\n",
       "| luna13 | am | luna    | 5442 | 33246 |  9893.1554 | 1486.685411 | 2.733361 | 36.27816 |\n",
       "| CG933613 | am | CG9336  |  445 |  1469 |   836.8392 |   62.200421 | 3.726944 | 36.19060 |\n",
       "| Ntan113 | am | Ntan1   |  863 |  3386 |  1589.8455 |  150.050682 | 3.395782 | 36.11610 |\n",
       "| Timp13 | am | Timp    |  306 |   886 |   572.9080 |   37.695194 | 3.888077 | 35.63297 |\n",
       "| CG316825 | am | CG3168  |  408 |  2370 |   768.6671 |   61.242660 | 3.626383 | 34.77009 |\n",
       "| CG1070213 | am | CG10702 |  163 |   414 |   306.9642 |   15.728294 | 4.197708 | 34.70085 |\n",
       "| Mdr6513 | am | Mdr65   |   67 |   103 |   117.5403 |    3.412272 | 4.735490 | 32.62390 |\n",
       "| Sur13 | am | Sur     |  195 |   558 |   360.9832 |   25.053336 | 3.792392 | 32.23449 |\n",
       "| trol13 | am | trol    | 6099 | 45540 | 11289.6513 | 2162.113029 | 2.383820 | 32.09299 |\n",
       "| sev13 | am | sev     |   73 |   133 |   134.7271 |    4.832070 | 4.529888 | 32.09228 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "pdm313    am          pdm3    2526               1368        4691.1372     \n",
       "CG1163513 am          CG11635  186                 58         346.1046     \n",
       "Drgx13    am          Drgx     473                592         872.6917     \n",
       "fipi13    am          fipi    1147               2284        1994.3340     \n",
       "CG4377813 am          CG43778  241                244         352.6458     \n",
       "CG1167013 am          CG11670  115                109         205.2060     \n",
       "CG3203213 am          CG32032  213                326         386.1673     \n",
       "prc13     am          prc      589               2155        1127.2325     \n",
       "zyd13     am          zyd      411               1135         760.0021     \n",
       "moody13   am          moody    546               1783         984.8754     \n",
       "luna13    am          luna    5442              33246        9893.1554     \n",
       "CG933613  am          CG9336   445               1469         836.8392     \n",
       "Ntan113   am          Ntan1    863               3386        1589.8455     \n",
       "Timp13    am          Timp     306                886         572.9080     \n",
       "CG316825  am          CG3168   408               2370         768.6671     \n",
       "CG1070213 am          CG10702  163                414         306.9642     \n",
       "Mdr6513   am          Mdr65     67                103         117.5403     \n",
       "Sur13     am          Sur      195                558         360.9832     \n",
       "trol13    am          trol    6099              45540       11289.6513     \n",
       "sev13     am          sev       73                133         134.7271     \n",
       "          other.tpm   log2.ratio heuristic.score\n",
       "pdm313      57.753709 6.319114   77.06810       \n",
       "CG1163513    2.234541 6.741503   56.89307       \n",
       "Drgx13      26.292682 4.998886   48.84402       \n",
       "fipi13     106.570141 4.212557   46.17980       \n",
       "CG4377813    9.573613 5.059680   42.83606       \n",
       "CG1167013    3.800027 5.417886   41.65240       \n",
       "CG3203213   13.007378 4.784967   41.13547       \n",
       "prc13       81.560291 3.771193   38.23933       \n",
       "zyd13       48.369396 3.944315   37.75402       \n",
       "moody13     74.513632 3.705132   36.84851       \n",
       "luna13    1486.685411 2.733361   36.27816       \n",
       "CG933613    62.200421 3.726944   36.19060       \n",
       "Ntan113    150.050682 3.395782   36.11610       \n",
       "Timp13      37.695194 3.888077   35.63297       \n",
       "CG316825    61.242660 3.626383   34.77009       \n",
       "CG1070213   15.728294 4.197708   34.70085       \n",
       "Mdr6513      3.412272 4.735490   32.62390       \n",
       "Sur13       25.053336 3.792392   32.23449       \n",
       "trol13    2162.113029 2.383820   32.09299       \n",
       "sev13        4.832070 4.529888   32.09228       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"am\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"pdm3\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"pdm3\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"pdm3\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Drgx\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"fipi\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ab' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ab' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1291,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Tracheal_system_2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Tracheal_system_2\", df_sub300$cluster %in% an)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'an' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'an' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(an)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "# 63 needs to be split into tracheal system with this and something else!\n",
    "clus = c(63)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1286,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Gasp14</th><td>an</td><td>Gasp    </td><td>6637</td><td>16886</td><td>10797.0949</td><td>676.031956</td><td>3.995275</td><td>53.53065</td></tr>\n",
       "\t<tr><th scope=row>CG1200914</th><td>an</td><td>CG12009 </td><td>2732</td><td> 5432</td><td> 4415.8814</td><td>210.039506</td><td>4.387116</td><td>53.12277</td></tr>\n",
       "\t<tr><th scope=row>wat14</th><td>an</td><td>wat     </td><td>1189</td><td> 1810</td><td> 1911.6972</td><td> 77.414342</td><td>4.607593</td><td>50.22918</td></tr>\n",
       "\t<tr><th scope=row>Osi1514</th><td>an</td><td>Osi15   </td><td>5594</td><td>18190</td><td> 8954.5910</td><td>704.462065</td><td>3.665987</td><td>48.12918</td></tr>\n",
       "\t<tr><th scope=row>Cht514</th><td>an</td><td>Cht5    </td><td> 725</td><td> 1060</td><td> 1191.3066</td><td> 45.882300</td><td>4.667358</td><td>47.69824</td></tr>\n",
       "\t<tr><th scope=row>Spn43Aa14</th><td>an</td><td>Spn43Aa </td><td>2692</td><td> 7049</td><td> 4456.2354</td><td>302.007359</td><td>3.878401</td><td>47.01372</td></tr>\n",
       "\t<tr><th scope=row>CG1327214</th><td>an</td><td>CG13272 </td><td> 653</td><td> 1165</td><td> 1042.3299</td><td> 49.389527</td><td>4.370544</td><td>43.82336</td></tr>\n",
       "\t<tr><th scope=row>CG1178614</th><td>an</td><td>CG11786 </td><td> 386</td><td>  529</td><td>  627.8403</td><td> 26.457797</td><td>4.515110</td><td>41.97494</td></tr>\n",
       "\t<tr><th scope=row>CG1367614</th><td>an</td><td>CG13676 </td><td> 664</td><td> 2015</td><td> 1107.1511</td><td> 81.282257</td><td>3.750127</td><td>37.92856</td></tr>\n",
       "\t<tr><th scope=row>Cyp313b114</th><td>an</td><td>Cyp313b1</td><td> 119</td><td>  187</td><td>  202.3016</td><td>  6.512551</td><td>4.751061</td><td>36.42865</td></tr>\n",
       "\t<tr><th scope=row>CG669614</th><td>an</td><td>CG6696  </td><td> 169</td><td>  325</td><td>  302.3376</td><td> 13.198244</td><td>4.412376</td><td>36.37907</td></tr>\n",
       "\t<tr><th scope=row>obst-A14</th><td>an</td><td>obst-A  </td><td>1617</td><td> 7168</td><td> 2627.2179</td><td>285.899200</td><td>3.194920</td><td>36.29387</td></tr>\n",
       "\t<tr><th scope=row>apn14</th><td>an</td><td>apn     </td><td> 620</td><td> 2215</td><td>  999.1396</td><td> 83.723942</td><td>3.559845</td><td>35.47736</td></tr>\n",
       "\t<tr><th scope=row>hiro14</th><td>an</td><td>hiro    </td><td> 150</td><td>  367</td><td>  237.2182</td><td>  9.804465</td><td>4.456515</td><td>35.18927</td></tr>\n",
       "\t<tr><th scope=row>obst-B14</th><td>an</td><td>obst-B  </td><td> 208</td><td>  458</td><td>  351.2666</td><td> 20.452895</td><td>4.033322</td><td>34.12402</td></tr>\n",
       "\t<tr><th scope=row>CG981214</th><td>an</td><td>CG9812  </td><td> 213</td><td>  583</td><td>  356.6797</td><td> 22.329211</td><td>3.934420</td><td>33.37381</td></tr>\n",
       "\t<tr><th scope=row>reb14</th><td>an</td><td>reb     </td><td> 150</td><td>  313</td><td>  240.6082</td><td> 12.234338</td><td>4.184328</td><td>33.12534</td></tr>\n",
       "\t<tr><th scope=row>CG201614</th><td>an</td><td>CG2016  </td><td> 875</td><td> 3615</td><td> 1420.6199</td><td>159.458790</td><td>3.146246</td><td>32.95164</td></tr>\n",
       "\t<tr><th scope=row>trh14</th><td>an</td><td>trh     </td><td> 842</td><td> 4018</td><td> 1371.7531</td><td>159.243416</td><td>3.097684</td><td>32.28671</td></tr>\n",
       "\t<tr><th scope=row>CG1093614</th><td>an</td><td>CG10936 </td><td>1374</td><td> 6753</td><td> 2247.5416</td><td>308.161489</td><td>2.861915</td><td>31.86678</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tGasp14 & an & Gasp     & 6637 & 16886 & 10797.0949 & 676.031956 & 3.995275 & 53.53065\\\\\n",
       "\tCG1200914 & an & CG12009  & 2732 &  5432 &  4415.8814 & 210.039506 & 4.387116 & 53.12277\\\\\n",
       "\twat14 & an & wat      & 1189 &  1810 &  1911.6972 &  77.414342 & 4.607593 & 50.22918\\\\\n",
       "\tOsi1514 & an & Osi15    & 5594 & 18190 &  8954.5910 & 704.462065 & 3.665987 & 48.12918\\\\\n",
       "\tCht514 & an & Cht5     &  725 &  1060 &  1191.3066 &  45.882300 & 4.667358 & 47.69824\\\\\n",
       "\tSpn43Aa14 & an & Spn43Aa  & 2692 &  7049 &  4456.2354 & 302.007359 & 3.878401 & 47.01372\\\\\n",
       "\tCG1327214 & an & CG13272  &  653 &  1165 &  1042.3299 &  49.389527 & 4.370544 & 43.82336\\\\\n",
       "\tCG1178614 & an & CG11786  &  386 &   529 &   627.8403 &  26.457797 & 4.515110 & 41.97494\\\\\n",
       "\tCG1367614 & an & CG13676  &  664 &  2015 &  1107.1511 &  81.282257 & 3.750127 & 37.92856\\\\\n",
       "\tCyp313b114 & an & Cyp313b1 &  119 &   187 &   202.3016 &   6.512551 & 4.751061 & 36.42865\\\\\n",
       "\tCG669614 & an & CG6696   &  169 &   325 &   302.3376 &  13.198244 & 4.412376 & 36.37907\\\\\n",
       "\tobst-A14 & an & obst-A   & 1617 &  7168 &  2627.2179 & 285.899200 & 3.194920 & 36.29387\\\\\n",
       "\tapn14 & an & apn      &  620 &  2215 &   999.1396 &  83.723942 & 3.559845 & 35.47736\\\\\n",
       "\thiro14 & an & hiro     &  150 &   367 &   237.2182 &   9.804465 & 4.456515 & 35.18927\\\\\n",
       "\tobst-B14 & an & obst-B   &  208 &   458 &   351.2666 &  20.452895 & 4.033322 & 34.12402\\\\\n",
       "\tCG981214 & an & CG9812   &  213 &   583 &   356.6797 &  22.329211 & 3.934420 & 33.37381\\\\\n",
       "\treb14 & an & reb      &  150 &   313 &   240.6082 &  12.234338 & 4.184328 & 33.12534\\\\\n",
       "\tCG201614 & an & CG2016   &  875 &  3615 &  1420.6199 & 159.458790 & 3.146246 & 32.95164\\\\\n",
       "\ttrh14 & an & trh      &  842 &  4018 &  1371.7531 & 159.243416 & 3.097684 & 32.28671\\\\\n",
       "\tCG1093614 & an & CG10936  & 1374 &  6753 &  2247.5416 & 308.161489 & 2.861915 & 31.86678\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Gasp14 | an | Gasp     | 6637 | 16886 | 10797.0949 | 676.031956 | 3.995275 | 53.53065 |\n",
       "| CG1200914 | an | CG12009  | 2732 |  5432 |  4415.8814 | 210.039506 | 4.387116 | 53.12277 |\n",
       "| wat14 | an | wat      | 1189 |  1810 |  1911.6972 |  77.414342 | 4.607593 | 50.22918 |\n",
       "| Osi1514 | an | Osi15    | 5594 | 18190 |  8954.5910 | 704.462065 | 3.665987 | 48.12918 |\n",
       "| Cht514 | an | Cht5     |  725 |  1060 |  1191.3066 |  45.882300 | 4.667358 | 47.69824 |\n",
       "| Spn43Aa14 | an | Spn43Aa  | 2692 |  7049 |  4456.2354 | 302.007359 | 3.878401 | 47.01372 |\n",
       "| CG1327214 | an | CG13272  |  653 |  1165 |  1042.3299 |  49.389527 | 4.370544 | 43.82336 |\n",
       "| CG1178614 | an | CG11786  |  386 |   529 |   627.8403 |  26.457797 | 4.515110 | 41.97494 |\n",
       "| CG1367614 | an | CG13676  |  664 |  2015 |  1107.1511 |  81.282257 | 3.750127 | 37.92856 |\n",
       "| Cyp313b114 | an | Cyp313b1 |  119 |   187 |   202.3016 |   6.512551 | 4.751061 | 36.42865 |\n",
       "| CG669614 | an | CG6696   |  169 |   325 |   302.3376 |  13.198244 | 4.412376 | 36.37907 |\n",
       "| obst-A14 | an | obst-A   | 1617 |  7168 |  2627.2179 | 285.899200 | 3.194920 | 36.29387 |\n",
       "| apn14 | an | apn      |  620 |  2215 |   999.1396 |  83.723942 | 3.559845 | 35.47736 |\n",
       "| hiro14 | an | hiro     |  150 |   367 |   237.2182 |   9.804465 | 4.456515 | 35.18927 |\n",
       "| obst-B14 | an | obst-B   |  208 |   458 |   351.2666 |  20.452895 | 4.033322 | 34.12402 |\n",
       "| CG981214 | an | CG9812   |  213 |   583 |   356.6797 |  22.329211 | 3.934420 | 33.37381 |\n",
       "| reb14 | an | reb      |  150 |   313 |   240.6082 |  12.234338 | 4.184328 | 33.12534 |\n",
       "| CG201614 | an | CG2016   |  875 |  3615 |  1420.6199 | 159.458790 | 3.146246 | 32.95164 |\n",
       "| trh14 | an | trh      |  842 |  4018 |  1371.7531 | 159.243416 | 3.097684 | 32.28671 |\n",
       "| CG1093614 | an | CG10936  | 1374 |  6753 |  2247.5416 | 308.161489 | 2.861915 | 31.86678 |\n",
       "\n"
      ],
      "text/plain": [
       "           cell.type.2 gene     cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "Gasp14     an          Gasp     6637              16886       10797.0949     \n",
       "CG1200914  an          CG12009  2732               5432        4415.8814     \n",
       "wat14      an          wat      1189               1810        1911.6972     \n",
       "Osi1514    an          Osi15    5594              18190        8954.5910     \n",
       "Cht514     an          Cht5      725               1060        1191.3066     \n",
       "Spn43Aa14  an          Spn43Aa  2692               7049        4456.2354     \n",
       "CG1327214  an          CG13272   653               1165        1042.3299     \n",
       "CG1178614  an          CG11786   386                529         627.8403     \n",
       "CG1367614  an          CG13676   664               2015        1107.1511     \n",
       "Cyp313b114 an          Cyp313b1  119                187         202.3016     \n",
       "CG669614   an          CG6696    169                325         302.3376     \n",
       "obst-A14   an          obst-A   1617               7168        2627.2179     \n",
       "apn14      an          apn       620               2215         999.1396     \n",
       "hiro14     an          hiro      150                367         237.2182     \n",
       "obst-B14   an          obst-B    208                458         351.2666     \n",
       "CG981214   an          CG9812    213                583         356.6797     \n",
       "reb14      an          reb       150                313         240.6082     \n",
       "CG201614   an          CG2016    875               3615        1420.6199     \n",
       "trh14      an          trh       842               4018        1371.7531     \n",
       "CG1093614  an          CG10936  1374               6753        2247.5416     \n",
       "           other.tpm  log2.ratio heuristic.score\n",
       "Gasp14     676.031956 3.995275   53.53065       \n",
       "CG1200914  210.039506 4.387116   53.12277       \n",
       "wat14       77.414342 4.607593   50.22918       \n",
       "Osi1514    704.462065 3.665987   48.12918       \n",
       "Cht514      45.882300 4.667358   47.69824       \n",
       "Spn43Aa14  302.007359 3.878401   47.01372       \n",
       "CG1327214   49.389527 4.370544   43.82336       \n",
       "CG1178614   26.457797 4.515110   41.97494       \n",
       "CG1367614   81.282257 3.750127   37.92856       \n",
       "Cyp313b114   6.512551 4.751061   36.42865       \n",
       "CG669614    13.198244 4.412376   36.37907       \n",
       "obst-A14   285.899200 3.194920   36.29387       \n",
       "apn14       83.723942 3.559845   35.47736       \n",
       "hiro14       9.804465 4.456515   35.18927       \n",
       "obst-B14    20.452895 4.033322   34.12402       \n",
       "CG981214    22.329211 3.934420   33.37381       \n",
       "reb14       12.234338 4.184328   33.12534       \n",
       "CG201614   159.458790 3.146246   32.95164       \n",
       "trh14      159.243416 3.097684   32.28671       \n",
       "CG1093614  308.161489 2.861915   31.86678       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"an\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Gasp\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Gasp\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Gasp\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"wat\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Osi15\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ab' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ab' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type ao\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## already assigned as chordotonal organ by Shendure paper, but this is adult antenna organ so more likel\n",
    "## some kind of touch neuron in larva, based on nompC and Tektin-C (microtubule associated)\n",
    "#df_sub300 = set.cell.type(df_sub300, \"ao\", df_sub300$cluster %in% ao)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ao' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ao' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ao)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bc' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bc' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bc)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1322,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG3103615</th><td>ao</td><td>CG31036 </td><td>276</td><td> 254</td><td>1487.4286</td><td> 9.9684644</td><td>7.083315</td><td>74.65512</td></tr>\n",
       "\t<tr><th scope=row>nompC15</th><td>ao</td><td>nompC   </td><td>192</td><td> 135</td><td>1021.9035</td><td> 5.0475182</td><td>7.400700</td><td>73.99556</td></tr>\n",
       "\t<tr><th scope=row>CG4333815</th><td>ao</td><td>CG43338 </td><td>135</td><td> 122</td><td> 716.5711</td><td> 3.8692931</td><td>7.201254</td><td>68.31813</td></tr>\n",
       "\t<tr><th scope=row>Tektin-C15</th><td>ao</td><td>Tektin-C</td><td>156</td><td> 169</td><td> 796.6092</td><td> 5.8833210</td><td>6.854624</td><td>66.07541</td></tr>\n",
       "\t<tr><th scope=row>CG949215</th><td>ao</td><td>CG9492  </td><td> 89</td><td>  67</td><td> 494.9308</td><td> 2.3213741</td><td>7.219303</td><td>64.64160</td></tr>\n",
       "\t<tr><th scope=row>CG4510515</th><td>ao</td><td>CG45105 </td><td>224</td><td> 459</td><td>1136.7793</td><td>16.4365020</td><td>6.026698</td><td>61.18307</td></tr>\n",
       "\t<tr><th scope=row>Dhc98D15</th><td>ao</td><td>Dhc98D  </td><td> 49</td><td>   9</td><td> 259.0329</td><td> 0.3557092</td><td>7.577944</td><td>60.79444</td></tr>\n",
       "\t<tr><th scope=row>CG3186915</th><td>ao</td><td>CG31869 </td><td>323</td><td> 926</td><td>1765.0474</td><td>41.8134116</td><td>5.365500</td><td>57.87394</td></tr>\n",
       "\t<tr><th scope=row>CG723615</th><td>ao</td><td>CG7236  </td><td> 83</td><td> 106</td><td> 433.4727</td><td> 3.5942727</td><td>6.559961</td><td>57.48574</td></tr>\n",
       "\t<tr><th scope=row>pHCl-115</th><td>ao</td><td>pHCl-1  </td><td>383</td><td>1266</td><td>2002.1074</td><td>54.6499640</td><td>5.168995</td><td>56.69366</td></tr>\n",
       "\t<tr><th scope=row>Dhc93AB15</th><td>ao</td><td>Dhc93AB </td><td> 95</td><td> 149</td><td> 543.4302</td><td> 6.2326008</td><td>6.231436</td><td>56.63505</td></tr>\n",
       "\t<tr><th scope=row>CG1434215</th><td>ao</td><td>CG14342 </td><td> 61</td><td>  49</td><td> 307.6379</td><td> 1.7663832</td><td>6.797088</td><td>56.21037</td></tr>\n",
       "\t<tr><th scope=row>CG1745015</th><td>ao</td><td>CG17450 </td><td> 40</td><td>  23</td><td> 231.5275</td><td> 0.6735817</td><td>7.112101</td><td>55.91005</td></tr>\n",
       "\t<tr><th scope=row>dtr15</th><td>ao</td><td>dtr     </td><td> 50</td><td>  41</td><td> 264.4511</td><td> 1.1836836</td><td>6.920093</td><td>55.72268</td></tr>\n",
       "\t<tr><th scope=row>eys15</th><td>ao</td><td>eys     </td><td>359</td><td>1172</td><td>1852.3015</td><td>53.1994119</td><td>5.094898</td><td>55.30961</td></tr>\n",
       "\t<tr><th scope=row>CG1033915</th><td>ao</td><td>CG10339 </td><td> 76</td><td> 131</td><td> 439.2233</td><td> 4.6043536</td><td>6.292263</td><td>55.25923</td></tr>\n",
       "\t<tr><th scope=row>Erk715</th><td>ao</td><td>Erk7    </td><td> 50</td><td>  50</td><td> 292.9121</td><td> 1.9359507</td><td>6.640496</td><td>54.44703</td></tr>\n",
       "\t<tr><th scope=row>CG1537315</th><td>ao</td><td>CG15373 </td><td> 33</td><td>   9</td><td> 190.8323</td><td> 0.4792366</td><td>7.011309</td><td>53.17168</td></tr>\n",
       "\t<tr><th scope=row>Crg-115</th><td>ao</td><td>Crg-1   </td><td> 57</td><td>  38</td><td> 299.4138</td><td> 2.4282371</td><td>6.448530</td><td>53.07661</td></tr>\n",
       "\t<tr><th scope=row>CG774215</th><td>ao</td><td>CG7742  </td><td> 36</td><td>  24</td><td> 207.1713</td><td> 0.9062100</td><td>6.763973</td><td>52.09360</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG3103615 & ao & CG31036  & 276 &  254 & 1487.4286 &  9.9684644 & 7.083315 & 74.65512\\\\\n",
       "\tnompC15 & ao & nompC    & 192 &  135 & 1021.9035 &  5.0475182 & 7.400700 & 73.99556\\\\\n",
       "\tCG4333815 & ao & CG43338  & 135 &  122 &  716.5711 &  3.8692931 & 7.201254 & 68.31813\\\\\n",
       "\tTektin-C15 & ao & Tektin-C & 156 &  169 &  796.6092 &  5.8833210 & 6.854624 & 66.07541\\\\\n",
       "\tCG949215 & ao & CG9492   &  89 &   67 &  494.9308 &  2.3213741 & 7.219303 & 64.64160\\\\\n",
       "\tCG4510515 & ao & CG45105  & 224 &  459 & 1136.7793 & 16.4365020 & 6.026698 & 61.18307\\\\\n",
       "\tDhc98D15 & ao & Dhc98D   &  49 &    9 &  259.0329 &  0.3557092 & 7.577944 & 60.79444\\\\\n",
       "\tCG3186915 & ao & CG31869  & 323 &  926 & 1765.0474 & 41.8134116 & 5.365500 & 57.87394\\\\\n",
       "\tCG723615 & ao & CG7236   &  83 &  106 &  433.4727 &  3.5942727 & 6.559961 & 57.48574\\\\\n",
       "\tpHCl-115 & ao & pHCl-1   & 383 & 1266 & 2002.1074 & 54.6499640 & 5.168995 & 56.69366\\\\\n",
       "\tDhc93AB15 & ao & Dhc93AB  &  95 &  149 &  543.4302 &  6.2326008 & 6.231436 & 56.63505\\\\\n",
       "\tCG1434215 & ao & CG14342  &  61 &   49 &  307.6379 &  1.7663832 & 6.797088 & 56.21037\\\\\n",
       "\tCG1745015 & ao & CG17450  &  40 &   23 &  231.5275 &  0.6735817 & 7.112101 & 55.91005\\\\\n",
       "\tdtr15 & ao & dtr      &  50 &   41 &  264.4511 &  1.1836836 & 6.920093 & 55.72268\\\\\n",
       "\teys15 & ao & eys      & 359 & 1172 & 1852.3015 & 53.1994119 & 5.094898 & 55.30961\\\\\n",
       "\tCG1033915 & ao & CG10339  &  76 &  131 &  439.2233 &  4.6043536 & 6.292263 & 55.25923\\\\\n",
       "\tErk715 & ao & Erk7     &  50 &   50 &  292.9121 &  1.9359507 & 6.640496 & 54.44703\\\\\n",
       "\tCG1537315 & ao & CG15373  &  33 &    9 &  190.8323 &  0.4792366 & 7.011309 & 53.17168\\\\\n",
       "\tCrg-115 & ao & Crg-1    &  57 &   38 &  299.4138 &  2.4282371 & 6.448530 & 53.07661\\\\\n",
       "\tCG774215 & ao & CG7742   &  36 &   24 &  207.1713 &  0.9062100 & 6.763973 & 52.09360\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG3103615 | ao | CG31036  | 276 |  254 | 1487.4286 |  9.9684644 | 7.083315 | 74.65512 |\n",
       "| nompC15 | ao | nompC    | 192 |  135 | 1021.9035 |  5.0475182 | 7.400700 | 73.99556 |\n",
       "| CG4333815 | ao | CG43338  | 135 |  122 |  716.5711 |  3.8692931 | 7.201254 | 68.31813 |\n",
       "| Tektin-C15 | ao | Tektin-C | 156 |  169 |  796.6092 |  5.8833210 | 6.854624 | 66.07541 |\n",
       "| CG949215 | ao | CG9492   |  89 |   67 |  494.9308 |  2.3213741 | 7.219303 | 64.64160 |\n",
       "| CG4510515 | ao | CG45105  | 224 |  459 | 1136.7793 | 16.4365020 | 6.026698 | 61.18307 |\n",
       "| Dhc98D15 | ao | Dhc98D   |  49 |    9 |  259.0329 |  0.3557092 | 7.577944 | 60.79444 |\n",
       "| CG3186915 | ao | CG31869  | 323 |  926 | 1765.0474 | 41.8134116 | 5.365500 | 57.87394 |\n",
       "| CG723615 | ao | CG7236   |  83 |  106 |  433.4727 |  3.5942727 | 6.559961 | 57.48574 |\n",
       "| pHCl-115 | ao | pHCl-1   | 383 | 1266 | 2002.1074 | 54.6499640 | 5.168995 | 56.69366 |\n",
       "| Dhc93AB15 | ao | Dhc93AB  |  95 |  149 |  543.4302 |  6.2326008 | 6.231436 | 56.63505 |\n",
       "| CG1434215 | ao | CG14342  |  61 |   49 |  307.6379 |  1.7663832 | 6.797088 | 56.21037 |\n",
       "| CG1745015 | ao | CG17450  |  40 |   23 |  231.5275 |  0.6735817 | 7.112101 | 55.91005 |\n",
       "| dtr15 | ao | dtr      |  50 |   41 |  264.4511 |  1.1836836 | 6.920093 | 55.72268 |\n",
       "| eys15 | ao | eys      | 359 | 1172 | 1852.3015 | 53.1994119 | 5.094898 | 55.30961 |\n",
       "| CG1033915 | ao | CG10339  |  76 |  131 |  439.2233 |  4.6043536 | 6.292263 | 55.25923 |\n",
       "| Erk715 | ao | Erk7     |  50 |   50 |  292.9121 |  1.9359507 | 6.640496 | 54.44703 |\n",
       "| CG1537315 | ao | CG15373  |  33 |    9 |  190.8323 |  0.4792366 | 7.011309 | 53.17168 |\n",
       "| Crg-115 | ao | Crg-1    |  57 |   38 |  299.4138 |  2.4282371 | 6.448530 | 53.07661 |\n",
       "| CG774215 | ao | CG7742   |  36 |   24 |  207.1713 |  0.9062100 | 6.763973 | 52.09360 |\n",
       "\n"
      ],
      "text/plain": [
       "           cell.type.2 gene     cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "CG3103615  ao          CG31036  276                254        1487.4286      \n",
       "nompC15    ao          nompC    192                135        1021.9035      \n",
       "CG4333815  ao          CG43338  135                122         716.5711      \n",
       "Tektin-C15 ao          Tektin-C 156                169         796.6092      \n",
       "CG949215   ao          CG9492    89                 67         494.9308      \n",
       "CG4510515  ao          CG45105  224                459        1136.7793      \n",
       "Dhc98D15   ao          Dhc98D    49                  9         259.0329      \n",
       "CG3186915  ao          CG31869  323                926        1765.0474      \n",
       "CG723615   ao          CG7236    83                106         433.4727      \n",
       "pHCl-115   ao          pHCl-1   383               1266        2002.1074      \n",
       "Dhc93AB15  ao          Dhc93AB   95                149         543.4302      \n",
       "CG1434215  ao          CG14342   61                 49         307.6379      \n",
       "CG1745015  ao          CG17450   40                 23         231.5275      \n",
       "dtr15      ao          dtr       50                 41         264.4511      \n",
       "eys15      ao          eys      359               1172        1852.3015      \n",
       "CG1033915  ao          CG10339   76                131         439.2233      \n",
       "Erk715     ao          Erk7      50                 50         292.9121      \n",
       "CG1537315  ao          CG15373   33                  9         190.8323      \n",
       "Crg-115    ao          Crg-1     57                 38         299.4138      \n",
       "CG774215   ao          CG7742    36                 24         207.1713      \n",
       "           other.tpm  log2.ratio heuristic.score\n",
       "CG3103615   9.9684644 7.083315   74.65512       \n",
       "nompC15     5.0475182 7.400700   73.99556       \n",
       "CG4333815   3.8692931 7.201254   68.31813       \n",
       "Tektin-C15  5.8833210 6.854624   66.07541       \n",
       "CG949215    2.3213741 7.219303   64.64160       \n",
       "CG4510515  16.4365020 6.026698   61.18307       \n",
       "Dhc98D15    0.3557092 7.577944   60.79444       \n",
       "CG3186915  41.8134116 5.365500   57.87394       \n",
       "CG723615    3.5942727 6.559961   57.48574       \n",
       "pHCl-115   54.6499640 5.168995   56.69366       \n",
       "Dhc93AB15   6.2326008 6.231436   56.63505       \n",
       "CG1434215   1.7663832 6.797088   56.21037       \n",
       "CG1745015   0.6735817 7.112101   55.91005       \n",
       "dtr15       1.1836836 6.920093   55.72268       \n",
       "eys15      53.1994119 5.094898   55.30961       \n",
       "CG1033915   4.6043536 6.292263   55.25923       \n",
       "Erk715      1.9359507 6.640496   54.44703       \n",
       "CG1537315   0.4792366 7.011309   53.17168       \n",
       "Crg-115     2.4282371 6.448530   53.07661       \n",
       "CG774215    0.9062100 6.763973   52.09360       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"ao\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG31036\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG31036\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG31036\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"nompC\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Tektin-C\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ab' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ab' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in set.cell.type(df_sub300, \"Oenocytes\", df_sub300$cluster %in% : could not find function \"set.cell.type\"\n",
     "output_type": "error",
     "traceback": [
      "Error in set.cell.type(df_sub300, \"Oenocytes\", df_sub300$cluster %in% : could not find function \"set.cell.type\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Oenocytes\", df_sub300$cluster %in% c(121) & \n",
    "                          df_sub300$umap_2 > 0.2*df_sub300$umap_1 -3.18 &\n",
    "              df_sub300$umap_1 >= 2\n",
    "                         )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1387,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Analpad_large_intestine\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Analpad_large_intestine\", df_sub300$cluster %in% c(121, 109) & \n",
    "                          (df_sub300$umap_2 <= 0.2*df_sub300$umap_1 -3.18 | df_sub300$umap_2 <= 0.6*df_sub300$umap_1 -4.12)&\n",
    "              df_sub300$umap_2 >  0.2*df_sub300$umap_1 -3.4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1399,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type large_intestine_hindgut_analpad\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"large_intestine_hindgut_analpad\", df_sub300$cluster %in% c( 109) & \n",
    "(df_sub300$umap_2 < 0.34*df_sub300$umap_1 -3.)&\n",
    "              df_sub300$umap_2 >=  0.34*df_sub300$umap_1 -3.3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ap' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ap' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ap)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(121)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(109)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "44. Gutierrez E, Wiggins D, Fielding B, Gould AP. Specialized hepatocyte-like cells regulate Drosophila lipid metabolism. Nature. 2007;445:275–80.\n",
    "[ 44. Shows that larval oenocytes express numerous lipid metabolic genes including Cyp4g1, accumulate lipid droplets upon fasting, are regulated by the fat body, and are required for molting. ] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1327,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>bond16</th><td>ap</td><td>bond   </td><td>  685</td><td>  434</td><td> 1009.17178</td><td>  13.977733</td><td>6.074209</td><td>60.62294</td></tr>\n",
       "\t<tr><th scope=row>Tsp74F16</th><td>ap</td><td>Tsp74F </td><td> 1076</td><td> 1007</td><td> 1547.08340</td><td>  36.284416</td><td>5.374834</td><td>56.95318</td></tr>\n",
       "\t<tr><th scope=row>FASN316</th><td>ap</td><td>FASN3  </td><td> 1025</td><td> 1945</td><td> 1444.83775</td><td>  47.719576</td><td>4.890262</td><td>51.33646</td></tr>\n",
       "\t<tr><th scope=row>CG215716</th><td>ap</td><td>CG2157 </td><td>  411</td><td>  389</td><td>  606.79636</td><td>  13.531372</td><td>5.383970</td><td>49.78796</td></tr>\n",
       "\t<tr><th scope=row>CG1321616</th><td>ap</td><td>CG13216</td><td>  172</td><td>  158</td><td>  257.57742</td><td>   5.995284</td><td>5.202480</td><td>41.69503</td></tr>\n",
       "\t<tr><th scope=row>CG717916</th><td>ap</td><td>CG7179 </td><td>  192</td><td>  215</td><td>  255.02886</td><td>   7.916008</td><td>4.838119</td><td>38.70574</td></tr>\n",
       "\t<tr><th scope=row>LpR116</th><td>ap</td><td>LpR1   </td><td>  464</td><td> 1000</td><td>  623.33351</td><td>  40.240412</td><td>3.917874</td><td>36.38205</td></tr>\n",
       "\t<tr><th scope=row>CG1860916</th><td>ap</td><td>CG18609</td><td>   80</td><td>   61</td><td>  104.17412</td><td>   1.996308</td><td>5.119667</td><td>34.38694</td></tr>\n",
       "\t<tr><th scope=row>Cyp4g117</th><td>ap</td><td>Cyp4g1 </td><td>  420</td><td> 1961</td><td>  567.44220</td><td>  45.520371</td><td>3.608539</td><td>33.02127</td></tr>\n",
       "\t<tr><th scope=row>Osi1416</th><td>ap</td><td>Osi14  </td><td> 1867</td><td> 8374</td><td> 2592.04528</td><td> 348.699894</td><td>2.889902</td><td>32.77273</td></tr>\n",
       "\t<tr><th scope=row>CG772016</th><td>ap</td><td>CG7720 </td><td> 1303</td><td> 4667</td><td> 1775.33532</td><td> 217.081380</td><td>3.025153</td><td>32.65558</td></tr>\n",
       "\t<tr><th scope=row>Sox100B16</th><td>ap</td><td>Sox100B</td><td>   83</td><td>  113</td><td>  120.12370</td><td>   4.427152</td><td>4.468182</td><td>30.92133</td></tr>\n",
       "\t<tr><th scope=row>CG1225616</th><td>ap</td><td>CG12256</td><td>   70</td><td>   76</td><td>   99.15118</td><td>   2.957089</td><td>4.647119</td><td>30.88492</td></tr>\n",
       "\t<tr><th scope=row>CG1590516</th><td>ap</td><td>CG15905</td><td>  106</td><td>  157</td><td>  156.67099</td><td>   7.501096</td><td>4.203945</td><td>30.69205</td></tr>\n",
       "\t<tr><th scope=row>CG1328516</th><td>ap</td><td>CG13285</td><td>   89</td><td>  159</td><td>  123.97786</td><td>   5.701337</td><td>4.209490</td><td>29.32132</td></tr>\n",
       "\t<tr><th scope=row>CG1756216</th><td>ap</td><td>CG17562</td><td>   54</td><td>   68</td><td>   77.52357</td><td>   2.180688</td><td>4.607224</td><td>29.00273</td></tr>\n",
       "\t<tr><th scope=row>CG791016</th><td>ap</td><td>CG7910 </td><td>   63</td><td>  141</td><td>   87.91781</td><td>   3.353226</td><td>4.335999</td><td>28.07299</td></tr>\n",
       "\t<tr><th scope=row>swi216</th><td>ap</td><td>swi2   </td><td>   66</td><td>   96</td><td>   91.15069</td><td>   3.644386</td><td>4.294694</td><td>28.02684</td></tr>\n",
       "\t<tr><th scope=row>Osi616</th><td>ap</td><td>Osi6   </td><td>11165</td><td>91464</td><td>15665.11185</td><td>3966.497369</td><td>1.981254</td><td>27.60949</td></tr>\n",
       "\t<tr><th scope=row>CG926916</th><td>ap</td><td>CG9269 </td><td>   49</td><td>   83</td><td>   73.10411</td><td>   2.503289</td><td>4.383171</td><td>27.22598</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tbond16 & ap & bond    &   685 &   434 &  1009.17178 &   13.977733 & 6.074209 & 60.62294\\\\\n",
       "\tTsp74F16 & ap & Tsp74F  &  1076 &  1007 &  1547.08340 &   36.284416 & 5.374834 & 56.95318\\\\\n",
       "\tFASN316 & ap & FASN3   &  1025 &  1945 &  1444.83775 &   47.719576 & 4.890262 & 51.33646\\\\\n",
       "\tCG215716 & ap & CG2157  &   411 &   389 &   606.79636 &   13.531372 & 5.383970 & 49.78796\\\\\n",
       "\tCG1321616 & ap & CG13216 &   172 &   158 &   257.57742 &    5.995284 & 5.202480 & 41.69503\\\\\n",
       "\tCG717916 & ap & CG7179  &   192 &   215 &   255.02886 &    7.916008 & 4.838119 & 38.70574\\\\\n",
       "\tLpR116 & ap & LpR1    &   464 &  1000 &   623.33351 &   40.240412 & 3.917874 & 36.38205\\\\\n",
       "\tCG1860916 & ap & CG18609 &    80 &    61 &   104.17412 &    1.996308 & 5.119667 & 34.38694\\\\\n",
       "\tCyp4g117 & ap & Cyp4g1  &   420 &  1961 &   567.44220 &   45.520371 & 3.608539 & 33.02127\\\\\n",
       "\tOsi1416 & ap & Osi14   &  1867 &  8374 &  2592.04528 &  348.699894 & 2.889902 & 32.77273\\\\\n",
       "\tCG772016 & ap & CG7720  &  1303 &  4667 &  1775.33532 &  217.081380 & 3.025153 & 32.65558\\\\\n",
       "\tSox100B16 & ap & Sox100B &    83 &   113 &   120.12370 &    4.427152 & 4.468182 & 30.92133\\\\\n",
       "\tCG1225616 & ap & CG12256 &    70 &    76 &    99.15118 &    2.957089 & 4.647119 & 30.88492\\\\\n",
       "\tCG1590516 & ap & CG15905 &   106 &   157 &   156.67099 &    7.501096 & 4.203945 & 30.69205\\\\\n",
       "\tCG1328516 & ap & CG13285 &    89 &   159 &   123.97786 &    5.701337 & 4.209490 & 29.32132\\\\\n",
       "\tCG1756216 & ap & CG17562 &    54 &    68 &    77.52357 &    2.180688 & 4.607224 & 29.00273\\\\\n",
       "\tCG791016 & ap & CG7910  &    63 &   141 &    87.91781 &    3.353226 & 4.335999 & 28.07299\\\\\n",
       "\tswi216 & ap & swi2    &    66 &    96 &    91.15069 &    3.644386 & 4.294694 & 28.02684\\\\\n",
       "\tOsi616 & ap & Osi6    & 11165 & 91464 & 15665.11185 & 3966.497369 & 1.981254 & 27.60949\\\\\n",
       "\tCG926916 & ap & CG9269  &    49 &    83 &    73.10411 &    2.503289 & 4.383171 & 27.22598\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| bond16 | ap | bond    |   685 |   434 |  1009.17178 |   13.977733 | 6.074209 | 60.62294 |\n",
       "| Tsp74F16 | ap | Tsp74F  |  1076 |  1007 |  1547.08340 |   36.284416 | 5.374834 | 56.95318 |\n",
       "| FASN316 | ap | FASN3   |  1025 |  1945 |  1444.83775 |   47.719576 | 4.890262 | 51.33646 |\n",
       "| CG215716 | ap | CG2157  |   411 |   389 |   606.79636 |   13.531372 | 5.383970 | 49.78796 |\n",
       "| CG1321616 | ap | CG13216 |   172 |   158 |   257.57742 |    5.995284 | 5.202480 | 41.69503 |\n",
       "| CG717916 | ap | CG7179  |   192 |   215 |   255.02886 |    7.916008 | 4.838119 | 38.70574 |\n",
       "| LpR116 | ap | LpR1    |   464 |  1000 |   623.33351 |   40.240412 | 3.917874 | 36.38205 |\n",
       "| CG1860916 | ap | CG18609 |    80 |    61 |   104.17412 |    1.996308 | 5.119667 | 34.38694 |\n",
       "| Cyp4g117 | ap | Cyp4g1  |   420 |  1961 |   567.44220 |   45.520371 | 3.608539 | 33.02127 |\n",
       "| Osi1416 | ap | Osi14   |  1867 |  8374 |  2592.04528 |  348.699894 | 2.889902 | 32.77273 |\n",
       "| CG772016 | ap | CG7720  |  1303 |  4667 |  1775.33532 |  217.081380 | 3.025153 | 32.65558 |\n",
       "| Sox100B16 | ap | Sox100B |    83 |   113 |   120.12370 |    4.427152 | 4.468182 | 30.92133 |\n",
       "| CG1225616 | ap | CG12256 |    70 |    76 |    99.15118 |    2.957089 | 4.647119 | 30.88492 |\n",
       "| CG1590516 | ap | CG15905 |   106 |   157 |   156.67099 |    7.501096 | 4.203945 | 30.69205 |\n",
       "| CG1328516 | ap | CG13285 |    89 |   159 |   123.97786 |    5.701337 | 4.209490 | 29.32132 |\n",
       "| CG1756216 | ap | CG17562 |    54 |    68 |    77.52357 |    2.180688 | 4.607224 | 29.00273 |\n",
       "| CG791016 | ap | CG7910  |    63 |   141 |    87.91781 |    3.353226 | 4.335999 | 28.07299 |\n",
       "| swi216 | ap | swi2    |    66 |    96 |    91.15069 |    3.644386 | 4.294694 | 28.02684 |\n",
       "| Osi616 | ap | Osi6    | 11165 | 91464 | 15665.11185 | 3966.497369 | 1.981254 | 27.60949 |\n",
       "| CG926916 | ap | CG9269  |    49 |    83 |    73.10411 |    2.503289 | 4.383171 | 27.22598 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "bond16    ap          bond      685               434        1009.17178    \n",
       "Tsp74F16  ap          Tsp74F   1076              1007        1547.08340    \n",
       "FASN316   ap          FASN3    1025              1945        1444.83775    \n",
       "CG215716  ap          CG2157    411               389         606.79636    \n",
       "CG1321616 ap          CG13216   172               158         257.57742    \n",
       "CG717916  ap          CG7179    192               215         255.02886    \n",
       "LpR116    ap          LpR1      464              1000         623.33351    \n",
       "CG1860916 ap          CG18609    80                61         104.17412    \n",
       "Cyp4g117  ap          Cyp4g1    420              1961         567.44220    \n",
       "Osi1416   ap          Osi14    1867              8374        2592.04528    \n",
       "CG772016  ap          CG7720   1303              4667        1775.33532    \n",
       "Sox100B16 ap          Sox100B    83               113         120.12370    \n",
       "CG1225616 ap          CG12256    70                76          99.15118    \n",
       "CG1590516 ap          CG15905   106               157         156.67099    \n",
       "CG1328516 ap          CG13285    89               159         123.97786    \n",
       "CG1756216 ap          CG17562    54                68          77.52357    \n",
       "CG791016  ap          CG7910     63               141          87.91781    \n",
       "swi216    ap          swi2       66                96          91.15069    \n",
       "Osi616    ap          Osi6    11165             91464       15665.11185    \n",
       "CG926916  ap          CG9269     49                83          73.10411    \n",
       "          other.tpm   log2.ratio heuristic.score\n",
       "bond16      13.977733 6.074209   60.62294       \n",
       "Tsp74F16    36.284416 5.374834   56.95318       \n",
       "FASN316     47.719576 4.890262   51.33646       \n",
       "CG215716    13.531372 5.383970   49.78796       \n",
       "CG1321616    5.995284 5.202480   41.69503       \n",
       "CG717916     7.916008 4.838119   38.70574       \n",
       "LpR116      40.240412 3.917874   36.38205       \n",
       "CG1860916    1.996308 5.119667   34.38694       \n",
       "Cyp4g117    45.520371 3.608539   33.02127       \n",
       "Osi1416    348.699894 2.889902   32.77273       \n",
       "CG772016   217.081380 3.025153   32.65558       \n",
       "Sox100B16    4.427152 4.468182   30.92133       \n",
       "CG1225616    2.957089 4.647119   30.88492       \n",
       "CG1590516    7.501096 4.203945   30.69205       \n",
       "CG1328516    5.701337 4.209490   29.32132       \n",
       "CG1756216    2.180688 4.607224   29.00273       \n",
       "CG791016     3.353226 4.335999   28.07299       \n",
       "swi216       3.644386 4.294694   28.02684       \n",
       "Osi616    3966.497369 1.981254   27.60949       \n",
       "CG926916     2.503289 4.383171   27.22598       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"ap\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"LpR1\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"LpR1\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"LpR1\"), \n",
    "                   size_pos = 2, size_neg = 0.9, stroke = 0.8, limit = 2)+\n",
    "xlim(0, 4) + ylim(-5, -1)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Cyp4g1\"), \n",
    "                   size_pos = 2, size_neg = 0.9, stroke = 0.8, limit = 2)+\n",
    "xlim(0, 4) + ylim(-5, -1)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG17562\"), \n",
    "                   size_pos = 2, size_neg = 0.9, stroke = 0.8, limit = 1)+\n",
    "xlim(0, 4) + ylim(-5, -1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(121)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.2, intercept = -3.18) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(0, 4) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(121)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "              df_sub300$umap_2 > 0.2*umap_1 -3.18 &\n",
    "              df_sub300$umap_1 >= 2), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.2, intercept = -3.18) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(0, 4) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(121, 109)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "              (df_sub300$umap_2 <= 0.2*umap_1 -3.18 | df_sub300$umap_2 <= 0.6*umap_1 -4.12)&\n",
    "              df_sub300$umap_2 >  0.2*umap_1 -3.4), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.8), stroke = 0.5) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.2, intercept = -3.18) +\n",
    "geom_abline(slope = 0.2, intercept = -3.4) +\n",
    "geom_abline(slope = 0.6, intercept = -4.12) +\n",
    "xlim(0, 4) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1384,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 734\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Tsp74F</th><td>Tsp74F </td><td>1208</td><td> 958</td><td>4738.9502</td><td> 35.696188</td><td>Set 1</td><td>7.012793</td><td>85.63081</td></tr>\n",
       "\t<tr><th scope=row>bond</th><td>bond   </td><td> 683</td><td> 424</td><td>2768.9956</td><td> 15.689418</td><td>Set 1</td><td>7.374285</td><td>84.32988</td></tr>\n",
       "\t<tr><th scope=row>CG2157</th><td>CG2157 </td><td> 445</td><td> 444</td><td>1807.1751</td><td> 14.975233</td><td>Set 1</td><td>6.821756</td><td>73.81357</td></tr>\n",
       "\t<tr><th scope=row>CG7179</th><td>CG7179 </td><td> 203</td><td> 202</td><td> 757.0008</td><td>  6.949523</td><td>Set 1</td><td>6.573283</td><td>62.88039</td></tr>\n",
       "\t<tr><th scope=row>CG13216</th><td>CG13216</td><td> 184</td><td> 204</td><td> 726.9638</td><td>  8.578737</td><td>Set 1</td><td>6.245904</td><td>59.38433</td></tr>\n",
       "\t<tr><th scope=row>CG13285</th><td>CG13285</td><td>  99</td><td> 116</td><td> 382.8871</td><td>  3.293653</td><td>Set 1</td><td>6.478570</td><td>55.61553</td></tr>\n",
       "\t<tr><th scope=row>CG15905</th><td>CG15905</td><td> 125</td><td> 198</td><td> 501.0075</td><td>  8.290548</td><td>Set 1</td><td>5.752925</td><td>51.61274</td></tr>\n",
       "\t<tr><th scope=row>Osi14</th><td>Osi14  </td><td>1449</td><td>8386</td><td>5573.5923</td><td>349.057598</td><td>Set 1</td><td>3.992943</td><td>49.69078</td></tr>\n",
       "\t<tr><th scope=row>swi2</th><td>swi2   </td><td>  74</td><td>  89</td><td> 270.3415</td><td>  3.447563</td><td>Set 1</td><td>5.925624</td><td>47.90254</td></tr>\n",
       "\t<tr><th scope=row>Sox100B</th><td>Sox100B</td><td>  85</td><td> 153</td><td> 341.4333</td><td>  6.334802</td><td>Set 1</td><td>5.540702</td><td>46.65093</td></tr>\n",
       "\t<tr><th scope=row>CG5070</th><td>CG5070 </td><td>  59</td><td> 143</td><td> 232.9561</td><td>  4.592190</td><td>Set 1</td><td>5.380501</td><td>42.34505</td></tr>\n",
       "\t<tr><th scope=row>CG9269</th><td>CG9269 </td><td>  50</td><td> 119</td><td> 203.4062</td><td>  4.051214</td><td>Set 1</td><td>5.331590</td><td>40.92153</td></tr>\n",
       "\t<tr><th scope=row>chas</th><td>chas   </td><td> 334</td><td>2305</td><td>1311.4086</td><td> 89.539778</td><td>Set 1</td><td>3.856422</td><td>39.94482</td></tr>\n",
       "\t<tr><th scope=row>ATP8A</th><td>ATP8A  </td><td> 283</td><td>1775</td><td>1123.0707</td><td> 78.678288</td><td>Set 1</td><td>3.817118</td><td>38.68465</td></tr>\n",
       "\t<tr><th scope=row>wat</th><td>wat    </td><td> 290</td><td>2007</td><td>1150.3354</td><td> 87.633827</td><td>Set 1</td><td>3.698053</td><td>37.60585</td></tr>\n",
       "\t<tr><th scope=row>CG14626</th><td>CG14626</td><td>  92</td><td> 463</td><td> 371.1021</td><td> 19.315196</td><td>Set 1</td><td>4.191185</td><td>35.79085</td></tr>\n",
       "\t<tr><th scope=row>ple</th><td>ple    </td><td> 165</td><td>1251</td><td> 633.2409</td><td> 44.294073</td><td>Set 1</td><td>3.805360</td><td>35.42367</td></tr>\n",
       "\t<tr><th scope=row>rgn</th><td>rgn    </td><td> 148</td><td>1059</td><td> 567.9503</td><td> 40.286128</td><td>Set 1</td><td>3.782036</td><td>34.61379</td></tr>\n",
       "\t<tr><th scope=row>CG43373</th><td>CG43373</td><td>  82</td><td> 440</td><td> 350.2266</td><td> 20.361771</td><td>Set 1</td><td>4.035185</td><td>34.12257</td></tr>\n",
       "\t<tr><th scope=row>CG7720</th><td>CG7720 </td><td> 499</td><td>4515</td><td>1858.5358</td><td>213.216132</td><td>Set 1</td><td>3.117027</td><td>33.85318</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tTsp74F & Tsp74F  & 1208 &  958 & 4738.9502 &  35.696188 & Set 1 & 7.012793 & 85.63081\\\\\n",
       "\tbond & bond    &  683 &  424 & 2768.9956 &  15.689418 & Set 1 & 7.374285 & 84.32988\\\\\n",
       "\tCG2157 & CG2157  &  445 &  444 & 1807.1751 &  14.975233 & Set 1 & 6.821756 & 73.81357\\\\\n",
       "\tCG7179 & CG7179  &  203 &  202 &  757.0008 &   6.949523 & Set 1 & 6.573283 & 62.88039\\\\\n",
       "\tCG13216 & CG13216 &  184 &  204 &  726.9638 &   8.578737 & Set 1 & 6.245904 & 59.38433\\\\\n",
       "\tCG13285 & CG13285 &   99 &  116 &  382.8871 &   3.293653 & Set 1 & 6.478570 & 55.61553\\\\\n",
       "\tCG15905 & CG15905 &  125 &  198 &  501.0075 &   8.290548 & Set 1 & 5.752925 & 51.61274\\\\\n",
       "\tOsi14 & Osi14   & 1449 & 8386 & 5573.5923 & 349.057598 & Set 1 & 3.992943 & 49.69078\\\\\n",
       "\tswi2 & swi2    &   74 &   89 &  270.3415 &   3.447563 & Set 1 & 5.925624 & 47.90254\\\\\n",
       "\tSox100B & Sox100B &   85 &  153 &  341.4333 &   6.334802 & Set 1 & 5.540702 & 46.65093\\\\\n",
       "\tCG5070 & CG5070  &   59 &  143 &  232.9561 &   4.592190 & Set 1 & 5.380501 & 42.34505\\\\\n",
       "\tCG9269 & CG9269  &   50 &  119 &  203.4062 &   4.051214 & Set 1 & 5.331590 & 40.92153\\\\\n",
       "\tchas & chas    &  334 & 2305 & 1311.4086 &  89.539778 & Set 1 & 3.856422 & 39.94482\\\\\n",
       "\tATP8A & ATP8A   &  283 & 1775 & 1123.0707 &  78.678288 & Set 1 & 3.817118 & 38.68465\\\\\n",
       "\twat & wat     &  290 & 2007 & 1150.3354 &  87.633827 & Set 1 & 3.698053 & 37.60585\\\\\n",
       "\tCG14626 & CG14626 &   92 &  463 &  371.1021 &  19.315196 & Set 1 & 4.191185 & 35.79085\\\\\n",
       "\tple & ple     &  165 & 1251 &  633.2409 &  44.294073 & Set 1 & 3.805360 & 35.42367\\\\\n",
       "\trgn & rgn     &  148 & 1059 &  567.9503 &  40.286128 & Set 1 & 3.782036 & 34.61379\\\\\n",
       "\tCG43373 & CG43373 &   82 &  440 &  350.2266 &  20.361771 & Set 1 & 4.035185 & 34.12257\\\\\n",
       "\tCG7720 & CG7720  &  499 & 4515 & 1858.5358 & 213.216132 & Set 1 & 3.117027 & 33.85318\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Tsp74F | Tsp74F  | 1208 |  958 | 4738.9502 |  35.696188 | Set 1 | 7.012793 | 85.63081 |\n",
       "| bond | bond    |  683 |  424 | 2768.9956 |  15.689418 | Set 1 | 7.374285 | 84.32988 |\n",
       "| CG2157 | CG2157  |  445 |  444 | 1807.1751 |  14.975233 | Set 1 | 6.821756 | 73.81357 |\n",
       "| CG7179 | CG7179  |  203 |  202 |  757.0008 |   6.949523 | Set 1 | 6.573283 | 62.88039 |\n",
       "| CG13216 | CG13216 |  184 |  204 |  726.9638 |   8.578737 | Set 1 | 6.245904 | 59.38433 |\n",
       "| CG13285 | CG13285 |   99 |  116 |  382.8871 |   3.293653 | Set 1 | 6.478570 | 55.61553 |\n",
       "| CG15905 | CG15905 |  125 |  198 |  501.0075 |   8.290548 | Set 1 | 5.752925 | 51.61274 |\n",
       "| Osi14 | Osi14   | 1449 | 8386 | 5573.5923 | 349.057598 | Set 1 | 3.992943 | 49.69078 |\n",
       "| swi2 | swi2    |   74 |   89 |  270.3415 |   3.447563 | Set 1 | 5.925624 | 47.90254 |\n",
       "| Sox100B | Sox100B |   85 |  153 |  341.4333 |   6.334802 | Set 1 | 5.540702 | 46.65093 |\n",
       "| CG5070 | CG5070  |   59 |  143 |  232.9561 |   4.592190 | Set 1 | 5.380501 | 42.34505 |\n",
       "| CG9269 | CG9269  |   50 |  119 |  203.4062 |   4.051214 | Set 1 | 5.331590 | 40.92153 |\n",
       "| chas | chas    |  334 | 2305 | 1311.4086 |  89.539778 | Set 1 | 3.856422 | 39.94482 |\n",
       "| ATP8A | ATP8A   |  283 | 1775 | 1123.0707 |  78.678288 | Set 1 | 3.817118 | 38.68465 |\n",
       "| wat | wat     |  290 | 2007 | 1150.3354 |  87.633827 | Set 1 | 3.698053 | 37.60585 |\n",
       "| CG14626 | CG14626 |   92 |  463 |  371.1021 |  19.315196 | Set 1 | 4.191185 | 35.79085 |\n",
       "| ple | ple     |  165 | 1251 |  633.2409 |  44.294073 | Set 1 | 3.805360 | 35.42367 |\n",
       "| rgn | rgn     |  148 | 1059 |  567.9503 |  40.286128 | Set 1 | 3.782036 | 34.61379 |\n",
       "| CG43373 | CG43373 |   82 |  440 |  350.2266 |  20.361771 | Set 1 | 4.035185 | 34.12257 |\n",
       "| CG7720 | CG7720  |  499 | 4515 | 1858.5358 | 213.216132 | Set 1 | 3.117027 | 33.85318 |\n",
       "\n"
      ],
      "text/plain": [
       "        gene    set.1.n.umi set.2.n.umi set.1.tpm set.2.tpm  higher.expr\n",
       "Tsp74F  Tsp74F  1208         958        4738.9502  35.696188 Set 1      \n",
       "bond    bond     683         424        2768.9956  15.689418 Set 1      \n",
       "CG2157  CG2157   445         444        1807.1751  14.975233 Set 1      \n",
       "CG7179  CG7179   203         202         757.0008   6.949523 Set 1      \n",
       "CG13216 CG13216  184         204         726.9638   8.578737 Set 1      \n",
       "CG13285 CG13285   99         116         382.8871   3.293653 Set 1      \n",
       "CG15905 CG15905  125         198         501.0075   8.290548 Set 1      \n",
       "Osi14   Osi14   1449        8386        5573.5923 349.057598 Set 1      \n",
       "swi2    swi2      74          89         270.3415   3.447563 Set 1      \n",
       "Sox100B Sox100B   85         153         341.4333   6.334802 Set 1      \n",
       "CG5070  CG5070    59         143         232.9561   4.592190 Set 1      \n",
       "CG9269  CG9269    50         119         203.4062   4.051214 Set 1      \n",
       "chas    chas     334        2305        1311.4086  89.539778 Set 1      \n",
       "ATP8A   ATP8A    283        1775        1123.0707  78.678288 Set 1      \n",
       "wat     wat      290        2007        1150.3354  87.633827 Set 1      \n",
       "CG14626 CG14626   92         463         371.1021  19.315196 Set 1      \n",
       "ple     ple      165        1251         633.2409  44.294073 Set 1      \n",
       "rgn     rgn      148        1059         567.9503  40.286128 Set 1      \n",
       "CG43373 CG43373   82         440         350.2266  20.361771 Set 1      \n",
       "CG7720  CG7720   499        4515        1858.5358 213.216132 Set 1      \n",
       "        log2.ratio heuristic.score\n",
       "Tsp74F  7.012793   85.63081       \n",
       "bond    7.374285   84.32988       \n",
       "CG2157  6.821756   73.81357       \n",
       "CG7179  6.573283   62.88039       \n",
       "CG13216 6.245904   59.38433       \n",
       "CG13285 6.478570   55.61553       \n",
       "CG15905 5.752925   51.61274       \n",
       "Osi14   3.992943   49.69078       \n",
       "swi2    5.925624   47.90254       \n",
       "Sox100B 5.540702   46.65093       \n",
       "CG5070  5.380501   42.34505       \n",
       "CG9269  5.331590   40.92153       \n",
       "chas    3.856422   39.94482       \n",
       "ATP8A   3.817118   38.68465       \n",
       "wat     3.698053   37.60585       \n",
       "CG14626 4.191185   35.79085       \n",
       "ple     3.805360   35.42367       \n",
       "rgn     3.782036   34.61379       \n",
       "CG43373 4.035185   34.12257       \n",
       "CG7720  3.117027   33.85318       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "two.set.differential.gene.test.3a(cds_sub, \n",
    "                                  df_sub300$cluster %in% c(121, 109) & \n",
    "                                  (df_sub300$umap_2 <= 0.2*df_sub300$umap_1 -3.18 | \n",
    "                                   df_sub300$umap_2 <= 0.6*df_sub300$umap_1 -4.12) &\n",
    "                                      df_sub300$umap_2 >  0.2*df_sub300$umap_1 -3.4, \n",
    "                                 !(df_sub300$cluster %in% c(121, 109) & \n",
    "                                  (df_sub300$umap_2 <= 0.2*df_sub300$umap_1 -3.18 | \n",
    "                                   df_sub300$umap_2 <= 0.6*df_sub300$umap_1 -4.12) &\n",
    "                                      df_sub300$umap_2 >  0.2*df_sub300$umap_1 -3.4),\n",
    "                                  sample = 50000\n",
    "                                )  %>% filter(higher.expr == \"Set 1\")  %>% \n",
    "head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"bond\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"bond\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"bond\"), \n",
    "                   size_pos = 2, size_neg = 0.9, stroke = 0.8, limit = 2)+\n",
    "xlim(0, 4) + ylim(-5, -1)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Tsp74F\"), \n",
    "                   size_pos = 2, size_neg = 0.9, stroke = 0.8, limit = 2)+\n",
    "xlim(0, 4) + ylim(-5, -1)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG2157\"), \n",
    "                   size_pos = 2, size_neg = 0.9, stroke = 0.8, limit = 1)+\n",
    "xlim(0, 4) + ylim(-5, -1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c( 109)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "              (df_sub300$umap_2 < 0.34*umap_1 -3.)&\n",
    "              df_sub300$umap_2 >=  0.34*umap_1 -3.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.8), stroke = 0.5) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.34, intercept = -3) +\n",
    "geom_abline(slope = 0.34, intercept = -3.32) +\n",
    "geom_abline(slope = 0.6, intercept = -4.12) +\n",
    "xlim(0, 4) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1412,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Posterior_spiracle\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Posterior_spiracle\", df_sub300$cluster %in% c(aq, 154)  &\n",
    "                         df_sub300$umap_2 < 1*df_sub300$umap_1 -8.1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'aq' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'aq' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(aq)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(154)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1404,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Cad96Cb17</th><td>aq</td><td>Cad96Cb</td><td>1109</td><td>  638</td><td> 2632.65884</td><td>  30.3855886</td><td>6.390275</td><td>72.61175</td></tr>\n",
       "\t<tr><th scope=row>CG1757217</th><td>aq</td><td>CG17572</td><td> 310</td><td>   82</td><td>  738.16252</td><td>   2.8421750</td><td>7.585871</td><td>72.29144</td></tr>\n",
       "\t<tr><th scope=row>CG266317</th><td>aq</td><td>CG2663 </td><td>1424</td><td> 1961</td><td> 3366.44804</td><td>  75.1547758</td><td>5.466149</td><td>64.04927</td></tr>\n",
       "\t<tr><th scope=row>CG3423217</th><td>aq</td><td>CG34232</td><td> 599</td><td>  617</td><td> 1471.85722</td><td>  27.2318689</td><td>5.704169</td><td>60.03297</td></tr>\n",
       "\t<tr><th scope=row>nyo17</th><td>aq</td><td>nyo    </td><td>1815</td><td> 3918</td><td> 4267.94688</td><td> 174.7398252</td><td>4.602029</td><td>55.49893</td></tr>\n",
       "\t<tr><th scope=row>CG1281417</th><td>aq</td><td>CG12814</td><td>1332</td><td> 3709</td><td> 3074.37678</td><td> 162.5850432</td><td>4.232181</td><td>49.03637</td></tr>\n",
       "\t<tr><th scope=row>CG894517</th><td>aq</td><td>CG8945 </td><td>  91</td><td>   40</td><td>  202.02313</td><td>   1.6574789</td><td>6.248318</td><td>47.89649</td></tr>\n",
       "\t<tr><th scope=row>CG4252417</th><td>aq</td><td>CG42524</td><td> 785</td><td> 1901</td><td> 1911.83185</td><td>  93.7679068</td><td>4.334413</td><td>47.25158</td></tr>\n",
       "\t<tr><th scope=row>CG1364017</th><td>aq</td><td>CG13640</td><td>  60</td><td>   19</td><td>  137.47710</td><td>   0.5987852</td><td>6.426071</td><td>45.71188</td></tr>\n",
       "\t<tr><th scope=row>CG1190517</th><td>aq</td><td>CG11905</td><td>1311</td><td> 4819</td><td> 3125.97954</td><td> 207.6170517</td><td>3.905379</td><td>45.34362</td></tr>\n",
       "\t<tr><th scope=row>Oaz17</th><td>aq</td><td>Oaz    </td><td> 202</td><td>  336</td><td>  479.15042</td><td>  13.4238801</td><td>5.053947</td><td>45.01724</td></tr>\n",
       "\t<tr><th scope=row>IRSp5317</th><td>aq</td><td>IRSp53 </td><td>1421</td><td> 5719</td><td> 3407.11531</td><td> 267.0409522</td><td>3.668025</td><td>43.04339</td></tr>\n",
       "\t<tr><th scope=row>ru17</th><td>aq</td><td>ru     </td><td> 126</td><td>  178</td><td>  280.23915</td><td>   8.0353814</td><td>4.954929</td><td>40.31159</td></tr>\n",
       "\t<tr><th scope=row>ect17</th><td>aq</td><td>ect    </td><td>4559</td><td>30396</td><td>10497.38966</td><td>1308.5410886</td><td>3.002897</td><td>40.11234</td></tr>\n",
       "\t<tr><th scope=row>ImpE117</th><td>aq</td><td>ImpE1  </td><td>2249</td><td>11984</td><td> 5282.02953</td><td> 558.7249269</td><td>3.238302</td><td>40.04857</td></tr>\n",
       "\t<tr><th scope=row>CG4229617</th><td>aq</td><td>CG42296</td><td> 333</td><td> 1000</td><td>  792.03437</td><td>  44.1881563</td><td>4.131546</td><td>39.79191</td></tr>\n",
       "\t<tr><th scope=row>CCHa117</th><td>aq</td><td>CCHa1  </td><td>  55</td><td>   29</td><td>  119.49592</td><td>   1.2589563</td><td>5.725161</td><td>39.57713</td></tr>\n",
       "\t<tr><th scope=row>CG1336117</th><td>aq</td><td>CG13361</td><td>  45</td><td>   23</td><td>   98.61008</td><td>   0.6372567</td><td>5.912383</td><td>39.24770</td></tr>\n",
       "\t<tr><th scope=row>Osi1717</th><td>aq</td><td>Osi17  </td><td> 342</td><td> 1087</td><td>  804.55486</td><td>  48.7031076</td><td>4.016783</td><td>38.77737</td></tr>\n",
       "\t<tr><th scope=row>CG643117</th><td>aq</td><td>CG6431 </td><td>  37</td><td>   13</td><td>   98.01586</td><td>   0.7429561</td><td>5.813407</td><td>38.54049</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCad96Cb17 & aq & Cad96Cb & 1109 &   638 &  2632.65884 &   30.3855886 & 6.390275 & 72.61175\\\\\n",
       "\tCG1757217 & aq & CG17572 &  310 &    82 &   738.16252 &    2.8421750 & 7.585871 & 72.29144\\\\\n",
       "\tCG266317 & aq & CG2663  & 1424 &  1961 &  3366.44804 &   75.1547758 & 5.466149 & 64.04927\\\\\n",
       "\tCG3423217 & aq & CG34232 &  599 &   617 &  1471.85722 &   27.2318689 & 5.704169 & 60.03297\\\\\n",
       "\tnyo17 & aq & nyo     & 1815 &  3918 &  4267.94688 &  174.7398252 & 4.602029 & 55.49893\\\\\n",
       "\tCG1281417 & aq & CG12814 & 1332 &  3709 &  3074.37678 &  162.5850432 & 4.232181 & 49.03637\\\\\n",
       "\tCG894517 & aq & CG8945  &   91 &    40 &   202.02313 &    1.6574789 & 6.248318 & 47.89649\\\\\n",
       "\tCG4252417 & aq & CG42524 &  785 &  1901 &  1911.83185 &   93.7679068 & 4.334413 & 47.25158\\\\\n",
       "\tCG1364017 & aq & CG13640 &   60 &    19 &   137.47710 &    0.5987852 & 6.426071 & 45.71188\\\\\n",
       "\tCG1190517 & aq & CG11905 & 1311 &  4819 &  3125.97954 &  207.6170517 & 3.905379 & 45.34362\\\\\n",
       "\tOaz17 & aq & Oaz     &  202 &   336 &   479.15042 &   13.4238801 & 5.053947 & 45.01724\\\\\n",
       "\tIRSp5317 & aq & IRSp53  & 1421 &  5719 &  3407.11531 &  267.0409522 & 3.668025 & 43.04339\\\\\n",
       "\tru17 & aq & ru      &  126 &   178 &   280.23915 &    8.0353814 & 4.954929 & 40.31159\\\\\n",
       "\tect17 & aq & ect     & 4559 & 30396 & 10497.38966 & 1308.5410886 & 3.002897 & 40.11234\\\\\n",
       "\tImpE117 & aq & ImpE1   & 2249 & 11984 &  5282.02953 &  558.7249269 & 3.238302 & 40.04857\\\\\n",
       "\tCG4229617 & aq & CG42296 &  333 &  1000 &   792.03437 &   44.1881563 & 4.131546 & 39.79191\\\\\n",
       "\tCCHa117 & aq & CCHa1   &   55 &    29 &   119.49592 &    1.2589563 & 5.725161 & 39.57713\\\\\n",
       "\tCG1336117 & aq & CG13361 &   45 &    23 &    98.61008 &    0.6372567 & 5.912383 & 39.24770\\\\\n",
       "\tOsi1717 & aq & Osi17   &  342 &  1087 &   804.55486 &   48.7031076 & 4.016783 & 38.77737\\\\\n",
       "\tCG643117 & aq & CG6431  &   37 &    13 &    98.01586 &    0.7429561 & 5.813407 & 38.54049\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Cad96Cb17 | aq | Cad96Cb | 1109 |   638 |  2632.65884 |   30.3855886 | 6.390275 | 72.61175 |\n",
       "| CG1757217 | aq | CG17572 |  310 |    82 |   738.16252 |    2.8421750 | 7.585871 | 72.29144 |\n",
       "| CG266317 | aq | CG2663  | 1424 |  1961 |  3366.44804 |   75.1547758 | 5.466149 | 64.04927 |\n",
       "| CG3423217 | aq | CG34232 |  599 |   617 |  1471.85722 |   27.2318689 | 5.704169 | 60.03297 |\n",
       "| nyo17 | aq | nyo     | 1815 |  3918 |  4267.94688 |  174.7398252 | 4.602029 | 55.49893 |\n",
       "| CG1281417 | aq | CG12814 | 1332 |  3709 |  3074.37678 |  162.5850432 | 4.232181 | 49.03637 |\n",
       "| CG894517 | aq | CG8945  |   91 |    40 |   202.02313 |    1.6574789 | 6.248318 | 47.89649 |\n",
       "| CG4252417 | aq | CG42524 |  785 |  1901 |  1911.83185 |   93.7679068 | 4.334413 | 47.25158 |\n",
       "| CG1364017 | aq | CG13640 |   60 |    19 |   137.47710 |    0.5987852 | 6.426071 | 45.71188 |\n",
       "| CG1190517 | aq | CG11905 | 1311 |  4819 |  3125.97954 |  207.6170517 | 3.905379 | 45.34362 |\n",
       "| Oaz17 | aq | Oaz     |  202 |   336 |   479.15042 |   13.4238801 | 5.053947 | 45.01724 |\n",
       "| IRSp5317 | aq | IRSp53  | 1421 |  5719 |  3407.11531 |  267.0409522 | 3.668025 | 43.04339 |\n",
       "| ru17 | aq | ru      |  126 |   178 |   280.23915 |    8.0353814 | 4.954929 | 40.31159 |\n",
       "| ect17 | aq | ect     | 4559 | 30396 | 10497.38966 | 1308.5410886 | 3.002897 | 40.11234 |\n",
       "| ImpE117 | aq | ImpE1   | 2249 | 11984 |  5282.02953 |  558.7249269 | 3.238302 | 40.04857 |\n",
       "| CG4229617 | aq | CG42296 |  333 |  1000 |   792.03437 |   44.1881563 | 4.131546 | 39.79191 |\n",
       "| CCHa117 | aq | CCHa1   |   55 |    29 |   119.49592 |    1.2589563 | 5.725161 | 39.57713 |\n",
       "| CG1336117 | aq | CG13361 |   45 |    23 |    98.61008 |    0.6372567 | 5.912383 | 39.24770 |\n",
       "| Osi1717 | aq | Osi17   |  342 |  1087 |   804.55486 |   48.7031076 | 4.016783 | 38.77737 |\n",
       "| CG643117 | aq | CG6431  |   37 |    13 |    98.01586 |    0.7429561 | 5.813407 | 38.54049 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "Cad96Cb17 aq          Cad96Cb 1109                638        2632.65884    \n",
       "CG1757217 aq          CG17572  310                 82         738.16252    \n",
       "CG266317  aq          CG2663  1424               1961        3366.44804    \n",
       "CG3423217 aq          CG34232  599                617        1471.85722    \n",
       "nyo17     aq          nyo     1815               3918        4267.94688    \n",
       "CG1281417 aq          CG12814 1332               3709        3074.37678    \n",
       "CG894517  aq          CG8945    91                 40         202.02313    \n",
       "CG4252417 aq          CG42524  785               1901        1911.83185    \n",
       "CG1364017 aq          CG13640   60                 19         137.47710    \n",
       "CG1190517 aq          CG11905 1311               4819        3125.97954    \n",
       "Oaz17     aq          Oaz      202                336         479.15042    \n",
       "IRSp5317  aq          IRSp53  1421               5719        3407.11531    \n",
       "ru17      aq          ru       126                178         280.23915    \n",
       "ect17     aq          ect     4559              30396       10497.38966    \n",
       "ImpE117   aq          ImpE1   2249              11984        5282.02953    \n",
       "CG4229617 aq          CG42296  333               1000         792.03437    \n",
       "CCHa117   aq          CCHa1     55                 29         119.49592    \n",
       "CG1336117 aq          CG13361   45                 23          98.61008    \n",
       "Osi1717   aq          Osi17    342               1087         804.55486    \n",
       "CG643117  aq          CG6431    37                 13          98.01586    \n",
       "          other.tpm    log2.ratio heuristic.score\n",
       "Cad96Cb17   30.3855886 6.390275   72.61175       \n",
       "CG1757217    2.8421750 7.585871   72.29144       \n",
       "CG266317    75.1547758 5.466149   64.04927       \n",
       "CG3423217   27.2318689 5.704169   60.03297       \n",
       "nyo17      174.7398252 4.602029   55.49893       \n",
       "CG1281417  162.5850432 4.232181   49.03637       \n",
       "CG894517     1.6574789 6.248318   47.89649       \n",
       "CG4252417   93.7679068 4.334413   47.25158       \n",
       "CG1364017    0.5987852 6.426071   45.71188       \n",
       "CG1190517  207.6170517 3.905379   45.34362       \n",
       "Oaz17       13.4238801 5.053947   45.01724       \n",
       "IRSp5317   267.0409522 3.668025   43.04339       \n",
       "ru17         8.0353814 4.954929   40.31159       \n",
       "ect17     1308.5410886 3.002897   40.11234       \n",
       "ImpE117    558.7249269 3.238302   40.04857       \n",
       "CG4229617   44.1881563 4.131546   39.79191       \n",
       "CCHa117      1.2589563 5.725161   39.57713       \n",
       "CG1336117    0.6372567 5.912383   39.24770       \n",
       "Osi1717     48.7031076 4.016783   38.77737       \n",
       "CG643117     0.7429561 5.813407   38.54049       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"aq\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Cad96Cb\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Cad96Cb\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Cad96Cb\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG17572\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG2663\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'aq' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'aq' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(154, aq)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 < 1*umap_1 -8.1), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) #+\n",
    "#xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1420,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Peripheral_nervous_system_early\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Peripheral_nervous_system_early\", df_sub300$cluster %in% ar)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ar)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ar)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1417,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>nrm18</th><td>ar</td><td>nrm      </td><td> 6997</td><td> 4384</td><td>6261.00984</td><td> 170.412278</td><td>5.190853</td><td>65.46917</td></tr>\n",
       "\t<tr><th scope=row>meru18</th><td>ar</td><td>meru     </td><td> 2322</td><td> 1190</td><td>2083.51115</td><td>  47.513687</td><td>5.424481</td><td>59.80758</td></tr>\n",
       "\t<tr><th scope=row>sv18</th><td>ar</td><td>sv       </td><td> 3195</td><td> 2021</td><td>2757.70946</td><td>  85.460695</td><td>4.995282</td><td>57.09497</td></tr>\n",
       "\t<tr><th scope=row>ss18</th><td>ar</td><td>ss       </td><td> 1729</td><td> 2018</td><td>1592.92704</td><td>  88.495298</td><td>4.153725</td><td>44.18886</td></tr>\n",
       "\t<tr><th scope=row>sens18</th><td>ar</td><td>sens     </td><td>  930</td><td>  963</td><td> 807.35037</td><td>  41.475929</td><td>4.248477</td><td>41.03535</td></tr>\n",
       "\t<tr><th scope=row>sca18</th><td>ar</td><td>sca      </td><td> 2172</td><td> 4754</td><td>1857.82833</td><td> 192.542677</td><td>3.262894</td><td>35.43560</td></tr>\n",
       "\t<tr><th scope=row>pb18</th><td>ar</td><td>pb       </td><td>  762</td><td> 1373</td><td> 692.54164</td><td>  58.970166</td><td>3.529584</td><td>33.31164</td></tr>\n",
       "\t<tr><th scope=row>cpo18</th><td>ar</td><td>cpo      </td><td>10042</td><td>38399</td><td>8992.99078</td><td>1629.298230</td><td>2.463665</td><td>32.35961</td></tr>\n",
       "\t<tr><th scope=row>Rfx18</th><td>ar</td><td>Rfx      </td><td>  678</td><td> 1448</td><td> 615.29590</td><td>  60.815720</td><td>3.315235</td><td>30.72387</td></tr>\n",
       "\t<tr><th scope=row>loco18</th><td>ar</td><td>loco     </td><td> 2560</td><td> 8119</td><td>2247.19415</td><td> 350.912280</td><td>2.674837</td><td>29.78311</td></tr>\n",
       "\t<tr><th scope=row>vn18</th><td>ar</td><td>vn       </td><td> 2370</td><td> 8462</td><td>2090.20231</td><td> 359.103880</td><td>2.537158</td><td>27.98514</td></tr>\n",
       "\t<tr><th scope=row>fd3F18</th><td>ar</td><td>fd3F     </td><td>   89</td><td>   87</td><td>  83.46276</td><td>   3.380252</td><td>4.252047</td><td>27.21413</td></tr>\n",
       "\t<tr><th scope=row>Dll18</th><td>ar</td><td>Dll      </td><td>  437</td><td> 1158</td><td> 398.21130</td><td>  45.839704</td><td>3.087730</td><td>26.68110</td></tr>\n",
       "\t<tr><th scope=row>beat-IIIc18</th><td>ar</td><td>beat-IIIc</td><td> 3884</td><td>16478</td><td>3471.52053</td><td> 737.772015</td><td>2.232367</td><td>26.25658</td></tr>\n",
       "\t<tr><th scope=row>ct18</th><td>ar</td><td>ct       </td><td> 5614</td><td>26309</td><td>5046.42037</td><td>1150.202747</td><td>2.132118</td><td>26.22789</td></tr>\n",
       "\t<tr><th scope=row>Notum18</th><td>ar</td><td>Notum    </td><td> 1126</td><td> 3943</td><td> 978.75267</td><td> 166.265683</td><td>2.548803</td><td>25.32560</td></tr>\n",
       "\t<tr><th scope=row>unc18</th><td>ar</td><td>unc      </td><td>   62</td><td>   55</td><td>  58.44251</td><td>   2.246182</td><td>4.170202</td><td>24.57677</td></tr>\n",
       "\t<tr><th scope=row>Traf418</th><td>ar</td><td>Traf4    </td><td> 1655</td><td> 7063</td><td>1445.73602</td><td> 290.914448</td><td>2.308187</td><td>24.23270</td></tr>\n",
       "\t<tr><th scope=row>spdo18</th><td>ar</td><td>spdo     </td><td>  425</td><td> 1434</td><td> 386.48655</td><td>  54.083052</td><td>2.810738</td><td>24.16673</td></tr>\n",
       "\t<tr><th scope=row>neur18</th><td>ar</td><td>neur     </td><td> 1963</td><td> 8700</td><td>1715.26769</td><td> 364.304832</td><td>2.231261</td><td>23.97503</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tnrm18 & ar & nrm       &  6997 &  4384 & 6261.00984 &  170.412278 & 5.190853 & 65.46917\\\\\n",
       "\tmeru18 & ar & meru      &  2322 &  1190 & 2083.51115 &   47.513687 & 5.424481 & 59.80758\\\\\n",
       "\tsv18 & ar & sv        &  3195 &  2021 & 2757.70946 &   85.460695 & 4.995282 & 57.09497\\\\\n",
       "\tss18 & ar & ss        &  1729 &  2018 & 1592.92704 &   88.495298 & 4.153725 & 44.18886\\\\\n",
       "\tsens18 & ar & sens      &   930 &   963 &  807.35037 &   41.475929 & 4.248477 & 41.03535\\\\\n",
       "\tsca18 & ar & sca       &  2172 &  4754 & 1857.82833 &  192.542677 & 3.262894 & 35.43560\\\\\n",
       "\tpb18 & ar & pb        &   762 &  1373 &  692.54164 &   58.970166 & 3.529584 & 33.31164\\\\\n",
       "\tcpo18 & ar & cpo       & 10042 & 38399 & 8992.99078 & 1629.298230 & 2.463665 & 32.35961\\\\\n",
       "\tRfx18 & ar & Rfx       &   678 &  1448 &  615.29590 &   60.815720 & 3.315235 & 30.72387\\\\\n",
       "\tloco18 & ar & loco      &  2560 &  8119 & 2247.19415 &  350.912280 & 2.674837 & 29.78311\\\\\n",
       "\tvn18 & ar & vn        &  2370 &  8462 & 2090.20231 &  359.103880 & 2.537158 & 27.98514\\\\\n",
       "\tfd3F18 & ar & fd3F      &    89 &    87 &   83.46276 &    3.380252 & 4.252047 & 27.21413\\\\\n",
       "\tDll18 & ar & Dll       &   437 &  1158 &  398.21130 &   45.839704 & 3.087730 & 26.68110\\\\\n",
       "\tbeat-IIIc18 & ar & beat-IIIc &  3884 & 16478 & 3471.52053 &  737.772015 & 2.232367 & 26.25658\\\\\n",
       "\tct18 & ar & ct        &  5614 & 26309 & 5046.42037 & 1150.202747 & 2.132118 & 26.22789\\\\\n",
       "\tNotum18 & ar & Notum     &  1126 &  3943 &  978.75267 &  166.265683 & 2.548803 & 25.32560\\\\\n",
       "\tunc18 & ar & unc       &    62 &    55 &   58.44251 &    2.246182 & 4.170202 & 24.57677\\\\\n",
       "\tTraf418 & ar & Traf4     &  1655 &  7063 & 1445.73602 &  290.914448 & 2.308187 & 24.23270\\\\\n",
       "\tspdo18 & ar & spdo      &   425 &  1434 &  386.48655 &   54.083052 & 2.810738 & 24.16673\\\\\n",
       "\tneur18 & ar & neur      &  1963 &  8700 & 1715.26769 &  364.304832 & 2.231261 & 23.97503\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| nrm18 | ar | nrm       |  6997 |  4384 | 6261.00984 |  170.412278 | 5.190853 | 65.46917 |\n",
       "| meru18 | ar | meru      |  2322 |  1190 | 2083.51115 |   47.513687 | 5.424481 | 59.80758 |\n",
       "| sv18 | ar | sv        |  3195 |  2021 | 2757.70946 |   85.460695 | 4.995282 | 57.09497 |\n",
       "| ss18 | ar | ss        |  1729 |  2018 | 1592.92704 |   88.495298 | 4.153725 | 44.18886 |\n",
       "| sens18 | ar | sens      |   930 |   963 |  807.35037 |   41.475929 | 4.248477 | 41.03535 |\n",
       "| sca18 | ar | sca       |  2172 |  4754 | 1857.82833 |  192.542677 | 3.262894 | 35.43560 |\n",
       "| pb18 | ar | pb        |   762 |  1373 |  692.54164 |   58.970166 | 3.529584 | 33.31164 |\n",
       "| cpo18 | ar | cpo       | 10042 | 38399 | 8992.99078 | 1629.298230 | 2.463665 | 32.35961 |\n",
       "| Rfx18 | ar | Rfx       |   678 |  1448 |  615.29590 |   60.815720 | 3.315235 | 30.72387 |\n",
       "| loco18 | ar | loco      |  2560 |  8119 | 2247.19415 |  350.912280 | 2.674837 | 29.78311 |\n",
       "| vn18 | ar | vn        |  2370 |  8462 | 2090.20231 |  359.103880 | 2.537158 | 27.98514 |\n",
       "| fd3F18 | ar | fd3F      |    89 |    87 |   83.46276 |    3.380252 | 4.252047 | 27.21413 |\n",
       "| Dll18 | ar | Dll       |   437 |  1158 |  398.21130 |   45.839704 | 3.087730 | 26.68110 |\n",
       "| beat-IIIc18 | ar | beat-IIIc |  3884 | 16478 | 3471.52053 |  737.772015 | 2.232367 | 26.25658 |\n",
       "| ct18 | ar | ct        |  5614 | 26309 | 5046.42037 | 1150.202747 | 2.132118 | 26.22789 |\n",
       "| Notum18 | ar | Notum     |  1126 |  3943 |  978.75267 |  166.265683 | 2.548803 | 25.32560 |\n",
       "| unc18 | ar | unc       |    62 |    55 |   58.44251 |    2.246182 | 4.170202 | 24.57677 |\n",
       "| Traf418 | ar | Traf4     |  1655 |  7063 | 1445.73602 |  290.914448 | 2.308187 | 24.23270 |\n",
       "| spdo18 | ar | spdo      |   425 |  1434 |  386.48655 |   54.083052 | 2.810738 | 24.16673 |\n",
       "| neur18 | ar | neur      |  1963 |  8700 | 1715.26769 |  364.304832 | 2.231261 | 23.97503 |\n",
       "\n"
      ],
      "text/plain": [
       "            cell.type.2 gene      cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "nrm18       ar          nrm        6997              4384       6261.00984     \n",
       "meru18      ar          meru       2322              1190       2083.51115     \n",
       "sv18        ar          sv         3195              2021       2757.70946     \n",
       "ss18        ar          ss         1729              2018       1592.92704     \n",
       "sens18      ar          sens        930               963        807.35037     \n",
       "sca18       ar          sca        2172              4754       1857.82833     \n",
       "pb18        ar          pb          762              1373        692.54164     \n",
       "cpo18       ar          cpo       10042             38399       8992.99078     \n",
       "Rfx18       ar          Rfx         678              1448        615.29590     \n",
       "loco18      ar          loco       2560              8119       2247.19415     \n",
       "vn18        ar          vn         2370              8462       2090.20231     \n",
       "fd3F18      ar          fd3F         89                87         83.46276     \n",
       "Dll18       ar          Dll         437              1158        398.21130     \n",
       "beat-IIIc18 ar          beat-IIIc  3884             16478       3471.52053     \n",
       "ct18        ar          ct         5614             26309       5046.42037     \n",
       "Notum18     ar          Notum      1126              3943        978.75267     \n",
       "unc18       ar          unc          62                55         58.44251     \n",
       "Traf418     ar          Traf4      1655              7063       1445.73602     \n",
       "spdo18      ar          spdo        425              1434        386.48655     \n",
       "neur18      ar          neur       1963              8700       1715.26769     \n",
       "            other.tpm   log2.ratio heuristic.score\n",
       "nrm18        170.412278 5.190853   65.46917       \n",
       "meru18        47.513687 5.424481   59.80758       \n",
       "sv18          85.460695 4.995282   57.09497       \n",
       "ss18          88.495298 4.153725   44.18886       \n",
       "sens18        41.475929 4.248477   41.03535       \n",
       "sca18        192.542677 3.262894   35.43560       \n",
       "pb18          58.970166 3.529584   33.31164       \n",
       "cpo18       1629.298230 2.463665   32.35961       \n",
       "Rfx18         60.815720 3.315235   30.72387       \n",
       "loco18       350.912280 2.674837   29.78311       \n",
       "vn18         359.103880 2.537158   27.98514       \n",
       "fd3F18         3.380252 4.252047   27.21413       \n",
       "Dll18         45.839704 3.087730   26.68110       \n",
       "beat-IIIc18  737.772015 2.232367   26.25658       \n",
       "ct18        1150.202747 2.132118   26.22789       \n",
       "Notum18      166.265683 2.548803   25.32560       \n",
       "unc18          2.246182 4.170202   24.57677       \n",
       "Traf418      290.914448 2.308187   24.23270       \n",
       "spdo18        54.083052 2.810738   24.16673       \n",
       "neur18       364.304832 2.231261   23.97503       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"ar\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"nrm\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"nrm\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"nrm\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"meru\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"sv\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### as cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1451,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Oenocytes_older\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Oenocytes_older\", df_sub300$cluster == as & \n",
    "                                  df_sub300$umap_2 < -1.5*df_sub300$umap_1 + 9.5 &\n",
    "                                  df_sub300$umap_2 <1*df_sub300$umap_1 -1.7)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1452,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Proventriculus_older\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Proventriculus_older\", df_sub300$cluster == as & \n",
    "                                  df_sub300$umap_2 < -1.5*df_sub300$umap_1 + 9.5 &\n",
    "                                  df_sub300$umap_2 >= 1*df_sub300$umap_1 -1.7)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "# upper half maybe proventriculus while bottom could be more eonocytes Split and check markers\n",
    "clus = c(as)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(as)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  \n",
    "           ifelse(df_sub300$cluster %in% clus, 1, 0.5), stroke = 0.1) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "xlim(3, 7) + ylim(1, 5)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1423,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG1354519</th><td>as</td><td>CG13545   </td><td> 848</td><td> 58</td><td> 759.2038</td><td> 1.5836748</td><td>8.198919</td><td>78.46564</td></tr>\n",
       "\t<tr><th scope=row>Cyp4g120</th><td>as</td><td>Cyp4g1    </td><td>2793</td><td>994</td><td>2604.7806</td><td>36.4730834</td><td>6.119163</td><td>69.43721</td></tr>\n",
       "\t<tr><th scope=row>FASN319</th><td>as</td><td>FASN3     </td><td>2868</td><td>975</td><td>2658.7175</td><td>44.8012765</td><td>5.859199</td><td>66.66044</td></tr>\n",
       "\t<tr><th scope=row>CG720319</th><td>as</td><td>CG7203    </td><td> 210</td><td> 42</td><td> 259.3241</td><td> 1.1934814</td><td>6.885390</td><td>55.24950</td></tr>\n",
       "\t<tr><th scope=row>lectin-22C19</th><td>as</td><td>lectin-22C</td><td> 260</td><td>  6</td><td> 170.2814</td><td> 0.1301897</td><td>7.235212</td><td>53.68690</td></tr>\n",
       "\t<tr><th scope=row>CG1072519</th><td>as</td><td>CG10725   </td><td> 176</td><td> 56</td><td> 270.8826</td><td> 2.0809875</td><td>6.458131</td><td>52.22587</td></tr>\n",
       "\t<tr><th scope=row>CG771419</th><td>as</td><td>CG7714    </td><td> 397</td><td>426</td><td> 677.8968</td><td>14.5780107</td><td>5.443483</td><td>51.20711</td></tr>\n",
       "\t<tr><th scope=row>CG1113119</th><td>as</td><td>CG11131   </td><td> 229</td><td>167</td><td> 386.7453</td><td> 5.3453349</td><td>5.929544</td><td>50.98794</td></tr>\n",
       "\t<tr><th scope=row>CG1304419</th><td>as</td><td>CG13044   </td><td> 429</td><td>283</td><td> 527.6020</td><td> 9.6488677</td><td>5.630678</td><td>50.93533</td></tr>\n",
       "\t<tr><th scope=row>CG1456819</th><td>as</td><td>CG14568   </td><td>  94</td><td> 49</td><td> 233.7863</td><td> 1.7054818</td><td>6.433161</td><td>50.66246</td></tr>\n",
       "\t<tr><th scope=row>CG1293419</th><td>as</td><td>CG12934   </td><td> 212</td><td>159</td><td> 336.2816</td><td> 4.4261444</td><td>5.953599</td><td>49.99719</td></tr>\n",
       "\t<tr><th scope=row>CG1553119</th><td>as</td><td>CG15531   </td><td> 328</td><td> 62</td><td> 257.2125</td><td> 2.4831205</td><td>6.206436</td><td>49.72854</td></tr>\n",
       "\t<tr><th scope=row>CG643219</th><td>as</td><td>CG6432    </td><td> 203</td><td>  6</td><td> 134.6565</td><td> 0.1457672</td><td>6.876827</td><td>48.71417</td></tr>\n",
       "\t<tr><th scope=row>FASN219</th><td>as</td><td>FASN2     </td><td> 256</td><td> 38</td><td> 207.9358</td><td> 1.6172751</td><td>6.311929</td><td>48.64550</td></tr>\n",
       "\t<tr><th scope=row>CG693319</th><td>as</td><td>CG6933    </td><td> 450</td><td>583</td><td> 656.1091</td><td>16.9540467</td><td>5.191555</td><td>48.59290</td></tr>\n",
       "\t<tr><th scope=row>CG1456519</th><td>as</td><td>CG14565   </td><td> 119</td><td> 59</td><td> 205.3430</td><td> 1.5724343</td><td>6.318758</td><td>48.58430</td></tr>\n",
       "\t<tr><th scope=row>Obp49a19</th><td>as</td><td>Obp49a    </td><td> 244</td><td>  8</td><td> 132.5956</td><td> 0.1486566</td><td>6.850942</td><td>48.37950</td></tr>\n",
       "\t<tr><th scope=row>CG1756019</th><td>as</td><td>CG17560   </td><td> 261</td><td> 55</td><td> 219.0143</td><td> 2.3294554</td><td>6.039595</td><td>46.99683</td></tr>\n",
       "\t<tr><th scope=row>Cpr31A19</th><td>as</td><td>Cpr31A    </td><td> 236</td><td>124</td><td> 301.6235</td><td> 4.8663441</td><td>5.684143</td><td>46.84519</td></tr>\n",
       "\t<tr><th scope=row>obst-J19</th><td>as</td><td>obst-J    </td><td> 101</td><td>106</td><td> 217.7832</td><td> 2.5016451</td><td>5.958716</td><td>46.31923</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG1354519 & as & CG13545    &  848 &  58 &  759.2038 &  1.5836748 & 8.198919 & 78.46564\\\\\n",
       "\tCyp4g120 & as & Cyp4g1     & 2793 & 994 & 2604.7806 & 36.4730834 & 6.119163 & 69.43721\\\\\n",
       "\tFASN319 & as & FASN3      & 2868 & 975 & 2658.7175 & 44.8012765 & 5.859199 & 66.66044\\\\\n",
       "\tCG720319 & as & CG7203     &  210 &  42 &  259.3241 &  1.1934814 & 6.885390 & 55.24950\\\\\n",
       "\tlectin-22C19 & as & lectin-22C &  260 &   6 &  170.2814 &  0.1301897 & 7.235212 & 53.68690\\\\\n",
       "\tCG1072519 & as & CG10725    &  176 &  56 &  270.8826 &  2.0809875 & 6.458131 & 52.22587\\\\\n",
       "\tCG771419 & as & CG7714     &  397 & 426 &  677.8968 & 14.5780107 & 5.443483 & 51.20711\\\\\n",
       "\tCG1113119 & as & CG11131    &  229 & 167 &  386.7453 &  5.3453349 & 5.929544 & 50.98794\\\\\n",
       "\tCG1304419 & as & CG13044    &  429 & 283 &  527.6020 &  9.6488677 & 5.630678 & 50.93533\\\\\n",
       "\tCG1456819 & as & CG14568    &   94 &  49 &  233.7863 &  1.7054818 & 6.433161 & 50.66246\\\\\n",
       "\tCG1293419 & as & CG12934    &  212 & 159 &  336.2816 &  4.4261444 & 5.953599 & 49.99719\\\\\n",
       "\tCG1553119 & as & CG15531    &  328 &  62 &  257.2125 &  2.4831205 & 6.206436 & 49.72854\\\\\n",
       "\tCG643219 & as & CG6432     &  203 &   6 &  134.6565 &  0.1457672 & 6.876827 & 48.71417\\\\\n",
       "\tFASN219 & as & FASN2      &  256 &  38 &  207.9358 &  1.6172751 & 6.311929 & 48.64550\\\\\n",
       "\tCG693319 & as & CG6933     &  450 & 583 &  656.1091 & 16.9540467 & 5.191555 & 48.59290\\\\\n",
       "\tCG1456519 & as & CG14565    &  119 &  59 &  205.3430 &  1.5724343 & 6.318758 & 48.58430\\\\\n",
       "\tObp49a19 & as & Obp49a     &  244 &   8 &  132.5956 &  0.1486566 & 6.850942 & 48.37950\\\\\n",
       "\tCG1756019 & as & CG17560    &  261 &  55 &  219.0143 &  2.3294554 & 6.039595 & 46.99683\\\\\n",
       "\tCpr31A19 & as & Cpr31A     &  236 & 124 &  301.6235 &  4.8663441 & 5.684143 & 46.84519\\\\\n",
       "\tobst-J19 & as & obst-J     &  101 & 106 &  217.7832 &  2.5016451 & 5.958716 & 46.31923\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG1354519 | as | CG13545    |  848 |  58 |  759.2038 |  1.5836748 | 8.198919 | 78.46564 |\n",
       "| Cyp4g120 | as | Cyp4g1     | 2793 | 994 | 2604.7806 | 36.4730834 | 6.119163 | 69.43721 |\n",
       "| FASN319 | as | FASN3      | 2868 | 975 | 2658.7175 | 44.8012765 | 5.859199 | 66.66044 |\n",
       "| CG720319 | as | CG7203     |  210 |  42 |  259.3241 |  1.1934814 | 6.885390 | 55.24950 |\n",
       "| lectin-22C19 | as | lectin-22C |  260 |   6 |  170.2814 |  0.1301897 | 7.235212 | 53.68690 |\n",
       "| CG1072519 | as | CG10725    |  176 |  56 |  270.8826 |  2.0809875 | 6.458131 | 52.22587 |\n",
       "| CG771419 | as | CG7714     |  397 | 426 |  677.8968 | 14.5780107 | 5.443483 | 51.20711 |\n",
       "| CG1113119 | as | CG11131    |  229 | 167 |  386.7453 |  5.3453349 | 5.929544 | 50.98794 |\n",
       "| CG1304419 | as | CG13044    |  429 | 283 |  527.6020 |  9.6488677 | 5.630678 | 50.93533 |\n",
       "| CG1456819 | as | CG14568    |   94 |  49 |  233.7863 |  1.7054818 | 6.433161 | 50.66246 |\n",
       "| CG1293419 | as | CG12934    |  212 | 159 |  336.2816 |  4.4261444 | 5.953599 | 49.99719 |\n",
       "| CG1553119 | as | CG15531    |  328 |  62 |  257.2125 |  2.4831205 | 6.206436 | 49.72854 |\n",
       "| CG643219 | as | CG6432     |  203 |   6 |  134.6565 |  0.1457672 | 6.876827 | 48.71417 |\n",
       "| FASN219 | as | FASN2      |  256 |  38 |  207.9358 |  1.6172751 | 6.311929 | 48.64550 |\n",
       "| CG693319 | as | CG6933     |  450 | 583 |  656.1091 | 16.9540467 | 5.191555 | 48.59290 |\n",
       "| CG1456519 | as | CG14565    |  119 |  59 |  205.3430 |  1.5724343 | 6.318758 | 48.58430 |\n",
       "| Obp49a19 | as | Obp49a     |  244 |   8 |  132.5956 |  0.1486566 | 6.850942 | 48.37950 |\n",
       "| CG1756019 | as | CG17560    |  261 |  55 |  219.0143 |  2.3294554 | 6.039595 | 46.99683 |\n",
       "| Cpr31A19 | as | Cpr31A     |  236 | 124 |  301.6235 |  4.8663441 | 5.684143 | 46.84519 |\n",
       "| obst-J19 | as | obst-J     |  101 | 106 |  217.7832 |  2.5016451 | 5.958716 | 46.31923 |\n",
       "\n"
      ],
      "text/plain": [
       "             cell.type.2 gene       cell.type.2.n.umi other.n.umi\n",
       "CG1354519    as          CG13545     848               58        \n",
       "Cyp4g120     as          Cyp4g1     2793              994        \n",
       "FASN319      as          FASN3      2868              975        \n",
       "CG720319     as          CG7203      210               42        \n",
       "lectin-22C19 as          lectin-22C  260                6        \n",
       "CG1072519    as          CG10725     176               56        \n",
       "CG771419     as          CG7714      397              426        \n",
       "CG1113119    as          CG11131     229              167        \n",
       "CG1304419    as          CG13044     429              283        \n",
       "CG1456819    as          CG14568      94               49        \n",
       "CG1293419    as          CG12934     212              159        \n",
       "CG1553119    as          CG15531     328               62        \n",
       "CG643219     as          CG6432      203                6        \n",
       "FASN219      as          FASN2       256               38        \n",
       "CG693319     as          CG6933      450              583        \n",
       "CG1456519    as          CG14565     119               59        \n",
       "Obp49a19     as          Obp49a      244                8        \n",
       "CG1756019    as          CG17560     261               55        \n",
       "Cpr31A19     as          Cpr31A      236              124        \n",
       "obst-J19     as          obst-J      101              106        \n",
       "             cell.type.2.tpm other.tpm  log2.ratio heuristic.score\n",
       "CG1354519     759.2038        1.5836748 8.198919   78.46564       \n",
       "Cyp4g120     2604.7806       36.4730834 6.119163   69.43721       \n",
       "FASN319      2658.7175       44.8012765 5.859199   66.66044       \n",
       "CG720319      259.3241        1.1934814 6.885390   55.24950       \n",
       "lectin-22C19  170.2814        0.1301897 7.235212   53.68690       \n",
       "CG1072519     270.8826        2.0809875 6.458131   52.22587       \n",
       "CG771419      677.8968       14.5780107 5.443483   51.20711       \n",
       "CG1113119     386.7453        5.3453349 5.929544   50.98794       \n",
       "CG1304419     527.6020        9.6488677 5.630678   50.93533       \n",
       "CG1456819     233.7863        1.7054818 6.433161   50.66246       \n",
       "CG1293419     336.2816        4.4261444 5.953599   49.99719       \n",
       "CG1553119     257.2125        2.4831205 6.206436   49.72854       \n",
       "CG643219      134.6565        0.1457672 6.876827   48.71417       \n",
       "FASN219       207.9358        1.6172751 6.311929   48.64550       \n",
       "CG693319      656.1091       16.9540467 5.191555   48.59290       \n",
       "CG1456519     205.3430        1.5724343 6.318758   48.58430       \n",
       "Obp49a19      132.5956        0.1486566 6.850942   48.37950       \n",
       "CG1756019     219.0143        2.3294554 6.039595   46.99683       \n",
       "Cpr31A19      301.6235        4.8663441 5.684143   46.84519       \n",
       "obst-J19      217.7832        2.5016451 5.958716   46.31923       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"as\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG13545\"), size_pos = 3, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG13545\"), size_pos = 3, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG13545\"), \n",
    "                   size_pos = 3, size_neg = 0.6, stroke = 0.8, limit = 8)+\n",
    "xlim(3, 7) + ylim(1, 5)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Cyp4g1\"), \n",
    "                   size_pos = 3, size_neg = 0.6, stroke = 0.8, limit = 8)+\n",
    "xlim(3, 7) + ylim(1, 5)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"FASN3\"), \n",
    "                   size_pos = 3, size_neg = 0.6, stroke = 0.8, limit = 4)+\n",
    "xlim(3, 7) + ylim(1, 5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(as)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "              df_sub300$umap_2 < -1.5*umap_1 + 9.5 &\n",
    "              df_sub300$umap_2 >= 1*umap_1 -1.7), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 1, intercept = -1.3) +\n",
    "geom_abline(slope = 1, intercept = -1.7) +\n",
    "geom_abline(slope = -1.5, intercept = 9.5) +\n",
    "xlim(3, 7) + ylim(1, 5)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(as)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "              df_sub300$umap_2 < -1.5*umap_1 + 9.5 &\n",
    "              df_sub300$umap_2 <1*umap_1 -1.7), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 1, intercept = -1.3) +\n",
    "geom_abline(slope = 1, intercept = -1.7) +\n",
    "geom_abline(slope = -1.5, intercept = 9.5) +\n",
    "xlim(3, 7) + ylim(1, 5)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1448,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 250\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG13545</th><td>CG13545   </td><td> 909</td><td>  91</td><td> 3030.8452</td><td> 3.6098462</td><td>Set 1</td><td>9.360786</td><td>108.26667</td></tr>\n",
       "\t<tr><th scope=row>Cyp4g1</th><td>Cyp4g1    </td><td>2999</td><td>1009</td><td>10146.7767</td><td>39.3121732</td><td>Set 1</td><td>7.975590</td><td>106.14614</td></tr>\n",
       "\t<tr><th scope=row>FASN3</th><td>FASN3     </td><td>3078</td><td> 960</td><td>10394.0117</td><td>43.1288400</td><td>Set 1</td><td>7.879815</td><td>105.14513</td></tr>\n",
       "\t<tr><th scope=row>FASN2</th><td>FASN2     </td><td> 307</td><td>  40</td><td> 1002.9520</td><td> 1.4522661</td><td>Set 1</td><td>8.675921</td><td> 86.51173</td></tr>\n",
       "\t<tr><th scope=row>lectin-22C</th><td>lectin-22C</td><td> 262</td><td>   6</td><td>  650.5100</td><td> 0.1924667</td><td>Set 1</td><td>9.091478</td><td> 84.98390</td></tr>\n",
       "\t<tr><th scope=row>CG15531</th><td>CG15531   </td><td> 368</td><td>  75</td><td> 1096.3000</td><td> 2.3819667</td><td>Set 1</td><td>8.340564</td><td> 84.23755</td></tr>\n",
       "\t<tr><th scope=row>CG11131</th><td>CG11131   </td><td> 265</td><td> 213</td><td> 1920.9309</td><td> 8.6290798</td><td>Set 1</td><td>7.640192</td><td> 83.34182</td></tr>\n",
       "\t<tr><th scope=row>CG13044</th><td>CG13044   </td><td> 461</td><td> 342</td><td> 2080.4287</td><td>10.3588155</td><td>Set 1</td><td>7.516925</td><td> 82.86175</td></tr>\n",
       "\t<tr><th scope=row>Cpr31A</th><td>Cpr31A    </td><td> 261</td><td> 112</td><td> 1345.8269</td><td> 4.4168247</td><td>Set 1</td><td>7.956830</td><td> 82.71402</td></tr>\n",
       "\t<tr><th scope=row>CG6432</th><td>CG6432    </td><td> 231</td><td>   5</td><td>  580.7315</td><td> 0.1354474</td><td>Set 1</td><td>8.998467</td><td> 82.64380</td></tr>\n",
       "\t<tr><th scope=row>CG7203</th><td>CG7203    </td><td> 221</td><td>  54</td><td>  955.6080</td><td> 2.2590338</td><td>Set 1</td><td>8.195831</td><td> 81.15335</td></tr>\n",
       "\t<tr><th scope=row>CG17560</th><td>CG17560   </td><td> 269</td><td>  58</td><td>  813.4936</td><td> 2.5557757</td><td>Set 1</td><td>7.837823</td><td> 75.78986</td></tr>\n",
       "\t<tr><th scope=row>CG16904</th><td>CG16904   </td><td> 122</td><td>   0</td><td>  397.3889</td><td> 0.0000000</td><td>Set 1</td><td>8.634408</td><td> 74.58431</td></tr>\n",
       "\t<tr><th scope=row>CG14565</th><td>CG14565   </td><td> 115</td><td>  66</td><td>  710.2476</td><td> 2.1878058</td><td>Set 1</td><td>7.799614</td><td> 73.89517</td></tr>\n",
       "\t<tr><th scope=row>Obp56h</th><td>Obp56h    </td><td> 165</td><td>  34</td><td>  538.3239</td><td> 1.5615260</td><td>Set 1</td><td>7.715327</td><td> 70.01666</td></tr>\n",
       "\t<tr><th scope=row>Obp49a</th><td>Obp49a    </td><td> 243</td><td>  24</td><td>  446.8804</td><td> 0.8682317</td><td>Set 1</td><td>7.902071</td><td> 69.59330</td></tr>\n",
       "\t<tr><th scope=row>Ccp84Ab</th><td>Ccp84Ab   </td><td> 148</td><td> 121</td><td>  723.8558</td><td> 3.9252355</td><td>Set 1</td><td>7.199366</td><td> 68.40514</td></tr>\n",
       "\t<tr><th scope=row>CG18609</th><td>CG18609   </td><td> 106</td><td>  43</td><td>  450.0625</td><td> 2.1134519</td><td>Set 1</td><td>7.175467</td><td> 63.26740</td></tr>\n",
       "\t<tr><th scope=row>CG7910</th><td>CG7910    </td><td> 173</td><td>  87</td><td>  548.1109</td><td> 3.4690466</td><td>Set 1</td><td>6.938357</td><td> 63.14567</td></tr>\n",
       "\t<tr><th scope=row>CG4020</th><td>CG4020    </td><td> 183</td><td>  45</td><td>  406.9724</td><td> 1.7529119</td><td>Set 1</td><td>7.207829</td><td> 62.50865</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG13545 & CG13545    &  909 &   91 &  3030.8452 &  3.6098462 & Set 1 & 9.360786 & 108.26667\\\\\n",
       "\tCyp4g1 & Cyp4g1     & 2999 & 1009 & 10146.7767 & 39.3121732 & Set 1 & 7.975590 & 106.14614\\\\\n",
       "\tFASN3 & FASN3      & 3078 &  960 & 10394.0117 & 43.1288400 & Set 1 & 7.879815 & 105.14513\\\\\n",
       "\tFASN2 & FASN2      &  307 &   40 &  1002.9520 &  1.4522661 & Set 1 & 8.675921 &  86.51173\\\\\n",
       "\tlectin-22C & lectin-22C &  262 &    6 &   650.5100 &  0.1924667 & Set 1 & 9.091478 &  84.98390\\\\\n",
       "\tCG15531 & CG15531    &  368 &   75 &  1096.3000 &  2.3819667 & Set 1 & 8.340564 &  84.23755\\\\\n",
       "\tCG11131 & CG11131    &  265 &  213 &  1920.9309 &  8.6290798 & Set 1 & 7.640192 &  83.34182\\\\\n",
       "\tCG13044 & CG13044    &  461 &  342 &  2080.4287 & 10.3588155 & Set 1 & 7.516925 &  82.86175\\\\\n",
       "\tCpr31A & Cpr31A     &  261 &  112 &  1345.8269 &  4.4168247 & Set 1 & 7.956830 &  82.71402\\\\\n",
       "\tCG6432 & CG6432     &  231 &    5 &   580.7315 &  0.1354474 & Set 1 & 8.998467 &  82.64380\\\\\n",
       "\tCG7203 & CG7203     &  221 &   54 &   955.6080 &  2.2590338 & Set 1 & 8.195831 &  81.15335\\\\\n",
       "\tCG17560 & CG17560    &  269 &   58 &   813.4936 &  2.5557757 & Set 1 & 7.837823 &  75.78986\\\\\n",
       "\tCG16904 & CG16904    &  122 &    0 &   397.3889 &  0.0000000 & Set 1 & 8.634408 &  74.58431\\\\\n",
       "\tCG14565 & CG14565    &  115 &   66 &   710.2476 &  2.1878058 & Set 1 & 7.799614 &  73.89517\\\\\n",
       "\tObp56h & Obp56h     &  165 &   34 &   538.3239 &  1.5615260 & Set 1 & 7.715327 &  70.01666\\\\\n",
       "\tObp49a & Obp49a     &  243 &   24 &   446.8804 &  0.8682317 & Set 1 & 7.902071 &  69.59330\\\\\n",
       "\tCcp84Ab & Ccp84Ab    &  148 &  121 &   723.8558 &  3.9252355 & Set 1 & 7.199366 &  68.40514\\\\\n",
       "\tCG18609 & CG18609    &  106 &   43 &   450.0625 &  2.1134519 & Set 1 & 7.175467 &  63.26740\\\\\n",
       "\tCG7910 & CG7910     &  173 &   87 &   548.1109 &  3.4690466 & Set 1 & 6.938357 &  63.14567\\\\\n",
       "\tCG4020 & CG4020     &  183 &   45 &   406.9724 &  1.7529119 & Set 1 & 7.207829 &  62.50865\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG13545 | CG13545    |  909 |   91 |  3030.8452 |  3.6098462 | Set 1 | 9.360786 | 108.26667 |\n",
       "| Cyp4g1 | Cyp4g1     | 2999 | 1009 | 10146.7767 | 39.3121732 | Set 1 | 7.975590 | 106.14614 |\n",
       "| FASN3 | FASN3      | 3078 |  960 | 10394.0117 | 43.1288400 | Set 1 | 7.879815 | 105.14513 |\n",
       "| FASN2 | FASN2      |  307 |   40 |  1002.9520 |  1.4522661 | Set 1 | 8.675921 |  86.51173 |\n",
       "| lectin-22C | lectin-22C |  262 |    6 |   650.5100 |  0.1924667 | Set 1 | 9.091478 |  84.98390 |\n",
       "| CG15531 | CG15531    |  368 |   75 |  1096.3000 |  2.3819667 | Set 1 | 8.340564 |  84.23755 |\n",
       "| CG11131 | CG11131    |  265 |  213 |  1920.9309 |  8.6290798 | Set 1 | 7.640192 |  83.34182 |\n",
       "| CG13044 | CG13044    |  461 |  342 |  2080.4287 | 10.3588155 | Set 1 | 7.516925 |  82.86175 |\n",
       "| Cpr31A | Cpr31A     |  261 |  112 |  1345.8269 |  4.4168247 | Set 1 | 7.956830 |  82.71402 |\n",
       "| CG6432 | CG6432     |  231 |    5 |   580.7315 |  0.1354474 | Set 1 | 8.998467 |  82.64380 |\n",
       "| CG7203 | CG7203     |  221 |   54 |   955.6080 |  2.2590338 | Set 1 | 8.195831 |  81.15335 |\n",
       "| CG17560 | CG17560    |  269 |   58 |   813.4936 |  2.5557757 | Set 1 | 7.837823 |  75.78986 |\n",
       "| CG16904 | CG16904    |  122 |    0 |   397.3889 |  0.0000000 | Set 1 | 8.634408 |  74.58431 |\n",
       "| CG14565 | CG14565    |  115 |   66 |   710.2476 |  2.1878058 | Set 1 | 7.799614 |  73.89517 |\n",
       "| Obp56h | Obp56h     |  165 |   34 |   538.3239 |  1.5615260 | Set 1 | 7.715327 |  70.01666 |\n",
       "| Obp49a | Obp49a     |  243 |   24 |   446.8804 |  0.8682317 | Set 1 | 7.902071 |  69.59330 |\n",
       "| Ccp84Ab | Ccp84Ab    |  148 |  121 |   723.8558 |  3.9252355 | Set 1 | 7.199366 |  68.40514 |\n",
       "| CG18609 | CG18609    |  106 |   43 |   450.0625 |  2.1134519 | Set 1 | 7.175467 |  63.26740 |\n",
       "| CG7910 | CG7910     |  173 |   87 |   548.1109 |  3.4690466 | Set 1 | 6.938357 |  63.14567 |\n",
       "| CG4020 | CG4020     |  183 |   45 |   406.9724 |  1.7529119 | Set 1 | 7.207829 |  62.50865 |\n",
       "\n"
      ],
      "text/plain": [
       "           gene       set.1.n.umi set.2.n.umi set.1.tpm  set.2.tpm  higher.expr\n",
       "CG13545    CG13545     909          91         3030.8452  3.6098462 Set 1      \n",
       "Cyp4g1     Cyp4g1     2999        1009        10146.7767 39.3121732 Set 1      \n",
       "FASN3      FASN3      3078         960        10394.0117 43.1288400 Set 1      \n",
       "FASN2      FASN2       307          40         1002.9520  1.4522661 Set 1      \n",
       "lectin-22C lectin-22C  262           6          650.5100  0.1924667 Set 1      \n",
       "CG15531    CG15531     368          75         1096.3000  2.3819667 Set 1      \n",
       "CG11131    CG11131     265         213         1920.9309  8.6290798 Set 1      \n",
       "CG13044    CG13044     461         342         2080.4287 10.3588155 Set 1      \n",
       "Cpr31A     Cpr31A      261         112         1345.8269  4.4168247 Set 1      \n",
       "CG6432     CG6432      231           5          580.7315  0.1354474 Set 1      \n",
       "CG7203     CG7203      221          54          955.6080  2.2590338 Set 1      \n",
       "CG17560    CG17560     269          58          813.4936  2.5557757 Set 1      \n",
       "CG16904    CG16904     122           0          397.3889  0.0000000 Set 1      \n",
       "CG14565    CG14565     115          66          710.2476  2.1878058 Set 1      \n",
       "Obp56h     Obp56h      165          34          538.3239  1.5615260 Set 1      \n",
       "Obp49a     Obp49a      243          24          446.8804  0.8682317 Set 1      \n",
       "Ccp84Ab    Ccp84Ab     148         121          723.8558  3.9252355 Set 1      \n",
       "CG18609    CG18609     106          43          450.0625  2.1134519 Set 1      \n",
       "CG7910     CG7910      173          87          548.1109  3.4690466 Set 1      \n",
       "CG4020     CG4020      183          45          406.9724  1.7529119 Set 1      \n",
       "           log2.ratio heuristic.score\n",
       "CG13545    9.360786   108.26667      \n",
       "Cyp4g1     7.975590   106.14614      \n",
       "FASN3      7.879815   105.14513      \n",
       "FASN2      8.675921    86.51173      \n",
       "lectin-22C 9.091478    84.98390      \n",
       "CG15531    8.340564    84.23755      \n",
       "CG11131    7.640192    83.34182      \n",
       "CG13044    7.516925    82.86175      \n",
       "Cpr31A     7.956830    82.71402      \n",
       "CG6432     8.998467    82.64380      \n",
       "CG7203     8.195831    81.15335      \n",
       "CG17560    7.837823    75.78986      \n",
       "CG16904    8.634408    74.58431      \n",
       "CG14565    7.799614    73.89517      \n",
       "Obp56h     7.715327    70.01666      \n",
       "Obp49a     7.902071    69.59330      \n",
       "Ccp84Ab    7.199366    68.40514      \n",
       "CG18609    7.175467    63.26740      \n",
       "CG7910     6.938357    63.14567      \n",
       "CG4020     7.207829    62.50865      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "two.set.differential.gene.test.3a(cds_sub, \n",
    "                                  df_sub300$cluster == as & \n",
    "                                  df_sub300$umap_2 < -1.5*df_sub300$umap_1 + 9.5 &\n",
    "                                  df_sub300$umap_2 <1*df_sub300$umap_1 -1.7, \n",
    "                                 !(df_sub300$cluster == as & \n",
    "                                  df_sub300$umap_2 < -1.5*df_sub300$umap_1 + 9.5 &\n",
    "                                  df_sub300$umap_2 <1*df_sub300$umap_1 -1.7),\n",
    "                                  sample = 50000\n",
    "                                )  %>% filter(higher.expr == \"Set 1\")  %>% \n",
    "head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1449,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 253\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG7714</th><td>CG7714 </td><td>562</td><td> 363</td><td>4099.1844</td><td>14.1259502</td><td>Set 1</td><td>8.082167</td><td>96.99791</td></tr>\n",
       "\t<tr><th scope=row>CG12934</th><td>CG12934</td><td>274</td><td> 108</td><td>1843.1714</td><td> 4.3268001</td><td>Set 1</td><td>8.434705</td><td>91.50607</td></tr>\n",
       "\t<tr><th scope=row>CG6933</th><td>CG6933 </td><td>628</td><td> 593</td><td>3958.4668</td><td>19.4028040</td><td>Set 1</td><td>7.600031</td><td>90.82865</td></tr>\n",
       "\t<tr><th scope=row>CG10725</th><td>CG10725</td><td>208</td><td>  62</td><td>1368.2242</td><td> 2.2740037</td><td>Set 1</td><td>8.707033</td><td>90.71982</td></tr>\n",
       "\t<tr><th scope=row>CG5084</th><td>CG5084 </td><td>199</td><td>  55</td><td>1196.3735</td><td> 2.5135347</td><td>Set 1</td><td>8.411529</td><td>86.01341</td></tr>\n",
       "\t<tr><th scope=row>CG7252</th><td>CG7252 </td><td> 75</td><td>  30</td><td> 500.7664</td><td> 1.0052698</td><td>Set 1</td><td>7.964198</td><td>71.44580</td></tr>\n",
       "\t<tr><th scope=row>obst-J</th><td>obst-J </td><td>116</td><td> 100</td><td> 825.5549</td><td> 4.2324714</td><td>Set 1</td><td>7.301728</td><td>70.76080</td></tr>\n",
       "\t<tr><th scope=row>CG10154</th><td>CG10154</td><td> 50</td><td>  16</td><td> 386.3770</td><td> 0.5090056</td><td>Set 1</td><td>8.000267</td><td>68.78305</td></tr>\n",
       "\t<tr><th scope=row>Ms</th><td>Ms     </td><td>110</td><td>  28</td><td> 463.5209</td><td> 1.2332980</td><td>Set 1</td><td>7.697315</td><td>68.19513</td></tr>\n",
       "\t<tr><th scope=row>CG11131</th><td>CG11131</td><td> 61</td><td> 213</td><td> 754.8787</td><td> 5.0081838</td><td>Set 1</td><td>6.973172</td><td>66.67755</td></tr>\n",
       "\t<tr><th scope=row>Gat</th><td>Gat    </td><td> 97</td><td> 569</td><td>1107.1347</td><td>23.5879915</td><td>Set 1</td><td>5.492733</td><td>55.55305</td></tr>\n",
       "\t<tr><th scope=row>nompA</th><td>nompA  </td><td>183</td><td>1759</td><td>2359.6171</td><td>76.7779882</td><td>Set 1</td><td>4.923047</td><td>55.16249</td></tr>\n",
       "\t<tr><th scope=row>Mco4</th><td>Mco4   </td><td> 26</td><td>  11</td><td> 201.0537</td><td> 0.5387964</td><td>Set 1</td><td>7.029635</td><td>53.83713</td></tr>\n",
       "\t<tr><th scope=row>CG4409</th><td>CG4409 </td><td> 26</td><td>  83</td><td> 331.4640</td><td> 3.0076534</td><td>Set 1</td><td>6.369951</td><td>53.36142</td></tr>\n",
       "\t<tr><th scope=row>CG12507</th><td>CG12507</td><td> 33</td><td> 129</td><td> 399.5588</td><td> 6.1893984</td><td>Set 1</td><td>5.796393</td><td>50.11486</td></tr>\n",
       "\t<tr><th scope=row>CG7017</th><td>CG7017 </td><td> 82</td><td> 796</td><td> 911.4121</td><td>30.3864522</td><td>Set 1</td><td>4.859890</td><td>47.78993</td></tr>\n",
       "\t<tr><th scope=row>Ag5r</th><td>Ag5r   </td><td> 40</td><td> 101</td><td> 258.0083</td><td> 3.6790332</td><td>Set 1</td><td>5.785063</td><td>46.37801</td></tr>\n",
       "\t<tr><th scope=row>Nplp1</th><td>Nplp1  </td><td> 26</td><td> 160</td><td> 346.5632</td><td> 6.7308692</td><td>Set 1</td><td>5.486344</td><td>46.31095</td></tr>\n",
       "\t<tr><th scope=row>alrm</th><td>alrm   </td><td> 73</td><td> 701</td><td> 734.8620</td><td>24.3060897</td><td>Set 1</td><td>4.859917</td><td>46.28240</td></tr>\n",
       "\t<tr><th scope=row>CG13318</th><td>CG13318</td><td> 31</td><td> 148</td><td> 335.7383</td><td> 6.7363450</td><td>Set 1</td><td>5.439541</td><td>45.66758</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG7714 & CG7714  & 562 &  363 & 4099.1844 & 14.1259502 & Set 1 & 8.082167 & 96.99791\\\\\n",
       "\tCG12934 & CG12934 & 274 &  108 & 1843.1714 &  4.3268001 & Set 1 & 8.434705 & 91.50607\\\\\n",
       "\tCG6933 & CG6933  & 628 &  593 & 3958.4668 & 19.4028040 & Set 1 & 7.600031 & 90.82865\\\\\n",
       "\tCG10725 & CG10725 & 208 &   62 & 1368.2242 &  2.2740037 & Set 1 & 8.707033 & 90.71982\\\\\n",
       "\tCG5084 & CG5084  & 199 &   55 & 1196.3735 &  2.5135347 & Set 1 & 8.411529 & 86.01341\\\\\n",
       "\tCG7252 & CG7252  &  75 &   30 &  500.7664 &  1.0052698 & Set 1 & 7.964198 & 71.44580\\\\\n",
       "\tobst-J & obst-J  & 116 &  100 &  825.5549 &  4.2324714 & Set 1 & 7.301728 & 70.76080\\\\\n",
       "\tCG10154 & CG10154 &  50 &   16 &  386.3770 &  0.5090056 & Set 1 & 8.000267 & 68.78305\\\\\n",
       "\tMs & Ms      & 110 &   28 &  463.5209 &  1.2332980 & Set 1 & 7.697315 & 68.19513\\\\\n",
       "\tCG11131 & CG11131 &  61 &  213 &  754.8787 &  5.0081838 & Set 1 & 6.973172 & 66.67755\\\\\n",
       "\tGat & Gat     &  97 &  569 & 1107.1347 & 23.5879915 & Set 1 & 5.492733 & 55.55305\\\\\n",
       "\tnompA & nompA   & 183 & 1759 & 2359.6171 & 76.7779882 & Set 1 & 4.923047 & 55.16249\\\\\n",
       "\tMco4 & Mco4    &  26 &   11 &  201.0537 &  0.5387964 & Set 1 & 7.029635 & 53.83713\\\\\n",
       "\tCG4409 & CG4409  &  26 &   83 &  331.4640 &  3.0076534 & Set 1 & 6.369951 & 53.36142\\\\\n",
       "\tCG12507 & CG12507 &  33 &  129 &  399.5588 &  6.1893984 & Set 1 & 5.796393 & 50.11486\\\\\n",
       "\tCG7017 & CG7017  &  82 &  796 &  911.4121 & 30.3864522 & Set 1 & 4.859890 & 47.78993\\\\\n",
       "\tAg5r & Ag5r    &  40 &  101 &  258.0083 &  3.6790332 & Set 1 & 5.785063 & 46.37801\\\\\n",
       "\tNplp1 & Nplp1   &  26 &  160 &  346.5632 &  6.7308692 & Set 1 & 5.486344 & 46.31095\\\\\n",
       "\talrm & alrm    &  73 &  701 &  734.8620 & 24.3060897 & Set 1 & 4.859917 & 46.28240\\\\\n",
       "\tCG13318 & CG13318 &  31 &  148 &  335.7383 &  6.7363450 & Set 1 & 5.439541 & 45.66758\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG7714 | CG7714  | 562 |  363 | 4099.1844 | 14.1259502 | Set 1 | 8.082167 | 96.99791 |\n",
       "| CG12934 | CG12934 | 274 |  108 | 1843.1714 |  4.3268001 | Set 1 | 8.434705 | 91.50607 |\n",
       "| CG6933 | CG6933  | 628 |  593 | 3958.4668 | 19.4028040 | Set 1 | 7.600031 | 90.82865 |\n",
       "| CG10725 | CG10725 | 208 |   62 | 1368.2242 |  2.2740037 | Set 1 | 8.707033 | 90.71982 |\n",
       "| CG5084 | CG5084  | 199 |   55 | 1196.3735 |  2.5135347 | Set 1 | 8.411529 | 86.01341 |\n",
       "| CG7252 | CG7252  |  75 |   30 |  500.7664 |  1.0052698 | Set 1 | 7.964198 | 71.44580 |\n",
       "| obst-J | obst-J  | 116 |  100 |  825.5549 |  4.2324714 | Set 1 | 7.301728 | 70.76080 |\n",
       "| CG10154 | CG10154 |  50 |   16 |  386.3770 |  0.5090056 | Set 1 | 8.000267 | 68.78305 |\n",
       "| Ms | Ms      | 110 |   28 |  463.5209 |  1.2332980 | Set 1 | 7.697315 | 68.19513 |\n",
       "| CG11131 | CG11131 |  61 |  213 |  754.8787 |  5.0081838 | Set 1 | 6.973172 | 66.67755 |\n",
       "| Gat | Gat     |  97 |  569 | 1107.1347 | 23.5879915 | Set 1 | 5.492733 | 55.55305 |\n",
       "| nompA | nompA   | 183 | 1759 | 2359.6171 | 76.7779882 | Set 1 | 4.923047 | 55.16249 |\n",
       "| Mco4 | Mco4    |  26 |   11 |  201.0537 |  0.5387964 | Set 1 | 7.029635 | 53.83713 |\n",
       "| CG4409 | CG4409  |  26 |   83 |  331.4640 |  3.0076534 | Set 1 | 6.369951 | 53.36142 |\n",
       "| CG12507 | CG12507 |  33 |  129 |  399.5588 |  6.1893984 | Set 1 | 5.796393 | 50.11486 |\n",
       "| CG7017 | CG7017  |  82 |  796 |  911.4121 | 30.3864522 | Set 1 | 4.859890 | 47.78993 |\n",
       "| Ag5r | Ag5r    |  40 |  101 |  258.0083 |  3.6790332 | Set 1 | 5.785063 | 46.37801 |\n",
       "| Nplp1 | Nplp1   |  26 |  160 |  346.5632 |  6.7308692 | Set 1 | 5.486344 | 46.31095 |\n",
       "| alrm | alrm    |  73 |  701 |  734.8620 | 24.3060897 | Set 1 | 4.859917 | 46.28240 |\n",
       "| CG13318 | CG13318 |  31 |  148 |  335.7383 |  6.7363450 | Set 1 | 5.439541 | 45.66758 |\n",
       "\n"
      ],
      "text/plain": [
       "        gene    set.1.n.umi set.2.n.umi set.1.tpm set.2.tpm  higher.expr\n",
       "CG7714  CG7714  562          363        4099.1844 14.1259502 Set 1      \n",
       "CG12934 CG12934 274          108        1843.1714  4.3268001 Set 1      \n",
       "CG6933  CG6933  628          593        3958.4668 19.4028040 Set 1      \n",
       "CG10725 CG10725 208           62        1368.2242  2.2740037 Set 1      \n",
       "CG5084  CG5084  199           55        1196.3735  2.5135347 Set 1      \n",
       "CG7252  CG7252   75           30         500.7664  1.0052698 Set 1      \n",
       "obst-J  obst-J  116          100         825.5549  4.2324714 Set 1      \n",
       "CG10154 CG10154  50           16         386.3770  0.5090056 Set 1      \n",
       "Ms      Ms      110           28         463.5209  1.2332980 Set 1      \n",
       "CG11131 CG11131  61          213         754.8787  5.0081838 Set 1      \n",
       "Gat     Gat      97          569        1107.1347 23.5879915 Set 1      \n",
       "nompA   nompA   183         1759        2359.6171 76.7779882 Set 1      \n",
       "Mco4    Mco4     26           11         201.0537  0.5387964 Set 1      \n",
       "CG4409  CG4409   26           83         331.4640  3.0076534 Set 1      \n",
       "CG12507 CG12507  33          129         399.5588  6.1893984 Set 1      \n",
       "CG7017  CG7017   82          796         911.4121 30.3864522 Set 1      \n",
       "Ag5r    Ag5r     40          101         258.0083  3.6790332 Set 1      \n",
       "Nplp1   Nplp1    26          160         346.5632  6.7308692 Set 1      \n",
       "alrm    alrm     73          701         734.8620 24.3060897 Set 1      \n",
       "CG13318 CG13318  31          148         335.7383  6.7363450 Set 1      \n",
       "        log2.ratio heuristic.score\n",
       "CG7714  8.082167   96.99791       \n",
       "CG12934 8.434705   91.50607       \n",
       "CG6933  7.600031   90.82865       \n",
       "CG10725 8.707033   90.71982       \n",
       "CG5084  8.411529   86.01341       \n",
       "CG7252  7.964198   71.44580       \n",
       "obst-J  7.301728   70.76080       \n",
       "CG10154 8.000267   68.78305       \n",
       "Ms      7.697315   68.19513       \n",
       "CG11131 6.973172   66.67755       \n",
       "Gat     5.492733   55.55305       \n",
       "nompA   4.923047   55.16249       \n",
       "Mco4    7.029635   53.83713       \n",
       "CG4409  6.369951   53.36142       \n",
       "CG12507 5.796393   50.11486       \n",
       "CG7017  4.859890   47.78993       \n",
       "Ag5r    5.785063   46.37801       \n",
       "Nplp1   5.486344   46.31095       \n",
       "alrm    4.859917   46.28240       \n",
       "CG13318 5.439541   45.66758       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "two.set.differential.gene.test.3a(cds_sub, \n",
    "                                  df_sub300$cluster == as & \n",
    "                                  df_sub300$umap_2 < -1.5*df_sub300$umap_1 + 9.5 &\n",
    "                                  df_sub300$umap_2 >= 1*df_sub300$umap_1 -1.7, \n",
    "                                 !(df_sub300$cluster == as & \n",
    "                                  df_sub300$umap_2 < -1.5*df_sub300$umap_1 + 9.5 &\n",
    "                                  df_sub300$umap_2 >= 1*df_sub300$umap_1 -1.7),\n",
    "                                  sample = 50000\n",
    "                                )  %>% filter(higher.expr == \"Set 1\")  %>% \n",
    "head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG7714\"), size_pos = 3, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG7714\"), size_pos = 3, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "## these have signal in 2 which is proventriculus and flybase says\n",
    "### \"embryonic midgut chamber; embryonic proventriculus; and embryonic/larval midgut\"\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG7714\"), \n",
    "                   size_pos = 3, size_neg = 0.6, stroke = 0.8, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG12934\"), \n",
    "                   size_pos = 3, size_neg = 0.6, stroke = 0.8, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG6933\"), \n",
    "                   size_pos = 3, size_neg = 0.6, stroke = 0.8, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1459,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Midgut_4th_chamber\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Midgut_4th_chamber\", df_sub300$cluster %in% at)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'at' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'at' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(at)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'at' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'at' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(at)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1457,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG310622</th><td>at</td><td>CG3106   </td><td> 930</td><td> 630</td><td>1512.8971</td><td> 20.657465</td><td>6.126306</td><td>64.71861</td></tr>\n",
       "\t<tr><th scope=row>Mal-A320</th><td>at</td><td>Mal-A3   </td><td> 582</td><td> 548</td><td> 952.4360</td><td> 18.416347</td><td>5.616278</td><td>55.58426</td></tr>\n",
       "\t<tr><th scope=row>CG3119820</th><td>at</td><td>CG31198  </td><td> 278</td><td> 274</td><td> 449.3827</td><td>  8.904097</td><td>5.503775</td><td>48.51582</td></tr>\n",
       "\t<tr><th scope=row>Eglp220</th><td>at</td><td>Eglp2    </td><td> 262</td><td> 266</td><td> 431.4238</td><td>  9.134088</td><td>5.411818</td><td>47.38751</td></tr>\n",
       "\t<tr><th scope=row>ham20</th><td>at</td><td>ham      </td><td>1822</td><td>3956</td><td>2955.8674</td><td>172.354329</td><td>4.091786</td><td>47.17769</td></tr>\n",
       "\t<tr><th scope=row>CG1349220</th><td>at</td><td>CG13492  </td><td> 356</td><td> 478</td><td> 594.2824</td><td> 16.630801</td><td>5.074979</td><td>46.77826</td></tr>\n",
       "\t<tr><th scope=row>CG483920</th><td>at</td><td>CG4839   </td><td> 281</td><td> 250</td><td> 469.4386</td><td> 11.293737</td><td>5.254941</td><td>46.65264</td></tr>\n",
       "\t<tr><th scope=row>CG956820</th><td>at</td><td>CG9568   </td><td> 281</td><td> 308</td><td> 471.4803</td><td> 13.491342</td><td>5.023934</td><td>44.63319</td></tr>\n",
       "\t<tr><th scope=row>CG1858520</th><td>at</td><td>CG18585  </td><td> 124</td><td>  94</td><td> 220.5966</td><td>  3.690665</td><td>5.555475</td><td>43.28710</td></tr>\n",
       "\t<tr><th scope=row>CG1534720</th><td>at</td><td>CG15347  </td><td>1375</td><td>3411</td><td>2263.8805</td><td>158.403489</td><td>3.828043</td><td>42.66438</td></tr>\n",
       "\t<tr><th scope=row>CG951220</th><td>at</td><td>CG9512   </td><td> 405</td><td> 906</td><td> 683.3212</td><td> 33.382971</td><td>4.312798</td><td>40.62022</td></tr>\n",
       "\t<tr><th scope=row>Alp620</th><td>at</td><td>Alp6     </td><td>  93</td><td> 109</td><td> 156.7652</td><td>  3.374548</td><td>5.163327</td><td>37.70073</td></tr>\n",
       "\t<tr><th scope=row>CG967320</th><td>at</td><td>CG9673   </td><td> 127</td><td> 140</td><td> 200.5070</td><td>  5.717182</td><td>4.899653</td><td>37.50530</td></tr>\n",
       "\t<tr><th scope=row>Ugt37B120</th><td>at</td><td>Ugt37B1  </td><td>  97</td><td> 106</td><td> 159.5959</td><td>  3.682651</td><td>5.090954</td><td>37.30290</td></tr>\n",
       "\t<tr><th scope=row>Agpat420</th><td>at</td><td>Agpat4   </td><td> 930</td><td>3110</td><td>1540.0703</td><td>133.860464</td><td>3.513457</td><td>37.20651</td></tr>\n",
       "\t<tr><th scope=row>LManV20</th><td>at</td><td>LManV    </td><td> 112</td><td> 162</td><td> 185.1154</td><td>  5.382358</td><td>4.858192</td><td>36.63103</td></tr>\n",
       "\t<tr><th scope=row>CNT220</th><td>at</td><td>CNT2     </td><td> 143</td><td> 205</td><td> 234.2629</td><td>  8.795771</td><td>4.579826</td><td>36.08046</td></tr>\n",
       "\t<tr><th scope=row>Cyp4ad120</th><td>at</td><td>Cyp4ad1  </td><td>  89</td><td>  91</td><td> 149.4542</td><td>  3.765727</td><td>4.970864</td><td>35.95516</td></tr>\n",
       "\t<tr><th scope=row>lambdaTry20</th><td>at</td><td>lambdaTry</td><td>  83</td><td>  97</td><td> 139.2277</td><td>  3.286279</td><td>5.021577</td><td>35.81202</td></tr>\n",
       "\t<tr><th scope=row>CG1328420</th><td>at</td><td>CG13284  </td><td> 561</td><td>1728</td><td> 929.3373</td><td> 74.027741</td><td>3.630706</td><td>35.80461</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG310622 & at & CG3106    &  930 &  630 & 1512.8971 &  20.657465 & 6.126306 & 64.71861\\\\\n",
       "\tMal-A320 & at & Mal-A3    &  582 &  548 &  952.4360 &  18.416347 & 5.616278 & 55.58426\\\\\n",
       "\tCG3119820 & at & CG31198   &  278 &  274 &  449.3827 &   8.904097 & 5.503775 & 48.51582\\\\\n",
       "\tEglp220 & at & Eglp2     &  262 &  266 &  431.4238 &   9.134088 & 5.411818 & 47.38751\\\\\n",
       "\tham20 & at & ham       & 1822 & 3956 & 2955.8674 & 172.354329 & 4.091786 & 47.17769\\\\\n",
       "\tCG1349220 & at & CG13492   &  356 &  478 &  594.2824 &  16.630801 & 5.074979 & 46.77826\\\\\n",
       "\tCG483920 & at & CG4839    &  281 &  250 &  469.4386 &  11.293737 & 5.254941 & 46.65264\\\\\n",
       "\tCG956820 & at & CG9568    &  281 &  308 &  471.4803 &  13.491342 & 5.023934 & 44.63319\\\\\n",
       "\tCG1858520 & at & CG18585   &  124 &   94 &  220.5966 &   3.690665 & 5.555475 & 43.28710\\\\\n",
       "\tCG1534720 & at & CG15347   & 1375 & 3411 & 2263.8805 & 158.403489 & 3.828043 & 42.66438\\\\\n",
       "\tCG951220 & at & CG9512    &  405 &  906 &  683.3212 &  33.382971 & 4.312798 & 40.62022\\\\\n",
       "\tAlp620 & at & Alp6      &   93 &  109 &  156.7652 &   3.374548 & 5.163327 & 37.70073\\\\\n",
       "\tCG967320 & at & CG9673    &  127 &  140 &  200.5070 &   5.717182 & 4.899653 & 37.50530\\\\\n",
       "\tUgt37B120 & at & Ugt37B1   &   97 &  106 &  159.5959 &   3.682651 & 5.090954 & 37.30290\\\\\n",
       "\tAgpat420 & at & Agpat4    &  930 & 3110 & 1540.0703 & 133.860464 & 3.513457 & 37.20651\\\\\n",
       "\tLManV20 & at & LManV     &  112 &  162 &  185.1154 &   5.382358 & 4.858192 & 36.63103\\\\\n",
       "\tCNT220 & at & CNT2      &  143 &  205 &  234.2629 &   8.795771 & 4.579826 & 36.08046\\\\\n",
       "\tCyp4ad120 & at & Cyp4ad1   &   89 &   91 &  149.4542 &   3.765727 & 4.970864 & 35.95516\\\\\n",
       "\tlambdaTry20 & at & lambdaTry &   83 &   97 &  139.2277 &   3.286279 & 5.021577 & 35.81202\\\\\n",
       "\tCG1328420 & at & CG13284   &  561 & 1728 &  929.3373 &  74.027741 & 3.630706 & 35.80461\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG310622 | at | CG3106    |  930 |  630 | 1512.8971 |  20.657465 | 6.126306 | 64.71861 |\n",
       "| Mal-A320 | at | Mal-A3    |  582 |  548 |  952.4360 |  18.416347 | 5.616278 | 55.58426 |\n",
       "| CG3119820 | at | CG31198   |  278 |  274 |  449.3827 |   8.904097 | 5.503775 | 48.51582 |\n",
       "| Eglp220 | at | Eglp2     |  262 |  266 |  431.4238 |   9.134088 | 5.411818 | 47.38751 |\n",
       "| ham20 | at | ham       | 1822 | 3956 | 2955.8674 | 172.354329 | 4.091786 | 47.17769 |\n",
       "| CG1349220 | at | CG13492   |  356 |  478 |  594.2824 |  16.630801 | 5.074979 | 46.77826 |\n",
       "| CG483920 | at | CG4839    |  281 |  250 |  469.4386 |  11.293737 | 5.254941 | 46.65264 |\n",
       "| CG956820 | at | CG9568    |  281 |  308 |  471.4803 |  13.491342 | 5.023934 | 44.63319 |\n",
       "| CG1858520 | at | CG18585   |  124 |   94 |  220.5966 |   3.690665 | 5.555475 | 43.28710 |\n",
       "| CG1534720 | at | CG15347   | 1375 | 3411 | 2263.8805 | 158.403489 | 3.828043 | 42.66438 |\n",
       "| CG951220 | at | CG9512    |  405 |  906 |  683.3212 |  33.382971 | 4.312798 | 40.62022 |\n",
       "| Alp620 | at | Alp6      |   93 |  109 |  156.7652 |   3.374548 | 5.163327 | 37.70073 |\n",
       "| CG967320 | at | CG9673    |  127 |  140 |  200.5070 |   5.717182 | 4.899653 | 37.50530 |\n",
       "| Ugt37B120 | at | Ugt37B1   |   97 |  106 |  159.5959 |   3.682651 | 5.090954 | 37.30290 |\n",
       "| Agpat420 | at | Agpat4    |  930 | 3110 | 1540.0703 | 133.860464 | 3.513457 | 37.20651 |\n",
       "| LManV20 | at | LManV     |  112 |  162 |  185.1154 |   5.382358 | 4.858192 | 36.63103 |\n",
       "| CNT220 | at | CNT2      |  143 |  205 |  234.2629 |   8.795771 | 4.579826 | 36.08046 |\n",
       "| Cyp4ad120 | at | Cyp4ad1   |   89 |   91 |  149.4542 |   3.765727 | 4.970864 | 35.95516 |\n",
       "| lambdaTry20 | at | lambdaTry |   83 |   97 |  139.2277 |   3.286279 | 5.021577 | 35.81202 |\n",
       "| CG1328420 | at | CG13284   |  561 | 1728 |  929.3373 |  74.027741 | 3.630706 | 35.80461 |\n",
       "\n"
      ],
      "text/plain": [
       "            cell.type.2 gene      cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "CG310622    at          CG3106     930               630        1512.8971      \n",
       "Mal-A320    at          Mal-A3     582               548         952.4360      \n",
       "CG3119820   at          CG31198    278               274         449.3827      \n",
       "Eglp220     at          Eglp2      262               266         431.4238      \n",
       "ham20       at          ham       1822              3956        2955.8674      \n",
       "CG1349220   at          CG13492    356               478         594.2824      \n",
       "CG483920    at          CG4839     281               250         469.4386      \n",
       "CG956820    at          CG9568     281               308         471.4803      \n",
       "CG1858520   at          CG18585    124                94         220.5966      \n",
       "CG1534720   at          CG15347   1375              3411        2263.8805      \n",
       "CG951220    at          CG9512     405               906         683.3212      \n",
       "Alp620      at          Alp6        93               109         156.7652      \n",
       "CG967320    at          CG9673     127               140         200.5070      \n",
       "Ugt37B120   at          Ugt37B1     97               106         159.5959      \n",
       "Agpat420    at          Agpat4     930              3110        1540.0703      \n",
       "LManV20     at          LManV      112               162         185.1154      \n",
       "CNT220      at          CNT2       143               205         234.2629      \n",
       "Cyp4ad120   at          Cyp4ad1     89                91         149.4542      \n",
       "lambdaTry20 at          lambdaTry   83                97         139.2277      \n",
       "CG1328420   at          CG13284    561              1728         929.3373      \n",
       "            other.tpm  log2.ratio heuristic.score\n",
       "CG310622     20.657465 6.126306   64.71861       \n",
       "Mal-A320     18.416347 5.616278   55.58426       \n",
       "CG3119820     8.904097 5.503775   48.51582       \n",
       "Eglp220       9.134088 5.411818   47.38751       \n",
       "ham20       172.354329 4.091786   47.17769       \n",
       "CG1349220    16.630801 5.074979   46.77826       \n",
       "CG483920     11.293737 5.254941   46.65264       \n",
       "CG956820     13.491342 5.023934   44.63319       \n",
       "CG1858520     3.690665 5.555475   43.28710       \n",
       "CG1534720   158.403489 3.828043   42.66438       \n",
       "CG951220     33.382971 4.312798   40.62022       \n",
       "Alp620        3.374548 5.163327   37.70073       \n",
       "CG967320      5.717182 4.899653   37.50530       \n",
       "Ugt37B120     3.682651 5.090954   37.30290       \n",
       "Agpat420    133.860464 3.513457   37.20651       \n",
       "LManV20       5.382358 4.858192   36.63103       \n",
       "CNT220        8.795771 4.579826   36.08046       \n",
       "Cyp4ad120     3.765727 4.970864   35.95516       \n",
       "lambdaTry20   3.286279 5.021577   35.81202       \n",
       "CG1328420    74.027741 3.630706   35.80461       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"at\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG3106\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG3106\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG3106\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Mal-A3\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG31198\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ab' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ab' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1315,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Tracheal_system_2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Tracheal_system_2\", df_sub300$cluster %in% au &\n",
    "                         df_sub300$umap_2 < -0.1*df_sub300$umap_1 -4.4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1318,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Maxillary_sensory_complex\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## could extend farther to right\n",
    "df_sub300 = set.cell.type(df_sub300, \"Maxillary_sensory_complex\", df_sub300$cluster %in% au &\n",
    "                         df_sub300$umap_2 >= -0.1*df_sub300$umap_1 -4.4 &\n",
    "                         df_sub300$umap_2 < -0.1*df_sub300$umap_1 -4.1 )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'au' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'au' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(au)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(63)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1294,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG730021</th><td>au</td><td>CG7300  </td><td> 601</td><td>  538</td><td>1335.55703</td><td>  19.643682</td><td>6.015597</td><td>62.46780</td></tr>\n",
       "\t<tr><th scope=row>twz21</th><td>au</td><td>twz     </td><td> 254</td><td>  664</td><td> 544.11068</td><td>  27.139776</td><td>4.273217</td><td>38.84528</td></tr>\n",
       "\t<tr><th scope=row>Dfd21</th><td>au</td><td>Dfd     </td><td> 219</td><td>  632</td><td> 451.98694</td><td>  24.202394</td><td>4.164648</td><td>36.74605</td></tr>\n",
       "\t<tr><th scope=row>CG1336221</th><td>au</td><td>CG13362 </td><td> 744</td><td> 3603</td><td>1652.33852</td><td> 152.067612</td><td>3.432268</td><td>36.69495</td></tr>\n",
       "\t<tr><th scope=row>CG1710821</th><td>au</td><td>CG17108 </td><td>  39</td><td>   29</td><td>  98.38083</td><td>   1.187024</td><td>5.491336</td><td>36.43444</td></tr>\n",
       "\t<tr><th scope=row>TfAP-221</th><td>au</td><td>TfAP-2  </td><td> 134</td><td>  335</td><td> 284.12911</td><td>  13.168154</td><td>4.325823</td><td>35.27913</td></tr>\n",
       "\t<tr><th scope=row>CG729621</th><td>au</td><td>CG7296  </td><td> 434</td><td> 2883</td><td>1004.98267</td><td> 116.195679</td><td>3.100179</td><td>30.92240</td></tr>\n",
       "\t<tr><th scope=row>Spn43Aa21</th><td>au</td><td>Spn43Aa </td><td>1024</td><td> 7726</td><td>2286.70824</td><td> 338.882299</td><td>2.750165</td><td>30.69098</td></tr>\n",
       "\t<tr><th scope=row>dsx-c73A21</th><td>au</td><td>dsx-c73A</td><td>1106</td><td> 8072</td><td>2389.67108</td><td> 363.966652</td><td>2.710976</td><td>30.42582</td></tr>\n",
       "\t<tr><th scope=row>knrl21</th><td>au</td><td>knrl    </td><td> 765</td><td> 5271</td><td>1554.94816</td><td> 225.051048</td><td>2.782146</td><td>29.50070</td></tr>\n",
       "\t<tr><th scope=row>rn21</th><td>au</td><td>rn      </td><td> 155</td><td>  457</td><td> 282.39788</td><td>  22.231435</td><td>3.603579</td><td>29.35722</td></tr>\n",
       "\t<tr><th scope=row>Cht521</th><td>au</td><td>Cht5    </td><td> 225</td><td> 1106</td><td> 481.99354</td><td>  50.138300</td><td>3.236538</td><td>28.85652</td></tr>\n",
       "\t<tr><th scope=row>verm21</th><td>au</td><td>verm    </td><td>3053</td><td>34866</td><td>6556.81741</td><td>1459.448162</td><td>2.166585</td><td>27.47013</td></tr>\n",
       "\t<tr><th scope=row>nw21</th><td>au</td><td>nw      </td><td>1278</td><td>13762</td><td>2819.95487</td><td> 548.853480</td><td>2.358553</td><td>27.03366</td></tr>\n",
       "\t<tr><th scope=row>CG1281421</th><td>au</td><td>CG12814 </td><td> 507</td><td> 3990</td><td>1050.28017</td><td> 173.958615</td><td>2.585689</td><td>25.95497</td></tr>\n",
       "\t<tr><th scope=row>CG4252421</th><td>au</td><td>CG42524 </td><td> 330</td><td> 2092</td><td> 681.63031</td><td> 102.891826</td><td>2.713907</td><td>25.55133</td></tr>\n",
       "\t<tr><th scope=row>pb21</th><td>au</td><td>pb      </td><td> 240</td><td> 1512</td><td> 482.55178</td><td>  66.483650</td><td>2.838074</td><td>25.30860</td></tr>\n",
       "\t<tr><th scope=row>sano21</th><td>au</td><td>sano    </td><td> 986</td><td> 9661</td><td>2073.47932</td><td> 432.996740</td><td>2.256298</td><td>24.86109</td></tr>\n",
       "\t<tr><th scope=row>CG4274121</th><td>au</td><td>CG42741 </td><td> 279</td><td> 1785</td><td> 561.02290</td><td>  85.066192</td><td>2.704541</td><td>24.70459</td></tr>\n",
       "\t<tr><th scope=row>serp21</th><td>au</td><td>serp    </td><td>1717</td><td>21363</td><td>3657.92483</td><td> 865.865382</td><td>2.077146</td><td>24.58760</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG730021 & au & CG7300   &  601 &   538 & 1335.55703 &   19.643682 & 6.015597 & 62.46780\\\\\n",
       "\ttwz21 & au & twz      &  254 &   664 &  544.11068 &   27.139776 & 4.273217 & 38.84528\\\\\n",
       "\tDfd21 & au & Dfd      &  219 &   632 &  451.98694 &   24.202394 & 4.164648 & 36.74605\\\\\n",
       "\tCG1336221 & au & CG13362  &  744 &  3603 & 1652.33852 &  152.067612 & 3.432268 & 36.69495\\\\\n",
       "\tCG1710821 & au & CG17108  &   39 &    29 &   98.38083 &    1.187024 & 5.491336 & 36.43444\\\\\n",
       "\tTfAP-221 & au & TfAP-2   &  134 &   335 &  284.12911 &   13.168154 & 4.325823 & 35.27913\\\\\n",
       "\tCG729621 & au & CG7296   &  434 &  2883 & 1004.98267 &  116.195679 & 3.100179 & 30.92240\\\\\n",
       "\tSpn43Aa21 & au & Spn43Aa  & 1024 &  7726 & 2286.70824 &  338.882299 & 2.750165 & 30.69098\\\\\n",
       "\tdsx-c73A21 & au & dsx-c73A & 1106 &  8072 & 2389.67108 &  363.966652 & 2.710976 & 30.42582\\\\\n",
       "\tknrl21 & au & knrl     &  765 &  5271 & 1554.94816 &  225.051048 & 2.782146 & 29.50070\\\\\n",
       "\trn21 & au & rn       &  155 &   457 &  282.39788 &   22.231435 & 3.603579 & 29.35722\\\\\n",
       "\tCht521 & au & Cht5     &  225 &  1106 &  481.99354 &   50.138300 & 3.236538 & 28.85652\\\\\n",
       "\tverm21 & au & verm     & 3053 & 34866 & 6556.81741 & 1459.448162 & 2.166585 & 27.47013\\\\\n",
       "\tnw21 & au & nw       & 1278 & 13762 & 2819.95487 &  548.853480 & 2.358553 & 27.03366\\\\\n",
       "\tCG1281421 & au & CG12814  &  507 &  3990 & 1050.28017 &  173.958615 & 2.585689 & 25.95497\\\\\n",
       "\tCG4252421 & au & CG42524  &  330 &  2092 &  681.63031 &  102.891826 & 2.713907 & 25.55133\\\\\n",
       "\tpb21 & au & pb       &  240 &  1512 &  482.55178 &   66.483650 & 2.838074 & 25.30860\\\\\n",
       "\tsano21 & au & sano     &  986 &  9661 & 2073.47932 &  432.996740 & 2.256298 & 24.86109\\\\\n",
       "\tCG4274121 & au & CG42741  &  279 &  1785 &  561.02290 &   85.066192 & 2.704541 & 24.70459\\\\\n",
       "\tserp21 & au & serp     & 1717 & 21363 & 3657.92483 &  865.865382 & 2.077146 & 24.58760\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG730021 | au | CG7300   |  601 |   538 | 1335.55703 |   19.643682 | 6.015597 | 62.46780 |\n",
       "| twz21 | au | twz      |  254 |   664 |  544.11068 |   27.139776 | 4.273217 | 38.84528 |\n",
       "| Dfd21 | au | Dfd      |  219 |   632 |  451.98694 |   24.202394 | 4.164648 | 36.74605 |\n",
       "| CG1336221 | au | CG13362  |  744 |  3603 | 1652.33852 |  152.067612 | 3.432268 | 36.69495 |\n",
       "| CG1710821 | au | CG17108  |   39 |    29 |   98.38083 |    1.187024 | 5.491336 | 36.43444 |\n",
       "| TfAP-221 | au | TfAP-2   |  134 |   335 |  284.12911 |   13.168154 | 4.325823 | 35.27913 |\n",
       "| CG729621 | au | CG7296   |  434 |  2883 | 1004.98267 |  116.195679 | 3.100179 | 30.92240 |\n",
       "| Spn43Aa21 | au | Spn43Aa  | 1024 |  7726 | 2286.70824 |  338.882299 | 2.750165 | 30.69098 |\n",
       "| dsx-c73A21 | au | dsx-c73A | 1106 |  8072 | 2389.67108 |  363.966652 | 2.710976 | 30.42582 |\n",
       "| knrl21 | au | knrl     |  765 |  5271 | 1554.94816 |  225.051048 | 2.782146 | 29.50070 |\n",
       "| rn21 | au | rn       |  155 |   457 |  282.39788 |   22.231435 | 3.603579 | 29.35722 |\n",
       "| Cht521 | au | Cht5     |  225 |  1106 |  481.99354 |   50.138300 | 3.236538 | 28.85652 |\n",
       "| verm21 | au | verm     | 3053 | 34866 | 6556.81741 | 1459.448162 | 2.166585 | 27.47013 |\n",
       "| nw21 | au | nw       | 1278 | 13762 | 2819.95487 |  548.853480 | 2.358553 | 27.03366 |\n",
       "| CG1281421 | au | CG12814  |  507 |  3990 | 1050.28017 |  173.958615 | 2.585689 | 25.95497 |\n",
       "| CG4252421 | au | CG42524  |  330 |  2092 |  681.63031 |  102.891826 | 2.713907 | 25.55133 |\n",
       "| pb21 | au | pb       |  240 |  1512 |  482.55178 |   66.483650 | 2.838074 | 25.30860 |\n",
       "| sano21 | au | sano     |  986 |  9661 | 2073.47932 |  432.996740 | 2.256298 | 24.86109 |\n",
       "| CG4274121 | au | CG42741  |  279 |  1785 |  561.02290 |   85.066192 | 2.704541 | 24.70459 |\n",
       "| serp21 | au | serp     | 1717 | 21363 | 3657.92483 |  865.865382 | 2.077146 | 24.58760 |\n",
       "\n"
      ],
      "text/plain": [
       "           cell.type.2 gene     cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "CG730021   au          CG7300    601                538       1335.55703     \n",
       "twz21      au          twz       254                664        544.11068     \n",
       "Dfd21      au          Dfd       219                632        451.98694     \n",
       "CG1336221  au          CG13362   744               3603       1652.33852     \n",
       "CG1710821  au          CG17108    39                 29         98.38083     \n",
       "TfAP-221   au          TfAP-2    134                335        284.12911     \n",
       "CG729621   au          CG7296    434               2883       1004.98267     \n",
       "Spn43Aa21  au          Spn43Aa  1024               7726       2286.70824     \n",
       "dsx-c73A21 au          dsx-c73A 1106               8072       2389.67108     \n",
       "knrl21     au          knrl      765               5271       1554.94816     \n",
       "rn21       au          rn        155                457        282.39788     \n",
       "Cht521     au          Cht5      225               1106        481.99354     \n",
       "verm21     au          verm     3053              34866       6556.81741     \n",
       "nw21       au          nw       1278              13762       2819.95487     \n",
       "CG1281421  au          CG12814   507               3990       1050.28017     \n",
       "CG4252421  au          CG42524   330               2092        681.63031     \n",
       "pb21       au          pb        240               1512        482.55178     \n",
       "sano21     au          sano      986               9661       2073.47932     \n",
       "CG4274121  au          CG42741   279               1785        561.02290     \n",
       "serp21     au          serp     1717              21363       3657.92483     \n",
       "           other.tpm   log2.ratio heuristic.score\n",
       "CG730021     19.643682 6.015597   62.46780       \n",
       "twz21        27.139776 4.273217   38.84528       \n",
       "Dfd21        24.202394 4.164648   36.74605       \n",
       "CG1336221   152.067612 3.432268   36.69495       \n",
       "CG1710821     1.187024 5.491336   36.43444       \n",
       "TfAP-221     13.168154 4.325823   35.27913       \n",
       "CG729621    116.195679 3.100179   30.92240       \n",
       "Spn43Aa21   338.882299 2.750165   30.69098       \n",
       "dsx-c73A21  363.966652 2.710976   30.42582       \n",
       "knrl21      225.051048 2.782146   29.50070       \n",
       "rn21         22.231435 3.603579   29.35722       \n",
       "Cht521       50.138300 3.236538   28.85652       \n",
       "verm21     1459.448162 2.166585   27.47013       \n",
       "nw21        548.853480 2.358553   27.03366       \n",
       "CG1281421   173.958615 2.585689   25.95497       \n",
       "CG4252421   102.891826 2.713907   25.55133       \n",
       "pb21         66.483650 2.838074   25.30860       \n",
       "sano21      432.996740 2.256298   24.86109       \n",
       "CG4274121    85.066192 2.704541   24.70459       \n",
       "serp21      865.865382 2.077146   24.58760       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"au\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG7300\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG7300\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "# These ar all from upper branch, not tracheal system branch which may connect with epidermis\n",
    "# flybase says CG7300 \"is expressed in embryonic maxillary sensory complex.\" and Sue agrees.\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG7300\"), \n",
    "                   size_pos = 2, size_neg = 0.8, stroke = 0.08, limit = 8)+\n",
    "xlim(-2, 2 ) + ylim(-6, -2)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"twz\"), \n",
    "                   size_pos = 2, size_neg = 0.8, stroke = 0.2, limit = 8)+\n",
    "xlim(-2, 2 ) + ylim(-6, -2)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Dfd\"), \n",
    "                   size_pos = 2, size_neg = 0.8, stroke = 0.2, limit = 4)+\n",
    "xlim(-2, 2 ) + ylim(-6, -2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'au' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'au' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(au)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 < -0.1*umap_1 -4.4), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.8), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = -0.1, intercept = -4.1) +\n",
    "geom_abline(slope = -0.1, intercept = -4.4) +\n",
    "xlim(-2, 2 ) + ylim(-6, -2)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'au' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'au' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(au)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "              df_sub300$umap_2 < -0.1*umap_1 -4.1 &\n",
    "              df_sub300$umap_2 >= -0.1*umap_1 -4.4), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.8), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = -0.1, intercept = -4.1) +\n",
    "geom_abline(slope = -0.1, intercept = -4.4) +\n",
    "xlim(-2, 2 ) + ylim(-6, -2)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1463,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Germ_cells_ovaries_query\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Germ_cells_ovaries_query\", df_sub300$cluster %in% av)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'av' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'av' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(av)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'av' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'av' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(av)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1461,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>dhd22</th><td>av</td><td>dhd        </td><td>1176</td><td>590</td><td>1425.65676</td><td>23.052980</td><td>5.889267</td><td>61.71023</td></tr>\n",
       "\t<tr><th scope=row>yl22</th><td>av</td><td>yl         </td><td> 245</td><td>139</td><td> 303.76097</td><td> 5.191471</td><td>5.616510</td><td>46.34483</td></tr>\n",
       "\t<tr><th scope=row>Pxt22</th><td>av</td><td>Pxt        </td><td> 245</td><td>138</td><td> 296.54945</td><td> 5.202930</td><td>5.579179</td><td>45.84403</td></tr>\n",
       "\t<tr><th scope=row>alphaTub67C22</th><td>av</td><td>alphaTub67C</td><td> 197</td><td>105</td><td> 249.75207</td><td> 4.226743</td><td>5.578441</td><td>44.46083</td></tr>\n",
       "\t<tr><th scope=row>Cyp6a1922</th><td>av</td><td>Cyp6a19    </td><td> 118</td><td> 32</td><td> 140.02111</td><td> 1.350599</td><td>5.896472</td><td>42.09944</td></tr>\n",
       "\t<tr><th scope=row>fs(1)Ya22</th><td>av</td><td>fs(1)Ya    </td><td> 119</td><td> 67</td><td> 143.91818</td><td> 2.499293</td><td>5.362041</td><td>38.49460</td></tr>\n",
       "\t<tr><th scope=row>CG720822</th><td>av</td><td>CG7208     </td><td>  88</td><td> 41</td><td> 109.07732</td><td> 1.454002</td><td>5.474071</td><td>37.12719</td></tr>\n",
       "\t<tr><th scope=row>Jabba22</th><td>av</td><td>Jabba      </td><td> 233</td><td>225</td><td> 244.94891</td><td> 8.689748</td><td>4.659878</td><td>37.00975</td></tr>\n",
       "\t<tr><th scope=row>mtrm22</th><td>av</td><td>mtrm       </td><td> 205</td><td>237</td><td> 254.67878</td><td> 9.493299</td><td>4.601139</td><td>36.80077</td></tr>\n",
       "\t<tr><th scope=row>CG499122</th><td>av</td><td>CG4991     </td><td> 118</td><td> 77</td><td> 139.70277</td><td> 3.326247</td><td>5.013101</td><td>35.77603</td></tr>\n",
       "\t<tr><th scope=row>CG1167422</th><td>av</td><td>CG11674    </td><td> 150</td><td>145</td><td> 180.87002</td><td> 5.696920</td><td>4.755312</td><td>35.69700</td></tr>\n",
       "\t<tr><th scope=row>fs(1)M322</th><td>av</td><td>fs(1)M3    </td><td>  76</td><td> 41</td><td>  98.18860</td><td> 1.409007</td><td>5.349045</td><td>35.47541</td></tr>\n",
       "\t<tr><th scope=row>exu22</th><td>av</td><td>exu        </td><td> 193</td><td>254</td><td> 221.79663</td><td> 8.713716</td><td>4.513070</td><td>35.20007</td></tr>\n",
       "\t<tr><th scope=row>CG447622</th><td>av</td><td>CG4476     </td><td>  62</td><td> 35</td><td>  92.22651</td><td> 1.396529</td><td>5.266163</td><td>34.45476</td></tr>\n",
       "\t<tr><th scope=row>vas22</th><td>av</td><td>vas        </td><td> 120</td><td>132</td><td> 152.48148</td><td> 4.970738</td><td>4.674581</td><td>33.94644</td></tr>\n",
       "\t<tr><th scope=row>wisp22</th><td>av</td><td>wisp       </td><td> 317</td><td>702</td><td> 363.06414</td><td>22.676776</td><td>3.938680</td><td>33.51048</td></tr>\n",
       "\t<tr><th scope=row>BigH122</th><td>av</td><td>BigH1      </td><td> 131</td><td>177</td><td> 167.39856</td><td> 6.282128</td><td>4.522783</td><td>33.44931</td></tr>\n",
       "\t<tr><th scope=row>zpg22</th><td>av</td><td>zpg        </td><td> 149</td><td>221</td><td> 197.11389</td><td> 8.950697</td><td>4.308088</td><td>32.87151</td></tr>\n",
       "\t<tr><th scope=row>swa22</th><td>av</td><td>swa        </td><td> 164</td><td>299</td><td> 220.39693</td><td>11.023102</td><td>4.196223</td><td>32.69064</td></tr>\n",
       "\t<tr><th scope=row>krimp22</th><td>av</td><td>krimp      </td><td>  73</td><td> 59</td><td>  93.27223</td><td> 2.016900</td><td>4.950309</td><td>32.46789</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tdhd22 & av & dhd         & 1176 & 590 & 1425.65676 & 23.052980 & 5.889267 & 61.71023\\\\\n",
       "\tyl22 & av & yl          &  245 & 139 &  303.76097 &  5.191471 & 5.616510 & 46.34483\\\\\n",
       "\tPxt22 & av & Pxt         &  245 & 138 &  296.54945 &  5.202930 & 5.579179 & 45.84403\\\\\n",
       "\talphaTub67C22 & av & alphaTub67C &  197 & 105 &  249.75207 &  4.226743 & 5.578441 & 44.46083\\\\\n",
       "\tCyp6a1922 & av & Cyp6a19     &  118 &  32 &  140.02111 &  1.350599 & 5.896472 & 42.09944\\\\\n",
       "\tfs(1)Ya22 & av & fs(1)Ya     &  119 &  67 &  143.91818 &  2.499293 & 5.362041 & 38.49460\\\\\n",
       "\tCG720822 & av & CG7208      &   88 &  41 &  109.07732 &  1.454002 & 5.474071 & 37.12719\\\\\n",
       "\tJabba22 & av & Jabba       &  233 & 225 &  244.94891 &  8.689748 & 4.659878 & 37.00975\\\\\n",
       "\tmtrm22 & av & mtrm        &  205 & 237 &  254.67878 &  9.493299 & 4.601139 & 36.80077\\\\\n",
       "\tCG499122 & av & CG4991      &  118 &  77 &  139.70277 &  3.326247 & 5.013101 & 35.77603\\\\\n",
       "\tCG1167422 & av & CG11674     &  150 & 145 &  180.87002 &  5.696920 & 4.755312 & 35.69700\\\\\n",
       "\tfs(1)M322 & av & fs(1)M3     &   76 &  41 &   98.18860 &  1.409007 & 5.349045 & 35.47541\\\\\n",
       "\texu22 & av & exu         &  193 & 254 &  221.79663 &  8.713716 & 4.513070 & 35.20007\\\\\n",
       "\tCG447622 & av & CG4476      &   62 &  35 &   92.22651 &  1.396529 & 5.266163 & 34.45476\\\\\n",
       "\tvas22 & av & vas         &  120 & 132 &  152.48148 &  4.970738 & 4.674581 & 33.94644\\\\\n",
       "\twisp22 & av & wisp        &  317 & 702 &  363.06414 & 22.676776 & 3.938680 & 33.51048\\\\\n",
       "\tBigH122 & av & BigH1       &  131 & 177 &  167.39856 &  6.282128 & 4.522783 & 33.44931\\\\\n",
       "\tzpg22 & av & zpg         &  149 & 221 &  197.11389 &  8.950697 & 4.308088 & 32.87151\\\\\n",
       "\tswa22 & av & swa         &  164 & 299 &  220.39693 & 11.023102 & 4.196223 & 32.69064\\\\\n",
       "\tkrimp22 & av & krimp       &   73 &  59 &   93.27223 &  2.016900 & 4.950309 & 32.46789\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| dhd22 | av | dhd         | 1176 | 590 | 1425.65676 | 23.052980 | 5.889267 | 61.71023 |\n",
       "| yl22 | av | yl          |  245 | 139 |  303.76097 |  5.191471 | 5.616510 | 46.34483 |\n",
       "| Pxt22 | av | Pxt         |  245 | 138 |  296.54945 |  5.202930 | 5.579179 | 45.84403 |\n",
       "| alphaTub67C22 | av | alphaTub67C |  197 | 105 |  249.75207 |  4.226743 | 5.578441 | 44.46083 |\n",
       "| Cyp6a1922 | av | Cyp6a19     |  118 |  32 |  140.02111 |  1.350599 | 5.896472 | 42.09944 |\n",
       "| fs(1)Ya22 | av | fs(1)Ya     |  119 |  67 |  143.91818 |  2.499293 | 5.362041 | 38.49460 |\n",
       "| CG720822 | av | CG7208      |   88 |  41 |  109.07732 |  1.454002 | 5.474071 | 37.12719 |\n",
       "| Jabba22 | av | Jabba       |  233 | 225 |  244.94891 |  8.689748 | 4.659878 | 37.00975 |\n",
       "| mtrm22 | av | mtrm        |  205 | 237 |  254.67878 |  9.493299 | 4.601139 | 36.80077 |\n",
       "| CG499122 | av | CG4991      |  118 |  77 |  139.70277 |  3.326247 | 5.013101 | 35.77603 |\n",
       "| CG1167422 | av | CG11674     |  150 | 145 |  180.87002 |  5.696920 | 4.755312 | 35.69700 |\n",
       "| fs(1)M322 | av | fs(1)M3     |   76 |  41 |   98.18860 |  1.409007 | 5.349045 | 35.47541 |\n",
       "| exu22 | av | exu         |  193 | 254 |  221.79663 |  8.713716 | 4.513070 | 35.20007 |\n",
       "| CG447622 | av | CG4476      |   62 |  35 |   92.22651 |  1.396529 | 5.266163 | 34.45476 |\n",
       "| vas22 | av | vas         |  120 | 132 |  152.48148 |  4.970738 | 4.674581 | 33.94644 |\n",
       "| wisp22 | av | wisp        |  317 | 702 |  363.06414 | 22.676776 | 3.938680 | 33.51048 |\n",
       "| BigH122 | av | BigH1       |  131 | 177 |  167.39856 |  6.282128 | 4.522783 | 33.44931 |\n",
       "| zpg22 | av | zpg         |  149 | 221 |  197.11389 |  8.950697 | 4.308088 | 32.87151 |\n",
       "| swa22 | av | swa         |  164 | 299 |  220.39693 | 11.023102 | 4.196223 | 32.69064 |\n",
       "| krimp22 | av | krimp       |   73 |  59 |   93.27223 |  2.016900 | 4.950309 | 32.46789 |\n",
       "\n"
      ],
      "text/plain": [
       "              cell.type.2 gene        cell.type.2.n.umi other.n.umi\n",
       "dhd22         av          dhd         1176              590        \n",
       "yl22          av          yl           245              139        \n",
       "Pxt22         av          Pxt          245              138        \n",
       "alphaTub67C22 av          alphaTub67C  197              105        \n",
       "Cyp6a1922     av          Cyp6a19      118               32        \n",
       "fs(1)Ya22     av          fs(1)Ya      119               67        \n",
       "CG720822      av          CG7208        88               41        \n",
       "Jabba22       av          Jabba        233              225        \n",
       "mtrm22        av          mtrm         205              237        \n",
       "CG499122      av          CG4991       118               77        \n",
       "CG1167422     av          CG11674      150              145        \n",
       "fs(1)M322     av          fs(1)M3       76               41        \n",
       "exu22         av          exu          193              254        \n",
       "CG447622      av          CG4476        62               35        \n",
       "vas22         av          vas          120              132        \n",
       "wisp22        av          wisp         317              702        \n",
       "BigH122       av          BigH1        131              177        \n",
       "zpg22         av          zpg          149              221        \n",
       "swa22         av          swa          164              299        \n",
       "krimp22       av          krimp         73               59        \n",
       "              cell.type.2.tpm other.tpm log2.ratio heuristic.score\n",
       "dhd22         1425.65676      23.052980 5.889267   61.71023       \n",
       "yl22           303.76097       5.191471 5.616510   46.34483       \n",
       "Pxt22          296.54945       5.202930 5.579179   45.84403       \n",
       "alphaTub67C22  249.75207       4.226743 5.578441   44.46083       \n",
       "Cyp6a1922      140.02111       1.350599 5.896472   42.09944       \n",
       "fs(1)Ya22      143.91818       2.499293 5.362041   38.49460       \n",
       "CG720822       109.07732       1.454002 5.474071   37.12719       \n",
       "Jabba22        244.94891       8.689748 4.659878   37.00975       \n",
       "mtrm22         254.67878       9.493299 4.601139   36.80077       \n",
       "CG499122       139.70277       3.326247 5.013101   35.77603       \n",
       "CG1167422      180.87002       5.696920 4.755312   35.69700       \n",
       "fs(1)M322       98.18860       1.409007 5.349045   35.47541       \n",
       "exu22          221.79663       8.713716 4.513070   35.20007       \n",
       "CG447622        92.22651       1.396529 5.266163   34.45476       \n",
       "vas22          152.48148       4.970738 4.674581   33.94644       \n",
       "wisp22         363.06414      22.676776 3.938680   33.51048       \n",
       "BigH122        167.39856       6.282128 4.522783   33.44931       \n",
       "zpg22          197.11389       8.950697 4.308088   32.87151       \n",
       "swa22          220.39693      11.023102 4.196223   32.69064       \n",
       "krimp22         93.27223       2.016900 4.950309   32.46789       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"av\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"dhd\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"dhd\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"dhd\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"yl\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Pxt\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type aw\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# already assigned with clster 107 part as nephrocyte query -- should be Garland cell nephrocyte\n",
    "# df_sub300 = set.cell.type(df_sub300, \"aw\", df_sub300$cluster %in% aw)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 237,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'aw' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'aw' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(aw)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'aw' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'aw' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(aw)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1465,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG4225523</th><td>aw</td><td>CG42255   </td><td>254</td><td>  243</td><td>2602.44085</td><td>1.097149e+01</td><td>7.764119</td><td>88.09327</td></tr>\n",
       "\t<tr><th scope=row>Amnionless23</th><td>aw</td><td>Amnionless</td><td>245</td><td>  265</td><td>2465.74382</td><td>1.007212e+01</td><td>7.798948</td><td>87.88161</td></tr>\n",
       "\t<tr><th scope=row>Cubn23</th><td>aw</td><td>Cubn      </td><td>451</td><td> 1002</td><td>4443.21343</td><td>4.333554e+01</td><td>6.646996</td><td>80.54639</td></tr>\n",
       "\t<tr><th scope=row>CG1520923</th><td>aw</td><td>CG15209   </td><td> 29</td><td>  275</td><td> 297.63488</td><td>1.134813e+01</td><td>4.591179</td><td>37.74977</td></tr>\n",
       "\t<tr><th scope=row>CG1559923</th><td>aw</td><td>CG15599   </td><td> 10</td><td>   34</td><td> 109.51322</td><td>1.474417e+00</td><td>5.467873</td><td>37.11633</td></tr>\n",
       "\t<tr><th scope=row>eIF4E623</th><td>aw</td><td>eIF4E6    </td><td>  7</td><td>    0</td><td>  50.77040</td><td>0.000000e+00</td><td>5.665916</td><td>32.26204</td></tr>\n",
       "\t<tr><th scope=row>Klf1523</th><td>aw</td><td>Klf15     </td><td>  5</td><td>    2</td><td>  51.76960</td><td>4.129939e-02</td><td>5.635648</td><td>32.24512</td></tr>\n",
       "\t<tr><th scope=row>CG911723</th><td>aw</td><td>CG9117    </td><td>  6</td><td>   22</td><td>  62.84368</td><td>1.025332e+00</td><td>4.955537</td><td>29.71574</td></tr>\n",
       "\t<tr><th scope=row>CG460723</th><td>aw</td><td>CG4607    </td><td>  5</td><td>   11</td><td>  45.73831</td><td>2.512329e-01</td><td>5.191981</td><td>28.79750</td></tr>\n",
       "\t<tr><th scope=row>CG1379423</th><td>aw</td><td>CG13794   </td><td>  6</td><td>   13</td><td>  49.05131</td><td>5.900474e-01</td><td>4.947150</td><td>27.92832</td></tr>\n",
       "\t<tr><th scope=row>CG1359423</th><td>aw</td><td>CG13594   </td><td> 10</td><td>  198</td><td> 113.40990</td><td>8.663901e+00</td><td>3.552797</td><td>24.29427</td></tr>\n",
       "\t<tr><th scope=row>CG182725</th><td>aw</td><td>CG1827    </td><td>  8</td><td>  186</td><td> 105.83455</td><td>7.785807e+00</td><td>3.590492</td><td>24.19717</td></tr>\n",
       "\t<tr><th scope=row>CG3443423</th><td>aw</td><td>CG34434   </td><td>  3</td><td>   19</td><td>  32.58921</td><td>5.063570e-01</td><td>4.435259</td><td>22.48643</td></tr>\n",
       "\t<tr><th scope=row>GstE223</th><td>aw</td><td>GstE2     </td><td>  3</td><td>   32</td><td>  39.37323</td><td>1.245903e+00</td><td>4.131847</td><td>22.04476</td></tr>\n",
       "\t<tr><th scope=row>Act57B23</th><td>aw</td><td>Act57B    </td><td>574</td><td>54854</td><td>5573.04513</td><td>1.649797e+03</td><td>1.755303</td><td>21.84388</td></tr>\n",
       "\t<tr><th scope=row>CG3433023</th><td>aw</td><td>CG34330   </td><td>  5</td><td>   66</td><td>  43.46485</td><td>1.786811e+00</td><td>3.963162</td><td>21.69670</td></tr>\n",
       "\t<tr><th scope=row>Mtpbeta23</th><td>aw</td><td>Mtpbeta   </td><td> 32</td><td> 1245</td><td> 263.69744</td><td>4.213168e+01</td><td>2.612064</td><td>21.02241</td></tr>\n",
       "\t<tr><th scope=row>Cog723</th><td>aw</td><td>Cog7      </td><td> 18</td><td>  690</td><td> 179.76026</td><td>2.549003e+01</td><td>2.762553</td><td>20.71344</td></tr>\n",
       "\t<tr><th scope=row>fbp23</th><td>aw</td><td>fbp       </td><td> 43</td><td> 2172</td><td> 394.61468</td><td>7.501023e+01</td><td>2.376179</td><td>20.50156</td></tr>\n",
       "\t<tr><th scope=row>CG276723</th><td>aw</td><td>CG2767    </td><td> 38</td><td> 2138</td><td> 400.02161</td><td>7.690775e+01</td><td>2.360239</td><td>20.41025</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG4225523 & aw & CG42255    & 254 &   243 & 2602.44085 & 1.097149e+01 & 7.764119 & 88.09327\\\\\n",
       "\tAmnionless23 & aw & Amnionless & 245 &   265 & 2465.74382 & 1.007212e+01 & 7.798948 & 87.88161\\\\\n",
       "\tCubn23 & aw & Cubn       & 451 &  1002 & 4443.21343 & 4.333554e+01 & 6.646996 & 80.54639\\\\\n",
       "\tCG1520923 & aw & CG15209    &  29 &   275 &  297.63488 & 1.134813e+01 & 4.591179 & 37.74977\\\\\n",
       "\tCG1559923 & aw & CG15599    &  10 &    34 &  109.51322 & 1.474417e+00 & 5.467873 & 37.11633\\\\\n",
       "\teIF4E623 & aw & eIF4E6     &   7 &     0 &   50.77040 & 0.000000e+00 & 5.665916 & 32.26204\\\\\n",
       "\tKlf1523 & aw & Klf15      &   5 &     2 &   51.76960 & 4.129939e-02 & 5.635648 & 32.24512\\\\\n",
       "\tCG911723 & aw & CG9117     &   6 &    22 &   62.84368 & 1.025332e+00 & 4.955537 & 29.71574\\\\\n",
       "\tCG460723 & aw & CG4607     &   5 &    11 &   45.73831 & 2.512329e-01 & 5.191981 & 28.79750\\\\\n",
       "\tCG1379423 & aw & CG13794    &   6 &    13 &   49.05131 & 5.900474e-01 & 4.947150 & 27.92832\\\\\n",
       "\tCG1359423 & aw & CG13594    &  10 &   198 &  113.40990 & 8.663901e+00 & 3.552797 & 24.29427\\\\\n",
       "\tCG182725 & aw & CG1827     &   8 &   186 &  105.83455 & 7.785807e+00 & 3.590492 & 24.19717\\\\\n",
       "\tCG3443423 & aw & CG34434    &   3 &    19 &   32.58921 & 5.063570e-01 & 4.435259 & 22.48643\\\\\n",
       "\tGstE223 & aw & GstE2      &   3 &    32 &   39.37323 & 1.245903e+00 & 4.131847 & 22.04476\\\\\n",
       "\tAct57B23 & aw & Act57B     & 574 & 54854 & 5573.04513 & 1.649797e+03 & 1.755303 & 21.84388\\\\\n",
       "\tCG3433023 & aw & CG34330    &   5 &    66 &   43.46485 & 1.786811e+00 & 3.963162 & 21.69670\\\\\n",
       "\tMtpbeta23 & aw & Mtpbeta    &  32 &  1245 &  263.69744 & 4.213168e+01 & 2.612064 & 21.02241\\\\\n",
       "\tCog723 & aw & Cog7       &  18 &   690 &  179.76026 & 2.549003e+01 & 2.762553 & 20.71344\\\\\n",
       "\tfbp23 & aw & fbp        &  43 &  2172 &  394.61468 & 7.501023e+01 & 2.376179 & 20.50156\\\\\n",
       "\tCG276723 & aw & CG2767     &  38 &  2138 &  400.02161 & 7.690775e+01 & 2.360239 & 20.41025\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG4225523 | aw | CG42255    | 254 |   243 | 2602.44085 | 1.097149e+01 | 7.764119 | 88.09327 |\n",
       "| Amnionless23 | aw | Amnionless | 245 |   265 | 2465.74382 | 1.007212e+01 | 7.798948 | 87.88161 |\n",
       "| Cubn23 | aw | Cubn       | 451 |  1002 | 4443.21343 | 4.333554e+01 | 6.646996 | 80.54639 |\n",
       "| CG1520923 | aw | CG15209    |  29 |   275 |  297.63488 | 1.134813e+01 | 4.591179 | 37.74977 |\n",
       "| CG1559923 | aw | CG15599    |  10 |    34 |  109.51322 | 1.474417e+00 | 5.467873 | 37.11633 |\n",
       "| eIF4E623 | aw | eIF4E6     |   7 |     0 |   50.77040 | 0.000000e+00 | 5.665916 | 32.26204 |\n",
       "| Klf1523 | aw | Klf15      |   5 |     2 |   51.76960 | 4.129939e-02 | 5.635648 | 32.24512 |\n",
       "| CG911723 | aw | CG9117     |   6 |    22 |   62.84368 | 1.025332e+00 | 4.955537 | 29.71574 |\n",
       "| CG460723 | aw | CG4607     |   5 |    11 |   45.73831 | 2.512329e-01 | 5.191981 | 28.79750 |\n",
       "| CG1379423 | aw | CG13794    |   6 |    13 |   49.05131 | 5.900474e-01 | 4.947150 | 27.92832 |\n",
       "| CG1359423 | aw | CG13594    |  10 |   198 |  113.40990 | 8.663901e+00 | 3.552797 | 24.29427 |\n",
       "| CG182725 | aw | CG1827     |   8 |   186 |  105.83455 | 7.785807e+00 | 3.590492 | 24.19717 |\n",
       "| CG3443423 | aw | CG34434    |   3 |    19 |   32.58921 | 5.063570e-01 | 4.435259 | 22.48643 |\n",
       "| GstE223 | aw | GstE2      |   3 |    32 |   39.37323 | 1.245903e+00 | 4.131847 | 22.04476 |\n",
       "| Act57B23 | aw | Act57B     | 574 | 54854 | 5573.04513 | 1.649797e+03 | 1.755303 | 21.84388 |\n",
       "| CG3433023 | aw | CG34330    |   5 |    66 |   43.46485 | 1.786811e+00 | 3.963162 | 21.69670 |\n",
       "| Mtpbeta23 | aw | Mtpbeta    |  32 |  1245 |  263.69744 | 4.213168e+01 | 2.612064 | 21.02241 |\n",
       "| Cog723 | aw | Cog7       |  18 |   690 |  179.76026 | 2.549003e+01 | 2.762553 | 20.71344 |\n",
       "| fbp23 | aw | fbp        |  43 |  2172 |  394.61468 | 7.501023e+01 | 2.376179 | 20.50156 |\n",
       "| CG276723 | aw | CG2767     |  38 |  2138 |  400.02161 | 7.690775e+01 | 2.360239 | 20.41025 |\n",
       "\n"
      ],
      "text/plain": [
       "             cell.type.2 gene       cell.type.2.n.umi other.n.umi\n",
       "CG4225523    aw          CG42255    254                 243      \n",
       "Amnionless23 aw          Amnionless 245                 265      \n",
       "Cubn23       aw          Cubn       451                1002      \n",
       "CG1520923    aw          CG15209     29                 275      \n",
       "CG1559923    aw          CG15599     10                  34      \n",
       "eIF4E623     aw          eIF4E6       7                   0      \n",
       "Klf1523      aw          Klf15        5                   2      \n",
       "CG911723     aw          CG9117       6                  22      \n",
       "CG460723     aw          CG4607       5                  11      \n",
       "CG1379423    aw          CG13794      6                  13      \n",
       "CG1359423    aw          CG13594     10                 198      \n",
       "CG182725     aw          CG1827       8                 186      \n",
       "CG3443423    aw          CG34434      3                  19      \n",
       "GstE223      aw          GstE2        3                  32      \n",
       "Act57B23     aw          Act57B     574               54854      \n",
       "CG3433023    aw          CG34330      5                  66      \n",
       "Mtpbeta23    aw          Mtpbeta     32                1245      \n",
       "Cog723       aw          Cog7        18                 690      \n",
       "fbp23        aw          fbp         43                2172      \n",
       "CG276723     aw          CG2767      38                2138      \n",
       "             cell.type.2.tpm other.tpm    log2.ratio heuristic.score\n",
       "CG4225523    2602.44085      1.097149e+01 7.764119   88.09327       \n",
       "Amnionless23 2465.74382      1.007212e+01 7.798948   87.88161       \n",
       "Cubn23       4443.21343      4.333554e+01 6.646996   80.54639       \n",
       "CG1520923     297.63488      1.134813e+01 4.591179   37.74977       \n",
       "CG1559923     109.51322      1.474417e+00 5.467873   37.11633       \n",
       "eIF4E623       50.77040      0.000000e+00 5.665916   32.26204       \n",
       "Klf1523        51.76960      4.129939e-02 5.635648   32.24512       \n",
       "CG911723       62.84368      1.025332e+00 4.955537   29.71574       \n",
       "CG460723       45.73831      2.512329e-01 5.191981   28.79750       \n",
       "CG1379423      49.05131      5.900474e-01 4.947150   27.92832       \n",
       "CG1359423     113.40990      8.663901e+00 3.552797   24.29427       \n",
       "CG182725      105.83455      7.785807e+00 3.590492   24.19717       \n",
       "CG3443423      32.58921      5.063570e-01 4.435259   22.48643       \n",
       "GstE223        39.37323      1.245903e+00 4.131847   22.04476       \n",
       "Act57B23     5573.04513      1.649797e+03 1.755303   21.84388       \n",
       "CG3433023      43.46485      1.786811e+00 3.963162   21.69670       \n",
       "Mtpbeta23     263.69744      4.213168e+01 2.612064   21.02241       \n",
       "Cog723        179.76026      2.549003e+01 2.762553   20.71344       \n",
       "fbp23         394.61468      7.501023e+01 2.376179   20.50156       \n",
       "CG276723      400.02161      7.690775e+01 2.360239   20.41025       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"aw\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG42255\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG42255\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG42255\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Amnionless\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Cubn\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1195,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Copper_cells\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Copper_cells\", df_sub300$cluster %in% ax)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ax' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ax' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ax)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "clus = c(113)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1191,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Ptx124</th><td>ax</td><td>Ptx1   </td><td> 4491</td><td> 2232</td><td> 3311.0703</td><td>  94.006046</td><td>5.123135</td><td>59.90746</td></tr>\n",
       "\t<tr><th scope=row>HHEX24</th><td>ax</td><td>HHEX   </td><td>  942</td><td>  336</td><td>  723.9475</td><td>  15.219785</td><td>5.480058</td><td>52.07005</td></tr>\n",
       "\t<tr><th scope=row>5-HT724</th><td>ax</td><td>5-HT7  </td><td> 2123</td><td> 1126</td><td> 1497.3070</td><td>  53.128577</td><td>4.789836</td><td>50.52854</td></tr>\n",
       "\t<tr><th scope=row>lab24</th><td>ax</td><td>lab    </td><td> 2838</td><td> 2318</td><td> 2035.5743</td><td> 104.666432</td><td>4.267847</td><td>46.91187</td></tr>\n",
       "\t<tr><th scope=row>dve24</th><td>ax</td><td>dve    </td><td> 2524</td><td> 3176</td><td> 1834.8667</td><td> 137.924853</td><td>3.723299</td><td>40.36892</td></tr>\n",
       "\t<tr><th scope=row>kcc24</th><td>ax</td><td>kcc    </td><td> 2112</td><td> 2379</td><td> 1410.7723</td><td> 101.076906</td><td>3.788757</td><td>39.64287</td></tr>\n",
       "\t<tr><th scope=row>fa2h24</th><td>ax</td><td>fa2h   </td><td>  882</td><td>  821</td><td>  662.1973</td><td>  35.256650</td><td>4.190944</td><td>39.28295</td></tr>\n",
       "\t<tr><th scope=row>CG3434724</th><td>ax</td><td>CG34347</td><td> 5626</td><td> 9291</td><td> 4085.3241</td><td> 439.022408</td><td>3.214802</td><td>38.56665</td></tr>\n",
       "\t<tr><th scope=row>rgr24</th><td>ax</td><td>rgr    </td><td>  943</td><td> 1064</td><td>  660.9783</td><td>  44.802330</td><td>3.851110</td><td>36.08737</td></tr>\n",
       "\t<tr><th scope=row>CG3175124</th><td>ax</td><td>CG31751</td><td>  743</td><td>  873</td><td>  519.2845</td><td>  33.970611</td><td>3.892310</td><td>35.12093</td></tr>\n",
       "\t<tr><th scope=row>CG3100724</th><td>ax</td><td>CG31007</td><td>  249</td><td>  108</td><td>  160.2359</td><td>   5.369431</td><td>4.652889</td><td>34.11977</td></tr>\n",
       "\t<tr><th scope=row>Itgbn24</th><td>ax</td><td>Itgbn  </td><td>  675</td><td>  763</td><td>  505.7177</td><td>  35.386759</td><td>3.796847</td><td>34.11481</td></tr>\n",
       "\t<tr><th scope=row>trol24</th><td>ax</td><td>trol   </td><td>15827</td><td>43473</td><td>11551.4716</td><td>2063.775978</td><td>2.484019</td><td>33.52411</td></tr>\n",
       "\t<tr><th scope=row>CG1297424</th><td>ax</td><td>CG12974</td><td>  397</td><td>  417</td><td>  296.6145</td><td>  17.723640</td><td>3.985656</td><td>32.75134</td></tr>\n",
       "\t<tr><th scope=row>bwa24</th><td>ax</td><td>bwa    </td><td> 1200</td><td> 2174</td><td>  922.8649</td><td>  96.096451</td><td>3.248629</td><td>32.00399</td></tr>\n",
       "\t<tr><th scope=row>Gmap24</th><td>ax</td><td>Gmap   </td><td> 5041</td><td>13161</td><td> 3763.6350</td><td> 624.284057</td><td>2.589543</td><td>30.75936</td></tr>\n",
       "\t<tr><th scope=row>mesh24</th><td>ax</td><td>mesh   </td><td> 2359</td><td> 5852</td><td> 1684.5979</td><td> 241.330071</td><td>2.797359</td><td>29.98502</td></tr>\n",
       "\t<tr><th scope=row>slo24</th><td>ax</td><td>slo    </td><td> 3508</td><td> 7689</td><td> 2234.7319</td><td> 350.007332</td><td>2.670529</td><td>29.71372</td></tr>\n",
       "\t<tr><th scope=row>ush24</th><td>ax</td><td>ush    </td><td> 2779</td><td> 7163</td><td> 2037.8001</td><td> 319.582744</td><td>2.668244</td><td>29.33335</td></tr>\n",
       "\t<tr><th scope=row>MESR324</th><td>ax</td><td>MESR3  </td><td> 1664</td><td> 3693</td><td> 1181.8889</td><td> 165.815607</td><td>2.824768</td><td>28.83551</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tPtx124 & ax & Ptx1    &  4491 &  2232 &  3311.0703 &   94.006046 & 5.123135 & 59.90746\\\\\n",
       "\tHHEX24 & ax & HHEX    &   942 &   336 &   723.9475 &   15.219785 & 5.480058 & 52.07005\\\\\n",
       "\t5-HT724 & ax & 5-HT7   &  2123 &  1126 &  1497.3070 &   53.128577 & 4.789836 & 50.52854\\\\\n",
       "\tlab24 & ax & lab     &  2838 &  2318 &  2035.5743 &  104.666432 & 4.267847 & 46.91187\\\\\n",
       "\tdve24 & ax & dve     &  2524 &  3176 &  1834.8667 &  137.924853 & 3.723299 & 40.36892\\\\\n",
       "\tkcc24 & ax & kcc     &  2112 &  2379 &  1410.7723 &  101.076906 & 3.788757 & 39.64287\\\\\n",
       "\tfa2h24 & ax & fa2h    &   882 &   821 &   662.1973 &   35.256650 & 4.190944 & 39.28295\\\\\n",
       "\tCG3434724 & ax & CG34347 &  5626 &  9291 &  4085.3241 &  439.022408 & 3.214802 & 38.56665\\\\\n",
       "\trgr24 & ax & rgr     &   943 &  1064 &   660.9783 &   44.802330 & 3.851110 & 36.08737\\\\\n",
       "\tCG3175124 & ax & CG31751 &   743 &   873 &   519.2845 &   33.970611 & 3.892310 & 35.12093\\\\\n",
       "\tCG3100724 & ax & CG31007 &   249 &   108 &   160.2359 &    5.369431 & 4.652889 & 34.11977\\\\\n",
       "\tItgbn24 & ax & Itgbn   &   675 &   763 &   505.7177 &   35.386759 & 3.796847 & 34.11481\\\\\n",
       "\ttrol24 & ax & trol    & 15827 & 43473 & 11551.4716 & 2063.775978 & 2.484019 & 33.52411\\\\\n",
       "\tCG1297424 & ax & CG12974 &   397 &   417 &   296.6145 &   17.723640 & 3.985656 & 32.75134\\\\\n",
       "\tbwa24 & ax & bwa     &  1200 &  2174 &   922.8649 &   96.096451 & 3.248629 & 32.00399\\\\\n",
       "\tGmap24 & ax & Gmap    &  5041 & 13161 &  3763.6350 &  624.284057 & 2.589543 & 30.75936\\\\\n",
       "\tmesh24 & ax & mesh    &  2359 &  5852 &  1684.5979 &  241.330071 & 2.797359 & 29.98502\\\\\n",
       "\tslo24 & ax & slo     &  3508 &  7689 &  2234.7319 &  350.007332 & 2.670529 & 29.71372\\\\\n",
       "\tush24 & ax & ush     &  2779 &  7163 &  2037.8001 &  319.582744 & 2.668244 & 29.33335\\\\\n",
       "\tMESR324 & ax & MESR3   &  1664 &  3693 &  1181.8889 &  165.815607 & 2.824768 & 28.83551\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Ptx124 | ax | Ptx1    |  4491 |  2232 |  3311.0703 |   94.006046 | 5.123135 | 59.90746 |\n",
       "| HHEX24 | ax | HHEX    |   942 |   336 |   723.9475 |   15.219785 | 5.480058 | 52.07005 |\n",
       "| 5-HT724 | ax | 5-HT7   |  2123 |  1126 |  1497.3070 |   53.128577 | 4.789836 | 50.52854 |\n",
       "| lab24 | ax | lab     |  2838 |  2318 |  2035.5743 |  104.666432 | 4.267847 | 46.91187 |\n",
       "| dve24 | ax | dve     |  2524 |  3176 |  1834.8667 |  137.924853 | 3.723299 | 40.36892 |\n",
       "| kcc24 | ax | kcc     |  2112 |  2379 |  1410.7723 |  101.076906 | 3.788757 | 39.64287 |\n",
       "| fa2h24 | ax | fa2h    |   882 |   821 |   662.1973 |   35.256650 | 4.190944 | 39.28295 |\n",
       "| CG3434724 | ax | CG34347 |  5626 |  9291 |  4085.3241 |  439.022408 | 3.214802 | 38.56665 |\n",
       "| rgr24 | ax | rgr     |   943 |  1064 |   660.9783 |   44.802330 | 3.851110 | 36.08737 |\n",
       "| CG3175124 | ax | CG31751 |   743 |   873 |   519.2845 |   33.970611 | 3.892310 | 35.12093 |\n",
       "| CG3100724 | ax | CG31007 |   249 |   108 |   160.2359 |    5.369431 | 4.652889 | 34.11977 |\n",
       "| Itgbn24 | ax | Itgbn   |   675 |   763 |   505.7177 |   35.386759 | 3.796847 | 34.11481 |\n",
       "| trol24 | ax | trol    | 15827 | 43473 | 11551.4716 | 2063.775978 | 2.484019 | 33.52411 |\n",
       "| CG1297424 | ax | CG12974 |   397 |   417 |   296.6145 |   17.723640 | 3.985656 | 32.75134 |\n",
       "| bwa24 | ax | bwa     |  1200 |  2174 |   922.8649 |   96.096451 | 3.248629 | 32.00399 |\n",
       "| Gmap24 | ax | Gmap    |  5041 | 13161 |  3763.6350 |  624.284057 | 2.589543 | 30.75936 |\n",
       "| mesh24 | ax | mesh    |  2359 |  5852 |  1684.5979 |  241.330071 | 2.797359 | 29.98502 |\n",
       "| slo24 | ax | slo     |  3508 |  7689 |  2234.7319 |  350.007332 | 2.670529 | 29.71372 |\n",
       "| ush24 | ax | ush     |  2779 |  7163 |  2037.8001 |  319.582744 | 2.668244 | 29.33335 |\n",
       "| MESR324 | ax | MESR3   |  1664 |  3693 |  1181.8889 |  165.815607 | 2.824768 | 28.83551 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "Ptx124    ax          Ptx1     4491              2232        3311.0703     \n",
       "HHEX24    ax          HHEX      942               336         723.9475     \n",
       "5-HT724   ax          5-HT7    2123              1126        1497.3070     \n",
       "lab24     ax          lab      2838              2318        2035.5743     \n",
       "dve24     ax          dve      2524              3176        1834.8667     \n",
       "kcc24     ax          kcc      2112              2379        1410.7723     \n",
       "fa2h24    ax          fa2h      882               821         662.1973     \n",
       "CG3434724 ax          CG34347  5626              9291        4085.3241     \n",
       "rgr24     ax          rgr       943              1064         660.9783     \n",
       "CG3175124 ax          CG31751   743               873         519.2845     \n",
       "CG3100724 ax          CG31007   249               108         160.2359     \n",
       "Itgbn24   ax          Itgbn     675               763         505.7177     \n",
       "trol24    ax          trol    15827             43473       11551.4716     \n",
       "CG1297424 ax          CG12974   397               417         296.6145     \n",
       "bwa24     ax          bwa      1200              2174         922.8649     \n",
       "Gmap24    ax          Gmap     5041             13161        3763.6350     \n",
       "mesh24    ax          mesh     2359              5852        1684.5979     \n",
       "slo24     ax          slo      3508              7689        2234.7319     \n",
       "ush24     ax          ush      2779              7163        2037.8001     \n",
       "MESR324   ax          MESR3    1664              3693        1181.8889     \n",
       "          other.tpm   log2.ratio heuristic.score\n",
       "Ptx124      94.006046 5.123135   59.90746       \n",
       "HHEX24      15.219785 5.480058   52.07005       \n",
       "5-HT724     53.128577 4.789836   50.52854       \n",
       "lab24      104.666432 4.267847   46.91187       \n",
       "dve24      137.924853 3.723299   40.36892       \n",
       "kcc24      101.076906 3.788757   39.64287       \n",
       "fa2h24      35.256650 4.190944   39.28295       \n",
       "CG3434724  439.022408 3.214802   38.56665       \n",
       "rgr24       44.802330 3.851110   36.08737       \n",
       "CG3175124   33.970611 3.892310   35.12093       \n",
       "CG3100724    5.369431 4.652889   34.11977       \n",
       "Itgbn24     35.386759 3.796847   34.11481       \n",
       "trol24    2063.775978 2.484019   33.52411       \n",
       "CG1297424   17.723640 3.985656   32.75134       \n",
       "bwa24       96.096451 3.248629   32.00399       \n",
       "Gmap24     624.284057 2.589543   30.75936       \n",
       "mesh24     241.330071 2.797359   29.98502       \n",
       "slo24      350.007332 2.670529   29.71372       \n",
       "ush24      319.582744 2.668244   29.33335       \n",
       "MESR324    165.815607 2.824768   28.83551       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"ax\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Ptx1\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Ptx1\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Ptx1\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"HHEX\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"5-HT7\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 242,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"dve\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"dve\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"dve\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"kcc\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"fa2h\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1472,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Muscle_system\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Muscle_system\", df_sub300$cluster %in% ay)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ay' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ay' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ay)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ay' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ay' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ay)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1470,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>up25</th><td>ay</td><td>up       </td><td>22358</td><td>25378</td><td>13769.3911</td><td> 982.45115</td><td>3.807468</td><td>52.34994</td></tr>\n",
       "\t<tr><th scope=row>bt25</th><td>ay</td><td>bt       </td><td>28283</td><td>41951</td><td>16887.2548</td><td>1395.97036</td><td>3.595562</td><td>50.49510</td></tr>\n",
       "\t<tr><th scope=row>Sh25</th><td>ay</td><td>Sh       </td><td> 8966</td><td> 8593</td><td> 5475.7467</td><td> 373.11780</td><td>3.871491</td><td>48.08045</td></tr>\n",
       "\t<tr><th scope=row>wupA25</th><td>ay</td><td>wupA     </td><td>13792</td><td>18894</td><td> 8349.8766</td><td> 704.90658</td><td>3.564206</td><td>46.43345</td></tr>\n",
       "\t<tr><th scope=row>Mhc25</th><td>ay</td><td>Mhc      </td><td>22946</td><td>47636</td><td>13896.7641</td><td>1556.24510</td><td>3.157681</td><td>43.45779</td></tr>\n",
       "\t<tr><th scope=row>CG3352125</th><td>ay</td><td>CG33521  </td><td> 2446</td><td> 2291</td><td> 1491.0209</td><td>  91.90377</td><td>4.004420</td><td>42.21880</td></tr>\n",
       "\t<tr><th scope=row>Nlg125</th><td>ay</td><td>Nlg1     </td><td>  930</td><td>  631</td><td>  564.4749</td><td>  26.27962</td><td>4.371014</td><td>39.96557</td></tr>\n",
       "\t<tr><th scope=row>Neto25</th><td>ay</td><td>Neto     </td><td> 1302</td><td> 1094</td><td>  801.4888</td><td>  47.08634</td><td>4.058983</td><td>39.16244</td></tr>\n",
       "\t<tr><th scope=row>CG167445</th><td>ay</td><td>CG1674   </td><td> 1746</td><td> 2079</td><td> 1065.0881</td><td>  75.70511</td><td>3.795506</td><td>38.17562</td></tr>\n",
       "\t<tr><th scope=row>mspo25</th><td>ay</td><td>mspo     </td><td> 4351</td><td> 5865</td><td> 2593.9663</td><td> 252.98335</td><td>3.352354</td><td>38.02072</td></tr>\n",
       "\t<tr><th scope=row>Eip63F-125</th><td>ay</td><td>Eip63F-1 </td><td> 1964</td><td> 2399</td><td> 1166.2818</td><td> 102.37490</td><td>3.495959</td><td>35.62010</td></tr>\n",
       "\t<tr><th scope=row>Mrtf25</th><td>ay</td><td>Mrtf     </td><td> 4242</td><td> 7031</td><td> 2550.9637</td><td> 290.09031</td><td>3.131504</td><td>35.44045</td></tr>\n",
       "\t<tr><th scope=row>Mf25</th><td>ay</td><td>Mf       </td><td> 2650</td><td> 6533</td><td> 1587.7474</td><td> 169.21139</td><td>3.221582</td><td>34.25725</td></tr>\n",
       "\t<tr><th scope=row>sls25</th><td>ay</td><td>sls      </td><td>11482</td><td>32823</td><td> 6790.2901</td><td>1053.94573</td><td>2.686304</td><td>34.19523</td></tr>\n",
       "\t<tr><th scope=row>sals25</th><td>ay</td><td>sals     </td><td> 1615</td><td> 2230</td><td>  971.9315</td><td>  89.00055</td><td>3.432849</td><td>34.07513</td></tr>\n",
       "\t<tr><th scope=row>tn25</th><td>ay</td><td>tn       </td><td> 1306</td><td> 2069</td><td>  791.4483</td><td>  70.63789</td><td>3.465700</td><td>33.37529</td></tr>\n",
       "\t<tr><th scope=row>CG888825</th><td>ay</td><td>CG8888   </td><td> 1645</td><td> 2684</td><td>  976.1423</td><td>  94.06146</td><td>3.360159</td><td>33.37453</td></tr>\n",
       "\t<tr><th scope=row>l(2)0128925</th><td>ay</td><td>l(2)01289</td><td> 1815</td><td> 3143</td><td> 1078.7070</td><td> 111.09108</td><td>3.266560</td><td>32.91524</td></tr>\n",
       "\t<tr><th scope=row>CG3441725</th><td>ay</td><td>CG34417  </td><td> 4646</td><td> 9495</td><td> 2837.7830</td><td> 389.69774</td><td>2.860639</td><td>32.81456</td></tr>\n",
       "\t<tr><th scope=row>dpr1025</th><td>ay</td><td>dpr10    </td><td> 1331</td><td> 1763</td><td>  808.7869</td><td>  79.44960</td><td>3.329602</td><td>32.16862</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tup25 & ay & up        & 22358 & 25378 & 13769.3911 &  982.45115 & 3.807468 & 52.34994\\\\\n",
       "\tbt25 & ay & bt        & 28283 & 41951 & 16887.2548 & 1395.97036 & 3.595562 & 50.49510\\\\\n",
       "\tSh25 & ay & Sh        &  8966 &  8593 &  5475.7467 &  373.11780 & 3.871491 & 48.08045\\\\\n",
       "\twupA25 & ay & wupA      & 13792 & 18894 &  8349.8766 &  704.90658 & 3.564206 & 46.43345\\\\\n",
       "\tMhc25 & ay & Mhc       & 22946 & 47636 & 13896.7641 & 1556.24510 & 3.157681 & 43.45779\\\\\n",
       "\tCG3352125 & ay & CG33521   &  2446 &  2291 &  1491.0209 &   91.90377 & 4.004420 & 42.21880\\\\\n",
       "\tNlg125 & ay & Nlg1      &   930 &   631 &   564.4749 &   26.27962 & 4.371014 & 39.96557\\\\\n",
       "\tNeto25 & ay & Neto      &  1302 &  1094 &   801.4888 &   47.08634 & 4.058983 & 39.16244\\\\\n",
       "\tCG167445 & ay & CG1674    &  1746 &  2079 &  1065.0881 &   75.70511 & 3.795506 & 38.17562\\\\\n",
       "\tmspo25 & ay & mspo      &  4351 &  5865 &  2593.9663 &  252.98335 & 3.352354 & 38.02072\\\\\n",
       "\tEip63F-125 & ay & Eip63F-1  &  1964 &  2399 &  1166.2818 &  102.37490 & 3.495959 & 35.62010\\\\\n",
       "\tMrtf25 & ay & Mrtf      &  4242 &  7031 &  2550.9637 &  290.09031 & 3.131504 & 35.44045\\\\\n",
       "\tMf25 & ay & Mf        &  2650 &  6533 &  1587.7474 &  169.21139 & 3.221582 & 34.25725\\\\\n",
       "\tsls25 & ay & sls       & 11482 & 32823 &  6790.2901 & 1053.94573 & 2.686304 & 34.19523\\\\\n",
       "\tsals25 & ay & sals      &  1615 &  2230 &   971.9315 &   89.00055 & 3.432849 & 34.07513\\\\\n",
       "\ttn25 & ay & tn        &  1306 &  2069 &   791.4483 &   70.63789 & 3.465700 & 33.37529\\\\\n",
       "\tCG888825 & ay & CG8888    &  1645 &  2684 &   976.1423 &   94.06146 & 3.360159 & 33.37453\\\\\n",
       "\tl(2)0128925 & ay & l(2)01289 &  1815 &  3143 &  1078.7070 &  111.09108 & 3.266560 & 32.91524\\\\\n",
       "\tCG3441725 & ay & CG34417   &  4646 &  9495 &  2837.7830 &  389.69774 & 2.860639 & 32.81456\\\\\n",
       "\tdpr1025 & ay & dpr10     &  1331 &  1763 &   808.7869 &   79.44960 & 3.329602 & 32.16862\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| up25 | ay | up        | 22358 | 25378 | 13769.3911 |  982.45115 | 3.807468 | 52.34994 |\n",
       "| bt25 | ay | bt        | 28283 | 41951 | 16887.2548 | 1395.97036 | 3.595562 | 50.49510 |\n",
       "| Sh25 | ay | Sh        |  8966 |  8593 |  5475.7467 |  373.11780 | 3.871491 | 48.08045 |\n",
       "| wupA25 | ay | wupA      | 13792 | 18894 |  8349.8766 |  704.90658 | 3.564206 | 46.43345 |\n",
       "| Mhc25 | ay | Mhc       | 22946 | 47636 | 13896.7641 | 1556.24510 | 3.157681 | 43.45779 |\n",
       "| CG3352125 | ay | CG33521   |  2446 |  2291 |  1491.0209 |   91.90377 | 4.004420 | 42.21880 |\n",
       "| Nlg125 | ay | Nlg1      |   930 |   631 |   564.4749 |   26.27962 | 4.371014 | 39.96557 |\n",
       "| Neto25 | ay | Neto      |  1302 |  1094 |   801.4888 |   47.08634 | 4.058983 | 39.16244 |\n",
       "| CG167445 | ay | CG1674    |  1746 |  2079 |  1065.0881 |   75.70511 | 3.795506 | 38.17562 |\n",
       "| mspo25 | ay | mspo      |  4351 |  5865 |  2593.9663 |  252.98335 | 3.352354 | 38.02072 |\n",
       "| Eip63F-125 | ay | Eip63F-1  |  1964 |  2399 |  1166.2818 |  102.37490 | 3.495959 | 35.62010 |\n",
       "| Mrtf25 | ay | Mrtf      |  4242 |  7031 |  2550.9637 |  290.09031 | 3.131504 | 35.44045 |\n",
       "| Mf25 | ay | Mf        |  2650 |  6533 |  1587.7474 |  169.21139 | 3.221582 | 34.25725 |\n",
       "| sls25 | ay | sls       | 11482 | 32823 |  6790.2901 | 1053.94573 | 2.686304 | 34.19523 |\n",
       "| sals25 | ay | sals      |  1615 |  2230 |   971.9315 |   89.00055 | 3.432849 | 34.07513 |\n",
       "| tn25 | ay | tn        |  1306 |  2069 |   791.4483 |   70.63789 | 3.465700 | 33.37529 |\n",
       "| CG888825 | ay | CG8888    |  1645 |  2684 |   976.1423 |   94.06146 | 3.360159 | 33.37453 |\n",
       "| l(2)0128925 | ay | l(2)01289 |  1815 |  3143 |  1078.7070 |  111.09108 | 3.266560 | 32.91524 |\n",
       "| CG3441725 | ay | CG34417   |  4646 |  9495 |  2837.7830 |  389.69774 | 2.860639 | 32.81456 |\n",
       "| dpr1025 | ay | dpr10     |  1331 |  1763 |   808.7869 |   79.44960 | 3.329602 | 32.16862 |\n",
       "\n"
      ],
      "text/plain": [
       "            cell.type.2 gene      cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "up25        ay          up        22358             25378       13769.3911     \n",
       "bt25        ay          bt        28283             41951       16887.2548     \n",
       "Sh25        ay          Sh         8966              8593        5475.7467     \n",
       "wupA25      ay          wupA      13792             18894        8349.8766     \n",
       "Mhc25       ay          Mhc       22946             47636       13896.7641     \n",
       "CG3352125   ay          CG33521    2446              2291        1491.0209     \n",
       "Nlg125      ay          Nlg1        930               631         564.4749     \n",
       "Neto25      ay          Neto       1302              1094         801.4888     \n",
       "CG167445    ay          CG1674     1746              2079        1065.0881     \n",
       "mspo25      ay          mspo       4351              5865        2593.9663     \n",
       "Eip63F-125  ay          Eip63F-1   1964              2399        1166.2818     \n",
       "Mrtf25      ay          Mrtf       4242              7031        2550.9637     \n",
       "Mf25        ay          Mf         2650              6533        1587.7474     \n",
       "sls25       ay          sls       11482             32823        6790.2901     \n",
       "sals25      ay          sals       1615              2230         971.9315     \n",
       "tn25        ay          tn         1306              2069         791.4483     \n",
       "CG888825    ay          CG8888     1645              2684         976.1423     \n",
       "l(2)0128925 ay          l(2)01289  1815              3143        1078.7070     \n",
       "CG3441725   ay          CG34417    4646              9495        2837.7830     \n",
       "dpr1025     ay          dpr10      1331              1763         808.7869     \n",
       "            other.tpm  log2.ratio heuristic.score\n",
       "up25         982.45115 3.807468   52.34994       \n",
       "bt25        1395.97036 3.595562   50.49510       \n",
       "Sh25         373.11780 3.871491   48.08045       \n",
       "wupA25       704.90658 3.564206   46.43345       \n",
       "Mhc25       1556.24510 3.157681   43.45779       \n",
       "CG3352125     91.90377 4.004420   42.21880       \n",
       "Nlg125        26.27962 4.371014   39.96557       \n",
       "Neto25        47.08634 4.058983   39.16244       \n",
       "CG167445      75.70511 3.795506   38.17562       \n",
       "mspo25       252.98335 3.352354   38.02072       \n",
       "Eip63F-125   102.37490 3.495959   35.62010       \n",
       "Mrtf25       290.09031 3.131504   35.44045       \n",
       "Mf25         169.21139 3.221582   34.25725       \n",
       "sls25       1053.94573 2.686304   34.19523       \n",
       "sals25        89.00055 3.432849   34.07513       \n",
       "tn25          70.63789 3.465700   33.37529       \n",
       "CG888825      94.06146 3.360159   33.37453       \n",
       "l(2)0128925  111.09108 3.266560   32.91524       \n",
       "CG3441725    389.69774 2.860639   32.81456       \n",
       "dpr1025       79.44960 3.329602   32.16862       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"ay\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"up\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"up\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"up\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"bt\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Sh\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1477,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Central_brain_ventral_nerve_cord_primorium\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Central_brain_ventral_nerve_cord_primorium\", df_sub300$cluster %in% az)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'az' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'az' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(az)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 247,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'az' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'az' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(az)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1473,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>cas26</th><td>az</td><td>cas    </td><td> 6276</td><td> 4634</td><td>3616.4622</td><td> 193.18429</td><td>4.219081</td><td>49.87275</td></tr>\n",
       "\t<tr><th scope=row>jigr126</th><td>az</td><td>jigr1  </td><td> 3737</td><td> 4019</td><td>2109.8951</td><td> 157.73247</td><td>3.732502</td><td>41.22041</td></tr>\n",
       "\t<tr><th scope=row>nab26</th><td>az</td><td>nab    </td><td> 1002</td><td>  764</td><td> 626.5084</td><td>  29.41010</td><td>4.364711</td><td>40.56340</td></tr>\n",
       "\t<tr><th scope=row>klu26</th><td>az</td><td>klu    </td><td>10109</td><td>15169</td><td>5795.1566</td><td> 624.97172</td><td>3.210678</td><td>40.13631</td></tr>\n",
       "\t<tr><th scope=row>E2f126</th><td>az</td><td>E2f1   </td><td> 9740</td><td>19125</td><td>5419.8189</td><td> 740.67100</td><td>2.869393</td><td>35.59280</td></tr>\n",
       "\t<tr><th scope=row>ktub26</th><td>az</td><td>ktub   </td><td> 1103</td><td> 1077</td><td> 602.1780</td><td>  41.65751</td><td>3.819318</td><td>35.27690</td></tr>\n",
       "\t<tr><th scope=row>Sp126</th><td>az</td><td>Sp1    </td><td> 2483</td><td> 3837</td><td>1392.8530</td><td> 161.27798</td><td>3.101504</td><td>32.39478</td></tr>\n",
       "\t<tr><th scope=row>Pex726</th><td>az</td><td>Pex7   </td><td>  500</td><td>  476</td><td> 321.8340</td><td>  21.07623</td><td>3.865751</td><td>32.21967</td></tr>\n",
       "\t<tr><th scope=row>pros26</th><td>az</td><td>pros   </td><td> 2908</td><td> 6103</td><td>1699.4675</td><td> 221.06116</td><td>2.936054</td><td>31.50889</td></tr>\n",
       "\t<tr><th scope=row>Tet26</th><td>az</td><td>Tet    </td><td>16585</td><td>45019</td><td>9245.9552</td><td>1785.09673</td><td>2.372012</td><td>31.25070</td></tr>\n",
       "\t<tr><th scope=row>CG965026</th><td>az</td><td>CG9650 </td><td> 6983</td><td>15333</td><td>3968.7947</td><td> 648.19108</td><td>2.611986</td><td>31.22590</td></tr>\n",
       "\t<tr><th scope=row>sca26</th><td>az</td><td>sca    </td><td> 2393</td><td> 4661</td><td>1268.1146</td><td> 188.86542</td><td>2.739636</td><td>28.24457</td></tr>\n",
       "\t<tr><th scope=row>CycE26</th><td>az</td><td>CycE   </td><td> 2463</td><td> 5858</td><td>1347.0654</td><td> 209.26162</td><td>2.679562</td><td>27.85854</td></tr>\n",
       "\t<tr><th scope=row>seq26</th><td>az</td><td>seq    </td><td> 2430</td><td> 5510</td><td>1357.3460</td><td> 212.06803</td><td>2.671402</td><td>27.80298</td></tr>\n",
       "\t<tr><th scope=row>CG1564626</th><td>az</td><td>CG15646</td><td>  326</td><td>  400</td><td> 190.2632</td><td>  14.45424</td><td>3.621922</td><td>27.45205</td></tr>\n",
       "\t<tr><th scope=row>CG1771626</th><td>az</td><td>CG17716</td><td> 3415</td><td> 7635</td><td>1940.8627</td><td> 339.87958</td><td>2.509364</td><td>27.41035</td></tr>\n",
       "\t<tr><th scope=row>Kdm4B26</th><td>az</td><td>Kdm4B  </td><td> 3456</td><td> 8669</td><td>1948.7321</td><td> 365.02588</td><td>2.412518</td><td>26.36656</td></tr>\n",
       "\t<tr><th scope=row>tll26</th><td>az</td><td>tll    </td><td>  333</td><td>  453</td><td> 185.7926</td><td>  15.66867</td><td>3.478482</td><td>26.24617</td></tr>\n",
       "\t<tr><th scope=row>msi26</th><td>az</td><td>msi    </td><td> 5565</td><td>16715</td><td>3207.5199</td><td> 708.38055</td><td>2.176827</td><td>25.35501</td></tr>\n",
       "\t<tr><th scope=row>sktl26</th><td>az</td><td>sktl   </td><td> 1644</td><td> 3868</td><td> 938.1810</td><td> 157.31639</td><td>2.567056</td><td>25.35034</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tcas26 & az & cas     &  6276 &  4634 & 3616.4622 &  193.18429 & 4.219081 & 49.87275\\\\\n",
       "\tjigr126 & az & jigr1   &  3737 &  4019 & 2109.8951 &  157.73247 & 3.732502 & 41.22041\\\\\n",
       "\tnab26 & az & nab     &  1002 &   764 &  626.5084 &   29.41010 & 4.364711 & 40.56340\\\\\n",
       "\tklu26 & az & klu     & 10109 & 15169 & 5795.1566 &  624.97172 & 3.210678 & 40.13631\\\\\n",
       "\tE2f126 & az & E2f1    &  9740 & 19125 & 5419.8189 &  740.67100 & 2.869393 & 35.59280\\\\\n",
       "\tktub26 & az & ktub    &  1103 &  1077 &  602.1780 &   41.65751 & 3.819318 & 35.27690\\\\\n",
       "\tSp126 & az & Sp1     &  2483 &  3837 & 1392.8530 &  161.27798 & 3.101504 & 32.39478\\\\\n",
       "\tPex726 & az & Pex7    &   500 &   476 &  321.8340 &   21.07623 & 3.865751 & 32.21967\\\\\n",
       "\tpros26 & az & pros    &  2908 &  6103 & 1699.4675 &  221.06116 & 2.936054 & 31.50889\\\\\n",
       "\tTet26 & az & Tet     & 16585 & 45019 & 9245.9552 & 1785.09673 & 2.372012 & 31.25070\\\\\n",
       "\tCG965026 & az & CG9650  &  6983 & 15333 & 3968.7947 &  648.19108 & 2.611986 & 31.22590\\\\\n",
       "\tsca26 & az & sca     &  2393 &  4661 & 1268.1146 &  188.86542 & 2.739636 & 28.24457\\\\\n",
       "\tCycE26 & az & CycE    &  2463 &  5858 & 1347.0654 &  209.26162 & 2.679562 & 27.85854\\\\\n",
       "\tseq26 & az & seq     &  2430 &  5510 & 1357.3460 &  212.06803 & 2.671402 & 27.80298\\\\\n",
       "\tCG1564626 & az & CG15646 &   326 &   400 &  190.2632 &   14.45424 & 3.621922 & 27.45205\\\\\n",
       "\tCG1771626 & az & CG17716 &  3415 &  7635 & 1940.8627 &  339.87958 & 2.509364 & 27.41035\\\\\n",
       "\tKdm4B26 & az & Kdm4B   &  3456 &  8669 & 1948.7321 &  365.02588 & 2.412518 & 26.36656\\\\\n",
       "\ttll26 & az & tll     &   333 &   453 &  185.7926 &   15.66867 & 3.478482 & 26.24617\\\\\n",
       "\tmsi26 & az & msi     &  5565 & 16715 & 3207.5199 &  708.38055 & 2.176827 & 25.35501\\\\\n",
       "\tsktl26 & az & sktl    &  1644 &  3868 &  938.1810 &  157.31639 & 2.567056 & 25.35034\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| cas26 | az | cas     |  6276 |  4634 | 3616.4622 |  193.18429 | 4.219081 | 49.87275 |\n",
       "| jigr126 | az | jigr1   |  3737 |  4019 | 2109.8951 |  157.73247 | 3.732502 | 41.22041 |\n",
       "| nab26 | az | nab     |  1002 |   764 |  626.5084 |   29.41010 | 4.364711 | 40.56340 |\n",
       "| klu26 | az | klu     | 10109 | 15169 | 5795.1566 |  624.97172 | 3.210678 | 40.13631 |\n",
       "| E2f126 | az | E2f1    |  9740 | 19125 | 5419.8189 |  740.67100 | 2.869393 | 35.59280 |\n",
       "| ktub26 | az | ktub    |  1103 |  1077 |  602.1780 |   41.65751 | 3.819318 | 35.27690 |\n",
       "| Sp126 | az | Sp1     |  2483 |  3837 | 1392.8530 |  161.27798 | 3.101504 | 32.39478 |\n",
       "| Pex726 | az | Pex7    |   500 |   476 |  321.8340 |   21.07623 | 3.865751 | 32.21967 |\n",
       "| pros26 | az | pros    |  2908 |  6103 | 1699.4675 |  221.06116 | 2.936054 | 31.50889 |\n",
       "| Tet26 | az | Tet     | 16585 | 45019 | 9245.9552 | 1785.09673 | 2.372012 | 31.25070 |\n",
       "| CG965026 | az | CG9650  |  6983 | 15333 | 3968.7947 |  648.19108 | 2.611986 | 31.22590 |\n",
       "| sca26 | az | sca     |  2393 |  4661 | 1268.1146 |  188.86542 | 2.739636 | 28.24457 |\n",
       "| CycE26 | az | CycE    |  2463 |  5858 | 1347.0654 |  209.26162 | 2.679562 | 27.85854 |\n",
       "| seq26 | az | seq     |  2430 |  5510 | 1357.3460 |  212.06803 | 2.671402 | 27.80298 |\n",
       "| CG1564626 | az | CG15646 |   326 |   400 |  190.2632 |   14.45424 | 3.621922 | 27.45205 |\n",
       "| CG1771626 | az | CG17716 |  3415 |  7635 | 1940.8627 |  339.87958 | 2.509364 | 27.41035 |\n",
       "| Kdm4B26 | az | Kdm4B   |  3456 |  8669 | 1948.7321 |  365.02588 | 2.412518 | 26.36656 |\n",
       "| tll26 | az | tll     |   333 |   453 |  185.7926 |   15.66867 | 3.478482 | 26.24617 |\n",
       "| msi26 | az | msi     |  5565 | 16715 | 3207.5199 |  708.38055 | 2.176827 | 25.35501 |\n",
       "| sktl26 | az | sktl    |  1644 |  3868 |  938.1810 |  157.31639 | 2.567056 | 25.35034 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "cas26     az          cas      6276              4634       3616.4622      \n",
       "jigr126   az          jigr1    3737              4019       2109.8951      \n",
       "nab26     az          nab      1002               764        626.5084      \n",
       "klu26     az          klu     10109             15169       5795.1566      \n",
       "E2f126    az          E2f1     9740             19125       5419.8189      \n",
       "ktub26    az          ktub     1103              1077        602.1780      \n",
       "Sp126     az          Sp1      2483              3837       1392.8530      \n",
       "Pex726    az          Pex7      500               476        321.8340      \n",
       "pros26    az          pros     2908              6103       1699.4675      \n",
       "Tet26     az          Tet     16585             45019       9245.9552      \n",
       "CG965026  az          CG9650   6983             15333       3968.7947      \n",
       "sca26     az          sca      2393              4661       1268.1146      \n",
       "CycE26    az          CycE     2463              5858       1347.0654      \n",
       "seq26     az          seq      2430              5510       1357.3460      \n",
       "CG1564626 az          CG15646   326               400        190.2632      \n",
       "CG1771626 az          CG17716  3415              7635       1940.8627      \n",
       "Kdm4B26   az          Kdm4B    3456              8669       1948.7321      \n",
       "tll26     az          tll       333               453        185.7926      \n",
       "msi26     az          msi      5565             16715       3207.5199      \n",
       "sktl26    az          sktl     1644              3868        938.1810      \n",
       "          other.tpm  log2.ratio heuristic.score\n",
       "cas26      193.18429 4.219081   49.87275       \n",
       "jigr126    157.73247 3.732502   41.22041       \n",
       "nab26       29.41010 4.364711   40.56340       \n",
       "klu26      624.97172 3.210678   40.13631       \n",
       "E2f126     740.67100 2.869393   35.59280       \n",
       "ktub26      41.65751 3.819318   35.27690       \n",
       "Sp126      161.27798 3.101504   32.39478       \n",
       "Pex726      21.07623 3.865751   32.21967       \n",
       "pros26     221.06116 2.936054   31.50889       \n",
       "Tet26     1785.09673 2.372012   31.25070       \n",
       "CG965026   648.19108 2.611986   31.22590       \n",
       "sca26      188.86542 2.739636   28.24457       \n",
       "CycE26     209.26162 2.679562   27.85854       \n",
       "seq26      212.06803 2.671402   27.80298       \n",
       "CG1564626   14.45424 3.621922   27.45205       \n",
       "CG1771626  339.87958 2.509364   27.41035       \n",
       "Kdm4B26    365.02588 2.412518   26.36656       \n",
       "tll26       15.66867 3.478482   26.24617       \n",
       "msi26      708.38055 2.176827   25.35501       \n",
       "sktl26     157.31639 2.567056   25.35034       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"az\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 248,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"cas\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"cas\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"cas\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"jigr1\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"nab\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1484,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Digestive_system\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Digestive_system\", df_sub300$cluster %in% ba)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 249,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ba' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ba' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ba)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 250,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ba' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ba' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ba)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1479,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Lim328</th><td>ba</td><td>Lim3   </td><td>1891</td><td> 2733</td><td>2978.4440</td><td>119.979833</td><td>4.621720</td><td>53.33848</td></tr>\n",
       "\t<tr><th scope=row>CG456228</th><td>ba</td><td>CG4562 </td><td> 622</td><td>  711</td><td> 959.1802</td><td> 26.773968</td><td>5.109997</td><td>50.62556</td></tr>\n",
       "\t<tr><th scope=row>Jheh328</th><td>ba</td><td>Jheh3  </td><td> 971</td><td> 1860</td><td>1572.7622</td><td> 80.372302</td><td>4.272619</td><td>45.37522</td></tr>\n",
       "\t<tr><th scope=row>CG473428</th><td>ba</td><td>CG4734 </td><td> 153</td><td>  149</td><td> 263.8954</td><td>  6.117015</td><td>5.212550</td><td>41.95727</td></tr>\n",
       "\t<tr><th scope=row>CG1534728</th><td>ba</td><td>CG15347</td><td>1347</td><td> 3348</td><td>2160.6806</td><td>157.565114</td><td>3.768339</td><td>41.74542</td></tr>\n",
       "\t<tr><th scope=row>CG3399328</th><td>ba</td><td>CG33993</td><td>1506</td><td> 4527</td><td>2408.0650</td><td>204.686708</td><td>3.549354</td><td>39.87436</td></tr>\n",
       "\t<tr><th scope=row>grn28</th><td>ba</td><td>grn    </td><td>4268</td><td>17145</td><td>6592.5322</td><td>753.401411</td><td>3.127428</td><td>39.67717</td></tr>\n",
       "\t<tr><th scope=row>CG1536528</th><td>ba</td><td>CG15365</td><td> 917</td><td> 2851</td><td>1441.2191</td><td>134.747304</td><td>3.408294</td><td>35.76689</td></tr>\n",
       "\t<tr><th scope=row>Lfg28</th><td>ba</td><td>Lfg    </td><td> 502</td><td> 1467</td><td> 800.1733</td><td> 65.912156</td><td>3.579972</td><td>34.53231</td></tr>\n",
       "\t<tr><th scope=row>CG1581828</th><td>ba</td><td>CG15818</td><td> 287</td><td>  675</td><td> 454.5534</td><td> 30.450067</td><td>3.853315</td><td>34.03046</td></tr>\n",
       "\t<tr><th scope=row>CG1763628</th><td>ba</td><td>CG17636</td><td>  80</td><td>   99</td><td> 135.6902</td><td>  3.904068</td><td>4.790194</td><td>33.98530</td></tr>\n",
       "\t<tr><th scope=row>ana57</th><td>ba</td><td>ana    </td><td> 256</td><td>  624</td><td> 421.0107</td><td> 28.954694</td><td>3.813003</td><td>33.25372</td></tr>\n",
       "\t<tr><th scope=row>CG430128</th><td>ba</td><td>CG4301 </td><td> 306</td><td>  840</td><td> 494.5218</td><td> 39.372595</td><td>3.614586</td><td>32.36068</td></tr>\n",
       "\t<tr><th scope=row>SLC22A28</th><td>ba</td><td>SLC22A </td><td> 132</td><td>  297</td><td> 220.7942</td><td> 12.229042</td><td>4.060921</td><td>31.64708</td></tr>\n",
       "\t<tr><th scope=row>al28</th><td>ba</td><td>al     </td><td> 347</td><td> 1066</td><td> 535.1364</td><td> 46.810234</td><td>3.484515</td><td>31.59221</td></tr>\n",
       "\t<tr><th scope=row>sog28</th><td>ba</td><td>sog    </td><td>1685</td><td> 9075</td><td>2612.5670</td><td>380.269540</td><td>2.776585</td><td>31.51925</td></tr>\n",
       "\t<tr><th scope=row>Jheh128</th><td>ba</td><td>Jheh1  </td><td> 602</td><td> 2823</td><td> 978.3783</td><td>112.403214</td><td>3.108931</td><td>30.88947</td></tr>\n",
       "\t<tr><th scope=row>CG478128</th><td>ba</td><td>CG4781 </td><td> 346</td><td> 1165</td><td> 539.1020</td><td> 51.804672</td><td>3.351821</td><td>30.42477</td></tr>\n",
       "\t<tr><th scope=row>CG1065028</th><td>ba</td><td>CG10650</td><td> 437</td><td> 1808</td><td> 714.7974</td><td> 78.658457</td><td>3.165635</td><td>30.02101</td></tr>\n",
       "\t<tr><th scope=row>Agpat428</th><td>ba</td><td>Agpat4 </td><td> 685</td><td> 3213</td><td>1107.1772</td><td>143.415513</td><td>2.938588</td><td>29.72080</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tLim328 & ba & Lim3    & 1891 &  2733 & 2978.4440 & 119.979833 & 4.621720 & 53.33848\\\\\n",
       "\tCG456228 & ba & CG4562  &  622 &   711 &  959.1802 &  26.773968 & 5.109997 & 50.62556\\\\\n",
       "\tJheh328 & ba & Jheh3   &  971 &  1860 & 1572.7622 &  80.372302 & 4.272619 & 45.37522\\\\\n",
       "\tCG473428 & ba & CG4734  &  153 &   149 &  263.8954 &   6.117015 & 5.212550 & 41.95727\\\\\n",
       "\tCG1534728 & ba & CG15347 & 1347 &  3348 & 2160.6806 & 157.565114 & 3.768339 & 41.74542\\\\\n",
       "\tCG3399328 & ba & CG33993 & 1506 &  4527 & 2408.0650 & 204.686708 & 3.549354 & 39.87436\\\\\n",
       "\tgrn28 & ba & grn     & 4268 & 17145 & 6592.5322 & 753.401411 & 3.127428 & 39.67717\\\\\n",
       "\tCG1536528 & ba & CG15365 &  917 &  2851 & 1441.2191 & 134.747304 & 3.408294 & 35.76689\\\\\n",
       "\tLfg28 & ba & Lfg     &  502 &  1467 &  800.1733 &  65.912156 & 3.579972 & 34.53231\\\\\n",
       "\tCG1581828 & ba & CG15818 &  287 &   675 &  454.5534 &  30.450067 & 3.853315 & 34.03046\\\\\n",
       "\tCG1763628 & ba & CG17636 &   80 &    99 &  135.6902 &   3.904068 & 4.790194 & 33.98530\\\\\n",
       "\tana57 & ba & ana     &  256 &   624 &  421.0107 &  28.954694 & 3.813003 & 33.25372\\\\\n",
       "\tCG430128 & ba & CG4301  &  306 &   840 &  494.5218 &  39.372595 & 3.614586 & 32.36068\\\\\n",
       "\tSLC22A28 & ba & SLC22A  &  132 &   297 &  220.7942 &  12.229042 & 4.060921 & 31.64708\\\\\n",
       "\tal28 & ba & al      &  347 &  1066 &  535.1364 &  46.810234 & 3.484515 & 31.59221\\\\\n",
       "\tsog28 & ba & sog     & 1685 &  9075 & 2612.5670 & 380.269540 & 2.776585 & 31.51925\\\\\n",
       "\tJheh128 & ba & Jheh1   &  602 &  2823 &  978.3783 & 112.403214 & 3.108931 & 30.88947\\\\\n",
       "\tCG478128 & ba & CG4781  &  346 &  1165 &  539.1020 &  51.804672 & 3.351821 & 30.42477\\\\\n",
       "\tCG1065028 & ba & CG10650 &  437 &  1808 &  714.7974 &  78.658457 & 3.165635 & 30.02101\\\\\n",
       "\tAgpat428 & ba & Agpat4  &  685 &  3213 & 1107.1772 & 143.415513 & 2.938588 & 29.72080\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Lim328 | ba | Lim3    | 1891 |  2733 | 2978.4440 | 119.979833 | 4.621720 | 53.33848 |\n",
       "| CG456228 | ba | CG4562  |  622 |   711 |  959.1802 |  26.773968 | 5.109997 | 50.62556 |\n",
       "| Jheh328 | ba | Jheh3   |  971 |  1860 | 1572.7622 |  80.372302 | 4.272619 | 45.37522 |\n",
       "| CG473428 | ba | CG4734  |  153 |   149 |  263.8954 |   6.117015 | 5.212550 | 41.95727 |\n",
       "| CG1534728 | ba | CG15347 | 1347 |  3348 | 2160.6806 | 157.565114 | 3.768339 | 41.74542 |\n",
       "| CG3399328 | ba | CG33993 | 1506 |  4527 | 2408.0650 | 204.686708 | 3.549354 | 39.87436 |\n",
       "| grn28 | ba | grn     | 4268 | 17145 | 6592.5322 | 753.401411 | 3.127428 | 39.67717 |\n",
       "| CG1536528 | ba | CG15365 |  917 |  2851 | 1441.2191 | 134.747304 | 3.408294 | 35.76689 |\n",
       "| Lfg28 | ba | Lfg     |  502 |  1467 |  800.1733 |  65.912156 | 3.579972 | 34.53231 |\n",
       "| CG1581828 | ba | CG15818 |  287 |   675 |  454.5534 |  30.450067 | 3.853315 | 34.03046 |\n",
       "| CG1763628 | ba | CG17636 |   80 |    99 |  135.6902 |   3.904068 | 4.790194 | 33.98530 |\n",
       "| ana57 | ba | ana     |  256 |   624 |  421.0107 |  28.954694 | 3.813003 | 33.25372 |\n",
       "| CG430128 | ba | CG4301  |  306 |   840 |  494.5218 |  39.372595 | 3.614586 | 32.36068 |\n",
       "| SLC22A28 | ba | SLC22A  |  132 |   297 |  220.7942 |  12.229042 | 4.060921 | 31.64708 |\n",
       "| al28 | ba | al      |  347 |  1066 |  535.1364 |  46.810234 | 3.484515 | 31.59221 |\n",
       "| sog28 | ba | sog     | 1685 |  9075 | 2612.5670 | 380.269540 | 2.776585 | 31.51925 |\n",
       "| Jheh128 | ba | Jheh1   |  602 |  2823 |  978.3783 | 112.403214 | 3.108931 | 30.88947 |\n",
       "| CG478128 | ba | CG4781  |  346 |  1165 |  539.1020 |  51.804672 | 3.351821 | 30.42477 |\n",
       "| CG1065028 | ba | CG10650 |  437 |  1808 |  714.7974 |  78.658457 | 3.165635 | 30.02101 |\n",
       "| Agpat428 | ba | Agpat4  |  685 |  3213 | 1107.1772 | 143.415513 | 2.938588 | 29.72080 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "Lim328    ba          Lim3    1891               2733       2978.4440      \n",
       "CG456228  ba          CG4562   622                711        959.1802      \n",
       "Jheh328   ba          Jheh3    971               1860       1572.7622      \n",
       "CG473428  ba          CG4734   153                149        263.8954      \n",
       "CG1534728 ba          CG15347 1347               3348       2160.6806      \n",
       "CG3399328 ba          CG33993 1506               4527       2408.0650      \n",
       "grn28     ba          grn     4268              17145       6592.5322      \n",
       "CG1536528 ba          CG15365  917               2851       1441.2191      \n",
       "Lfg28     ba          Lfg      502               1467        800.1733      \n",
       "CG1581828 ba          CG15818  287                675        454.5534      \n",
       "CG1763628 ba          CG17636   80                 99        135.6902      \n",
       "ana57     ba          ana      256                624        421.0107      \n",
       "CG430128  ba          CG4301   306                840        494.5218      \n",
       "SLC22A28  ba          SLC22A   132                297        220.7942      \n",
       "al28      ba          al       347               1066        535.1364      \n",
       "sog28     ba          sog     1685               9075       2612.5670      \n",
       "Jheh128   ba          Jheh1    602               2823        978.3783      \n",
       "CG478128  ba          CG4781   346               1165        539.1020      \n",
       "CG1065028 ba          CG10650  437               1808        714.7974      \n",
       "Agpat428  ba          Agpat4   685               3213       1107.1772      \n",
       "          other.tpm  log2.ratio heuristic.score\n",
       "Lim328    119.979833 4.621720   53.33848       \n",
       "CG456228   26.773968 5.109997   50.62556       \n",
       "Jheh328    80.372302 4.272619   45.37522       \n",
       "CG473428    6.117015 5.212550   41.95727       \n",
       "CG1534728 157.565114 3.768339   41.74542       \n",
       "CG3399328 204.686708 3.549354   39.87436       \n",
       "grn28     753.401411 3.127428   39.67717       \n",
       "CG1536528 134.747304 3.408294   35.76689       \n",
       "Lfg28      65.912156 3.579972   34.53231       \n",
       "CG1581828  30.450067 3.853315   34.03046       \n",
       "CG1763628   3.904068 4.790194   33.98530       \n",
       "ana57      28.954694 3.813003   33.25372       \n",
       "CG430128   39.372595 3.614586   32.36068       \n",
       "SLC22A28   12.229042 4.060921   31.64708       \n",
       "al28       46.810234 3.484515   31.59221       \n",
       "sog28     380.269540 2.776585   31.51925       \n",
       "Jheh128   112.403214 3.108931   30.88947       \n",
       "CG478128   51.804672 3.351821   30.42477       \n",
       "CG1065028  78.658457 3.165635   30.02101       \n",
       "Agpat428  143.415513 2.938588   29.72080       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"ba\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 251,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Lim3\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Lim3\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Lim3\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG4562\"), # Mrp2\n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Jheh3\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### cell.type.3 counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1784,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 70 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cell.type.3</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Amnioserosa                               </td><td> 7091</td></tr>\n",
       "\t<tr><td>Analpad                                   </td><td>  850</td></tr>\n",
       "\t<tr><td>Analpad_large_intestine                   </td><td>  734</td></tr>\n",
       "\t<tr><td>Antennal_sense_organs                     </td><td>  748</td></tr>\n",
       "\t<tr><td>Bolwig_organ                              </td><td>  115</td></tr>\n",
       "\t<tr><td>Central_brain_ventral_nerve_cord_1        </td><td> 6846</td></tr>\n",
       "\t<tr><td>Central_brain_ventral_nerve_cord_2        </td><td>10161</td></tr>\n",
       "\t<tr><td>Central_brain_ventral_nerve_cord_early    </td><td> 5058</td></tr>\n",
       "\t<tr><td>Central_brain_ventral_nerve_cord_primorium</td><td> 5819</td></tr>\n",
       "\t<tr><td>Cholinergic                               </td><td> 8949</td></tr>\n",
       "\t<tr><td>Chordotonal_organ                         </td><td>  121</td></tr>\n",
       "\t<tr><td>Chordotonal_sensory                       </td><td>  750</td></tr>\n",
       "\t<tr><td>Copper_cells                              </td><td> 4327</td></tr>\n",
       "\t<tr><td>Crystal_cell                              </td><td>  432</td></tr>\n",
       "\t<tr><td>Cuprophilic_cells                         </td><td>  581</td></tr>\n",
       "\t<tr><td>Digestive_system                          </td><td> 2212</td></tr>\n",
       "\t<tr><td>Embryonic_body_wall                       </td><td>42392</td></tr>\n",
       "\t<tr><td>Epidermis                                 </td><td>17230</td></tr>\n",
       "\t<tr><td>Epidermis_late                            </td><td>  184</td></tr>\n",
       "\t<tr><td>External_sensory                          </td><td>  647</td></tr>\n",
       "\t<tr><td>Fat_body                                  </td><td> 5694</td></tr>\n",
       "\t<tr><td>Garland_cell_nephrocyte                   </td><td>  536</td></tr>\n",
       "\t<tr><td>Germ_cells_ovaries_query                  </td><td> 2139</td></tr>\n",
       "\t<tr><td>Glutamatergic                             </td><td> 5107</td></tr>\n",
       "\t<tr><td>Late_cells                                </td><td> 1565</td></tr>\n",
       "\t<tr><td>Longitudinal_visceral_muscle              </td><td>  617</td></tr>\n",
       "\t<tr><td>Malpighian tubule                         </td><td> 1731</td></tr>\n",
       "\t<tr><td>Maxillary_sensory_complex                 </td><td>  533</td></tr>\n",
       "\t<tr><td>Midgut                                    </td><td> 1674</td></tr>\n",
       "\t<tr><td>Midgut_4th_chamber                        </td><td> 2128</td></tr>\n",
       "\t<tr><td>Midgut_primordium                         </td><td>  689</td></tr>\n",
       "\t<tr><td>Monoaminergic                             </td><td>  454</td></tr>\n",
       "\t<tr><td>Multidendritic_sensory                    </td><td> 2069</td></tr>\n",
       "\t<tr><td>Muscle_system                             </td><td> 5347</td></tr>\n",
       "\t<tr><td>Neuron_unknown_1                          </td><td>  602</td></tr>\n",
       "\t<tr><td>Neuron_unknown_2                          </td><td>  225</td></tr>\n",
       "\t<tr><td>Oenocytes                                 </td><td>  142</td></tr>\n",
       "\t<tr><td>Oenocytes_older                           </td><td>  250</td></tr>\n",
       "\t<tr><td>Olfactory_neurons                         </td><td> 1864</td></tr>\n",
       "\t<tr><td>Peptidergic_neuron                        </td><td>  147</td></tr>\n",
       "\t<tr><td>Peripheral_nervous_system_early           </td><td> 3851</td></tr>\n",
       "\t<tr><td>Pharynx                                   </td><td> 1704</td></tr>\n",
       "\t<tr><td>Pharynx_2                                 </td><td>  975</td></tr>\n",
       "\t<tr><td>Pharynx_primordium_query                  </td><td> 1990</td></tr>\n",
       "\t<tr><td>Plasmatocytes                             </td><td> 6021</td></tr>\n",
       "\t<tr><td>Posterior_spiracle                        </td><td> 1436</td></tr>\n",
       "\t<tr><td>Proventriculus                            </td><td> 2423</td></tr>\n",
       "\t<tr><td>Proventriculus_older                      </td><td>  253</td></tr>\n",
       "\t<tr><td>Salivary_gland                            </td><td> 4237</td></tr>\n",
       "\t<tr><td>Somatic_muscle                            </td><td> 2766</td></tr>\n",
       "\t<tr><td>Somatic_muscle_primordium                 </td><td> 5139</td></tr>\n",
       "\t<tr><td>Tracheal_system_1                         </td><td> 1101</td></tr>\n",
       "\t<tr><td>Tracheal_system_2                         </td><td> 2642</td></tr>\n",
       "\t<tr><td>Tracheal_system_early                     </td><td> 1797</td></tr>\n",
       "\t<tr><td>Tracheal_system_older                     </td><td> 2821</td></tr>\n",
       "\t<tr><td>Undifferentiated_1                        </td><td> 3354</td></tr>\n",
       "\t<tr><td>Undifferentiated_2                        </td><td> 1700</td></tr>\n",
       "\t<tr><td>Undifferentiated_3                        </td><td> 2511</td></tr>\n",
       "\t<tr><td>Ventral_midline                           </td><td> 2159</td></tr>\n",
       "\t<tr><td>Visceral_mesoderm                         </td><td> 1495</td></tr>\n",
       "\t<tr><td>Visceral_muscle                           </td><td> 2308</td></tr>\n",
       "\t<tr><td>Visceral_muscle_2                         </td><td> 1508</td></tr>\n",
       "\t<tr><td>gut_query                                 </td><td>  887</td></tr>\n",
       "\t<tr><td>large_intestine_hindgut_analpad           </td><td> 2226</td></tr>\n",
       "\t<tr><td>large_intestine_hindgut_analpad_2         </td><td> 1467</td></tr>\n",
       "\t<tr><td>lateral_cord_glia                         </td><td>  636</td></tr>\n",
       "\t<tr><td>uncertain_1                               </td><td>   59</td></tr>\n",
       "\t<tr><td>unknown_2                                 </td><td> 1629</td></tr>\n",
       "\t<tr><td>unknown_3                                 </td><td>  180</td></tr>\n",
       "\t<tr><td>NA                                        </td><td>28410</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 70 × 2\n",
       "\\begin{tabular}{ll}\n",
       " cell.type.3 & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Amnioserosa                                &  7091\\\\\n",
       "\t Analpad                                    &   850\\\\\n",
       "\t Analpad\\_large\\_intestine                    &   734\\\\\n",
       "\t Antennal\\_sense\\_organs                      &   748\\\\\n",
       "\t Bolwig\\_organ                               &   115\\\\\n",
       "\t Central\\_brain\\_ventral\\_nerve\\_cord\\_1         &  6846\\\\\n",
       "\t Central\\_brain\\_ventral\\_nerve\\_cord\\_2         & 10161\\\\\n",
       "\t Central\\_brain\\_ventral\\_nerve\\_cord\\_early     &  5058\\\\\n",
       "\t Central\\_brain\\_ventral\\_nerve\\_cord\\_primorium &  5819\\\\\n",
       "\t Cholinergic                                &  8949\\\\\n",
       "\t Chordotonal\\_organ                          &   121\\\\\n",
       "\t Chordotonal\\_sensory                        &   750\\\\\n",
       "\t Copper\\_cells                               &  4327\\\\\n",
       "\t Crystal\\_cell                               &   432\\\\\n",
       "\t Cuprophilic\\_cells                          &   581\\\\\n",
       "\t Digestive\\_system                           &  2212\\\\\n",
       "\t Embryonic\\_body\\_wall                        & 42392\\\\\n",
       "\t Epidermis                                  & 17230\\\\\n",
       "\t Epidermis\\_late                             &   184\\\\\n",
       "\t External\\_sensory                           &   647\\\\\n",
       "\t Fat\\_body                                   &  5694\\\\\n",
       "\t Garland\\_cell\\_nephrocyte                    &   536\\\\\n",
       "\t Germ\\_cells\\_ovaries\\_query                   &  2139\\\\\n",
       "\t Glutamatergic                              &  5107\\\\\n",
       "\t Late\\_cells                                 &  1565\\\\\n",
       "\t Longitudinal\\_visceral\\_muscle               &   617\\\\\n",
       "\t Malpighian tubule                          &  1731\\\\\n",
       "\t Maxillary\\_sensory\\_complex                  &   533\\\\\n",
       "\t Midgut                                     &  1674\\\\\n",
       "\t Midgut\\_4th\\_chamber                         &  2128\\\\\n",
       "\t Midgut\\_primordium                          &   689\\\\\n",
       "\t Monoaminergic                              &   454\\\\\n",
       "\t Multidendritic\\_sensory                     &  2069\\\\\n",
       "\t Muscle\\_system                              &  5347\\\\\n",
       "\t Neuron\\_unknown\\_1                           &   602\\\\\n",
       "\t Neuron\\_unknown\\_2                           &   225\\\\\n",
       "\t Oenocytes                                  &   142\\\\\n",
       "\t Oenocytes\\_older                            &   250\\\\\n",
       "\t Olfactory\\_neurons                          &  1864\\\\\n",
       "\t Peptidergic\\_neuron                         &   147\\\\\n",
       "\t Peripheral\\_nervous\\_system\\_early            &  3851\\\\\n",
       "\t Pharynx                                    &  1704\\\\\n",
       "\t Pharynx\\_2                                  &   975\\\\\n",
       "\t Pharynx\\_primordium\\_query                   &  1990\\\\\n",
       "\t Plasmatocytes                              &  6021\\\\\n",
       "\t Posterior\\_spiracle                         &  1436\\\\\n",
       "\t Proventriculus                             &  2423\\\\\n",
       "\t Proventriculus\\_older                       &   253\\\\\n",
       "\t Salivary\\_gland                             &  4237\\\\\n",
       "\t Somatic\\_muscle                             &  2766\\\\\n",
       "\t Somatic\\_muscle\\_primordium                  &  5139\\\\\n",
       "\t Tracheal\\_system\\_1                          &  1101\\\\\n",
       "\t Tracheal\\_system\\_2                          &  2642\\\\\n",
       "\t Tracheal\\_system\\_early                      &  1797\\\\\n",
       "\t Tracheal\\_system\\_older                      &  2821\\\\\n",
       "\t Undifferentiated\\_1                         &  3354\\\\\n",
       "\t Undifferentiated\\_2                         &  1700\\\\\n",
       "\t Undifferentiated\\_3                         &  2511\\\\\n",
       "\t Ventral\\_midline                            &  2159\\\\\n",
       "\t Visceral\\_mesoderm                          &  1495\\\\\n",
       "\t Visceral\\_muscle                            &  2308\\\\\n",
       "\t Visceral\\_muscle\\_2                          &  1508\\\\\n",
       "\t gut\\_query                                  &   887\\\\\n",
       "\t large\\_intestine\\_hindgut\\_analpad            &  2226\\\\\n",
       "\t large\\_intestine\\_hindgut\\_analpad\\_2          &  1467\\\\\n",
       "\t lateral\\_cord\\_glia                          &   636\\\\\n",
       "\t uncertain\\_1                                &    59\\\\\n",
       "\t unknown\\_2                                  &  1629\\\\\n",
       "\t unknown\\_3                                  &   180\\\\\n",
       "\t NA                                         & 28410\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 70 × 2\n",
       "\n",
       "| cell.type.3 &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Amnioserosa                                |  7091 |\n",
       "| Analpad                                    |   850 |\n",
       "| Analpad_large_intestine                    |   734 |\n",
       "| Antennal_sense_organs                      |   748 |\n",
       "| Bolwig_organ                               |   115 |\n",
       "| Central_brain_ventral_nerve_cord_1         |  6846 |\n",
       "| Central_brain_ventral_nerve_cord_2         | 10161 |\n",
       "| Central_brain_ventral_nerve_cord_early     |  5058 |\n",
       "| Central_brain_ventral_nerve_cord_primorium |  5819 |\n",
       "| Cholinergic                                |  8949 |\n",
       "| Chordotonal_organ                          |   121 |\n",
       "| Chordotonal_sensory                        |   750 |\n",
       "| Copper_cells                               |  4327 |\n",
       "| Crystal_cell                               |   432 |\n",
       "| Cuprophilic_cells                          |   581 |\n",
       "| Digestive_system                           |  2212 |\n",
       "| Embryonic_body_wall                        | 42392 |\n",
       "| Epidermis                                  | 17230 |\n",
       "| Epidermis_late                             |   184 |\n",
       "| External_sensory                           |   647 |\n",
       "| Fat_body                                   |  5694 |\n",
       "| Garland_cell_nephrocyte                    |   536 |\n",
       "| Germ_cells_ovaries_query                   |  2139 |\n",
       "| Glutamatergic                              |  5107 |\n",
       "| Late_cells                                 |  1565 |\n",
       "| Longitudinal_visceral_muscle               |   617 |\n",
       "| Malpighian tubule                          |  1731 |\n",
       "| Maxillary_sensory_complex                  |   533 |\n",
       "| Midgut                                     |  1674 |\n",
       "| Midgut_4th_chamber                         |  2128 |\n",
       "| Midgut_primordium                          |   689 |\n",
       "| Monoaminergic                              |   454 |\n",
       "| Multidendritic_sensory                     |  2069 |\n",
       "| Muscle_system                              |  5347 |\n",
       "| Neuron_unknown_1                           |   602 |\n",
       "| Neuron_unknown_2                           |   225 |\n",
       "| Oenocytes                                  |   142 |\n",
       "| Oenocytes_older                            |   250 |\n",
       "| Olfactory_neurons                          |  1864 |\n",
       "| Peptidergic_neuron                         |   147 |\n",
       "| Peripheral_nervous_system_early            |  3851 |\n",
       "| Pharynx                                    |  1704 |\n",
       "| Pharynx_2                                  |   975 |\n",
       "| Pharynx_primordium_query                   |  1990 |\n",
       "| Plasmatocytes                              |  6021 |\n",
       "| Posterior_spiracle                         |  1436 |\n",
       "| Proventriculus                             |  2423 |\n",
       "| Proventriculus_older                       |   253 |\n",
       "| Salivary_gland                             |  4237 |\n",
       "| Somatic_muscle                             |  2766 |\n",
       "| Somatic_muscle_primordium                  |  5139 |\n",
       "| Tracheal_system_1                          |  1101 |\n",
       "| Tracheal_system_2                          |  2642 |\n",
       "| Tracheal_system_early                      |  1797 |\n",
       "| Tracheal_system_older                      |  2821 |\n",
       "| Undifferentiated_1                         |  3354 |\n",
       "| Undifferentiated_2                         |  1700 |\n",
       "| Undifferentiated_3                         |  2511 |\n",
       "| Ventral_midline                            |  2159 |\n",
       "| Visceral_mesoderm                          |  1495 |\n",
       "| Visceral_muscle                            |  2308 |\n",
       "| Visceral_muscle_2                          |  1508 |\n",
       "| gut_query                                  |   887 |\n",
       "| large_intestine_hindgut_analpad            |  2226 |\n",
       "| large_intestine_hindgut_analpad_2          |  1467 |\n",
       "| lateral_cord_glia                          |   636 |\n",
       "| uncertain_1                                |    59 |\n",
       "| unknown_2                                  |  1629 |\n",
       "| unknown_3                                  |   180 |\n",
       "| NA                                         | 28410 |\n",
       "\n"
      ],
      "text/plain": [
       "   cell.type.3                                n    \n",
       "1  Amnioserosa                                 7091\n",
       "2  Analpad                                      850\n",
       "3  Analpad_large_intestine                      734\n",
       "4  Antennal_sense_organs                        748\n",
       "5  Bolwig_organ                                 115\n",
       "6  Central_brain_ventral_nerve_cord_1          6846\n",
       "7  Central_brain_ventral_nerve_cord_2         10161\n",
       "8  Central_brain_ventral_nerve_cord_early      5058\n",
       "9  Central_brain_ventral_nerve_cord_primorium  5819\n",
       "10 Cholinergic                                 8949\n",
       "11 Chordotonal_organ                            121\n",
       "12 Chordotonal_sensory                          750\n",
       "13 Copper_cells                                4327\n",
       "14 Crystal_cell                                 432\n",
       "15 Cuprophilic_cells                            581\n",
       "16 Digestive_system                            2212\n",
       "17 Embryonic_body_wall                        42392\n",
       "18 Epidermis                                  17230\n",
       "19 Epidermis_late                               184\n",
       "20 External_sensory                             647\n",
       "21 Fat_body                                    5694\n",
       "22 Garland_cell_nephrocyte                      536\n",
       "23 Germ_cells_ovaries_query                    2139\n",
       "24 Glutamatergic                               5107\n",
       "25 Late_cells                                  1565\n",
       "26 Longitudinal_visceral_muscle                 617\n",
       "27 Malpighian tubule                           1731\n",
       "28 Maxillary_sensory_complex                    533\n",
       "29 Midgut                                      1674\n",
       "30 Midgut_4th_chamber                          2128\n",
       "31 Midgut_primordium                            689\n",
       "32 Monoaminergic                                454\n",
       "33 Multidendritic_sensory                      2069\n",
       "34 Muscle_system                               5347\n",
       "35 Neuron_unknown_1                             602\n",
       "36 Neuron_unknown_2                             225\n",
       "37 Oenocytes                                    142\n",
       "38 Oenocytes_older                              250\n",
       "39 Olfactory_neurons                           1864\n",
       "40 Peptidergic_neuron                           147\n",
       "41 Peripheral_nervous_system_early             3851\n",
       "42 Pharynx                                     1704\n",
       "43 Pharynx_2                                    975\n",
       "44 Pharynx_primordium_query                    1990\n",
       "45 Plasmatocytes                               6021\n",
       "46 Posterior_spiracle                          1436\n",
       "47 Proventriculus                              2423\n",
       "48 Proventriculus_older                         253\n",
       "49 Salivary_gland                              4237\n",
       "50 Somatic_muscle                              2766\n",
       "51 Somatic_muscle_primordium                   5139\n",
       "52 Tracheal_system_1                           1101\n",
       "53 Tracheal_system_2                           2642\n",
       "54 Tracheal_system_early                       1797\n",
       "55 Tracheal_system_older                       2821\n",
       "56 Undifferentiated_1                          3354\n",
       "57 Undifferentiated_2                          1700\n",
       "58 Undifferentiated_3                          2511\n",
       "59 Ventral_midline                             2159\n",
       "60 Visceral_mesoderm                           1495\n",
       "61 Visceral_muscle                             2308\n",
       "62 Visceral_muscle_2                           1508\n",
       "63 gut_query                                    887\n",
       "64 large_intestine_hindgut_analpad             2226\n",
       "65 large_intestine_hindgut_analpad_2           1467\n",
       "66 lateral_cord_glia                            636\n",
       "67 uncertain_1                                   59\n",
       "68 unknown_2                                   1629\n",
       "69 unknown_3                                    180\n",
       "70 NA                                         28410"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub300  %>% count(cell.type.3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1505,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Visceral_muscle\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Visceral_muscle\", df_sub300$cluster %in% bb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bb' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bb' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bb)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 253,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(18,46)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1485,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG4442229</th><td>bb</td><td>CG44422</td><td>1744</td><td> 3467</td><td> 3507.6104</td><td> 149.341023</td><td>4.544178</td><td>53.51535</td></tr>\n",
       "\t<tr><th scope=row>disco-r29</th><td>bb</td><td>disco-r</td><td>2741</td><td> 7077</td><td> 5473.9731</td><td> 305.776910</td><td>4.157327</td><td>51.62833</td></tr>\n",
       "\t<tr><th scope=row>CG3325329</th><td>bb</td><td>CG33253</td><td> 298</td><td>  488</td><td>  598.5538</td><td>  20.605880</td><td>4.791985</td><td>44.21922</td></tr>\n",
       "\t<tr><th scope=row>side-IV29</th><td>bb</td><td>side-IV</td><td>1335</td><td> 4172</td><td> 2685.4678</td><td> 185.489971</td><td>3.848003</td><td>43.83451</td></tr>\n",
       "\t<tr><th scope=row>bru329</th><td>bb</td><td>bru3   </td><td>7363</td><td>41873</td><td>14429.4171</td><td>1850.352999</td><td>2.962361</td><td>40.93043</td></tr>\n",
       "\t<tr><th scope=row>CG482929</th><td>bb</td><td>CG4829 </td><td>1074</td><td> 3934</td><td> 2155.1859</td><td> 171.532618</td><td>3.642871</td><td>40.34213</td></tr>\n",
       "\t<tr><th scope=row>NetA29</th><td>bb</td><td>NetA   </td><td>1683</td><td> 7445</td><td> 3400.1073</td><td> 327.692188</td><td>3.370771</td><td>39.54518</td></tr>\n",
       "\t<tr><th scope=row>sqa29</th><td>bb</td><td>sqa    </td><td> 498</td><td> 1575</td><td>  973.8925</td><td>  63.322972</td><td>3.920357</td><td>38.92561</td></tr>\n",
       "\t<tr><th scope=row>CG3132329</th><td>bb</td><td>CG31323</td><td> 232</td><td>  499</td><td>  449.5786</td><td>  20.165243</td><td>4.408804</td><td>38.86641</td></tr>\n",
       "\t<tr><th scope=row>CG3182329</th><td>bb</td><td>CG31823</td><td>  83</td><td>   71</td><td>  162.8667</td><td>   3.388348</td><td>5.213870</td><td>38.35520</td></tr>\n",
       "\t<tr><th scope=row>Con29</th><td>bb</td><td>Con    </td><td>2023</td><td>11153</td><td> 4066.5294</td><td> 493.255110</td><td>3.040470</td><td>36.45505</td></tr>\n",
       "\t<tr><th scope=row>Sdr29</th><td>bb</td><td>Sdr    </td><td> 120</td><td>  260</td><td>  249.5316</td><td>  10.624171</td><td>4.424023</td><td>35.25437</td></tr>\n",
       "\t<tr><th scope=row>Zasp5229</th><td>bb</td><td>Zasp52 </td><td>1449</td><td> 8553</td><td> 2849.9958</td><td> 353.572266</td><td>3.006808</td><td>34.50989</td></tr>\n",
       "\t<tr><th scope=row>org-129</th><td>bb</td><td>org-1  </td><td> 302</td><td> 1094</td><td>  596.7217</td><td>  47.177958</td><td>3.630613</td><td>33.48634</td></tr>\n",
       "\t<tr><th scope=row>Hand29</th><td>bb</td><td>Hand   </td><td>  77</td><td>  141</td><td>  161.1644</td><td>   5.843087</td><td>4.557742</td><td>33.45981</td></tr>\n",
       "\t<tr><th scope=row>AstC-R229</th><td>bb</td><td>AstC-R2</td><td> 175</td><td>  486</td><td>  341.2005</td><td>  21.098535</td><td>3.948597</td><td>33.24205</td></tr>\n",
       "\t<tr><th scope=row>wupA29</th><td>bb</td><td>wupA   </td><td>2703</td><td>21434</td><td> 5335.0986</td><td> 851.211925</td><td>2.646231</td><td>32.76449</td></tr>\n",
       "\t<tr><th scope=row>H2.029</th><td>bb</td><td>H2.0   </td><td> 165</td><td>  528</td><td>  330.2561</td><td>  21.168446</td><td>3.897006</td><td>32.62496</td></tr>\n",
       "\t<tr><th scope=row>RunxA29</th><td>bb</td><td>RunxA  </td><td> 106</td><td>  237</td><td>  207.5914</td><td>  10.129358</td><td>4.221305</td><td>32.52319</td></tr>\n",
       "\t<tr><th scope=row>otk39</th><td>bb</td><td>otk    </td><td>1378</td><td> 9249</td><td> 2773.4429</td><td> 401.510672</td><td>2.784579</td><td>31.84997</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG4442229 & bb & CG44422 & 1744 &  3467 &  3507.6104 &  149.341023 & 4.544178 & 53.51535\\\\\n",
       "\tdisco-r29 & bb & disco-r & 2741 &  7077 &  5473.9731 &  305.776910 & 4.157327 & 51.62833\\\\\n",
       "\tCG3325329 & bb & CG33253 &  298 &   488 &   598.5538 &   20.605880 & 4.791985 & 44.21922\\\\\n",
       "\tside-IV29 & bb & side-IV & 1335 &  4172 &  2685.4678 &  185.489971 & 3.848003 & 43.83451\\\\\n",
       "\tbru329 & bb & bru3    & 7363 & 41873 & 14429.4171 & 1850.352999 & 2.962361 & 40.93043\\\\\n",
       "\tCG482929 & bb & CG4829  & 1074 &  3934 &  2155.1859 &  171.532618 & 3.642871 & 40.34213\\\\\n",
       "\tNetA29 & bb & NetA    & 1683 &  7445 &  3400.1073 &  327.692188 & 3.370771 & 39.54518\\\\\n",
       "\tsqa29 & bb & sqa     &  498 &  1575 &   973.8925 &   63.322972 & 3.920357 & 38.92561\\\\\n",
       "\tCG3132329 & bb & CG31323 &  232 &   499 &   449.5786 &   20.165243 & 4.408804 & 38.86641\\\\\n",
       "\tCG3182329 & bb & CG31823 &   83 &    71 &   162.8667 &    3.388348 & 5.213870 & 38.35520\\\\\n",
       "\tCon29 & bb & Con     & 2023 & 11153 &  4066.5294 &  493.255110 & 3.040470 & 36.45505\\\\\n",
       "\tSdr29 & bb & Sdr     &  120 &   260 &   249.5316 &   10.624171 & 4.424023 & 35.25437\\\\\n",
       "\tZasp5229 & bb & Zasp52  & 1449 &  8553 &  2849.9958 &  353.572266 & 3.006808 & 34.50989\\\\\n",
       "\torg-129 & bb & org-1   &  302 &  1094 &   596.7217 &   47.177958 & 3.630613 & 33.48634\\\\\n",
       "\tHand29 & bb & Hand    &   77 &   141 &   161.1644 &    5.843087 & 4.557742 & 33.45981\\\\\n",
       "\tAstC-R229 & bb & AstC-R2 &  175 &   486 &   341.2005 &   21.098535 & 3.948597 & 33.24205\\\\\n",
       "\twupA29 & bb & wupA    & 2703 & 21434 &  5335.0986 &  851.211925 & 2.646231 & 32.76449\\\\\n",
       "\tH2.029 & bb & H2.0    &  165 &   528 &   330.2561 &   21.168446 & 3.897006 & 32.62496\\\\\n",
       "\tRunxA29 & bb & RunxA   &  106 &   237 &   207.5914 &   10.129358 & 4.221305 & 32.52319\\\\\n",
       "\totk39 & bb & otk     & 1378 &  9249 &  2773.4429 &  401.510672 & 2.784579 & 31.84997\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG4442229 | bb | CG44422 | 1744 |  3467 |  3507.6104 |  149.341023 | 4.544178 | 53.51535 |\n",
       "| disco-r29 | bb | disco-r | 2741 |  7077 |  5473.9731 |  305.776910 | 4.157327 | 51.62833 |\n",
       "| CG3325329 | bb | CG33253 |  298 |   488 |   598.5538 |   20.605880 | 4.791985 | 44.21922 |\n",
       "| side-IV29 | bb | side-IV | 1335 |  4172 |  2685.4678 |  185.489971 | 3.848003 | 43.83451 |\n",
       "| bru329 | bb | bru3    | 7363 | 41873 | 14429.4171 | 1850.352999 | 2.962361 | 40.93043 |\n",
       "| CG482929 | bb | CG4829  | 1074 |  3934 |  2155.1859 |  171.532618 | 3.642871 | 40.34213 |\n",
       "| NetA29 | bb | NetA    | 1683 |  7445 |  3400.1073 |  327.692188 | 3.370771 | 39.54518 |\n",
       "| sqa29 | bb | sqa     |  498 |  1575 |   973.8925 |   63.322972 | 3.920357 | 38.92561 |\n",
       "| CG3132329 | bb | CG31323 |  232 |   499 |   449.5786 |   20.165243 | 4.408804 | 38.86641 |\n",
       "| CG3182329 | bb | CG31823 |   83 |    71 |   162.8667 |    3.388348 | 5.213870 | 38.35520 |\n",
       "| Con29 | bb | Con     | 2023 | 11153 |  4066.5294 |  493.255110 | 3.040470 | 36.45505 |\n",
       "| Sdr29 | bb | Sdr     |  120 |   260 |   249.5316 |   10.624171 | 4.424023 | 35.25437 |\n",
       "| Zasp5229 | bb | Zasp52  | 1449 |  8553 |  2849.9958 |  353.572266 | 3.006808 | 34.50989 |\n",
       "| org-129 | bb | org-1   |  302 |  1094 |   596.7217 |   47.177958 | 3.630613 | 33.48634 |\n",
       "| Hand29 | bb | Hand    |   77 |   141 |   161.1644 |    5.843087 | 4.557742 | 33.45981 |\n",
       "| AstC-R229 | bb | AstC-R2 |  175 |   486 |   341.2005 |   21.098535 | 3.948597 | 33.24205 |\n",
       "| wupA29 | bb | wupA    | 2703 | 21434 |  5335.0986 |  851.211925 | 2.646231 | 32.76449 |\n",
       "| H2.029 | bb | H2.0    |  165 |   528 |   330.2561 |   21.168446 | 3.897006 | 32.62496 |\n",
       "| RunxA29 | bb | RunxA   |  106 |   237 |   207.5914 |   10.129358 | 4.221305 | 32.52319 |\n",
       "| otk39 | bb | otk     | 1378 |  9249 |  2773.4429 |  401.510672 | 2.784579 | 31.84997 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "CG4442229 bb          CG44422 1744               3467        3507.6104     \n",
       "disco-r29 bb          disco-r 2741               7077        5473.9731     \n",
       "CG3325329 bb          CG33253  298                488         598.5538     \n",
       "side-IV29 bb          side-IV 1335               4172        2685.4678     \n",
       "bru329    bb          bru3    7363              41873       14429.4171     \n",
       "CG482929  bb          CG4829  1074               3934        2155.1859     \n",
       "NetA29    bb          NetA    1683               7445        3400.1073     \n",
       "sqa29     bb          sqa      498               1575         973.8925     \n",
       "CG3132329 bb          CG31323  232                499         449.5786     \n",
       "CG3182329 bb          CG31823   83                 71         162.8667     \n",
       "Con29     bb          Con     2023              11153        4066.5294     \n",
       "Sdr29     bb          Sdr      120                260         249.5316     \n",
       "Zasp5229  bb          Zasp52  1449               8553        2849.9958     \n",
       "org-129   bb          org-1    302               1094         596.7217     \n",
       "Hand29    bb          Hand      77                141         161.1644     \n",
       "AstC-R229 bb          AstC-R2  175                486         341.2005     \n",
       "wupA29    bb          wupA    2703              21434        5335.0986     \n",
       "H2.029    bb          H2.0     165                528         330.2561     \n",
       "RunxA29   bb          RunxA    106                237         207.5914     \n",
       "otk39     bb          otk     1378               9249        2773.4429     \n",
       "          other.tpm   log2.ratio heuristic.score\n",
       "CG4442229  149.341023 4.544178   53.51535       \n",
       "disco-r29  305.776910 4.157327   51.62833       \n",
       "CG3325329   20.605880 4.791985   44.21922       \n",
       "side-IV29  185.489971 3.848003   43.83451       \n",
       "bru329    1850.352999 2.962361   40.93043       \n",
       "CG482929   171.532618 3.642871   40.34213       \n",
       "NetA29     327.692188 3.370771   39.54518       \n",
       "sqa29       63.322972 3.920357   38.92561       \n",
       "CG3132329   20.165243 4.408804   38.86641       \n",
       "CG3182329    3.388348 5.213870   38.35520       \n",
       "Con29      493.255110 3.040470   36.45505       \n",
       "Sdr29       10.624171 4.424023   35.25437       \n",
       "Zasp5229   353.572266 3.006808   34.50989       \n",
       "org-129     47.177958 3.630613   33.48634       \n",
       "Hand29       5.843087 4.557742   33.45981       \n",
       "AstC-R229   21.098535 3.948597   33.24205       \n",
       "wupA29     851.211925 2.646231   32.76449       \n",
       "H2.029      21.168446 3.897006   32.62496       \n",
       "RunxA29     10.129358 4.221305   32.52319       \n",
       "otk39      401.510672 2.784579   31.84997       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"bb\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"side-IV\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"side-IV\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"side-IV\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"disco-r\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"bru3\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(18)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.7), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -3.3) +\n",
    "geom_abline(slope = 1, intercept = -3.3) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    " xlim(0, 4) + ylim(-3, 1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### bc cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bc\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bc\", df_sub300$cluster %in% bc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1669,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Visceral_mesoderm\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Longitudinal_visceral_muscle\", df_sub300$cluster %in% bc &\n",
    "                          df_sub300$umap_2 < -0.1*df_sub300$umap_1 -0.1\n",
    "                         )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1673,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Visceral_mesoderm\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Visceral_mesoderm\", df_sub300$cluster %in% bc &\n",
    "                          df_sub300$umap_2 >= -0.1*df_sub300$umap_1 -0.1\n",
    "                         )\n",
    "# based on Kah expression  doi: 10.1242/dev.199465 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 256,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bc' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bc' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bc)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 257,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(113)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1241,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>tey30</th><td>bc</td><td>tey     </td><td> 854</td><td> 1707</td><td>1979.00447</td><td> 71.1894839</td><td>4.776842</td><td>52.31258</td></tr>\n",
       "\t<tr><th scope=row>sns30</th><td>bc</td><td>sns     </td><td>1716</td><td> 5586</td><td>4245.56777</td><td>237.1779838</td><td>4.155846</td><td>50.08659</td></tr>\n",
       "\t<tr><th scope=row>HLH54F30</th><td>bc</td><td>HLH54F  </td><td> 228</td><td>  343</td><td> 534.43234</td><td> 13.2222349</td><td>5.231787</td><td>47.42385</td></tr>\n",
       "\t<tr><th scope=row>Kah30</th><td>bc</td><td>Kah     </td><td> 486</td><td> 2153</td><td>1242.70110</td><td> 89.1600192</td><td>3.784848</td><td>38.90984</td></tr>\n",
       "\t<tr><th scope=row>CG3437730</th><td>bc</td><td>CG34377 </td><td> 151</td><td>  379</td><td> 360.51816</td><td> 15.2306256</td><td>4.473281</td><td>38.01361</td></tr>\n",
       "\t<tr><th scope=row>beat-IIb30</th><td>bc</td><td>beat-IIb</td><td> 602</td><td> 2393</td><td>1368.69188</td><td>109.0783068</td><td>3.636196</td><td>37.88783</td></tr>\n",
       "\t<tr><th scope=row>NK7.130</th><td>bc</td><td>NK7.1   </td><td>1408</td><td> 8264</td><td>3497.71352</td><td>375.6909445</td><td>3.214959</td><td>37.84845</td></tr>\n",
       "\t<tr><th scope=row>gol30</th><td>bc</td><td>gol     </td><td> 517</td><td> 2626</td><td>1255.99837</td><td>113.6723025</td><td>3.453246</td><td>35.55381</td></tr>\n",
       "\t<tr><th scope=row>beat-IIa30</th><td>bc</td><td>beat-IIa</td><td> 379</td><td> 1656</td><td> 891.28705</td><td> 72.1371877</td><td>3.607213</td><td>35.35561</td></tr>\n",
       "\t<tr><th scope=row>MCU30</th><td>bc</td><td>MCU     </td><td> 361</td><td> 2045</td><td> 922.39027</td><td> 87.6406611</td><td>3.379337</td><td>33.28916</td></tr>\n",
       "\t<tr><th scope=row>Vrp130</th><td>bc</td><td>Vrp1    </td><td> 703</td><td> 5071</td><td>1721.59605</td><td>204.5691637</td><td>3.066051</td><td>32.96118</td></tr>\n",
       "\t<tr><th scope=row>Mef230</th><td>bc</td><td>Mef2    </td><td>1429</td><td>13095</td><td>3472.75645</td><td>519.2098483</td><td>2.738916</td><td>32.21589</td></tr>\n",
       "\t<tr><th scope=row>Adgf-A230</th><td>bc</td><td>Adgf-A2 </td><td> 246</td><td> 1128</td><td> 587.18679</td><td> 51.0822462</td><td>3.494956</td><td>32.15405</td></tr>\n",
       "\t<tr><th scope=row>tok30</th><td>bc</td><td>tok     </td><td> 318</td><td> 1747</td><td> 750.69627</td><td> 75.1905725</td><td>3.300545</td><td>31.53343</td></tr>\n",
       "\t<tr><th scope=row>HEATR230</th><td>bc</td><td>HEATR2  </td><td> 150</td><td>  653</td><td> 369.99383</td><td> 27.6012301</td><td>3.693352</td><td>31.52369</td></tr>\n",
       "\t<tr><th scope=row>lmd30</th><td>bc</td><td>lmd     </td><td> 459</td><td> 3169</td><td>1141.84968</td><td>138.5942210</td><td>3.032062</td><td>30.80096</td></tr>\n",
       "\t<tr><th scope=row>CG967430</th><td>bc</td><td>CG9674  </td><td> 700</td><td> 5702</td><td>1673.15678</td><td>234.4597339</td><td>2.829020</td><td>30.29660</td></tr>\n",
       "\t<tr><th scope=row>CG1810930</th><td>bc</td><td>CG18109 </td><td>  19</td><td>   15</td><td>  51.43406</td><td>  0.4750871</td><td>5.123852</td><td>29.26966</td></tr>\n",
       "\t<tr><th scope=row>net30</th><td>bc</td><td>net     </td><td> 115</td><td>  564</td><td> 284.38147</td><td> 24.9099580</td><td>3.456249</td><td>28.19175</td></tr>\n",
       "\t<tr><th scope=row>ewg30</th><td>bc</td><td>ewg     </td><td>1035</td><td>11203</td><td>2567.05502</td><td>487.7430411</td><td>2.392966</td><td>27.10384</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\ttey30 & bc & tey      &  854 &  1707 & 1979.00447 &  71.1894839 & 4.776842 & 52.31258\\\\\n",
       "\tsns30 & bc & sns      & 1716 &  5586 & 4245.56777 & 237.1779838 & 4.155846 & 50.08659\\\\\n",
       "\tHLH54F30 & bc & HLH54F   &  228 &   343 &  534.43234 &  13.2222349 & 5.231787 & 47.42385\\\\\n",
       "\tKah30 & bc & Kah      &  486 &  2153 & 1242.70110 &  89.1600192 & 3.784848 & 38.90984\\\\\n",
       "\tCG3437730 & bc & CG34377  &  151 &   379 &  360.51816 &  15.2306256 & 4.473281 & 38.01361\\\\\n",
       "\tbeat-IIb30 & bc & beat-IIb &  602 &  2393 & 1368.69188 & 109.0783068 & 3.636196 & 37.88783\\\\\n",
       "\tNK7.130 & bc & NK7.1    & 1408 &  8264 & 3497.71352 & 375.6909445 & 3.214959 & 37.84845\\\\\n",
       "\tgol30 & bc & gol      &  517 &  2626 & 1255.99837 & 113.6723025 & 3.453246 & 35.55381\\\\\n",
       "\tbeat-IIa30 & bc & beat-IIa &  379 &  1656 &  891.28705 &  72.1371877 & 3.607213 & 35.35561\\\\\n",
       "\tMCU30 & bc & MCU      &  361 &  2045 &  922.39027 &  87.6406611 & 3.379337 & 33.28916\\\\\n",
       "\tVrp130 & bc & Vrp1     &  703 &  5071 & 1721.59605 & 204.5691637 & 3.066051 & 32.96118\\\\\n",
       "\tMef230 & bc & Mef2     & 1429 & 13095 & 3472.75645 & 519.2098483 & 2.738916 & 32.21589\\\\\n",
       "\tAdgf-A230 & bc & Adgf-A2  &  246 &  1128 &  587.18679 &  51.0822462 & 3.494956 & 32.15405\\\\\n",
       "\ttok30 & bc & tok      &  318 &  1747 &  750.69627 &  75.1905725 & 3.300545 & 31.53343\\\\\n",
       "\tHEATR230 & bc & HEATR2   &  150 &   653 &  369.99383 &  27.6012301 & 3.693352 & 31.52369\\\\\n",
       "\tlmd30 & bc & lmd      &  459 &  3169 & 1141.84968 & 138.5942210 & 3.032062 & 30.80096\\\\\n",
       "\tCG967430 & bc & CG9674   &  700 &  5702 & 1673.15678 & 234.4597339 & 2.829020 & 30.29660\\\\\n",
       "\tCG1810930 & bc & CG18109  &   19 &    15 &   51.43406 &   0.4750871 & 5.123852 & 29.26966\\\\\n",
       "\tnet30 & bc & net      &  115 &   564 &  284.38147 &  24.9099580 & 3.456249 & 28.19175\\\\\n",
       "\tewg30 & bc & ewg      & 1035 & 11203 & 2567.05502 & 487.7430411 & 2.392966 & 27.10384\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| tey30 | bc | tey      |  854 |  1707 | 1979.00447 |  71.1894839 | 4.776842 | 52.31258 |\n",
       "| sns30 | bc | sns      | 1716 |  5586 | 4245.56777 | 237.1779838 | 4.155846 | 50.08659 |\n",
       "| HLH54F30 | bc | HLH54F   |  228 |   343 |  534.43234 |  13.2222349 | 5.231787 | 47.42385 |\n",
       "| Kah30 | bc | Kah      |  486 |  2153 | 1242.70110 |  89.1600192 | 3.784848 | 38.90984 |\n",
       "| CG3437730 | bc | CG34377  |  151 |   379 |  360.51816 |  15.2306256 | 4.473281 | 38.01361 |\n",
       "| beat-IIb30 | bc | beat-IIb |  602 |  2393 | 1368.69188 | 109.0783068 | 3.636196 | 37.88783 |\n",
       "| NK7.130 | bc | NK7.1    | 1408 |  8264 | 3497.71352 | 375.6909445 | 3.214959 | 37.84845 |\n",
       "| gol30 | bc | gol      |  517 |  2626 | 1255.99837 | 113.6723025 | 3.453246 | 35.55381 |\n",
       "| beat-IIa30 | bc | beat-IIa |  379 |  1656 |  891.28705 |  72.1371877 | 3.607213 | 35.35561 |\n",
       "| MCU30 | bc | MCU      |  361 |  2045 |  922.39027 |  87.6406611 | 3.379337 | 33.28916 |\n",
       "| Vrp130 | bc | Vrp1     |  703 |  5071 | 1721.59605 | 204.5691637 | 3.066051 | 32.96118 |\n",
       "| Mef230 | bc | Mef2     | 1429 | 13095 | 3472.75645 | 519.2098483 | 2.738916 | 32.21589 |\n",
       "| Adgf-A230 | bc | Adgf-A2  |  246 |  1128 |  587.18679 |  51.0822462 | 3.494956 | 32.15405 |\n",
       "| tok30 | bc | tok      |  318 |  1747 |  750.69627 |  75.1905725 | 3.300545 | 31.53343 |\n",
       "| HEATR230 | bc | HEATR2   |  150 |   653 |  369.99383 |  27.6012301 | 3.693352 | 31.52369 |\n",
       "| lmd30 | bc | lmd      |  459 |  3169 | 1141.84968 | 138.5942210 | 3.032062 | 30.80096 |\n",
       "| CG967430 | bc | CG9674   |  700 |  5702 | 1673.15678 | 234.4597339 | 2.829020 | 30.29660 |\n",
       "| CG1810930 | bc | CG18109  |   19 |    15 |   51.43406 |   0.4750871 | 5.123852 | 29.26966 |\n",
       "| net30 | bc | net      |  115 |   564 |  284.38147 |  24.9099580 | 3.456249 | 28.19175 |\n",
       "| ewg30 | bc | ewg      | 1035 | 11203 | 2567.05502 | 487.7430411 | 2.392966 | 27.10384 |\n",
       "\n"
      ],
      "text/plain": [
       "           cell.type.2 gene     cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "tey30      bc          tey       854               1707       1979.00447     \n",
       "sns30      bc          sns      1716               5586       4245.56777     \n",
       "HLH54F30   bc          HLH54F    228                343        534.43234     \n",
       "Kah30      bc          Kah       486               2153       1242.70110     \n",
       "CG3437730  bc          CG34377   151                379        360.51816     \n",
       "beat-IIb30 bc          beat-IIb  602               2393       1368.69188     \n",
       "NK7.130    bc          NK7.1    1408               8264       3497.71352     \n",
       "gol30      bc          gol       517               2626       1255.99837     \n",
       "beat-IIa30 bc          beat-IIa  379               1656        891.28705     \n",
       "MCU30      bc          MCU       361               2045        922.39027     \n",
       "Vrp130     bc          Vrp1      703               5071       1721.59605     \n",
       "Mef230     bc          Mef2     1429              13095       3472.75645     \n",
       "Adgf-A230  bc          Adgf-A2   246               1128        587.18679     \n",
       "tok30      bc          tok       318               1747        750.69627     \n",
       "HEATR230   bc          HEATR2    150                653        369.99383     \n",
       "lmd30      bc          lmd       459               3169       1141.84968     \n",
       "CG967430   bc          CG9674    700               5702       1673.15678     \n",
       "CG1810930  bc          CG18109    19                 15         51.43406     \n",
       "net30      bc          net       115                564        284.38147     \n",
       "ewg30      bc          ewg      1035              11203       2567.05502     \n",
       "           other.tpm   log2.ratio heuristic.score\n",
       "tey30       71.1894839 4.776842   52.31258       \n",
       "sns30      237.1779838 4.155846   50.08659       \n",
       "HLH54F30    13.2222349 5.231787   47.42385       \n",
       "Kah30       89.1600192 3.784848   38.90984       \n",
       "CG3437730   15.2306256 4.473281   38.01361       \n",
       "beat-IIb30 109.0783068 3.636196   37.88783       \n",
       "NK7.130    375.6909445 3.214959   37.84845       \n",
       "gol30      113.6723025 3.453246   35.55381       \n",
       "beat-IIa30  72.1371877 3.607213   35.35561       \n",
       "MCU30       87.6406611 3.379337   33.28916       \n",
       "Vrp130     204.5691637 3.066051   32.96118       \n",
       "Mef230     519.2098483 2.738916   32.21589       \n",
       "Adgf-A230   51.0822462 3.494956   32.15405       \n",
       "tok30       75.1905725 3.300545   31.53343       \n",
       "HEATR230    27.6012301 3.693352   31.52369       \n",
       "lmd30      138.5942210 3.032062   30.80096       \n",
       "CG967430   234.4597339 2.829020   30.29660       \n",
       "CG1810930    0.4750871 5.123852   29.26966       \n",
       "net30       24.9099580 3.456249   28.19175       \n",
       "ewg30      487.7430411 2.392966   27.10384       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"bc\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 258,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"tey\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"tey\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"tey\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"sns\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"HLH54F\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 259,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(75, 69)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 < -0.1*umap_1 -0.1), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 0.3, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = -0.1, intercept = -.1) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(-5, -1) + ylim(-2, 2)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1270,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 579\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2931 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>tey</th><td>tey        </td><td> 967</td><td> 1836</td><td> 4611.02394</td><td>  78.52727</td><td>Set 1</td><td>5.857494</td><td>71.29264</td></tr>\n",
       "\t<tr><th scope=row>HLH54F</th><td>HLH54F     </td><td> 251</td><td>  336</td><td> 1223.38930</td><td>  13.18234</td><td>Set 1</td><td>6.430644</td><td>65.96456</td></tr>\n",
       "\t<tr><th scope=row>beat-IIb</th><td>beat-IIb   </td><td> 641</td><td> 2319</td><td> 3062.79663</td><td> 105.41408</td><td>Set 1</td><td>4.847089</td><td>56.13464</td></tr>\n",
       "\t<tr><th scope=row>beat-IIa</th><td>beat-IIa   </td><td> 423</td><td> 1642</td><td> 2054.21747</td><td>  72.15760</td><td>Set 1</td><td>4.811437</td><td>52.95023</td></tr>\n",
       "\t<tr><th scope=row>bru3</th><td>bru3       </td><td>3545</td><td>41521</td><td>16857.46794</td><td>1860.61444</td><td>Set 1</td><td>3.178762</td><td>44.63358</td></tr>\n",
       "\t<tr><th scope=row>tok</th><td>tok        </td><td> 234</td><td> 1776</td><td> 1139.77013</td><td>  78.64160</td><td>Set 1</td><td>3.839077</td><td>38.98887</td></tr>\n",
       "\t<tr><th scope=row>H2.0</th><td>H2.0       </td><td>  87</td><td>  543</td><td>  409.42302</td><td>  22.36169</td><td>Set 1</td><td>4.131375</td><td>35.86434</td></tr>\n",
       "\t<tr><th scope=row>ATP8B</th><td>ATP8B      </td><td> 134</td><td> 1051</td><td>  644.78794</td><td>  47.51049</td><td>Set 1</td><td>3.732456</td><td>34.84217</td></tr>\n",
       "\t<tr><th scope=row>RapGAP1</th><td>RapGAP1    </td><td>2368</td><td>44286</td><td>11055.15800</td><td>1997.87862</td><td>Set 1</td><td>2.467457</td><td>33.14427</td></tr>\n",
       "\t<tr><th scope=row>Syn2</th><td>Syn2       </td><td>  64</td><td>  472</td><td>  322.69397</td><td>  20.38014</td><td>Set 1</td><td>3.915824</td><td>32.65204</td></tr>\n",
       "\t<tr><th scope=row>kon</th><td>kon        </td><td> 393</td><td> 5765</td><td> 1834.81104</td><td> 245.10225</td><td>Set 1</td><td>2.898302</td><td>31.42397</td></tr>\n",
       "\t<tr><th scope=row>Doc2</th><td>Doc2       </td><td> 155</td><td> 1798</td><td>  708.59679</td><td>  76.67323</td><td>Set 1</td><td>3.189476</td><td>30.20706</td></tr>\n",
       "\t<tr><th scope=row>Egfr</th><td>Egfr       </td><td> 767</td><td>14291</td><td> 3622.63139</td><td> 632.11507</td><td>Set 1</td><td>2.516498</td><td>29.75312</td></tr>\n",
       "\t<tr><th scope=row>Tet</th><td>Tet        </td><td>2013</td><td>48636</td><td> 9405.50468</td><td>1975.91807</td><td>Set 1</td><td>2.250252</td><td>29.70208</td></tr>\n",
       "\t<tr><th scope=row>AdamTS-B</th><td>AdamTS-B   </td><td> 103</td><td> 1253</td><td>  494.50129</td><td>  52.04845</td><td>Set 1</td><td>3.220592</td><td>28.83314</td></tr>\n",
       "\t<tr><th scope=row>kirre</th><td>kirre      </td><td> 761</td><td>15152</td><td> 3680.58968</td><td> 705.14779</td><td>Set 1</td><td>2.381895</td><td>28.21620</td></tr>\n",
       "\t<tr><th scope=row>DIP-epsilon</th><td>DIP-epsilon</td><td>  51</td><td>  480</td><td>  247.84092</td><td>  21.77440</td><td>Set 1</td><td>3.443929</td><td>27.41051</td></tr>\n",
       "\t<tr><th scope=row>stumps</th><td>stumps     </td><td> 317</td><td> 5785</td><td> 1490.06308</td><td> 248.36320</td><td>Set 1</td><td>2.579053</td><td>27.18870</td></tr>\n",
       "\t<tr><th scope=row>vn</th><td>vn         </td><td> 445</td><td> 8801</td><td> 2005.65104</td><td> 379.30314</td><td>Set 1</td><td>2.398849</td><td>26.31675</td></tr>\n",
       "\t<tr><th scope=row>mew</th><td>mew        </td><td> 874</td><td>20605</td><td> 4124.99469</td><td> 931.71114</td><td>Set 1</td><td>2.144890</td><td>25.76126</td></tr>\n",
       "\t<tr><th scope=row>dpr17</th><td>dpr17      </td><td>  33</td><td>  339</td><td>  169.37357</td><td>  15.15537</td><td>Set 1</td><td>3.390123</td><td>25.12949</td></tr>\n",
       "\t<tr><th scope=row>Grip</th><td>Grip       </td><td> 132</td><td> 2367</td><td>  650.40301</td><td> 101.02941</td><td>Set 1</td><td>2.672349</td><td>24.97953</td></tr>\n",
       "\t<tr><th scope=row>bin</th><td>bin        </td><td>  36</td><td>  404</td><td>  170.85063</td><td>  16.12752</td><td>Set 1</td><td>3.318347</td><td>24.63877</td></tr>\n",
       "\t<tr><th scope=row>htl</th><td>htl        </td><td> 110</td><td> 1863</td><td>  534.90162</td><td>  80.86442</td><td>Set 1</td><td>2.707965</td><td>24.54994</td></tr>\n",
       "\t<tr><th scope=row>beta-Man</th><td>beta-Man   </td><td>  90</td><td> 1460</td><td>  453.56967</td><td>  65.44308</td><td>Set 1</td><td>2.771133</td><td>24.46456</td></tr>\n",
       "\t<tr><th scope=row>scrib</th><td>scrib      </td><td> 715</td><td>18508</td><td> 3325.24371</td><td> 825.78871</td><td>Set 1</td><td>2.007869</td><td>23.49143</td></tr>\n",
       "\t<tr><th scope=row>rols</th><td>rols       </td><td> 520</td><td>12155</td><td> 2369.70052</td><td> 562.39810</td><td>Set 1</td><td>2.072478</td><td>23.23475</td></tr>\n",
       "\t<tr><th scope=row>nvy</th><td>nvy        </td><td>  73</td><td> 1330</td><td>  344.88258</td><td>  52.08342</td><td>Set 1</td><td>2.699772</td><td>22.77025</td></tr>\n",
       "\t<tr><th scope=row>CG5002</th><td>CG5002     </td><td>  36</td><td>  531</td><td>  158.64059</td><td>  18.40212</td><td>Set 1</td><td>3.031476</td><td>22.18641</td></tr>\n",
       "\t<tr><th scope=row>CG9650</th><td>CG9650     </td><td> 566</td><td>16454</td><td> 2712.18117</td><td> 709.37883</td><td>Set 1</td><td>1.932793</td><td>22.04499</td></tr>\n",
       "\t<tr><th scope=row>spri</th><td>spri       </td><td> 582</td><td>17210</td><td> 2858.17103</td><td> 776.55356</td><td>Set 1</td><td>1.878078</td><td>21.56293</td></tr>\n",
       "\t<tr><th scope=row>aos</th><td>aos        </td><td> 215</td><td> 5278</td><td> 1043.73123</td><td> 234.52877</td><td>Set 1</td><td>2.147775</td><td>21.53986</td></tr>\n",
       "\t<tr><th scope=row>pnt</th><td>pnt        </td><td> 674</td><td>20606</td><td> 3097.55987</td><td> 866.19249</td><td>Set 1</td><td>1.836708</td><td>21.30100</td></tr>\n",
       "\t<tr><th scope=row>be</th><td>be         </td><td> 261</td><td> 7125</td><td> 1280.34054</td><td> 312.47351</td><td>Set 1</td><td>2.030112</td><td>20.95774</td></tr>\n",
       "\t<tr><th scope=row>Amnionless</th><td>Amnionless </td><td>  19</td><td>  250</td><td>  104.52194</td><td>  11.09625</td><td>Set 1</td><td>3.111174</td><td>20.91145</td></tr>\n",
       "\t<tr><th scope=row>Sema1a</th><td>Sema1a     </td><td>1000</td><td>32060</td><td> 4655.05666</td><td>1415.98805</td><td>Set 1</td><td>1.715971</td><td>20.90892</td></tr>\n",
       "\t<tr><th scope=row>CadN</th><td>CadN       </td><td>1062</td><td>33747</td><td> 4894.53987</td><td>1500.31851</td><td>Set 1</td><td>1.704943</td><td>20.89792</td></tr>\n",
       "\t<tr><th scope=row>apt</th><td>apt        </td><td> 355</td><td>10277</td><td> 1704.98316</td><td> 441.72788</td><td>Set 1</td><td>1.945265</td><td>20.88512</td></tr>\n",
       "\t<tr><th scope=row>Doc1</th><td>Doc1       </td><td>  54</td><td>  993</td><td>  250.73715</td><td>  41.97202</td><td>Set 1</td><td>2.544706</td><td>20.29600</td></tr>\n",
       "\t<tr><th scope=row>cdi</th><td>cdi        </td><td> 500</td><td>15741</td><td> 2400.42520</td><td> 695.09258</td><td>Set 1</td><td>1.785939</td><td>20.05551</td></tr>\n",
       "\t<tr><th scope=row>Mef2</th><td>Mef2       </td><td> 414</td><td>13368</td><td> 1932.75822</td><td> 543.83147</td><td>Set 1</td><td>1.826779</td><td>19.94330</td></tr>\n",
       "\t<tr><th scope=row>sca</th><td>sca        </td><td> 173</td><td> 5333</td><td>  903.03694</td><td> 220.66083</td><td>Set 1</td><td>2.026431</td><td>19.90004</td></tr>\n",
       "\t<tr><th scope=row>TyrR</th><td>TyrR       </td><td>  28</td><td>  490</td><td>  154.63399</td><td>  22.40983</td><td>Set 1</td><td>2.723671</td><td>19.83381</td></tr>\n",
       "\t<tr><th scope=row>Btk29A</th><td>Btk29A     </td><td> 789</td><td>27675</td><td> 3668.98799</td><td>1161.77642</td><td>Set 1</td><td>1.657808</td><td>19.63104</td></tr>\n",
       "\t<tr><th scope=row>tacc</th><td>tacc       </td><td> 333</td><td>10326</td><td> 1606.60476</td><td> 447.29517</td><td>Set 1</td><td>1.841494</td><td>19.61320</td></tr>\n",
       "\t<tr><th scope=row>CG11357</th><td>CG11357    </td><td>  93</td><td> 2223</td><td>  446.74338</td><td>  96.58461</td><td>Set 1</td><td>2.194721</td><td>19.32787</td></tr>\n",
       "\t<tr><th scope=row>jvl</th><td>jvl        </td><td> 637</td><td>20401</td><td> 2966.77954</td><td> 928.61076</td><td>Set 1</td><td>1.674199</td><td>19.31217</td></tr>\n",
       "\t<tr><th scope=row>MESR3</th><td>MESR3      </td><td> 154</td><td> 3916</td><td>  717.09971</td><td> 175.75586</td><td>Set 1</td><td>2.020416</td><td>19.16979</td></tr>\n",
       "\t<tr><th scope=row>zfh1</th><td>zfh1       </td><td> 357</td><td>12405</td><td> 1704.80724</td><td> 514.78198</td><td>Set 1</td><td>1.724775</td><td>18.51760</td></tr>\n",
       "\t<tr><th scope=row>CG13722</th><td>CG13722    </td><td>  14</td><td>  228</td><td>   70.06575</td><td>   7.87755</td><td>Set 1</td><td>2.980476</td><td>18.33315</td></tr>\n",
       "\t<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><th scope=row>CG13287</th><td>CG13287</td><td>3</td><td>164</td><td> 6.440554</td><td> 6.0558456</td><td>Set 1</td><td>-0.1316342</td><td>-0.3811350</td></tr>\n",
       "\t<tr><th scope=row>CG15877</th><td>CG15877</td><td>2</td><td>169</td><td> 6.361384</td><td> 5.9767784</td><td>Set 1</td><td>-0.1332204</td><td>-0.3836716</td></tr>\n",
       "\t<tr><th scope=row>mrn</th><td>mrn    </td><td>2</td><td>326</td><td>11.161402</td><td>11.0410891</td><td>Set 1</td><td>-0.1094476</td><td>-0.3944752</td></tr>\n",
       "\t<tr><th scope=row>CG5056</th><td>CG5056 </td><td>2</td><td>238</td><td> 8.691202</td><td> 8.4534143</td><td>Set 1</td><td>-0.1212797</td><td>-0.3973943</td></tr>\n",
       "\t<tr><th scope=row>CG18428</th><td>CG18428</td><td>2</td><td>281</td><td>12.701564</td><td>12.6914699</td><td>Set 1</td><td>-0.1082712</td><td>-0.4088611</td></tr>\n",
       "\t<tr><th scope=row>Ugt49C1</th><td>Ugt49C1</td><td>2</td><td>103</td><td> 4.899619</td><td> 4.4748316</td><td>Set 1</td><td>-0.1601450</td><td>-0.4100708</td></tr>\n",
       "\t<tr><th scope=row>CG14490</th><td>CG14490</td><td>1</td><td> 35</td><td> 2.284545</td><td> 1.7016100</td><td>Set 1</td><td>-0.2419128</td><td>-0.4150482</td></tr>\n",
       "\t<tr><th scope=row>CG42489</th><td>CG42489</td><td>1</td><td> 53</td><td> 2.570113</td><td> 2.0078865</td><td>Set 1</td><td>-0.2269185</td><td>-0.4166155</td></tr>\n",
       "\t<tr><th scope=row>CG32536</th><td>CG32536</td><td>1</td><td>139</td><td> 6.826544</td><td> 6.5250981</td><td>Set 1</td><td>-0.1405549</td><td>-0.4172198</td></tr>\n",
       "\t<tr><th scope=row>Atac1</th><td>Atac1  </td><td>1</td><td>123</td><td> 4.642784</td><td> 4.2187544</td><td>Set 1</td><td>-0.1687153</td><td>-0.4211820</td></tr>\n",
       "\t<tr><th scope=row>CG15784</th><td>CG15784</td><td>2</td><td>309</td><td>11.515378</td><td>11.4954447</td><td>Set 1</td><td>-0.1178405</td><td>-0.4296028</td></tr>\n",
       "\t<tr><th scope=row>Ssu72</th><td>Ssu72  </td><td>3</td><td>243</td><td> 9.016918</td><td> 8.8659394</td><td>Set 1</td><td>-0.1298221</td><td>-0.4315762</td></tr>\n",
       "\t<tr><th scope=row>GstE9</th><td>GstE9  </td><td>1</td><td>108</td><td> 3.510398</td><td> 3.0318245</td><td>Set 1</td><td>-0.1997983</td><td>-0.4342126</td></tr>\n",
       "\t<tr><th scope=row>CG8908</th><td>CG8908 </td><td>1</td><td> 90</td><td> 3.997953</td><td> 3.5584482</td><td>Set 1</td><td>-0.1892812</td><td>-0.4393856</td></tr>\n",
       "\t<tr><th scope=row>CG4409</th><td>CG4409 </td><td>1</td><td> 70</td><td> 3.517547</td><td> 3.0553674</td><td>Set 1</td><td>-0.2052628</td><td>-0.4465573</td></tr>\n",
       "\t<tr><th scope=row>CG3594</th><td>CG3594 </td><td>3</td><td>247</td><td> 9.113292</td><td> 9.0043894</td><td>Set 1</td><td>-0.1345889</td><td>-0.4492821</td></tr>\n",
       "\t<tr><th scope=row>Cpr73D</th><td>Cpr73D </td><td>1</td><td> 49</td><td> 2.686028</td><td> 2.1700670</td><td>Set 1</td><td>-0.2390392</td><td>-0.4498878</td></tr>\n",
       "\t<tr><th scope=row>ItgaPS4</th><td>ItgaPS4</td><td>1</td><td> 18</td><td> 1.315397</td><td> 0.7098502</td><td>Set 1</td><td>-0.3783721</td><td>-0.4583068</td></tr>\n",
       "\t<tr><th scope=row>CG12268</th><td>CG12268</td><td>1</td><td>138</td><td> 5.482907</td><td> 5.1780722</td><td>Set 1</td><td>-0.1722157</td><td>-0.4644039</td></tr>\n",
       "\t<tr><th scope=row>CG13151</th><td>CG13151</td><td>1</td><td>151</td><td> 5.955571</td><td> 5.7031899</td><td>Set 1</td><td>-0.1706079</td><td>-0.4773896</td></tr>\n",
       "\t<tr><th scope=row>CG1946</th><td>CG1946 </td><td>1</td><td>165</td><td> 6.257666</td><td> 6.0370289</td><td>Set 1</td><td>-0.1693419</td><td>-0.4842340</td></tr>\n",
       "\t<tr><th scope=row>cer</th><td>cer    </td><td>1</td><td> 87</td><td> 4.439886</td><td> 4.0992364</td><td>Set 1</td><td>-0.1997585</td><td>-0.4881252</td></tr>\n",
       "\t<tr><th scope=row>CG44385</th><td>CG44385</td><td>1</td><td> 78</td><td> 3.674714</td><td> 3.2923932</td><td>Set 1</td><td>-0.2241501</td><td>-0.4987068</td></tr>\n",
       "\t<tr><th scope=row>CG33230</th><td>CG33230</td><td>1</td><td>210</td><td> 7.287408</td><td> 7.1866026</td><td>Set 1</td><td>-0.1678591</td><td>-0.5121248</td></tr>\n",
       "\t<tr><th scope=row>Zmynd10</th><td>Zmynd10</td><td>1</td><td>104</td><td> 4.824346</td><td> 4.5523527</td><td>Set 1</td><td>-0.2027660</td><td>-0.5154505</td></tr>\n",
       "\t<tr><th scope=row>CG33494</th><td>CG33494</td><td>1</td><td>106</td><td> 5.124972</td><td> 4.9003573</td><td>Set 1</td><td>-0.2032581</td><td>-0.5314596</td></tr>\n",
       "\t<tr><th scope=row>CG32182</th><td>CG32182</td><td>1</td><td> 82</td><td> 3.327776</td><td> 2.9680063</td><td>Set 1</td><td>-0.2538560</td><td>-0.5365567</td></tr>\n",
       "\t<tr><th scope=row>Rsph1</th><td>Rsph1  </td><td>1</td><td>141</td><td> 6.038866</td><td> 5.8918577</td><td>Set 1</td><td>-0.1906152</td><td>-0.5366472</td></tr>\n",
       "\t<tr><th scope=row>Tap42</th><td>Tap42  </td><td>2</td><td>194</td><td> 7.422430</td><td> 7.3878435</td><td>Set 1</td><td>-0.1764084</td><td>-0.5423211</td></tr>\n",
       "\t<tr><th scope=row>CG14977</th><td>CG14977</td><td>1</td><td>101</td><td> 4.141764</td><td> 3.8581433</td><td>Set 1</td><td>-0.2301596</td><td>-0.5436976</td></tr>\n",
       "\t<tr><th scope=row>CG5359</th><td>CG5359 </td><td>3</td><td>149</td><td> 6.015768</td><td> 5.8797037</td><td>Set 1</td><td>-0.1935974</td><td>-0.5441251</td></tr>\n",
       "\t<tr><th scope=row>fzo</th><td>fzo    </td><td>1</td><td> 82</td><td> 4.102413</td><td> 3.8254926</td><td>Set 1</td><td>-0.2342036</td><td>-0.5506548</td></tr>\n",
       "\t<tr><th scope=row>CG9903</th><td>CG9903 </td><td>1</td><td>143</td><td> 6.373121</td><td> 6.2829359</td><td>Set 1</td><td>-0.1925201</td><td>-0.5548961</td></tr>\n",
       "\t<tr><th scope=row>CG12105</th><td>CG12105</td><td>1</td><td>156</td><td> 5.330604</td><td> 5.1732895</td><td>Set 1</td><td>-0.2117404</td><td>-0.5637257</td></tr>\n",
       "\t<tr><th scope=row>CG11454</th><td>CG11454</td><td>1</td><td>104</td><td> 3.925263</td><td> 3.6663117</td><td>Set 1</td><td>-0.2494934</td><td>-0.5738848</td></tr>\n",
       "\t<tr><th scope=row>PCNA2</th><td>PCNA2  </td><td>1</td><td>137</td><td> 5.363713</td><td> 5.2315727</td><td>Set 1</td><td>-0.2163641</td><td>-0.5776639</td></tr>\n",
       "\t<tr><th scope=row>CG10208</th><td>CG10208</td><td>2</td><td>141</td><td> 5.448315</td><td> 5.3291836</td><td>Set 1</td><td>-0.2162094</td><td>-0.5813703</td></tr>\n",
       "\t<tr><th scope=row>CG17364</th><td>CG17364</td><td>1</td><td>103</td><td> 4.655298</td><td> 4.5142193</td><td>Set 1</td><td>-0.2442830</td><td>-0.6106106</td></tr>\n",
       "\t<tr><th scope=row>CG6985</th><td>CG6985 </td><td>1</td><td>155</td><td> 5.815204</td><td> 5.7863254</td><td>Set 1</td><td>-0.2228007</td><td>-0.6168810</td></tr>\n",
       "\t<tr><th scope=row>Vsx2</th><td>Vsx2   </td><td>1</td><td>110</td><td> 4.906579</td><td> 4.8082809</td><td>Set 1</td><td>-0.2433938</td><td>-0.6236535</td></tr>\n",
       "\t<tr><th scope=row>CG8191</th><td>CG8191 </td><td>1</td><td> 98</td><td> 3.682550</td><td> 3.4920553</td><td>Set 1</td><td>-0.2866708</td><td>-0.6385001</td></tr>\n",
       "\t<tr><th scope=row>mei-38</th><td>mei-38 </td><td>1</td><td>151</td><td> 5.064855</td><td> 5.0444964</td><td>Set 1</td><td>-0.2551010</td><td>-0.6633834</td></tr>\n",
       "\t<tr><th scope=row>CG31957</th><td>CG31957</td><td>1</td><td> 74</td><td> 3.754599</td><td> 3.6069260</td><td>Set 1</td><td>-0.2951455</td><td>-0.6638777</td></tr>\n",
       "\t<tr><th scope=row>MED10</th><td>MED10  </td><td>1</td><td>127</td><td> 4.642784</td><td> 4.5887821</td><td>Set 1</td><td>-0.2675437</td><td>-0.6678980</td></tr>\n",
       "\t<tr><th scope=row>CG9394</th><td>CG9394 </td><td>1</td><td> 39</td><td> 1.942762</td><td> 1.6230231</td><td>Set 1</td><td>-0.4331211</td><td>-0.6744436</td></tr>\n",
       "\t<tr><th scope=row>CG12253</th><td>CG12253</td><td>1</td><td>107</td><td> 4.383542</td><td> 4.3663049</td><td>Set 1</td><td>-0.2918318</td><td>-0.7087298</td></tr>\n",
       "\t<tr><th scope=row>CG12713</th><td>CG12713</td><td>1</td><td> 87</td><td> 3.468104</td><td> 3.3753788</td><td>Set 1</td><td>-0.3352608</td><td>-0.7240503</td></tr>\n",
       "\t<tr><th scope=row>CG31223</th><td>CG31223</td><td>1</td><td> 69</td><td> 2.724157</td><td> 2.5899814</td><td>Set 1</td><td>-0.3981664</td><td>-0.7552874</td></tr>\n",
       "\t<tr><th scope=row>TrpA1</th><td>TrpA1  </td><td>1</td><td> 45</td><td> 1.929738</td><td> 1.7933341</td><td>Set 1</td><td>-0.5335830</td><td>-0.8274654</td></tr>\n",
       "\t<tr><th scope=row>Fign</th><td>Fign   </td><td>1</td><td> 65</td><td> 2.506699</td><td> 2.4681571</td><td>Set 1</td><td>-0.4683804</td><td>-0.8478218</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2931 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\ttey & tey         &  967 &  1836 &  4611.02394 &   78.52727 & Set 1 & 5.857494 & 71.29264\\\\\n",
       "\tHLH54F & HLH54F      &  251 &   336 &  1223.38930 &   13.18234 & Set 1 & 6.430644 & 65.96456\\\\\n",
       "\tbeat-IIb & beat-IIb    &  641 &  2319 &  3062.79663 &  105.41408 & Set 1 & 4.847089 & 56.13464\\\\\n",
       "\tbeat-IIa & beat-IIa    &  423 &  1642 &  2054.21747 &   72.15760 & Set 1 & 4.811437 & 52.95023\\\\\n",
       "\tbru3 & bru3        & 3545 & 41521 & 16857.46794 & 1860.61444 & Set 1 & 3.178762 & 44.63358\\\\\n",
       "\ttok & tok         &  234 &  1776 &  1139.77013 &   78.64160 & Set 1 & 3.839077 & 38.98887\\\\\n",
       "\tH2.0 & H2.0        &   87 &   543 &   409.42302 &   22.36169 & Set 1 & 4.131375 & 35.86434\\\\\n",
       "\tATP8B & ATP8B       &  134 &  1051 &   644.78794 &   47.51049 & Set 1 & 3.732456 & 34.84217\\\\\n",
       "\tRapGAP1 & RapGAP1     & 2368 & 44286 & 11055.15800 & 1997.87862 & Set 1 & 2.467457 & 33.14427\\\\\n",
       "\tSyn2 & Syn2        &   64 &   472 &   322.69397 &   20.38014 & Set 1 & 3.915824 & 32.65204\\\\\n",
       "\tkon & kon         &  393 &  5765 &  1834.81104 &  245.10225 & Set 1 & 2.898302 & 31.42397\\\\\n",
       "\tDoc2 & Doc2        &  155 &  1798 &   708.59679 &   76.67323 & Set 1 & 3.189476 & 30.20706\\\\\n",
       "\tEgfr & Egfr        &  767 & 14291 &  3622.63139 &  632.11507 & Set 1 & 2.516498 & 29.75312\\\\\n",
       "\tTet & Tet         & 2013 & 48636 &  9405.50468 & 1975.91807 & Set 1 & 2.250252 & 29.70208\\\\\n",
       "\tAdamTS-B & AdamTS-B    &  103 &  1253 &   494.50129 &   52.04845 & Set 1 & 3.220592 & 28.83314\\\\\n",
       "\tkirre & kirre       &  761 & 15152 &  3680.58968 &  705.14779 & Set 1 & 2.381895 & 28.21620\\\\\n",
       "\tDIP-epsilon & DIP-epsilon &   51 &   480 &   247.84092 &   21.77440 & Set 1 & 3.443929 & 27.41051\\\\\n",
       "\tstumps & stumps      &  317 &  5785 &  1490.06308 &  248.36320 & Set 1 & 2.579053 & 27.18870\\\\\n",
       "\tvn & vn          &  445 &  8801 &  2005.65104 &  379.30314 & Set 1 & 2.398849 & 26.31675\\\\\n",
       "\tmew & mew         &  874 & 20605 &  4124.99469 &  931.71114 & Set 1 & 2.144890 & 25.76126\\\\\n",
       "\tdpr17 & dpr17       &   33 &   339 &   169.37357 &   15.15537 & Set 1 & 3.390123 & 25.12949\\\\\n",
       "\tGrip & Grip        &  132 &  2367 &   650.40301 &  101.02941 & Set 1 & 2.672349 & 24.97953\\\\\n",
       "\tbin & bin         &   36 &   404 &   170.85063 &   16.12752 & Set 1 & 3.318347 & 24.63877\\\\\n",
       "\thtl & htl         &  110 &  1863 &   534.90162 &   80.86442 & Set 1 & 2.707965 & 24.54994\\\\\n",
       "\tbeta-Man & beta-Man    &   90 &  1460 &   453.56967 &   65.44308 & Set 1 & 2.771133 & 24.46456\\\\\n",
       "\tscrib & scrib       &  715 & 18508 &  3325.24371 &  825.78871 & Set 1 & 2.007869 & 23.49143\\\\\n",
       "\trols & rols        &  520 & 12155 &  2369.70052 &  562.39810 & Set 1 & 2.072478 & 23.23475\\\\\n",
       "\tnvy & nvy         &   73 &  1330 &   344.88258 &   52.08342 & Set 1 & 2.699772 & 22.77025\\\\\n",
       "\tCG5002 & CG5002      &   36 &   531 &   158.64059 &   18.40212 & Set 1 & 3.031476 & 22.18641\\\\\n",
       "\tCG9650 & CG9650      &  566 & 16454 &  2712.18117 &  709.37883 & Set 1 & 1.932793 & 22.04499\\\\\n",
       "\tspri & spri        &  582 & 17210 &  2858.17103 &  776.55356 & Set 1 & 1.878078 & 21.56293\\\\\n",
       "\taos & aos         &  215 &  5278 &  1043.73123 &  234.52877 & Set 1 & 2.147775 & 21.53986\\\\\n",
       "\tpnt & pnt         &  674 & 20606 &  3097.55987 &  866.19249 & Set 1 & 1.836708 & 21.30100\\\\\n",
       "\tbe & be          &  261 &  7125 &  1280.34054 &  312.47351 & Set 1 & 2.030112 & 20.95774\\\\\n",
       "\tAmnionless & Amnionless  &   19 &   250 &   104.52194 &   11.09625 & Set 1 & 3.111174 & 20.91145\\\\\n",
       "\tSema1a & Sema1a      & 1000 & 32060 &  4655.05666 & 1415.98805 & Set 1 & 1.715971 & 20.90892\\\\\n",
       "\tCadN & CadN        & 1062 & 33747 &  4894.53987 & 1500.31851 & Set 1 & 1.704943 & 20.89792\\\\\n",
       "\tapt & apt         &  355 & 10277 &  1704.98316 &  441.72788 & Set 1 & 1.945265 & 20.88512\\\\\n",
       "\tDoc1 & Doc1        &   54 &   993 &   250.73715 &   41.97202 & Set 1 & 2.544706 & 20.29600\\\\\n",
       "\tcdi & cdi         &  500 & 15741 &  2400.42520 &  695.09258 & Set 1 & 1.785939 & 20.05551\\\\\n",
       "\tMef2 & Mef2        &  414 & 13368 &  1932.75822 &  543.83147 & Set 1 & 1.826779 & 19.94330\\\\\n",
       "\tsca & sca         &  173 &  5333 &   903.03694 &  220.66083 & Set 1 & 2.026431 & 19.90004\\\\\n",
       "\tTyrR & TyrR        &   28 &   490 &   154.63399 &   22.40983 & Set 1 & 2.723671 & 19.83381\\\\\n",
       "\tBtk29A & Btk29A      &  789 & 27675 &  3668.98799 & 1161.77642 & Set 1 & 1.657808 & 19.63104\\\\\n",
       "\ttacc & tacc        &  333 & 10326 &  1606.60476 &  447.29517 & Set 1 & 1.841494 & 19.61320\\\\\n",
       "\tCG11357 & CG11357     &   93 &  2223 &   446.74338 &   96.58461 & Set 1 & 2.194721 & 19.32787\\\\\n",
       "\tjvl & jvl         &  637 & 20401 &  2966.77954 &  928.61076 & Set 1 & 1.674199 & 19.31217\\\\\n",
       "\tMESR3 & MESR3       &  154 &  3916 &   717.09971 &  175.75586 & Set 1 & 2.020416 & 19.16979\\\\\n",
       "\tzfh1 & zfh1        &  357 & 12405 &  1704.80724 &  514.78198 & Set 1 & 1.724775 & 18.51760\\\\\n",
       "\tCG13722 & CG13722     &   14 &   228 &    70.06575 &    7.87755 & Set 1 & 2.980476 & 18.33315\\\\\n",
       "\t⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\tCG13287 & CG13287 & 3 & 164 &  6.440554 &  6.0558456 & Set 1 & -0.1316342 & -0.3811350\\\\\n",
       "\tCG15877 & CG15877 & 2 & 169 &  6.361384 &  5.9767784 & Set 1 & -0.1332204 & -0.3836716\\\\\n",
       "\tmrn & mrn     & 2 & 326 & 11.161402 & 11.0410891 & Set 1 & -0.1094476 & -0.3944752\\\\\n",
       "\tCG5056 & CG5056  & 2 & 238 &  8.691202 &  8.4534143 & Set 1 & -0.1212797 & -0.3973943\\\\\n",
       "\tCG18428 & CG18428 & 2 & 281 & 12.701564 & 12.6914699 & Set 1 & -0.1082712 & -0.4088611\\\\\n",
       "\tUgt49C1 & Ugt49C1 & 2 & 103 &  4.899619 &  4.4748316 & Set 1 & -0.1601450 & -0.4100708\\\\\n",
       "\tCG14490 & CG14490 & 1 &  35 &  2.284545 &  1.7016100 & Set 1 & -0.2419128 & -0.4150482\\\\\n",
       "\tCG42489 & CG42489 & 1 &  53 &  2.570113 &  2.0078865 & Set 1 & -0.2269185 & -0.4166155\\\\\n",
       "\tCG32536 & CG32536 & 1 & 139 &  6.826544 &  6.5250981 & Set 1 & -0.1405549 & -0.4172198\\\\\n",
       "\tAtac1 & Atac1   & 1 & 123 &  4.642784 &  4.2187544 & Set 1 & -0.1687153 & -0.4211820\\\\\n",
       "\tCG15784 & CG15784 & 2 & 309 & 11.515378 & 11.4954447 & Set 1 & -0.1178405 & -0.4296028\\\\\n",
       "\tSsu72 & Ssu72   & 3 & 243 &  9.016918 &  8.8659394 & Set 1 & -0.1298221 & -0.4315762\\\\\n",
       "\tGstE9 & GstE9   & 1 & 108 &  3.510398 &  3.0318245 & Set 1 & -0.1997983 & -0.4342126\\\\\n",
       "\tCG8908 & CG8908  & 1 &  90 &  3.997953 &  3.5584482 & Set 1 & -0.1892812 & -0.4393856\\\\\n",
       "\tCG4409 & CG4409  & 1 &  70 &  3.517547 &  3.0553674 & Set 1 & -0.2052628 & -0.4465573\\\\\n",
       "\tCG3594 & CG3594  & 3 & 247 &  9.113292 &  9.0043894 & Set 1 & -0.1345889 & -0.4492821\\\\\n",
       "\tCpr73D & Cpr73D  & 1 &  49 &  2.686028 &  2.1700670 & Set 1 & -0.2390392 & -0.4498878\\\\\n",
       "\tItgaPS4 & ItgaPS4 & 1 &  18 &  1.315397 &  0.7098502 & Set 1 & -0.3783721 & -0.4583068\\\\\n",
       "\tCG12268 & CG12268 & 1 & 138 &  5.482907 &  5.1780722 & Set 1 & -0.1722157 & -0.4644039\\\\\n",
       "\tCG13151 & CG13151 & 1 & 151 &  5.955571 &  5.7031899 & Set 1 & -0.1706079 & -0.4773896\\\\\n",
       "\tCG1946 & CG1946  & 1 & 165 &  6.257666 &  6.0370289 & Set 1 & -0.1693419 & -0.4842340\\\\\n",
       "\tcer & cer     & 1 &  87 &  4.439886 &  4.0992364 & Set 1 & -0.1997585 & -0.4881252\\\\\n",
       "\tCG44385 & CG44385 & 1 &  78 &  3.674714 &  3.2923932 & Set 1 & -0.2241501 & -0.4987068\\\\\n",
       "\tCG33230 & CG33230 & 1 & 210 &  7.287408 &  7.1866026 & Set 1 & -0.1678591 & -0.5121248\\\\\n",
       "\tZmynd10 & Zmynd10 & 1 & 104 &  4.824346 &  4.5523527 & Set 1 & -0.2027660 & -0.5154505\\\\\n",
       "\tCG33494 & CG33494 & 1 & 106 &  5.124972 &  4.9003573 & Set 1 & -0.2032581 & -0.5314596\\\\\n",
       "\tCG32182 & CG32182 & 1 &  82 &  3.327776 &  2.9680063 & Set 1 & -0.2538560 & -0.5365567\\\\\n",
       "\tRsph1 & Rsph1   & 1 & 141 &  6.038866 &  5.8918577 & Set 1 & -0.1906152 & -0.5366472\\\\\n",
       "\tTap42 & Tap42   & 2 & 194 &  7.422430 &  7.3878435 & Set 1 & -0.1764084 & -0.5423211\\\\\n",
       "\tCG14977 & CG14977 & 1 & 101 &  4.141764 &  3.8581433 & Set 1 & -0.2301596 & -0.5436976\\\\\n",
       "\tCG5359 & CG5359  & 3 & 149 &  6.015768 &  5.8797037 & Set 1 & -0.1935974 & -0.5441251\\\\\n",
       "\tfzo & fzo     & 1 &  82 &  4.102413 &  3.8254926 & Set 1 & -0.2342036 & -0.5506548\\\\\n",
       "\tCG9903 & CG9903  & 1 & 143 &  6.373121 &  6.2829359 & Set 1 & -0.1925201 & -0.5548961\\\\\n",
       "\tCG12105 & CG12105 & 1 & 156 &  5.330604 &  5.1732895 & Set 1 & -0.2117404 & -0.5637257\\\\\n",
       "\tCG11454 & CG11454 & 1 & 104 &  3.925263 &  3.6663117 & Set 1 & -0.2494934 & -0.5738848\\\\\n",
       "\tPCNA2 & PCNA2   & 1 & 137 &  5.363713 &  5.2315727 & Set 1 & -0.2163641 & -0.5776639\\\\\n",
       "\tCG10208 & CG10208 & 2 & 141 &  5.448315 &  5.3291836 & Set 1 & -0.2162094 & -0.5813703\\\\\n",
       "\tCG17364 & CG17364 & 1 & 103 &  4.655298 &  4.5142193 & Set 1 & -0.2442830 & -0.6106106\\\\\n",
       "\tCG6985 & CG6985  & 1 & 155 &  5.815204 &  5.7863254 & Set 1 & -0.2228007 & -0.6168810\\\\\n",
       "\tVsx2 & Vsx2    & 1 & 110 &  4.906579 &  4.8082809 & Set 1 & -0.2433938 & -0.6236535\\\\\n",
       "\tCG8191 & CG8191  & 1 &  98 &  3.682550 &  3.4920553 & Set 1 & -0.2866708 & -0.6385001\\\\\n",
       "\tmei-38 & mei-38  & 1 & 151 &  5.064855 &  5.0444964 & Set 1 & -0.2551010 & -0.6633834\\\\\n",
       "\tCG31957 & CG31957 & 1 &  74 &  3.754599 &  3.6069260 & Set 1 & -0.2951455 & -0.6638777\\\\\n",
       "\tMED10 & MED10   & 1 & 127 &  4.642784 &  4.5887821 & Set 1 & -0.2675437 & -0.6678980\\\\\n",
       "\tCG9394 & CG9394  & 1 &  39 &  1.942762 &  1.6230231 & Set 1 & -0.4331211 & -0.6744436\\\\\n",
       "\tCG12253 & CG12253 & 1 & 107 &  4.383542 &  4.3663049 & Set 1 & -0.2918318 & -0.7087298\\\\\n",
       "\tCG12713 & CG12713 & 1 &  87 &  3.468104 &  3.3753788 & Set 1 & -0.3352608 & -0.7240503\\\\\n",
       "\tCG31223 & CG31223 & 1 &  69 &  2.724157 &  2.5899814 & Set 1 & -0.3981664 & -0.7552874\\\\\n",
       "\tTrpA1 & TrpA1   & 1 &  45 &  1.929738 &  1.7933341 & Set 1 & -0.5335830 & -0.8274654\\\\\n",
       "\tFign & Fign    & 1 &  65 &  2.506699 &  2.4681571 & Set 1 & -0.4683804 & -0.8478218\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2931 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| tey | tey         |  967 |  1836 |  4611.02394 |   78.52727 | Set 1 | 5.857494 | 71.29264 |\n",
       "| HLH54F | HLH54F      |  251 |   336 |  1223.38930 |   13.18234 | Set 1 | 6.430644 | 65.96456 |\n",
       "| beat-IIb | beat-IIb    |  641 |  2319 |  3062.79663 |  105.41408 | Set 1 | 4.847089 | 56.13464 |\n",
       "| beat-IIa | beat-IIa    |  423 |  1642 |  2054.21747 |   72.15760 | Set 1 | 4.811437 | 52.95023 |\n",
       "| bru3 | bru3        | 3545 | 41521 | 16857.46794 | 1860.61444 | Set 1 | 3.178762 | 44.63358 |\n",
       "| tok | tok         |  234 |  1776 |  1139.77013 |   78.64160 | Set 1 | 3.839077 | 38.98887 |\n",
       "| H2.0 | H2.0        |   87 |   543 |   409.42302 |   22.36169 | Set 1 | 4.131375 | 35.86434 |\n",
       "| ATP8B | ATP8B       |  134 |  1051 |   644.78794 |   47.51049 | Set 1 | 3.732456 | 34.84217 |\n",
       "| RapGAP1 | RapGAP1     | 2368 | 44286 | 11055.15800 | 1997.87862 | Set 1 | 2.467457 | 33.14427 |\n",
       "| Syn2 | Syn2        |   64 |   472 |   322.69397 |   20.38014 | Set 1 | 3.915824 | 32.65204 |\n",
       "| kon | kon         |  393 |  5765 |  1834.81104 |  245.10225 | Set 1 | 2.898302 | 31.42397 |\n",
       "| Doc2 | Doc2        |  155 |  1798 |   708.59679 |   76.67323 | Set 1 | 3.189476 | 30.20706 |\n",
       "| Egfr | Egfr        |  767 | 14291 |  3622.63139 |  632.11507 | Set 1 | 2.516498 | 29.75312 |\n",
       "| Tet | Tet         | 2013 | 48636 |  9405.50468 | 1975.91807 | Set 1 | 2.250252 | 29.70208 |\n",
       "| AdamTS-B | AdamTS-B    |  103 |  1253 |   494.50129 |   52.04845 | Set 1 | 3.220592 | 28.83314 |\n",
       "| kirre | kirre       |  761 | 15152 |  3680.58968 |  705.14779 | Set 1 | 2.381895 | 28.21620 |\n",
       "| DIP-epsilon | DIP-epsilon |   51 |   480 |   247.84092 |   21.77440 | Set 1 | 3.443929 | 27.41051 |\n",
       "| stumps | stumps      |  317 |  5785 |  1490.06308 |  248.36320 | Set 1 | 2.579053 | 27.18870 |\n",
       "| vn | vn          |  445 |  8801 |  2005.65104 |  379.30314 | Set 1 | 2.398849 | 26.31675 |\n",
       "| mew | mew         |  874 | 20605 |  4124.99469 |  931.71114 | Set 1 | 2.144890 | 25.76126 |\n",
       "| dpr17 | dpr17       |   33 |   339 |   169.37357 |   15.15537 | Set 1 | 3.390123 | 25.12949 |\n",
       "| Grip | Grip        |  132 |  2367 |   650.40301 |  101.02941 | Set 1 | 2.672349 | 24.97953 |\n",
       "| bin | bin         |   36 |   404 |   170.85063 |   16.12752 | Set 1 | 3.318347 | 24.63877 |\n",
       "| htl | htl         |  110 |  1863 |   534.90162 |   80.86442 | Set 1 | 2.707965 | 24.54994 |\n",
       "| beta-Man | beta-Man    |   90 |  1460 |   453.56967 |   65.44308 | Set 1 | 2.771133 | 24.46456 |\n",
       "| scrib | scrib       |  715 | 18508 |  3325.24371 |  825.78871 | Set 1 | 2.007869 | 23.49143 |\n",
       "| rols | rols        |  520 | 12155 |  2369.70052 |  562.39810 | Set 1 | 2.072478 | 23.23475 |\n",
       "| nvy | nvy         |   73 |  1330 |   344.88258 |   52.08342 | Set 1 | 2.699772 | 22.77025 |\n",
       "| CG5002 | CG5002      |   36 |   531 |   158.64059 |   18.40212 | Set 1 | 3.031476 | 22.18641 |\n",
       "| CG9650 | CG9650      |  566 | 16454 |  2712.18117 |  709.37883 | Set 1 | 1.932793 | 22.04499 |\n",
       "| spri | spri        |  582 | 17210 |  2858.17103 |  776.55356 | Set 1 | 1.878078 | 21.56293 |\n",
       "| aos | aos         |  215 |  5278 |  1043.73123 |  234.52877 | Set 1 | 2.147775 | 21.53986 |\n",
       "| pnt | pnt         |  674 | 20606 |  3097.55987 |  866.19249 | Set 1 | 1.836708 | 21.30100 |\n",
       "| be | be          |  261 |  7125 |  1280.34054 |  312.47351 | Set 1 | 2.030112 | 20.95774 |\n",
       "| Amnionless | Amnionless  |   19 |   250 |   104.52194 |   11.09625 | Set 1 | 3.111174 | 20.91145 |\n",
       "| Sema1a | Sema1a      | 1000 | 32060 |  4655.05666 | 1415.98805 | Set 1 | 1.715971 | 20.90892 |\n",
       "| CadN | CadN        | 1062 | 33747 |  4894.53987 | 1500.31851 | Set 1 | 1.704943 | 20.89792 |\n",
       "| apt | apt         |  355 | 10277 |  1704.98316 |  441.72788 | Set 1 | 1.945265 | 20.88512 |\n",
       "| Doc1 | Doc1        |   54 |   993 |   250.73715 |   41.97202 | Set 1 | 2.544706 | 20.29600 |\n",
       "| cdi | cdi         |  500 | 15741 |  2400.42520 |  695.09258 | Set 1 | 1.785939 | 20.05551 |\n",
       "| Mef2 | Mef2        |  414 | 13368 |  1932.75822 |  543.83147 | Set 1 | 1.826779 | 19.94330 |\n",
       "| sca | sca         |  173 |  5333 |   903.03694 |  220.66083 | Set 1 | 2.026431 | 19.90004 |\n",
       "| TyrR | TyrR        |   28 |   490 |   154.63399 |   22.40983 | Set 1 | 2.723671 | 19.83381 |\n",
       "| Btk29A | Btk29A      |  789 | 27675 |  3668.98799 | 1161.77642 | Set 1 | 1.657808 | 19.63104 |\n",
       "| tacc | tacc        |  333 | 10326 |  1606.60476 |  447.29517 | Set 1 | 1.841494 | 19.61320 |\n",
       "| CG11357 | CG11357     |   93 |  2223 |   446.74338 |   96.58461 | Set 1 | 2.194721 | 19.32787 |\n",
       "| jvl | jvl         |  637 | 20401 |  2966.77954 |  928.61076 | Set 1 | 1.674199 | 19.31217 |\n",
       "| MESR3 | MESR3       |  154 |  3916 |   717.09971 |  175.75586 | Set 1 | 2.020416 | 19.16979 |\n",
       "| zfh1 | zfh1        |  357 | 12405 |  1704.80724 |  514.78198 | Set 1 | 1.724775 | 18.51760 |\n",
       "| CG13722 | CG13722     |   14 |   228 |    70.06575 |    7.87755 | Set 1 | 2.980476 | 18.33315 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| CG13287 | CG13287 | 3 | 164 |  6.440554 |  6.0558456 | Set 1 | -0.1316342 | -0.3811350 |\n",
       "| CG15877 | CG15877 | 2 | 169 |  6.361384 |  5.9767784 | Set 1 | -0.1332204 | -0.3836716 |\n",
       "| mrn | mrn     | 2 | 326 | 11.161402 | 11.0410891 | Set 1 | -0.1094476 | -0.3944752 |\n",
       "| CG5056 | CG5056  | 2 | 238 |  8.691202 |  8.4534143 | Set 1 | -0.1212797 | -0.3973943 |\n",
       "| CG18428 | CG18428 | 2 | 281 | 12.701564 | 12.6914699 | Set 1 | -0.1082712 | -0.4088611 |\n",
       "| Ugt49C1 | Ugt49C1 | 2 | 103 |  4.899619 |  4.4748316 | Set 1 | -0.1601450 | -0.4100708 |\n",
       "| CG14490 | CG14490 | 1 |  35 |  2.284545 |  1.7016100 | Set 1 | -0.2419128 | -0.4150482 |\n",
       "| CG42489 | CG42489 | 1 |  53 |  2.570113 |  2.0078865 | Set 1 | -0.2269185 | -0.4166155 |\n",
       "| CG32536 | CG32536 | 1 | 139 |  6.826544 |  6.5250981 | Set 1 | -0.1405549 | -0.4172198 |\n",
       "| Atac1 | Atac1   | 1 | 123 |  4.642784 |  4.2187544 | Set 1 | -0.1687153 | -0.4211820 |\n",
       "| CG15784 | CG15784 | 2 | 309 | 11.515378 | 11.4954447 | Set 1 | -0.1178405 | -0.4296028 |\n",
       "| Ssu72 | Ssu72   | 3 | 243 |  9.016918 |  8.8659394 | Set 1 | -0.1298221 | -0.4315762 |\n",
       "| GstE9 | GstE9   | 1 | 108 |  3.510398 |  3.0318245 | Set 1 | -0.1997983 | -0.4342126 |\n",
       "| CG8908 | CG8908  | 1 |  90 |  3.997953 |  3.5584482 | Set 1 | -0.1892812 | -0.4393856 |\n",
       "| CG4409 | CG4409  | 1 |  70 |  3.517547 |  3.0553674 | Set 1 | -0.2052628 | -0.4465573 |\n",
       "| CG3594 | CG3594  | 3 | 247 |  9.113292 |  9.0043894 | Set 1 | -0.1345889 | -0.4492821 |\n",
       "| Cpr73D | Cpr73D  | 1 |  49 |  2.686028 |  2.1700670 | Set 1 | -0.2390392 | -0.4498878 |\n",
       "| ItgaPS4 | ItgaPS4 | 1 |  18 |  1.315397 |  0.7098502 | Set 1 | -0.3783721 | -0.4583068 |\n",
       "| CG12268 | CG12268 | 1 | 138 |  5.482907 |  5.1780722 | Set 1 | -0.1722157 | -0.4644039 |\n",
       "| CG13151 | CG13151 | 1 | 151 |  5.955571 |  5.7031899 | Set 1 | -0.1706079 | -0.4773896 |\n",
       "| CG1946 | CG1946  | 1 | 165 |  6.257666 |  6.0370289 | Set 1 | -0.1693419 | -0.4842340 |\n",
       "| cer | cer     | 1 |  87 |  4.439886 |  4.0992364 | Set 1 | -0.1997585 | -0.4881252 |\n",
       "| CG44385 | CG44385 | 1 |  78 |  3.674714 |  3.2923932 | Set 1 | -0.2241501 | -0.4987068 |\n",
       "| CG33230 | CG33230 | 1 | 210 |  7.287408 |  7.1866026 | Set 1 | -0.1678591 | -0.5121248 |\n",
       "| Zmynd10 | Zmynd10 | 1 | 104 |  4.824346 |  4.5523527 | Set 1 | -0.2027660 | -0.5154505 |\n",
       "| CG33494 | CG33494 | 1 | 106 |  5.124972 |  4.9003573 | Set 1 | -0.2032581 | -0.5314596 |\n",
       "| CG32182 | CG32182 | 1 |  82 |  3.327776 |  2.9680063 | Set 1 | -0.2538560 | -0.5365567 |\n",
       "| Rsph1 | Rsph1   | 1 | 141 |  6.038866 |  5.8918577 | Set 1 | -0.1906152 | -0.5366472 |\n",
       "| Tap42 | Tap42   | 2 | 194 |  7.422430 |  7.3878435 | Set 1 | -0.1764084 | -0.5423211 |\n",
       "| CG14977 | CG14977 | 1 | 101 |  4.141764 |  3.8581433 | Set 1 | -0.2301596 | -0.5436976 |\n",
       "| CG5359 | CG5359  | 3 | 149 |  6.015768 |  5.8797037 | Set 1 | -0.1935974 | -0.5441251 |\n",
       "| fzo | fzo     | 1 |  82 |  4.102413 |  3.8254926 | Set 1 | -0.2342036 | -0.5506548 |\n",
       "| CG9903 | CG9903  | 1 | 143 |  6.373121 |  6.2829359 | Set 1 | -0.1925201 | -0.5548961 |\n",
       "| CG12105 | CG12105 | 1 | 156 |  5.330604 |  5.1732895 | Set 1 | -0.2117404 | -0.5637257 |\n",
       "| CG11454 | CG11454 | 1 | 104 |  3.925263 |  3.6663117 | Set 1 | -0.2494934 | -0.5738848 |\n",
       "| PCNA2 | PCNA2   | 1 | 137 |  5.363713 |  5.2315727 | Set 1 | -0.2163641 | -0.5776639 |\n",
       "| CG10208 | CG10208 | 2 | 141 |  5.448315 |  5.3291836 | Set 1 | -0.2162094 | -0.5813703 |\n",
       "| CG17364 | CG17364 | 1 | 103 |  4.655298 |  4.5142193 | Set 1 | -0.2442830 | -0.6106106 |\n",
       "| CG6985 | CG6985  | 1 | 155 |  5.815204 |  5.7863254 | Set 1 | -0.2228007 | -0.6168810 |\n",
       "| Vsx2 | Vsx2    | 1 | 110 |  4.906579 |  4.8082809 | Set 1 | -0.2433938 | -0.6236535 |\n",
       "| CG8191 | CG8191  | 1 |  98 |  3.682550 |  3.4920553 | Set 1 | -0.2866708 | -0.6385001 |\n",
       "| mei-38 | mei-38  | 1 | 151 |  5.064855 |  5.0444964 | Set 1 | -0.2551010 | -0.6633834 |\n",
       "| CG31957 | CG31957 | 1 |  74 |  3.754599 |  3.6069260 | Set 1 | -0.2951455 | -0.6638777 |\n",
       "| MED10 | MED10   | 1 | 127 |  4.642784 |  4.5887821 | Set 1 | -0.2675437 | -0.6678980 |\n",
       "| CG9394 | CG9394  | 1 |  39 |  1.942762 |  1.6230231 | Set 1 | -0.4331211 | -0.6744436 |\n",
       "| CG12253 | CG12253 | 1 | 107 |  4.383542 |  4.3663049 | Set 1 | -0.2918318 | -0.7087298 |\n",
       "| CG12713 | CG12713 | 1 |  87 |  3.468104 |  3.3753788 | Set 1 | -0.3352608 | -0.7240503 |\n",
       "| CG31223 | CG31223 | 1 |  69 |  2.724157 |  2.5899814 | Set 1 | -0.3981664 | -0.7552874 |\n",
       "| TrpA1 | TrpA1   | 1 |  45 |  1.929738 |  1.7933341 | Set 1 | -0.5335830 | -0.8274654 |\n",
       "| Fign | Fign    | 1 |  65 |  2.506699 |  2.4681571 | Set 1 | -0.4683804 | -0.8478218 |\n",
       "\n"
      ],
      "text/plain": [
       "            gene        set.1.n.umi set.2.n.umi set.1.tpm   set.2.tpm \n",
       "tey         tey          967         1836        4611.02394   78.52727\n",
       "HLH54F      HLH54F       251          336        1223.38930   13.18234\n",
       "beat-IIb    beat-IIb     641         2319        3062.79663  105.41408\n",
       "beat-IIa    beat-IIa     423         1642        2054.21747   72.15760\n",
       "bru3        bru3        3545        41521       16857.46794 1860.61444\n",
       "tok         tok          234         1776        1139.77013   78.64160\n",
       "H2.0        H2.0          87          543         409.42302   22.36169\n",
       "ATP8B       ATP8B        134         1051         644.78794   47.51049\n",
       "RapGAP1     RapGAP1     2368        44286       11055.15800 1997.87862\n",
       "Syn2        Syn2          64          472         322.69397   20.38014\n",
       "kon         kon          393         5765        1834.81104  245.10225\n",
       "Doc2        Doc2         155         1798         708.59679   76.67323\n",
       "Egfr        Egfr         767        14291        3622.63139  632.11507\n",
       "Tet         Tet         2013        48636        9405.50468 1975.91807\n",
       "AdamTS-B    AdamTS-B     103         1253         494.50129   52.04845\n",
       "kirre       kirre        761        15152        3680.58968  705.14779\n",
       "DIP-epsilon DIP-epsilon   51          480         247.84092   21.77440\n",
       "stumps      stumps       317         5785        1490.06308  248.36320\n",
       "vn          vn           445         8801        2005.65104  379.30314\n",
       "mew         mew          874        20605        4124.99469  931.71114\n",
       "dpr17       dpr17         33          339         169.37357   15.15537\n",
       "Grip        Grip         132         2367         650.40301  101.02941\n",
       "bin         bin           36          404         170.85063   16.12752\n",
       "htl         htl          110         1863         534.90162   80.86442\n",
       "beta-Man    beta-Man      90         1460         453.56967   65.44308\n",
       "scrib       scrib        715        18508        3325.24371  825.78871\n",
       "rols        rols         520        12155        2369.70052  562.39810\n",
       "nvy         nvy           73         1330         344.88258   52.08342\n",
       "CG5002      CG5002        36          531         158.64059   18.40212\n",
       "CG9650      CG9650       566        16454        2712.18117  709.37883\n",
       "spri        spri         582        17210        2858.17103  776.55356\n",
       "aos         aos          215         5278        1043.73123  234.52877\n",
       "pnt         pnt          674        20606        3097.55987  866.19249\n",
       "be          be           261         7125        1280.34054  312.47351\n",
       "Amnionless  Amnionless    19          250         104.52194   11.09625\n",
       "Sema1a      Sema1a      1000        32060        4655.05666 1415.98805\n",
       "CadN        CadN        1062        33747        4894.53987 1500.31851\n",
       "apt         apt          355        10277        1704.98316  441.72788\n",
       "Doc1        Doc1          54          993         250.73715   41.97202\n",
       "cdi         cdi          500        15741        2400.42520  695.09258\n",
       "Mef2        Mef2         414        13368        1932.75822  543.83147\n",
       "sca         sca          173         5333         903.03694  220.66083\n",
       "TyrR        TyrR          28          490         154.63399   22.40983\n",
       "Btk29A      Btk29A       789        27675        3668.98799 1161.77642\n",
       "tacc        tacc         333        10326        1606.60476  447.29517\n",
       "CG11357     CG11357       93         2223         446.74338   96.58461\n",
       "jvl         jvl          637        20401        2966.77954  928.61076\n",
       "MESR3       MESR3        154         3916         717.09971  175.75586\n",
       "zfh1        zfh1         357        12405        1704.80724  514.78198\n",
       "CG13722     CG13722       14          228          70.06575    7.87755\n",
       "⋮           ⋮           ⋮           ⋮           ⋮           ⋮         \n",
       "CG13287     CG13287     3           164          6.440554    6.0558456\n",
       "CG15877     CG15877     2           169          6.361384    5.9767784\n",
       "mrn         mrn         2           326         11.161402   11.0410891\n",
       "CG5056      CG5056      2           238          8.691202    8.4534143\n",
       "CG18428     CG18428     2           281         12.701564   12.6914699\n",
       "Ugt49C1     Ugt49C1     2           103          4.899619    4.4748316\n",
       "CG14490     CG14490     1            35          2.284545    1.7016100\n",
       "CG42489     CG42489     1            53          2.570113    2.0078865\n",
       "CG32536     CG32536     1           139          6.826544    6.5250981\n",
       "Atac1       Atac1       1           123          4.642784    4.2187544\n",
       "CG15784     CG15784     2           309         11.515378   11.4954447\n",
       "Ssu72       Ssu72       3           243          9.016918    8.8659394\n",
       "GstE9       GstE9       1           108          3.510398    3.0318245\n",
       "CG8908      CG8908      1            90          3.997953    3.5584482\n",
       "CG4409      CG4409      1            70          3.517547    3.0553674\n",
       "CG3594      CG3594      3           247          9.113292    9.0043894\n",
       "Cpr73D      Cpr73D      1            49          2.686028    2.1700670\n",
       "ItgaPS4     ItgaPS4     1            18          1.315397    0.7098502\n",
       "CG12268     CG12268     1           138          5.482907    5.1780722\n",
       "CG13151     CG13151     1           151          5.955571    5.7031899\n",
       "CG1946      CG1946      1           165          6.257666    6.0370289\n",
       "cer         cer         1            87          4.439886    4.0992364\n",
       "CG44385     CG44385     1            78          3.674714    3.2923932\n",
       "CG33230     CG33230     1           210          7.287408    7.1866026\n",
       "Zmynd10     Zmynd10     1           104          4.824346    4.5523527\n",
       "CG33494     CG33494     1           106          5.124972    4.9003573\n",
       "CG32182     CG32182     1            82          3.327776    2.9680063\n",
       "Rsph1       Rsph1       1           141          6.038866    5.8918577\n",
       "Tap42       Tap42       2           194          7.422430    7.3878435\n",
       "CG14977     CG14977     1           101          4.141764    3.8581433\n",
       "CG5359      CG5359      3           149          6.015768    5.8797037\n",
       "fzo         fzo         1            82          4.102413    3.8254926\n",
       "CG9903      CG9903      1           143          6.373121    6.2829359\n",
       "CG12105     CG12105     1           156          5.330604    5.1732895\n",
       "CG11454     CG11454     1           104          3.925263    3.6663117\n",
       "PCNA2       PCNA2       1           137          5.363713    5.2315727\n",
       "CG10208     CG10208     2           141          5.448315    5.3291836\n",
       "CG17364     CG17364     1           103          4.655298    4.5142193\n",
       "CG6985      CG6985      1           155          5.815204    5.7863254\n",
       "Vsx2        Vsx2        1           110          4.906579    4.8082809\n",
       "CG8191      CG8191      1            98          3.682550    3.4920553\n",
       "mei-38      mei-38      1           151          5.064855    5.0444964\n",
       "CG31957     CG31957     1            74          3.754599    3.6069260\n",
       "MED10       MED10       1           127          4.642784    4.5887821\n",
       "CG9394      CG9394      1            39          1.942762    1.6230231\n",
       "CG12253     CG12253     1           107          4.383542    4.3663049\n",
       "CG12713     CG12713     1            87          3.468104    3.3753788\n",
       "CG31223     CG31223     1            69          2.724157    2.5899814\n",
       "TrpA1       TrpA1       1            45          1.929738    1.7933341\n",
       "Fign        Fign        1            65          2.506699    2.4681571\n",
       "            higher.expr log2.ratio heuristic.score\n",
       "tey         Set 1       5.857494   71.29264       \n",
       "HLH54F      Set 1       6.430644   65.96456       \n",
       "beat-IIb    Set 1       4.847089   56.13464       \n",
       "beat-IIa    Set 1       4.811437   52.95023       \n",
       "bru3        Set 1       3.178762   44.63358       \n",
       "tok         Set 1       3.839077   38.98887       \n",
       "H2.0        Set 1       4.131375   35.86434       \n",
       "ATP8B       Set 1       3.732456   34.84217       \n",
       "RapGAP1     Set 1       2.467457   33.14427       \n",
       "Syn2        Set 1       3.915824   32.65204       \n",
       "kon         Set 1       2.898302   31.42397       \n",
       "Doc2        Set 1       3.189476   30.20706       \n",
       "Egfr        Set 1       2.516498   29.75312       \n",
       "Tet         Set 1       2.250252   29.70208       \n",
       "AdamTS-B    Set 1       3.220592   28.83314       \n",
       "kirre       Set 1       2.381895   28.21620       \n",
       "DIP-epsilon Set 1       3.443929   27.41051       \n",
       "stumps      Set 1       2.579053   27.18870       \n",
       "vn          Set 1       2.398849   26.31675       \n",
       "mew         Set 1       2.144890   25.76126       \n",
       "dpr17       Set 1       3.390123   25.12949       \n",
       "Grip        Set 1       2.672349   24.97953       \n",
       "bin         Set 1       3.318347   24.63877       \n",
       "htl         Set 1       2.707965   24.54994       \n",
       "beta-Man    Set 1       2.771133   24.46456       \n",
       "scrib       Set 1       2.007869   23.49143       \n",
       "rols        Set 1       2.072478   23.23475       \n",
       "nvy         Set 1       2.699772   22.77025       \n",
       "CG5002      Set 1       3.031476   22.18641       \n",
       "CG9650      Set 1       1.932793   22.04499       \n",
       "spri        Set 1       1.878078   21.56293       \n",
       "aos         Set 1       2.147775   21.53986       \n",
       "pnt         Set 1       1.836708   21.30100       \n",
       "be          Set 1       2.030112   20.95774       \n",
       "Amnionless  Set 1       3.111174   20.91145       \n",
       "Sema1a      Set 1       1.715971   20.90892       \n",
       "CadN        Set 1       1.704943   20.89792       \n",
       "apt         Set 1       1.945265   20.88512       \n",
       "Doc1        Set 1       2.544706   20.29600       \n",
       "cdi         Set 1       1.785939   20.05551       \n",
       "Mef2        Set 1       1.826779   19.94330       \n",
       "sca         Set 1       2.026431   19.90004       \n",
       "TyrR        Set 1       2.723671   19.83381       \n",
       "Btk29A      Set 1       1.657808   19.63104       \n",
       "tacc        Set 1       1.841494   19.61320       \n",
       "CG11357     Set 1       2.194721   19.32787       \n",
       "jvl         Set 1       1.674199   19.31217       \n",
       "MESR3       Set 1       2.020416   19.16979       \n",
       "zfh1        Set 1       1.724775   18.51760       \n",
       "CG13722     Set 1       2.980476   18.33315       \n",
       "⋮           ⋮           ⋮          ⋮              \n",
       "CG13287     Set 1       -0.1316342 -0.3811350     \n",
       "CG15877     Set 1       -0.1332204 -0.3836716     \n",
       "mrn         Set 1       -0.1094476 -0.3944752     \n",
       "CG5056      Set 1       -0.1212797 -0.3973943     \n",
       "CG18428     Set 1       -0.1082712 -0.4088611     \n",
       "Ugt49C1     Set 1       -0.1601450 -0.4100708     \n",
       "CG14490     Set 1       -0.2419128 -0.4150482     \n",
       "CG42489     Set 1       -0.2269185 -0.4166155     \n",
       "CG32536     Set 1       -0.1405549 -0.4172198     \n",
       "Atac1       Set 1       -0.1687153 -0.4211820     \n",
       "CG15784     Set 1       -0.1178405 -0.4296028     \n",
       "Ssu72       Set 1       -0.1298221 -0.4315762     \n",
       "GstE9       Set 1       -0.1997983 -0.4342126     \n",
       "CG8908      Set 1       -0.1892812 -0.4393856     \n",
       "CG4409      Set 1       -0.2052628 -0.4465573     \n",
       "CG3594      Set 1       -0.1345889 -0.4492821     \n",
       "Cpr73D      Set 1       -0.2390392 -0.4498878     \n",
       "ItgaPS4     Set 1       -0.3783721 -0.4583068     \n",
       "CG12268     Set 1       -0.1722157 -0.4644039     \n",
       "CG13151     Set 1       -0.1706079 -0.4773896     \n",
       "CG1946      Set 1       -0.1693419 -0.4842340     \n",
       "cer         Set 1       -0.1997585 -0.4881252     \n",
       "CG44385     Set 1       -0.2241501 -0.4987068     \n",
       "CG33230     Set 1       -0.1678591 -0.5121248     \n",
       "Zmynd10     Set 1       -0.2027660 -0.5154505     \n",
       "CG33494     Set 1       -0.2032581 -0.5314596     \n",
       "CG32182     Set 1       -0.2538560 -0.5365567     \n",
       "Rsph1       Set 1       -0.1906152 -0.5366472     \n",
       "Tap42       Set 1       -0.1764084 -0.5423211     \n",
       "CG14977     Set 1       -0.2301596 -0.5436976     \n",
       "CG5359      Set 1       -0.1935974 -0.5441251     \n",
       "fzo         Set 1       -0.2342036 -0.5506548     \n",
       "CG9903      Set 1       -0.1925201 -0.5548961     \n",
       "CG12105     Set 1       -0.2117404 -0.5637257     \n",
       "CG11454     Set 1       -0.2494934 -0.5738848     \n",
       "PCNA2       Set 1       -0.2163641 -0.5776639     \n",
       "CG10208     Set 1       -0.2162094 -0.5813703     \n",
       "CG17364     Set 1       -0.2442830 -0.6106106     \n",
       "CG6985      Set 1       -0.2228007 -0.6168810     \n",
       "Vsx2        Set 1       -0.2433938 -0.6236535     \n",
       "CG8191      Set 1       -0.2866708 -0.6385001     \n",
       "mei-38      Set 1       -0.2551010 -0.6633834     \n",
       "CG31957     Set 1       -0.2951455 -0.6638777     \n",
       "MED10       Set 1       -0.2675437 -0.6678980     \n",
       "CG9394      Set 1       -0.4331211 -0.6744436     \n",
       "CG12253     Set 1       -0.2918318 -0.7087298     \n",
       "CG12713     Set 1       -0.3352608 -0.7240503     \n",
       "CG31223     Set 1       -0.3981664 -0.7552874     \n",
       "TrpA1       Set 1       -0.5335830 -0.8274654     \n",
       "Fign        Set 1       -0.4683804 -0.8478218     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "20"
      ],
      "text/latex": [
       "20"
      ],
      "text/markdown": [
       "20"
      ],
      "text/plain": [
       "[1] 20"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "two.set.differential.gene.test.3a(cds_sub, \n",
    "                                  df_sub300$cluster == 75 & \n",
    "                                  df_sub300$umap_2 < -0.1*df_sub300$umap_1 -0.1 , \n",
    "                                 !(df_sub300$cluster == 75 & \n",
    "                                  df_sub300$umap_2 < -0.1*df_sub300$umap_1 -0.1),\n",
    "                                  sample = 50000\n",
    "                                )  %>% filter(higher.expr == \"Set 1\")  %>% \n",
    "head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1663,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 916\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>sns</th><td>sns    </td><td>1882</td><td> 5610</td><td> 6312.26540</td><td> 240.7767416</td><td>Set 1</td><td>4.706410</td><td>59.41453</td></tr>\n",
       "\t<tr><th scope=row>NK7.1</th><td>NK7.1  </td><td>1562</td><td> 8219</td><td> 5208.74440</td><td> 378.4342898</td><td>Set 1</td><td>3.779014</td><td>46.65947</td></tr>\n",
       "\t<tr><th scope=row>CG34377</th><td>CG34377</td><td> 169</td><td>  394</td><td>  547.81018</td><td>  14.8422673</td><td>Set 1</td><td>5.111825</td><td>46.51845</td></tr>\n",
       "\t<tr><th scope=row>Kah</th><td>Kah    </td><td> 502</td><td> 2284</td><td> 1727.39389</td><td>  97.2727810</td><td>Set 1</td><td>4.135661</td><td>44.47993</td></tr>\n",
       "\t<tr><th scope=row>gol</th><td>gol    </td><td> 561</td><td> 2645</td><td> 1850.24954</td><td> 113.1379020</td><td>Set 1</td><td>4.018870</td><td>43.62196</td></tr>\n",
       "\t<tr><th scope=row>Vrp1</th><td>Vrp1   </td><td> 789</td><td> 5064</td><td> 2640.68498</td><td> 199.4421657</td><td>Set 1</td><td>3.719654</td><td>42.28221</td></tr>\n",
       "\t<tr><th scope=row>MCU</th><td>MCU    </td><td> 386</td><td> 1974</td><td> 1334.45571</td><td>  85.1868301</td><td>Set 1</td><td>3.952640</td><td>41.04072</td></tr>\n",
       "\t<tr><th scope=row>lmd</th><td>lmd    </td><td> 502</td><td> 3225</td><td> 1692.77080</td><td> 140.8134240</td><td>Set 1</td><td>3.577321</td><td>38.37042</td></tr>\n",
       "\t<tr><th scope=row>HEATR2</th><td>HEATR2 </td><td> 171</td><td>  774</td><td>  562.66661</td><td>  31.7782450</td><td>Set 1</td><td>4.101470</td><td>37.48210</td></tr>\n",
       "\t<tr><th scope=row>Adgf-A2</th><td>Adgf-A2</td><td> 248</td><td> 1243</td><td>  806.74703</td><td>  54.2976110</td><td>Set 1</td><td>3.866827</td><td>37.34489</td></tr>\n",
       "\t<tr><th scope=row>Mef2</th><td>Mef2   </td><td>1324</td><td>13213</td><td> 4379.27203</td><td> 525.7576492</td><td>Set 1</td><td>3.055480</td><td>36.96154</td></tr>\n",
       "\t<tr><th scope=row>CG9674</th><td>CG9674 </td><td> 695</td><td> 5609</td><td> 2274.88087</td><td> 231.4911083</td><td>Set 1</td><td>3.290544</td><td>36.69683</td></tr>\n",
       "\t<tr><th scope=row>ewg</th><td>ewg    </td><td>1151</td><td>11225</td><td> 3841.10357</td><td> 489.1734084</td><td>Set 1</td><td>2.970157</td><td>35.36768</td></tr>\n",
       "\t<tr><th scope=row>net</th><td>net    </td><td> 124</td><td>  590</td><td>  418.72934</td><td>  24.6629514</td><td>Set 1</td><td>4.028259</td><td>35.09949</td></tr>\n",
       "\t<tr><th scope=row>mbl</th><td>mbl    </td><td>4865</td><td>67442</td><td>16512.17538</td><td>3086.5381043</td><td>Set 1</td><td>2.419001</td><td>33.89343</td></tr>\n",
       "\t<tr><th scope=row>Rim2</th><td>Rim2   </td><td> 498</td><td> 4892</td><td> 1674.00816</td><td> 190.9193175</td><td>Set 1</td><td>3.124735</td><td>33.46576</td></tr>\n",
       "\t<tr><th scope=row>rdgA</th><td>rdgA   </td><td>1020</td><td>10688</td><td> 3461.59766</td><td> 485.1933696</td><td>Set 1</td><td>2.831836</td><td>33.29570</td></tr>\n",
       "\t<tr><th scope=row>rst</th><td>rst    </td><td>1333</td><td>16200</td><td> 4546.36918</td><td> 711.9156535</td><td>Set 1</td><td>2.672912</td><td>32.47806</td></tr>\n",
       "\t<tr><th scope=row>CG18109</th><td>CG18109</td><td>  18</td><td>   16</td><td>   66.14784</td><td>   0.6729098</td><td>Set 1</td><td>5.305262</td><td>32.19907</td></tr>\n",
       "\t<tr><th scope=row>sug</th><td>sug    </td><td> 131</td><td>  902</td><td>  458.34580</td><td>  38.7750360</td><td>Set 1</td><td>3.526501</td><td>31.18639</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tsns & sns     & 1882 &  5610 &  6312.26540 &  240.7767416 & Set 1 & 4.706410 & 59.41453\\\\\n",
       "\tNK7.1 & NK7.1   & 1562 &  8219 &  5208.74440 &  378.4342898 & Set 1 & 3.779014 & 46.65947\\\\\n",
       "\tCG34377 & CG34377 &  169 &   394 &   547.81018 &   14.8422673 & Set 1 & 5.111825 & 46.51845\\\\\n",
       "\tKah & Kah     &  502 &  2284 &  1727.39389 &   97.2727810 & Set 1 & 4.135661 & 44.47993\\\\\n",
       "\tgol & gol     &  561 &  2645 &  1850.24954 &  113.1379020 & Set 1 & 4.018870 & 43.62196\\\\\n",
       "\tVrp1 & Vrp1    &  789 &  5064 &  2640.68498 &  199.4421657 & Set 1 & 3.719654 & 42.28221\\\\\n",
       "\tMCU & MCU     &  386 &  1974 &  1334.45571 &   85.1868301 & Set 1 & 3.952640 & 41.04072\\\\\n",
       "\tlmd & lmd     &  502 &  3225 &  1692.77080 &  140.8134240 & Set 1 & 3.577321 & 38.37042\\\\\n",
       "\tHEATR2 & HEATR2  &  171 &   774 &   562.66661 &   31.7782450 & Set 1 & 4.101470 & 37.48210\\\\\n",
       "\tAdgf-A2 & Adgf-A2 &  248 &  1243 &   806.74703 &   54.2976110 & Set 1 & 3.866827 & 37.34489\\\\\n",
       "\tMef2 & Mef2    & 1324 & 13213 &  4379.27203 &  525.7576492 & Set 1 & 3.055480 & 36.96154\\\\\n",
       "\tCG9674 & CG9674  &  695 &  5609 &  2274.88087 &  231.4911083 & Set 1 & 3.290544 & 36.69683\\\\\n",
       "\tewg & ewg     & 1151 & 11225 &  3841.10357 &  489.1734084 & Set 1 & 2.970157 & 35.36768\\\\\n",
       "\tnet & net     &  124 &   590 &   418.72934 &   24.6629514 & Set 1 & 4.028259 & 35.09949\\\\\n",
       "\tmbl & mbl     & 4865 & 67442 & 16512.17538 & 3086.5381043 & Set 1 & 2.419001 & 33.89343\\\\\n",
       "\tRim2 & Rim2    &  498 &  4892 &  1674.00816 &  190.9193175 & Set 1 & 3.124735 & 33.46576\\\\\n",
       "\trdgA & rdgA    & 1020 & 10688 &  3461.59766 &  485.1933696 & Set 1 & 2.831836 & 33.29570\\\\\n",
       "\trst & rst     & 1333 & 16200 &  4546.36918 &  711.9156535 & Set 1 & 2.672912 & 32.47806\\\\\n",
       "\tCG18109 & CG18109 &   18 &    16 &    66.14784 &    0.6729098 & Set 1 & 5.305262 & 32.19907\\\\\n",
       "\tsug & sug     &  131 &   902 &   458.34580 &   38.7750360 & Set 1 & 3.526501 & 31.18639\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| sns | sns     | 1882 |  5610 |  6312.26540 |  240.7767416 | Set 1 | 4.706410 | 59.41453 |\n",
       "| NK7.1 | NK7.1   | 1562 |  8219 |  5208.74440 |  378.4342898 | Set 1 | 3.779014 | 46.65947 |\n",
       "| CG34377 | CG34377 |  169 |   394 |   547.81018 |   14.8422673 | Set 1 | 5.111825 | 46.51845 |\n",
       "| Kah | Kah     |  502 |  2284 |  1727.39389 |   97.2727810 | Set 1 | 4.135661 | 44.47993 |\n",
       "| gol | gol     |  561 |  2645 |  1850.24954 |  113.1379020 | Set 1 | 4.018870 | 43.62196 |\n",
       "| Vrp1 | Vrp1    |  789 |  5064 |  2640.68498 |  199.4421657 | Set 1 | 3.719654 | 42.28221 |\n",
       "| MCU | MCU     |  386 |  1974 |  1334.45571 |   85.1868301 | Set 1 | 3.952640 | 41.04072 |\n",
       "| lmd | lmd     |  502 |  3225 |  1692.77080 |  140.8134240 | Set 1 | 3.577321 | 38.37042 |\n",
       "| HEATR2 | HEATR2  |  171 |   774 |   562.66661 |   31.7782450 | Set 1 | 4.101470 | 37.48210 |\n",
       "| Adgf-A2 | Adgf-A2 |  248 |  1243 |   806.74703 |   54.2976110 | Set 1 | 3.866827 | 37.34489 |\n",
       "| Mef2 | Mef2    | 1324 | 13213 |  4379.27203 |  525.7576492 | Set 1 | 3.055480 | 36.96154 |\n",
       "| CG9674 | CG9674  |  695 |  5609 |  2274.88087 |  231.4911083 | Set 1 | 3.290544 | 36.69683 |\n",
       "| ewg | ewg     | 1151 | 11225 |  3841.10357 |  489.1734084 | Set 1 | 2.970157 | 35.36768 |\n",
       "| net | net     |  124 |   590 |   418.72934 |   24.6629514 | Set 1 | 4.028259 | 35.09949 |\n",
       "| mbl | mbl     | 4865 | 67442 | 16512.17538 | 3086.5381043 | Set 1 | 2.419001 | 33.89343 |\n",
       "| Rim2 | Rim2    |  498 |  4892 |  1674.00816 |  190.9193175 | Set 1 | 3.124735 | 33.46576 |\n",
       "| rdgA | rdgA    | 1020 | 10688 |  3461.59766 |  485.1933696 | Set 1 | 2.831836 | 33.29570 |\n",
       "| rst | rst     | 1333 | 16200 |  4546.36918 |  711.9156535 | Set 1 | 2.672912 | 32.47806 |\n",
       "| CG18109 | CG18109 |   18 |    16 |    66.14784 |    0.6729098 | Set 1 | 5.305262 | 32.19907 |\n",
       "| sug | sug     |  131 |   902 |   458.34580 |   38.7750360 | Set 1 | 3.526501 | 31.18639 |\n",
       "\n"
      ],
      "text/plain": [
       "        gene    set.1.n.umi set.2.n.umi set.1.tpm   set.2.tpm    higher.expr\n",
       "sns     sns     1882         5610        6312.26540  240.7767416 Set 1      \n",
       "NK7.1   NK7.1   1562         8219        5208.74440  378.4342898 Set 1      \n",
       "CG34377 CG34377  169          394         547.81018   14.8422673 Set 1      \n",
       "Kah     Kah      502         2284        1727.39389   97.2727810 Set 1      \n",
       "gol     gol      561         2645        1850.24954  113.1379020 Set 1      \n",
       "Vrp1    Vrp1     789         5064        2640.68498  199.4421657 Set 1      \n",
       "MCU     MCU      386         1974        1334.45571   85.1868301 Set 1      \n",
       "lmd     lmd      502         3225        1692.77080  140.8134240 Set 1      \n",
       "HEATR2  HEATR2   171          774         562.66661   31.7782450 Set 1      \n",
       "Adgf-A2 Adgf-A2  248         1243         806.74703   54.2976110 Set 1      \n",
       "Mef2    Mef2    1324        13213        4379.27203  525.7576492 Set 1      \n",
       "CG9674  CG9674   695         5609        2274.88087  231.4911083 Set 1      \n",
       "ewg     ewg     1151        11225        3841.10357  489.1734084 Set 1      \n",
       "net     net      124          590         418.72934   24.6629514 Set 1      \n",
       "mbl     mbl     4865        67442       16512.17538 3086.5381043 Set 1      \n",
       "Rim2    Rim2     498         4892        1674.00816  190.9193175 Set 1      \n",
       "rdgA    rdgA    1020        10688        3461.59766  485.1933696 Set 1      \n",
       "rst     rst     1333        16200        4546.36918  711.9156535 Set 1      \n",
       "CG18109 CG18109   18           16          66.14784    0.6729098 Set 1      \n",
       "sug     sug      131          902         458.34580   38.7750360 Set 1      \n",
       "        log2.ratio heuristic.score\n",
       "sns     4.706410   59.41453       \n",
       "NK7.1   3.779014   46.65947       \n",
       "CG34377 5.111825   46.51845       \n",
       "Kah     4.135661   44.47993       \n",
       "gol     4.018870   43.62196       \n",
       "Vrp1    3.719654   42.28221       \n",
       "MCU     3.952640   41.04072       \n",
       "lmd     3.577321   38.37042       \n",
       "HEATR2  4.101470   37.48210       \n",
       "Adgf-A2 3.866827   37.34489       \n",
       "Mef2    3.055480   36.96154       \n",
       "CG9674  3.290544   36.69683       \n",
       "ewg     2.970157   35.36768       \n",
       "net     4.028259   35.09949       \n",
       "mbl     2.419001   33.89343       \n",
       "Rim2    3.124735   33.46576       \n",
       "rdgA    2.831836   33.29570       \n",
       "rst     2.672912   32.47806       \n",
       "CG18109 5.305262   32.19907       \n",
       "sug     3.526501   31.18639       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# do the other part of 75 (bc)\n",
    "two.set.differential.gene.test.3a(cds_sub, \n",
    "                                  df_sub300$cluster == 75 & \n",
    "                                  df_sub300$umap_2 >= -0.1*df_sub300$umap_1 -0.1 , \n",
    "                                 !(df_sub300$cluster == 75 & \n",
    "                                  df_sub300$umap_2 >= -0.1*df_sub300$umap_1 -0.1),\n",
    "                                  sample = 50000\n",
    "                                )  %>% filter(higher.expr == \"Set 1\")  %>% \n",
    "head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 260,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"sns\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"sns\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"sns\"), \n",
    "                   size_pos = 2, size_neg = 0.5, stroke = 0.008, limit = 2)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"gol\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 2)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Kah\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1535,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type unknown_3\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"unknown_3\", df_sub300$cluster %in% bd &\n",
    "                         df_sub300$umap_2 < 0*df_sub300$umap_1 + 2.1 )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bd' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bd' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bd)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bd' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bd' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bd)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1521,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Desi31</th><td>bd</td><td>Desi         </td><td>  95</td><td>    68</td><td>  266.87731</td><td>2.539683e+00</td><td>6.236413</td><td>50.29934</td></tr>\n",
       "\t<tr><th scope=row>CG1372731</th><td>bd</td><td>CG13727      </td><td>  54</td><td>    36</td><td>  137.35042</td><td>1.072172e+00</td><td>6.050574</td><td>43.03279</td></tr>\n",
       "\t<tr><th scope=row>CG1331031</th><td>bd</td><td>CG13310      </td><td>  20</td><td>     3</td><td>   56.93845</td><td>6.838945e-02</td><td>5.735894</td><td>33.59197</td></tr>\n",
       "\t<tr><th scope=row>atk31</th><td>bd</td><td>atk          </td><td>  30</td><td>    96</td><td>  105.84229</td><td>4.012126e+00</td><td>4.400350</td><td>29.65545</td></tr>\n",
       "\t<tr><th scope=row>CG3204031</th><td>bd</td><td>CG32040      </td><td>  27</td><td>    55</td><td>   82.46591</td><td>2.383886e+00</td><td>4.607045</td><td>29.40730</td></tr>\n",
       "\t<tr><th scope=row>CG4507231</th><td>bd</td><td>CG45072      </td><td>  12</td><td>     4</td><td>   34.62260</td><td>2.417838e-01</td><td>4.801228</td><td>24.74899</td></tr>\n",
       "\t<tr><th scope=row>pAbp31</th><td>bd</td><td>pAbp         </td><td>2969</td><td>111562</td><td>10308.58419</td><td>3.401275e+03</td><td>1.599275</td><td>21.32105</td></tr>\n",
       "\t<tr><th scope=row>CG3446131</th><td>bd</td><td>CG34461      </td><td> 106</td><td>  1339</td><td>  262.29102</td><td>4.705980e+01</td><td>2.448266</td><td>19.68532</td></tr>\n",
       "\t<tr><th scope=row>Ccp84Ab31</th><td>bd</td><td>Ccp84Ab      </td><td>  17</td><td>   135</td><td>   54.60077</td><td>4.264440e+00</td><td>3.374569</td><td>19.56249</td></tr>\n",
       "\t<tr><th scope=row>CG1338631</th><td>bd</td><td>CG13386      </td><td>   9</td><td>     8</td><td>   24.01867</td><td>3.963017e-01</td><td>4.104474</td><td>19.06501</td></tr>\n",
       "\t<tr><th scope=row>CG296231</th><td>bd</td><td>CG2962       </td><td> 967</td><td> 30922</td><td> 3129.26935</td><td>1.094443e+03</td><td>1.514311</td><td>17.58429</td></tr>\n",
       "\t<tr><th scope=row>fne31</th><td>bd</td><td>fne          </td><td> 508</td><td> 17447</td><td> 1775.60564</td><td>6.067701e+02</td><td>1.546714</td><td>16.69663</td></tr>\n",
       "\t<tr><th scope=row>Ccp84Ac31</th><td>bd</td><td>Ccp84Ac      </td><td>  71</td><td>  1274</td><td>  197.19076</td><td>4.532648e+01</td><td>2.089683</td><td>15.94584</td></tr>\n",
       "\t<tr><th scope=row>etaTry31</th><td>bd</td><td>etaTry       </td><td>   4</td><td>    17</td><td>   19.20897</td><td>6.404629e-01</td><td>3.549606</td><td>15.39437</td></tr>\n",
       "\t<tr><th scope=row>Act57B31</th><td>bd</td><td>Act57B       </td><td>1187</td><td> 54486</td><td> 3966.23050</td><td>1.629691e+03</td><td>1.282285</td><td>15.32833</td></tr>\n",
       "\t<tr><th scope=row>Cpr31A31</th><td>bd</td><td>Cpr31A       </td><td>  13</td><td>   201</td><td>   48.22923</td><td>6.369060e+00</td><td>2.710355</td><td>15.23611</td></tr>\n",
       "\t<tr><th scope=row>Ccp84Af31</th><td>bd</td><td>Ccp84Af      </td><td>   8</td><td>    77</td><td>   29.34640</td><td>2.462360e+00</td><td>3.083356</td><td>15.18076</td></tr>\n",
       "\t<tr><th scope=row>Neurochondrin31</th><td>bd</td><td>Neurochondrin</td><td> 154</td><td>  5112</td><td>  535.53395</td><td>1.694069e+02</td><td>1.651994</td><td>14.97950</td></tr>\n",
       "\t<tr><th scope=row>fax31</th><td>bd</td><td>fax          </td><td>1746</td><td> 81545</td><td> 6148.00238</td><td>2.700803e+03</td><td>1.186195</td><td>14.92961</td></tr>\n",
       "\t<tr><th scope=row>Vajk131</th><td>bd</td><td>Vajk1        </td><td> 124</td><td>  3004</td><td>  382.11803</td><td>1.134906e+02</td><td>1.738789</td><td>14.92167</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tDesi31 & bd & Desi          &   95 &     68 &   266.87731 & 2.539683e+00 & 6.236413 & 50.29934\\\\\n",
       "\tCG1372731 & bd & CG13727       &   54 &     36 &   137.35042 & 1.072172e+00 & 6.050574 & 43.03279\\\\\n",
       "\tCG1331031 & bd & CG13310       &   20 &      3 &    56.93845 & 6.838945e-02 & 5.735894 & 33.59197\\\\\n",
       "\tatk31 & bd & atk           &   30 &     96 &   105.84229 & 4.012126e+00 & 4.400350 & 29.65545\\\\\n",
       "\tCG3204031 & bd & CG32040       &   27 &     55 &    82.46591 & 2.383886e+00 & 4.607045 & 29.40730\\\\\n",
       "\tCG4507231 & bd & CG45072       &   12 &      4 &    34.62260 & 2.417838e-01 & 4.801228 & 24.74899\\\\\n",
       "\tpAbp31 & bd & pAbp          & 2969 & 111562 & 10308.58419 & 3.401275e+03 & 1.599275 & 21.32105\\\\\n",
       "\tCG3446131 & bd & CG34461       &  106 &   1339 &   262.29102 & 4.705980e+01 & 2.448266 & 19.68532\\\\\n",
       "\tCcp84Ab31 & bd & Ccp84Ab       &   17 &    135 &    54.60077 & 4.264440e+00 & 3.374569 & 19.56249\\\\\n",
       "\tCG1338631 & bd & CG13386       &    9 &      8 &    24.01867 & 3.963017e-01 & 4.104474 & 19.06501\\\\\n",
       "\tCG296231 & bd & CG2962        &  967 &  30922 &  3129.26935 & 1.094443e+03 & 1.514311 & 17.58429\\\\\n",
       "\tfne31 & bd & fne           &  508 &  17447 &  1775.60564 & 6.067701e+02 & 1.546714 & 16.69663\\\\\n",
       "\tCcp84Ac31 & bd & Ccp84Ac       &   71 &   1274 &   197.19076 & 4.532648e+01 & 2.089683 & 15.94584\\\\\n",
       "\tetaTry31 & bd & etaTry        &    4 &     17 &    19.20897 & 6.404629e-01 & 3.549606 & 15.39437\\\\\n",
       "\tAct57B31 & bd & Act57B        & 1187 &  54486 &  3966.23050 & 1.629691e+03 & 1.282285 & 15.32833\\\\\n",
       "\tCpr31A31 & bd & Cpr31A        &   13 &    201 &    48.22923 & 6.369060e+00 & 2.710355 & 15.23611\\\\\n",
       "\tCcp84Af31 & bd & Ccp84Af       &    8 &     77 &    29.34640 & 2.462360e+00 & 3.083356 & 15.18076\\\\\n",
       "\tNeurochondrin31 & bd & Neurochondrin &  154 &   5112 &   535.53395 & 1.694069e+02 & 1.651994 & 14.97950\\\\\n",
       "\tfax31 & bd & fax           & 1746 &  81545 &  6148.00238 & 2.700803e+03 & 1.186195 & 14.92961\\\\\n",
       "\tVajk131 & bd & Vajk1         &  124 &   3004 &   382.11803 & 1.134906e+02 & 1.738789 & 14.92167\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Desi31 | bd | Desi          |   95 |     68 |   266.87731 | 2.539683e+00 | 6.236413 | 50.29934 |\n",
       "| CG1372731 | bd | CG13727       |   54 |     36 |   137.35042 | 1.072172e+00 | 6.050574 | 43.03279 |\n",
       "| CG1331031 | bd | CG13310       |   20 |      3 |    56.93845 | 6.838945e-02 | 5.735894 | 33.59197 |\n",
       "| atk31 | bd | atk           |   30 |     96 |   105.84229 | 4.012126e+00 | 4.400350 | 29.65545 |\n",
       "| CG3204031 | bd | CG32040       |   27 |     55 |    82.46591 | 2.383886e+00 | 4.607045 | 29.40730 |\n",
       "| CG4507231 | bd | CG45072       |   12 |      4 |    34.62260 | 2.417838e-01 | 4.801228 | 24.74899 |\n",
       "| pAbp31 | bd | pAbp          | 2969 | 111562 | 10308.58419 | 3.401275e+03 | 1.599275 | 21.32105 |\n",
       "| CG3446131 | bd | CG34461       |  106 |   1339 |   262.29102 | 4.705980e+01 | 2.448266 | 19.68532 |\n",
       "| Ccp84Ab31 | bd | Ccp84Ab       |   17 |    135 |    54.60077 | 4.264440e+00 | 3.374569 | 19.56249 |\n",
       "| CG1338631 | bd | CG13386       |    9 |      8 |    24.01867 | 3.963017e-01 | 4.104474 | 19.06501 |\n",
       "| CG296231 | bd | CG2962        |  967 |  30922 |  3129.26935 | 1.094443e+03 | 1.514311 | 17.58429 |\n",
       "| fne31 | bd | fne           |  508 |  17447 |  1775.60564 | 6.067701e+02 | 1.546714 | 16.69663 |\n",
       "| Ccp84Ac31 | bd | Ccp84Ac       |   71 |   1274 |   197.19076 | 4.532648e+01 | 2.089683 | 15.94584 |\n",
       "| etaTry31 | bd | etaTry        |    4 |     17 |    19.20897 | 6.404629e-01 | 3.549606 | 15.39437 |\n",
       "| Act57B31 | bd | Act57B        | 1187 |  54486 |  3966.23050 | 1.629691e+03 | 1.282285 | 15.32833 |\n",
       "| Cpr31A31 | bd | Cpr31A        |   13 |    201 |    48.22923 | 6.369060e+00 | 2.710355 | 15.23611 |\n",
       "| Ccp84Af31 | bd | Ccp84Af       |    8 |     77 |    29.34640 | 2.462360e+00 | 3.083356 | 15.18076 |\n",
       "| Neurochondrin31 | bd | Neurochondrin |  154 |   5112 |   535.53395 | 1.694069e+02 | 1.651994 | 14.97950 |\n",
       "| fax31 | bd | fax           | 1746 |  81545 |  6148.00238 | 2.700803e+03 | 1.186195 | 14.92961 |\n",
       "| Vajk131 | bd | Vajk1         |  124 |   3004 |   382.11803 | 1.134906e+02 | 1.738789 | 14.92167 |\n",
       "\n"
      ],
      "text/plain": [
       "                cell.type.2 gene          cell.type.2.n.umi other.n.umi\n",
       "Desi31          bd          Desi            95                  68     \n",
       "CG1372731       bd          CG13727         54                  36     \n",
       "CG1331031       bd          CG13310         20                   3     \n",
       "atk31           bd          atk             30                  96     \n",
       "CG3204031       bd          CG32040         27                  55     \n",
       "CG4507231       bd          CG45072         12                   4     \n",
       "pAbp31          bd          pAbp          2969              111562     \n",
       "CG3446131       bd          CG34461        106                1339     \n",
       "Ccp84Ab31       bd          Ccp84Ab         17                 135     \n",
       "CG1338631       bd          CG13386          9                   8     \n",
       "CG296231        bd          CG2962         967               30922     \n",
       "fne31           bd          fne            508               17447     \n",
       "Ccp84Ac31       bd          Ccp84Ac         71                1274     \n",
       "etaTry31        bd          etaTry           4                  17     \n",
       "Act57B31        bd          Act57B        1187               54486     \n",
       "Cpr31A31        bd          Cpr31A          13                 201     \n",
       "Ccp84Af31       bd          Ccp84Af          8                  77     \n",
       "Neurochondrin31 bd          Neurochondrin  154                5112     \n",
       "fax31           bd          fax           1746               81545     \n",
       "Vajk131         bd          Vajk1          124                3004     \n",
       "                cell.type.2.tpm other.tpm    log2.ratio heuristic.score\n",
       "Desi31            266.87731     2.539683e+00 6.236413   50.29934       \n",
       "CG1372731         137.35042     1.072172e+00 6.050574   43.03279       \n",
       "CG1331031          56.93845     6.838945e-02 5.735894   33.59197       \n",
       "atk31             105.84229     4.012126e+00 4.400350   29.65545       \n",
       "CG3204031          82.46591     2.383886e+00 4.607045   29.40730       \n",
       "CG4507231          34.62260     2.417838e-01 4.801228   24.74899       \n",
       "pAbp31          10308.58419     3.401275e+03 1.599275   21.32105       \n",
       "CG3446131         262.29102     4.705980e+01 2.448266   19.68532       \n",
       "Ccp84Ab31          54.60077     4.264440e+00 3.374569   19.56249       \n",
       "CG1338631          24.01867     3.963017e-01 4.104474   19.06501       \n",
       "CG296231         3129.26935     1.094443e+03 1.514311   17.58429       \n",
       "fne31            1775.60564     6.067701e+02 1.546714   16.69663       \n",
       "Ccp84Ac31         197.19076     4.532648e+01 2.089683   15.94584       \n",
       "etaTry31           19.20897     6.404629e-01 3.549606   15.39437       \n",
       "Act57B31         3966.23050     1.629691e+03 1.282285   15.32833       \n",
       "Cpr31A31           48.22923     6.369060e+00 2.710355   15.23611       \n",
       "Ccp84Af31          29.34640     2.462360e+00 3.083356   15.18076       \n",
       "Neurochondrin31   535.53395     1.694069e+02 1.651994   14.97950       \n",
       "fax31            6148.00238     2.700803e+03 1.186195   14.92961       \n",
       "Vajk131           382.11803     1.134906e+02 1.738789   14.92167       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"bd\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 263,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Desi\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Desi\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Desi\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG13727\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG32040\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 264,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(133)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 < 0*umap_1 + 2.1), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0, intercept = 2.1) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(4, 8) + ylim(1, 5)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1534,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 180\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Desi</th><td>Desi      </td><td>118</td><td> 109</td><td>1462.15337</td><td>  4.5872114</td><td>Set 1</td><td>8.031751</td><td>84.45277</td></tr>\n",
       "\t<tr><th scope=row>CG13727</th><td>CG13727   </td><td> 64</td><td>  56</td><td> 713.38143</td><td>  2.4560272</td><td>Set 1</td><td>7.689415</td><td>72.89989</td></tr>\n",
       "\t<tr><th scope=row>CG13310</th><td>CG13310   </td><td> 23</td><td>  12</td><td> 262.28598</td><td>  0.3391824</td><td>Set 1</td><td>7.613644</td><td>61.21741</td></tr>\n",
       "\t<tr><th scope=row>CG32040</th><td>CG32040   </td><td> 32</td><td>  73</td><td> 430.07555</td><td>  3.2541828</td><td>Set 1</td><td>6.659564</td><td>58.28315</td></tr>\n",
       "\t<tr><th scope=row>CG45072</th><td>CG45072   </td><td> 14</td><td>  28</td><td> 177.20362</td><td>  1.2591345</td><td>Set 1</td><td>6.293494</td><td>47.05886</td></tr>\n",
       "\t<tr><th scope=row>atk</th><td>atk       </td><td> 18</td><td> 121</td><td> 275.12744</td><td>  4.8699808</td><td>Set 1</td><td>5.550600</td><td>45.01088</td></tr>\n",
       "\t<tr><th scope=row>CG13386</th><td>CG13386   </td><td> 13</td><td>  13</td><td> 116.89719</td><td>  0.5098887</td><td>Set 1</td><td>6.274654</td><td>43.17831</td></tr>\n",
       "\t<tr><th scope=row>Ptth</th><td>Ptth      </td><td>  3</td><td>  20</td><td>  60.60606</td><td>  0.9442888</td><td>Set 1</td><td>4.962148</td><td>29.49997</td></tr>\n",
       "\t<tr><th scope=row>CG3301</th><td>CG3301    </td><td>  3</td><td>  42</td><td>  64.85084</td><td>  1.4013055</td><td>Set 1</td><td>4.755234</td><td>28.72699</td></tr>\n",
       "\t<tr><th scope=row>CG13196</th><td>CG13196   </td><td> 11</td><td> 354</td><td> 171.62101</td><td> 15.1215662</td><td>Set 1</td><td>3.412162</td><td>25.35736</td></tr>\n",
       "\t<tr><th scope=row>Vajk4</th><td>Vajk4     </td><td> 64</td><td>3893</td><td> 818.90149</td><td>138.6952636</td><td>Set 1</td><td>2.551407</td><td>24.69585</td></tr>\n",
       "\t<tr><th scope=row>CG43338</th><td>CG43338   </td><td>  6</td><td> 140</td><td>  79.14283</td><td>  5.1882173</td><td>Set 1</td><td>3.676863</td><td>23.25433</td></tr>\n",
       "\t<tr><th scope=row>CG14606</th><td>CG14606   </td><td>  3</td><td>  41</td><td>  42.73504</td><td>  1.3954268</td><td>Set 1</td><td>4.157065</td><td>22.65899</td></tr>\n",
       "\t<tr><th scope=row>Coq8</th><td>Coq8      </td><td>  7</td><td> 266</td><td>  99.22991</td><td>  8.7906728</td><td>Set 1</td><td>3.341295</td><td>22.21015</td></tr>\n",
       "\t<tr><th scope=row>Nmt</th><td>Nmt       </td><td>  7</td><td> 519</td><td> 156.09537</td><td> 18.1778252</td><td>Set 1</td><td>3.024917</td><td>22.06827</td></tr>\n",
       "\t<tr><th scope=row>CG42359</th><td>CG42359   </td><td>  7</td><td> 367</td><td> 129.64697</td><td> 14.0741028</td><td>Set 1</td><td>3.104444</td><td>21.82279</td></tr>\n",
       "\t<tr><th scope=row>Cog3</th><td>Cog3      </td><td> 15</td><td> 604</td><td> 180.21105</td><td> 24.4024147</td><td>Set 1</td><td>2.826650</td><td>21.20419</td></tr>\n",
       "\t<tr><th scope=row>mRpS24</th><td>mRpS24    </td><td>  3</td><td>  84</td><td>  52.24660</td><td>  3.0532663</td><td>Set 1</td><td>3.688180</td><td>21.15030</td></tr>\n",
       "\t<tr><th scope=row>eEF1alpha2</th><td>eEF1alpha2</td><td> 10</td><td> 334</td><td> 116.05675</td><td> 12.9668555</td><td>Set 1</td><td>3.054751</td><td>20.98939</td></tr>\n",
       "\t<tr><th scope=row>CG14971</th><td>CG14971   </td><td> 14</td><td> 618</td><td> 182.62682</td><td> 25.8401851</td><td>Set 1</td><td>2.766432</td><td>20.80532</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tDesi & Desi       & 118 &  109 & 1462.15337 &   4.5872114 & Set 1 & 8.031751 & 84.45277\\\\\n",
       "\tCG13727 & CG13727    &  64 &   56 &  713.38143 &   2.4560272 & Set 1 & 7.689415 & 72.89989\\\\\n",
       "\tCG13310 & CG13310    &  23 &   12 &  262.28598 &   0.3391824 & Set 1 & 7.613644 & 61.21741\\\\\n",
       "\tCG32040 & CG32040    &  32 &   73 &  430.07555 &   3.2541828 & Set 1 & 6.659564 & 58.28315\\\\\n",
       "\tCG45072 & CG45072    &  14 &   28 &  177.20362 &   1.2591345 & Set 1 & 6.293494 & 47.05886\\\\\n",
       "\tatk & atk        &  18 &  121 &  275.12744 &   4.8699808 & Set 1 & 5.550600 & 45.01088\\\\\n",
       "\tCG13386 & CG13386    &  13 &   13 &  116.89719 &   0.5098887 & Set 1 & 6.274654 & 43.17831\\\\\n",
       "\tPtth & Ptth       &   3 &   20 &   60.60606 &   0.9442888 & Set 1 & 4.962148 & 29.49997\\\\\n",
       "\tCG3301 & CG3301     &   3 &   42 &   64.85084 &   1.4013055 & Set 1 & 4.755234 & 28.72699\\\\\n",
       "\tCG13196 & CG13196    &  11 &  354 &  171.62101 &  15.1215662 & Set 1 & 3.412162 & 25.35736\\\\\n",
       "\tVajk4 & Vajk4      &  64 & 3893 &  818.90149 & 138.6952636 & Set 1 & 2.551407 & 24.69585\\\\\n",
       "\tCG43338 & CG43338    &   6 &  140 &   79.14283 &   5.1882173 & Set 1 & 3.676863 & 23.25433\\\\\n",
       "\tCG14606 & CG14606    &   3 &   41 &   42.73504 &   1.3954268 & Set 1 & 4.157065 & 22.65899\\\\\n",
       "\tCoq8 & Coq8       &   7 &  266 &   99.22991 &   8.7906728 & Set 1 & 3.341295 & 22.21015\\\\\n",
       "\tNmt & Nmt        &   7 &  519 &  156.09537 &  18.1778252 & Set 1 & 3.024917 & 22.06827\\\\\n",
       "\tCG42359 & CG42359    &   7 &  367 &  129.64697 &  14.0741028 & Set 1 & 3.104444 & 21.82279\\\\\n",
       "\tCog3 & Cog3       &  15 &  604 &  180.21105 &  24.4024147 & Set 1 & 2.826650 & 21.20419\\\\\n",
       "\tmRpS24 & mRpS24     &   3 &   84 &   52.24660 &   3.0532663 & Set 1 & 3.688180 & 21.15030\\\\\n",
       "\teEF1alpha2 & eEF1alpha2 &  10 &  334 &  116.05675 &  12.9668555 & Set 1 & 3.054751 & 20.98939\\\\\n",
       "\tCG14971 & CG14971    &  14 &  618 &  182.62682 &  25.8401851 & Set 1 & 2.766432 & 20.80532\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Desi | Desi       | 118 |  109 | 1462.15337 |   4.5872114 | Set 1 | 8.031751 | 84.45277 |\n",
       "| CG13727 | CG13727    |  64 |   56 |  713.38143 |   2.4560272 | Set 1 | 7.689415 | 72.89989 |\n",
       "| CG13310 | CG13310    |  23 |   12 |  262.28598 |   0.3391824 | Set 1 | 7.613644 | 61.21741 |\n",
       "| CG32040 | CG32040    |  32 |   73 |  430.07555 |   3.2541828 | Set 1 | 6.659564 | 58.28315 |\n",
       "| CG45072 | CG45072    |  14 |   28 |  177.20362 |   1.2591345 | Set 1 | 6.293494 | 47.05886 |\n",
       "| atk | atk        |  18 |  121 |  275.12744 |   4.8699808 | Set 1 | 5.550600 | 45.01088 |\n",
       "| CG13386 | CG13386    |  13 |   13 |  116.89719 |   0.5098887 | Set 1 | 6.274654 | 43.17831 |\n",
       "| Ptth | Ptth       |   3 |   20 |   60.60606 |   0.9442888 | Set 1 | 4.962148 | 29.49997 |\n",
       "| CG3301 | CG3301     |   3 |   42 |   64.85084 |   1.4013055 | Set 1 | 4.755234 | 28.72699 |\n",
       "| CG13196 | CG13196    |  11 |  354 |  171.62101 |  15.1215662 | Set 1 | 3.412162 | 25.35736 |\n",
       "| Vajk4 | Vajk4      |  64 | 3893 |  818.90149 | 138.6952636 | Set 1 | 2.551407 | 24.69585 |\n",
       "| CG43338 | CG43338    |   6 |  140 |   79.14283 |   5.1882173 | Set 1 | 3.676863 | 23.25433 |\n",
       "| CG14606 | CG14606    |   3 |   41 |   42.73504 |   1.3954268 | Set 1 | 4.157065 | 22.65899 |\n",
       "| Coq8 | Coq8       |   7 |  266 |   99.22991 |   8.7906728 | Set 1 | 3.341295 | 22.21015 |\n",
       "| Nmt | Nmt        |   7 |  519 |  156.09537 |  18.1778252 | Set 1 | 3.024917 | 22.06827 |\n",
       "| CG42359 | CG42359    |   7 |  367 |  129.64697 |  14.0741028 | Set 1 | 3.104444 | 21.82279 |\n",
       "| Cog3 | Cog3       |  15 |  604 |  180.21105 |  24.4024147 | Set 1 | 2.826650 | 21.20419 |\n",
       "| mRpS24 | mRpS24     |   3 |   84 |   52.24660 |   3.0532663 | Set 1 | 3.688180 | 21.15030 |\n",
       "| eEF1alpha2 | eEF1alpha2 |  10 |  334 |  116.05675 |  12.9668555 | Set 1 | 3.054751 | 20.98939 |\n",
       "| CG14971 | CG14971    |  14 |  618 |  182.62682 |  25.8401851 | Set 1 | 2.766432 | 20.80532 |\n",
       "\n"
      ],
      "text/plain": [
       "           gene       set.1.n.umi set.2.n.umi set.1.tpm  set.2.tpm  \n",
       "Desi       Desi       118          109        1462.15337   4.5872114\n",
       "CG13727    CG13727     64           56         713.38143   2.4560272\n",
       "CG13310    CG13310     23           12         262.28598   0.3391824\n",
       "CG32040    CG32040     32           73         430.07555   3.2541828\n",
       "CG45072    CG45072     14           28         177.20362   1.2591345\n",
       "atk        atk         18          121         275.12744   4.8699808\n",
       "CG13386    CG13386     13           13         116.89719   0.5098887\n",
       "Ptth       Ptth         3           20          60.60606   0.9442888\n",
       "CG3301     CG3301       3           42          64.85084   1.4013055\n",
       "CG13196    CG13196     11          354         171.62101  15.1215662\n",
       "Vajk4      Vajk4       64         3893         818.90149 138.6952636\n",
       "CG43338    CG43338      6          140          79.14283   5.1882173\n",
       "CG14606    CG14606      3           41          42.73504   1.3954268\n",
       "Coq8       Coq8         7          266          99.22991   8.7906728\n",
       "Nmt        Nmt          7          519         156.09537  18.1778252\n",
       "CG42359    CG42359      7          367         129.64697  14.0741028\n",
       "Cog3       Cog3        15          604         180.21105  24.4024147\n",
       "mRpS24     mRpS24       3           84          52.24660   3.0532663\n",
       "eEF1alpha2 eEF1alpha2  10          334         116.05675  12.9668555\n",
       "CG14971    CG14971     14          618         182.62682  25.8401851\n",
       "           higher.expr log2.ratio heuristic.score\n",
       "Desi       Set 1       8.031751   84.45277       \n",
       "CG13727    Set 1       7.689415   72.89989       \n",
       "CG13310    Set 1       7.613644   61.21741       \n",
       "CG32040    Set 1       6.659564   58.28315       \n",
       "CG45072    Set 1       6.293494   47.05886       \n",
       "atk        Set 1       5.550600   45.01088       \n",
       "CG13386    Set 1       6.274654   43.17831       \n",
       "Ptth       Set 1       4.962148   29.49997       \n",
       "CG3301     Set 1       4.755234   28.72699       \n",
       "CG13196    Set 1       3.412162   25.35736       \n",
       "Vajk4      Set 1       2.551407   24.69585       \n",
       "CG43338    Set 1       3.676863   23.25433       \n",
       "CG14606    Set 1       4.157065   22.65899       \n",
       "Coq8       Set 1       3.341295   22.21015       \n",
       "Nmt        Set 1       3.024917   22.06827       \n",
       "CG42359    Set 1       3.104444   21.82279       \n",
       "Cog3       Set 1       2.826650   21.20419       \n",
       "mRpS24     Set 1       3.688180   21.15030       \n",
       "eEF1alpha2 Set 1       3.054751   20.98939       \n",
       "CG14971    Set 1       2.766432   20.80532       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## no strong differences but keep them separate by name\n",
    "two.set.differential.gene.test.3a(cds_sub, \n",
    "                                  df_sub300$cluster == 133 & \n",
    "                                   df_sub300$umap_2 < 0*df_sub300$umap_1 + 2.1 , \n",
    "                                 !(df_sub300$cluster == 18 & \n",
    "                                   df_sub300$umap_2 < 0*df_sub300$umap_1 + 2.1),\n",
    "                                  sample = 50000\n",
    "                                )  %>% filter(higher.expr == \"Set 1\")  %>% \n",
    "head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1540,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Somatic_muscle_primordium\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Somatic_muscle_primordium\", df_sub300$cluster %in% be)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 265,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'be' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'be' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(be)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 266,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'be' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'be' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(be)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1536,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>lmd32</th><td>be</td><td>lmd    </td><td> 2521</td><td> 2688</td><td>1915.68285</td><td> 109.8841438</td><td>4.110734</td><td>44.82507</td></tr>\n",
       "\t<tr><th scope=row>Kah32</th><td>be</td><td>Kah    </td><td> 1015</td><td> 2002</td><td> 734.89371</td><td>  82.0770682</td><td>3.145013</td><td>29.95107</td></tr>\n",
       "\t<tr><th scope=row>sns32</th><td>be</td><td>sns    </td><td> 2329</td><td> 5668</td><td>1696.66298</td><td> 246.0002192</td><td>2.780116</td><td>29.82879</td></tr>\n",
       "\t<tr><th scope=row>Rim232</th><td>be</td><td>Rim2   </td><td> 1671</td><td> 4736</td><td>1201.20687</td><td> 178.2374103</td><td>2.744541</td><td>28.08069</td></tr>\n",
       "\t<tr><th scope=row>Gyc32E32</th><td>be</td><td>Gyc32E </td><td>  306</td><td>  487</td><td> 218.19646</td><td>  17.8360264</td><td>3.534061</td><td>27.48114</td></tr>\n",
       "\t<tr><th scope=row>kirre32</th><td>be</td><td>kirre  </td><td> 4464</td><td>15028</td><td>3243.79946</td><td> 690.6964813</td><td>2.229474</td><td>26.00439</td></tr>\n",
       "\t<tr><th scope=row>sing32</th><td>be</td><td>sing   </td><td>  201</td><td>  289</td><td> 145.87586</td><td>  11.7929025</td><td>3.511326</td><td>25.27611</td></tr>\n",
       "\t<tr><th scope=row>Mef232</th><td>be</td><td>Mef2   </td><td> 3141</td><td>12961</td><td>2278.45077</td><td> 504.5078466</td><td>2.172248</td><td>24.23027</td></tr>\n",
       "\t<tr><th scope=row>Or69a32</th><td>be</td><td>Or69a  </td><td>   48</td><td>   22</td><td>  37.67684</td><td>   0.7583816</td><td>4.421358</td><td>23.31558</td></tr>\n",
       "\t<tr><th scope=row>ltl32</th><td>be</td><td>ltl    </td><td>  520</td><td> 1416</td><td> 379.05958</td><td>  56.5685860</td><td>2.719071</td><td>23.30266</td></tr>\n",
       "\t<tr><th scope=row>Rbfox132</th><td>be</td><td>Rbfox1 </td><td>10353</td><td>53052</td><td>7553.22894</td><td>2288.2871448</td><td>1.722195</td><td>22.18716</td></tr>\n",
       "\t<tr><th scope=row>CG4275032</th><td>be</td><td>CG42750</td><td>  531</td><td> 1649</td><td> 399.01378</td><td>  68.2179990</td><td>2.527219</td><td>21.84505</td></tr>\n",
       "\t<tr><th scope=row>CG1546832</th><td>be</td><td>CG15468</td><td>   68</td><td>   70</td><td>  53.60399</td><td>   3.0439852</td><td>3.728491</td><td>21.51688</td></tr>\n",
       "\t<tr><th scope=row>CG3403032</th><td>be</td><td>CG34030</td><td>  108</td><td>  195</td><td>  87.73728</td><td>   8.2326579</td><td>3.248372</td><td>21.02174</td></tr>\n",
       "\t<tr><th scope=row>CG965032</th><td>be</td><td>CG9650 </td><td> 3332</td><td>16327</td><td>2452.85830</td><td> 690.1307912</td><td>1.827433</td><td>20.57843</td></tr>\n",
       "\t<tr><th scope=row>Con32</th><td>be</td><td>Con    </td><td> 2444</td><td>11091</td><td>1807.66836</td><td> 489.1255043</td><td>1.882907</td><td>20.37439</td></tr>\n",
       "\t<tr><th scope=row>CG635632</th><td>be</td><td>CG6356 </td><td>  446</td><td> 1368</td><td> 330.80037</td><td>  60.6195290</td><td>2.424501</td><td>20.30319</td></tr>\n",
       "\t<tr><th scope=row>kl-532</th><td>be</td><td>kl-5   </td><td>   86</td><td>  146</td><td>  65.14618</td><td>   5.5399407</td><td>3.316331</td><td>20.05580</td></tr>\n",
       "\t<tr><th scope=row>CG864132</th><td>be</td><td>CG8641 </td><td>  199</td><td>  521</td><td> 156.59503</td><td>  22.8031397</td><td>2.717815</td><td>19.84026</td></tr>\n",
       "\t<tr><th scope=row>ewg32</th><td>be</td><td>ewg    </td><td> 2413</td><td>11100</td><td>1726.24210</td><td> 480.8060005</td><td>1.841111</td><td>19.79977</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tlmd32 & be & lmd     &  2521 &  2688 & 1915.68285 &  109.8841438 & 4.110734 & 44.82507\\\\\n",
       "\tKah32 & be & Kah     &  1015 &  2002 &  734.89371 &   82.0770682 & 3.145013 & 29.95107\\\\\n",
       "\tsns32 & be & sns     &  2329 &  5668 & 1696.66298 &  246.0002192 & 2.780116 & 29.82879\\\\\n",
       "\tRim232 & be & Rim2    &  1671 &  4736 & 1201.20687 &  178.2374103 & 2.744541 & 28.08069\\\\\n",
       "\tGyc32E32 & be & Gyc32E  &   306 &   487 &  218.19646 &   17.8360264 & 3.534061 & 27.48114\\\\\n",
       "\tkirre32 & be & kirre   &  4464 & 15028 & 3243.79946 &  690.6964813 & 2.229474 & 26.00439\\\\\n",
       "\tsing32 & be & sing    &   201 &   289 &  145.87586 &   11.7929025 & 3.511326 & 25.27611\\\\\n",
       "\tMef232 & be & Mef2    &  3141 & 12961 & 2278.45077 &  504.5078466 & 2.172248 & 24.23027\\\\\n",
       "\tOr69a32 & be & Or69a   &    48 &    22 &   37.67684 &    0.7583816 & 4.421358 & 23.31558\\\\\n",
       "\tltl32 & be & ltl     &   520 &  1416 &  379.05958 &   56.5685860 & 2.719071 & 23.30266\\\\\n",
       "\tRbfox132 & be & Rbfox1  & 10353 & 53052 & 7553.22894 & 2288.2871448 & 1.722195 & 22.18716\\\\\n",
       "\tCG4275032 & be & CG42750 &   531 &  1649 &  399.01378 &   68.2179990 & 2.527219 & 21.84505\\\\\n",
       "\tCG1546832 & be & CG15468 &    68 &    70 &   53.60399 &    3.0439852 & 3.728491 & 21.51688\\\\\n",
       "\tCG3403032 & be & CG34030 &   108 &   195 &   87.73728 &    8.2326579 & 3.248372 & 21.02174\\\\\n",
       "\tCG965032 & be & CG9650  &  3332 & 16327 & 2452.85830 &  690.1307912 & 1.827433 & 20.57843\\\\\n",
       "\tCon32 & be & Con     &  2444 & 11091 & 1807.66836 &  489.1255043 & 1.882907 & 20.37439\\\\\n",
       "\tCG635632 & be & CG6356  &   446 &  1368 &  330.80037 &   60.6195290 & 2.424501 & 20.30319\\\\\n",
       "\tkl-532 & be & kl-5    &    86 &   146 &   65.14618 &    5.5399407 & 3.316331 & 20.05580\\\\\n",
       "\tCG864132 & be & CG8641  &   199 &   521 &  156.59503 &   22.8031397 & 2.717815 & 19.84026\\\\\n",
       "\tewg32 & be & ewg     &  2413 & 11100 & 1726.24210 &  480.8060005 & 1.841111 & 19.79977\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| lmd32 | be | lmd     |  2521 |  2688 | 1915.68285 |  109.8841438 | 4.110734 | 44.82507 |\n",
       "| Kah32 | be | Kah     |  1015 |  2002 |  734.89371 |   82.0770682 | 3.145013 | 29.95107 |\n",
       "| sns32 | be | sns     |  2329 |  5668 | 1696.66298 |  246.0002192 | 2.780116 | 29.82879 |\n",
       "| Rim232 | be | Rim2    |  1671 |  4736 | 1201.20687 |  178.2374103 | 2.744541 | 28.08069 |\n",
       "| Gyc32E32 | be | Gyc32E  |   306 |   487 |  218.19646 |   17.8360264 | 3.534061 | 27.48114 |\n",
       "| kirre32 | be | kirre   |  4464 | 15028 | 3243.79946 |  690.6964813 | 2.229474 | 26.00439 |\n",
       "| sing32 | be | sing    |   201 |   289 |  145.87586 |   11.7929025 | 3.511326 | 25.27611 |\n",
       "| Mef232 | be | Mef2    |  3141 | 12961 | 2278.45077 |  504.5078466 | 2.172248 | 24.23027 |\n",
       "| Or69a32 | be | Or69a   |    48 |    22 |   37.67684 |    0.7583816 | 4.421358 | 23.31558 |\n",
       "| ltl32 | be | ltl     |   520 |  1416 |  379.05958 |   56.5685860 | 2.719071 | 23.30266 |\n",
       "| Rbfox132 | be | Rbfox1  | 10353 | 53052 | 7553.22894 | 2288.2871448 | 1.722195 | 22.18716 |\n",
       "| CG4275032 | be | CG42750 |   531 |  1649 |  399.01378 |   68.2179990 | 2.527219 | 21.84505 |\n",
       "| CG1546832 | be | CG15468 |    68 |    70 |   53.60399 |    3.0439852 | 3.728491 | 21.51688 |\n",
       "| CG3403032 | be | CG34030 |   108 |   195 |   87.73728 |    8.2326579 | 3.248372 | 21.02174 |\n",
       "| CG965032 | be | CG9650  |  3332 | 16327 | 2452.85830 |  690.1307912 | 1.827433 | 20.57843 |\n",
       "| Con32 | be | Con     |  2444 | 11091 | 1807.66836 |  489.1255043 | 1.882907 | 20.37439 |\n",
       "| CG635632 | be | CG6356  |   446 |  1368 |  330.80037 |   60.6195290 | 2.424501 | 20.30319 |\n",
       "| kl-532 | be | kl-5    |    86 |   146 |   65.14618 |    5.5399407 | 3.316331 | 20.05580 |\n",
       "| CG864132 | be | CG8641  |   199 |   521 |  156.59503 |   22.8031397 | 2.717815 | 19.84026 |\n",
       "| ewg32 | be | ewg     |  2413 | 11100 | 1726.24210 |  480.8060005 | 1.841111 | 19.79977 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "lmd32     be          lmd      2521              2688       1915.68285     \n",
       "Kah32     be          Kah      1015              2002        734.89371     \n",
       "sns32     be          sns      2329              5668       1696.66298     \n",
       "Rim232    be          Rim2     1671              4736       1201.20687     \n",
       "Gyc32E32  be          Gyc32E    306               487        218.19646     \n",
       "kirre32   be          kirre    4464             15028       3243.79946     \n",
       "sing32    be          sing      201               289        145.87586     \n",
       "Mef232    be          Mef2     3141             12961       2278.45077     \n",
       "Or69a32   be          Or69a      48                22         37.67684     \n",
       "ltl32     be          ltl       520              1416        379.05958     \n",
       "Rbfox132  be          Rbfox1  10353             53052       7553.22894     \n",
       "CG4275032 be          CG42750   531              1649        399.01378     \n",
       "CG1546832 be          CG15468    68                70         53.60399     \n",
       "CG3403032 be          CG34030   108               195         87.73728     \n",
       "CG965032  be          CG9650   3332             16327       2452.85830     \n",
       "Con32     be          Con      2444             11091       1807.66836     \n",
       "CG635632  be          CG6356    446              1368        330.80037     \n",
       "kl-532    be          kl-5       86               146         65.14618     \n",
       "CG864132  be          CG8641    199               521        156.59503     \n",
       "ewg32     be          ewg      2413             11100       1726.24210     \n",
       "          other.tpm    log2.ratio heuristic.score\n",
       "lmd32      109.8841438 4.110734   44.82507       \n",
       "Kah32       82.0770682 3.145013   29.95107       \n",
       "sns32      246.0002192 2.780116   29.82879       \n",
       "Rim232     178.2374103 2.744541   28.08069       \n",
       "Gyc32E32    17.8360264 3.534061   27.48114       \n",
       "kirre32    690.6964813 2.229474   26.00439       \n",
       "sing32      11.7929025 3.511326   25.27611       \n",
       "Mef232     504.5078466 2.172248   24.23027       \n",
       "Or69a32      0.7583816 4.421358   23.31558       \n",
       "ltl32       56.5685860 2.719071   23.30266       \n",
       "Rbfox132  2288.2871448 1.722195   22.18716       \n",
       "CG4275032   68.2179990 2.527219   21.84505       \n",
       "CG1546832    3.0439852 3.728491   21.51688       \n",
       "CG3403032    8.2326579 3.248372   21.02174       \n",
       "CG965032   690.1307912 1.827433   20.57843       \n",
       "Con32      489.1255043 1.882907   20.37439       \n",
       "CG635632    60.6195290 2.424501   20.30319       \n",
       "kl-532       5.5399407 3.316331   20.05580       \n",
       "CG864132    22.8031397 2.717815   19.84026       \n",
       "ewg32      480.8060005 1.841111   19.79977       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"be\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 267,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"lmd\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"lmd\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"lmd\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Kah\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"sns\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1545,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Midgut\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Midgut\", df_sub300$cluster %in% bf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 268,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bf' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bf' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bf)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 269,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bf' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bf' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bf)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1542,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>ham33</th><td>bf</td><td>ham     </td><td>1272</td><td> 4031</td><td>2670.4435</td><td> 180.00901</td><td>3.882946</td><td>44.20114</td></tr>\n",
       "\t<tr><th scope=row>CG3399333</th><td>bf</td><td>CG33993 </td><td>1174</td><td> 4459</td><td>2451.1117</td><td> 201.33156</td><td>3.598643</td><td>40.52003</td></tr>\n",
       "\t<tr><th scope=row>FucTC33</th><td>bf</td><td>FucTC   </td><td> 220</td><td>  587</td><td> 456.7397</td><td>  23.89620</td><td>4.197375</td><td>37.09801</td></tr>\n",
       "\t<tr><th scope=row>vnd33</th><td>bf</td><td>vnd     </td><td> 328</td><td> 1035</td><td> 663.1536</td><td>  42.55709</td><td>3.928364</td><td>36.82987</td></tr>\n",
       "\t<tr><th scope=row>grn33</th><td>bf</td><td>grn     </td><td>2636</td><td>17294</td><td>5582.3982</td><td> 757.87278</td><td>2.878955</td><td>35.83415</td></tr>\n",
       "\t<tr><th scope=row>sens-233</th><td>bf</td><td>sens-2  </td><td> 379</td><td> 1720</td><td> 799.3552</td><td>  75.55306</td><td>3.384305</td><td>32.63992</td></tr>\n",
       "\t<tr><th scope=row>CG1534733</th><td>bf</td><td>CG15347 </td><td> 634</td><td> 3388</td><td>1299.8011</td><td> 159.46232</td><td>3.017984</td><td>31.22161</td></tr>\n",
       "\t<tr><th scope=row>bru233</th><td>bf</td><td>bru2    </td><td>1776</td><td>13742</td><td>3696.2302</td><td> 633.97719</td><td>2.541278</td><td>30.11981</td></tr>\n",
       "\t<tr><th scope=row>CG430133</th><td>bf</td><td>CG4301  </td><td> 197</td><td>  781</td><td> 404.9602</td><td>  37.03425</td><td>3.412409</td><td>29.56919</td></tr>\n",
       "\t<tr><th scope=row>Hnf433</th><td>bf</td><td>Hnf4    </td><td> 455</td><td> 2979</td><td> 961.6012</td><td> 127.50292</td><td>2.903638</td><td>28.77735</td></tr>\n",
       "\t<tr><th scope=row>Oatp33Ea33</th><td>bf</td><td>Oatp33Ea</td><td> 140</td><td>  606</td><td> 292.9165</td><td>  25.23696</td><td>3.480817</td><td>28.54013</td></tr>\n",
       "\t<tr><th scope=row>CG1328433</th><td>bf</td><td>CG13284 </td><td> 319</td><td> 1796</td><td> 660.4561</td><td>  79.93914</td><td>3.028553</td><td>28.37603</td></tr>\n",
       "\t<tr><th scope=row>trol33</th><td>bf</td><td>trol    </td><td>4587</td><td>46864</td><td>9732.4421</td><td>2208.63624</td><td>2.138993</td><td>28.33895</td></tr>\n",
       "\t<tr><th scope=row>Trim933</th><td>bf</td><td>Trim9   </td><td>1284</td><td>11386</td><td>2713.1455</td><td> 511.46621</td><td>2.404438</td><td>27.42569</td></tr>\n",
       "\t<tr><th scope=row>Pde933</th><td>bf</td><td>Pde9    </td><td>1298</td><td>11321</td><td>2750.9840</td><td> 528.76670</td><td>2.376519</td><td>27.15471</td></tr>\n",
       "\t<tr><th scope=row>mesh33</th><td>bf</td><td>mesh    </td><td> 706</td><td> 6435</td><td>1489.9262</td><td> 258.06950</td><td>2.523830</td><td>26.60620</td></tr>\n",
       "\t<tr><th scope=row>CG1536533</th><td>bf</td><td>CG15365 </td><td> 437</td><td> 3003</td><td> 930.3203</td><td> 142.72435</td><td>2.694423</td><td>26.57545</td></tr>\n",
       "\t<tr><th scope=row>Itgbn33</th><td>bf</td><td>Itgbn   </td><td> 187</td><td>  971</td><td> 398.0983</td><td>  46.47721</td><td>3.067817</td><td>26.50778</td></tr>\n",
       "\t<tr><th scope=row>Mitf33</th><td>bf</td><td>Mitf    </td><td> 349</td><td> 2245</td><td> 721.2168</td><td> 103.70631</td><td>2.784085</td><td>26.43847</td></tr>\n",
       "\t<tr><th scope=row>Ance33</th><td>bf</td><td>Ance    </td><td> 929</td><td> 8980</td><td>1996.4292</td><td> 386.01946</td><td>2.366944</td><td>25.95100</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tham33 & bf & ham      & 1272 &  4031 & 2670.4435 &  180.00901 & 3.882946 & 44.20114\\\\\n",
       "\tCG3399333 & bf & CG33993  & 1174 &  4459 & 2451.1117 &  201.33156 & 3.598643 & 40.52003\\\\\n",
       "\tFucTC33 & bf & FucTC    &  220 &   587 &  456.7397 &   23.89620 & 4.197375 & 37.09801\\\\\n",
       "\tvnd33 & bf & vnd      &  328 &  1035 &  663.1536 &   42.55709 & 3.928364 & 36.82987\\\\\n",
       "\tgrn33 & bf & grn      & 2636 & 17294 & 5582.3982 &  757.87278 & 2.878955 & 35.83415\\\\\n",
       "\tsens-233 & bf & sens-2   &  379 &  1720 &  799.3552 &   75.55306 & 3.384305 & 32.63992\\\\\n",
       "\tCG1534733 & bf & CG15347  &  634 &  3388 & 1299.8011 &  159.46232 & 3.017984 & 31.22161\\\\\n",
       "\tbru233 & bf & bru2     & 1776 & 13742 & 3696.2302 &  633.97719 & 2.541278 & 30.11981\\\\\n",
       "\tCG430133 & bf & CG4301   &  197 &   781 &  404.9602 &   37.03425 & 3.412409 & 29.56919\\\\\n",
       "\tHnf433 & bf & Hnf4     &  455 &  2979 &  961.6012 &  127.50292 & 2.903638 & 28.77735\\\\\n",
       "\tOatp33Ea33 & bf & Oatp33Ea &  140 &   606 &  292.9165 &   25.23696 & 3.480817 & 28.54013\\\\\n",
       "\tCG1328433 & bf & CG13284  &  319 &  1796 &  660.4561 &   79.93914 & 3.028553 & 28.37603\\\\\n",
       "\ttrol33 & bf & trol     & 4587 & 46864 & 9732.4421 & 2208.63624 & 2.138993 & 28.33895\\\\\n",
       "\tTrim933 & bf & Trim9    & 1284 & 11386 & 2713.1455 &  511.46621 & 2.404438 & 27.42569\\\\\n",
       "\tPde933 & bf & Pde9     & 1298 & 11321 & 2750.9840 &  528.76670 & 2.376519 & 27.15471\\\\\n",
       "\tmesh33 & bf & mesh     &  706 &  6435 & 1489.9262 &  258.06950 & 2.523830 & 26.60620\\\\\n",
       "\tCG1536533 & bf & CG15365  &  437 &  3003 &  930.3203 &  142.72435 & 2.694423 & 26.57545\\\\\n",
       "\tItgbn33 & bf & Itgbn    &  187 &   971 &  398.0983 &   46.47721 & 3.067817 & 26.50778\\\\\n",
       "\tMitf33 & bf & Mitf     &  349 &  2245 &  721.2168 &  103.70631 & 2.784085 & 26.43847\\\\\n",
       "\tAnce33 & bf & Ance     &  929 &  8980 & 1996.4292 &  386.01946 & 2.366944 & 25.95100\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| ham33 | bf | ham      | 1272 |  4031 | 2670.4435 |  180.00901 | 3.882946 | 44.20114 |\n",
       "| CG3399333 | bf | CG33993  | 1174 |  4459 | 2451.1117 |  201.33156 | 3.598643 | 40.52003 |\n",
       "| FucTC33 | bf | FucTC    |  220 |   587 |  456.7397 |   23.89620 | 4.197375 | 37.09801 |\n",
       "| vnd33 | bf | vnd      |  328 |  1035 |  663.1536 |   42.55709 | 3.928364 | 36.82987 |\n",
       "| grn33 | bf | grn      | 2636 | 17294 | 5582.3982 |  757.87278 | 2.878955 | 35.83415 |\n",
       "| sens-233 | bf | sens-2   |  379 |  1720 |  799.3552 |   75.55306 | 3.384305 | 32.63992 |\n",
       "| CG1534733 | bf | CG15347  |  634 |  3388 | 1299.8011 |  159.46232 | 3.017984 | 31.22161 |\n",
       "| bru233 | bf | bru2     | 1776 | 13742 | 3696.2302 |  633.97719 | 2.541278 | 30.11981 |\n",
       "| CG430133 | bf | CG4301   |  197 |   781 |  404.9602 |   37.03425 | 3.412409 | 29.56919 |\n",
       "| Hnf433 | bf | Hnf4     |  455 |  2979 |  961.6012 |  127.50292 | 2.903638 | 28.77735 |\n",
       "| Oatp33Ea33 | bf | Oatp33Ea |  140 |   606 |  292.9165 |   25.23696 | 3.480817 | 28.54013 |\n",
       "| CG1328433 | bf | CG13284  |  319 |  1796 |  660.4561 |   79.93914 | 3.028553 | 28.37603 |\n",
       "| trol33 | bf | trol     | 4587 | 46864 | 9732.4421 | 2208.63624 | 2.138993 | 28.33895 |\n",
       "| Trim933 | bf | Trim9    | 1284 | 11386 | 2713.1455 |  511.46621 | 2.404438 | 27.42569 |\n",
       "| Pde933 | bf | Pde9     | 1298 | 11321 | 2750.9840 |  528.76670 | 2.376519 | 27.15471 |\n",
       "| mesh33 | bf | mesh     |  706 |  6435 | 1489.9262 |  258.06950 | 2.523830 | 26.60620 |\n",
       "| CG1536533 | bf | CG15365  |  437 |  3003 |  930.3203 |  142.72435 | 2.694423 | 26.57545 |\n",
       "| Itgbn33 | bf | Itgbn    |  187 |   971 |  398.0983 |   46.47721 | 3.067817 | 26.50778 |\n",
       "| Mitf33 | bf | Mitf     |  349 |  2245 |  721.2168 |  103.70631 | 2.784085 | 26.43847 |\n",
       "| Ance33 | bf | Ance     |  929 |  8980 | 1996.4292 |  386.01946 | 2.366944 | 25.95100 |\n",
       "\n"
      ],
      "text/plain": [
       "           cell.type.2 gene     cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "ham33      bf          ham      1272               4031       2670.4435      \n",
       "CG3399333  bf          CG33993  1174               4459       2451.1117      \n",
       "FucTC33    bf          FucTC     220                587        456.7397      \n",
       "vnd33      bf          vnd       328               1035        663.1536      \n",
       "grn33      bf          grn      2636              17294       5582.3982      \n",
       "sens-233   bf          sens-2    379               1720        799.3552      \n",
       "CG1534733  bf          CG15347   634               3388       1299.8011      \n",
       "bru233     bf          bru2     1776              13742       3696.2302      \n",
       "CG430133   bf          CG4301    197                781        404.9602      \n",
       "Hnf433     bf          Hnf4      455               2979        961.6012      \n",
       "Oatp33Ea33 bf          Oatp33Ea  140                606        292.9165      \n",
       "CG1328433  bf          CG13284   319               1796        660.4561      \n",
       "trol33     bf          trol     4587              46864       9732.4421      \n",
       "Trim933    bf          Trim9    1284              11386       2713.1455      \n",
       "Pde933     bf          Pde9     1298              11321       2750.9840      \n",
       "mesh33     bf          mesh      706               6435       1489.9262      \n",
       "CG1536533  bf          CG15365   437               3003        930.3203      \n",
       "Itgbn33    bf          Itgbn     187                971        398.0983      \n",
       "Mitf33     bf          Mitf      349               2245        721.2168      \n",
       "Ance33     bf          Ance      929               8980       1996.4292      \n",
       "           other.tpm  log2.ratio heuristic.score\n",
       "ham33       180.00901 3.882946   44.20114       \n",
       "CG3399333   201.33156 3.598643   40.52003       \n",
       "FucTC33      23.89620 4.197375   37.09801       \n",
       "vnd33        42.55709 3.928364   36.82987       \n",
       "grn33       757.87278 2.878955   35.83415       \n",
       "sens-233     75.55306 3.384305   32.63992       \n",
       "CG1534733   159.46232 3.017984   31.22161       \n",
       "bru233      633.97719 2.541278   30.11981       \n",
       "CG430133     37.03425 3.412409   29.56919       \n",
       "Hnf433      127.50292 2.903638   28.77735       \n",
       "Oatp33Ea33   25.23696 3.480817   28.54013       \n",
       "CG1328433    79.93914 3.028553   28.37603       \n",
       "trol33     2208.63624 2.138993   28.33895       \n",
       "Trim933     511.46621 2.404438   27.42569       \n",
       "Pde933      528.76670 2.376519   27.15471       \n",
       "mesh33      258.06950 2.523830   26.60620       \n",
       "CG1536533   142.72435 2.694423   26.57545       \n",
       "Itgbn33      46.47721 3.067817   26.50778       \n",
       "Mitf33      103.70631 2.784085   26.43847       \n",
       "Ance33      386.01946 2.366944   25.95100       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"bf\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 270,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"tey\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"tey\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"tey\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"AdamTS-A\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"beat-IIb\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1551,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Late_cells\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Late_cells\", df_sub300$cluster %in% bg)\n",
    "## top hits connected to pharynx but only at very tip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 271,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bg' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bg' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bg)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 272,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bg' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bg' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bg)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1548,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Ccp84Ac34</th><td>bg</td><td>Ccp84Ac</td><td>475</td><td>1028</td><td>972.66793</td><td> 38.9462062</td><td>4.605817</td><td>45.72326</td></tr>\n",
       "\t<tr><th scope=row>CG754834</th><td>bg</td><td>CG7548 </td><td> 94</td><td> 110</td><td>192.96764</td><td>  3.3584222</td><td>5.468409</td><td>41.55812</td></tr>\n",
       "\t<tr><th scope=row>Cpr56F34</th><td>bg</td><td>Cpr56F </td><td>329</td><td> 960</td><td>646.56628</td><td> 31.5248357</td><td>4.313185</td><td>40.28033</td></tr>\n",
       "\t<tr><th scope=row>Cpr92F34</th><td>bg</td><td>Cpr92F </td><td>113</td><td> 164</td><td>237.90139</td><td>  6.3026669</td><td>5.025796</td><td>39.70516</td></tr>\n",
       "\t<tr><th scope=row>CG3446134</th><td>bg</td><td>CG34461</td><td>328</td><td>1240</td><td>682.72649</td><td> 44.2191448</td><td>3.916302</td><td>36.88090</td></tr>\n",
       "\t<tr><th scope=row>CG507034</th><td>bg</td><td>CG5070 </td><td> 88</td><td> 145</td><td>172.43531</td><td>  4.7561169</td><td>4.904816</td><td>36.48326</td></tr>\n",
       "\t<tr><th scope=row>Cpr97Ea34</th><td>bg</td><td>Cpr97Ea</td><td> 86</td><td> 156</td><td>167.73288</td><td>  5.1451985</td><td>4.770562</td><td>35.29547</td></tr>\n",
       "\t<tr><th scope=row>CG926934</th><td>bg</td><td>CG9269 </td><td> 72</td><td> 107</td><td>141.84387</td><td>  3.7619523</td><td>4.896607</td><td>35.05136</td></tr>\n",
       "\t<tr><th scope=row>TwdlC34</th><td>bg</td><td>TwdlC  </td><td>319</td><td>1186</td><td>584.75983</td><td> 43.3012171</td><td>3.722426</td><td>34.22460</td></tr>\n",
       "\t<tr><th scope=row>Ccp84Ab34</th><td>bg</td><td>Ccp84Ab</td><td> 63</td><td> 176</td><td>138.08641</td><td>  4.1299525</td><td>4.750482</td><td>33.82266</td></tr>\n",
       "\t<tr><th scope=row>CG717934</th><td>bg</td><td>CG7179 </td><td>112</td><td> 215</td><td>202.67024</td><td>  8.8739713</td><td>4.359360</td><td>33.43669</td></tr>\n",
       "\t<tr><th scope=row>CG1304146</th><td>bg</td><td>CG13041</td><td> 90</td><td> 168</td><td>158.39488</td><td>  5.8903057</td><td>4.522814</td><td>33.09099</td></tr>\n",
       "\t<tr><th scope=row>Cpr62Bc34</th><td>bg</td><td>Cpr62Bc</td><td>239</td><td> 966</td><td>461.47700</td><td> 35.1577109</td><td>3.673883</td><td>32.52576</td></tr>\n",
       "\t<tr><th scope=row>Ccp84Aa34</th><td>bg</td><td>Ccp84Aa</td><td> 31</td><td>  31</td><td> 67.40467</td><td>  0.6917341</td><td>5.316274</td><td>32.40813</td></tr>\n",
       "\t<tr><th scope=row>CG1252334</th><td>bg</td><td>CG12523</td><td>159</td><td> 560</td><td>313.04517</td><td> 20.9281758</td><td>3.835513</td><td>31.81492</td></tr>\n",
       "\t<tr><th scope=row>CG1304334</th><td>bg</td><td>CG13043</td><td> 98</td><td> 353</td><td>192.78959</td><td> 10.4495332</td><td>4.073666</td><td>30.95313</td></tr>\n",
       "\t<tr><th scope=row>Muc91C34</th><td>bg</td><td>Muc91C </td><td>433</td><td>2681</td><td>884.17446</td><td>100.0958347</td><td>3.128608</td><td>30.62850</td></tr>\n",
       "\t<tr><th scope=row>CG1306034</th><td>bg</td><td>CG13060</td><td> 62</td><td> 143</td><td>113.49362</td><td>  5.1268556</td><td>4.211321</td><td>28.80174</td></tr>\n",
       "\t<tr><th scope=row>CG1305934</th><td>bg</td><td>CG13059</td><td>137</td><td> 571</td><td>251.10833</td><td> 20.7232531</td><td>3.530998</td><td>28.16995</td></tr>\n",
       "\t<tr><th scope=row>Ccp84Ad34</th><td>bg</td><td>Ccp84Ad</td><td> 99</td><td> 432</td><td>202.30688</td><td> 15.0565912</td><td>3.655308</td><td>28.02713</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCcp84Ac34 & bg & Ccp84Ac & 475 & 1028 & 972.66793 &  38.9462062 & 4.605817 & 45.72326\\\\\n",
       "\tCG754834 & bg & CG7548  &  94 &  110 & 192.96764 &   3.3584222 & 5.468409 & 41.55812\\\\\n",
       "\tCpr56F34 & bg & Cpr56F  & 329 &  960 & 646.56628 &  31.5248357 & 4.313185 & 40.28033\\\\\n",
       "\tCpr92F34 & bg & Cpr92F  & 113 &  164 & 237.90139 &   6.3026669 & 5.025796 & 39.70516\\\\\n",
       "\tCG3446134 & bg & CG34461 & 328 & 1240 & 682.72649 &  44.2191448 & 3.916302 & 36.88090\\\\\n",
       "\tCG507034 & bg & CG5070  &  88 &  145 & 172.43531 &   4.7561169 & 4.904816 & 36.48326\\\\\n",
       "\tCpr97Ea34 & bg & Cpr97Ea &  86 &  156 & 167.73288 &   5.1451985 & 4.770562 & 35.29547\\\\\n",
       "\tCG926934 & bg & CG9269  &  72 &  107 & 141.84387 &   3.7619523 & 4.896607 & 35.05136\\\\\n",
       "\tTwdlC34 & bg & TwdlC   & 319 & 1186 & 584.75983 &  43.3012171 & 3.722426 & 34.22460\\\\\n",
       "\tCcp84Ab34 & bg & Ccp84Ab &  63 &  176 & 138.08641 &   4.1299525 & 4.750482 & 33.82266\\\\\n",
       "\tCG717934 & bg & CG7179  & 112 &  215 & 202.67024 &   8.8739713 & 4.359360 & 33.43669\\\\\n",
       "\tCG1304146 & bg & CG13041 &  90 &  168 & 158.39488 &   5.8903057 & 4.522814 & 33.09099\\\\\n",
       "\tCpr62Bc34 & bg & Cpr62Bc & 239 &  966 & 461.47700 &  35.1577109 & 3.673883 & 32.52576\\\\\n",
       "\tCcp84Aa34 & bg & Ccp84Aa &  31 &   31 &  67.40467 &   0.6917341 & 5.316274 & 32.40813\\\\\n",
       "\tCG1252334 & bg & CG12523 & 159 &  560 & 313.04517 &  20.9281758 & 3.835513 & 31.81492\\\\\n",
       "\tCG1304334 & bg & CG13043 &  98 &  353 & 192.78959 &  10.4495332 & 4.073666 & 30.95313\\\\\n",
       "\tMuc91C34 & bg & Muc91C  & 433 & 2681 & 884.17446 & 100.0958347 & 3.128608 & 30.62850\\\\\n",
       "\tCG1306034 & bg & CG13060 &  62 &  143 & 113.49362 &   5.1268556 & 4.211321 & 28.80174\\\\\n",
       "\tCG1305934 & bg & CG13059 & 137 &  571 & 251.10833 &  20.7232531 & 3.530998 & 28.16995\\\\\n",
       "\tCcp84Ad34 & bg & Ccp84Ad &  99 &  432 & 202.30688 &  15.0565912 & 3.655308 & 28.02713\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Ccp84Ac34 | bg | Ccp84Ac | 475 | 1028 | 972.66793 |  38.9462062 | 4.605817 | 45.72326 |\n",
       "| CG754834 | bg | CG7548  |  94 |  110 | 192.96764 |   3.3584222 | 5.468409 | 41.55812 |\n",
       "| Cpr56F34 | bg | Cpr56F  | 329 |  960 | 646.56628 |  31.5248357 | 4.313185 | 40.28033 |\n",
       "| Cpr92F34 | bg | Cpr92F  | 113 |  164 | 237.90139 |   6.3026669 | 5.025796 | 39.70516 |\n",
       "| CG3446134 | bg | CG34461 | 328 | 1240 | 682.72649 |  44.2191448 | 3.916302 | 36.88090 |\n",
       "| CG507034 | bg | CG5070  |  88 |  145 | 172.43531 |   4.7561169 | 4.904816 | 36.48326 |\n",
       "| Cpr97Ea34 | bg | Cpr97Ea |  86 |  156 | 167.73288 |   5.1451985 | 4.770562 | 35.29547 |\n",
       "| CG926934 | bg | CG9269  |  72 |  107 | 141.84387 |   3.7619523 | 4.896607 | 35.05136 |\n",
       "| TwdlC34 | bg | TwdlC   | 319 | 1186 | 584.75983 |  43.3012171 | 3.722426 | 34.22460 |\n",
       "| Ccp84Ab34 | bg | Ccp84Ab |  63 |  176 | 138.08641 |   4.1299525 | 4.750482 | 33.82266 |\n",
       "| CG717934 | bg | CG7179  | 112 |  215 | 202.67024 |   8.8739713 | 4.359360 | 33.43669 |\n",
       "| CG1304146 | bg | CG13041 |  90 |  168 | 158.39488 |   5.8903057 | 4.522814 | 33.09099 |\n",
       "| Cpr62Bc34 | bg | Cpr62Bc | 239 |  966 | 461.47700 |  35.1577109 | 3.673883 | 32.52576 |\n",
       "| Ccp84Aa34 | bg | Ccp84Aa |  31 |   31 |  67.40467 |   0.6917341 | 5.316274 | 32.40813 |\n",
       "| CG1252334 | bg | CG12523 | 159 |  560 | 313.04517 |  20.9281758 | 3.835513 | 31.81492 |\n",
       "| CG1304334 | bg | CG13043 |  98 |  353 | 192.78959 |  10.4495332 | 4.073666 | 30.95313 |\n",
       "| Muc91C34 | bg | Muc91C  | 433 | 2681 | 884.17446 | 100.0958347 | 3.128608 | 30.62850 |\n",
       "| CG1306034 | bg | CG13060 |  62 |  143 | 113.49362 |   5.1268556 | 4.211321 | 28.80174 |\n",
       "| CG1305934 | bg | CG13059 | 137 |  571 | 251.10833 |  20.7232531 | 3.530998 | 28.16995 |\n",
       "| Ccp84Ad34 | bg | Ccp84Ad |  99 |  432 | 202.30688 |  15.0565912 | 3.655308 | 28.02713 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "Ccp84Ac34 bg          Ccp84Ac 475               1028        972.66793      \n",
       "CG754834  bg          CG7548   94                110        192.96764      \n",
       "Cpr56F34  bg          Cpr56F  329                960        646.56628      \n",
       "Cpr92F34  bg          Cpr92F  113                164        237.90139      \n",
       "CG3446134 bg          CG34461 328               1240        682.72649      \n",
       "CG507034  bg          CG5070   88                145        172.43531      \n",
       "Cpr97Ea34 bg          Cpr97Ea  86                156        167.73288      \n",
       "CG926934  bg          CG9269   72                107        141.84387      \n",
       "TwdlC34   bg          TwdlC   319               1186        584.75983      \n",
       "Ccp84Ab34 bg          Ccp84Ab  63                176        138.08641      \n",
       "CG717934  bg          CG7179  112                215        202.67024      \n",
       "CG1304146 bg          CG13041  90                168        158.39488      \n",
       "Cpr62Bc34 bg          Cpr62Bc 239                966        461.47700      \n",
       "Ccp84Aa34 bg          Ccp84Aa  31                 31         67.40467      \n",
       "CG1252334 bg          CG12523 159                560        313.04517      \n",
       "CG1304334 bg          CG13043  98                353        192.78959      \n",
       "Muc91C34  bg          Muc91C  433               2681        884.17446      \n",
       "CG1306034 bg          CG13060  62                143        113.49362      \n",
       "CG1305934 bg          CG13059 137                571        251.10833      \n",
       "Ccp84Ad34 bg          Ccp84Ad  99                432        202.30688      \n",
       "          other.tpm   log2.ratio heuristic.score\n",
       "Ccp84Ac34  38.9462062 4.605817   45.72326       \n",
       "CG754834    3.3584222 5.468409   41.55812       \n",
       "Cpr56F34   31.5248357 4.313185   40.28033       \n",
       "Cpr92F34    6.3026669 5.025796   39.70516       \n",
       "CG3446134  44.2191448 3.916302   36.88090       \n",
       "CG507034    4.7561169 4.904816   36.48326       \n",
       "Cpr97Ea34   5.1451985 4.770562   35.29547       \n",
       "CG926934    3.7619523 4.896607   35.05136       \n",
       "TwdlC34    43.3012171 3.722426   34.22460       \n",
       "Ccp84Ab34   4.1299525 4.750482   33.82266       \n",
       "CG717934    8.8739713 4.359360   33.43669       \n",
       "CG1304146   5.8903057 4.522814   33.09099       \n",
       "Cpr62Bc34  35.1577109 3.673883   32.52576       \n",
       "Ccp84Aa34   0.6917341 5.316274   32.40813       \n",
       "CG1252334  20.9281758 3.835513   31.81492       \n",
       "CG1304334  10.4495332 4.073666   30.95313       \n",
       "Muc91C34  100.0958347 3.128608   30.62850       \n",
       "CG1306034   5.1268556 4.211321   28.80174       \n",
       "CG1305934  20.7232531 3.530998   28.16995       \n",
       "Ccp84Ad34  15.0565912 3.655308   28.02713       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"bg\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 273,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Ccp84Ac\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Ccp84Ac\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Ccp84Ac\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG7548\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Cpr56F\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1573,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Pharynx_primordium_query\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Pharynx_primordium_query\", df_sub300$cluster %in% bh &\n",
    "                         df_sub300$umap_2 >= -.65*df_sub300$umap_1 -4.4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1577,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Tracheal_system_early\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Tracheal_system_early\", df_sub300$cluster %in% bh &\n",
    "        df_sub300$umap_2 < -.65*df_sub300$umap_1 -4.4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 274,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bh' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bh' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bh)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 275,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(72, 57)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1555,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>scro35</th><td>bh</td><td>scro   </td><td> 744</td><td> 1378</td><td> 862.6576</td><td>  60.78208</td><td>3.803528</td><td>37.10081</td></tr>\n",
       "\t<tr><th scope=row>btl35</th><td>bh</td><td>btl    </td><td> 450</td><td>  937</td><td> 505.3887</td><td>  37.92671</td><td>3.698561</td><td>33.22825</td></tr>\n",
       "\t<tr><th scope=row>knrl35</th><td>bh</td><td>knrl   </td><td>1603</td><td> 5043</td><td>1722.9310</td><td> 211.93745</td><td>3.016363</td><td>32.43039</td></tr>\n",
       "\t<tr><th scope=row>CG3244735</th><td>bh</td><td>CG32447</td><td> 923</td><td> 2460</td><td>1002.0158</td><td> 111.02878</td><td>3.160964</td><td>31.51522</td></tr>\n",
       "\t<tr><th scope=row>CG4274135</th><td>bh</td><td>CG42741</td><td> 563</td><td> 1708</td><td> 648.4632</td><td>  75.03016</td><td>3.092381</td><td>28.89244</td></tr>\n",
       "\t<tr><th scope=row>trh35</th><td>bh</td><td>trh    </td><td> 916</td><td> 4081</td><td> 995.5577</td><td> 163.61051</td><td>2.596448</td><td>25.86273</td></tr>\n",
       "\t<tr><th scope=row>ds35</th><td>bh</td><td>ds     </td><td>2748</td><td>16270</td><td>3066.7078</td><td> 683.64488</td><td>2.163263</td><td>25.05696</td></tr>\n",
       "\t<tr><th scope=row>dpy35</th><td>bh</td><td>dpy    </td><td>4564</td><td>28379</td><td>4978.7536</td><td>1209.65845</td><td>2.039993</td><td>25.05490</td></tr>\n",
       "\t<tr><th scope=row>Vsx135</th><td>bh</td><td>Vsx1   </td><td> 370</td><td> 1426</td><td> 422.2196</td><td>  59.81775</td><td>2.795429</td><td>24.39085</td></tr>\n",
       "\t<tr><th scope=row>CG1330035</th><td>bh</td><td>CG13300</td><td>1724</td><td> 9286</td><td>1883.6871</td><td> 401.89755</td><td>2.225074</td><td>24.20905</td></tr>\n",
       "\t<tr><th scope=row>Blimp-135</th><td>bh</td><td>Blimp-1</td><td> 879</td><td> 4696</td><td> 994.8391</td><td> 184.04076</td><td>2.426620</td><td>24.16858</td></tr>\n",
       "\t<tr><th scope=row>Ser39</th><td>bh</td><td>Ser    </td><td>1228</td><td> 6288</td><td>1374.5421</td><td> 280.75304</td><td>2.286448</td><td>23.83802</td></tr>\n",
       "\t<tr><th scope=row>stumps35</th><td>bh</td><td>stumps </td><td>1040</td><td> 5380</td><td>1140.0734</td><td> 223.88153</td><td>2.341890</td><td>23.78464</td></tr>\n",
       "\t<tr><th scope=row>heph35</th><td>bh</td><td>heph   </td><td>5021</td><td>35610</td><td>5616.6543</td><td>1494.86149</td><td>1.908734</td><td>23.77472</td></tr>\n",
       "\t<tr><th scope=row>CG4274735</th><td>bh</td><td>CG42747</td><td>1904</td><td>10508</td><td>2108.6305</td><td> 484.53255</td><td>2.118666</td><td>23.39596</td></tr>\n",
       "\t<tr><th scope=row>Spn43Aa35</th><td>bh</td><td>Spn43Aa</td><td>1318</td><td> 7646</td><td>1497.7549</td><td> 334.04804</td><td>2.160362</td><td>22.79084</td></tr>\n",
       "\t<tr><th scope=row>Pvf135</th><td>bh</td><td>Pvf1   </td><td> 346</td><td> 1440</td><td> 398.8190</td><td>  65.22387</td><td>2.590311</td><td>22.38859</td></tr>\n",
       "\t<tr><th scope=row>Reck35</th><td>bh</td><td>Reck   </td><td> 292</td><td> 1081</td><td> 303.7724</td><td>  47.32510</td><td>2.652146</td><td>21.88442</td></tr>\n",
       "\t<tr><th scope=row>Hs6st35</th><td>bh</td><td>Hs6st  </td><td>2178</td><td>13448</td><td>2364.0094</td><td> 616.34437</td><td>1.937088</td><td>21.71017</td></tr>\n",
       "\t<tr><th scope=row>verm35</th><td>bh</td><td>verm   </td><td>4322</td><td>35154</td><td>4838.3484</td><td>1448.27700</td><td>1.739181</td><td>21.28862</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tscro35 & bh & scro    &  744 &  1378 &  862.6576 &   60.78208 & 3.803528 & 37.10081\\\\\n",
       "\tbtl35 & bh & btl     &  450 &   937 &  505.3887 &   37.92671 & 3.698561 & 33.22825\\\\\n",
       "\tknrl35 & bh & knrl    & 1603 &  5043 & 1722.9310 &  211.93745 & 3.016363 & 32.43039\\\\\n",
       "\tCG3244735 & bh & CG32447 &  923 &  2460 & 1002.0158 &  111.02878 & 3.160964 & 31.51522\\\\\n",
       "\tCG4274135 & bh & CG42741 &  563 &  1708 &  648.4632 &   75.03016 & 3.092381 & 28.89244\\\\\n",
       "\ttrh35 & bh & trh     &  916 &  4081 &  995.5577 &  163.61051 & 2.596448 & 25.86273\\\\\n",
       "\tds35 & bh & ds      & 2748 & 16270 & 3066.7078 &  683.64488 & 2.163263 & 25.05696\\\\\n",
       "\tdpy35 & bh & dpy     & 4564 & 28379 & 4978.7536 & 1209.65845 & 2.039993 & 25.05490\\\\\n",
       "\tVsx135 & bh & Vsx1    &  370 &  1426 &  422.2196 &   59.81775 & 2.795429 & 24.39085\\\\\n",
       "\tCG1330035 & bh & CG13300 & 1724 &  9286 & 1883.6871 &  401.89755 & 2.225074 & 24.20905\\\\\n",
       "\tBlimp-135 & bh & Blimp-1 &  879 &  4696 &  994.8391 &  184.04076 & 2.426620 & 24.16858\\\\\n",
       "\tSer39 & bh & Ser     & 1228 &  6288 & 1374.5421 &  280.75304 & 2.286448 & 23.83802\\\\\n",
       "\tstumps35 & bh & stumps  & 1040 &  5380 & 1140.0734 &  223.88153 & 2.341890 & 23.78464\\\\\n",
       "\theph35 & bh & heph    & 5021 & 35610 & 5616.6543 & 1494.86149 & 1.908734 & 23.77472\\\\\n",
       "\tCG4274735 & bh & CG42747 & 1904 & 10508 & 2108.6305 &  484.53255 & 2.118666 & 23.39596\\\\\n",
       "\tSpn43Aa35 & bh & Spn43Aa & 1318 &  7646 & 1497.7549 &  334.04804 & 2.160362 & 22.79084\\\\\n",
       "\tPvf135 & bh & Pvf1    &  346 &  1440 &  398.8190 &   65.22387 & 2.590311 & 22.38859\\\\\n",
       "\tReck35 & bh & Reck    &  292 &  1081 &  303.7724 &   47.32510 & 2.652146 & 21.88442\\\\\n",
       "\tHs6st35 & bh & Hs6st   & 2178 & 13448 & 2364.0094 &  616.34437 & 1.937088 & 21.71017\\\\\n",
       "\tverm35 & bh & verm    & 4322 & 35154 & 4838.3484 & 1448.27700 & 1.739181 & 21.28862\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| scro35 | bh | scro    |  744 |  1378 |  862.6576 |   60.78208 | 3.803528 | 37.10081 |\n",
       "| btl35 | bh | btl     |  450 |   937 |  505.3887 |   37.92671 | 3.698561 | 33.22825 |\n",
       "| knrl35 | bh | knrl    | 1603 |  5043 | 1722.9310 |  211.93745 | 3.016363 | 32.43039 |\n",
       "| CG3244735 | bh | CG32447 |  923 |  2460 | 1002.0158 |  111.02878 | 3.160964 | 31.51522 |\n",
       "| CG4274135 | bh | CG42741 |  563 |  1708 |  648.4632 |   75.03016 | 3.092381 | 28.89244 |\n",
       "| trh35 | bh | trh     |  916 |  4081 |  995.5577 |  163.61051 | 2.596448 | 25.86273 |\n",
       "| ds35 | bh | ds      | 2748 | 16270 | 3066.7078 |  683.64488 | 2.163263 | 25.05696 |\n",
       "| dpy35 | bh | dpy     | 4564 | 28379 | 4978.7536 | 1209.65845 | 2.039993 | 25.05490 |\n",
       "| Vsx135 | bh | Vsx1    |  370 |  1426 |  422.2196 |   59.81775 | 2.795429 | 24.39085 |\n",
       "| CG1330035 | bh | CG13300 | 1724 |  9286 | 1883.6871 |  401.89755 | 2.225074 | 24.20905 |\n",
       "| Blimp-135 | bh | Blimp-1 |  879 |  4696 |  994.8391 |  184.04076 | 2.426620 | 24.16858 |\n",
       "| Ser39 | bh | Ser     | 1228 |  6288 | 1374.5421 |  280.75304 | 2.286448 | 23.83802 |\n",
       "| stumps35 | bh | stumps  | 1040 |  5380 | 1140.0734 |  223.88153 | 2.341890 | 23.78464 |\n",
       "| heph35 | bh | heph    | 5021 | 35610 | 5616.6543 | 1494.86149 | 1.908734 | 23.77472 |\n",
       "| CG4274735 | bh | CG42747 | 1904 | 10508 | 2108.6305 |  484.53255 | 2.118666 | 23.39596 |\n",
       "| Spn43Aa35 | bh | Spn43Aa | 1318 |  7646 | 1497.7549 |  334.04804 | 2.160362 | 22.79084 |\n",
       "| Pvf135 | bh | Pvf1    |  346 |  1440 |  398.8190 |   65.22387 | 2.590311 | 22.38859 |\n",
       "| Reck35 | bh | Reck    |  292 |  1081 |  303.7724 |   47.32510 | 2.652146 | 21.88442 |\n",
       "| Hs6st35 | bh | Hs6st   | 2178 | 13448 | 2364.0094 |  616.34437 | 1.937088 | 21.71017 |\n",
       "| verm35 | bh | verm    | 4322 | 35154 | 4838.3484 | 1448.27700 | 1.739181 | 21.28862 |\n",
       "\n"
      ],
      "text/plain": [
       "          cell.type.2 gene    cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "scro35    bh          scro     744               1378        862.6576      \n",
       "btl35     bh          btl      450                937        505.3887      \n",
       "knrl35    bh          knrl    1603               5043       1722.9310      \n",
       "CG3244735 bh          CG32447  923               2460       1002.0158      \n",
       "CG4274135 bh          CG42741  563               1708        648.4632      \n",
       "trh35     bh          trh      916               4081        995.5577      \n",
       "ds35      bh          ds      2748              16270       3066.7078      \n",
       "dpy35     bh          dpy     4564              28379       4978.7536      \n",
       "Vsx135    bh          Vsx1     370               1426        422.2196      \n",
       "CG1330035 bh          CG13300 1724               9286       1883.6871      \n",
       "Blimp-135 bh          Blimp-1  879               4696        994.8391      \n",
       "Ser39     bh          Ser     1228               6288       1374.5421      \n",
       "stumps35  bh          stumps  1040               5380       1140.0734      \n",
       "heph35    bh          heph    5021              35610       5616.6543      \n",
       "CG4274735 bh          CG42747 1904              10508       2108.6305      \n",
       "Spn43Aa35 bh          Spn43Aa 1318               7646       1497.7549      \n",
       "Pvf135    bh          Pvf1     346               1440        398.8190      \n",
       "Reck35    bh          Reck     292               1081        303.7724      \n",
       "Hs6st35   bh          Hs6st   2178              13448       2364.0094      \n",
       "verm35    bh          verm    4322              35154       4838.3484      \n",
       "          other.tpm  log2.ratio heuristic.score\n",
       "scro35      60.78208 3.803528   37.10081       \n",
       "btl35       37.92671 3.698561   33.22825       \n",
       "knrl35     211.93745 3.016363   32.43039       \n",
       "CG3244735  111.02878 3.160964   31.51522       \n",
       "CG4274135   75.03016 3.092381   28.89244       \n",
       "trh35      163.61051 2.596448   25.86273       \n",
       "ds35       683.64488 2.163263   25.05696       \n",
       "dpy35     1209.65845 2.039993   25.05490       \n",
       "Vsx135      59.81775 2.795429   24.39085       \n",
       "CG1330035  401.89755 2.225074   24.20905       \n",
       "Blimp-135  184.04076 2.426620   24.16858       \n",
       "Ser39      280.75304 2.286448   23.83802       \n",
       "stumps35   223.88153 2.341890   23.78464       \n",
       "heph35    1494.86149 1.908734   23.77472       \n",
       "CG4274735  484.53255 2.118666   23.39596       \n",
       "Spn43Aa35  334.04804 2.160362   22.79084       \n",
       "Pvf135      65.22387 2.590311   22.38859       \n",
       "Reck35      47.32510 2.652146   21.88442       \n",
       "Hs6st35    616.34437 1.937088   21.71017       \n",
       "verm35    1448.27700 1.739181   21.28862       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"bh\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 276,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"scro\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"scro\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"scro\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"btl\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"knrl\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 277,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bh' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bh' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bh)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= -.65*umap_1 -4.4), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -5) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -0.65, intercept = -4.4) +\n",
    "xlim(-4, 0) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1570,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 1371\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>scro</th><td>scro     </td><td> 783</td><td> 1437</td><td>1698.2191</td><td>  62.68135</td><td>Set 1</td><td>4.737008</td><td>50.83120</td></tr>\n",
       "\t<tr><th scope=row>CG42741</th><td>CG42741  </td><td> 606</td><td> 1711</td><td>1316.4031</td><td>  77.60279</td><td>Set 1</td><td>4.065877</td><td>42.13664</td></tr>\n",
       "\t<tr><th scope=row>Vsx1</th><td>Vsx1     </td><td> 401</td><td> 1508</td><td> 865.0783</td><td>  61.65082</td><td>Set 1</td><td>3.787425</td><td>36.95904</td></tr>\n",
       "\t<tr><th scope=row>Ser</th><td>Ser      </td><td>1145</td><td> 6547</td><td>2411.8339</td><td> 289.66732</td><td>Set 1</td><td>3.052690</td><td>34.30159</td></tr>\n",
       "\t<tr><th scope=row>sr</th><td>sr       </td><td> 504</td><td> 3868</td><td>1049.1706</td><td> 169.86971</td><td>Set 1</td><td>2.618281</td><td>26.27813</td></tr>\n",
       "\t<tr><th scope=row>ds</th><td>ds       </td><td>1559</td><td>16288</td><td>3303.1970</td><td> 701.84804</td><td>Set 1</td><td>2.232578</td><td>26.09903</td></tr>\n",
       "\t<tr><th scope=row>CG42747</th><td>CG42747  </td><td>1132</td><td>10810</td><td>2445.5860</td><td> 501.24278</td><td>Set 1</td><td>2.283723</td><td>25.70686</td></tr>\n",
       "\t<tr><th scope=row>CG13300</th><td>CG13300  </td><td> 936</td><td> 9532</td><td>1986.4469</td><td> 411.59973</td><td>Set 1</td><td>2.267375</td><td>24.84295</td></tr>\n",
       "\t<tr><th scope=row>Fas3</th><td>Fas3     </td><td>3793</td><td>50129</td><td>8052.7295</td><td>2235.61297</td><td>Set 1</td><td>1.848162</td><td>23.98072</td></tr>\n",
       "\t<tr><th scope=row>heph</th><td>heph     </td><td>2580</td><td>35793</td><td>5431.5734</td><td>1520.43138</td><td>Set 1</td><td>1.835941</td><td>22.77929</td></tr>\n",
       "\t<tr><th scope=row>Blimp-1</th><td>Blimp-1  </td><td> 431</td><td> 4777</td><td> 919.6360</td><td> 193.14945</td><td>Set 1</td><td>2.243895</td><td>22.09449</td></tr>\n",
       "\t<tr><th scope=row>shd</th><td>shd      </td><td>  60</td><td>  355</td><td> 129.3388</td><td>  14.00120</td><td>Set 1</td><td>3.108005</td><td>21.83723</td></tr>\n",
       "\t<tr><th scope=row>yellow-e3</th><td>yellow-e3</td><td> 165</td><td> 1577</td><td> 369.5044</td><td>  65.16284</td><td>Set 1</td><td>2.481498</td><td>21.17549</td></tr>\n",
       "\t<tr><th scope=row>mgl</th><td>mgl      </td><td>3385</td><td>52155</td><td>7213.5611</td><td>2301.26841</td><td>Set 1</td><td>1.647656</td><td>21.11750</td></tr>\n",
       "\t<tr><th scope=row>side-V</th><td>side-V   </td><td> 718</td><td> 8722</td><td>1527.7004</td><td> 386.04980</td><td>Set 1</td><td>1.980771</td><td>20.95277</td></tr>\n",
       "\t<tr><th scope=row>cnc</th><td>cnc      </td><td> 905</td><td>12563</td><td>1921.6234</td><td> 516.36366</td><td>Set 1</td><td>1.893075</td><td>20.65129</td></tr>\n",
       "\t<tr><th scope=row>fng</th><td>fng      </td><td> 873</td><td>10649</td><td>1855.1037</td><td> 509.63801</td><td>Set 1</td><td>1.861127</td><td>20.20823</td></tr>\n",
       "\t<tr><th scope=row>Gsc</th><td>Gsc      </td><td>  47</td><td>  339</td><td> 111.7421</td><td>  13.62518</td><td>Set 1</td><td>2.933647</td><td>19.99833</td></tr>\n",
       "\t<tr><th scope=row>Awh</th><td>Awh      </td><td> 301</td><td> 3129</td><td> 624.6466</td><td> 140.28792</td><td>Set 1</td><td>2.144402</td><td>19.91979</td></tr>\n",
       "\t<tr><th scope=row>CG32447</th><td>CG32447  </td><td> 282</td><td> 2702</td><td> 565.8295</td><td> 125.15490</td><td>Set 1</td><td>2.165171</td><td>19.80432</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tscro & scro      &  783 &  1437 & 1698.2191 &   62.68135 & Set 1 & 4.737008 & 50.83120\\\\\n",
       "\tCG42741 & CG42741   &  606 &  1711 & 1316.4031 &   77.60279 & Set 1 & 4.065877 & 42.13664\\\\\n",
       "\tVsx1 & Vsx1      &  401 &  1508 &  865.0783 &   61.65082 & Set 1 & 3.787425 & 36.95904\\\\\n",
       "\tSer & Ser       & 1145 &  6547 & 2411.8339 &  289.66732 & Set 1 & 3.052690 & 34.30159\\\\\n",
       "\tsr & sr        &  504 &  3868 & 1049.1706 &  169.86971 & Set 1 & 2.618281 & 26.27813\\\\\n",
       "\tds & ds        & 1559 & 16288 & 3303.1970 &  701.84804 & Set 1 & 2.232578 & 26.09903\\\\\n",
       "\tCG42747 & CG42747   & 1132 & 10810 & 2445.5860 &  501.24278 & Set 1 & 2.283723 & 25.70686\\\\\n",
       "\tCG13300 & CG13300   &  936 &  9532 & 1986.4469 &  411.59973 & Set 1 & 2.267375 & 24.84295\\\\\n",
       "\tFas3 & Fas3      & 3793 & 50129 & 8052.7295 & 2235.61297 & Set 1 & 1.848162 & 23.98072\\\\\n",
       "\theph & heph      & 2580 & 35793 & 5431.5734 & 1520.43138 & Set 1 & 1.835941 & 22.77929\\\\\n",
       "\tBlimp-1 & Blimp-1   &  431 &  4777 &  919.6360 &  193.14945 & Set 1 & 2.243895 & 22.09449\\\\\n",
       "\tshd & shd       &   60 &   355 &  129.3388 &   14.00120 & Set 1 & 3.108005 & 21.83723\\\\\n",
       "\tyellow-e3 & yellow-e3 &  165 &  1577 &  369.5044 &   65.16284 & Set 1 & 2.481498 & 21.17549\\\\\n",
       "\tmgl & mgl       & 3385 & 52155 & 7213.5611 & 2301.26841 & Set 1 & 1.647656 & 21.11750\\\\\n",
       "\tside-V & side-V    &  718 &  8722 & 1527.7004 &  386.04980 & Set 1 & 1.980771 & 20.95277\\\\\n",
       "\tcnc & cnc       &  905 & 12563 & 1921.6234 &  516.36366 & Set 1 & 1.893075 & 20.65129\\\\\n",
       "\tfng & fng       &  873 & 10649 & 1855.1037 &  509.63801 & Set 1 & 1.861127 & 20.20823\\\\\n",
       "\tGsc & Gsc       &   47 &   339 &  111.7421 &   13.62518 & Set 1 & 2.933647 & 19.99833\\\\\n",
       "\tAwh & Awh       &  301 &  3129 &  624.6466 &  140.28792 & Set 1 & 2.144402 & 19.91979\\\\\n",
       "\tCG32447 & CG32447   &  282 &  2702 &  565.8295 &  125.15490 & Set 1 & 2.165171 & 19.80432\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| scro | scro      |  783 |  1437 | 1698.2191 |   62.68135 | Set 1 | 4.737008 | 50.83120 |\n",
       "| CG42741 | CG42741   |  606 |  1711 | 1316.4031 |   77.60279 | Set 1 | 4.065877 | 42.13664 |\n",
       "| Vsx1 | Vsx1      |  401 |  1508 |  865.0783 |   61.65082 | Set 1 | 3.787425 | 36.95904 |\n",
       "| Ser | Ser       | 1145 |  6547 | 2411.8339 |  289.66732 | Set 1 | 3.052690 | 34.30159 |\n",
       "| sr | sr        |  504 |  3868 | 1049.1706 |  169.86971 | Set 1 | 2.618281 | 26.27813 |\n",
       "| ds | ds        | 1559 | 16288 | 3303.1970 |  701.84804 | Set 1 | 2.232578 | 26.09903 |\n",
       "| CG42747 | CG42747   | 1132 | 10810 | 2445.5860 |  501.24278 | Set 1 | 2.283723 | 25.70686 |\n",
       "| CG13300 | CG13300   |  936 |  9532 | 1986.4469 |  411.59973 | Set 1 | 2.267375 | 24.84295 |\n",
       "| Fas3 | Fas3      | 3793 | 50129 | 8052.7295 | 2235.61297 | Set 1 | 1.848162 | 23.98072 |\n",
       "| heph | heph      | 2580 | 35793 | 5431.5734 | 1520.43138 | Set 1 | 1.835941 | 22.77929 |\n",
       "| Blimp-1 | Blimp-1   |  431 |  4777 |  919.6360 |  193.14945 | Set 1 | 2.243895 | 22.09449 |\n",
       "| shd | shd       |   60 |   355 |  129.3388 |   14.00120 | Set 1 | 3.108005 | 21.83723 |\n",
       "| yellow-e3 | yellow-e3 |  165 |  1577 |  369.5044 |   65.16284 | Set 1 | 2.481498 | 21.17549 |\n",
       "| mgl | mgl       | 3385 | 52155 | 7213.5611 | 2301.26841 | Set 1 | 1.647656 | 21.11750 |\n",
       "| side-V | side-V    |  718 |  8722 | 1527.7004 |  386.04980 | Set 1 | 1.980771 | 20.95277 |\n",
       "| cnc | cnc       |  905 | 12563 | 1921.6234 |  516.36366 | Set 1 | 1.893075 | 20.65129 |\n",
       "| fng | fng       |  873 | 10649 | 1855.1037 |  509.63801 | Set 1 | 1.861127 | 20.20823 |\n",
       "| Gsc | Gsc       |   47 |   339 |  111.7421 |   13.62518 | Set 1 | 2.933647 | 19.99833 |\n",
       "| Awh | Awh       |  301 |  3129 |  624.6466 |  140.28792 | Set 1 | 2.144402 | 19.91979 |\n",
       "| CG32447 | CG32447   |  282 |  2702 |  565.8295 |  125.15490 | Set 1 | 2.165171 | 19.80432 |\n",
       "\n"
      ],
      "text/plain": [
       "          gene      set.1.n.umi set.2.n.umi set.1.tpm set.2.tpm  higher.expr\n",
       "scro      scro       783         1437       1698.2191   62.68135 Set 1      \n",
       "CG42741   CG42741    606         1711       1316.4031   77.60279 Set 1      \n",
       "Vsx1      Vsx1       401         1508        865.0783   61.65082 Set 1      \n",
       "Ser       Ser       1145         6547       2411.8339  289.66732 Set 1      \n",
       "sr        sr         504         3868       1049.1706  169.86971 Set 1      \n",
       "ds        ds        1559        16288       3303.1970  701.84804 Set 1      \n",
       "CG42747   CG42747   1132        10810       2445.5860  501.24278 Set 1      \n",
       "CG13300   CG13300    936         9532       1986.4469  411.59973 Set 1      \n",
       "Fas3      Fas3      3793        50129       8052.7295 2235.61297 Set 1      \n",
       "heph      heph      2580        35793       5431.5734 1520.43138 Set 1      \n",
       "Blimp-1   Blimp-1    431         4777        919.6360  193.14945 Set 1      \n",
       "shd       shd         60          355        129.3388   14.00120 Set 1      \n",
       "yellow-e3 yellow-e3  165         1577        369.5044   65.16284 Set 1      \n",
       "mgl       mgl       3385        52155       7213.5611 2301.26841 Set 1      \n",
       "side-V    side-V     718         8722       1527.7004  386.04980 Set 1      \n",
       "cnc       cnc        905        12563       1921.6234  516.36366 Set 1      \n",
       "fng       fng        873        10649       1855.1037  509.63801 Set 1      \n",
       "Gsc       Gsc         47          339        111.7421   13.62518 Set 1      \n",
       "Awh       Awh        301         3129        624.6466  140.28792 Set 1      \n",
       "CG32447   CG32447    282         2702        565.8295  125.15490 Set 1      \n",
       "          log2.ratio heuristic.score\n",
       "scro      4.737008   50.83120       \n",
       "CG42741   4.065877   42.13664       \n",
       "Vsx1      3.787425   36.95904       \n",
       "Ser       3.052690   34.30159       \n",
       "sr        2.618281   26.27813       \n",
       "ds        2.232578   26.09903       \n",
       "CG42747   2.283723   25.70686       \n",
       "CG13300   2.267375   24.84295       \n",
       "Fas3      1.848162   23.98072       \n",
       "heph      1.835941   22.77929       \n",
       "Blimp-1   2.243895   22.09449       \n",
       "shd       3.108005   21.83723       \n",
       "yellow-e3 2.481498   21.17549       \n",
       "mgl       1.647656   21.11750       \n",
       "side-V    1.980771   20.95277       \n",
       "cnc       1.893075   20.65129       \n",
       "fng       1.861127   20.20823       \n",
       "Gsc       2.933647   19.99833       \n",
       "Awh       2.144402   19.91979       \n",
       "CG32447   2.165171   19.80432       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "two.set.differential.gene.test.3a(cds_sub, \n",
    "                                  df_sub300$cluster %in% bh & \n",
    "                                  df_sub300$umap_2 >= -.65*df_sub300$umap_1 -4.4 , \n",
    "                                 !(df_sub300$cluster  %in% bh & \n",
    "                                  df_sub300$umap_2 >= -.65*df_sub300$umap_1 -4.4),\n",
    "                                  sample = 50000\n",
    "                                )  %>% filter(higher.expr == \"Set 1\")  %>% \n",
    "head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 278,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"scro\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"scro\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"scro\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG42741\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Vsx1\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 279,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Blimp-1\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Blimp-1\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Blimp-1\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1574,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 1797\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>btl</th><td>btl    </td><td> 528</td><td>  954</td><td> 842.55653</td><td>  38.76600</td><td>Set 1</td><td>4.405166</td><td>42.81971</td></tr>\n",
       "\t<tr><th scope=row>knrl</th><td>knrl   </td><td>1633</td><td> 5095</td><td>2512.32097</td><td> 219.67868</td><td>Set 1</td><td>3.509002</td><td>39.63550</td></tr>\n",
       "\t<tr><th scope=row>CG32447</th><td>CG32447</td><td> 908</td><td> 2600</td><td>1405.95938</td><td> 120.72412</td><td>Set 1</td><td>3.529868</td><td>36.91665</td></tr>\n",
       "\t<tr><th scope=row>trh</th><td>trh    </td><td>1004</td><td> 3895</td><td>1553.82627</td><td> 154.67893</td><td>Set 1</td><td>3.319180</td><td>35.19173</td></tr>\n",
       "\t<tr><th scope=row>Hs6st</th><td>Hs6st  </td><td>2439</td><td>12907</td><td>3797.48259</td><td> 598.95041</td><td>Set 1</td><td>2.662128</td><td>31.65592</td></tr>\n",
       "\t<tr><th scope=row>stumps</th><td>stumps </td><td>1157</td><td> 5616</td><td>1808.79742</td><td> 239.54276</td><td>Set 1</td><td>2.910666</td><td>31.49810</td></tr>\n",
       "\t<tr><th scope=row>dpy</th><td>dpy    </td><td>4206</td><td>27964</td><td>6513.30715</td><td>1214.66647</td><td>Set 1</td><td>2.421643</td><td>30.68075</td></tr>\n",
       "\t<tr><th scope=row>Reck</th><td>Reck   </td><td> 329</td><td> 1122</td><td> 490.36081</td><td>  48.32521</td><td>Set 1</td><td>3.313447</td><td>29.62433</td></tr>\n",
       "\t<tr><th scope=row>Pvf1</th><td>Pvf1   </td><td> 378</td><td> 1521</td><td> 618.78967</td><td>  69.56386</td><td>Set 1</td><td>3.132448</td><td>29.05544</td></tr>\n",
       "\t<tr><th scope=row>verm</th><td>verm   </td><td>4267</td><td>34335</td><td>6751.03609</td><td>1438.41617</td><td>Set 1</td><td>2.229625</td><td>28.36330</td></tr>\n",
       "\t<tr><th scope=row>Spn43Aa</th><td>Spn43Aa</td><td>1190</td><td> 7646</td><td>1870.37638</td><td> 339.83949</td><td>Set 1</td><td>2.456164</td><td>26.69822</td></tr>\n",
       "\t<tr><th scope=row>form3</th><td>form3  </td><td> 385</td><td> 1922</td><td> 625.48931</td><td>  85.44868</td><td>Set 1</td><td>2.855069</td><td>26.52687</td></tr>\n",
       "\t<tr><th scope=row>CG16959</th><td>CG16959</td><td> 147</td><td>  563</td><td> 232.87507</td><td>  23.17878</td><td>Set 1</td><td>3.267743</td><td>25.71581</td></tr>\n",
       "\t<tr><th scope=row>aop</th><td>aop    </td><td>2482</td><td>22651</td><td>3894.21581</td><td> 886.00921</td><td>Set 1</td><td>2.134312</td><td>25.45698</td></tr>\n",
       "\t<tr><th scope=row>Pura</th><td>Pura   </td><td> 668</td><td> 4131</td><td>1061.21442</td><td> 186.80941</td><td>Set 1</td><td>2.498375</td><td>25.11581</td></tr>\n",
       "\t<tr><th scope=row>rdo</th><td>rdo    </td><td> 346</td><td> 1526</td><td> 495.49368</td><td>  74.33770</td><td>Set 1</td><td>2.717423</td><td>24.33624</td></tr>\n",
       "\t<tr><th scope=row>sano</th><td>sano   </td><td>1266</td><td> 9750</td><td>1974.24804</td><td> 432.39271</td><td>Set 1</td><td>2.187556</td><td>23.94897</td></tr>\n",
       "\t<tr><th scope=row>Blimp-1</th><td>Blimp-1</td><td> 629</td><td> 4684</td><td>1001.48287</td><td> 191.35194</td><td>Set 1</td><td>2.380317</td><td>23.73025</td></tr>\n",
       "\t<tr><th scope=row>heph</th><td>heph   </td><td>3559</td><td>35088</td><td>5613.82245</td><td>1498.63823</td><td>Set 1</td><td>1.904369</td><td>23.71896</td></tr>\n",
       "\t<tr><th scope=row>CG14301</th><td>CG14301</td><td>  58</td><td>  155</td><td>  93.05248</td><td>   6.58735</td><td>Set 1</td><td>3.616377</td><td>23.70677</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tbtl & btl     &  528 &   954 &  842.55653 &   38.76600 & Set 1 & 4.405166 & 42.81971\\\\\n",
       "\tknrl & knrl    & 1633 &  5095 & 2512.32097 &  219.67868 & Set 1 & 3.509002 & 39.63550\\\\\n",
       "\tCG32447 & CG32447 &  908 &  2600 & 1405.95938 &  120.72412 & Set 1 & 3.529868 & 36.91665\\\\\n",
       "\ttrh & trh     & 1004 &  3895 & 1553.82627 &  154.67893 & Set 1 & 3.319180 & 35.19173\\\\\n",
       "\tHs6st & Hs6st   & 2439 & 12907 & 3797.48259 &  598.95041 & Set 1 & 2.662128 & 31.65592\\\\\n",
       "\tstumps & stumps  & 1157 &  5616 & 1808.79742 &  239.54276 & Set 1 & 2.910666 & 31.49810\\\\\n",
       "\tdpy & dpy     & 4206 & 27964 & 6513.30715 & 1214.66647 & Set 1 & 2.421643 & 30.68075\\\\\n",
       "\tReck & Reck    &  329 &  1122 &  490.36081 &   48.32521 & Set 1 & 3.313447 & 29.62433\\\\\n",
       "\tPvf1 & Pvf1    &  378 &  1521 &  618.78967 &   69.56386 & Set 1 & 3.132448 & 29.05544\\\\\n",
       "\tverm & verm    & 4267 & 34335 & 6751.03609 & 1438.41617 & Set 1 & 2.229625 & 28.36330\\\\\n",
       "\tSpn43Aa & Spn43Aa & 1190 &  7646 & 1870.37638 &  339.83949 & Set 1 & 2.456164 & 26.69822\\\\\n",
       "\tform3 & form3   &  385 &  1922 &  625.48931 &   85.44868 & Set 1 & 2.855069 & 26.52687\\\\\n",
       "\tCG16959 & CG16959 &  147 &   563 &  232.87507 &   23.17878 & Set 1 & 3.267743 & 25.71581\\\\\n",
       "\taop & aop     & 2482 & 22651 & 3894.21581 &  886.00921 & Set 1 & 2.134312 & 25.45698\\\\\n",
       "\tPura & Pura    &  668 &  4131 & 1061.21442 &  186.80941 & Set 1 & 2.498375 & 25.11581\\\\\n",
       "\trdo & rdo     &  346 &  1526 &  495.49368 &   74.33770 & Set 1 & 2.717423 & 24.33624\\\\\n",
       "\tsano & sano    & 1266 &  9750 & 1974.24804 &  432.39271 & Set 1 & 2.187556 & 23.94897\\\\\n",
       "\tBlimp-1 & Blimp-1 &  629 &  4684 & 1001.48287 &  191.35194 & Set 1 & 2.380317 & 23.73025\\\\\n",
       "\theph & heph    & 3559 & 35088 & 5613.82245 & 1498.63823 & Set 1 & 1.904369 & 23.71896\\\\\n",
       "\tCG14301 & CG14301 &   58 &   155 &   93.05248 &    6.58735 & Set 1 & 3.616377 & 23.70677\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| btl | btl     |  528 |   954 |  842.55653 |   38.76600 | Set 1 | 4.405166 | 42.81971 |\n",
       "| knrl | knrl    | 1633 |  5095 | 2512.32097 |  219.67868 | Set 1 | 3.509002 | 39.63550 |\n",
       "| CG32447 | CG32447 |  908 |  2600 | 1405.95938 |  120.72412 | Set 1 | 3.529868 | 36.91665 |\n",
       "| trh | trh     | 1004 |  3895 | 1553.82627 |  154.67893 | Set 1 | 3.319180 | 35.19173 |\n",
       "| Hs6st | Hs6st   | 2439 | 12907 | 3797.48259 |  598.95041 | Set 1 | 2.662128 | 31.65592 |\n",
       "| stumps | stumps  | 1157 |  5616 | 1808.79742 |  239.54276 | Set 1 | 2.910666 | 31.49810 |\n",
       "| dpy | dpy     | 4206 | 27964 | 6513.30715 | 1214.66647 | Set 1 | 2.421643 | 30.68075 |\n",
       "| Reck | Reck    |  329 |  1122 |  490.36081 |   48.32521 | Set 1 | 3.313447 | 29.62433 |\n",
       "| Pvf1 | Pvf1    |  378 |  1521 |  618.78967 |   69.56386 | Set 1 | 3.132448 | 29.05544 |\n",
       "| verm | verm    | 4267 | 34335 | 6751.03609 | 1438.41617 | Set 1 | 2.229625 | 28.36330 |\n",
       "| Spn43Aa | Spn43Aa | 1190 |  7646 | 1870.37638 |  339.83949 | Set 1 | 2.456164 | 26.69822 |\n",
       "| form3 | form3   |  385 |  1922 |  625.48931 |   85.44868 | Set 1 | 2.855069 | 26.52687 |\n",
       "| CG16959 | CG16959 |  147 |   563 |  232.87507 |   23.17878 | Set 1 | 3.267743 | 25.71581 |\n",
       "| aop | aop     | 2482 | 22651 | 3894.21581 |  886.00921 | Set 1 | 2.134312 | 25.45698 |\n",
       "| Pura | Pura    |  668 |  4131 | 1061.21442 |  186.80941 | Set 1 | 2.498375 | 25.11581 |\n",
       "| rdo | rdo     |  346 |  1526 |  495.49368 |   74.33770 | Set 1 | 2.717423 | 24.33624 |\n",
       "| sano | sano    | 1266 |  9750 | 1974.24804 |  432.39271 | Set 1 | 2.187556 | 23.94897 |\n",
       "| Blimp-1 | Blimp-1 |  629 |  4684 | 1001.48287 |  191.35194 | Set 1 | 2.380317 | 23.73025 |\n",
       "| heph | heph    | 3559 | 35088 | 5613.82245 | 1498.63823 | Set 1 | 1.904369 | 23.71896 |\n",
       "| CG14301 | CG14301 |   58 |   155 |   93.05248 |    6.58735 | Set 1 | 3.616377 | 23.70677 |\n",
       "\n"
      ],
      "text/plain": [
       "        gene    set.1.n.umi set.2.n.umi set.1.tpm  set.2.tpm  higher.expr\n",
       "btl     btl      528          954        842.55653   38.76600 Set 1      \n",
       "knrl    knrl    1633         5095       2512.32097  219.67868 Set 1      \n",
       "CG32447 CG32447  908         2600       1405.95938  120.72412 Set 1      \n",
       "trh     trh     1004         3895       1553.82627  154.67893 Set 1      \n",
       "Hs6st   Hs6st   2439        12907       3797.48259  598.95041 Set 1      \n",
       "stumps  stumps  1157         5616       1808.79742  239.54276 Set 1      \n",
       "dpy     dpy     4206        27964       6513.30715 1214.66647 Set 1      \n",
       "Reck    Reck     329         1122        490.36081   48.32521 Set 1      \n",
       "Pvf1    Pvf1     378         1521        618.78967   69.56386 Set 1      \n",
       "verm    verm    4267        34335       6751.03609 1438.41617 Set 1      \n",
       "Spn43Aa Spn43Aa 1190         7646       1870.37638  339.83949 Set 1      \n",
       "form3   form3    385         1922        625.48931   85.44868 Set 1      \n",
       "CG16959 CG16959  147          563        232.87507   23.17878 Set 1      \n",
       "aop     aop     2482        22651       3894.21581  886.00921 Set 1      \n",
       "Pura    Pura     668         4131       1061.21442  186.80941 Set 1      \n",
       "rdo     rdo      346         1526        495.49368   74.33770 Set 1      \n",
       "sano    sano    1266         9750       1974.24804  432.39271 Set 1      \n",
       "Blimp-1 Blimp-1  629         4684       1001.48287  191.35194 Set 1      \n",
       "heph    heph    3559        35088       5613.82245 1498.63823 Set 1      \n",
       "CG14301 CG14301   58          155         93.05248    6.58735 Set 1      \n",
       "        log2.ratio heuristic.score\n",
       "btl     4.405166   42.81971       \n",
       "knrl    3.509002   39.63550       \n",
       "CG32447 3.529868   36.91665       \n",
       "trh     3.319180   35.19173       \n",
       "Hs6st   2.662128   31.65592       \n",
       "stumps  2.910666   31.49810       \n",
       "dpy     2.421643   30.68075       \n",
       "Reck    3.313447   29.62433       \n",
       "Pvf1    3.132448   29.05544       \n",
       "verm    2.229625   28.36330       \n",
       "Spn43Aa 2.456164   26.69822       \n",
       "form3   2.855069   26.52687       \n",
       "CG16959 3.267743   25.71581       \n",
       "aop     2.134312   25.45698       \n",
       "Pura    2.498375   25.11581       \n",
       "rdo     2.717423   24.33624       \n",
       "sano    2.187556   23.94897       \n",
       "Blimp-1 2.380317   23.73025       \n",
       "heph    1.904369   23.71896       \n",
       "CG14301 3.616377   23.70677       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "two.set.differential.gene.test.3a(cds_sub, \n",
    "                                  df_sub300$cluster %in% bh & \n",
    "                                  df_sub300$umap_2 < -.65*df_sub300$umap_1 -4.4 , \n",
    "                                 !(df_sub300$cluster  %in% bh & \n",
    "                                  df_sub300$umap_2 < -.65*df_sub300$umap_1 -4.4),\n",
    "                                  sample = 50000\n",
    "                                )  %>% filter(higher.expr == \"Set 1\")  %>% \n",
    "head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 280,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"btl\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"btl\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"btl\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"knrl\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"trh\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1585,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Fat_body\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Fat_body\", df_sub300$cluster %in% bi)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 281,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bi' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bi' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bi)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 282,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bi' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bi' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bi)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1583,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG722736</th><td>bi</td><td>CG7227     </td><td> 565</td><td> 2622</td><td> 1864.2348</td><td> 126.64160</td><td>3.868413</td><td>42.03086</td></tr>\n",
       "\t<tr><th scope=row>Idgf236</th><td>bi</td><td>Idgf2      </td><td> 401</td><td> 1888</td><td> 1345.2300</td><td>  84.99897</td><td>3.967390</td><td>41.23986</td></tr>\n",
       "\t<tr><th scope=row>CG273636</th><td>bi</td><td>CG2736     </td><td> 198</td><td>  680</td><td>  663.5002</td><td>  32.25217</td><td>4.318576</td><td>40.49152</td></tr>\n",
       "\t<tr><th scope=row>apolpp36</th><td>bi</td><td>apolpp     </td><td>6182</td><td>78013</td><td>19901.0719</td><td>2830.11553</td><td>2.813404</td><td>40.17718</td></tr>\n",
       "\t<tr><th scope=row>NimB236</th><td>bi</td><td>NimB2      </td><td> 318</td><td> 1533</td><td> 1000.3929</td><td>  68.40678</td><td>3.849346</td><td>38.36949</td></tr>\n",
       "\t<tr><th scope=row>Gbp236</th><td>bi</td><td>Gbp2       </td><td> 132</td><td>  664</td><td>  438.7670</td><td>  29.49265</td><td>3.846922</td><td>33.77826</td></tr>\n",
       "\t<tr><th scope=row>Apoltp36</th><td>bi</td><td>Apoltp     </td><td> 391</td><td> 3532</td><td> 1308.3100</td><td> 139.51681</td><td>3.218890</td><td>33.33029</td></tr>\n",
       "\t<tr><th scope=row>Idgf436</th><td>bi</td><td>Idgf4      </td><td> 737</td><td> 7690</td><td> 2408.4207</td><td> 308.87167</td><td>2.958345</td><td>33.23544</td></tr>\n",
       "\t<tr><th scope=row>Tep436</th><td>bi</td><td>Tep4       </td><td> 391</td><td> 3192</td><td> 1325.0141</td><td> 142.78726</td><td>3.204000</td><td>33.23471</td></tr>\n",
       "\t<tr><th scope=row>teq36</th><td>bi</td><td>teq        </td><td> 199</td><td> 1264</td><td>  672.6837</td><td>  57.80424</td><td>3.515936</td><td>33.03548</td></tr>\n",
       "\t<tr><th scope=row>Glt36</th><td>bi</td><td>Glt        </td><td> 533</td><td> 4406</td><td> 1717.4489</td><td> 206.82979</td><td>3.046793</td><td>32.74355</td></tr>\n",
       "\t<tr><th scope=row>Plod36</th><td>bi</td><td>Plod       </td><td>1314</td><td>15008</td><td> 4242.6632</td><td> 678.18710</td><td>2.643089</td><td>31.85212</td></tr>\n",
       "\t<tr><th scope=row>fat-spondin36</th><td>bi</td><td>fat-spondin</td><td> 391</td><td> 3269</td><td> 1291.7786</td><td> 152.82007</td><td>3.070043</td><td>31.73276</td></tr>\n",
       "\t<tr><th scope=row>CG223336</th><td>bi</td><td>CG2233     </td><td> 104</td><td>  642</td><td>  370.9450</td><td>  28.22342</td><td>3.666008</td><td>31.30384</td></tr>\n",
       "\t<tr><th scope=row>CG324655</th><td>bi</td><td>CG3246     </td><td> 273</td><td> 2265</td><td>  908.5946</td><td> 100.38599</td><td>3.163778</td><td>31.09703</td></tr>\n",
       "\t<tr><th scope=row>Nepl1436</th><td>bi</td><td>Nepl14     </td><td>  64</td><td>  255</td><td>  214.1053</td><td>  12.29149</td><td>4.009746</td><td>31.07112</td></tr>\n",
       "\t<tr><th scope=row>CG993236</th><td>bi</td><td>CG9932     </td><td> 767</td><td> 9092</td><td> 2513.8753</td><td> 381.30433</td><td>2.717120</td><td>30.69332</td></tr>\n",
       "\t<tr><th scope=row>Hayan36</th><td>bi</td><td>Hayan      </td><td> 115</td><td>  665</td><td>  373.8048</td><td>  30.88011</td><td>3.551557</td><td>30.36579</td></tr>\n",
       "\t<tr><th scope=row>CG3252136</th><td>bi</td><td>CG32521    </td><td> 431</td><td> 4369</td><td> 1429.1724</td><td> 194.65483</td><td>2.868797</td><td>30.07066</td></tr>\n",
       "\t<tr><th scope=row>CG350536</th><td>bi</td><td>CG3505     </td><td>  71</td><td>  356</td><td>  235.8829</td><td>  16.33524</td><td>3.766291</td><td>29.70862</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG722736 & bi & CG7227      &  565 &  2622 &  1864.2348 &  126.64160 & 3.868413 & 42.03086\\\\\n",
       "\tIdgf236 & bi & Idgf2       &  401 &  1888 &  1345.2300 &   84.99897 & 3.967390 & 41.23986\\\\\n",
       "\tCG273636 & bi & CG2736      &  198 &   680 &   663.5002 &   32.25217 & 4.318576 & 40.49152\\\\\n",
       "\tapolpp36 & bi & apolpp      & 6182 & 78013 & 19901.0719 & 2830.11553 & 2.813404 & 40.17718\\\\\n",
       "\tNimB236 & bi & NimB2       &  318 &  1533 &  1000.3929 &   68.40678 & 3.849346 & 38.36949\\\\\n",
       "\tGbp236 & bi & Gbp2        &  132 &   664 &   438.7670 &   29.49265 & 3.846922 & 33.77826\\\\\n",
       "\tApoltp36 & bi & Apoltp      &  391 &  3532 &  1308.3100 &  139.51681 & 3.218890 & 33.33029\\\\\n",
       "\tIdgf436 & bi & Idgf4       &  737 &  7690 &  2408.4207 &  308.87167 & 2.958345 & 33.23544\\\\\n",
       "\tTep436 & bi & Tep4        &  391 &  3192 &  1325.0141 &  142.78726 & 3.204000 & 33.23471\\\\\n",
       "\tteq36 & bi & teq         &  199 &  1264 &   672.6837 &   57.80424 & 3.515936 & 33.03548\\\\\n",
       "\tGlt36 & bi & Glt         &  533 &  4406 &  1717.4489 &  206.82979 & 3.046793 & 32.74355\\\\\n",
       "\tPlod36 & bi & Plod        & 1314 & 15008 &  4242.6632 &  678.18710 & 2.643089 & 31.85212\\\\\n",
       "\tfat-spondin36 & bi & fat-spondin &  391 &  3269 &  1291.7786 &  152.82007 & 3.070043 & 31.73276\\\\\n",
       "\tCG223336 & bi & CG2233      &  104 &   642 &   370.9450 &   28.22342 & 3.666008 & 31.30384\\\\\n",
       "\tCG324655 & bi & CG3246      &  273 &  2265 &   908.5946 &  100.38599 & 3.163778 & 31.09703\\\\\n",
       "\tNepl1436 & bi & Nepl14      &   64 &   255 &   214.1053 &   12.29149 & 4.009746 & 31.07112\\\\\n",
       "\tCG993236 & bi & CG9932      &  767 &  9092 &  2513.8753 &  381.30433 & 2.717120 & 30.69332\\\\\n",
       "\tHayan36 & bi & Hayan       &  115 &   665 &   373.8048 &   30.88011 & 3.551557 & 30.36579\\\\\n",
       "\tCG3252136 & bi & CG32521     &  431 &  4369 &  1429.1724 &  194.65483 & 2.868797 & 30.07066\\\\\n",
       "\tCG350536 & bi & CG3505      &   71 &   356 &   235.8829 &   16.33524 & 3.766291 & 29.70862\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG722736 | bi | CG7227      |  565 |  2622 |  1864.2348 |  126.64160 | 3.868413 | 42.03086 |\n",
       "| Idgf236 | bi | Idgf2       |  401 |  1888 |  1345.2300 |   84.99897 | 3.967390 | 41.23986 |\n",
       "| CG273636 | bi | CG2736      |  198 |   680 |   663.5002 |   32.25217 | 4.318576 | 40.49152 |\n",
       "| apolpp36 | bi | apolpp      | 6182 | 78013 | 19901.0719 | 2830.11553 | 2.813404 | 40.17718 |\n",
       "| NimB236 | bi | NimB2       |  318 |  1533 |  1000.3929 |   68.40678 | 3.849346 | 38.36949 |\n",
       "| Gbp236 | bi | Gbp2        |  132 |   664 |   438.7670 |   29.49265 | 3.846922 | 33.77826 |\n",
       "| Apoltp36 | bi | Apoltp      |  391 |  3532 |  1308.3100 |  139.51681 | 3.218890 | 33.33029 |\n",
       "| Idgf436 | bi | Idgf4       |  737 |  7690 |  2408.4207 |  308.87167 | 2.958345 | 33.23544 |\n",
       "| Tep436 | bi | Tep4        |  391 |  3192 |  1325.0141 |  142.78726 | 3.204000 | 33.23471 |\n",
       "| teq36 | bi | teq         |  199 |  1264 |   672.6837 |   57.80424 | 3.515936 | 33.03548 |\n",
       "| Glt36 | bi | Glt         |  533 |  4406 |  1717.4489 |  206.82979 | 3.046793 | 32.74355 |\n",
       "| Plod36 | bi | Plod        | 1314 | 15008 |  4242.6632 |  678.18710 | 2.643089 | 31.85212 |\n",
       "| fat-spondin36 | bi | fat-spondin |  391 |  3269 |  1291.7786 |  152.82007 | 3.070043 | 31.73276 |\n",
       "| CG223336 | bi | CG2233      |  104 |   642 |   370.9450 |   28.22342 | 3.666008 | 31.30384 |\n",
       "| CG324655 | bi | CG3246      |  273 |  2265 |   908.5946 |  100.38599 | 3.163778 | 31.09703 |\n",
       "| Nepl1436 | bi | Nepl14      |   64 |   255 |   214.1053 |   12.29149 | 4.009746 | 31.07112 |\n",
       "| CG993236 | bi | CG9932      |  767 |  9092 |  2513.8753 |  381.30433 | 2.717120 | 30.69332 |\n",
       "| Hayan36 | bi | Hayan       |  115 |   665 |   373.8048 |   30.88011 | 3.551557 | 30.36579 |\n",
       "| CG3252136 | bi | CG32521     |  431 |  4369 |  1429.1724 |  194.65483 | 2.868797 | 30.07066 |\n",
       "| CG350536 | bi | CG3505      |   71 |   356 |   235.8829 |   16.33524 | 3.766291 | 29.70862 |\n",
       "\n"
      ],
      "text/plain": [
       "              cell.type.2 gene        cell.type.2.n.umi other.n.umi\n",
       "CG722736      bi          CG7227       565               2622      \n",
       "Idgf236       bi          Idgf2        401               1888      \n",
       "CG273636      bi          CG2736       198                680      \n",
       "apolpp36      bi          apolpp      6182              78013      \n",
       "NimB236       bi          NimB2        318               1533      \n",
       "Gbp236        bi          Gbp2         132                664      \n",
       "Apoltp36      bi          Apoltp       391               3532      \n",
       "Idgf436       bi          Idgf4        737               7690      \n",
       "Tep436        bi          Tep4         391               3192      \n",
       "teq36         bi          teq          199               1264      \n",
       "Glt36         bi          Glt          533               4406      \n",
       "Plod36        bi          Plod        1314              15008      \n",
       "fat-spondin36 bi          fat-spondin  391               3269      \n",
       "CG223336      bi          CG2233       104                642      \n",
       "CG324655      bi          CG3246       273               2265      \n",
       "Nepl1436      bi          Nepl14        64                255      \n",
       "CG993236      bi          CG9932       767               9092      \n",
       "Hayan36       bi          Hayan        115                665      \n",
       "CG3252136     bi          CG32521      431               4369      \n",
       "CG350536      bi          CG3505        71                356      \n",
       "              cell.type.2.tpm other.tpm  log2.ratio heuristic.score\n",
       "CG722736       1864.2348       126.64160 3.868413   42.03086       \n",
       "Idgf236        1345.2300        84.99897 3.967390   41.23986       \n",
       "CG273636        663.5002        32.25217 4.318576   40.49152       \n",
       "apolpp36      19901.0719      2830.11553 2.813404   40.17718       \n",
       "NimB236        1000.3929        68.40678 3.849346   38.36949       \n",
       "Gbp236          438.7670        29.49265 3.846922   33.77826       \n",
       "Apoltp36       1308.3100       139.51681 3.218890   33.33029       \n",
       "Idgf436        2408.4207       308.87167 2.958345   33.23544       \n",
       "Tep436         1325.0141       142.78726 3.204000   33.23471       \n",
       "teq36           672.6837        57.80424 3.515936   33.03548       \n",
       "Glt36          1717.4489       206.82979 3.046793   32.74355       \n",
       "Plod36         4242.6632       678.18710 2.643089   31.85212       \n",
       "fat-spondin36  1291.7786       152.82007 3.070043   31.73276       \n",
       "CG223336        370.9450        28.22342 3.666008   31.30384       \n",
       "CG324655        908.5946       100.38599 3.163778   31.09703       \n",
       "Nepl1436        214.1053        12.29149 4.009746   31.07112       \n",
       "CG993236       2513.8753       381.30433 2.717120   30.69332       \n",
       "Hayan36         373.8048        30.88011 3.551557   30.36579       \n",
       "CG3252136      1429.1724       194.65483 2.868797   30.07066       \n",
       "CG350536        235.8829        16.33524 3.766291   29.70862       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"bi\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 283,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG7227\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG7227\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG7227\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Idgf2\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"NimB2\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 284,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ab' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ab' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ab)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= 0.7*umap_1 -6.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -6.3) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    "xlim(3, 7) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1617,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Central_brain_ventral_nerve_cord_early\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Central_brain_ventral_nerve_cord_early\", df_sub300$cluster %in% bj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 285,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bj' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bj' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bj)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 286,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bj' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bj' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bj)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1612,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Sep537</th><td>bj</td><td>Sep5      </td><td>  594</td><td>  364</td><td> 280.20712</td><td>  11.69696</td><td>4.463938</td><td>36.31632</td></tr>\n",
       "\t<tr><th scope=row>kuk37</th><td>bj</td><td>kuk       </td><td> 3424</td><td> 4888</td><td>1587.27462</td><td> 162.53511</td><td>3.278879</td><td>34.86513</td></tr>\n",
       "\t<tr><th scope=row>CG1003537</th><td>bj</td><td>CG10035   </td><td> 1630</td><td> 1594</td><td> 764.50987</td><td>  63.12637</td><td>3.575545</td><td>34.25472</td></tr>\n",
       "\t<tr><th scope=row>CG1562837</th><td>bj</td><td>CG15628   </td><td> 1193</td><td> 1197</td><td> 531.53144</td><td>  44.94570</td><td>3.532153</td><td>31.98973</td></tr>\n",
       "\t<tr><th scope=row>dan37</th><td>bj</td><td>dan       </td><td>  738</td><td>  616</td><td> 342.95420</td><td>  23.69375</td><td>3.795798</td><td>31.98367</td></tr>\n",
       "\t<tr><th scope=row>Sr-CII37</th><td>bj</td><td>Sr-CII    </td><td>  700</td><td>  535</td><td> 343.03010</td><td>  23.80041</td><td>3.789899</td><td>31.93517</td></tr>\n",
       "\t<tr><th scope=row>CycB337</th><td>bj</td><td>CycB3     </td><td> 2333</td><td> 3815</td><td>1098.37610</td><td> 122.98774</td><td>3.147103</td><td>31.79351</td></tr>\n",
       "\t<tr><th scope=row>Top237</th><td>bj</td><td>Top2      </td><td> 2767</td><td> 4448</td><td>1262.64231</td><td> 150.62495</td><td>3.057867</td><td>31.50634</td></tr>\n",
       "\t<tr><th scope=row>borr37</th><td>bj</td><td>borr      </td><td> 1337</td><td> 1974</td><td> 639.46164</td><td>  64.35291</td><td>3.290534</td><td>30.67755</td></tr>\n",
       "\t<tr><th scope=row>slam37</th><td>bj</td><td>slam      </td><td>  346</td><td>  236</td><td> 158.19268</td><td>   7.81187</td><td>4.166091</td><td>30.47341</td></tr>\n",
       "\t<tr><th scope=row>Df3137</th><td>bj</td><td>Df31      </td><td> 9769</td><td>25282</td><td>4517.76258</td><td> 795.95232</td><td>2.503043</td><td>30.39123</td></tr>\n",
       "\t<tr><th scope=row>betaTub56D37</th><td>bj</td><td>betaTub56D</td><td>11846</td><td>33376</td><td>5545.13674</td><td>1084.63238</td><td>2.352687</td><td>29.26100</td></tr>\n",
       "\t<tr><th scope=row>CG1309637</th><td>bj</td><td>CG13096   </td><td> 1250</td><td> 2120</td><td> 582.81995</td><td>  71.13752</td><td>3.014229</td><td>27.69889</td></tr>\n",
       "\t<tr><th scope=row>Mapmodulin37</th><td>bj</td><td>Mapmodulin</td><td> 2728</td><td> 5890</td><td>1259.72295</td><td> 198.53626</td><td>2.658384</td><td>27.38145</td></tr>\n",
       "\t<tr><th scope=row>Mcm737</th><td>bj</td><td>Mcm7      </td><td> 1294</td><td> 2340</td><td> 604.46900</td><td>  78.18911</td><td>2.932294</td><td>27.10000</td></tr>\n",
       "\t<tr><th scope=row>fzy37</th><td>bj</td><td>fzy       </td><td>  767</td><td> 1326</td><td> 379.20310</td><td>  43.15605</td><td>3.102288</td><td>26.58855</td></tr>\n",
       "\t<tr><th scope=row>polo37</th><td>bj</td><td>polo      </td><td> 1220</td><td> 2309</td><td> 573.28198</td><td>  76.09624</td><td>2.894513</td><td>26.52999</td></tr>\n",
       "\t<tr><th scope=row>prd47</th><td>bj</td><td>prd       </td><td>  182</td><td>  120</td><td>  83.72168</td><td>   4.04938</td><td>4.051423</td><td>25.94798</td></tr>\n",
       "\t<tr><th scope=row>sti37</th><td>bj</td><td>sti       </td><td> 1196</td><td> 2393</td><td> 564.98322</td><td>  78.60352</td><td>2.827304</td><td>25.85461</td></tr>\n",
       "\t<tr><th scope=row>gt37</th><td>bj</td><td>gt        </td><td>  302</td><td>  276</td><td> 139.04847</td><td>  10.34364</td><td>3.615632</td><td>25.77867</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tSep537 & bj & Sep5       &   594 &   364 &  280.20712 &   11.69696 & 4.463938 & 36.31632\\\\\n",
       "\tkuk37 & bj & kuk        &  3424 &  4888 & 1587.27462 &  162.53511 & 3.278879 & 34.86513\\\\\n",
       "\tCG1003537 & bj & CG10035    &  1630 &  1594 &  764.50987 &   63.12637 & 3.575545 & 34.25472\\\\\n",
       "\tCG1562837 & bj & CG15628    &  1193 &  1197 &  531.53144 &   44.94570 & 3.532153 & 31.98973\\\\\n",
       "\tdan37 & bj & dan        &   738 &   616 &  342.95420 &   23.69375 & 3.795798 & 31.98367\\\\\n",
       "\tSr-CII37 & bj & Sr-CII     &   700 &   535 &  343.03010 &   23.80041 & 3.789899 & 31.93517\\\\\n",
       "\tCycB337 & bj & CycB3      &  2333 &  3815 & 1098.37610 &  122.98774 & 3.147103 & 31.79351\\\\\n",
       "\tTop237 & bj & Top2       &  2767 &  4448 & 1262.64231 &  150.62495 & 3.057867 & 31.50634\\\\\n",
       "\tborr37 & bj & borr       &  1337 &  1974 &  639.46164 &   64.35291 & 3.290534 & 30.67755\\\\\n",
       "\tslam37 & bj & slam       &   346 &   236 &  158.19268 &    7.81187 & 4.166091 & 30.47341\\\\\n",
       "\tDf3137 & bj & Df31       &  9769 & 25282 & 4517.76258 &  795.95232 & 2.503043 & 30.39123\\\\\n",
       "\tbetaTub56D37 & bj & betaTub56D & 11846 & 33376 & 5545.13674 & 1084.63238 & 2.352687 & 29.26100\\\\\n",
       "\tCG1309637 & bj & CG13096    &  1250 &  2120 &  582.81995 &   71.13752 & 3.014229 & 27.69889\\\\\n",
       "\tMapmodulin37 & bj & Mapmodulin &  2728 &  5890 & 1259.72295 &  198.53626 & 2.658384 & 27.38145\\\\\n",
       "\tMcm737 & bj & Mcm7       &  1294 &  2340 &  604.46900 &   78.18911 & 2.932294 & 27.10000\\\\\n",
       "\tfzy37 & bj & fzy        &   767 &  1326 &  379.20310 &   43.15605 & 3.102288 & 26.58855\\\\\n",
       "\tpolo37 & bj & polo       &  1220 &  2309 &  573.28198 &   76.09624 & 2.894513 & 26.52999\\\\\n",
       "\tprd47 & bj & prd        &   182 &   120 &   83.72168 &    4.04938 & 4.051423 & 25.94798\\\\\n",
       "\tsti37 & bj & sti        &  1196 &  2393 &  564.98322 &   78.60352 & 2.827304 & 25.85461\\\\\n",
       "\tgt37 & bj & gt         &   302 &   276 &  139.04847 &   10.34364 & 3.615632 & 25.77867\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Sep537 | bj | Sep5       |   594 |   364 |  280.20712 |   11.69696 | 4.463938 | 36.31632 |\n",
       "| kuk37 | bj | kuk        |  3424 |  4888 | 1587.27462 |  162.53511 | 3.278879 | 34.86513 |\n",
       "| CG1003537 | bj | CG10035    |  1630 |  1594 |  764.50987 |   63.12637 | 3.575545 | 34.25472 |\n",
       "| CG1562837 | bj | CG15628    |  1193 |  1197 |  531.53144 |   44.94570 | 3.532153 | 31.98973 |\n",
       "| dan37 | bj | dan        |   738 |   616 |  342.95420 |   23.69375 | 3.795798 | 31.98367 |\n",
       "| Sr-CII37 | bj | Sr-CII     |   700 |   535 |  343.03010 |   23.80041 | 3.789899 | 31.93517 |\n",
       "| CycB337 | bj | CycB3      |  2333 |  3815 | 1098.37610 |  122.98774 | 3.147103 | 31.79351 |\n",
       "| Top237 | bj | Top2       |  2767 |  4448 | 1262.64231 |  150.62495 | 3.057867 | 31.50634 |\n",
       "| borr37 | bj | borr       |  1337 |  1974 |  639.46164 |   64.35291 | 3.290534 | 30.67755 |\n",
       "| slam37 | bj | slam       |   346 |   236 |  158.19268 |    7.81187 | 4.166091 | 30.47341 |\n",
       "| Df3137 | bj | Df31       |  9769 | 25282 | 4517.76258 |  795.95232 | 2.503043 | 30.39123 |\n",
       "| betaTub56D37 | bj | betaTub56D | 11846 | 33376 | 5545.13674 | 1084.63238 | 2.352687 | 29.26100 |\n",
       "| CG1309637 | bj | CG13096    |  1250 |  2120 |  582.81995 |   71.13752 | 3.014229 | 27.69889 |\n",
       "| Mapmodulin37 | bj | Mapmodulin |  2728 |  5890 | 1259.72295 |  198.53626 | 2.658384 | 27.38145 |\n",
       "| Mcm737 | bj | Mcm7       |  1294 |  2340 |  604.46900 |   78.18911 | 2.932294 | 27.10000 |\n",
       "| fzy37 | bj | fzy        |   767 |  1326 |  379.20310 |   43.15605 | 3.102288 | 26.58855 |\n",
       "| polo37 | bj | polo       |  1220 |  2309 |  573.28198 |   76.09624 | 2.894513 | 26.52999 |\n",
       "| prd47 | bj | prd        |   182 |   120 |   83.72168 |    4.04938 | 4.051423 | 25.94798 |\n",
       "| sti37 | bj | sti        |  1196 |  2393 |  564.98322 |   78.60352 | 2.827304 | 25.85461 |\n",
       "| gt37 | bj | gt         |   302 |   276 |  139.04847 |   10.34364 | 3.615632 | 25.77867 |\n",
       "\n"
      ],
      "text/plain": [
       "             cell.type.2 gene       cell.type.2.n.umi other.n.umi\n",
       "Sep537       bj          Sep5         594               364      \n",
       "kuk37        bj          kuk         3424              4888      \n",
       "CG1003537    bj          CG10035     1630              1594      \n",
       "CG1562837    bj          CG15628     1193              1197      \n",
       "dan37        bj          dan          738               616      \n",
       "Sr-CII37     bj          Sr-CII       700               535      \n",
       "CycB337      bj          CycB3       2333              3815      \n",
       "Top237       bj          Top2        2767              4448      \n",
       "borr37       bj          borr        1337              1974      \n",
       "slam37       bj          slam         346               236      \n",
       "Df3137       bj          Df31        9769             25282      \n",
       "betaTub56D37 bj          betaTub56D 11846             33376      \n",
       "CG1309637    bj          CG13096     1250              2120      \n",
       "Mapmodulin37 bj          Mapmodulin  2728              5890      \n",
       "Mcm737       bj          Mcm7        1294              2340      \n",
       "fzy37        bj          fzy          767              1326      \n",
       "polo37       bj          polo        1220              2309      \n",
       "prd47        bj          prd          182               120      \n",
       "sti37        bj          sti         1196              2393      \n",
       "gt37         bj          gt           302               276      \n",
       "             cell.type.2.tpm other.tpm  log2.ratio heuristic.score\n",
       "Sep537        280.20712        11.69696 4.463938   36.31632       \n",
       "kuk37        1587.27462       162.53511 3.278879   34.86513       \n",
       "CG1003537     764.50987        63.12637 3.575545   34.25472       \n",
       "CG1562837     531.53144        44.94570 3.532153   31.98973       \n",
       "dan37         342.95420        23.69375 3.795798   31.98367       \n",
       "Sr-CII37      343.03010        23.80041 3.789899   31.93517       \n",
       "CycB337      1098.37610       122.98774 3.147103   31.79351       \n",
       "Top237       1262.64231       150.62495 3.057867   31.50634       \n",
       "borr37        639.46164        64.35291 3.290534   30.67755       \n",
       "slam37        158.19268         7.81187 4.166091   30.47341       \n",
       "Df3137       4517.76258       795.95232 2.503043   30.39123       \n",
       "betaTub56D37 5545.13674      1084.63238 2.352687   29.26100       \n",
       "CG1309637     582.81995        71.13752 3.014229   27.69889       \n",
       "Mapmodulin37 1259.72295       198.53626 2.658384   27.38145       \n",
       "Mcm737        604.46900        78.18911 2.932294   27.10000       \n",
       "fzy37         379.20310        43.15605 3.102288   26.58855       \n",
       "polo37        573.28198        76.09624 2.894513   26.52999       \n",
       "prd47          83.72168         4.04938 4.051423   25.94798       \n",
       "sti37         564.98322        78.60352 2.827304   25.85461       \n",
       "gt37          139.04847        10.34364 3.615632   25.77867       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"bj\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 287,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Sep5\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Sep5\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Sep5\"), \n",
    "                   size_pos = 2, size_neg = 0.5, stroke = 0.08, limit = 2)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"kuk\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 2)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG10035\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": false
   },
   "source": [
    "clus = c(bh)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= -.65*umap_1 -4.4), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -5) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -0.65, intercept = -4.4) +\n",
    "xlim(-4, 0) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1627,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Central_brain_ventral_nerve_cord_1\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Central_brain_ventral_nerve_cord_1\", df_sub300$cluster %in% bk)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 288,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bk' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bk' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bk)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 289,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bk' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bk' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bk)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1620,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Df3138</th><td>bk</td><td>Df31      </td><td>17794</td><td>23008</td><td>2804.4980</td><td> 816.34329</td><td>1.778729</td><td>20.37363</td></tr>\n",
       "\t<tr><th scope=row>stg48</th><td>bk</td><td>stg       </td><td> 8867</td><td>11266</td><td>1509.6917</td><td> 427.30331</td><td>1.817549</td><td>19.19513</td></tr>\n",
       "\t<tr><th scope=row>CG4223238</th><td>bk</td><td>CG42232   </td><td> 2999</td><td> 2907</td><td> 480.2010</td><td> 107.32405</td><td>2.148285</td><td>19.14228</td></tr>\n",
       "\t<tr><th scope=row>fax38</th><td>bk</td><td>fax       </td><td>41605</td><td>73599</td><td>7110.4074</td><td>2601.53893</td><td>1.450012</td><td>18.55424</td></tr>\n",
       "\t<tr><th scope=row>akirin38</th><td>bk</td><td>akirin    </td><td>10507</td><td>14399</td><td>1717.0873</td><td> 525.70268</td><td>1.704903</td><td>18.32189</td></tr>\n",
       "\t<tr><th scope=row>dap38</th><td>bk</td><td>dap       </td><td> 4165</td><td> 4629</td><td> 688.4098</td><td> 180.27197</td><td>1.925112</td><td>18.15230</td></tr>\n",
       "\t<tr><th scope=row>RhoGEF438</th><td>bk</td><td>RhoGEF4   </td><td> 1222</td><td> 1146</td><td> 212.0748</td><td>  42.83461</td><td>2.274431</td><td>17.59321</td></tr>\n",
       "\t<tr><th scope=row>CycB48</th><td>bk</td><td>CycB      </td><td> 5319</td><td> 6891</td><td> 852.7267</td><td> 254.10946</td><td>1.740967</td><td>16.95289</td></tr>\n",
       "\t<tr><th scope=row>Mapmodulin38</th><td>bk</td><td>Mapmodulin</td><td> 4285</td><td> 5446</td><td> 718.2133</td><td> 208.11045</td><td>1.780147</td><td>16.89409</td></tr>\n",
       "\t<tr><th scope=row>CycB338</th><td>bk</td><td>CycB3     </td><td> 3102</td><td> 3682</td><td> 501.0640</td><td> 138.17725</td><td>1.848071</td><td>16.58039</td></tr>\n",
       "\t<tr><th scope=row>CNBP38</th><td>bk</td><td>CNBP      </td><td>19743</td><td>34481</td><td>3333.0730</td><td>1250.28693</td><td>1.413440</td><td>16.54159</td></tr>\n",
       "\t<tr><th scope=row>Lam38</th><td>bk</td><td>Lam       </td><td> 7623</td><td>11726</td><td>1282.5611</td><td> 423.44777</td><td>1.595369</td><td>16.47367</td></tr>\n",
       "\t<tr><th scope=row>sti38</th><td>bk</td><td>sti       </td><td> 1995</td><td> 2301</td><td> 344.2905</td><td>  88.08789</td><td>1.950325</td><td>16.44449</td></tr>\n",
       "\t<tr><th scope=row>Ppa38</th><td>bk</td><td>Ppa       </td><td>10805</td><td>16508</td><td>1791.4458</td><td> 648.02587</td><td>1.464777</td><td>15.83089</td></tr>\n",
       "\t<tr><th scope=row>aqz38</th><td>bk</td><td>aqz       </td><td>32148</td><td>65730</td><td>5574.2756</td><td>2350.79928</td><td>1.245019</td><td>15.49405</td></tr>\n",
       "\t<tr><th scope=row>Amun38</th><td>bk</td><td>Amun      </td><td> 7042</td><td>11368</td><td>1192.2804</td><td> 418.36876</td><td>1.507432</td><td>15.40704</td></tr>\n",
       "\t<tr><th scope=row>feo38</th><td>bk</td><td>feo       </td><td> 1414</td><td> 1504</td><td> 221.8111</td><td>  55.72072</td><td>1.967384</td><td>15.34496</td></tr>\n",
       "\t<tr><th scope=row>l(3)neo3838</th><td>bk</td><td>l(3)neo38 </td><td>22667</td><td>40867</td><td>3825.3433</td><td>1571.88787</td><td>1.282173</td><td>15.26011</td></tr>\n",
       "\t<tr><th scope=row>Hrb27C38</th><td>bk</td><td>Hrb27C    </td><td>22166</td><td>42098</td><td>3783.0011</td><td>1554.66649</td><td>1.281998</td><td>15.23744</td></tr>\n",
       "\t<tr><th scope=row>D148</th><td>bk</td><td>D1        </td><td> 4299</td><td> 6858</td><td> 729.0853</td><td> 239.65715</td><td>1.599108</td><td>15.21059</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tDf3138 & bk & Df31       & 17794 & 23008 & 2804.4980 &  816.34329 & 1.778729 & 20.37363\\\\\n",
       "\tstg48 & bk & stg        &  8867 & 11266 & 1509.6917 &  427.30331 & 1.817549 & 19.19513\\\\\n",
       "\tCG4223238 & bk & CG42232    &  2999 &  2907 &  480.2010 &  107.32405 & 2.148285 & 19.14228\\\\\n",
       "\tfax38 & bk & fax        & 41605 & 73599 & 7110.4074 & 2601.53893 & 1.450012 & 18.55424\\\\\n",
       "\takirin38 & bk & akirin     & 10507 & 14399 & 1717.0873 &  525.70268 & 1.704903 & 18.32189\\\\\n",
       "\tdap38 & bk & dap        &  4165 &  4629 &  688.4098 &  180.27197 & 1.925112 & 18.15230\\\\\n",
       "\tRhoGEF438 & bk & RhoGEF4    &  1222 &  1146 &  212.0748 &   42.83461 & 2.274431 & 17.59321\\\\\n",
       "\tCycB48 & bk & CycB       &  5319 &  6891 &  852.7267 &  254.10946 & 1.740967 & 16.95289\\\\\n",
       "\tMapmodulin38 & bk & Mapmodulin &  4285 &  5446 &  718.2133 &  208.11045 & 1.780147 & 16.89409\\\\\n",
       "\tCycB338 & bk & CycB3      &  3102 &  3682 &  501.0640 &  138.17725 & 1.848071 & 16.58039\\\\\n",
       "\tCNBP38 & bk & CNBP       & 19743 & 34481 & 3333.0730 & 1250.28693 & 1.413440 & 16.54159\\\\\n",
       "\tLam38 & bk & Lam        &  7623 & 11726 & 1282.5611 &  423.44777 & 1.595369 & 16.47367\\\\\n",
       "\tsti38 & bk & sti        &  1995 &  2301 &  344.2905 &   88.08789 & 1.950325 & 16.44449\\\\\n",
       "\tPpa38 & bk & Ppa        & 10805 & 16508 & 1791.4458 &  648.02587 & 1.464777 & 15.83089\\\\\n",
       "\taqz38 & bk & aqz        & 32148 & 65730 & 5574.2756 & 2350.79928 & 1.245019 & 15.49405\\\\\n",
       "\tAmun38 & bk & Amun       &  7042 & 11368 & 1192.2804 &  418.36876 & 1.507432 & 15.40704\\\\\n",
       "\tfeo38 & bk & feo        &  1414 &  1504 &  221.8111 &   55.72072 & 1.967384 & 15.34496\\\\\n",
       "\tl(3)neo3838 & bk & l(3)neo38  & 22667 & 40867 & 3825.3433 & 1571.88787 & 1.282173 & 15.26011\\\\\n",
       "\tHrb27C38 & bk & Hrb27C     & 22166 & 42098 & 3783.0011 & 1554.66649 & 1.281998 & 15.23744\\\\\n",
       "\tD148 & bk & D1         &  4299 &  6858 &  729.0853 &  239.65715 & 1.599108 & 15.21059\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Df3138 | bk | Df31       | 17794 | 23008 | 2804.4980 |  816.34329 | 1.778729 | 20.37363 |\n",
       "| stg48 | bk | stg        |  8867 | 11266 | 1509.6917 |  427.30331 | 1.817549 | 19.19513 |\n",
       "| CG4223238 | bk | CG42232    |  2999 |  2907 |  480.2010 |  107.32405 | 2.148285 | 19.14228 |\n",
       "| fax38 | bk | fax        | 41605 | 73599 | 7110.4074 | 2601.53893 | 1.450012 | 18.55424 |\n",
       "| akirin38 | bk | akirin     | 10507 | 14399 | 1717.0873 |  525.70268 | 1.704903 | 18.32189 |\n",
       "| dap38 | bk | dap        |  4165 |  4629 |  688.4098 |  180.27197 | 1.925112 | 18.15230 |\n",
       "| RhoGEF438 | bk | RhoGEF4    |  1222 |  1146 |  212.0748 |   42.83461 | 2.274431 | 17.59321 |\n",
       "| CycB48 | bk | CycB       |  5319 |  6891 |  852.7267 |  254.10946 | 1.740967 | 16.95289 |\n",
       "| Mapmodulin38 | bk | Mapmodulin |  4285 |  5446 |  718.2133 |  208.11045 | 1.780147 | 16.89409 |\n",
       "| CycB338 | bk | CycB3      |  3102 |  3682 |  501.0640 |  138.17725 | 1.848071 | 16.58039 |\n",
       "| CNBP38 | bk | CNBP       | 19743 | 34481 | 3333.0730 | 1250.28693 | 1.413440 | 16.54159 |\n",
       "| Lam38 | bk | Lam        |  7623 | 11726 | 1282.5611 |  423.44777 | 1.595369 | 16.47367 |\n",
       "| sti38 | bk | sti        |  1995 |  2301 |  344.2905 |   88.08789 | 1.950325 | 16.44449 |\n",
       "| Ppa38 | bk | Ppa        | 10805 | 16508 | 1791.4458 |  648.02587 | 1.464777 | 15.83089 |\n",
       "| aqz38 | bk | aqz        | 32148 | 65730 | 5574.2756 | 2350.79928 | 1.245019 | 15.49405 |\n",
       "| Amun38 | bk | Amun       |  7042 | 11368 | 1192.2804 |  418.36876 | 1.507432 | 15.40704 |\n",
       "| feo38 | bk | feo        |  1414 |  1504 |  221.8111 |   55.72072 | 1.967384 | 15.34496 |\n",
       "| l(3)neo3838 | bk | l(3)neo38  | 22667 | 40867 | 3825.3433 | 1571.88787 | 1.282173 | 15.26011 |\n",
       "| Hrb27C38 | bk | Hrb27C     | 22166 | 42098 | 3783.0011 | 1554.66649 | 1.281998 | 15.23744 |\n",
       "| D148 | bk | D1         |  4299 |  6858 |  729.0853 |  239.65715 | 1.599108 | 15.21059 |\n",
       "\n"
      ],
      "text/plain": [
       "             cell.type.2 gene       cell.type.2.n.umi other.n.umi\n",
       "Df3138       bk          Df31       17794             23008      \n",
       "stg48        bk          stg         8867             11266      \n",
       "CG4223238    bk          CG42232     2999              2907      \n",
       "fax38        bk          fax        41605             73599      \n",
       "akirin38     bk          akirin     10507             14399      \n",
       "dap38        bk          dap         4165              4629      \n",
       "RhoGEF438    bk          RhoGEF4     1222              1146      \n",
       "CycB48       bk          CycB        5319              6891      \n",
       "Mapmodulin38 bk          Mapmodulin  4285              5446      \n",
       "CycB338      bk          CycB3       3102              3682      \n",
       "CNBP38       bk          CNBP       19743             34481      \n",
       "Lam38        bk          Lam         7623             11726      \n",
       "sti38        bk          sti         1995              2301      \n",
       "Ppa38        bk          Ppa        10805             16508      \n",
       "aqz38        bk          aqz        32148             65730      \n",
       "Amun38       bk          Amun        7042             11368      \n",
       "feo38        bk          feo         1414              1504      \n",
       "l(3)neo3838  bk          l(3)neo38  22667             40867      \n",
       "Hrb27C38     bk          Hrb27C     22166             42098      \n",
       "D148         bk          D1          4299              6858      \n",
       "             cell.type.2.tpm other.tpm  log2.ratio heuristic.score\n",
       "Df3138       2804.4980        816.34329 1.778729   20.37363       \n",
       "stg48        1509.6917        427.30331 1.817549   19.19513       \n",
       "CG4223238     480.2010        107.32405 2.148285   19.14228       \n",
       "fax38        7110.4074       2601.53893 1.450012   18.55424       \n",
       "akirin38     1717.0873        525.70268 1.704903   18.32189       \n",
       "dap38         688.4098        180.27197 1.925112   18.15230       \n",
       "RhoGEF438     212.0748         42.83461 2.274431   17.59321       \n",
       "CycB48        852.7267        254.10946 1.740967   16.95289       \n",
       "Mapmodulin38  718.2133        208.11045 1.780147   16.89409       \n",
       "CycB338       501.0640        138.17725 1.848071   16.58039       \n",
       "CNBP38       3333.0730       1250.28693 1.413440   16.54159       \n",
       "Lam38        1282.5611        423.44777 1.595369   16.47367       \n",
       "sti38         344.2905         88.08789 1.950325   16.44449       \n",
       "Ppa38        1791.4458        648.02587 1.464777   15.83089       \n",
       "aqz38        5574.2756       2350.79928 1.245019   15.49405       \n",
       "Amun38       1192.2804        418.36876 1.507432   15.40704       \n",
       "feo38         221.8111         55.72072 1.967384   15.34496       \n",
       "l(3)neo3838  3825.3433       1571.88787 1.282173   15.26011       \n",
       "Hrb27C38     3783.0011       1554.66649 1.281998   15.23744       \n",
       "D148          729.0853        239.65715 1.599108   15.21059       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"bk\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 290,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Df31\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Df31\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Df31\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"stg\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG42232\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": false
   },
   "source": [
    "clus = c(bh)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= -.65*umap_1 -4.4), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -5) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -0.65, intercept = -4.4) +\n",
    "xlim(-4, 0) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1628,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Central_brain_ventral_nerve_cord_2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Central_brain_ventral_nerve_cord_2\", df_sub300$cluster %in% bl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 291,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bl' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bl' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bl)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 292,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(72, 57)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1625,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Brd40</th><td>bl</td><td>Brd       </td><td> 1380</td><td>   766</td><td>  216.0523</td><td>  33.27688</td><td>2.656073</td><td>20.61617</td></tr>\n",
       "\t<tr><th scope=row>Tom42</th><td>bl</td><td>Tom       </td><td> 3864</td><td>  3910</td><td>  696.4962</td><td> 154.34260</td><td>2.164662</td><td>20.44749</td></tr>\n",
       "\t<tr><th scope=row>HmgD39</th><td>bl</td><td>HmgD      </td><td> 8396</td><td> 11087</td><td> 1500.5803</td><td> 396.69402</td><td>1.915790</td><td>20.21592</td></tr>\n",
       "\t<tr><th scope=row>RhoGEF439</th><td>bl</td><td>RhoGEF4   </td><td> 1225</td><td>  1217</td><td>  238.8639</td><td>  40.43466</td><td>2.527279</td><td>19.98085</td></tr>\n",
       "\t<tr><th scope=row>polo39</th><td>bl</td><td>polo      </td><td> 1751</td><td>  2043</td><td>  348.0008</td><td>  73.08635</td><td>2.231811</td><td>18.85230</td></tr>\n",
       "\t<tr><th scope=row>CycB339</th><td>bl</td><td>CycB3     </td><td> 2974</td><td>  3769</td><td>  551.2583</td><td> 131.36374</td><td>2.058221</td><td>18.74874</td></tr>\n",
       "\t<tr><th scope=row>CycB49</th><td>bl</td><td>CycB      </td><td> 4709</td><td>  7051</td><td>  924.7275</td><td> 248.33623</td><td>1.890936</td><td>18.63412</td></tr>\n",
       "\t<tr><th scope=row>eEF1alpha139</th><td>bl</td><td>eEF1alpha1</td><td>92257</td><td>214249</td><td>18176.4765</td><td>7328.17816</td><td>1.310349</td><td>18.54126</td></tr>\n",
       "\t<tr><th scope=row>borr39</th><td>bl</td><td>borr      </td><td> 1633</td><td>  1849</td><td>  307.8084</td><td>  65.13914</td><td>2.218456</td><td>18.34789</td></tr>\n",
       "\t<tr><th scope=row>Incenp39</th><td>bl</td><td>Incenp    </td><td> 1457</td><td>  1611</td><td>  279.0077</td><td>  58.18852</td><td>2.236916</td><td>18.18461</td></tr>\n",
       "\t<tr><th scope=row>Fib39</th><td>bl</td><td>Fib       </td><td> 5868</td><td>  9813</td><td> 1168.9732</td><td> 342.27249</td><td>1.767816</td><td>18.01804</td></tr>\n",
       "\t<tr><th scope=row>Oatp58Dc39</th><td>bl</td><td>Oatp58Dc  </td><td> 1541</td><td>  1922</td><td>  294.9961</td><td>  63.69694</td><td>2.188926</td><td>17.96985</td></tr>\n",
       "\t<tr><th scope=row>CG828939</th><td>bl</td><td>CG8289    </td><td> 2514</td><td>  3438</td><td>  497.3332</td><td> 123.61096</td><td>1.996782</td><td>17.89310</td></tr>\n",
       "\t<tr><th scope=row>Rcc139</th><td>bl</td><td>Rcc1      </td><td> 2483</td><td>  3503</td><td>  482.0764</td><td> 119.98725</td><td>1.994407</td><td>17.78235</td></tr>\n",
       "\t<tr><th scope=row>tum39</th><td>bl</td><td>tum       </td><td> 1176</td><td>  1471</td><td>  234.5421</td><td>  48.44596</td><td>2.245922</td><td>17.69751</td></tr>\n",
       "\t<tr><th scope=row>sti39</th><td>bl</td><td>sti       </td><td> 1816</td><td>  2342</td><td>  353.7287</td><td>  82.83981</td><td>2.076936</td><td>17.59284</td></tr>\n",
       "\t<tr><th scope=row>lmd39</th><td>bl</td><td>lmd       </td><td> 2080</td><td>  2839</td><td>  514.3450</td><td> 132.31298</td><td>1.947919</td><td>17.54957</td></tr>\n",
       "\t<tr><th scope=row>fzy39</th><td>bl</td><td>fzy       </td><td> 1162</td><td>  1196</td><td>  217.0809</td><td>  44.65628</td><td>2.249347</td><td>17.47455</td></tr>\n",
       "\t<tr><th scope=row>geminin39</th><td>bl</td><td>geminin   </td><td> 1242</td><td>  1456</td><td>  231.2830</td><td>  48.82646</td><td>2.214675</td><td>17.40677</td></tr>\n",
       "\t<tr><th scope=row>Df3139</th><td>bl</td><td>Df31      </td><td>12836</td><td> 24343</td><td> 2378.3994</td><td> 810.56638</td><td>1.551210</td><td>17.39896</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tBrd40 & bl & Brd        &  1380 &    766 &   216.0523 &   33.27688 & 2.656073 & 20.61617\\\\\n",
       "\tTom42 & bl & Tom        &  3864 &   3910 &   696.4962 &  154.34260 & 2.164662 & 20.44749\\\\\n",
       "\tHmgD39 & bl & HmgD       &  8396 &  11087 &  1500.5803 &  396.69402 & 1.915790 & 20.21592\\\\\n",
       "\tRhoGEF439 & bl & RhoGEF4    &  1225 &   1217 &   238.8639 &   40.43466 & 2.527279 & 19.98085\\\\\n",
       "\tpolo39 & bl & polo       &  1751 &   2043 &   348.0008 &   73.08635 & 2.231811 & 18.85230\\\\\n",
       "\tCycB339 & bl & CycB3      &  2974 &   3769 &   551.2583 &  131.36374 & 2.058221 & 18.74874\\\\\n",
       "\tCycB49 & bl & CycB       &  4709 &   7051 &   924.7275 &  248.33623 & 1.890936 & 18.63412\\\\\n",
       "\teEF1alpha139 & bl & eEF1alpha1 & 92257 & 214249 & 18176.4765 & 7328.17816 & 1.310349 & 18.54126\\\\\n",
       "\tborr39 & bl & borr       &  1633 &   1849 &   307.8084 &   65.13914 & 2.218456 & 18.34789\\\\\n",
       "\tIncenp39 & bl & Incenp     &  1457 &   1611 &   279.0077 &   58.18852 & 2.236916 & 18.18461\\\\\n",
       "\tFib39 & bl & Fib        &  5868 &   9813 &  1168.9732 &  342.27249 & 1.767816 & 18.01804\\\\\n",
       "\tOatp58Dc39 & bl & Oatp58Dc   &  1541 &   1922 &   294.9961 &   63.69694 & 2.188926 & 17.96985\\\\\n",
       "\tCG828939 & bl & CG8289     &  2514 &   3438 &   497.3332 &  123.61096 & 1.996782 & 17.89310\\\\\n",
       "\tRcc139 & bl & Rcc1       &  2483 &   3503 &   482.0764 &  119.98725 & 1.994407 & 17.78235\\\\\n",
       "\ttum39 & bl & tum        &  1176 &   1471 &   234.5421 &   48.44596 & 2.245922 & 17.69751\\\\\n",
       "\tsti39 & bl & sti        &  1816 &   2342 &   353.7287 &   82.83981 & 2.076936 & 17.59284\\\\\n",
       "\tlmd39 & bl & lmd        &  2080 &   2839 &   514.3450 &  132.31298 & 1.947919 & 17.54957\\\\\n",
       "\tfzy39 & bl & fzy        &  1162 &   1196 &   217.0809 &   44.65628 & 2.249347 & 17.47455\\\\\n",
       "\tgeminin39 & bl & geminin    &  1242 &   1456 &   231.2830 &   48.82646 & 2.214675 & 17.40677\\\\\n",
       "\tDf3139 & bl & Df31       & 12836 &  24343 &  2378.3994 &  810.56638 & 1.551210 & 17.39896\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Brd40 | bl | Brd        |  1380 |    766 |   216.0523 |   33.27688 | 2.656073 | 20.61617 |\n",
       "| Tom42 | bl | Tom        |  3864 |   3910 |   696.4962 |  154.34260 | 2.164662 | 20.44749 |\n",
       "| HmgD39 | bl | HmgD       |  8396 |  11087 |  1500.5803 |  396.69402 | 1.915790 | 20.21592 |\n",
       "| RhoGEF439 | bl | RhoGEF4    |  1225 |   1217 |   238.8639 |   40.43466 | 2.527279 | 19.98085 |\n",
       "| polo39 | bl | polo       |  1751 |   2043 |   348.0008 |   73.08635 | 2.231811 | 18.85230 |\n",
       "| CycB339 | bl | CycB3      |  2974 |   3769 |   551.2583 |  131.36374 | 2.058221 | 18.74874 |\n",
       "| CycB49 | bl | CycB       |  4709 |   7051 |   924.7275 |  248.33623 | 1.890936 | 18.63412 |\n",
       "| eEF1alpha139 | bl | eEF1alpha1 | 92257 | 214249 | 18176.4765 | 7328.17816 | 1.310349 | 18.54126 |\n",
       "| borr39 | bl | borr       |  1633 |   1849 |   307.8084 |   65.13914 | 2.218456 | 18.34789 |\n",
       "| Incenp39 | bl | Incenp     |  1457 |   1611 |   279.0077 |   58.18852 | 2.236916 | 18.18461 |\n",
       "| Fib39 | bl | Fib        |  5868 |   9813 |  1168.9732 |  342.27249 | 1.767816 | 18.01804 |\n",
       "| Oatp58Dc39 | bl | Oatp58Dc   |  1541 |   1922 |   294.9961 |   63.69694 | 2.188926 | 17.96985 |\n",
       "| CG828939 | bl | CG8289     |  2514 |   3438 |   497.3332 |  123.61096 | 1.996782 | 17.89310 |\n",
       "| Rcc139 | bl | Rcc1       |  2483 |   3503 |   482.0764 |  119.98725 | 1.994407 | 17.78235 |\n",
       "| tum39 | bl | tum        |  1176 |   1471 |   234.5421 |   48.44596 | 2.245922 | 17.69751 |\n",
       "| sti39 | bl | sti        |  1816 |   2342 |   353.7287 |   82.83981 | 2.076936 | 17.59284 |\n",
       "| lmd39 | bl | lmd        |  2080 |   2839 |   514.3450 |  132.31298 | 1.947919 | 17.54957 |\n",
       "| fzy39 | bl | fzy        |  1162 |   1196 |   217.0809 |   44.65628 | 2.249347 | 17.47455 |\n",
       "| geminin39 | bl | geminin    |  1242 |   1456 |   231.2830 |   48.82646 | 2.214675 | 17.40677 |\n",
       "| Df3139 | bl | Df31       | 12836 |  24343 |  2378.3994 |  810.56638 | 1.551210 | 17.39896 |\n",
       "\n"
      ],
      "text/plain": [
       "             cell.type.2 gene       cell.type.2.n.umi other.n.umi\n",
       "Brd40        bl          Brd         1380                766     \n",
       "Tom42        bl          Tom         3864               3910     \n",
       "HmgD39       bl          HmgD        8396              11087     \n",
       "RhoGEF439    bl          RhoGEF4     1225               1217     \n",
       "polo39       bl          polo        1751               2043     \n",
       "CycB339      bl          CycB3       2974               3769     \n",
       "CycB49       bl          CycB        4709               7051     \n",
       "eEF1alpha139 bl          eEF1alpha1 92257             214249     \n",
       "borr39       bl          borr        1633               1849     \n",
       "Incenp39     bl          Incenp      1457               1611     \n",
       "Fib39        bl          Fib         5868               9813     \n",
       "Oatp58Dc39   bl          Oatp58Dc    1541               1922     \n",
       "CG828939     bl          CG8289      2514               3438     \n",
       "Rcc139       bl          Rcc1        2483               3503     \n",
       "tum39        bl          tum         1176               1471     \n",
       "sti39        bl          sti         1816               2342     \n",
       "lmd39        bl          lmd         2080               2839     \n",
       "fzy39        bl          fzy         1162               1196     \n",
       "geminin39    bl          geminin     1242               1456     \n",
       "Df3139       bl          Df31       12836              24343     \n",
       "             cell.type.2.tpm other.tpm  log2.ratio heuristic.score\n",
       "Brd40          216.0523        33.27688 2.656073   20.61617       \n",
       "Tom42          696.4962       154.34260 2.164662   20.44749       \n",
       "HmgD39        1500.5803       396.69402 1.915790   20.21592       \n",
       "RhoGEF439      238.8639        40.43466 2.527279   19.98085       \n",
       "polo39         348.0008        73.08635 2.231811   18.85230       \n",
       "CycB339        551.2583       131.36374 2.058221   18.74874       \n",
       "CycB49         924.7275       248.33623 1.890936   18.63412       \n",
       "eEF1alpha139 18176.4765      7328.17816 1.310349   18.54126       \n",
       "borr39         307.8084        65.13914 2.218456   18.34789       \n",
       "Incenp39       279.0077        58.18852 2.236916   18.18461       \n",
       "Fib39         1168.9732       342.27249 1.767816   18.01804       \n",
       "Oatp58Dc39     294.9961        63.69694 2.188926   17.96985       \n",
       "CG828939       497.3332       123.61096 1.996782   17.89310       \n",
       "Rcc139         482.0764       119.98725 1.994407   17.78235       \n",
       "tum39          234.5421        48.44596 2.245922   17.69751       \n",
       "sti39          353.7287        82.83981 2.076936   17.59284       \n",
       "lmd39          514.3450       132.31298 1.947919   17.54957       \n",
       "fzy39          217.0809        44.65628 2.249347   17.47455       \n",
       "geminin39      231.2830        48.82646 2.214675   17.40677       \n",
       "Df3139        2378.3994       810.56638 1.551210   17.39896       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"bl\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 293,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Brd\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Brd\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Brd\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Tom\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"HmgD\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 294,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bh' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bh' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bh)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= -.65*umap_1 -4.4), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -5) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -0.65, intercept = -4.4) +\n",
    "xlim(-4, 0) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1632,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Undifferentiated_1\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Undifferentiated_1\", df_sub300$cluster %in% bm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 295,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bm' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bm' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bm)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 296,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bm' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bm' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bm)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1629,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>eEF240</th><td>bm</td><td>eEF2         </td><td>15933</td><td> 82767</td><td> 8084.7580</td><td>2731.34968</td><td>1.565062</td><td>20.31634</td></tr>\n",
       "\t<tr><th scope=row>betaTub60D40</th><td>bm</td><td>betaTub60D   </td><td> 3426</td><td> 14535</td><td> 1723.4942</td><td> 515.49430</td><td>1.738512</td><td>18.69241</td></tr>\n",
       "\t<tr><th scope=row>eEF1alpha140</th><td>bm</td><td>eEF1alpha1   </td><td>37214</td><td>223504</td><td>18278.3932</td><td>7565.11548</td><td>1.272515</td><td>18.01617</td></tr>\n",
       "\t<tr><th scope=row>Hsc70-440</th><td>bm</td><td>Hsc70-4      </td><td> 9909</td><td> 55407</td><td> 4957.9620</td><td>1859.00434</td><td>1.414441</td><td>17.36343</td></tr>\n",
       "\t<tr><th scope=row>Non140</th><td>bm</td><td>Non1         </td><td> 4092</td><td> 21259</td><td> 1997.7208</td><td> 709.18214</td><td>1.492094</td><td>16.36060</td></tr>\n",
       "\t<tr><th scope=row>RpL450</th><td>bm</td><td>RpL4         </td><td> 7051</td><td> 40709</td><td> 3499.0902</td><td>1348.69270</td><td>1.374349</td><td>16.18045</td></tr>\n",
       "\t<tr><th scope=row>eIF3c40</th><td>bm</td><td>eIF3c        </td><td> 1468</td><td>  7093</td><td>  750.8986</td><td> 231.45128</td><td>1.691690</td><td>16.16307</td></tr>\n",
       "\t<tr><th scope=row>RpL7A40</th><td>bm</td><td>RpL7A        </td><td>10554</td><td> 64427</td><td> 5159.1336</td><td>2085.48360</td><td>1.306055</td><td>16.10783</td></tr>\n",
       "\t<tr><th scope=row>eIF4A40</th><td>bm</td><td>eIF4A        </td><td> 7323</td><td> 44396</td><td> 3589.2292</td><td>1455.22444</td><td>1.301441</td><td>15.36984</td></tr>\n",
       "\t<tr><th scope=row>RpL378</th><td>bm</td><td>RpL3         </td><td> 9724</td><td> 59841</td><td> 4780.5943</td><td>1999.71287</td><td>1.256676</td><td>15.36070</td></tr>\n",
       "\t<tr><th scope=row>CG436443</th><td>bm</td><td>CG4364       </td><td> 1265</td><td>  6614</td><td>  661.1250</td><td> 218.19321</td><td>1.592720</td><td>14.92532</td></tr>\n",
       "\t<tr><th scope=row>SerRS40</th><td>bm</td><td>SerRS        </td><td>  442</td><td>  1963</td><td>  240.3248</td><td>  64.45107</td><td>1.876497</td><td>14.85216</td></tr>\n",
       "\t<tr><th scope=row>Prx2540-140</th><td>bm</td><td>Prx2540-1    </td><td>  603</td><td>  2587</td><td>  309.4307</td><td>  88.42007</td><td>1.790946</td><td>14.82568</td></tr>\n",
       "\t<tr><th scope=row>eEF1gamma40</th><td>bm</td><td>eEF1gamma    </td><td> 5075</td><td> 29648</td><td> 2499.2167</td><td>1005.89346</td><td>1.311565</td><td>14.80473</td></tr>\n",
       "\t<tr><th scope=row>eIF3a40</th><td>bm</td><td>eIF3a        </td><td> 2073</td><td> 11000</td><td> 1016.9136</td><td> 369.91736</td><td>1.455027</td><td>14.53776</td></tr>\n",
       "\t<tr><th scope=row>Neurochondrin40</th><td>bm</td><td>Neurochondrin</td><td>  961</td><td>  5001</td><td>  505.8119</td><td> 165.04718</td><td>1.607008</td><td>14.43946</td></tr>\n",
       "\t<tr><th scope=row>Fib40</th><td>bm</td><td>Fib          </td><td> 1983</td><td> 10602</td><td>  980.1815</td><td> 358.38971</td><td>1.447500</td><td>14.38580</td></tr>\n",
       "\t<tr><th scope=row>CCT140</th><td>bm</td><td>CCT1         </td><td>  702</td><td>  3535</td><td>  389.5810</td><td> 121.53030</td><td>1.668785</td><td>14.36736</td></tr>\n",
       "\t<tr><th scope=row>alphaTub84B40</th><td>bm</td><td>alphaTub84B  </td><td> 6958</td><td> 43878</td><td> 3517.5141</td><td>1511.35416</td><td>1.217760</td><td>14.34613</td></tr>\n",
       "\t<tr><th scope=row>RpL640</th><td>bm</td><td>RpL6         </td><td> 9993</td><td> 63933</td><td> 4713.8964</td><td>2092.40810</td><td>1.171066</td><td>14.29054</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\teEF240 & bm & eEF2          & 15933 &  82767 &  8084.7580 & 2731.34968 & 1.565062 & 20.31634\\\\\n",
       "\tbetaTub60D40 & bm & betaTub60D    &  3426 &  14535 &  1723.4942 &  515.49430 & 1.738512 & 18.69241\\\\\n",
       "\teEF1alpha140 & bm & eEF1alpha1    & 37214 & 223504 & 18278.3932 & 7565.11548 & 1.272515 & 18.01617\\\\\n",
       "\tHsc70-440 & bm & Hsc70-4       &  9909 &  55407 &  4957.9620 & 1859.00434 & 1.414441 & 17.36343\\\\\n",
       "\tNon140 & bm & Non1          &  4092 &  21259 &  1997.7208 &  709.18214 & 1.492094 & 16.36060\\\\\n",
       "\tRpL450 & bm & RpL4          &  7051 &  40709 &  3499.0902 & 1348.69270 & 1.374349 & 16.18045\\\\\n",
       "\teIF3c40 & bm & eIF3c         &  1468 &   7093 &   750.8986 &  231.45128 & 1.691690 & 16.16307\\\\\n",
       "\tRpL7A40 & bm & RpL7A         & 10554 &  64427 &  5159.1336 & 2085.48360 & 1.306055 & 16.10783\\\\\n",
       "\teIF4A40 & bm & eIF4A         &  7323 &  44396 &  3589.2292 & 1455.22444 & 1.301441 & 15.36984\\\\\n",
       "\tRpL378 & bm & RpL3          &  9724 &  59841 &  4780.5943 & 1999.71287 & 1.256676 & 15.36070\\\\\n",
       "\tCG436443 & bm & CG4364        &  1265 &   6614 &   661.1250 &  218.19321 & 1.592720 & 14.92532\\\\\n",
       "\tSerRS40 & bm & SerRS         &   442 &   1963 &   240.3248 &   64.45107 & 1.876497 & 14.85216\\\\\n",
       "\tPrx2540-140 & bm & Prx2540-1     &   603 &   2587 &   309.4307 &   88.42007 & 1.790946 & 14.82568\\\\\n",
       "\teEF1gamma40 & bm & eEF1gamma     &  5075 &  29648 &  2499.2167 & 1005.89346 & 1.311565 & 14.80473\\\\\n",
       "\teIF3a40 & bm & eIF3a         &  2073 &  11000 &  1016.9136 &  369.91736 & 1.455027 & 14.53776\\\\\n",
       "\tNeurochondrin40 & bm & Neurochondrin &   961 &   5001 &   505.8119 &  165.04718 & 1.607008 & 14.43946\\\\\n",
       "\tFib40 & bm & Fib           &  1983 &  10602 &   980.1815 &  358.38971 & 1.447500 & 14.38580\\\\\n",
       "\tCCT140 & bm & CCT1          &   702 &   3535 &   389.5810 &  121.53030 & 1.668785 & 14.36736\\\\\n",
       "\talphaTub84B40 & bm & alphaTub84B   &  6958 &  43878 &  3517.5141 & 1511.35416 & 1.217760 & 14.34613\\\\\n",
       "\tRpL640 & bm & RpL6          &  9993 &  63933 &  4713.8964 & 2092.40810 & 1.171066 & 14.29054\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| eEF240 | bm | eEF2          | 15933 |  82767 |  8084.7580 | 2731.34968 | 1.565062 | 20.31634 |\n",
       "| betaTub60D40 | bm | betaTub60D    |  3426 |  14535 |  1723.4942 |  515.49430 | 1.738512 | 18.69241 |\n",
       "| eEF1alpha140 | bm | eEF1alpha1    | 37214 | 223504 | 18278.3932 | 7565.11548 | 1.272515 | 18.01617 |\n",
       "| Hsc70-440 | bm | Hsc70-4       |  9909 |  55407 |  4957.9620 | 1859.00434 | 1.414441 | 17.36343 |\n",
       "| Non140 | bm | Non1          |  4092 |  21259 |  1997.7208 |  709.18214 | 1.492094 | 16.36060 |\n",
       "| RpL450 | bm | RpL4          |  7051 |  40709 |  3499.0902 | 1348.69270 | 1.374349 | 16.18045 |\n",
       "| eIF3c40 | bm | eIF3c         |  1468 |   7093 |   750.8986 |  231.45128 | 1.691690 | 16.16307 |\n",
       "| RpL7A40 | bm | RpL7A         | 10554 |  64427 |  5159.1336 | 2085.48360 | 1.306055 | 16.10783 |\n",
       "| eIF4A40 | bm | eIF4A         |  7323 |  44396 |  3589.2292 | 1455.22444 | 1.301441 | 15.36984 |\n",
       "| RpL378 | bm | RpL3          |  9724 |  59841 |  4780.5943 | 1999.71287 | 1.256676 | 15.36070 |\n",
       "| CG436443 | bm | CG4364        |  1265 |   6614 |   661.1250 |  218.19321 | 1.592720 | 14.92532 |\n",
       "| SerRS40 | bm | SerRS         |   442 |   1963 |   240.3248 |   64.45107 | 1.876497 | 14.85216 |\n",
       "| Prx2540-140 | bm | Prx2540-1     |   603 |   2587 |   309.4307 |   88.42007 | 1.790946 | 14.82568 |\n",
       "| eEF1gamma40 | bm | eEF1gamma     |  5075 |  29648 |  2499.2167 | 1005.89346 | 1.311565 | 14.80473 |\n",
       "| eIF3a40 | bm | eIF3a         |  2073 |  11000 |  1016.9136 |  369.91736 | 1.455027 | 14.53776 |\n",
       "| Neurochondrin40 | bm | Neurochondrin |   961 |   5001 |   505.8119 |  165.04718 | 1.607008 | 14.43946 |\n",
       "| Fib40 | bm | Fib           |  1983 |  10602 |   980.1815 |  358.38971 | 1.447500 | 14.38580 |\n",
       "| CCT140 | bm | CCT1          |   702 |   3535 |   389.5810 |  121.53030 | 1.668785 | 14.36736 |\n",
       "| alphaTub84B40 | bm | alphaTub84B   |  6958 |  43878 |  3517.5141 | 1511.35416 | 1.217760 | 14.34613 |\n",
       "| RpL640 | bm | RpL6          |  9993 |  63933 |  4713.8964 | 2092.40810 | 1.171066 | 14.29054 |\n",
       "\n"
      ],
      "text/plain": [
       "                cell.type.2 gene          cell.type.2.n.umi other.n.umi\n",
       "eEF240          bm          eEF2          15933              82767     \n",
       "betaTub60D40    bm          betaTub60D     3426              14535     \n",
       "eEF1alpha140    bm          eEF1alpha1    37214             223504     \n",
       "Hsc70-440       bm          Hsc70-4        9909              55407     \n",
       "Non140          bm          Non1           4092              21259     \n",
       "RpL450          bm          RpL4           7051              40709     \n",
       "eIF3c40         bm          eIF3c          1468               7093     \n",
       "RpL7A40         bm          RpL7A         10554              64427     \n",
       "eIF4A40         bm          eIF4A          7323              44396     \n",
       "RpL378          bm          RpL3           9724              59841     \n",
       "CG436443        bm          CG4364         1265               6614     \n",
       "SerRS40         bm          SerRS           442               1963     \n",
       "Prx2540-140     bm          Prx2540-1       603               2587     \n",
       "eEF1gamma40     bm          eEF1gamma      5075              29648     \n",
       "eIF3a40         bm          eIF3a          2073              11000     \n",
       "Neurochondrin40 bm          Neurochondrin   961               5001     \n",
       "Fib40           bm          Fib            1983              10602     \n",
       "CCT140          bm          CCT1            702               3535     \n",
       "alphaTub84B40   bm          alphaTub84B    6958              43878     \n",
       "RpL640          bm          RpL6           9993              63933     \n",
       "                cell.type.2.tpm other.tpm  log2.ratio heuristic.score\n",
       "eEF240           8084.7580      2731.34968 1.565062   20.31634       \n",
       "betaTub60D40     1723.4942       515.49430 1.738512   18.69241       \n",
       "eEF1alpha140    18278.3932      7565.11548 1.272515   18.01617       \n",
       "Hsc70-440        4957.9620      1859.00434 1.414441   17.36343       \n",
       "Non140           1997.7208       709.18214 1.492094   16.36060       \n",
       "RpL450           3499.0902      1348.69270 1.374349   16.18045       \n",
       "eIF3c40           750.8986       231.45128 1.691690   16.16307       \n",
       "RpL7A40          5159.1336      2085.48360 1.306055   16.10783       \n",
       "eIF4A40          3589.2292      1455.22444 1.301441   15.36984       \n",
       "RpL378           4780.5943      1999.71287 1.256676   15.36070       \n",
       "CG436443          661.1250       218.19321 1.592720   14.92532       \n",
       "SerRS40           240.3248        64.45107 1.876497   14.85216       \n",
       "Prx2540-140       309.4307        88.42007 1.790946   14.82568       \n",
       "eEF1gamma40      2499.2167      1005.89346 1.311565   14.80473       \n",
       "eIF3a40          1016.9136       369.91736 1.455027   14.53776       \n",
       "Neurochondrin40   505.8119       165.04718 1.607008   14.43946       \n",
       "Fib40             980.1815       358.38971 1.447500   14.38580       \n",
       "CCT140            389.5810       121.53030 1.668785   14.36736       \n",
       "alphaTub84B40    3517.5141      1511.35416 1.217760   14.34613       \n",
       "RpL640           4713.8964      2092.40810 1.171066   14.29054       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"bm\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 297,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"scro\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"scro\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"scro\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"btl\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"knrl\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": false
   },
   "source": [
    "clus = c(bh)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= -.65*umap_1 -4.4), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -5) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -0.65, intercept = -4.4) +\n",
    "xlim(-4, 0) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1636,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Undifferentiated_2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Undifferentiated_2\", df_sub300$cluster %in% bn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 298,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bn' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bn' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bn)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 299,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bn' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bn' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bn)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1634,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG517141</th><td>bn</td><td>CG5171    </td><td>  365</td><td>  1497</td><td>  342.5247</td><td>  52.52668</td><td>2.677878</td><td>22.55917</td></tr>\n",
       "\t<tr><th scope=row>eEF1alpha141</th><td>bn</td><td>eEF1alpha1</td><td>18811</td><td>227912</td><td>19434.8958</td><td>7606.12842</td><td>1.353226</td><td>19.27864</td></tr>\n",
       "\t<tr><th scope=row>c12.141</th><td>bn</td><td>c12.1     </td><td>  163</td><td>  1197</td><td>  191.7397</td><td>  39.95890</td><td>2.226900</td><td>16.90331</td></tr>\n",
       "\t<tr><th scope=row>RpS641</th><td>bn</td><td>RpS6      </td><td> 4502</td><td> 52001</td><td> 4445.3439</td><td>1717.44532</td><td>1.371191</td><td>16.61665</td></tr>\n",
       "\t<tr><th scope=row>RpS3A41</th><td>bn</td><td>RpS3A     </td><td> 3224</td><td> 37862</td><td> 3324.1579</td><td>1247.76134</td><td>1.412491</td><td>16.52503</td></tr>\n",
       "\t<tr><th scope=row>Act42A41</th><td>bn</td><td>Act42A    </td><td> 1981</td><td> 19016</td><td> 1815.7351</td><td> 635.21926</td><td>1.512958</td><td>16.38099</td></tr>\n",
       "\t<tr><th scope=row>Hsp2341</th><td>bn</td><td>Hsp23     </td><td>  165</td><td>  1497</td><td>  224.0413</td><td>  51.67445</td><td>2.088589</td><td>16.32033</td></tr>\n",
       "\t<tr><th scope=row>betaTub60D41</th><td>bn</td><td>betaTub60D</td><td> 1733</td><td> 15134</td><td> 1537.9638</td><td> 528.47768</td><td>1.538380</td><td>16.28797</td></tr>\n",
       "\t<tr><th scope=row>RpL741</th><td>bn</td><td>RpL7      </td><td> 2547</td><td> 29001</td><td> 2521.2205</td><td> 955.60025</td><td>1.398134</td><td>15.79959</td></tr>\n",
       "\t<tr><th scope=row>RpL641</th><td>bn</td><td>RpL6      </td><td> 5362</td><td> 65739</td><td> 5140.3499</td><td>2121.17961</td><td>1.276320</td><td>15.73438</td></tr>\n",
       "\t<tr><th scope=row>Prx2540-141</th><td>bn</td><td>Prx2540-1 </td><td>  407</td><td>  2648</td><td>  324.9392</td><td>  91.30573</td><td>1.815678</td><td>15.15811</td></tr>\n",
       "\t<tr><th scope=row>Hsp67Bc41</th><td>bn</td><td>Hsp67Bc   </td><td>   83</td><td>   699</td><td>  111.1307</td><td>  23.86733</td><td>2.159933</td><td>14.70707</td></tr>\n",
       "\t<tr><th scope=row>ATPsynbeta41</th><td>bn</td><td>ATPsynbeta</td><td> 1182</td><td> 12897</td><td> 1154.3226</td><td> 423.35637</td><td>1.443698</td><td>14.68830</td></tr>\n",
       "\t<tr><th scope=row>RpL23A41</th><td>bn</td><td>RpL23A    </td><td> 4987</td><td> 65849</td><td> 4892.8259</td><td>2160.28467</td><td>1.178779</td><td>14.44799</td></tr>\n",
       "\t<tr><th scope=row>Rack141</th><td>bn</td><td>Rack1     </td><td> 2915</td><td> 36475</td><td> 2840.7248</td><td>1200.04229</td><td>1.241972</td><td>14.24859</td></tr>\n",
       "\t<tr><th scope=row>Hsc70-441</th><td>bn</td><td>Hsc70-4   </td><td> 4122</td><td> 56787</td><td> 4218.2732</td><td>1887.26731</td><td>1.159590</td><td>13.96468</td></tr>\n",
       "\t<tr><th scope=row>eIF4E141</th><td>bn</td><td>eIF4E1    </td><td>  655</td><td>  7154</td><td>  714.1668</td><td> 256.92717</td><td>1.469297</td><td>13.93208</td></tr>\n",
       "\t<tr><th scope=row>P3241</th><td>bn</td><td>P32       </td><td>  245</td><td>  2186</td><td>  248.2310</td><td>  73.07375</td><td>1.744649</td><td>13.88974</td></tr>\n",
       "\t<tr><th scope=row>eIF3l41</th><td>bn</td><td>eIF3l     </td><td>  517</td><td>  5932</td><td>  569.8906</td><td> 198.15787</td><td>1.516772</td><td>13.88919</td></tr>\n",
       "\t<tr><th scope=row>Vta141</th><td>bn</td><td>Vta1      </td><td>   89</td><td>   775</td><td>  110.1941</td><td>  25.74068</td><td>2.042940</td><td>13.88573</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG517141 & bn & CG5171     &   365 &   1497 &   342.5247 &   52.52668 & 2.677878 & 22.55917\\\\\n",
       "\teEF1alpha141 & bn & eEF1alpha1 & 18811 & 227912 & 19434.8958 & 7606.12842 & 1.353226 & 19.27864\\\\\n",
       "\tc12.141 & bn & c12.1      &   163 &   1197 &   191.7397 &   39.95890 & 2.226900 & 16.90331\\\\\n",
       "\tRpS641 & bn & RpS6       &  4502 &  52001 &  4445.3439 & 1717.44532 & 1.371191 & 16.61665\\\\\n",
       "\tRpS3A41 & bn & RpS3A      &  3224 &  37862 &  3324.1579 & 1247.76134 & 1.412491 & 16.52503\\\\\n",
       "\tAct42A41 & bn & Act42A     &  1981 &  19016 &  1815.7351 &  635.21926 & 1.512958 & 16.38099\\\\\n",
       "\tHsp2341 & bn & Hsp23      &   165 &   1497 &   224.0413 &   51.67445 & 2.088589 & 16.32033\\\\\n",
       "\tbetaTub60D41 & bn & betaTub60D &  1733 &  15134 &  1537.9638 &  528.47768 & 1.538380 & 16.28797\\\\\n",
       "\tRpL741 & bn & RpL7       &  2547 &  29001 &  2521.2205 &  955.60025 & 1.398134 & 15.79959\\\\\n",
       "\tRpL641 & bn & RpL6       &  5362 &  65739 &  5140.3499 & 2121.17961 & 1.276320 & 15.73438\\\\\n",
       "\tPrx2540-141 & bn & Prx2540-1  &   407 &   2648 &   324.9392 &   91.30573 & 1.815678 & 15.15811\\\\\n",
       "\tHsp67Bc41 & bn & Hsp67Bc    &    83 &    699 &   111.1307 &   23.86733 & 2.159933 & 14.70707\\\\\n",
       "\tATPsynbeta41 & bn & ATPsynbeta &  1182 &  12897 &  1154.3226 &  423.35637 & 1.443698 & 14.68830\\\\\n",
       "\tRpL23A41 & bn & RpL23A     &  4987 &  65849 &  4892.8259 & 2160.28467 & 1.178779 & 14.44799\\\\\n",
       "\tRack141 & bn & Rack1      &  2915 &  36475 &  2840.7248 & 1200.04229 & 1.241972 & 14.24859\\\\\n",
       "\tHsc70-441 & bn & Hsc70-4    &  4122 &  56787 &  4218.2732 & 1887.26731 & 1.159590 & 13.96468\\\\\n",
       "\teIF4E141 & bn & eIF4E1     &   655 &   7154 &   714.1668 &  256.92717 & 1.469297 & 13.93208\\\\\n",
       "\tP3241 & bn & P32        &   245 &   2186 &   248.2310 &   73.07375 & 1.744649 & 13.88974\\\\\n",
       "\teIF3l41 & bn & eIF3l      &   517 &   5932 &   569.8906 &  198.15787 & 1.516772 & 13.88919\\\\\n",
       "\tVta141 & bn & Vta1       &    89 &    775 &   110.1941 &   25.74068 & 2.042940 & 13.88573\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG517141 | bn | CG5171     |   365 |   1497 |   342.5247 |   52.52668 | 2.677878 | 22.55917 |\n",
       "| eEF1alpha141 | bn | eEF1alpha1 | 18811 | 227912 | 19434.8958 | 7606.12842 | 1.353226 | 19.27864 |\n",
       "| c12.141 | bn | c12.1      |   163 |   1197 |   191.7397 |   39.95890 | 2.226900 | 16.90331 |\n",
       "| RpS641 | bn | RpS6       |  4502 |  52001 |  4445.3439 | 1717.44532 | 1.371191 | 16.61665 |\n",
       "| RpS3A41 | bn | RpS3A      |  3224 |  37862 |  3324.1579 | 1247.76134 | 1.412491 | 16.52503 |\n",
       "| Act42A41 | bn | Act42A     |  1981 |  19016 |  1815.7351 |  635.21926 | 1.512958 | 16.38099 |\n",
       "| Hsp2341 | bn | Hsp23      |   165 |   1497 |   224.0413 |   51.67445 | 2.088589 | 16.32033 |\n",
       "| betaTub60D41 | bn | betaTub60D |  1733 |  15134 |  1537.9638 |  528.47768 | 1.538380 | 16.28797 |\n",
       "| RpL741 | bn | RpL7       |  2547 |  29001 |  2521.2205 |  955.60025 | 1.398134 | 15.79959 |\n",
       "| RpL641 | bn | RpL6       |  5362 |  65739 |  5140.3499 | 2121.17961 | 1.276320 | 15.73438 |\n",
       "| Prx2540-141 | bn | Prx2540-1  |   407 |   2648 |   324.9392 |   91.30573 | 1.815678 | 15.15811 |\n",
       "| Hsp67Bc41 | bn | Hsp67Bc    |    83 |    699 |   111.1307 |   23.86733 | 2.159933 | 14.70707 |\n",
       "| ATPsynbeta41 | bn | ATPsynbeta |  1182 |  12897 |  1154.3226 |  423.35637 | 1.443698 | 14.68830 |\n",
       "| RpL23A41 | bn | RpL23A     |  4987 |  65849 |  4892.8259 | 2160.28467 | 1.178779 | 14.44799 |\n",
       "| Rack141 | bn | Rack1      |  2915 |  36475 |  2840.7248 | 1200.04229 | 1.241972 | 14.24859 |\n",
       "| Hsc70-441 | bn | Hsc70-4    |  4122 |  56787 |  4218.2732 | 1887.26731 | 1.159590 | 13.96468 |\n",
       "| eIF4E141 | bn | eIF4E1     |   655 |   7154 |   714.1668 |  256.92717 | 1.469297 | 13.93208 |\n",
       "| P3241 | bn | P32        |   245 |   2186 |   248.2310 |   73.07375 | 1.744649 | 13.88974 |\n",
       "| eIF3l41 | bn | eIF3l      |   517 |   5932 |   569.8906 |  198.15787 | 1.516772 | 13.88919 |\n",
       "| Vta141 | bn | Vta1       |    89 |    775 |   110.1941 |   25.74068 | 2.042940 | 13.88573 |\n",
       "\n"
      ],
      "text/plain": [
       "             cell.type.2 gene       cell.type.2.n.umi other.n.umi\n",
       "CG517141     bn          CG5171       365               1497     \n",
       "eEF1alpha141 bn          eEF1alpha1 18811             227912     \n",
       "c12.141      bn          c12.1        163               1197     \n",
       "RpS641       bn          RpS6        4502              52001     \n",
       "RpS3A41      bn          RpS3A       3224              37862     \n",
       "Act42A41     bn          Act42A      1981              19016     \n",
       "Hsp2341      bn          Hsp23        165               1497     \n",
       "betaTub60D41 bn          betaTub60D  1733              15134     \n",
       "RpL741       bn          RpL7        2547              29001     \n",
       "RpL641       bn          RpL6        5362              65739     \n",
       "Prx2540-141  bn          Prx2540-1    407               2648     \n",
       "Hsp67Bc41    bn          Hsp67Bc       83                699     \n",
       "ATPsynbeta41 bn          ATPsynbeta  1182              12897     \n",
       "RpL23A41     bn          RpL23A      4987              65849     \n",
       "Rack141      bn          Rack1       2915              36475     \n",
       "Hsc70-441    bn          Hsc70-4     4122              56787     \n",
       "eIF4E141     bn          eIF4E1       655               7154     \n",
       "P3241        bn          P32          245               2186     \n",
       "eIF3l41      bn          eIF3l        517               5932     \n",
       "Vta141       bn          Vta1          89                775     \n",
       "             cell.type.2.tpm other.tpm  log2.ratio heuristic.score\n",
       "CG517141       342.5247        52.52668 2.677878   22.55917       \n",
       "eEF1alpha141 19434.8958      7606.12842 1.353226   19.27864       \n",
       "c12.141        191.7397        39.95890 2.226900   16.90331       \n",
       "RpS641        4445.3439      1717.44532 1.371191   16.61665       \n",
       "RpS3A41       3324.1579      1247.76134 1.412491   16.52503       \n",
       "Act42A41      1815.7351       635.21926 1.512958   16.38099       \n",
       "Hsp2341        224.0413        51.67445 2.088589   16.32033       \n",
       "betaTub60D41  1537.9638       528.47768 1.538380   16.28797       \n",
       "RpL741        2521.2205       955.60025 1.398134   15.79959       \n",
       "RpL641        5140.3499      2121.17961 1.276320   15.73438       \n",
       "Prx2540-141    324.9392        91.30573 1.815678   15.15811       \n",
       "Hsp67Bc41      111.1307        23.86733 2.159933   14.70707       \n",
       "ATPsynbeta41  1154.3226       423.35637 1.443698   14.68830       \n",
       "RpL23A41      4892.8259      2160.28467 1.178779   14.44799       \n",
       "Rack141       2840.7248      1200.04229 1.241972   14.24859       \n",
       "Hsc70-441     4218.2732      1887.26731 1.159590   13.96468       \n",
       "eIF4E141       714.1668       256.92717 1.469297   13.93208       \n",
       "P3241          248.2310        73.07375 1.744649   13.88974       \n",
       "eIF3l41        569.8906       198.15787 1.516772   13.88919       \n",
       "Vta141         110.1941        25.74068 2.042940   13.88573       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"bn\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 300,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"scro\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"scro\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"scro\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"btl\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"knrl\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": false
   },
   "source": [
    "clus = c(bh)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= -.65*umap_1 -4.4), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -5) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -0.65, intercept = -4.4) +\n",
    "xlim(-4, 0) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1639,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Undifferentiated_3\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Undifferentiated_3\", df_sub300$cluster %in% bo)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 301,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bo' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bo' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bo)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 302,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(72, 57)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1638,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>mt:CoI42</th><td>bo</td><td>mt:CoI</td><td>3939</td><td>21895</td><td>3410.85985</td><td> 808.452709</td><td>2.075117</td><td>24.35428</td></tr>\n",
       "\t<tr><th scope=row>RpL1942</th><td>bo</td><td>RpL19 </td><td>5101</td><td>33700</td><td>3874.61540</td><td>1079.976481</td><td>1.841718</td><td>21.95366</td></tr>\n",
       "\t<tr><th scope=row>RpL542</th><td>bo</td><td>RpL5  </td><td>4693</td><td>31777</td><td>3792.33919</td><td>1102.369352</td><td>1.781172</td><td>21.17681</td></tr>\n",
       "\t<tr><th scope=row>RpL1342</th><td>bo</td><td>RpL13 </td><td>3897</td><td>25275</td><td>3118.73949</td><td> 880.648630</td><td>1.822687</td><td>21.15631</td></tr>\n",
       "\t<tr><th scope=row>eEF542</th><td>bo</td><td>eEF5  </td><td>1780</td><td>10597</td><td>1442.71511</td><td> 373.415021</td><td>1.946076</td><td>20.42518</td></tr>\n",
       "\t<tr><th scope=row>RpL1042</th><td>bo</td><td>RpL10 </td><td>3343</td><td>21503</td><td>2643.35711</td><td> 763.993723</td><td>1.788852</td><td>20.33692</td></tr>\n",
       "\t<tr><th scope=row>mt:ND442</th><td>bo</td><td>mt:ND4</td><td> 904</td><td> 4988</td><td> 831.17402</td><td> 193.966310</td><td>2.091926</td><td>20.29323</td></tr>\n",
       "\t<tr><th scope=row>yl42</th><td>bo</td><td>yl    </td><td>  88</td><td>  199</td><td>  79.95737</td><td>   7.849171</td><td>3.175617</td><td>20.13052</td></tr>\n",
       "\t<tr><th scope=row>RpS2442</th><td>bo</td><td>RpS24 </td><td>2767</td><td>18183</td><td>2210.37212</td><td> 629.417643</td><td>1.809909</td><td>20.10941</td></tr>\n",
       "\t<tr><th scope=row>roh42</th><td>bo</td><td>roh   </td><td> 367</td><td> 1297</td><td> 268.94604</td><td>  47.146042</td><td>2.481828</td><td>20.04455</td></tr>\n",
       "\t<tr><th scope=row>CG517142</th><td>bo</td><td>CG5171</td><td> 376</td><td> 1515</td><td> 290.01059</td><td>  53.780227</td><td>2.404378</td><td>19.67966</td></tr>\n",
       "\t<tr><th scope=row>RpL34b42</th><td>bo</td><td>RpL34b</td><td>2249</td><td>15014</td><td>1736.28119</td><td> 488.684069</td><td>1.826077</td><td>19.65337</td></tr>\n",
       "\t<tr><th scope=row>RpS3A42</th><td>bo</td><td>RpS3A </td><td>4871</td><td>37308</td><td>3852.69404</td><td>1227.594615</td><td>1.648859</td><td>19.64125</td></tr>\n",
       "\t<tr><th scope=row>RpS2042</th><td>bo</td><td>RpS20 </td><td>1497</td><td> 8717</td><td>1175.39753</td><td> 308.884454</td><td>1.923346</td><td>19.61844</td></tr>\n",
       "\t<tr><th scope=row>RpL18A42</th><td>bo</td><td>RpL18A</td><td>3509</td><td>25676</td><td>2688.86986</td><td> 846.401774</td><td>1.665882</td><td>18.97993</td></tr>\n",
       "\t<tr><th scope=row>RpL742</th><td>bo</td><td>RpL7  </td><td>3736</td><td>28637</td><td>2942.75472</td><td> 941.247202</td><td>1.642990</td><td>18.93290</td></tr>\n",
       "\t<tr><th scope=row>RpS642</th><td>bo</td><td>RpS6  </td><td>6129</td><td>51619</td><td>4937.80668</td><td>1724.400340</td><td>1.516939</td><td>18.61276</td></tr>\n",
       "\t<tr><th scope=row>RpS2642</th><td>bo</td><td>RpS26 </td><td>3729</td><td>27511</td><td>2810.19114</td><td> 912.313875</td><td>1.621486</td><td>18.57731</td></tr>\n",
       "\t<tr><th scope=row>bic42</th><td>bo</td><td>bic   </td><td>1210</td><td> 7152</td><td> 953.75928</td><td> 260.444664</td><td>1.867120</td><td>18.48260</td></tr>\n",
       "\t<tr><th scope=row>RpL1842</th><td>bo</td><td>RpL18 </td><td>2969</td><td>23109</td><td>2392.50874</td><td> 770.595037</td><td>1.632608</td><td>18.32588</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tmt:CoI42 & bo & mt:CoI & 3939 & 21895 & 3410.85985 &  808.452709 & 2.075117 & 24.35428\\\\\n",
       "\tRpL1942 & bo & RpL19  & 5101 & 33700 & 3874.61540 & 1079.976481 & 1.841718 & 21.95366\\\\\n",
       "\tRpL542 & bo & RpL5   & 4693 & 31777 & 3792.33919 & 1102.369352 & 1.781172 & 21.17681\\\\\n",
       "\tRpL1342 & bo & RpL13  & 3897 & 25275 & 3118.73949 &  880.648630 & 1.822687 & 21.15631\\\\\n",
       "\teEF542 & bo & eEF5   & 1780 & 10597 & 1442.71511 &  373.415021 & 1.946076 & 20.42518\\\\\n",
       "\tRpL1042 & bo & RpL10  & 3343 & 21503 & 2643.35711 &  763.993723 & 1.788852 & 20.33692\\\\\n",
       "\tmt:ND442 & bo & mt:ND4 &  904 &  4988 &  831.17402 &  193.966310 & 2.091926 & 20.29323\\\\\n",
       "\tyl42 & bo & yl     &   88 &   199 &   79.95737 &    7.849171 & 3.175617 & 20.13052\\\\\n",
       "\tRpS2442 & bo & RpS24  & 2767 & 18183 & 2210.37212 &  629.417643 & 1.809909 & 20.10941\\\\\n",
       "\troh42 & bo & roh    &  367 &  1297 &  268.94604 &   47.146042 & 2.481828 & 20.04455\\\\\n",
       "\tCG517142 & bo & CG5171 &  376 &  1515 &  290.01059 &   53.780227 & 2.404378 & 19.67966\\\\\n",
       "\tRpL34b42 & bo & RpL34b & 2249 & 15014 & 1736.28119 &  488.684069 & 1.826077 & 19.65337\\\\\n",
       "\tRpS3A42 & bo & RpS3A  & 4871 & 37308 & 3852.69404 & 1227.594615 & 1.648859 & 19.64125\\\\\n",
       "\tRpS2042 & bo & RpS20  & 1497 &  8717 & 1175.39753 &  308.884454 & 1.923346 & 19.61844\\\\\n",
       "\tRpL18A42 & bo & RpL18A & 3509 & 25676 & 2688.86986 &  846.401774 & 1.665882 & 18.97993\\\\\n",
       "\tRpL742 & bo & RpL7   & 3736 & 28637 & 2942.75472 &  941.247202 & 1.642990 & 18.93290\\\\\n",
       "\tRpS642 & bo & RpS6   & 6129 & 51619 & 4937.80668 & 1724.400340 & 1.516939 & 18.61276\\\\\n",
       "\tRpS2642 & bo & RpS26  & 3729 & 27511 & 2810.19114 &  912.313875 & 1.621486 & 18.57731\\\\\n",
       "\tbic42 & bo & bic    & 1210 &  7152 &  953.75928 &  260.444664 & 1.867120 & 18.48260\\\\\n",
       "\tRpL1842 & bo & RpL18  & 2969 & 23109 & 2392.50874 &  770.595037 & 1.632608 & 18.32588\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| mt:CoI42 | bo | mt:CoI | 3939 | 21895 | 3410.85985 |  808.452709 | 2.075117 | 24.35428 |\n",
       "| RpL1942 | bo | RpL19  | 5101 | 33700 | 3874.61540 | 1079.976481 | 1.841718 | 21.95366 |\n",
       "| RpL542 | bo | RpL5   | 4693 | 31777 | 3792.33919 | 1102.369352 | 1.781172 | 21.17681 |\n",
       "| RpL1342 | bo | RpL13  | 3897 | 25275 | 3118.73949 |  880.648630 | 1.822687 | 21.15631 |\n",
       "| eEF542 | bo | eEF5   | 1780 | 10597 | 1442.71511 |  373.415021 | 1.946076 | 20.42518 |\n",
       "| RpL1042 | bo | RpL10  | 3343 | 21503 | 2643.35711 |  763.993723 | 1.788852 | 20.33692 |\n",
       "| mt:ND442 | bo | mt:ND4 |  904 |  4988 |  831.17402 |  193.966310 | 2.091926 | 20.29323 |\n",
       "| yl42 | bo | yl     |   88 |   199 |   79.95737 |    7.849171 | 3.175617 | 20.13052 |\n",
       "| RpS2442 | bo | RpS24  | 2767 | 18183 | 2210.37212 |  629.417643 | 1.809909 | 20.10941 |\n",
       "| roh42 | bo | roh    |  367 |  1297 |  268.94604 |   47.146042 | 2.481828 | 20.04455 |\n",
       "| CG517142 | bo | CG5171 |  376 |  1515 |  290.01059 |   53.780227 | 2.404378 | 19.67966 |\n",
       "| RpL34b42 | bo | RpL34b | 2249 | 15014 | 1736.28119 |  488.684069 | 1.826077 | 19.65337 |\n",
       "| RpS3A42 | bo | RpS3A  | 4871 | 37308 | 3852.69404 | 1227.594615 | 1.648859 | 19.64125 |\n",
       "| RpS2042 | bo | RpS20  | 1497 |  8717 | 1175.39753 |  308.884454 | 1.923346 | 19.61844 |\n",
       "| RpL18A42 | bo | RpL18A | 3509 | 25676 | 2688.86986 |  846.401774 | 1.665882 | 18.97993 |\n",
       "| RpL742 | bo | RpL7   | 3736 | 28637 | 2942.75472 |  941.247202 | 1.642990 | 18.93290 |\n",
       "| RpS642 | bo | RpS6   | 6129 | 51619 | 4937.80668 | 1724.400340 | 1.516939 | 18.61276 |\n",
       "| RpS2642 | bo | RpS26  | 3729 | 27511 | 2810.19114 |  912.313875 | 1.621486 | 18.57731 |\n",
       "| bic42 | bo | bic    | 1210 |  7152 |  953.75928 |  260.444664 | 1.867120 | 18.48260 |\n",
       "| RpL1842 | bo | RpL18  | 2969 | 23109 | 2392.50874 |  770.595037 | 1.632608 | 18.32588 |\n",
       "\n"
      ],
      "text/plain": [
       "         cell.type.2 gene   cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "mt:CoI42 bo          mt:CoI 3939              21895       3410.85985     \n",
       "RpL1942  bo          RpL19  5101              33700       3874.61540     \n",
       "RpL542   bo          RpL5   4693              31777       3792.33919     \n",
       "RpL1342  bo          RpL13  3897              25275       3118.73949     \n",
       "eEF542   bo          eEF5   1780              10597       1442.71511     \n",
       "RpL1042  bo          RpL10  3343              21503       2643.35711     \n",
       "mt:ND442 bo          mt:ND4  904               4988        831.17402     \n",
       "yl42     bo          yl       88                199         79.95737     \n",
       "RpS2442  bo          RpS24  2767              18183       2210.37212     \n",
       "roh42    bo          roh     367               1297        268.94604     \n",
       "CG517142 bo          CG5171  376               1515        290.01059     \n",
       "RpL34b42 bo          RpL34b 2249              15014       1736.28119     \n",
       "RpS3A42  bo          RpS3A  4871              37308       3852.69404     \n",
       "RpS2042  bo          RpS20  1497               8717       1175.39753     \n",
       "RpL18A42 bo          RpL18A 3509              25676       2688.86986     \n",
       "RpL742   bo          RpL7   3736              28637       2942.75472     \n",
       "RpS642   bo          RpS6   6129              51619       4937.80668     \n",
       "RpS2642  bo          RpS26  3729              27511       2810.19114     \n",
       "bic42    bo          bic    1210               7152        953.75928     \n",
       "RpL1842  bo          RpL18  2969              23109       2392.50874     \n",
       "         other.tpm   log2.ratio heuristic.score\n",
       "mt:CoI42  808.452709 2.075117   24.35428       \n",
       "RpL1942  1079.976481 1.841718   21.95366       \n",
       "RpL542   1102.369352 1.781172   21.17681       \n",
       "RpL1342   880.648630 1.822687   21.15631       \n",
       "eEF542    373.415021 1.946076   20.42518       \n",
       "RpL1042   763.993723 1.788852   20.33692       \n",
       "mt:ND442  193.966310 2.091926   20.29323       \n",
       "yl42        7.849171 3.175617   20.13052       \n",
       "RpS2442   629.417643 1.809909   20.10941       \n",
       "roh42      47.146042 2.481828   20.04455       \n",
       "CG517142   53.780227 2.404378   19.67966       \n",
       "RpL34b42  488.684069 1.826077   19.65337       \n",
       "RpS3A42  1227.594615 1.648859   19.64125       \n",
       "RpS2042   308.884454 1.923346   19.61844       \n",
       "RpL18A42  846.401774 1.665882   18.97993       \n",
       "RpL742    941.247202 1.642990   18.93290       \n",
       "RpS642   1724.400340 1.516939   18.61276       \n",
       "RpS2642   912.313875 1.621486   18.57731       \n",
       "bic42     260.444664 1.867120   18.48260       \n",
       "RpL1842   770.595037 1.632608   18.32588       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"bo\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 303,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"scro\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"scro\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"scro\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"btl\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"knrl\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": false
   },
   "source": [
    "clus = c(bh)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 >= -.65*umap_1 -4.4), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -5) +\n",
    "geom_abline(slope = 0.5, intercept = -3.8) +\n",
    "geom_abline(slope = -0.65, intercept = -4.4) +\n",
    "xlim(-4, 0) + ylim(-5, -1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1698,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Pharynx_primordium_query\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Pharynx_primordium_query\", df_sub300$cluster %in% bp &\n",
    "                         df_sub300$umap_2 < -.5*df_sub300$umap_1 -4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1705,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Epidermis\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Epidermis\", df_sub300$cluster %in% bp &\n",
    "df_sub300$umap_2 < -0.4*df_sub300$umap_1 -3.4 &\n",
    "                                  df_sub300$umap_2 >= -.5*df_sub300$umap_1 -4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 304,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bp' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bp' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bp)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 305,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bp' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bp' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bp)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1642,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG1059143</th><td>bp</td><td>CG10591  </td><td>14608</td><td>21471</td><td>6951.2411</td><td> 850.50420</td><td>3.029185</td><td>38.66228</td></tr>\n",
       "\t<tr><th scope=row>CG918443</th><td>bp</td><td>CG9184   </td><td>20829</td><td>33318</td><td>9956.5393</td><td>1334.39999</td><td>2.898372</td><td>38.49495</td></tr>\n",
       "\t<tr><th scope=row>sha43</th><td>bp</td><td>sha      </td><td> 2251</td><td> 2523</td><td>1089.5775</td><td> 106.16884</td><td>3.345811</td><td>33.76217</td></tr>\n",
       "\t<tr><th scope=row>bbg43</th><td>bp</td><td>bbg      </td><td> 8374</td><td>15453</td><td>3870.8853</td><td> 675.32178</td><td>2.516882</td><td>29.99826</td></tr>\n",
       "\t<tr><th scope=row>CG1411043</th><td>bp</td><td>CG14110  </td><td> 2919</td><td> 4963</td><td>1420.1206</td><td> 202.07180</td><td>2.805952</td><td>29.38621</td></tr>\n",
       "\t<tr><th scope=row>tyn43</th><td>bp</td><td>tyn      </td><td>14960</td><td>32414</td><td>6966.1264</td><td>1427.61175</td><td>2.285743</td><td>29.18059</td></tr>\n",
       "\t<tr><th scope=row>ImpE243</th><td>bp</td><td>ImpE2    </td><td> 6694</td><td>14367</td><td>3173.4271</td><td> 559.89954</td><td>2.500227</td><td>29.08335</td></tr>\n",
       "\t<tr><th scope=row>haf43</th><td>bp</td><td>haf      </td><td> 4511</td><td> 8632</td><td>2137.3825</td><td> 366.27735</td><td>2.540903</td><td>28.10825</td></tr>\n",
       "\t<tr><th scope=row>CG3044043</th><td>bp</td><td>CG30440  </td><td> 2371</td><td> 3631</td><td>1111.8606</td><td> 163.98659</td><td>2.752555</td><td>27.85602</td></tr>\n",
       "\t<tr><th scope=row>dsx-c73A43</th><td>bp</td><td>dsx-c73A </td><td> 3975</td><td> 7637</td><td>1854.7224</td><td> 333.38279</td><td>2.471631</td><td>26.83639</td></tr>\n",
       "\t<tr><th scope=row>ab43</th><td>bp</td><td>ab       </td><td>11442</td><td>29540</td><td>5444.3552</td><td>1241.98668</td><td>2.130950</td><td>26.44682</td></tr>\n",
       "\t<tr><th scope=row>CG1721143</th><td>bp</td><td>CG17211  </td><td>  894</td><td> 1177</td><td> 412.8276</td><td>  49.30669</td><td>3.036717</td><td>26.39784</td></tr>\n",
       "\t<tr><th scope=row>RhoGEF64C43</th><td>bp</td><td>RhoGEF64C</td><td> 4238</td><td> 8624</td><td>1997.3797</td><td> 389.76460</td><td>2.353737</td><td>25.80782</td></tr>\n",
       "\t<tr><th scope=row>sr43</th><td>bp</td><td>sr       </td><td> 1900</td><td> 3560</td><td> 885.5784</td><td> 150.20938</td><td>2.550072</td><td>24.97058</td></tr>\n",
       "\t<tr><th scope=row>spz643</th><td>bp</td><td>spz6     </td><td> 1586</td><td> 3259</td><td> 762.1826</td><td> 135.25102</td><td>2.483870</td><td>23.78525</td></tr>\n",
       "\t<tr><th scope=row>ich43</th><td>bp</td><td>ich      </td><td> 2241</td><td> 5772</td><td>1050.9468</td><td> 218.89156</td><td>2.256826</td><td>22.65592</td></tr>\n",
       "\t<tr><th scope=row>CG4274743</th><td>bp</td><td>CG42747  </td><td> 4032</td><td>10070</td><td>1879.7027</td><td> 460.29896</td><td>2.026731</td><td>22.04486</td></tr>\n",
       "\t<tr><th scope=row>CG1430443</th><td>bp</td><td>CG14304  </td><td>  561</td><td>  916</td><td> 256.6995</td><td>  37.86283</td><td>2.723618</td><td>21.81494</td></tr>\n",
       "\t<tr><th scope=row>CG1439543</th><td>bp</td><td>CG14395  </td><td> 1160</td><td> 2504</td><td> 553.9007</td><td> 104.61042</td><td>2.390875</td><td>21.79542</td></tr>\n",
       "\t<tr><th scope=row>mey43</th><td>bp</td><td>mey      </td><td> 2540</td><td> 6407</td><td>1202.4662</td><td> 276.89324</td><td>2.113394</td><td>21.62631</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG1059143 & bp & CG10591   & 14608 & 21471 & 6951.2411 &  850.50420 & 3.029185 & 38.66228\\\\\n",
       "\tCG918443 & bp & CG9184    & 20829 & 33318 & 9956.5393 & 1334.39999 & 2.898372 & 38.49495\\\\\n",
       "\tsha43 & bp & sha       &  2251 &  2523 & 1089.5775 &  106.16884 & 3.345811 & 33.76217\\\\\n",
       "\tbbg43 & bp & bbg       &  8374 & 15453 & 3870.8853 &  675.32178 & 2.516882 & 29.99826\\\\\n",
       "\tCG1411043 & bp & CG14110   &  2919 &  4963 & 1420.1206 &  202.07180 & 2.805952 & 29.38621\\\\\n",
       "\ttyn43 & bp & tyn       & 14960 & 32414 & 6966.1264 & 1427.61175 & 2.285743 & 29.18059\\\\\n",
       "\tImpE243 & bp & ImpE2     &  6694 & 14367 & 3173.4271 &  559.89954 & 2.500227 & 29.08335\\\\\n",
       "\thaf43 & bp & haf       &  4511 &  8632 & 2137.3825 &  366.27735 & 2.540903 & 28.10825\\\\\n",
       "\tCG3044043 & bp & CG30440   &  2371 &  3631 & 1111.8606 &  163.98659 & 2.752555 & 27.85602\\\\\n",
       "\tdsx-c73A43 & bp & dsx-c73A  &  3975 &  7637 & 1854.7224 &  333.38279 & 2.471631 & 26.83639\\\\\n",
       "\tab43 & bp & ab        & 11442 & 29540 & 5444.3552 & 1241.98668 & 2.130950 & 26.44682\\\\\n",
       "\tCG1721143 & bp & CG17211   &   894 &  1177 &  412.8276 &   49.30669 & 3.036717 & 26.39784\\\\\n",
       "\tRhoGEF64C43 & bp & RhoGEF64C &  4238 &  8624 & 1997.3797 &  389.76460 & 2.353737 & 25.80782\\\\\n",
       "\tsr43 & bp & sr        &  1900 &  3560 &  885.5784 &  150.20938 & 2.550072 & 24.97058\\\\\n",
       "\tspz643 & bp & spz6      &  1586 &  3259 &  762.1826 &  135.25102 & 2.483870 & 23.78525\\\\\n",
       "\tich43 & bp & ich       &  2241 &  5772 & 1050.9468 &  218.89156 & 2.256826 & 22.65592\\\\\n",
       "\tCG4274743 & bp & CG42747   &  4032 & 10070 & 1879.7027 &  460.29896 & 2.026731 & 22.04486\\\\\n",
       "\tCG1430443 & bp & CG14304   &   561 &   916 &  256.6995 &   37.86283 & 2.723618 & 21.81494\\\\\n",
       "\tCG1439543 & bp & CG14395   &  1160 &  2504 &  553.9007 &  104.61042 & 2.390875 & 21.79542\\\\\n",
       "\tmey43 & bp & mey       &  2540 &  6407 & 1202.4662 &  276.89324 & 2.113394 & 21.62631\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG1059143 | bp | CG10591   | 14608 | 21471 | 6951.2411 |  850.50420 | 3.029185 | 38.66228 |\n",
       "| CG918443 | bp | CG9184    | 20829 | 33318 | 9956.5393 | 1334.39999 | 2.898372 | 38.49495 |\n",
       "| sha43 | bp | sha       |  2251 |  2523 | 1089.5775 |  106.16884 | 3.345811 | 33.76217 |\n",
       "| bbg43 | bp | bbg       |  8374 | 15453 | 3870.8853 |  675.32178 | 2.516882 | 29.99826 |\n",
       "| CG1411043 | bp | CG14110   |  2919 |  4963 | 1420.1206 |  202.07180 | 2.805952 | 29.38621 |\n",
       "| tyn43 | bp | tyn       | 14960 | 32414 | 6966.1264 | 1427.61175 | 2.285743 | 29.18059 |\n",
       "| ImpE243 | bp | ImpE2     |  6694 | 14367 | 3173.4271 |  559.89954 | 2.500227 | 29.08335 |\n",
       "| haf43 | bp | haf       |  4511 |  8632 | 2137.3825 |  366.27735 | 2.540903 | 28.10825 |\n",
       "| CG3044043 | bp | CG30440   |  2371 |  3631 | 1111.8606 |  163.98659 | 2.752555 | 27.85602 |\n",
       "| dsx-c73A43 | bp | dsx-c73A  |  3975 |  7637 | 1854.7224 |  333.38279 | 2.471631 | 26.83639 |\n",
       "| ab43 | bp | ab        | 11442 | 29540 | 5444.3552 | 1241.98668 | 2.130950 | 26.44682 |\n",
       "| CG1721143 | bp | CG17211   |   894 |  1177 |  412.8276 |   49.30669 | 3.036717 | 26.39784 |\n",
       "| RhoGEF64C43 | bp | RhoGEF64C |  4238 |  8624 | 1997.3797 |  389.76460 | 2.353737 | 25.80782 |\n",
       "| sr43 | bp | sr        |  1900 |  3560 |  885.5784 |  150.20938 | 2.550072 | 24.97058 |\n",
       "| spz643 | bp | spz6      |  1586 |  3259 |  762.1826 |  135.25102 | 2.483870 | 23.78525 |\n",
       "| ich43 | bp | ich       |  2241 |  5772 | 1050.9468 |  218.89156 | 2.256826 | 22.65592 |\n",
       "| CG4274743 | bp | CG42747   |  4032 | 10070 | 1879.7027 |  460.29896 | 2.026731 | 22.04486 |\n",
       "| CG1430443 | bp | CG14304   |   561 |   916 |  256.6995 |   37.86283 | 2.723618 | 21.81494 |\n",
       "| CG1439543 | bp | CG14395   |  1160 |  2504 |  553.9007 |  104.61042 | 2.390875 | 21.79542 |\n",
       "| mey43 | bp | mey       |  2540 |  6407 | 1202.4662 |  276.89324 | 2.113394 | 21.62631 |\n",
       "\n"
      ],
      "text/plain": [
       "            cell.type.2 gene      cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "CG1059143   bp          CG10591   14608             21471       6951.2411      \n",
       "CG918443    bp          CG9184    20829             33318       9956.5393      \n",
       "sha43       bp          sha        2251              2523       1089.5775      \n",
       "bbg43       bp          bbg        8374             15453       3870.8853      \n",
       "CG1411043   bp          CG14110    2919              4963       1420.1206      \n",
       "tyn43       bp          tyn       14960             32414       6966.1264      \n",
       "ImpE243     bp          ImpE2      6694             14367       3173.4271      \n",
       "haf43       bp          haf        4511              8632       2137.3825      \n",
       "CG3044043   bp          CG30440    2371              3631       1111.8606      \n",
       "dsx-c73A43  bp          dsx-c73A   3975              7637       1854.7224      \n",
       "ab43        bp          ab        11442             29540       5444.3552      \n",
       "CG1721143   bp          CG17211     894              1177        412.8276      \n",
       "RhoGEF64C43 bp          RhoGEF64C  4238              8624       1997.3797      \n",
       "sr43        bp          sr         1900              3560        885.5784      \n",
       "spz643      bp          spz6       1586              3259        762.1826      \n",
       "ich43       bp          ich        2241              5772       1050.9468      \n",
       "CG4274743   bp          CG42747    4032             10070       1879.7027      \n",
       "CG1430443   bp          CG14304     561               916        256.6995      \n",
       "CG1439543   bp          CG14395    1160              2504        553.9007      \n",
       "mey43       bp          mey        2540              6407       1202.4662      \n",
       "            other.tpm  log2.ratio heuristic.score\n",
       "CG1059143    850.50420 3.029185   38.66228       \n",
       "CG918443    1334.39999 2.898372   38.49495       \n",
       "sha43        106.16884 3.345811   33.76217       \n",
       "bbg43        675.32178 2.516882   29.99826       \n",
       "CG1411043    202.07180 2.805952   29.38621       \n",
       "tyn43       1427.61175 2.285743   29.18059       \n",
       "ImpE243      559.89954 2.500227   29.08335       \n",
       "haf43        366.27735 2.540903   28.10825       \n",
       "CG3044043    163.98659 2.752555   27.85602       \n",
       "dsx-c73A43   333.38279 2.471631   26.83639       \n",
       "ab43        1241.98668 2.130950   26.44682       \n",
       "CG1721143     49.30669 3.036717   26.39784       \n",
       "RhoGEF64C43  389.76460 2.353737   25.80782       \n",
       "sr43         150.20938 2.550072   24.97058       \n",
       "spz643       135.25102 2.483870   23.78525       \n",
       "ich43        218.89156 2.256826   22.65592       \n",
       "CG4274743    460.29896 2.026731   22.04486       \n",
       "CG1430443     37.86283 2.723618   21.81494       \n",
       "CG1439543    104.61042 2.390875   21.79542       \n",
       "mey43        276.89324 2.113394   21.62631       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"bp\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 306,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG10591\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG10591\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG10591\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"sha\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"bbg\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG10591\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"sha\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"bbg\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 307,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bp' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bp' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bp)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "              df_sub300$umap_2 < -0.4*umap_1 -3.4 &\n",
    "              df_sub300$umap_2 >= -.5*umap_1 -4), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -5) +\n",
    "geom_abline(slope = -0.5, intercept = -4.0) +\n",
    "geom_abline(slope = -0.4, intercept = -3.4)  +\n",
    "xlim(-3, 2.5) + ylim(-6, -2)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1696,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 124\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG10348</th><td>CG10348</td><td> 64</td><td> 1213</td><td>1597.02130</td><td> 51.5013888</td><td>Set 1</td><td>4.926884</td><td>52.43225</td></tr>\n",
       "\t<tr><th scope=row>rn</th><td>rn     </td><td> 19</td><td>  524</td><td> 558.36504</td><td> 24.3560693</td><td>Set 1</td><td>4.460806</td><td>40.71666</td></tr>\n",
       "\t<tr><th scope=row>mt:ND4L</th><td>mt:ND4L</td><td>  4</td><td>   64</td><td> 133.85089</td><td>  2.2568238</td><td>Set 1</td><td>5.361017</td><td>37.93038</td></tr>\n",
       "\t<tr><th scope=row>CG34398</th><td>CG34398</td><td> 54</td><td> 2413</td><td>1370.22387</td><td>111.5270181</td><td>Set 1</td><td>3.606068</td><td>37.57973</td></tr>\n",
       "\t<tr><th scope=row>CG32073</th><td>CG32073</td><td>  5</td><td>    6</td><td>  70.24840</td><td>  0.3090386</td><td>Set 1</td><td>5.745886</td><td>35.36470</td></tr>\n",
       "\t<tr><th scope=row>Sox21b</th><td>Sox21b </td><td> 33</td><td> 1514</td><td> 815.11339</td><td> 66.4012628</td><td>Set 1</td><td>3.596153</td><td>34.78424</td></tr>\n",
       "\t<tr><th scope=row>CG13675</th><td>CG13675</td><td>  5</td><td>  122</td><td> 146.51375</td><td>  4.6136895</td><td>Set 1</td><td>4.705943</td><td>33.90493</td></tr>\n",
       "\t<tr><th scope=row>CG13643</th><td>CG13643</td><td> 10</td><td>  240</td><td> 209.05299</td><td>  9.7319550</td><td>Set 1</td><td>4.283884</td><td>33.04849</td></tr>\n",
       "\t<tr><th scope=row>sano</th><td>sano   </td><td>118</td><td> 9864</td><td>2908.10442</td><td>444.0639745</td><td>Set 1</td><td>2.707994</td><td>31.15916</td></tr>\n",
       "\t<tr><th scope=row>upd3</th><td>upd3   </td><td>  2</td><td>   29</td><td>  62.10916</td><td>  1.0258149</td><td>Set 1</td><td>4.938232</td><td>29.52953</td></tr>\n",
       "\t<tr><th scope=row>CG32318</th><td>CG32318</td><td>  2</td><td>   27</td><td>  59.73716</td><td>  0.9497062</td><td>Set 1</td><td>4.937300</td><td>29.25107</td></tr>\n",
       "\t<tr><th scope=row>CG32302</th><td>CG32302</td><td>  5</td><td>  216</td><td> 160.64773</td><td>  9.2827831</td><td>Set 1</td><td>3.965598</td><td>29.09444</td></tr>\n",
       "\t<tr><th scope=row>Ser</th><td>Ser    </td><td> 75</td><td> 6480</td><td>1886.16843</td><td>295.3315612</td><td>Set 1</td><td>2.670174</td><td>29.05686</td></tr>\n",
       "\t<tr><th scope=row>CG30284</th><td>CG30284</td><td>  2</td><td>   27</td><td>  64.77523</td><td>  1.3255086</td><td>Set 1</td><td>4.799824</td><td>28.98841</td></tr>\n",
       "\t<tr><th scope=row>CG7365</th><td>CG7365 </td><td>  3</td><td>   61</td><td>  77.79276</td><td>  2.3839984</td><td>Set 1</td><td>4.522835</td><td>28.49382</td></tr>\n",
       "\t<tr><th scope=row>CG17999</th><td>CG17999</td><td>  3</td><td>   91</td><td>  97.16284</td><td>  4.1124868</td><td>Set 1</td><td>4.248308</td><td>28.11150</td></tr>\n",
       "\t<tr><th scope=row>CG4374</th><td>CG4374 </td><td> 19</td><td> 1432</td><td> 558.31579</td><td> 65.3341423</td><td>Set 1</td><td>3.073258</td><td>28.05122</td></tr>\n",
       "\t<tr><th scope=row>cnc</th><td>cnc    </td><td>107</td><td>12752</td><td>2778.62478</td><td>533.4219854</td><td>Set 1</td><td>2.378320</td><td>27.20958</td></tr>\n",
       "\t<tr><th scope=row>CG4653</th><td>CG4653 </td><td>  3</td><td>   41</td><td>  63.46349</td><td>  1.7845271</td><td>Set 1</td><td>4.510423</td><td>27.10949</td></tr>\n",
       "\t<tr><th scope=row>CG43861</th><td>CG43861</td><td> 42</td><td> 2889</td><td> 914.79386</td><td>137.4108067</td><td>Set 1</td><td>2.724490</td><td>26.80593</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG10348 & CG10348 &  64 &  1213 & 1597.02130 &  51.5013888 & Set 1 & 4.926884 & 52.43225\\\\\n",
       "\trn & rn      &  19 &   524 &  558.36504 &  24.3560693 & Set 1 & 4.460806 & 40.71666\\\\\n",
       "\tmt:ND4L & mt:ND4L &   4 &    64 &  133.85089 &   2.2568238 & Set 1 & 5.361017 & 37.93038\\\\\n",
       "\tCG34398 & CG34398 &  54 &  2413 & 1370.22387 & 111.5270181 & Set 1 & 3.606068 & 37.57973\\\\\n",
       "\tCG32073 & CG32073 &   5 &     6 &   70.24840 &   0.3090386 & Set 1 & 5.745886 & 35.36470\\\\\n",
       "\tSox21b & Sox21b  &  33 &  1514 &  815.11339 &  66.4012628 & Set 1 & 3.596153 & 34.78424\\\\\n",
       "\tCG13675 & CG13675 &   5 &   122 &  146.51375 &   4.6136895 & Set 1 & 4.705943 & 33.90493\\\\\n",
       "\tCG13643 & CG13643 &  10 &   240 &  209.05299 &   9.7319550 & Set 1 & 4.283884 & 33.04849\\\\\n",
       "\tsano & sano    & 118 &  9864 & 2908.10442 & 444.0639745 & Set 1 & 2.707994 & 31.15916\\\\\n",
       "\tupd3 & upd3    &   2 &    29 &   62.10916 &   1.0258149 & Set 1 & 4.938232 & 29.52953\\\\\n",
       "\tCG32318 & CG32318 &   2 &    27 &   59.73716 &   0.9497062 & Set 1 & 4.937300 & 29.25107\\\\\n",
       "\tCG32302 & CG32302 &   5 &   216 &  160.64773 &   9.2827831 & Set 1 & 3.965598 & 29.09444\\\\\n",
       "\tSer & Ser     &  75 &  6480 & 1886.16843 & 295.3315612 & Set 1 & 2.670174 & 29.05686\\\\\n",
       "\tCG30284 & CG30284 &   2 &    27 &   64.77523 &   1.3255086 & Set 1 & 4.799824 & 28.98841\\\\\n",
       "\tCG7365 & CG7365  &   3 &    61 &   77.79276 &   2.3839984 & Set 1 & 4.522835 & 28.49382\\\\\n",
       "\tCG17999 & CG17999 &   3 &    91 &   97.16284 &   4.1124868 & Set 1 & 4.248308 & 28.11150\\\\\n",
       "\tCG4374 & CG4374  &  19 &  1432 &  558.31579 &  65.3341423 & Set 1 & 3.073258 & 28.05122\\\\\n",
       "\tcnc & cnc     & 107 & 12752 & 2778.62478 & 533.4219854 & Set 1 & 2.378320 & 27.20958\\\\\n",
       "\tCG4653 & CG4653  &   3 &    41 &   63.46349 &   1.7845271 & Set 1 & 4.510423 & 27.10949\\\\\n",
       "\tCG43861 & CG43861 &  42 &  2889 &  914.79386 & 137.4108067 & Set 1 & 2.724490 & 26.80593\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG10348 | CG10348 |  64 |  1213 | 1597.02130 |  51.5013888 | Set 1 | 4.926884 | 52.43225 |\n",
       "| rn | rn      |  19 |   524 |  558.36504 |  24.3560693 | Set 1 | 4.460806 | 40.71666 |\n",
       "| mt:ND4L | mt:ND4L |   4 |    64 |  133.85089 |   2.2568238 | Set 1 | 5.361017 | 37.93038 |\n",
       "| CG34398 | CG34398 |  54 |  2413 | 1370.22387 | 111.5270181 | Set 1 | 3.606068 | 37.57973 |\n",
       "| CG32073 | CG32073 |   5 |     6 |   70.24840 |   0.3090386 | Set 1 | 5.745886 | 35.36470 |\n",
       "| Sox21b | Sox21b  |  33 |  1514 |  815.11339 |  66.4012628 | Set 1 | 3.596153 | 34.78424 |\n",
       "| CG13675 | CG13675 |   5 |   122 |  146.51375 |   4.6136895 | Set 1 | 4.705943 | 33.90493 |\n",
       "| CG13643 | CG13643 |  10 |   240 |  209.05299 |   9.7319550 | Set 1 | 4.283884 | 33.04849 |\n",
       "| sano | sano    | 118 |  9864 | 2908.10442 | 444.0639745 | Set 1 | 2.707994 | 31.15916 |\n",
       "| upd3 | upd3    |   2 |    29 |   62.10916 |   1.0258149 | Set 1 | 4.938232 | 29.52953 |\n",
       "| CG32318 | CG32318 |   2 |    27 |   59.73716 |   0.9497062 | Set 1 | 4.937300 | 29.25107 |\n",
       "| CG32302 | CG32302 |   5 |   216 |  160.64773 |   9.2827831 | Set 1 | 3.965598 | 29.09444 |\n",
       "| Ser | Ser     |  75 |  6480 | 1886.16843 | 295.3315612 | Set 1 | 2.670174 | 29.05686 |\n",
       "| CG30284 | CG30284 |   2 |    27 |   64.77523 |   1.3255086 | Set 1 | 4.799824 | 28.98841 |\n",
       "| CG7365 | CG7365  |   3 |    61 |   77.79276 |   2.3839984 | Set 1 | 4.522835 | 28.49382 |\n",
       "| CG17999 | CG17999 |   3 |    91 |   97.16284 |   4.1124868 | Set 1 | 4.248308 | 28.11150 |\n",
       "| CG4374 | CG4374  |  19 |  1432 |  558.31579 |  65.3341423 | Set 1 | 3.073258 | 28.05122 |\n",
       "| cnc | cnc     | 107 | 12752 | 2778.62478 | 533.4219854 | Set 1 | 2.378320 | 27.20958 |\n",
       "| CG4653 | CG4653  |   3 |    41 |   63.46349 |   1.7845271 | Set 1 | 4.510423 | 27.10949 |\n",
       "| CG43861 | CG43861 |  42 |  2889 |  914.79386 | 137.4108067 | Set 1 | 2.724490 | 26.80593 |\n",
       "\n"
      ],
      "text/plain": [
       "        gene    set.1.n.umi set.2.n.umi set.1.tpm  set.2.tpm   higher.expr\n",
       "CG10348 CG10348  64          1213       1597.02130  51.5013888 Set 1      \n",
       "rn      rn       19           524        558.36504  24.3560693 Set 1      \n",
       "mt:ND4L mt:ND4L   4            64        133.85089   2.2568238 Set 1      \n",
       "CG34398 CG34398  54          2413       1370.22387 111.5270181 Set 1      \n",
       "CG32073 CG32073   5             6         70.24840   0.3090386 Set 1      \n",
       "Sox21b  Sox21b   33          1514        815.11339  66.4012628 Set 1      \n",
       "CG13675 CG13675   5           122        146.51375   4.6136895 Set 1      \n",
       "CG13643 CG13643  10           240        209.05299   9.7319550 Set 1      \n",
       "sano    sano    118          9864       2908.10442 444.0639745 Set 1      \n",
       "upd3    upd3      2            29         62.10916   1.0258149 Set 1      \n",
       "CG32318 CG32318   2            27         59.73716   0.9497062 Set 1      \n",
       "CG32302 CG32302   5           216        160.64773   9.2827831 Set 1      \n",
       "Ser     Ser      75          6480       1886.16843 295.3315612 Set 1      \n",
       "CG30284 CG30284   2            27         64.77523   1.3255086 Set 1      \n",
       "CG7365  CG7365    3            61         77.79276   2.3839984 Set 1      \n",
       "CG17999 CG17999   3            91         97.16284   4.1124868 Set 1      \n",
       "CG4374  CG4374   19          1432        558.31579  65.3341423 Set 1      \n",
       "cnc     cnc     107         12752       2778.62478 533.4219854 Set 1      \n",
       "CG4653  CG4653    3            41         63.46349   1.7845271 Set 1      \n",
       "CG43861 CG43861  42          2889        914.79386 137.4108067 Set 1      \n",
       "        log2.ratio heuristic.score\n",
       "CG10348 4.926884   52.43225       \n",
       "rn      4.460806   40.71666       \n",
       "mt:ND4L 5.361017   37.93038       \n",
       "CG34398 3.606068   37.57973       \n",
       "CG32073 5.745886   35.36470       \n",
       "Sox21b  3.596153   34.78424       \n",
       "CG13675 4.705943   33.90493       \n",
       "CG13643 4.283884   33.04849       \n",
       "sano    2.707994   31.15916       \n",
       "upd3    4.938232   29.52953       \n",
       "CG32318 4.937300   29.25107       \n",
       "CG32302 3.965598   29.09444       \n",
       "Ser     2.670174   29.05686       \n",
       "CG30284 4.799824   28.98841       \n",
       "CG7365  4.522835   28.49382       \n",
       "CG17999 4.248308   28.11150       \n",
       "CG4374  3.073258   28.05122       \n",
       "cnc     2.378320   27.20958       \n",
       "CG4653  4.510423   27.10949       \n",
       "CG43861 2.724490   26.80593       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "two.set.differential.gene.test.3a(cds_sub, \n",
    "                                  df_sub300$cluster == bp & \n",
    "                                  df_sub300$umap_2 < -.5*df_sub300$umap_1 -4 , \n",
    "                                 !(df_sub300$cluster == bp & \n",
    "                                  df_sub300$umap_2 < -.5*df_sub300$umap_1 -4),\n",
    "                                  sample = 50000\n",
    "                                )  %>% filter(higher.expr == \"Set 1\")  %>% \n",
    "head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 308,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG10348\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG10348\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG10348\"), \n",
    "                   size_pos = 2, size_neg = 0.5, stroke = 0.008, limit = 2)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"rn\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 2)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG34398\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1702,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 767\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>bbg</th><td>bbg      </td><td>1312</td><td>16165</td><td>4936.7152</td><td> 743.141753</td><td>Set 1</td><td>2.729902</td><td>33.49488</td></tr>\n",
       "\t<tr><th scope=row>CG9184</th><td>CG9184   </td><td>2294</td><td>37041</td><td>8868.3756</td><td>1601.805215</td><td>Set 1</td><td>2.468071</td><td>32.36780</td></tr>\n",
       "\t<tr><th scope=row>CG30440</th><td>CG30440  </td><td> 416</td><td> 4143</td><td>1584.8964</td><td> 195.834456</td><td>Set 1</td><td>3.009334</td><td>31.99248</td></tr>\n",
       "\t<tr><th scope=row>sha</th><td>sha      </td><td> 304</td><td> 2856</td><td>1169.7647</td><td> 131.930527</td><td>Set 1</td><td>3.137474</td><td>31.98101</td></tr>\n",
       "\t<tr><th scope=row>haf</th><td>haf      </td><td> 734</td><td> 9203</td><td>2853.5429</td><td> 414.269083</td><td>Set 1</td><td>2.780636</td><td>31.91904</td></tr>\n",
       "\t<tr><th scope=row>tyn</th><td>tyn      </td><td>2274</td><td>34448</td><td>8633.8552</td><td>1589.563878</td><td>Set 1</td><td>2.440467</td><td>31.91143</td></tr>\n",
       "\t<tr><th scope=row>CG10591</th><td>CG10591  </td><td>1541</td><td>24411</td><td>5941.8520</td><td>1049.633881</td><td>Set 1</td><td>2.499653</td><td>31.33799</td></tr>\n",
       "\t<tr><th scope=row>ab</th><td>ab       </td><td>1903</td><td>31425</td><td>7426.4713</td><td>1387.779482</td><td>Set 1</td><td>2.418859</td><td>31.10328</td></tr>\n",
       "\t<tr><th scope=row>dsx-c73A</th><td>dsx-c73A </td><td> 653</td><td> 7932</td><td>2404.2975</td><td> 359.945884</td><td>Set 1</td><td>2.735761</td><td>30.72807</td></tr>\n",
       "\t<tr><th scope=row>RhoGEF64C</th><td>RhoGEF64C</td><td> 689</td><td> 9221</td><td>2647.9659</td><td> 436.302448</td><td>Set 1</td><td>2.598181</td><td>29.54447</td></tr>\n",
       "\t<tr><th scope=row>CG17211</th><td>CG17211  </td><td> 147</td><td> 1422</td><td> 544.9679</td><td>  64.225371</td><td>Set 1</td><td>3.062666</td><td>27.84782</td></tr>\n",
       "\t<tr><th scope=row>CG42747</th><td>CG42747  </td><td> 667</td><td>10871</td><td>2634.4487</td><td> 503.228512</td><td>Set 1</td><td>2.385351</td><td>27.10674</td></tr>\n",
       "\t<tr><th scope=row>CG4374</th><td>CG4374   </td><td> 135</td><td> 1440</td><td> 506.9589</td><td>  65.954037</td><td>Set 1</td><td>2.920626</td><td>26.25224</td></tr>\n",
       "\t<tr><th scope=row>ich</th><td>ich      </td><td> 350</td><td> 5841</td><td>1322.7375</td><td> 243.035322</td><td>Set 1</td><td>2.438365</td><td>25.28682</td></tr>\n",
       "\t<tr><th scope=row>CG9514</th><td>CG9514   </td><td> 137</td><td> 1694</td><td> 501.1667</td><td>  78.386995</td><td>Set 1</td><td>2.658316</td><td>23.85047</td></tr>\n",
       "\t<tr><th scope=row>CG13744</th><td>CG13744  </td><td>  25</td><td>  199</td><td> 105.9840</td><td>   8.225669</td><td>Set 1</td><td>3.522049</td><td>23.74302</td></tr>\n",
       "\t<tr><th scope=row>ImpE1</th><td>ImpE1    </td><td> 617</td><td>11851</td><td>2341.7257</td><td> 551.856406</td><td>Set 1</td><td>2.082595</td><td>23.31252</td></tr>\n",
       "\t<tr><th scope=row>ovo</th><td>ovo      </td><td> 634</td><td>13652</td><td>2452.4531</td><td> 583.751677</td><td>Set 1</td><td>2.068330</td><td>23.29063</td></tr>\n",
       "\t<tr><th scope=row>CG43980</th><td>CG43980  </td><td> 143</td><td> 2337</td><td> 560.8612</td><td>  96.452387</td><td>Set 1</td><td>2.524874</td><td>23.06238</td></tr>\n",
       "\t<tr><th scope=row>CG9095</th><td>CG9095   </td><td>1122</td><td>27359</td><td>4365.9144</td><td>1190.955536</td><td>Set 1</td><td>1.872953</td><td>22.64850</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tbbg & bbg       & 1312 & 16165 & 4936.7152 &  743.141753 & Set 1 & 2.729902 & 33.49488\\\\\n",
       "\tCG9184 & CG9184    & 2294 & 37041 & 8868.3756 & 1601.805215 & Set 1 & 2.468071 & 32.36780\\\\\n",
       "\tCG30440 & CG30440   &  416 &  4143 & 1584.8964 &  195.834456 & Set 1 & 3.009334 & 31.99248\\\\\n",
       "\tsha & sha       &  304 &  2856 & 1169.7647 &  131.930527 & Set 1 & 3.137474 & 31.98101\\\\\n",
       "\thaf & haf       &  734 &  9203 & 2853.5429 &  414.269083 & Set 1 & 2.780636 & 31.91904\\\\\n",
       "\ttyn & tyn       & 2274 & 34448 & 8633.8552 & 1589.563878 & Set 1 & 2.440467 & 31.91143\\\\\n",
       "\tCG10591 & CG10591   & 1541 & 24411 & 5941.8520 & 1049.633881 & Set 1 & 2.499653 & 31.33799\\\\\n",
       "\tab & ab        & 1903 & 31425 & 7426.4713 & 1387.779482 & Set 1 & 2.418859 & 31.10328\\\\\n",
       "\tdsx-c73A & dsx-c73A  &  653 &  7932 & 2404.2975 &  359.945884 & Set 1 & 2.735761 & 30.72807\\\\\n",
       "\tRhoGEF64C & RhoGEF64C &  689 &  9221 & 2647.9659 &  436.302448 & Set 1 & 2.598181 & 29.54447\\\\\n",
       "\tCG17211 & CG17211   &  147 &  1422 &  544.9679 &   64.225371 & Set 1 & 3.062666 & 27.84782\\\\\n",
       "\tCG42747 & CG42747   &  667 & 10871 & 2634.4487 &  503.228512 & Set 1 & 2.385351 & 27.10674\\\\\n",
       "\tCG4374 & CG4374    &  135 &  1440 &  506.9589 &   65.954037 & Set 1 & 2.920626 & 26.25224\\\\\n",
       "\tich & ich       &  350 &  5841 & 1322.7375 &  243.035322 & Set 1 & 2.438365 & 25.28682\\\\\n",
       "\tCG9514 & CG9514    &  137 &  1694 &  501.1667 &   78.386995 & Set 1 & 2.658316 & 23.85047\\\\\n",
       "\tCG13744 & CG13744   &   25 &   199 &  105.9840 &    8.225669 & Set 1 & 3.522049 & 23.74302\\\\\n",
       "\tImpE1 & ImpE1     &  617 & 11851 & 2341.7257 &  551.856406 & Set 1 & 2.082595 & 23.31252\\\\\n",
       "\tovo & ovo       &  634 & 13652 & 2452.4531 &  583.751677 & Set 1 & 2.068330 & 23.29063\\\\\n",
       "\tCG43980 & CG43980   &  143 &  2337 &  560.8612 &   96.452387 & Set 1 & 2.524874 & 23.06238\\\\\n",
       "\tCG9095 & CG9095    & 1122 & 27359 & 4365.9144 & 1190.955536 & Set 1 & 1.872953 & 22.64850\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| bbg | bbg       | 1312 | 16165 | 4936.7152 |  743.141753 | Set 1 | 2.729902 | 33.49488 |\n",
       "| CG9184 | CG9184    | 2294 | 37041 | 8868.3756 | 1601.805215 | Set 1 | 2.468071 | 32.36780 |\n",
       "| CG30440 | CG30440   |  416 |  4143 | 1584.8964 |  195.834456 | Set 1 | 3.009334 | 31.99248 |\n",
       "| sha | sha       |  304 |  2856 | 1169.7647 |  131.930527 | Set 1 | 3.137474 | 31.98101 |\n",
       "| haf | haf       |  734 |  9203 | 2853.5429 |  414.269083 | Set 1 | 2.780636 | 31.91904 |\n",
       "| tyn | tyn       | 2274 | 34448 | 8633.8552 | 1589.563878 | Set 1 | 2.440467 | 31.91143 |\n",
       "| CG10591 | CG10591   | 1541 | 24411 | 5941.8520 | 1049.633881 | Set 1 | 2.499653 | 31.33799 |\n",
       "| ab | ab        | 1903 | 31425 | 7426.4713 | 1387.779482 | Set 1 | 2.418859 | 31.10328 |\n",
       "| dsx-c73A | dsx-c73A  |  653 |  7932 | 2404.2975 |  359.945884 | Set 1 | 2.735761 | 30.72807 |\n",
       "| RhoGEF64C | RhoGEF64C |  689 |  9221 | 2647.9659 |  436.302448 | Set 1 | 2.598181 | 29.54447 |\n",
       "| CG17211 | CG17211   |  147 |  1422 |  544.9679 |   64.225371 | Set 1 | 3.062666 | 27.84782 |\n",
       "| CG42747 | CG42747   |  667 | 10871 | 2634.4487 |  503.228512 | Set 1 | 2.385351 | 27.10674 |\n",
       "| CG4374 | CG4374    |  135 |  1440 |  506.9589 |   65.954037 | Set 1 | 2.920626 | 26.25224 |\n",
       "| ich | ich       |  350 |  5841 | 1322.7375 |  243.035322 | Set 1 | 2.438365 | 25.28682 |\n",
       "| CG9514 | CG9514    |  137 |  1694 |  501.1667 |   78.386995 | Set 1 | 2.658316 | 23.85047 |\n",
       "| CG13744 | CG13744   |   25 |   199 |  105.9840 |    8.225669 | Set 1 | 3.522049 | 23.74302 |\n",
       "| ImpE1 | ImpE1     |  617 | 11851 | 2341.7257 |  551.856406 | Set 1 | 2.082595 | 23.31252 |\n",
       "| ovo | ovo       |  634 | 13652 | 2452.4531 |  583.751677 | Set 1 | 2.068330 | 23.29063 |\n",
       "| CG43980 | CG43980   |  143 |  2337 |  560.8612 |   96.452387 | Set 1 | 2.524874 | 23.06238 |\n",
       "| CG9095 | CG9095    | 1122 | 27359 | 4365.9144 | 1190.955536 | Set 1 | 1.872953 | 22.64850 |\n",
       "\n"
      ],
      "text/plain": [
       "          gene      set.1.n.umi set.2.n.umi set.1.tpm set.2.tpm   higher.expr\n",
       "bbg       bbg       1312        16165       4936.7152  743.141753 Set 1      \n",
       "CG9184    CG9184    2294        37041       8868.3756 1601.805215 Set 1      \n",
       "CG30440   CG30440    416         4143       1584.8964  195.834456 Set 1      \n",
       "sha       sha        304         2856       1169.7647  131.930527 Set 1      \n",
       "haf       haf        734         9203       2853.5429  414.269083 Set 1      \n",
       "tyn       tyn       2274        34448       8633.8552 1589.563878 Set 1      \n",
       "CG10591   CG10591   1541        24411       5941.8520 1049.633881 Set 1      \n",
       "ab        ab        1903        31425       7426.4713 1387.779482 Set 1      \n",
       "dsx-c73A  dsx-c73A   653         7932       2404.2975  359.945884 Set 1      \n",
       "RhoGEF64C RhoGEF64C  689         9221       2647.9659  436.302448 Set 1      \n",
       "CG17211   CG17211    147         1422        544.9679   64.225371 Set 1      \n",
       "CG42747   CG42747    667        10871       2634.4487  503.228512 Set 1      \n",
       "CG4374    CG4374     135         1440        506.9589   65.954037 Set 1      \n",
       "ich       ich        350         5841       1322.7375  243.035322 Set 1      \n",
       "CG9514    CG9514     137         1694        501.1667   78.386995 Set 1      \n",
       "CG13744   CG13744     25          199        105.9840    8.225669 Set 1      \n",
       "ImpE1     ImpE1      617        11851       2341.7257  551.856406 Set 1      \n",
       "ovo       ovo        634        13652       2452.4531  583.751677 Set 1      \n",
       "CG43980   CG43980    143         2337        560.8612   96.452387 Set 1      \n",
       "CG9095    CG9095    1122        27359       4365.9144 1190.955536 Set 1      \n",
       "          log2.ratio heuristic.score\n",
       "bbg       2.729902   33.49488       \n",
       "CG9184    2.468071   32.36780       \n",
       "CG30440   3.009334   31.99248       \n",
       "sha       3.137474   31.98101       \n",
       "haf       2.780636   31.91904       \n",
       "tyn       2.440467   31.91143       \n",
       "CG10591   2.499653   31.33799       \n",
       "ab        2.418859   31.10328       \n",
       "dsx-c73A  2.735761   30.72807       \n",
       "RhoGEF64C 2.598181   29.54447       \n",
       "CG17211   3.062666   27.84782       \n",
       "CG42747   2.385351   27.10674       \n",
       "CG4374    2.920626   26.25224       \n",
       "ich       2.438365   25.28682       \n",
       "CG9514    2.658316   23.85047       \n",
       "CG13744   3.522049   23.74302       \n",
       "ImpE1     2.082595   23.31252       \n",
       "ovo       2.068330   23.29063       \n",
       "CG43980   2.524874   23.06238       \n",
       "CG9095    1.872953   22.64850       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "two.set.differential.gene.test.3a(cds_sub, \n",
    "                                  df_sub300$cluster == bp & \n",
    "                                  df_sub300$umap_2 < -0.4*df_sub300$umap_1 -3.4 &\n",
    "                                  df_sub300$umap_2 >= -.5*df_sub300$umap_1 -4 , \n",
    "                                 !(df_sub300$cluster == bp & \n",
    "                                   df_sub300$umap_2 < -0.4*df_sub300$umap_1 -3.4 &\n",
    "                                  df_sub300$umap_2 >= -.5*df_sub300$umap_1 -4),\n",
    "                                  sample = 50000\n",
    "                                )  %>% filter(higher.expr == \"Set 1\")  %>% \n",
    "head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 309,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"bbg\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"bbg\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"bbg\"), \n",
    "                   size_pos = 2, size_neg = 0.5, stroke = 0.008, limit = 12)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG9184\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 12)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"sha\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "two.set.differential.gene.test.3a(cds_sub, \n",
    "                                  df_sub300$cluster == bp & \n",
    "                                  df_sub300$umap_2 >= -0.4*df_sub300$umap_1 -3.4 , \n",
    "                                 !(df_sub300$cluster == bp & \n",
    "                                   df_sub300$umap_2 < -0.4*df_sub300$umap_1 -3.4 &\n",
    "                                  df_sub300$umap_2 >= -.5*df_sub300$umap_1 -4),\n",
    "                                  sample = 50000\n",
    "                                )  %>% filter(higher.expr == \"Set 1\")  %>% \n",
    "head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1660,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Embryonic_body_wall\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Embryonic_body_wall\", df_sub300$cluster %in% c(bq, br,bs, bt, bu, bv, bw, bx, by))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1644,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Act57B44</th><td>bq</td><td>Act57B    </td><td>24201</td><td> 49032</td><td> 6408.3207</td><td>1439.01846</td><td>2.153859</td><td>27.23761</td></tr>\n",
       "\t<tr><th scope=row>CG296244</th><td>bq</td><td>CG2962    </td><td>14862</td><td> 28955</td><td> 3852.2991</td><td>1015.30233</td><td>1.922390</td><td>22.89928</td></tr>\n",
       "\t<tr><th scope=row>Mlc244</th><td>bq</td><td>Mlc2      </td><td>12441</td><td> 28458</td><td> 3299.5489</td><td> 866.05437</td><td>1.928074</td><td>22.53628</td></tr>\n",
       "\t<tr><th scope=row>eEF1alpha144</th><td>bq</td><td>eEF1alpha1</td><td>76454</td><td>218993</td><td>20543.3019</td><td>7206.71742</td><td>1.511054</td><td>21.64804</td></tr>\n",
       "\t<tr><th scope=row>CG1573144</th><td>bq</td><td>CG15731   </td><td> 2958</td><td>  5182</td><td>  782.8430</td><td> 168.63752</td><td>2.206268</td><td>21.21199</td></tr>\n",
       "\t<tr><th scope=row>RpL23A44</th><td>bq</td><td>RpL23A    </td><td>23945</td><td> 63966</td><td> 6321.6670</td><td>2065.07794</td><td>1.613410</td><td>20.37143</td></tr>\n",
       "\t<tr><th scope=row>TwdlF44</th><td>bq</td><td>TwdlF     </td><td> 6697</td><td> 13997</td><td> 1776.8570</td><td> 492.98094</td><td>1.846800</td><td>19.93791</td></tr>\n",
       "\t<tr><th scope=row>RpL644</th><td>bq</td><td>RpL6      </td><td>22539</td><td> 62854</td><td> 5951.1989</td><td>1986.58711</td><td>1.582162</td><td>19.83906</td></tr>\n",
       "\t<tr><th scope=row>RpS644</th><td>bq</td><td>RpS6      </td><td>18692</td><td> 50047</td><td> 4918.5066</td><td>1630.96020</td><td>1.591614</td><td>19.52003</td></tr>\n",
       "\t<tr><th scope=row>Hsc70-444</th><td>bq</td><td>Hsc70-4   </td><td>19542</td><td> 54591</td><td> 5251.8047</td><td>1790.73474</td><td>1.551456</td><td>19.17425</td></tr>\n",
       "\t<tr><th scope=row>RpL7A44</th><td>bq</td><td>RpL7A     </td><td>21373</td><td> 62912</td><td> 5587.7042</td><td>1969.40485</td><td>1.503764</td><td>18.71930</td></tr>\n",
       "\t<tr><th scope=row>eEF244</th><td>bq</td><td>eEF2      </td><td>27581</td><td> 82036</td><td> 7219.6870</td><td>2627.25166</td><td>1.457833</td><td>18.68639</td></tr>\n",
       "\t<tr><th scope=row>RpL454</th><td>bq</td><td>RpL4      </td><td>14388</td><td> 40203</td><td> 3855.1943</td><td>1299.19926</td><td>1.568071</td><td>18.68037</td></tr>\n",
       "\t<tr><th scope=row>Gabat44</th><td>bq</td><td>Gabat     </td><td> 1392</td><td>  2557</td><td>  375.6893</td><td>  81.77015</td><td>2.182357</td><td>18.67494</td></tr>\n",
       "\t<tr><th scope=row>CG908344</th><td>bq</td><td>CG9083    </td><td> 5361</td><td> 11400</td><td> 1406.4989</td><td> 413.48149</td><td>1.762729</td><td>18.43624</td></tr>\n",
       "\t<tr><th scope=row>CG767544</th><td>bq</td><td>CG7675    </td><td> 2668</td><td>  5580</td><td>  717.7727</td><td> 186.19533</td><td>1.938983</td><td>18.39976</td></tr>\n",
       "\t<tr><th scope=row>Tm244</th><td>bq</td><td>Tm2       </td><td> 3845</td><td>  9615</td><td> 1041.4820</td><td> 298.07987</td><td>1.800035</td><td>18.04681</td></tr>\n",
       "\t<tr><th scope=row>Gdh44</th><td>bq</td><td>Gdh       </td><td> 4394</td><td>  9732</td><td> 1180.6039</td><td> 346.38616</td><td>1.764913</td><td>18.01364</td></tr>\n",
       "\t<tr><th scope=row>RpL3100</th><td>bq</td><td>RpL3      </td><td>19347</td><td> 58621</td><td> 5215.0858</td><td>1898.14933</td><td>1.457338</td><td>17.99630</td></tr>\n",
       "\t<tr><th scope=row>LamC44</th><td>bq</td><td>LamC      </td><td> 2474</td><td>  5218</td><td>  668.7581</td><td> 176.29974</td><td>1.915294</td><td>17.97981</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tAct57B44 & bq & Act57B     & 24201 &  49032 &  6408.3207 & 1439.01846 & 2.153859 & 27.23761\\\\\n",
       "\tCG296244 & bq & CG2962     & 14862 &  28955 &  3852.2991 & 1015.30233 & 1.922390 & 22.89928\\\\\n",
       "\tMlc244 & bq & Mlc2       & 12441 &  28458 &  3299.5489 &  866.05437 & 1.928074 & 22.53628\\\\\n",
       "\teEF1alpha144 & bq & eEF1alpha1 & 76454 & 218993 & 20543.3019 & 7206.71742 & 1.511054 & 21.64804\\\\\n",
       "\tCG1573144 & bq & CG15731    &  2958 &   5182 &   782.8430 &  168.63752 & 2.206268 & 21.21199\\\\\n",
       "\tRpL23A44 & bq & RpL23A     & 23945 &  63966 &  6321.6670 & 2065.07794 & 1.613410 & 20.37143\\\\\n",
       "\tTwdlF44 & bq & TwdlF      &  6697 &  13997 &  1776.8570 &  492.98094 & 1.846800 & 19.93791\\\\\n",
       "\tRpL644 & bq & RpL6       & 22539 &  62854 &  5951.1989 & 1986.58711 & 1.582162 & 19.83906\\\\\n",
       "\tRpS644 & bq & RpS6       & 18692 &  50047 &  4918.5066 & 1630.96020 & 1.591614 & 19.52003\\\\\n",
       "\tHsc70-444 & bq & Hsc70-4    & 19542 &  54591 &  5251.8047 & 1790.73474 & 1.551456 & 19.17425\\\\\n",
       "\tRpL7A44 & bq & RpL7A      & 21373 &  62912 &  5587.7042 & 1969.40485 & 1.503764 & 18.71930\\\\\n",
       "\teEF244 & bq & eEF2       & 27581 &  82036 &  7219.6870 & 2627.25166 & 1.457833 & 18.68639\\\\\n",
       "\tRpL454 & bq & RpL4       & 14388 &  40203 &  3855.1943 & 1299.19926 & 1.568071 & 18.68037\\\\\n",
       "\tGabat44 & bq & Gabat      &  1392 &   2557 &   375.6893 &   81.77015 & 2.182357 & 18.67494\\\\\n",
       "\tCG908344 & bq & CG9083     &  5361 &  11400 &  1406.4989 &  413.48149 & 1.762729 & 18.43624\\\\\n",
       "\tCG767544 & bq & CG7675     &  2668 &   5580 &   717.7727 &  186.19533 & 1.938983 & 18.39976\\\\\n",
       "\tTm244 & bq & Tm2        &  3845 &   9615 &  1041.4820 &  298.07987 & 1.800035 & 18.04681\\\\\n",
       "\tGdh44 & bq & Gdh        &  4394 &   9732 &  1180.6039 &  346.38616 & 1.764913 & 18.01364\\\\\n",
       "\tRpL3100 & bq & RpL3       & 19347 &  58621 &  5215.0858 & 1898.14933 & 1.457338 & 17.99630\\\\\n",
       "\tLamC44 & bq & LamC       &  2474 &   5218 &   668.7581 &  176.29974 & 1.915294 & 17.97981\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Act57B44 | bq | Act57B     | 24201 |  49032 |  6408.3207 | 1439.01846 | 2.153859 | 27.23761 |\n",
       "| CG296244 | bq | CG2962     | 14862 |  28955 |  3852.2991 | 1015.30233 | 1.922390 | 22.89928 |\n",
       "| Mlc244 | bq | Mlc2       | 12441 |  28458 |  3299.5489 |  866.05437 | 1.928074 | 22.53628 |\n",
       "| eEF1alpha144 | bq | eEF1alpha1 | 76454 | 218993 | 20543.3019 | 7206.71742 | 1.511054 | 21.64804 |\n",
       "| CG1573144 | bq | CG15731    |  2958 |   5182 |   782.8430 |  168.63752 | 2.206268 | 21.21199 |\n",
       "| RpL23A44 | bq | RpL23A     | 23945 |  63966 |  6321.6670 | 2065.07794 | 1.613410 | 20.37143 |\n",
       "| TwdlF44 | bq | TwdlF      |  6697 |  13997 |  1776.8570 |  492.98094 | 1.846800 | 19.93791 |\n",
       "| RpL644 | bq | RpL6       | 22539 |  62854 |  5951.1989 | 1986.58711 | 1.582162 | 19.83906 |\n",
       "| RpS644 | bq | RpS6       | 18692 |  50047 |  4918.5066 | 1630.96020 | 1.591614 | 19.52003 |\n",
       "| Hsc70-444 | bq | Hsc70-4    | 19542 |  54591 |  5251.8047 | 1790.73474 | 1.551456 | 19.17425 |\n",
       "| RpL7A44 | bq | RpL7A      | 21373 |  62912 |  5587.7042 | 1969.40485 | 1.503764 | 18.71930 |\n",
       "| eEF244 | bq | eEF2       | 27581 |  82036 |  7219.6870 | 2627.25166 | 1.457833 | 18.68639 |\n",
       "| RpL454 | bq | RpL4       | 14388 |  40203 |  3855.1943 | 1299.19926 | 1.568071 | 18.68037 |\n",
       "| Gabat44 | bq | Gabat      |  1392 |   2557 |   375.6893 |   81.77015 | 2.182357 | 18.67494 |\n",
       "| CG908344 | bq | CG9083     |  5361 |  11400 |  1406.4989 |  413.48149 | 1.762729 | 18.43624 |\n",
       "| CG767544 | bq | CG7675     |  2668 |   5580 |   717.7727 |  186.19533 | 1.938983 | 18.39976 |\n",
       "| Tm244 | bq | Tm2        |  3845 |   9615 |  1041.4820 |  298.07987 | 1.800035 | 18.04681 |\n",
       "| Gdh44 | bq | Gdh        |  4394 |   9732 |  1180.6039 |  346.38616 | 1.764913 | 18.01364 |\n",
       "| RpL3100 | bq | RpL3       | 19347 |  58621 |  5215.0858 | 1898.14933 | 1.457338 | 17.99630 |\n",
       "| LamC44 | bq | LamC       |  2474 |   5218 |   668.7581 |  176.29974 | 1.915294 | 17.97981 |\n",
       "\n"
      ],
      "text/plain": [
       "             cell.type.2 gene       cell.type.2.n.umi other.n.umi\n",
       "Act57B44     bq          Act57B     24201              49032     \n",
       "CG296244     bq          CG2962     14862              28955     \n",
       "Mlc244       bq          Mlc2       12441              28458     \n",
       "eEF1alpha144 bq          eEF1alpha1 76454             218993     \n",
       "CG1573144    bq          CG15731     2958               5182     \n",
       "RpL23A44     bq          RpL23A     23945              63966     \n",
       "TwdlF44      bq          TwdlF       6697              13997     \n",
       "RpL644       bq          RpL6       22539              62854     \n",
       "RpS644       bq          RpS6       18692              50047     \n",
       "Hsc70-444    bq          Hsc70-4    19542              54591     \n",
       "RpL7A44      bq          RpL7A      21373              62912     \n",
       "eEF244       bq          eEF2       27581              82036     \n",
       "RpL454       bq          RpL4       14388              40203     \n",
       "Gabat44      bq          Gabat       1392               2557     \n",
       "CG908344     bq          CG9083      5361              11400     \n",
       "CG767544     bq          CG7675      2668               5580     \n",
       "Tm244        bq          Tm2         3845               9615     \n",
       "Gdh44        bq          Gdh         4394               9732     \n",
       "RpL3100      bq          RpL3       19347              58621     \n",
       "LamC44       bq          LamC        2474               5218     \n",
       "             cell.type.2.tpm other.tpm  log2.ratio heuristic.score\n",
       "Act57B44      6408.3207      1439.01846 2.153859   27.23761       \n",
       "CG296244      3852.2991      1015.30233 1.922390   22.89928       \n",
       "Mlc244        3299.5489       866.05437 1.928074   22.53628       \n",
       "eEF1alpha144 20543.3019      7206.71742 1.511054   21.64804       \n",
       "CG1573144      782.8430       168.63752 2.206268   21.21199       \n",
       "RpL23A44      6321.6670      2065.07794 1.613410   20.37143       \n",
       "TwdlF44       1776.8570       492.98094 1.846800   19.93791       \n",
       "RpL644        5951.1989      1986.58711 1.582162   19.83906       \n",
       "RpS644        4918.5066      1630.96020 1.591614   19.52003       \n",
       "Hsc70-444     5251.8047      1790.73474 1.551456   19.17425       \n",
       "RpL7A44       5587.7042      1969.40485 1.503764   18.71930       \n",
       "eEF244        7219.6870      2627.25166 1.457833   18.68639       \n",
       "RpL454        3855.1943      1299.19926 1.568071   18.68037       \n",
       "Gabat44        375.6893        81.77015 2.182357   18.67494       \n",
       "CG908344      1406.4989       413.48149 1.762729   18.43624       \n",
       "CG767544       717.7727       186.19533 1.938983   18.39976       \n",
       "Tm244         1041.4820       298.07987 1.800035   18.04681       \n",
       "Gdh44         1180.6039       346.38616 1.764913   18.01364       \n",
       "RpL3100       5215.0858      1898.14933 1.457338   17.99630       \n",
       "LamC44         668.7581       176.29974 1.915294   17.97981       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \n",
    "                                              \"bq\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1655,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 42392\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Act57B</th><td>Act57B       </td><td>148265</td><td>28363</td><td>5202.0650</td><td> 877.41574</td><td>Set 1</td><td>2.566109</td><td>31.67899</td></tr>\n",
       "\t<tr><th scope=row>CG2962</th><td>CG2962       </td><td> 99075</td><td>12169</td><td>3508.8653</td><td> 602.85115</td><td>Set 1</td><td>2.538740</td><td>29.89924</td></tr>\n",
       "\t<tr><th scope=row>CG15731</th><td>CG15731      </td><td> 19656</td><td> 1894</td><td> 689.3785</td><td>  86.86101</td><td>Set 1</td><td>2.972001</td><td>28.02967</td></tr>\n",
       "\t<tr><th scope=row>Mlc2</th><td>Mlc2         </td><td> 79663</td><td>17297</td><td>2743.0415</td><td> 561.59088</td><td>Set 1</td><td>2.285618</td><td>26.10653</td></tr>\n",
       "\t<tr><th scope=row>TwdlF</th><td>TwdlF        </td><td> 45244</td><td> 6161</td><td>1582.2296</td><td> 312.78750</td><td>Set 1</td><td>2.334099</td><td>24.80833</td></tr>\n",
       "\t<tr><th scope=row>Ipod</th><td>Ipod         </td><td> 33599</td><td> 4423</td><td>1166.9931</td><td> 224.21123</td><td>Set 1</td><td>2.373445</td><td>24.18497</td></tr>\n",
       "\t<tr><th scope=row>Neurochondrin</th><td>Neurochondrin</td><td> 15657</td><td> 2254</td><td> 545.3709</td><td>  93.77742</td><td>Set 1</td><td>2.524622</td><td>22.95825</td></tr>\n",
       "\t<tr><th scope=row>CG9083</th><td>CG9083       </td><td> 35113</td><td> 5337</td><td>1244.7225</td><td> 271.68212</td><td>Set 1</td><td>2.190532</td><td>22.52473</td></tr>\n",
       "\t<tr><th scope=row>CG15740</th><td>CG15740      </td><td> 10511</td><td> 1200</td><td> 366.7385</td><td>  58.21386</td><td>Set 1</td><td>2.630745</td><td>22.42062</td></tr>\n",
       "\t<tr><th scope=row>TwdlM</th><td>TwdlM        </td><td> 11652</td><td> 1573</td><td> 395.1466</td><td>  69.44464</td><td>Set 1</td><td>2.487826</td><td>21.46967</td></tr>\n",
       "\t<tr><th scope=row>Vajk4</th><td>Vajk4        </td><td> 12855</td><td> 1615</td><td> 431.1784</td><td>  80.46736</td><td>Set 1</td><td>2.403991</td><td>21.04810</td></tr>\n",
       "\t<tr><th scope=row>RpL23A</th><td>RpL23A       </td><td>149790</td><td>43447</td><td>5205.2589</td><td>1601.83560</td><td>Set 1</td><td>1.699343</td><td>20.98015</td></tr>\n",
       "\t<tr><th scope=row>pAbp</th><td>pAbp         </td><td>262026</td><td>70653</td><td>7896.0454</td><td>2582.30659</td><td>Set 1</td><td>1.611911</td><td>20.86957</td></tr>\n",
       "\t<tr><th scope=row>Tm2</th><td>Tm2          </td><td> 25555</td><td> 6125</td><td> 887.7708</td><td> 203.88137</td><td>Set 1</td><td>2.115398</td><td>20.72174</td></tr>\n",
       "\t<tr><th scope=row>TwdlD</th><td>TwdlD        </td><td> 10267</td><td> 1273</td><td> 349.4682</td><td>  63.44438</td><td>Set 1</td><td>2.439035</td><td>20.61750</td></tr>\n",
       "\t<tr><th scope=row>CG15022</th><td>CG15022      </td><td> 14823</td><td> 1939</td><td> 501.9904</td><td> 101.67095</td><td>Set 1</td><td>2.289632</td><td>20.54804</td></tr>\n",
       "\t<tr><th scope=row>Gabat</th><td>Gabat        </td><td>  8129</td><td> 1223</td><td> 288.2433</td><td>  50.03099</td><td>Set 1</td><td>2.497842</td><td>20.42270</td></tr>\n",
       "\t<tr><th scope=row>slbo</th><td>slbo         </td><td> 22290</td><td> 3410</td><td> 755.3329</td><td> 171.20036</td><td>Set 1</td><td>2.133024</td><td>20.39785</td></tr>\n",
       "\t<tr><th scope=row>CG7675</th><td>CG7675       </td><td> 17068</td><td> 2898</td><td> 591.7014</td><td> 129.40820</td><td>Set 1</td><td>2.181835</td><td>20.09723</td></tr>\n",
       "\t<tr><th scope=row>RpL6</th><td>RpL6         </td><td>143914</td><td>44043</td><td>4937.6167</td><td>1588.14555</td><td>Set 1</td><td>1.635564</td><td>20.06819</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tAct57B & Act57B        & 148265 & 28363 & 5202.0650 &  877.41574 & Set 1 & 2.566109 & 31.67899\\\\\n",
       "\tCG2962 & CG2962        &  99075 & 12169 & 3508.8653 &  602.85115 & Set 1 & 2.538740 & 29.89924\\\\\n",
       "\tCG15731 & CG15731       &  19656 &  1894 &  689.3785 &   86.86101 & Set 1 & 2.972001 & 28.02967\\\\\n",
       "\tMlc2 & Mlc2          &  79663 & 17297 & 2743.0415 &  561.59088 & Set 1 & 2.285618 & 26.10653\\\\\n",
       "\tTwdlF & TwdlF         &  45244 &  6161 & 1582.2296 &  312.78750 & Set 1 & 2.334099 & 24.80833\\\\\n",
       "\tIpod & Ipod          &  33599 &  4423 & 1166.9931 &  224.21123 & Set 1 & 2.373445 & 24.18497\\\\\n",
       "\tNeurochondrin & Neurochondrin &  15657 &  2254 &  545.3709 &   93.77742 & Set 1 & 2.524622 & 22.95825\\\\\n",
       "\tCG9083 & CG9083        &  35113 &  5337 & 1244.7225 &  271.68212 & Set 1 & 2.190532 & 22.52473\\\\\n",
       "\tCG15740 & CG15740       &  10511 &  1200 &  366.7385 &   58.21386 & Set 1 & 2.630745 & 22.42062\\\\\n",
       "\tTwdlM & TwdlM         &  11652 &  1573 &  395.1466 &   69.44464 & Set 1 & 2.487826 & 21.46967\\\\\n",
       "\tVajk4 & Vajk4         &  12855 &  1615 &  431.1784 &   80.46736 & Set 1 & 2.403991 & 21.04810\\\\\n",
       "\tRpL23A & RpL23A        & 149790 & 43447 & 5205.2589 & 1601.83560 & Set 1 & 1.699343 & 20.98015\\\\\n",
       "\tpAbp & pAbp          & 262026 & 70653 & 7896.0454 & 2582.30659 & Set 1 & 1.611911 & 20.86957\\\\\n",
       "\tTm2 & Tm2           &  25555 &  6125 &  887.7708 &  203.88137 & Set 1 & 2.115398 & 20.72174\\\\\n",
       "\tTwdlD & TwdlD         &  10267 &  1273 &  349.4682 &   63.44438 & Set 1 & 2.439035 & 20.61750\\\\\n",
       "\tCG15022 & CG15022       &  14823 &  1939 &  501.9904 &  101.67095 & Set 1 & 2.289632 & 20.54804\\\\\n",
       "\tGabat & Gabat         &   8129 &  1223 &  288.2433 &   50.03099 & Set 1 & 2.497842 & 20.42270\\\\\n",
       "\tslbo & slbo          &  22290 &  3410 &  755.3329 &  171.20036 & Set 1 & 2.133024 & 20.39785\\\\\n",
       "\tCG7675 & CG7675        &  17068 &  2898 &  591.7014 &  129.40820 & Set 1 & 2.181835 & 20.09723\\\\\n",
       "\tRpL6 & RpL6          & 143914 & 44043 & 4937.6167 & 1588.14555 & Set 1 & 1.635564 & 20.06819\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Act57B | Act57B        | 148265 | 28363 | 5202.0650 |  877.41574 | Set 1 | 2.566109 | 31.67899 |\n",
       "| CG2962 | CG2962        |  99075 | 12169 | 3508.8653 |  602.85115 | Set 1 | 2.538740 | 29.89924 |\n",
       "| CG15731 | CG15731       |  19656 |  1894 |  689.3785 |   86.86101 | Set 1 | 2.972001 | 28.02967 |\n",
       "| Mlc2 | Mlc2          |  79663 | 17297 | 2743.0415 |  561.59088 | Set 1 | 2.285618 | 26.10653 |\n",
       "| TwdlF | TwdlF         |  45244 |  6161 | 1582.2296 |  312.78750 | Set 1 | 2.334099 | 24.80833 |\n",
       "| Ipod | Ipod          |  33599 |  4423 | 1166.9931 |  224.21123 | Set 1 | 2.373445 | 24.18497 |\n",
       "| Neurochondrin | Neurochondrin |  15657 |  2254 |  545.3709 |   93.77742 | Set 1 | 2.524622 | 22.95825 |\n",
       "| CG9083 | CG9083        |  35113 |  5337 | 1244.7225 |  271.68212 | Set 1 | 2.190532 | 22.52473 |\n",
       "| CG15740 | CG15740       |  10511 |  1200 |  366.7385 |   58.21386 | Set 1 | 2.630745 | 22.42062 |\n",
       "| TwdlM | TwdlM         |  11652 |  1573 |  395.1466 |   69.44464 | Set 1 | 2.487826 | 21.46967 |\n",
       "| Vajk4 | Vajk4         |  12855 |  1615 |  431.1784 |   80.46736 | Set 1 | 2.403991 | 21.04810 |\n",
       "| RpL23A | RpL23A        | 149790 | 43447 | 5205.2589 | 1601.83560 | Set 1 | 1.699343 | 20.98015 |\n",
       "| pAbp | pAbp          | 262026 | 70653 | 7896.0454 | 2582.30659 | Set 1 | 1.611911 | 20.86957 |\n",
       "| Tm2 | Tm2           |  25555 |  6125 |  887.7708 |  203.88137 | Set 1 | 2.115398 | 20.72174 |\n",
       "| TwdlD | TwdlD         |  10267 |  1273 |  349.4682 |   63.44438 | Set 1 | 2.439035 | 20.61750 |\n",
       "| CG15022 | CG15022       |  14823 |  1939 |  501.9904 |  101.67095 | Set 1 | 2.289632 | 20.54804 |\n",
       "| Gabat | Gabat         |   8129 |  1223 |  288.2433 |   50.03099 | Set 1 | 2.497842 | 20.42270 |\n",
       "| slbo | slbo          |  22290 |  3410 |  755.3329 |  171.20036 | Set 1 | 2.133024 | 20.39785 |\n",
       "| CG7675 | CG7675        |  17068 |  2898 |  591.7014 |  129.40820 | Set 1 | 2.181835 | 20.09723 |\n",
       "| RpL6 | RpL6          | 143914 | 44043 | 4937.6167 | 1588.14555 | Set 1 | 1.635564 | 20.06819 |\n",
       "\n"
      ],
      "text/plain": [
       "              gene          set.1.n.umi set.2.n.umi set.1.tpm set.2.tpm \n",
       "Act57B        Act57B        148265      28363       5202.0650  877.41574\n",
       "CG2962        CG2962         99075      12169       3508.8653  602.85115\n",
       "CG15731       CG15731        19656       1894        689.3785   86.86101\n",
       "Mlc2          Mlc2           79663      17297       2743.0415  561.59088\n",
       "TwdlF         TwdlF          45244       6161       1582.2296  312.78750\n",
       "Ipod          Ipod           33599       4423       1166.9931  224.21123\n",
       "Neurochondrin Neurochondrin  15657       2254        545.3709   93.77742\n",
       "CG9083        CG9083         35113       5337       1244.7225  271.68212\n",
       "CG15740       CG15740        10511       1200        366.7385   58.21386\n",
       "TwdlM         TwdlM          11652       1573        395.1466   69.44464\n",
       "Vajk4         Vajk4          12855       1615        431.1784   80.46736\n",
       "RpL23A        RpL23A        149790      43447       5205.2589 1601.83560\n",
       "pAbp          pAbp          262026      70653       7896.0454 2582.30659\n",
       "Tm2           Tm2            25555       6125        887.7708  203.88137\n",
       "TwdlD         TwdlD          10267       1273        349.4682   63.44438\n",
       "CG15022       CG15022        14823       1939        501.9904  101.67095\n",
       "Gabat         Gabat           8129       1223        288.2433   50.03099\n",
       "slbo          slbo           22290       3410        755.3329  171.20036\n",
       "CG7675        CG7675         17068       2898        591.7014  129.40820\n",
       "RpL6          RpL6          143914      44043       4937.6167 1588.14555\n",
       "              higher.expr log2.ratio heuristic.score\n",
       "Act57B        Set 1       2.566109   31.67899       \n",
       "CG2962        Set 1       2.538740   29.89924       \n",
       "CG15731       Set 1       2.972001   28.02967       \n",
       "Mlc2          Set 1       2.285618   26.10653       \n",
       "TwdlF         Set 1       2.334099   24.80833       \n",
       "Ipod          Set 1       2.373445   24.18497       \n",
       "Neurochondrin Set 1       2.524622   22.95825       \n",
       "CG9083        Set 1       2.190532   22.52473       \n",
       "CG15740       Set 1       2.630745   22.42062       \n",
       "TwdlM         Set 1       2.487826   21.46967       \n",
       "Vajk4         Set 1       2.403991   21.04810       \n",
       "RpL23A        Set 1       1.699343   20.98015       \n",
       "pAbp          Set 1       1.611911   20.86957       \n",
       "Tm2           Set 1       2.115398   20.72174       \n",
       "TwdlD         Set 1       2.439035   20.61750       \n",
       "CG15022       Set 1       2.289632   20.54804       \n",
       "Gabat         Set 1       2.497842   20.42270       \n",
       "slbo          Set 1       2.133024   20.39785       \n",
       "CG7675        Set 1       2.181835   20.09723       \n",
       "RpL6          Set 1       1.635564   20.06819       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "two.set.differential.gene.test.3a(cds_sub, \n",
    "                                  df_sub300$cluster %in% c(bq, br,bs, bt, bu, bv, bw, bx, by) , \n",
    "                                 !(df_sub300$cluster %in% c(bq, br,bs, bt, bu, bv, bw, bx, by)),\n",
    "                                  sample = 50000\n",
    "                                )  %>% filter(higher.expr == \"Set 1\")  %>% \n",
    "head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 310,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Act57B\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Act57B\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Act57B\"), \n",
    "                   size_pos = 2, size_neg = 0.5, stroke = 0.008, limit = 18)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG15731\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Ipod\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 311,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Neurochondrin\"), : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Neurochondrin\"), : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Neurochondrin\"), \n",
    "                   size_pos = 2, size_neg = 0.5, stroke = 0.008, limit = 18)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Vajk4\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Gabat\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type br\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"br\", df_sub300$cluster %in% br)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1645,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Act57B44</th><td>bq</td><td>Act57B    </td><td>24201</td><td> 49032</td><td> 6408.3207</td><td>1439.01846</td><td>2.153859</td><td>27.23761</td></tr>\n",
       "\t<tr><th scope=row>CG296244</th><td>bq</td><td>CG2962    </td><td>14862</td><td> 28955</td><td> 3852.2991</td><td>1015.30233</td><td>1.922390</td><td>22.89928</td></tr>\n",
       "\t<tr><th scope=row>Mlc244</th><td>bq</td><td>Mlc2      </td><td>12441</td><td> 28458</td><td> 3299.5489</td><td> 866.05437</td><td>1.928074</td><td>22.53628</td></tr>\n",
       "\t<tr><th scope=row>eEF1alpha144</th><td>bq</td><td>eEF1alpha1</td><td>76454</td><td>218993</td><td>20543.3019</td><td>7206.71742</td><td>1.511054</td><td>21.64804</td></tr>\n",
       "\t<tr><th scope=row>CG1573144</th><td>bq</td><td>CG15731   </td><td> 2958</td><td>  5182</td><td>  782.8430</td><td> 168.63752</td><td>2.206268</td><td>21.21199</td></tr>\n",
       "\t<tr><th scope=row>RpL23A44</th><td>bq</td><td>RpL23A    </td><td>23945</td><td> 63966</td><td> 6321.6670</td><td>2065.07794</td><td>1.613410</td><td>20.37143</td></tr>\n",
       "\t<tr><th scope=row>TwdlF44</th><td>bq</td><td>TwdlF     </td><td> 6697</td><td> 13997</td><td> 1776.8570</td><td> 492.98094</td><td>1.846800</td><td>19.93791</td></tr>\n",
       "\t<tr><th scope=row>RpL644</th><td>bq</td><td>RpL6      </td><td>22539</td><td> 62854</td><td> 5951.1989</td><td>1986.58711</td><td>1.582162</td><td>19.83906</td></tr>\n",
       "\t<tr><th scope=row>RpS644</th><td>bq</td><td>RpS6      </td><td>18692</td><td> 50047</td><td> 4918.5066</td><td>1630.96020</td><td>1.591614</td><td>19.52003</td></tr>\n",
       "\t<tr><th scope=row>Hsc70-444</th><td>bq</td><td>Hsc70-4   </td><td>19542</td><td> 54591</td><td> 5251.8047</td><td>1790.73474</td><td>1.551456</td><td>19.17425</td></tr>\n",
       "\t<tr><th scope=row>RpL7A44</th><td>bq</td><td>RpL7A     </td><td>21373</td><td> 62912</td><td> 5587.7042</td><td>1969.40485</td><td>1.503764</td><td>18.71930</td></tr>\n",
       "\t<tr><th scope=row>eEF244</th><td>bq</td><td>eEF2      </td><td>27581</td><td> 82036</td><td> 7219.6870</td><td>2627.25166</td><td>1.457833</td><td>18.68639</td></tr>\n",
       "\t<tr><th scope=row>RpL454</th><td>bq</td><td>RpL4      </td><td>14388</td><td> 40203</td><td> 3855.1943</td><td>1299.19926</td><td>1.568071</td><td>18.68037</td></tr>\n",
       "\t<tr><th scope=row>Gabat44</th><td>bq</td><td>Gabat     </td><td> 1392</td><td>  2557</td><td>  375.6893</td><td>  81.77015</td><td>2.182357</td><td>18.67494</td></tr>\n",
       "\t<tr><th scope=row>CG908344</th><td>bq</td><td>CG9083    </td><td> 5361</td><td> 11400</td><td> 1406.4989</td><td> 413.48149</td><td>1.762729</td><td>18.43624</td></tr>\n",
       "\t<tr><th scope=row>CG767544</th><td>bq</td><td>CG7675    </td><td> 2668</td><td>  5580</td><td>  717.7727</td><td> 186.19533</td><td>1.938983</td><td>18.39976</td></tr>\n",
       "\t<tr><th scope=row>Tm244</th><td>bq</td><td>Tm2       </td><td> 3845</td><td>  9615</td><td> 1041.4820</td><td> 298.07987</td><td>1.800035</td><td>18.04681</td></tr>\n",
       "\t<tr><th scope=row>Gdh44</th><td>bq</td><td>Gdh       </td><td> 4394</td><td>  9732</td><td> 1180.6039</td><td> 346.38616</td><td>1.764913</td><td>18.01364</td></tr>\n",
       "\t<tr><th scope=row>RpL3100</th><td>bq</td><td>RpL3      </td><td>19347</td><td> 58621</td><td> 5215.0858</td><td>1898.14933</td><td>1.457338</td><td>17.99630</td></tr>\n",
       "\t<tr><th scope=row>LamC44</th><td>bq</td><td>LamC      </td><td> 2474</td><td>  5218</td><td>  668.7581</td><td> 176.29974</td><td>1.915294</td><td>17.97981</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tAct57B44 & bq & Act57B     & 24201 &  49032 &  6408.3207 & 1439.01846 & 2.153859 & 27.23761\\\\\n",
       "\tCG296244 & bq & CG2962     & 14862 &  28955 &  3852.2991 & 1015.30233 & 1.922390 & 22.89928\\\\\n",
       "\tMlc244 & bq & Mlc2       & 12441 &  28458 &  3299.5489 &  866.05437 & 1.928074 & 22.53628\\\\\n",
       "\teEF1alpha144 & bq & eEF1alpha1 & 76454 & 218993 & 20543.3019 & 7206.71742 & 1.511054 & 21.64804\\\\\n",
       "\tCG1573144 & bq & CG15731    &  2958 &   5182 &   782.8430 &  168.63752 & 2.206268 & 21.21199\\\\\n",
       "\tRpL23A44 & bq & RpL23A     & 23945 &  63966 &  6321.6670 & 2065.07794 & 1.613410 & 20.37143\\\\\n",
       "\tTwdlF44 & bq & TwdlF      &  6697 &  13997 &  1776.8570 &  492.98094 & 1.846800 & 19.93791\\\\\n",
       "\tRpL644 & bq & RpL6       & 22539 &  62854 &  5951.1989 & 1986.58711 & 1.582162 & 19.83906\\\\\n",
       "\tRpS644 & bq & RpS6       & 18692 &  50047 &  4918.5066 & 1630.96020 & 1.591614 & 19.52003\\\\\n",
       "\tHsc70-444 & bq & Hsc70-4    & 19542 &  54591 &  5251.8047 & 1790.73474 & 1.551456 & 19.17425\\\\\n",
       "\tRpL7A44 & bq & RpL7A      & 21373 &  62912 &  5587.7042 & 1969.40485 & 1.503764 & 18.71930\\\\\n",
       "\teEF244 & bq & eEF2       & 27581 &  82036 &  7219.6870 & 2627.25166 & 1.457833 & 18.68639\\\\\n",
       "\tRpL454 & bq & RpL4       & 14388 &  40203 &  3855.1943 & 1299.19926 & 1.568071 & 18.68037\\\\\n",
       "\tGabat44 & bq & Gabat      &  1392 &   2557 &   375.6893 &   81.77015 & 2.182357 & 18.67494\\\\\n",
       "\tCG908344 & bq & CG9083     &  5361 &  11400 &  1406.4989 &  413.48149 & 1.762729 & 18.43624\\\\\n",
       "\tCG767544 & bq & CG7675     &  2668 &   5580 &   717.7727 &  186.19533 & 1.938983 & 18.39976\\\\\n",
       "\tTm244 & bq & Tm2        &  3845 &   9615 &  1041.4820 &  298.07987 & 1.800035 & 18.04681\\\\\n",
       "\tGdh44 & bq & Gdh        &  4394 &   9732 &  1180.6039 &  346.38616 & 1.764913 & 18.01364\\\\\n",
       "\tRpL3100 & bq & RpL3       & 19347 &  58621 &  5215.0858 & 1898.14933 & 1.457338 & 17.99630\\\\\n",
       "\tLamC44 & bq & LamC       &  2474 &   5218 &   668.7581 &  176.29974 & 1.915294 & 17.97981\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Act57B44 | bq | Act57B     | 24201 |  49032 |  6408.3207 | 1439.01846 | 2.153859 | 27.23761 |\n",
       "| CG296244 | bq | CG2962     | 14862 |  28955 |  3852.2991 | 1015.30233 | 1.922390 | 22.89928 |\n",
       "| Mlc244 | bq | Mlc2       | 12441 |  28458 |  3299.5489 |  866.05437 | 1.928074 | 22.53628 |\n",
       "| eEF1alpha144 | bq | eEF1alpha1 | 76454 | 218993 | 20543.3019 | 7206.71742 | 1.511054 | 21.64804 |\n",
       "| CG1573144 | bq | CG15731    |  2958 |   5182 |   782.8430 |  168.63752 | 2.206268 | 21.21199 |\n",
       "| RpL23A44 | bq | RpL23A     | 23945 |  63966 |  6321.6670 | 2065.07794 | 1.613410 | 20.37143 |\n",
       "| TwdlF44 | bq | TwdlF      |  6697 |  13997 |  1776.8570 |  492.98094 | 1.846800 | 19.93791 |\n",
       "| RpL644 | bq | RpL6       | 22539 |  62854 |  5951.1989 | 1986.58711 | 1.582162 | 19.83906 |\n",
       "| RpS644 | bq | RpS6       | 18692 |  50047 |  4918.5066 | 1630.96020 | 1.591614 | 19.52003 |\n",
       "| Hsc70-444 | bq | Hsc70-4    | 19542 |  54591 |  5251.8047 | 1790.73474 | 1.551456 | 19.17425 |\n",
       "| RpL7A44 | bq | RpL7A      | 21373 |  62912 |  5587.7042 | 1969.40485 | 1.503764 | 18.71930 |\n",
       "| eEF244 | bq | eEF2       | 27581 |  82036 |  7219.6870 | 2627.25166 | 1.457833 | 18.68639 |\n",
       "| RpL454 | bq | RpL4       | 14388 |  40203 |  3855.1943 | 1299.19926 | 1.568071 | 18.68037 |\n",
       "| Gabat44 | bq | Gabat      |  1392 |   2557 |   375.6893 |   81.77015 | 2.182357 | 18.67494 |\n",
       "| CG908344 | bq | CG9083     |  5361 |  11400 |  1406.4989 |  413.48149 | 1.762729 | 18.43624 |\n",
       "| CG767544 | bq | CG7675     |  2668 |   5580 |   717.7727 |  186.19533 | 1.938983 | 18.39976 |\n",
       "| Tm244 | bq | Tm2        |  3845 |   9615 |  1041.4820 |  298.07987 | 1.800035 | 18.04681 |\n",
       "| Gdh44 | bq | Gdh        |  4394 |   9732 |  1180.6039 |  346.38616 | 1.764913 | 18.01364 |\n",
       "| RpL3100 | bq | RpL3       | 19347 |  58621 |  5215.0858 | 1898.14933 | 1.457338 | 17.99630 |\n",
       "| LamC44 | bq | LamC       |  2474 |   5218 |   668.7581 |  176.29974 | 1.915294 | 17.97981 |\n",
       "\n"
      ],
      "text/plain": [
       "             cell.type.2 gene       cell.type.2.n.umi other.n.umi\n",
       "Act57B44     bq          Act57B     24201              49032     \n",
       "CG296244     bq          CG2962     14862              28955     \n",
       "Mlc244       bq          Mlc2       12441              28458     \n",
       "eEF1alpha144 bq          eEF1alpha1 76454             218993     \n",
       "CG1573144    bq          CG15731     2958               5182     \n",
       "RpL23A44     bq          RpL23A     23945              63966     \n",
       "TwdlF44      bq          TwdlF       6697              13997     \n",
       "RpL644       bq          RpL6       22539              62854     \n",
       "RpS644       bq          RpS6       18692              50047     \n",
       "Hsc70-444    bq          Hsc70-4    19542              54591     \n",
       "RpL7A44      bq          RpL7A      21373              62912     \n",
       "eEF244       bq          eEF2       27581              82036     \n",
       "RpL454       bq          RpL4       14388              40203     \n",
       "Gabat44      bq          Gabat       1392               2557     \n",
       "CG908344     bq          CG9083      5361              11400     \n",
       "CG767544     bq          CG7675      2668               5580     \n",
       "Tm244        bq          Tm2         3845               9615     \n",
       "Gdh44        bq          Gdh         4394               9732     \n",
       "RpL3100      bq          RpL3       19347              58621     \n",
       "LamC44       bq          LamC        2474               5218     \n",
       "             cell.type.2.tpm other.tpm  log2.ratio heuristic.score\n",
       "Act57B44      6408.3207      1439.01846 2.153859   27.23761       \n",
       "CG296244      3852.2991      1015.30233 1.922390   22.89928       \n",
       "Mlc244        3299.5489       866.05437 1.928074   22.53628       \n",
       "eEF1alpha144 20543.3019      7206.71742 1.511054   21.64804       \n",
       "CG1573144      782.8430       168.63752 2.206268   21.21199       \n",
       "RpL23A44      6321.6670      2065.07794 1.613410   20.37143       \n",
       "TwdlF44       1776.8570       492.98094 1.846800   19.93791       \n",
       "RpL644        5951.1989      1986.58711 1.582162   19.83906       \n",
       "RpS644        4918.5066      1630.96020 1.591614   19.52003       \n",
       "Hsc70-444     5251.8047      1790.73474 1.551456   19.17425       \n",
       "RpL7A44       5587.7042      1969.40485 1.503764   18.71930       \n",
       "eEF244        7219.6870      2627.25166 1.457833   18.68639       \n",
       "RpL454        3855.1943      1299.19926 1.568071   18.68037       \n",
       "Gabat44        375.6893        81.77015 2.182357   18.67494       \n",
       "CG908344      1406.4989       413.48149 1.762729   18.43624       \n",
       "CG767544       717.7727       186.19533 1.938983   18.39976       \n",
       "Tm244         1041.4820       298.07987 1.800035   18.04681       \n",
       "Gdh44         1180.6039       346.38616 1.764913   18.01364       \n",
       "RpL3100       5215.0858      1898.14933 1.457338   17.99630       \n",
       "LamC44         668.7581       176.29974 1.915294   17.97981       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \n",
    "                                              \"bq\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bs\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bs\", df_sub300$cluster %in% bs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1646,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Act57B44</th><td>bq</td><td>Act57B    </td><td>24201</td><td> 49032</td><td> 6408.3207</td><td>1439.01846</td><td>2.153859</td><td>27.23761</td></tr>\n",
       "\t<tr><th scope=row>CG296244</th><td>bq</td><td>CG2962    </td><td>14862</td><td> 28955</td><td> 3852.2991</td><td>1015.30233</td><td>1.922390</td><td>22.89928</td></tr>\n",
       "\t<tr><th scope=row>Mlc244</th><td>bq</td><td>Mlc2      </td><td>12441</td><td> 28458</td><td> 3299.5489</td><td> 866.05437</td><td>1.928074</td><td>22.53628</td></tr>\n",
       "\t<tr><th scope=row>eEF1alpha144</th><td>bq</td><td>eEF1alpha1</td><td>76454</td><td>218993</td><td>20543.3019</td><td>7206.71742</td><td>1.511054</td><td>21.64804</td></tr>\n",
       "\t<tr><th scope=row>CG1573144</th><td>bq</td><td>CG15731   </td><td> 2958</td><td>  5182</td><td>  782.8430</td><td> 168.63752</td><td>2.206268</td><td>21.21199</td></tr>\n",
       "\t<tr><th scope=row>RpL23A44</th><td>bq</td><td>RpL23A    </td><td>23945</td><td> 63966</td><td> 6321.6670</td><td>2065.07794</td><td>1.613410</td><td>20.37143</td></tr>\n",
       "\t<tr><th scope=row>TwdlF44</th><td>bq</td><td>TwdlF     </td><td> 6697</td><td> 13997</td><td> 1776.8570</td><td> 492.98094</td><td>1.846800</td><td>19.93791</td></tr>\n",
       "\t<tr><th scope=row>RpL644</th><td>bq</td><td>RpL6      </td><td>22539</td><td> 62854</td><td> 5951.1989</td><td>1986.58711</td><td>1.582162</td><td>19.83906</td></tr>\n",
       "\t<tr><th scope=row>RpS644</th><td>bq</td><td>RpS6      </td><td>18692</td><td> 50047</td><td> 4918.5066</td><td>1630.96020</td><td>1.591614</td><td>19.52003</td></tr>\n",
       "\t<tr><th scope=row>Hsc70-444</th><td>bq</td><td>Hsc70-4   </td><td>19542</td><td> 54591</td><td> 5251.8047</td><td>1790.73474</td><td>1.551456</td><td>19.17425</td></tr>\n",
       "\t<tr><th scope=row>RpL7A44</th><td>bq</td><td>RpL7A     </td><td>21373</td><td> 62912</td><td> 5587.7042</td><td>1969.40485</td><td>1.503764</td><td>18.71930</td></tr>\n",
       "\t<tr><th scope=row>eEF244</th><td>bq</td><td>eEF2      </td><td>27581</td><td> 82036</td><td> 7219.6870</td><td>2627.25166</td><td>1.457833</td><td>18.68639</td></tr>\n",
       "\t<tr><th scope=row>RpL454</th><td>bq</td><td>RpL4      </td><td>14388</td><td> 40203</td><td> 3855.1943</td><td>1299.19926</td><td>1.568071</td><td>18.68037</td></tr>\n",
       "\t<tr><th scope=row>Gabat44</th><td>bq</td><td>Gabat     </td><td> 1392</td><td>  2557</td><td>  375.6893</td><td>  81.77015</td><td>2.182357</td><td>18.67494</td></tr>\n",
       "\t<tr><th scope=row>CG908344</th><td>bq</td><td>CG9083    </td><td> 5361</td><td> 11400</td><td> 1406.4989</td><td> 413.48149</td><td>1.762729</td><td>18.43624</td></tr>\n",
       "\t<tr><th scope=row>CG767544</th><td>bq</td><td>CG7675    </td><td> 2668</td><td>  5580</td><td>  717.7727</td><td> 186.19533</td><td>1.938983</td><td>18.39976</td></tr>\n",
       "\t<tr><th scope=row>Tm244</th><td>bq</td><td>Tm2       </td><td> 3845</td><td>  9615</td><td> 1041.4820</td><td> 298.07987</td><td>1.800035</td><td>18.04681</td></tr>\n",
       "\t<tr><th scope=row>Gdh44</th><td>bq</td><td>Gdh       </td><td> 4394</td><td>  9732</td><td> 1180.6039</td><td> 346.38616</td><td>1.764913</td><td>18.01364</td></tr>\n",
       "\t<tr><th scope=row>RpL3100</th><td>bq</td><td>RpL3      </td><td>19347</td><td> 58621</td><td> 5215.0858</td><td>1898.14933</td><td>1.457338</td><td>17.99630</td></tr>\n",
       "\t<tr><th scope=row>LamC44</th><td>bq</td><td>LamC      </td><td> 2474</td><td>  5218</td><td>  668.7581</td><td> 176.29974</td><td>1.915294</td><td>17.97981</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tAct57B44 & bq & Act57B     & 24201 &  49032 &  6408.3207 & 1439.01846 & 2.153859 & 27.23761\\\\\n",
       "\tCG296244 & bq & CG2962     & 14862 &  28955 &  3852.2991 & 1015.30233 & 1.922390 & 22.89928\\\\\n",
       "\tMlc244 & bq & Mlc2       & 12441 &  28458 &  3299.5489 &  866.05437 & 1.928074 & 22.53628\\\\\n",
       "\teEF1alpha144 & bq & eEF1alpha1 & 76454 & 218993 & 20543.3019 & 7206.71742 & 1.511054 & 21.64804\\\\\n",
       "\tCG1573144 & bq & CG15731    &  2958 &   5182 &   782.8430 &  168.63752 & 2.206268 & 21.21199\\\\\n",
       "\tRpL23A44 & bq & RpL23A     & 23945 &  63966 &  6321.6670 & 2065.07794 & 1.613410 & 20.37143\\\\\n",
       "\tTwdlF44 & bq & TwdlF      &  6697 &  13997 &  1776.8570 &  492.98094 & 1.846800 & 19.93791\\\\\n",
       "\tRpL644 & bq & RpL6       & 22539 &  62854 &  5951.1989 & 1986.58711 & 1.582162 & 19.83906\\\\\n",
       "\tRpS644 & bq & RpS6       & 18692 &  50047 &  4918.5066 & 1630.96020 & 1.591614 & 19.52003\\\\\n",
       "\tHsc70-444 & bq & Hsc70-4    & 19542 &  54591 &  5251.8047 & 1790.73474 & 1.551456 & 19.17425\\\\\n",
       "\tRpL7A44 & bq & RpL7A      & 21373 &  62912 &  5587.7042 & 1969.40485 & 1.503764 & 18.71930\\\\\n",
       "\teEF244 & bq & eEF2       & 27581 &  82036 &  7219.6870 & 2627.25166 & 1.457833 & 18.68639\\\\\n",
       "\tRpL454 & bq & RpL4       & 14388 &  40203 &  3855.1943 & 1299.19926 & 1.568071 & 18.68037\\\\\n",
       "\tGabat44 & bq & Gabat      &  1392 &   2557 &   375.6893 &   81.77015 & 2.182357 & 18.67494\\\\\n",
       "\tCG908344 & bq & CG9083     &  5361 &  11400 &  1406.4989 &  413.48149 & 1.762729 & 18.43624\\\\\n",
       "\tCG767544 & bq & CG7675     &  2668 &   5580 &   717.7727 &  186.19533 & 1.938983 & 18.39976\\\\\n",
       "\tTm244 & bq & Tm2        &  3845 &   9615 &  1041.4820 &  298.07987 & 1.800035 & 18.04681\\\\\n",
       "\tGdh44 & bq & Gdh        &  4394 &   9732 &  1180.6039 &  346.38616 & 1.764913 & 18.01364\\\\\n",
       "\tRpL3100 & bq & RpL3       & 19347 &  58621 &  5215.0858 & 1898.14933 & 1.457338 & 17.99630\\\\\n",
       "\tLamC44 & bq & LamC       &  2474 &   5218 &   668.7581 &  176.29974 & 1.915294 & 17.97981\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Act57B44 | bq | Act57B     | 24201 |  49032 |  6408.3207 | 1439.01846 | 2.153859 | 27.23761 |\n",
       "| CG296244 | bq | CG2962     | 14862 |  28955 |  3852.2991 | 1015.30233 | 1.922390 | 22.89928 |\n",
       "| Mlc244 | bq | Mlc2       | 12441 |  28458 |  3299.5489 |  866.05437 | 1.928074 | 22.53628 |\n",
       "| eEF1alpha144 | bq | eEF1alpha1 | 76454 | 218993 | 20543.3019 | 7206.71742 | 1.511054 | 21.64804 |\n",
       "| CG1573144 | bq | CG15731    |  2958 |   5182 |   782.8430 |  168.63752 | 2.206268 | 21.21199 |\n",
       "| RpL23A44 | bq | RpL23A     | 23945 |  63966 |  6321.6670 | 2065.07794 | 1.613410 | 20.37143 |\n",
       "| TwdlF44 | bq | TwdlF      |  6697 |  13997 |  1776.8570 |  492.98094 | 1.846800 | 19.93791 |\n",
       "| RpL644 | bq | RpL6       | 22539 |  62854 |  5951.1989 | 1986.58711 | 1.582162 | 19.83906 |\n",
       "| RpS644 | bq | RpS6       | 18692 |  50047 |  4918.5066 | 1630.96020 | 1.591614 | 19.52003 |\n",
       "| Hsc70-444 | bq | Hsc70-4    | 19542 |  54591 |  5251.8047 | 1790.73474 | 1.551456 | 19.17425 |\n",
       "| RpL7A44 | bq | RpL7A      | 21373 |  62912 |  5587.7042 | 1969.40485 | 1.503764 | 18.71930 |\n",
       "| eEF244 | bq | eEF2       | 27581 |  82036 |  7219.6870 | 2627.25166 | 1.457833 | 18.68639 |\n",
       "| RpL454 | bq | RpL4       | 14388 |  40203 |  3855.1943 | 1299.19926 | 1.568071 | 18.68037 |\n",
       "| Gabat44 | bq | Gabat      |  1392 |   2557 |   375.6893 |   81.77015 | 2.182357 | 18.67494 |\n",
       "| CG908344 | bq | CG9083     |  5361 |  11400 |  1406.4989 |  413.48149 | 1.762729 | 18.43624 |\n",
       "| CG767544 | bq | CG7675     |  2668 |   5580 |   717.7727 |  186.19533 | 1.938983 | 18.39976 |\n",
       "| Tm244 | bq | Tm2        |  3845 |   9615 |  1041.4820 |  298.07987 | 1.800035 | 18.04681 |\n",
       "| Gdh44 | bq | Gdh        |  4394 |   9732 |  1180.6039 |  346.38616 | 1.764913 | 18.01364 |\n",
       "| RpL3100 | bq | RpL3       | 19347 |  58621 |  5215.0858 | 1898.14933 | 1.457338 | 17.99630 |\n",
       "| LamC44 | bq | LamC       |  2474 |   5218 |   668.7581 |  176.29974 | 1.915294 | 17.97981 |\n",
       "\n"
      ],
      "text/plain": [
       "             cell.type.2 gene       cell.type.2.n.umi other.n.umi\n",
       "Act57B44     bq          Act57B     24201              49032     \n",
       "CG296244     bq          CG2962     14862              28955     \n",
       "Mlc244       bq          Mlc2       12441              28458     \n",
       "eEF1alpha144 bq          eEF1alpha1 76454             218993     \n",
       "CG1573144    bq          CG15731     2958               5182     \n",
       "RpL23A44     bq          RpL23A     23945              63966     \n",
       "TwdlF44      bq          TwdlF       6697              13997     \n",
       "RpL644       bq          RpL6       22539              62854     \n",
       "RpS644       bq          RpS6       18692              50047     \n",
       "Hsc70-444    bq          Hsc70-4    19542              54591     \n",
       "RpL7A44      bq          RpL7A      21373              62912     \n",
       "eEF244       bq          eEF2       27581              82036     \n",
       "RpL454       bq          RpL4       14388              40203     \n",
       "Gabat44      bq          Gabat       1392               2557     \n",
       "CG908344     bq          CG9083      5361              11400     \n",
       "CG767544     bq          CG7675      2668               5580     \n",
       "Tm244        bq          Tm2         3845               9615     \n",
       "Gdh44        bq          Gdh         4394               9732     \n",
       "RpL3100      bq          RpL3       19347              58621     \n",
       "LamC44       bq          LamC        2474               5218     \n",
       "             cell.type.2.tpm other.tpm  log2.ratio heuristic.score\n",
       "Act57B44      6408.3207      1439.01846 2.153859   27.23761       \n",
       "CG296244      3852.2991      1015.30233 1.922390   22.89928       \n",
       "Mlc244        3299.5489       866.05437 1.928074   22.53628       \n",
       "eEF1alpha144 20543.3019      7206.71742 1.511054   21.64804       \n",
       "CG1573144      782.8430       168.63752 2.206268   21.21199       \n",
       "RpL23A44      6321.6670      2065.07794 1.613410   20.37143       \n",
       "TwdlF44       1776.8570       492.98094 1.846800   19.93791       \n",
       "RpL644        5951.1989      1986.58711 1.582162   19.83906       \n",
       "RpS644        4918.5066      1630.96020 1.591614   19.52003       \n",
       "Hsc70-444     5251.8047      1790.73474 1.551456   19.17425       \n",
       "RpL7A44       5587.7042      1969.40485 1.503764   18.71930       \n",
       "eEF244        7219.6870      2627.25166 1.457833   18.68639       \n",
       "RpL454        3855.1943      1299.19926 1.568071   18.68037       \n",
       "Gabat44        375.6893        81.77015 2.182357   18.67494       \n",
       "CG908344      1406.4989       413.48149 1.762729   18.43624       \n",
       "CG767544       717.7727       186.19533 1.938983   18.39976       \n",
       "Tm244         1041.4820       298.07987 1.800035   18.04681       \n",
       "Gdh44         1180.6039       346.38616 1.764913   18.01364       \n",
       "RpL3100       5215.0858      1898.14933 1.457338   17.99630       \n",
       "LamC44         668.7581       176.29974 1.915294   17.97981       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \n",
    "                                              \"bq\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bt\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bt\", df_sub300$cluster %in% bt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1647,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Act57B44</th><td>bq</td><td>Act57B    </td><td>24201</td><td> 49032</td><td> 6408.3207</td><td>1439.01846</td><td>2.153859</td><td>27.23761</td></tr>\n",
       "\t<tr><th scope=row>CG296244</th><td>bq</td><td>CG2962    </td><td>14862</td><td> 28955</td><td> 3852.2991</td><td>1015.30233</td><td>1.922390</td><td>22.89928</td></tr>\n",
       "\t<tr><th scope=row>Mlc244</th><td>bq</td><td>Mlc2      </td><td>12441</td><td> 28458</td><td> 3299.5489</td><td> 866.05437</td><td>1.928074</td><td>22.53628</td></tr>\n",
       "\t<tr><th scope=row>eEF1alpha144</th><td>bq</td><td>eEF1alpha1</td><td>76454</td><td>218993</td><td>20543.3019</td><td>7206.71742</td><td>1.511054</td><td>21.64804</td></tr>\n",
       "\t<tr><th scope=row>CG1573144</th><td>bq</td><td>CG15731   </td><td> 2958</td><td>  5182</td><td>  782.8430</td><td> 168.63752</td><td>2.206268</td><td>21.21199</td></tr>\n",
       "\t<tr><th scope=row>RpL23A44</th><td>bq</td><td>RpL23A    </td><td>23945</td><td> 63966</td><td> 6321.6670</td><td>2065.07794</td><td>1.613410</td><td>20.37143</td></tr>\n",
       "\t<tr><th scope=row>TwdlF44</th><td>bq</td><td>TwdlF     </td><td> 6697</td><td> 13997</td><td> 1776.8570</td><td> 492.98094</td><td>1.846800</td><td>19.93791</td></tr>\n",
       "\t<tr><th scope=row>RpL644</th><td>bq</td><td>RpL6      </td><td>22539</td><td> 62854</td><td> 5951.1989</td><td>1986.58711</td><td>1.582162</td><td>19.83906</td></tr>\n",
       "\t<tr><th scope=row>RpS644</th><td>bq</td><td>RpS6      </td><td>18692</td><td> 50047</td><td> 4918.5066</td><td>1630.96020</td><td>1.591614</td><td>19.52003</td></tr>\n",
       "\t<tr><th scope=row>Hsc70-444</th><td>bq</td><td>Hsc70-4   </td><td>19542</td><td> 54591</td><td> 5251.8047</td><td>1790.73474</td><td>1.551456</td><td>19.17425</td></tr>\n",
       "\t<tr><th scope=row>RpL7A44</th><td>bq</td><td>RpL7A     </td><td>21373</td><td> 62912</td><td> 5587.7042</td><td>1969.40485</td><td>1.503764</td><td>18.71930</td></tr>\n",
       "\t<tr><th scope=row>eEF244</th><td>bq</td><td>eEF2      </td><td>27581</td><td> 82036</td><td> 7219.6870</td><td>2627.25166</td><td>1.457833</td><td>18.68639</td></tr>\n",
       "\t<tr><th scope=row>RpL454</th><td>bq</td><td>RpL4      </td><td>14388</td><td> 40203</td><td> 3855.1943</td><td>1299.19926</td><td>1.568071</td><td>18.68037</td></tr>\n",
       "\t<tr><th scope=row>Gabat44</th><td>bq</td><td>Gabat     </td><td> 1392</td><td>  2557</td><td>  375.6893</td><td>  81.77015</td><td>2.182357</td><td>18.67494</td></tr>\n",
       "\t<tr><th scope=row>CG908344</th><td>bq</td><td>CG9083    </td><td> 5361</td><td> 11400</td><td> 1406.4989</td><td> 413.48149</td><td>1.762729</td><td>18.43624</td></tr>\n",
       "\t<tr><th scope=row>CG767544</th><td>bq</td><td>CG7675    </td><td> 2668</td><td>  5580</td><td>  717.7727</td><td> 186.19533</td><td>1.938983</td><td>18.39976</td></tr>\n",
       "\t<tr><th scope=row>Tm244</th><td>bq</td><td>Tm2       </td><td> 3845</td><td>  9615</td><td> 1041.4820</td><td> 298.07987</td><td>1.800035</td><td>18.04681</td></tr>\n",
       "\t<tr><th scope=row>Gdh44</th><td>bq</td><td>Gdh       </td><td> 4394</td><td>  9732</td><td> 1180.6039</td><td> 346.38616</td><td>1.764913</td><td>18.01364</td></tr>\n",
       "\t<tr><th scope=row>RpL3100</th><td>bq</td><td>RpL3      </td><td>19347</td><td> 58621</td><td> 5215.0858</td><td>1898.14933</td><td>1.457338</td><td>17.99630</td></tr>\n",
       "\t<tr><th scope=row>LamC44</th><td>bq</td><td>LamC      </td><td> 2474</td><td>  5218</td><td>  668.7581</td><td> 176.29974</td><td>1.915294</td><td>17.97981</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tAct57B44 & bq & Act57B     & 24201 &  49032 &  6408.3207 & 1439.01846 & 2.153859 & 27.23761\\\\\n",
       "\tCG296244 & bq & CG2962     & 14862 &  28955 &  3852.2991 & 1015.30233 & 1.922390 & 22.89928\\\\\n",
       "\tMlc244 & bq & Mlc2       & 12441 &  28458 &  3299.5489 &  866.05437 & 1.928074 & 22.53628\\\\\n",
       "\teEF1alpha144 & bq & eEF1alpha1 & 76454 & 218993 & 20543.3019 & 7206.71742 & 1.511054 & 21.64804\\\\\n",
       "\tCG1573144 & bq & CG15731    &  2958 &   5182 &   782.8430 &  168.63752 & 2.206268 & 21.21199\\\\\n",
       "\tRpL23A44 & bq & RpL23A     & 23945 &  63966 &  6321.6670 & 2065.07794 & 1.613410 & 20.37143\\\\\n",
       "\tTwdlF44 & bq & TwdlF      &  6697 &  13997 &  1776.8570 &  492.98094 & 1.846800 & 19.93791\\\\\n",
       "\tRpL644 & bq & RpL6       & 22539 &  62854 &  5951.1989 & 1986.58711 & 1.582162 & 19.83906\\\\\n",
       "\tRpS644 & bq & RpS6       & 18692 &  50047 &  4918.5066 & 1630.96020 & 1.591614 & 19.52003\\\\\n",
       "\tHsc70-444 & bq & Hsc70-4    & 19542 &  54591 &  5251.8047 & 1790.73474 & 1.551456 & 19.17425\\\\\n",
       "\tRpL7A44 & bq & RpL7A      & 21373 &  62912 &  5587.7042 & 1969.40485 & 1.503764 & 18.71930\\\\\n",
       "\teEF244 & bq & eEF2       & 27581 &  82036 &  7219.6870 & 2627.25166 & 1.457833 & 18.68639\\\\\n",
       "\tRpL454 & bq & RpL4       & 14388 &  40203 &  3855.1943 & 1299.19926 & 1.568071 & 18.68037\\\\\n",
       "\tGabat44 & bq & Gabat      &  1392 &   2557 &   375.6893 &   81.77015 & 2.182357 & 18.67494\\\\\n",
       "\tCG908344 & bq & CG9083     &  5361 &  11400 &  1406.4989 &  413.48149 & 1.762729 & 18.43624\\\\\n",
       "\tCG767544 & bq & CG7675     &  2668 &   5580 &   717.7727 &  186.19533 & 1.938983 & 18.39976\\\\\n",
       "\tTm244 & bq & Tm2        &  3845 &   9615 &  1041.4820 &  298.07987 & 1.800035 & 18.04681\\\\\n",
       "\tGdh44 & bq & Gdh        &  4394 &   9732 &  1180.6039 &  346.38616 & 1.764913 & 18.01364\\\\\n",
       "\tRpL3100 & bq & RpL3       & 19347 &  58621 &  5215.0858 & 1898.14933 & 1.457338 & 17.99630\\\\\n",
       "\tLamC44 & bq & LamC       &  2474 &   5218 &   668.7581 &  176.29974 & 1.915294 & 17.97981\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Act57B44 | bq | Act57B     | 24201 |  49032 |  6408.3207 | 1439.01846 | 2.153859 | 27.23761 |\n",
       "| CG296244 | bq | CG2962     | 14862 |  28955 |  3852.2991 | 1015.30233 | 1.922390 | 22.89928 |\n",
       "| Mlc244 | bq | Mlc2       | 12441 |  28458 |  3299.5489 |  866.05437 | 1.928074 | 22.53628 |\n",
       "| eEF1alpha144 | bq | eEF1alpha1 | 76454 | 218993 | 20543.3019 | 7206.71742 | 1.511054 | 21.64804 |\n",
       "| CG1573144 | bq | CG15731    |  2958 |   5182 |   782.8430 |  168.63752 | 2.206268 | 21.21199 |\n",
       "| RpL23A44 | bq | RpL23A     | 23945 |  63966 |  6321.6670 | 2065.07794 | 1.613410 | 20.37143 |\n",
       "| TwdlF44 | bq | TwdlF      |  6697 |  13997 |  1776.8570 |  492.98094 | 1.846800 | 19.93791 |\n",
       "| RpL644 | bq | RpL6       | 22539 |  62854 |  5951.1989 | 1986.58711 | 1.582162 | 19.83906 |\n",
       "| RpS644 | bq | RpS6       | 18692 |  50047 |  4918.5066 | 1630.96020 | 1.591614 | 19.52003 |\n",
       "| Hsc70-444 | bq | Hsc70-4    | 19542 |  54591 |  5251.8047 | 1790.73474 | 1.551456 | 19.17425 |\n",
       "| RpL7A44 | bq | RpL7A      | 21373 |  62912 |  5587.7042 | 1969.40485 | 1.503764 | 18.71930 |\n",
       "| eEF244 | bq | eEF2       | 27581 |  82036 |  7219.6870 | 2627.25166 | 1.457833 | 18.68639 |\n",
       "| RpL454 | bq | RpL4       | 14388 |  40203 |  3855.1943 | 1299.19926 | 1.568071 | 18.68037 |\n",
       "| Gabat44 | bq | Gabat      |  1392 |   2557 |   375.6893 |   81.77015 | 2.182357 | 18.67494 |\n",
       "| CG908344 | bq | CG9083     |  5361 |  11400 |  1406.4989 |  413.48149 | 1.762729 | 18.43624 |\n",
       "| CG767544 | bq | CG7675     |  2668 |   5580 |   717.7727 |  186.19533 | 1.938983 | 18.39976 |\n",
       "| Tm244 | bq | Tm2        |  3845 |   9615 |  1041.4820 |  298.07987 | 1.800035 | 18.04681 |\n",
       "| Gdh44 | bq | Gdh        |  4394 |   9732 |  1180.6039 |  346.38616 | 1.764913 | 18.01364 |\n",
       "| RpL3100 | bq | RpL3       | 19347 |  58621 |  5215.0858 | 1898.14933 | 1.457338 | 17.99630 |\n",
       "| LamC44 | bq | LamC       |  2474 |   5218 |   668.7581 |  176.29974 | 1.915294 | 17.97981 |\n",
       "\n"
      ],
      "text/plain": [
       "             cell.type.2 gene       cell.type.2.n.umi other.n.umi\n",
       "Act57B44     bq          Act57B     24201              49032     \n",
       "CG296244     bq          CG2962     14862              28955     \n",
       "Mlc244       bq          Mlc2       12441              28458     \n",
       "eEF1alpha144 bq          eEF1alpha1 76454             218993     \n",
       "CG1573144    bq          CG15731     2958               5182     \n",
       "RpL23A44     bq          RpL23A     23945              63966     \n",
       "TwdlF44      bq          TwdlF       6697              13997     \n",
       "RpL644       bq          RpL6       22539              62854     \n",
       "RpS644       bq          RpS6       18692              50047     \n",
       "Hsc70-444    bq          Hsc70-4    19542              54591     \n",
       "RpL7A44      bq          RpL7A      21373              62912     \n",
       "eEF244       bq          eEF2       27581              82036     \n",
       "RpL454       bq          RpL4       14388              40203     \n",
       "Gabat44      bq          Gabat       1392               2557     \n",
       "CG908344     bq          CG9083      5361              11400     \n",
       "CG767544     bq          CG7675      2668               5580     \n",
       "Tm244        bq          Tm2         3845               9615     \n",
       "Gdh44        bq          Gdh         4394               9732     \n",
       "RpL3100      bq          RpL3       19347              58621     \n",
       "LamC44       bq          LamC        2474               5218     \n",
       "             cell.type.2.tpm other.tpm  log2.ratio heuristic.score\n",
       "Act57B44      6408.3207      1439.01846 2.153859   27.23761       \n",
       "CG296244      3852.2991      1015.30233 1.922390   22.89928       \n",
       "Mlc244        3299.5489       866.05437 1.928074   22.53628       \n",
       "eEF1alpha144 20543.3019      7206.71742 1.511054   21.64804       \n",
       "CG1573144      782.8430       168.63752 2.206268   21.21199       \n",
       "RpL23A44      6321.6670      2065.07794 1.613410   20.37143       \n",
       "TwdlF44       1776.8570       492.98094 1.846800   19.93791       \n",
       "RpL644        5951.1989      1986.58711 1.582162   19.83906       \n",
       "RpS644        4918.5066      1630.96020 1.591614   19.52003       \n",
       "Hsc70-444     5251.8047      1790.73474 1.551456   19.17425       \n",
       "RpL7A44       5587.7042      1969.40485 1.503764   18.71930       \n",
       "eEF244        7219.6870      2627.25166 1.457833   18.68639       \n",
       "RpL454        3855.1943      1299.19926 1.568071   18.68037       \n",
       "Gabat44        375.6893        81.77015 2.182357   18.67494       \n",
       "CG908344      1406.4989       413.48149 1.762729   18.43624       \n",
       "CG767544       717.7727       186.19533 1.938983   18.39976       \n",
       "Tm244         1041.4820       298.07987 1.800035   18.04681       \n",
       "Gdh44         1180.6039       346.38616 1.764913   18.01364       \n",
       "RpL3100       5215.0858      1898.14933 1.457338   17.99630       \n",
       "LamC44         668.7581       176.29974 1.915294   17.97981       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \n",
    "                                              \"bq\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bu\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bu\", df_sub300$cluster %in% bu)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1648,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Act57B44</th><td>bq</td><td>Act57B    </td><td>24201</td><td> 49032</td><td> 6408.3207</td><td>1439.01846</td><td>2.153859</td><td>27.23761</td></tr>\n",
       "\t<tr><th scope=row>CG296244</th><td>bq</td><td>CG2962    </td><td>14862</td><td> 28955</td><td> 3852.2991</td><td>1015.30233</td><td>1.922390</td><td>22.89928</td></tr>\n",
       "\t<tr><th scope=row>Mlc244</th><td>bq</td><td>Mlc2      </td><td>12441</td><td> 28458</td><td> 3299.5489</td><td> 866.05437</td><td>1.928074</td><td>22.53628</td></tr>\n",
       "\t<tr><th scope=row>eEF1alpha144</th><td>bq</td><td>eEF1alpha1</td><td>76454</td><td>218993</td><td>20543.3019</td><td>7206.71742</td><td>1.511054</td><td>21.64804</td></tr>\n",
       "\t<tr><th scope=row>CG1573144</th><td>bq</td><td>CG15731   </td><td> 2958</td><td>  5182</td><td>  782.8430</td><td> 168.63752</td><td>2.206268</td><td>21.21199</td></tr>\n",
       "\t<tr><th scope=row>RpL23A44</th><td>bq</td><td>RpL23A    </td><td>23945</td><td> 63966</td><td> 6321.6670</td><td>2065.07794</td><td>1.613410</td><td>20.37143</td></tr>\n",
       "\t<tr><th scope=row>TwdlF44</th><td>bq</td><td>TwdlF     </td><td> 6697</td><td> 13997</td><td> 1776.8570</td><td> 492.98094</td><td>1.846800</td><td>19.93791</td></tr>\n",
       "\t<tr><th scope=row>RpL644</th><td>bq</td><td>RpL6      </td><td>22539</td><td> 62854</td><td> 5951.1989</td><td>1986.58711</td><td>1.582162</td><td>19.83906</td></tr>\n",
       "\t<tr><th scope=row>RpS644</th><td>bq</td><td>RpS6      </td><td>18692</td><td> 50047</td><td> 4918.5066</td><td>1630.96020</td><td>1.591614</td><td>19.52003</td></tr>\n",
       "\t<tr><th scope=row>Hsc70-444</th><td>bq</td><td>Hsc70-4   </td><td>19542</td><td> 54591</td><td> 5251.8047</td><td>1790.73474</td><td>1.551456</td><td>19.17425</td></tr>\n",
       "\t<tr><th scope=row>RpL7A44</th><td>bq</td><td>RpL7A     </td><td>21373</td><td> 62912</td><td> 5587.7042</td><td>1969.40485</td><td>1.503764</td><td>18.71930</td></tr>\n",
       "\t<tr><th scope=row>eEF244</th><td>bq</td><td>eEF2      </td><td>27581</td><td> 82036</td><td> 7219.6870</td><td>2627.25166</td><td>1.457833</td><td>18.68639</td></tr>\n",
       "\t<tr><th scope=row>RpL454</th><td>bq</td><td>RpL4      </td><td>14388</td><td> 40203</td><td> 3855.1943</td><td>1299.19926</td><td>1.568071</td><td>18.68037</td></tr>\n",
       "\t<tr><th scope=row>Gabat44</th><td>bq</td><td>Gabat     </td><td> 1392</td><td>  2557</td><td>  375.6893</td><td>  81.77015</td><td>2.182357</td><td>18.67494</td></tr>\n",
       "\t<tr><th scope=row>CG908344</th><td>bq</td><td>CG9083    </td><td> 5361</td><td> 11400</td><td> 1406.4989</td><td> 413.48149</td><td>1.762729</td><td>18.43624</td></tr>\n",
       "\t<tr><th scope=row>CG767544</th><td>bq</td><td>CG7675    </td><td> 2668</td><td>  5580</td><td>  717.7727</td><td> 186.19533</td><td>1.938983</td><td>18.39976</td></tr>\n",
       "\t<tr><th scope=row>Tm244</th><td>bq</td><td>Tm2       </td><td> 3845</td><td>  9615</td><td> 1041.4820</td><td> 298.07987</td><td>1.800035</td><td>18.04681</td></tr>\n",
       "\t<tr><th scope=row>Gdh44</th><td>bq</td><td>Gdh       </td><td> 4394</td><td>  9732</td><td> 1180.6039</td><td> 346.38616</td><td>1.764913</td><td>18.01364</td></tr>\n",
       "\t<tr><th scope=row>RpL3100</th><td>bq</td><td>RpL3      </td><td>19347</td><td> 58621</td><td> 5215.0858</td><td>1898.14933</td><td>1.457338</td><td>17.99630</td></tr>\n",
       "\t<tr><th scope=row>LamC44</th><td>bq</td><td>LamC      </td><td> 2474</td><td>  5218</td><td>  668.7581</td><td> 176.29974</td><td>1.915294</td><td>17.97981</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tAct57B44 & bq & Act57B     & 24201 &  49032 &  6408.3207 & 1439.01846 & 2.153859 & 27.23761\\\\\n",
       "\tCG296244 & bq & CG2962     & 14862 &  28955 &  3852.2991 & 1015.30233 & 1.922390 & 22.89928\\\\\n",
       "\tMlc244 & bq & Mlc2       & 12441 &  28458 &  3299.5489 &  866.05437 & 1.928074 & 22.53628\\\\\n",
       "\teEF1alpha144 & bq & eEF1alpha1 & 76454 & 218993 & 20543.3019 & 7206.71742 & 1.511054 & 21.64804\\\\\n",
       "\tCG1573144 & bq & CG15731    &  2958 &   5182 &   782.8430 &  168.63752 & 2.206268 & 21.21199\\\\\n",
       "\tRpL23A44 & bq & RpL23A     & 23945 &  63966 &  6321.6670 & 2065.07794 & 1.613410 & 20.37143\\\\\n",
       "\tTwdlF44 & bq & TwdlF      &  6697 &  13997 &  1776.8570 &  492.98094 & 1.846800 & 19.93791\\\\\n",
       "\tRpL644 & bq & RpL6       & 22539 &  62854 &  5951.1989 & 1986.58711 & 1.582162 & 19.83906\\\\\n",
       "\tRpS644 & bq & RpS6       & 18692 &  50047 &  4918.5066 & 1630.96020 & 1.591614 & 19.52003\\\\\n",
       "\tHsc70-444 & bq & Hsc70-4    & 19542 &  54591 &  5251.8047 & 1790.73474 & 1.551456 & 19.17425\\\\\n",
       "\tRpL7A44 & bq & RpL7A      & 21373 &  62912 &  5587.7042 & 1969.40485 & 1.503764 & 18.71930\\\\\n",
       "\teEF244 & bq & eEF2       & 27581 &  82036 &  7219.6870 & 2627.25166 & 1.457833 & 18.68639\\\\\n",
       "\tRpL454 & bq & RpL4       & 14388 &  40203 &  3855.1943 & 1299.19926 & 1.568071 & 18.68037\\\\\n",
       "\tGabat44 & bq & Gabat      &  1392 &   2557 &   375.6893 &   81.77015 & 2.182357 & 18.67494\\\\\n",
       "\tCG908344 & bq & CG9083     &  5361 &  11400 &  1406.4989 &  413.48149 & 1.762729 & 18.43624\\\\\n",
       "\tCG767544 & bq & CG7675     &  2668 &   5580 &   717.7727 &  186.19533 & 1.938983 & 18.39976\\\\\n",
       "\tTm244 & bq & Tm2        &  3845 &   9615 &  1041.4820 &  298.07987 & 1.800035 & 18.04681\\\\\n",
       "\tGdh44 & bq & Gdh        &  4394 &   9732 &  1180.6039 &  346.38616 & 1.764913 & 18.01364\\\\\n",
       "\tRpL3100 & bq & RpL3       & 19347 &  58621 &  5215.0858 & 1898.14933 & 1.457338 & 17.99630\\\\\n",
       "\tLamC44 & bq & LamC       &  2474 &   5218 &   668.7581 &  176.29974 & 1.915294 & 17.97981\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Act57B44 | bq | Act57B     | 24201 |  49032 |  6408.3207 | 1439.01846 | 2.153859 | 27.23761 |\n",
       "| CG296244 | bq | CG2962     | 14862 |  28955 |  3852.2991 | 1015.30233 | 1.922390 | 22.89928 |\n",
       "| Mlc244 | bq | Mlc2       | 12441 |  28458 |  3299.5489 |  866.05437 | 1.928074 | 22.53628 |\n",
       "| eEF1alpha144 | bq | eEF1alpha1 | 76454 | 218993 | 20543.3019 | 7206.71742 | 1.511054 | 21.64804 |\n",
       "| CG1573144 | bq | CG15731    |  2958 |   5182 |   782.8430 |  168.63752 | 2.206268 | 21.21199 |\n",
       "| RpL23A44 | bq | RpL23A     | 23945 |  63966 |  6321.6670 | 2065.07794 | 1.613410 | 20.37143 |\n",
       "| TwdlF44 | bq | TwdlF      |  6697 |  13997 |  1776.8570 |  492.98094 | 1.846800 | 19.93791 |\n",
       "| RpL644 | bq | RpL6       | 22539 |  62854 |  5951.1989 | 1986.58711 | 1.582162 | 19.83906 |\n",
       "| RpS644 | bq | RpS6       | 18692 |  50047 |  4918.5066 | 1630.96020 | 1.591614 | 19.52003 |\n",
       "| Hsc70-444 | bq | Hsc70-4    | 19542 |  54591 |  5251.8047 | 1790.73474 | 1.551456 | 19.17425 |\n",
       "| RpL7A44 | bq | RpL7A      | 21373 |  62912 |  5587.7042 | 1969.40485 | 1.503764 | 18.71930 |\n",
       "| eEF244 | bq | eEF2       | 27581 |  82036 |  7219.6870 | 2627.25166 | 1.457833 | 18.68639 |\n",
       "| RpL454 | bq | RpL4       | 14388 |  40203 |  3855.1943 | 1299.19926 | 1.568071 | 18.68037 |\n",
       "| Gabat44 | bq | Gabat      |  1392 |   2557 |   375.6893 |   81.77015 | 2.182357 | 18.67494 |\n",
       "| CG908344 | bq | CG9083     |  5361 |  11400 |  1406.4989 |  413.48149 | 1.762729 | 18.43624 |\n",
       "| CG767544 | bq | CG7675     |  2668 |   5580 |   717.7727 |  186.19533 | 1.938983 | 18.39976 |\n",
       "| Tm244 | bq | Tm2        |  3845 |   9615 |  1041.4820 |  298.07987 | 1.800035 | 18.04681 |\n",
       "| Gdh44 | bq | Gdh        |  4394 |   9732 |  1180.6039 |  346.38616 | 1.764913 | 18.01364 |\n",
       "| RpL3100 | bq | RpL3       | 19347 |  58621 |  5215.0858 | 1898.14933 | 1.457338 | 17.99630 |\n",
       "| LamC44 | bq | LamC       |  2474 |   5218 |   668.7581 |  176.29974 | 1.915294 | 17.97981 |\n",
       "\n"
      ],
      "text/plain": [
       "             cell.type.2 gene       cell.type.2.n.umi other.n.umi\n",
       "Act57B44     bq          Act57B     24201              49032     \n",
       "CG296244     bq          CG2962     14862              28955     \n",
       "Mlc244       bq          Mlc2       12441              28458     \n",
       "eEF1alpha144 bq          eEF1alpha1 76454             218993     \n",
       "CG1573144    bq          CG15731     2958               5182     \n",
       "RpL23A44     bq          RpL23A     23945              63966     \n",
       "TwdlF44      bq          TwdlF       6697              13997     \n",
       "RpL644       bq          RpL6       22539              62854     \n",
       "RpS644       bq          RpS6       18692              50047     \n",
       "Hsc70-444    bq          Hsc70-4    19542              54591     \n",
       "RpL7A44      bq          RpL7A      21373              62912     \n",
       "eEF244       bq          eEF2       27581              82036     \n",
       "RpL454       bq          RpL4       14388              40203     \n",
       "Gabat44      bq          Gabat       1392               2557     \n",
       "CG908344     bq          CG9083      5361              11400     \n",
       "CG767544     bq          CG7675      2668               5580     \n",
       "Tm244        bq          Tm2         3845               9615     \n",
       "Gdh44        bq          Gdh         4394               9732     \n",
       "RpL3100      bq          RpL3       19347              58621     \n",
       "LamC44       bq          LamC        2474               5218     \n",
       "             cell.type.2.tpm other.tpm  log2.ratio heuristic.score\n",
       "Act57B44      6408.3207      1439.01846 2.153859   27.23761       \n",
       "CG296244      3852.2991      1015.30233 1.922390   22.89928       \n",
       "Mlc244        3299.5489       866.05437 1.928074   22.53628       \n",
       "eEF1alpha144 20543.3019      7206.71742 1.511054   21.64804       \n",
       "CG1573144      782.8430       168.63752 2.206268   21.21199       \n",
       "RpL23A44      6321.6670      2065.07794 1.613410   20.37143       \n",
       "TwdlF44       1776.8570       492.98094 1.846800   19.93791       \n",
       "RpL644        5951.1989      1986.58711 1.582162   19.83906       \n",
       "RpS644        4918.5066      1630.96020 1.591614   19.52003       \n",
       "Hsc70-444     5251.8047      1790.73474 1.551456   19.17425       \n",
       "RpL7A44       5587.7042      1969.40485 1.503764   18.71930       \n",
       "eEF244        7219.6870      2627.25166 1.457833   18.68639       \n",
       "RpL454        3855.1943      1299.19926 1.568071   18.68037       \n",
       "Gabat44        375.6893        81.77015 2.182357   18.67494       \n",
       "CG908344      1406.4989       413.48149 1.762729   18.43624       \n",
       "CG767544       717.7727       186.19533 1.938983   18.39976       \n",
       "Tm244         1041.4820       298.07987 1.800035   18.04681       \n",
       "Gdh44         1180.6039       346.38616 1.764913   18.01364       \n",
       "RpL3100       5215.0858      1898.14933 1.457338   17.99630       \n",
       "LamC44         668.7581       176.29974 1.915294   17.97981       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \n",
    "                                              \"bq\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bv\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bv\", df_sub300$cluster %in% bv)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1649,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Act57B44</th><td>bq</td><td>Act57B    </td><td>24201</td><td> 49032</td><td> 6408.3207</td><td>1439.01846</td><td>2.153859</td><td>27.23761</td></tr>\n",
       "\t<tr><th scope=row>CG296244</th><td>bq</td><td>CG2962    </td><td>14862</td><td> 28955</td><td> 3852.2991</td><td>1015.30233</td><td>1.922390</td><td>22.89928</td></tr>\n",
       "\t<tr><th scope=row>Mlc244</th><td>bq</td><td>Mlc2      </td><td>12441</td><td> 28458</td><td> 3299.5489</td><td> 866.05437</td><td>1.928074</td><td>22.53628</td></tr>\n",
       "\t<tr><th scope=row>eEF1alpha144</th><td>bq</td><td>eEF1alpha1</td><td>76454</td><td>218993</td><td>20543.3019</td><td>7206.71742</td><td>1.511054</td><td>21.64804</td></tr>\n",
       "\t<tr><th scope=row>CG1573144</th><td>bq</td><td>CG15731   </td><td> 2958</td><td>  5182</td><td>  782.8430</td><td> 168.63752</td><td>2.206268</td><td>21.21199</td></tr>\n",
       "\t<tr><th scope=row>RpL23A44</th><td>bq</td><td>RpL23A    </td><td>23945</td><td> 63966</td><td> 6321.6670</td><td>2065.07794</td><td>1.613410</td><td>20.37143</td></tr>\n",
       "\t<tr><th scope=row>TwdlF44</th><td>bq</td><td>TwdlF     </td><td> 6697</td><td> 13997</td><td> 1776.8570</td><td> 492.98094</td><td>1.846800</td><td>19.93791</td></tr>\n",
       "\t<tr><th scope=row>RpL644</th><td>bq</td><td>RpL6      </td><td>22539</td><td> 62854</td><td> 5951.1989</td><td>1986.58711</td><td>1.582162</td><td>19.83906</td></tr>\n",
       "\t<tr><th scope=row>RpS644</th><td>bq</td><td>RpS6      </td><td>18692</td><td> 50047</td><td> 4918.5066</td><td>1630.96020</td><td>1.591614</td><td>19.52003</td></tr>\n",
       "\t<tr><th scope=row>Hsc70-444</th><td>bq</td><td>Hsc70-4   </td><td>19542</td><td> 54591</td><td> 5251.8047</td><td>1790.73474</td><td>1.551456</td><td>19.17425</td></tr>\n",
       "\t<tr><th scope=row>RpL7A44</th><td>bq</td><td>RpL7A     </td><td>21373</td><td> 62912</td><td> 5587.7042</td><td>1969.40485</td><td>1.503764</td><td>18.71930</td></tr>\n",
       "\t<tr><th scope=row>eEF244</th><td>bq</td><td>eEF2      </td><td>27581</td><td> 82036</td><td> 7219.6870</td><td>2627.25166</td><td>1.457833</td><td>18.68639</td></tr>\n",
       "\t<tr><th scope=row>RpL454</th><td>bq</td><td>RpL4      </td><td>14388</td><td> 40203</td><td> 3855.1943</td><td>1299.19926</td><td>1.568071</td><td>18.68037</td></tr>\n",
       "\t<tr><th scope=row>Gabat44</th><td>bq</td><td>Gabat     </td><td> 1392</td><td>  2557</td><td>  375.6893</td><td>  81.77015</td><td>2.182357</td><td>18.67494</td></tr>\n",
       "\t<tr><th scope=row>CG908344</th><td>bq</td><td>CG9083    </td><td> 5361</td><td> 11400</td><td> 1406.4989</td><td> 413.48149</td><td>1.762729</td><td>18.43624</td></tr>\n",
       "\t<tr><th scope=row>CG767544</th><td>bq</td><td>CG7675    </td><td> 2668</td><td>  5580</td><td>  717.7727</td><td> 186.19533</td><td>1.938983</td><td>18.39976</td></tr>\n",
       "\t<tr><th scope=row>Tm244</th><td>bq</td><td>Tm2       </td><td> 3845</td><td>  9615</td><td> 1041.4820</td><td> 298.07987</td><td>1.800035</td><td>18.04681</td></tr>\n",
       "\t<tr><th scope=row>Gdh44</th><td>bq</td><td>Gdh       </td><td> 4394</td><td>  9732</td><td> 1180.6039</td><td> 346.38616</td><td>1.764913</td><td>18.01364</td></tr>\n",
       "\t<tr><th scope=row>RpL3100</th><td>bq</td><td>RpL3      </td><td>19347</td><td> 58621</td><td> 5215.0858</td><td>1898.14933</td><td>1.457338</td><td>17.99630</td></tr>\n",
       "\t<tr><th scope=row>LamC44</th><td>bq</td><td>LamC      </td><td> 2474</td><td>  5218</td><td>  668.7581</td><td> 176.29974</td><td>1.915294</td><td>17.97981</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tAct57B44 & bq & Act57B     & 24201 &  49032 &  6408.3207 & 1439.01846 & 2.153859 & 27.23761\\\\\n",
       "\tCG296244 & bq & CG2962     & 14862 &  28955 &  3852.2991 & 1015.30233 & 1.922390 & 22.89928\\\\\n",
       "\tMlc244 & bq & Mlc2       & 12441 &  28458 &  3299.5489 &  866.05437 & 1.928074 & 22.53628\\\\\n",
       "\teEF1alpha144 & bq & eEF1alpha1 & 76454 & 218993 & 20543.3019 & 7206.71742 & 1.511054 & 21.64804\\\\\n",
       "\tCG1573144 & bq & CG15731    &  2958 &   5182 &   782.8430 &  168.63752 & 2.206268 & 21.21199\\\\\n",
       "\tRpL23A44 & bq & RpL23A     & 23945 &  63966 &  6321.6670 & 2065.07794 & 1.613410 & 20.37143\\\\\n",
       "\tTwdlF44 & bq & TwdlF      &  6697 &  13997 &  1776.8570 &  492.98094 & 1.846800 & 19.93791\\\\\n",
       "\tRpL644 & bq & RpL6       & 22539 &  62854 &  5951.1989 & 1986.58711 & 1.582162 & 19.83906\\\\\n",
       "\tRpS644 & bq & RpS6       & 18692 &  50047 &  4918.5066 & 1630.96020 & 1.591614 & 19.52003\\\\\n",
       "\tHsc70-444 & bq & Hsc70-4    & 19542 &  54591 &  5251.8047 & 1790.73474 & 1.551456 & 19.17425\\\\\n",
       "\tRpL7A44 & bq & RpL7A      & 21373 &  62912 &  5587.7042 & 1969.40485 & 1.503764 & 18.71930\\\\\n",
       "\teEF244 & bq & eEF2       & 27581 &  82036 &  7219.6870 & 2627.25166 & 1.457833 & 18.68639\\\\\n",
       "\tRpL454 & bq & RpL4       & 14388 &  40203 &  3855.1943 & 1299.19926 & 1.568071 & 18.68037\\\\\n",
       "\tGabat44 & bq & Gabat      &  1392 &   2557 &   375.6893 &   81.77015 & 2.182357 & 18.67494\\\\\n",
       "\tCG908344 & bq & CG9083     &  5361 &  11400 &  1406.4989 &  413.48149 & 1.762729 & 18.43624\\\\\n",
       "\tCG767544 & bq & CG7675     &  2668 &   5580 &   717.7727 &  186.19533 & 1.938983 & 18.39976\\\\\n",
       "\tTm244 & bq & Tm2        &  3845 &   9615 &  1041.4820 &  298.07987 & 1.800035 & 18.04681\\\\\n",
       "\tGdh44 & bq & Gdh        &  4394 &   9732 &  1180.6039 &  346.38616 & 1.764913 & 18.01364\\\\\n",
       "\tRpL3100 & bq & RpL3       & 19347 &  58621 &  5215.0858 & 1898.14933 & 1.457338 & 17.99630\\\\\n",
       "\tLamC44 & bq & LamC       &  2474 &   5218 &   668.7581 &  176.29974 & 1.915294 & 17.97981\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Act57B44 | bq | Act57B     | 24201 |  49032 |  6408.3207 | 1439.01846 | 2.153859 | 27.23761 |\n",
       "| CG296244 | bq | CG2962     | 14862 |  28955 |  3852.2991 | 1015.30233 | 1.922390 | 22.89928 |\n",
       "| Mlc244 | bq | Mlc2       | 12441 |  28458 |  3299.5489 |  866.05437 | 1.928074 | 22.53628 |\n",
       "| eEF1alpha144 | bq | eEF1alpha1 | 76454 | 218993 | 20543.3019 | 7206.71742 | 1.511054 | 21.64804 |\n",
       "| CG1573144 | bq | CG15731    |  2958 |   5182 |   782.8430 |  168.63752 | 2.206268 | 21.21199 |\n",
       "| RpL23A44 | bq | RpL23A     | 23945 |  63966 |  6321.6670 | 2065.07794 | 1.613410 | 20.37143 |\n",
       "| TwdlF44 | bq | TwdlF      |  6697 |  13997 |  1776.8570 |  492.98094 | 1.846800 | 19.93791 |\n",
       "| RpL644 | bq | RpL6       | 22539 |  62854 |  5951.1989 | 1986.58711 | 1.582162 | 19.83906 |\n",
       "| RpS644 | bq | RpS6       | 18692 |  50047 |  4918.5066 | 1630.96020 | 1.591614 | 19.52003 |\n",
       "| Hsc70-444 | bq | Hsc70-4    | 19542 |  54591 |  5251.8047 | 1790.73474 | 1.551456 | 19.17425 |\n",
       "| RpL7A44 | bq | RpL7A      | 21373 |  62912 |  5587.7042 | 1969.40485 | 1.503764 | 18.71930 |\n",
       "| eEF244 | bq | eEF2       | 27581 |  82036 |  7219.6870 | 2627.25166 | 1.457833 | 18.68639 |\n",
       "| RpL454 | bq | RpL4       | 14388 |  40203 |  3855.1943 | 1299.19926 | 1.568071 | 18.68037 |\n",
       "| Gabat44 | bq | Gabat      |  1392 |   2557 |   375.6893 |   81.77015 | 2.182357 | 18.67494 |\n",
       "| CG908344 | bq | CG9083     |  5361 |  11400 |  1406.4989 |  413.48149 | 1.762729 | 18.43624 |\n",
       "| CG767544 | bq | CG7675     |  2668 |   5580 |   717.7727 |  186.19533 | 1.938983 | 18.39976 |\n",
       "| Tm244 | bq | Tm2        |  3845 |   9615 |  1041.4820 |  298.07987 | 1.800035 | 18.04681 |\n",
       "| Gdh44 | bq | Gdh        |  4394 |   9732 |  1180.6039 |  346.38616 | 1.764913 | 18.01364 |\n",
       "| RpL3100 | bq | RpL3       | 19347 |  58621 |  5215.0858 | 1898.14933 | 1.457338 | 17.99630 |\n",
       "| LamC44 | bq | LamC       |  2474 |   5218 |   668.7581 |  176.29974 | 1.915294 | 17.97981 |\n",
       "\n"
      ],
      "text/plain": [
       "             cell.type.2 gene       cell.type.2.n.umi other.n.umi\n",
       "Act57B44     bq          Act57B     24201              49032     \n",
       "CG296244     bq          CG2962     14862              28955     \n",
       "Mlc244       bq          Mlc2       12441              28458     \n",
       "eEF1alpha144 bq          eEF1alpha1 76454             218993     \n",
       "CG1573144    bq          CG15731     2958               5182     \n",
       "RpL23A44     bq          RpL23A     23945              63966     \n",
       "TwdlF44      bq          TwdlF       6697              13997     \n",
       "RpL644       bq          RpL6       22539              62854     \n",
       "RpS644       bq          RpS6       18692              50047     \n",
       "Hsc70-444    bq          Hsc70-4    19542              54591     \n",
       "RpL7A44      bq          RpL7A      21373              62912     \n",
       "eEF244       bq          eEF2       27581              82036     \n",
       "RpL454       bq          RpL4       14388              40203     \n",
       "Gabat44      bq          Gabat       1392               2557     \n",
       "CG908344     bq          CG9083      5361              11400     \n",
       "CG767544     bq          CG7675      2668               5580     \n",
       "Tm244        bq          Tm2         3845               9615     \n",
       "Gdh44        bq          Gdh         4394               9732     \n",
       "RpL3100      bq          RpL3       19347              58621     \n",
       "LamC44       bq          LamC        2474               5218     \n",
       "             cell.type.2.tpm other.tpm  log2.ratio heuristic.score\n",
       "Act57B44      6408.3207      1439.01846 2.153859   27.23761       \n",
       "CG296244      3852.2991      1015.30233 1.922390   22.89928       \n",
       "Mlc244        3299.5489       866.05437 1.928074   22.53628       \n",
       "eEF1alpha144 20543.3019      7206.71742 1.511054   21.64804       \n",
       "CG1573144      782.8430       168.63752 2.206268   21.21199       \n",
       "RpL23A44      6321.6670      2065.07794 1.613410   20.37143       \n",
       "TwdlF44       1776.8570       492.98094 1.846800   19.93791       \n",
       "RpL644        5951.1989      1986.58711 1.582162   19.83906       \n",
       "RpS644        4918.5066      1630.96020 1.591614   19.52003       \n",
       "Hsc70-444     5251.8047      1790.73474 1.551456   19.17425       \n",
       "RpL7A44       5587.7042      1969.40485 1.503764   18.71930       \n",
       "eEF244        7219.6870      2627.25166 1.457833   18.68639       \n",
       "RpL454        3855.1943      1299.19926 1.568071   18.68037       \n",
       "Gabat44        375.6893        81.77015 2.182357   18.67494       \n",
       "CG908344      1406.4989       413.48149 1.762729   18.43624       \n",
       "CG767544       717.7727       186.19533 1.938983   18.39976       \n",
       "Tm244         1041.4820       298.07987 1.800035   18.04681       \n",
       "Gdh44         1180.6039       346.38616 1.764913   18.01364       \n",
       "RpL3100       5215.0858      1898.14933 1.457338   17.99630       \n",
       "LamC44         668.7581       176.29974 1.915294   17.97981       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \n",
    "                                              \"bq\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bw\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bw\", df_sub300$cluster %in% bw)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1650,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Act57B44</th><td>bq</td><td>Act57B    </td><td>24201</td><td> 49032</td><td> 6408.3207</td><td>1439.01846</td><td>2.153859</td><td>27.23761</td></tr>\n",
       "\t<tr><th scope=row>CG296244</th><td>bq</td><td>CG2962    </td><td>14862</td><td> 28955</td><td> 3852.2991</td><td>1015.30233</td><td>1.922390</td><td>22.89928</td></tr>\n",
       "\t<tr><th scope=row>Mlc244</th><td>bq</td><td>Mlc2      </td><td>12441</td><td> 28458</td><td> 3299.5489</td><td> 866.05437</td><td>1.928074</td><td>22.53628</td></tr>\n",
       "\t<tr><th scope=row>eEF1alpha144</th><td>bq</td><td>eEF1alpha1</td><td>76454</td><td>218993</td><td>20543.3019</td><td>7206.71742</td><td>1.511054</td><td>21.64804</td></tr>\n",
       "\t<tr><th scope=row>CG1573144</th><td>bq</td><td>CG15731   </td><td> 2958</td><td>  5182</td><td>  782.8430</td><td> 168.63752</td><td>2.206268</td><td>21.21199</td></tr>\n",
       "\t<tr><th scope=row>RpL23A44</th><td>bq</td><td>RpL23A    </td><td>23945</td><td> 63966</td><td> 6321.6670</td><td>2065.07794</td><td>1.613410</td><td>20.37143</td></tr>\n",
       "\t<tr><th scope=row>TwdlF44</th><td>bq</td><td>TwdlF     </td><td> 6697</td><td> 13997</td><td> 1776.8570</td><td> 492.98094</td><td>1.846800</td><td>19.93791</td></tr>\n",
       "\t<tr><th scope=row>RpL644</th><td>bq</td><td>RpL6      </td><td>22539</td><td> 62854</td><td> 5951.1989</td><td>1986.58711</td><td>1.582162</td><td>19.83906</td></tr>\n",
       "\t<tr><th scope=row>RpS644</th><td>bq</td><td>RpS6      </td><td>18692</td><td> 50047</td><td> 4918.5066</td><td>1630.96020</td><td>1.591614</td><td>19.52003</td></tr>\n",
       "\t<tr><th scope=row>Hsc70-444</th><td>bq</td><td>Hsc70-4   </td><td>19542</td><td> 54591</td><td> 5251.8047</td><td>1790.73474</td><td>1.551456</td><td>19.17425</td></tr>\n",
       "\t<tr><th scope=row>RpL7A44</th><td>bq</td><td>RpL7A     </td><td>21373</td><td> 62912</td><td> 5587.7042</td><td>1969.40485</td><td>1.503764</td><td>18.71930</td></tr>\n",
       "\t<tr><th scope=row>eEF244</th><td>bq</td><td>eEF2      </td><td>27581</td><td> 82036</td><td> 7219.6870</td><td>2627.25166</td><td>1.457833</td><td>18.68639</td></tr>\n",
       "\t<tr><th scope=row>RpL454</th><td>bq</td><td>RpL4      </td><td>14388</td><td> 40203</td><td> 3855.1943</td><td>1299.19926</td><td>1.568071</td><td>18.68037</td></tr>\n",
       "\t<tr><th scope=row>Gabat44</th><td>bq</td><td>Gabat     </td><td> 1392</td><td>  2557</td><td>  375.6893</td><td>  81.77015</td><td>2.182357</td><td>18.67494</td></tr>\n",
       "\t<tr><th scope=row>CG908344</th><td>bq</td><td>CG9083    </td><td> 5361</td><td> 11400</td><td> 1406.4989</td><td> 413.48149</td><td>1.762729</td><td>18.43624</td></tr>\n",
       "\t<tr><th scope=row>CG767544</th><td>bq</td><td>CG7675    </td><td> 2668</td><td>  5580</td><td>  717.7727</td><td> 186.19533</td><td>1.938983</td><td>18.39976</td></tr>\n",
       "\t<tr><th scope=row>Tm244</th><td>bq</td><td>Tm2       </td><td> 3845</td><td>  9615</td><td> 1041.4820</td><td> 298.07987</td><td>1.800035</td><td>18.04681</td></tr>\n",
       "\t<tr><th scope=row>Gdh44</th><td>bq</td><td>Gdh       </td><td> 4394</td><td>  9732</td><td> 1180.6039</td><td> 346.38616</td><td>1.764913</td><td>18.01364</td></tr>\n",
       "\t<tr><th scope=row>RpL3100</th><td>bq</td><td>RpL3      </td><td>19347</td><td> 58621</td><td> 5215.0858</td><td>1898.14933</td><td>1.457338</td><td>17.99630</td></tr>\n",
       "\t<tr><th scope=row>LamC44</th><td>bq</td><td>LamC      </td><td> 2474</td><td>  5218</td><td>  668.7581</td><td> 176.29974</td><td>1.915294</td><td>17.97981</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tAct57B44 & bq & Act57B     & 24201 &  49032 &  6408.3207 & 1439.01846 & 2.153859 & 27.23761\\\\\n",
       "\tCG296244 & bq & CG2962     & 14862 &  28955 &  3852.2991 & 1015.30233 & 1.922390 & 22.89928\\\\\n",
       "\tMlc244 & bq & Mlc2       & 12441 &  28458 &  3299.5489 &  866.05437 & 1.928074 & 22.53628\\\\\n",
       "\teEF1alpha144 & bq & eEF1alpha1 & 76454 & 218993 & 20543.3019 & 7206.71742 & 1.511054 & 21.64804\\\\\n",
       "\tCG1573144 & bq & CG15731    &  2958 &   5182 &   782.8430 &  168.63752 & 2.206268 & 21.21199\\\\\n",
       "\tRpL23A44 & bq & RpL23A     & 23945 &  63966 &  6321.6670 & 2065.07794 & 1.613410 & 20.37143\\\\\n",
       "\tTwdlF44 & bq & TwdlF      &  6697 &  13997 &  1776.8570 &  492.98094 & 1.846800 & 19.93791\\\\\n",
       "\tRpL644 & bq & RpL6       & 22539 &  62854 &  5951.1989 & 1986.58711 & 1.582162 & 19.83906\\\\\n",
       "\tRpS644 & bq & RpS6       & 18692 &  50047 &  4918.5066 & 1630.96020 & 1.591614 & 19.52003\\\\\n",
       "\tHsc70-444 & bq & Hsc70-4    & 19542 &  54591 &  5251.8047 & 1790.73474 & 1.551456 & 19.17425\\\\\n",
       "\tRpL7A44 & bq & RpL7A      & 21373 &  62912 &  5587.7042 & 1969.40485 & 1.503764 & 18.71930\\\\\n",
       "\teEF244 & bq & eEF2       & 27581 &  82036 &  7219.6870 & 2627.25166 & 1.457833 & 18.68639\\\\\n",
       "\tRpL454 & bq & RpL4       & 14388 &  40203 &  3855.1943 & 1299.19926 & 1.568071 & 18.68037\\\\\n",
       "\tGabat44 & bq & Gabat      &  1392 &   2557 &   375.6893 &   81.77015 & 2.182357 & 18.67494\\\\\n",
       "\tCG908344 & bq & CG9083     &  5361 &  11400 &  1406.4989 &  413.48149 & 1.762729 & 18.43624\\\\\n",
       "\tCG767544 & bq & CG7675     &  2668 &   5580 &   717.7727 &  186.19533 & 1.938983 & 18.39976\\\\\n",
       "\tTm244 & bq & Tm2        &  3845 &   9615 &  1041.4820 &  298.07987 & 1.800035 & 18.04681\\\\\n",
       "\tGdh44 & bq & Gdh        &  4394 &   9732 &  1180.6039 &  346.38616 & 1.764913 & 18.01364\\\\\n",
       "\tRpL3100 & bq & RpL3       & 19347 &  58621 &  5215.0858 & 1898.14933 & 1.457338 & 17.99630\\\\\n",
       "\tLamC44 & bq & LamC       &  2474 &   5218 &   668.7581 &  176.29974 & 1.915294 & 17.97981\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Act57B44 | bq | Act57B     | 24201 |  49032 |  6408.3207 | 1439.01846 | 2.153859 | 27.23761 |\n",
       "| CG296244 | bq | CG2962     | 14862 |  28955 |  3852.2991 | 1015.30233 | 1.922390 | 22.89928 |\n",
       "| Mlc244 | bq | Mlc2       | 12441 |  28458 |  3299.5489 |  866.05437 | 1.928074 | 22.53628 |\n",
       "| eEF1alpha144 | bq | eEF1alpha1 | 76454 | 218993 | 20543.3019 | 7206.71742 | 1.511054 | 21.64804 |\n",
       "| CG1573144 | bq | CG15731    |  2958 |   5182 |   782.8430 |  168.63752 | 2.206268 | 21.21199 |\n",
       "| RpL23A44 | bq | RpL23A     | 23945 |  63966 |  6321.6670 | 2065.07794 | 1.613410 | 20.37143 |\n",
       "| TwdlF44 | bq | TwdlF      |  6697 |  13997 |  1776.8570 |  492.98094 | 1.846800 | 19.93791 |\n",
       "| RpL644 | bq | RpL6       | 22539 |  62854 |  5951.1989 | 1986.58711 | 1.582162 | 19.83906 |\n",
       "| RpS644 | bq | RpS6       | 18692 |  50047 |  4918.5066 | 1630.96020 | 1.591614 | 19.52003 |\n",
       "| Hsc70-444 | bq | Hsc70-4    | 19542 |  54591 |  5251.8047 | 1790.73474 | 1.551456 | 19.17425 |\n",
       "| RpL7A44 | bq | RpL7A      | 21373 |  62912 |  5587.7042 | 1969.40485 | 1.503764 | 18.71930 |\n",
       "| eEF244 | bq | eEF2       | 27581 |  82036 |  7219.6870 | 2627.25166 | 1.457833 | 18.68639 |\n",
       "| RpL454 | bq | RpL4       | 14388 |  40203 |  3855.1943 | 1299.19926 | 1.568071 | 18.68037 |\n",
       "| Gabat44 | bq | Gabat      |  1392 |   2557 |   375.6893 |   81.77015 | 2.182357 | 18.67494 |\n",
       "| CG908344 | bq | CG9083     |  5361 |  11400 |  1406.4989 |  413.48149 | 1.762729 | 18.43624 |\n",
       "| CG767544 | bq | CG7675     |  2668 |   5580 |   717.7727 |  186.19533 | 1.938983 | 18.39976 |\n",
       "| Tm244 | bq | Tm2        |  3845 |   9615 |  1041.4820 |  298.07987 | 1.800035 | 18.04681 |\n",
       "| Gdh44 | bq | Gdh        |  4394 |   9732 |  1180.6039 |  346.38616 | 1.764913 | 18.01364 |\n",
       "| RpL3100 | bq | RpL3       | 19347 |  58621 |  5215.0858 | 1898.14933 | 1.457338 | 17.99630 |\n",
       "| LamC44 | bq | LamC       |  2474 |   5218 |   668.7581 |  176.29974 | 1.915294 | 17.97981 |\n",
       "\n"
      ],
      "text/plain": [
       "             cell.type.2 gene       cell.type.2.n.umi other.n.umi\n",
       "Act57B44     bq          Act57B     24201              49032     \n",
       "CG296244     bq          CG2962     14862              28955     \n",
       "Mlc244       bq          Mlc2       12441              28458     \n",
       "eEF1alpha144 bq          eEF1alpha1 76454             218993     \n",
       "CG1573144    bq          CG15731     2958               5182     \n",
       "RpL23A44     bq          RpL23A     23945              63966     \n",
       "TwdlF44      bq          TwdlF       6697              13997     \n",
       "RpL644       bq          RpL6       22539              62854     \n",
       "RpS644       bq          RpS6       18692              50047     \n",
       "Hsc70-444    bq          Hsc70-4    19542              54591     \n",
       "RpL7A44      bq          RpL7A      21373              62912     \n",
       "eEF244       bq          eEF2       27581              82036     \n",
       "RpL454       bq          RpL4       14388              40203     \n",
       "Gabat44      bq          Gabat       1392               2557     \n",
       "CG908344     bq          CG9083      5361              11400     \n",
       "CG767544     bq          CG7675      2668               5580     \n",
       "Tm244        bq          Tm2         3845               9615     \n",
       "Gdh44        bq          Gdh         4394               9732     \n",
       "RpL3100      bq          RpL3       19347              58621     \n",
       "LamC44       bq          LamC        2474               5218     \n",
       "             cell.type.2.tpm other.tpm  log2.ratio heuristic.score\n",
       "Act57B44      6408.3207      1439.01846 2.153859   27.23761       \n",
       "CG296244      3852.2991      1015.30233 1.922390   22.89928       \n",
       "Mlc244        3299.5489       866.05437 1.928074   22.53628       \n",
       "eEF1alpha144 20543.3019      7206.71742 1.511054   21.64804       \n",
       "CG1573144      782.8430       168.63752 2.206268   21.21199       \n",
       "RpL23A44      6321.6670      2065.07794 1.613410   20.37143       \n",
       "TwdlF44       1776.8570       492.98094 1.846800   19.93791       \n",
       "RpL644        5951.1989      1986.58711 1.582162   19.83906       \n",
       "RpS644        4918.5066      1630.96020 1.591614   19.52003       \n",
       "Hsc70-444     5251.8047      1790.73474 1.551456   19.17425       \n",
       "RpL7A44       5587.7042      1969.40485 1.503764   18.71930       \n",
       "eEF244        7219.6870      2627.25166 1.457833   18.68639       \n",
       "RpL454        3855.1943      1299.19926 1.568071   18.68037       \n",
       "Gabat44        375.6893        81.77015 2.182357   18.67494       \n",
       "CG908344      1406.4989       413.48149 1.762729   18.43624       \n",
       "CG767544       717.7727       186.19533 1.938983   18.39976       \n",
       "Tm244         1041.4820       298.07987 1.800035   18.04681       \n",
       "Gdh44         1180.6039       346.38616 1.764913   18.01364       \n",
       "RpL3100       5215.0858      1898.14933 1.457338   17.99630       \n",
       "LamC44         668.7581       176.29974 1.915294   17.97981       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \n",
    "                                              \"bq\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type bx\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"bx\", df_sub300$cluster %in% bx)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1651,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Act57B44</th><td>bq</td><td>Act57B    </td><td>24201</td><td> 49032</td><td> 6408.3207</td><td>1439.01846</td><td>2.153859</td><td>27.23761</td></tr>\n",
       "\t<tr><th scope=row>CG296244</th><td>bq</td><td>CG2962    </td><td>14862</td><td> 28955</td><td> 3852.2991</td><td>1015.30233</td><td>1.922390</td><td>22.89928</td></tr>\n",
       "\t<tr><th scope=row>Mlc244</th><td>bq</td><td>Mlc2      </td><td>12441</td><td> 28458</td><td> 3299.5489</td><td> 866.05437</td><td>1.928074</td><td>22.53628</td></tr>\n",
       "\t<tr><th scope=row>eEF1alpha144</th><td>bq</td><td>eEF1alpha1</td><td>76454</td><td>218993</td><td>20543.3019</td><td>7206.71742</td><td>1.511054</td><td>21.64804</td></tr>\n",
       "\t<tr><th scope=row>CG1573144</th><td>bq</td><td>CG15731   </td><td> 2958</td><td>  5182</td><td>  782.8430</td><td> 168.63752</td><td>2.206268</td><td>21.21199</td></tr>\n",
       "\t<tr><th scope=row>RpL23A44</th><td>bq</td><td>RpL23A    </td><td>23945</td><td> 63966</td><td> 6321.6670</td><td>2065.07794</td><td>1.613410</td><td>20.37143</td></tr>\n",
       "\t<tr><th scope=row>TwdlF44</th><td>bq</td><td>TwdlF     </td><td> 6697</td><td> 13997</td><td> 1776.8570</td><td> 492.98094</td><td>1.846800</td><td>19.93791</td></tr>\n",
       "\t<tr><th scope=row>RpL644</th><td>bq</td><td>RpL6      </td><td>22539</td><td> 62854</td><td> 5951.1989</td><td>1986.58711</td><td>1.582162</td><td>19.83906</td></tr>\n",
       "\t<tr><th scope=row>RpS644</th><td>bq</td><td>RpS6      </td><td>18692</td><td> 50047</td><td> 4918.5066</td><td>1630.96020</td><td>1.591614</td><td>19.52003</td></tr>\n",
       "\t<tr><th scope=row>Hsc70-444</th><td>bq</td><td>Hsc70-4   </td><td>19542</td><td> 54591</td><td> 5251.8047</td><td>1790.73474</td><td>1.551456</td><td>19.17425</td></tr>\n",
       "\t<tr><th scope=row>RpL7A44</th><td>bq</td><td>RpL7A     </td><td>21373</td><td> 62912</td><td> 5587.7042</td><td>1969.40485</td><td>1.503764</td><td>18.71930</td></tr>\n",
       "\t<tr><th scope=row>eEF244</th><td>bq</td><td>eEF2      </td><td>27581</td><td> 82036</td><td> 7219.6870</td><td>2627.25166</td><td>1.457833</td><td>18.68639</td></tr>\n",
       "\t<tr><th scope=row>RpL454</th><td>bq</td><td>RpL4      </td><td>14388</td><td> 40203</td><td> 3855.1943</td><td>1299.19926</td><td>1.568071</td><td>18.68037</td></tr>\n",
       "\t<tr><th scope=row>Gabat44</th><td>bq</td><td>Gabat     </td><td> 1392</td><td>  2557</td><td>  375.6893</td><td>  81.77015</td><td>2.182357</td><td>18.67494</td></tr>\n",
       "\t<tr><th scope=row>CG908344</th><td>bq</td><td>CG9083    </td><td> 5361</td><td> 11400</td><td> 1406.4989</td><td> 413.48149</td><td>1.762729</td><td>18.43624</td></tr>\n",
       "\t<tr><th scope=row>CG767544</th><td>bq</td><td>CG7675    </td><td> 2668</td><td>  5580</td><td>  717.7727</td><td> 186.19533</td><td>1.938983</td><td>18.39976</td></tr>\n",
       "\t<tr><th scope=row>Tm244</th><td>bq</td><td>Tm2       </td><td> 3845</td><td>  9615</td><td> 1041.4820</td><td> 298.07987</td><td>1.800035</td><td>18.04681</td></tr>\n",
       "\t<tr><th scope=row>Gdh44</th><td>bq</td><td>Gdh       </td><td> 4394</td><td>  9732</td><td> 1180.6039</td><td> 346.38616</td><td>1.764913</td><td>18.01364</td></tr>\n",
       "\t<tr><th scope=row>RpL3100</th><td>bq</td><td>RpL3      </td><td>19347</td><td> 58621</td><td> 5215.0858</td><td>1898.14933</td><td>1.457338</td><td>17.99630</td></tr>\n",
       "\t<tr><th scope=row>LamC44</th><td>bq</td><td>LamC      </td><td> 2474</td><td>  5218</td><td>  668.7581</td><td> 176.29974</td><td>1.915294</td><td>17.97981</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tAct57B44 & bq & Act57B     & 24201 &  49032 &  6408.3207 & 1439.01846 & 2.153859 & 27.23761\\\\\n",
       "\tCG296244 & bq & CG2962     & 14862 &  28955 &  3852.2991 & 1015.30233 & 1.922390 & 22.89928\\\\\n",
       "\tMlc244 & bq & Mlc2       & 12441 &  28458 &  3299.5489 &  866.05437 & 1.928074 & 22.53628\\\\\n",
       "\teEF1alpha144 & bq & eEF1alpha1 & 76454 & 218993 & 20543.3019 & 7206.71742 & 1.511054 & 21.64804\\\\\n",
       "\tCG1573144 & bq & CG15731    &  2958 &   5182 &   782.8430 &  168.63752 & 2.206268 & 21.21199\\\\\n",
       "\tRpL23A44 & bq & RpL23A     & 23945 &  63966 &  6321.6670 & 2065.07794 & 1.613410 & 20.37143\\\\\n",
       "\tTwdlF44 & bq & TwdlF      &  6697 &  13997 &  1776.8570 &  492.98094 & 1.846800 & 19.93791\\\\\n",
       "\tRpL644 & bq & RpL6       & 22539 &  62854 &  5951.1989 & 1986.58711 & 1.582162 & 19.83906\\\\\n",
       "\tRpS644 & bq & RpS6       & 18692 &  50047 &  4918.5066 & 1630.96020 & 1.591614 & 19.52003\\\\\n",
       "\tHsc70-444 & bq & Hsc70-4    & 19542 &  54591 &  5251.8047 & 1790.73474 & 1.551456 & 19.17425\\\\\n",
       "\tRpL7A44 & bq & RpL7A      & 21373 &  62912 &  5587.7042 & 1969.40485 & 1.503764 & 18.71930\\\\\n",
       "\teEF244 & bq & eEF2       & 27581 &  82036 &  7219.6870 & 2627.25166 & 1.457833 & 18.68639\\\\\n",
       "\tRpL454 & bq & RpL4       & 14388 &  40203 &  3855.1943 & 1299.19926 & 1.568071 & 18.68037\\\\\n",
       "\tGabat44 & bq & Gabat      &  1392 &   2557 &   375.6893 &   81.77015 & 2.182357 & 18.67494\\\\\n",
       "\tCG908344 & bq & CG9083     &  5361 &  11400 &  1406.4989 &  413.48149 & 1.762729 & 18.43624\\\\\n",
       "\tCG767544 & bq & CG7675     &  2668 &   5580 &   717.7727 &  186.19533 & 1.938983 & 18.39976\\\\\n",
       "\tTm244 & bq & Tm2        &  3845 &   9615 &  1041.4820 &  298.07987 & 1.800035 & 18.04681\\\\\n",
       "\tGdh44 & bq & Gdh        &  4394 &   9732 &  1180.6039 &  346.38616 & 1.764913 & 18.01364\\\\\n",
       "\tRpL3100 & bq & RpL3       & 19347 &  58621 &  5215.0858 & 1898.14933 & 1.457338 & 17.99630\\\\\n",
       "\tLamC44 & bq & LamC       &  2474 &   5218 &   668.7581 &  176.29974 & 1.915294 & 17.97981\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Act57B44 | bq | Act57B     | 24201 |  49032 |  6408.3207 | 1439.01846 | 2.153859 | 27.23761 |\n",
       "| CG296244 | bq | CG2962     | 14862 |  28955 |  3852.2991 | 1015.30233 | 1.922390 | 22.89928 |\n",
       "| Mlc244 | bq | Mlc2       | 12441 |  28458 |  3299.5489 |  866.05437 | 1.928074 | 22.53628 |\n",
       "| eEF1alpha144 | bq | eEF1alpha1 | 76454 | 218993 | 20543.3019 | 7206.71742 | 1.511054 | 21.64804 |\n",
       "| CG1573144 | bq | CG15731    |  2958 |   5182 |   782.8430 |  168.63752 | 2.206268 | 21.21199 |\n",
       "| RpL23A44 | bq | RpL23A     | 23945 |  63966 |  6321.6670 | 2065.07794 | 1.613410 | 20.37143 |\n",
       "| TwdlF44 | bq | TwdlF      |  6697 |  13997 |  1776.8570 |  492.98094 | 1.846800 | 19.93791 |\n",
       "| RpL644 | bq | RpL6       | 22539 |  62854 |  5951.1989 | 1986.58711 | 1.582162 | 19.83906 |\n",
       "| RpS644 | bq | RpS6       | 18692 |  50047 |  4918.5066 | 1630.96020 | 1.591614 | 19.52003 |\n",
       "| Hsc70-444 | bq | Hsc70-4    | 19542 |  54591 |  5251.8047 | 1790.73474 | 1.551456 | 19.17425 |\n",
       "| RpL7A44 | bq | RpL7A      | 21373 |  62912 |  5587.7042 | 1969.40485 | 1.503764 | 18.71930 |\n",
       "| eEF244 | bq | eEF2       | 27581 |  82036 |  7219.6870 | 2627.25166 | 1.457833 | 18.68639 |\n",
       "| RpL454 | bq | RpL4       | 14388 |  40203 |  3855.1943 | 1299.19926 | 1.568071 | 18.68037 |\n",
       "| Gabat44 | bq | Gabat      |  1392 |   2557 |   375.6893 |   81.77015 | 2.182357 | 18.67494 |\n",
       "| CG908344 | bq | CG9083     |  5361 |  11400 |  1406.4989 |  413.48149 | 1.762729 | 18.43624 |\n",
       "| CG767544 | bq | CG7675     |  2668 |   5580 |   717.7727 |  186.19533 | 1.938983 | 18.39976 |\n",
       "| Tm244 | bq | Tm2        |  3845 |   9615 |  1041.4820 |  298.07987 | 1.800035 | 18.04681 |\n",
       "| Gdh44 | bq | Gdh        |  4394 |   9732 |  1180.6039 |  346.38616 | 1.764913 | 18.01364 |\n",
       "| RpL3100 | bq | RpL3       | 19347 |  58621 |  5215.0858 | 1898.14933 | 1.457338 | 17.99630 |\n",
       "| LamC44 | bq | LamC       |  2474 |   5218 |   668.7581 |  176.29974 | 1.915294 | 17.97981 |\n",
       "\n"
      ],
      "text/plain": [
       "             cell.type.2 gene       cell.type.2.n.umi other.n.umi\n",
       "Act57B44     bq          Act57B     24201              49032     \n",
       "CG296244     bq          CG2962     14862              28955     \n",
       "Mlc244       bq          Mlc2       12441              28458     \n",
       "eEF1alpha144 bq          eEF1alpha1 76454             218993     \n",
       "CG1573144    bq          CG15731     2958               5182     \n",
       "RpL23A44     bq          RpL23A     23945              63966     \n",
       "TwdlF44      bq          TwdlF       6697              13997     \n",
       "RpL644       bq          RpL6       22539              62854     \n",
       "RpS644       bq          RpS6       18692              50047     \n",
       "Hsc70-444    bq          Hsc70-4    19542              54591     \n",
       "RpL7A44      bq          RpL7A      21373              62912     \n",
       "eEF244       bq          eEF2       27581              82036     \n",
       "RpL454       bq          RpL4       14388              40203     \n",
       "Gabat44      bq          Gabat       1392               2557     \n",
       "CG908344     bq          CG9083      5361              11400     \n",
       "CG767544     bq          CG7675      2668               5580     \n",
       "Tm244        bq          Tm2         3845               9615     \n",
       "Gdh44        bq          Gdh         4394               9732     \n",
       "RpL3100      bq          RpL3       19347              58621     \n",
       "LamC44       bq          LamC        2474               5218     \n",
       "             cell.type.2.tpm other.tpm  log2.ratio heuristic.score\n",
       "Act57B44      6408.3207      1439.01846 2.153859   27.23761       \n",
       "CG296244      3852.2991      1015.30233 1.922390   22.89928       \n",
       "Mlc244        3299.5489       866.05437 1.928074   22.53628       \n",
       "eEF1alpha144 20543.3019      7206.71742 1.511054   21.64804       \n",
       "CG1573144      782.8430       168.63752 2.206268   21.21199       \n",
       "RpL23A44      6321.6670      2065.07794 1.613410   20.37143       \n",
       "TwdlF44       1776.8570       492.98094 1.846800   19.93791       \n",
       "RpL644        5951.1989      1986.58711 1.582162   19.83906       \n",
       "RpS644        4918.5066      1630.96020 1.591614   19.52003       \n",
       "Hsc70-444     5251.8047      1790.73474 1.551456   19.17425       \n",
       "RpL7A44       5587.7042      1969.40485 1.503764   18.71930       \n",
       "eEF244        7219.6870      2627.25166 1.457833   18.68639       \n",
       "RpL454        3855.1943      1299.19926 1.568071   18.68037       \n",
       "Gabat44        375.6893        81.77015 2.182357   18.67494       \n",
       "CG908344      1406.4989       413.48149 1.762729   18.43624       \n",
       "CG767544       717.7727       186.19533 1.938983   18.39976       \n",
       "Tm244         1041.4820       298.07987 1.800035   18.04681       \n",
       "Gdh44         1180.6039       346.38616 1.764913   18.01364       \n",
       "RpL3100       5215.0858      1898.14933 1.457338   17.99630       \n",
       "LamC44         668.7581       176.29974 1.915294   17.97981       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \n",
    "                                              \"bq\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type by\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"by\", df_sub300$cluster %in% by)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1652,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Act57B44</th><td>bq</td><td>Act57B    </td><td>24201</td><td> 49032</td><td> 6408.3207</td><td>1439.01846</td><td>2.153859</td><td>27.23761</td></tr>\n",
       "\t<tr><th scope=row>CG296244</th><td>bq</td><td>CG2962    </td><td>14862</td><td> 28955</td><td> 3852.2991</td><td>1015.30233</td><td>1.922390</td><td>22.89928</td></tr>\n",
       "\t<tr><th scope=row>Mlc244</th><td>bq</td><td>Mlc2      </td><td>12441</td><td> 28458</td><td> 3299.5489</td><td> 866.05437</td><td>1.928074</td><td>22.53628</td></tr>\n",
       "\t<tr><th scope=row>eEF1alpha144</th><td>bq</td><td>eEF1alpha1</td><td>76454</td><td>218993</td><td>20543.3019</td><td>7206.71742</td><td>1.511054</td><td>21.64804</td></tr>\n",
       "\t<tr><th scope=row>CG1573144</th><td>bq</td><td>CG15731   </td><td> 2958</td><td>  5182</td><td>  782.8430</td><td> 168.63752</td><td>2.206268</td><td>21.21199</td></tr>\n",
       "\t<tr><th scope=row>RpL23A44</th><td>bq</td><td>RpL23A    </td><td>23945</td><td> 63966</td><td> 6321.6670</td><td>2065.07794</td><td>1.613410</td><td>20.37143</td></tr>\n",
       "\t<tr><th scope=row>TwdlF44</th><td>bq</td><td>TwdlF     </td><td> 6697</td><td> 13997</td><td> 1776.8570</td><td> 492.98094</td><td>1.846800</td><td>19.93791</td></tr>\n",
       "\t<tr><th scope=row>RpL644</th><td>bq</td><td>RpL6      </td><td>22539</td><td> 62854</td><td> 5951.1989</td><td>1986.58711</td><td>1.582162</td><td>19.83906</td></tr>\n",
       "\t<tr><th scope=row>RpS644</th><td>bq</td><td>RpS6      </td><td>18692</td><td> 50047</td><td> 4918.5066</td><td>1630.96020</td><td>1.591614</td><td>19.52003</td></tr>\n",
       "\t<tr><th scope=row>Hsc70-444</th><td>bq</td><td>Hsc70-4   </td><td>19542</td><td> 54591</td><td> 5251.8047</td><td>1790.73474</td><td>1.551456</td><td>19.17425</td></tr>\n",
       "\t<tr><th scope=row>RpL7A44</th><td>bq</td><td>RpL7A     </td><td>21373</td><td> 62912</td><td> 5587.7042</td><td>1969.40485</td><td>1.503764</td><td>18.71930</td></tr>\n",
       "\t<tr><th scope=row>eEF244</th><td>bq</td><td>eEF2      </td><td>27581</td><td> 82036</td><td> 7219.6870</td><td>2627.25166</td><td>1.457833</td><td>18.68639</td></tr>\n",
       "\t<tr><th scope=row>RpL454</th><td>bq</td><td>RpL4      </td><td>14388</td><td> 40203</td><td> 3855.1943</td><td>1299.19926</td><td>1.568071</td><td>18.68037</td></tr>\n",
       "\t<tr><th scope=row>Gabat44</th><td>bq</td><td>Gabat     </td><td> 1392</td><td>  2557</td><td>  375.6893</td><td>  81.77015</td><td>2.182357</td><td>18.67494</td></tr>\n",
       "\t<tr><th scope=row>CG908344</th><td>bq</td><td>CG9083    </td><td> 5361</td><td> 11400</td><td> 1406.4989</td><td> 413.48149</td><td>1.762729</td><td>18.43624</td></tr>\n",
       "\t<tr><th scope=row>CG767544</th><td>bq</td><td>CG7675    </td><td> 2668</td><td>  5580</td><td>  717.7727</td><td> 186.19533</td><td>1.938983</td><td>18.39976</td></tr>\n",
       "\t<tr><th scope=row>Tm244</th><td>bq</td><td>Tm2       </td><td> 3845</td><td>  9615</td><td> 1041.4820</td><td> 298.07987</td><td>1.800035</td><td>18.04681</td></tr>\n",
       "\t<tr><th scope=row>Gdh44</th><td>bq</td><td>Gdh       </td><td> 4394</td><td>  9732</td><td> 1180.6039</td><td> 346.38616</td><td>1.764913</td><td>18.01364</td></tr>\n",
       "\t<tr><th scope=row>RpL3100</th><td>bq</td><td>RpL3      </td><td>19347</td><td> 58621</td><td> 5215.0858</td><td>1898.14933</td><td>1.457338</td><td>17.99630</td></tr>\n",
       "\t<tr><th scope=row>LamC44</th><td>bq</td><td>LamC      </td><td> 2474</td><td>  5218</td><td>  668.7581</td><td> 176.29974</td><td>1.915294</td><td>17.97981</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tAct57B44 & bq & Act57B     & 24201 &  49032 &  6408.3207 & 1439.01846 & 2.153859 & 27.23761\\\\\n",
       "\tCG296244 & bq & CG2962     & 14862 &  28955 &  3852.2991 & 1015.30233 & 1.922390 & 22.89928\\\\\n",
       "\tMlc244 & bq & Mlc2       & 12441 &  28458 &  3299.5489 &  866.05437 & 1.928074 & 22.53628\\\\\n",
       "\teEF1alpha144 & bq & eEF1alpha1 & 76454 & 218993 & 20543.3019 & 7206.71742 & 1.511054 & 21.64804\\\\\n",
       "\tCG1573144 & bq & CG15731    &  2958 &   5182 &   782.8430 &  168.63752 & 2.206268 & 21.21199\\\\\n",
       "\tRpL23A44 & bq & RpL23A     & 23945 &  63966 &  6321.6670 & 2065.07794 & 1.613410 & 20.37143\\\\\n",
       "\tTwdlF44 & bq & TwdlF      &  6697 &  13997 &  1776.8570 &  492.98094 & 1.846800 & 19.93791\\\\\n",
       "\tRpL644 & bq & RpL6       & 22539 &  62854 &  5951.1989 & 1986.58711 & 1.582162 & 19.83906\\\\\n",
       "\tRpS644 & bq & RpS6       & 18692 &  50047 &  4918.5066 & 1630.96020 & 1.591614 & 19.52003\\\\\n",
       "\tHsc70-444 & bq & Hsc70-4    & 19542 &  54591 &  5251.8047 & 1790.73474 & 1.551456 & 19.17425\\\\\n",
       "\tRpL7A44 & bq & RpL7A      & 21373 &  62912 &  5587.7042 & 1969.40485 & 1.503764 & 18.71930\\\\\n",
       "\teEF244 & bq & eEF2       & 27581 &  82036 &  7219.6870 & 2627.25166 & 1.457833 & 18.68639\\\\\n",
       "\tRpL454 & bq & RpL4       & 14388 &  40203 &  3855.1943 & 1299.19926 & 1.568071 & 18.68037\\\\\n",
       "\tGabat44 & bq & Gabat      &  1392 &   2557 &   375.6893 &   81.77015 & 2.182357 & 18.67494\\\\\n",
       "\tCG908344 & bq & CG9083     &  5361 &  11400 &  1406.4989 &  413.48149 & 1.762729 & 18.43624\\\\\n",
       "\tCG767544 & bq & CG7675     &  2668 &   5580 &   717.7727 &  186.19533 & 1.938983 & 18.39976\\\\\n",
       "\tTm244 & bq & Tm2        &  3845 &   9615 &  1041.4820 &  298.07987 & 1.800035 & 18.04681\\\\\n",
       "\tGdh44 & bq & Gdh        &  4394 &   9732 &  1180.6039 &  346.38616 & 1.764913 & 18.01364\\\\\n",
       "\tRpL3100 & bq & RpL3       & 19347 &  58621 &  5215.0858 & 1898.14933 & 1.457338 & 17.99630\\\\\n",
       "\tLamC44 & bq & LamC       &  2474 &   5218 &   668.7581 &  176.29974 & 1.915294 & 17.97981\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Act57B44 | bq | Act57B     | 24201 |  49032 |  6408.3207 | 1439.01846 | 2.153859 | 27.23761 |\n",
       "| CG296244 | bq | CG2962     | 14862 |  28955 |  3852.2991 | 1015.30233 | 1.922390 | 22.89928 |\n",
       "| Mlc244 | bq | Mlc2       | 12441 |  28458 |  3299.5489 |  866.05437 | 1.928074 | 22.53628 |\n",
       "| eEF1alpha144 | bq | eEF1alpha1 | 76454 | 218993 | 20543.3019 | 7206.71742 | 1.511054 | 21.64804 |\n",
       "| CG1573144 | bq | CG15731    |  2958 |   5182 |   782.8430 |  168.63752 | 2.206268 | 21.21199 |\n",
       "| RpL23A44 | bq | RpL23A     | 23945 |  63966 |  6321.6670 | 2065.07794 | 1.613410 | 20.37143 |\n",
       "| TwdlF44 | bq | TwdlF      |  6697 |  13997 |  1776.8570 |  492.98094 | 1.846800 | 19.93791 |\n",
       "| RpL644 | bq | RpL6       | 22539 |  62854 |  5951.1989 | 1986.58711 | 1.582162 | 19.83906 |\n",
       "| RpS644 | bq | RpS6       | 18692 |  50047 |  4918.5066 | 1630.96020 | 1.591614 | 19.52003 |\n",
       "| Hsc70-444 | bq | Hsc70-4    | 19542 |  54591 |  5251.8047 | 1790.73474 | 1.551456 | 19.17425 |\n",
       "| RpL7A44 | bq | RpL7A      | 21373 |  62912 |  5587.7042 | 1969.40485 | 1.503764 | 18.71930 |\n",
       "| eEF244 | bq | eEF2       | 27581 |  82036 |  7219.6870 | 2627.25166 | 1.457833 | 18.68639 |\n",
       "| RpL454 | bq | RpL4       | 14388 |  40203 |  3855.1943 | 1299.19926 | 1.568071 | 18.68037 |\n",
       "| Gabat44 | bq | Gabat      |  1392 |   2557 |   375.6893 |   81.77015 | 2.182357 | 18.67494 |\n",
       "| CG908344 | bq | CG9083     |  5361 |  11400 |  1406.4989 |  413.48149 | 1.762729 | 18.43624 |\n",
       "| CG767544 | bq | CG7675     |  2668 |   5580 |   717.7727 |  186.19533 | 1.938983 | 18.39976 |\n",
       "| Tm244 | bq | Tm2        |  3845 |   9615 |  1041.4820 |  298.07987 | 1.800035 | 18.04681 |\n",
       "| Gdh44 | bq | Gdh        |  4394 |   9732 |  1180.6039 |  346.38616 | 1.764913 | 18.01364 |\n",
       "| RpL3100 | bq | RpL3       | 19347 |  58621 |  5215.0858 | 1898.14933 | 1.457338 | 17.99630 |\n",
       "| LamC44 | bq | LamC       |  2474 |   5218 |   668.7581 |  176.29974 | 1.915294 | 17.97981 |\n",
       "\n"
      ],
      "text/plain": [
       "             cell.type.2 gene       cell.type.2.n.umi other.n.umi\n",
       "Act57B44     bq          Act57B     24201              49032     \n",
       "CG296244     bq          CG2962     14862              28955     \n",
       "Mlc244       bq          Mlc2       12441              28458     \n",
       "eEF1alpha144 bq          eEF1alpha1 76454             218993     \n",
       "CG1573144    bq          CG15731     2958               5182     \n",
       "RpL23A44     bq          RpL23A     23945              63966     \n",
       "TwdlF44      bq          TwdlF       6697              13997     \n",
       "RpL644       bq          RpL6       22539              62854     \n",
       "RpS644       bq          RpS6       18692              50047     \n",
       "Hsc70-444    bq          Hsc70-4    19542              54591     \n",
       "RpL7A44      bq          RpL7A      21373              62912     \n",
       "eEF244       bq          eEF2       27581              82036     \n",
       "RpL454       bq          RpL4       14388              40203     \n",
       "Gabat44      bq          Gabat       1392               2557     \n",
       "CG908344     bq          CG9083      5361              11400     \n",
       "CG767544     bq          CG7675      2668               5580     \n",
       "Tm244        bq          Tm2         3845               9615     \n",
       "Gdh44        bq          Gdh         4394               9732     \n",
       "RpL3100      bq          RpL3       19347              58621     \n",
       "LamC44       bq          LamC        2474               5218     \n",
       "             cell.type.2.tpm other.tpm  log2.ratio heuristic.score\n",
       "Act57B44      6408.3207      1439.01846 2.153859   27.23761       \n",
       "CG296244      3852.2991      1015.30233 1.922390   22.89928       \n",
       "Mlc244        3299.5489       866.05437 1.928074   22.53628       \n",
       "eEF1alpha144 20543.3019      7206.71742 1.511054   21.64804       \n",
       "CG1573144      782.8430       168.63752 2.206268   21.21199       \n",
       "RpL23A44      6321.6670      2065.07794 1.613410   20.37143       \n",
       "TwdlF44       1776.8570       492.98094 1.846800   19.93791       \n",
       "RpL644        5951.1989      1986.58711 1.582162   19.83906       \n",
       "RpS644        4918.5066      1630.96020 1.591614   19.52003       \n",
       "Hsc70-444     5251.8047      1790.73474 1.551456   19.17425       \n",
       "RpL7A44       5587.7042      1969.40485 1.503764   18.71930       \n",
       "eEF244        7219.6870      2627.25166 1.457833   18.68639       \n",
       "RpL454        3855.1943      1299.19926 1.568071   18.68037       \n",
       "Gabat44        375.6893        81.77015 2.182357   18.67494       \n",
       "CG908344      1406.4989       413.48149 1.762729   18.43624       \n",
       "CG767544       717.7727       186.19533 1.938983   18.39976       \n",
       "Tm244         1041.4820       298.07987 1.800035   18.04681       \n",
       "Gdh44         1180.6039       346.38616 1.764913   18.01364       \n",
       "RpL3100       5215.0858      1898.14933 1.457338   17.99630       \n",
       "LamC44         668.7581       176.29974 1.915294   17.97981       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \n",
    "                                              \"bq\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1516,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Visceral_muscle_2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Visceral_muscle_2\", df_sub300$cluster %in% bz &\n",
    "                         df_sub300$umap_2 < 1*df_sub300$umap_1 -3.3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1517,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Visceral_muscle\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Visceral_muscle\", df_sub300$cluster %in% bz &\n",
    "                         df_sub300$umap_2 >= 1*df_sub300$umap_1 -3.3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 312,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bz' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bz' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bz)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 313,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'bz' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'bz' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(bz)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1508,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>up53</th><td>bz</td><td>up       </td><td>6626</td><td>28694</td><td> 9532.6511</td><td>1202.03845</td><td>2.986195</td><td>39.47395</td></tr>\n",
       "\t<tr><th scope=row>bt53</th><td>bz</td><td>bt       </td><td>8603</td><td>46215</td><td>11957.0915</td><td>1651.15272</td><td>2.855448</td><td>38.67903</td></tr>\n",
       "\t<tr><th scope=row>wupA53</th><td>bz</td><td>wupA     </td><td>4161</td><td>21040</td><td> 5918.9464</td><td> 828.25226</td><td>2.835457</td><td>35.53216</td></tr>\n",
       "\t<tr><th scope=row>Sh53</th><td>bz</td><td>Sh       </td><td>2468</td><td> 9715</td><td> 3405.5853</td><td> 445.37387</td><td>2.931578</td><td>34.39946</td></tr>\n",
       "\t<tr><th scope=row>Mhc53</th><td>bz</td><td>Mhc      </td><td>7437</td><td>51099</td><td>10562.4954</td><td>1784.74478</td><td>2.564353</td><td>34.27719</td></tr>\n",
       "\t<tr><th scope=row>CG3352153</th><td>bz</td><td>CG33521  </td><td> 744</td><td> 2636</td><td> 1051.7574</td><td> 115.18077</td><td>3.178359</td><td>31.91059</td></tr>\n",
       "\t<tr><th scope=row>jp53</th><td>bz</td><td>jp       </td><td> 427</td><td> 1694</td><td>  623.4231</td><td>  65.70320</td><td>3.224384</td><td>29.94285</td></tr>\n",
       "\t<tr><th scope=row>Neto53</th><td>bz</td><td>Neto     </td><td> 397</td><td> 1337</td><td>  580.8132</td><td>  59.94962</td><td>3.252385</td><td>29.87125</td></tr>\n",
       "\t<tr><th scope=row>CG4442253</th><td>bz</td><td>CG44422  </td><td> 795</td><td> 3587</td><td> 1194.7991</td><td> 162.27534</td><td>2.871389</td><td>29.35639</td></tr>\n",
       "\t<tr><th scope=row>l(2)0128953</th><td>bz</td><td>l(2)01289</td><td> 697</td><td> 3320</td><td>  961.4359</td><td> 122.64449</td><td>2.958993</td><td>29.32523</td></tr>\n",
       "\t<tr><th scope=row>Rgk353</th><td>bz</td><td>Rgk3     </td><td> 238</td><td>  749</td><td>  331.8769</td><td>  30.74347</td><td>3.386116</td><td>28.37174</td></tr>\n",
       "\t<tr><th scope=row>Nlg153</th><td>bz</td><td>Nlg1     </td><td> 248</td><td>  770</td><td>  352.3538</td><td>  34.63533</td><td>3.305645</td><td>27.98218</td></tr>\n",
       "\t<tr><th scope=row>CG167473</th><td>bz</td><td>CG1674   </td><td> 499</td><td> 2316</td><td>  714.2636</td><td>  93.26415</td><td>2.921676</td><td>27.70429</td></tr>\n",
       "\t<tr><th scope=row>mspo53</th><td>bz</td><td>mspo     </td><td>1136</td><td> 6393</td><td> 1632.7361</td><td> 285.87776</td><td>2.508784</td><td>26.77865</td></tr>\n",
       "\t<tr><th scope=row>Eip63F-153</th><td>bz</td><td>Eip63F-1 </td><td> 564</td><td> 2626</td><td>  797.4296</td><td> 116.70798</td><td>2.760145</td><td>26.61062</td></tr>\n",
       "\t<tr><th scope=row>CG3441753</th><td>bz</td><td>CG34417  </td><td>1528</td><td>10215</td><td> 2195.9268</td><td> 432.80950</td><td>2.339696</td><td>25.97360</td></tr>\n",
       "\t<tr><th scope=row>sals53</th><td>bz</td><td>sals     </td><td> 506</td><td> 2513</td><td>  708.0793</td><td> 108.99761</td><td>2.686439</td><td>25.44005</td></tr>\n",
       "\t<tr><th scope=row>Strn-Mlck53</th><td>bz</td><td>Strn-Mlck</td><td> 926</td><td> 6323</td><td> 1279.5360</td><td> 238.78180</td><td>2.415827</td><td>24.93745</td></tr>\n",
       "\t<tr><th scope=row>Fhos53</th><td>bz</td><td>Fhos     </td><td>1070</td><td> 6482</td><td> 1512.5323</td><td> 294.50384</td><td>2.355717</td><td>24.88510</td></tr>\n",
       "\t<tr><th scope=row>dpr1053</th><td>bz</td><td>dpr10    </td><td> 409</td><td> 1871</td><td>  556.6907</td><td>  83.63592</td><td>2.717534</td><td>24.79294</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tup53 & bz & up        & 6626 & 28694 &  9532.6511 & 1202.03845 & 2.986195 & 39.47395\\\\\n",
       "\tbt53 & bz & bt        & 8603 & 46215 & 11957.0915 & 1651.15272 & 2.855448 & 38.67903\\\\\n",
       "\twupA53 & bz & wupA      & 4161 & 21040 &  5918.9464 &  828.25226 & 2.835457 & 35.53216\\\\\n",
       "\tSh53 & bz & Sh        & 2468 &  9715 &  3405.5853 &  445.37387 & 2.931578 & 34.39946\\\\\n",
       "\tMhc53 & bz & Mhc       & 7437 & 51099 & 10562.4954 & 1784.74478 & 2.564353 & 34.27719\\\\\n",
       "\tCG3352153 & bz & CG33521   &  744 &  2636 &  1051.7574 &  115.18077 & 3.178359 & 31.91059\\\\\n",
       "\tjp53 & bz & jp        &  427 &  1694 &   623.4231 &   65.70320 & 3.224384 & 29.94285\\\\\n",
       "\tNeto53 & bz & Neto      &  397 &  1337 &   580.8132 &   59.94962 & 3.252385 & 29.87125\\\\\n",
       "\tCG4442253 & bz & CG44422   &  795 &  3587 &  1194.7991 &  162.27534 & 2.871389 & 29.35639\\\\\n",
       "\tl(2)0128953 & bz & l(2)01289 &  697 &  3320 &   961.4359 &  122.64449 & 2.958993 & 29.32523\\\\\n",
       "\tRgk353 & bz & Rgk3      &  238 &   749 &   331.8769 &   30.74347 & 3.386116 & 28.37174\\\\\n",
       "\tNlg153 & bz & Nlg1      &  248 &   770 &   352.3538 &   34.63533 & 3.305645 & 27.98218\\\\\n",
       "\tCG167473 & bz & CG1674    &  499 &  2316 &   714.2636 &   93.26415 & 2.921676 & 27.70429\\\\\n",
       "\tmspo53 & bz & mspo      & 1136 &  6393 &  1632.7361 &  285.87776 & 2.508784 & 26.77865\\\\\n",
       "\tEip63F-153 & bz & Eip63F-1  &  564 &  2626 &   797.4296 &  116.70798 & 2.760145 & 26.61062\\\\\n",
       "\tCG3441753 & bz & CG34417   & 1528 & 10215 &  2195.9268 &  432.80950 & 2.339696 & 25.97360\\\\\n",
       "\tsals53 & bz & sals      &  506 &  2513 &   708.0793 &  108.99761 & 2.686439 & 25.44005\\\\\n",
       "\tStrn-Mlck53 & bz & Strn-Mlck &  926 &  6323 &  1279.5360 &  238.78180 & 2.415827 & 24.93745\\\\\n",
       "\tFhos53 & bz & Fhos      & 1070 &  6482 &  1512.5323 &  294.50384 & 2.355717 & 24.88510\\\\\n",
       "\tdpr1053 & bz & dpr10     &  409 &  1871 &   556.6907 &   83.63592 & 2.717534 & 24.79294\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| up53 | bz | up        | 6626 | 28694 |  9532.6511 | 1202.03845 | 2.986195 | 39.47395 |\n",
       "| bt53 | bz | bt        | 8603 | 46215 | 11957.0915 | 1651.15272 | 2.855448 | 38.67903 |\n",
       "| wupA53 | bz | wupA      | 4161 | 21040 |  5918.9464 |  828.25226 | 2.835457 | 35.53216 |\n",
       "| Sh53 | bz | Sh        | 2468 |  9715 |  3405.5853 |  445.37387 | 2.931578 | 34.39946 |\n",
       "| Mhc53 | bz | Mhc       | 7437 | 51099 | 10562.4954 | 1784.74478 | 2.564353 | 34.27719 |\n",
       "| CG3352153 | bz | CG33521   |  744 |  2636 |  1051.7574 |  115.18077 | 3.178359 | 31.91059 |\n",
       "| jp53 | bz | jp        |  427 |  1694 |   623.4231 |   65.70320 | 3.224384 | 29.94285 |\n",
       "| Neto53 | bz | Neto      |  397 |  1337 |   580.8132 |   59.94962 | 3.252385 | 29.87125 |\n",
       "| CG4442253 | bz | CG44422   |  795 |  3587 |  1194.7991 |  162.27534 | 2.871389 | 29.35639 |\n",
       "| l(2)0128953 | bz | l(2)01289 |  697 |  3320 |   961.4359 |  122.64449 | 2.958993 | 29.32523 |\n",
       "| Rgk353 | bz | Rgk3      |  238 |   749 |   331.8769 |   30.74347 | 3.386116 | 28.37174 |\n",
       "| Nlg153 | bz | Nlg1      |  248 |   770 |   352.3538 |   34.63533 | 3.305645 | 27.98218 |\n",
       "| CG167473 | bz | CG1674    |  499 |  2316 |   714.2636 |   93.26415 | 2.921676 | 27.70429 |\n",
       "| mspo53 | bz | mspo      | 1136 |  6393 |  1632.7361 |  285.87776 | 2.508784 | 26.77865 |\n",
       "| Eip63F-153 | bz | Eip63F-1  |  564 |  2626 |   797.4296 |  116.70798 | 2.760145 | 26.61062 |\n",
       "| CG3441753 | bz | CG34417   | 1528 | 10215 |  2195.9268 |  432.80950 | 2.339696 | 25.97360 |\n",
       "| sals53 | bz | sals      |  506 |  2513 |   708.0793 |  108.99761 | 2.686439 | 25.44005 |\n",
       "| Strn-Mlck53 | bz | Strn-Mlck |  926 |  6323 |  1279.5360 |  238.78180 | 2.415827 | 24.93745 |\n",
       "| Fhos53 | bz | Fhos      | 1070 |  6482 |  1512.5323 |  294.50384 | 2.355717 | 24.88510 |\n",
       "| dpr1053 | bz | dpr10     |  409 |  1871 |   556.6907 |   83.63592 | 2.717534 | 24.79294 |\n",
       "\n"
      ],
      "text/plain": [
       "            cell.type.2 gene      cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "up53        bz          up        6626              28694        9532.6511     \n",
       "bt53        bz          bt        8603              46215       11957.0915     \n",
       "wupA53      bz          wupA      4161              21040        5918.9464     \n",
       "Sh53        bz          Sh        2468               9715        3405.5853     \n",
       "Mhc53       bz          Mhc       7437              51099       10562.4954     \n",
       "CG3352153   bz          CG33521    744               2636        1051.7574     \n",
       "jp53        bz          jp         427               1694         623.4231     \n",
       "Neto53      bz          Neto       397               1337         580.8132     \n",
       "CG4442253   bz          CG44422    795               3587        1194.7991     \n",
       "l(2)0128953 bz          l(2)01289  697               3320         961.4359     \n",
       "Rgk353      bz          Rgk3       238                749         331.8769     \n",
       "Nlg153      bz          Nlg1       248                770         352.3538     \n",
       "CG167473    bz          CG1674     499               2316         714.2636     \n",
       "mspo53      bz          mspo      1136               6393        1632.7361     \n",
       "Eip63F-153  bz          Eip63F-1   564               2626         797.4296     \n",
       "CG3441753   bz          CG34417   1528              10215        2195.9268     \n",
       "sals53      bz          sals       506               2513         708.0793     \n",
       "Strn-Mlck53 bz          Strn-Mlck  926               6323        1279.5360     \n",
       "Fhos53      bz          Fhos      1070               6482        1512.5323     \n",
       "dpr1053     bz          dpr10      409               1871         556.6907     \n",
       "            other.tpm  log2.ratio heuristic.score\n",
       "up53        1202.03845 2.986195   39.47395       \n",
       "bt53        1651.15272 2.855448   38.67903       \n",
       "wupA53       828.25226 2.835457   35.53216       \n",
       "Sh53         445.37387 2.931578   34.39946       \n",
       "Mhc53       1784.74478 2.564353   34.27719       \n",
       "CG3352153    115.18077 3.178359   31.91059       \n",
       "jp53          65.70320 3.224384   29.94285       \n",
       "Neto53        59.94962 3.252385   29.87125       \n",
       "CG4442253    162.27534 2.871389   29.35639       \n",
       "l(2)0128953  122.64449 2.958993   29.32523       \n",
       "Rgk353        30.74347 3.386116   28.37174       \n",
       "Nlg153        34.63533 3.305645   27.98218       \n",
       "CG167473      93.26415 2.921676   27.70429       \n",
       "mspo53       285.87776 2.508784   26.77865       \n",
       "Eip63F-153   116.70798 2.760145   26.61062       \n",
       "CG3441753    432.80950 2.339696   25.97360       \n",
       "sals53       108.99761 2.686439   25.44005       \n",
       "Strn-Mlck53  238.78180 2.415827   24.93745       \n",
       "Fhos53       294.50384 2.355717   24.88510       \n",
       "dpr1053       83.63592 2.717534   24.79294       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"bz\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 314,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"up\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"up\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"up\"), \n",
    "                   size_pos = 2, size_neg = 0.05, stroke = 0.008, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"bt\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"wupA\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 315,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(18)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus &  \n",
    "             # df_sub300$umap_2 < 0.7*umap_1 -6 &\n",
    "              df_sub300$umap_2 < 1*umap_1 -3.3), \n",
    "           size =  ifelse(df_sub300$cluster %in% clus, 1, 0.7), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic() +\n",
    "geom_abline(slope = 0.7, intercept = -3.3) +\n",
    "geom_abline(slope = 1, intercept = -3.3) +\n",
    "geom_abline(slope = -1., intercept = -6.7) +\n",
    " xlim(0, 4) + ylim(-3, 1)\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1514,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 1508\n",
      "\n",
      "# of cells sampled in set 2: 615\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG14304</th><td>CG14304   </td><td>  70</td><td>  1</td><td> 101.56334</td><td>  3.226223</td><td>Set 1</td><td>4.586867</td><td>30.64198</td></tr>\n",
       "\t<tr><th scope=row>modSP</th><td>modSP     </td><td>  25</td><td>  0</td><td>  45.44194</td><td>  0.000000</td><td>Set 1</td><td>5.505953</td><td>30.48842</td></tr>\n",
       "\t<tr><th scope=row>Snmp2</th><td>Snmp2     </td><td>  23</td><td>  0</td><td>  41.85317</td><td>  0.000000</td><td>Set 1</td><td>5.387265</td><td>29.20614</td></tr>\n",
       "\t<tr><th scope=row>CG5835</th><td>CG5835    </td><td>  25</td><td>  0</td><td>  39.91086</td><td>  0.000000</td><td>Set 1</td><td>5.318710</td><td>28.47856</td></tr>\n",
       "\t<tr><th scope=row>Rab40</th><td>Rab40     </td><td>  19</td><td>  0</td><td>  34.55481</td><td>  0.000000</td><td>Set 1</td><td>5.110815</td><td>26.33078</td></tr>\n",
       "\t<tr><th scope=row>ph-d</th><td>ph-d      </td><td>  19</td><td>  0</td><td>  33.51014</td><td>  0.000000</td><td>Set 1</td><td>5.066526</td><td>25.88462</td></tr>\n",
       "\t<tr><th scope=row>ine</th><td>ine       </td><td>  43</td><td>  1</td><td>  78.45769</td><td>  3.637620</td><td>Set 1</td><td>4.080458</td><td>25.75632</td></tr>\n",
       "\t<tr><th scope=row>CG7011</th><td>CG7011    </td><td>  19</td><td>  0</td><td>  33.05476</td><td>  0.000000</td><td>Set 1</td><td>5.046786</td><td>25.68705</td></tr>\n",
       "\t<tr><th scope=row>Syx5</th><td>Syx5      </td><td>  17</td><td>  0</td><td>  32.19570</td><td>  0.000000</td><td>Set 1</td><td>5.008796</td><td>25.30907</td></tr>\n",
       "\t<tr><th scope=row>CG10237</th><td>CG10237   </td><td>  23</td><td>  1</td><td>  45.38121</td><td>  1.006821</td><td>Set 1</td><td>4.499111</td><td>24.90469</td></tr>\n",
       "\t<tr><th scope=row>MICAL-like</th><td>MICAL-like</td><td>  16</td><td>  0</td><td>  30.56956</td><td>  0.000000</td><td>Set 1</td><td>4.934024</td><td>24.57372</td></tr>\n",
       "\t<tr><th scope=row>CG11438</th><td>CG11438   </td><td>  17</td><td>  0</td><td>  29.97227</td><td>  0.000000</td><td>Set 1</td><td>4.905557</td><td>24.29676</td></tr>\n",
       "\t<tr><th scope=row>ND-ASHI</th><td>ND-ASHI   </td><td>  16</td><td>  0</td><td>  29.57181</td><td>  0.000000</td><td>Set 1</td><td>4.886150</td><td>24.10890</td></tr>\n",
       "\t<tr><th scope=row>NT1</th><td>NT1       </td><td> 155</td><td>  9</td><td> 290.27552</td><td> 37.459748</td><td>Set 1</td><td>2.916002</td><td>23.87109</td></tr>\n",
       "\t<tr><th scope=row>CG1307</th><td>CG1307    </td><td>  16</td><td>  0</td><td>  29.04102</td><td>  0.000000</td><td>Set 1</td><td>4.860020</td><td>23.85717</td></tr>\n",
       "\t<tr><th scope=row>Kal1</th><td>Kal1      </td><td>  15</td><td>  0</td><td>  28.81075</td><td>  0.000000</td><td>Set 1</td><td>4.848536</td><td>23.74697</td></tr>\n",
       "\t<tr><th scope=row>unc-4</th><td>unc-4     </td><td>  12</td><td>  0</td><td>  27.36481</td><td>  0.000000</td><td>Set 1</td><td>4.774250</td><td>23.04068</td></tr>\n",
       "\t<tr><th scope=row>CG9095</th><td>CG9095    </td><td>1349</td><td>124</td><td>2301.51187</td><td>554.365400</td><td>Set 1</td><td>2.051073</td><td>22.90842</td></tr>\n",
       "\t<tr><th scope=row>CG4928</th><td>CG4928    </td><td> 168</td><td> 10</td><td> 295.99841</td><td> 41.931575</td><td>Set 1</td><td>2.785478</td><td>22.88079</td></tr>\n",
       "\t<tr><th scope=row>dsx-c73A</th><td>dsx-c73A  </td><td> 520</td><td> 47</td><td> 926.04380</td><td>186.010363</td><td>Set 1</td><td>2.307962</td><td>22.74841</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG14304 & CG14304    &   70 &   1 &  101.56334 &   3.226223 & Set 1 & 4.586867 & 30.64198\\\\\n",
       "\tmodSP & modSP      &   25 &   0 &   45.44194 &   0.000000 & Set 1 & 5.505953 & 30.48842\\\\\n",
       "\tSnmp2 & Snmp2      &   23 &   0 &   41.85317 &   0.000000 & Set 1 & 5.387265 & 29.20614\\\\\n",
       "\tCG5835 & CG5835     &   25 &   0 &   39.91086 &   0.000000 & Set 1 & 5.318710 & 28.47856\\\\\n",
       "\tRab40 & Rab40      &   19 &   0 &   34.55481 &   0.000000 & Set 1 & 5.110815 & 26.33078\\\\\n",
       "\tph-d & ph-d       &   19 &   0 &   33.51014 &   0.000000 & Set 1 & 5.066526 & 25.88462\\\\\n",
       "\tine & ine        &   43 &   1 &   78.45769 &   3.637620 & Set 1 & 4.080458 & 25.75632\\\\\n",
       "\tCG7011 & CG7011     &   19 &   0 &   33.05476 &   0.000000 & Set 1 & 5.046786 & 25.68705\\\\\n",
       "\tSyx5 & Syx5       &   17 &   0 &   32.19570 &   0.000000 & Set 1 & 5.008796 & 25.30907\\\\\n",
       "\tCG10237 & CG10237    &   23 &   1 &   45.38121 &   1.006821 & Set 1 & 4.499111 & 24.90469\\\\\n",
       "\tMICAL-like & MICAL-like &   16 &   0 &   30.56956 &   0.000000 & Set 1 & 4.934024 & 24.57372\\\\\n",
       "\tCG11438 & CG11438    &   17 &   0 &   29.97227 &   0.000000 & Set 1 & 4.905557 & 24.29676\\\\\n",
       "\tND-ASHI & ND-ASHI    &   16 &   0 &   29.57181 &   0.000000 & Set 1 & 4.886150 & 24.10890\\\\\n",
       "\tNT1 & NT1        &  155 &   9 &  290.27552 &  37.459748 & Set 1 & 2.916002 & 23.87109\\\\\n",
       "\tCG1307 & CG1307     &   16 &   0 &   29.04102 &   0.000000 & Set 1 & 4.860020 & 23.85717\\\\\n",
       "\tKal1 & Kal1       &   15 &   0 &   28.81075 &   0.000000 & Set 1 & 4.848536 & 23.74697\\\\\n",
       "\tunc-4 & unc-4      &   12 &   0 &   27.36481 &   0.000000 & Set 1 & 4.774250 & 23.04068\\\\\n",
       "\tCG9095 & CG9095     & 1349 & 124 & 2301.51187 & 554.365400 & Set 1 & 2.051073 & 22.90842\\\\\n",
       "\tCG4928 & CG4928     &  168 &  10 &  295.99841 &  41.931575 & Set 1 & 2.785478 & 22.88079\\\\\n",
       "\tdsx-c73A & dsx-c73A   &  520 &  47 &  926.04380 & 186.010363 & Set 1 & 2.307962 & 22.74841\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG14304 | CG14304    |   70 |   1 |  101.56334 |   3.226223 | Set 1 | 4.586867 | 30.64198 |\n",
       "| modSP | modSP      |   25 |   0 |   45.44194 |   0.000000 | Set 1 | 5.505953 | 30.48842 |\n",
       "| Snmp2 | Snmp2      |   23 |   0 |   41.85317 |   0.000000 | Set 1 | 5.387265 | 29.20614 |\n",
       "| CG5835 | CG5835     |   25 |   0 |   39.91086 |   0.000000 | Set 1 | 5.318710 | 28.47856 |\n",
       "| Rab40 | Rab40      |   19 |   0 |   34.55481 |   0.000000 | Set 1 | 5.110815 | 26.33078 |\n",
       "| ph-d | ph-d       |   19 |   0 |   33.51014 |   0.000000 | Set 1 | 5.066526 | 25.88462 |\n",
       "| ine | ine        |   43 |   1 |   78.45769 |   3.637620 | Set 1 | 4.080458 | 25.75632 |\n",
       "| CG7011 | CG7011     |   19 |   0 |   33.05476 |   0.000000 | Set 1 | 5.046786 | 25.68705 |\n",
       "| Syx5 | Syx5       |   17 |   0 |   32.19570 |   0.000000 | Set 1 | 5.008796 | 25.30907 |\n",
       "| CG10237 | CG10237    |   23 |   1 |   45.38121 |   1.006821 | Set 1 | 4.499111 | 24.90469 |\n",
       "| MICAL-like | MICAL-like |   16 |   0 |   30.56956 |   0.000000 | Set 1 | 4.934024 | 24.57372 |\n",
       "| CG11438 | CG11438    |   17 |   0 |   29.97227 |   0.000000 | Set 1 | 4.905557 | 24.29676 |\n",
       "| ND-ASHI | ND-ASHI    |   16 |   0 |   29.57181 |   0.000000 | Set 1 | 4.886150 | 24.10890 |\n",
       "| NT1 | NT1        |  155 |   9 |  290.27552 |  37.459748 | Set 1 | 2.916002 | 23.87109 |\n",
       "| CG1307 | CG1307     |   16 |   0 |   29.04102 |   0.000000 | Set 1 | 4.860020 | 23.85717 |\n",
       "| Kal1 | Kal1       |   15 |   0 |   28.81075 |   0.000000 | Set 1 | 4.848536 | 23.74697 |\n",
       "| unc-4 | unc-4      |   12 |   0 |   27.36481 |   0.000000 | Set 1 | 4.774250 | 23.04068 |\n",
       "| CG9095 | CG9095     | 1349 | 124 | 2301.51187 | 554.365400 | Set 1 | 2.051073 | 22.90842 |\n",
       "| CG4928 | CG4928     |  168 |  10 |  295.99841 |  41.931575 | Set 1 | 2.785478 | 22.88079 |\n",
       "| dsx-c73A | dsx-c73A   |  520 |  47 |  926.04380 | 186.010363 | Set 1 | 2.307962 | 22.74841 |\n",
       "\n"
      ],
      "text/plain": [
       "           gene       set.1.n.umi set.2.n.umi set.1.tpm  set.2.tpm  higher.expr\n",
       "CG14304    CG14304      70          1          101.56334   3.226223 Set 1      \n",
       "modSP      modSP        25          0           45.44194   0.000000 Set 1      \n",
       "Snmp2      Snmp2        23          0           41.85317   0.000000 Set 1      \n",
       "CG5835     CG5835       25          0           39.91086   0.000000 Set 1      \n",
       "Rab40      Rab40        19          0           34.55481   0.000000 Set 1      \n",
       "ph-d       ph-d         19          0           33.51014   0.000000 Set 1      \n",
       "ine        ine          43          1           78.45769   3.637620 Set 1      \n",
       "CG7011     CG7011       19          0           33.05476   0.000000 Set 1      \n",
       "Syx5       Syx5         17          0           32.19570   0.000000 Set 1      \n",
       "CG10237    CG10237      23          1           45.38121   1.006821 Set 1      \n",
       "MICAL-like MICAL-like   16          0           30.56956   0.000000 Set 1      \n",
       "CG11438    CG11438      17          0           29.97227   0.000000 Set 1      \n",
       "ND-ASHI    ND-ASHI      16          0           29.57181   0.000000 Set 1      \n",
       "NT1        NT1         155          9          290.27552  37.459748 Set 1      \n",
       "CG1307     CG1307       16          0           29.04102   0.000000 Set 1      \n",
       "Kal1       Kal1         15          0           28.81075   0.000000 Set 1      \n",
       "unc-4      unc-4        12          0           27.36481   0.000000 Set 1      \n",
       "CG9095     CG9095     1349        124         2301.51187 554.365400 Set 1      \n",
       "CG4928     CG4928      168         10          295.99841  41.931575 Set 1      \n",
       "dsx-c73A   dsx-c73A    520         47          926.04380 186.010363 Set 1      \n",
       "           log2.ratio heuristic.score\n",
       "CG14304    4.586867   30.64198       \n",
       "modSP      5.505953   30.48842       \n",
       "Snmp2      5.387265   29.20614       \n",
       "CG5835     5.318710   28.47856       \n",
       "Rab40      5.110815   26.33078       \n",
       "ph-d       5.066526   25.88462       \n",
       "ine        4.080458   25.75632       \n",
       "CG7011     5.046786   25.68705       \n",
       "Syx5       5.008796   25.30907       \n",
       "CG10237    4.499111   24.90469       \n",
       "MICAL-like 4.934024   24.57372       \n",
       "CG11438    4.905557   24.29676       \n",
       "ND-ASHI    4.886150   24.10890       \n",
       "NT1        2.916002   23.87109       \n",
       "CG1307     4.860020   23.85717       \n",
       "Kal1       4.848536   23.74697       \n",
       "unc-4      4.774250   23.04068       \n",
       "CG9095     2.051073   22.90842       \n",
       "CG4928     2.785478   22.88079       \n",
       "dsx-c73A   2.307962   22.74841       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## no strong differences but keep them separate by name\n",
    "two.set.differential.gene.test.3a(cds_sub, \n",
    "                                  df_sub300$cluster == 18 & \n",
    "                                  df_sub300$umap_2 < 1*df_sub300$umap_1 -3.3 , \n",
    "                                 (df_sub300$cluster == 18 & \n",
    "                                  df_sub300$umap_2 >= 1*df_sub300$umap_1 -3.3),\n",
    "                                  sample = 50000\n",
    "                                )  %>% filter(higher.expr == \"Set 1\")  %>% \n",
    "head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1515,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 1: 1508\n",
      "\n",
      "# of cells sampled in set 2: 615\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>set.1.n.umi</th><th scope=col>set.2.n.umi</th><th scope=col>set.1.tpm</th><th scope=col>set.2.tpm</th><th scope=col>higher.expr</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG44422</th><td>CG44422   </td><td>211</td><td> 716</td><td> 313.729354</td><td>3430.53707</td><td>Set 2</td><td>3.446251</td><td>40.47497</td></tr>\n",
       "\t<tr><th scope=row>CG13739</th><td>CG13739   </td><td>  2</td><td>  46</td><td>   3.671678</td><td> 185.01418</td><td>Set 2</td><td>5.307551</td><td>40.01505</td></tr>\n",
       "\t<tr><th scope=row>CG31323</th><td>CG31323   </td><td> 11</td><td>  74</td><td>  14.217173</td><td> 349.12730</td><td>Set 2</td><td>4.519981</td><td>38.20168</td></tr>\n",
       "\t<tr><th scope=row>CG13954</th><td>CG13954   </td><td> 12</td><td>  64</td><td>  14.832382</td><td> 321.34180</td><td>Set 2</td><td>4.343158</td><td>36.18914</td></tr>\n",
       "\t<tr><th scope=row>Ca-alpha1T</th><td>Ca-alpha1T</td><td>106</td><td> 370</td><td> 170.145540</td><td>1710.53159</td><td>Set 2</td><td>3.321149</td><td>35.67270</td></tr>\n",
       "\t<tr><th scope=row>CG33253</th><td>CG33253   </td><td> 16</td><td>  90</td><td>  25.504338</td><td> 394.45680</td><td>Set 2</td><td>3.895567</td><td>33.60852</td></tr>\n",
       "\t<tr><th scope=row>CG30116</th><td>CG30116   </td><td> 40</td><td> 130</td><td>  57.009062</td><td> 640.86113</td><td>Set 2</td><td>3.465662</td><td>32.32117</td></tr>\n",
       "\t<tr><th scope=row>Sur</th><td>Sur       </td><td> 30</td><td> 114</td><td>  44.014088</td><td> 522.95874</td><td>Set 2</td><td>3.538249</td><td>31.96209</td></tr>\n",
       "\t<tr><th scope=row>Ppn</th><td>Ppn       </td><td>167</td><td> 460</td><td> 276.624625</td><td>2051.42136</td><td>Set 2</td><td>2.885416</td><td>31.74856</td></tr>\n",
       "\t<tr><th scope=row>Frq1</th><td>Frq1      </td><td> 37</td><td> 152</td><td>  68.468606</td><td> 697.99442</td><td>Set 2</td><td>3.328782</td><td>31.45412</td></tr>\n",
       "\t<tr><th scope=row>bru3</th><td>bru3      </td><td>805</td><td>1619</td><td>1249.622012</td><td>6889.38536</td><td>Set 2</td><td>2.461729</td><td>31.38796</td></tr>\n",
       "\t<tr><th scope=row>Pdfr</th><td>Pdfr      </td><td>  9</td><td>  42</td><td>  14.771994</td><td> 225.12323</td><td>Set 2</td><td>3.835278</td><td>29.99558</td></tr>\n",
       "\t<tr><th scope=row>CG12541</th><td>CG12541   </td><td>  4</td><td>  33</td><td>   7.617228</td><td> 145.03984</td><td>Set 2</td><td>4.073082</td><td>29.28635</td></tr>\n",
       "\t<tr><th scope=row>disco-r</th><td>disco-r   </td><td>246</td><td> 565</td><td> 407.343230</td><td>2445.01760</td><td>Set 2</td><td>2.581991</td><td>29.06345</td></tr>\n",
       "\t<tr><th scope=row>CG15431</th><td>CG15431   </td><td> 10</td><td>  53</td><td>  18.559028</td><td> 241.98525</td><td>Set 2</td><td>3.629013</td><td>28.75893</td></tr>\n",
       "\t<tr><th scope=row>CG30377</th><td>CG30377   </td><td>  8</td><td>  34</td><td>  10.251726</td><td> 159.16510</td><td>Set 2</td><td>3.822306</td><td>27.99233</td></tr>\n",
       "\t<tr><th scope=row>AdamTS-A</th><td>AdamTS-A  </td><td>169</td><td> 369</td><td> 271.265610</td><td>1632.39961</td><td>Set 2</td><td>2.583908</td><td>27.57976</td></tr>\n",
       "\t<tr><th scope=row>Pde1c</th><td>Pde1c     </td><td>118</td><td> 251</td><td> 183.387509</td><td>1158.21056</td><td>Set 2</td><td>2.651085</td><td>26.98520</td></tr>\n",
       "\t<tr><th scope=row>PGRP-LB</th><td>PGRP-LB   </td><td>  0</td><td>   7</td><td>   0.000000</td><td>  36.05007</td><td>Set 2</td><td>5.171930</td><td>26.95302</td></tr>\n",
       "\t<tr><th scope=row>CG31823</th><td>CG31823   </td><td>  2</td><td>  20</td><td>   4.623305</td><td>  93.51717</td><td>Set 2</td><td>4.055741</td><td>26.61582</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene & set.1.n.umi & set.2.n.umi & set.1.tpm & set.2.tpm & higher.expr & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG44422 & CG44422    & 211 &  716 &  313.729354 & 3430.53707 & Set 2 & 3.446251 & 40.47497\\\\\n",
       "\tCG13739 & CG13739    &   2 &   46 &    3.671678 &  185.01418 & Set 2 & 5.307551 & 40.01505\\\\\n",
       "\tCG31323 & CG31323    &  11 &   74 &   14.217173 &  349.12730 & Set 2 & 4.519981 & 38.20168\\\\\n",
       "\tCG13954 & CG13954    &  12 &   64 &   14.832382 &  321.34180 & Set 2 & 4.343158 & 36.18914\\\\\n",
       "\tCa-alpha1T & Ca-alpha1T & 106 &  370 &  170.145540 & 1710.53159 & Set 2 & 3.321149 & 35.67270\\\\\n",
       "\tCG33253 & CG33253    &  16 &   90 &   25.504338 &  394.45680 & Set 2 & 3.895567 & 33.60852\\\\\n",
       "\tCG30116 & CG30116    &  40 &  130 &   57.009062 &  640.86113 & Set 2 & 3.465662 & 32.32117\\\\\n",
       "\tSur & Sur        &  30 &  114 &   44.014088 &  522.95874 & Set 2 & 3.538249 & 31.96209\\\\\n",
       "\tPpn & Ppn        & 167 &  460 &  276.624625 & 2051.42136 & Set 2 & 2.885416 & 31.74856\\\\\n",
       "\tFrq1 & Frq1       &  37 &  152 &   68.468606 &  697.99442 & Set 2 & 3.328782 & 31.45412\\\\\n",
       "\tbru3 & bru3       & 805 & 1619 & 1249.622012 & 6889.38536 & Set 2 & 2.461729 & 31.38796\\\\\n",
       "\tPdfr & Pdfr       &   9 &   42 &   14.771994 &  225.12323 & Set 2 & 3.835278 & 29.99558\\\\\n",
       "\tCG12541 & CG12541    &   4 &   33 &    7.617228 &  145.03984 & Set 2 & 4.073082 & 29.28635\\\\\n",
       "\tdisco-r & disco-r    & 246 &  565 &  407.343230 & 2445.01760 & Set 2 & 2.581991 & 29.06345\\\\\n",
       "\tCG15431 & CG15431    &  10 &   53 &   18.559028 &  241.98525 & Set 2 & 3.629013 & 28.75893\\\\\n",
       "\tCG30377 & CG30377    &   8 &   34 &   10.251726 &  159.16510 & Set 2 & 3.822306 & 27.99233\\\\\n",
       "\tAdamTS-A & AdamTS-A   & 169 &  369 &  271.265610 & 1632.39961 & Set 2 & 2.583908 & 27.57976\\\\\n",
       "\tPde1c & Pde1c      & 118 &  251 &  183.387509 & 1158.21056 & Set 2 & 2.651085 & 26.98520\\\\\n",
       "\tPGRP-LB & PGRP-LB    &   0 &    7 &    0.000000 &   36.05007 & Set 2 & 5.171930 & 26.95302\\\\\n",
       "\tCG31823 & CG31823    &   2 &   20 &    4.623305 &   93.51717 & Set 2 & 4.055741 & 26.61582\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | set.1.n.umi &lt;dbl&gt; | set.2.n.umi &lt;dbl&gt; | set.1.tpm &lt;dbl&gt; | set.2.tpm &lt;dbl&gt; | higher.expr &lt;chr&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG44422 | CG44422    | 211 |  716 |  313.729354 | 3430.53707 | Set 2 | 3.446251 | 40.47497 |\n",
       "| CG13739 | CG13739    |   2 |   46 |    3.671678 |  185.01418 | Set 2 | 5.307551 | 40.01505 |\n",
       "| CG31323 | CG31323    |  11 |   74 |   14.217173 |  349.12730 | Set 2 | 4.519981 | 38.20168 |\n",
       "| CG13954 | CG13954    |  12 |   64 |   14.832382 |  321.34180 | Set 2 | 4.343158 | 36.18914 |\n",
       "| Ca-alpha1T | Ca-alpha1T | 106 |  370 |  170.145540 | 1710.53159 | Set 2 | 3.321149 | 35.67270 |\n",
       "| CG33253 | CG33253    |  16 |   90 |   25.504338 |  394.45680 | Set 2 | 3.895567 | 33.60852 |\n",
       "| CG30116 | CG30116    |  40 |  130 |   57.009062 |  640.86113 | Set 2 | 3.465662 | 32.32117 |\n",
       "| Sur | Sur        |  30 |  114 |   44.014088 |  522.95874 | Set 2 | 3.538249 | 31.96209 |\n",
       "| Ppn | Ppn        | 167 |  460 |  276.624625 | 2051.42136 | Set 2 | 2.885416 | 31.74856 |\n",
       "| Frq1 | Frq1       |  37 |  152 |   68.468606 |  697.99442 | Set 2 | 3.328782 | 31.45412 |\n",
       "| bru3 | bru3       | 805 | 1619 | 1249.622012 | 6889.38536 | Set 2 | 2.461729 | 31.38796 |\n",
       "| Pdfr | Pdfr       |   9 |   42 |   14.771994 |  225.12323 | Set 2 | 3.835278 | 29.99558 |\n",
       "| CG12541 | CG12541    |   4 |   33 |    7.617228 |  145.03984 | Set 2 | 4.073082 | 29.28635 |\n",
       "| disco-r | disco-r    | 246 |  565 |  407.343230 | 2445.01760 | Set 2 | 2.581991 | 29.06345 |\n",
       "| CG15431 | CG15431    |  10 |   53 |   18.559028 |  241.98525 | Set 2 | 3.629013 | 28.75893 |\n",
       "| CG30377 | CG30377    |   8 |   34 |   10.251726 |  159.16510 | Set 2 | 3.822306 | 27.99233 |\n",
       "| AdamTS-A | AdamTS-A   | 169 |  369 |  271.265610 | 1632.39961 | Set 2 | 2.583908 | 27.57976 |\n",
       "| Pde1c | Pde1c      | 118 |  251 |  183.387509 | 1158.21056 | Set 2 | 2.651085 | 26.98520 |\n",
       "| PGRP-LB | PGRP-LB    |   0 |    7 |    0.000000 |   36.05007 | Set 2 | 5.171930 | 26.95302 |\n",
       "| CG31823 | CG31823    |   2 |   20 |    4.623305 |   93.51717 | Set 2 | 4.055741 | 26.61582 |\n",
       "\n"
      ],
      "text/plain": [
       "           gene       set.1.n.umi set.2.n.umi set.1.tpm   set.2.tpm \n",
       "CG44422    CG44422    211          716         313.729354 3430.53707\n",
       "CG13739    CG13739      2           46           3.671678  185.01418\n",
       "CG31323    CG31323     11           74          14.217173  349.12730\n",
       "CG13954    CG13954     12           64          14.832382  321.34180\n",
       "Ca-alpha1T Ca-alpha1T 106          370         170.145540 1710.53159\n",
       "CG33253    CG33253     16           90          25.504338  394.45680\n",
       "CG30116    CG30116     40          130          57.009062  640.86113\n",
       "Sur        Sur         30          114          44.014088  522.95874\n",
       "Ppn        Ppn        167          460         276.624625 2051.42136\n",
       "Frq1       Frq1        37          152          68.468606  697.99442\n",
       "bru3       bru3       805         1619        1249.622012 6889.38536\n",
       "Pdfr       Pdfr         9           42          14.771994  225.12323\n",
       "CG12541    CG12541      4           33           7.617228  145.03984\n",
       "disco-r    disco-r    246          565         407.343230 2445.01760\n",
       "CG15431    CG15431     10           53          18.559028  241.98525\n",
       "CG30377    CG30377      8           34          10.251726  159.16510\n",
       "AdamTS-A   AdamTS-A   169          369         271.265610 1632.39961\n",
       "Pde1c      Pde1c      118          251         183.387509 1158.21056\n",
       "PGRP-LB    PGRP-LB      0            7           0.000000   36.05007\n",
       "CG31823    CG31823      2           20           4.623305   93.51717\n",
       "           higher.expr log2.ratio heuristic.score\n",
       "CG44422    Set 2       3.446251   40.47497       \n",
       "CG13739    Set 2       5.307551   40.01505       \n",
       "CG31323    Set 2       4.519981   38.20168       \n",
       "CG13954    Set 2       4.343158   36.18914       \n",
       "Ca-alpha1T Set 2       3.321149   35.67270       \n",
       "CG33253    Set 2       3.895567   33.60852       \n",
       "CG30116    Set 2       3.465662   32.32117       \n",
       "Sur        Set 2       3.538249   31.96209       \n",
       "Ppn        Set 2       2.885416   31.74856       \n",
       "Frq1       Set 2       3.328782   31.45412       \n",
       "bru3       Set 2       2.461729   31.38796       \n",
       "Pdfr       Set 2       3.835278   29.99558       \n",
       "CG12541    Set 2       4.073082   29.28635       \n",
       "disco-r    Set 2       2.581991   29.06345       \n",
       "CG15431    Set 2       3.629013   28.75893       \n",
       "CG30377    Set 2       3.822306   27.99233       \n",
       "AdamTS-A   Set 2       2.583908   27.57976       \n",
       "Pde1c      Set 2       2.651085   26.98520       \n",
       "PGRP-LB    Set 2       5.171930   26.95302       \n",
       "CG31823    Set 2       4.055741   26.61582       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "two.set.differential.gene.test.3a(cds_sub, \n",
    "                                  df_sub300$cluster == 18 & \n",
    "                                  df_sub300$umap_2 < 1*df_sub300$umap_1 -3.3 , \n",
    "                                 (df_sub300$cluster == 18 & \n",
    "                                  df_sub300$umap_2 >= 1*df_sub300$umap_1 -3.3),\n",
    "                                  sample = 50000\n",
    "                                )  %>% filter(higher.expr == \"Set 2\")  %>% \n",
    "head(20)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### ca cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type ca\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Primordial_cells_1\", df_sub300$cluster %in% ca)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 316,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ca' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ca' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ca)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 317,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'ca' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'ca' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(ca)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.4, 0.05), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1801,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Brd56</th><td>ca</td><td>Brd             </td><td> 471</td><td>  942</td><td> 284.47076</td><td>  33.60678</td><td>3.039154</td><td>24.79098</td></tr>\n",
       "\t<tr><th scope=row>E(spl)m4-BFM55</th><td>ca</td><td>E(spl)m4-BFM    </td><td> 278</td><td>  502</td><td> 170.17356</td><td>  18.09074</td><td>3.156062</td><td>23.41582</td></tr>\n",
       "\t<tr><th scope=row>Tom58</th><td>ca</td><td>Tom             </td><td>1237</td><td> 4333</td><td> 746.04440</td><td> 156.40094</td><td>2.244817</td><td>21.42689</td></tr>\n",
       "\t<tr><th scope=row>link55</th><td>ca</td><td>link            </td><td>1337</td><td> 5089</td><td> 846.65940</td><td> 202.42380</td><td>2.057293</td><td>20.01199</td></tr>\n",
       "\t<tr><th scope=row>Ama55</th><td>ca</td><td>Ama             </td><td>2260</td><td> 9768</td><td>1408.93770</td><td> 393.80348</td><td>1.835401</td><td>19.20090</td></tr>\n",
       "\t<tr><th scope=row>Hsp70Bc55</th><td>ca</td><td>Hsp70Bc         </td><td> 275</td><td>  803</td><td> 172.64602</td><td>  28.69892</td><td>2.539335</td><td>18.89266</td></tr>\n",
       "\t<tr><th scope=row>Obp99a55</th><td>ca</td><td>Obp99a          </td><td>1591</td><td> 6636</td><td> 963.71498</td><td> 260.62622</td><td>1.881099</td><td>18.64914</td></tr>\n",
       "\t<tr><th scope=row>lmd55</th><td>ca</td><td>lmd             </td><td> 826</td><td> 3074</td><td> 537.61124</td><td> 134.07745</td><td>1.992776</td><td>18.08066</td></tr>\n",
       "\t<tr><th scope=row>Hsp6855</th><td>ca</td><td>Hsp68           </td><td> 234</td><td>  856</td><td> 149.17504</td><td>  29.72471</td><td>2.279535</td><td>16.48218</td></tr>\n",
       "\t<tr><th scope=row>E(spl)mgamma-HLH55</th><td>ca</td><td>E(spl)mgamma-HLH</td><td> 289</td><td> 1001</td><td> 168.32943</td><td>  36.76289</td><td>2.156247</td><td>15.96418</td></tr>\n",
       "\t<tr><th scope=row>CG445555</th><td>ca</td><td>CG4455          </td><td> 782</td><td> 3544</td><td> 501.86728</td><td> 147.86066</td><td>1.753343</td><td>15.73456</td></tr>\n",
       "\t<tr><th scope=row>E(spl)m8-HLH55</th><td>ca</td><td>E(spl)m8-HLH    </td><td> 122</td><td>  360</td><td>  78.69835</td><td>  13.70919</td><td>2.419616</td><td>15.28345</td></tr>\n",
       "\t<tr><th scope=row>bib55</th><td>ca</td><td>bib             </td><td> 882</td><td> 3835</td><td> 519.22160</td><td> 159.96952</td><td>1.689563</td><td>15.24490</td></tr>\n",
       "\t<tr><th scope=row>E(spl)m5-HLH55</th><td>ca</td><td>E(spl)m5-HLH    </td><td> 142</td><td>  390</td><td>  83.89865</td><td>  15.22031</td><td>2.370846</td><td>15.19160</td></tr>\n",
       "\t<tr><th scope=row>E(spl)m7-HLH55</th><td>ca</td><td>E(spl)m7-HLH    </td><td> 309</td><td> 1263</td><td> 194.43926</td><td>  48.59433</td><td>1.971072</td><td>15.00100</td></tr>\n",
       "\t<tr><th scope=row>E(spl)malpha-BFM55</th><td>ca</td><td>E(spl)malpha-BFM</td><td> 215</td><td>  749</td><td> 121.91336</td><td>  28.18740</td><td>2.062439</td><td>14.31641</td></tr>\n",
       "\t<tr><th scope=row>CG1342755</th><td>ca</td><td>CG13427         </td><td> 178</td><td>  537</td><td> 103.28021</td><td>  23.71819</td><td>2.062919</td><td>13.83047</td></tr>\n",
       "\t<tr><th scope=row>N55</th><td>ca</td><td>N               </td><td>5681</td><td>32482</td><td>3120.06093</td><td>1375.31107</td><td>1.180768</td><td>13.70614</td></tr>\n",
       "\t<tr><th scope=row>E(spl)mdelta-HLH55</th><td>ca</td><td>E(spl)mdelta-HLH</td><td>  99</td><td>  341</td><td>  62.92622</td><td>  12.23357</td><td>2.249459</td><td>13.49301</td></tr>\n",
       "\t<tr><th scope=row>ihog55</th><td>ca</td><td>ihog            </td><td> 341</td><td> 1676</td><td> 227.55863</td><td>  68.53673</td><td>1.710391</td><td>13.40334</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tBrd56 & ca & Brd              &  471 &   942 &  284.47076 &   33.60678 & 3.039154 & 24.79098\\\\\n",
       "\tE(spl)m4-BFM55 & ca & E(spl)m4-BFM     &  278 &   502 &  170.17356 &   18.09074 & 3.156062 & 23.41582\\\\\n",
       "\tTom58 & ca & Tom              & 1237 &  4333 &  746.04440 &  156.40094 & 2.244817 & 21.42689\\\\\n",
       "\tlink55 & ca & link             & 1337 &  5089 &  846.65940 &  202.42380 & 2.057293 & 20.01199\\\\\n",
       "\tAma55 & ca & Ama              & 2260 &  9768 & 1408.93770 &  393.80348 & 1.835401 & 19.20090\\\\\n",
       "\tHsp70Bc55 & ca & Hsp70Bc          &  275 &   803 &  172.64602 &   28.69892 & 2.539335 & 18.89266\\\\\n",
       "\tObp99a55 & ca & Obp99a           & 1591 &  6636 &  963.71498 &  260.62622 & 1.881099 & 18.64914\\\\\n",
       "\tlmd55 & ca & lmd              &  826 &  3074 &  537.61124 &  134.07745 & 1.992776 & 18.08066\\\\\n",
       "\tHsp6855 & ca & Hsp68            &  234 &   856 &  149.17504 &   29.72471 & 2.279535 & 16.48218\\\\\n",
       "\tE(spl)mgamma-HLH55 & ca & E(spl)mgamma-HLH &  289 &  1001 &  168.32943 &   36.76289 & 2.156247 & 15.96418\\\\\n",
       "\tCG445555 & ca & CG4455           &  782 &  3544 &  501.86728 &  147.86066 & 1.753343 & 15.73456\\\\\n",
       "\tE(spl)m8-HLH55 & ca & E(spl)m8-HLH     &  122 &   360 &   78.69835 &   13.70919 & 2.419616 & 15.28345\\\\\n",
       "\tbib55 & ca & bib              &  882 &  3835 &  519.22160 &  159.96952 & 1.689563 & 15.24490\\\\\n",
       "\tE(spl)m5-HLH55 & ca & E(spl)m5-HLH     &  142 &   390 &   83.89865 &   15.22031 & 2.370846 & 15.19160\\\\\n",
       "\tE(spl)m7-HLH55 & ca & E(spl)m7-HLH     &  309 &  1263 &  194.43926 &   48.59433 & 1.971072 & 15.00100\\\\\n",
       "\tE(spl)malpha-BFM55 & ca & E(spl)malpha-BFM &  215 &   749 &  121.91336 &   28.18740 & 2.062439 & 14.31641\\\\\n",
       "\tCG1342755 & ca & CG13427          &  178 &   537 &  103.28021 &   23.71819 & 2.062919 & 13.83047\\\\\n",
       "\tN55 & ca & N                & 5681 & 32482 & 3120.06093 & 1375.31107 & 1.180768 & 13.70614\\\\\n",
       "\tE(spl)mdelta-HLH55 & ca & E(spl)mdelta-HLH &   99 &   341 &   62.92622 &   12.23357 & 2.249459 & 13.49301\\\\\n",
       "\tihog55 & ca & ihog             &  341 &  1676 &  227.55863 &   68.53673 & 1.710391 & 13.40334\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Brd56 | ca | Brd              |  471 |   942 |  284.47076 |   33.60678 | 3.039154 | 24.79098 |\n",
       "| E(spl)m4-BFM55 | ca | E(spl)m4-BFM     |  278 |   502 |  170.17356 |   18.09074 | 3.156062 | 23.41582 |\n",
       "| Tom58 | ca | Tom              | 1237 |  4333 |  746.04440 |  156.40094 | 2.244817 | 21.42689 |\n",
       "| link55 | ca | link             | 1337 |  5089 |  846.65940 |  202.42380 | 2.057293 | 20.01199 |\n",
       "| Ama55 | ca | Ama              | 2260 |  9768 | 1408.93770 |  393.80348 | 1.835401 | 19.20090 |\n",
       "| Hsp70Bc55 | ca | Hsp70Bc          |  275 |   803 |  172.64602 |   28.69892 | 2.539335 | 18.89266 |\n",
       "| Obp99a55 | ca | Obp99a           | 1591 |  6636 |  963.71498 |  260.62622 | 1.881099 | 18.64914 |\n",
       "| lmd55 | ca | lmd              |  826 |  3074 |  537.61124 |  134.07745 | 1.992776 | 18.08066 |\n",
       "| Hsp6855 | ca | Hsp68            |  234 |   856 |  149.17504 |   29.72471 | 2.279535 | 16.48218 |\n",
       "| E(spl)mgamma-HLH55 | ca | E(spl)mgamma-HLH |  289 |  1001 |  168.32943 |   36.76289 | 2.156247 | 15.96418 |\n",
       "| CG445555 | ca | CG4455           |  782 |  3544 |  501.86728 |  147.86066 | 1.753343 | 15.73456 |\n",
       "| E(spl)m8-HLH55 | ca | E(spl)m8-HLH     |  122 |   360 |   78.69835 |   13.70919 | 2.419616 | 15.28345 |\n",
       "| bib55 | ca | bib              |  882 |  3835 |  519.22160 |  159.96952 | 1.689563 | 15.24490 |\n",
       "| E(spl)m5-HLH55 | ca | E(spl)m5-HLH     |  142 |   390 |   83.89865 |   15.22031 | 2.370846 | 15.19160 |\n",
       "| E(spl)m7-HLH55 | ca | E(spl)m7-HLH     |  309 |  1263 |  194.43926 |   48.59433 | 1.971072 | 15.00100 |\n",
       "| E(spl)malpha-BFM55 | ca | E(spl)malpha-BFM |  215 |   749 |  121.91336 |   28.18740 | 2.062439 | 14.31641 |\n",
       "| CG1342755 | ca | CG13427          |  178 |   537 |  103.28021 |   23.71819 | 2.062919 | 13.83047 |\n",
       "| N55 | ca | N                | 5681 | 32482 | 3120.06093 | 1375.31107 | 1.180768 | 13.70614 |\n",
       "| E(spl)mdelta-HLH55 | ca | E(spl)mdelta-HLH |   99 |   341 |   62.92622 |   12.23357 | 2.249459 | 13.49301 |\n",
       "| ihog55 | ca | ihog             |  341 |  1676 |  227.55863 |   68.53673 | 1.710391 | 13.40334 |\n",
       "\n"
      ],
      "text/plain": [
       "                   cell.type.2 gene             cell.type.2.n.umi other.n.umi\n",
       "Brd56              ca          Brd               471                942      \n",
       "E(spl)m4-BFM55     ca          E(spl)m4-BFM      278                502      \n",
       "Tom58              ca          Tom              1237               4333      \n",
       "link55             ca          link             1337               5089      \n",
       "Ama55              ca          Ama              2260               9768      \n",
       "Hsp70Bc55          ca          Hsp70Bc           275                803      \n",
       "Obp99a55           ca          Obp99a           1591               6636      \n",
       "lmd55              ca          lmd               826               3074      \n",
       "Hsp6855            ca          Hsp68             234                856      \n",
       "E(spl)mgamma-HLH55 ca          E(spl)mgamma-HLH  289               1001      \n",
       "CG445555           ca          CG4455            782               3544      \n",
       "E(spl)m8-HLH55     ca          E(spl)m8-HLH      122                360      \n",
       "bib55              ca          bib               882               3835      \n",
       "E(spl)m5-HLH55     ca          E(spl)m5-HLH      142                390      \n",
       "E(spl)m7-HLH55     ca          E(spl)m7-HLH      309               1263      \n",
       "E(spl)malpha-BFM55 ca          E(spl)malpha-BFM  215                749      \n",
       "CG1342755          ca          CG13427           178                537      \n",
       "N55                ca          N                5681              32482      \n",
       "E(spl)mdelta-HLH55 ca          E(spl)mdelta-HLH   99                341      \n",
       "ihog55             ca          ihog              341               1676      \n",
       "                   cell.type.2.tpm other.tpm  log2.ratio heuristic.score\n",
       "Brd56               284.47076        33.60678 3.039154   24.79098       \n",
       "E(spl)m4-BFM55      170.17356        18.09074 3.156062   23.41582       \n",
       "Tom58               746.04440       156.40094 2.244817   21.42689       \n",
       "link55              846.65940       202.42380 2.057293   20.01199       \n",
       "Ama55              1408.93770       393.80348 1.835401   19.20090       \n",
       "Hsp70Bc55           172.64602        28.69892 2.539335   18.89266       \n",
       "Obp99a55            963.71498       260.62622 1.881099   18.64914       \n",
       "lmd55               537.61124       134.07745 1.992776   18.08066       \n",
       "Hsp6855             149.17504        29.72471 2.279535   16.48218       \n",
       "E(spl)mgamma-HLH55  168.32943        36.76289 2.156247   15.96418       \n",
       "CG445555            501.86728       147.86066 1.753343   15.73456       \n",
       "E(spl)m8-HLH55       78.69835        13.70919 2.419616   15.28345       \n",
       "bib55               519.22160       159.96952 1.689563   15.24490       \n",
       "E(spl)m5-HLH55       83.89865        15.22031 2.370846   15.19160       \n",
       "E(spl)m7-HLH55      194.43926        48.59433 1.971072   15.00100       \n",
       "E(spl)malpha-BFM55  121.91336        28.18740 2.062439   14.31641       \n",
       "CG1342755           103.28021        23.71819 2.062919   13.83047       \n",
       "N55                3120.06093      1375.31107 1.180768   13.70614       \n",
       "E(spl)mdelta-HLH55   62.92622        12.23357 2.249459   13.49301       \n",
       "ihog55              227.55863        68.53673 1.710391   13.40334       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"ca\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 318,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Brd\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Brd\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Brd\"), \n",
    "                   size_pos = 2, size_neg = 0.5, stroke = 0.008, limit = 1)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"E(spl)m4-BFM\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 1)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Tom\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1811,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type Primordial_cells_2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"Primordial_cells_2\", df_sub300$cluster %in% cb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 319,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'cb' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'cb' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(cb)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 320,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'cb' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'cb' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(cb)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.4, 0.2), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1807,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CG1003556</th><td>cb</td><td>CG10035  </td><td>1345</td><td> 1607</td><td> 857.47043</td><td>  58.905895</td><td>3.839317</td><td>37.41654</td></tr>\n",
       "\t<tr><th scope=row>sca56</th><td>cb</td><td>sca      </td><td>3098</td><td> 4564</td><td>1973.03039</td><td> 186.937089</td><td>3.392091</td><td>37.13298</td></tr>\n",
       "\t<tr><th scope=row>Abd-B56</th><td>cb</td><td>Abd-B    </td><td>2988</td><td> 7747</td><td>1982.32564</td><td> 326.630866</td><td>2.597051</td><td>28.44733</td></tr>\n",
       "\t<tr><th scope=row>5-HT2A56</th><td>cb</td><td>5-HT2A   </td><td> 279</td><td>  269</td><td> 173.65133</td><td>  11.743355</td><td>3.768376</td><td>28.06813</td></tr>\n",
       "\t<tr><th scope=row>Sr-CII56</th><td>cb</td><td>Sr-CII   </td><td> 393</td><td>  617</td><td> 261.48554</td><td>  23.921433</td><td>3.391272</td><td>27.25258</td></tr>\n",
       "\t<tr><th scope=row>CG788456</th><td>cb</td><td>CG7884   </td><td> 674</td><td> 1221</td><td> 441.15350</td><td>  51.786011</td><td>3.063053</td><td>26.91935</td></tr>\n",
       "\t<tr><th scope=row>cas56</th><td>cb</td><td>cas      </td><td>2137</td><td> 5455</td><td>1410.00316</td><td> 241.624947</td><td>2.538899</td><td>26.56324</td></tr>\n",
       "\t<tr><th scope=row>link56</th><td>cb</td><td>link     </td><td>1812</td><td> 5182</td><td>1199.41711</td><td> 200.722742</td><td>2.571888</td><td>26.30866</td></tr>\n",
       "\t<tr><th scope=row>bib56</th><td>cb</td><td>bib      </td><td>1347</td><td> 3782</td><td> 873.35872</td><td> 151.604983</td><td>2.516772</td><td>24.59410</td></tr>\n",
       "\t<tr><th scope=row>jigr156</th><td>cb</td><td>jigr1    </td><td>1603</td><td> 4612</td><td> 987.55313</td><td> 189.839017</td><td>2.371502</td><td>23.59449</td></tr>\n",
       "\t<tr><th scope=row>opa56</th><td>cb</td><td>opa      </td><td> 652</td><td> 1722</td><td> 410.22710</td><td>  65.998498</td><td>2.614222</td><td>22.70136</td></tr>\n",
       "\t<tr><th scope=row>DIP-alpha56</th><td>cb</td><td>DIP-alpha</td><td> 810</td><td> 2058</td><td> 507.74042</td><td>  89.322876</td><td>2.490928</td><td>22.39540</td></tr>\n",
       "\t<tr><th scope=row>wntD56</th><td>cb</td><td>wntD     </td><td> 115</td><td>  158</td><td>  81.29847</td><td>   6.372233</td><td>3.463055</td><td>22.03470</td></tr>\n",
       "\t<tr><th scope=row>Notum56</th><td>cb</td><td>Notum    </td><td>1284</td><td> 3930</td><td> 806.99471</td><td> 166.195405</td><td>2.271024</td><td>21.93401</td></tr>\n",
       "\t<tr><th scope=row>dan56</th><td>cb</td><td>dan      </td><td> 327</td><td>  749</td><td> 204.09145</td><td>  27.964900</td><td>2.816838</td><td>21.63366</td></tr>\n",
       "\t<tr><th scope=row>upd156</th><td>cb</td><td>upd1     </td><td> 306</td><td>  710</td><td> 203.05107</td><td>  28.322989</td><td>2.791739</td><td>21.42041</td></tr>\n",
       "\t<tr><th scope=row>CG1562856</th><td>cb</td><td>CG15628  </td><td> 497</td><td> 1450</td><td> 313.24764</td><td>  51.659306</td><td>2.572543</td><td>21.34120</td></tr>\n",
       "\t<tr><th scope=row>neur56</th><td>cb</td><td>neur     </td><td>2339</td><td> 8665</td><td>1470.08504</td><td> 361.437383</td><td>2.020096</td><td>21.25679</td></tr>\n",
       "\t<tr><th scope=row>BobA56</th><td>cb</td><td>BobA     </td><td> 127</td><td>  199</td><td>  85.68271</td><td>   8.357711</td><td>3.194777</td><td>20.56692</td></tr>\n",
       "\t<tr><th scope=row>Tet56</th><td>cb</td><td>Tet      </td><td>9377</td><td>47226</td><td>5802.72068</td><td>1880.645994</td><td>1.624734</td><td>20.31367</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tCG1003556 & cb & CG10035   & 1345 &  1607 &  857.47043 &   58.905895 & 3.839317 & 37.41654\\\\\n",
       "\tsca56 & cb & sca       & 3098 &  4564 & 1973.03039 &  186.937089 & 3.392091 & 37.13298\\\\\n",
       "\tAbd-B56 & cb & Abd-B     & 2988 &  7747 & 1982.32564 &  326.630866 & 2.597051 & 28.44733\\\\\n",
       "\t5-HT2A56 & cb & 5-HT2A    &  279 &   269 &  173.65133 &   11.743355 & 3.768376 & 28.06813\\\\\n",
       "\tSr-CII56 & cb & Sr-CII    &  393 &   617 &  261.48554 &   23.921433 & 3.391272 & 27.25258\\\\\n",
       "\tCG788456 & cb & CG7884    &  674 &  1221 &  441.15350 &   51.786011 & 3.063053 & 26.91935\\\\\n",
       "\tcas56 & cb & cas       & 2137 &  5455 & 1410.00316 &  241.624947 & 2.538899 & 26.56324\\\\\n",
       "\tlink56 & cb & link      & 1812 &  5182 & 1199.41711 &  200.722742 & 2.571888 & 26.30866\\\\\n",
       "\tbib56 & cb & bib       & 1347 &  3782 &  873.35872 &  151.604983 & 2.516772 & 24.59410\\\\\n",
       "\tjigr156 & cb & jigr1     & 1603 &  4612 &  987.55313 &  189.839017 & 2.371502 & 23.59449\\\\\n",
       "\topa56 & cb & opa       &  652 &  1722 &  410.22710 &   65.998498 & 2.614222 & 22.70136\\\\\n",
       "\tDIP-alpha56 & cb & DIP-alpha &  810 &  2058 &  507.74042 &   89.322876 & 2.490928 & 22.39540\\\\\n",
       "\twntD56 & cb & wntD      &  115 &   158 &   81.29847 &    6.372233 & 3.463055 & 22.03470\\\\\n",
       "\tNotum56 & cb & Notum     & 1284 &  3930 &  806.99471 &  166.195405 & 2.271024 & 21.93401\\\\\n",
       "\tdan56 & cb & dan       &  327 &   749 &  204.09145 &   27.964900 & 2.816838 & 21.63366\\\\\n",
       "\tupd156 & cb & upd1      &  306 &   710 &  203.05107 &   28.322989 & 2.791739 & 21.42041\\\\\n",
       "\tCG1562856 & cb & CG15628   &  497 &  1450 &  313.24764 &   51.659306 & 2.572543 & 21.34120\\\\\n",
       "\tneur56 & cb & neur      & 2339 &  8665 & 1470.08504 &  361.437383 & 2.020096 & 21.25679\\\\\n",
       "\tBobA56 & cb & BobA      &  127 &   199 &   85.68271 &    8.357711 & 3.194777 & 20.56692\\\\\n",
       "\tTet56 & cb & Tet       & 9377 & 47226 & 5802.72068 & 1880.645994 & 1.624734 & 20.31367\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| CG1003556 | cb | CG10035   | 1345 |  1607 |  857.47043 |   58.905895 | 3.839317 | 37.41654 |\n",
       "| sca56 | cb | sca       | 3098 |  4564 | 1973.03039 |  186.937089 | 3.392091 | 37.13298 |\n",
       "| Abd-B56 | cb | Abd-B     | 2988 |  7747 | 1982.32564 |  326.630866 | 2.597051 | 28.44733 |\n",
       "| 5-HT2A56 | cb | 5-HT2A    |  279 |   269 |  173.65133 |   11.743355 | 3.768376 | 28.06813 |\n",
       "| Sr-CII56 | cb | Sr-CII    |  393 |   617 |  261.48554 |   23.921433 | 3.391272 | 27.25258 |\n",
       "| CG788456 | cb | CG7884    |  674 |  1221 |  441.15350 |   51.786011 | 3.063053 | 26.91935 |\n",
       "| cas56 | cb | cas       | 2137 |  5455 | 1410.00316 |  241.624947 | 2.538899 | 26.56324 |\n",
       "| link56 | cb | link      | 1812 |  5182 | 1199.41711 |  200.722742 | 2.571888 | 26.30866 |\n",
       "| bib56 | cb | bib       | 1347 |  3782 |  873.35872 |  151.604983 | 2.516772 | 24.59410 |\n",
       "| jigr156 | cb | jigr1     | 1603 |  4612 |  987.55313 |  189.839017 | 2.371502 | 23.59449 |\n",
       "| opa56 | cb | opa       |  652 |  1722 |  410.22710 |   65.998498 | 2.614222 | 22.70136 |\n",
       "| DIP-alpha56 | cb | DIP-alpha |  810 |  2058 |  507.74042 |   89.322876 | 2.490928 | 22.39540 |\n",
       "| wntD56 | cb | wntD      |  115 |   158 |   81.29847 |    6.372233 | 3.463055 | 22.03470 |\n",
       "| Notum56 | cb | Notum     | 1284 |  3930 |  806.99471 |  166.195405 | 2.271024 | 21.93401 |\n",
       "| dan56 | cb | dan       |  327 |   749 |  204.09145 |   27.964900 | 2.816838 | 21.63366 |\n",
       "| upd156 | cb | upd1      |  306 |   710 |  203.05107 |   28.322989 | 2.791739 | 21.42041 |\n",
       "| CG1562856 | cb | CG15628   |  497 |  1450 |  313.24764 |   51.659306 | 2.572543 | 21.34120 |\n",
       "| neur56 | cb | neur      | 2339 |  8665 | 1470.08504 |  361.437383 | 2.020096 | 21.25679 |\n",
       "| BobA56 | cb | BobA      |  127 |   199 |   85.68271 |    8.357711 | 3.194777 | 20.56692 |\n",
       "| Tet56 | cb | Tet       | 9377 | 47226 | 5802.72068 | 1880.645994 | 1.624734 | 20.31367 |\n",
       "\n"
      ],
      "text/plain": [
       "            cell.type.2 gene      cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "CG1003556   cb          CG10035   1345               1607        857.47043     \n",
       "sca56       cb          sca       3098               4564       1973.03039     \n",
       "Abd-B56     cb          Abd-B     2988               7747       1982.32564     \n",
       "5-HT2A56    cb          5-HT2A     279                269        173.65133     \n",
       "Sr-CII56    cb          Sr-CII     393                617        261.48554     \n",
       "CG788456    cb          CG7884     674               1221        441.15350     \n",
       "cas56       cb          cas       2137               5455       1410.00316     \n",
       "link56      cb          link      1812               5182       1199.41711     \n",
       "bib56       cb          bib       1347               3782        873.35872     \n",
       "jigr156     cb          jigr1     1603               4612        987.55313     \n",
       "opa56       cb          opa        652               1722        410.22710     \n",
       "DIP-alpha56 cb          DIP-alpha  810               2058        507.74042     \n",
       "wntD56      cb          wntD       115                158         81.29847     \n",
       "Notum56     cb          Notum     1284               3930        806.99471     \n",
       "dan56       cb          dan        327                749        204.09145     \n",
       "upd156      cb          upd1       306                710        203.05107     \n",
       "CG1562856   cb          CG15628    497               1450        313.24764     \n",
       "neur56      cb          neur      2339               8665       1470.08504     \n",
       "BobA56      cb          BobA       127                199         85.68271     \n",
       "Tet56       cb          Tet       9377              47226       5802.72068     \n",
       "            other.tpm   log2.ratio heuristic.score\n",
       "CG1003556     58.905895 3.839317   37.41654       \n",
       "sca56        186.937089 3.392091   37.13298       \n",
       "Abd-B56      326.630866 2.597051   28.44733       \n",
       "5-HT2A56      11.743355 3.768376   28.06813       \n",
       "Sr-CII56      23.921433 3.391272   27.25258       \n",
       "CG788456      51.786011 3.063053   26.91935       \n",
       "cas56        241.624947 2.538899   26.56324       \n",
       "link56       200.722742 2.571888   26.30866       \n",
       "bib56        151.604983 2.516772   24.59410       \n",
       "jigr156      189.839017 2.371502   23.59449       \n",
       "opa56         65.998498 2.614222   22.70136       \n",
       "DIP-alpha56   89.322876 2.490928   22.39540       \n",
       "wntD56         6.372233 3.463055   22.03470       \n",
       "Notum56      166.195405 2.271024   21.93401       \n",
       "dan56         27.964900 2.816838   21.63366       \n",
       "upd156        28.322989 2.791739   21.42041       \n",
       "CG1562856     51.659306 2.572543   21.34120       \n",
       "neur56       361.437383 2.020096   21.25679       \n",
       "BobA56         8.357711 3.194777   20.56692       \n",
       "Tet56       1880.645994 1.624734   20.31367       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"cb\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 321,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG10035\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG10035\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"CG10035\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.008, limit = 1)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"sca\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 4)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Abd-B\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type cc\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"cc\", df_sub300$cluster %in% cc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 322,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'cc' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'cc' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(cc)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 323,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'cc' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'cc' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "clus = c(cc)\n",
    "peedeef = ggplot(df_sub300 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_sub300$cluster %in% clus), size =  ifelse(df_sub300$cluster %in% clus, 0.2, 0.02), stroke = 0.005) +\n",
    "scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1815,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>ds57</th><td>cc</td><td>ds   </td><td> 9963</td><td>14410</td><td>3206.9115</td><td> 619.96869</td><td>2.368592</td><td>27.58798</td></tr>\n",
       "\t<tr><th scope=row>otp57</th><td>cc</td><td>otp  </td><td> 3116</td><td> 4680</td><td>1251.3588</td><td> 218.77044</td><td>2.509426</td><td>25.82308</td></tr>\n",
       "\t<tr><th scope=row>Pvf357</th><td>cc</td><td>Pvf3 </td><td> 6120</td><td>11847</td><td>2010.4898</td><td> 509.92641</td><td>1.976360</td><td>21.68867</td></tr>\n",
       "\t<tr><th scope=row>robo257</th><td>cc</td><td>robo2</td><td> 6690</td><td>11692</td><td>2010.1194</td><td> 525.03955</td><td>1.934038</td><td>21.22371</td></tr>\n",
       "\t<tr><th scope=row>Notum57</th><td>cc</td><td>Notum</td><td> 2219</td><td> 3892</td><td> 749.4497</td><td> 166.00330</td><td>2.165955</td><td>20.68836</td></tr>\n",
       "\t<tr><th scope=row>Poxm57</th><td>cc</td><td>Poxm </td><td>  580</td><td>  793</td><td> 206.3057</td><td>  32.51788</td><td>2.621780</td><td>20.17621</td></tr>\n",
       "\t<tr><th scope=row>nkd57</th><td>cc</td><td>nkd  </td><td>16948</td><td>38596</td><td>5280.4399</td><td>1706.77591</td><td>1.628539</td><td>20.13968</td></tr>\n",
       "\t<tr><th scope=row>byn57</th><td>cc</td><td>byn  </td><td>  373</td><td>  560</td><td> 148.2083</td><td>  21.97528</td><td>2.689472</td><td>19.42117</td></tr>\n",
       "\t<tr><th scope=row>dally57</th><td>cc</td><td>dally</td><td> 6530</td><td>13999</td><td>2080.4428</td><td> 629.45349</td><td>1.722429</td><td>18.98697</td></tr>\n",
       "\t<tr><th scope=row>ed57</th><td>cc</td><td>ed   </td><td>25479</td><td>68004</td><td>8207.6110</td><td>2987.98922</td><td>1.457305</td><td>18.94922</td></tr>\n",
       "\t<tr><th scope=row>mgl57</th><td>cc</td><td>mgl  </td><td>18913</td><td>49172</td><td>6132.5250</td><td>2178.96099</td><td>1.492179</td><td>18.77534</td></tr>\n",
       "\t<tr><th scope=row>N57</th><td>cc</td><td>N    </td><td>11778</td><td>30829</td><td>3823.4703</td><td>1281.73298</td><td>1.575662</td><td>18.75202</td></tr>\n",
       "\t<tr><th scope=row>heph57</th><td>cc</td><td>heph </td><td>13056</td><td>34869</td><td>4289.5803</td><td>1499.35584</td><td>1.515532</td><td>18.28786</td></tr>\n",
       "\t<tr><th scope=row>grh57</th><td>cc</td><td>grh  </td><td> 6016</td><td>15283</td><td>1985.8798</td><td> 628.25228</td><td>1.658068</td><td>18.16627</td></tr>\n",
       "\t<tr><th scope=row>ab57</th><td>cc</td><td>ab   </td><td>11781</td><td>30184</td><td>3815.7688</td><td>1333.20342</td><td>1.515995</td><td>18.03752</td></tr>\n",
       "\t<tr><th scope=row>Abd-B57</th><td>cc</td><td>Abd-B</td><td> 3094</td><td> 7938</td><td>1136.2949</td><td> 338.24731</td><td>1.743928</td><td>17.70330</td></tr>\n",
       "\t<tr><th scope=row>Ser70</th><td>cc</td><td>Ser  </td><td> 2819</td><td> 5904</td><td> 929.2477</td><td> 266.95383</td><td>1.794079</td><td>17.69226</td></tr>\n",
       "\t<tr><th scope=row>inv57</th><td>cc</td><td>inv  </td><td> 1908</td><td> 3881</td><td> 643.5439</td><td> 172.92837</td><td>1.887545</td><td>17.61483</td></tr>\n",
       "\t<tr><th scope=row>hh57</th><td>cc</td><td>hh   </td><td> 1499</td><td> 3195</td><td> 518.5385</td><td> 134.55242</td><td>1.935600</td><td>17.46122</td></tr>\n",
       "\t<tr><th scope=row>ft57</th><td>cc</td><td>ft   </td><td> 4932</td><td>13268</td><td>1638.9368</td><td> 531.83541</td><td>1.620998</td><td>17.31133</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tds57 & cc & ds    &  9963 & 14410 & 3206.9115 &  619.96869 & 2.368592 & 27.58798\\\\\n",
       "\totp57 & cc & otp   &  3116 &  4680 & 1251.3588 &  218.77044 & 2.509426 & 25.82308\\\\\n",
       "\tPvf357 & cc & Pvf3  &  6120 & 11847 & 2010.4898 &  509.92641 & 1.976360 & 21.68867\\\\\n",
       "\trobo257 & cc & robo2 &  6690 & 11692 & 2010.1194 &  525.03955 & 1.934038 & 21.22371\\\\\n",
       "\tNotum57 & cc & Notum &  2219 &  3892 &  749.4497 &  166.00330 & 2.165955 & 20.68836\\\\\n",
       "\tPoxm57 & cc & Poxm  &   580 &   793 &  206.3057 &   32.51788 & 2.621780 & 20.17621\\\\\n",
       "\tnkd57 & cc & nkd   & 16948 & 38596 & 5280.4399 & 1706.77591 & 1.628539 & 20.13968\\\\\n",
       "\tbyn57 & cc & byn   &   373 &   560 &  148.2083 &   21.97528 & 2.689472 & 19.42117\\\\\n",
       "\tdally57 & cc & dally &  6530 & 13999 & 2080.4428 &  629.45349 & 1.722429 & 18.98697\\\\\n",
       "\ted57 & cc & ed    & 25479 & 68004 & 8207.6110 & 2987.98922 & 1.457305 & 18.94922\\\\\n",
       "\tmgl57 & cc & mgl   & 18913 & 49172 & 6132.5250 & 2178.96099 & 1.492179 & 18.77534\\\\\n",
       "\tN57 & cc & N     & 11778 & 30829 & 3823.4703 & 1281.73298 & 1.575662 & 18.75202\\\\\n",
       "\theph57 & cc & heph  & 13056 & 34869 & 4289.5803 & 1499.35584 & 1.515532 & 18.28786\\\\\n",
       "\tgrh57 & cc & grh   &  6016 & 15283 & 1985.8798 &  628.25228 & 1.658068 & 18.16627\\\\\n",
       "\tab57 & cc & ab    & 11781 & 30184 & 3815.7688 & 1333.20342 & 1.515995 & 18.03752\\\\\n",
       "\tAbd-B57 & cc & Abd-B &  3094 &  7938 & 1136.2949 &  338.24731 & 1.743928 & 17.70330\\\\\n",
       "\tSer70 & cc & Ser   &  2819 &  5904 &  929.2477 &  266.95383 & 1.794079 & 17.69226\\\\\n",
       "\tinv57 & cc & inv   &  1908 &  3881 &  643.5439 &  172.92837 & 1.887545 & 17.61483\\\\\n",
       "\thh57 & cc & hh    &  1499 &  3195 &  518.5385 &  134.55242 & 1.935600 & 17.46122\\\\\n",
       "\tft57 & cc & ft    &  4932 & 13268 & 1638.9368 &  531.83541 & 1.620998 & 17.31133\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| ds57 | cc | ds    |  9963 | 14410 | 3206.9115 |  619.96869 | 2.368592 | 27.58798 |\n",
       "| otp57 | cc | otp   |  3116 |  4680 | 1251.3588 |  218.77044 | 2.509426 | 25.82308 |\n",
       "| Pvf357 | cc | Pvf3  |  6120 | 11847 | 2010.4898 |  509.92641 | 1.976360 | 21.68867 |\n",
       "| robo257 | cc | robo2 |  6690 | 11692 | 2010.1194 |  525.03955 | 1.934038 | 21.22371 |\n",
       "| Notum57 | cc | Notum |  2219 |  3892 |  749.4497 |  166.00330 | 2.165955 | 20.68836 |\n",
       "| Poxm57 | cc | Poxm  |   580 |   793 |  206.3057 |   32.51788 | 2.621780 | 20.17621 |\n",
       "| nkd57 | cc | nkd   | 16948 | 38596 | 5280.4399 | 1706.77591 | 1.628539 | 20.13968 |\n",
       "| byn57 | cc | byn   |   373 |   560 |  148.2083 |   21.97528 | 2.689472 | 19.42117 |\n",
       "| dally57 | cc | dally |  6530 | 13999 | 2080.4428 |  629.45349 | 1.722429 | 18.98697 |\n",
       "| ed57 | cc | ed    | 25479 | 68004 | 8207.6110 | 2987.98922 | 1.457305 | 18.94922 |\n",
       "| mgl57 | cc | mgl   | 18913 | 49172 | 6132.5250 | 2178.96099 | 1.492179 | 18.77534 |\n",
       "| N57 | cc | N     | 11778 | 30829 | 3823.4703 | 1281.73298 | 1.575662 | 18.75202 |\n",
       "| heph57 | cc | heph  | 13056 | 34869 | 4289.5803 | 1499.35584 | 1.515532 | 18.28786 |\n",
       "| grh57 | cc | grh   |  6016 | 15283 | 1985.8798 |  628.25228 | 1.658068 | 18.16627 |\n",
       "| ab57 | cc | ab    | 11781 | 30184 | 3815.7688 | 1333.20342 | 1.515995 | 18.03752 |\n",
       "| Abd-B57 | cc | Abd-B |  3094 |  7938 | 1136.2949 |  338.24731 | 1.743928 | 17.70330 |\n",
       "| Ser70 | cc | Ser   |  2819 |  5904 |  929.2477 |  266.95383 | 1.794079 | 17.69226 |\n",
       "| inv57 | cc | inv   |  1908 |  3881 |  643.5439 |  172.92837 | 1.887545 | 17.61483 |\n",
       "| hh57 | cc | hh    |  1499 |  3195 |  518.5385 |  134.55242 | 1.935600 | 17.46122 |\n",
       "| ft57 | cc | ft    |  4932 | 13268 | 1638.9368 |  531.83541 | 1.620998 | 17.31133 |\n",
       "\n"
      ],
      "text/plain": [
       "        cell.type.2 gene  cell.type.2.n.umi other.n.umi cell.type.2.tpm\n",
       "ds57    cc          ds     9963             14410       3206.9115      \n",
       "otp57   cc          otp    3116              4680       1251.3588      \n",
       "Pvf357  cc          Pvf3   6120             11847       2010.4898      \n",
       "robo257 cc          robo2  6690             11692       2010.1194      \n",
       "Notum57 cc          Notum  2219              3892        749.4497      \n",
       "Poxm57  cc          Poxm    580               793        206.3057      \n",
       "nkd57   cc          nkd   16948             38596       5280.4399      \n",
       "byn57   cc          byn     373               560        148.2083      \n",
       "dally57 cc          dally  6530             13999       2080.4428      \n",
       "ed57    cc          ed    25479             68004       8207.6110      \n",
       "mgl57   cc          mgl   18913             49172       6132.5250      \n",
       "N57     cc          N     11778             30829       3823.4703      \n",
       "heph57  cc          heph  13056             34869       4289.5803      \n",
       "grh57   cc          grh    6016             15283       1985.8798      \n",
       "ab57    cc          ab    11781             30184       3815.7688      \n",
       "Abd-B57 cc          Abd-B  3094              7938       1136.2949      \n",
       "Ser70   cc          Ser    2819              5904        929.2477      \n",
       "inv57   cc          inv    1908              3881        643.5439      \n",
       "hh57    cc          hh     1499              3195        518.5385      \n",
       "ft57    cc          ft     4932             13268       1638.9368      \n",
       "        other.tpm  log2.ratio heuristic.score\n",
       "ds57     619.96869 2.368592   27.58798       \n",
       "otp57    218.77044 2.509426   25.82308       \n",
       "Pvf357   509.92641 1.976360   21.68867       \n",
       "robo257  525.03955 1.934038   21.22371       \n",
       "Notum57  166.00330 2.165955   20.68836       \n",
       "Poxm57    32.51788 2.621780   20.17621       \n",
       "nkd57   1706.77591 1.628539   20.13968       \n",
       "byn57     21.97528 2.689472   19.42117       \n",
       "dally57  629.45349 1.722429   18.98697       \n",
       "ed57    2987.98922 1.457305   18.94922       \n",
       "mgl57   2178.96099 1.492179   18.77534       \n",
       "N57     1281.73298 1.575662   18.75202       \n",
       "heph57  1499.35584 1.515532   18.28786       \n",
       "grh57    628.25228 1.658068   18.16627       \n",
       "ab57    1333.20342 1.515995   18.03752       \n",
       "Abd-B57  338.24731 1.743928   17.70330       \n",
       "Ser70    266.95383 1.794079   17.69226       \n",
       "inv57    172.92837 1.887545   17.61483       \n",
       "hh57     134.55242 1.935600   17.46122       \n",
       "ft57     531.83541 1.620998   17.31133       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.2.markers_no_doubletons  %>% filter(cell.type.2 == \"cc\" & cell.type.2.tpm > other.tpm)  %>% head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 324,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"ds\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_sub_no_doubletons, c(\"ds\"), size_pos = 2, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"ds\"), \n",
    "                   size_pos = 2, size_neg = 0.5, stroke = 0.08, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"otp\"), \n",
    "                   size_pos = 2, size_neg = 0.4, stroke = 0.02, limit = 8)\n",
    "plot.expr_clus.br2(cds_sub_no_doubletons, c(\"Pvf3\"), \n",
    "                   size_pos = 2, size_neg = 0.2, stroke = 0.02, limit = 4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type cd\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"cd\", df_sub300$cluster %in% cd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type ce\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"ce\", df_sub300$cluster %in% ce)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Assigning cells for cell type cf\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df_sub300 = set.cell.type(df_sub300, \"cf\", df_sub300$cluster %in% cf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 771,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 179 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cell.type.3</th><th scope=col>cluster</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Chordotonal_organ     </td><td>172</td><td> 121</td></tr>\n",
       "\t<tr><td>Chordotonal_sensory   </td><td>14 </td><td> 144</td></tr>\n",
       "\t<tr><td>Chordotonal_sensory   </td><td>158</td><td> 606</td></tr>\n",
       "\t<tr><td>Malpighian tubule     </td><td>103</td><td>1265</td></tr>\n",
       "\t<tr><td>Malpighian tubule     </td><td>163</td><td> 466</td></tr>\n",
       "\t<tr><td>multidendritic_sensory</td><td>8  </td><td> 929</td></tr>\n",
       "\t<tr><td>multidendritic_sensory</td><td>14 </td><td> 995</td></tr>\n",
       "\t<tr><td>multidendritic_sensory</td><td>170</td><td> 145</td></tr>\n",
       "\t<tr><td>proventriculus        </td><td>160</td><td> 528</td></tr>\n",
       "\t<tr><td>unknown_1             </td><td>168</td><td> 225</td></tr>\n",
       "\t<tr><td>NA                    </td><td>1  </td><td>2548</td></tr>\n",
       "\t<tr><td>NA                    </td><td>2  </td><td>2423</td></tr>\n",
       "\t<tr><td>NA                    </td><td>3  </td><td>2409</td></tr>\n",
       "\t<tr><td>NA                    </td><td>4  </td><td>2403</td></tr>\n",
       "\t<tr><td>NA                    </td><td>5  </td><td>2376</td></tr>\n",
       "\t<tr><td>NA                    </td><td>6  </td><td>2357</td></tr>\n",
       "\t<tr><td>NA                    </td><td>7  </td><td>2346</td></tr>\n",
       "\t<tr><td>NA                    </td><td>8  </td><td>1415</td></tr>\n",
       "\t<tr><td>NA                    </td><td>9  </td><td>2336</td></tr>\n",
       "\t<tr><td>NA                    </td><td>10 </td><td>2316</td></tr>\n",
       "\t<tr><td>NA                    </td><td>11 </td><td>2236</td></tr>\n",
       "\t<tr><td>NA                    </td><td>12 </td><td>2212</td></tr>\n",
       "\t<tr><td>NA                    </td><td>13 </td><td>2156</td></tr>\n",
       "\t<tr><td>NA                    </td><td>14 </td><td>1007</td></tr>\n",
       "\t<tr><td>NA                    </td><td>15 </td><td>2141</td></tr>\n",
       "\t<tr><td>NA                    </td><td>16 </td><td>2140</td></tr>\n",
       "\t<tr><td>NA                    </td><td>17 </td><td>2128</td></tr>\n",
       "\t<tr><td>NA                    </td><td>18 </td><td>2123</td></tr>\n",
       "\t<tr><td>NA                    </td><td>19 </td><td>2073</td></tr>\n",
       "\t<tr><td>NA                    </td><td>20 </td><td>2047</td></tr>\n",
       "\t<tr><td>NA                    </td><td>21 </td><td>2041</td></tr>\n",
       "\t<tr><td>NA                    </td><td>22 </td><td>2004</td></tr>\n",
       "\t<tr><td>NA                    </td><td>23 </td><td>2001</td></tr>\n",
       "\t<tr><td>NA                    </td><td>24 </td><td>2000</td></tr>\n",
       "\t<tr><td>NA                    </td><td>25 </td><td>1977</td></tr>\n",
       "\t<tr><td>NA                    </td><td>26 </td><td>1960</td></tr>\n",
       "\t<tr><td>NA                    </td><td>27 </td><td>1949</td></tr>\n",
       "\t<tr><td>NA                    </td><td>28 </td><td>1882</td></tr>\n",
       "\t<tr><td>NA                    </td><td>29 </td><td>1878</td></tr>\n",
       "\t<tr><td>NA                    </td><td>30 </td><td>1864</td></tr>\n",
       "\t<tr><td>NA                    </td><td>31 </td><td>1857</td></tr>\n",
       "\t<tr><td>NA                    </td><td>32 </td><td>1854</td></tr>\n",
       "\t<tr><td>NA                    </td><td>33 </td><td>1838</td></tr>\n",
       "\t<tr><td>NA                    </td><td>34 </td><td>1812</td></tr>\n",
       "\t<tr><td>NA                    </td><td>35 </td><td>1801</td></tr>\n",
       "\t<tr><td>NA                    </td><td>36 </td><td>1789</td></tr>\n",
       "\t<tr><td>NA                    </td><td>37 </td><td>1784</td></tr>\n",
       "\t<tr><td>NA                    </td><td>38 </td><td>1768</td></tr>\n",
       "\t<tr><td>NA                    </td><td>39 </td><td>1767</td></tr>\n",
       "\t<tr><td>NA                    </td><td>40 </td><td>1766</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>NA</td><td>121</td><td>1146</td></tr>\n",
       "\t<tr><td>NA</td><td>122</td><td>1140</td></tr>\n",
       "\t<tr><td>NA</td><td>123</td><td>1125</td></tr>\n",
       "\t<tr><td>NA</td><td>124</td><td>1101</td></tr>\n",
       "\t<tr><td>NA</td><td>125</td><td>1101</td></tr>\n",
       "\t<tr><td>NA</td><td>126</td><td>1081</td></tr>\n",
       "\t<tr><td>NA</td><td>127</td><td>1063</td></tr>\n",
       "\t<tr><td>NA</td><td>128</td><td>1062</td></tr>\n",
       "\t<tr><td>NA</td><td>129</td><td>1056</td></tr>\n",
       "\t<tr><td>NA</td><td>130</td><td>1052</td></tr>\n",
       "\t<tr><td>NA</td><td>131</td><td>1043</td></tr>\n",
       "\t<tr><td>NA</td><td>132</td><td>1028</td></tr>\n",
       "\t<tr><td>NA</td><td>133</td><td>1003</td></tr>\n",
       "\t<tr><td>NA</td><td>134</td><td>1000</td></tr>\n",
       "\t<tr><td>NA</td><td>135</td><td> 998</td></tr>\n",
       "\t<tr><td>NA</td><td>136</td><td> 987</td></tr>\n",
       "\t<tr><td>NA</td><td>137</td><td> 981</td></tr>\n",
       "\t<tr><td>NA</td><td>138</td><td> 979</td></tr>\n",
       "\t<tr><td>NA</td><td>139</td><td> 970</td></tr>\n",
       "\t<tr><td>NA</td><td>140</td><td> 962</td></tr>\n",
       "\t<tr><td>NA</td><td>141</td><td> 956</td></tr>\n",
       "\t<tr><td>NA</td><td>142</td><td> 931</td></tr>\n",
       "\t<tr><td>NA</td><td>143</td><td> 898</td></tr>\n",
       "\t<tr><td>NA</td><td>144</td><td> 890</td></tr>\n",
       "\t<tr><td>NA</td><td>145</td><td> 888</td></tr>\n",
       "\t<tr><td>NA</td><td>146</td><td> 887</td></tr>\n",
       "\t<tr><td>NA</td><td>147</td><td> 850</td></tr>\n",
       "\t<tr><td>NA</td><td>148</td><td> 748</td></tr>\n",
       "\t<tr><td>NA</td><td>149</td><td> 717</td></tr>\n",
       "\t<tr><td>NA</td><td>150</td><td> 697</td></tr>\n",
       "\t<tr><td>NA</td><td>151</td><td> 673</td></tr>\n",
       "\t<tr><td>NA</td><td>152</td><td> 651</td></tr>\n",
       "\t<tr><td>NA</td><td>153</td><td> 645</td></tr>\n",
       "\t<tr><td>NA</td><td>154</td><td> 640</td></tr>\n",
       "\t<tr><td>NA</td><td>155</td><td> 636</td></tr>\n",
       "\t<tr><td>NA</td><td>156</td><td> 617</td></tr>\n",
       "\t<tr><td>NA</td><td>157</td><td> 610</td></tr>\n",
       "\t<tr><td>NA</td><td>158</td><td>   1</td></tr>\n",
       "\t<tr><td>NA</td><td>159</td><td> 581</td></tr>\n",
       "\t<tr><td>NA</td><td>161</td><td> 526</td></tr>\n",
       "\t<tr><td>NA</td><td>162</td><td> 485</td></tr>\n",
       "\t<tr><td>NA</td><td>164</td><td> 453</td></tr>\n",
       "\t<tr><td>NA</td><td>165</td><td> 432</td></tr>\n",
       "\t<tr><td>NA</td><td>166</td><td> 371</td></tr>\n",
       "\t<tr><td>NA</td><td>167</td><td> 370</td></tr>\n",
       "\t<tr><td>NA</td><td>169</td><td> 184</td></tr>\n",
       "\t<tr><td>NA</td><td>170</td><td>   4</td></tr>\n",
       "\t<tr><td>NA</td><td>171</td><td> 147</td></tr>\n",
       "\t<tr><td>NA</td><td>173</td><td> 115</td></tr>\n",
       "\t<tr><td>NA</td><td>174</td><td>  59</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 179 × 3\n",
       "\\begin{tabular}{lll}\n",
       " cell.type.3 & cluster & n\\\\\n",
       " <chr> & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t Chordotonal\\_organ      & 172 &  121\\\\\n",
       "\t Chordotonal\\_sensory    & 14  &  144\\\\\n",
       "\t Chordotonal\\_sensory    & 158 &  606\\\\\n",
       "\t Malpighian tubule      & 103 & 1265\\\\\n",
       "\t Malpighian tubule      & 163 &  466\\\\\n",
       "\t multidendritic\\_sensory & 8   &  929\\\\\n",
       "\t multidendritic\\_sensory & 14  &  995\\\\\n",
       "\t multidendritic\\_sensory & 170 &  145\\\\\n",
       "\t proventriculus         & 160 &  528\\\\\n",
       "\t unknown\\_1              & 168 &  225\\\\\n",
       "\t NA                     & 1   & 2548\\\\\n",
       "\t NA                     & 2   & 2423\\\\\n",
       "\t NA                     & 3   & 2409\\\\\n",
       "\t NA                     & 4   & 2403\\\\\n",
       "\t NA                     & 5   & 2376\\\\\n",
       "\t NA                     & 6   & 2357\\\\\n",
       "\t NA                     & 7   & 2346\\\\\n",
       "\t NA                     & 8   & 1415\\\\\n",
       "\t NA                     & 9   & 2336\\\\\n",
       "\t NA                     & 10  & 2316\\\\\n",
       "\t NA                     & 11  & 2236\\\\\n",
       "\t NA                     & 12  & 2212\\\\\n",
       "\t NA                     & 13  & 2156\\\\\n",
       "\t NA                     & 14  & 1007\\\\\n",
       "\t NA                     & 15  & 2141\\\\\n",
       "\t NA                     & 16  & 2140\\\\\n",
       "\t NA                     & 17  & 2128\\\\\n",
       "\t NA                     & 18  & 2123\\\\\n",
       "\t NA                     & 19  & 2073\\\\\n",
       "\t NA                     & 20  & 2047\\\\\n",
       "\t NA                     & 21  & 2041\\\\\n",
       "\t NA                     & 22  & 2004\\\\\n",
       "\t NA                     & 23  & 2001\\\\\n",
       "\t NA                     & 24  & 2000\\\\\n",
       "\t NA                     & 25  & 1977\\\\\n",
       "\t NA                     & 26  & 1960\\\\\n",
       "\t NA                     & 27  & 1949\\\\\n",
       "\t NA                     & 28  & 1882\\\\\n",
       "\t NA                     & 29  & 1878\\\\\n",
       "\t NA                     & 30  & 1864\\\\\n",
       "\t NA                     & 31  & 1857\\\\\n",
       "\t NA                     & 32  & 1854\\\\\n",
       "\t NA                     & 33  & 1838\\\\\n",
       "\t NA                     & 34  & 1812\\\\\n",
       "\t NA                     & 35  & 1801\\\\\n",
       "\t NA                     & 36  & 1789\\\\\n",
       "\t NA                     & 37  & 1784\\\\\n",
       "\t NA                     & 38  & 1768\\\\\n",
       "\t NA                     & 39  & 1767\\\\\n",
       "\t NA                     & 40  & 1766\\\\\n",
       "\t ⋮ & ⋮ & ⋮\\\\\n",
       "\t NA & 121 & 1146\\\\\n",
       "\t NA & 122 & 1140\\\\\n",
       "\t NA & 123 & 1125\\\\\n",
       "\t NA & 124 & 1101\\\\\n",
       "\t NA & 125 & 1101\\\\\n",
       "\t NA & 126 & 1081\\\\\n",
       "\t NA & 127 & 1063\\\\\n",
       "\t NA & 128 & 1062\\\\\n",
       "\t NA & 129 & 1056\\\\\n",
       "\t NA & 130 & 1052\\\\\n",
       "\t NA & 131 & 1043\\\\\n",
       "\t NA & 132 & 1028\\\\\n",
       "\t NA & 133 & 1003\\\\\n",
       "\t NA & 134 & 1000\\\\\n",
       "\t NA & 135 &  998\\\\\n",
       "\t NA & 136 &  987\\\\\n",
       "\t NA & 137 &  981\\\\\n",
       "\t NA & 138 &  979\\\\\n",
       "\t NA & 139 &  970\\\\\n",
       "\t NA & 140 &  962\\\\\n",
       "\t NA & 141 &  956\\\\\n",
       "\t NA & 142 &  931\\\\\n",
       "\t NA & 143 &  898\\\\\n",
       "\t NA & 144 &  890\\\\\n",
       "\t NA & 145 &  888\\\\\n",
       "\t NA & 146 &  887\\\\\n",
       "\t NA & 147 &  850\\\\\n",
       "\t NA & 148 &  748\\\\\n",
       "\t NA & 149 &  717\\\\\n",
       "\t NA & 150 &  697\\\\\n",
       "\t NA & 151 &  673\\\\\n",
       "\t NA & 152 &  651\\\\\n",
       "\t NA & 153 &  645\\\\\n",
       "\t NA & 154 &  640\\\\\n",
       "\t NA & 155 &  636\\\\\n",
       "\t NA & 156 &  617\\\\\n",
       "\t NA & 157 &  610\\\\\n",
       "\t NA & 158 &    1\\\\\n",
       "\t NA & 159 &  581\\\\\n",
       "\t NA & 161 &  526\\\\\n",
       "\t NA & 162 &  485\\\\\n",
       "\t NA & 164 &  453\\\\\n",
       "\t NA & 165 &  432\\\\\n",
       "\t NA & 166 &  371\\\\\n",
       "\t NA & 167 &  370\\\\\n",
       "\t NA & 169 &  184\\\\\n",
       "\t NA & 170 &    4\\\\\n",
       "\t NA & 171 &  147\\\\\n",
       "\t NA & 173 &  115\\\\\n",
       "\t NA & 174 &   59\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 179 × 3\n",
       "\n",
       "| cell.type.3 &lt;chr&gt; | cluster &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| Chordotonal_organ      | 172 |  121 |\n",
       "| Chordotonal_sensory    | 14  |  144 |\n",
       "| Chordotonal_sensory    | 158 |  606 |\n",
       "| Malpighian tubule      | 103 | 1265 |\n",
       "| Malpighian tubule      | 163 |  466 |\n",
       "| multidendritic_sensory | 8   |  929 |\n",
       "| multidendritic_sensory | 14  |  995 |\n",
       "| multidendritic_sensory | 170 |  145 |\n",
       "| proventriculus         | 160 |  528 |\n",
       "| unknown_1              | 168 |  225 |\n",
       "| NA                     | 1   | 2548 |\n",
       "| NA                     | 2   | 2423 |\n",
       "| NA                     | 3   | 2409 |\n",
       "| NA                     | 4   | 2403 |\n",
       "| NA                     | 5   | 2376 |\n",
       "| NA                     | 6   | 2357 |\n",
       "| NA                     | 7   | 2346 |\n",
       "| NA                     | 8   | 1415 |\n",
       "| NA                     | 9   | 2336 |\n",
       "| NA                     | 10  | 2316 |\n",
       "| NA                     | 11  | 2236 |\n",
       "| NA                     | 12  | 2212 |\n",
       "| NA                     | 13  | 2156 |\n",
       "| NA                     | 14  | 1007 |\n",
       "| NA                     | 15  | 2141 |\n",
       "| NA                     | 16  | 2140 |\n",
       "| NA                     | 17  | 2128 |\n",
       "| NA                     | 18  | 2123 |\n",
       "| NA                     | 19  | 2073 |\n",
       "| NA                     | 20  | 2047 |\n",
       "| NA                     | 21  | 2041 |\n",
       "| NA                     | 22  | 2004 |\n",
       "| NA                     | 23  | 2001 |\n",
       "| NA                     | 24  | 2000 |\n",
       "| NA                     | 25  | 1977 |\n",
       "| NA                     | 26  | 1960 |\n",
       "| NA                     | 27  | 1949 |\n",
       "| NA                     | 28  | 1882 |\n",
       "| NA                     | 29  | 1878 |\n",
       "| NA                     | 30  | 1864 |\n",
       "| NA                     | 31  | 1857 |\n",
       "| NA                     | 32  | 1854 |\n",
       "| NA                     | 33  | 1838 |\n",
       "| NA                     | 34  | 1812 |\n",
       "| NA                     | 35  | 1801 |\n",
       "| NA                     | 36  | 1789 |\n",
       "| NA                     | 37  | 1784 |\n",
       "| NA                     | 38  | 1768 |\n",
       "| NA                     | 39  | 1767 |\n",
       "| NA                     | 40  | 1766 |\n",
       "| ⋮ | ⋮ | ⋮ |\n",
       "| NA | 121 | 1146 |\n",
       "| NA | 122 | 1140 |\n",
       "| NA | 123 | 1125 |\n",
       "| NA | 124 | 1101 |\n",
       "| NA | 125 | 1101 |\n",
       "| NA | 126 | 1081 |\n",
       "| NA | 127 | 1063 |\n",
       "| NA | 128 | 1062 |\n",
       "| NA | 129 | 1056 |\n",
       "| NA | 130 | 1052 |\n",
       "| NA | 131 | 1043 |\n",
       "| NA | 132 | 1028 |\n",
       "| NA | 133 | 1003 |\n",
       "| NA | 134 | 1000 |\n",
       "| NA | 135 |  998 |\n",
       "| NA | 136 |  987 |\n",
       "| NA | 137 |  981 |\n",
       "| NA | 138 |  979 |\n",
       "| NA | 139 |  970 |\n",
       "| NA | 140 |  962 |\n",
       "| NA | 141 |  956 |\n",
       "| NA | 142 |  931 |\n",
       "| NA | 143 |  898 |\n",
       "| NA | 144 |  890 |\n",
       "| NA | 145 |  888 |\n",
       "| NA | 146 |  887 |\n",
       "| NA | 147 |  850 |\n",
       "| NA | 148 |  748 |\n",
       "| NA | 149 |  717 |\n",
       "| NA | 150 |  697 |\n",
       "| NA | 151 |  673 |\n",
       "| NA | 152 |  651 |\n",
       "| NA | 153 |  645 |\n",
       "| NA | 154 |  640 |\n",
       "| NA | 155 |  636 |\n",
       "| NA | 156 |  617 |\n",
       "| NA | 157 |  610 |\n",
       "| NA | 158 |    1 |\n",
       "| NA | 159 |  581 |\n",
       "| NA | 161 |  526 |\n",
       "| NA | 162 |  485 |\n",
       "| NA | 164 |  453 |\n",
       "| NA | 165 |  432 |\n",
       "| NA | 166 |  371 |\n",
       "| NA | 167 |  370 |\n",
       "| NA | 169 |  184 |\n",
       "| NA | 170 |    4 |\n",
       "| NA | 171 |  147 |\n",
       "| NA | 173 |  115 |\n",
       "| NA | 174 |   59 |\n",
       "\n"
      ],
      "text/plain": [
       "    cell.type.3            cluster n   \n",
       "1   Chordotonal_organ      172      121\n",
       "2   Chordotonal_sensory    14       144\n",
       "3   Chordotonal_sensory    158      606\n",
       "4   Malpighian tubule      103     1265\n",
       "5   Malpighian tubule      163      466\n",
       "6   multidendritic_sensory 8        929\n",
       "7   multidendritic_sensory 14       995\n",
       "8   multidendritic_sensory 170      145\n",
       "9   proventriculus         160      528\n",
       "10  unknown_1              168      225\n",
       "11  NA                     1       2548\n",
       "12  NA                     2       2423\n",
       "13  NA                     3       2409\n",
       "14  NA                     4       2403\n",
       "15  NA                     5       2376\n",
       "16  NA                     6       2357\n",
       "17  NA                     7       2346\n",
       "18  NA                     8       1415\n",
       "19  NA                     9       2336\n",
       "20  NA                     10      2316\n",
       "21  NA                     11      2236\n",
       "22  NA                     12      2212\n",
       "23  NA                     13      2156\n",
       "24  NA                     14      1007\n",
       "25  NA                     15      2141\n",
       "26  NA                     16      2140\n",
       "27  NA                     17      2128\n",
       "28  NA                     18      2123\n",
       "29  NA                     19      2073\n",
       "30  NA                     20      2047\n",
       "31  NA                     21      2041\n",
       "32  NA                     22      2004\n",
       "33  NA                     23      2001\n",
       "34  NA                     24      2000\n",
       "35  NA                     25      1977\n",
       "36  NA                     26      1960\n",
       "37  NA                     27      1949\n",
       "38  NA                     28      1882\n",
       "39  NA                     29      1878\n",
       "40  NA                     30      1864\n",
       "41  NA                     31      1857\n",
       "42  NA                     32      1854\n",
       "43  NA                     33      1838\n",
       "44  NA                     34      1812\n",
       "45  NA                     35      1801\n",
       "46  NA                     36      1789\n",
       "47  NA                     37      1784\n",
       "48  NA                     38      1768\n",
       "49  NA                     39      1767\n",
       "50  NA                     40      1766\n",
       "⋮   ⋮                      ⋮       ⋮   \n",
       "130 NA                     121     1146\n",
       "131 NA                     122     1140\n",
       "132 NA                     123     1125\n",
       "133 NA                     124     1101\n",
       "134 NA                     125     1101\n",
       "135 NA                     126     1081\n",
       "136 NA                     127     1063\n",
       "137 NA                     128     1062\n",
       "138 NA                     129     1056\n",
       "139 NA                     130     1052\n",
       "140 NA                     131     1043\n",
       "141 NA                     132     1028\n",
       "142 NA                     133     1003\n",
       "143 NA                     134     1000\n",
       "144 NA                     135      998\n",
       "145 NA                     136      987\n",
       "146 NA                     137      981\n",
       "147 NA                     138      979\n",
       "148 NA                     139      970\n",
       "149 NA                     140      962\n",
       "150 NA                     141      956\n",
       "151 NA                     142      931\n",
       "152 NA                     143      898\n",
       "153 NA                     144      890\n",
       "154 NA                     145      888\n",
       "155 NA                     146      887\n",
       "156 NA                     147      850\n",
       "157 NA                     148      748\n",
       "158 NA                     149      717\n",
       "159 NA                     150      697\n",
       "160 NA                     151      673\n",
       "161 NA                     152      651\n",
       "162 NA                     153      645\n",
       "163 NA                     154      640\n",
       "164 NA                     155      636\n",
       "165 NA                     156      617\n",
       "166 NA                     157      610\n",
       "167 NA                     158        1\n",
       "168 NA                     159      581\n",
       "169 NA                     161      526\n",
       "170 NA                     162      485\n",
       "171 NA                     164      453\n",
       "172 NA                     165      432\n",
       "173 NA                     166      371\n",
       "174 NA                     167      370\n",
       "175 NA                     169      184\n",
       "176 NA                     170        4\n",
       "177 NA                     171      147\n",
       "178 NA                     173      115\n",
       "179 NA                     174       59"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub300  %>% count(cell.type.3, cluster)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1601,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 62 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cell.type.3</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Amnioserosa                               </td><td>  7091</td></tr>\n",
       "\t<tr><td>Analpad                                   </td><td>   850</td></tr>\n",
       "\t<tr><td>Analpad_large_intestine                   </td><td>   734</td></tr>\n",
       "\t<tr><td>Antennal_sense_organs                     </td><td>   748</td></tr>\n",
       "\t<tr><td>Bolwig_organ                              </td><td>   115</td></tr>\n",
       "\t<tr><td>Central_brain_ventral_nerve_cord_primorium</td><td>  5819</td></tr>\n",
       "\t<tr><td>Cholinergic                               </td><td>  8949</td></tr>\n",
       "\t<tr><td>Chordotonal_organ                         </td><td>   121</td></tr>\n",
       "\t<tr><td>Chordotonal_sensory                       </td><td>   750</td></tr>\n",
       "\t<tr><td>Copper_cells                              </td><td>  4327</td></tr>\n",
       "\t<tr><td>Crystal_cell                              </td><td>   432</td></tr>\n",
       "\t<tr><td>Cuprophilic_cells                         </td><td>   581</td></tr>\n",
       "\t<tr><td>Digestive_system                          </td><td>  2212</td></tr>\n",
       "\t<tr><td>Epidermis                                 </td><td> 10373</td></tr>\n",
       "\t<tr><td>Epidermis_late                            </td><td>   184</td></tr>\n",
       "\t<tr><td>External_sensory                          </td><td>   647</td></tr>\n",
       "\t<tr><td>Fat_body                                  </td><td>  5694</td></tr>\n",
       "\t<tr><td>Germ_cells_ovaries_query                  </td><td>  2139</td></tr>\n",
       "\t<tr><td>Glutamatergic                             </td><td>  5107</td></tr>\n",
       "\t<tr><td>Late_cells                                </td><td>  1565</td></tr>\n",
       "\t<tr><td>Longitudinal_visceral_muscle              </td><td>  1196</td></tr>\n",
       "\t<tr><td>Malpighian tubule                         </td><td>  1731</td></tr>\n",
       "\t<tr><td>Maxillary_sensory_complex                 </td><td>   533</td></tr>\n",
       "\t<tr><td>Midgut                                    </td><td>  1674</td></tr>\n",
       "\t<tr><td>Midgut_4th_chamber                        </td><td>  2128</td></tr>\n",
       "\t<tr><td>Midgut_primordium                         </td><td>   689</td></tr>\n",
       "\t<tr><td>Monoaminergic                             </td><td>   454</td></tr>\n",
       "\t<tr><td>Multidendritic_sensory                    </td><td>  2069</td></tr>\n",
       "\t<tr><td>Muscle_system                             </td><td>  5347</td></tr>\n",
       "\t<tr><td>Neuron_unknown_1                          </td><td>   602</td></tr>\n",
       "\t<tr><td>Oenocytes                                 </td><td>   142</td></tr>\n",
       "\t<tr><td>Oenocytes_older                           </td><td>   250</td></tr>\n",
       "\t<tr><td>Olfactory_neurons                         </td><td>  1864</td></tr>\n",
       "\t<tr><td>Peptidergic_neuron                        </td><td>   147</td></tr>\n",
       "\t<tr><td>Peripheral_nervous_system_early           </td><td>  3851</td></tr>\n",
       "\t<tr><td>Pharynx                                   </td><td>  1704</td></tr>\n",
       "\t<tr><td>Pharynx_2                                 </td><td>   975</td></tr>\n",
       "\t<tr><td>Pharynx_primordium_query                  </td><td>  1371</td></tr>\n",
       "\t<tr><td>Plasmatocytes                             </td><td>  6021</td></tr>\n",
       "\t<tr><td>Posterior_spiracle                        </td><td>  1436</td></tr>\n",
       "\t<tr><td>Proventriculus                            </td><td>  2423</td></tr>\n",
       "\t<tr><td>Proventriculus_older                      </td><td>   253</td></tr>\n",
       "\t<tr><td>Salivary_gland                            </td><td>  4237</td></tr>\n",
       "\t<tr><td>Somatic_muscle                            </td><td>  2766</td></tr>\n",
       "\t<tr><td>Somatic_muscle_primordium                 </td><td>  5139</td></tr>\n",
       "\t<tr><td>Tracheal_system_1                         </td><td>  1101</td></tr>\n",
       "\t<tr><td>Tracheal_system_2                         </td><td>  2642</td></tr>\n",
       "\t<tr><td>Tracheal_system_early                     </td><td>  1797</td></tr>\n",
       "\t<tr><td>Tracheal_system_older                     </td><td>  2821</td></tr>\n",
       "\t<tr><td>Ventral_midline                           </td><td>  2159</td></tr>\n",
       "\t<tr><td>Visceral_muscle                           </td><td>  2308</td></tr>\n",
       "\t<tr><td>Visceral_muscle_2                         </td><td>  1508</td></tr>\n",
       "\t<tr><td>gut_query                                 </td><td>   887</td></tr>\n",
       "\t<tr><td>large_intestine_hindgut_analpad           </td><td>  2226</td></tr>\n",
       "\t<tr><td>large_intestine_hindgut_analpad_2         </td><td>  1467</td></tr>\n",
       "\t<tr><td>lateral_cord_glia                         </td><td>   636</td></tr>\n",
       "\t<tr><td>nephrocyte_query                          </td><td>   536</td></tr>\n",
       "\t<tr><td>proventriculus                            </td><td>   528</td></tr>\n",
       "\t<tr><td>uncertain_1                               </td><td>    59</td></tr>\n",
       "\t<tr><td>unknown_2                                 </td><td>  1629</td></tr>\n",
       "\t<tr><td>unknown_3                                 </td><td>   180</td></tr>\n",
       "\t<tr><td>NA                                        </td><td>108521</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 62 × 2\n",
       "\\begin{tabular}{ll}\n",
       " cell.type.3 & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Amnioserosa                                &   7091\\\\\n",
       "\t Analpad                                    &    850\\\\\n",
       "\t Analpad\\_large\\_intestine                    &    734\\\\\n",
       "\t Antennal\\_sense\\_organs                      &    748\\\\\n",
       "\t Bolwig\\_organ                               &    115\\\\\n",
       "\t Central\\_brain\\_ventral\\_nerve\\_cord\\_primorium &   5819\\\\\n",
       "\t Cholinergic                                &   8949\\\\\n",
       "\t Chordotonal\\_organ                          &    121\\\\\n",
       "\t Chordotonal\\_sensory                        &    750\\\\\n",
       "\t Copper\\_cells                               &   4327\\\\\n",
       "\t Crystal\\_cell                               &    432\\\\\n",
       "\t Cuprophilic\\_cells                          &    581\\\\\n",
       "\t Digestive\\_system                           &   2212\\\\\n",
       "\t Epidermis                                  &  10373\\\\\n",
       "\t Epidermis\\_late                             &    184\\\\\n",
       "\t External\\_sensory                           &    647\\\\\n",
       "\t Fat\\_body                                   &   5694\\\\\n",
       "\t Germ\\_cells\\_ovaries\\_query                   &   2139\\\\\n",
       "\t Glutamatergic                              &   5107\\\\\n",
       "\t Late\\_cells                                 &   1565\\\\\n",
       "\t Longitudinal\\_visceral\\_muscle               &   1196\\\\\n",
       "\t Malpighian tubule                          &   1731\\\\\n",
       "\t Maxillary\\_sensory\\_complex                  &    533\\\\\n",
       "\t Midgut                                     &   1674\\\\\n",
       "\t Midgut\\_4th\\_chamber                         &   2128\\\\\n",
       "\t Midgut\\_primordium                          &    689\\\\\n",
       "\t Monoaminergic                              &    454\\\\\n",
       "\t Multidendritic\\_sensory                     &   2069\\\\\n",
       "\t Muscle\\_system                              &   5347\\\\\n",
       "\t Neuron\\_unknown\\_1                           &    602\\\\\n",
       "\t Oenocytes                                  &    142\\\\\n",
       "\t Oenocytes\\_older                            &    250\\\\\n",
       "\t Olfactory\\_neurons                          &   1864\\\\\n",
       "\t Peptidergic\\_neuron                         &    147\\\\\n",
       "\t Peripheral\\_nervous\\_system\\_early            &   3851\\\\\n",
       "\t Pharynx                                    &   1704\\\\\n",
       "\t Pharynx\\_2                                  &    975\\\\\n",
       "\t Pharynx\\_primordium\\_query                   &   1371\\\\\n",
       "\t Plasmatocytes                              &   6021\\\\\n",
       "\t Posterior\\_spiracle                         &   1436\\\\\n",
       "\t Proventriculus                             &   2423\\\\\n",
       "\t Proventriculus\\_older                       &    253\\\\\n",
       "\t Salivary\\_gland                             &   4237\\\\\n",
       "\t Somatic\\_muscle                             &   2766\\\\\n",
       "\t Somatic\\_muscle\\_primordium                  &   5139\\\\\n",
       "\t Tracheal\\_system\\_1                          &   1101\\\\\n",
       "\t Tracheal\\_system\\_2                          &   2642\\\\\n",
       "\t Tracheal\\_system\\_early                      &   1797\\\\\n",
       "\t Tracheal\\_system\\_older                      &   2821\\\\\n",
       "\t Ventral\\_midline                            &   2159\\\\\n",
       "\t Visceral\\_muscle                            &   2308\\\\\n",
       "\t Visceral\\_muscle\\_2                          &   1508\\\\\n",
       "\t gut\\_query                                  &    887\\\\\n",
       "\t large\\_intestine\\_hindgut\\_analpad            &   2226\\\\\n",
       "\t large\\_intestine\\_hindgut\\_analpad\\_2          &   1467\\\\\n",
       "\t lateral\\_cord\\_glia                          &    636\\\\\n",
       "\t nephrocyte\\_query                           &    536\\\\\n",
       "\t proventriculus                             &    528\\\\\n",
       "\t uncertain\\_1                                &     59\\\\\n",
       "\t unknown\\_2                                  &   1629\\\\\n",
       "\t unknown\\_3                                  &    180\\\\\n",
       "\t NA                                         & 108521\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 62 × 2\n",
       "\n",
       "| cell.type.3 &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Amnioserosa                                |   7091 |\n",
       "| Analpad                                    |    850 |\n",
       "| Analpad_large_intestine                    |    734 |\n",
       "| Antennal_sense_organs                      |    748 |\n",
       "| Bolwig_organ                               |    115 |\n",
       "| Central_brain_ventral_nerve_cord_primorium |   5819 |\n",
       "| Cholinergic                                |   8949 |\n",
       "| Chordotonal_organ                          |    121 |\n",
       "| Chordotonal_sensory                        |    750 |\n",
       "| Copper_cells                               |   4327 |\n",
       "| Crystal_cell                               |    432 |\n",
       "| Cuprophilic_cells                          |    581 |\n",
       "| Digestive_system                           |   2212 |\n",
       "| Epidermis                                  |  10373 |\n",
       "| Epidermis_late                             |    184 |\n",
       "| External_sensory                           |    647 |\n",
       "| Fat_body                                   |   5694 |\n",
       "| Germ_cells_ovaries_query                   |   2139 |\n",
       "| Glutamatergic                              |   5107 |\n",
       "| Late_cells                                 |   1565 |\n",
       "| Longitudinal_visceral_muscle               |   1196 |\n",
       "| Malpighian tubule                          |   1731 |\n",
       "| Maxillary_sensory_complex                  |    533 |\n",
       "| Midgut                                     |   1674 |\n",
       "| Midgut_4th_chamber                         |   2128 |\n",
       "| Midgut_primordium                          |    689 |\n",
       "| Monoaminergic                              |    454 |\n",
       "| Multidendritic_sensory                     |   2069 |\n",
       "| Muscle_system                              |   5347 |\n",
       "| Neuron_unknown_1                           |    602 |\n",
       "| Oenocytes                                  |    142 |\n",
       "| Oenocytes_older                            |    250 |\n",
       "| Olfactory_neurons                          |   1864 |\n",
       "| Peptidergic_neuron                         |    147 |\n",
       "| Peripheral_nervous_system_early            |   3851 |\n",
       "| Pharynx                                    |   1704 |\n",
       "| Pharynx_2                                  |    975 |\n",
       "| Pharynx_primordium_query                   |   1371 |\n",
       "| Plasmatocytes                              |   6021 |\n",
       "| Posterior_spiracle                         |   1436 |\n",
       "| Proventriculus                             |   2423 |\n",
       "| Proventriculus_older                       |    253 |\n",
       "| Salivary_gland                             |   4237 |\n",
       "| Somatic_muscle                             |   2766 |\n",
       "| Somatic_muscle_primordium                  |   5139 |\n",
       "| Tracheal_system_1                          |   1101 |\n",
       "| Tracheal_system_2                          |   2642 |\n",
       "| Tracheal_system_early                      |   1797 |\n",
       "| Tracheal_system_older                      |   2821 |\n",
       "| Ventral_midline                            |   2159 |\n",
       "| Visceral_muscle                            |   2308 |\n",
       "| Visceral_muscle_2                          |   1508 |\n",
       "| gut_query                                  |    887 |\n",
       "| large_intestine_hindgut_analpad            |   2226 |\n",
       "| large_intestine_hindgut_analpad_2          |   1467 |\n",
       "| lateral_cord_glia                          |    636 |\n",
       "| nephrocyte_query                           |    536 |\n",
       "| proventriculus                             |    528 |\n",
       "| uncertain_1                                |     59 |\n",
       "| unknown_2                                  |   1629 |\n",
       "| unknown_3                                  |    180 |\n",
       "| NA                                         | 108521 |\n",
       "\n"
      ],
      "text/plain": [
       "   cell.type.3                                n     \n",
       "1  Amnioserosa                                  7091\n",
       "2  Analpad                                       850\n",
       "3  Analpad_large_intestine                       734\n",
       "4  Antennal_sense_organs                         748\n",
       "5  Bolwig_organ                                  115\n",
       "6  Central_brain_ventral_nerve_cord_primorium   5819\n",
       "7  Cholinergic                                  8949\n",
       "8  Chordotonal_organ                             121\n",
       "9  Chordotonal_sensory                           750\n",
       "10 Copper_cells                                 4327\n",
       "11 Crystal_cell                                  432\n",
       "12 Cuprophilic_cells                             581\n",
       "13 Digestive_system                             2212\n",
       "14 Epidermis                                   10373\n",
       "15 Epidermis_late                                184\n",
       "16 External_sensory                              647\n",
       "17 Fat_body                                     5694\n",
       "18 Germ_cells_ovaries_query                     2139\n",
       "19 Glutamatergic                                5107\n",
       "20 Late_cells                                   1565\n",
       "21 Longitudinal_visceral_muscle                 1196\n",
       "22 Malpighian tubule                            1731\n",
       "23 Maxillary_sensory_complex                     533\n",
       "24 Midgut                                       1674\n",
       "25 Midgut_4th_chamber                           2128\n",
       "26 Midgut_primordium                             689\n",
       "27 Monoaminergic                                 454\n",
       "28 Multidendritic_sensory                       2069\n",
       "29 Muscle_system                                5347\n",
       "30 Neuron_unknown_1                              602\n",
       "31 Oenocytes                                     142\n",
       "32 Oenocytes_older                               250\n",
       "33 Olfactory_neurons                            1864\n",
       "34 Peptidergic_neuron                            147\n",
       "35 Peripheral_nervous_system_early              3851\n",
       "36 Pharynx                                      1704\n",
       "37 Pharynx_2                                     975\n",
       "38 Pharynx_primordium_query                     1371\n",
       "39 Plasmatocytes                                6021\n",
       "40 Posterior_spiracle                           1436\n",
       "41 Proventriculus                               2423\n",
       "42 Proventriculus_older                          253\n",
       "43 Salivary_gland                               4237\n",
       "44 Somatic_muscle                               2766\n",
       "45 Somatic_muscle_primordium                    5139\n",
       "46 Tracheal_system_1                            1101\n",
       "47 Tracheal_system_2                            2642\n",
       "48 Tracheal_system_early                        1797\n",
       "49 Tracheal_system_older                        2821\n",
       "50 Ventral_midline                              2159\n",
       "51 Visceral_muscle                              2308\n",
       "52 Visceral_muscle_2                            1508\n",
       "53 gut_query                                     887\n",
       "54 large_intestine_hindgut_analpad              2226\n",
       "55 large_intestine_hindgut_analpad_2            1467\n",
       "56 lateral_cord_glia                             636\n",
       "57 nephrocyte_query                              536\n",
       "58 proventriculus                                528\n",
       "59 uncertain_1                                    59\n",
       "60 unknown_2                                    1629\n",
       "61 unknown_3                                     180\n",
       "62 NA                                         108521"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub300  %>% count(cell.type.3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "rename.cell.type = function(data_df, cell.type, x) {\n",
    "       message(\"Reassigning cells for cell type \", cell.type)\n",
    "       data_df$cell.type.3 = ifelse(x, cell.type, data_df$cell.type.3)\n",
    "       return(data_df)\n",
    "    }\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1778,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sub300_copy = df_sub300"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1605,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sub300_copy$cell.type.3 <- \n",
    "gsub(\"nephrocyte_query\", \"Garland_nephrocyte\", df_sub300_copy$cell.type.3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1779,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sub300_copy$cell.type.3 <- \n",
    "gsub(\"proventriculus\", NA, df_sub300_copy$cell.type.3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1780,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 70 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cell.type.3</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Amnioserosa                               </td><td> 7091</td></tr>\n",
       "\t<tr><td>Analpad                                   </td><td>  850</td></tr>\n",
       "\t<tr><td>Analpad_large_intestine                   </td><td>  734</td></tr>\n",
       "\t<tr><td>Antennal_sense_organs                     </td><td>  748</td></tr>\n",
       "\t<tr><td>Bolwig_organ                              </td><td>  115</td></tr>\n",
       "\t<tr><td>Central_brain_ventral_nerve_cord_1        </td><td> 6846</td></tr>\n",
       "\t<tr><td>Central_brain_ventral_nerve_cord_2        </td><td>10161</td></tr>\n",
       "\t<tr><td>Central_brain_ventral_nerve_cord_early    </td><td> 5058</td></tr>\n",
       "\t<tr><td>Central_brain_ventral_nerve_cord_primorium</td><td> 5819</td></tr>\n",
       "\t<tr><td>Cholinergic                               </td><td> 8949</td></tr>\n",
       "\t<tr><td>Chordotonal_organ                         </td><td>  121</td></tr>\n",
       "\t<tr><td>Chordotonal_sensory                       </td><td>  750</td></tr>\n",
       "\t<tr><td>Copper_cells                              </td><td> 4327</td></tr>\n",
       "\t<tr><td>Crystal_cell                              </td><td>  432</td></tr>\n",
       "\t<tr><td>Cuprophilic_cells                         </td><td>  581</td></tr>\n",
       "\t<tr><td>Digestive_system                          </td><td> 2212</td></tr>\n",
       "\t<tr><td>Embryonic_body_wall                       </td><td>42392</td></tr>\n",
       "\t<tr><td>Epidermis                                 </td><td>17230</td></tr>\n",
       "\t<tr><td>Epidermis_late                            </td><td>  184</td></tr>\n",
       "\t<tr><td>External_sensory                          </td><td>  647</td></tr>\n",
       "\t<tr><td>Fat_body                                  </td><td> 5694</td></tr>\n",
       "\t<tr><td>Garland_cell_nephrocyte                   </td><td>  536</td></tr>\n",
       "\t<tr><td>Germ_cells_ovaries_query                  </td><td> 2139</td></tr>\n",
       "\t<tr><td>Glutamatergic                             </td><td> 5107</td></tr>\n",
       "\t<tr><td>Late_cells                                </td><td> 1565</td></tr>\n",
       "\t<tr><td>Longitudinal_visceral_muscle              </td><td>  617</td></tr>\n",
       "\t<tr><td>Malpighian tubule                         </td><td> 1731</td></tr>\n",
       "\t<tr><td>Maxillary_sensory_complex                 </td><td>  533</td></tr>\n",
       "\t<tr><td>Midgut                                    </td><td> 1674</td></tr>\n",
       "\t<tr><td>Midgut_4th_chamber                        </td><td> 2128</td></tr>\n",
       "\t<tr><td>Midgut_primordium                         </td><td>  689</td></tr>\n",
       "\t<tr><td>Monoaminergic                             </td><td>  454</td></tr>\n",
       "\t<tr><td>Multidendritic_sensory                    </td><td> 2069</td></tr>\n",
       "\t<tr><td>Muscle_system                             </td><td> 5347</td></tr>\n",
       "\t<tr><td>Neuron_unknown_1                          </td><td>  602</td></tr>\n",
       "\t<tr><td>Neuron_unknown_2                          </td><td>  225</td></tr>\n",
       "\t<tr><td>Oenocytes                                 </td><td>  142</td></tr>\n",
       "\t<tr><td>Oenocytes_older                           </td><td>  250</td></tr>\n",
       "\t<tr><td>Olfactory_neurons                         </td><td> 1864</td></tr>\n",
       "\t<tr><td>Peptidergic_neuron                        </td><td>  147</td></tr>\n",
       "\t<tr><td>Peripheral_nervous_system_early           </td><td> 3851</td></tr>\n",
       "\t<tr><td>Pharynx                                   </td><td> 1704</td></tr>\n",
       "\t<tr><td>Pharynx_2                                 </td><td>  975</td></tr>\n",
       "\t<tr><td>Pharynx_primordium_query                  </td><td> 1990</td></tr>\n",
       "\t<tr><td>Plasmatocytes                             </td><td> 6021</td></tr>\n",
       "\t<tr><td>Posterior_spiracle                        </td><td> 1436</td></tr>\n",
       "\t<tr><td>Proventriculus                            </td><td> 2423</td></tr>\n",
       "\t<tr><td>Proventriculus_older                      </td><td>  253</td></tr>\n",
       "\t<tr><td>Salivary_gland                            </td><td> 4237</td></tr>\n",
       "\t<tr><td>Somatic_muscle                            </td><td> 2766</td></tr>\n",
       "\t<tr><td>Somatic_muscle_primordium                 </td><td> 5139</td></tr>\n",
       "\t<tr><td>Tracheal_system_1                         </td><td> 1101</td></tr>\n",
       "\t<tr><td>Tracheal_system_2                         </td><td> 2642</td></tr>\n",
       "\t<tr><td>Tracheal_system_early                     </td><td> 1797</td></tr>\n",
       "\t<tr><td>Tracheal_system_older                     </td><td> 2821</td></tr>\n",
       "\t<tr><td>Undifferentiated_1                        </td><td> 3354</td></tr>\n",
       "\t<tr><td>Undifferentiated_2                        </td><td> 1700</td></tr>\n",
       "\t<tr><td>Undifferentiated_3                        </td><td> 2511</td></tr>\n",
       "\t<tr><td>Ventral_midline                           </td><td> 2159</td></tr>\n",
       "\t<tr><td>Visceral_mesoderm                         </td><td> 1495</td></tr>\n",
       "\t<tr><td>Visceral_muscle                           </td><td> 2308</td></tr>\n",
       "\t<tr><td>Visceral_muscle_2                         </td><td> 1508</td></tr>\n",
       "\t<tr><td>gut_query                                 </td><td>  887</td></tr>\n",
       "\t<tr><td>large_intestine_hindgut_analpad           </td><td> 2226</td></tr>\n",
       "\t<tr><td>large_intestine_hindgut_analpad_2         </td><td> 1467</td></tr>\n",
       "\t<tr><td>lateral_cord_glia                         </td><td>  636</td></tr>\n",
       "\t<tr><td>uncertain_1                               </td><td>   59</td></tr>\n",
       "\t<tr><td>unknown_2                                 </td><td> 1629</td></tr>\n",
       "\t<tr><td>unknown_3                                 </td><td>  180</td></tr>\n",
       "\t<tr><td>NA                                        </td><td>28410</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 70 × 2\n",
       "\\begin{tabular}{ll}\n",
       " cell.type.3 & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Amnioserosa                                &  7091\\\\\n",
       "\t Analpad                                    &   850\\\\\n",
       "\t Analpad\\_large\\_intestine                    &   734\\\\\n",
       "\t Antennal\\_sense\\_organs                      &   748\\\\\n",
       "\t Bolwig\\_organ                               &   115\\\\\n",
       "\t Central\\_brain\\_ventral\\_nerve\\_cord\\_1         &  6846\\\\\n",
       "\t Central\\_brain\\_ventral\\_nerve\\_cord\\_2         & 10161\\\\\n",
       "\t Central\\_brain\\_ventral\\_nerve\\_cord\\_early     &  5058\\\\\n",
       "\t Central\\_brain\\_ventral\\_nerve\\_cord\\_primorium &  5819\\\\\n",
       "\t Cholinergic                                &  8949\\\\\n",
       "\t Chordotonal\\_organ                          &   121\\\\\n",
       "\t Chordotonal\\_sensory                        &   750\\\\\n",
       "\t Copper\\_cells                               &  4327\\\\\n",
       "\t Crystal\\_cell                               &   432\\\\\n",
       "\t Cuprophilic\\_cells                          &   581\\\\\n",
       "\t Digestive\\_system                           &  2212\\\\\n",
       "\t Embryonic\\_body\\_wall                        & 42392\\\\\n",
       "\t Epidermis                                  & 17230\\\\\n",
       "\t Epidermis\\_late                             &   184\\\\\n",
       "\t External\\_sensory                           &   647\\\\\n",
       "\t Fat\\_body                                   &  5694\\\\\n",
       "\t Garland\\_cell\\_nephrocyte                    &   536\\\\\n",
       "\t Germ\\_cells\\_ovaries\\_query                   &  2139\\\\\n",
       "\t Glutamatergic                              &  5107\\\\\n",
       "\t Late\\_cells                                 &  1565\\\\\n",
       "\t Longitudinal\\_visceral\\_muscle               &   617\\\\\n",
       "\t Malpighian tubule                          &  1731\\\\\n",
       "\t Maxillary\\_sensory\\_complex                  &   533\\\\\n",
       "\t Midgut                                     &  1674\\\\\n",
       "\t Midgut\\_4th\\_chamber                         &  2128\\\\\n",
       "\t Midgut\\_primordium                          &   689\\\\\n",
       "\t Monoaminergic                              &   454\\\\\n",
       "\t Multidendritic\\_sensory                     &  2069\\\\\n",
       "\t Muscle\\_system                              &  5347\\\\\n",
       "\t Neuron\\_unknown\\_1                           &   602\\\\\n",
       "\t Neuron\\_unknown\\_2                           &   225\\\\\n",
       "\t Oenocytes                                  &   142\\\\\n",
       "\t Oenocytes\\_older                            &   250\\\\\n",
       "\t Olfactory\\_neurons                          &  1864\\\\\n",
       "\t Peptidergic\\_neuron                         &   147\\\\\n",
       "\t Peripheral\\_nervous\\_system\\_early            &  3851\\\\\n",
       "\t Pharynx                                    &  1704\\\\\n",
       "\t Pharynx\\_2                                  &   975\\\\\n",
       "\t Pharynx\\_primordium\\_query                   &  1990\\\\\n",
       "\t Plasmatocytes                              &  6021\\\\\n",
       "\t Posterior\\_spiracle                         &  1436\\\\\n",
       "\t Proventriculus                             &  2423\\\\\n",
       "\t Proventriculus\\_older                       &   253\\\\\n",
       "\t Salivary\\_gland                             &  4237\\\\\n",
       "\t Somatic\\_muscle                             &  2766\\\\\n",
       "\t Somatic\\_muscle\\_primordium                  &  5139\\\\\n",
       "\t Tracheal\\_system\\_1                          &  1101\\\\\n",
       "\t Tracheal\\_system\\_2                          &  2642\\\\\n",
       "\t Tracheal\\_system\\_early                      &  1797\\\\\n",
       "\t Tracheal\\_system\\_older                      &  2821\\\\\n",
       "\t Undifferentiated\\_1                         &  3354\\\\\n",
       "\t Undifferentiated\\_2                         &  1700\\\\\n",
       "\t Undifferentiated\\_3                         &  2511\\\\\n",
       "\t Ventral\\_midline                            &  2159\\\\\n",
       "\t Visceral\\_mesoderm                          &  1495\\\\\n",
       "\t Visceral\\_muscle                            &  2308\\\\\n",
       "\t Visceral\\_muscle\\_2                          &  1508\\\\\n",
       "\t gut\\_query                                  &   887\\\\\n",
       "\t large\\_intestine\\_hindgut\\_analpad            &  2226\\\\\n",
       "\t large\\_intestine\\_hindgut\\_analpad\\_2          &  1467\\\\\n",
       "\t lateral\\_cord\\_glia                          &   636\\\\\n",
       "\t uncertain\\_1                                &    59\\\\\n",
       "\t unknown\\_2                                  &  1629\\\\\n",
       "\t unknown\\_3                                  &   180\\\\\n",
       "\t NA                                         & 28410\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 70 × 2\n",
       "\n",
       "| cell.type.3 &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Amnioserosa                                |  7091 |\n",
       "| Analpad                                    |   850 |\n",
       "| Analpad_large_intestine                    |   734 |\n",
       "| Antennal_sense_organs                      |   748 |\n",
       "| Bolwig_organ                               |   115 |\n",
       "| Central_brain_ventral_nerve_cord_1         |  6846 |\n",
       "| Central_brain_ventral_nerve_cord_2         | 10161 |\n",
       "| Central_brain_ventral_nerve_cord_early     |  5058 |\n",
       "| Central_brain_ventral_nerve_cord_primorium |  5819 |\n",
       "| Cholinergic                                |  8949 |\n",
       "| Chordotonal_organ                          |   121 |\n",
       "| Chordotonal_sensory                        |   750 |\n",
       "| Copper_cells                               |  4327 |\n",
       "| Crystal_cell                               |   432 |\n",
       "| Cuprophilic_cells                          |   581 |\n",
       "| Digestive_system                           |  2212 |\n",
       "| Embryonic_body_wall                        | 42392 |\n",
       "| Epidermis                                  | 17230 |\n",
       "| Epidermis_late                             |   184 |\n",
       "| External_sensory                           |   647 |\n",
       "| Fat_body                                   |  5694 |\n",
       "| Garland_cell_nephrocyte                    |   536 |\n",
       "| Germ_cells_ovaries_query                   |  2139 |\n",
       "| Glutamatergic                              |  5107 |\n",
       "| Late_cells                                 |  1565 |\n",
       "| Longitudinal_visceral_muscle               |   617 |\n",
       "| Malpighian tubule                          |  1731 |\n",
       "| Maxillary_sensory_complex                  |   533 |\n",
       "| Midgut                                     |  1674 |\n",
       "| Midgut_4th_chamber                         |  2128 |\n",
       "| Midgut_primordium                          |   689 |\n",
       "| Monoaminergic                              |   454 |\n",
       "| Multidendritic_sensory                     |  2069 |\n",
       "| Muscle_system                              |  5347 |\n",
       "| Neuron_unknown_1                           |   602 |\n",
       "| Neuron_unknown_2                           |   225 |\n",
       "| Oenocytes                                  |   142 |\n",
       "| Oenocytes_older                            |   250 |\n",
       "| Olfactory_neurons                          |  1864 |\n",
       "| Peptidergic_neuron                         |   147 |\n",
       "| Peripheral_nervous_system_early            |  3851 |\n",
       "| Pharynx                                    |  1704 |\n",
       "| Pharynx_2                                  |   975 |\n",
       "| Pharynx_primordium_query                   |  1990 |\n",
       "| Plasmatocytes                              |  6021 |\n",
       "| Posterior_spiracle                         |  1436 |\n",
       "| Proventriculus                             |  2423 |\n",
       "| Proventriculus_older                       |   253 |\n",
       "| Salivary_gland                             |  4237 |\n",
       "| Somatic_muscle                             |  2766 |\n",
       "| Somatic_muscle_primordium                  |  5139 |\n",
       "| Tracheal_system_1                          |  1101 |\n",
       "| Tracheal_system_2                          |  2642 |\n",
       "| Tracheal_system_early                      |  1797 |\n",
       "| Tracheal_system_older                      |  2821 |\n",
       "| Undifferentiated_1                         |  3354 |\n",
       "| Undifferentiated_2                         |  1700 |\n",
       "| Undifferentiated_3                         |  2511 |\n",
       "| Ventral_midline                            |  2159 |\n",
       "| Visceral_mesoderm                          |  1495 |\n",
       "| Visceral_muscle                            |  2308 |\n",
       "| Visceral_muscle_2                          |  1508 |\n",
       "| gut_query                                  |   887 |\n",
       "| large_intestine_hindgut_analpad            |  2226 |\n",
       "| large_intestine_hindgut_analpad_2          |  1467 |\n",
       "| lateral_cord_glia                          |   636 |\n",
       "| uncertain_1                                |    59 |\n",
       "| unknown_2                                  |  1629 |\n",
       "| unknown_3                                  |   180 |\n",
       "| NA                                         | 28410 |\n",
       "\n"
      ],
      "text/plain": [
       "   cell.type.3                                n    \n",
       "1  Amnioserosa                                 7091\n",
       "2  Analpad                                      850\n",
       "3  Analpad_large_intestine                      734\n",
       "4  Antennal_sense_organs                        748\n",
       "5  Bolwig_organ                                 115\n",
       "6  Central_brain_ventral_nerve_cord_1          6846\n",
       "7  Central_brain_ventral_nerve_cord_2         10161\n",
       "8  Central_brain_ventral_nerve_cord_early      5058\n",
       "9  Central_brain_ventral_nerve_cord_primorium  5819\n",
       "10 Cholinergic                                 8949\n",
       "11 Chordotonal_organ                            121\n",
       "12 Chordotonal_sensory                          750\n",
       "13 Copper_cells                                4327\n",
       "14 Crystal_cell                                 432\n",
       "15 Cuprophilic_cells                            581\n",
       "16 Digestive_system                            2212\n",
       "17 Embryonic_body_wall                        42392\n",
       "18 Epidermis                                  17230\n",
       "19 Epidermis_late                               184\n",
       "20 External_sensory                             647\n",
       "21 Fat_body                                    5694\n",
       "22 Garland_cell_nephrocyte                      536\n",
       "23 Germ_cells_ovaries_query                    2139\n",
       "24 Glutamatergic                               5107\n",
       "25 Late_cells                                  1565\n",
       "26 Longitudinal_visceral_muscle                 617\n",
       "27 Malpighian tubule                           1731\n",
       "28 Maxillary_sensory_complex                    533\n",
       "29 Midgut                                      1674\n",
       "30 Midgut_4th_chamber                          2128\n",
       "31 Midgut_primordium                            689\n",
       "32 Monoaminergic                                454\n",
       "33 Multidendritic_sensory                      2069\n",
       "34 Muscle_system                               5347\n",
       "35 Neuron_unknown_1                             602\n",
       "36 Neuron_unknown_2                             225\n",
       "37 Oenocytes                                    142\n",
       "38 Oenocytes_older                              250\n",
       "39 Olfactory_neurons                           1864\n",
       "40 Peptidergic_neuron                           147\n",
       "41 Peripheral_nervous_system_early             3851\n",
       "42 Pharynx                                     1704\n",
       "43 Pharynx_2                                    975\n",
       "44 Pharynx_primordium_query                    1990\n",
       "45 Plasmatocytes                               6021\n",
       "46 Posterior_spiracle                          1436\n",
       "47 Proventriculus                              2423\n",
       "48 Proventriculus_older                         253\n",
       "49 Salivary_gland                              4237\n",
       "50 Somatic_muscle                              2766\n",
       "51 Somatic_muscle_primordium                   5139\n",
       "52 Tracheal_system_1                           1101\n",
       "53 Tracheal_system_2                           2642\n",
       "54 Tracheal_system_early                       1797\n",
       "55 Tracheal_system_older                       2821\n",
       "56 Undifferentiated_1                          3354\n",
       "57 Undifferentiated_2                          1700\n",
       "58 Undifferentiated_3                          2511\n",
       "59 Ventral_midline                             2159\n",
       "60 Visceral_mesoderm                           1495\n",
       "61 Visceral_muscle                             2308\n",
       "62 Visceral_muscle_2                           1508\n",
       "63 gut_query                                    887\n",
       "64 large_intestine_hindgut_analpad             2226\n",
       "65 large_intestine_hindgut_analpad_2           1467\n",
       "66 lateral_cord_glia                            636\n",
       "67 uncertain_1                                   59\n",
       "68 unknown_2                                   1629\n",
       "69 unknown_3                                    180\n",
       "70 NA                                         28410"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_sub300_copy  %>% count(cell.type.3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## DE genes on new annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finding markers for cell.type a\n",
      "\n",
      "# of cells sampled in set 1: 528\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type aa\n",
      "\n",
      "# of cells sampled in set 1: 1798\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ab\n",
      "\n",
      "# of cells sampled in set 1: 1189\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ac\n",
      "\n",
      "# of cells sampled in set 1: 4237\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ad\n",
      "\n",
      "# of cells sampled in set 1: 850\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ae\n",
      "\n",
      "# of cells sampled in set 1: 1241\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type af\n",
      "\n",
      "# of cells sampled in set 1: 115\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ag\n",
      "\n",
      "# of cells sampled in set 1: 636\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ah\n",
      "\n",
      "# of cells sampled in set 1: 2766\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ai\n",
      "\n",
      "# of cells sampled in set 1: 1467\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type aj\n",
      "\n",
      "# of cells sampled in set 1: 2159\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ak\n",
      "\n",
      "# of cells sampled in set 1: 617\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type al\n",
      "\n",
      "# of cells sampled in set 1: 2821\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type am\n",
      "\n",
      "# of cells sampled in set 1: 1864\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type an\n",
      "\n",
      "# of cells sampled in set 1: 2043\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ao\n",
      "\n",
      "# of cells sampled in set 1: 607\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ap\n",
      "\n",
      "# of cells sampled in set 1: 2367\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type aq\n",
      "\n",
      "# of cells sampled in set 1: 1355\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ar\n",
      "\n",
      "# of cells sampled in set 1: 3851\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type as\n",
      "\n",
      "# of cells sampled in set 1: 1186\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type at\n",
      "\n",
      "# of cells sampled in set 1: 2128\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type au\n",
      "\n",
      "# of cells sampled in set 1: 1570\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type av\n",
      "\n",
      "# of cells sampled in set 1: 2139\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type aw\n",
      "\n",
      "# of cells sampled in set 1: 371\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ax\n",
      "\n",
      "# of cells sampled in set 1: 3940\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ay\n",
      "\n",
      "# of cells sampled in set 1: 5347\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type az\n",
      "\n",
      "# of cells sampled in set 1: 5819\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type b\n",
      "\n",
      "# of cells sampled in set 1: 1731\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ba\n",
      "\n",
      "# of cells sampled in set 1: 2212\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bb\n",
      "\n",
      "# of cells sampled in set 1: 1693\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bc\n",
      "\n",
      "# of cells sampled in set 1: 1495\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bd\n",
      "\n",
      "# of cells sampled in set 1: 1003\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type be\n",
      "\n",
      "# of cells sampled in set 1: 5139\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bf\n",
      "\n",
      "# of cells sampled in set 1: 1674\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bg\n",
      "\n",
      "# of cells sampled in set 1: 1565\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bh\n",
      "\n",
      "# of cells sampled in set 1: 3168\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bi\n",
      "\n",
      "# of cells sampled in set 1: 962\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bj\n",
      "\n",
      "# of cells sampled in set 1: 5058\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bk\n",
      "\n",
      "# of cells sampled in set 1: 6846\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bl\n",
      "\n",
      "# of cells sampled in set 1: 8547\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bm\n",
      "\n",
      "# of cells sampled in set 1: 3354\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bn\n",
      "\n",
      "# of cells sampled in set 1: 1700\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bo\n",
      "\n",
      "# of cells sampled in set 1: 2511\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bp\n",
      "\n",
      "# of cells sampled in set 1: 7476\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bq\n",
      "\n",
      "# of cells sampled in set 1: 9416\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type br\n",
      "\n",
      "# of cells sampled in set 1: 4103\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bs\n",
      "\n",
      "# of cells sampled in set 1: 4348\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bt\n",
      "\n",
      "# of cells sampled in set 1: 9401\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bu\n",
      "\n",
      "# of cells sampled in set 1: 3626\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bv\n",
      "\n",
      "# of cells sampled in set 1: 3444\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bw\n",
      "\n",
      "# of cells sampled in set 1: 3137\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bx\n",
      "\n",
      "# of cells sampled in set 1: 2876\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type by\n",
      "\n",
      "# of cells sampled in set 1: 2041\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type bz\n",
      "\n",
      "# of cells sampled in set 1: 2123\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type c\n",
      "\n",
      "# of cells sampled in set 1: 121\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ca\n",
      "\n",
      "# of cells sampled in set 1: 4430\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type cb\n",
      "\n",
      "# of cells sampled in set 1: 5016\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type cc\n",
      "\n",
      "# of cells sampled in set 1: 8699\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type cd\n",
      "\n",
      "# of cells sampled in set 1: 3002\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type ce\n",
      "\n",
      "# of cells sampled in set 1: 1377\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type cf\n",
      "\n",
      "# of cells sampled in set 1: 697\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type d\n",
      "\n",
      "# of cells sampled in set 1: 225\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type e\n",
      "\n",
      "# of cells sampled in set 1: 5537\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type f\n",
      "\n",
      "# of cells sampled in set 1: 1629\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type g\n",
      "\n",
      "# of cells sampled in set 1: 887\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type h\n",
      "\n",
      "# of cells sampled in set 1: 432\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type i\n",
      "\n",
      "# of cells sampled in set 1: 10353\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type j\n",
      "\n",
      "# of cells sampled in set 1: 184\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type k\n",
      "\n",
      "# of cells sampled in set 1: 59\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type l\n",
      "\n",
      "# of cells sampled in set 1: 1663\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type m\n",
      "\n",
      "# of cells sampled in set 1: 2880\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type n\n",
      "\n",
      "# of cells sampled in set 1: 2548\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type o\n",
      "\n",
      "# of cells sampled in set 1: 2423\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type p\n",
      "\n",
      "# of cells sampled in set 1: 147\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type q\n",
      "\n",
      "# of cells sampled in set 1: 10255\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type r\n",
      "\n",
      "# of cells sampled in set 1: 3801\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type s\n",
      "\n",
      "# of cells sampled in set 1: 4732\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type t\n",
      "\n",
      "# of cells sampled in set 1: 1056\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type u\n",
      "\n",
      "# of cells sampled in set 1: 610\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type v\n",
      "\n",
      "# of cells sampled in set 1: 6021\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type w\n",
      "\n",
      "# of cells sampled in set 1: 748\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type x\n",
      "\n",
      "# of cells sampled in set 1: 581\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type y\n",
      "\n",
      "# of cells sampled in set 1: 1571\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n",
      "Finding markers for cell.type z\n",
      "\n",
      "# of cells sampled in set 1: 1101\n",
      "\n",
      "# of cells sampled in set 2: 50000\n",
      "\n"
     ]
    }
   ],
   "source": [
    "DEG.results = lapply(sort((unique(df_sub300$cell.type.3))),\n",
    "                     function(this.cell.type) {\n",
    "    message(\"Finding markers for cell.type \", this.cell.type)\n",
    "    cbind(\n",
    "        two.set.differential.gene.test.3a(\n",
    "            cds_sub,\n",
    "            df_sub300$cell.type.3 == this.cell.type,\n",
    "            df_sub300$cell.type.3 != this.cell.type,\n",
    "            sample.n.max = 50000),\n",
    "        data.frame(cell.type.3 = this.cell.type))\n",
    "                         })"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "metadata": {},
   "outputs": [],
   "source": [
    "cell.type.3.markers = do.call(rbind, DEG.results)  %>%\n",
    "    select(\n",
    "        cell.type.3 = cell.type.3, gene, \n",
    "        cell.type.3.n.umi = set.1.n.umi,\n",
    "        other.n.umi = set.2.n.umi,\n",
    "        cell.type.3.tpm = set.1.tpm,\n",
    "        other.tpm = set.2.tpm,\n",
    "        log2.ratio, heuristic.score) %>%\n",
    "    arrange(-heuristic.score)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell.type.2</th><th scope=col>gene</th><th scope=col>cell.type.2.n.umi</th><th scope=col>other.n.umi</th><th scope=col>cell.type.2.tpm</th><th scope=col>other.tpm</th><th scope=col>log2.ratio</th><th scope=col>heuristic.score</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Muc26B</th><td>a</td><td>Muc26B     </td><td> 3993</td><td> 111</td><td>10368.322</td><td>  4.1576480</td><td>10.973182</td><td>146.3826</td></tr>\n",
       "\t<tr><th scope=row>CG740954</th><td>c</td><td>CG7409     </td><td>  600</td><td>  24</td><td> 4121.228</td><td>  0.4739611</td><td>11.449160</td><td>137.4954</td></tr>\n",
       "\t<tr><th scope=row>CG1137061</th><td>d</td><td>CG11370    </td><td> 3708</td><td> 446</td><td>15737.252</td><td> 17.5042494</td><td> 9.732111</td><td>135.6850</td></tr>\n",
       "\t<tr><th scope=row>CG746563</th><td>f</td><td>CG7465     </td><td>23068</td><td> 761</td><td>18549.567</td><td> 23.9239377</td><td> 9.539638</td><td>135.2642</td></tr>\n",
       "\t<tr><th scope=row>CG7017</th><td>a</td><td>CG7017     </td><td> 3759</td><td> 265</td><td>11685.814</td><td> 10.6958361</td><td> 9.964547</td><td>134.6469</td></tr>\n",
       "\t<tr><th scope=row>CG1135063</th><td>f</td><td>CG11350    </td><td> 8044</td><td> 159</td><td> 6300.220</td><td>  4.2137333</td><td>10.238870</td><td>129.2290</td></tr>\n",
       "\t<tr><th scope=row>Muc96D</th><td>a</td><td>Muc96D     </td><td> 2590</td><td> 118</td><td> 6427.809</td><td>  5.1071969</td><td>10.039601</td><td>127.0043</td></tr>\n",
       "\t<tr><th scope=row>CG1313354</th><td>c</td><td>CG13133    </td><td>  405</td><td>  14</td><td> 2679.090</td><td>  0.1829864</td><td>11.145094</td><td>126.9211</td></tr>\n",
       "\t<tr><th scope=row>CG1456661</th><td>d</td><td>CG14566    </td><td>  518</td><td>  60</td><td> 3985.539</td><td>  1.6730948</td><td>10.542048</td><td>126.0926</td></tr>\n",
       "\t<tr><th scope=row>CG3260363</th><td>f</td><td>CG32603    </td><td>10610</td><td> 353</td><td> 9231.304</td><td> 11.4313571</td><td> 9.536407</td><td>125.6181</td></tr>\n",
       "\t<tr><th scope=row>CG4314954</th><td>c</td><td>CG43149    </td><td>  527</td><td>  40</td><td> 3407.622</td><td>  1.1778353</td><td>10.611655</td><td>124.5275</td></tr>\n",
       "\t<tr><th scope=row>Nplp461</th><td>d</td><td>Nplp4      </td><td>  585</td><td>  98</td><td> 3518.479</td><td>  1.4273695</td><td>10.501343</td><td>123.7179</td></tr>\n",
       "\t<tr><th scope=row>Pde1c65</th><td>h</td><td>Pde1c      </td><td> 6587</td><td>4496</td><td>42193.293</td><td>197.9561181</td><td> 7.728420</td><td>118.7453</td></tr>\n",
       "\t<tr><th scope=row>Jon99Cii64</th><td>g</td><td>Jon99Cii   </td><td> 7795</td><td> 347</td><td> 7722.641</td><td> 12.5731343</td><td> 9.152197</td><td>118.2012</td></tr>\n",
       "\t<tr><th scope=row>CG1442163</th><td>f</td><td>CG14421    </td><td> 4232</td><td>  61</td><td> 3341.966</td><td>  2.2776488</td><td> 9.993820</td><td>116.9968</td></tr>\n",
       "\t<tr><th scope=row>CG1158167</th><td>j</td><td>CG11581    </td><td>  283</td><td>  36</td><td> 2455.322</td><td>  0.9543971</td><td>10.294973</td><td>115.9449</td></tr>\n",
       "\t<tr><th scope=row>CG17826</th><td>a</td><td>CG17826    </td><td> 1107</td><td>  41</td><td> 2902.487</td><td>  1.7851269</td><td>10.025331</td><td>115.3271</td></tr>\n",
       "\t<tr><th scope=row>alphaTub85E54</th><td>c</td><td>alphaTub85E</td><td> 3378</td><td>2390</td><td>21520.691</td><td> 82.5317555</td><td> 8.009184</td><td>115.2802</td></tr>\n",
       "\t<tr><th scope=row>CG1818064</th><td>g</td><td>CG18180    </td><td> 4374</td><td> 104</td><td> 3639.959</td><td>  3.3577182</td><td> 9.706134</td><td>114.8246</td></tr>\n",
       "\t<tr><th scope=row>CG1141361</th><td>d</td><td>CG11413    </td><td>  908</td><td>  67</td><td> 3428.035</td><td>  2.9631017</td><td> 9.756536</td><td>114.5767</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & cell.type.2 & gene & cell.type.2.n.umi & other.n.umi & cell.type.2.tpm & other.tpm & log2.ratio & heuristic.score\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tMuc26B & a & Muc26B      &  3993 &  111 & 10368.322 &   4.1576480 & 10.973182 & 146.3826\\\\\n",
       "\tCG740954 & c & CG7409      &   600 &   24 &  4121.228 &   0.4739611 & 11.449160 & 137.4954\\\\\n",
       "\tCG1137061 & d & CG11370     &  3708 &  446 & 15737.252 &  17.5042494 &  9.732111 & 135.6850\\\\\n",
       "\tCG746563 & f & CG7465      & 23068 &  761 & 18549.567 &  23.9239377 &  9.539638 & 135.2642\\\\\n",
       "\tCG7017 & a & CG7017      &  3759 &  265 & 11685.814 &  10.6958361 &  9.964547 & 134.6469\\\\\n",
       "\tCG1135063 & f & CG11350     &  8044 &  159 &  6300.220 &   4.2137333 & 10.238870 & 129.2290\\\\\n",
       "\tMuc96D & a & Muc96D      &  2590 &  118 &  6427.809 &   5.1071969 & 10.039601 & 127.0043\\\\\n",
       "\tCG1313354 & c & CG13133     &   405 &   14 &  2679.090 &   0.1829864 & 11.145094 & 126.9211\\\\\n",
       "\tCG1456661 & d & CG14566     &   518 &   60 &  3985.539 &   1.6730948 & 10.542048 & 126.0926\\\\\n",
       "\tCG3260363 & f & CG32603     & 10610 &  353 &  9231.304 &  11.4313571 &  9.536407 & 125.6181\\\\\n",
       "\tCG4314954 & c & CG43149     &   527 &   40 &  3407.622 &   1.1778353 & 10.611655 & 124.5275\\\\\n",
       "\tNplp461 & d & Nplp4       &   585 &   98 &  3518.479 &   1.4273695 & 10.501343 & 123.7179\\\\\n",
       "\tPde1c65 & h & Pde1c       &  6587 & 4496 & 42193.293 & 197.9561181 &  7.728420 & 118.7453\\\\\n",
       "\tJon99Cii64 & g & Jon99Cii    &  7795 &  347 &  7722.641 &  12.5731343 &  9.152197 & 118.2012\\\\\n",
       "\tCG1442163 & f & CG14421     &  4232 &   61 &  3341.966 &   2.2776488 &  9.993820 & 116.9968\\\\\n",
       "\tCG1158167 & j & CG11581     &   283 &   36 &  2455.322 &   0.9543971 & 10.294973 & 115.9449\\\\\n",
       "\tCG17826 & a & CG17826     &  1107 &   41 &  2902.487 &   1.7851269 & 10.025331 & 115.3271\\\\\n",
       "\talphaTub85E54 & c & alphaTub85E &  3378 & 2390 & 21520.691 &  82.5317555 &  8.009184 & 115.2802\\\\\n",
       "\tCG1818064 & g & CG18180     &  4374 &  104 &  3639.959 &   3.3577182 &  9.706134 & 114.8246\\\\\n",
       "\tCG1141361 & d & CG11413     &   908 &   67 &  3428.035 &   2.9631017 &  9.756536 & 114.5767\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 × 8\n",
       "\n",
       "| <!--/--> | cell.type.2 &lt;chr&gt; | gene &lt;chr&gt; | cell.type.2.n.umi &lt;dbl&gt; | other.n.umi &lt;dbl&gt; | cell.type.2.tpm &lt;dbl&gt; | other.tpm &lt;dbl&gt; | log2.ratio &lt;dbl&gt; | heuristic.score &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Muc26B | a | Muc26B      |  3993 |  111 | 10368.322 |   4.1576480 | 10.973182 | 146.3826 |\n",
       "| CG740954 | c | CG7409      |   600 |   24 |  4121.228 |   0.4739611 | 11.449160 | 137.4954 |\n",
       "| CG1137061 | d | CG11370     |  3708 |  446 | 15737.252 |  17.5042494 |  9.732111 | 135.6850 |\n",
       "| CG746563 | f | CG7465      | 23068 |  761 | 18549.567 |  23.9239377 |  9.539638 | 135.2642 |\n",
       "| CG7017 | a | CG7017      |  3759 |  265 | 11685.814 |  10.6958361 |  9.964547 | 134.6469 |\n",
       "| CG1135063 | f | CG11350     |  8044 |  159 |  6300.220 |   4.2137333 | 10.238870 | 129.2290 |\n",
       "| Muc96D | a | Muc96D      |  2590 |  118 |  6427.809 |   5.1071969 | 10.039601 | 127.0043 |\n",
       "| CG1313354 | c | CG13133     |   405 |   14 |  2679.090 |   0.1829864 | 11.145094 | 126.9211 |\n",
       "| CG1456661 | d | CG14566     |   518 |   60 |  3985.539 |   1.6730948 | 10.542048 | 126.0926 |\n",
       "| CG3260363 | f | CG32603     | 10610 |  353 |  9231.304 |  11.4313571 |  9.536407 | 125.6181 |\n",
       "| CG4314954 | c | CG43149     |   527 |   40 |  3407.622 |   1.1778353 | 10.611655 | 124.5275 |\n",
       "| Nplp461 | d | Nplp4       |   585 |   98 |  3518.479 |   1.4273695 | 10.501343 | 123.7179 |\n",
       "| Pde1c65 | h | Pde1c       |  6587 | 4496 | 42193.293 | 197.9561181 |  7.728420 | 118.7453 |\n",
       "| Jon99Cii64 | g | Jon99Cii    |  7795 |  347 |  7722.641 |  12.5731343 |  9.152197 | 118.2012 |\n",
       "| CG1442163 | f | CG14421     |  4232 |   61 |  3341.966 |   2.2776488 |  9.993820 | 116.9968 |\n",
       "| CG1158167 | j | CG11581     |   283 |   36 |  2455.322 |   0.9543971 | 10.294973 | 115.9449 |\n",
       "| CG17826 | a | CG17826     |  1107 |   41 |  2902.487 |   1.7851269 | 10.025331 | 115.3271 |\n",
       "| alphaTub85E54 | c | alphaTub85E |  3378 | 2390 | 21520.691 |  82.5317555 |  8.009184 | 115.2802 |\n",
       "| CG1818064 | g | CG18180     |  4374 |  104 |  3639.959 |   3.3577182 |  9.706134 | 114.8246 |\n",
       "| CG1141361 | d | CG11413     |   908 |   67 |  3428.035 |   2.9631017 |  9.756536 | 114.5767 |\n",
       "\n"
      ],
      "text/plain": [
       "              cell.type.2 gene        cell.type.2.n.umi other.n.umi\n",
       "Muc26B        a           Muc26B       3993              111       \n",
       "CG740954      c           CG7409        600               24       \n",
       "CG1137061     d           CG11370      3708              446       \n",
       "CG746563      f           CG7465      23068              761       \n",
       "CG7017        a           CG7017       3759              265       \n",
       "CG1135063     f           CG11350      8044              159       \n",
       "Muc96D        a           Muc96D       2590              118       \n",
       "CG1313354     c           CG13133       405               14       \n",
       "CG1456661     d           CG14566       518               60       \n",
       "CG3260363     f           CG32603     10610              353       \n",
       "CG4314954     c           CG43149       527               40       \n",
       "Nplp461       d           Nplp4         585               98       \n",
       "Pde1c65       h           Pde1c        6587             4496       \n",
       "Jon99Cii64    g           Jon99Cii     7795              347       \n",
       "CG1442163     f           CG14421      4232               61       \n",
       "CG1158167     j           CG11581       283               36       \n",
       "CG17826       a           CG17826      1107               41       \n",
       "alphaTub85E54 c           alphaTub85E  3378             2390       \n",
       "CG1818064     g           CG18180      4374              104       \n",
       "CG1141361     d           CG11413       908               67       \n",
       "              cell.type.2.tpm other.tpm   log2.ratio heuristic.score\n",
       "Muc26B        10368.322         4.1576480 10.973182  146.3826       \n",
       "CG740954       4121.228         0.4739611 11.449160  137.4954       \n",
       "CG1137061     15737.252        17.5042494  9.732111  135.6850       \n",
       "CG746563      18549.567        23.9239377  9.539638  135.2642       \n",
       "CG7017        11685.814        10.6958361  9.964547  134.6469       \n",
       "CG1135063      6300.220         4.2137333 10.238870  129.2290       \n",
       "Muc96D         6427.809         5.1071969 10.039601  127.0043       \n",
       "CG1313354      2679.090         0.1829864 11.145094  126.9211       \n",
       "CG1456661      3985.539         1.6730948 10.542048  126.0926       \n",
       "CG3260363      9231.304        11.4313571  9.536407  125.6181       \n",
       "CG4314954      3407.622         1.1778353 10.611655  124.5275       \n",
       "Nplp461        3518.479         1.4273695 10.501343  123.7179       \n",
       "Pde1c65       42193.293       197.9561181  7.728420  118.7453       \n",
       "Jon99Cii64     7722.641        12.5731343  9.152197  118.2012       \n",
       "CG1442163      3341.966         2.2776488  9.993820  116.9968       \n",
       "CG1158167      2455.322         0.9543971 10.294973  115.9449       \n",
       "CG17826        2902.487         1.7851269 10.025331  115.3271       \n",
       "alphaTub85E54 21520.691        82.5317555  8.009184  115.2802       \n",
       "CG1818064      3639.959         3.3577182  9.706134  114.8246       \n",
       "CG1141361      3428.035         2.9631017  9.756536  114.5767       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(cell.type.3.markers, 20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>399</li><li>8</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 399\n",
       "\\item 8\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 399\n",
       "2. 8\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 399   8"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.3.markers  %>% filter(heuristic.score > 60)  %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 280,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>10</li><li>8</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 10\n",
       "\\item 8\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 10\n",
       "2. 8\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 10  8"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.3.markers  %>% filter(heuristic.score > 60 & gene %in% fly_TFs)  %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 284,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>96</li><li>8</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 96\n",
       "\\item 8\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 96\n",
       "2. 8\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 96  8"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cell.type.3.markers  %>% filter(heuristic.score > 40 & gene %in% fly_TFs)  %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'a'</li><li>'c'</li><li>'d'</li><li>'f'</li><li>'h'</li><li>'g'</li><li>'j'</li><li>'k'</li><li>'o'</li><li>'l'</li><li>'b'</li><li>'ac'</li><li>'p'</li><li>'t'</li><li>'u'</li><li>'x'</li><li>'z'</li><li>'w'</li><li>'v'</li><li>'m'</li><li>'aw'</li><li>'ad'</li><li>'ag'</li><li>'n'</li><li>'af'</li><li>'y'</li><li>'al'</li><li>'ai'</li><li>'ak'</li><li>'aa'</li><li>'as'</li><li>'ab'</li><li>'ao'</li><li>'aq'</li><li>'bc'</li><li>'ce'</li><li>'am'</li><li>'aj'</li><li>'ah'</li><li>'s'</li><li>'cd'</li><li>'ar'</li><li>'au'</li><li>'at'</li><li>'bj'</li><li>'cb'</li><li>'av'</li><li>'ca'</li><li>'ae'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'a'\n",
       "\\item 'c'\n",
       "\\item 'd'\n",
       "\\item 'f'\n",
       "\\item 'h'\n",
       "\\item 'g'\n",
       "\\item 'j'\n",
       "\\item 'k'\n",
       "\\item 'o'\n",
       "\\item 'l'\n",
       "\\item 'b'\n",
       "\\item 'ac'\n",
       "\\item 'p'\n",
       "\\item 't'\n",
       "\\item 'u'\n",
       "\\item 'x'\n",
       "\\item 'z'\n",
       "\\item 'w'\n",
       "\\item 'v'\n",
       "\\item 'm'\n",
       "\\item 'aw'\n",
       "\\item 'ad'\n",
       "\\item 'ag'\n",
       "\\item 'n'\n",
       "\\item 'af'\n",
       "\\item 'y'\n",
       "\\item 'al'\n",
       "\\item 'ai'\n",
       "\\item 'ak'\n",
       "\\item 'aa'\n",
       "\\item 'as'\n",
       "\\item 'ab'\n",
       "\\item 'ao'\n",
       "\\item 'aq'\n",
       "\\item 'bc'\n",
       "\\item 'ce'\n",
       "\\item 'am'\n",
       "\\item 'aj'\n",
       "\\item 'ah'\n",
       "\\item 's'\n",
       "\\item 'cd'\n",
       "\\item 'ar'\n",
       "\\item 'au'\n",
       "\\item 'at'\n",
       "\\item 'bj'\n",
       "\\item 'cb'\n",
       "\\item 'av'\n",
       "\\item 'ca'\n",
       "\\item 'ae'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'a'\n",
       "2. 'c'\n",
       "3. 'd'\n",
       "4. 'f'\n",
       "5. 'h'\n",
       "6. 'g'\n",
       "7. 'j'\n",
       "8. 'k'\n",
       "9. 'o'\n",
       "10. 'l'\n",
       "11. 'b'\n",
       "12. 'ac'\n",
       "13. 'p'\n",
       "14. 't'\n",
       "15. 'u'\n",
       "16. 'x'\n",
       "17. 'z'\n",
       "18. 'w'\n",
       "19. 'v'\n",
       "20. 'm'\n",
       "21. 'aw'\n",
       "22. 'ad'\n",
       "23. 'ag'\n",
       "24. 'n'\n",
       "25. 'af'\n",
       "26. 'y'\n",
       "27. 'al'\n",
       "28. 'ai'\n",
       "29. 'ak'\n",
       "30. 'aa'\n",
       "31. 'as'\n",
       "32. 'ab'\n",
       "33. 'ao'\n",
       "34. 'aq'\n",
       "35. 'bc'\n",
       "36. 'ce'\n",
       "37. 'am'\n",
       "38. 'aj'\n",
       "39. 'ah'\n",
       "40. 's'\n",
       "41. 'cd'\n",
       "42. 'ar'\n",
       "43. 'au'\n",
       "44. 'at'\n",
       "45. 'bj'\n",
       "46. 'cb'\n",
       "47. 'av'\n",
       "48. 'ca'\n",
       "49. 'ae'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"a\"  \"c\"  \"d\"  \"f\"  \"h\"  \"g\"  \"j\"  \"k\"  \"o\"  \"l\"  \"b\"  \"ac\" \"p\"  \"t\"  \"u\" \n",
       "[16] \"x\"  \"z\"  \"w\"  \"v\"  \"m\"  \"aw\" \"ad\" \"ag\" \"n\"  \"af\" \"y\"  \"al\" \"ai\" \"ak\" \"aa\"\n",
       "[31] \"as\" \"ab\" \"ao\" \"aq\" \"bc\" \"ce\" \"am\" \"aj\" \"ah\" \"s\"  \"cd\" \"ar\" \"au\" \"at\" \"bj\"\n",
       "[46] \"cb\" \"av\" \"ca\" \"ae\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique((cell.type.3.markers  %>% filter(heuristic.score > 60))$cell.type.3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "49"
      ],
      "text/latex": [
       "49"
      ],
      "text/markdown": [
       "49"
      ],
      "text/plain": [
       "[1] 49"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(unique((cell.type.3.markers  %>% filter(heuristic.score > 60))$cell.type.3))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### umap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 793,
   "metadata": {},
   "outputs": [],
   "source": [
    "colData(cds_sub)$cell.type.3 = NULL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 794,
   "metadata": {},
   "outputs": [],
   "source": [
    "colData(cds_sub)$cell.type.3 = rep(NA, ncol(cds_sub))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1791,
   "metadata": {},
   "outputs": [],
   "source": [
    "colData(cds_sub)$cell.type.3 = df_sub300$cell.type.3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No trajectory to plot. Has learn_graph() been called yet?\n",
      "\n"
     ]
    }
   ],
   "source": [
    "plot_cells(cds_sub, color_cells_by = \"cell.type.3\", label_groups_by_cluster = FALSE,  \n",
    "           cell_size = 0.1,group_label_size = 5) + \n",
    "# theme_classic() +\n",
    "theme(text = element_text(color=\"black\", size=20, face=\"bold\")) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 325,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot_cells(cds_sub, color_cells_by = \"cell.type.3\", label_groups_by_cluster = FALSE, : could not find function \"plot_cells\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot_cells(cds_sub, color_cells_by = \"cell.type.3\", label_groups_by_cluster = FALSE, : could not find function \"plot_cells\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot_cells(cds_sub, color_cells_by = \"cell.type.3\", label_groups_by_cluster = FALSE,  \n",
    "           cell_size = 0.1,group_label_size = 5) + \n",
    "# theme_classic() +\n",
    "theme(text = element_text(color=\"black\", size=20, face=\"bold\")) +\n",
    "xlim(0, 7.5) + ylim(-7.5, 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 326,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot_cells(cds_sub, color_cells_by = \"cell.type.3\", label_groups_by_cluster = FALSE, : could not find function \"plot_cells\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot_cells(cds_sub, color_cells_by = \"cell.type.3\", label_groups_by_cluster = FALSE, : could not find function \"plot_cells\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot_cells(cds_sub, color_cells_by = \"cell.type.3\", label_groups_by_cluster = FALSE,  \n",
    "           cell_size = 0.1,group_label_size = 5) + \n",
    "# theme_classic() +\n",
    "theme(text = element_text(color=\"black\", size=20, face=\"bold\")) +\n",
    "xlim(0, 7.5) + ylim(0, 7.5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 327,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot_cells(cds_sub, color_cells_by = \"cell.type.3\", label_groups_by_cluster = FALSE, : could not find function \"plot_cells\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot_cells(cds_sub, color_cells_by = \"cell.type.3\", label_groups_by_cluster = FALSE, : could not find function \"plot_cells\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot_cells(cds_sub, color_cells_by = \"cell.type.3\", label_groups_by_cluster = FALSE,  \n",
    "           cell_size = 0.1,group_label_size = 5) + \n",
    "# theme_classic() +\n",
    "theme(text = element_text(color=\"black\", size=20, face=\"bold\")) +\n",
    "xlim(-7.5, 0) + ylim(-7.5, 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 328,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot_cells(cds_sub, color_cells_by = \"cell.type.3\", label_groups_by_cluster = FALSE, : could not find function \"plot_cells\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot_cells(cds_sub, color_cells_by = \"cell.type.3\", label_groups_by_cluster = FALSE, : could not find function \"plot_cells\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot_cells(cds_sub, color_cells_by = \"cell.type.3\", label_groups_by_cluster = FALSE,  \n",
    "           cell_size = 0.1,group_label_size = 5) + \n",
    "# theme_classic() +\n",
    "theme(text = element_text(color=\"black\", size=20, face=\"bold\")) +\n",
    "xlim(-8, 0) + ylim(0, 8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1599,
   "metadata": {},
   "outputs": [],
   "source": [
    "options(repr.plot.width=28, repr.plot.height=16)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 329,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot_cells(cds_sub, color_cells_by = \"cell.type.3\", label_groups_by_cluster = FALSE, : could not find function \"plot_cells\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot_cells(cds_sub, color_cells_by = \"cell.type.3\", label_groups_by_cluster = FALSE, : could not find function \"plot_cells\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot_cells(cds_sub, color_cells_by = \"cell.type.3\", label_groups_by_cluster = FALSE,  \n",
    "           cell_size = 0.1,group_label_size = 5) + \n",
    "# theme_classic() +\n",
    "theme(text = element_text(color=\"black\", size=20, face=\"bold\")) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 330,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot_cells(cds_sub, color_cells_by = \"inferred_time_window\", : could not find function \"plot_cells\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot_cells(cds_sub, color_cells_by = \"inferred_time_window\", : could not find function \"plot_cells\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot_cells(cds_sub, color_cells_by = \"inferred_time_window\", cell_size = 0.05, group_label_size = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "metadata": {},
   "outputs": [],
   "source": [
    "saveRDS(cds_sub, \"data/cds_fly_sub_PCA300_older_cells_reannotated.rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {},
   "outputs": [],
   "source": [
    "saveRDS(cell.type.2.markers, \"data/fly_sub300_cell.type.2.markers\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 331,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_sub300, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "ggplot(df_sub300, aes(x = umap_1, y = umap_2)) +\n",
    "        geom_point(aes(color = cell.type.3), size = 0.1, stroke = 0.03) +\n",
    "        geom_label(aes(label = cell.type.3), color = \"black\", size = 2.5, data = df_sub300 %>%\n",
    "            group_by(cell.type.3) %>%\n",
    "            summarize(umap_1 = median(umap_1), umap_2 = median(umap_2))) +\n",
    "        guides(color = F) +\n",
    "# geom_text_repel() +\n",
    "        monocle3:::monocle_theme_opts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## check Lou's fuzzy clustering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1039,
   "metadata": {},
   "outputs": [],
   "source": [
    "cds_n21 = readRDS(\"data/cds_fly_pca300_no_doubletons.nn21.rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1042,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1mRows: \u001b[22m\u001b[34m216570\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m5\u001b[39m\n",
      "\u001b[36m──\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \"\\t\"\n",
      "\u001b[31mchr\u001b[39m (1): cell\n",
      "\u001b[32mdbl\u001b[39m (4): z1, x, y, pct\n",
      "\n",
      "\u001b[36mℹ\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36mℹ\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n"
     ]
    }
   ],
   "source": [
    "fuzzy_percent = read_tsv(\"data/cds_fly_pca300_no_doubletons.nn21.fuzzy.tsv\") "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1043,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_n21 = make_df(cds_n21)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1044,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>216570</li><li>33</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 216570\n",
       "\\item 33\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 216570\n",
       "2. 33\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 216570     33"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>216570</li><li>5</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 216570\n",
       "\\item 5\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 216570\n",
       "2. 5\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 216570      5"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(df_n21)\n",
    "dim(fuzzy_percent)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1045,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>cell</th><th scope=col>z1</th><th scope=col>x</th><th scope=col>y</th><th scope=col>pct</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td> 41</td><td>-4.1837888</td><td> 0.4881140</td><td>0.26496218</td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA</td><td>164</td><td>-3.0353165</td><td>-2.7561928</td><td>0.15352179</td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG</td><td> 22</td><td>-6.3859172</td><td> 3.5901138</td><td>0.01011936</td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA</td><td> 80</td><td>-3.1897362</td><td>-2.6267887</td><td>0.11677301</td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA</td><td>141</td><td>-0.3154551</td><td> 1.6942367</td><td>0.01940591</td></tr>\n",
       "\t<tr><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA</td><td>152</td><td>-6.8989954</td><td> 0.6930812</td><td>0.51285236</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " cell & z1 & x & y & pct\\\\\n",
       " <chr> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT &  41 & -4.1837888 &  0.4881140 & 0.26496218\\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA & 164 & -3.0353165 & -2.7561928 & 0.15352179\\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG &  22 & -6.3859172 &  3.5901138 & 0.01011936\\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA &  80 & -3.1897362 & -2.6267887 & 0.11677301\\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA & 141 & -0.3154551 &  1.6942367 & 0.01940591\\\\\n",
       "\t exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA & 152 & -6.8989954 &  0.6930812 & 0.51285236\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 5\n",
       "\n",
       "| cell &lt;chr&gt; | z1 &lt;dbl&gt; | x &lt;dbl&gt; | y &lt;dbl&gt; | pct &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT |  41 | -4.1837888 |  0.4881140 | 0.26496218 |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA | 164 | -3.0353165 | -2.7561928 | 0.15352179 |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG |  22 | -6.3859172 |  3.5901138 | 0.01011936 |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA |  80 | -3.1897362 | -2.6267887 | 0.11677301 |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA | 141 | -0.3154551 |  1.6942367 | 0.01940591 |\n",
       "| exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA | 152 | -6.8989954 |  0.6930812 | 0.51285236 |\n",
       "\n"
      ],
      "text/plain": [
       "  cell                                                   z1  x         \n",
       "1 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT  41 -4.1837888\n",
       "2 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA 164 -3.0353165\n",
       "3 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG  22 -6.3859172\n",
       "4 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA  80 -3.1897362\n",
       "5 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA 141 -0.3154551\n",
       "6 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA 152 -6.8989954\n",
       "  y          pct       \n",
       "1  0.4881140 0.26496218\n",
       "2 -2.7561928 0.15352179\n",
       "3  3.5901138 0.01011936\n",
       "4 -2.6267887 0.11677301\n",
       "5  1.6942367 0.01940591\n",
       "6  0.6930812 0.51285236"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(fuzzy_percent)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1048,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 34</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>orig.ident</th><th scope=col>nCount_RNA</th><th scope=col>nFeature_RNA</th><th scope=col>rt_bc</th><th scope=col>cell</th><th scope=col>time</th><th scope=col>percent.mt</th><th scope=col>percent.ribo</th><th scope=col>barn_dro</th><th scope=col>barn_hu</th><th scope=col>barn_mus</th><th scope=col>RNA_snn_res.0.8</th><th scope=col>doublet_score</th><th scope=col>predicted_doublet</th><th scope=col>doublet_cluster</th><th scope=col>seurat_clusters</th><th scope=col>doublet_subcluster</th><th scope=col>RNA_snn_res.0.08</th><th scope=col>lasso_age</th><th scope=col>NNv1_age</th><th scope=col>NNv2_age</th><th scope=col>lasso_shift</th><th scope=col>NNv1_shift</th><th scope=col>NNv2_shift</th><th scope=col>annot</th><th scope=col>ident</th><th scope=col>inferred_time_window</th><th scope=col>cell.type_time.bin</th><th scope=col>Size_Factor</th><th scope=col>cell.type.2</th><th scope=col>cluster</th><th scope=col>umap_1</th><th scope=col>umap_2</th><th scope=col>pct</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>exp1</td><td>346</td><td>247</td><td>AAGTTGCCAT</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT</td><td>hrs_06_10</td><td>1.4450867</td><td>4.3352601</td><td>371</td><td>0</td><td>0</td><td>16</td><td>0.05276064</td><td>Singlet</td><td>FALSE</td><td>21</td><td>FALSE</td><td>21</td><td> 8.402794</td><td> 7.885203</td><td> 8.119637</td><td> 0.0000000</td><td> 0.0000000</td><td> 0.000000</td><td>sensory nervous system primordium</td><td>21</td><td>6-8  </td><td>sensory nervous system primordium:6-8</td><td>0.8108612</td><td>v </td><td>89 </td><td>-2.0067012</td><td> 1.2184464</td><td>0.26496218</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>exp1</td><td>586</td><td>425</td><td>CAGGTATGGA</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA</td><td>hrs_08_12</td><td>1.0238908</td><td>6.4846416</td><td>630</td><td>0</td><td>0</td><td>4 </td><td>0.09008910</td><td>Singlet</td><td>FALSE</td><td>11</td><td>FALSE</td><td>11</td><td>11.266422</td><td>10.062633</td><td>11.156040</td><td> 0.0000000</td><td> 0.0000000</td><td> 0.000000</td><td>brain                            </td><td>11</td><td>10-12</td><td>brain:10-12                          </td><td>1.4059888</td><td>bl</td><td>22 </td><td>-3.6831634</td><td> 2.8012645</td><td>0.15352179</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>exp1</td><td>292</td><td>202</td><td>AAGTAGTCAG</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG</td><td>hrs_10_14</td><td>0.3424658</td><td>0.6849315</td><td>320</td><td>3</td><td>0</td><td>23</td><td>0.06585398</td><td>Singlet</td><td>FALSE</td><td>12</td><td>FALSE</td><td>12</td><td>12.544636</td><td>12.211529</td><td>13.830982</td><td> 0.0000000</td><td> 0.0000000</td><td> 0.000000</td><td>epidermis                        </td><td>12</td><td>12-14</td><td>epidermis:12-14                      </td><td>0.6967951</td><td>h </td><td>165</td><td>-1.3580737</td><td> 0.3582040</td><td>0.01011936</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>exp1</td><td>335</td><td>243</td><td>TGAGGTAGAA</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA</td><td>hrs_14_18</td><td>4.4776119</td><td>5.6716418</td><td>386</td><td>0</td><td>0</td><td>8 </td><td>0.08478501</td><td>Singlet</td><td>FALSE</td><td>11</td><td>FALSE</td><td>11</td><td>12.542527</td><td>13.739003</td><td>12.519031</td><td>-1.4574728</td><td>-0.2609968</td><td>-1.480969</td><td>sensory nervous system           </td><td>11</td><td>12-14</td><td>sensory nervous system:12-14         </td><td>0.7811049</td><td>bk</td><td>102</td><td>-3.8563070</td><td> 3.3158269</td><td>0.11677301</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>exp1</td><td>354</td><td>272</td><td>TCGATATTGA</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA</td><td>hrs_14_18</td><td>0.5649718</td><td>1.6949153</td><td>380</td><td>0</td><td>1</td><td>8 </td><td>0.08257165</td><td>Singlet</td><td>FALSE</td><td>11</td><td>FALSE</td><td>11</td><td>13.458162</td><td>15.126904</td><td>14.558554</td><td>-0.5418379</td><td> 0.0000000</td><td> 0.000000</td><td>sensory nervous system           </td><td>11</td><td>14-16</td><td>sensory nervous system:14-16         </td><td>0.8306988</td><td>w </td><td>148</td><td>-0.6711318</td><td>-2.9462104</td><td>0.01940591</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>exp1</td><td>287</td><td>221</td><td>TGCGCCAGAA</td><td>exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA</td><td>hrs_08_12</td><td>0.0000000</td><td>3.4843206</td><td>307</td><td>0</td><td>0</td><td>6 </td><td>0.06023494</td><td>Singlet</td><td>FALSE</td><td>9 </td><td>FALSE</td><td>9 </td><td> 9.715162</td><td> 9.000631</td><td>11.355991</td><td> 0.0000000</td><td> 0.0000000</td><td> 0.000000</td><td>midline primordium               </td><td>9 </td><td>8-10 </td><td>midline primordium:8-10              </td><td>0.6893560</td><td>aj</td><td>120</td><td>-7.5437593</td><td>-0.6611074</td><td>0.51285236</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 34\n",
       "\\begin{tabular}{r|llllllllllllllllllllllllllllllllll}\n",
       "  & orig.ident & nCount\\_RNA & nFeature\\_RNA & rt\\_bc & cell & time & percent.mt & percent.ribo & barn\\_dro & barn\\_hu & barn\\_mus & RNA\\_snn\\_res.0.8 & doublet\\_score & predicted\\_doublet & doublet\\_cluster & seurat\\_clusters & doublet\\_subcluster & RNA\\_snn\\_res.0.08 & lasso\\_age & NNv1\\_age & NNv2\\_age & lasso\\_shift & NNv1\\_shift & NNv2\\_shift & annot & ident & inferred\\_time\\_window & cell.type\\_time.bin & Size\\_Factor & cell.type.2 & cluster & umap\\_1 & umap\\_2 & pct\\\\\n",
       "  & <chr> & <dbl> & <int> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <int> & <int> & <int> & <chr> & <dbl> & <chr> & <lgl> & <fct> & <lgl> & <fct> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <fct> & <chr> & <chr> & <dbl> & <chr> & <fct> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & exp1 & 346 & 247 & AAGTTGCCAT & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT & hrs\\_06\\_10 & 1.4450867 & 4.3352601 & 371 & 0 & 0 & 16 & 0.05276064 & Singlet & FALSE & 21 & FALSE & 21 &  8.402794 &  7.885203 &  8.119637 &  0.0000000 &  0.0000000 &  0.000000 & sensory nervous system primordium & 21 & 6-8   & sensory nervous system primordium:6-8 & 0.8108612 & v  & 89  & -2.0067012 &  1.2184464 & 0.26496218\\\\\n",
       "\t2 & exp1 & 586 & 425 & CAGGTATGGA & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA & hrs\\_08\\_12 & 1.0238908 & 6.4846416 & 630 & 0 & 0 & 4  & 0.09008910 & Singlet & FALSE & 11 & FALSE & 11 & 11.266422 & 10.062633 & 11.156040 &  0.0000000 &  0.0000000 &  0.000000 & brain                             & 11 & 10-12 & brain:10-12                           & 1.4059888 & bl & 22  & -3.6831634 &  2.8012645 & 0.15352179\\\\\n",
       "\t3 & exp1 & 292 & 202 & AAGTAGTCAG & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG & hrs\\_10\\_14 & 0.3424658 & 0.6849315 & 320 & 3 & 0 & 23 & 0.06585398 & Singlet & FALSE & 12 & FALSE & 12 & 12.544636 & 12.211529 & 13.830982 &  0.0000000 &  0.0000000 &  0.000000 & epidermis                         & 12 & 12-14 & epidermis:12-14                       & 0.6967951 & h  & 165 & -1.3580737 &  0.3582040 & 0.01011936\\\\\n",
       "\t4 & exp1 & 335 & 243 & TGAGGTAGAA & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA & hrs\\_14\\_18 & 4.4776119 & 5.6716418 & 386 & 0 & 0 & 8  & 0.08478501 & Singlet & FALSE & 11 & FALSE & 11 & 12.542527 & 13.739003 & 12.519031 & -1.4574728 & -0.2609968 & -1.480969 & sensory nervous system            & 11 & 12-14 & sensory nervous system:12-14          & 0.7811049 & bk & 102 & -3.8563070 &  3.3158269 & 0.11677301\\\\\n",
       "\t5 & exp1 & 354 & 272 & TCGATATTGA & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA & hrs\\_14\\_18 & 0.5649718 & 1.6949153 & 380 & 0 & 1 & 8  & 0.08257165 & Singlet & FALSE & 11 & FALSE & 11 & 13.458162 & 15.126904 & 14.558554 & -0.5418379 &  0.0000000 &  0.000000 & sensory nervous system            & 11 & 14-16 & sensory nervous system:14-16          & 0.8306988 & w  & 148 & -0.6711318 & -2.9462104 & 0.01940591\\\\\n",
       "\t6 & exp1 & 287 & 221 & TGCGCCAGAA & exp1\\_idx1AGACCATATC\\_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA & hrs\\_08\\_12 & 0.0000000 & 3.4843206 & 307 & 0 & 0 & 6  & 0.06023494 & Singlet & FALSE & 9  & FALSE & 9  &  9.715162 &  9.000631 & 11.355991 &  0.0000000 &  0.0000000 &  0.000000 & midline primordium                & 9  & 8-10  & midline primordium:8-10               & 0.6893560 & aj & 120 & -7.5437593 & -0.6611074 & 0.51285236\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 34\n",
       "\n",
       "| <!--/--> | orig.ident &lt;chr&gt; | nCount_RNA &lt;dbl&gt; | nFeature_RNA &lt;int&gt; | rt_bc &lt;chr&gt; | cell &lt;chr&gt; | time &lt;chr&gt; | percent.mt &lt;dbl&gt; | percent.ribo &lt;dbl&gt; | barn_dro &lt;int&gt; | barn_hu &lt;int&gt; | barn_mus &lt;int&gt; | RNA_snn_res.0.8 &lt;chr&gt; | doublet_score &lt;dbl&gt; | predicted_doublet &lt;chr&gt; | doublet_cluster &lt;lgl&gt; | seurat_clusters &lt;fct&gt; | doublet_subcluster &lt;lgl&gt; | RNA_snn_res.0.08 &lt;fct&gt; | lasso_age &lt;dbl&gt; | NNv1_age &lt;dbl&gt; | NNv2_age &lt;dbl&gt; | lasso_shift &lt;dbl&gt; | NNv1_shift &lt;dbl&gt; | NNv2_shift &lt;dbl&gt; | annot &lt;chr&gt; | ident &lt;fct&gt; | inferred_time_window &lt;chr&gt; | cell.type_time.bin &lt;chr&gt; | Size_Factor &lt;dbl&gt; | cell.type.2 &lt;chr&gt; | cluster &lt;fct&gt; | umap_1 &lt;dbl&gt; | umap_2 &lt;dbl&gt; | pct &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | exp1 | 346 | 247 | AAGTTGCCAT | exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT | hrs_06_10 | 1.4450867 | 4.3352601 | 371 | 0 | 0 | 16 | 0.05276064 | Singlet | FALSE | 21 | FALSE | 21 |  8.402794 |  7.885203 |  8.119637 |  0.0000000 |  0.0000000 |  0.000000 | sensory nervous system primordium | 21 | 6-8   | sensory nervous system primordium:6-8 | 0.8108612 | v  | 89  | -2.0067012 |  1.2184464 | 0.26496218 |\n",
       "| 2 | exp1 | 586 | 425 | CAGGTATGGA | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA | hrs_08_12 | 1.0238908 | 6.4846416 | 630 | 0 | 0 | 4  | 0.09008910 | Singlet | FALSE | 11 | FALSE | 11 | 11.266422 | 10.062633 | 11.156040 |  0.0000000 |  0.0000000 |  0.000000 | brain                             | 11 | 10-12 | brain:10-12                           | 1.4059888 | bl | 22  | -3.6831634 |  2.8012645 | 0.15352179 |\n",
       "| 3 | exp1 | 292 | 202 | AAGTAGTCAG | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG | hrs_10_14 | 0.3424658 | 0.6849315 | 320 | 3 | 0 | 23 | 0.06585398 | Singlet | FALSE | 12 | FALSE | 12 | 12.544636 | 12.211529 | 13.830982 |  0.0000000 |  0.0000000 |  0.000000 | epidermis                         | 12 | 12-14 | epidermis:12-14                       | 0.6967951 | h  | 165 | -1.3580737 |  0.3582040 | 0.01011936 |\n",
       "| 4 | exp1 | 335 | 243 | TGAGGTAGAA | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA | hrs_14_18 | 4.4776119 | 5.6716418 | 386 | 0 | 0 | 8  | 0.08478501 | Singlet | FALSE | 11 | FALSE | 11 | 12.542527 | 13.739003 | 12.519031 | -1.4574728 | -0.2609968 | -1.480969 | sensory nervous system            | 11 | 12-14 | sensory nervous system:12-14          | 0.7811049 | bk | 102 | -3.8563070 |  3.3158269 | 0.11677301 |\n",
       "| 5 | exp1 | 354 | 272 | TCGATATTGA | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA | hrs_14_18 | 0.5649718 | 1.6949153 | 380 | 0 | 1 | 8  | 0.08257165 | Singlet | FALSE | 11 | FALSE | 11 | 13.458162 | 15.126904 | 14.558554 | -0.5418379 |  0.0000000 |  0.000000 | sensory nervous system            | 11 | 14-16 | sensory nervous system:14-16          | 0.8306988 | w  | 148 | -0.6711318 | -2.9462104 | 0.01940591 |\n",
       "| 6 | exp1 | 287 | 221 | TGCGCCAGAA | exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA | hrs_08_12 | 0.0000000 | 3.4843206 | 307 | 0 | 0 | 6  | 0.06023494 | Singlet | FALSE | 9  | FALSE | 9  |  9.715162 |  9.000631 | 11.355991 |  0.0000000 |  0.0000000 |  0.000000 | midline primordium                | 9  | 8-10  | midline primordium:8-10               | 0.6893560 | aj | 120 | -7.5437593 | -0.6611074 | 0.51285236 |\n",
       "\n"
      ],
      "text/plain": [
       "  orig.ident nCount_RNA nFeature_RNA rt_bc     \n",
       "1 exp1       346        247          AAGTTGCCAT\n",
       "2 exp1       586        425          CAGGTATGGA\n",
       "3 exp1       292        202          AAGTAGTCAG\n",
       "4 exp1       335        243          TGAGGTAGAA\n",
       "5 exp1       354        272          TCGATATTGA\n",
       "6 exp1       287        221          TGCGCCAGAA\n",
       "  cell                                                   time      percent.mt\n",
       "1 exp1_idx1AGACCATATC_idx2AACGAGGCTA.AACCGCTGTAAGTTGCCAT hrs_06_10 1.4450867 \n",
       "2 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCAGAACCTCAGGTATGGA hrs_08_12 1.0238908 \n",
       "3 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGAGAGATAAGTAGTCAG hrs_10_14 0.3424658 \n",
       "4 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGGAGTTTTGAGGTAGAA hrs_14_18 4.4776119 \n",
       "5 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCGTCTCATTCGATATTGA hrs_14_18 0.5649718 \n",
       "6 exp1_idx1AGACCATATC_idx2AACGAGGCTA.CCTCCTGATTGCGCCAGAA hrs_08_12 0.0000000 \n",
       "  percent.ribo barn_dro barn_hu barn_mus RNA_snn_res.0.8 doublet_score\n",
       "1 4.3352601    371      0       0        16              0.05276064   \n",
       "2 6.4846416    630      0       0        4               0.09008910   \n",
       "3 0.6849315    320      3       0        23              0.06585398   \n",
       "4 5.6716418    386      0       0        8               0.08478501   \n",
       "5 1.6949153    380      0       1        8               0.08257165   \n",
       "6 3.4843206    307      0       0        6               0.06023494   \n",
       "  predicted_doublet doublet_cluster seurat_clusters doublet_subcluster\n",
       "1 Singlet           FALSE           21              FALSE             \n",
       "2 Singlet           FALSE           11              FALSE             \n",
       "3 Singlet           FALSE           12              FALSE             \n",
       "4 Singlet           FALSE           11              FALSE             \n",
       "5 Singlet           FALSE           11              FALSE             \n",
       "6 Singlet           FALSE           9               FALSE             \n",
       "  RNA_snn_res.0.08 lasso_age NNv1_age  NNv2_age  lasso_shift NNv1_shift\n",
       "1 21                8.402794  7.885203  8.119637  0.0000000   0.0000000\n",
       "2 11               11.266422 10.062633 11.156040  0.0000000   0.0000000\n",
       "3 12               12.544636 12.211529 13.830982  0.0000000   0.0000000\n",
       "4 11               12.542527 13.739003 12.519031 -1.4574728  -0.2609968\n",
       "5 11               13.458162 15.126904 14.558554 -0.5418379   0.0000000\n",
       "6 9                 9.715162  9.000631 11.355991  0.0000000   0.0000000\n",
       "  NNv2_shift annot                             ident inferred_time_window\n",
       "1  0.000000  sensory nervous system primordium 21    6-8                 \n",
       "2  0.000000  brain                             11    10-12               \n",
       "3  0.000000  epidermis                         12    12-14               \n",
       "4 -1.480969  sensory nervous system            11    12-14               \n",
       "5  0.000000  sensory nervous system            11    14-16               \n",
       "6  0.000000  midline primordium                9     8-10                \n",
       "  cell.type_time.bin                    Size_Factor cell.type.2 cluster\n",
       "1 sensory nervous system primordium:6-8 0.8108612   v           89     \n",
       "2 brain:10-12                           1.4059888   bl          22     \n",
       "3 epidermis:12-14                       0.6967951   h           165    \n",
       "4 sensory nervous system:12-14          0.7811049   bk          102    \n",
       "5 sensory nervous system:14-16          0.8306988   w           148    \n",
       "6 midline primordium:8-10               0.6893560   aj          120    \n",
       "  umap_1     umap_2     pct       \n",
       "1 -2.0067012  1.2184464 0.26496218\n",
       "2 -3.6831634  2.8012645 0.15352179\n",
       "3 -1.3580737  0.3582040 0.01011936\n",
       "4 -3.8563070  3.3158269 0.11677301\n",
       "5 -0.6711318 -2.9462104 0.01940591\n",
       "6 -7.5437593 -0.6611074 0.51285236"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "left_join(df_n21, select(fuzzy_percent, cell, pct), by = \"cell\")  %>%  head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1049,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_n21 = left_join(df_n21, select(fuzzy_percent, cell, pct), by = \"cell\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1064,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'orig.ident'</li><li>'nCount_RNA'</li><li>'nFeature_RNA'</li><li>'rt_bc'</li><li>'cell'</li><li>'time'</li><li>'percent.mt'</li><li>'percent.ribo'</li><li>'barn_dro'</li><li>'barn_hu'</li><li>'barn_mus'</li><li>'RNA_snn_res.0.8'</li><li>'doublet_score'</li><li>'predicted_doublet'</li><li>'doublet_cluster'</li><li>'seurat_clusters'</li><li>'doublet_subcluster'</li><li>'RNA_snn_res.0.08'</li><li>'lasso_age'</li><li>'NNv1_age'</li><li>'NNv2_age'</li><li>'lasso_shift'</li><li>'NNv1_shift'</li><li>'NNv2_shift'</li><li>'annot'</li><li>'ident'</li><li>'inferred_time_window'</li><li>'cell.type_time.bin'</li><li>'Size_Factor'</li><li>'cell.type.2'</li><li>'cluster'</li><li>'umap_1'</li><li>'umap_2'</li><li>'pct'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'orig.ident'\n",
       "\\item 'nCount\\_RNA'\n",
       "\\item 'nFeature\\_RNA'\n",
       "\\item 'rt\\_bc'\n",
       "\\item 'cell'\n",
       "\\item 'time'\n",
       "\\item 'percent.mt'\n",
       "\\item 'percent.ribo'\n",
       "\\item 'barn\\_dro'\n",
       "\\item 'barn\\_hu'\n",
       "\\item 'barn\\_mus'\n",
       "\\item 'RNA\\_snn\\_res.0.8'\n",
       "\\item 'doublet\\_score'\n",
       "\\item 'predicted\\_doublet'\n",
       "\\item 'doublet\\_cluster'\n",
       "\\item 'seurat\\_clusters'\n",
       "\\item 'doublet\\_subcluster'\n",
       "\\item 'RNA\\_snn\\_res.0.08'\n",
       "\\item 'lasso\\_age'\n",
       "\\item 'NNv1\\_age'\n",
       "\\item 'NNv2\\_age'\n",
       "\\item 'lasso\\_shift'\n",
       "\\item 'NNv1\\_shift'\n",
       "\\item 'NNv2\\_shift'\n",
       "\\item 'annot'\n",
       "\\item 'ident'\n",
       "\\item 'inferred\\_time\\_window'\n",
       "\\item 'cell.type\\_time.bin'\n",
       "\\item 'Size\\_Factor'\n",
       "\\item 'cell.type.2'\n",
       "\\item 'cluster'\n",
       "\\item 'umap\\_1'\n",
       "\\item 'umap\\_2'\n",
       "\\item 'pct'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'orig.ident'\n",
       "2. 'nCount_RNA'\n",
       "3. 'nFeature_RNA'\n",
       "4. 'rt_bc'\n",
       "5. 'cell'\n",
       "6. 'time'\n",
       "7. 'percent.mt'\n",
       "8. 'percent.ribo'\n",
       "9. 'barn_dro'\n",
       "10. 'barn_hu'\n",
       "11. 'barn_mus'\n",
       "12. 'RNA_snn_res.0.8'\n",
       "13. 'doublet_score'\n",
       "14. 'predicted_doublet'\n",
       "15. 'doublet_cluster'\n",
       "16. 'seurat_clusters'\n",
       "17. 'doublet_subcluster'\n",
       "18. 'RNA_snn_res.0.08'\n",
       "19. 'lasso_age'\n",
       "20. 'NNv1_age'\n",
       "21. 'NNv2_age'\n",
       "22. 'lasso_shift'\n",
       "23. 'NNv1_shift'\n",
       "24. 'NNv2_shift'\n",
       "25. 'annot'\n",
       "26. 'ident'\n",
       "27. 'inferred_time_window'\n",
       "28. 'cell.type_time.bin'\n",
       "29. 'Size_Factor'\n",
       "30. 'cell.type.2'\n",
       "31. 'cluster'\n",
       "32. 'umap_1'\n",
       "33. 'umap_2'\n",
       "34. 'pct'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"orig.ident\"           \"nCount_RNA\"           \"nFeature_RNA\"        \n",
       " [4] \"rt_bc\"                \"cell\"                 \"time\"                \n",
       " [7] \"percent.mt\"           \"percent.ribo\"         \"barn_dro\"            \n",
       "[10] \"barn_hu\"              \"barn_mus\"             \"RNA_snn_res.0.8\"     \n",
       "[13] \"doublet_score\"        \"predicted_doublet\"    \"doublet_cluster\"     \n",
       "[16] \"seurat_clusters\"      \"doublet_subcluster\"   \"RNA_snn_res.0.08\"    \n",
       "[19] \"lasso_age\"            \"NNv1_age\"             \"NNv2_age\"            \n",
       "[22] \"lasso_shift\"          \"NNv1_shift\"           \"NNv2_shift\"          \n",
       "[25] \"annot\"                \"ident\"                \"inferred_time_window\"\n",
       "[28] \"cell.type_time.bin\"   \"Size_Factor\"          \"cell.type.2\"         \n",
       "[31] \"cluster\"              \"umap_1\"               \"umap_2\"              \n",
       "[34] \"pct\"                 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "colnames(df_n21)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1065,
   "metadata": {},
   "outputs": [],
   "source": [
    "colData(cds_sub)$fuz_pct = NULL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1066,
   "metadata": {},
   "outputs": [],
   "source": [
    "colData(cds_n21)$fuz_pct = rep(NA, ncol(cds_n21))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1067,
   "metadata": {},
   "outputs": [],
   "source": [
    "colData(cds_n21)$fuz_pct = df_n21$pct"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 332,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in plot.expr_clus.br2(cds_n21, c(\"apn\"), size_pos = 1, size_neg = 0.3, : could not find function \"plot.expr_clus.br2\"\n",
     "output_type": "error",
     "traceback": [
      "Error in plot.expr_clus.br2(cds_n21, c(\"apn\"), size_pos = 1, size_neg = 0.3, : could not find function \"plot.expr_clus.br2\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_n21, c(\"apn\"), \n",
    "                   size_pos = 1, size_neg = 0.3, stroke = 0.008, limit = 8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1069,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in h(simpleError(msg, call)): error in evaluating the argument 'x' in selecting a method for function 'reducedDims': unable to find an inherited method for function ‘NSBS’ for signature ‘\"data.frame\"’\n",
     "output_type": "error",
     "traceback": [
      "Error in h(simpleError(msg, call)): error in evaluating the argument 'x' in selecting a method for function 'reducedDims': unable to find an inherited method for function ‘NSBS’ for signature ‘\"data.frame\"’\nTraceback:\n",
      "1. plot.expr_clus.br2(cds_n21[, df_n21 %>% filter(df_n21$pct > 0.2)], \n .     c(\"apn\"), size_pos = 1, size_neg = 0.3, stroke = 0.008, limit = 8)",
      "2. reducedDims(cds)   # at line 11 of file <text>",
      "3. cds_n21[, df_n21 %>% filter(df_n21$pct > 0.2)]   # at line 11 of file <text>",
      "4. cds_n21[, df_n21 %>% filter(df_n21$pct > 0.2)]",
      "5. int_colData(x)[jj, , drop = FALSE]",
      "6. int_colData(x)[jj, , drop = FALSE]",
      "7. extractROWS(x, i)",
      "8. extractROWS(x, i)",
      "9. normalizeSingleBracketSubscript(i, x, allow.NAs = TRUE, as.NSBS = TRUE)",
      "10. NSBS(i, x, exact = exact, strict.upper.bound = !allow.append, \n  .     allow.NAs = allow.NAs)",
      "11. (function (classes, fdef, mtable) \n  . {\n  .     methods <- .findInheritedMethods(classes, fdef, mtable)\n  .     if (length(methods) == 1L) \n  .         return(methods[[1L]])\n  .     else if (length(methods) == 0L) {\n  .         cnames <- paste0(\"\\\"\", vapply(classes, as.character, \n  .             \"\"), \"\\\"\", collapse = \", \")\n  .         stop(gettextf(\"unable to find an inherited method for function %s for signature %s\", \n  .             sQuote(fdef@generic), sQuote(cnames)), domain = NA)\n  .     }\n  .     else stop(\"Internal error in finding inherited methods; didn't return a unique method\", \n  .         domain = NA)\n  . })(list(\"data.frame\"), new(\"standardGeneric\", .Data = function (i, \n  .     x, exact = TRUE, strict.upper.bound = TRUE, allow.NAs = FALSE) \n  . standardGeneric(\"NSBS\"), generic = structure(\"NSBS\", package = \"S4Vectors\"), \n  .     package = \"S4Vectors\", group = list(), valueClass = character(0), \n  .     signature = \"i\", default = NULL, skeleton = (function (i, \n  .         x, exact = TRUE, strict.upper.bound = TRUE, allow.NAs = FALSE) \n  .     stop(gettextf(\"invalid call in method dispatch to '%s' (no default method)\", \n  .         \"NSBS\"), domain = NA))(i, x, exact, strict.upper.bound, \n  .         allow.NAs)), <environment>)",
      "12. stop(gettextf(\"unable to find an inherited method for function %s for signature %s\", \n  .     sQuote(fdef@generic), sQuote(cnames)), domain = NA)",
      "13. .handleSimpleError(function (cond) \n  . .Internal(C_tryCatchHelper(addr, 1L, cond)), \"unable to find an inherited method for function ‘NSBS’ for signature ‘\\\"data.frame\\\"’\", \n  .     base::quote((function (classes, fdef, mtable) \n  .     {\n  .         methods <- .findInheritedMethods(classes, fdef, mtable)\n  .         if (length(methods) == 1L) \n  .             return(methods[[1L]])\n  .         else if (length(methods) == 0L) {\n  .             cnames <- paste0(\"\\\"\", vapply(classes, as.character, \n  .                 \"\"), \"\\\"\", collapse = \", \")\n  .             stop(gettextf(\"unable to find an inherited method for function %s for signature %s\", \n  .                 sQuote(fdef@generic), sQuote(cnames)), domain = NA)\n  .         }\n  .         else stop(\"Internal error in finding inherited methods; didn't return a unique method\", \n  .             domain = NA)\n  .     })(list(\"data.frame\"), new(\"standardGeneric\", .Data = function (i, \n  .         x, exact = TRUE, strict.upper.bound = TRUE, allow.NAs = FALSE) \n  .     standardGeneric(\"NSBS\"), generic = structure(\"NSBS\", package = \"S4Vectors\"), \n  .         package = \"S4Vectors\", group = list(), valueClass = character(0), \n  .         signature = \"i\", default = NULL, skeleton = (function (i, \n  .             x, exact = TRUE, strict.upper.bound = TRUE, allow.NAs = FALSE) \n  .         stop(gettextf(\"invalid call in method dispatch to '%s' (no default method)\", \n  .             \"NSBS\"), domain = NA))(i, x, exact, strict.upper.bound, \n  .             allow.NAs)), <environment>)))",
      "14. h(simpleError(msg, call))"
     ]
    }
   ],
   "source": [
    "plot.expr_clus.br2(cds_n21[, df_n21  %>% filter(df_n21$pct > 0.2)], c(\"apn\"), \n",
    "                   size_pos = 1, size_neg = 0.3, stroke = 0.008, limit = 8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1070,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>141573</li><li>34</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 141573\n",
       "\\item 34\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 141573\n",
       "2. 34\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 141573     34"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_n21  %>% filter(pct > 0.2)  %>%  dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 333,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_n21, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_n21, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_n21 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_n21$pct), size =  0.02, stroke = 0.005) +\n",
    "# scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "#      labs(color = cell_type) +\n",
    "guides(color = guide_legend(override.aes = list(size = 4))) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 334,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_n21 %>% filter(pct > 0.2), aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_n21 %>% filter(pct > 0.2), aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_n21  %>% filter(pct > 0.2), aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = (df_n21  %>% filter(pct > 0.2))$pct), size =  0.02, stroke = 0.005) +\n",
    "# scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "#      labs(color = cell_type) +\n",
    "guides(color = guide_legend(override.aes = list(size = 4))) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 335,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_n21 %>% filter(pct < 0.2), aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_n21 %>% filter(pct < 0.2), aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_n21  %>% filter(pct < 0.2), aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = (df_n21  %>% filter(pct < 0.2))$pct), size =  0.02, stroke = 0.005) +\n",
    "# scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "#      labs(color = cell_type) +\n",
    "guides(color = guide_legend(override.aes = list(size = 4))) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 336,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in ggplot(df_n21, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in ggplot(df_n21, aes(x = umap_1, y = umap_2)): could not find function \"ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "peedeef = ggplot(df_n21 , aes(x = umap_1, y = umap_2)) +\n",
    "geom_point(aes(color = df_n21$pct < 0.1), size =  0.02, stroke = 0.005) +\n",
    " scale_color_manual(values = c(\"grey70\", \"firebrick\")) +\n",
    "      labs(color = cell_type) +\n",
    "# guides(color = guide_legend(override.aes = list(size = 4))) +\n",
    "theme_classic()\n",
    "print(peedeef)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1782,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<pre class=language-r><code>function (cds_subset, min_expr = NULL, cell_size = 0.75, nrow = NULL, \n",
       "<span style=white-space:pre-wrap>    ncol = 1, panel_order = NULL, color_cells_by = \"pseudotime\", </span>\n",
       "<span style=white-space:pre-wrap>    trend_formula = \"~ splines::ns(pseudotime, df=3)\", label_by_short_name = TRUE, </span>\n",
       "<span style=white-space:pre-wrap>    vertical_jitter = NULL, horizontal_jitter = NULL) </span>\n",
       "NULL</code></pre>"
      ],
      "text/latex": [
       "\\begin{minted}{r}\n",
       "function (cds\\_subset, min\\_expr = NULL, cell\\_size = 0.75, nrow = NULL, \n",
       "    ncol = 1, panel\\_order = NULL, color\\_cells\\_by = \"pseudotime\", \n",
       "    trend\\_formula = \"\\textasciitilde{} splines::ns(pseudotime, df=3)\", label\\_by\\_short\\_name = TRUE, \n",
       "    vertical\\_jitter = NULL, horizontal\\_jitter = NULL) \n",
       "NULL\n",
       "\\end{minted}"
      ],
      "text/markdown": [
       "```r\n",
       "function (cds_subset, min_expr = NULL, cell_size = 0.75, nrow = NULL, \n",
       "    ncol = 1, panel_order = NULL, color_cells_by = \"pseudotime\", \n",
       "    trend_formula = \"~ splines::ns(pseudotime, df=3)\", label_by_short_name = TRUE, \n",
       "    vertical_jitter = NULL, horizontal_jitter = NULL) \n",
       "NULL\n",
       "```"
      ],
      "text/plain": [
       "function (cds_subset, min_expr = NULL, cell_size = 0.75, nrow = NULL, \n",
       "    ncol = 1, panel_order = NULL, color_cells_by = \"pseudotime\", \n",
       "    trend_formula = \"~ splines::ns(pseudotime, df=3)\", label_by_short_name = TRUE, \n",
       "    vertical_jitter = NULL, horizontal_jitter = NULL) \n",
       "NULL"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "args(plot_genes_in_pseudotime)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.2"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "279.261px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
